// igniteui-react-excel/esm2015/lib/ActiveXBinaryPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { PartManagerBase } from "./PartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ActiveXBinaryPartManager = /*@__PURE__*/ (() => {
    class ActiveXBinaryPartManager extends PartManagerBase {
        get_m() {
            return "application/vnd.ms-office.activeX";
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/activeX.bin";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return ActiveXBinaryPartManager.u;
                case 1: return ActiveXBinaryPartManager.t;
                default: return ActiveXBinaryPartManager.u;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    ActiveXBinaryPartManager.$t = /*@__PURE__*/ markType(ActiveXBinaryPartManager, 'ActiveXBinaryPartManager', PartManagerBase.$);
    ActiveXBinaryPartManager.u = "http://schemas.microsoft.com/office/2006/relationships/activeXControlBinary";
    ActiveXBinaryPartManager.t = "http://schemas.microsoft.com/office/2006/relationships/activeXControlBinary";
    return ActiveXBinaryPartManager;
})();
