// igniteui-react-excel/esm2015/lib/ActiveXPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { PartManagerBase } from "./PartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ActiveXPartManager = /*@__PURE__*/ (() => {
    class ActiveXPartManager extends PartManagerBase {
        get_m() {
            return "application/vnd.ms-office.activeX+xml";
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/activeX.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return ActiveXPartManager.u;
                case 1: return ActiveXPartManager.t;
                default: return ActiveXPartManager.u;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    ActiveXPartManager.$t = /*@__PURE__*/ markType(ActiveXPartManager, 'ActiveXPartManager', PartManagerBase.$);
    ActiveXPartManager.u = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/control";
    ActiveXPartManager.t = "http://purl.oclc.org/ooxml/officeDocument/relationships/control";
    return ActiveXPartManager;
})();
