// igniteui-react-excel/esm2015/lib/Async_ForSliceHelper.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, runOn, markType } from "igniteui-react-core";
import { WorkItem } from "./WorkItem";
/**
 * @hidden
 */
export let Async_ForSliceHelper = /*@__PURE__*/ (() => {
    class Async_ForSliceHelper extends Base {
        constructor(a, b, c, d) {
            super();
            this.d = 0;
            this.b = null;
            this.e = 0;
            this.c = null;
            this.f = 0;
            this.f = a;
            this.d = b;
            this.e = c;
            this.c = d;
            this.b = runOn(this, this.a);
        }
        a() {
            while (this.f < this.d) {
                let a = this.c(this.f);
                this.f += this.e;
                if (!a) {
                    return WorkItem.d(this.b, 2);
                }
            }
            return null;
        }
    }
    Async_ForSliceHelper.$t = /*@__PURE__*/ markType(Async_ForSliceHelper, 'Async_ForSliceHelper');
    return Async_ForSliceHelper;
})();
