// igniteui-react-excel/esm2015/lib/Async_ForHelper.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, runOn, markType } from "igniteui-react-core";
import { WorkItemExtensions } from "./WorkItemExtensions";
/**
 * @hidden
 */
export let Async_ForHelper = /*@__PURE__*/ (() => {
    class Async_ForHelper extends Base {
        constructor(a, b, c, d) {
            super();
            this.e = 0;
            this.c = null;
            this.f = 0;
            this.d = null;
            this.g = 0;
            this.g = a;
            this.e = b;
            this.f = c;
            this.d = d;
            this.c = runOn(this, this.b);
        }
        a() {
            if (this.e <= this.g) {
                return null;
            }
            return WorkItemExtensions.c(this.d(this.g), this.c);
        }
        b() {
            this.g += this.f;
            return this.a();
        }
    }
    Async_ForHelper.$t = /*@__PURE__*/ markType(Async_ForHelper, 'Async_ForHelper');
    return Async_ForHelper;
})();
