// igniteui-react-excel/esm2015/lib/AxisPosition.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which determine the placement of an [[Axis]],
 * as relative to the chart space.
 */
export var AxisPosition = /*@__PURE__*/ (function (AxisPosition) {
    /**
     * The axis is positioned automatically, with the value axes
     * appearing on the left/right, and the category axes appearing
     * on the top/bottom.
     */
    AxisPosition[AxisPosition["Automatic"] = 0] = "Automatic";
    /**
     * The axis appears alongside the bottom edge of the chart.
     */
    AxisPosition[AxisPosition["Bottom"] = 1] = "Bottom";
    /**
     * The axis appears alongside the left edge of the chart.
     */
    AxisPosition[AxisPosition["Left"] = 2] = "Left";
    /**
     * The axis appears alongside the right edge of the chart.
     */
    AxisPosition[AxisPosition["Right"] = 3] = "Right";
    /**
     * The axis appears alongside the top edge of the chart.
     */
    AxisPosition[AxisPosition["Top"] = 4] = "Top";
    return AxisPosition;
})({});
/**
 * @hidden
 */
export let AxisPosition_$type = /*@__PURE__*/ markEnum('AxisPosition', 'Automatic,0|Bottom,1|Left,2|Right,3|Top,4');
