// igniteui-react-excel/esm2015/lib/BiffRecordStream_RecordBlockInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let BiffRecordStream_RecordBlockInfo = /*@__PURE__*/ (() => {
    class BiffRecordStream_RecordBlockInfo extends Base {
        constructor(a, b, c) {
            super();
            this.f = 0;
            this.a = 0;
            this.c = 0;
            this.f = a;
            this.a = b;
            this.c = c;
        }
        h() {
            this.c = this.a;
        }
        get e() {
            return this.f + this.a;
        }
        get b() {
            return this.a;
        }
        set b(a) {
            this.a = a;
        }
        get g() {
            return this.f;
        }
        get d() {
            return this.c;
        }
    }
    BiffRecordStream_RecordBlockInfo.$t = /*@__PURE__*/ markType(BiffRecordStream_RecordBlockInfo, 'BiffRecordStream_RecordBlockInfo');
    return BiffRecordStream_RecordBlockInfo;
})();
