// igniteui-react-excel/esm2015/lib/Biff8RecordStream_Stxp.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let Biff8RecordStream_Stxp = /*@__PURE__*/ (() => {
    class Biff8RecordStream_Stxp extends Base {
        constructor() {
            super(...arguments);
            this.e = 0;
            this.a = null;
            this.f = 0;
            this.g = 0;
            this.d = 0;
            this.c = 0;
            this.b = 0;
        }
    }
    Biff8RecordStream_Stxp.$t = /*@__PURE__*/ markType(Biff8RecordStream_Stxp, 'Biff8RecordStream_Stxp');
    return Biff8RecordStream_Stxp;
})();
