// igniteui-react-excel/esm2015/lib/BmpPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ImagePartManagerBase } from "./ImagePartManagerBase";
import { ImageFormat } from "./ImageFormat";
import { IOPath } from "igniteui-react-core";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let BmpPartManager = /*@__PURE__*/ (() => {
    class BmpPartManager extends ImagePartManagerBase {
        constructor(a) {
            super();
            this.aa = false;
            this.aa = a;
        }
        get_s() {
            return this.aa ? "/xl/media/image.dib" : "/xl/media/image.bmp";
        }
        get s() {
            return this.get_s();
        }
        get_t() {
            return this.aa ? "/ppt/media/image.dib" : "/ppt/media/image.bmp";
        }
        get t() {
            return this.get_t();
        }
        get_u() {
            return this.aa ? "/word/media/image.dib" : "/word/media/image.bmp";
        }
        get u() {
            return this.get_u();
        }
        get_m() {
            return "image/bmp";
        }
        get m() {
            return this.get_m();
        }
        r(a) {
            let b = IOPath.getExtension(a);
            if (b.toLowerCase() == ".dib") {
                return ImageFormat.b;
            }
            return ImageFormat.a;
        }
    }
    BmpPartManager.$t = /*@__PURE__*/ markType(BmpPartManager, 'BmpPartManager', ImagePartManagerBase.$);
    BmpPartManager.y = /*@__PURE__*/ new BmpPartManager(false);
    BmpPartManager.z = /*@__PURE__*/ new BmpPartManager(true);
    return BmpPartManager;
})();
