// igniteui-react-excel/esm2015/lib/BorderStyleInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType } from "igniteui-react-core";
import { ColorInfo } from "./ColorInfo";
/**
 * @hidden
 */
export let BorderStyleInfo = /*@__PURE__*/ (() => {
    class BorderStyleInfo extends Base {
        constructor() {
            super(...arguments);
            this.c = null;
            this.b = -1;
        }
        equals(a) {
            if (Base.referenceEquals(this, a)) {
                return true;
            }
            let b = typeCast(BorderStyleInfo.$, a);
            if (b == null) {
                return false;
            }
            return this.b == b.b && Base.equalsStatic(this.c, b.c);
        }
        getHashCode() {
            let a = Base.getHashCodeStatic(this.b);
            if (this.c != null) {
                a ^= this.c.getHashCode();
            }
            return a;
        }
        get f() {
            if (this.c != null && this.c.f == false) {
                return false;
            }
            if (this.b != -1) {
                return false;
            }
            return true;
        }
        get d() {
            if (this.c == null) {
                this.c = new ColorInfo();
            }
            return this.c;
        }
        set d(a) {
            this.c = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
    }
    BorderStyleInfo.$t = /*@__PURE__*/ markType(BorderStyleInfo, 'BorderStyleInfo');
    return BorderStyleInfo;
})();
