// igniteui-react-excel/esm2015/lib/CalcManagerUtilities.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType, EnumUtil } from "igniteui-react-core";
import { SilverlightFixes } from "./SilverlightFixes";
/**
 * @hidden
 */
export let CalcManagerUtilities = /*@__PURE__*/ (() => {
    class CalcManagerUtilities extends Base {
        static c(a, b, c) {
            return SilverlightFixes.e(SilverlightFixes.a(a), b, c);
        }
        static b(a, b, c) {
            return SilverlightFixes.e(a, b, c);
        }
        static e(a, b, c, d, e) {
            return SilverlightFixes.q(SilverlightFixes.a(a), b, c, d, e);
        }
        static d(a, b, c, d, e) {
            return SilverlightFixes.q(a, b, c, d, e);
        }
        static f(a) {
        }
        static g(a, b) {
        }
        static a(a) {
            return EnumUtil.getValues(a);
        }
    }
    CalcManagerUtilities.$t = /*@__PURE__*/ markType(CalcManagerUtilities, 'CalcManagerUtilities');
    return CalcManagerUtilities;
})();
