// igniteui-react-excel/esm2015/lib/CalcChainPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CalcChainPartManager = /*@__PURE__*/ (() => {
    class CalcChainPartManager extends XmlPartManagerBase {
        get_m() {
            return CalcChainPartManager.a3;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/calcChain.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return CalcChainPartManager.a6;
                case 1: return CalcChainPartManager.a5;
                default: return CalcChainPartManager.a6;
            }
        }
        get_g() {
            return false;
        }
        get g() {
            return this.get_g();
        }
    }
    CalcChainPartManager.$t = /*@__PURE__*/ markType(CalcChainPartManager, 'CalcChainPartManager', XmlPartManagerBase.$);
    CalcChainPartManager.a3 = "application/vnd.openxmlformats-officedocument.spreadsheetml.calcChain+xml";
    CalcChainPartManager.a6 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/calcChain";
    CalcChainPartManager.a5 = "http://purl.oclc.org/ooxml/officeDocument/relationships/calcChain";
    return CalcChainPartManager;
})();
