// igniteui-react-excel/esm2015/lib/CalendarEraInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { ArgumentOutOfRangeException } from "igniteui-react-core";
import { dateFromValues } from "igniteui-react-core";
/**
 * @hidden
 */
export let CalendarEraInfo = /*@__PURE__*/ (() => {
    class CalendarEraInfo extends Base {
        constructor(a, b, c, d, e, f, g, h, i, j) {
            super();
            this.b = 0;
            this.a = new Date();
            this.e = 0;
            this.g = 0;
            this.f = 0;
            this.j = null;
            this.h = null;
            this.i = null;
            this.b = a;
            this.a = dateFromValues(b, c, d, 0, 0, 0, 0);
            this.e = e;
            this.g = f;
            this.f = g;
            this.j = h;
            this.h = i;
            this.i = j;
        }
        static c(a, b) {
            for (let c = 0; c < a.length; c++) {
                if (b == a[c].b) {
                    return c;
                }
            }
            return -1;
        }
        static d(a, b, c) {
            let d = CalendarEraInfo.c(a, c);
            if (d < 0) {
                throw new ArgumentOutOfRangeException(0);
            }
            let e = a[d];
            if (b < e.g || e.f < b) {
                throw new ArgumentOutOfRangeException(0);
            }
            return e.e + b;
        }
    }
    CalendarEraInfo.$t = /*@__PURE__*/ markType(CalendarEraInfo, 'CalendarEraInfo');
    return CalendarEraInfo;
})();
