// igniteui-react-excel/esm2015/lib/CalendarType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various calendar types available for the [[FixedValuesFilter]]
 * @see [[FixedValuesFilter.calendarType]]
 */
export var CalendarType = /*@__PURE__*/ (function (CalendarType) {
    /**
     * The Gregorian calendar should be used.
     */
    CalendarType[CalendarType["Gregorian"] = 0] = "Gregorian";
    /**
     * The Arabic version of the Gregorian calendar should be used.
     */
    CalendarType[CalendarType["GregorianArabic"] = 1] = "GregorianArabic";
    /**
     * The Middle East French version of the Gregorian calendar should be used.
     */
    CalendarType[CalendarType["GregorianMeFrench"] = 2] = "GregorianMeFrench";
    /**
     * The US English version of the Gregorian calendar should be used.
     */
    CalendarType[CalendarType["GregorianUs"] = 3] = "GregorianUs";
    /**
     * The transliterated English version of the Gregorian calendar should be used.
     */
    CalendarType[CalendarType["GregorianXlitEnglish"] = 4] = "GregorianXlitEnglish";
    /**
     * The transliterated French version of the Gregorian calendar should be used.
     */
    CalendarType[CalendarType["GregorianXlitFrench"] = 5] = "GregorianXlitFrench";
    /**
     * The Hebrew lunar calendar, as described by the Gauss formula for Passover and The Complete Restatement of Oral Law,
     * should be used.
     */
    CalendarType[CalendarType["Hebrew"] = 6] = "Hebrew";
    /**
     * The Hijri lunar calendar, as described by the Kingdom of Saudi Arabia, Ministry of Islamic Affairs, Endowments, Da‘wah
     * and Guidance, should be used.
     */
    CalendarType[CalendarType["Hijri"] = 7] = "Hijri";
    /**
     * The Japanese Emperor Era calendar, as described by Japanese Industrial Standard JIS X 0301, should be used.
     */
    CalendarType[CalendarType["Japan"] = 8] = "Japan";
    /**
     * The Korean Tangun Era calendar, as described by Korean Law Enactment No. 4, should be used.
     */
    CalendarType[CalendarType["Korea"] = 9] = "Korea";
    /**
     * Specifies that no calendar should be used.
     */
    CalendarType[CalendarType["None"] = 10] = "None";
    /**
     * The Saka Era calendar, as described by the Calendar Reform Committee of India, as part of the Indian Ephemeris and Nautical
     * Almanac, should be used.
     */
    CalendarType[CalendarType["Saka"] = 11] = "Saka";
    /**
     * The Taiwanese calendar, as defined by the Chinese National Standard CNS 7648, should be used.
     */
    CalendarType[CalendarType["Taiwan"] = 12] = "Taiwan";
    /**
     * The Thai calendar, as defined by the Royal Decree of H.M. King Vajiravudh (Rama VI) in Royal Gazette B. E. 2456 (1913 A.D.)
     * and by the decree of Prime Minister Phibunsongkhram (1941 A.D.) to start the year on the Gregorian January 1 and to map year
     * zero to Gregorian year 543 B.C., should be used.
     */
    CalendarType[CalendarType["Thai"] = 13] = "Thai";
    return CalendarType;
})({});
/**
 * @hidden
 */
export let CalendarType_$type = /*@__PURE__*/ markEnum('CalendarType', 'Gregorian,0|GregorianArabic,1|GregorianMeFrench,2|GregorianUs,3|GregorianXlitEnglish,4|GregorianXlitFrench,5|Hebrew,6|Hijri,7|Japan,8|Korea,9|None,10|Saka,11|Taiwan,12|Thai,13');
