// igniteui-react-excel/esm2015/lib/CellBorderLineStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the different cell border line styles.
 */
export var CellBorderLineStyle = /*@__PURE__*/ (function (CellBorderLineStyle) {
    /**
     * Use the default border line style.
     */
    CellBorderLineStyle[CellBorderLineStyle["Default"] = -1] = "Default";
    /**
     * No border.
     */
    CellBorderLineStyle[CellBorderLineStyle["None"] = 0] = "None";
    /**
     * Thin border.
     */
    CellBorderLineStyle[CellBorderLineStyle["Thin"] = 1] = "Thin";
    /**
     * Medium border.
     */
    CellBorderLineStyle[CellBorderLineStyle["Medium"] = 2] = "Medium";
    /**
     * Dashed border.
     */
    CellBorderLineStyle[CellBorderLineStyle["Dashed"] = 3] = "Dashed";
    /**
     * Dotted border.
     */
    CellBorderLineStyle[CellBorderLineStyle["Dotted"] = 4] = "Dotted";
    /**
     * Thick border.
     */
    CellBorderLineStyle[CellBorderLineStyle["Thick"] = 5] = "Thick";
    /**
     * Double-line border.
     */
    CellBorderLineStyle[CellBorderLineStyle["double1"] = 6] = "double1";
    /**
     * Dotted border with small dots.
     */
    CellBorderLineStyle[CellBorderLineStyle["Hair"] = 7] = "Hair";
    /**
     * Dotted border with big dots.
     */
    CellBorderLineStyle[CellBorderLineStyle["MediumDashed"] = 8] = "MediumDashed";
    /**
     * Dash-dot border.
     */
    CellBorderLineStyle[CellBorderLineStyle["DashDot"] = 9] = "DashDot";
    /**
     * Medium dash-dot border.
     */
    CellBorderLineStyle[CellBorderLineStyle["MediumDashDot"] = 10] = "MediumDashDot";
    /**
     * Dash-dot-dot border.
     */
    CellBorderLineStyle[CellBorderLineStyle["DashDotDot"] = 11] = "DashDotDot";
    /**
     * Medium dash-dot-dot border.
     */
    CellBorderLineStyle[CellBorderLineStyle["MediumDashDotDot"] = 12] = "MediumDashDotDot";
    /**
     * Slanted dash-dot border.
     */
    CellBorderLineStyle[CellBorderLineStyle["SlantedDashDot"] = 13] = "SlantedDashDot";
    return CellBorderLineStyle;
})({});
/**
 * @hidden
 */
export let CellBorderLineStyle_$type = /*@__PURE__*/ markEnum('CellBorderLineStyle', 'Default,-1|None,0|Thin,1|Medium,2|Dashed,3|Dotted,4|Thick,5|Double:double1,6|Hair,7|MediumDashed,8|DashDot,9|MediumDashDot,10|DashDotDot,11|MediumDashDotDot,12|SlantedDashDot,13');
