// igniteui-react-excel/esm2015/lib/CfExNonCf12Wrapper.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { CfCommonProps } from "./CfCommonProps";
/**
 * @hidden
 */
export let CfExNonCf12Wrapper = /*@__PURE__*/ (() => {
    class CfExNonCf12Wrapper extends Base {
        constructor() {
            super();
            this.a = null;
            this.c = null;
            this.e = 0;
            this.b = 0;
            this.d = null;
            this.a = new CfCommonProps();
        }
    }
    CfExNonCf12Wrapper.$t = /*@__PURE__*/ markType(CfExNonCf12Wrapper, 'CfExNonCf12Wrapper');
    return CfExNonCf12Wrapper;
})();
