// igniteui-react-excel/esm2015/lib/ChartPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ChartPartManager = /*@__PURE__*/ (() => {
    class ChartPartManager extends XmlPartManagerBase {
        get_m() {
            return ChartPartManager.a5;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/charts/chart.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return ChartPartManager.a7;
                case 1: return ChartPartManager.a6;
                default: return ChartPartManager.a7;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    ChartPartManager.$t = /*@__PURE__*/ markType(ChartPartManager, 'ChartPartManager', XmlPartManagerBase.$);
    ChartPartManager.a5 = "application/vnd.openxmlformats-officedocument.drawingml.chart+xml";
    ChartPartManager.a7 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/chart";
    ChartPartManager.a6 = "http://purl.oclc.org/ooxml/officeDocument/relationships/chart";
    return ChartPartManager;
})();
