// igniteui-react-excel/esm2015/lib/ChartUserShapePartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { PartManagerBase } from "./PartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ChartUserShapePartManager = /*@__PURE__*/ (() => {
    class ChartUserShapePartManager extends PartManagerBase {
        get_m() {
            return ChartUserShapePartManager.s;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/drawings/drawing.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return ChartUserShapePartManager.u;
                case 1: return ChartUserShapePartManager.t;
                default: return ChartUserShapePartManager.u;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    ChartUserShapePartManager.$t = /*@__PURE__*/ markType(ChartUserShapePartManager, 'ChartUserShapePartManager', PartManagerBase.$);
    ChartUserShapePartManager.s = "application/vnd.openxmlformats-officedocument.drawingml.chartshapes+xml";
    ChartUserShapePartManager.u = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/chartUserShapes";
    ChartUserShapePartManager.t = "http://purl.oclc.org/ooxml/officeDocument/relationships/chart";
    return ChartUserShapePartManager;
})();
