// igniteui-react-excel/esm2015/lib/ChartColorStylePartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ChartColorStylePartManager = /*@__PURE__*/ (() => {
    class ChartColorStylePartManager extends XmlPartManagerBase {
        get_m() {
            return ChartColorStylePartManager.a4;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/charts/colors.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return ChartColorStylePartManager.a6;
                case 1: return ChartColorStylePartManager.a5;
                default: return ChartColorStylePartManager.a6;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    ChartColorStylePartManager.$t = /*@__PURE__*/ markType(ChartColorStylePartManager, 'ChartColorStylePartManager', XmlPartManagerBase.$);
    ChartColorStylePartManager.a4 = "application/vnd.ms-office.chartcolorstyle+xml";
    ChartColorStylePartManager.a6 = "http://schemas.microsoft.com/office/2011/relationships/chartColorStyle";
    ChartColorStylePartManager.a5 = "http://schemas.microsoft.com/office/2011/relationships/chartColorStyle";
    return ChartColorStylePartManager;
})();
