// igniteui-react-excel/esm2015/lib/ChartExPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ChartExPartManager = /*@__PURE__*/ (() => {
    class ChartExPartManager extends XmlPartManagerBase {
        get_m() {
            return "application/vnd.ms-office.chartex+xml";
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/charts/chartEx.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return ChartExPartManager.a7;
                case 1: return ChartExPartManager.a6;
                default: return ChartExPartManager.a7;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    ChartExPartManager.$t = /*@__PURE__*/ markType(ChartExPartManager, 'ChartExPartManager', XmlPartManagerBase.$);
    ChartExPartManager.a7 = "http://schemas.microsoft.com/office/2014/relationships/chartEx";
    ChartExPartManager.a6 = "http://purl.oclc.org/ooxml/office/2014/relationships/chartEx";
    return ChartExPartManager;
})();
