// igniteui-react-excel/esm2015/lib/ChartType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which identify the chart's type, i.e.,
 * area chart, bar chart, pie chart, etc.
 */
export var ChartType = /*@__PURE__*/ (function (ChartType) {
    /**
     * Not an actual value; used for variable initialization.
     */
    ChartType[ChartType["Unknown"] = 0] = "Unknown";
    /**
     * 3D Area.
     */
    ChartType[ChartType["Area3D"] = -4098] = "Area3D";
    /**
     * 3D Stacked Area.
     */
    ChartType[ChartType["Area3DStacked"] = 78] = "Area3DStacked";
    /**
     * 100% Stacked Area.
     */
    ChartType[ChartType["Area3DStacked100"] = 79] = "Area3DStacked100";
    /**
     * 3D Clustered Bar.
     */
    ChartType[ChartType["Bar3DClustered"] = 60] = "Bar3DClustered";
    /**
     * 3D Stacked Bar.
     */
    ChartType[ChartType["Bar3DStacked"] = 61] = "Bar3DStacked";
    /**
     * 3D 100% Stacked Bar.
     */
    ChartType[ChartType["Bar3DStacked100"] = 62] = "Bar3DStacked100";
    /**
     * 3D Column.
     */
    ChartType[ChartType["Column3D"] = -4100] = "Column3D";
    /**
     * 3D Clustered Column.
     */
    ChartType[ChartType["Column3DClustered"] = 54] = "Column3DClustered";
    /**
     * 3D Stacked Column.
     */
    ChartType[ChartType["Column3DStacked"] = 55] = "Column3DStacked";
    /**
     * 3D 100% Stacked Column.
     */
    ChartType[ChartType["Column3DStacked100"] = 56] = "Column3DStacked100";
    /**
     * 3D Line.
     */
    ChartType[ChartType["Line3D"] = -4101] = "Line3D";
    /**
     * 3D Pie.
     */
    ChartType[ChartType["Pie3D"] = -4102] = "Pie3D";
    /**
     * Exploded 3D Pie.
     */
    ChartType[ChartType["Pie3DExploded"] = 70] = "Pie3DExploded";
    /**
     * Area
     */
    ChartType[ChartType["Area"] = 1] = "Area";
    /**
     * Stacked Area.
     */
    ChartType[ChartType["AreaStacked"] = 76] = "AreaStacked";
    /**
     * 100% Stacked Area.
     */
    ChartType[ChartType["AreaStacked100"] = 77] = "AreaStacked100";
    /**
     * Clustered Bar.
     */
    ChartType[ChartType["BarClustered"] = 57] = "BarClustered";
    /**
     * Bar of Pie.
     */
    ChartType[ChartType["BarOfPie"] = 71] = "BarOfPie";
    /**
     * Stacked Bar.
     */
    ChartType[ChartType["BarStacked"] = 58] = "BarStacked";
    /**
     * 100% Stacked Bar.
     */
    ChartType[ChartType["BarStacked100"] = 59] = "BarStacked100";
    /**
     * Bubble.
     */
    ChartType[ChartType["Bubble"] = 15] = "Bubble";
    /**
     * Bubble with 3D effects.
     */
    ChartType[ChartType["Bubble3DEffect"] = 87] = "Bubble3DEffect";
    /**
     * Clustered Column.
     */
    ChartType[ChartType["ColumnClustered"] = 51] = "ColumnClustered";
    /**
     * Stacked Column.
     */
    ChartType[ChartType["ColumnStacked"] = 52] = "ColumnStacked";
    /**
     * 100% Stacked Column.
     */
    ChartType[ChartType["ColumnStacked100"] = 53] = "ColumnStacked100";
    /**
     * Clustered Cone Bar.
     */
    ChartType[ChartType["ConeBarClustered"] = 102] = "ConeBarClustered";
    /**
     * Stacked Cone Bar.
     */
    ChartType[ChartType["ConeBarStacked"] = 103] = "ConeBarStacked";
    /**
     * 100% Stacked Cone Bar.
     */
    ChartType[ChartType["ConeBarStacked100"] = 104] = "ConeBarStacked100";
    /**
     * 3D Cone Column.
     */
    ChartType[ChartType["ConeCol"] = 105] = "ConeCol";
    /**
     * Clustered Cone Column.
     */
    ChartType[ChartType["ConeColClustered"] = 99] = "ConeColClustered";
    /**
     * Stacked Cone Column.
     */
    ChartType[ChartType["ConeColStacked"] = 100] = "ConeColStacked";
    /**
     * 100% Stacked Cone Column.
     */
    ChartType[ChartType["ConeColStacked100"] = 101] = "ConeColStacked100";
    /**
     * Clustered Cylinder Bar.
     */
    ChartType[ChartType["CylinderBarClustered"] = 95] = "CylinderBarClustered";
    /**
     * Stacked Cylinder Bar.
     */
    ChartType[ChartType["CylinderBarStacked"] = 96] = "CylinderBarStacked";
    /**
     * 100% Stacked Cylinder Bar.
     */
    ChartType[ChartType["CylinderBarStacked100"] = 97] = "CylinderBarStacked100";
    /**
     * 3D Cylinder Column.
     */
    ChartType[ChartType["CylinderCol"] = 98] = "CylinderCol";
    /**
     * Clustered Cone Column.
     */
    ChartType[ChartType["CylinderColClustered"] = 92] = "CylinderColClustered";
    /**
     * Stacked Cone Column.
     */
    ChartType[ChartType["CylinderColStacked"] = 93] = "CylinderColStacked";
    /**
     * 100% Stacked Cylinder Column.
     */
    ChartType[ChartType["CylinderColStacked100"] = 94] = "CylinderColStacked100";
    /**
     * Doughnut.
     */
    ChartType[ChartType["Doughnut"] = -4120] = "Doughnut";
    /**
     * Exploded Doughnut.
     */
    ChartType[ChartType["DoughnutExploded"] = 80] = "DoughnutExploded";
    /**
     * Line.
     */
    ChartType[ChartType["Line"] = 4] = "Line";
    /**
     * Line with Markers.
     */
    ChartType[ChartType["LineMarkers"] = 65] = "LineMarkers";
    /**
     * Stacked Line with Markers.
     */
    ChartType[ChartType["LineMarkersStacked"] = 66] = "LineMarkersStacked";
    /**
     * 100% Stacked Line with Markers.
     */
    ChartType[ChartType["LineMarkersStacked100"] = 67] = "LineMarkersStacked100";
    /**
     * Stacked Line.
     */
    ChartType[ChartType["LineStacked"] = 63] = "LineStacked";
    /**
     * 100% Stacked Line.
     */
    ChartType[ChartType["LineStacked100"] = 64] = "LineStacked100";
    /**
     * Pie.
     */
    ChartType[ChartType["Pie"] = 5] = "Pie";
    /**
     * Exploded Pie.
     */
    ChartType[ChartType["PieExploded"] = 69] = "PieExploded";
    /**
     * Pie of Pie.
     */
    ChartType[ChartType["PieOfPie"] = 68] = "PieOfPie";
    /**
     * Clustered Pyramid Bar.
     */
    ChartType[ChartType["PyramidBarClustered"] = 109] = "PyramidBarClustered";
    /**
     * Stacked Pyramid Bar.
     */
    ChartType[ChartType["PyramidBarStacked"] = 110] = "PyramidBarStacked";
    /**
     * 100% Stacked Pyramid Bar.
     */
    ChartType[ChartType["PyramidBarStacked100"] = 111] = "PyramidBarStacked100";
    /**
     * 3D Pyramid Column.
     */
    ChartType[ChartType["PyramidCol"] = 112] = "PyramidCol";
    /**
     * Clustered Pyramid Column.
     */
    ChartType[ChartType["PyramidColClustered"] = 106] = "PyramidColClustered";
    /**
     * Stacked Pyramid Column.
     */
    ChartType[ChartType["PyramidColStacked"] = 107] = "PyramidColStacked";
    /**
     * 100% Stacked Pyramid Column.
     */
    ChartType[ChartType["PyramidColStacked100"] = 108] = "PyramidColStacked100";
    /**
     * Radar.
     */
    ChartType[ChartType["Radar"] = -4151] = "Radar";
    /**
     * Filled Radar.
     */
    ChartType[ChartType["RadarFilled"] = 82] = "RadarFilled";
    /**
     * Radar with Data Markers.
     */
    ChartType[ChartType["RadarMarkers"] = 81] = "RadarMarkers";
    /**
     * High-Low-Close.
     */
    ChartType[ChartType["StockHLC"] = 88] = "StockHLC";
    /**
     * Open-High-Low-Close.
     */
    ChartType[ChartType["StockOHLC"] = 89] = "StockOHLC";
    /**
     * Volume-High-Low-Close.
     */
    ChartType[ChartType["StockVHLC"] = 90] = "StockVHLC";
    /**
     * Volume-Open-High-Low-Close.
     */
    ChartType[ChartType["StockVOHLC"] = 91] = "StockVOHLC";
    /**
     * Surface (a.k.a. Contour).
     */
    ChartType[ChartType["Surface"] = 83] = "Surface";
    /**
     * 3D Surface (Top View).
     */
    ChartType[ChartType["SurfaceTopView"] = 85] = "SurfaceTopView";
    /**
     * 3D Surface (Top View wireframe).
     */
    ChartType[ChartType["SurfaceTopViewWireframe"] = 86] = "SurfaceTopViewWireframe";
    /**
     * Surface (a.k.a. Contour - wireframe).
     */
    ChartType[ChartType["SurfaceWireframe"] = 84] = "SurfaceWireframe";
    /**
     * Scatter.
     */
    ChartType[ChartType["XYScatter"] = -4169] = "XYScatter";
    /**
     * Scatter with Lines.
     */
    ChartType[ChartType["XYScatterLines"] = 74] = "XYScatterLines";
    /**
     * Scatter with Lines and No Data Markers.
     */
    ChartType[ChartType["XYScatterLinesNoMarkers"] = 75] = "XYScatterLinesNoMarkers";
    /**
     * Scatter with Smoothed Lines.
     */
    ChartType[ChartType["XYScatterSmooth"] = 72] = "XYScatterSmooth";
    /**
     * Scatter with Smoothed Lines and No Data Markers.
     */
    ChartType[ChartType["XYScatterSmoothNoMarkers"] = 73] = "XYScatterSmoothNoMarkers";
    /**
     * Combination of different chart types.
     */
    ChartType[ChartType["Combo"] = 113] = "Combo";
    /**
     * Box and whisker (Office2016 or later)
     */
    ChartType[ChartType["BoxAndWhisker"] = 114] = "BoxAndWhisker";
    /**
     * Funnel (Office365 only)
     */
    ChartType[ChartType["Funnel"] = 115] = "Funnel";
    /**
     * Pareto chart with bars and lines (Office2016 or later)
     */
    ChartType[ChartType["Pareto"] = 116] = "Pareto";
    /**
     * Histogram chart (Office2016 or later)
     */
    ChartType[ChartType["Histogram"] = 117] = "Histogram";
    /**
     * Sunburst chart (Office2016 or later)
     */
    ChartType[ChartType["Sunburst"] = 118] = "Sunburst";
    /**
     * Tree map chart (Office2016 or later)
     */
    ChartType[ChartType["Treemap"] = 119] = "Treemap";
    /**
     * Waterfall chart (Office2016 or later)
     */
    ChartType[ChartType["Waterfall"] = 120] = "Waterfall";
    /**
     * Region map (geographic) chart (Office365 only)
     */
    ChartType[ChartType["RegionMap"] = 121] = "RegionMap";
    return ChartType;
})({});
/**
 * @hidden
 */
export let ChartType_$type = /*@__PURE__*/ markEnum('ChartType', 'Unknown,0|Area3D,-4098|Area3DStacked,78|Area3DStacked100,79|Bar3DClustered,60|Bar3DStacked,61|Bar3DStacked100,62|Column3D,-4100|Column3DClustered,54|Column3DStacked,55|Column3DStacked100,56|Line3D,-4101|Pie3D,-4102|Pie3DExploded,70|Area,1|AreaStacked,76|AreaStacked100,77|BarClustered,57|BarOfPie,71|BarStacked,58|BarStacked100,59|Bubble,15|Bubble3DEffect,87|ColumnClustered,51|ColumnStacked,52|ColumnStacked100,53|ConeBarClustered,102|ConeBarStacked,103|ConeBarStacked100,104|ConeCol,105|ConeColClustered,99|ConeColStacked,100|ConeColStacked100,101|CylinderBarClustered,95|CylinderBarStacked,96|CylinderBarStacked100,97|CylinderCol,98|CylinderColClustered,92|CylinderColStacked,93|CylinderColStacked100,94|Doughnut,-4120|DoughnutExploded,80|Line,4|LineMarkers,65|LineMarkersStacked,66|LineMarkersStacked100,67|LineStacked,63|LineStacked100,64|Pie,5|PieExploded,69|PieOfPie,68|PyramidBarClustered,109|PyramidBarStacked,110|PyramidBarStacked100,111|PyramidCol,112|PyramidColClustered,106|PyramidColStacked,107|PyramidColStacked100,108|Radar,-4151|RadarFilled,82|RadarMarkers,81|StockHLC,88|StockOHLC,89|StockVHLC,90|StockVOHLC,91|Surface,83|SurfaceTopView,85|SurfaceTopViewWireframe,86|SurfaceWireframe,84|XYScatter,-4169|XYScatterLines,74|XYScatterLinesNoMarkers,75|XYScatterSmooth,72|XYScatterSmoothNoMarkers,73|Combo,113|BoxAndWhisker,114|Funnel,115|Pareto,116|Histogram,117|Sunburst,118|Treemap,119|Waterfall,120|RegionMap,121');
