// igniteui-react-excel/esm2015/lib/ChartStylePartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ChartStylePartManager = /*@__PURE__*/ (() => {
    class ChartStylePartManager extends XmlPartManagerBase {
        get_m() {
            return ChartStylePartManager.a4;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/charts/style.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return ChartStylePartManager.a6;
                case 1: return ChartStylePartManager.a5;
                default: return ChartStylePartManager.a6;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    ChartStylePartManager.$t = /*@__PURE__*/ markType(ChartStylePartManager, 'ChartStylePartManager', XmlPartManagerBase.$);
    ChartStylePartManager.a4 = "application/vnd.ms-office.chartstyle+xml";
    ChartStylePartManager.a6 = "http://schemas.microsoft.com/office/2011/relationships/chartStyle";
    ChartStylePartManager.a5 = "http://schemas.microsoft.com/office/2011/relationships/chartStyle";
    return ChartStylePartManager;
})();
