// igniteui-react-excel/esm2015/lib/Colors.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { Color } from "igniteui-react-core";
/**
 * @hidden
 */
export let Colors = /*@__PURE__*/ (() => {
    class Colors extends Base {
        static get a() {
            return Color.u(255, 0, 0, 0);
        }
        static get b() {
            return Color.u(255, 0, 0, 255);
        }
        static get c() {
            return Color.u(255, 165, 42, 42);
        }
        static get d() {
            return Color.u(255, 0, 255, 255);
        }
        static get e() {
            return Color.u(255, 169, 169, 169);
        }
        static get f() {
            return Color.u(255, 128, 128, 128);
        }
        static get g() {
            return Color.u(255, 0, 128, 0);
        }
        static get h() {
            return Color.u(255, 173, 255, 47);
        }
        static get i() {
            return Color.u(255, 240, 255, 240);
        }
        static get j() {
            return Color.u(255, 211, 211, 211);
        }
        static get k() {
            return Color.u(255, 255, 0, 255);
        }
        static get l() {
            return Color.u(255, 255, 165, 0);
        }
        static get m() {
            return Color.u(255, 255, 192, 203);
        }
        static get n() {
            return Color.u(255, 128, 0, 128);
        }
        static get o() {
            return Color.u(255, 255, 0, 0);
        }
        static get p() {
            return Color.u(255, 65, 105, 225);
        }
        static get q() {
            return Color.u(255, 0, 128, 128);
        }
        static get r() {
            return Color.u(0, 255, 255, 255);
        }
        static get s() {
            return Color.u(255, 245, 222, 179);
        }
        static get t() {
            return Color.u(255, 255, 255, 255);
        }
        static get u() {
            return Color.u(255, 255, 255, 0);
        }
    }
    Colors.$t = /*@__PURE__*/ markType(Colors, 'Colors');
    return Colors;
})();
