// igniteui-react-excel/esm2015/lib/CondFmtBase.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { ConditionBase } from "./ConditionBase";
import { stringFormat } from "igniteui-react-core";
/**
 * @hidden
 */
export let CondFmtBase = /*@__PURE__*/ (() => {
    class CondFmtBase extends Base {
        constructor() {
            super(...arguments);
            this.e = 0;
            this.b = false;
            this.d = 0;
            this.a = null;
            this.c = null;
        }
        toString() {
            if (this.c.count == 0) {
                return super.toString();
            }
            let a = this.c.item(0).worksheet;
            let b = ConditionBase._a1(a, this.c);
            return stringFormat("ID={0} ({1})", this.d, b);
        }
    }
    CondFmtBase.$t = /*@__PURE__*/ markType(CondFmtBase, 'CondFmtBase');
    return CondFmtBase;
})();
