// igniteui-react-excel/esm2015/lib/CondFmtWrapper_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CondFmtBase } from "./CondFmtBase";
import { Base, markType } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
import { CfCommonProps } from "./CfCommonProps";
/**
 * @hidden
 */
export let CondFmtWrapper = /*@__PURE__*/ (() => {
    class CondFmtWrapper extends CondFmtBase {
        constructor() {
            super();
            this.i = null;
            this.g = null;
            this.h = false;
            this.i = new List$1(CfWrapper.$, 0);
        }
    }
    CondFmtWrapper.$t = /*@__PURE__*/ markType(CondFmtWrapper, 'CondFmtWrapper', CondFmtBase.$);
    return CondFmtWrapper;
})();
/**
 * @hidden
 */
export let CfWrapper = /*@__PURE__*/ (() => {
    class CfWrapper extends Base {
        constructor() {
            super();
            this.e = null;
            this.l = null;
            this.m = 0;
            this.k = 0;
            this.n = null;
            this.a = null;
            this.b = null;
            this.c = null;
            this.i = null;
            this.h = null;
            this.d = null;
            this.g = null;
            this.f = null;
            this.j = null;
            this.e = new CfCommonProps();
        }
        o(a) {
            let b = a.b.i.indexOf(this);
        }
    }
    CfWrapper.$t = /*@__PURE__*/ markType(CfWrapper, 'CfWrapper');
    return CfWrapper;
})();
/**
 * @hidden
 */
export let CfExWrapper = /*@__PURE__*/ (() => {
    class CfExWrapper extends Base {
        constructor() {
            super(...arguments);
            this.f = 0;
            this.e = false;
            this.d = false;
            this.c = null;
            this.b = null;
            this.a = null;
        }
    }
    CfExWrapper.$t = /*@__PURE__*/ markType(CfExWrapper, 'CfExWrapper');
    return CfExWrapper;
})();
