// igniteui-react-excel/esm2015/lib/ConnectionsPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ConnectionsPartManager = /*@__PURE__*/ (() => {
    class ConnectionsPartManager extends XmlPartManagerBase {
        get_m() {
            return ConnectionsPartManager.a3;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/connections.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return ConnectionsPartManager.a6;
                case 1: return ConnectionsPartManager.a5;
                default: return ConnectionsPartManager.a6;
            }
        }
        get_g() {
            return false;
        }
        get g() {
            return this.get_g();
        }
    }
    ConnectionsPartManager.$t = /*@__PURE__*/ markType(ConnectionsPartManager, 'ConnectionsPartManager', XmlPartManagerBase.$);
    ConnectionsPartManager.a3 = "application/vnd.openxmlformats-officedocument.spreadsheetml.connections+xml";
    ConnectionsPartManager.a6 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/connections";
    ConnectionsPartManager.a5 = "http://purl.oclc.org/ooxml/officeDocument/relationships/connections";
    return ConnectionsPartManager;
})();
