// igniteui-react-excel/esm2015/lib/CorePropertiesPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CorePropertiesPartManager = /*@__PURE__*/ (() => {
    class CorePropertiesPartManager extends XmlPartManagerBase {
        get_m() {
            return CorePropertiesPartManager.a3;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            return "/docProps/core.xml";
        }
        o(a) {
            switch (a) {
                case 0: return CorePropertiesPartManager.a6;
                case 1: return CorePropertiesPartManager.a5;
                default: return CorePropertiesPartManager.a6;
            }
        }
        get_g() {
            return false;
        }
        get g() {
            return this.get_g();
        }
    }
    CorePropertiesPartManager.$t = /*@__PURE__*/ markType(CorePropertiesPartManager, 'CorePropertiesPartManager', XmlPartManagerBase.$);
    CorePropertiesPartManager.a3 = "application/vnd.openxmlformats-package.core-properties+xml";
    CorePropertiesPartManager.a6 = "http://schemas.openxmlformats.org/package/2006/relationships/metadata/core-properties";
    CorePropertiesPartManager.a5 = "http://schemas.openxmlformats.org/package/2006/relationships/metadata/core-properties";
    return CorePropertiesPartManager;
})();
