// igniteui-react-excel/esm2015/lib/CT_AudioCDTime.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt8_1, parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_AudioCDTime = /*@__PURE__*/ (() => {
    class CT_AudioCDTime extends Base {
        constructor() {
            super();
            this.b = 0;
            this.d = toNullable(Number_$type, null);
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        static g(a, b, c) {
            b = 0;
            c = toNullable(Number_$type, null);
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "track":
                        b = parseUInt8_1(d.d, CultureInfo.invariantCulture);
                        break;
                    case "time":
                        c = toNullable(Number_$type, parseUInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static h(a, b, c, d) {
            a.ad.p(b, "a");
            let e = null;
            let f = null;
            if (c.hasValue) {
                f = "time";
                e = c.value.toString();
                a.ad.h(f, e);
            }
            f = "track";
            e = d.toString();
            a.ad.h(f, e);
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_AudioCDTime.$t = /*@__PURE__*/ markType(CT_AudioCDTime, 'CT_AudioCDTime');
    return CT_AudioCDTime;
})();
