// igniteui-react-excel/esm2015/lib/CT_AudioCD.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_AudioCDTime } from "./CT_AudioCDTime";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
export let CT_AudioCD = /*@__PURE__*/ (() => {
    class CT_AudioCD extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
            this.f = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 1588:
                    this.c = typeCast(CT_AudioCDTime.$, b);
                    break;
                case 573:
                    this.a = typeCast(CT_AudioCDTime.$, b);
                    break;
                case 601:
                    this.e = typeCast(CT_OfficeArtExtensionList.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_AudioCD.$t = /*@__PURE__*/ markType(CT_AudioCD, 'CT_AudioCD', Base.$, [ISupportSetProperty_$type]);
    return CT_AudioCD;
})();
