// igniteui-react-excel/esm2015/lib/CT_AnimationDgmBuildProperties.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_AnimationDgmBuildType } from "./ST_AnimationDgmBuildType";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_AnimationDgmBuildProperties = /*@__PURE__*/ (() => {
    class CT_AnimationDgmBuildProperties extends Base {
        constructor() {
            super();
            this.b = null;
            this.e = toNullable(Boolean_$type, null);
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        static g(a, b, c) {
            b = null;
            c = toNullable(Boolean_$type, null);
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "bld":
                        b = ST_AnimationDgmBuildType.a(d.d);
                        break;
                    case "rev":
                        c = toNullable(Boolean_$type, ST_OnOff.e(d.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static h(a, b, c = null, d = toNullable(Boolean_$type, null)) {
            a.ad.p(b, "a");
            let e = null;
            let f = null;
            if (c != null) {
                f = "bld";
                e = c.g();
                a.ad.h(f, e);
            }
            if (d.hasValue) {
                f = "rev";
                e = ST_OnOff.g(d.value);
                a.ad.h(f, e);
            }
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_AnimationDgmBuildProperties.$t = /*@__PURE__*/ markType(CT_AnimationDgmBuildProperties, 'CT_AnimationDgmBuildProperties');
    return CT_AnimationDgmBuildProperties;
})();
