// igniteui-react-excel/esm2015/lib/CT_BorderPr.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { ColorInfo } from "./ColorInfo";
import { ST_BorderStyle_$type } from "./ST_BorderStyle";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
export let CT_BorderPr = /*@__PURE__*/ (() => {
    class CT_BorderPr extends Base {
        constructor() {
            super();
            this.a = null;
            this.b = toNullable(ST_BorderStyle_$type, null);
        }
        static d(a, b) {
            b = toNullable(ST_BorderStyle_$type, null);
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "style":
                        b = toNullable(ST_BorderStyle_$type, XmlNamespaceDefinitionBase.b(ST_BorderStyle_$type, c.d));
                        break;
                    default: break;
                }
            }
            return {
                p1: b
            };
        }
        static e(a, b, c = toNullable(ST_BorderStyle_$type, null)) {
            a.ad.p(b, null);
            let d = null;
            let e = null;
            if (c.hasValue) {
                e = "style";
                d = XmlNamespaceDefinitionBase.p(ST_BorderStyle_$type, c.value);
                a.ad.h(e, d);
            }
        }
        setProperty(a, b) {
            switch (a) {
                case 302:
                    this.a = typeCast(ColorInfo.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_BorderPr.$t = /*@__PURE__*/ markType(CT_BorderPr, 'CT_BorderPr', Base.$, [ISupportSetProperty_$type]);
    return CT_BorderPr;
})();
