// igniteui-react-excel/esm2015/lib/CT_BaseStyles.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_ColorScheme } from "./CT_ColorScheme";
import { CT_FontScheme } from "./CT_FontScheme";
import { CT_StyleMatrix } from "./CT_StyleMatrix";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
export let CT_BaseStyles = /*@__PURE__*/ (() => {
    class CT_BaseStyles extends Base {
        constructor() {
            super();
            this.b = null;
            this.d = null;
            this.h = null;
            this.f = null;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 286:
                    this.a = typeCast(CT_ColorScheme.$, b);
                    break;
                case 661:
                    this.c = typeCast(CT_FontScheme.$, b);
                    break;
                case 657:
                    this.g = typeCast(CT_StyleMatrix.$, b);
                    break;
                case 601:
                    this.e = typeCast(CT_OfficeArtExtensionList.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_BaseStyles.$t = /*@__PURE__*/ markType(CT_BaseStyles, 'CT_BaseStyles', Base.$, [ISupportSetProperty_$type]);
    return CT_BaseStyles;
})();
