// igniteui-react-excel/esm2015/lib/CT_BackgroundFormatting.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { EG_FillProperties } from "./EG_FillProperties";
import { EG_EffectProperties } from "./EG_EffectProperties";
/**
 * @hidden
 */
export let CT_BackgroundFormatting = /*@__PURE__*/ (() => {
    class CT_BackgroundFormatting extends Base {
        constructor() {
            super();
            this.c = null;
            this.a = null;
        }
        get d() {
            if (this.c == null) {
                this.c = new EG_FillProperties();
            }
            return this.c;
        }
        get b() {
            if (this.a == null) {
                this.a = new EG_EffectProperties();
            }
            return this.a;
        }
        setProperty(a, b) {
            switch (a) {
                default:
                    if (this.d.f(a)) {
                        this.d.j(a, b);
                        return;
                    }
                    else if (this.b.c(a)) {
                        this.b.e(a, b);
                        return;
                    }
                    break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_BackgroundFormatting.$t = /*@__PURE__*/ markType(CT_BackgroundFormatting, 'CT_BackgroundFormatting', Base.$, [ISupportSetProperty_$type]);
    return CT_BackgroundFormatting;
})();
