// igniteui-react-excel/esm2015/lib/CT_Blip.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, markType } from "igniteui-react-core";
import { ST_BlipCompression_$type } from "./ST_BlipCompression";
/**
 * @hidden
 */
export let CT_Blip = /*@__PURE__*/ (() => {
    class CT_Blip extends Base {
        constructor() {
            super(...arguments);
            this.c = toNullable(ST_BlipCompression_$type, null);
            this.d = null;
            this.e = null;
            this.a = null;
            this.b = null;
        }
    }
    CT_Blip.$t = /*@__PURE__*/ markType(CT_Blip, 'CT_Blip');
    return CT_Blip;
})();
