// igniteui-react-excel/esm2015/lib/CT_BooleanProperty.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_BooleanProperty = /*@__PURE__*/ (() => {
    class CT_BooleanProperty extends Base {
        constructor() {
            super();
            this.c = toNullable(Boolean_$type, null);
        }
        get b() {
            return this.c.hasValue ? this.c.value : true;
        }
        static e(a, b) {
            b = toNullable(Boolean_$type, null);
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "val":
                        b = toNullable(Boolean_$type, ST_OnOff.e(c.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static f(a, b, c = toNullable(Boolean_$type, null)) {
            a.ad.p(b, null);
            let d = null;
            let e = null;
            if (c.hasValue) {
                e = "val";
                d = ST_OnOff.g(c.value);
                a.ad.h(e, d);
            }
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_BooleanProperty.$t = /*@__PURE__*/ markType(CT_BooleanProperty, 'CT_BooleanProperty');
    return CT_BooleanProperty;
})();
