// igniteui-react-excel/esm2015/lib/CT_CustomFilters_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_CustomFilters_SML2009 = /*@__PURE__*/ (() => {
    class CT_CustomFilters_SML2009 extends Base {
        constructor() {
            super();
            this.c = null;
            this.e = toNullable(Boolean_$type, null);
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        static f(a, b) {
            b = toNullable(Boolean_$type, null);
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "and":
                        b = toNullable(Boolean_$type, ST_OnOff.e(c.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static g(a, b, c = toNullable(Boolean_$type, null)) {
            a.ad.p(b, null);
            let d = null;
            let e = null;
            if (c.hasValue) {
                e = "and";
                d = ST_OnOff.g(c.value);
                a.ad.h(e, d);
            }
            ;
        }
    }
    CT_CustomFilters_SML2009.$t = /*@__PURE__*/ markType(CT_CustomFilters_SML2009, 'CT_CustomFilters_SML2009');
    return CT_CustomFilters_SML2009;
})();
