// igniteui-react-excel/esm2015/lib/CT_ColorScheme.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_Color_DML } from "./CT_Color_DML";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
export let CT_ColorScheme = /*@__PURE__*/ (() => {
    class CT_ColorScheme extends Base {
        constructor() {
            super();
            this.n = null;
            this.v = null;
            this.p = null;
            this.x = null;
            this.b = null;
            this.d = null;
            this.f = null;
            this.h = null;
            this.j = null;
            this.l = null;
            this.t = null;
            this.r = null;
            this.z = null;
            this.ab = null;
        }
        get m() {
            return this.n;
        }
        set m(a) {
            this.n = a;
        }
        get u() {
            return this.v;
        }
        set u(a) {
            this.v = a;
        }
        get o() {
            return this.p;
        }
        set o(a) {
            this.p = a;
        }
        get w() {
            return this.x;
        }
        set w(a) {
            this.x = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get i() {
            return this.j;
        }
        set i(a) {
            this.j = a;
        }
        get k() {
            return this.l;
        }
        set k(a) {
            this.l = a;
        }
        get s() {
            return this.t;
        }
        set s(a) {
            this.t = a;
        }
        get q() {
            return this.r;
        }
        set q(a) {
            this.r = a;
        }
        get y() {
            return this.z;
        }
        set y(a) {
            this.z = a;
        }
        get aa() {
            return this.ab;
        }
        set aa(a) {
            this.ab = a;
        }
        static ad(a, b) {
            b = null;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "name":
                        b = c.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static ae(a, b, c) {
            a.ad.p(b, "a");
            let d = null;
            let e = null;
            if (c != null) {
                e = "name";
                d = c;
                a.ad.h(e, d);
            }
            ;
        }
        setProperty(a, b) {
            switch (a) {
                case 503:
                    this.m = typeCast(CT_Color_DML.$, b);
                    break;
                case 939:
                    this.u = typeCast(CT_Color_DML.$, b);
                    break;
                case 504:
                    this.o = typeCast(CT_Color_DML.$, b);
                    break;
                case 940:
                    this.w = typeCast(CT_Color_DML.$, b);
                    break;
                case 7:
                    this.a = typeCast(CT_Color_DML.$, b);
                    break;
                case 8:
                    this.c = typeCast(CT_Color_DML.$, b);
                    break;
                case 9:
                    this.e = typeCast(CT_Color_DML.$, b);
                    break;
                case 10:
                    this.g = typeCast(CT_Color_DML.$, b);
                    break;
                case 11:
                    this.i = typeCast(CT_Color_DML.$, b);
                    break;
                case 12:
                    this.k = typeCast(CT_Color_DML.$, b);
                    break;
                case 790:
                    this.s = typeCast(CT_Color_DML.$, b);
                    break;
                case 658:
                    this.q = typeCast(CT_Color_DML.$, b);
                    break;
                case 601:
                    this.y = typeCast(CT_OfficeArtExtensionList.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_ColorScheme.$t = /*@__PURE__*/ markType(CT_ColorScheme, 'CT_ColorScheme', Base.$, [ISupportSetProperty_$type]);
    return CT_ColorScheme;
})();
