// igniteui-react-excel/esm2015/lib/CT_CellProtection.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_CellProtection = /*@__PURE__*/ (() => {
    class CT_CellProtection extends Base {
        constructor() {
            super(...arguments);
            this.b = toNullable(Boolean_$type, null);
            this.a = toNullable(Boolean_$type, null);
        }
        static d(a, b, c) {
            b = toNullable(Boolean_$type, null);
            c = toNullable(Boolean_$type, null);
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "locked":
                        b = toNullable(Boolean_$type, ST_OnOff.e(d.d));
                        break;
                    case "hidden":
                        c = toNullable(Boolean_$type, ST_OnOff.e(d.d));
                        break;
                    default: break;
                }
            }
            return {
                p1: b,
                p2: c
            };
        }
        static e(a, b, c = toNullable(Boolean_$type, null), d = toNullable(Boolean_$type, null)) {
            a.ad.p(b, null);
            let e = null;
            let f = null;
            if (c.hasValue) {
                f = "hidden";
                e = ST_OnOff.g(c.value);
                a.ad.h(f, e);
            }
            if (d.hasValue) {
                f = "locked";
                e = ST_OnOff.g(d.value);
                a.ad.h(f, e);
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_CellProtection.$t = /*@__PURE__*/ markType(CT_CellProtection, 'CT_CellProtection');
    return CT_CellProtection;
})();
