// igniteui-react-excel/esm2015/lib/CT_ColorSchemeAndMapping.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_ColorScheme } from "./CT_ColorScheme";
import { CT_ColorMapping } from "./CT_ColorMapping";
/**
 * @hidden
 */
export let CT_ColorSchemeAndMapping = /*@__PURE__*/ (() => {
    class CT_ColorSchemeAndMapping extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 286:
                    this.c = typeCast(CT_ColorScheme.$, b);
                    break;
                case 282:
                    this.a = typeCast(CT_ColorMapping.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_ColorSchemeAndMapping.$t = /*@__PURE__*/ markType(CT_ColorSchemeAndMapping, 'CT_ColorSchemeAndMapping', Base.$, [ISupportSetProperty_$type]);
    return CT_ColorSchemeAndMapping;
})();
