// igniteui-react-excel/esm2015/lib/CT_CacheHierarchy_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_CacheHierarchy_SML2009 = /*@__PURE__*/ (() => {
    class CT_CacheHierarchy_SML2009 extends Base {
        constructor() {
            super();
            this.b = null;
            this.h = toNullable(Boolean_$type, null);
            this.n = toNullable(Boolean_$type, null);
            this.j = toNullable(Boolean_$type, null);
            this.l = toNullable(Boolean_$type, null);
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get m() {
            return this.n;
        }
        set m(a) {
            this.n = a;
        }
        get i() {
            return this.j;
        }
        set i(a) {
            this.j = a;
        }
        get k() {
            return this.l;
        }
        set k(a) {
            this.l = a;
        }
        static o(a, b, c, d, e) {
            b = toNullable(Boolean_$type, null);
            c = toNullable(Boolean_$type, null);
            d = toNullable(Boolean_$type, null);
            e = toNullable(Boolean_$type, null);
            for (let f of fromEnum(a.ab.u)) {
                switch (f.a.e(a.d)) {
                    case "flattenHierarchies":
                        b = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "measuresSet":
                        c = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "hierarchizeDistinct":
                        d = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "ignore":
                        e = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d,
                p4: e
            };
        }
        static p(a, b, c = toNullable(Boolean_$type, null), d = toNullable(Boolean_$type, null), e = toNullable(Boolean_$type, null), f = toNullable(Boolean_$type, null)) {
            a.ad.p(b, null);
            let g = null;
            let h = null;
            if (c.hasValue) {
                h = "flattenHierarchies";
                g = ST_OnOff.g(c.value);
                a.ad.h(h, g);
            }
            if (d.hasValue) {
                h = "hierarchizeDistinct";
                g = ST_OnOff.g(d.value);
                a.ad.h(h, g);
            }
            if (e.hasValue) {
                h = "ignore";
                g = ST_OnOff.g(e.value);
                a.ad.h(h, g);
            }
            if (f.hasValue) {
                h = "measuresSet";
                g = ST_OnOff.g(f.value);
                a.ad.h(h, g);
            }
            ;
        }
    }
    CT_CacheHierarchy_SML2009.$t = /*@__PURE__*/ markType(CT_CacheHierarchy_SML2009, 'CT_CacheHierarchy_SML2009');
    return CT_CacheHierarchy_SML2009;
})();
