// igniteui-react-excel/esm2015/lib/CT_ConditionalFormat.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_PivotAreas } from "./CT_PivotAreas";
import { CT_ExtensionList } from "./CT_ExtensionList";
import { ST_Scope_$type } from "./ST_Scope";
import { ST_Type_$type } from "./ST_Type";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_ConditionalFormat = /*@__PURE__*/ (() => {
    class CT_ConditionalFormat extends Base {
        constructor() {
            super();
            this.b = null;
            this.a = null;
            this.d = 0;
            this.f = 0;
            this.h = 0;
        }
        static i(a, b, c, d) {
            b = 0;
            c = 0;
            d = 0;
            for (let e of fromEnum(a.ab.u)) {
                switch (e.a.e(a.d)) {
                    case "scope":
                        b = XmlNamespaceDefinitionBase.b(ST_Scope_$type, e.d);
                        break;
                    case "type":
                        c = XmlNamespaceDefinitionBase.b(ST_Type_$type, e.d);
                        break;
                    case "priority":
                        d = parseUInt32_1(e.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            return {
                p1: b,
                p2: c,
                p3: d
            };
        }
        static j(a, b, c, d, e) {
            a.ad.p(b, null);
            let f = null;
            let g = null;
            g = "priority";
            f = c.toString();
            a.ad.h(g, f);
            g = "scope";
            f = XmlNamespaceDefinitionBase.p(ST_Scope_$type, d);
            a.ad.h(g, f);
            g = "type";
            f = XmlNamespaceDefinitionBase.p(ST_Type_$type, e);
            a.ad.h(g, f);
        }
        setProperty(a, b) {
            switch (a) {
                case 1204:
                    this.b = typeCast(CT_PivotAreas.$, b);
                    break;
                case 601:
                    this.a = typeCast(CT_ExtensionList.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_ConditionalFormat.$t = /*@__PURE__*/ markType(CT_ConditionalFormat, 'CT_ConditionalFormat', Base.$, [ISupportSetProperty_$type]);
    return CT_ConditionalFormat;
})();
