// igniteui-react-excel/esm2015/lib/CT_ConditionalFormats.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { List$1 } from "igniteui-react-core";
import { CT_ConditionalFormat } from "./CT_ConditionalFormat";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_ConditionalFormats = /*@__PURE__*/ (() => {
    class CT_ConditionalFormats extends Base {
        constructor() {
            super();
            this.a = null;
            this.c = 0;
            this.a = new List$1(CT_ConditionalFormat.$, 0);
        }
        static e(a, b) {
            b = 0;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "count":
                        b = parseUInt32_1(c.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            return {
                p1: b
            };
        }
        static f(a, b, c) {
            a.ad.p(b, null);
            let d = null;
            let e = null;
            e = "count";
            d = c.toString();
            a.ad.h(e, d);
        }
        setProperty(a, b) {
            switch (a) {
                case 333:
                    this.a.add(typeCast(CT_ConditionalFormat.$, b));
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_ConditionalFormats.$t = /*@__PURE__*/ markType(CT_ConditionalFormats, 'CT_ConditionalFormats', Base.$, [ISupportSetProperty_$type]);
    return CT_ConditionalFormats;
})();
