// igniteui-react-excel/esm2015/lib/CT_CategoryFilterException.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, String_$type, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_ShapeProperties_DML } from "./CT_ShapeProperties_DML";
import { CT_UnsignedInt } from "./CT_UnsignedInt";
import { CT_Boolean } from "./CT_Boolean";
import { CT_Marker_Chart } from "./CT_Marker_Chart";
import { CT_DLbl } from "./CT_DLbl";
/**
 * @hidden
 */
export let CT_CategoryFilterException = /*@__PURE__*/ (() => {
    class CT_CategoryFilterException extends Base {
        constructor() {
            super();
            this.n = null;
            this.l = null;
            this.f = null;
            this.d = null;
            this.b = null;
            this.j = null;
            this.h = null;
        }
        get m() {
            return this.n;
        }
        set m(a) {
            this.n = a;
        }
        get k() {
            return this.l;
        }
        set k(a) {
            this.l = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get i() {
            return this.j;
        }
        set i(a) {
            this.j = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 1583:
                    this.m = typeCast(String_$type, b);
                    break;
                case 1570:
                    this.k = typeCast(CT_ShapeProperties_DML.$, b);
                    break;
                case 598:
                    this.e = typeCast(CT_UnsignedInt.$, b);
                    break;
                case 848:
                    this.c = typeCast(CT_Boolean.$, b);
                    break;
                case 183:
                    this.a = typeCast(CT_Boolean.$, b);
                    break;
                case 979:
                    this.i = typeCast(CT_Marker_Chart.$, b);
                    break;
                case 401:
                    this.g = typeCast(CT_DLbl.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_CategoryFilterException.$t = /*@__PURE__*/ markType(CT_CategoryFilterException, 'CT_CategoryFilterException', Base.$, [ISupportSetProperty_$type]);
    return CT_CategoryFilterException;
})();
