// igniteui-react-excel/esm2015/lib/CT_CellAlignment.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_HorizontalAlignment_$type } from "./ST_HorizontalAlignment";
import { ST_VerticalAlignment_$type } from "./ST_VerticalAlignment";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { CultureInfo } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
import { parseUInt32_1, parseInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_CellAlignment = /*@__PURE__*/ (() => {
    class CT_CellAlignment extends Base {
        constructor() {
            super(...arguments);
            this.a = toNullable(ST_HorizontalAlignment_$type, null);
            this.h = toNullable(ST_VerticalAlignment_$type, null);
            this.g = toNullable(Number_$type, null);
            this.i = toNullable(Boolean_$type, null);
            this.b = toNullable(Number_$type, null);
            this.e = toNullable(Number_$type, null);
            this.c = toNullable(Boolean_$type, null);
            this.f = toNullable(Boolean_$type, null);
            this.d = toNullable(Number_$type, null);
        }
        static k(a, b, c, d, e, f, g, h, i, j) {
            b = toNullable(ST_HorizontalAlignment_$type, null);
            c = toNullable(ST_VerticalAlignment_$type, null);
            d = toNullable(Number_$type, null);
            e = toNullable(Boolean_$type, null);
            f = toNullable(Number_$type, null);
            g = toNullable(Number_$type, null);
            h = toNullable(Boolean_$type, null);
            i = toNullable(Boolean_$type, null);
            j = toNullable(Number_$type, null);
            for (let k of fromEnum(a.ab.u)) {
                switch (k.a.e(a.d)) {
                    case "horizontal":
                        b = toNullable(ST_HorizontalAlignment_$type, XmlNamespaceDefinitionBase.b(ST_HorizontalAlignment_$type, k.d));
                        break;
                    case "vertical":
                        c = toNullable(ST_VerticalAlignment_$type, XmlNamespaceDefinitionBase.b(ST_VerticalAlignment_$type, k.d));
                        break;
                    case "textRotation":
                        d = toNullable(Number_$type, parseUInt32_1(k.d, CultureInfo.invariantCulture));
                        break;
                    case "wrapText":
                        e = toNullable(Boolean_$type, ST_OnOff.e(k.d));
                        break;
                    case "indent":
                        f = toNullable(Number_$type, parseUInt32_1(k.d, CultureInfo.invariantCulture));
                        break;
                    case "relativeIndent":
                        g = toNullable(Number_$type, parseInt32_1(k.d, CultureInfo.invariantCulture));
                        break;
                    case "justifyLastLine":
                        h = toNullable(Boolean_$type, ST_OnOff.e(k.d));
                        break;
                    case "shrinkToFit":
                        i = toNullable(Boolean_$type, ST_OnOff.e(k.d));
                        break;
                    case "readingOrder":
                        j = toNullable(Number_$type, parseUInt32_1(k.d, CultureInfo.invariantCulture));
                        break;
                    default: break;
                }
            }
            return {
                p1: b,
                p2: c,
                p3: d,
                p4: e,
                p5: f,
                p6: g,
                p7: h,
                p8: i,
                p9: j
            };
        }
        static l(a, b, c = toNullable(ST_HorizontalAlignment_$type, null), d = toNullable(Number_$type, null), e = toNullable(Boolean_$type, null), f = toNullable(Number_$type, null), g = toNullable(Number_$type, null), h = toNullable(Boolean_$type, null), i = toNullable(Number_$type, null), j = toNullable(ST_VerticalAlignment_$type, null), k = toNullable(Boolean_$type, null)) {
            a.ad.p(b, null);
            let l = null;
            let m = null;
            if (c.hasValue) {
                m = "horizontal";
                l = XmlNamespaceDefinitionBase.p(ST_HorizontalAlignment_$type, c.value);
                a.ad.h(m, l);
            }
            if (d.hasValue) {
                m = "indent";
                l = d.value.toString();
                a.ad.h(m, l);
            }
            if (e.hasValue) {
                m = "justifyLastLine";
                l = ST_OnOff.g(e.value);
                a.ad.h(m, l);
            }
            if (f.hasValue) {
                m = "readingOrder";
                l = f.value.toString();
                a.ad.h(m, l);
            }
            if (g.hasValue) {
                m = "relativeIndent";
                l = g.value.toString();
                a.ad.h(m, l);
            }
            if (h.hasValue) {
                m = "shrinkToFit";
                l = ST_OnOff.g(h.value);
                a.ad.h(m, l);
            }
            if (i.hasValue) {
                m = "textRotation";
                l = i.value.toString();
                a.ad.h(m, l);
            }
            if (j.hasValue) {
                m = "vertical";
                l = XmlNamespaceDefinitionBase.p(ST_VerticalAlignment_$type, j.value);
                a.ad.h(m, l);
            }
            if (k.hasValue) {
                m = "wrapText";
                l = ST_OnOff.g(k.value);
                a.ad.h(m, l);
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_CellAlignment.$t = /*@__PURE__*/ markType(CT_CellAlignment, 'CT_CellAlignment');
    return CT_CellAlignment;
})();
