// igniteui-react-excel/esm2015/lib/CT_ColorScale_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, NotSupportedException, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_ColorScale_SML2009 = /*@__PURE__*/ (() => {
    class CT_ColorScale_SML2009 extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static e(a) {
            throw new NotSupportedException(1, "The corresponding XML element for this class has no attributes, so there is nothing to read.");
        }
        static f(a, b) {
            throw new NotSupportedException(1, "The corresponding XML element for this class has no attributes, so there is nothing to write.");
        }
    }
    CT_ColorScale_SML2009.$t = /*@__PURE__*/ markType(CT_ColorScale_SML2009, 'CT_ColorScale_SML2009');
    return CT_ColorScale_SML2009;
})();
