// igniteui-react-excel/esm2015/lib/CT_ConditionalFormat_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_Scope_$type } from "./ST_Scope";
import { ST_Type_$type } from "./ST_Type";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_ConditionalFormat_SML2009 = /*@__PURE__*/ (() => {
    class CT_ConditionalFormat_SML2009 extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
            this.j = toNullable(ST_Scope_$type, null);
            this.l = toNullable(ST_Type_$type, null);
            this.h = toNullable(Number_$type, null);
            this.n = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get i() {
            return this.j;
        }
        set i(a) {
            this.j = a;
        }
        get k() {
            return this.l;
        }
        set k(a) {
            this.l = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get m() {
            return this.n;
        }
        set m(a) {
            this.n = a;
        }
        static o(a, b, c, d, e) {
            b = toNullable(ST_Scope_$type, null);
            c = toNullable(ST_Type_$type, null);
            d = toNullable(Number_$type, null);
            e = null;
            for (let f of fromEnum(a.ab.u)) {
                switch (f.a.e(a.d)) {
                    case "scope":
                        b = toNullable(ST_Scope_$type, XmlNamespaceDefinitionBase.b(ST_Scope_$type, f.d));
                        break;
                    case "type":
                        c = toNullable(ST_Type_$type, XmlNamespaceDefinitionBase.b(ST_Type_$type, f.d));
                        break;
                    case "priority":
                        d = toNullable(Number_$type, parseUInt32_1(f.d, CultureInfo.invariantCulture));
                        break;
                    case "id":
                        e = f.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d,
                p4: e
            };
        }
        static p(a, b, c, d, e, f) {
            a.ad.p(b, null);
            let g = null;
            let h = null;
            if (c.hasValue) {
                h = "priority";
                g = c.value.toString();
                a.ad.h(h, g);
            }
            if (d.hasValue) {
                h = "scope";
                g = XmlNamespaceDefinitionBase.p(ST_Scope_$type, d.value);
                a.ad.h(h, g);
            }
            if (e.hasValue) {
                h = "type";
                g = XmlNamespaceDefinitionBase.p(ST_Type_$type, e.value);
                a.ad.h(h, g);
            }
            if (f != null) {
                h = "id";
                g = f;
                a.ad.h(h, g);
            }
            ;
        }
    }
    CT_ConditionalFormat_SML2009.$t = /*@__PURE__*/ markType(CT_ConditionalFormat_SML2009, 'CT_ConditionalFormat_SML2009');
    return CT_ConditionalFormat_SML2009;
})();
