// igniteui-react-excel/esm2015/lib/CT_ConnectorLocking.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_ConnectorLocking = /*@__PURE__*/ (() => {
    class CT_ConnectorLocking extends Base {
        constructor() {
            super(...arguments);
            this.f = false;
            this.j = false;
            this.i = false;
            this.c = false;
            this.g = false;
            this.h = false;
            this.e = false;
            this.a = false;
            this.b = false;
            this.d = false;
        }
    }
    CT_ConnectorLocking.$t = /*@__PURE__*/ markType(CT_ConnectorLocking, 'CT_ConnectorLocking');
    return CT_ConnectorLocking;
})();
