// igniteui-react-excel/esm2015/lib/CT_ConditionalFormatting_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_ConditionalFormatting_SML2009 = /*@__PURE__*/ (() => {
    class CT_ConditionalFormatting_SML2009 extends Base {
        constructor() {
            super();
            this.g = null;
            this.d = null;
            this.b = null;
            this.i = toNullable(Boolean_$type, null);
        }
        get f() {
            return this.g;
        }
        set f(a) {
            this.g = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get h() {
            return this.i;
        }
        set h(a) {
            this.i = a;
        }
        static j(a, b) {
            b = toNullable(Boolean_$type, null);
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "pivot":
                        b = toNullable(Boolean_$type, ST_OnOff.e(c.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static k(a, b, c = toNullable(Boolean_$type, null)) {
            a.ad.p(b, null);
            let d = null;
            let e = null;
            if (c.hasValue) {
                e = "pivot";
                d = ST_OnOff.g(c.value);
                a.ad.h(e, d);
            }
            ;
        }
    }
    CT_ConditionalFormatting_SML2009.$t = /*@__PURE__*/ markType(CT_ConditionalFormatting_SML2009, 'CT_ConditionalFormatting_SML2009');
    return CT_ConditionalFormatting_SML2009;
})();
