// igniteui-react-excel/esm2015/lib/CT_Cell3D.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_Bevel } from "./CT_Bevel";
import { CT_LightRig } from "./CT_LightRig";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
import { ST_PresetMaterialType_$type } from "./ST_PresetMaterialType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
export let CT_Cell3D = /*@__PURE__*/ (() => {
    class CT_Cell3D extends Base {
        constructor() {
            super();
            this.b = null;
            this.d = null;
            this.f = null;
            this.i = toNullable(ST_PresetMaterialType_$type, null);
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get h() {
            return this.i;
        }
        set h(a) {
            this.i = a;
        }
        static k(a, b) {
            b = toNullable(ST_PresetMaterialType_$type, null);
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "prstMaterial":
                        b = toNullable(ST_PresetMaterialType_$type, XmlNamespaceDefinitionBase.b(ST_PresetMaterialType_$type, c.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static l(a, b, c = toNullable(ST_PresetMaterialType_$type, null)) {
            a.ad.p(b, "a");
            let d = null;
            let e = null;
            if (c.hasValue) {
                e = "prstMaterial";
                d = XmlNamespaceDefinitionBase.p(ST_PresetMaterialType_$type, c.value);
                a.ad.h(e, d);
            }
            ;
        }
        setProperty(a, b) {
            switch (a) {
                case 108:
                    this.a = typeCast(CT_Bevel.$, b);
                    break;
                case 896:
                    this.c = typeCast(CT_LightRig.$, b);
                    break;
                case 601:
                    this.e = typeCast(CT_OfficeArtExtensionList.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_Cell3D.$t = /*@__PURE__*/ markType(CT_Cell3D, 'CT_Cell3D', Base.$, [ISupportSetProperty_$type]);
    return CT_Cell3D;
})();
