// igniteui-react-excel/esm2015/lib/CT_CustomColor.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { EG_ColorChoice_DML } from "./EG_ColorChoice_DML";
/**
 * @hidden
 */
export let CT_CustomColor = /*@__PURE__*/ (() => {
    class CT_CustomColor extends Base {
        constructor() {
            super();
            this.d = null;
            this.a = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get b() {
            if (this.a == null) {
                this.a = new EG_ColorChoice_DML();
            }
            return this.a;
        }
        static f(a, b) {
            b = null;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "name":
                        b = c.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static g(a, b, c = null) {
            a.ad.p(b, "a");
            let d = null;
            let e = null;
            if (c != null) {
                e = "name";
                d = c;
                a.ad.h(e, d);
            }
            ;
        }
        setProperty(a, b) {
            switch (a) {
                default:
                    if (this.b.g(a)) {
                        this.b.i(a, b);
                        return;
                    }
                    break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_CustomColor.$t = /*@__PURE__*/ markType(CT_CustomColor, 'CT_CustomColor', Base.$, [ISupportSetProperty_$type]);
    return CT_CustomColor;
})();
