// igniteui-react-excel/esm2015/lib/CT_Dxf.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_Font } from "./CT_Font";
import { CT_NumFmt } from "./CT_NumFmt";
import { CT_Fill } from "./CT_Fill";
import { CT_CellAlignment } from "./CT_CellAlignment";
import { CT_Border } from "./CT_Border";
import { CT_CellProtection } from "./CT_CellProtection";
import { CT_ExtensionList } from "./CT_ExtensionList";
/**
 * @hidden
 */
export let CT_Dxf = /*@__PURE__*/ (() => {
    class CT_Dxf extends Base {
        constructor() {
            super();
            this.l = null;
            this.n = null;
            this.j = null;
            this.f = null;
            this.d = null;
            this.h = null;
            this.b = null;
        }
        get k() {
            return this.l;
        }
        set k(a) {
            this.l = a;
        }
        get m() {
            return this.n;
        }
        set m(a) {
            this.n = a;
        }
        get i() {
            return this.j;
        }
        set i(a) {
            this.j = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 659:
                    this.k = typeCast(CT_Font.$, b);
                    break;
                case 1088:
                    this.m = typeCast(CT_NumFmt.$, b);
                    break;
                case 630:
                    this.i = typeCast(CT_Fill.$, b);
                    break;
                case 27:
                    this.e = typeCast(CT_CellAlignment.$, b);
                    break;
                case 149:
                    this.c = typeCast(CT_Border.$, b);
                    break;
                case 1271:
                    this.g = typeCast(CT_CellProtection.$, b);
                    break;
                case 601:
                    this.a = typeCast(CT_ExtensionList.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_Dxf.$t = /*@__PURE__*/ markType(CT_Dxf, 'CT_Dxf', Base.$, [ISupportSetProperty_$type]);
    return CT_Dxf;
})();
