// igniteui-react-excel/esm2015/lib/CT_DataField_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_PivotShowAs_$type } from "./ST_PivotShowAs";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_DataField_SML2009 = /*@__PURE__*/ (() => {
    class CT_DataField_SML2009 extends Base {
        constructor() {
            super(...arguments);
            this.b = toNullable(ST_PivotShowAs_$type, null);
            this.d = toNullable(Number_$type, null);
            this.f = null;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        static g(a, b, c, d) {
            b = toNullable(ST_PivotShowAs_$type, null);
            c = toNullable(Number_$type, null);
            d = null;
            for (let e of fromEnum(a.ab.u)) {
                switch (e.a.e(a.d)) {
                    case "pivotShowAs":
                        b = toNullable(ST_PivotShowAs_$type, XmlNamespaceDefinitionBase.b(ST_PivotShowAs_$type, e.d));
                        break;
                    case "sourceField":
                        c = toNullable(Number_$type, parseUInt32_1(e.d, CultureInfo.invariantCulture));
                        break;
                    case "uniqueName":
                        d = e.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d
            };
        }
        static h(a, b, c = toNullable(ST_PivotShowAs_$type, null), d = toNullable(Number_$type, null), e = null) {
            a.ad.p(b, null);
            let f = null;
            let g = null;
            if (c.hasValue) {
                g = "pivotShowAs";
                f = XmlNamespaceDefinitionBase.p(ST_PivotShowAs_$type, c.value);
                a.ad.h(g, f);
            }
            if (d.hasValue) {
                g = "sourceField";
                f = d.value.toString();
                a.ad.h(g, f);
            }
            if (e != null) {
                g = "uniqueName";
                f = e;
                a.ad.h(g, f);
            }
            ;
        }
    }
    CT_DataField_SML2009.$t = /*@__PURE__*/ markType(CT_DataField_SML2009, 'CT_DataField_SML2009');
    return CT_DataField_SML2009;
})();
