// igniteui-react-excel/esm2015/lib/CT_DataLabelVisibilities.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_DataLabelVisibilities = /*@__PURE__*/ (() => {
    class CT_DataLabelVisibilities extends Base {
        constructor() {
            super(...arguments);
            this.d = toNullable(Boolean_$type, null);
            this.b = toNullable(Boolean_$type, null);
            this.f = toNullable(Boolean_$type, null);
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        static h(a, b, c, d) {
            b = toNullable(Boolean_$type, null);
            c = toNullable(Boolean_$type, null);
            d = toNullable(Boolean_$type, null);
            for (let e of fromEnum(a.ab.u)) {
                switch (e.a.e(a.d)) {
                    case "seriesName":
                        b = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    case "categoryName":
                        c = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    case "value":
                        d = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d
            };
        }
        static i(a, b, c = toNullable(Boolean_$type, null), d = toNullable(Boolean_$type, null), e = toNullable(Boolean_$type, null)) {
            a.ad.p(b, "cx");
            let f = null;
            let g = null;
            if (c.hasValue) {
                g = "categoryName";
                f = ST_OnOff.g(c.value);
                a.ad.h(g, f);
            }
            if (d.hasValue) {
                g = "seriesName";
                f = ST_OnOff.g(d.value);
                a.ad.h(g, f);
            }
            if (e.hasValue) {
                g = "value";
                f = ST_OnOff.g(e.value);
                a.ad.h(g, f);
            }
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_DataLabelVisibilities.$t = /*@__PURE__*/ markType(CT_DataLabelVisibilities, 'CT_DataLabelVisibilities');
    return CT_DataLabelVisibilities;
})();
