// igniteui-react-excel/esm2015/lib/CT_DataLabelFieldTableEntry.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, String_$type, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_StrData } from "./CT_StrData";
/**
 * @hidden
 */
export let CT_DataLabelFieldTableEntry = /*@__PURE__*/ (() => {
    class CT_DataLabelFieldTableEntry extends Base {
        constructor() {
            super();
            this.g = null;
            this.d = null;
            this.b = null;
        }
        get f() {
            return this.g;
        }
        set f(a) {
            this.g = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 1787:
                    this.f = typeCast(String_$type, b);
                    break;
                case 612:
                    this.c = typeCast(String_$type, b);
                    break;
                case 402:
                    this.a = typeCast(CT_StrData.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_DataLabelFieldTableEntry.$t = /*@__PURE__*/ markType(CT_DataLabelFieldTableEntry, 'CT_DataLabelFieldTableEntry', Base.$, [ISupportSetProperty_$type]);
    return CT_DataLabelFieldTableEntry;
})();
