// igniteui-react-excel/esm2015/lib/CT_DataValidationFormula.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, NotSupportedException, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_DataValidationFormula = /*@__PURE__*/ (() => {
    class CT_DataValidationFormula extends Base {
        constructor() {
            super(...arguments);
            this.b = null;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static c(a) {
            throw new NotSupportedException(1, "The corresponding XML element for this class has no attributes, so there is nothing to read.");
        }
        static d(a, b) {
            throw new NotSupportedException(1, "The corresponding XML element for this class has no attributes, so there is nothing to write.");
        }
    }
    CT_DataValidationFormula.$t = /*@__PURE__*/ markType(CT_DataValidationFormula, 'CT_DataValidationFormula');
    return CT_DataValidationFormula;
})();
