// igniteui-react-excel/esm2015/lib/CT_DdeValue.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, fromEnum, typeCast, String_$type, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { ST_DdeValueType_$type } from "./ST_DdeValueType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
export let CT_DdeValue = /*@__PURE__*/ (() => {
    class CT_DdeValue extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = toNullable(ST_DdeValueType_$type, null);
        }
        static e(a, b) {
            b = toNullable(ST_DdeValueType_$type, null);
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "t":
                        b = toNullable(ST_DdeValueType_$type, XmlNamespaceDefinitionBase.b(ST_DdeValueType_$type, c.d));
                        break;
                    default: break;
                }
            }
            return {
                p1: b
            };
        }
        static f(a, b, c = toNullable(ST_DdeValueType_$type, null)) {
            a.ad.p(b, null);
            let d = null;
            let e = null;
            if (c.hasValue) {
                e = "t";
                d = XmlNamespaceDefinitionBase.p(ST_DdeValueType_$type, c.value);
                a.ad.h(e, d);
            }
        }
        setProperty(a, b) {
            switch (a) {
                case 1821:
                    this.d = typeCast(String_$type, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_DdeValue.$t = /*@__PURE__*/ markType(CT_DdeValue, 'CT_DdeValue', Base.$, [ISupportSetProperty_$type]);
    return CT_DdeValue;
})();
