// igniteui-react-excel/esm2015/lib/CT_FillStyleList.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { List$1 } from "igniteui-react-core";
import { ISaveToXLSX_$type } from "./ISaveToXLSX";
/**
 * @hidden
 */
export let CT_FillStyleList = /*@__PURE__*/ (() => {
    class CT_FillStyleList extends Base {
        constructor() {
            super();
            this.a = null;
        }
        get b() {
            if (this.a == null) {
                this.a = new List$1(ISaveToXLSX_$type, 0);
            }
            return this.a;
        }
        static c(a, b) {
            b = a > 1000;
            let c = b ? (a - 1001) : (a - 1);
            return {
                ret: c,
                p1: b
            };
        }
        setProperty(a, b) {
            let c = typeCast(ISaveToXLSX_$type, b);
            if (c == null) {
            }
            else {
                this.b.add(c);
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_FillStyleList.$t = /*@__PURE__*/ markType(CT_FillStyleList, 'CT_FillStyleList', Base.$, [ISupportSetProperty_$type]);
    return CT_FillStyleList;
})();
