// igniteui-react-excel/esm2015/lib/CT_FontCollection.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_TextFont } from "./CT_TextFont";
import { CT_SupplementalFont } from "./CT_SupplementalFont";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
export let CT_FontCollection = /*@__PURE__*/ (() => {
    class CT_FontCollection extends Base {
        constructor() {
            super();
            this.h = null;
            this.f = null;
            this.d = null;
            this.j = null;
            this.b = null;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get i() {
            return this.j;
        }
        set i(a) {
            this.j = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 875:
                    this.g = typeCast(CT_TextFont.$, b);
                    break;
                case 551:
                    this.e = typeCast(CT_TextFont.$, b);
                    break;
                case 363:
                    this.c = typeCast(CT_TextFont.$, b);
                    break;
                case 659:
                    this.i.add(typeCast(CT_SupplementalFont.$, b));
                    break;
                case 601:
                    this.a = typeCast(CT_OfficeArtExtensionList.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_FontCollection.$t = /*@__PURE__*/ markType(CT_FontCollection, 'CT_FontCollection', Base.$, [ISupportSetProperty_$type]);
    return CT_FontCollection;
})();
