// igniteui-react-excel/esm2015/lib/CT_FilteredBubbleSer.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_BubbleSer } from "./CT_BubbleSer";
/**
 * @hidden
 */
export let CT_FilteredBubbleSer = /*@__PURE__*/ (() => {
    class CT_FilteredBubbleSer extends Base {
        constructor() {
            super();
            this.c = null;
            this.a = null;
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 1446:
                    this.b = typeCast(CT_BubbleSer.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_FilteredBubbleSer.$t = /*@__PURE__*/ markType(CT_FilteredBubbleSer, 'CT_FilteredBubbleSer', Base.$, [ISupportSetProperty_$type]);
    return CT_FilteredBubbleSer;
})();
