// igniteui-react-excel/esm2015/lib/CT_GvmlTextShape.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { NotImplementedException } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_GvmlTextShape = /*@__PURE__*/ (() => {
    class CT_GvmlTextShape extends Base {
        constructor() {
            super();
            this.d = null;
            this.f = null;
            this.b = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get item() {
            return this.f;
        }
        set item(a) {
            this.f = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
        setProperty(a, b) {
            throw new NotImplementedException(0);
        }
    }
    CT_GvmlTextShape.$t = /*@__PURE__*/ markType(CT_GvmlTextShape, 'CT_GvmlTextShape', Base.$, [ISupportSetProperty_$type]);
    return CT_GvmlTextShape;
})();
