// igniteui-react-excel/esm2015/lib/CT_GvmlGraphicalObjectFrame.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_GvmlGraphicFrameNonVisual } from "./CT_GvmlGraphicFrameNonVisual";
import { CT_GraphicalObject } from "./CT_GraphicalObject";
import { CT_Transform2D } from "./CT_Transform2D";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
export let CT_GvmlGraphicalObjectFrame = /*@__PURE__*/ (() => {
    class CT_GvmlGraphicalObjectFrame extends Base {
        constructor() {
            super();
            this.b = null;
            this.f = null;
            this.h = null;
            this.d = null;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 1103:
                    this.a = typeCast(CT_GvmlGraphicFrameNonVisual.$, b);
                    break;
                case 727:
                    this.e = typeCast(CT_GraphicalObject.$, b);
                    break;
                case 1892:
                    this.g = typeCast(CT_Transform2D.$, b);
                    break;
                case 601:
                    this.c = typeCast(CT_OfficeArtExtensionList.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_GvmlGraphicalObjectFrame.$t = /*@__PURE__*/ markType(CT_GvmlGraphicalObjectFrame, 'CT_GvmlGraphicalObjectFrame', Base.$, [ISupportSetProperty_$type]);
    return CT_GvmlGraphicalObjectFrame;
})();
