// igniteui-react-excel/esm2015/lib/CT_GvmlConnectorNonVisual.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_NonVisualDrawingProps } from "./CT_NonVisualDrawingProps";
import { CT_NonVisualConnectorProperties } from "./CT_NonVisualConnectorProperties";
/**
 * @hidden
 */
export let CT_GvmlConnectorNonVisual = /*@__PURE__*/ (() => {
    class CT_GvmlConnectorNonVisual extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 202:
                    this.c = typeCast(CT_NonVisualDrawingProps.$, b);
                    break;
                case 198:
                    this.a = typeCast(CT_NonVisualConnectorProperties.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_GvmlConnectorNonVisual.$t = /*@__PURE__*/ markType(CT_GvmlConnectorNonVisual, 'CT_GvmlConnectorNonVisual', Base.$, [ISupportSetProperty_$type]);
    return CT_GvmlConnectorNonVisual;
})();
