// igniteui-react-excel/esm2015/lib/CT_GeoChildEntitiesQuery.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, EnumUtil, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { List$1 } from "igniteui-react-core";
import { ST_EntityType_$type } from "./ST_EntityType";
/**
 * @hidden
 */
export let CT_GeoChildEntitiesQuery = /*@__PURE__*/ (() => {
    class CT_GeoChildEntitiesQuery extends Base {
        constructor() {
            super();
            this.c = null;
            this.a = null;
            this.a = new List$1(ST_EntityType_$type, 0);
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        static e(a, b) {
            b = null;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "entityId":
                        b = c.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static f(a, b, c) {
            a.ad.p(b, "cx");
            let d = null;
            let e = null;
            if (c != null) {
                e = "entityId";
                d = c;
                a.ad.h(e, d);
            }
            ;
        }
        setProperty(a, b) {
            switch (a) {
                case 705:
                    this.a.add(EnumUtil.getEnumValue(ST_EntityType_$type, b));
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_GeoChildEntitiesQuery.$t = /*@__PURE__*/ markType(CT_GeoChildEntitiesQuery, 'CT_GeoChildEntitiesQuery', Base.$, [ISupportSetProperty_$type]);
    return CT_GeoChildEntitiesQuery;
})();
