// igniteui-react-excel/esm2015/lib/CT_GeoDataEntityQueryResult.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_GeoDataEntityQuery } from "./CT_GeoDataEntityQuery";
import { CT_GeoData } from "./CT_GeoData";
/**
 * @hidden
 */
export let CT_GeoDataEntityQueryResult = /*@__PURE__*/ (() => {
    class CT_GeoDataEntityQueryResult extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 707:
                    this.c = typeCast(CT_GeoDataEntityQuery.$, b);
                    break;
                case 706:
                    this.a = typeCast(CT_GeoData.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_GeoDataEntityQueryResult.$t = /*@__PURE__*/ markType(CT_GeoDataEntityQueryResult, 'CT_GeoDataEntityQueryResult', Base.$, [ISupportSetProperty_$type]);
    return CT_GeoDataEntityQueryResult;
})();
