// igniteui-react-excel/esm2015/lib/CT_GeomGuide.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_GeomGuide = /*@__PURE__*/ (() => {
    class CT_GeomGuide extends Base {
        constructor() {
            super(...arguments);
            this.a = null;
            this.b = null;
        }
        static c(a, b, c) {
            b = null;
            c = null;
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "name":
                        b = d.d;
                        break;
                    case "fmla":
                        c = d.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
    }
    CT_GeomGuide.$t = /*@__PURE__*/ markType(CT_GeomGuide, 'CT_GeomGuide');
    return CT_GeomGuide;
})();
