// igniteui-react-excel/esm2015/lib/CT_GeoHierarchyEntity.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_EntityType_$type } from "./ST_EntityType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
export let CT_GeoHierarchyEntity = /*@__PURE__*/ (() => {
    class CT_GeoHierarchyEntity extends Base {
        constructor() {
            super(...arguments);
            this.g = null;
            this.e = null;
            this.c = 0;
        }
        get f() {
            return this.g;
        }
        set f(a) {
            this.g = a;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        static a(a) {
            let b = null;
            let c = null;
            let d = 11;
            let e = CT_GeoHierarchyEntity.i(a, b, c, d);
            b = e.p1;
            c = e.p2;
            d = e.p3;
            return ((() => {
                let $ret = new CT_GeoHierarchyEntity();
                $ret.d = c;
                $ret.b = d;
                $ret.f = b;
                return $ret;
            })());
        }
        static i(a, b, c, d) {
            b = null;
            c = null;
            d = 0;
            for (let e of fromEnum(a.ab.u)) {
                switch (e.a.e(a.d)) {
                    case "entityName":
                        b = e.d;
                        break;
                    case "entityId":
                        c = e.d;
                        break;
                    case "entityType":
                        d = XmlNamespaceDefinitionBase.b(ST_EntityType_$type, e.d);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d
            };
        }
        static j(a, b, c, d, e) {
            a.ad.p(b, "cx");
            let f = null;
            let g = null;
            if (c != null) {
                g = "entityId";
                f = c;
                a.ad.h(g, f);
            }
            if (d != null) {
                g = "entityName";
                f = d;
                a.ad.h(g, f);
            }
            g = "entityType";
            f = XmlNamespaceDefinitionBase.p(ST_EntityType_$type, e);
            a.ad.h(g, f);
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_GeoHierarchyEntity.$t = /*@__PURE__*/ markType(CT_GeoHierarchyEntity, 'CT_GeoHierarchyEntity');
    return CT_GeoHierarchyEntity;
})();
