// igniteui-react-excel/esm2015/lib/CT_GvmlGroupShapeNonVisual.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_NonVisualDrawingProps } from "./CT_NonVisualDrawingProps";
import { CT_NonVisualGroupDrawingShapeProps } from "./CT_NonVisualGroupDrawingShapeProps";
/**
 * @hidden
 */
export let CT_GvmlGroupShapeNonVisual = /*@__PURE__*/ (() => {
    class CT_GvmlGroupShapeNonVisual extends Base {
        constructor() {
            super();
            this.b = null;
            this.d = null;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 202:
                    this.a = typeCast(CT_NonVisualDrawingProps.$, b);
                    break;
                case 200:
                    this.c = typeCast(CT_NonVisualGroupDrawingShapeProps.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_GvmlGroupShapeNonVisual.$t = /*@__PURE__*/ markType(CT_GvmlGroupShapeNonVisual, 'CT_GvmlGroupShapeNonVisual', Base.$, [ISupportSetProperty_$type]);
    return CT_GvmlGroupShapeNonVisual;
})();
