// igniteui-react-excel/esm2015/lib/CT_GeoChildEntitiesQueryResult.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_GeoChildEntitiesQuery } from "./CT_GeoChildEntitiesQuery";
import { List$1 } from "igniteui-react-core";
import { CT_GeoHierarchyEntity } from "./CT_GeoHierarchyEntity";
/**
 * @hidden
 */
export let CT_GeoChildEntitiesQueryResult = /*@__PURE__*/ (() => {
    class CT_GeoChildEntitiesQueryResult extends Base {
        constructor() {
            super();
            this.b = null;
            this.c = null;
            this.c = new List$1(CT_GeoHierarchyEntity.$, 0);
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 702:
                    this.a = typeCast(CT_GeoChildEntitiesQuery.$, b);
                    break;
                case 701:
                    this.c.add(typeCast(CT_GeoHierarchyEntity.$, b));
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_GeoChildEntitiesQueryResult.$t = /*@__PURE__*/ markType(CT_GeoChildEntitiesQueryResult, 'CT_GeoChildEntitiesQueryResult', Base.$, [ISupportSetProperty_$type]);
    return CT_GeoChildEntitiesQueryResult;
})();
