// igniteui-react-excel/esm2015/lib/CT_GvmlPictureNonVisual.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_NonVisualDrawingProps } from "./CT_NonVisualDrawingProps";
import { CT_NonVisualPictureProperties } from "./CT_NonVisualPictureProperties";
/**
 * @hidden
 */
export let CT_GvmlPictureNonVisual = /*@__PURE__*/ (() => {
    class CT_GvmlPictureNonVisual extends Base {
        constructor() {
            super();
            this.b = null;
            this.d = null;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 202:
                    this.a = typeCast(CT_NonVisualDrawingProps.$, b);
                    break;
                case 201:
                    this.c = typeCast(CT_NonVisualPictureProperties.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_GvmlPictureNonVisual.$t = /*@__PURE__*/ markType(CT_GvmlPictureNonVisual, 'CT_GvmlPictureNonVisual', Base.$, [ISupportSetProperty_$type]);
    return CT_GvmlPictureNonVisual;
})();
