// igniteui-react-excel/esm2015/lib/CT_GeoCache.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, typeCast, String_$type, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_Clear } from "./CT_Clear";
/**
 * @hidden
 */
export let CT_GeoCache = /*@__PURE__*/ (() => {
    class CT_GeoCache extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
            this.a = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        static f(a, b) {
            b = null;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "provider":
                        b = c.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static g(a, b, c) {
            a.ad.p(b, "cx");
            let d = null;
            let e = null;
            if (c != null) {
                e = "provider";
                d = c;
                a.ad.h(e, d);
            }
            ;
        }
        setProperty(a, b) {
            switch (a) {
                case 1917:
                    this.b = typeCast(String_$type, b);
                    break;
                case 1918:
                    this.a = typeCast(CT_Clear.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_GeoCache.$t = /*@__PURE__*/ markType(CT_GeoCache, 'CT_GeoCache', Base.$, [ISupportSetProperty_$type]);
    return CT_GeoCache;
})();
