// igniteui-react-excel/esm2015/lib/CT_GeoDataPointToEntityQuery.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_EntityType_$type } from "./ST_EntityType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
export let CT_GeoDataPointToEntityQuery = /*@__PURE__*/ (() => {
    class CT_GeoDataPointToEntityQuery extends Base {
        constructor() {
            super(...arguments);
            this.c = 0;
            this.e = null;
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        static a(a) {
            let b = 0;
            let c = null;
            let d = CT_GeoDataPointToEntityQuery.g(a, b, c);
            b = d.p1;
            c = d.p2;
            return ((() => {
                let $ret = new CT_GeoDataPointToEntityQuery();
                $ret.d = c;
                $ret.b = b;
                return $ret;
            })());
        }
        static g(a, b, c) {
            b = 0;
            c = null;
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "entityType":
                        b = XmlNamespaceDefinitionBase.b(ST_EntityType_$type, d.d);
                        break;
                    case "entityId":
                        c = d.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static h(a, b, c, d) {
            a.ad.p(b, "cx");
            let e = null;
            let f = null;
            if (c != null) {
                f = "entityId";
                e = c;
                a.ad.h(f, e);
            }
            f = "entityType";
            e = XmlNamespaceDefinitionBase.p(ST_EntityType_$type, d);
            a.ad.h(f, e);
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_GeoDataPointToEntityQuery.$t = /*@__PURE__*/ markType(CT_GeoDataPointToEntityQuery, 'CT_GeoDataPointToEntityQuery');
    return CT_GeoDataPointToEntityQuery;
})();
