// igniteui-react-excel/esm2015/lib/CT_HeaderFooter.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, typeCast, String_$type, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_HeaderFooter = /*@__PURE__*/ (() => {
    class CT_HeaderFooter extends Base {
        constructor() {
            super();
            this.u = null;
            this.s = null;
            this.m = null;
            this.k = null;
            this.q = null;
            this.o = null;
            this.e = toNullable(Boolean_$type, null);
            this.i = toNullable(Boolean_$type, null);
            this.g = toNullable(Boolean_$type, null);
        }
        get t() {
            return this.u;
        }
        set t(a) {
            this.u = a;
        }
        get r() {
            return this.s;
        }
        set r(a) {
            this.s = a;
        }
        get l() {
            return this.m;
        }
        set l(a) {
            this.m = a;
        }
        get j() {
            return this.k;
        }
        set j(a) {
            this.k = a;
        }
        get p() {
            return this.q;
        }
        set p(a) {
            this.q = a;
        }
        get n() {
            return this.o;
        }
        set n(a) {
            this.o = a;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        get h() {
            return this.i;
        }
        set h(a) {
            this.i = a;
        }
        get f() {
            return this.g;
        }
        set f(a) {
            this.g = a;
        }
        static w(a, b, c, d) {
            b = toNullable(Boolean_$type, null);
            c = toNullable(Boolean_$type, null);
            d = toNullable(Boolean_$type, null);
            for (let e of fromEnum(a.ab.u)) {
                switch (e.a.e(a.d)) {
                    case "alignWithMargins":
                        b = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    case "differentOddEven":
                        c = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    case "differentFirst":
                        d = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d
            };
        }
        static x(a, b, c = toNullable(Boolean_$type, null), d = toNullable(Boolean_$type, null), e = toNullable(Boolean_$type, null)) {
            a.ad.p(b, "cx");
            let f = null;
            let g = null;
            if (c.hasValue) {
                g = "alignWithMargins";
                f = ST_OnOff.g(c.value);
                a.ad.h(g, f);
            }
            if (d.hasValue) {
                g = "differentFirst";
                f = ST_OnOff.g(d.value);
                a.ad.h(g, f);
            }
            if (e.hasValue) {
                g = "differentOddEven";
                f = ST_OnOff.g(e.value);
                a.ad.h(g, f);
            }
            ;
        }
        setProperty(a, b) {
            switch (a) {
                case 1118:
                    this.t = typeCast(String_$type, b);
                    break;
                case 1117:
                    this.r = typeCast(String_$type, b);
                    break;
                case 597:
                    this.l = typeCast(String_$type, b);
                    break;
                case 596:
                    this.j = typeCast(String_$type, b);
                    break;
                case 645:
                    this.p = typeCast(String_$type, b);
                    break;
                case 644:
                    this.n = typeCast(String_$type, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_HeaderFooter.$t = /*@__PURE__*/ markType(CT_HeaderFooter, 'CT_HeaderFooter', Base.$, [ISupportSetProperty_$type]);
    return CT_HeaderFooter;
})();
