// igniteui-react-excel/esm2015/lib/CT_Hyperlink.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, typeCast, markType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_EmbeddedWAVAudioFile } from "./CT_EmbeddedWAVAudioFile";
/**
 * @hidden
 */
export let CT_Hyperlink = /*@__PURE__*/ (() => {
    class CT_Hyperlink extends Base {
        constructor() {
            super();
            this.f = null;
            this.b = toNullable(Boolean_$type, null);
            this.c = toNullable(Boolean_$type, null);
            this.d = toNullable(Boolean_$type, null);
            this.g = null;
            this.h = null;
            this.a = null;
            this.i = null;
            this.j = null;
            this.e = null;
            this.h = "";
            this.f = "";
            this.i = "";
            this.j = "";
            this.d = toNullable(Boolean_$type, true);
        }
        setProperty(a, b) {
            switch (a) {
                case 601:
                    this.e = b;
                    break;
                case 1549:
                    this.a = typeCast(CT_EmbeddedWAVAudioFile.$, b);
                    break;
            }
        }
    }
    CT_Hyperlink.$t = /*@__PURE__*/ markType(CT_Hyperlink, 'CT_Hyperlink', Base.$, [ISupportSetProperty_$type]);
    return CT_Hyperlink;
})();
