// igniteui-react-excel/esm2015/lib/CT_IgnoredError.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_Sqref } from "./ST_Sqref";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_IgnoredError = /*@__PURE__*/ (() => {
    class CT_IgnoredError extends Base {
        constructor() {
            super();
            this.b = null;
            this.q = toNullable(Boolean_$type, null);
            this.aa = toNullable(Boolean_$type, null);
            this.y = toNullable(Boolean_$type, null);
            this.s = toNullable(Boolean_$type, null);
            this.u = toNullable(Boolean_$type, null);
            this.ac = toNullable(Boolean_$type, null);
            this.o = toNullable(Boolean_$type, null);
            this.w = toNullable(Boolean_$type, null);
            this.m = toNullable(Boolean_$type, null);
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get p() {
            return this.q;
        }
        set p(a) {
            this.q = a;
        }
        get z() {
            return this.aa;
        }
        set z(a) {
            this.aa = a;
        }
        get x() {
            return this.y;
        }
        set x(a) {
            this.y = a;
        }
        get r() {
            return this.s;
        }
        set r(a) {
            this.s = a;
        }
        get t() {
            return this.u;
        }
        set t(a) {
            this.u = a;
        }
        get ab() {
            return this.ac;
        }
        set ab(a) {
            this.ac = a;
        }
        get n() {
            return this.o;
        }
        set n(a) {
            this.o = a;
        }
        get v() {
            return this.w;
        }
        set v(a) {
            this.w = a;
        }
        get l() {
            return this.m;
        }
        set l(a) {
            this.m = a;
        }
        static ae(a, b, c, d, e, f, g, h, i, j, k) {
            b = null;
            c = toNullable(Boolean_$type, null);
            d = toNullable(Boolean_$type, null);
            e = toNullable(Boolean_$type, null);
            f = toNullable(Boolean_$type, null);
            g = toNullable(Boolean_$type, null);
            h = toNullable(Boolean_$type, null);
            i = toNullable(Boolean_$type, null);
            j = toNullable(Boolean_$type, null);
            k = toNullable(Boolean_$type, null);
            for (let l of fromEnum(a.ab.u)) {
                switch (l.a.e(a.d)) {
                    case "sqref":
                        b = ST_Sqref.a(l.d);
                        break;
                    case "evalError":
                        c = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "twoDigitTextYear":
                        d = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "numberStoredAsText":
                        e = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "formula":
                        f = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "formulaRange":
                        g = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "unlockedFormula":
                        h = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "emptyCellReference":
                        i = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "listDataValidation":
                        j = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "calculatedColumn":
                        k = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    default: break;
                }
            }
            return {
                p1: b,
                p2: c,
                p3: d,
                p4: e,
                p5: f,
                p6: g,
                p7: h,
                p8: i,
                p9: j,
                p10: k
            };
        }
        static af(a, b, c, d, e, f, g, h, i, j, k, l) {
            a.ad.p(b, null);
            let m = null;
            let n = null;
            if (c.hasValue) {
                n = "calculatedColumn";
                m = ST_OnOff.g(c.value);
                a.ad.h(n, m);
            }
            if (d.hasValue) {
                n = "emptyCellReference";
                m = ST_OnOff.g(d.value);
                a.ad.h(n, m);
            }
            if (e.hasValue) {
                n = "evalError";
                m = ST_OnOff.g(e.value);
                a.ad.h(n, m);
            }
            if (f.hasValue) {
                n = "formula";
                m = ST_OnOff.g(f.value);
                a.ad.h(n, m);
            }
            if (g.hasValue) {
                n = "formulaRange";
                m = ST_OnOff.g(g.value);
                a.ad.h(n, m);
            }
            if (h.hasValue) {
                n = "listDataValidation";
                m = ST_OnOff.g(h.value);
                a.ad.h(n, m);
            }
            if (i.hasValue) {
                n = "numberStoredAsText";
                m = ST_OnOff.g(i.value);
                a.ad.h(n, m);
            }
            if (j.hasValue) {
                n = "twoDigitTextYear";
                m = ST_OnOff.g(j.value);
                a.ad.h(n, m);
            }
            if (k.hasValue) {
                n = "unlockedFormula";
                m = ST_OnOff.g(k.value);
                a.ad.h(n, m);
            }
            if (l != null) {
                n = "sqref";
                m = l.e();
                a.ad.h(n, m);
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_IgnoredError.$t = /*@__PURE__*/ markType(CT_IgnoredError, 'CT_IgnoredError');
    return CT_IgnoredError;
})();
