// igniteui-react-excel/esm2015/lib/CT_LinePropertiesBase.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, markType } from "igniteui-react-core";
import { ST_PenAlignment_$type } from "./ST_PenAlignment";
import { ST_LineCap_$type } from "./ST_LineCap";
import { ST_CompoundLine_$type } from "./ST_CompoundLine";
import { ST_LineWidth } from "./ST_LineWidth";
/**
 * @hidden
 */
export let CT_LinePropertiesBase = /*@__PURE__*/ (() => {
    class CT_LinePropertiesBase extends Base {
        constructor() {
            super(...arguments);
            this.e = toNullable(ST_PenAlignment_$type, null);
            this.f = toNullable(ST_LineCap_$type, null);
            this.g = toNullable(ST_CompoundLine_$type, null);
            this.a = null;
            this.c = null;
            this.d = null;
            this.b = null;
            this.h = toNullable(ST_LineWidth.$, null);
        }
    }
    CT_LinePropertiesBase.$t = /*@__PURE__*/ markType(CT_LinePropertiesBase, 'CT_LinePropertiesBase');
    return CT_LinePropertiesBase;
})();
