// igniteui-react-excel/esm2015/lib/CT_LuminanceEffect.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_LuminanceEffect = /*@__PURE__*/ (() => {
    class CT_LuminanceEffect extends Base {
        constructor() {
            super();
            this.d = toNullable(Number_$type, null);
            this.f = toNullable(Number_$type, null);
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        static h(a, b, c) {
            b = toNullable(Number_$type, null);
            c = toNullable(Number_$type, null);
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "bright":
                        b = toNullable(Number_$type, parseInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    case "contrast":
                        c = toNullable(Number_$type, parseInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static i(a, b, c = toNullable(Number_$type, null), d = toNullable(Number_$type, null)) {
            a.ad.p(b, "a");
            let e = null;
            let f = null;
            if (c.hasValue) {
                f = "bright";
                e = c.value.toString();
                a.ad.h(f, e);
            }
            if (d.hasValue) {
                f = "contrast";
                e = d.value.toString();
                a.ad.h(f, e);
            }
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_LuminanceEffect.$t = /*@__PURE__*/ markType(CT_LuminanceEffect, 'CT_LuminanceEffect');
    return CT_LuminanceEffect;
})();
