// igniteui-react-excel/esm2015/lib/CT_ListItems.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, NotSupportedException, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_ListItems = /*@__PURE__*/ (() => {
    class CT_ListItems extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
        }
        get item() {
            return this.d;
        }
        set item(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static e(a) {
            throw new NotSupportedException(1, "The corresponding XML element for this class has no attributes, so there is nothing to read.");
        }
        static f(a, b) {
            throw new NotSupportedException(1, "The corresponding XML element for this class has no attributes, so there is nothing to write.");
        }
    }
    CT_ListItems.$t = /*@__PURE__*/ markType(CT_ListItems, 'CT_ListItems');
    return CT_ListItems;
})();
