// igniteui-react-excel/esm2015/lib/CT_NumberColorPosition.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseNumber } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_NumberColorPosition = /*@__PURE__*/ (() => {
    class CT_NumberColorPosition extends Base {
        constructor() {
            super(...arguments);
            this.b = 0;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static d(a, b) {
            b = 0;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "val":
                        b = parseNumber(c.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static e(a, b, c) {
            a.ad.p(b, "cx");
            let d = null;
            let e = null;
            e = "val";
            d = c.toString();
            a.ad.h(e, d);
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_NumberColorPosition.$t = /*@__PURE__*/ markType(CT_NumberColorPosition, 'CT_NumberColorPosition');
    return CT_NumberColorPosition;
})();
