// igniteui-react-excel/esm2015/lib/CT_NumVal.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, typeCast, String_$type, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_NumVal = /*@__PURE__*/ (() => {
    class CT_NumVal extends Base {
        constructor() {
            super(...arguments);
            this.e = null;
            this.g = 0;
            this.b = null;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        get f() {
            return this.g;
        }
        set f(a) {
            this.g = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static h(a, b, c) {
            b = 0;
            c = null;
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "idx":
                        b = parseUInt32_1(d.d, CultureInfo.invariantCulture);
                        break;
                    case "formatCode":
                        c = d.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static i(a, b, c, d) {
            a.ad.p(b, "c");
            let e = null;
            let f = null;
            if (c != null) {
                f = "formatCode";
                e = c;
                a.ad.h(f, e);
            }
            f = "idx";
            e = d.toString();
            a.ad.h(f, e);
            ;
        }
        setProperty(a, b) {
            switch (a) {
                case 1818:
                    this.d = typeCast(String_$type, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_NumVal.$t = /*@__PURE__*/ markType(CT_NumVal, 'CT_NumVal', Base.$, [ISupportSetProperty_$type]);
    return CT_NumVal;
})();
