// igniteui-react-excel/esm2015/lib/CT_NumberFormat.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_NumberFormat = /*@__PURE__*/ (() => {
    class CT_NumberFormat extends Base {
        constructor() {
            super(...arguments);
            this.d = null;
            this.b = toNullable(Boolean_$type, null);
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static f(a, b, c) {
            b = null;
            c = toNullable(Boolean_$type, null);
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "formatCode":
                        b = d.d;
                        break;
                    case "sourceLinked":
                        c = toNullable(Boolean_$type, ST_OnOff.e(d.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static g(a, b, c, d) {
            a.ad.p(b, "cx");
            let e = null;
            let f = null;
            if (c.hasValue) {
                f = "sourceLinked";
                e = ST_OnOff.g(c.value);
                a.ad.h(f, e);
            }
            if (d != null) {
                f = "formatCode";
                e = d;
                a.ad.h(f, e);
            }
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_NumberFormat.$t = /*@__PURE__*/ markType(CT_NumberFormat, 'CT_NumberFormat');
    return CT_NumberFormat;
})();
