// igniteui-react-excel/esm2015/lib/CT_NonVisualDrawingProps.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { ST_DrawingElementId } from "./ST_DrawingElementId";
/**
 * @hidden
 */
export let CT_NonVisualDrawingProps = /*@__PURE__*/ (() => {
    class CT_NonVisualDrawingProps extends Base {
        constructor() {
            super(...arguments);
            this.e = null;
            this.d = false;
            this.a = null;
            this.b = null;
            this.c = new ST_DrawingElementId();
            this.f = null;
            this.g = null;
        }
    }
    CT_NonVisualDrawingProps.$t = /*@__PURE__*/ markType(CT_NonVisualDrawingProps, 'CT_NonVisualDrawingProps');
    return CT_NonVisualDrawingProps;
})();
