// igniteui-react-excel/esm2015/lib/CT_NumFmt.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_NumFmt = /*@__PURE__*/ (() => {
    class CT_NumFmt extends Base {
        constructor() {
            super(...arguments);
            this.c = 0;
            this.a = null;
        }
        static d(a, b, c) {
            b = 0;
            c = null;
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "numFmtId":
                        b = parseUInt32_1(d.d, CultureInfo.invariantCulture);
                        break;
                    case "formatCode":
                        c = d.d;
                        break;
                    default: break;
                }
            }
            return {
                p1: b,
                p2: c
            };
        }
        static e(a, b, c, d) {
            a.ad.p(b, null);
            let e = null;
            let f = null;
            if (c != null) {
                f = "formatCode";
                e = c;
                a.ad.h(f, e);
            }
            f = "numFmtId";
            e = d.toString();
            a.ad.h(f, e);
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_NumFmt.$t = /*@__PURE__*/ markType(CT_NumFmt, 'CT_NumFmt');
    return CT_NumFmt;
})();
