// igniteui-react-excel/esm2015/lib/CT_OleItem.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_OleItem = /*@__PURE__*/ (() => {
    class CT_OleItem extends Base {
        constructor() {
            super();
            this.g = null;
            this.e = toNullable(Boolean_$type, null);
            this.d = toNullable(Boolean_$type, null);
            this.f = toNullable(Boolean_$type, null);
        }
        static i(a, b, c, d, e) {
            b = null;
            c = toNullable(Boolean_$type, null);
            d = toNullable(Boolean_$type, null);
            e = toNullable(Boolean_$type, null);
            for (let f of fromEnum(a.ab.u)) {
                switch (f.a.e(a.d)) {
                    case "name":
                        b = f.d;
                        break;
                    case "icon":
                        c = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "advise":
                        d = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "preferPic":
                        e = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    default: break;
                }
            }
            return {
                p1: b,
                p2: c,
                p3: d,
                p4: e
            };
        }
        static j(a, b, c, d, e, f) {
            a.ad.p(b, null);
            let g = null;
            let h = null;
            if (c.hasValue) {
                h = "advise";
                g = ST_OnOff.g(c.value);
                a.ad.h(h, g);
            }
            if (d.hasValue) {
                h = "icon";
                g = ST_OnOff.g(d.value);
                a.ad.h(h, g);
            }
            if (e.hasValue) {
                h = "preferPic";
                g = ST_OnOff.g(e.value);
                a.ad.h(h, g);
            }
            if (f != null) {
                h = "name";
                g = f;
                a.ad.h(h, g);
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_OleItem.$t = /*@__PURE__*/ markType(CT_OleItem, 'CT_OleItem');
    return CT_OleItem;
})();
