// igniteui-react-excel/esm2015/lib/CT_OlapSlicerCacheItem.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_OlapSlicerCacheItem = /*@__PURE__*/ (() => {
    class CT_OlapSlicerCacheItem extends Base {
        constructor() {
            super();
            this.c = null;
            this.i = null;
            this.g = null;
            this.e = toNullable(Boolean_$type, null);
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        get h() {
            return this.i;
        }
        set h(a) {
            this.i = a;
        }
        get f() {
            return this.g;
        }
        set f(a) {
            this.g = a;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        static j(a, b, c, d) {
            b = null;
            c = null;
            d = toNullable(Boolean_$type, null);
            for (let e of fromEnum(a.ab.u)) {
                switch (e.a.e(a.d)) {
                    case "n":
                        b = e.d;
                        break;
                    case "c":
                        c = e.d;
                        break;
                    case "nd":
                        d = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d
            };
        }
        static k(a, b, c, d, e) {
            a.ad.p(b, null);
            let f = null;
            let g = null;
            if (c != null) {
                g = "c";
                f = c;
                a.ad.h(g, f);
            }
            if (d.hasValue) {
                g = "nd";
                f = ST_OnOff.g(d.value);
                a.ad.h(g, f);
            }
            if (e != null) {
                g = "n";
                f = e;
                a.ad.h(g, f);
            }
            ;
        }
    }
    CT_OlapSlicerCacheItem.$t = /*@__PURE__*/ markType(CT_OlapSlicerCacheItem, 'CT_OlapSlicerCacheItem');
    return CT_OlapSlicerCacheItem;
})();
