// igniteui-react-excel/esm2015/lib/CT_OlapSlicerCacheLevelsData.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, fromEnum, markType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_OlapSlicerCacheLevelsData = /*@__PURE__*/ (() => {
    class CT_OlapSlicerCacheLevelsData extends Base {
        constructor() {
            super();
            this.b = null;
            this.d = toNullable(Number_$type, null);
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        static e(a, b) {
            b = toNullable(Number_$type, null);
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "count":
                        b = toNullable(Number_$type, parseUInt32_1(c.d, CultureInfo.invariantCulture));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static f(a, b, c = toNullable(Number_$type, null)) {
            a.ad.p(b, null);
            let d = null;
            let e = null;
            if (c.hasValue) {
                e = "count";
                d = c.value.toString();
                a.ad.h(e, d);
            }
            ;
        }
    }
    CT_OlapSlicerCacheLevelsData.$t = /*@__PURE__*/ markType(CT_OlapSlicerCacheLevelsData, 'CT_OlapSlicerCacheLevelsData');
    return CT_OlapSlicerCacheLevelsData;
})();
