// igniteui-react-excel/esm2015/lib/CT_OleItem_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CT_OleItem } from "./CT_OleItem";
import { toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_OleItem_SML2009 = /*@__PURE__*/ (() => {
    class CT_OleItem_SML2009 extends CT_OleItem {
        constructor() {
            super();
            this.l = null;
        }
        get k() {
            return this.l;
        }
        set k(a) {
            this.l = a;
        }
        static p(a, b, c, d, e) {
            b = null;
            c = toNullable(Boolean_$type, null);
            d = toNullable(Boolean_$type, null);
            e = toNullable(Boolean_$type, null);
            for (let f of fromEnum(a.ab.u)) {
                switch (f.a.e(a.d)) {
                    case "name":
                        b = f.d;
                        break;
                    case "icon":
                        c = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "advise":
                        d = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "preferPic":
                        e = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d,
                p4: e
            };
        }
        static q(a, b, c, d, e, f) {
            a.ad.p(b, null);
            let g = null;
            let h = null;
            if (c.hasValue) {
                h = "advise";
                g = ST_OnOff.g(c.value);
                a.ad.h(h, g);
            }
            if (d.hasValue) {
                h = "icon";
                g = ST_OnOff.g(d.value);
                a.ad.h(h, g);
            }
            if (e.hasValue) {
                h = "preferPic";
                g = ST_OnOff.g(e.value);
                a.ad.h(h, g);
            }
            if (f != null) {
                h = "name";
                g = f;
                a.ad.h(h, g);
            }
            ;
        }
    }
    CT_OleItem_SML2009.$t = /*@__PURE__*/ markType(CT_OleItem_SML2009, 'CT_OleItem_SML2009', CT_OleItem.$);
    return CT_OleItem_SML2009;
})();
