// igniteui-react-excel/esm2015/lib/CT_OlapSlicerCacheRange.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_OlapSlicerCacheRange = /*@__PURE__*/ (() => {
    class CT_OlapSlicerCacheRange extends Base {
        constructor() {
            super();
            this.b = null;
            this.d = 0;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        static e(a, b) {
            b = 0;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "startItem":
                        b = parseUInt32_1(c.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static f(a, b, c) {
            a.ad.p(b, null);
            let d = null;
            let e = null;
            e = "startItem";
            d = c.toString();
            a.ad.h(e, d);
            ;
        }
    }
    CT_OlapSlicerCacheRange.$t = /*@__PURE__*/ markType(CT_OlapSlicerCacheRange, 'CT_OlapSlicerCacheRange');
    return CT_OlapSlicerCacheRange;
})();
