// igniteui-react-excel/esm2015/lib/CT_PageMargins_Chart.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseNumber } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_PageMargins_Chart = /*@__PURE__*/ (() => {
    class CT_PageMargins_Chart extends Base {
        constructor() {
            super(...arguments);
            this.i = 0;
            this.k = 0;
            this.m = 0;
            this.c = 0;
            this.g = 0;
            this.e = 0;
        }
        get h() {
            return this.i;
        }
        set h(a) {
            this.i = a;
        }
        get j() {
            return this.k;
        }
        set j(a) {
            this.k = a;
        }
        get l() {
            return this.m;
        }
        set l(a) {
            this.m = a;
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        get f() {
            return this.g;
        }
        set f(a) {
            this.g = a;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        static a(a) {
            let b = 0;
            let c = 0;
            let d = 0;
            let e = 0;
            let f = 0;
            let g = 0;
            let h = CT_PageMargins_Chart.o(a, b, c, d, e, f, g);
            b = h.p1;
            c = h.p2;
            d = h.p3;
            e = h.p4;
            f = h.p5;
            g = h.p6;
            return ((() => {
                let $ret = new CT_PageMargins_Chart();
                $ret.b = e;
                $ret.d = g;
                $ret.f = f;
                $ret.h = b;
                $ret.j = c;
                $ret.l = d;
                return $ret;
            })());
        }
        static o(a, b, c, d, e, f, g) {
            b = 0;
            c = 0;
            d = 0;
            e = 0;
            f = 0;
            g = 0;
            for (let h of fromEnum(a.ab.u)) {
                switch (h.a.e(a.d)) {
                    case "l":
                        b = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "r":
                        c = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "t":
                        d = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "b":
                        e = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "header":
                        f = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "footer":
                        g = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d,
                p4: e,
                p5: f,
                p6: g
            };
        }
        static p(a, b, c, d, e, f, g, h) {
            a.ad.p(b, "c");
            let i = null;
            let j = null;
            j = "b";
            i = c.toString();
            a.ad.h(j, i);
            j = "footer";
            i = d.toString();
            a.ad.h(j, i);
            j = "header";
            i = e.toString();
            a.ad.h(j, i);
            j = "l";
            i = f.toString();
            a.ad.h(j, i);
            j = "r";
            i = g.toString();
            a.ad.h(j, i);
            j = "t";
            i = h.toString();
            a.ad.h(j, i);
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_PageMargins_Chart.$t = /*@__PURE__*/ markType(CT_PageMargins_Chart, 'CT_PageMargins_Chart');
    return CT_PageMargins_Chart;
})();
