// igniteui-react-excel/esm2015/lib/CT_PresetTextShape.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { ST_TextShapeType_$type } from "./ST_TextShapeType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { NotImplementedException } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_PresetTextShape = /*@__PURE__*/ (() => {
    class CT_PresetTextShape extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = 0;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static f(a, b) {
            b = 0;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "prst":
                        b = XmlNamespaceDefinitionBase.b(ST_TextShapeType_$type, c.d);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static g(a, b, c) {
            a.ad.p(b, "a");
            let d = null;
            let e = null;
            e = "prst";
            d = XmlNamespaceDefinitionBase.p(ST_TextShapeType_$type, c);
            a.ad.h(e, d);
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
        setProperty(a, b) {
            throw new NotImplementedException(0);
        }
    }
    CT_PresetTextShape.$t = /*@__PURE__*/ markType(CT_PresetTextShape, 'CT_PresetTextShape', Base.$, [ISupportSetProperty_$type]);
    return CT_PresetTextShape;
})();
