// igniteui-react-excel/esm2015/lib/CT_Path2D.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, markType } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
import { IPath2DCommand_$type } from "./IPath2DCommand";
import { ST_PositiveCoordinate } from "./ST_PositiveCoordinate";
import { ST_PathFillMode_$type } from "./ST_PathFillMode";
/**
 * @hidden
 */
export let CT_Path2D = /*@__PURE__*/ (() => {
    class CT_Path2D extends Base {
        constructor() {
            super(...arguments);
            this.a = new List$1(IPath2DCommand_$type, 0);
            this.g = toNullable(ST_PositiveCoordinate.$, null);
            this.e = toNullable(ST_PositiveCoordinate.$, null);
            this.d = toNullable(ST_PathFillMode_$type, null);
            this.f = toNullable(Boolean_$type, null);
            this.c = toNullable(Boolean_$type, null);
        }
        get b() {
            return this.a;
        }
    }
    CT_Path2D.$t = /*@__PURE__*/ markType(CT_Path2D, 'CT_Path2D');
    return CT_Path2D;
})();
