// igniteui-react-excel/esm2015/lib/CT_PivotChange.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType } from "igniteui-react-core";
import { ST_AllocationMethod_$type } from "./ST_AllocationMethod";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
export let CT_PivotChange = /*@__PURE__*/ (() => {
    class CT_PivotChange extends Base {
        constructor() {
            super();
            this.d = null;
            this.i = null;
            this.b = null;
            this.f = 0;
            this.k = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get h() {
            return this.i;
        }
        set h(a) {
            this.i = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get j() {
            return this.k;
        }
        set j(a) {
            this.k = a;
        }
        static l(a, b, c) {
            b = 0;
            c = null;
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "allocationMethod":
                        b = XmlNamespaceDefinitionBase.b(ST_AllocationMethod_$type, d.d);
                        break;
                    case "weightExpression":
                        c = d.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static m(a, b, c, d) {
            a.ad.p(b, null);
            let e = null;
            let f = null;
            if (c != null) {
                f = "weightExpression";
                e = c;
                a.ad.h(f, e);
            }
            f = "allocationMethod";
            e = XmlNamespaceDefinitionBase.p(ST_AllocationMethod_$type, d);
            a.ad.h(f, e);
            ;
        }
    }
    CT_PivotChange.$t = /*@__PURE__*/ markType(CT_PivotChange, 'CT_PivotChange');
    return CT_PivotChange;
})();
