// igniteui-react-excel/esm2015/lib/CT_PrintSettings.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_HeaderFooter_Chart } from "./CT_HeaderFooter_Chart";
import { CT_PageMargins_Chart } from "./CT_PageMargins_Chart";
import { CT_PageSetup_Chart } from "./CT_PageSetup_Chart";
import { CT_RelId } from "./CT_RelId";
/**
 * @hidden
 */
export let CT_PrintSettings = /*@__PURE__*/ (() => {
    class CT_PrintSettings extends Base {
        constructor() {
            super();
            this.b = null;
            this.d = null;
            this.f = null;
            this.h = null;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 774:
                    this.a = typeCast(CT_HeaderFooter_Chart.$, b);
                    break;
                case 1162:
                    this.c = typeCast(CT_PageMargins_Chart.$, b);
                    break;
                case 1164:
                    this.e = typeCast(CT_PageSetup_Chart.$, b);
                    break;
                case 889:
                    this.g = typeCast(CT_RelId.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_PrintSettings.$t = /*@__PURE__*/ markType(CT_PrintSettings, 'CT_PrintSettings', Base.$, [ISupportSetProperty_$type]);
    return CT_PrintSettings;
})();
