// igniteui-react-excel/esm2015/lib/CT_PivotField_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
export let CT_PivotField_SML2009 = /*@__PURE__*/ (() => {
    class CT_PivotField_SML2009 extends Base {
        constructor() {
            super();
            this.d = toNullable(Boolean_$type, null);
            this.f = toNullable(Boolean_$type, null);
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        static g(a, b, c) {
            b = toNullable(Boolean_$type, null);
            c = toNullable(Boolean_$type, null);
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "fillDownLabels":
                        b = toNullable(Boolean_$type, ST_OnOff.e(d.d));
                        break;
                    case "ignore":
                        c = toNullable(Boolean_$type, ST_OnOff.e(d.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static h(a, b, c = toNullable(Boolean_$type, null), d = toNullable(Boolean_$type, null)) {
            a.ad.p(b, null);
            let e = null;
            let f = null;
            if (c.hasValue) {
                f = "fillDownLabels";
                e = ST_OnOff.g(c.value);
                a.ad.h(f, e);
            }
            if (d.hasValue) {
                f = "ignore";
                e = ST_OnOff.g(d.value);
                a.ad.h(f, e);
            }
            ;
        }
    }
    CT_PivotField_SML2009.$t = /*@__PURE__*/ markType(CT_PivotField_SML2009, 'CT_PivotField_SML2009');
    return CT_PivotField_SML2009;
})();
