// igniteui-react-excel/esm2015/lib/CT_PolarAdjustHandle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, markType } from "igniteui-react-core";
import { IAdjustHandle_$type } from "./IAdjustHandle";
import { ST_GeomGuideName } from "./ST_GeomGuideName";
import { ST_AdjCoordinate } from "./ST_AdjCoordinate";
import { ST_AdjAngle } from "./ST_AdjAngle";
/**
 * @hidden
 */
export let CT_PolarAdjustHandle = /*@__PURE__*/ (() => {
    class CT_PolarAdjustHandle extends Base {
        constructor() {
            super(...arguments);
            this.a = null;
            this.c = toNullable(ST_GeomGuideName.$, null);
            this.g = toNullable(ST_AdjCoordinate.$, null);
            this.e = toNullable(ST_AdjCoordinate.$, null);
            this.b = toNullable(ST_GeomGuideName.$, null);
            this.f = toNullable(ST_AdjAngle.$, null);
            this.d = toNullable(ST_AdjAngle.$, null);
        }
    }
    CT_PolarAdjustHandle.$t = /*@__PURE__*/ markType(CT_PolarAdjustHandle, 'CT_PolarAdjustHandle', Base.$, [IAdjustHandle_$type]);
    return CT_PolarAdjustHandle;
})();
