// igniteui-react-excel/esm2015/lib/CT_PivotCacheDefinition_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, Number_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_PivotCacheDefinition_SML2009 = /*@__PURE__*/ (() => {
    class CT_PivotCacheDefinition_SML2009 extends Base {
        constructor() {
            super();
            this.h = toNullable(Boolean_$type, null);
            this.f = toNullable(Number_$type, null);
            this.n = toNullable(Boolean_$type, null);
            this.l = toNullable(Boolean_$type, null);
            this.j = toNullable(Boolean_$type, null);
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get m() {
            return this.n;
        }
        set m(a) {
            this.n = a;
        }
        get k() {
            return this.l;
        }
        set k(a) {
            this.l = a;
        }
        get i() {
            return this.j;
        }
        set i(a) {
            this.j = a;
        }
        static o(a, b, c, d, e, f) {
            b = toNullable(Boolean_$type, null);
            c = toNullable(Number_$type, null);
            d = toNullable(Boolean_$type, null);
            e = toNullable(Boolean_$type, null);
            f = toNullable(Boolean_$type, null);
            for (let g of fromEnum(a.ab.u)) {
                switch (g.a.e(a.d)) {
                    case "slicerData":
                        b = toNullable(Boolean_$type, ST_OnOff.e(g.d));
                        break;
                    case "pivotCacheId":
                        c = toNullable(Number_$type, parseUInt32_1(g.d, CultureInfo.invariantCulture));
                        break;
                    case "supportSubqueryNonVisual":
                        d = toNullable(Boolean_$type, ST_OnOff.e(g.d));
                        break;
                    case "supportSubqueryCalcMem":
                        e = toNullable(Boolean_$type, ST_OnOff.e(g.d));
                        break;
                    case "supportAddCalcMems":
                        f = toNullable(Boolean_$type, ST_OnOff.e(g.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d,
                p4: e,
                p5: f
            };
        }
        static p(a, b, c = toNullable(Number_$type, null), d = toNullable(Boolean_$type, null), e = toNullable(Boolean_$type, null), f = toNullable(Boolean_$type, null), g = toNullable(Boolean_$type, null)) {
            a.ad.p(b, null);
            let h = null;
            let i = null;
            if (c.hasValue) {
                i = "pivotCacheId";
                h = c.value.toString();
                a.ad.h(i, h);
            }
            if (d.hasValue) {
                i = "slicerData";
                h = ST_OnOff.g(d.value);
                a.ad.h(i, h);
            }
            if (e.hasValue) {
                i = "supportAddCalcMems";
                h = ST_OnOff.g(e.value);
                a.ad.h(i, h);
            }
            if (f.hasValue) {
                i = "supportSubqueryCalcMem";
                h = ST_OnOff.g(f.value);
                a.ad.h(i, h);
            }
            if (g.hasValue) {
                i = "supportSubqueryNonVisual";
                h = ST_OnOff.g(g.value);
                a.ad.h(i, h);
            }
            ;
        }
    }
    CT_PivotCacheDefinition_SML2009.$t = /*@__PURE__*/ markType(CT_PivotCacheDefinition_SML2009, 'CT_PivotCacheDefinition_SML2009');
    return CT_PivotCacheDefinition_SML2009;
})();
