// igniteui-react-excel/esm2015/lib/CT_PivotEdit.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, NotSupportedException, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_PivotEdit = /*@__PURE__*/ (() => {
    class CT_PivotEdit extends Base {
        constructor() {
            super();
            this.f = null;
            this.h = null;
            this.d = null;
            this.b = null;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static i(a) {
            throw new NotSupportedException(1, "The corresponding XML element for this class has no attributes, so there is nothing to read.");
        }
        static j(a, b) {
            throw new NotSupportedException(1, "The corresponding XML element for this class has no attributes, so there is nothing to write.");
        }
    }
    CT_PivotEdit.$t = /*@__PURE__*/ markType(CT_PivotEdit, 'CT_PivotEdit');
    return CT_PivotEdit;
})();
