// igniteui-react-excel/esm2015/lib/CT_RelativeRect.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { ST_Percentage } from "./ST_Percentage";
/**
 * @hidden
 */
export let CT_RelativeRect = /*@__PURE__*/ (() => {
    class CT_RelativeRect extends Base {
        constructor() {
            super(...arguments);
            this.a = new ST_Percentage();
            this.b = new ST_Percentage();
            this.c = new ST_Percentage();
            this.d = new ST_Percentage();
        }
        get e() {
            return ST_Percentage.e(this.a, ST_Percentage.b(0)) || ST_Percentage.e(this.b, ST_Percentage.b(0)) || ST_Percentage.e(this.c, ST_Percentage.b(0)) || ST_Percentage.e(this.d, ST_Percentage.b(0));
        }
    }
    CT_RelativeRect.$t = /*@__PURE__*/ markType(CT_RelativeRect, 'CT_RelativeRect');
    return CT_RelativeRect;
})();
