// igniteui-react-excel/esm2015/lib/CT_Ref.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_Ref = /*@__PURE__*/ (() => {
    class CT_Ref extends Base {
        constructor() {
            super(...arguments);
            this.f = false;
            this.c = false;
            this.d = false;
            this.h = null;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        get a() {
            return this.d;
        }
        set a(a) {
            this.d = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
    }
    CT_Ref.$t = /*@__PURE__*/ markType(CT_Ref, 'CT_Ref');
    return CT_Ref;
})();
