// igniteui-react-excel/esm2015/lib/CT_Ratio.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseInt64_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_Ratio = /*@__PURE__*/ (() => {
    class CT_Ratio extends Base {
        constructor() {
            super(...arguments);
            this.d = 0;
            this.b = 0;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static f(a, b, c) {
            b = 0;
            c = 0;
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "n":
                        b = parseInt64_1(d.d, CultureInfo.invariantCulture);
                        break;
                    case "d":
                        c = parseInt64_1(d.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static g(a, b, c, d) {
            a.ad.p(b, "a");
            let e = null;
            let f = null;
            f = "d";
            e = c.toString();
            a.ad.h(f, e);
            f = "n";
            e = d.toString();
            a.ad.h(f, e);
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_Ratio.$t = /*@__PURE__*/ markType(CT_Ratio, 'CT_Ratio');
    return CT_Ratio;
})();
