// igniteui-react-excel/esm2015/lib/CT_SphereCoords.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_SphereCoords = /*@__PURE__*/ (() => {
    class CT_SphereCoords extends Base {
        constructor() {
            super(...arguments);
            this.b = 0;
            this.d = 0;
            this.f = 0;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        static h(a, b, c, d) {
            b = 0;
            c = 0;
            d = 0;
            for (let e of fromEnum(a.ab.u)) {
                switch (e.a.e(a.d)) {
                    case "lat":
                        b = parseInt32_1(e.d, CultureInfo.invariantCulture);
                        break;
                    case "lon":
                        c = parseInt32_1(e.d, CultureInfo.invariantCulture);
                        break;
                    case "rev":
                        d = parseInt32_1(e.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d
            };
        }
        static i(a, b, c, d, e) {
            a.ad.p(b, "a");
            let f = null;
            let g = null;
            g = "lat";
            f = c.toString();
            a.ad.h(g, f);
            g = "lon";
            f = d.toString();
            a.ad.h(g, f);
            g = "rev";
            f = e.toString();
            a.ad.h(g, f);
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_SphereCoords.$t = /*@__PURE__*/ markType(CT_SphereCoords, 'CT_SphereCoords');
    return CT_SphereCoords;
})();
