// igniteui-react-excel/esm2015/lib/CT_StringDimension.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_Formula } from "./CT_Formula";
import { List$1 } from "igniteui-react-core";
import { CT_StringLevel } from "./CT_StringLevel";
import { ST_StringDimensionType_$type } from "./ST_StringDimensionType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
export let CT_StringDimension = /*@__PURE__*/ (() => {
    class CT_StringDimension extends Base {
        constructor() {
            super();
            this.d = 0;
            this.a = null;
            this.b = null;
            this.e = null;
            this.e = new List$1(CT_StringLevel.$, 0);
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        static g(a, b) {
            b = 0;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "type":
                        b = XmlNamespaceDefinitionBase.b(ST_StringDimensionType_$type, c.d);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static h(a, b, c) {
            a.ad.p(b, "cx");
            let d = null;
            let e = null;
            e = "type";
            d = XmlNamespaceDefinitionBase.p(ST_StringDimensionType_$type, c);
            a.ad.h(e, d);
            ;
        }
        setProperty(a, b) {
            switch (a) {
                case 612:
                    this.a = typeCast(CT_Formula.$, b);
                    break;
                case 1907:
                    this.b = typeCast(CT_Formula.$, b);
                    break;
                case 944:
                    this.e.add(typeCast(CT_StringLevel.$, b));
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_StringDimension.$t = /*@__PURE__*/ markType(CT_StringDimension, 'CT_StringDimension', Base.$, [ISupportSetProperty_$type]);
    return CT_StringDimension;
})();
