// igniteui-react-excel/esm2015/lib/CT_Sqref.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_Sqref = /*@__PURE__*/ (() => {
    class CT_Sqref extends Base {
        constructor() {
            super();
            this.f = false;
            this.h = false;
            this.c = false;
            this.d = false;
            this.j = null;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        get a() {
            return this.d;
        }
        set a(a) {
            this.d = a;
        }
        get i() {
            return this.j;
        }
        set i(a) {
            this.j = a;
        }
    }
    CT_Sqref.$t = /*@__PURE__*/ markType(CT_Sqref, 'CT_Sqref');
    return CT_Sqref;
})();
