// igniteui-react-excel/esm2015/lib/CT_Skip.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt16_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_Skip = /*@__PURE__*/ (() => {
    class CT_Skip extends Base {
        constructor() {
            super(...arguments);
            this.c = 0;
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        static d(a, b) {
            b = 0;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "val":
                        b = parseUInt16_1(c.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static e(a, b, c) {
            a.ad.p(b, "c");
            let d = null;
            let e = null;
            e = "val";
            d = c.toString();
            a.ad.h(e, d);
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_Skip.$t = /*@__PURE__*/ markType(CT_Skip, 'CT_Skip');
    return CT_Skip;
})();
