// igniteui-react-excel/esm2015/lib/CT_SlicerCachePivotTable.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_SlicerCachePivotTable = /*@__PURE__*/ (() => {
    class CT_SlicerCachePivotTable extends Base {
        constructor() {
            super(...arguments);
            this.d = 0;
            this.b = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static e(a, b, c) {
            b = 0;
            c = null;
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "tabId":
                        b = parseUInt32_1(d.d, CultureInfo.invariantCulture);
                        break;
                    case "name":
                        c = d.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static f(a, b, c, d) {
            a.ad.p(b, null);
            let e = null;
            let f = null;
            if (c != null) {
                f = "name";
                e = c;
                a.ad.h(f, e);
            }
            f = "tabId";
            e = d.toString();
            a.ad.h(f, e);
            ;
        }
    }
    CT_SlicerCachePivotTable.$t = /*@__PURE__*/ markType(CT_SlicerCachePivotTable, 'CT_SlicerCachePivotTable');
    return CT_SlicerCachePivotTable;
})();
