// igniteui-react-excel/esm2015/lib/CT_StringValue.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { ChartExElements } from "./ChartExElements";
/**
 * @hidden
 */
export let CT_StringValue = /*@__PURE__*/ (() => {
    class CT_StringValue extends Base {
        constructor() {
            super(...arguments);
            this.e = 0;
            this.c = null;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        toString() {
            return this.b;
        }
        static f(a, b, c) {
            a.ad.p(c, "cx");
            a.ad.h(ChartExElements.a7, b.toString());
        }
    }
    CT_StringValue.$t = /*@__PURE__*/ markType(CT_StringValue, 'CT_StringValue');
    return CT_StringValue;
})();
