// igniteui-react-excel/esm2015/lib/CT_ShapeProperties.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CT_ShapePropertiesBase } from "./CT_ShapePropertiesBase";
import { toNullable, fromEnum, markType } from "igniteui-react-core";
import { ST_BlackWhiteMode_$type } from "./ST_BlackWhiteMode";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
export let CT_ShapeProperties = /*@__PURE__*/ (() => {
    class CT_ShapeProperties extends CT_ShapePropertiesBase {
        constructor() {
            super(...arguments);
            this.e = null;
            this.c = null;
            this.d = null;
        }
        static f(a, b) {
            b = toNullable(ST_BlackWhiteMode_$type, null);
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "bwMode":
                        b = toNullable(ST_BlackWhiteMode_$type, XmlNamespaceDefinitionBase.b(ST_BlackWhiteMode_$type, c.d));
                        break;
                    default: break;
                }
            }
            return {
                p1: b
            };
        }
    }
    CT_ShapeProperties.$t = /*@__PURE__*/ markType(CT_ShapeProperties, 'CT_ShapeProperties', CT_ShapePropertiesBase.$);
    return CT_ShapeProperties;
})();
