// igniteui-react-excel/esm2015/lib/CT_SlicerRef.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_SlicerRef = /*@__PURE__*/ (() => {
    class CT_SlicerRef extends Base {
        constructor() {
            super(...arguments);
            this.b = null;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static c(a, b) {
            b = null;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "http://schemas.openxmlformats.org/officeDocument/2006/relationships:id":
                        b = c.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static d(a, b, c) {
            a.ad.p(b, null);
            let d = null;
            let e = null;
            if (c != null) {
                e = "http://schemas.openxmlformats.org/officeDocument/2006/relationships:id";
                d = c;
                a.ad.h(e, d);
            }
            ;
        }
    }
    CT_SlicerRef.$t = /*@__PURE__*/ markType(CT_SlicerRef, 'CT_SlicerRef');
    return CT_SlicerRef;
})();
