// igniteui-react-excel/esm2015/lib/CT_SetLevel.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_SetLevel = /*@__PURE__*/ (() => {
    class CT_SetLevel extends Base {
        constructor() {
            super(...arguments);
            this.b = 0;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static c(a, b) {
            b = 0;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "hierarchy":
                        b = parseInt32_1(c.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static d(a, b, c) {
            a.ad.p(b, null);
            let d = null;
            let e = null;
            e = "hierarchy";
            d = c.toString();
            a.ad.h(e, d);
            ;
        }
    }
    CT_SetLevel.$t = /*@__PURE__*/ markType(CT_SetLevel, 'CT_SetLevel');
    return CT_SetLevel;
})();
