// igniteui-react-excel/esm2015/lib/CT_StrVal.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, typeCast, String_$type, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_StrVal = /*@__PURE__*/ (() => {
    class CT_StrVal extends Base {
        constructor() {
            super(...arguments);
            this.c = null;
            this.e = 0;
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        static f(a, b) {
            b = 0;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "idx":
                        b = parseUInt32_1(c.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static g(a, b, c) {
            a.ad.p(b, "c");
            let d = null;
            let e = null;
            e = "idx";
            d = c.toString();
            a.ad.h(e, d);
            ;
        }
        setProperty(a, b) {
            switch (a) {
                case 1818:
                    this.b = typeCast(String_$type, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_StrVal.$t = /*@__PURE__*/ markType(CT_StrVal, 'CT_StrVal', Base.$, [ISupportSetProperty_$type]);
    return CT_StrVal;
})();
