// igniteui-react-excel/esm2015/lib/CT_Statistics.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_QuartileMethod_$type } from "./ST_QuartileMethod";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
export let CT_Statistics = /*@__PURE__*/ (() => {
    class CT_Statistics extends Base {
        constructor() {
            super(...arguments);
            this.b = toNullable(ST_QuartileMethod_$type, null);
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static d(a, b) {
            b = toNullable(ST_QuartileMethod_$type, null);
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "quartileMethod":
                        b = toNullable(ST_QuartileMethod_$type, XmlNamespaceDefinitionBase.b(ST_QuartileMethod_$type, c.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static e(a, b, c = toNullable(ST_QuartileMethod_$type, null)) {
            a.ad.p(b, "cx");
            let d = null;
            let e = null;
            if (c.hasValue) {
                e = "quartileMethod";
                d = XmlNamespaceDefinitionBase.p(ST_QuartileMethod_$type, c.value);
                a.ad.h(e, d);
            }
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_Statistics.$t = /*@__PURE__*/ markType(CT_Statistics, 'CT_Statistics');
    return CT_Statistics;
})();
