// igniteui-react-excel/esm2015/lib/CT_StringLevel.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_StringValue } from "./CT_StringValue";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_StringLevel = /*@__PURE__*/ (() => {
    class CT_StringLevel extends Base {
        constructor() {
            super();
            this.b = null;
            this.g = 0;
            this.d = null;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get f() {
            return this.g;
        }
        set f(a) {
            this.g = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        static h(a, b, c) {
            b = 0;
            c = null;
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "ptCount":
                        b = parseUInt32_1(d.d, CultureInfo.invariantCulture);
                        break;
                    case "name":
                        c = d.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static i(a, b, c, d) {
            a.ad.p(b, "cx");
            let e = null;
            let f = null;
            if (c != null) {
                f = "name";
                e = c;
                a.ad.h(f, e);
            }
            f = "ptCount";
            e = d.toString();
            a.ad.h(f, e);
            ;
        }
        setProperty(a, b) {
            switch (a) {
                case 1277:
                    this.a.add(typeCast(CT_StringValue.$, b));
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_StringLevel.$t = /*@__PURE__*/ markType(CT_StringLevel, 'CT_StringLevel', Base.$, [ISupportSetProperty_$type]);
    return CT_StringLevel;
})();
