// igniteui-react-excel/esm2015/lib/CT_TextCharacterPropertiesBase.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, Number_$type, markType } from "igniteui-react-core";
import { ST_TextUnderlineType_$type } from "./ST_TextUnderlineType";
import { ST_TextStrikeType_$type } from "./ST_TextStrikeType";
import { ST_TextCapsType_$type } from "./ST_TextCapsType";
import { ST_TextPoint } from "./ST_TextPoint";
import { ST_Percentage } from "./ST_Percentage";
/**
 * @hidden
 */
export let CT_TextCharacterPropertiesBase = /*@__PURE__*/ (() => {
    class CT_TextCharacterPropertiesBase extends Base {
        constructor() {
            super(...arguments);
            this.c = null;
            this.b = null;
            this.a = null;
            this.k = toNullable(Boolean_$type, null);
            this.v = null;
            this.t = null;
            this.r = toNullable(Number_$type, null);
            this.d = toNullable(Boolean_$type, null);
            this.i = toNullable(Boolean_$type, null);
            this.s = toNullable(ST_TextUnderlineType_$type, null);
            this.q = toNullable(ST_TextStrikeType_$type, null);
            this.j = toNullable(Number_$type, null);
            this.f = toNullable(ST_TextCapsType_$type, null);
            this.p = toNullable(ST_TextPoint.$, null);
            this.m = toNullable(Boolean_$type, null);
            this.e = toNullable(ST_Percentage.$, null);
            this.l = toNullable(Boolean_$type, null);
            this.g = toNullable(Boolean_$type, null);
            this.h = toNullable(Boolean_$type, null);
            this.n = toNullable(Boolean_$type, null);
            this.o = toNullable(Number_$type, null);
            this.u = null;
        }
    }
    CT_TextCharacterPropertiesBase.$t = /*@__PURE__*/ markType(CT_TextCharacterPropertiesBase, 'CT_TextCharacterPropertiesBase');
    return CT_TextCharacterPropertiesBase;
})();
