// igniteui-react-excel/esm2015/lib/CT_TupleSetRowItem.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_TupleSetRowItem = /*@__PURE__*/ (() => {
    class CT_TupleSetRowItem extends Base {
        constructor() {
            super(...arguments);
            this.d = null;
            this.b = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static e(a, b, c) {
            b = null;
            c = null;
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "u":
                        b = d.d;
                        break;
                    case "d":
                        c = d.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static f(a, b, c = null, d = null) {
            a.ad.p(b, null);
            let e = null;
            let f = null;
            if (c != null) {
                f = "d";
                e = c;
                a.ad.h(f, e);
            }
            if (d != null) {
                f = "u";
                e = d;
                a.ad.h(f, e);
            }
            ;
        }
    }
    CT_TupleSetRowItem.$t = /*@__PURE__*/ markType(CT_TupleSetRowItem, 'CT_TupleSetRowItem');
    return CT_TupleSetRowItem;
})();
