// igniteui-react-excel/esm2015/lib/CT_TimeUnit.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_TimeUnit_$type } from "./ST_TimeUnit";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
export let CT_TimeUnit = /*@__PURE__*/ (() => {
    class CT_TimeUnit extends Base {
        constructor() {
            super();
            this.c = 0;
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        static e(a, b) {
            b = 0;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "val":
                        b = XmlNamespaceDefinitionBase.b(ST_TimeUnit_$type, c.d);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static f(a, b, c) {
            a.ad.p(b, "c");
            let d = null;
            let e = null;
            e = "val";
            d = XmlNamespaceDefinitionBase.p(ST_TimeUnit_$type, c);
            a.ad.h(e, d);
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_TimeUnit.$t = /*@__PURE__*/ markType(CT_TimeUnit, 'CT_TimeUnit');
    return CT_TimeUnit;
})();
