// igniteui-react-excel/esm2015/lib/CT_TextTabStop.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_TextTabAlignType_$type } from "./ST_TextTabAlignType";
import { CultureInfo } from "igniteui-react-core";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { parseInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_TextTabStop = /*@__PURE__*/ (() => {
    class CT_TextTabStop extends Base {
        constructor() {
            super(...arguments);
            this.d = toNullable(Number_$type, null);
            this.b = toNullable(ST_TextTabAlignType_$type, null);
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        static f(a, b, c) {
            b = toNullable(Number_$type, null);
            c = toNullable(ST_TextTabAlignType_$type, null);
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "pos":
                        b = toNullable(Number_$type, parseInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    case "algn":
                        c = toNullable(ST_TextTabAlignType_$type, XmlNamespaceDefinitionBase.b(ST_TextTabAlignType_$type, d.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static g(a, b, c = toNullable(ST_TextTabAlignType_$type, null), d = toNullable(Number_$type, null)) {
            a.ad.p(b, "a");
            let e = null;
            let f = null;
            if (c.hasValue) {
                f = "algn";
                e = XmlNamespaceDefinitionBase.p(ST_TextTabAlignType_$type, c.value);
                a.ad.h(f, e);
            }
            if (d.hasValue) {
                f = "pos";
                e = d.value.toString();
                a.ad.h(f, e);
            }
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_TextTabStop.$t = /*@__PURE__*/ markType(CT_TextTabStop, 'CT_TextTabStop');
    return CT_TextTabStop;
})();
