// igniteui-react-excel/esm2015/lib/CT_TextAutonumberBullet.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_TextAutonumberScheme_$type } from "./ST_TextAutonumberScheme";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { CultureInfo } from "igniteui-react-core";
import { parseInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_TextAutonumberBullet = /*@__PURE__*/ (() => {
    class CT_TextAutonumberBullet extends Base {
        constructor() {
            super();
            this.b = 0;
            this.e = toNullable(Number_$type, null);
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        static g(a, b, c) {
            b = 0;
            c = toNullable(Number_$type, null);
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "type":
                        b = XmlNamespaceDefinitionBase.b(ST_TextAutonumberScheme_$type, d.d);
                        break;
                    case "startAt":
                        c = toNullable(Number_$type, parseInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static h(a, b, c, d) {
            a.ad.p(b, "a");
            let e = null;
            let f = null;
            if (c.hasValue) {
                f = "startAt";
                e = c.value.toString();
                a.ad.h(f, e);
            }
            f = "type";
            e = XmlNamespaceDefinitionBase.p(ST_TextAutonumberScheme_$type, d);
            a.ad.h(f, e);
            ;
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_TextAutonumberBullet.$t = /*@__PURE__*/ markType(CT_TextAutonumberBullet, 'CT_TextAutonumberBullet');
    return CT_TextAutonumberBullet;
})();
