// igniteui-react-excel/esm2015/lib/CT_TablePartStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_TableStyleTextStyle } from "./CT_TableStyleTextStyle";
import { CT_TableStyleCellStyle } from "./CT_TableStyleCellStyle";
/**
 * @hidden
 */
export let CT_TablePartStyle = /*@__PURE__*/ (() => {
    class CT_TablePartStyle extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 1703:
                    this.c = typeCast(CT_TableStyleTextStyle.$, b);
                    break;
                case 1702:
                    this.a = typeCast(CT_TableStyleCellStyle.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_TablePartStyle.$t = /*@__PURE__*/ markType(CT_TablePartStyle, 'CT_TablePartStyle', Base.$, [ISupportSetProperty_$type]);
    return CT_TablePartStyle;
})();
