// igniteui-react-excel/esm2015/lib/CT_TabularSlicerCache.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_TabularSlicerCacheSortOrder_$type } from "./ST_TabularSlicerCacheSortOrder";
import { ST_SlicerCacheCrossFilter_$type } from "./ST_SlicerCacheCrossFilter";
import { CultureInfo } from "igniteui-react-core";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { ST_OnOff } from "./ST_OnOff";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_TabularSlicerCache = /*@__PURE__*/ (() => {
    class CT_TabularSlicerCache extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
            this.r = 0;
            this.p = toNullable(ST_TabularSlicerCacheSortOrder_$type, null);
            this.l = toNullable(Boolean_$type, null);
            this.n = toNullable(Boolean_$type, null);
            this.j = toNullable(ST_SlicerCacheCrossFilter_$type, null);
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get q() {
            return this.r;
        }
        set q(a) {
            this.r = a;
        }
        get o() {
            return this.p;
        }
        set o(a) {
            this.p = a;
        }
        get k() {
            return this.l;
        }
        set k(a) {
            this.l = a;
        }
        get m() {
            return this.n;
        }
        set m(a) {
            this.n = a;
        }
        get i() {
            return this.j;
        }
        set i(a) {
            this.j = a;
        }
        static s(a, b, c, d, e, f) {
            b = 0;
            c = toNullable(ST_TabularSlicerCacheSortOrder_$type, null);
            d = toNullable(Boolean_$type, null);
            e = toNullable(Boolean_$type, null);
            f = toNullable(ST_SlicerCacheCrossFilter_$type, null);
            for (let g of fromEnum(a.ab.u)) {
                switch (g.a.e(a.d)) {
                    case "pivotCacheId":
                        b = parseUInt32_1(g.d, CultureInfo.invariantCulture);
                        break;
                    case "sortOrder":
                        c = toNullable(ST_TabularSlicerCacheSortOrder_$type, XmlNamespaceDefinitionBase.b(ST_TabularSlicerCacheSortOrder_$type, g.d));
                        break;
                    case "customListSort":
                        d = toNullable(Boolean_$type, ST_OnOff.e(g.d));
                        break;
                    case "showMissing":
                        e = toNullable(Boolean_$type, ST_OnOff.e(g.d));
                        break;
                    case "crossFilter":
                        f = toNullable(ST_SlicerCacheCrossFilter_$type, XmlNamespaceDefinitionBase.b(ST_SlicerCacheCrossFilter_$type, g.d));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d,
                p4: e,
                p5: f
            };
        }
        static t(a, b, c, d, e, f, g) {
            a.ad.p(b, null);
            let h = null;
            let i = null;
            if (c.hasValue) {
                i = "crossFilter";
                h = XmlNamespaceDefinitionBase.p(ST_SlicerCacheCrossFilter_$type, c.value);
                a.ad.h(i, h);
            }
            if (d.hasValue) {
                i = "customListSort";
                h = ST_OnOff.g(d.value);
                a.ad.h(i, h);
            }
            if (e.hasValue) {
                i = "showMissing";
                h = ST_OnOff.g(e.value);
                a.ad.h(i, h);
            }
            if (f.hasValue) {
                i = "sortOrder";
                h = XmlNamespaceDefinitionBase.p(ST_TabularSlicerCacheSortOrder_$type, f.value);
                a.ad.h(i, h);
            }
            i = "pivotCacheId";
            h = g.toString();
            a.ad.h(i, h);
            ;
        }
    }
    CT_TabularSlicerCache.$t = /*@__PURE__*/ markType(CT_TabularSlicerCache, 'CT_TabularSlicerCache');
    return CT_TabularSlicerCache;
})();
