// igniteui-react-excel/esm2015/lib/CT_Transform2D.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, markType } from "igniteui-react-core";
import { ST_Angle } from "./ST_Angle";
/**
 * @hidden
 */
export let CT_Transform2D = /*@__PURE__*/ (() => {
    class CT_Transform2D extends Base {
        constructor() {
            super();
            this.b = null;
            this.i = toNullable(Boolean_$type, null);
            this.a = null;
            this.j = toNullable(ST_Angle.$, null);
            this.k = toNullable(Boolean_$type, null);
            this.i = toNullable(Boolean_$type, null);
            this.k = toNullable(Boolean_$type, null);
            this.j = toNullable(ST_Angle.$, null);
        }
        g() {
            return this.i.hasValue ? this.i.value : false;
        }
        h() {
            return this.k.hasValue ? this.k.value : false;
        }
        d() {
            return this.j.hasValue ? this.j.value : CT_Transform2D.c;
        }
    }
    CT_Transform2D.$t = /*@__PURE__*/ markType(CT_Transform2D, 'CT_Transform2D');
    CT_Transform2D.c = /*@__PURE__*/ ST_Angle.a(0);
    return CT_Transform2D;
})();
