// igniteui-react-excel/esm2015/lib/CT_TableStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_TableBackgroundStyle } from "./CT_TableBackgroundStyle";
import { CT_TablePartStyle } from "./CT_TablePartStyle";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
export let CT_TableStyle = /*@__PURE__*/ (() => {
    class CT_TableStyle extends Base {
        constructor() {
            super();
            this.d = null;
            this.ad = null;
            this.f = null;
            this.j = null;
            this.h = null;
            this.l = null;
            this.r = null;
            this.n = null;
            this.t = null;
            this.z = null;
            this.ab = null;
            this.p = null;
            this.v = null;
            this.x = null;
            this.b = null;
            this.af = null;
            this.ah = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get ac() {
            return this.ad;
        }
        set ac(a) {
            this.ad = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get i() {
            return this.j;
        }
        set i(a) {
            this.j = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get k() {
            return this.l;
        }
        set k(a) {
            this.l = a;
        }
        get q() {
            return this.r;
        }
        set q(a) {
            this.r = a;
        }
        get m() {
            return this.n;
        }
        set m(a) {
            this.n = a;
        }
        get s() {
            return this.t;
        }
        set s(a) {
            this.t = a;
        }
        get y() {
            return this.z;
        }
        set y(a) {
            this.z = a;
        }
        get aa() {
            return this.ab;
        }
        set aa(a) {
            this.ab = a;
        }
        get o() {
            return this.p;
        }
        set o(a) {
            this.p = a;
        }
        get u() {
            return this.v;
        }
        set u(a) {
            this.v = a;
        }
        get w() {
            return this.x;
        }
        set w(a) {
            this.x = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get ae() {
            return this.af;
        }
        set ae(a) {
            this.af = a;
        }
        get ag() {
            return this.ah;
        }
        set ag(a) {
            this.ah = a;
        }
        static aj(a, b, c) {
            b = null;
            c = null;
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "styleId":
                        b = d.d;
                        break;
                    case "styleName":
                        c = d.d;
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static ak(a, b, c, d) {
            a.ad.p(b, "a");
            let e = null;
            let f = null;
            if (c != null) {
                f = "styleId";
                e = c;
                a.ad.h(f, e);
            }
            if (d != null) {
                f = "styleName";
                e = d;
                a.ad.h(f, e);
            }
            ;
        }
        setProperty(a, b) {
            switch (a) {
                case 1680:
                    this.c = typeCast(CT_TableBackgroundStyle.$, b);
                    break;
                case 1866:
                    this.ac = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 92:
                    this.e = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 94:
                    this.i = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 93:
                    this.g = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 95:
                    this.k = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 871:
                    this.q = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 643:
                    this.m = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 873:
                    this.s = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 1437:
                    this.y = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 1647:
                    this.aa = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 646:
                    this.o = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 1052:
                    this.u = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 1107:
                    this.w = typeCast(CT_TablePartStyle.$, b);
                    break;
                case 601:
                    this.a = typeCast(CT_OfficeArtExtensionList.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_TableStyle.$t = /*@__PURE__*/ markType(CT_TableStyle, 'CT_TableStyle', Base.$, [ISupportSetProperty_$type]);
    return CT_TableStyle;
})();
