// igniteui-react-excel/esm2015/lib/CT_TableStyleCellStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_TableCellBorderStyle } from "./CT_TableCellBorderStyle";
import { CT_Cell3D } from "./CT_Cell3D";
import { EG_ThemeableFillStyle } from "./EG_ThemeableFillStyle";
/**
 * @hidden
 */
export let CT_TableStyleCellStyle = /*@__PURE__*/ (() => {
    class CT_TableStyleCellStyle extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
            this.e = null;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get f() {
            if (this.e == null) {
                this.e = new EG_ThemeableFillStyle();
            }
            return this.e;
        }
        setProperty(a, b) {
            switch (a) {
                case 1697:
                    this.c = typeCast(CT_TableCellBorderStyle.$, b);
                    break;
                case 235:
                    this.a = typeCast(CT_Cell3D.$, b);
                    break;
                default:
                    if (this.f.c(a)) {
                        this.f.e(a, b);
                        return;
                    }
                    break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_TableStyleCellStyle.$t = /*@__PURE__*/ markType(CT_TableStyleCellStyle, 'CT_TableStyleCellStyle', Base.$, [ISupportSetProperty_$type]);
    return CT_TableStyleCellStyle;
})();
