// igniteui-react-excel/esm2015/lib/CT_TupleSet.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, fromEnum, markType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_TupleSet = /*@__PURE__*/ (() => {
    class CT_TupleSet extends Base {
        constructor() {
            super();
            this.b = null;
            this.d = null;
            this.h = toNullable(Number_$type, null);
            this.f = toNullable(Number_$type, null);
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        static k(a, b, c) {
            b = toNullable(Number_$type, null);
            c = toNullable(Number_$type, null);
            for (let d of fromEnum(a.ab.u)) {
                switch (d.a.e(a.d)) {
                    case "rowCount":
                        b = toNullable(Number_$type, parseUInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    case "columnCount":
                        c = toNullable(Number_$type, parseUInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c
            };
        }
        static l(a, b, c = toNullable(Number_$type, null), d = toNullable(Number_$type, null)) {
            a.ad.p(b, null);
            let e = null;
            let f = null;
            if (c.hasValue) {
                f = "columnCount";
                e = c.value.toString();
                a.ad.h(f, e);
            }
            if (d.hasValue) {
                f = "rowCount";
                e = d.value.toString();
                a.ad.h(f, e);
            }
            ;
        }
    }
    CT_TupleSet.$t = /*@__PURE__*/ markType(CT_TupleSet, 'CT_TupleSet');
    return CT_TupleSet;
})();
