// igniteui-react-excel/esm2015/lib/CT_TextTabStopList.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_TextTabStop } from "./CT_TextTabStop";
/**
 * @hidden
 */
export let CT_TextTabStopList = /*@__PURE__*/ (() => {
    class CT_TextTabStopList extends Base {
        constructor() {
            super();
            this.f = null;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 1656:
                    this.e.add(typeCast(CT_TextTabStop.$, b));
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
        static c(a) {
            return a != null ? a.e : null;
        }
        static a(a) {
            let b = new CT_TextTabStopList();
            b.f = a;
            return b;
        }
    }
    CT_TextTabStopList.$t = /*@__PURE__*/ markType(CT_TextTabStopList, 'CT_TextTabStopList', Base.$, [ISupportSetProperty_$type]);
    return CT_TextTabStopList;
})();
