// igniteui-react-excel/esm2015/lib/CT_TableCellBorderStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_ThemeableLineStyle } from "./CT_ThemeableLineStyle";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
export let CT_TableCellBorderStyle = /*@__PURE__*/ (() => {
    class CT_TableCellBorderStyle extends Base {
        constructor() {
            super();
            this.j = null;
            this.l = null;
            this.p = null;
            this.d = null;
            this.f = null;
            this.h = null;
            this.n = null;
            this.r = null;
            this.b = null;
        }
        get i() {
            return this.j;
        }
        set i(a) {
            this.j = a;
        }
        get k() {
            return this.l;
        }
        set k(a) {
            this.l = a;
        }
        get o() {
            return this.p;
        }
        set o(a) {
            this.p = a;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        get g() {
            return this.h;
        }
        set g(a) {
            this.h = a;
        }
        get m() {
            return this.n;
        }
        set m(a) {
            this.n = a;
        }
        get q() {
            return this.r;
        }
        set q(a) {
            this.r = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        setProperty(a, b) {
            switch (a) {
                case 887:
                    this.i = typeCast(CT_ThemeableLineStyle.$, b);
                    break;
                case 1363:
                    this.k = typeCast(CT_ThemeableLineStyle.$, b);
                    break;
                case 1754:
                    this.o = typeCast(CT_ThemeableLineStyle.$, b);
                    break;
                case 160:
                    this.c = typeCast(CT_ThemeableLineStyle.$, b);
                    break;
                case 833:
                    this.e = typeCast(CT_ThemeableLineStyle.$, b);
                    break;
                case 834:
                    this.g = typeCast(CT_ThemeableLineStyle.$, b);
                    break;
                case 1749:
                    this.m = typeCast(CT_ThemeableLineStyle.$, b);
                    break;
                case 1761:
                    this.q = typeCast(CT_ThemeableLineStyle.$, b);
                    break;
                case 601:
                    this.a = typeCast(CT_OfficeArtExtensionList.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_TableCellBorderStyle.$t = /*@__PURE__*/ markType(CT_TableCellBorderStyle, 'CT_TableCellBorderStyle', Base.$, [ISupportSetProperty_$type]);
    return CT_TableCellBorderStyle;
})();
