// igniteui-react-excel/esm2015/lib/CT_TableRow.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_TableCell } from "./CT_TableCell";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
import { CultureInfo } from "igniteui-react-core";
import { parseInt64_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_TableRow = /*@__PURE__*/ (() => {
    class CT_TableRow extends Base {
        constructor() {
            super();
            this.d = null;
            this.b = null;
            this.f = 0;
        }
        get c() {
            return this.d;
        }
        set c(a) {
            this.d = a;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
        static h(a, b) {
            b = 0;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "h":
                        b = parseInt64_1(c.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b
            };
        }
        static i(a, b, c) {
            a.ad.p(b, "a");
            let d = null;
            let e = null;
            e = "h";
            d = c.toString();
            a.ad.h(e, d);
            ;
        }
        setProperty(a, b) {
            switch (a) {
                case 1696:
                    this.c.add(typeCast(CT_TableCell.$, b));
                    break;
                case 601:
                    this.a = typeCast(CT_OfficeArtExtensionList.$, b);
                    break;
                default: break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_TableRow.$t = /*@__PURE__*/ markType(CT_TableRow, 'CT_TableRow', Base.$, [ISupportSetProperty_$type]);
    return CT_TableRow;
})();
