// igniteui-react-excel/esm2015/lib/CT_VerticalAlignFontProperty.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_VerticalAlignRun_$type } from "./ST_VerticalAlignRun";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
export let CT_VerticalAlignFontProperty = /*@__PURE__*/ (() => {
    class CT_VerticalAlignFontProperty extends Base {
        constructor() {
            super(...arguments);
            this.a = 0;
        }
        static c(a, b) {
            b = 0;
            for (let c of fromEnum(a.ab.u)) {
                switch (c.a.e(a.d)) {
                    case "val":
                        b = XmlNamespaceDefinitionBase.b(ST_VerticalAlignRun_$type, c.d);
                        break;
                    default: break;
                }
            }
            return {
                p1: b
            };
        }
        static d(a, b, c) {
            a.ad.p(b, null);
            let d = null;
            let e = null;
            e = "val";
            d = XmlNamespaceDefinitionBase.p(ST_VerticalAlignRun_$type, c);
            a.ad.h(e, d);
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_VerticalAlignFontProperty.$t = /*@__PURE__*/ markType(CT_VerticalAlignFontProperty, 'CT_VerticalAlignFontProperty');
    return CT_VerticalAlignFontProperty;
})();
