// igniteui-react-excel/esm2015/lib/CT_WorkbookPr_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, fromEnum, markType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let CT_WorkbookPr_SML2009 = /*@__PURE__*/ (() => {
    class CT_WorkbookPr_SML2009 extends Base {
        constructor() {
            super();
            this.i = 0;
            this.c = false;
            this.e = 0;
        }
        get h() {
            return this.i;
        }
        set h(a) {
            this.i = a;
        }
        get b() {
            return this.c;
        }
        set b(a) {
            this.c = a;
        }
        get d() {
            return this.e;
        }
        set d(a) {
            this.e = a;
        }
        static j(a, b, c, d) {
            b = 0;
            c = false;
            d = 0;
            for (let e of fromEnum(a.ab.u)) {
                switch (e.a.e(a.d)) {
                    case "defaultImageDpi":
                        b = parseUInt32_1(e.d, CultureInfo.invariantCulture);
                        break;
                    case "discardImageEditData":
                        c = ST_OnOff.e(e.d);
                        break;
                    case "accuracyVersion":
                        d = parseUInt32_1(e.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
            ;
            return {
                p1: b,
                p2: c,
                p3: d
            };
        }
        static k(a, b, c, d, e) {
            a.ad.p(b, null);
            let f = null;
            let g = null;
            g = "accuracyVersion";
            f = c.toString();
            a.ad.h(g, f);
            g = "defaultImageDpi";
            f = d.toString();
            a.ad.h(g, f);
            g = "discardImageEditData";
            f = ST_OnOff.g(e);
            a.ad.h(g, f);
            ;
        }
    }
    CT_WorkbookPr_SML2009.$t = /*@__PURE__*/ markType(CT_WorkbookPr_SML2009, 'CT_WorkbookPr_SML2009');
    return CT_WorkbookPr_SML2009;
})();
