// igniteui-react-excel/esm2015/lib/CT_WholeE2oFormatting.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_LineProperties_DML } from "./CT_LineProperties_DML";
import { EG_EffectProperties } from "./EG_EffectProperties";
/**
 * @hidden
 */
export let CT_WholeE2oFormatting = /*@__PURE__*/ (() => {
    class CT_WholeE2oFormatting extends Base {
        constructor() {
            super();
            this.b = null;
            this.c = null;
        }
        get a() {
            return this.b;
        }
        set a(a) {
            this.b = a;
        }
        get d() {
            if (this.c == null) {
                this.c = new EG_EffectProperties();
            }
            return this.c;
        }
        setProperty(a, b) {
            switch (a) {
                case 918:
                    this.a = typeCast(CT_LineProperties_DML.$, b);
                    break;
                default:
                    if (this.d.c(a)) {
                        this.d.e(a, b);
                        return;
                    }
                    break;
            }
        }
        toString() {
            let a = getInstanceType(this);
            return a != null ? a.typeName : super.toString();
        }
    }
    CT_WholeE2oFormatting.$t = /*@__PURE__*/ markType(CT_WholeE2oFormatting, 'CT_WholeE2oFormatting', Base.$, [ISupportSetProperty_$type]);
    return CT_WholeE2oFormatting;
})();
