// igniteui-react-excel/esm2015/lib/CtrlPropPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { PartManagerBase } from "./PartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CtrlPropPartManager = /*@__PURE__*/ (() => {
    class CtrlPropPartManager extends PartManagerBase {
        get_m() {
            return CtrlPropPartManager.s;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/ctrlProp.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return CtrlPropPartManager.u;
                case 1: return CtrlPropPartManager.t;
                default: return CtrlPropPartManager.u;
            }
        }
        get_g() {
            return false;
        }
        get g() {
            return this.get_g();
        }
    }
    CtrlPropPartManager.$t = /*@__PURE__*/ markType(CtrlPropPartManager, 'CtrlPropPartManager', PartManagerBase.$);
    CtrlPropPartManager.s = "application/vnd.ms-excel.controlproperties+xml";
    CtrlPropPartManager.u = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/ctrlProp";
    CtrlPropPartManager.t = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/ctrlProp";
    return CtrlPropPartManager;
})();
