// igniteui-react-excel/esm2015/lib/CustomXmlItemPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { PartManagerBase } from "./PartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CustomXmlItemPartManager = /*@__PURE__*/ (() => {
    class CustomXmlItemPartManager extends PartManagerBase {
        get_m() {
            return "application/xml";
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/customXml/item.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return CustomXmlItemPartManager.u;
                case 1: return CustomXmlItemPartManager.t;
                default: return CustomXmlItemPartManager.u;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    CustomXmlItemPartManager.$t = /*@__PURE__*/ markType(CustomXmlItemPartManager, 'CustomXmlItemPartManager', PartManagerBase.$);
    CustomXmlItemPartManager.u = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/customXml";
    CustomXmlItemPartManager.t = "http://purl.oclc.org/ooxml/officeDocument/relationships/customXml";
    return CustomXmlItemPartManager;
})();
