// igniteui-react-excel/esm2015/lib/CustomXmlItemPropsPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { PartManagerBase } from "./PartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CustomXmlItemPropsPartManager = /*@__PURE__*/ (() => {
    class CustomXmlItemPropsPartManager extends PartManagerBase {
        get_m() {
            return CustomXmlItemPropsPartManager.s;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/customXml/itemProps.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return CustomXmlItemPropsPartManager.u;
                case 1: return CustomXmlItemPropsPartManager.t;
                default: return CustomXmlItemPropsPartManager.u;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    CustomXmlItemPropsPartManager.$t = /*@__PURE__*/ markType(CustomXmlItemPropsPartManager, 'CustomXmlItemPropsPartManager', PartManagerBase.$);
    CustomXmlItemPropsPartManager.s = "application/vnd.openxmlformats-officedocument.customXmlProperties+xml";
    CustomXmlItemPropsPartManager.u = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/customXmlProps";
    CustomXmlItemPropsPartManager.t = "http://purl.oclc.org/ooxml/officeDocument/relationships/customXmlProps";
    return CustomXmlItemPropsPartManager;
})();
