// igniteui-react-excel/esm2015/lib/DataValidationCriteria.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Determines what types of cell values are allowed and how the cell value is validated against the constraint(s).
 * @see [[ValueConstraintDataValidationRule.validationCriteria]]
 * @see [[OneConstraintDataValidationRule]]
 * @see [[TwoConstraintDataValidationRule]]
 */
export var DataValidationCriteria = /*@__PURE__*/ (function (DataValidationCriteria) {
    /**
     * Only numbers are allowed and they cannot have a fractional part.
     * When comparing against constraints, the cell value is used.
     */
    DataValidationCriteria[DataValidationCriteria["WholeNumber"] = 1] = "WholeNumber";
    /**
     * Only numbers are allowed.
     * When comparing against constraints, the cell value is used.
     */
    DataValidationCriteria[DataValidationCriteria["Decimal"] = 2] = "Decimal";
    /**
     * Dates with or without time portions are allowed as well as numbers equivalent to valid dates.
     * When comparing against constraints, the cell value is used.
     */
    DataValidationCriteria[DataValidationCriteria["Date"] = 4] = "Date";
    /**
     * Times are allowed are well as numbers equivalent to valid times without a date portion.
     * When comparing against constraints, the cell value is used.
     */
    DataValidationCriteria[DataValidationCriteria["Time"] = 5] = "Time";
    /**
     * Non-error values are allowed.
     * When comparing against constraints, the length of the cell value's text equivalent is used.
     */
    DataValidationCriteria[DataValidationCriteria["TextLength"] = 6] = "TextLength";
    return DataValidationCriteria;
})({});
/**
 * @hidden
 */
export let DataValidationCriteria_$type = /*@__PURE__*/ markEnum('DataValidationCriteria', 'WholeNumber,1|Decimal,2|Date,4|Time,5|TextLength,6');
