// igniteui-react-excel/esm2015/lib/DataLabelPosition.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which define the location at which a [[DataLabel]] is positioned.
 * <p class="body">
 * Note that which constants in this enumeration are applicable is dependent on the chart type.
 * </p>
 * <p class="body">
 * The following table lists the supported constants for each chart type.
 * </p>
 * <p class="body">
 * <table border="1">
 * <tr>
 * <th><b>Chart Type</b></th><th><b>Supported Constant(s)</b></th>
 * </tr>
 * <tr>
 * <td>3D charts (except pie/doughnut)</td><td>Custom</td>
 * </tr>
 * <tr>
 * <td>Area, Radar</td><td>Custom</td>
 * </tr>
 * <tr>
 * <td>Surface</td><td>Data labels are not supported</td>
 * </tr>
 * <tr>
 * <td>Bar/Column (stacked)</td><td>Center, InsideEnd, InsideBase</td>
 * </tr>
 * <tr>
 * <td>Bar/Column (clustered)</td><td>Center, InsideEnd, InsideBase, OutsideEnd</td>
 * </tr>
 * <tr>
 * <td>Line/Scatter/Bubble</td><td>Center, Left, Right, Above, Below</td>
 * </tr>
 * <tr>
 * <td>Pie/Doughnut</td><td>Center, InsideEnd, OutsideEnd, BestFit</td>
 * </tr>
 * </table>
 * </p>
 * @see [[DataLabel.labelPosition]]
 * @see [[Series.dataLabels]]
 * @see [[DataLabel]]
 */
export var DataLabelPosition = /*@__PURE__*/ (function (DataLabelPosition) {
    /**
     * The label position is not specified, and is determined automatically by Excel.
     */
    DataLabelPosition[DataLabelPosition["Default"] = 0] = "Default";
    /**
     * The label is centered with respect to the associated data point.
     */
    DataLabelPosition[DataLabelPosition["Center"] = 1] = "Center";
    /**
     * The label's left edge coincides with the left edge of the series bar/line.
     */
    DataLabelPosition[DataLabelPosition["InsideBase"] = 2] = "InsideBase";
    /**
     * The label's right edge coincides with the right edge of the series bar/line.
     */
    DataLabelPosition[DataLabelPosition["InsideEnd"] = 3] = "InsideEnd";
    /**
     * The label's left edge coincides with the right edge of the series bar/line.
     */
    DataLabelPosition[DataLabelPosition["OutsideEnd"] = 4] = "OutsideEnd";
    /**
     * The label's position is explicitly defined via the [[ChartTextAreaBase.left]]
     * and [[ChartTextAreaBase.top]] properties.
     */
    DataLabelPosition[DataLabelPosition["Custom"] = 5] = "Custom";
    /**
     * The label is positioned to the left of the associated data point.
     */
    DataLabelPosition[DataLabelPosition["Left"] = 6] = "Left";
    /**
     * The label is positioned to the right of the associated data point.
     */
    DataLabelPosition[DataLabelPosition["Right"] = 7] = "Right";
    /**
     * The label is positioned above the associated data point.
     */
    DataLabelPosition[DataLabelPosition["Above"] = 8] = "Above";
    /**
     * The label is positioned below the associated data point.
     */
    DataLabelPosition[DataLabelPosition["Below"] = 9] = "Below";
    /**
     * The label is positioned where it best fits with relation to the associated pie slice.
     */
    DataLabelPosition[DataLabelPosition["BestFit"] = 10] = "BestFit";
    return DataLabelPosition;
})({});
/**
 * @hidden
 */
export let DataLabelPosition_$type = /*@__PURE__*/ markEnum('DataLabelPosition', 'Default,0|Center,1|InsideBase,2|InsideEnd,3|OutsideEnd,4|Custom,5|Left,6|Right,7|Above,8|Below,9|BestFit,10');
