// igniteui-react-excel/esm2015/lib/DataValidationImeMode.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the description of the Japanese input rules.
 */
export var DataValidationImeMode = /*@__PURE__*/ (function (DataValidationImeMode) {
    /**
     * The ime mode is not controlled.
     */
    DataValidationImeMode[DataValidationImeMode["NoControl"] = 0] = "NoControl";
    /**
     * The ime mode is on.
     */
    DataValidationImeMode[DataValidationImeMode["On"] = 1] = "On";
    /**
     * The ime mode is off.
     */
    DataValidationImeMode[DataValidationImeMode["Off"] = 2] = "Off";
    /**
     * The ime mode is disabled.
     */
    DataValidationImeMode[DataValidationImeMode["Disabled"] = 3] = "Disabled";
    /**
     * The ime mode is set to Hiragana.
     */
    DataValidationImeMode[DataValidationImeMode["Hiragana"] = 4] = "Hiragana";
    /**
     * The ime mode is set to Katakana
     */
    DataValidationImeMode[DataValidationImeMode["FullKatakana"] = 5] = "FullKatakana";
    /**
     * The ime mode is set to half-width Katakana.
     */
    DataValidationImeMode[DataValidationImeMode["HalfKatakana"] = 6] = "HalfKatakana";
    /**
     * The ime mode is set to full-width alphanumeric.
     */
    DataValidationImeMode[DataValidationImeMode["FullAlpha"] = 7] = "FullAlpha";
    /**
     * The ime mode is set to half-width alphanumeric.
     */
    DataValidationImeMode[DataValidationImeMode["HalfAlpha"] = 8] = "HalfAlpha";
    /**
     * The ime mode is set to full-width Hangul.
     */
    DataValidationImeMode[DataValidationImeMode["FullHangul"] = 9] = "FullHangul";
    /**
     * The ime mode is set to half-width Hangul.
     */
    DataValidationImeMode[DataValidationImeMode["HalfHangul"] = 10] = "HalfHangul";
    return DataValidationImeMode;
})({});
/**
 * @hidden
 */
export let DataValidationImeMode_$type = /*@__PURE__*/ markEnum('DataValidationImeMode', 'NoControl,0|On,1|Off,2|Disabled,3|Hiragana,4|FullKatakana,5|HalfKatakana,6|FullAlpha,7|HalfAlpha,8|FullHangul,9|HalfHangul,10');
