// igniteui-react-excel/esm2015/lib/DiagonalBorders.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the diagonal borders which can be displayed in cells.
 */
export var DiagonalBorders = /*@__PURE__*/ (function (DiagonalBorders) {
    /**
     * The default value for the diagonal borders.
     */
    DiagonalBorders[DiagonalBorders["Default"] = 0] = "Default";
    /**
     * No diagonal borders will be displayed in the cell.
     */
    DiagonalBorders[DiagonalBorders["None"] = 1] = "None";
    /**
     * A diagonal border going from the top-left to bottom-right corner will be displayed in the cell.
     */
    DiagonalBorders[DiagonalBorders["DiagonalDown"] = 3] = "DiagonalDown";
    /**
     * A diagonal border going from the bottom-left to top-right corner will be displayed in the cell.
     */
    DiagonalBorders[DiagonalBorders["DiagonalUp"] = 5] = "DiagonalUp";
    /**
     * Both diagonal borders will be displayed in the cell.
     */
    DiagonalBorders[DiagonalBorders["All"] = 7] = "All";
    return DiagonalBorders;
})({});
/**
 * @hidden
 */
export let DiagonalBorders_$type = /*@__PURE__*/ markEnum('DiagonalBorders', 'Default,0|None,1|DiagonalDown,3|DiagonalUp,5|All,7');
