// igniteui-react-excel/esm2015/lib/DocumentsCoreLocalePt.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
export class DocumentsCoreLocalePt {
    constructor() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Valor {0} inválido. Deve estar entre {1} e {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "O documento está encriptado e deve ser aberto com uma palavra-passe.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "O documento está encriptado com um algoritmo de encriptação não suportado e não pode ser desencriptado.";
        this["LE_FormatException_TypeError"] = "Formato {0} incorreto: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "A palavra-passe utilizada para abrir o documento encriptado está incorreta.";
    }
}
