// igniteui-react-excel/esm2015/lib/DocumentsCoreLocalePl.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
export class DocumentsCoreLocalePl {
    constructor() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Nieprawidłowa wartość {0}. Musi należeć do przedziału od {1} do {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Dokument jest zaszyfrowany i należy go otworzyć hasłem.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Dokument jest zaszyfrowany nieobsługiwanym algorytmem szyfrowania i nie można go odszyfrować.";
        this["LE_FormatException_TypeError"] = "Nieprawidłowy format {0}: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "Hasło użyte do otwarcia zaszyfrowanego dokumentu jest nieprawidłowe.";
    }
}
