// igniteui-react-excel/esm2015/lib/DocumentsCoreLocaleJa.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
export class DocumentsCoreLocaleJa {
    constructor() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "無効な {0} 値です。値は {1} と {2} の間である必要があります。";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "文書は暗号化されているため、パスワードで開く必要があります。";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "文書はサポートされていない暗号化アルゴリズムで暗号化されているため、暗号化の解除を実行できません。";
        this["LE_FormatException_TypeError"] = "無効な {0} 形式: {0}。";
        this["LE_InvalidPasswordException_DefaultMessage"] = "暗号化された文書を開くためのパスワードが無効です。";
    }
}
