// igniteui-react-excel/esm2015/lib/DocumentsCoreLocaleIt.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
export class DocumentsCoreLocaleIt {
    constructor() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Valore {0} non valido. Deve essere compreso tra {1} e {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Il documento è crittografato e deve essere aperto con una password.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Il documento è crittografato con un algoritmo di crittografia non supportato e non può essere decrittografato.";
        this["LE_FormatException_TypeError"] = "Formato {0} errato: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "La password utilizzata per aprire il documento crittografato non è corretta.";
    }
}
