// igniteui-react-excel/esm2015/lib/DocumentsCoreLocaleFr.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
export class DocumentsCoreLocaleFr {
    constructor() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Valeur {0} non valide.  Elle doit être comprise entre {1} et {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Le document est crypté et doit être ouvert avec un mot de passe.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Le document est crypté avec un algorithme de cryptage non pris en charge et ne peut pas être déchiffré.";
        this["LE_FormatException_TypeError"] = "Format {0} incorrect:  {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "Le mot de passe utilisé pour ouvrir le document crypté est incorrect.";
    }
}
