// igniteui-react-excel/esm2015/lib/DocumentsCoreLocaleNb.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
export class DocumentsCoreLocaleNb {
    constructor() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Ugyldig {0}-verdi. Den må være mellom {1} og {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Dokumentet er kryptert og må åpnes med et passord.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Dokumentet er kryptert med en krypteringsalgoritme som ikke støttes og kan ikke dekrypteres.";
        this["LE_FormatException_TypeError"] = "Feil {0} format: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "Passordet som brukes til å åpne det krypterte dokumentet er feil.";
    }
}
