// igniteui-react-excel/esm2015/lib/DocumentsCoreLocaleCs.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
export class DocumentsCoreLocaleCs {
    constructor() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Neplatná hodnota {0}.  Musí být mezi {1} a {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Dokument je šifrovaný a musí být otevřen pomocí hesla.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Dokument je šifrován nepodporovaným šifrovacím algoritmem a nelze jej dešifrovat.";
        this["LE_FormatException_TypeError"] = "Nesprávný formát {0}: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "Heslo použité k otevření šifrovaného dokumentu je nesprávné.";
    }
}
