// igniteui-react-excel/esm2015/lib/DownRevPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let DownRevPartManager = /*@__PURE__*/ (() => {
    class DownRevPartManager extends XmlPartManagerBase {
        get_m() {
            return DownRevPartManager.a3;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            return "/drs/downrev.xml";
        }
        o(a) {
            switch (a) {
                case 0: return DownRevPartManager.a5;
                case 1: return DownRevPartManager.a5;
                default: return DownRevPartManager.a5;
            }
        }
        get_g() {
            return false;
        }
        get g() {
            return this.get_g();
        }
    }
    DownRevPartManager.$t = /*@__PURE__*/ markType(DownRevPartManager, 'DownRevPartManager', XmlPartManagerBase.$);
    DownRevPartManager.a3 = "application/vnd.ms-office.DrsDownRev+xml";
    DownRevPartManager.a5 = "http://schemas.microsoft.com/office/2006/relationships/downRev";
    return DownRevPartManager;
})();
