// igniteui-react-excel/esm2015/lib/DrawingPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let DrawingPartManager = /*@__PURE__*/ (() => {
    class DrawingPartManager extends XmlPartManagerBase {
        get_m() {
            return DrawingPartManager.a4;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/drawings/drawing.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return DrawingPartManager.a6;
                case 1: return DrawingPartManager.a5;
                default: return DrawingPartManager.a6;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    DrawingPartManager.$t = /*@__PURE__*/ markType(DrawingPartManager, 'DrawingPartManager', XmlPartManagerBase.$);
    DrawingPartManager.a4 = "application/vnd.openxmlformats-officedocument.drawing+xml";
    DrawingPartManager.a6 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/drawing";
    DrawingPartManager.a5 = "http://purl.oclc.org/ooxml/officeDocument/relationships/drawing";
    return DrawingPartManager;
})();
