// igniteui-react-excel/esm2015/lib/EG_Anchor_ChartDrawing.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType } from "igniteui-react-core";
import { CT_RelSizeAnchor } from "./CT_RelSizeAnchor";
import { CT_AbsSizeAnchor } from "./CT_AbsSizeAnchor";
import { NotImplementedException } from "igniteui-react-core";
/**
 * @hidden
 */
export let EG_Anchor_ChartDrawing = /*@__PURE__*/ (() => {
    class EG_Anchor_ChartDrawing extends Base {
        constructor() {
            super();
            this.b = null;
            this.a = null;
        }
        d(a, b, c, d) {
            throw new NotImplementedException(0);
        }
        e(a, b) {
            switch (a) {
                case 1344:
                    this.b = typeCast(CT_RelSizeAnchor.$, b);
                    break;
                case 1:
                    this.a = typeCast(CT_AbsSizeAnchor.$, b);
                    break;
                default: break;
            }
        }
        c(a) {
            switch (a) {
                case 1344:
                case 1: return true;
            }
            return false;
        }
    }
    EG_Anchor_ChartDrawing.$t = /*@__PURE__*/ markType(EG_Anchor_ChartDrawing, 'EG_Anchor_ChartDrawing');
    return EG_Anchor_ChartDrawing;
})();
