// igniteui-react-excel/esm2015/lib/EG_Anchor_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, Type, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let EG_Anchor = /*@__PURE__*/ (() => {
    class EG_Anchor extends Base {
        constructor() {
            super(...arguments);
            this.a = null;
        }
    }
    EG_Anchor.$t = /*@__PURE__*/ markType(EG_Anchor, 'EG_Anchor');
    return EG_Anchor;
})();
/**
 * @hidden
 */
export let CT_AbsoluteAnchor = /*@__PURE__*/ (() => {
    class CT_AbsoluteAnchor extends EG_Anchor {
        constructor() {
            super(...arguments);
            this.c = null;
            this.d = null;
        }
        b($tArg, $tResult, a, b = Type.getDefaultValue($tArg)) {
            return a._visitCT_AbsoluteAnchor$i(this, b);
        }
    }
    CT_AbsoluteAnchor.$t = /*@__PURE__*/ markType(CT_AbsoluteAnchor, 'CT_AbsoluteAnchor', EG_Anchor.$);
    return CT_AbsoluteAnchor;
})();
/**
 * @hidden
 */
export let CT_OneCellAnchor = /*@__PURE__*/ (() => {
    class CT_OneCellAnchor extends EG_Anchor {
        constructor() {
            super(...arguments);
            this.d = null;
            this.c = null;
        }
        b($tArg, $tResult, a, b = Type.getDefaultValue($tArg)) {
            return a._visitCT_OneCellAnchor$i(this, b);
        }
    }
    CT_OneCellAnchor.$t = /*@__PURE__*/ markType(CT_OneCellAnchor, 'CT_OneCellAnchor', EG_Anchor.$);
    return CT_OneCellAnchor;
})();
/**
 * @hidden
 */
export let CT_TwoCellAnchor = /*@__PURE__*/ (() => {
    class CT_TwoCellAnchor extends EG_Anchor {
        constructor() {
            super(...arguments);
            this.e = 0;
            this.c = null;
            this.d = null;
        }
        b($tArg, $tResult, a, b = Type.getDefaultValue($tArg)) {
            return a._visitCT_TwoCellAnchor$i(this, b);
        }
    }
    CT_TwoCellAnchor.$t = /*@__PURE__*/ markType(CT_TwoCellAnchor, 'CT_TwoCellAnchor', EG_Anchor.$);
    return CT_TwoCellAnchor;
})();
/**
 * @hidden
 */
export let IEG_AnchorVisitor$2_$type = /*@__PURE__*/ new Type(null, 'IEG_AnchorVisitor$2');
