// igniteui-react-excel/esm2015/lib/EG_ObjectChoices.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType } from "igniteui-react-core";
import { CT_Shape } from "./CT_Shape";
import { CT_GroupShape } from "./CT_GroupShape";
import { CT_GraphicFrame } from "./CT_GraphicFrame";
import { CT_Connector } from "./CT_Connector";
import { CT_Picture } from "./CT_Picture";
import { NotImplementedException } from "igniteui-react-core";
/**
 * @hidden
 */
export let EG_ObjectChoices = /*@__PURE__*/ (() => {
    class EG_ObjectChoices extends Base {
        constructor() {
            super();
            this.e = null;
            this.c = null;
            this.a = null;
            this.b = null;
            this.d = null;
        }
        g(a, b, c, d) {
            throw new NotImplementedException(0);
        }
        h(a, b) {
            switch (a) {
                case 1563:
                    this.e = typeCast(CT_Shape.$, b);
                    break;
                case 753:
                    this.c = typeCast(CT_GroupShape.$, b);
                    break;
                case 729:
                    this.a = typeCast(CT_GraphicFrame.$, b);
                    break;
                case 396:
                    this.b = typeCast(CT_Connector.$, b);
                    break;
                case 1194:
                    this.d = typeCast(CT_Picture.$, b);
                    break;
                default: break;
            }
        }
        f(a) {
            switch (a) {
                case 1563:
                case 753:
                case 729:
                case 396:
                case 1194: return true;
            }
            return false;
        }
    }
    EG_ObjectChoices.$t = /*@__PURE__*/ markType(EG_ObjectChoices, 'EG_ObjectChoices');
    return EG_ObjectChoices;
})();
