// igniteui-react-excel/esm2015/lib/EG_ThemeableFontStyles.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType } from "igniteui-react-core";
import { CT_FontCollection } from "./CT_FontCollection";
import { CT_FontReference } from "./CT_FontReference";
import { NotImplementedException } from "igniteui-react-core";
/**
 * @hidden
 */
export let EG_ThemeableFontStyles = /*@__PURE__*/ (() => {
    class EG_ThemeableFontStyles extends Base {
        constructor() {
            super(...arguments);
            this.a = null;
            this.b = null;
        }
        d(a, b, c, d) {
            throw new NotImplementedException(0);
        }
        e(a, b) {
            switch (a) {
                case 659:
                    this.a = typeCast(CT_FontCollection.$, b);
                    break;
                case 660:
                    this.b = typeCast(CT_FontReference.$, b);
                    break;
                default: break;
            }
        }
        c(a) {
            switch (a) {
                case 659:
                case 660: return true;
            }
            return false;
        }
    }
    EG_ThemeableFontStyles.$t = /*@__PURE__*/ markType(EG_ThemeableFontStyles, 'EG_ThemeableFontStyles');
    return EG_ThemeableFontStyles;
})();
