// igniteui-react-excel/esm2015/lib/EG_ThemeableEffectStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType } from "igniteui-react-core";
import { CT_EffectProperties } from "./CT_EffectProperties";
import { CT_StyleMatrixReference_DML } from "./CT_StyleMatrixReference_DML";
import { DMLExtensions } from "./DMLExtensions";
import { DrawingMLMain } from "./DrawingMLMain";
/**
 * @hidden
 */
export let EG_ThemeableEffectStyle = /*@__PURE__*/ (() => {
    class EG_ThemeableEffectStyle extends Base {
        constructor() {
            super(...arguments);
            this.a = null;
            this.b = null;
        }
        d(a, b, c, d) {
            if (this.a != null) {
                DMLExtensions.x(this.a, a, b, c, DrawingMLMain.dh);
            }
            if (this.b != null) {
                DMLExtensions.ba(this.b, a, b, c, DrawingMLMain.dk);
            }
        }
        e(a, b) {
            switch (a) {
                case 556:
                    this.a = typeCast(CT_EffectProperties.$, b);
                    break;
                case 561:
                    this.b = typeCast(CT_StyleMatrixReference_DML.$, b);
                    break;
                default: break;
            }
        }
        c(a) {
            switch (a) {
                case 556:
                case 561: return true;
            }
            return false;
        }
    }
    EG_ThemeableEffectStyle.$t = /*@__PURE__*/ markType(EG_ThemeableEffectStyle, 'EG_ThemeableEffectStyle');
    return EG_ThemeableEffectStyle;
})();
