// igniteui-react-excel/esm2015/lib/EG_ThemeableFillStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType } from "igniteui-react-core";
import { CT_FillProperties } from "./CT_FillProperties";
import { CT_StyleMatrixReference_DML } from "./CT_StyleMatrixReference_DML";
import { DMLExtensions } from "./DMLExtensions";
import { DrawingMLMain } from "./DrawingMLMain";
/**
 * @hidden
 */
export let EG_ThemeableFillStyle = /*@__PURE__*/ (() => {
    class EG_ThemeableFillStyle extends Base {
        constructor() {
            super(...arguments);
            this.a = null;
            this.b = null;
        }
        d(a, b, c, d) {
            if (this.a != null) {
                DMLExtensions.ad(this.a, a, b, c, DrawingMLMain.dx);
            }
            if (this.b != null) {
                DMLExtensions.ba(this.b, a, b, c, DrawingMLMain.d0);
            }
        }
        e(a, b) {
            switch (a) {
                case 630:
                    this.a = typeCast(CT_FillProperties.$, b);
                    break;
                case 635:
                    this.b = typeCast(CT_StyleMatrixReference_DML.$, b);
                    break;
                default: break;
            }
        }
        c(a) {
            switch (a) {
                case 630:
                case 635: return true;
            }
            return false;
        }
    }
    EG_ThemeableFillStyle.$t = /*@__PURE__*/ markType(EG_ThemeableFillStyle, 'EG_ThemeableFillStyle');
    return EG_ThemeableFillStyle;
})();
