// igniteui-react-excel/esm2015/lib/ErrorValueType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which specify the units used to depict errors on an
 * <see cref="ErrorBars">error bar</see>.
 */
export var ErrorValueType = /*@__PURE__*/ (function (ErrorValueType) {
    /**
     * The error range is defined by a fixed numerical value.
     */
    ErrorValueType[ErrorValueType["FixedValue"] = 0] = "FixedValue";
    /**
     * The error range is defined by a percentage.
     */
    ErrorValueType[ErrorValueType["Percentage"] = 1] = "Percentage";
    /**
     * The error range is defined by standard deviation.
     */
    ErrorValueType[ErrorValueType["StandardDeviation"] = 2] = "StandardDeviation";
    /**
     * The error range is defined by standard error.
     */
    ErrorValueType[ErrorValueType["StandardError"] = 3] = "StandardError";
    return ErrorValueType;
})({});
/**
 * @hidden
 */
export let ErrorValueType_$type = /*@__PURE__*/ markEnum('ErrorValueType', 'FixedValue,0|Percentage,1|StandardDeviation,2|StandardError,3');
