// igniteui-react-excel/esm2015/lib/EvaluationResult$1.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, Type, toNullable, Number_$type, markStruct } from "igniteui-react-core";
/**
 * @hidden
 */
export let EvaluationResult$1 = /*@__PURE__*/ (() => {
    class EvaluationResult$1 extends ValueType {
        constructor($t, a, ..._rest) {
            super();
            this.$t = null;
            this.a = null;
            this.c = false;
            this.e = toNullable(Number_$type, null);
            this.$t = $t;
            this.$type = this.$type.specialize(this.$t);
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.c = true;
                        this.a = c;
                        this.e = toNullable(Number_$type, null);
                    }
                    break;
                case 2:
                    {
                        let c = _rest[0];
                        this.c = false;
                        this.a = Type.getDefaultValue(this.$t);
                        this.e = c;
                    }
                    break;
            }
        }
        get d() {
            return this.c;
        }
        get b() {
            return this.a;
        }
        get f() {
            return this.e;
        }
    }
    EvaluationResult$1.$t = /*@__PURE__*/ markStruct(EvaluationResult$1, 'EvaluationResult$1');
    return EvaluationResult$1;
})();
