// igniteui-react-excel/esm2015/lib/Excel2006Main.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let Excel2006Main = /*@__PURE__*/ (() => {
    class Excel2006Main extends XmlNamespaceDefinitionBase {
        constructor() {
            super();
        }
        s(a) {
            switch (a) {
                case 0:
                case 1: return Excel2006Main.ar;
                default: return Excel2006Main.ar;
            }
        }
        get_t() {
            return "xm";
        }
        get t() {
            return this.get_t();
        }
        static at(a, b) {
            b = XmlNamespaceDefinitionBase.u(a, Excel2006Main.aq);
            return {
                p1: b
            };
        }
        static av(a, b) {
            XmlNamespaceDefinitionBase.aj(a, Excel2006Main.aq, "xm", b);
        }
        static au(a, b) {
            b = XmlNamespaceDefinitionBase.u(a, Excel2006Main.as);
            return {
                p1: b
            };
        }
        static aw(a, b) {
            XmlNamespaceDefinitionBase.aj(a, Excel2006Main.as, "xm", b);
        }
    }
    Excel2006Main.$t = /*@__PURE__*/ markType(Excel2006Main, 'Excel2006Main', XmlNamespaceDefinitionBase.$);
    Excel2006Main.ar = "http://schemas.microsoft.com/office/excel/2006/main";
    Excel2006Main.aq = Excel2006Main.ar + "/f";
    Excel2006Main.as = Excel2006Main.ar + "/sqref";
    Excel2006Main.ao = /*@__PURE__*/ new Excel2006Main();
    return Excel2006Main;
})();
