// igniteui-react-excel/esm2015/lib/Excel2007ChartData.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let Excel2007ChartData = /*@__PURE__*/ (() => {
    class Excel2007ChartData extends Base {
        constructor() {
            super(...arguments);
            this.a = null;
            this.c = null;
            this.d = null;
            this.e = null;
            this.b = null;
        }
    }
    Excel2007ChartData.$t = /*@__PURE__*/ markType(Excel2007ChartData, 'Excel2007ChartData');
    return Excel2007ChartData;
})();
