// igniteui-react-excel/esm2015/lib/ExcelLocaleRo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
export class ExcelLocaleRo {
    constructor() {
        this["DefaultTotalLabel"] = "Total";
        this["Error_AnchoredReference"] = "Încercați să accesați {0} ca referință neacoperită";
        this["Error_ExplicitThis"] = "{0} - Acest lucru nu este acceptat explicit";
        this["Error_ExternalRange"] = "Intervalul conține referință externă";
        this["Error_FormulaNotSupported"] = "{0} - Încercați să accesați formula pe o referință care nu acceptă formule";
        this["Error_Internal"] = "A apărut o eroare internă în metoda {0}.";
        this["Error_InvalidFunction"] = "Funcția nevalidă '{0}' la {1}.";
        this["Error_InvalidFunctionArgCountTooFew"] = "Funcția '{0}' la care se face referire la {1} nu conține numărul corect de argumente. Funcția necesită minimum {2} parametri.";
        this["Error_InvalidFunctionArgCountTooMany"] = "Funcția '{0}' la care se face referire la {1} nu conține numărul corect de argumente. Numărul argumentelor nu trebuie să depășească {2} parametri.";
        this["Error_InvalidNumberSystemDigit"] = "Cifră nevalidă din '{0}' întâlnită în valoarea '{1}' pentru sistemul numeric cu baza {2}.";
        this["Error_InvalidOperand"] = "Operand nevalid";
        this["Error_InvalidOperatorArgCount"] = "Funcția '{0}' trebuie să accepte {1} argumente.";
        this["Error_InvalidReference"] = "Referință nevalidă sau indisponibilă";
        this["Error_Location"] = "linie {0} col {1}";
        this["Error_MethodNotSupported"] = "{0} nu acceptă {1}.";
        this["Error_NoCurrentRangeElement"] = "Fără element de interval curent";
        this["Error_NullFunctionResult"] = "Evaluarea '{0}' a returnat o valoare nulă.";
        this["Error_NullOperand"] = "Operand nul";
        this["Error_ParseRef"] = "Referință necunoscută la {0}.";
        this["Error_RangeFromLast"] = "din T. Nu a fost setat";
        this["Error_RangeFromNull"] = "RefRange: de la este nul";
        this["Error_RangeFromRefBase"] = "RefRange: de la nu este un RefBase: {0}";
        this["Error_RangeLast"] = "Validare RefRange: ultimele tupluri [{0}] != [{1}]";
        this["Error_RangeNormalizeScope"] = "RefRange Normalize: scop greșit [{0} / {1}]";
        this["Error_RangeToNull"] = "RefRange: la este nul";
        this["Error_RangeToRefBase"] = "RefRange: la nu este un RefBase: {0}";
        this["Error_RangeTuple"] = "Validare RefRange: tupluri [{0}] != [{1}]";
        this["Error_RangeValidateCount"] = "Validare RefRange: de la numărare {0}! = La numărare {1}";
        this["Error_ReadOnlyReference"] = "{0} - Încercați să setați valoarea referinței numai în citire";
        this["Error_ReferenceNotEumerable"] = "{0} nu este o referință enumerabilă";
        this["Error_RefMalformed"] = "{0} - Referință greșită {1}";
        this["Error_RP_ContainsBase"] = "RefParser.Contains - Baza goală";
        this["Error_RP_ContainsTarget"] = "RefParser.Contains - Țintă goală";
        this["Error_RP_EmptyElement"] = "Referință nevalidă {0}. Element gol întâlnit la locația {1}.";
        this["Error_RP_EmptyReference"] = "Referință nevalidă {0}. Referința este goală.";
        this["Error_RP_MalformedExtra"] = "Referință Malformată RefParser: Multiple [{0}] găsite în [{1}]";
        this["Error_RP_MalformedUnexpected"] = "Referință Malformată RefParser: neașteptat [{0}] găsit în [{1}]";
        this["Error_RP_MergeRelativeBase"] = "RefParser.Merge: Baza este relativă";
        this["Error_RP_MissingCloseParenthesis"] = "Referință nevalidă {0}. Lipsește o paranteză închisă.";
        this["Error_RP_MissingEndQuotes"] = "Referință nevalidă {0}. Ghilimele de final lipsă.";
        this["Error_RT_Expected"] = "RefTuple: Așteptat [{0}] în [{1}]";
        this["Error_RT_InvalidScope"] = "Domeniu de aplicare nevalid: {0}.";
        this["Error_RT_InvalidScope_CharactersAfterClosingQuotes"] = "Caracterele întâlnite după închiderea ghilimelelor duble în {0} domeniul de aplicare la locație {1}.";
        this["Error_RT_InvalidTuple_CharactersAfterScopeEnd"] = "Element nevalid {0}. Caracterele întâlnite după sfârșitul domeniului.";
        this["Error_RT_InvalidTuple_MissingCloseParenthesis"] = "Element nevalid {0}. Lipsește o paranteză închisă.";
        this["Error_RT_InvalidTuple_MissingEndQuotes"] = "Element nevalid {0}. Ghilimele de final lipsă.";
        this["Error_RT_InvalidTuple_NamePortionEmpty"] = "Element nevalid {0}. Porțiunea de nume de element este goală.";
        this["Error_RT_InvalidTuple_ScopeIndexLarge"] = "Element nevalid {0}. Indexul {1} este nevalid: {2}.";
        this["Error_RT_InvalidTuple_UnescapedCharacter"] = "{0} caracter neevacuat întâlnit în {1} la locația {2}.";
        this["Error_RT_NullRefName"] = "RefTuple: Nul RefName";
        this["Error_RT_Unexpected"] = "RefTuple: neașteptat [{0}] în [{1}]";
        this["Error_ScopeAllNotLast"] = "{0} - Domeniul de aplicare Toate trebuie să apară în ultimul element {1}";
        this["Error_UCErrorCode_Div"] = "Împarte la zero";
        this["Error_UCErrorCode_Fail"] = "Eșuat";
        this["Error_UCErrorCode_NA"] = "Nu se aplică";
        this["Error_UCErrorCode_Name"] = "Nume nevalid";
        this["Error_UCErrorCode_Null"] = "Valoare nulă";
        this["Error_UCErrorCode_Num"] = "Valori numerice nevalide în funcție sau formulă.";
        this["Error_UCErrorCode_Ok"] = "A reuși";
        this["Error_UCErrorCode_Reference"] = "Referință nevalidă sau indisponibilă";
        this["Error_UCErrorCode_Unknown"] = "Cod invalid";
        this["Error_UCErrorCode_Value"] = "Tipul de argument sau operand incorect.";
        this["Error_UnexpectedScope"] = "{0} - Domeniu de aplicare neașteptat pentru părinte {1}";
        this["Error_UnknownRange"] = "Tipul de interval necunoscut";
        this["Error_UnknownRefType"] = "{0} - RefType necunoscut {1}";
        this["Error_UnknownScope"] = "{0} - Domeniu de aplicare necunoscut {1}";
        this["Error_UnresolvedExternal"] = "Referință externă nerezolvată";
        this["Func_abs_Arg_0"] = "număr";
        this["Func_abs_ArgDesc_0"] = "Numărul real din care se obține valoarea absolută.";
        this["Func_abs_Category"] = "Matematica";
        this["Func_abs_CategoryURL"] = "Matematica";
        this["Func_abs_Desc"] = "Returnează valoarea absolută a unui număr. Valoarea absolută a unui număr este numărul fără semnul său.";
        this["Func_acos_Arg_0"] = "număr";
        this["Func_acos_ArgDesc_0"] = "Cosinusul unghiului de obținut, care trebuie să fie de la -1 la 1.";
        this["Func_acos_Category"] = "Matematica";
        this["Func_acos_CategoryURL"] = "Matematica";
        this["Func_acos_Desc"] = "Returnează arc cosinus sau cosinusul invers al unui număr. Arc cosinus este unghiul al cărui cosinus este numărul specificat. Unghiul returnat este dat în radiani în intervalul zero până la pi. Pentru a converti rezultatul din radiani în grade, înmulțiți cu 180 / PI ().";
        this["Func_acosh_Arg_0"] = "număr";
        this["Func_acosh_ArgDesc_0"] = "Un număr real mai mare de 1 pentru care să se găsească cosinusul hiperbolic invers.";
        this["Func_acosh_Category"] = "Matematica";
        this["Func_acosh_CategoryURL"] = "Matematica";
        this["Func_acosh_Desc"] = "Returnează cosinusul hiperbolic invers al numărului dat. Cosinusul hiperbolic invers este unghiul al cărui cosinus hiperbolic este numărul specificat.";
        this["Func_and_Arg_0"] = "{0} logic";
        this["Func_and_ArgDesc_0"] = "O valoare logică (adevărată sau falsă).";
        this["Func_and_Category"] = "Logic";
        this["Func_and_CategoryURL"] = "Logic";
        this["Func_and_Desc"] = "Returnează adevărat dacă toate valorile sunt adevărate sau returnează fals dacă una sau mai multe valori sunt false.";
        this["Func_asin_Arg_0"] = "număr";
        this["Func_asin_ArgDesc_0"] = "Sinusul unghiului de obținut, care trebuie să fie de la -1 la 1 radiani.";
        this["Func_asin_Category"] = "Matematica";
        this["Func_asin_CategoryURL"] = "Matematica";
        this["Func_asin_Desc"] = "Returnează arc sinusul, sau sinusul invers, al unui număr. Arc sinusul este unghiul al cărui sinus este numărul specificat. Unghiul returnat este dat în radiani în intervalul -pi / 2 până la pi / 2. Pentru a exprima arcul în grade, înmulțiți rezultatul cu 180 / PI ().";
        this["Func_asinh_Arg_0"] = "număr";
        this["Func_asinh_ArgDesc_0"] = "Un număr real pentru care să se găsească sinusul hiperbolic invers.";
        this["Func_asinh_Category"] = "Matematica";
        this["Func_asinh_CategoryURL"] = "Matematica";
        this["Func_asinh_Desc"] = "Returnează sinusul hiperbolic invers al numărului dat. Sinusul hiperbolic invers este unghiul al cărui sinus hiperbolic este numărul specificat.";
        this["Func_atan_Arg_0"] = "număr";
        this["Func_atan_ArgDesc_0"] = "Tangenta unghiului de obținut.";
        this["Func_atan_Category"] = "Matematica";
        this["Func_atan_CategoryURL"] = "Matematica";
        this["Func_atan_Desc"] = "Returnează arc tangenta sau tangenta inversă a unui număr. Arc tangenta este unghiul a cărui tangentă este numărul specificat. Unghiul returnat este dat în radiani în intervalul -pi / 2 până la pi / 2. Pentru a exprima arc tangenta în grade, înmulțiți rezultatul cu 180 / PI ().";
        this["Func_atan2_Arg_0"] = "X";
        this["Func_atan2_Arg_1"] = "y";
        this["Func_atan2_ArgDesc_0"] = "Coordonata x a punctului.";
        this["Func_atan2_ArgDesc_1"] = "Coordonata y a punctului.";
        this["Func_atan2_Category"] = "Matematica";
        this["Func_atan2_CategoryURL"] = "Matematica";
        this["Func_atan2_Desc"] = "Returnează arc tangenta sau tangenta inversă a coordonatelor x și y specificate. Arc tangenta este unghiul de la axa x la o linie care conține originea (0, 0) și un punct cu coordonate (x_num, y_num). Unghiul este dat în radiani între -pi și pi, excluzând -pi.";
        this["Func_atanh_Arg_0"] = "număr";
        this["Func_atanh_ArgDesc_0"] = "Un număr real între -1 și 1 pentru care se obține tangenta hiperbolică inversă.";
        this["Func_atanh_Category"] = "Matematica";
        this["Func_atanh_CategoryURL"] = "Matematica";
        this["Func_atanh_Desc"] = "Returnează tangenta hiperbolică inversă a numărului dat. Tangenta hiperbolică inversă este unghiul a cărui tangentă hiperbolică este numărul specificat.";
        this["Func_avedev_Arg_0"] = "numărul {0}";
        this["Func_avedev_ArgDesc_0"] = "Numărul1, numărul2, ... sunt una sau mai multe valori numerice sau referințe la valori numerice.";
        this["Func_avedev_Category"] = "Statistic";
        this["Func_avedev_CategoryURL"] = "Statistic";
        this["Func_avedev_Desc"] = "Returnează deviația medie față de media tuturor valorilor numerice.";
        this["Func_average_Arg_0"] = "numărul {0}";
        this["Func_average_ArgDesc_0"] = "Un număr sau o gamă de numere pentru care să se găsească media cu celelalte argumente.";
        this["Func_average_Category"] = "Statistic";
        this["Func_average_CategoryURL"] = "Statistic";
        this["Func_average_Desc"] = "Returnează media (media aritmetică) a uneia sau mai multor valori.";
        this["Func_bin2dec_Arg_0"] = "Număr";
        this["Func_bin2dec_ArgDesc_0"] = "O expresie text care reprezintă un număr binar.";
        this["Func_bin2dec_Category"] = "Inginerie";
        this["Func_bin2dec_CategoryURL"] = "Inginerie";
        this["Func_bin2dec_Desc"] = "Returnează valoarea zecimală a unei reprezentări text a unui număr în sistemul numeric binar.";
        this["Func_bin2hex_Arg_0"] = "Număr";
        this["Func_bin2hex_Arg_1"] = "Locuri";
        this["Func_bin2hex_ArgDesc_0"] = "O expresie text care reprezintă un număr binar.";
        this["Func_bin2hex_ArgDesc_1"] = "Numărul de caractere de utilizat pentru reprezentarea valorii. Dacă acest lucru nu este specificat, va fi utilizat numărul minim de caractere necesar pentru a reprezenta valoarea. Acest lucru vă permite să specificați zerouri de la început.";
        this["Func_bin2hex_Category"] = "Inginerie";
        this["Func_bin2hex_CategoryURL"] = "Inginerie";
        this["Func_bin2hex_Desc"] = "Convertește o expresie text care reprezintă un număr binar într-o expresie text care reprezintă același număr în sistemul numeric hexazecimal.";
        this["Func_bin2oct_Arg_0"] = "Număr";
        this["Func_bin2oct_Arg_1"] = "Locuri";
        this["Func_bin2oct_ArgDesc_0"] = "O expresie text care reprezintă un număr binar.";
        this["Func_bin2oct_ArgDesc_1"] = "Numărul de caractere de utilizat pentru reprezentarea valorii. Dacă acest lucru nu este specificat, va fi utilizat numărul minim de caractere necesar pentru a reprezenta valoarea. Acest lucru vă permite să specificați zerouri de la început.";
        this["Func_bin2oct_Category"] = "Inginerie";
        this["Func_bin2oct_CategoryURL"] = "Inginerie";
        this["Func_bin2oct_Desc"] = "Convertește o expresie text care reprezintă un număr binar într-o expresie text care reprezintă același număr în sistemul numeric octal.";
        this["Func_binomdist_Arg_0"] = "Număr";
        this["Func_binomdist_Arg_1"] = "Încercări";
        this["Func_binomdist_Arg_2"] = "Probabilitate";
        this["Func_binomdist_Arg_3"] = "Cumulativ";
        this["Func_binomdist_ArgDesc_0"] = "Numărul de succese în încercări.";
        this["Func_binomdist_ArgDesc_1"] = "Numărul de studii independente.";
        this["Func_binomdist_ArgDesc_2"] = "Probabilitatea de succes pe fiecare proces.";
        this["Func_binomdist_ArgDesc_3"] = "O valoare logică care determină forma funcției. Dacă este adevărat, funcția va returna distribuția cumulativă, care este probabilitatea ca există cel mult succesul \"numărului\", altfel va returna masa probabilității, care este probabilitatea că există succesele \"numărului\".";
        this["Func_binomdist_Category"] = "Statistic";
        this["Func_binomdist_CategoryURL"] = "Statistic";
        this["Func_binomdist_Desc"] = "Returnează numărul de combinații posibile (cu repetări) dat unui set de articole și un număr de articole alese din acel set.";
        this["Func_CategoryURL_Template"] = "ms-help: // INFRAGISTICS_HELP / NetAdvantagev {2} {3} / Win / WinElements / WinCalcManager_ {0} _Functions.html # FUNC_ {1}";
        this["Func_ceiling_Arg_0"] = "număr";
        this["Func_ceiling_Arg_1"] = "semnificaţie";
        this["Func_ceiling_ArgDesc_0"] = "Numărul de rotunjit.";
        this["Func_ceiling_ArgDesc_1"] = "Multiplu la care se rotunjește. Dacă este omis, se presupune că acesta este unul.";
        this["Func_ceiling_Category"] = "Matematica";
        this["Func_ceiling_CategoryURL"] = "Matematica";
        this["Func_ceiling_Desc"] = "Returnează cel mai mic multiplu al semnificației (implicit la 1) mai mare sau egal cu numărul specificat.";
        this["Func_char_Arg_0"] = "charCode";
        this["Func_char_ArgDesc_0"] = "Numărul care trebuie convertit la un caracter.";
        this["Func_char_Category"] = "TextAndData";
        this["Func_char_CategoryURL"] = "Text_and_Data";
        this["Func_char_Desc"] = "Returnează caracterul specificat de numărul de cod din setul de caractere pentru computer.";
        this["Func_choose_Arg_0"] = "IndexNum";
        this["Func_choose_Arg_1"] = "Value{0}";
        this["Func_choose_ArgDesc_0"] = "Indexul bazat pe 1 în restul argumentelor pentru valoarea de utilizat.";
        this["Func_choose_ArgDesc_1"] = "O valoare care poate fi aleasă în funcție de valoarea IndexNum.";
        this["Func_choose_Category"] = "LookupAndReference";
        this["Func_choose_CategoryURL"] = "Lookup_And_Reference";
        this["Func_choose_Desc"] = "Returnează una dintre valorile furnizate în argumentele sale pe baza numărului din primul argument.";
        this["Func_code_Arg_0"] = "text";
        this["Func_code_ArgDesc_0"] = "Expresia textului care va avea primul său caracter evaluat.";
        this["Func_code_Category"] = "TextAndData";
        this["Func_code_CategoryURL"] = "Text_and_Data";
        this["Func_code_Desc"] = "Returnează un cod numeric pentru primul caracter dintr-o expresie text utilizând setul de caractere utilizat de computer.";
        this["Func_combin_Arg_0"] = "Număr";
        this["Func_combin_Arg_1"] = "NumberChosen";
        this["Func_combin_ArgDesc_0"] = "Număr de elemente.";
        this["Func_combin_ArgDesc_1"] = "Numărul de elemente alese în fiecare combinație.";
        this["Func_combin_Category"] = "Matematica";
        this["Func_combin_CategoryURL"] = "Matematica";
        this["Func_combin_Desc"] = "Returnează numărul de combinații posibile, dat fiind un set de articole și un număr de articole alese din acel set.";
        this["Func_combina_Arg_0"] = "Număr";
        this["Func_combina_Arg_1"] = "NumberChosen";
        this["Func_combina_ArgDesc_0"] = "Număr de elemente.";
        this["Func_combina_ArgDesc_1"] = "Numărul de elemente alese în fiecare combinație.";
        this["Func_combina_Category"] = "Matematica";
        this["Func_combina_CategoryURL"] = "Matematica";
        this["Func_combina_Desc"] = "Returnează numărul de combinații posibile (cu repetări) dat unui set de articole și un număr de articole alese din acel set.";
        this["Func_complex_Arg_0"] = "RealNumber";
        this["Func_complex_Arg_1"] = "ImaginaryNumber";
        this["Func_complex_Arg_2"] = "Sufix";
        this["Func_complex_ArgDesc_0"] = "Coeficientul real al numărului complex.";
        this["Func_complex_ArgDesc_1"] = "Coeficientul imaginar al numărului.";
        this["Func_complex_ArgDesc_2"] = "Specifică sufixul de utilizat. Valorile acceptabile sunt \"i\" sau \"j\". Dacă este omis, se utilizează \"i\". Rețineți că sufixul este sensibil la majuscule; majusculele \"I\" și \"J\" nu sunt permise și vor avea ca rezultat o eroare #VALUE.";
        this["Func_complex_Category"] = "Inginerie";
        this["Func_complex_CategoryURL"] = "Inginerie";
        this["Func_complex_Desc"] = "Returnează un număr complex reprezentat ca o expresie text în formatul \"x + yi\" sau \"x + yj\" prin combinarea unui număr real și imaginar.";
        this["Func_concatenate_Arg_0"] = "text {0}";
        this["Func_concatenate_ArgDesc_0"] = "O expresie de text sau o gamă de expresii de text care trebuie alăturate într-o singură expresie de text.";
        this["Func_concatenate_Category"] = "TextAndData";
        this["Func_concatenate_CategoryURL"] = "Text_and_Data";
        this["Func_concatenate_Desc"] = "Alătură toate argumentele într-o expresie de text.";
        this["Func_convert_Arg_0"] = "Număr";
        this["Func_convert_Arg_1"] = "FromUnit";
        this["Func_convert_Arg_2"] = "ToUnit";
        this["Func_convert_ArgDesc_0"] = "Valoarea de convertit.";
        this["Func_convert_ArgDesc_1"] = "Unitatea în care este dat numărul.\r\n\r\nUnități de greutate și masă: \" g \" = gram; \" sg \" = slug; \" lbm \" = masa lirei (avoirdupois); \" u \" = U (unitate de masă atomică); \" ozm \" = Masa de uncie (avoirdupois)\r\nUnități de distanță: \" m \" = Contor; \" mi \" = Statute mile; \" Nmi \" = Nautical mile; \" in \" = Inch; \" ft \" = Picior; \" yd \" = Yard; \" ang \" = Angstrom; \" Pica \" = Pica (1/72 in.)\r\nUnități de timp: \" yr \" = An; \" day \" = Ziua; \" hr \" = Ora; \" mn \" = Minut; \" sec \" = Second\r\nUnități de presiune: \" Pa \" (sau \" p \") = Pascal; \" atm \" (sau \" la \") = Atmosfera; \" mmHg \" = mm de mercur\r\nUnități de forță: \" N \" = Newton; \" dyn \" (sau \" dy \") = Dyne; \" lbf \" = Forța lirei\r\nUnități de energie: \" J \" = Joule; \" e \" = Erg; \" c \" = Calorie termodinamică; \" cal \" = calorie IT; \" eV \" (sau \" ev \") = Electron volt; \" HPh \" (sau \" hh \") = Horsepower-hour; \" Wh \" (sau \" wh \") = Watt-oră; \" flb \" = Picior-kilogram; \" BTU \" (sau \" btu \") = BTU\r\nUnități de putere: \" HP \" (sau \" h \") = Putere; \" W \" (sau \" w \") = Watt\r\nUnități de magnetism: \" T \" = Tesla; \" ga \" = Gauss\r\nUnități de temperatură: \" C \" (sau \" cel \") = Grad Celsius; \" F \" (sau \" fah \") = Grad Fahrenheit; \" K \" (sau \" kel \") = Kelvin\r\nUnități de măsură lichide: \" linguriță \" = linguriță; \" tbs \" = lingură; \" oz \" = uncie fluidă; \" cup \" = Cupă; \" pt \" (sau \" us_pt \") = US pint; \" uk_pt \" = UK pint; \" qt \" = Quart; \" gal \" = Gallon; \" l \" (sau \" lt \") = Litru\r\n\r\nOrice unitate metrică poate fi prefixată cu una dintre următoarele: exa (\" E \") = 1E + 18; peta (\" P \") = 1E + 15; tera (\" T \") = 1E + 12; giga (\" G \") = 1E + 09; mega (\" M \") = 1E + 06; kilo (\" k \") = 1E + 03; hecto (\" h \") = 1E + 02; dekao (\" e \") = 1E + 01; deci (\" d \") = 1E-01; centi (\" c \") = 1E-02; milli (\" m \") = 1E-03; micro (\" u \") = 1E-06; nano (\" n \") = 1E-09; pico (\" p \") = 1E-12; femto (\" f \") = 1E-15; atto (\" a \") = 1E-18\r\n    ";
        this["Func_convert_ArgDesc_2"] = "Unitățile în care trebuie returnată valoarea convertită.\r\n\r\nUnități de greutate și masă: \" g \" = gram; \" sg \" = slug; \" lbm \" = masa lirei (avoirdupois); \" u \" = U (unitate de masă atomică); \" ozm \" = Masa de uncie (avoirdupois)\r\nUnități de distanță: \" m \" = Contor; \" mi \" = Statute mile; \" Nmi \" = Nautical mile; \" in \" = Inch; \" ft \" = Picior; \" yd \" = Yard; \" ang \" = Angstrom; \" Pica \" = Pica (1/72 in.)\r\nUnități de timp: \" yr \" = An; \" day \" = Ziua; \" hr \" = Ora; \" mn \" = Minut; \" sec \" = Secundă\r\nUnități de presiune: \" Pa \" (sau \" p \") = Pascal; \" atm \" (sau \" la \") = Atmosfera; \" mmHg \" = mm de mercur\r\nUnități de forță: \" N \" = Newton; \" dyn \" (sau \" dy \") = Dyne; \" lbf \" = Forța lirei\r\nUnități de energie: \" J \" = Joule; \" e \" = Erg; \" c \" = Calorie termodinamică; \" cal \" = calorie IT; \" eV \" (sau \" ev \") = Electron volt; \" HPh \" (sau \" hh \") = Horsepower-hour; \" Wh \" (sau \" wh \") = Watt-oră; \" flb \" = Picior-kilogram; \" BTU \" (sau \" btu \") = BTU\r\nUnități de putere: \" HP \" (sau \" h \") = Putere; \" W \" (sau \" w \") = Watt\r\nUnități de magnetism: \" T \" = Tesla; \" ga \" = Gauss\r\nUnități de temperatură: \" C \" (sau \" cel \") = Grad Celsius; \" F \" (sau \" fah \") = Grad Fahrenheit; \" K \" (sau \" kel \") = Kelvin\r\nUnități de măsură lichide: \" linguriță \" = linguriță; \" tbs \" = lingură; \" oz \" = uncie fluidă; \" cup \" = Cupă; \" pt \" (sau \" us_pt \") = US pint; \" uk_pt \" = UK pint; \" qt \" = Quart; \" gal \" = Gallon; \" l \" (sau \" lt \") = Litru\r\n\r\nOrice unitate metrică poate fi prefixată cu una dintre următoarele: exa (\" E \") = 1E + 18; peta (\" P \") = 1E + 15; tera (\" T \") = 1E + 12; giga (\" G \") = 1E + 09; mega (\" M \") = 1E + 06; kilo (\" k \") = 1E + 03; hecto (\" h \") = 1E + 02; dekao (\" e \") = 1E + 01; deci (\" d \") = 1E-01; centi (\" c \") = 1E-02; milli (\" m \") = 1E-03; micro (\" u \") = 1E-06; nano (\" n \") = 1E-09; pico (\" p \") = 1E-12; femto (\" f \") = 1E-15; atto (\" a \") = 1E-18\r\n    ";
        this["Func_convert_Category"] = "Inginerie";
        this["Func_convert_CategoryURL"] = "Inginerie";
        this["Func_convert_Desc"] = "Convertește o valoare dintr-un sistem de măsurare în altul, cum ar fi metri în inci sau ore în secunde.";
        this["Func_cos_Arg_0"] = "număr";
        this["Func_cos_ArgDesc_0"] = "Unghiul în radiani pentru care se obține cosinusul.";
        this["Func_cos_Category"] = "Matematica";
        this["Func_cos_CategoryURL"] = "Matematica";
        this["Func_cos_Desc"] = "Returnează cosinusul unghiului dat.";
        this["Func_cosh_Arg_0"] = "număr";
        this["Func_cosh_ArgDesc_0"] = "Un număr real pentru care să se găsească cosinusul hiperbolic.";
        this["Func_cosh_Category"] = "Matematica";
        this["Func_cosh_CategoryURL"] = "Matematica";
        this["Func_cosh_Desc"] = "Returnează cosinusul hiperbolic al unui număr.";
        this["Func_count_Arg_0"] = "valoare {0}";
        this["Func_count_ArgDesc_0"] = "O referință sau o gamă de referințe care pot conține numere.";
        this["Func_count_Category"] = "Statistic";
        this["Func_count_CategoryURL"] = "Statistic";
        this["Func_count_Desc"] = "Numără numărul de referințe care conțin numere.";
        this["Func_counta_Arg_0"] = "valoare {0}";
        this["Func_counta_ArgDesc_0"] = "O referință sau o gamă de referințe care pot conține numere.";
        this["Func_counta_Category"] = "Statistic";
        this["Func_counta_CategoryURL"] = "Statistic";
        this["Func_counta_Desc"] = "Numără numărul de referințe care nu sunt necompletate.";
        this["Func_critbinom_Arg_0"] = "Încercări";
        this["Func_critbinom_Arg_1"] = "Probabilitate";
        this["Func_critbinom_Arg_2"] = "Alfa";
        this["Func_critbinom_ArgDesc_0"] = "Numărul de studii independente.";
        this["Func_critbinom_ArgDesc_1"] = "Probabilitatea de succes pe fiecare proces.";
        this["Func_critbinom_ArgDesc_2"] = "Valoarea criteriului.";
        this["Func_critbinom_Category"] = "Statistic";
        this["Func_critbinom_CategoryURL"] = "Statistic";
        this["Func_critbinom_Desc"] = "Returnează cea mai mică valoare pentru care distribuția binomială cumulativă este mai mare sau egală cu o valoare de criteriu.";
        this["Func_date_Arg_0"] = "an";
        this["Func_date_Arg_1"] = "lună";
        this["Func_date_Arg_2"] = "zi";
        this["Func_date_ArgDesc_0"] = "Un număr care reprezintă anul.";
        this["Func_date_ArgDesc_1"] = "Un număr care reprezintă luna anului.";
        this["Func_date_ArgDesc_2"] = "Un număr care reprezintă ziua lunii.";
        this["Func_date_Category"] = "Data si ora";
        this["Func_date_CategoryURL"] = "Date_and_Time";
        this["Func_date_Desc"] = "Returnează numărul de serie secvențial care reprezintă o anumită dată.";
        this["Func_dateadd_Arg_0"] = "interval";
        this["Func_dateadd_Arg_1"] = "număr";
        this["Func_dateadd_Arg_2"] = "date";
        this["Func_dateadd_ArgDesc_0"] = "O expresie text care reprezintă intervalul de timp de adăugat.\r\nd = Ziua\r\ny = An\r\nh = Ora\r\nn = Minut\r\nm = lună\r\nq = trimestru\r\ns = Secund\r\nw = Săptămâna\r\nww = Săptămâna calendaristică\r\nyyyy = An\r\n";
        this["Func_dateadd_ArgDesc_1"] = "Numărul de intervale de adăugat. Această valoare poate fi pozitivă (pentru a obține un DateTime în viitor) sau negativă (pentru a obține un DateTime în trecut). Poate conține o parte fracționată atunci când Intervalul specifică ore, minute sau secunde. Pentru alte valori ale intervalului, orice parte fracționată a numărului este ignorată.";
        this["Func_dateadd_ArgDesc_2"] = "Data la care trebuie adăugat intervalul.";
        this["Func_dateadd_Category"] = "Data si ora";
        this["Func_dateadd_CategoryURL"] = "Date_and_Time";
        this["Func_dateadd_Desc"] = "Returnează o dată la care a fost adăugat intervalul specificat.";
        this["Func_datediff_Arg_0"] = "interval";
        this["Func_datediff_Arg_1"] = "date1";
        this["Func_datediff_Arg_2"] = "date2";
        this["Func_datediff_ArgDesc_0"] = "O expresie text care reprezintă intervalul de timp de utilizat ca unitate de diferență între data1 și data2.\r\nd = Ziua\r\ny = An\r\nh = Ora\r\nn = Minut\r\nm = lună\r\nq = trimestru\r\ns = Secund\r\nw = Săptămâna\r\nww = Săptămâna calendaristică\r\nyyyy = An\r\n";
        this["Func_datediff_ArgDesc_1"] = "Prima dată de utilizat în calculul diferenței.";
        this["Func_datediff_ArgDesc_2"] = "A doua dată de utilizat în calculul diferenței.";
        this["Func_datediff_Category"] = "Data si ora";
        this["Func_datediff_CategoryURL"] = "Date_and_Time";
        this["Func_datediff_Desc"] = "Returnează o valoare care indică numărul de intervale între două valori DateTime.";
        this["Func_datevalue_Arg_0"] = "date_text";
        this["Func_datevalue_ArgDesc_0"] = "O expresie text care reprezintă o dată.";
        this["Func_datevalue_Category"] = "Data si ora";
        this["Func_datevalue_CategoryURL"] = "Date_and_Time";
        this["Func_datevalue_Desc"] = "Returnează valoarea datei textului de dată specificat.";
        this["Func_day_Arg_0"] = "Date_value";
        this["Func_day_ArgDesc_0"] = "O valoare de dată din care să obțineți ziua.";
        this["Func_day_Category"] = "Data si ora";
        this["Func_day_CategoryURL"] = "Date_and_Time";
        this["Func_day_Desc"] = "Returnează ziua lunii valorii datei.";
        this["Func_days360_Arg_0"] = "start_date";
        this["Func_days360_Arg_1"] = "end_date";
        this["Func_days360_Arg_2"] = "metodă";
        this["Func_days360_ArgDesc_0"] = "Data de începere a intervalului de zile. Dacă aceasta este după data de încheiere, funcția va returna un număr negativ.";
        this["Func_days360_ArgDesc_1"] = "Data de încheiere a intervalului de zile. Dacă aceasta este înainte de data de începere, funcția va returna un număr negativ.";
        this["Func_days360_ArgDesc_2"] = "O valoare logică care specifică dacă se utilizează metoda SUA (falsă) sau europeană (adevărată) în calcul.";
        this["Func_days360_Category"] = "Data si ora";
        this["Func_days360_CategoryURL"] = "Date_and_Time";
        this["Func_days360_Desc"] = "Returnează numărul de zile dintre două date pe baza unui an de 360 de zile.";
        this["Func_db_Arg_0"] = "cost";
        this["Func_db_Arg_1"] = "salvare";
        this["Func_db_Arg_2"] = "viaţă";
        this["Func_db_Arg_3"] = "perioadă";
        this["Func_db_Arg_4"] = "lună";
        this["Func_db_ArgDesc_0"] = "Costul inițial al activului.";
        this["Func_db_ArgDesc_1"] = "Valoarea la sfârșitul amortizării (numită uneori valoarea de recuperare a activului).";
        this["Func_db_ArgDesc_2"] = "Numărul de perioade în care activul este depreciat (denumit uneori viața utilă a activului).";
        this["Func_db_ArgDesc_3"] = "Perioada pentru care se calculează amortizarea. Perioada trebuie să utilizeze aceleași unități ca și viața.";
        this["Func_db_ArgDesc_4"] = "Numărul de luni din primul an. Dacă luna este omisă, se presupune că este 12.";
        this["Func_db_Category"] = "Financiar";
        this["Func_db_CategoryURL"] = "Financiar";
        this["Func_db_Desc"] = "Returnează amortizarea unui activ pentru o perioadă specificată utilizând metoda soldului cu scădere fixă.";
        this["Func_dbnull_Category"] = "Informație";
        this["Func_dbnull_CategoryURL"] = "Informație";
        this["Func_dbnull_Desc"] = "Returnează ceva care reprezintă o valoare inexistentă.";
        this["Func_ddb_Arg_0"] = "cost";
        this["Func_ddb_Arg_1"] = "salvare";
        this["Func_ddb_Arg_2"] = "viaţă";
        this["Func_ddb_Arg_3"] = "perioadă";
        this["Func_ddb_Arg_4"] = "factor";
        this["Func_ddb_ArgDesc_0"] = "Costul inițial al activului.";
        this["Func_ddb_ArgDesc_1"] = "Valoarea la sfârșitul amortizării (numită uneori valoarea de recuperare a activului).";
        this["Func_ddb_ArgDesc_2"] = "Numărul de perioade în care activul este depreciat (denumit uneori viața utilă a activului).";
        this["Func_ddb_ArgDesc_3"] = "Perioada pentru care se calculează amortizarea. Perioada trebuie să utilizeze aceleași unități ca și viața.";
        this["Func_ddb_ArgDesc_4"] = "Rata la care scade soldul. Dacă factorul este omis, se presupune că este 2 (metoda echilibrului dublu declinant).";
        this["Func_ddb_Category"] = "Financiar";
        this["Func_ddb_CategoryURL"] = "Financiar";
        this["Func_ddb_Desc"] = "Returnează amortizarea unui activ pentru o perioadă specificată utilizând metoda soldului cu scădere dublă sau altă metodă specificată.";
        this["Func_dec2bin_Arg_0"] = "Număr";
        this["Func_dec2bin_Arg_1"] = "Locuri";
        this["Func_dec2bin_ArgDesc_0"] = "Valoarea zecimală care urmează să fie convertită.";
        this["Func_dec2bin_ArgDesc_1"] = "Numărul de caractere de utilizat pentru reprezentarea valorii. Dacă locurile nu sunt specificate, va fi utilizat numărul minim de caractere necesar pentru a reprezenta valoarea. Acest lucru vă permite să specificați zerouri din partea de sus.";
        this["Func_dec2bin_Category"] = "Inginerie";
        this["Func_dec2bin_CategoryURL"] = "Inginerie";
        this["Func_dec2bin_Desc"] = "Returnează o expresie text care reprezintă valoarea zecimală specificată ca număr binar.";
        this["Func_dec2hex_Arg_0"] = "Număr";
        this["Func_dec2hex_Arg_1"] = "Locuri";
        this["Func_dec2hex_ArgDesc_0"] = "Valoarea zecimală care urmează să fie convertită.";
        this["Func_dec2hex_ArgDesc_1"] = "Numărul de caractere de utilizat pentru reprezentarea valorii. Dacă locurile nu sunt specificate, va fi utilizat numărul minim de caractere necesar pentru a reprezenta valoarea. Acest lucru vă permite să specificați zerouri din partea de sus.";
        this["Func_dec2hex_Category"] = "Inginerie";
        this["Func_dec2hex_CategoryURL"] = "Inginerie";
        this["Func_dec2hex_Desc"] = "Returnează o expresie text care reprezintă valoarea zecimală specificată ca număr hexazecimal.";
        this["Func_dec2oct_Arg_0"] = "Număr";
        this["Func_dec2oct_Arg_1"] = "Locuri";
        this["Func_dec2oct_ArgDesc_0"] = "Valoarea zecimală care urmează să fie convertită.";
        this["Func_dec2oct_ArgDesc_1"] = "Numărul de caractere de utilizat pentru reprezentarea valorii. Dacă locurile nu sunt specificate, va fi utilizat numărul minim de caractere necesar pentru a reprezenta valoarea. Acest lucru vă permite să specificați zerouri din partea de sus.";
        this["Func_dec2oct_Category"] = "Inginerie";
        this["Func_dec2oct_CategoryURL"] = "Inginerie";
        this["Func_dec2oct_Desc"] = "Returnează o expresie text care reprezintă valoarea zecimală specificată ca număr octal.";
        this["Func_degrees_Arg_0"] = "Radiani";
        this["Func_degrees_ArgDesc_0"] = "O valoare în radiani care va fi convertită în grade.";
        this["Func_degrees_Category"] = "Matematica";
        this["Func_degrees_CategoryURL"] = "Matematica";
        this["Func_degrees_Desc"] = "Convertește radianii în grade.";
        this["Func_delta_Arg_0"] = "Number1";
        this["Func_delta_Arg_1"] = "Number2";
        this["Func_delta_ArgDesc_0"] = "Primul număr de comparat.";
        this["Func_delta_ArgDesc_1"] = "Al doilea număr de comparat. Dacă nu este specificat, se va presupune că este zero.";
        this["Func_delta_Category"] = "Inginerie";
        this["Func_delta_CategoryURL"] = "Inginerie";
        this["Func_delta_Desc"] = "Compară două numere și returnează unul dacă sunt egale sau zero dacă nu sunt.";
        this["Func_dollarde_Arg_0"] = "FractionDollarAmount";
        this["Func_dollarde_Arg_1"] = "Fracțiune";
        this["Func_dollarde_ArgDesc_0"] = "Suma în dolari exprimată ca fracție.";
        this["Func_dollarde_ArgDesc_1"] = "Valoarea utilizată ca numitor al fracției. Dacă această valoare nu este un număr întreg, va fi trunchiată.";
        this["Func_dollarde_Category"] = "Financiar";
        this["Func_dollarde_CategoryURL"] = "Financiar";
        this["Func_dollarde_Desc"] = "Convertește o sumă în dolari exprimată ca fracție într-o sumă în dolari exprimată ca zecimală.";
        this["Func_dollarfr_Arg_0"] = "DecimalDollarAmount";
        this["Func_dollarfr_Arg_1"] = "Fracțiune";
        this["Func_dollarfr_ArgDesc_0"] = "Suma în dolari exprimată ca zecimală.";
        this["Func_dollarfr_ArgDesc_1"] = "Valoarea utilizată ca numitor al fracției. Dacă această valoare nu este un număr întreg, va fi trunchiată.";
        this["Func_dollarfr_Category"] = "Financiar";
        this["Func_dollarfr_CategoryURL"] = "Financiar";
        this["Func_dollarfr_Desc"] = "Convertește o sumă în dolari exprimată ca zecimală într-o sumă în dolari exprimată ca fracție.";
        this["Func_edate_Arg_0"] = "date";
        this["Func_edate_Arg_1"] = "numberOfMonths";
        this["Func_edate_ArgDesc_0"] = "Data de începere.";
        this["Func_edate_ArgDesc_1"] = "Numărul de luni pentru schimbarea datei. Un număr pozitiv înseamnă o dată după data de începere. Un număr negativ indică o dată anterioară datei de începere.";
        this["Func_edate_Category"] = "Data si ora";
        this["Func_edate_CategoryURL"] = "Data si ora";
        this["Func_edate_Desc"] = "Returnează o dată care este numărul specificat de luni înainte sau după data de începere.";
        this["Func_eomonth_Arg_0"] = "date";
        this["Func_eomonth_Arg_1"] = "numberOfMonths";
        this["Func_eomonth_ArgDesc_0"] = "Data de începere.";
        this["Func_eomonth_ArgDesc_1"] = "Numărul de luni pentru schimbarea datei. Un număr pozitiv înseamnă o dată după data de începere. Un număr negativ indică o dată anterioară datei de începere.";
        this["Func_eomonth_Category"] = "Data si ora";
        this["Func_eomonth_CategoryURL"] = "Data si ora";
        this["Func_eomonth_Desc"] = "Returnează o dată care este la sfârșitul lunii, care este numărul specificat de luni înainte sau după data de începere.";
        this["Func_errortype_Arg_0"] = "error_value";
        this["Func_errortype_ArgDesc_0"] = "Valoarea erorii al cărei număr de identificare ar trebui găsit.";
        this["Func_errortype_Category"] = "Informație";
        this["Func_errortype_CategoryURL"] = "Informație";
        this["Func_errortype_Desc"] = "Returnează un număr corespunzător unei valori de eroare.";
        this["Func_even_Arg_0"] = "număr";
        this["Func_even_ArgDesc_0"] = "Valoarea de rotunjit.";
        this["Func_even_Category"] = "Matematica";
        this["Func_even_CategoryURL"] = "Matematica";
        this["Func_even_Desc"] = "Rotunjește un număr pozitiv în sus și un număr negativ în jos la cel mai apropiat număr întreg.";
        this["Func_exp_Arg_0"] = "număr";
        this["Func_exp_ArgDesc_0"] = "Exponentul aplicat la care să crească e.";
        this["Func_exp_Category"] = "Matematica";
        this["Func_exp_CategoryURL"] = "Matematica";
        this["Func_exp_Desc"] = "Returnează e ridicat la puterea numărului specificat. Constanta e este egală cu 2,71828182845904.";
        this["Func_fact_Arg_0"] = "Număr";
        this["Func_fact_ArgDesc_0"] = "Un număr pozitiv pentru care se va calcula factorialul.";
        this["Func_fact_Category"] = "Matematica";
        this["Func_fact_CategoryURL"] = "Matematica";
        this["Func_fact_Desc"] = "Returnează factorialul unui număr.";
        this["Func_factdouble_Arg_0"] = "Număr";
        this["Func_factdouble_ArgDesc_0"] = "Un număr pozitiv pentru care se va calcula factorialul dublu.";
        this["Func_factdouble_Category"] = "Matematica";
        this["Func_factdouble_CategoryURL"] = "Matematica";
        this["Func_factdouble_Desc"] = "Returnează factorialul dublu al unui număr.";
        this["Func_false_Category"] = "Logic";
        this["Func_false_CategoryURL"] = "Logic";
        this["Func_false_Desc"] = "Returnează valoarea logică false.";
        this["Func_find_Arg_0"] = "find_text";
        this["Func_find_Arg_1"] = "within_text";
        this["Func_find_Arg_2"] = "start_num";
        this["Func_find_ArgDesc_0"] = "Textul de găsit.";
        this["Func_find_ArgDesc_1"] = "Textul care conține textul de găsit.";
        this["Func_find_ArgDesc_2"] = "Caracterul index 1 bazat pe care să începeți căutarea. Dacă este omis, se presupune că este unul.";
        this["Func_find_Category"] = "TextAndData";
        this["Func_find_CategoryURL"] = "Text_and_Data";
        this["Func_find_Desc"] = "Returnează poziția inițială a unei expresii de text în alta, cu sensibilitate la majuscule.";
        this["Func_fixed_Arg_0"] = "număr";
        this["Func_fixed_Arg_1"] = "decimals";
        this["Func_fixed_Arg_2"] = "no_comas";
        this["Func_fixed_ArgDesc_0"] = "Numărul de rotunjit și de convertit în text.";
        this["Func_fixed_ArgDesc_1"] = "Numărul de cifre din dreapta punctului zecimal din valoarea returnată. Dacă această valoare este negativă, \"număr\" este rotunjit la stânga punctului zecimal. Valoarea implicită este 2.";
        this["Func_fixed_ArgDesc_2"] = "Adevărat dacă virgulele nu trebuie să fie în valoarea returnată. Dacă este omis, se presupune că este fals.";
        this["Func_fixed_Category"] = "TextAndData";
        this["Func_fixed_CategoryURL"] = "Text_and_Data";
        this["Func_fixed_Desc"] = "Rotunjește un număr la numărul specificat de zecimale și returnează rezultatul ca text cu sau fără virgule.";
        this["Func_floor_Arg_0"] = "număr";
        this["Func_floor_Arg_1"] = "semnificaţie";
        this["Func_floor_ArgDesc_0"] = "Numărul de rotunjit.";
        this["Func_floor_ArgDesc_1"] = "Multiplu la care se rotunjește. Dacă este omis, se presupune că acesta este unul.";
        this["Func_floor_Category"] = "Matematica";
        this["Func_floor_CategoryURL"] = "Matematica";
        this["Func_floor_Desc"] = "Returnează cel mai mare multiplu al semnificației (implicit la 1) mai mic sau egal cu numărul dat.";
        this["Func_fv_Arg_0"] = "rată";
        this["Func_fv_Arg_1"] = "nper";
        this["Func_fv_Arg_2"] = "pmt";
        this["Func_fv_Arg_3"] = "pv";
        this["Func_fv_Arg_4"] = "type";
        this["Func_fv_ArgDesc_0"] = "Rata dobânzii pe perioadă.";
        this["Func_fv_ArgDesc_1"] = "Numărul total de perioade de plată dintr-o anuitate.";
        this["Func_fv_ArgDesc_2"] = "Plata efectuată în fiecare perioadă, care nu se poate modifica pe durata de viață a anuității.";
        this["Func_fv_ArgDesc_3"] = "Valoarea actuală sau suma forfetară pe care o valoare o serie de plăți viitoare o merită acum. Dacă este angajat, se presupune că este zero.";
        this["Func_fv_ArgDesc_4"] = "Numărul zero sau unul care indică dacă plățile sunt datorate la sfârșitul perioadei (zero, valoarea implicită) sau la începutul (una).";
        this["Func_fv_Category"] = "Financiar";
        this["Func_fv_CategoryURL"] = "Financiar";
        this["Func_fv_Desc"] = "Returnează valoarea viitoare a unei investiții pe baza periodică.";
        this["Func_gcd_Arg_0"] = "numărul {0}";
        this["Func_gcd_ArgDesc_0"] = "O valoare întreagă. O valoare neîntreagă va fi trunchiată.";
        this["Func_gcd_Category"] = "Matematica";
        this["Func_gcd_CategoryURL"] = "Matematica";
        this["Func_gcd_Desc"] = "Returnează cel mai mare divizor comun al uneia sau mai multor valori întregi.";
        this["Func_gestep_Arg_0"] = "Number1";
        this["Func_gestep_Arg_1"] = "Pas";
        this["Func_gestep_ArgDesc_0"] = "Primul număr de comparat.";
        this["Func_gestep_ArgDesc_1"] = "Al doilea număr de comparat. Dacă nu este specificat, se va presupune că este zero.";
        this["Func_gestep_Category"] = "Inginerie";
        this["Func_gestep_CategoryURL"] = "Inginerie";
        this["Func_gestep_Desc"] = "Compară două numere și returnează unul dacă primul număr este mai mare sau egal cu al doilea sau returnează zero dacă nu.";
        this["Func_hex2bin_Arg_0"] = "Număr";
        this["Func_hex2bin_Arg_1"] = "Locuri";
        this["Func_hex2bin_ArgDesc_0"] = "O expresie text care reprezintă un număr hexazecimal.";
        this["Func_hex2bin_ArgDesc_1"] = "Numărul de caractere de utilizat pentru reprezentarea valorii. Dacă acest lucru nu este specificat, va fi utilizat numărul minim de caractere necesar pentru a reprezenta valoarea. Acest lucru vă permite să specificați zerouri de la început.";
        this["Func_hex2bin_Category"] = "Inginerie";
        this["Func_hex2bin_CategoryURL"] = "Inginerie";
        this["Func_hex2bin_Desc"] = "Convertește o expresie text care reprezintă un număr hexazecimal într-o expresie text care reprezintă același număr în sistemul numeric binar.";
        this["Func_hex2dec_Arg_0"] = "Număr";
        this["Func_hex2dec_ArgDesc_0"] = "O expresie text care reprezintă un număr hexazecimal.";
        this["Func_hex2dec_Category"] = "Inginerie";
        this["Func_hex2dec_CategoryURL"] = "Inginerie";
        this["Func_hex2dec_Desc"] = "Returnează valoarea zecimală a unei reprezentări text ale unui număr în sistemul numeric hexazecimal.";
        this["Func_hex2oct_Arg_0"] = "Număr";
        this["Func_hex2oct_Arg_1"] = "Locuri";
        this["Func_hex2oct_ArgDesc_0"] = "O expresie text care reprezintă un număr hexazecimal.";
        this["Func_hex2oct_ArgDesc_1"] = "Numărul de caractere de utilizat pentru reprezentarea valorii. Dacă acest lucru nu este specificat, va fi utilizat numărul minim de caractere necesar pentru a reprezenta valoarea. Acest lucru vă permite să specificați zerouri de la început.";
        this["Func_hex2oct_Category"] = "Inginerie";
        this["Func_hex2oct_CategoryURL"] = "Inginerie";
        this["Func_hex2oct_Desc"] = "Convertește o expresie text care reprezintă un număr hexazecimal într-o expresie text care reprezintă același număr în sistemul numeric octal.";
        this["Func_hour_Arg_0"] = "Time_value";
        this["Func_hour_ArgDesc_0"] = "O valoare de timp din care să obțineți ora.";
        this["Func_hour_Category"] = "Data si ora";
        this["Func_hour_CategoryURL"] = "Date_and_Time";
        this["Func_hour_Desc"] = "Returnează ora unei valori de timp.";
        this["Func_if_Arg_0"] = "test_logic";
        this["Func_if_Arg_1"] = "value_if_true";
        this["Func_if_Arg_2"] = "value_if_false";
        this["Func_if_ArgDesc_0"] = "Condiția de testat.";
        this["Func_if_ArgDesc_1"] = "Rezultatul dacă condiția se evaluează la adevărat.";
        this["Func_if_ArgDesc_2"] = "Rezultatul dacă condiția se evaluează la fals. Dacă este omis, se presupune că este valoarea logică falsă.";
        this["Func_if_Category"] = "Logic";
        this["Func_if_CategoryURL"] = "Logic";
        this["Func_if_Desc"] = "Returnează o valoare dacă condiția specificată se evaluează la adevărat și o altă valoare dacă se evaluează la fals.";
        this["Func_iferror_Arg_0"] = "valoare";
        this["Func_iferror_Arg_1"] = "value_if_error";
        this["Func_iferror_ArgDesc_0"] = "Valoarea de utilizat atunci când nu este o eroare.";
        this["Func_iferror_ArgDesc_1"] = "Valoarea de utilizat dacă argumentul valoare este o eroare.";
        this["Func_iferror_Category"] = "Logic";
        this["Func_iferror_CategoryURL"] = "Logic";
        this["Func_iferror_Desc"] = "Returnează una din cele două valori, în funcție de faptul dacă prima valoare este sau nu o eroare.";
        this["Func_imabs_Arg_0"] = "ComplexNumber";
        this["Func_imabs_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imabs_Category"] = "Inginerie";
        this["Func_imabs_CategoryURL"] = "Inginerie";
        this["Func_imabs_Desc"] = "Returnează valoarea absolută a unui număr complex.";
        this["Func_imaginary_Arg_0"] = "ComplexNumber";
        this["Func_imaginary_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imaginary_Category"] = "Inginerie";
        this["Func_imaginary_CategoryURL"] = "Inginerie";
        this["Func_imaginary_Desc"] = "Returnează coeficientul imaginar al unui număr complex.";
        this["Func_imargument_Arg_0"] = "ComplexNumber";
        this["Func_imargument_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imargument_Category"] = "Inginerie";
        this["Func_imargument_CategoryURL"] = "Inginerie";
        this["Func_imargument_Desc"] = "Returnează argumentul theta, un unghi exprimat în radiani.";
        this["Func_imconjugate_Arg_0"] = "ComplexNumber";
        this["Func_imconjugate_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imconjugate_Category"] = "Inginerie";
        this["Func_imconjugate_CategoryURL"] = "Inginerie";
        this["Func_imconjugate_Desc"] = "Returnează conjugatul unui număr complex.";
        this["Func_imcos_Arg_0"] = "ComplexNumber";
        this["Func_imcos_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imcos_Category"] = "Inginerie";
        this["Func_imcos_CategoryURL"] = "Inginerie";
        this["Func_imcos_Desc"] = "Returnează cosinusul unui număr complex.";
        this["Func_imdiv_Arg_0"] = "Dividend";
        this["Func_imdiv_Arg_1"] = "divizor";
        this["Func_imdiv_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imdiv_ArgDesc_1"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imdiv_Category"] = "Inginerie";
        this["Func_imdiv_CategoryURL"] = "Inginerie";
        this["Func_imdiv_Desc"] = "Returnează coeficientul a două numere complexe (Dividend / Divizor).";
        this["Func_imexp_Arg_0"] = "ComplexNumber";
        this["Func_imexp_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imexp_Category"] = "Inginerie";
        this["Func_imexp_CategoryURL"] = "Inginerie";
        this["Func_imexp_Desc"] = "Returnează exponențialul unui număr complex.";
        this["Func_imln_Arg_0"] = "ComplexNumber";
        this["Func_imln_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imln_Category"] = "Inginerie";
        this["Func_imln_CategoryURL"] = "Inginerie";
        this["Func_imln_Desc"] = "Returnează logaritmul natural al unui număr complex.";
        this["Func_imlog10_Arg_0"] = "ComplexNumber";
        this["Func_imlog10_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imlog10_Category"] = "Inginerie";
        this["Func_imlog10_CategoryURL"] = "Inginerie";
        this["Func_imlog10_Desc"] = "Returnează logaritmul de bază-10 al unui număr complex.";
        this["Func_imlog2_Arg_0"] = "ComplexNumber";
        this["Func_imlog2_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imlog2_Category"] = "Inginerie";
        this["Func_imlog2_CategoryURL"] = "Inginerie";
        this["Func_imlog2_Desc"] = "Returnează logaritmul de bază-2 al unui număr complex.";
        this["Func_impower_Arg_0"] = "ComplexNumber";
        this["Func_impower_Arg_1"] = "putere";
        this["Func_impower_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_impower_ArgDesc_1"] = "Puterea la care va fi ridicat ComplexNumber.";
        this["Func_impower_Category"] = "Inginerie";
        this["Func_impower_CategoryURL"] = "Inginerie";
        this["Func_impower_Desc"] = "Returnează numărul complex ridicat la puterea specificată.";
        this["Func_improduct_Arg_0"] = "Număr complex {0}";
        this["Func_improduct_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_improduct_Category"] = "Inginerie";
        this["Func_improduct_CategoryURL"] = "Inginerie";
        this["Func_improduct_Desc"] = "Returnează produsul unuia sau mai multor numere complexe.";
        this["Func_imreal_Arg_0"] = "ComplexNumber";
        this["Func_imreal_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imreal_Category"] = "Inginerie";
        this["Func_imreal_CategoryURL"] = "Inginerie";
        this["Func_imreal_Desc"] = "Returnează coeficientul real al unui număr complex.";
        this["Func_imsin_Arg_0"] = "ComplexNumber";
        this["Func_imsin_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imsin_Category"] = "Inginerie";
        this["Func_imsin_CategoryURL"] = "Inginerie";
        this["Func_imsin_Desc"] = "Returnează sinusul unui număr complex.";
        this["Func_imsqrt_Arg_0"] = "ComplexNumber";
        this["Func_imsqrt_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imsqrt_Category"] = "Inginerie";
        this["Func_imsqrt_CategoryURL"] = "Inginerie";
        this["Func_imsqrt_Desc"] = "Returnează rădăcina pătrată a unui număr complex.";
        this["Func_imsub_Arg_0"] = "ComplexNumber1";
        this["Func_imsub_Arg_1"] = "ComplexNumber2";
        this["Func_imsub_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\" din care se va scădea Numărul Complex2.";
        this["Func_imsub_ArgDesc_1"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\" care va fi scăzut din ComplexNumber1.";
        this["Func_imsub_Category"] = "Inginerie";
        this["Func_imsub_CategoryURL"] = "Inginerie";
        this["Func_imsub_Desc"] = "Returnează diferența dintre două numere complexe.";
        this["Func_imsum_Arg_0"] = "Număr complex {0}";
        this["Func_imsum_ArgDesc_0"] = "Reprezentarea text a unui număr complex în format: \"x + yi\" sau \"x + yj\".";
        this["Func_imsum_Category"] = "Inginerie";
        this["Func_imsum_CategoryURL"] = "Inginerie";
        this["Func_imsum_Desc"] = "Returnează suma unuia sau mai multor numere complexe.";
        this["Func_info_Arg_0"] = "Tip";
        this["Func_info_ArgDesc_0"] = "O expresie text care indică tipul de informații de returnat.\r\n\"director\" = Calea completă către folderul curent.\r\n\"osversion\" = Versiunea curentă a sistemului de operare ca expresie text.\r\n\"sistem\" = Sistemul de operare curent. Macintosh = \"mac\", Windows = \"pcdos\".\r\n    ";
        this["Func_info_Category"] = "Informație";
        this["Func_info_CategoryURL"] = "Informație";
        this["Func_info_Desc"] = "Returnează informații despre mediul de operare actual.";
        this["Func_int_Arg_0"] = "număr";
        this["Func_int_ArgDesc_0"] = "Numărul real de rotunjit în jos la un număr întreg.";
        this["Func_int_Category"] = "Matematica";
        this["Func_int_CategoryURL"] = "Matematica";
        this["Func_int_Desc"] = "Rotunjește un număr până la cel mai apropiat număr întreg.";
        this["Func_intrate_Arg_0"] = "așezare";
        this["Func_intrate_Arg_1"] = "maturitate";
        this["Func_intrate_Arg_2"] = "investiție";
        this["Func_intrate_Arg_3"] = "răscumpărare";
        this["Func_intrate_Arg_4"] = "bază";
        this["Func_intrate_ArgDesc_0"] = "Data decontării garanției.";
        this["Func_intrate_ArgDesc_1"] = "Data scadenței garanției.";
        this["Func_intrate_ArgDesc_2"] = "Suma investită în garanție.";
        this["Func_intrate_ArgDesc_3"] = "Suma de primit la scadență.";
        this["Func_intrate_ArgDesc_4"] = "Tipul bazei de numărare a zilei de utilizat.\r\n0 sau omis = SUA (NASD) 30 / 360,\r\n1 = Actual / actual,\r\n2 = Actual / 360,\r\n3 = Actual / 365,\r\n4 = European 30 / 360\r\n    ";
        this["Func_intrate_Category"] = "Financiar";
        this["Func_intrate_CategoryURL"] = "Financiar";
        this["Func_intrate_Desc"] = "Returnează rata dobânzii pentru un titlu total investit.";
        this["Func_ipmt_Arg_0"] = "rată";
        this["Func_ipmt_Arg_1"] = "per";
        this["Func_ipmt_Arg_2"] = "nper";
        this["Func_ipmt_Arg_3"] = "pv";
        this["Func_ipmt_Arg_4"] = "fv";
        this["Func_ipmt_Arg_5"] = "type";
        this["Func_ipmt_ArgDesc_0"] = "Rata dobânzii pe perioadă.";
        this["Func_ipmt_ArgDesc_1"] = "Perioada, care trebuie să fie cuprinsă între 1 și nper.";
        this["Func_ipmt_ArgDesc_2"] = "Numărul total de perioade de plată dintr-o anuitate.";
        this["Func_ipmt_ArgDesc_3"] = "Valoarea actuală sau suma forfetară pe care o valorează acum o serie de plăți viitoare.";
        this["Func_ipmt_ArgDesc_4"] = "Valoarea viitoare sau un sold de numerar pe care doriți să îl obțineți după efectuarea ultimei plăți. Dacă este angajat, se presupune că este zero.";
        this["Func_ipmt_ArgDesc_5"] = "Numărul zero sau unul care indică dacă plățile sunt datorate la sfârșitul perioadei (zero, valoarea implicită) sau la începutul (una).";
        this["Func_ipmt_Category"] = "Financiar";
        this["Func_ipmt_CategoryURL"] = "Financiar";
        this["Func_ipmt_Desc"] = "Returnează plata dobânzii pentru o anumită perioadă pentru o investiție, pe baza plăților periodice, constante și a unei rate a dobânzii constante.";
        this["Func_irr_Arg_0"] = "valori";
        this["Func_irr_Arg_1"] = "ghici";
        this["Func_irr_ArgDesc_0"] = "O colecție de valori de plată (valori negative) și venituri (valori pozitive) care apar la intervale regulate.";
        this["Func_irr_ArgDesc_1"] = "Un număr pe care ghiciți că este aproape de rezultat. Valoarea implicită este 0,1 (10%).";
        this["Func_irr_Category"] = "Financiar";
        this["Func_irr_CategoryURL"] = "Financiar";
        this["Func_irr_Desc"] = "Returnează rata internă de rentabilitate pentru o serie de fluxuri de numerar.";
        this["Func_isblank_Arg_0"] = "valoare";
        this["Func_isblank_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_isblank_Category"] = "Informație";
        this["Func_isblank_CategoryURL"] = "Informație";
        this["Func_isblank_Desc"] = "Returnează adevărat dacă valoarea este necompletată.";
        this["Func_isdbnull_Arg_0"] = "valoare";
        this["Func_isdbnull_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_isdbnull_Category"] = "Informație";
        this["Func_isdbnull_CategoryURL"] = "Informație";
        this["Func_isdbnull_Desc"] = "Returnează adevărat dacă valoarea reprezintă o valoare inexistentă.";
        this["Func_iserr_Arg_0"] = "valoare";
        this["Func_iserr_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_iserr_Category"] = "Informație";
        this["Func_iserr_CategoryURL"] = "Informație";
        this["Func_iserr_Desc"] = "Returnează adevărat dacă valoarea este orice valoare de eroare, cu excepția #N/A.";
        this["Func_iserror_Arg_0"] = "valoare";
        this["Func_iserror_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_iserror_Category"] = "Informație";
        this["Func_iserror_CategoryURL"] = "Informație";
        this["Func_iserror_Desc"] = "Returnează adevărat dacă valoarea este orice valoare de eroare.";
        this["Func_iseven_Arg_0"] = "valoare";
        this["Func_iseven_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_iseven_Category"] = "Informație";
        this["Func_iseven_CategoryURL"] = "Informație";
        this["Func_iseven_Desc"] = "Returnează true dacă valoarea este un număr par, altfel returnează false.";
        this["Func_islogical_Arg_0"] = "valoare";
        this["Func_islogical_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_islogical_Category"] = "Informație";
        this["Func_islogical_CategoryURL"] = "Informație";
        this["Func_islogical_Desc"] = "Returnează adevărat dacă valoarea este o valoare logică (adevărată sau falsă).";
        this["Func_isna_Arg_0"] = "valoare";
        this["Func_isna_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_isna_Category"] = "Informație";
        this["Func_isna_CategoryURL"] = "Informație";
        this["Func_isna_Desc"] = "Returnează adevărat dacă valoarea este valoarea de eroare #N/A (valoarea nu este disponibilă).";
        this["Func_isnontext_Arg_0"] = "valoare";
        this["Func_isnontext_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_isnontext_Category"] = "Informație";
        this["Func_isnontext_CategoryURL"] = "Informație";
        this["Func_isnontext_Desc"] = "Returnează adevărat dacă valoarea este un element care nu este text.";
        this["Func_isnull_Arg_0"] = "valoare";
        this["Func_isnull_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_isnull_Category"] = "Informație";
        this["Func_isnull_CategoryURL"] = "Informație";
        this["Func_isnull_Desc"] = "Returnează adevărat dacă valoarea este inexistentă.";
        this["Func_isnumber_Arg_0"] = "valoare";
        this["Func_isnumber_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_isnumber_Category"] = "Informație";
        this["Func_isnumber_CategoryURL"] = "Informație";
        this["Func_isnumber_Desc"] = "Returnează adevărat dacă valoarea este un număr.";
        this["Func_isodd_Arg_0"] = "valoare";
        this["Func_isodd_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_isodd_Category"] = "Informație";
        this["Func_isodd_CategoryURL"] = "Informație";
        this["Func_isodd_Desc"] = "Returnează true dacă valoarea este un număr impar, altfel returnează false.";
        this["Func_isref_Arg_0"] = "valoare";
        this["Func_isref_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_isref_Category"] = "Informație";
        this["Func_isref_CategoryURL"] = "Informație";
        this["Func_isref_Desc"] = "Returnează adevărat dacă valoarea este o referință.";
        this["Func_istext_Arg_0"] = "valoare";
        this["Func_istext_ArgDesc_0"] = "Valoarea de testat.";
        this["Func_istext_Category"] = "Informație";
        this["Func_istext_CategoryURL"] = "Informație";
        this["Func_istext_Desc"] = "Returnează adevărat dacă valoarea este o valoare text.";
        this["Func_lcm_Arg_0"] = "numărul {0}";
        this["Func_lcm_ArgDesc_0"] = "O valoare întreagă. O valoare neîntreagă va fi trunchiată.";
        this["Func_lcm_Category"] = "Matematica";
        this["Func_lcm_CategoryURL"] = "Matematica";
        this["Func_lcm_Desc"] = "Returnează cel mai mic multiplu comun al uneia sau mai multor valori întregi.";
        this["Func_left_Arg_0"] = "text";
        this["Func_left_Arg_1"] = "num_chars";
        this["Func_left_ArgDesc_0"] = "Expresia text care conține caracterele de extras.";
        this["Func_left_ArgDesc_1"] = "Numărul de caractere de extras.";
        this["Func_left_Category"] = "TextAndData";
        this["Func_left_CategoryURL"] = "Text_and_Data";
        this["Func_left_Desc"] = "Returnează primul sau primele caractere dintr-o expresie text.";
        this["Func_len_Arg_0"] = "text";
        this["Func_len_ArgDesc_0"] = "Expresia de text a cărei lungime trebuie găsită. Spațiile se numără caractere.";
        this["Func_len_Category"] = "TextAndData";
        this["Func_len_CategoryURL"] = "Text_and_Data";
        this["Func_len_Desc"] = "Returnează numărul de caractere dintr-o expresie text.";
        this["Func_ln_Arg_0"] = "număr";
        this["Func_ln_ArgDesc_0"] = "Numărul real pozitiv pentru care se obține logaritmul natural.";
        this["Func_ln_Category"] = "Matematica";
        this["Func_ln_CategoryURL"] = "Matematica";
        this["Func_ln_Desc"] = "Returnează logaritmul natural al unui număr.";
        this["Func_log_Arg_0"] = "număr";
        this["Func_log_Arg_1"] = "baza";
        this["Func_log_ArgDesc_0"] = "Numărul real pozitiv pentru care se obține logaritmul.";
        this["Func_log_ArgDesc_1"] = "Baza logaritmului.";
        this["Func_log_Category"] = "Matematica";
        this["Func_log_CategoryURL"] = "Matematica";
        this["Func_log_Desc"] = "Returnează logaritmul unui număr la baza specificată.";
        this["Func_log10_Arg_0"] = "număr";
        this["Func_log10_ArgDesc_0"] = "Numărul real pozitiv pentru care se obține logaritmul de bază-10.";
        this["Func_log10_Category"] = "Matematica";
        this["Func_log10_CategoryURL"] = "Matematica";
        this["Func_log10_Desc"] = "Returnează logaritmul de bază-10 al unui număr.";
        this["Func_lower_Arg_0"] = "text";
        this["Func_lower_ArgDesc_0"] = "Textul de convertit în minuscule.";
        this["Func_lower_Category"] = "TextAndData";
        this["Func_lower_CategoryURL"] = "Text_and_Data";
        this["Func_lower_Desc"] = "Convertește toate literele majuscule dintr-o expresie text în minuscule.";
        this["Func_max_Arg_0"] = "numărul {0}";
        this["Func_max_ArgDesc_0"] = "Un număr sau o gamă de numere.";
        this["Func_max_Category"] = "Statistic";
        this["Func_max_CategoryURL"] = "Statistic";
        this["Func_max_Desc"] = "Returnează cel mai mare număr dintr-un set de numere.";
        this["Func_median_Arg_0"] = "numărul {0}";
        this["Func_median_ArgDesc_0"] = "Un număr sau o gamă de numere.";
        this["Func_median_Category"] = "Statistic";
        this["Func_median_CategoryURL"] = "Statistic";
        this["Func_median_Desc"] = "Returnează mediana unui set de numere. Mediana este numărul de mijloc atunci când numerele sunt sortate sau media celor două numere de mijloc dacă există un număr par de numere specificate.";
        this["Func_mid_Arg_0"] = "text";
        this["Func_mid_Arg_1"] = "start_num";
        this["Func_mid_Arg_2"] = "num_chars";
        this["Func_mid_ArgDesc_0"] = "Expresia text care conține caracterele de extras.";
        this["Func_mid_ArgDesc_1"] = "Indexul bazat pe 1 al primului caracter extras.";
        this["Func_mid_ArgDesc_2"] = "Numărul de caractere de extras.";
        this["Func_mid_Category"] = "TextAndData";
        this["Func_mid_CategoryURL"] = "Text_and_Data";
        this["Func_mid_Desc"] = "Returnează un anumit număr de caractere dintr-o expresie text.";
        this["Func_min_Arg_0"] = "numărul {0}";
        this["Func_min_ArgDesc_0"] = "Un număr sau o gamă de numere.";
        this["Func_min_Category"] = "Statistic";
        this["Func_min_CategoryURL"] = "Statistic";
        this["Func_min_Desc"] = "Returnează cel mai mic număr dintr-un set de numere.";
        this["Func_minute_Arg_0"] = "Time_value";
        this["Func_minute_ArgDesc_0"] = "O valoare de timp din care să obțineți minutul.";
        this["Func_minute_Category"] = "Data si ora";
        this["Func_minute_CategoryURL"] = "Date_and_Time";
        this["Func_minute_Desc"] = "Returnează minutul unei valori de timp.";
        this["Func_mod_Arg_0"] = "număr";
        this["Func_mod_Arg_1"] = "divizor";
        this["Func_mod_ArgDesc_0"] = "Numărul pentru care se găsește restul.";
        this["Func_mod_ArgDesc_1"] = "Valoarea cu care se împarte numărul.";
        this["Func_mod_Category"] = "Matematica";
        this["Func_mod_CategoryURL"] = "Matematica";
        this["Func_mod_Desc"] = "Returnează restul după ce un număr este împărțit la un divizor. Rezultatul are același semn cu divizorul.";
        this["Func_month_Arg_0"] = "Date_value";
        this["Func_month_ArgDesc_0"] = "O valoare de dată din care să obțineți luna.";
        this["Func_month_Category"] = "Data si ora";
        this["Func_month_CategoryURL"] = "Date_and_Time";
        this["Func_month_Desc"] = "Returnează luna valorii datei.";
        this["Func_mround_Arg_0"] = "Număr";
        this["Func_mround_Arg_1"] = "Multiplu";
        this["Func_mround_ArgDesc_0"] = "Numărul care trebuie rotunjit.";
        this["Func_mround_ArgDesc_1"] = "Un număr care indică multiplu la care se rotunjește.";
        this["Func_mround_Category"] = "Matematica";
        this["Func_mround_CategoryURL"] = "Matematica";
        this["Func_mround_Desc"] = "Rotunjește un număr la cel mai apropiat multiplu al unui alt număr.";
        this["Func_multinomial_Arg_0"] = "numărul {0}";
        this["Func_multinomial_ArgDesc_0"] = "Unul dintre numerele pentru care se obține multinomialul.";
        this["Func_multinomial_Category"] = "Matematica";
        this["Func_multinomial_CategoryURL"] = "Matematica";
        this["Func_multinomial_Desc"] = "Returnează multinomialul unui set de numere. Multinomialul este raportul dintre factorialul unei sume de valori și produsul factorialelor lor.";
        this["Func_n_Arg_0"] = "Valoare";
        this["Func_n_ArgDesc_0"] = "Valoarea care urmează să fie convertită într-un număr. Următoarele sunt valori acceptabile:\r\nUn număr - returnează numărul.\r\nO dată - returnează data.\r\nAdevărat - returnează 1.\r\nFals - returnează 0.\r\nO valoare de eroare - returnează valoarea de eroare.\r\nOrice altă valoare - returnează 0.\r\n    ";
        this["Func_n_Category"] = "Informație";
        this["Func_n_CategoryURL"] = "Informație";
        this["Func_n_Desc"] = "Convertește o valoare într-un număr.";
        this["Func_na_Category"] = "Informație";
        this["Func_na_CategoryURL"] = "Informație";
        this["Func_na_Desc"] = "Returnează valoarea de eroare #N/A (valoarea nu este disponibilă).";
        this["Func_networkdays_Arg_0"] = "startDate";
        this["Func_networkdays_Arg_1"] = "endDate";
        this["Func_networkdays_Arg_2"] = "sărbători";
        this["Func_networkdays_ArgDesc_0"] = "Data de la care să înceapă.";
        this["Func_networkdays_ArgDesc_1"] = "Data la care trebuie să se încheie.";
        this["Func_networkdays_ArgDesc_2"] = "O gamă de una sau mai multe referințe care conțin date care trebuie considerate zile nelucrătoare.";
        this["Func_networkdays_Category"] = "Data si ora";
        this["Func_networkdays_CategoryURL"] = "Data si ora";
        this["Func_networkdays_Desc"] = "Returnează numărul total de zile întregi de lucru între datele specificate, excluzând orice sărbători specificate.";
        this["Func_not_Arg_0"] = "Logic";
        this["Func_not_ArgDesc_0"] = "O valoare logică (adevărată sau falsă).";
        this["Func_not_Category"] = "Logic";
        this["Func_not_CategoryURL"] = "Logic";
        this["Func_not_Desc"] = "Inversează o valoare logică (se schimbă adevărat în fals și fals în adevărat).";
        this["Func_now_Category"] = "Data si ora";
        this["Func_now_CategoryURL"] = "Date_and_Time";
        this["Func_now_Desc"] = "Returnează o valoare care reprezintă data și ora curente.";
        this["Func_nper_Arg_0"] = "rată";
        this["Func_nper_Arg_1"] = "pmt";
        this["Func_nper_Arg_2"] = "pv";
        this["Func_nper_Arg_3"] = "fv";
        this["Func_nper_Arg_4"] = "type";
        this["Func_nper_ArgDesc_0"] = "Rata dobânzii pe perioadă.";
        this["Func_nper_ArgDesc_1"] = "Plata efectuată în fiecare perioadă, care nu se poate modifica pe durata de viață a anuității.";
        this["Func_nper_ArgDesc_2"] = "Valoarea actuală sau suma forfetară pe care o valoare o serie de plăți viitoare o merită acum.";
        this["Func_nper_ArgDesc_3"] = "Valoarea viitoare sau un sold de numerar pe care doriți să îl obțineți după efectuarea ultimei plăți.";
        this["Func_nper_ArgDesc_4"] = "Numărul zero sau unul care indică dacă plățile sunt datorate la sfârșitul perioadei (zero, valoarea implicită) sau la începutul (una).";
        this["Func_nper_Category"] = "Financiar";
        this["Func_nper_CategoryURL"] = "Financiar";
        this["Func_nper_Desc"] = "Returnează numărul de perioade pentru o investiție pe baza plăților periodice, constante și a unei rate a dobânzii constante.";
        this["Func_npv_Arg_0"] = "rată";
        this["Func_npv_Arg_1"] = "valoare {0}";
        this["Func_npv_ArgDesc_0"] = "Rata de reducere pe durata unei perioade.";
        this["Func_npv_ArgDesc_1"] = "O plată (valoare negativă) sau un venit (valoare pozitivă).";
        this["Func_npv_Category"] = "Financiar";
        this["Func_npv_CategoryURL"] = "Financiar";
        this["Func_npv_Desc"] = "Calculează valoarea actuală netă a unei investiții utilizând o rată de actualizare și o serie de plăți viitoare (valori negative) și venituri (valoare pozitivă).";
        this["Func_null_Category"] = "Informație";
        this["Func_null_CategoryURL"] = "Informație";
        this["Func_null_Desc"] = "Returnează o valoare inexistentă.";
        this["Func_oct2bin_Arg_0"] = "Număr";
        this["Func_oct2bin_Arg_1"] = "Locuri";
        this["Func_oct2bin_ArgDesc_0"] = "O expresie text care reprezintă un număr octal.";
        this["Func_oct2bin_ArgDesc_1"] = "Numărul de caractere de utilizat pentru reprezentarea valorii. Dacă acest lucru nu este specificat, va fi utilizat numărul minim de caractere necesar pentru a reprezenta valoarea. Acest lucru vă permite să specificați zerouri de la început.";
        this["Func_oct2bin_Category"] = "Inginerie";
        this["Func_oct2bin_CategoryURL"] = "Inginerie";
        this["Func_oct2bin_Desc"] = "Convertește o expresie text care reprezintă un număr octal într-o expresie text care reprezintă același număr în sistemul numeric binar.";
        this["Func_oct2dec_Arg_0"] = "Număr";
        this["Func_oct2dec_ArgDesc_0"] = "O expresie text care reprezintă un număr octal.";
        this["Func_oct2dec_Category"] = "Inginerie";
        this["Func_oct2dec_CategoryURL"] = "Inginerie";
        this["Func_oct2dec_Desc"] = "Returnează valoarea zecimală a unei reprezentări text a unui număr în sistemul numeric octal.";
        this["Func_oct2hex_Arg_0"] = "Număr";
        this["Func_oct2hex_Arg_1"] = "Locuri";
        this["Func_oct2hex_ArgDesc_0"] = "O expresie text care reprezintă un număr octal.";
        this["Func_oct2hex_ArgDesc_1"] = "Numărul de caractere de utilizat pentru reprezentarea valorii. Dacă acest lucru nu este specificat, va fi utilizat numărul minim de caractere necesar pentru a reprezenta valoarea. Acest lucru vă permite să specificați zerouri de la început.";
        this["Func_oct2hex_Category"] = "Inginerie";
        this["Func_oct2hex_CategoryURL"] = "Inginerie";
        this["Func_oct2hex_Desc"] = "Convertește o expresie text care reprezintă un număr octal într-o expresie text care reprezintă același număr în sistemul numeric hexazecimal.";
        this["Func_odd_Arg_0"] = "număr";
        this["Func_odd_ArgDesc_0"] = "Valoarea de rotunjit.";
        this["Func_odd_Category"] = "Matematica";
        this["Func_odd_CategoryURL"] = "Matematica";
        this["Func_odd_Desc"] = "Rotunjește un număr pozitiv în sus și un număr negativ în jos la cel mai apropiat număr impar.";
        this["Func_or_Arg_0"] = "{0} logic";
        this["Func_or_ArgDesc_0"] = "O valoare logică (adevărată sau falsă).";
        this["Func_or_Category"] = "Logic";
        this["Func_or_CategoryURL"] = "Logic";
        this["Func_or_Desc"] = "Returnează adevărat dacă una sau mai multe valori sunt adevărate; returnează false dacă toate valorile sunt false.";
        this["Func_pi_Category"] = "Matematica";
        this["Func_pi_CategoryURL"] = "Matematica";
        this["Func_pi_Desc"] = "Returnează constanta pi, care este 3.14159265358979.";
        this["Func_pmt_Arg_0"] = "rată";
        this["Func_pmt_Arg_1"] = "nper";
        this["Func_pmt_Arg_2"] = "pv";
        this["Func_pmt_Arg_3"] = "fv";
        this["Func_pmt_Arg_4"] = "type";
        this["Func_pmt_ArgDesc_0"] = "Rata dobânzii pentru împrumut.";
        this["Func_pmt_ArgDesc_1"] = "Numărul total de plăți pentru împrumut.";
        this["Func_pmt_ArgDesc_2"] = "Valoarea actuală sau suma totală pe care o valorează acum o serie de plăți viitoare; cunoscut și sub numele de director.";
        this["Func_pmt_ArgDesc_3"] = "Valoarea viitoare sau un sold de numerar pe care doriți să îl obțineți după efectuarea ultimei plăți.";
        this["Func_pmt_ArgDesc_4"] = "Numărul zero sau unul care indică dacă plățile sunt datorate la sfârșitul perioadei (zero, valoarea implicită) sau la începutul (una).";
        this["Func_pmt_Category"] = "Financiar";
        this["Func_pmt_CategoryURL"] = "Financiar";
        this["Func_pmt_Desc"] = "Calculează plata pentru un împrumut pe baza plăților constante și a unei rate a dobânzii constante.";
        this["Func_power_Arg_0"] = "număr";
        this["Func_power_Arg_1"] = "putere";
        this["Func_power_ArgDesc_0"] = "Numărul de bază de ridicat la o putere.";
        this["Func_power_ArgDesc_1"] = "Exponentul la care este ridicat numărul de bază.";
        this["Func_power_Category"] = "Matematica";
        this["Func_power_CategoryURL"] = "Matematica";
        this["Func_power_Desc"] = "Returnează rezultatul unui număr ridicat la o putere.";
        this["Func_ppmt_Arg_0"] = "rată";
        this["Func_ppmt_Arg_1"] = "per";
        this["Func_ppmt_Arg_2"] = "nper";
        this["Func_ppmt_Arg_3"] = "pv";
        this["Func_ppmt_Arg_4"] = "fv";
        this["Func_ppmt_Arg_5"] = "type";
        this["Func_ppmt_ArgDesc_0"] = "Rata dobânzii pe perioadă.";
        this["Func_ppmt_ArgDesc_1"] = "Perioada, care trebuie să fie cuprinsă între 1 și nper.";
        this["Func_ppmt_ArgDesc_2"] = "Numărul total de perioade de plată dintr-o anuitate.";
        this["Func_ppmt_ArgDesc_3"] = "Valoarea actuală sau suma totală pe care o valorează acum o serie de plăți viitoare.";
        this["Func_ppmt_ArgDesc_4"] = "Valoarea viitoare sau un sold de numerar pe care doriți să îl obțineți după efectuarea ultimei plăți.";
        this["Func_ppmt_ArgDesc_5"] = "Numărul zero sau unul care indică dacă plățile sunt datorate la sfârșitul perioadei (zero, valoarea implicită) sau la începutul (una).";
        this["Func_ppmt_Category"] = "Financiar";
        this["Func_ppmt_CategoryURL"] = "Financiar";
        this["Func_ppmt_Desc"] = "Returnează plata principalului pentru o perioadă dată pentru o investiție bazată pe plăți periodice, constante și o rată a dobânzii constantă.";
        this["Func_product_Arg_0"] = "numărul {0}";
        this["Func_product_ArgDesc_0"] = "Un număr de înmulțit cu celelalte numere.";
        this["Func_product_Category"] = "Matematica";
        this["Func_product_CategoryURL"] = "Matematica";
        this["Func_product_Desc"] = "Multiplică toate numerele date ca argumente și returnează produsul.";
        this["Func_pv_Arg_0"] = "rată";
        this["Func_pv_Arg_1"] = "nper";
        this["Func_pv_Arg_2"] = "pmt";
        this["Func_pv_Arg_3"] = "fv";
        this["Func_pv_Arg_4"] = "type";
        this["Func_pv_ArgDesc_0"] = "Rata dobânzii pe perioadă.";
        this["Func_pv_ArgDesc_1"] = "Numărul total de perioade de plată dintr-o anuitate.";
        this["Func_pv_ArgDesc_2"] = "Plata efectuată în fiecare perioadă, care nu se poate modifica pe durata de viață a anuității.";
        this["Func_pv_ArgDesc_3"] = "Valoarea viitoare sau un sold de numerar pe care doriți să îl obțineți după efectuarea ultimei plăți.";
        this["Func_pv_ArgDesc_4"] = "Numărul zero sau unul care indică dacă plățile sunt datorate la sfârșitul perioadei (zero, valoarea implicită) sau la începutul (una).";
        this["Func_pv_Category"] = "Financiar";
        this["Func_pv_CategoryURL"] = "Financiar";
        this["Func_pv_Desc"] = "Returnează valoarea actuală a unei investiții.";
        this["Func_quotient_Arg_0"] = "numărător";
        this["Func_quotient_Arg_1"] = "numitor";
        this["Func_quotient_ArgDesc_0"] = "Dividendul operației de divizare.";
        this["Func_quotient_ArgDesc_1"] = "Divizorul operației de divizare.";
        this["Func_quotient_Category"] = "Matematica";
        this["Func_quotient_CategoryURL"] = "Matematica";
        this["Func_quotient_Desc"] = "Returnează porțiunea întreagă a unei diviziuni. Acest lucru va elimina restul diviziei.";
        this["Func_radians_Arg_0"] = "Grade";
        this["Func_radians_ArgDesc_0"] = "O valoare în grade care va fi convertită în radiani.";
        this["Func_radians_Category"] = "Matematica";
        this["Func_radians_CategoryURL"] = "Matematica";
        this["Func_radians_Desc"] = "Convertește grade în radiani.";
        this["Func_rand_Category"] = "Matematica";
        this["Func_rand_CategoryURL"] = "Matematica";
        this["Func_rand_Desc"] = "Returnează un număr aleatoriu distribuit uniform mai mare sau egal cu 0 și mai mic decât 1";
        this["Func_randbetween_Arg_0"] = "Jos";
        this["Func_randbetween_Arg_1"] = "Sus";
        this["Func_randbetween_ArgDesc_0"] = "Valoarea minimă care va fi returnată.";
        this["Func_randbetween_ArgDesc_1"] = "Valoarea maximă care va fi returnată.";
        this["Func_randbetween_Category"] = "Matematica";
        this["Func_randbetween_CategoryURL"] = "Matematica";
        this["Func_randbetween_Desc"] = "Generează un întreg pseudoaleator între două numere specificate.";
        this["Func_rate_Arg_0"] = "NPer";
        this["Func_rate_Arg_1"] = "Pmt";
        this["Func_rate_Arg_2"] = "Pv";
        this["Func_rate_Arg_3"] = "Fv";
        this["Func_rate_Arg_4"] = "Tip";
        this["Func_rate_Arg_5"] = "Guess";
        this["Func_rate_ArgDesc_0"] = "Numărul total de perioade de plată dintr-o anuitate.";
        this["Func_rate_ArgDesc_1"] = "Plata a fost efectuată în fiecare perioadă și nu se poate modifica pe durata de viață a anuității.";
        this["Func_rate_ArgDesc_2"] = "Valoarea actuală sau suma totală pe care o valorează acum o serie de plăți viitoare.";
        this["Func_rate_ArgDesc_3"] = "Valoarea viitoare sau un sold de numerar pe care doriți să îl obțineți după efectuarea ultimei plăți. Dacă fv este omis, se presupune că este 0 (valoarea viitoare a unui împrumut, de exemplu, este 0). De exemplu, dacă doriți să economisiți 50.000 USD pentru a plăti pentru un proiect special în 18 ani, atunci 50.000 USD este valoarea viitoare. Ați putea face apoi o presupunere conservatoare la o rată a dobânzii și să determinați cât de mult trebuie să economisiți în fiecare lună. Dacă fv este omis, trebuie să includeți argumentul pmt. Valoarea implicită pentru fv este zero.";
        this["Func_rate_ArgDesc_4"] = "Numărul zero sau unul care indică dacă plățile sunt datorate la sfârșitul perioadei (zero, valoarea implicită) sau la începutul (una).";
        this["Func_rate_ArgDesc_5"] = "Un număr despre care ghiciți că este aproape de rezultat. Dacă este angajat, se presupune că este 0,1 (10%).";
        this["Func_rate_Category"] = "Financiar";
        this["Func_rate_CategoryURL"] = "Financiar";
        this["Func_rate_Desc"] = "Returnează rata dobânzii pe perioadă a unei anuități.";
        this["Func_replace_Arg_0"] = "Old_text";
        this["Func_replace_Arg_1"] = "Start_num";
        this["Func_replace_Arg_2"] = "Num_chars";
        this["Func_replace_Arg_3"] = "New_text";
        this["Func_replace_ArgDesc_0"] = "Expresia textului în care să înlocuiți unele caractere.";
        this["Func_replace_ArgDesc_1"] = "Indexul bazat pe 1 al caracterului din text_vechi pentru a fi înlocuit cu new_text.";
        this["Func_replace_ArgDesc_2"] = "Numărul de caractere din textul_vechi de înlocuit cu textul_nou.";
        this["Func_replace_ArgDesc_3"] = "Textul care va înlocui caracterele din textul_vechi.";
        this["Func_replace_Category"] = "TextAndData";
        this["Func_replace_CategoryURL"] = "Text_and_Data";
        this["Func_replace_Desc"] = "Înlocuiește o parte a unei expresii de text.";
        this["Func_rept_Arg_0"] = "TextValue";
        this["Func_rept_Arg_1"] = "RepeatCount";
        this["Func_rept_ArgDesc_0"] = "Textul de repetat.";
        this["Func_rept_ArgDesc_1"] = "Numărul pozitiv de repetări a textului.";
        this["Func_rept_Category"] = "TextAndData";
        this["Func_rept_CategoryURL"] = "Text_and_Data";
        this["Func_rept_Desc"] = "Repetă textul de un anumit număr de ori.";
        this["Func_right_Arg_0"] = "text";
        this["Func_right_Arg_1"] = "num_chars";
        this["Func_right_ArgDesc_0"] = "Expresia text care conține caracterele de extras.";
        this["Func_right_ArgDesc_1"] = "Numărul de caractere de extras.";
        this["Func_right_Category"] = "TextAndData";
        this["Func_right_CategoryURL"] = "Text_and_Data";
        this["Func_right_Desc"] = "Returnează ultimul sau ultimele caractere dintr-o expresie text.";
        this["Func_roman_Arg_0"] = "Număr";
        this["Func_roman_Arg_1"] = "Formă";
        this["Func_roman_ArgDesc_0"] = "Numărul de convertit în cifre romane.";
        this["Func_roman_ArgDesc_1"] = "Un număr sau o valoare booleană care indică dacă se utilizează cifre romane clasice sau o versiune mai concisă.\r\n0 = (implicit) Clasic. (499 = \" CDXCIX \")\r\n1 = Mai concis. (499 = \" LDVLIV \")\r\n2 = Mai concis. (499 = \" XDIX \")\r\n3 = Mai concis. (499 = \" VDIV \")\r\n4 = Simplificat. (499 = \" ID \")\r\nAdevărat = Clasic.\r\nFals = Simplificat.\r\n    ";
        this["Func_roman_Category"] = "Matematica";
        this["Func_roman_CategoryURL"] = "Matematica";
        this["Func_roman_Desc"] = "Convertește un număr într-un număr roman ca expresie text.";
        this["Func_round_Arg_0"] = "număr";
        this["Func_round_Arg_1"] = "num_digits";
        this["Func_round_Arg_2"] = "round_Style";
        this["Func_round_ArgDesc_0"] = "Numărul de rotunjit.";
        this["Func_round_ArgDesc_1"] = "Numărul de cifre la care se rotunjește numărul.";
        this["Func_round_ArgDesc_2"] = "Un număr care indică stilul de rotunjire de utilizat.\r\n0 - Rotunjire aritmetică (implicit) - Dacă ultima cifră semnificativă din rotunjire este 5, numărul este rotunjit în sus.\r\n1 - Rotunjirea bancherului - Când ultima cifră semnificativă este 5, cifra precedentă este utilizată pentru a determina dacă se rotunjește în sus sau în jos. Dacă cifra este pară, numărul este rotunjit în jos. Dacă este impar, atunci numărul este rotunjit în sus.\r\n    ";
        this["Func_round_Category"] = "Matematica";
        this["Func_round_CategoryURL"] = "Matematica";
        this["Func_round_Desc"] = "Rotunjește un număr la un număr specificat de cifre.";
        this["Func_rounddown_Arg_0"] = "Număr";
        this["Func_rounddown_Arg_1"] = "Cifre";
        this["Func_rounddown_ArgDesc_0"] = "Numărul care trebuie rotunjit în jos.";
        this["Func_rounddown_ArgDesc_1"] = "Numărul de zecimale de rotunjit. Numerele pozitive indică locurile după punctul zecimal, numerele negative indică locurile înainte de punctul zecimal.";
        this["Func_rounddown_Category"] = "Matematica";
        this["Func_rounddown_CategoryURL"] = "Matematica";
        this["Func_rounddown_Desc"] = "Rotunjește un număr până la numărul specificat de cifre.";
        this["Func_roundup_Arg_0"] = "Număr";
        this["Func_roundup_Arg_1"] = "Cifre";
        this["Func_roundup_ArgDesc_0"] = "Numărul care trebuie rotunjit în sus.";
        this["Func_roundup_ArgDesc_1"] = "Numărul de zecimale de rotunjit. Numerele pozitive indică locurile după punctul zecimal, numerele negative indică locurile înainte de punctul zecimal.";
        this["Func_roundup_Category"] = "Matematica";
        this["Func_roundup_CategoryURL"] = "Matematica";
        this["Func_roundup_Desc"] = "Rotunjește un număr până la numărul specificat de cifre.";
        this["Func_search_Arg_0"] = "search_text";
        this["Func_search_Arg_1"] = "valoare";
        this["Func_search_Arg_2"] = "start_index";
        this["Func_search_ArgDesc_0"] = "Textul de găsit în valoare.";
        this["Func_search_ArgDesc_1"] = "Textul în care să găsiți search_text.";
        this["Func_search_ArgDesc_2"] = "Indexul bazat pe 1 în care să începeți căutarea (dacă este omis, start_index este 1).";
        this["Func_search_Category"] = "TextAndData";
        this["Func_search_CategoryURL"] = "TextAndData";
        this["Func_search_Desc"] = "Returnează indexul bazat pe 1 al unui șir în altul, căutând insensibil majuscule.";
        this["Func_searchb_Arg_0"] = "search_text";
        this["Func_searchb_Arg_1"] = "valoare";
        this["Func_searchb_Arg_2"] = "start_index";
        this["Func_searchb_ArgDesc_0"] = "Textul de găsit în valoare.";
        this["Func_searchb_ArgDesc_1"] = "Textul în care să găsiți search_text.";
        this["Func_searchb_ArgDesc_2"] = "Indexul bazat pe 1 în care să începeți căutarea (dacă este omis, start_index este 1).";
        this["Func_searchb_Category"] = "TextAndData";
        this["Func_searchb_CategoryURL"] = "TextAndData";
        this["Func_searchb_Desc"] = "Returnează indexul bazat pe 1 al unui șir în altul, căutând insensibil majuscule.";
        this["Func_second_Arg_0"] = "Time_value";
        this["Func_second_ArgDesc_0"] = "O valoare de timp din care să obțineți a doua.";
        this["Func_second_Category"] = "Data si ora";
        this["Func_second_CategoryURL"] = "Date_and_Time";
        this["Func_second_Desc"] = "Returnează secunda unei valori de timp.";
        this["Func_seriessum_Arg_0"] = "InputValue";
        this["Func_seriessum_Arg_1"] = "InitialPower";
        this["Func_seriessum_Arg_2"] = "Pas";
        this["Func_seriessum_Arg_3"] = "Coeficienți";
        this["Func_seriessum_ArgDesc_0"] = "Valoarea de intrare pentru seria de putere.";
        this["Func_seriessum_ArgDesc_1"] = "Puterea inițială la care va fi ridicată Valoarea de intrare.";
        this["Func_seriessum_ArgDesc_2"] = "Pasul care va fi folosit pentru a crește puterea pentru fiecare termen din serie.";
        this["Func_seriessum_ArgDesc_3"] = "Un set de coeficienți prin care fiecare putere succesivă a InputValue este multiplicată.";
        this["Func_seriessum_Category"] = "Matematica";
        this["Func_seriessum_CategoryURL"] = "Matematica";
        this["Func_seriessum_Desc"] = "Returnează suma unei serii de putere.";
        this["Func_sign_Arg_0"] = "Număr";
        this["Func_sign_ArgDesc_0"] = "Numărul al cărui semn va fi returnat.";
        this["Func_sign_Category"] = "Matematica";
        this["Func_sign_CategoryURL"] = "Matematica";
        this["Func_sign_Desc"] = "Returnează semnul unui număr (-1, 0 sau 1).";
        this["Func_sin_Arg_0"] = "număr";
        this["Func_sin_ArgDesc_0"] = "Unghiul în radiani pentru care se obține sinusul.";
        this["Func_sin_Category"] = "Matematica";
        this["Func_sin_CategoryURL"] = "Matematica";
        this["Func_sin_Desc"] = "Returnează sinusul unghiului specificat.";
        this["Func_sinh_Arg_0"] = "număr";
        this["Func_sinh_ArgDesc_0"] = "Un număr real din care se obține sinusul hiperbolic.";
        this["Func_sinh_Category"] = "Matematica";
        this["Func_sinh_CategoryURL"] = "Matematica";
        this["Func_sinh_Desc"] = "Returnează sinusul hiperbolic al unui număr.";
        this["Func_sln_Arg_0"] = "cost";
        this["Func_sln_Arg_1"] = "salvare";
        this["Func_sln_Arg_2"] = "viaţă";
        this["Func_sln_ArgDesc_0"] = "Costul inițial al activului.";
        this["Func_sln_ArgDesc_1"] = "Valoarea la sfârșitul amortizării.";
        this["Func_sln_ArgDesc_2"] = "Numărul de perioade în care activul este amortizat.";
        this["Func_sln_Category"] = "Financiar";
        this["Func_sln_CategoryURL"] = "Financiar";
        this["Func_sln_Desc"] = "Returnează amortizarea liniară a unui activ pentru o perioadă.";
        this["Func_sqrt_Arg_0"] = "număr";
        this["Func_sqrt_ArgDesc_0"] = "Numărul pentru care se obține rădăcina pătrată.";
        this["Func_sqrt_Category"] = "Matematica";
        this["Func_sqrt_CategoryURL"] = "Matematica";
        this["Func_sqrt_Desc"] = "Returnează o rădăcină pătrată pozitivă a unui număr.";
        this["Func_sqrtpi_Arg_0"] = "Număr";
        this["Func_sqrtpi_ArgDesc_0"] = "Numărul care trebuie înmulțit cu pi.";
        this["Func_sqrtpi_Category"] = "Matematica";
        this["Func_sqrtpi_CategoryURL"] = "Matematica";
        this["Func_sqrtpi_Desc"] = "Multiplică numărul specificat cu pi și returnează rădăcina pătrată a rezultatului.";
        this["Func_stdev_Arg_0"] = "numărul {0}";
        this["Func_stdev_ArgDesc_0"] = "Un număr sau o gamă de numere care specifică un eșantion dintr-o populație.";
        this["Func_stdev_Category"] = "Statistic";
        this["Func_stdev_CategoryURL"] = "Statistic";
        this["Func_stdev_Desc"] = "Returnează abaterea standard a unui eșantion de populație.";
        this["Func_subtotal_Arg_0"] = "FunctionNum";
        this["Func_subtotal_Arg_1"] = "Ref {0}";
        this["Func_subtotal_ArgDesc_0"] = "Un număr care reprezintă funcția de calculat. Adăugați 100 la valoare pentru a ignora valorile ascunse.\r\n1 (sau 101) - MEDIE\r\n2 (sau 102) - CONT\r\n4 (sau 104) - MAX\r\n5 (sau 105) - MIN\r\n6 (sau 106) - PRODUS\r\n7 (sau 107) - STDEV\r\n9 (sau 109) - SUMĂ\r\n10 (sau 110) - VAR\r\n    ";
        this["Func_subtotal_ArgDesc_1"] = "Una sau mai multe referințe pentru care să găsiți subtotalul.";
        this["Func_subtotal_Category"] = "Matematica";
        this["Func_subtotal_CategoryURL"] = "Matematica";
        this["Func_subtotal_Desc"] = "Calculează subtotalul într-una din mai multe referințe.";
        this["Func_sum_Arg_0"] = "numărul {0}";
        this["Func_sum_ArgDesc_0"] = "Un număr de adăugat la celelalte numere.";
        this["Func_sum_Category"] = "Matematica";
        this["Func_sum_CategoryURL"] = "Matematica";
        this["Func_sum_Desc"] = "Sumează toate numerele date ca argumente și returnează suma.";
        this["Func_syd_Arg_0"] = "cost";
        this["Func_syd_Arg_1"] = "salvare";
        this["Func_syd_Arg_2"] = "viaţă";
        this["Func_syd_Arg_3"] = "per";
        this["Func_syd_ArgDesc_0"] = "Costul inițial al activului.";
        this["Func_syd_ArgDesc_1"] = "Valoarea la sfârșitul amortizării.";
        this["Func_syd_ArgDesc_2"] = "Numărul de perioade în care activul este amortizat.";
        this["Func_syd_ArgDesc_3"] = "Perioada, care trebuie să utilizeze aceleași unități ca și viața.";
        this["Func_syd_Category"] = "Financiar";
        this["Func_syd_CategoryURL"] = "Financiar";
        this["Func_syd_Desc"] = "Returnează suma amortizării cifrelor unui an pentru o perioadă specificată.";
        this["Func_tan_Arg_0"] = "număr";
        this["Func_tan_ArgDesc_0"] = "Unghiul, în radiani, pentru care se obține tangenta.";
        this["Func_tan_Category"] = "Matematica";
        this["Func_tan_CategoryURL"] = "Matematica";
        this["Func_tan_Desc"] = "Returnează tangenta unghiului specificat.";
        this["Func_tanh_Arg_0"] = "număr";
        this["Func_tanh_ArgDesc_0"] = "Unghiul, în radiani, pentru care se obține tangenta hiperbolică.";
        this["Func_tanh_Category"] = "Matematica";
        this["Func_tanh_CategoryURL"] = "Matematica";
        this["Func_tanh_Desc"] = "Returnează tangenta hiperbolică a unghiului specificat.";
        this["Func_text_Arg_0"] = "Valoare";
        this["Func_text_Arg_1"] = "FormatText";
        this["Func_text_ArgDesc_0"] = "Valoarea numerică care ar trebui formatată.";
        this["Func_text_ArgDesc_1"] = "Formatul numeric cu care să formatați valoarea.";
        this["Func_text_Category"] = "TextAndData";
        this["Func_text_CategoryURL"] = "Text_and_Data";
        this["Func_text_Desc"] = "Convertește valoarea în text pe baza formatului specificat.";
        this["Func_time_Arg_0"] = "ora";
        this["Func_time_Arg_1"] = "minut";
        this["Func_time_Arg_2"] = "secundă";
        this["Func_time_ArgDesc_0"] = "Un număr care reprezintă ora timpului.";
        this["Func_time_ArgDesc_1"] = "Un număr care reprezintă minutul timpului.";
        this["Func_time_ArgDesc_2"] = "Un număr care reprezintă secunda timpului.";
        this["Func_time_Category"] = "Data si ora";
        this["Func_time_CategoryURL"] = "Date_and_Time";
        this["Func_time_Desc"] = "Returnează un număr care reprezintă o anumită valoare de timp.";
        this["Func_timevalue_Arg_0"] = "time_text";
        this["Func_timevalue_ArgDesc_0"] = "O expresie text care reprezintă un timp.";
        this["Func_timevalue_Category"] = "Data si ora";
        this["Func_timevalue_CategoryURL"] = "Date_and_Time";
        this["Func_timevalue_Desc"] = "Convertește o expresie text care reprezintă un timp la valoarea sa echivalentă a timpului.";
        this["Func_today_Category"] = "Data si ora";
        this["Func_today_CategoryURL"] = "Date_and_Time";
        this["Func_today_Desc"] = "Returnează data curentă.";
        this["Func_trim_Arg_0"] = "text";
        this["Func_trim_ArgDesc_0"] = "Textul din care să eliminați spațiile suplimentare.";
        this["Func_trim_Category"] = "TextAndData";
        this["Func_trim_CategoryURL"] = "Text_and_Data";
        this["Func_trim_Desc"] = "Elimină toate spațiile din text, cu excepția spațiilor unice dintre cuvinte.";
        this["Func_true_Category"] = "Logic";
        this["Func_true_CategoryURL"] = "Logic";
        this["Func_true_Desc"] = "Returnează valoarea logică adevărată.";
        this["Func_trunc_Arg_0"] = "număr";
        this["Func_trunc_Arg_1"] = "num_digits";
        this["Func_trunc_ArgDesc_0"] = "Numărul de trunchiat.";
        this["Func_trunc_ArgDesc_1"] = "Un număr care specifică precizia trunchierii. Valoarea implicită este zero.";
        this["Func_trunc_Category"] = "Matematica";
        this["Func_trunc_CategoryURL"] = "Matematica";
        this["Func_trunc_Desc"] = "Trunchiază un număr într-un număr întreg, eliminând partea fracțională a numărului.";
        this["Func_type_Arg_0"] = "valoare";
        this["Func_type_ArgDesc_0"] = "Valoarea căreia se obține tipul.";
        this["Func_type_Category"] = "Informație";
        this["Func_type_CategoryURL"] = "Informație";
        this["Func_type_Desc"] = "Returnează un număr reprezentând tipul de valoare: număr = 1; text = 2; valoare logică (adevărată sau falsă) = 4; valoarea erorii = 16; matrice = 64.";
        this["Func_upper_Arg_0"] = "text";
        this["Func_upper_ArgDesc_0"] = "Textul de convertit în majuscule.";
        this["Func_upper_Category"] = "TextAndData";
        this["Func_upper_CategoryURL"] = "Text_and_Data";
        this["Func_upper_Desc"] = "Convertește toate literele minuscule dintr-o expresie text în majuscule.";
        this["Func_value_Arg_0"] = "text";
        this["Func_value_ArgDesc_0"] = "Expresia de text de convertit.";
        this["Func_value_Category"] = "TextAndData";
        this["Func_value_CategoryURL"] = "Text_and_Data";
        this["Func_value_Desc"] = "Convertește o expresie text care reprezintă un număr într-un număr.";
        this["Func_var_Arg_0"] = "numărul {0}";
        this["Func_var_ArgDesc_0"] = "Un număr sau o gamă de numere care specifică un eșantion dintr-o populație.";
        this["Func_var_Category"] = "Statistic";
        this["Func_var_CategoryURL"] = "Statistic";
        this["Func_var_Desc"] = "Returnează varianța unui eșantion de populație.";
        this["Func_weekday_Arg_0"] = "Data";
        this["Func_weekday_Arg_1"] = "ReturnType";
        this["Func_weekday_ArgDesc_0"] = "Data pentru care se găsește numărul din ziua săptămânii.";
        this["Func_weekday_ArgDesc_1"] = "Determină schema de numerotare utilizată pentru zilele săptămânii.\r\n1 (implicit) = duminică (1) până sâmbătă (7)\r\n2 = luni (1) până duminică (7)\r\n3 = luni (0) până duminică (6)\r\n    ";
        this["Func_weekday_Category"] = "Data si ora";
        this["Func_weekday_CategoryURL"] = "Data si ora";
        this["Func_weekday_Desc"] = "Returnează un număr care reprezintă ziua săptămânii datei specificate.";
        this["Func_weeknum_Arg_0"] = "date";
        this["Func_weeknum_Arg_1"] = "returnType";
        this["Func_weeknum_ArgDesc_0"] = "Data pentru care se găsește numărul săptămânii.";
        this["Func_weeknum_ArgDesc_1"] = "Determină care zi începe săptămâna.\r\n1 (implicit) = Duminica este prima zi a săptămânii.\r\n2 = Luni este prima zi a săptămânii.\r\n    ";
        this["Func_weeknum_Category"] = "Data si ora";
        this["Func_weeknum_CategoryURL"] = "Data si ora";
        this["Func_weeknum_Desc"] = "Returnează numărul săptămânii în care se încadrează data specificată.";
        this["Func_workday_Arg_0"] = "startDate";
        this["Func_workday_Arg_1"] = "zile";
        this["Func_workday_Arg_2"] = "sărbători";
        this["Func_workday_ArgDesc_0"] = "Data de la care să înceapă.";
        this["Func_workday_ArgDesc_1"] = "Numărul de zile lucrătoare care trebuie compensate de la data de începere.";
        this["Func_workday_ArgDesc_2"] = "O gamă de una sau mai multe referințe care conțin date care trebuie considerate zile nelucrătoare.";
        this["Func_workday_Category"] = "Data si ora";
        this["Func_workday_CategoryURL"] = "Data si ora";
        this["Func_workday_Desc"] = "Returnează o zi lucrătoare numărul specificat de zile de la data specificată, excluzând orice sărbători specificate.";
        this["Func_year_Arg_0"] = "Date_value";
        this["Func_year_ArgDesc_0"] = "O valoare de dată din care să obțineți anul.";
        this["Func_year_Category"] = "Data si ora";
        this["Func_year_CategoryURL"] = "Date_and_Time";
        this["Func_year_Desc"] = "Returnează anul valorii datei.";
        this["Func_yearfrac_Arg_0"] = "StartDate";
        this["Func_yearfrac_Arg_1"] = "EndDate";
        this["Func_yearfrac_Arg_2"] = "bază";
        this["Func_yearfrac_ArgDesc_0"] = "Data de începere este data de începere din interval.";
        this["Func_yearfrac_ArgDesc_1"] = "Data de încheiere este data de încheiere din interval.";
        this["Func_yearfrac_ArgDesc_2"] = "\r\n      Baza determină baza pentru numărul de zile de utilizat.\r\n0 (Implicit = SUA (NASD) 30 / 360\r\n1 = Actual / actual\r\n2 = Actual / 360\r\n3 = Actual / 365\r\n4 = European 30 / 360\r\n    ";
        this["Func_yearfrac_Category"] = "Data si ora";
        this["Func_yearfrac_CategoryURL"] = "Data si ora";
        this["Func_yearfrac_Desc"] = "Returnează o fracțiune care reprezintă partea de an specificată între data de începere specificată și data de încheiere.";
        this["GenerateTableColumnName"] = "Coloană";
        this["GenerateTableName"] = "Tabel";
        this["LD_Chart_AxisDisplayUnits"] = "Sute: Sute, Mii: Mii, Zece Mii: x 10000, Sute Mii: x 100000, Milioane: Milioane, Zece Milioane: x 10000000, Sute Milioane: x 100000000, Mii Milioane: Miliarde, Milioane Milioane: Trilioane";
        this["LD_Fallback_ChartEx_Line1"] = "Această diagramă nu este disponibilă în versiunea dvs. de Excel.";
        this["LD_Fallback_ChartEx_Line2"] = "Editarea acestei forme sau salvarea acestui registru de lucru într-un alt format de fișier va sparge definitiv graficul.";
        this["LE_AllMustHavGuidsIfAnyHasThem"] = "Nu se poate seta GenerateGuidsForGroups la fals dacă vreun grup existent are un Guid.";
        this["LE_ArgumentException_AnchorCellFromOtherWorksheet"] = "Celula specificată nu există pe aceeași foaie de lucru cu forma.";
        this["LE_ArgumentException_ArrayFormulaMustHaveSingleRegion"] = "O formulă matrice poate fi aplicată numai unei regiuni de celule.";
        this["LE_ArgumentException_ArrayTooSmall"] = "Matricea nu este suficient de mare pentru a copia toate valorile pe ea.";
        this["LE_ArgumentException_CannotAddStandardTableStyle"] = "Un stil de tabel standard nu poate fi adăugat la CustomTableStyleCollection.";
        this["LE_ArgumentException_CannotApplyDVRuleToTotalCell"] = "Una sau mai multe celule din referința specificată sunt celule totale dintr-un tabel și nu pot aplica o regulă de validare a datelor.";
        this["LE_ArgumentException_CannotCreateEmptyColorInfo"] = "Culoarea goală nu poate fi utilizată pentru a crea o instanță WorkbookColorInfo.";
        this["LE_ArgumentException_CellMustBeOnSameWorksheetAsReferencesCollection"] = "Celula trebuie să fie din aceeași foaie de lucru ca și colecția de referință.";
        this["LE_ArgumentException_CellShiftedOffWorksheet"] = "Celula a fost ștearsă sau mutată de pe foaia de lucru și nu mai reprezintă o poziție a foii de lucru.";
        this["LE_ArgumentException_CellsInTableFromOtherWorksheet"] = "Regiunea specificată nu provine din aceeași foaie de lucru cu tabelul de date.";
        this["LE_ArgumentException_CellValueStringLength"] = "Valorile șirului atribuite unei celule nu pot avea o lungime mai mare de {0}.";
        this["LE_ArgumentException_ColumnFormulaAlreadyAppliedToCell"] = "Formula se aplică deja unei celule. Clonați-l pentru a-l utiliza ca formulă de coloană.";
        this["LE_ArgumentException_ColumnFromDifferentWorksheet"] = "Coloana nu aparține aceleiași foi de lucru cu această colecție.";
        this["LE_ArgumentException_ColumnInputCellFromOtherWorksheet"] = "Celula de introducere a coloanei specificată nu există în aceeași foaie de lucru ca și tabelul de date.";
        this["LE_ArgumentException_ColumnNotInTable"] = "Coloana provine dintr-un alt tabel.";
        this["LE_ArgumentException_ColumnRemovedFromWorksheet"] = "Coloana a fost eliminată din foaia de lucru.";
        this["LE_ArgumentException_ColumnsToRepeatAtLeftOutsideRange"] = "Intervalul este în afara intervalului de coloane disponibile din foaia de lucru.";
        this["LE_ArgumentException_CommentTooLong"] = "Lungimea comentariului trebuie să fie mai mică de 255 de caractere.";
        this["LE_ArgumentException_CustomViewNameAlreadyExists"] = "Există deja o vizualizare personalizată cu numele \"{0}\".";
        this["LE_ArgumentException_DefaultTableStyleNotInWorkbook"] = "Stilul de tabel specificat nu se află în colecțiile CustomTableStyles sau StandardTableStyles.";
        this["LE_ArgumentException_DisplayTextTooLong"] = "Textul afișat prin hiperlink trebuie să aibă mai puțin de 256 de caractere.";
        this["LE_ArgumentException_DuplicateDisplayValue"] = "Valoarea de afișare specificată este deja în colecție.";
        this["LE_ArgumentException_DuplicateFixedDateGroup"] = "FixedDateGroup specificat este deja în colecție.";
        this["LE_ArgumentException_DuplicateTableStyle"] = "Stilul specificat se află deja într-o colecție CustomTableStyle.";
        this["LE_ArgumentException_DuplicateTableStyleName_Existing"] = "Numele specificat care se potrivește cu cel al unui alt stil din colecție.";
        this["LE_ArgumentException_DuplicateTableStyleName_New"] = "Stilul specificat are un nume care se potrivește cu un alt stil din colecție.";
        this["LE_ArgumentException_DuplicateWorksheetProtectedRangeTitle"] = "Există deja un interval cu titlul '{0}'.";
        this["LE_ArgumentException_DV_ArrayFormulaCannotBeUsed"] = "ArrayFormula nu poate fi aplicată ca {0}.";
        this["LE_ArgumentException_DV_CellFromOtherWorksheet"] = "Celula nu aparține aceleiași foi de lucru ca și colecția de reguli de validare a datelor.";
        this["LE_ArgumentException_DV_ConstraintFormulaMustBeNumberOrReference"] = "{0} trebuie setat la un număr, o expresie numerică sau o referință la o singură celulă.";
        this["LE_ArgumentException_DV_FormulaCannotBeNull"] = "{0} nu poate fi setat la nul în timp ce regula de validare a datelor este aplicată unei foi de lucru.";
        this["LE_ArgumentException_DV_FormulaCannotFindNamedReference"] = "{0} conține o referință numită care nu există în registrul de lucru.";
        this["LE_ArgumentException_DV_FormulaCannotFindWorksheetReference"] = "{0} conține o referință pe o foaie de lucru care nu există în registrul de lucru.";
        this["LE_ArgumentException_DV_FormulaCannotReferenceOtherWorkbook"] = "{0} nu poate conține o referință la un registru de lucru extern.";
        this["LE_ArgumentException_DV_FormulaInvalidForWorkbookFormat"] = "{0} face referire la celule care nu sunt valide în formatul de registru de lucru {1}.";
        this["LE_ArgumentException_DV_InvalidAddress"] = "Adresa specificată nu este o adresă validă pentru celule sau regiuni.";
        this["LE_ArgumentException_DV_InvalidExcelDate"] = "Data specificată nu poate fi reprezentată în Excel.";
        this["LE_ArgumentException_DV_InvalidFormula"] = "Formula nevalidă utilizată în regula de validare a datelor.";
        this["LE_ArgumentException_DV_InvalidMessageLength"] = "{0} nu poate avea mai mult de {1} caractere.";
        this["LE_ArgumentException_DV_ListFormulaCannotHaveEmptyString"] = "Șirul constant din ValuesFormula nu poate fi un șir gol.";
        this["LE_ArgumentException_DV_ListFormulaMustBeStringOrReference"] = "ValuesFormula trebuie setată la o constantă de șir sau la o referință la o singură celulă sau regiune din același registru de lucru.";
        this["LE_ArgumentException_DV_ListFormulaReferenceMustBeOneDimensional"] = "ValuesFormula conține o referință de regiune care are mai multe rânduri și coloane. Regiunea trebuie să conțină un rând sau o coloană.";
        this["LE_ArgumentException_DV_LowerGreaterThanUpperContraint"] = "Valoarea constrângerii inferioare trebuie să fie mai mică sau egală cu constrângerea superioară.";
        this["LE_ArgumentException_DV_MustHaveOneAcceptedValue"] = "Cel puțin o valoare trebuie specificată ca valoare acceptată.";
        this["LE_ArgumentException_DV_ReferencesAlreadyContainDVs"] = "Referințele specificate conțin deja una sau mai multe reguli de validare a datelor.";
        this["LE_ArgumentException_DV_ReferencesFromOtherWorksheet"] = "Colecția de referințe nu aparține aceleiași foi de lucru ca și colecția de reguli de validare a datelor.";
        this["LE_ArgumentException_DV_RegionFromOtherWorksheet"] = "Regiunea nu aparține aceleiași foi de lucru ca și colecția de reguli de validare a datelor.";
        this["LE_ArgumentException_DV_RuleAppliedOnOtherWorksheet"] = "Regula de validare a datelor se aplică deja unei alte foi de lucru.";
        this["LE_ArgumentException_DV_UpperLessThanLowerContraint"] = "Valoarea constrângerii superioare trebuie să fie mai mare sau egală cu constrângerea inferioară.";
        this["LE_ArgumentException_EditRangeAlreadyOwned"] = "Gama este deja în AllowedEditRanges a unei foi de lucru.";
        this["LE_ArgumentException_EditRangeHasRangeFromOtherWorksheet"] = "Intervalul de editare are o gamă de la o foaie de lucru diferită de AllowedEditRanges.";
        this["LE_ArgumentException_EmptyCustomList"] = "Trebuie să existe cel puțin un șir în listă.";
        this["LE_ArgumentException_EndIndexLessThanZero"] = "Sfârșitul intervalului trebuie să fie mai mare sau egal cu zero.";
        this["LE_ArgumentException_FileDoesntContainsWorkbookStream"] = "Fluxul sau fișierul specificat nu conține un registru de lucru valid.";
        this["LE_ArgumentException_FormulaReferencesInvalidCells"] = "Formula specificată face referire la celule care nu sunt acceptate în formatul de fișier curent al registrului de lucru.";
        this["LE_ArgumentException_HashSetFull"] = "Există prea multe articole în setul de hash.";
        this["LE_ArgumentException_HyperlinkAlreadyOwned"] = "Hiperlinkul aparține deja unei colecții de hiperlinkuri.";
        this["LE_ArgumentException_HyperlinkSourceFromOtherWorksheet"] = "Hiperlinkul are o celulă sursă sau o regiune dintr-o foaie de lucru diferită.";
        this["LE_ArgumentException_HyperlinkTargetFromOtherWorkbook"] = "Ținta provine dintr-un registru de lucru diferit de cel al proprietarului hiperlinkului.";
        this["LE_ArgumentException_InfiniteColumnWidth"] = "Lățimea coloanei nu poate fi infinită.";
        this["LE_ArgumentException_Interval"] = "Parametrul Interval ar trebui să fie una dintre valorile din enumerarea DateInterval";
        this["LE_ArgumentException_IntervalStr"] = "Parametrul Interval nu a putut fi convertit la una dintre valorile din enumerarea DateInterval";
        this["LE_ArgumentException_InvalidCellAddress"] = "Adresa specificată nu este o adresă de celulă validă.";
        this["LE_ArgumentException_InvalidCommentPositioningMode"] = "Numai DontMoveOrSizeWithCells și MoveAndSizeWithCells ShapePositionModes pot fi aplicate unei forme de comentariu.";
        this["LE_ArgumentException_InvalidCustomFilterOperandNumber"] = "Valoarea nu poate fi infinit sau NaN.";
        this["LE_ArgumentException_InvalidCustomFilterOperator"] = "Atunci când valoarea comparației nu este un șir, operatorul de comparație nu poate fi BeginsWith, DoesNotBeginWith, Contains, DoesNotContain, EndsWith sau DoesNotEndWith.";
        this["LE_ArgumentException_InvalidDatePeriodFilterValue"] = "Valoarea nu se încadrează. Pentru un tip de filtru de lună, acesta trebuie să fie 1-12. Pentru un tip de filtru de trimestru, acesta trebuie să fie 1-4.";
        this["LE_ArgumentException_InvalidDpi"] = "Valoarea Dpi specificată este nevalidă.";
        this["LE_ArgumentException_InvalidExcelDate"] = "Valoarea nu poate fi exprimată ca dată în Excel.";
        this["LE_ArgumentException_InvalidFileFormat"] = "Format de fișier nevalid sau nerecunoscut.";
        this["LE_ArgumentException_InvalidFormula"] = "Formula specificată nu era validă.";
        this["LE_ArgumentException_InvalidGradientStopColor"] = "Culorile automate și de sistem nu pot fi utilizate în opriri de gradient.";
        this["LE_ArgumentException_InvalidNamedReferenceName"] = "Numele nu este un nume de referință valid.";
        this["LE_ArgumentException_InvalidPaletteColor_EmptyOrSystem"] = "Culorile goale sau ale sistemului nu pot fi adăugate în paletă.";
        this["LE_ArgumentException_InvalidPaletteColor_NonOpaque"] = "Doar culorile opace pot fi adăugate la paletă.";
        this["LE_ArgumentException_InvalidReferencesString"] = "Valoarea referințelor nu este formatată corect.";
        this["LE_ArgumentException_InvalidRegionAddress"] = "Adresa specificată nu este o adresă de regiune validă.";
        this["LE_ArgumentException_InvalidRowOrColumnRange"] = "Interval de rânduri sau coloane nevalide.";
        this["LE_ArgumentException_InvalidSeriesValues"] = "Valori de serie nevalide. Trebuie să existe cel puțin o valoare și fiecare valoare trebuie să fie fie un șir, fie un obiect care poate fi convertit într-o dublă.";
        this["LE_ArgumentException_InvalidTarget"] = "Ținta hiperlink trebuie să fie un șir, WorksheetCell, WorksheetRegion sau NamedReference.";
        this["LE_ArgumentException_InvalidTopOrBottomFilterValue"] = "Valoarea trebuie să fie între 1 și 500.";
        this["LE_ArgumentException_InvalidWorksheetName"] = "Numele foii de lucru nu poate conține niciunul dintre următoarele caractere: : \\ /? * [ ]";
        this["LE_ArgumentException_InvalidWorksheetNameStartingQuote"] = "Numele foii de lucru nu poate începe sau nu se poate termina cu ghilimele simple.";
        this["LE_ArgumentException_LessThanTwoGradientStops"] = "Trebuie specificate cel puțin două opriri de gradient.";
        this["LE_ArgumentException_NamedReferenceNameTooLong"] = "Numele referinței numite nu poate fi mai mare de 255 de caractere.";
        this["LE_ArgumentException_NaNDefaultColumnWidth"] = "NaN nu poate fi utilizat ca lățime implicită a coloanei.";
        this["LE_ArgumentException_NewRegionOverlapsFilterArea"] = "Operațiunea încearcă să schimbe un interval filtrat în foaia dvs. de lucru și nu poate fi finalizată. Pentru a finaliza această operațiune, filtrele automate din foaie trebuie eliminate.";
        this["LE_ArgumentException_NewTableRegionCannotMoveHeaders"] = "Noua regiune a tabelului nu poate muta anteturile pe un rând nou.";
        this["LE_ArgumentException_NewTableRegionFromWrongWorksheet"] = "Regiunea specificată nu este din aceeași foaie de lucru ca tabelul.";
        this["LE_ArgumentException_NewTableRegionMustHaveDataRows"] = "Noua regiune a tabelului trebuie să conțină cel puțin un rând de date.";
        this["LE_ArgumentException_NewTableRegionOverlapOld"] = "Noua regiune de tabel trebuie să se suprapună cu regiunea de tabel anterioară.";
        this["LE_ArgumentException_NewTableRegionOverlapsBlockingValue"] = "Tabelul nu poate fi redimensionat la regiunea specificată, deoarece include o formulă matrice cu mai multe celule sau un tabel de date.";
        this["LE_ArgumentException_NewTableRegionOverlapsMergedRegion"] = "Tabelul nu poate fi redimensionat la regiunea specificată, deoarece include o celulă îmbinată.";
        this["LE_ArgumentException_NewTableRegionOverlapsOtherTable"] = "Tabelul nu poate fi redimensionat la regiunea specificată, deoarece include un alt tabel.";
        this["LE_ArgumentException_NoRegionsInArray"] = "Matricea specificată nu are regiuni.";
        this["LE_ArgumentException_NotEnum"] = "Tipul '{0}' nu este o enumerare";
        this["LE_ArgumentException_NotSupportedCodePage"] = "Pagina de cod nu este acceptată";
        this["LE_ArgumentException_NPer"] = "nPer nu poate fi 0";
        this["LE_ArgumentException_NPerFunction"] = "Combinație greșită de parametri pentru funcția NPer";
        this["LE_ArgumentException_PA_PrintAreasCannotOverlap"] = "Zona de imprimare specificată se suprapune cu una dintre zonele de imprimare existente.";
        this["LE_ArgumentException_ParentStyleFromOtherWorkbook"] = "Stilul specificat nu provine din același registru de lucru ca formatul celulei.";
        this["LE_ArgumentException_PB_CantInsertBreakAtIndex"] = "Această colecție nu acceptă inserarea articolelor la un index.";
        this["LE_ArgumentException_PB_CantSetBreakAtIndex"] = "Această colecție nu acceptă setarea articolelor la un index.";
        this["LE_ArgumentException_PB_MustBeWithinPrintArea"] = "Sfârșitul de pagină trebuie să se afle în zona de imprimare.";
        this["LE_ArgumentException_PB_PageBreakCantBeA1Cell"] = "Nu se poate insera sfârșit de pagină în partea stângă sus a foii de lucru.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeLeftColumn"] = "Nu se poate insera sfârșit de pagină în stânga foii de lucru.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeTopRow"] = "Nu se poate insera sfârșit de pagină în partea de sus a foii de lucru.";
        this["LE_ArgumentException_PB_PageBreaksCannotAddExistingInstance"] = "Instanța specificată există deja în colecție.";
        this["LE_ArgumentException_PB_PageBreaksCannotOverlap"] = "Pauza de pagină nu se poate suprapune cu alte pauze de pagină din foaia de lucru.";
        this["LE_ArgumentException_PB_PrintAreaMustBeInPrintAreas"] = "Sfârșitul de pagină se află într-o zonă de imprimare care nu se află în colecția PrintAreas.";
        this["LE_ArgumentException_PB_PrintAreaMustBeOnSameWorksheet"] = "Sfârșitul de pagină se află într-o zonă de imprimare care nu se află pe aceeași foaie de lucru cu colecția de pauze de pagină.";
        this["LE_ArgumentException_Pmt"] = "Pmt nu poate fi 0";
        this["LE_ArgumentException_PrintAreaShiftedOffWorksheet"] = "Pauza de pagină se află într-o zonă de imprimare care a fost mutată de pe foaia de lucru și nu mai reprezintă un interval valid de celule.";
        this["LE_ArgumentException_RangeFromOtherWorksheetThanProtectedRange"] = "Intervalul specificat provine dintr-o foaie de lucru diferită de intervalul de editare.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheet"] = "Regiunea trebuie să provină din aceeași foaie de lucru.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheetAsReferencesCollection"] = "Regiunea trebuie să provină din aceeași foaie de lucru cu colecția de referință.";
        this["LE_ArgumentException_RegionsFromMixedWorksheets"] = "Toate regiunile specificate în matrice trebuie să provină din aceeași foaie de lucru.";
        this["LE_ArgumentException_RegionShiftedOffWorksheet"] = "Regiunea a fost mutată de pe foaia de lucru și nu mai reprezintă un interval valid de celule.";
        this["LE_ArgumentException_RegionsShiftedOffWorksheet"] = "Una sau mai multe regiuni specificate au fost mutate de pe foaia de lucru și nu mai reprezintă un interval valid de celule.";
        this["LE_ArgumentException_RowFromDifferentWorksheet"] = "Rândul nu aparține aceleiași foi de lucru cu această colecție.";
        this["LE_ArgumentException_RowInputCellFromOtherWorksheet"] = "Celula de intrare rând specificată nu există pe aceeași foaie de lucru ca tabelul de date.";
        this["LE_ArgumentException_RowRemovedFromWorksheet"] = "Rândul a fost eliminat din foaia de lucru.";
        this["LE_ArgumentException_SelectedWorksheetFromOtherWorkbook"] = "Foaia de lucru nu aparține registrului de lucru asociat cu aceste opțiuni ale ferestrei.";
        this["LE_ArgumentException_ShapeCannotBeAdded"] = "Forma specificată nu poate fi adăugată la o colecție de forme.";
        this["LE_ArgumentException_SourceFromOtherWorksheetThanHyperlink"] = "Celula sau regiunea sursă specificată provine dintr-o foaie de lucru diferită de cea a hyperlinkului.";
        this["LE_ArgumentException_StartIndexLessThanZero"] = "Începutul intervalului trebuie să fie mai mare sau egal cu zero.";
        this["LE_ArgumentException_StyleNameAlreadyExists"] = "În registrul de lucru există deja un stil cu numele \"{0}\".";
        this["LE_ArgumentException_StyleNameTooLong"] = "Numele stilului nu poate fi mai lung de 255 de caractere.";
        this["LE_ArgumentException_TableColumnRemovedFromTable"] = "Coloana tabelului a fost eliminată din tabel.";
        this["LE_ArgumentException_TableStyleFromOtherWorkbook"] = "WorksheetTableStyle specificat nu aparține aceluiași registru de lucru ca tabelul.";
        this["LE_ArgumentException_TargetTooLong"] = "Adresa țintă a hiperlinkului trebuie să aibă mai puțin de 256 de caractere.";
        this["LE_ArgumentException_TextLengthGreaterThanMax"] = "Lungimea textului într-o formă sau comentariu nu poate fi mai mare de {0}.";
        this["LE_ArgumentException_ToolTipTooLong"] = "Sfatul instrumentului de hiperlink trebuie să aibă mai puțin de 256 de caractere.";
        this["LE_ArgumentException_TopLeftWindowBounds"] = "Limitele stânga și de sus ale ferestrei trebuie să fie între -32768 și 32767.";
        this["LE_ArgumentException_WeekdayFirst"] = "Parametrul WeekdayFirst ar trebui să fie între FirstDayOfWeek.System și FirstDayOfWeek.Saturday";
        this["LE_ArgumentException_WidthHeightWindowBounds"] = "Lățimea și înălțimea limitelor ferestrei trebuie să fie între 0 și 65535.";
        this["LE_ArgumentException_WorksheetNameAlreadyExists"] = "Există deja o foaie de lucru cu numele \"{0}\" în registrul de lucru.";
        this["LE_ArgumentException_WorksheetNameTooLong"] = "Numele foii de lucru trebuie să aibă între 1 și 31 de caractere în lungime.";
        this["LE_ArgumentException_WorksheetProtectedRangeTitleTooLong"] = "Titlul unui interval de editare nu poate fi mai mare de 255 de caractere.";
        this["LE_ArgumentException_WorksheetScopeFromOtherWorkbook"] = "Foaia de lucru nu aparține aceluiași registru de lucru ca această colecție.";
        this["LE_ArgumentNullException_AnchorCell"] = "Celula nu poate fi nulă atunci când forma există pe o foaie de lucru sau un grup.";
        this["LE_ArgumentNullException_CustomViewName"] = "Numele vizualizării personalizate nu poate fi nul sau gol.";
        this["LE_ArgumentNullException_Encoder"] = "Codificatorul de imagine nu poate fi nul.";
        this["LE_ArgumentNullException_FindNamedReference"] = "Numele de găsit nu poate fi nul sau gol.";
        this["LE_ArgumentNullException_FormulaCantBeNull"] = "Formula pentru o referință numită nu poate fi nulă sau goală.";
        this["LE_ArgumentNullException_HiddenColumn"] = "O coloană nulă nu poate fi adăugată la colecție.";
        this["LE_ArgumentNullException_HiddenRow"] = "Un rând nul nu poate fi adăugat la colecție.";
        this["LE_ArgumentNullException_Image"] = "Imaginea nu poate fi nulă.";
        this["LE_ArgumentNullException_NamedReferenceNameCantBeNull"] = "Numele unei referințe numite nu poate fi nul sau gol.";
        this["LE_ArgumentNullException_SaveStream"] = "Nu se poate salva registrul de lucru într-un flux nul.";
        this["LE_ArgumentNullException_SelectedWorksheet"] = "Registrul de lucru trebuie să aibă o foaie de lucru selectată validă.";
        this["LE_ArgumentNullException_Shape"] = "O formă nulă nu poate fi adăugată la colecție.";
        this["LE_ArgumentNullException_SourceFont"] = "Fontul sursă nu poate fi nul.";
        this["LE_ArgumentNullException_SourceFormatting"] = "Formatarea sursă nu poate fi nulă.";
        this["LE_ArgumentNullException_StyleName"] = "Numele stilului nu poate fi nul sau gol.";
        this["LE_ArgumentNullException_UnformattedString"] = "Șirul neformatat nu poate fi nul.";
        this["LE_ArgumentNullException_Workbook"] = "Registrul de lucru nu poate fi nul.";
        this["LE_ArgumentNullException_WorkbookRequiredToResolveThemeColor"] = "Trebuie specificat un registru de lucru pentru a rezolva o culoare tematică.";
        this["LE_ArgumentNullException_WorksheetName"] = "Numele foii de lucru nu poate fi nul sau gol.";
        this["LE_ArgumentNullException_WorksheetProtectedRangeTitleCantBeNull"] = "Titlul unui interval de editare nu poate fi nul sau gol.";
        this["LE_ArgumentNullException_WorksheetScope"] = "Domeniul foii de lucru nu poate fi nul.";
        this["LE_ArgumentOutOfRange_RelativeColumnIndex"] = "Indicele coloanei relative se află în afara regiunii specificate.";
        this["LE_ArgumentOutOfRange_RelativeRowIndex"] = "Indicele de rând relativ se află în afara regiunii specificate.";
        this["LE_ArgumentOutOfRangeException_AnchorPosition"] = "Unul dintre procentele din locația specificată este mai mic de 0.";
        this["LE_ArgumentOutOfRangeException_BadInsertIndex"] = "Indicele specificat trebuie să fie mai mare sau egal cu zero și mai mic sau egal cu numărul de paragrafe existente.";
        this["LE_ArgumentOutOfRangeException_BadRemoveAtIndex"] = "Indexul trebuie să fie mai mare decât zero și mai mic decât numărul de paragrafe din șirul formatat.";
        this["LE_ArgumentOutOfRangeException_CollectionIndex"] = "Indicele a fost în afara intervalului. Nu trebuie să fie negativ și mai mic decât dimensiunea colecției.";
        this["LE_ArgumentOutOfRangeException_ColumnWidth"] = "Lățimea coloanei trebuie să fie între 0 și 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultColumnWidth"] = "Lățimea implicită a coloanei trebuie să fie între 0 și 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultFontHeight"] = "Înălțimea implicită a fontului trebuie să fie între 20 și 8180.";
        this["LE_ArgumentOutOfRangeException_DefaultRowHeight"] = "Înălțimea implicită a rândului trebuie să fie între -1 și 8192.";
        this["LE_ArgumentOutOfRangeException_DuplicateItemSorted"] = "Elementul specificat are deja aplicată o condiție de sortare.";
        this["LE_ArgumentOutOfRangeException_FirstVisibleTabIndex"] = "Primul index de filă vizibil trebuie să fie mai mare sau egal cu zero.";
        this["LE_ArgumentOutOfRangeException_FontHeight"] = "Înălțimea fontului trebuie să fie între 20 și 8180.";
        this["LE_ArgumentOutOfRangeException_GroupAddedToSelf"] = "Un grup nu poate fi adăugat la propria colecție de forme.";
        this["LE_ArgumentOutOfRangeException_Indent"] = "Nivelul de indentare trebuie să fie între 0 și 250.";
        this["LE_ArgumentOutOfRangeException_IndexNegative"] = "Indicele trebuie să fie non-negativ.";
        this["LE_ArgumentOutOfRangeException_InvalidCollectionIndex"] = "Indexul specificat nu este valid în colecție.";
        this["LE_ArgumentOutOfRangeException_InvalidColorInfoTint"] = "Valoarea nuanței trebuie să fie între -1,0 și 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnCount"] = "{0} nu este un număr valid de coloane în formatul curent. Valoarea trebuie să fie între {1} și {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnIndex"] = "{0} nu este un index de coloană valid în formatul curent. Valoarea trebuie să fie între {1} și {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidGradientStopOffset"] = "Decalajul opririi gradientului trebuie să fie între 0,0 și 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRelativeRectangleValueForGradient"] = "Valorile dreptunghiului interior al gradientului dreptunghiular trebuie să fie cuprinse între 0,0 și 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRemoveAtIndex_SortConditions"] = "Indicele la care se elimină condiția de sortare este în afara intervalului.";
        this["LE_ArgumentOutOfRangeException_InvalidRowCount"] = "{0} nu este un număr valid de rânduri în formatul curent. Valoarea trebuie să fie între {1} și {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidRowIndex"] = "{0} nu este un index de rând valid în formatul curent. Valoarea trebuie să fie între {1} și {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidTableStyleAreaStripeExtent"] = "{0} trebuie să fie între 1 și 9, inclusiv.";
        this["LE_ArgumentOutOfRangeException_LeftPaneWidth"] = "Lățimea panoului specificată este în afara intervalului valid de {0} și {1}.";
        this["LE_ArgumentOutOfRangeException_LengthMustBePositive"] = "Lungimea trebuie să fie mai mare sau egală cu una.";
        this["LE_ArgumentOutOfRangeException_MagnificationLevel"] = "Nivelurile de mărire a vizualizării trebuie să fie între 10 și 400.";
        this["LE_ArgumentOutOfRangeException_Margins"] = "Marjele trebuie să fie cuprinse între 0 și 100.";
        this["LE_ArgumentOutOfRangeException_MaxPagesHorizontally"] = "Numărul maxim de pagini pe orizontală trebuie să fie între 0 și 32767.";
        this["LE_ArgumentOutOfRangeException_MaxPagesVertically"] = "Numărul maxim de pagini pe verticală trebuie să fie între 0 și 32767.";
        this["LE_ArgumentOutOfRangeException_MaxRecursionIterations"] = "Iterațiile recursive maxime trebuie să fie cuprinse între 1 și 32767.";
        this["LE_ArgumentOutOfRangeException_MaxSortConditions"] = "Nu pot exista mai mult de {0} condiții de sortare în colecție.";
        this["LE_ArgumentOutOfRangeException_NegativeStartIndex"] = "Indicele de start nu poate fi negativ.";
        this["LE_ArgumentOutOfRangeException_NumberOfCopies"] = "Numărul de copii trebuie să fie între 1 și 65535.";
        this["LE_ArgumentOutOfRangeException_Per"] = "per valoare ar trebui să fie între 0 și nPer + 1";
        this["LE_ArgumentOutOfRangeException_Rate"] = "Rata ar trebui să fie mai mare de -1";
        this["LE_ArgumentOutOfRangeException_Resolution"] = "Rezoluția trebuie să fie între 0 și 65535.";
        this["LE_ArgumentOutOfRangeException_RowHeight"] = "Înălțimea rândului trebuie să fie între 0 și 8192.";
        this["LE_ArgumentOutOfRangeException_ScalingFactor"] = "Factorul de scalare trebuie să fie între 10 și 400.";
        this["LE_ArgumentOutOfRangeException_StartPageNumber"] = "Numărul paginii de pornire trebuie să fie între -32765 și 32767.";
        this["LE_ArgumentOutOfRangeException_TabBarWidth"] = "Lățimea barei de file trebuie să fie între 0 și 1000.";
        this["LE_ArgumentOutOfRangeException_TopPaneHeight"] = "Înălțimea panoului specificată este în afara intervalului valid de {0} și {1}.";
        this["LE_AutoColorNotAllowed"] = "Culoarea nu poate fi \"Automată\".";
        this["LE_Axis_MajorMinorUnit"] = "Valoarea parametrului majorUnit trebuie să fie mai mare decât valoarea parametrului minorUnit.";
        this["LE_Axis_NoCrossAxis"] = "Dacă proprietatea CrossAxis este setată pe un obiect Axis, aceasta trebuie setată pe toate. Dacă diagrama dvs. are doar două axe, axa de trecere poate fi determinată automat, dar numai dacă proprietatea CrossAxis este lăsată nesetată pe toate obiectele Axis.";
        this["LE_Biff8SerializerNotLoaded"] = "Formatul '{0}' necesită un serializator care nu este încărcat, în JavaScript acesta este conținut în fișierul infragistics.excel_serialization_biff8.js";
        this["LE_CategoryAxisBinning_BinWidthAndNumberOfBins"] = "Parametrii \"binWidth\" și \"numberOfBins\" se exclud reciproc. Ambele pot fi nespecificate, dar dacă este specificat unul, celălalt nu trebuie să fie.";
        this["LE_ChartGradientFill_EmptyStops"] = "Trebuie să conțină cel puțin o oprire de gradient.";
        this["LE_ChartObject_DifferentChart"] = "Instanța obiect alocată este asociată cu o diagramă diferită și nu poate fi atribuită acestei instanțe.";
        this["LE_ColumnsToRepeatAtLeftAreOutsideAvailableRange"] = "Opțiunile de imprimare au coloane care se repetă în stânga, care se află în afara gamei de coloane disponibile în noul format.";
        this["LE_ComboChartGroupCollection_DoesNotExist"] = "ComboChartGroup specificat nu există în această colecție.";
        this["LE_FillPercentInvalid"] = "Valoarea {0} nu este o valoare validă pentru proprietatea {1}.";
        this["LE_FormatLimitError_Indent"] = "Indentarea depășește valoarea maximă de {0} furnizată de formatul curent.";
        this["LE_FormatLimitError_MacroWorkbook"] = "Registrul de lucru are cod VBA și nu poate fi salvată ca registru de lucru activat pentru macro.";
        this["LE_FormatLimitError_MaxColumnIndex"] = "Indicele coloanei {0} este mai mare decât {1} maximul acceptat de formatul specificat.";
        this["LE_FormatLimitError_MaxRowIndex"] = "Indicele rândului {0} depășește {1} maximul acceptat de acest format.";
        this["LE_FormulaParseException_ArrayContainsConstants"] = "O matrice poate conține doar expresii constante.";
        this["LE_FormulaParseException_ArrayHasEmptyFirstRow"] = "O matrice constantă din formulă nu are elemente în primul rând.";
        this["LE_FormulaParseException_ArrayHasMisalignedRows"] = "O matrice constantă din formulă are rânduri nealiniate. Fiecare rând dintr-o matrice trebuie să aibă același număr de coloane.";
        this["LE_FormulaParseException_CellReferenceAfterWorkbookName"] = "O referință de registru de lucru extern trebuie să fie urmată de o referință numită, nu de o referință de celulă.";
        this["LE_FormulaParseException_ExtraExpressions"] = "Formula are expresii suplimentare după sfârșitul acesteia.";
        this["LE_FormulaParseException_FunctionMissingClosingParen"] = "O funcție din formulă nu are paranteză de închidere.";
        this["LE_FormulaParseException_FunctionNestingTooDeep"] = "O funcție din formulă este cuibărită prea adânc în alte funcții.";
        this["LE_FormulaParseException_IncorrectNumberOfArguments"] = "O funcție din formulă nu a primit numărul corect de argumente.";
        this["LE_FormulaParseException_InvalidArguments"] = "Un operator sau o funcție din formulă are argumente nevalide.";
        this["LE_FormulaParseException_InvalidConditionalFormatFormula"] = "O formulă de format condițional nu poate utiliza operatori de referință, cum ar fi uniuni sau intersecții, constante de matrice sau referințe de tabel structurat.";
        this["LE_FormulaParseException_InvalidErrorValue"] = "Tipul de eroare specificat nu este valid. Trebuie să fie una dintre următoarele valori: \"#NULL!\", \"#DIV/0!\", \"#VALUE!\", \"#REF!\", \"#NAME?\", \"#NUM!\" Sau \"#N/A\".";
        this["LE_FormulaParseException_InvalidFileNameInBrackets"] = "Un nume de fișier între paranteze pătrate nu era valid.";
        this["LE_FormulaParseException_InvalidStructuredTableReference"] = "Referința tabelului structurat este incorectă.";
        this["LE_FormulaParseException_InvalidWorkbookName"] = "Referința registrului de lucru în ghilimele nu este un nume de fișier valid al registrului de lucru.";
        this["LE_FormulaParseException_InvalidWorksheetName"] = "Numele foii de lucru este nevalid.";
        this["LE_FormulaParseException_MacroFunctionNotAllowed"] = "Funcțiile '{0}' sunt valabile numai într-o referință numită.";
        this["LE_FormulaParseException_Message_PortionWithError"] = "Porțiune cu eroare: {0}";
        this["LE_FormulaParseException_MissingArgumentAfterBinary"] = "Unui operator binar din formulă îi lipsește un operand după acesta.";
        this["LE_FormulaParseException_MissingArgumentAfterParen"] = "Nu există nicio expresie în urma unei paranteze deschise.";
        this["LE_FormulaParseException_MissingArgumentAfterUnary"] = "Un operator unar din formulă nu a fost urmat de o expresie validă.";
        this["LE_FormulaParseException_MissingArgumentBeforeBinary"] = "Unui operator binar din formulă îi lipsește un operand înaintea acestuia.";
        this["LE_FormulaParseException_NamedReferenceRefsNeedSheetName"] = "Toate referințele de celule dintr-o formulă de referință numită trebuie să identifice foaia de lucru.";
        this["LE_FormulaParseException_NoElementAfterArraySerapator"] = "Nu există niciun element de matrice după un separator de element în formulă.";
        this["LE_FormulaParseException_NoEqualsSign"] = "Formula trebuie să înceapă cu un semn egal.";
        this["LE_FormulaParseException_NoExclamationAfterWorkbookName"] = "Referința registrului de lucru trebuie să fie urmată de semnul exclamării.";
        this["LE_FormulaParseException_NoExclamationAfterWorksheetName"] = "Referința foii de lucru trebuie să fie urmată de semnul exclamării.";
        this["LE_FormulaParseException_NoExpressions"] = "Formula nu are expresii valide după semnul egal.";
        this["LE_FormulaParseException_NoFileNameAfterBracket"] = "Se aștepta un nume de fișier valid după paranteza pătrată deschisă.";
        this["LE_FormulaParseException_NoValidTermAfterWorkbookName"] = "O referință externă a registrului de lucru trebuie să fie urmată de o referință numită.";
        this["LE_FormulaParseException_NoValidTermAfterWorksheetName"] = "O referință de foaie de lucru trebuie să fie urmată de o celulă sau o referință numită.";
        this["LE_FormulaParseException_NoWorksheetAfterWorkbookName"] = "Un nume de fișier de registru de lucru între paranteze pătrate trebuie să fie urmat de un nume de foaie de lucru.";
        this["LE_FormulaParseException_StringConstantLengthTooLong"] = "Formula specificată nu poate conține un șir mai lung de {0} caractere.";
        this["LE_FormulaParseException_TooLong"] = "Lungimea maximă a unei formule nu poate depăși {0} pentru formatul de registru de lucru specificat.";
        this["LE_FormulaParseException_UnknownFunction"] = "O funcție din formulă nu are un nume de funcție valid.";
        this["LE_FormulaParseException_UnmatchedOpenBracket"] = "O matrice constantă din formulă nu are o paranteză de închidere.";
        this["LE_FormulaParseException_UnmatchedOpenParen"] = "O paranteză deschisă din formulă nu are o paranteză de închidere potrivită.";
        this["LE_FormulaParseException_UnmatchedOpenSquareBracket"] = "O paranteză pătrată de închidere era de așteptat după numele fișierului din registrul de lucru.";
        this["LE_FormulaParseException_WorkbookNameMissingEndQuote"] = "Referința registrului de lucru nu are ghilimele de închidere.";
        this["LE_FormulaParseException_WorksheetRangeMissingEndingName"] = "Gama de foi de lucru nu specifică un al doilea nume de foaie de lucru.";
        this["LE_GradientStop_InvalidPosition"] = "Trebuie să specificați un număr între 0 și 1.";
        this["LE_IconSetInvalid"] = "IconSetNoIcon nu este o setare validă pentru proprietatea IconSet.";
        this["LE_IndexOutOfRangeException_ArrayBounds"] = "Indexul se afla în afara limitelor matricei.";
        this["LE_InvalidEnumArgumentException_DefaultPatternCannotBeUsed"] = "Implicit FillPatternStyle nu poate fi utilizat pentru a crea un CellFillPattern.";
        this["LE_InvalidLineWeight"] = "LineWeight nu trebuie să fie mai mic de 0 sau mai mare de 1584.";
        this["LE_InvalidOperationException_AnchorCommentBeforeApplyingToCell"] = "Comentariul trebuie să aibă setul TopLeftCornerCell și BottomRightCornerCell sau trebuie să fie ambele nule înainte ca acesta să fie aplicat unei celule.";
        this["LE_InvalidOperationException_AnchorShapeBeforeAddingToCollection"] = "Forma trebuie să aibă setul TopLeftCornerCell și BottomRightCornerCell înainte de a fi adăugat la o foaie de lucru sau la un grup.";
        this["LE_InvalidOperationException_ArrayFormulaAlreadyApplied"] = "Formula matricei este deja aplicată unui interval de celule. Ștergeți mai întâi gama de celule existente.";
        this["LE_InvalidOperationException_ArrayFormulaAppliedInTable"] = "Formulele matrice nu pot fi aplicate unei zone în care este prezent un tabel.";
        this["LE_InvalidOperationException_ArrayFormulaInMergedCell"] = "Formulele matrice nu sunt valabile în celulele combinate.";
        this["LE_InvalidOperationException_BottomRightAnchorFromOtherWorksheet"] = "Adăugarea formei la această colecție o va plasa pe o foaie de lucru diferită de BottomRightCornerCell.";
        this["LE_InvalidOperationException_BuiltInStyleNameCannotBeChanged"] = "Numele unui stil construit nu poate fi schimbat.";
        this["LE_InvalidOperationException_CannotAddColorsToPaletteDirectly"] = "Culorile nu pot fi adăugate la WorkbookColorPalette. Utilizați indexerul pentru a seta o culoare la un anumit index.";
        this["LE_InvalidOperationException_CannotAddParagraphDirectly"] = "Paragrafele nu pot fi adăugate direct.";
        this["LE_InvalidOperationException_CannotAddTableDirectly"] = "Instanțele WorksheetTable nu pot fi inserate direct în colecție. Utilizați metoda Adăugați.";
        this["LE_InvalidOperationException_CannotAddTableToRemovedWorksheet"] = "Tabelele nu pot fi adăugate la o foaie de lucru care a fost eliminată din registrul de lucru propriu.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileHeaderRowIsHidden"] = "Un filtru nu poate fi aplicat în timp ce rândul antetului tabelului este ascuns.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileUIIsHidden"] = "Un filtru nu poate fi aplicat în timp ce interfața de utilizare a filtrului este ascunsă.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfData"] = "{0} nu poate fi inserat deoarece ar schimba datele din partea de jos a foii de lucru.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfObject"] = "{0} nu poate fi inserat deoarece ar schimba o formă sau ar putea comenta în partea de jos a foii de lucru.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitBlockingValue"] = "{0} nu poate fi inserat deoarece ar schimba celulele într-o formulă matrice sau un tabel de date.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitMergedRegion"] = "{0} nu poate fi inserat deoarece ar schimba celulele într-o regiune fuzionată.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitTable"] = "{0} nu poate fi inserat deoarece ar schimba celulele dintr-un alt tabel.";
        this["LE_InvalidOperationException_CannotInsertTableRow_TableOnBottomOfWorksheet"] = "{0} nu poate fi inserat deoarece tabelul se află în partea de jos a foii de lucru.";
        this["LE_InvalidOperationException_CannotModifyKeysCollection"] = "Colecția Chei nu poate fi modificată.";
        this["LE_InvalidOperationException_CannotModifyStandardTableStyle"] = "Stilurile standard de masă nu pot fi modificate.";
        this["LE_InvalidOperationException_CannotModifyValuesCollection"] = "Colecția de valori nu poate fi modificată.";
        this["LE_InvalidOperationException_CannotMoveDisconnectedWorksheet"] = "Foaia de lucru nu poate fi mutată deoarece nu aparține unui registru de lucru.";
        this["LE_InvalidOperationException_CannotRemoveColorsToPaletteDirectly"] = "Culorile nu pot fi eliminate din WorkbookColorPalette. Utilizați metoda Reset pentru a șterge toate culorile personalizate.";
        this["LE_InvalidOperationException_CannotRemoveNormalStyle"] = "Stilul Normal nu poate fi eliminat.";
        this["LE_InvalidOperationException_CannotSetParentStyleOnStyle"] = "Stilul nu poate fi setat pe un format care aparține altui stil.";
        this["LE_InvalidOperationException_CannotShift_LossOfData"] = "Nu se poate schimba datele de pe foaie.";
        this["LE_InvalidOperationException_CannotShift_LossOfObject"] = "Nu se pot deplasa obiecte de pe foaie.";
        this["LE_InvalidOperationException_CannotShift_SplitBlockingValue"] = "Nu se poate schimba într-o formulă matrice sau un tabel de date.";
        this["LE_InvalidOperationException_CannotShift_SplitMergedRegion"] = "Nu se poate schimba într-o regiune fuzionată.";
        this["LE_InvalidOperationException_CannotShift_SplitTable"] = "Nu se poate schimba în tabel.";
        this["LE_InvalidOperationException_CantAddChartInChart"] = "O diagramă nu poate fi adăugată la colecția Shapes a altei diagrame.";
        this["LE_InvalidOperationException_CantAddCustomView"] = "Vizualizările personalizate nu pot fi adăugate direct la colecție.";
        this["LE_InvalidOperationException_CantAddDataTable"] = "Tabelele de date nu pot fi adăugate direct.";
        this["LE_InvalidOperationException_CantAddMergedRegion"] = "Regiunile de celule fuzionate nu pot fi adăugate direct.";
        this["LE_InvalidOperationException_CantAddNamedReference"] = "Referințele denumite nu pot fi adăugate direct.";
        this["LE_InvalidOperationException_CantAddWorksheet"] = "Foile de lucru nu pot fi adăugate direct la această colecție.";
        this["LE_InvalidOperationException_CantApplyRemovedCustomView"] = "Vizualizarea personalizată nu poate fi aplicată dacă a fost eliminată din colecția de vizualizări personalizate a registrului de lucru.";
        this["LE_InvalidOperationException_CantChangeArrayFormula"] = "Celulele unei formule matrice nu pot fi modificate.";
        this["LE_InvalidOperationException_CantChangeDataTable"] = "Celulele interioare ale unui tabel de date nu pot fi modificate.";
        this["LE_InvalidOperationException_CantModifyCollection"] = "Colecția nu poate fi modificată.";
        this["LE_InvalidOperationException_CantOverlapArrayFormula"] = "O regiune de celulă fuzionată nu se poate suprapune peste o regiune de formulă matrice.";
        this["LE_InvalidOperationException_CantOverlapDataTableInterior"] = "O regiune de celulă îmbinată nu se poate suprapune peste celulele dintr-un tabel de date dacă se extinde în afara rândului de sus sau a coloanei din stânga a acelor celule.";
        this["LE_InvalidOperationException_CantSaveEditRangeWithoutRange"] = "AllowedEditRanges al foilor de lucru trebuie să aibă cel puțin un interval.";
        this["LE_InvalidOperationException_CantSaveWithNoVisibleWorksheets"] = "Cel puțin o foaie de lucru din registrul de lucru trebuie să fie vizibilă.";
        this["LE_InvalidOperationException_CantSaveWithNoWorksheets"] = "Un registru de lucru fără foi de lucru nu poate fi salvat";
        this["LE_InvalidOperationException_CantSetCircularityErrorDirectly"] = "Valoarea erorii de circularitate nu poate fi aplicată direct celulelor.";
        this["LE_InvalidOperationException_CantSetDataTableDirectly"] = "Tabelele de date nu pot fi setate ca valoare pe o celulă.";
        this["LE_InvalidOperationException_CantSetFormulaDirectly"] = "Formulele nu pot fi setate direct pe o celulă, utilizați în schimb metoda ApplyFormula sau una dintre metodele ApplyTo ale instanței formulei.";
        this["LE_InvalidOperationException_CellsInTableMinSize"] = "Regiunea celulelor dintr-un tabel de date trebuie să aibă cel puțin două celule în fiecare dimensiune.";
        this["LE_InvalidOperationException_ChartNotSupported"] = "Foaia de lucru WorksheetChart nu sunt acceptate în acest format.";
        this["LE_InvalidOperationException_ClearUnknownShapeData"] = "Toate datele pentru această formă sunt necunoscute, eliminați forma din colecția sa.";
        this["LE_InvalidOperationException_CollectionLongerThanMaxValue"] = "Colecția conține mai multe intrări decât permite valoarea maximă specificată.";
        this["LE_InvalidOperationException_CollectionModifiedWhileEnumerating"] = "Colecția a fost modificată în timp ce era enumerată.";
        this["LE_InvalidOperationException_CurrentTableRowAddressNeedsOriginCell"] = "Pentru a analiza o referință de tabel structurat utilizând rândul curent, trebuie specificată o celulă de origine.";
        this["LE_InvalidOperationException_CustomViewNoVisibleWorksheets"] = "Cel puțin o foaie de lucru în vizualizarea personalizată numită \"{0}\" trebuie să fie vizibilă.";
        this["LE_InvalidOperationException_DataTableAppliedInTable"] = "Un tabel de date nu poate fi într-o zonă în care este prezent un tabel.";
        this["LE_InvalidOperationException_DataTableFormulaCannotBeApplied"] = "O formulă de tabel de date nu poate fi aplicată direct celulelor.";
        this["LE_InvalidOperationException_DataTableInMergedCell"] = "Celulele interioare ale unui tabel de date nu pot face parte dintr-o regiune de celule îmbinată.";
        this["LE_InvalidOperationException_DataTableRemoved"] = "Celulele de intrare rând și coloană nu pot fi setate după ce tabelul de date a fost eliminat din foaia de lucru.";
        this["LE_InvalidOperationException_DeprecatedFillColorSetWithGradientFill"] = "O culoare a modelului de umplere nu poate fi setată în timp ce un gradient este setat pe celulă.";
        this["LE_InvalidOperationException_EncryptedWorkbooksNotSupported"] = "Cartea de lucru încărcată este criptată. Registrele de lucru criptate nu sunt acceptate.";
        this["LE_InvalidOperationException_ExtensionDoesntMatchCurrentFormat"] = "Numele fișierului în care trebuie salvat fișierul are o extensie care nu se potrivește cu formatul actual al registrului de lucru.";
        this["LE_InvalidOperationException_FixedValuesFilterMustAcceptAValue"] = "Nu pot avea un FixedValuesFilter care nu permite valori.";
        this["LE_InvalidOperationException_FormattedStringAlreadyOwned"] = "Șirul formatat este deja valoarea unei alte celule sau comentarii. Clonați-o mai întâi și aplicați clona pe noua celulă sau formă.";
        this["LE_InvalidOperationException_FormattedStringNotOwned"] = "Proprietățile fontului nu pot fi accesate în timp ce șirul sau textul formatat nu sunt aplicate unei celule sau forme.";
        this["LE_InvalidOperationException_FormattedTextAlreadyOwned"] = "Șirul formatat este deja valoarea unei alte forme. Clonați-o mai întâi și aplicați clona pe noua celulă sau formă.";
        this["LE_InvalidOperationException_FormulaAlreadyOwned"] = "Formula este deja aplicată unei alte celule.";
        this["LE_InvalidOperationException_FormulaReferencesInvalidCells"] = "Formula face referire la celule care nu sunt acceptate în formatul de fișier curent al registrului de lucru.";
        this["LE_InvalidOperationException_GetBoundsBeforeAnchorsSet"] = "Forma nu are limite înainte de a fi setate celulele de ancorare.";
        this["LE_InvalidOperationException_HiddenWorksheetCannotBeSelected"] = "O foaie de lucru ascunsă nu poate fi selectată.";
        this["LE_InvalidOperationException_HyperlinkSealed"] = "Hiperlinkul nu poate fi modificat.";
        this["LE_InvalidOperationException_ImageDisposed"] = "O imagine a unei foi de lucru Imagine a fost eliminată sau este nevalidă.";
        this["LE_InvalidOperationException_InputCellsBothNull"] = "Celulele de introducere a coloanei și rândul de intrare nu pot fi ambele nule.";
        this["LE_InvalidOperationException_InputCellsInTable"] = "Celulele de intrare nu pot fi în celulele tabelului de date.";
        this["LE_InvalidOperationException_InputCellsSame"] = "Celulele de introducere a coloanei și rândul de intrare nu pot fi aceeași celulă.";
        this["LE_InvalidOperationException_InvalidCharacterRange"] = "Gama de caractere acoperite de font nu mai există în șirul formatat.";
        this["LE_InvalidOperationException_InvalidFormatOptionsInWholeTableArea"] = "Valoarea formatului FormatOptions specificată nu este validă în zona tabelului WholeTable.";
        this["LE_InvalidOperationException_InvalidFormatString"] = "Șirul de format specificat este nevalid.";
        this["LE_InvalidOperationException_InvalidFormatString_GetTextCall"] = "Șirul de format al celulei este nevalid. Pentru a fi notificat în acest sens când FormatString este setat, setați Workbook.ValidateFormatStrings la True.";
        this["LE_InvalidOperationException_InvalidForWorkbookFormat"] = "Formula nu este validă pentru formatul specificat.";
        this["LE_InvalidOperationException_InvalidHeaderCellColumnAreaFormatProperty"] = "Valorile de margine de sus nu pot fi setate în formatul zonei coloanei tabelului HeaderCell.";
        this["LE_InvalidOperationException_InvalidHeaderRowAreaFormatProperty"] = "Valorile de margine de sus nu pot fi setate în formatul zonei de tabel HeaderRow.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFontProperty"] = "Valoarea fontului {0} nu poate fi setată pe zonele stilului tabelului.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatOptions"] = "Valoarea formatului FormatOptions specificată este nevalidă în zonele stilului tabelului.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatProperty"] = "Valoarea formatului {0} nu poate fi setată pe zonele de stil de masă.";
        this["LE_InvalidOperationException_InvalidTotalsCellColumnAreaFormatProperty"] = "Valorile marginii inferioare nu pot fi setate în formatul zonei coloanei tabelului TotalCell.";
        this["LE_InvalidOperationException_InvalidTotalsRowAreaFormatProperty"] = "Valorile marginii inferioare nu pot fi setate în formatul zonei tabelului TotalsRow.";
        this["LE_InvalidOperationException_InvalidWholeTableAreaFormatProperty"] = "Numai valorile marginilor de margine pot fi setate în formatul zonei de tabel WholeTable.";
        this["LE_InvalidOperationException_LastColumnBeforeFirst"] = "Ultima coloană din regiune trebuie să fie mai mare sau egală cu prima coloană.";
        this["LE_InvalidOperationException_LastRowBeforeFirst"] = "Ultimul rând din regiune ar trebui să fie mai mare decât primul rând.";
        this["LE_InvalidOperationException_MaxCellFormats"] = "Numărul maxim de formate de celule distincte din fișierul Excel este: {0}";
        this["LE_InvalidOperationException_MaxColors"] = "Numărul maxim de culori dintr-un registru de lucru Excel este: {0}";
        this["LE_InvalidOperationException_MaxColumns"] = "Numărul maxim de coloane dintr-o foaie de lucru Excel cu formatul curent este: {0}";
        this["LE_InvalidOperationException_MaxFonts"] = "Numărul de fonturi dintr-un registru de lucru nu poate depăși {0}.";
        this["LE_InvalidOperationException_MaxFormattedStrings"] = "Numărul de șiruri unice dintr-un registru de lucru nu poate depăși {0}.";
        this["LE_InvalidOperationException_MaxRows"] = "Numărul maxim de rânduri dintr-o foaie de lucru Excel cu formatul curent este: {0}";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableLeftColumn"] = "Coloana de celule din partea stângă specificată pentru tabelul de date conține o regiune de celule îmbinată care se extinde în afara zonei acelor celule.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopLeftCell"] = "Celula din stânga sus a celulelor specificate pentru tabelul de date face parte dintr-o celulă îmbinată care se extinde în afara zonei acelor celule.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopRow"] = "Rândul superior de celule specificat pentru tabelul de date conține o regiune de celule combinată care se extinde în afara zonei acelor celule.";
        this["LE_InvalidOperationException_MergedCellsAppliedInTable"] = "Celulele dintr-un tabel nu pot fi combinate.";
        this["LE_InvalidOperationException_NamedReferenceNameAlreadyExists"] = "Numele \"{0}\" este utilizat de un tabel sau o referință numită în registrul de lucru cu același domeniu.";
        this["LE_InvalidOperationException_NoRegionHasBeenSpecified"] = "Nu a fost specificată nicio regiune.";
        this["LE_InvalidOperationException_NoWorksheetContextToSetShapeBounds"] = "Această suprasarcină de Adăugare nu poate fi apelată atunci când colecția de forme nu este asociată cu o Foaie de lucru. Apelați supraîncărcarea Adăugării care necesită o instanță de foaie de lucru.";
        this["LE_InvalidOperationException_OverlappingTable"] = "Există un alt tabel în regiunea specificată.";
        this["LE_InvalidOperationException_ReadOnlyFont"] = "Obiectul fontului este doar în citire și nu poate fi modificat.";
        this["LE_InvalidOperationException_ReadOnlyFormat"] = "Obiectul format este numai în citire și nu poate fi modificat.";
        this["LE_InvalidOperationException_RelativeR1C1AddressNeedsOriginCell"] = "Pentru a analiza o adresă relativă R1C1, trebuie specificată o celulă de origine.";
        this["LE_InvalidOperationException_ResizedTableMustBeOnWorksheet"] = "Tabelul trebuie să fie pe o foaie de lucru pentru a fi redimensionat.";
        this["LE_InvalidOperationException_ResolvedFormatCannotBeModified"] = "Formatul de celulă rezolvat este numai în citire și nu poate fi modificat.";
        this["LE_InvalidOperationException_ShapeCantChangeOrientation"] = "Această formă nu poate fi rotită sau răsturnată.";
        this["LE_InvalidOperationException_ShapeFillSolidIsImmutable"] = "ShapeFillSolid este imuabil. Creați în schimb un nou ShapeFillSolid.";
        this["LE_InvalidOperationException_ShapeInAnotherCollection"] = "Forma există deja pe o altă foaie de lucru sau un alt grup de forme.";
        this["LE_InvalidOperationException_ShapeOutlineSolidIsImmutable"] = "ShapeOutlineSolid este imuabil. Creați în schimb un nou ShapeOutlineSolid.";
        this["LE_InvalidOperationException_ShowFilterUIWhileHeaderRowHidden"] = "UI-ul filtrului nu poate fi afișat în timp ce rândul de antet este ascuns.";
        this["LE_InvalidOperationException_SystemColorsAccessedFromWrongThread"] = "Nu se poate accesa SystemColors dintr-un fir de fundal.";
        this["LE_InvalidOperationException_TableReferenceMustBeUsedFromInsideTable"] = "Formula are o referință de tabel structurat care poate fi utilizată numai din interiorul unui tabel.";
        this["LE_InvalidOperationException_TableReferenceToMissingTable"] = "Formula are o referință la un tabel care nu există.";
        this["LE_InvalidOperationException_TableReferenceToMissingTableColumn"] = "Formula are o referință la o coloană de tabel care nu există.";
        this["LE_InvalidOperationException_TopLeftAnchorFromOtherWorksheet"] = "Adăugarea formei la această colecție o va plasa pe o foaie de lucru diferită de TopLeftCornerCell.";
        this["LE_InvalidOperationException_TotalsRowCannotBeShownInLastRow"] = "Rândul totaluri nu poate fi afișat atunci când tabelul ocupă ultimul rând din foaia de lucru.";
        this["LE_InvalidOperationException_UnknownBIFFFormat"] = "Fișierul registrului de lucru se afla într-un format BIFF necunoscut.";
        this["LE_InvalidOperationException_UnsupportedBIFFFormat"] = "Fișierul din registrul de lucru a fost salvat în {0}, un format care nu este acceptat.";
        this["LE_InvalidOperationException_WorkbookDataViolatesFormatLimits"] = "Registrul de lucru conține date care ar depăși limitele impuse de formatul specificat.";
        this["LE_InvalidRegion"] = "Adresă de regiune nevalidă.";
        this["LE_InvalidValueType"] = "{0} nu este o valoare validă pentru proprietatea ValueType.";
        this["LE_LocationNotSingleRowOrColumn"] = "Referința locației nu este validă, deoarece celulele nu sunt toate în aceeași coloană sau rând. Selectați celule care sunt toate într-o singură coloană sau rând.";
        this["LE_LocationOrDataAreaInvalid"] = "Referința pentru locație sau interval de date nu este validă.";
        this["LE_LocationWorksheetMismatch"] = "Locația trebuie să fie din această foaie de lucru.";
        this["LE_NotSingleTargetFormula"] = "Poate accepta doar formule țintă unice.";
        this["LE_NotSupportedException_CellType"] = "{0} nu este un tip de valoare de celulă acceptat.";
        this["LE_NotSupportedException_NoPackageFactory"] = "packageFactory nu poate fi nul. Când încărcați din formatul de registru de lucru Excel2007 și utilizați ansamblul Infragistics2.Documents.Excel, trebuie să furnizați un IPackageFactory pentru a gestiona ambalarea datelor. Dacă utilizați DotNet Framework 3.0 sau o versiune ulterioară, utilizați în schimb ansamblul Infragistics3.Documents.Excel, iar ambalajul va fi gestionat de clasa WindowsBase.";
        this["LE_OpenPackagingNonConformanceException_ContainsDublinCoreRefinements"] = "IPackagePart este relația de proprietăți de bază și conține îmbunătățiri pentru elementele Dublin Core, altele decât \"dcterms: created\" și \"dcterms: modified\".";
        this["LE_OpenPackagingNonConformanceException_ContainsXmlLanguageAttribute"] = "IPackagePart este relația de proprietăți de bază și conține elemente care conțin atributul \"xml: lang\".";
        this["LE_OpenPackagingNonConformanceException_CorePropertiesRelationshipAlreadyProcessed"] = "IPackagePart este relația de proprietăți de bază, dar una a fost deja procesată pentru IPackage-ul asociat.";
        this["LE_OpenPackagingNonConformanceException_DuplicatePartName"] = "Un IPackagePart cu același nume aparține deja acestui IPackage. Un IPackage nu poate conține părți cu nume identice. Rețineți că verificarea valorilor duplicate nu face diferență între majuscule și minuscule.";
        this["LE_OpenPackagingNonConformanceException_UsesMarkupCompatibilityNamespace"] = "IPackagePart este relația de proprietăți de bază și folosește spațiul de nume de compatibilitate de marcare.";
        this["LE_OpenPackagingNonConformanceException_XmlContainsDocumentTypeDefinition"] = "Un IPackagePart conținea o declarație DTD (Document Type Definition). Regulile Convenției Open Packaging interzic utilizarea conținutului XML cu declarațiile DTD.";
        this["LE_OpenPackagingNonConformanceException_XsiTypeAttributeInvalid"] = "IPackagePart este relația de proprietăți de bază și conține elemente care conțin atributul \"xsi: type\" sau conține elemente care sunt necesare și nu conțin atributul \"xsi: type\".";
        this["LE_OpenXmlSerializerNotLoaded"] = "Formatul '{0}' necesită un serializator care nu este încărcat, în JavaScript acesta este conținut în fișierul infragistics.excel_serialization_openxml.js";
        this["LE_PercentageValueOutOfRange"] = "Valoarea procentuală {0} trebuie să fie între 0 și 100.";
        this["LE_PercentileValueOutOfRange"] = "Valoarea percentilă {0} trebuie să fie între 0 și 100.";
        this["LE_PercentRangeInvalid"] = "{0} trebuie să fie mai mic decât {1}.";
        this["LE_RowsToRepeatAtTopAreOutsideAvailableRange"] = "Opțiunile de imprimare au rânduri care se repetă în partea de sus, care se află în afara gamei de rânduri disponibile în noul format.";
        this["LE_Serialization_NoPrimaryCategoryAxis"] = "Tipul de diagramă '{0}' necesită o axă a categoriei principale. Adăugați o axă de categorie principală în colecția Axis a diagramei.";
        this["LE_Serialization_NoPrimaryValueAxis"] = "Tipul de diagramă '{0}' necesită o axă a valorii principale. Adăugați o axă de valoare primară la colecția Axis a diagramei.";
        this["LE_Series_ChartType_ChartEx"] = "{0} este nevalid pentru o serie ChartType dintr-un grafic '{1}'.";
        this["LE_Series_ChartType_InvalidForComboChart"] = "Tipul de diagramă '{0}' nu este valid pentru utilizarea într-o diagramă combinată.";
        this["LE_Series_ChartType_NonCombo"] = "Tipul graficului pentru această serie este determinat de tipul graficului părinte ('{0}') și nu poate fi setat. Setați proprietatea ChartType a diagramei părinte la '{1}'.";
        this["LE_Series_Explosion_OutOfRange"] = "Proprietatea Explosion acceptă valori cuprinse între 0 și 400, inclusiv.";
        this["LE_Series_InvalidAxisGroup"] = "Proprietatea AxisGroup nu poate fi setată la '{0}' deoarece există o serie de tip '{1}' pe aceeași axă, care nu poate împărtăși aceeași axă ca o serie de tip '{2}'.";
        this["LE_Series_InvalidSeriesTypeForComboChart"] = "O serie de tip '{0}' nu poate fi utilizată într-o diagramă combinată.";
        this["LE_Series_NoPrimaryAxis"] = "Cel puțin o serie trebuie să fie trasată pe axa primară. Schimbați proprietatea AxisGroup la \"Principal\" pentru cel puțin o serie.";
        this["LE_Series_OwningSeries_NotClusteredColumn"] = "Proprietatea OwningSeries poate fi setată doar la o serie de coloane grupate într-o diagramă Pareto.";
        this["LE_Series_OwningSeries_NotParetoLine"] = "Proprietatea OwningSeries poate fi setată numai pentru o serie de linii dintr-o diagramă Pareto.";
        this["LE_Series_OwningSeries_Same"] = "OwningSeries nu se poate referi la sine.";
        this["LE_Series_SeriesType_ChartEx"] = "Valoarea SeriesType '{0}' nu este validă pentru un grafic '{1}'.";
        this["LE_Series_SeriesType_NonCombo"] = "Proprietatea Type se aplică numai pentru diagramele combinate și nu poate fi setată pentru o diagramă de tip '{0}'.";
        this["LE_SetValue_Formula"] = "Tipul de valoare '{0}' este valid numai pentru formule. Folosiți în schimb metoda SetFormula.";
        this["LE_SetValue_NonNumeric"] = "Parametrul de valoare nu se aplică pentru tipul de valoare '{0}'. Utilizați suprasarcina care necesită doar un tip de valoare.";
        this["LE_SetValue_Numeric"] = "Parametrul de valoare trebuie specificat pentru tipul de valoare '{0}'. Utilizați suprasarcina care ia atât valoarea, cât și tipul de valoare.";
        this["LE_WorksheetChart_ChartType_ChartEx_CannotTransition"] = "Proprietatea ChartType nu poate fi schimbată din '{0}' în '{1}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_From"] = "Proprietatea ChartType a unei diagrame combo nu poate fi schimbată în '{0}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_To"] = "O diagramă de tip '{0}' nu poate fi schimbată în \"Combo\". O nouă diagramă combinată poate fi creată utilizând metoda AddChart, dar schimbarea unei diagrame existente într-o diagramă combinată nu este acceptată.";
        this["LE_WorksheetChart_ComboChart_NotEnoughSeries"] = "O diagramă combinată necesită cel puțin 2 serii. Adăugați mai multe articole în SeriesCollection a graficului.";
        this["LE_WorksheetChart_MissingRequiredAxis_StockCharts"] = "O diagramă '{0}' necesită o axă de categorie principală, o axă de valoare primară, o axă de categorie secundară și o axă de valoare secundară.";
        this["LE_WorksheetChart_NotEnoughSeries"] = "Un grafic '{0}' necesită minimum {1} serii. Pentru a utiliza acest tip de diagramă, trebuie să adăugați mai multe articole la SeriesCollection.";
        this["LE_WorksheetChart_Save_Series"] = "O diagramă '{0}' trebuie să conțină cel puțin o serie. Adăugați mai multe serii în diagramă înainte de a încerca să salvați registrul de lucru.";
        this["LE_WorksheetChart_Save_Series_Pareto"] = "O diagramă \"Pareto\" trebuie să conțină o serie \"ColumnClustered\" și o serie \"Pareto\". Adăugați seria corespunzătoare în diagramă înainte de a încerca să salvați registrul de lucru.";
        this["LE_WorksheetChart_SetComboChartSourceData_InvalidChartTypes"] = "Unul sau mai multe dintre tipurile de diagrame specificate nu sunt valabile pentru utilizare într-o diagramă combinată: {0}";
        this["LE_WorksheetChart_SetComboChartSourceData_NotAComboChart"] = "SetComboChartSourceData nu poate fi apelat deoarece tipul graficului curent este '{0}'. Schimbați ChartType la \"Combo\".";
        this["LE_WorksheetChart_SetComboChartSourceData_SeriesChartTypesNotSupported"] = "Parametrul seriesChartTypes trebuie să fie nul pentru o diagramă '{0}'.";
        this["LE_WorksheetChart_SetSourceData_NotEnoughSeries"] = "Diagramele combinate necesită cel puțin 2 serii. Schimbați parametrul \"dataRange\" într-un interval care va avea ca rezultat cel puțin 2 serii.";
        this["LE_WorksheetChart_SetSourceData_TooManyAxes"] = "Combinația specificată de tipuri de diagrame necesită mai mult decât numărul maxim admis de axe.";
        this["LE_WorksheetChart_StockChartNotSupported"] = "Editarea tipului de diagramă '{0}' nu este acceptată în prezent.";
        this["LE_WorksheetChart_UnknownChartNotSupported"] = "Această diagramă a fost încărcată dintr-o foaie de lucru existentă, dar ChartType nu a putut fi determinată. Editarea tipurilor de diagrame necunoscute nu este acceptată în prezent.";
        this["LER_ArgumentOutOfRangeException_OutlineLevel"] = "Nivelul conturului trebuie să fie între 0 și 7.";
        this["LER_Exception_KeyNotFound"] = "Cheia nu a fost găsită.";
        this["LER_Exception_MergedRegionsOverlap"] = "Regiunile celulare combinate nu se pot suprapune.";
        this["TableHeaderRowDescription"] = "rândul de antet";
        this["TableInsertRowDescription"] = "introduceți rândul";
        this["TableTotalsRowDescription"] = "rânduri totaluri";
        this["Value_UCErrorCode_Div"] = "#DIV/0!";
        this["Value_UCErrorCode_Fail"] = "#FAIL";
        this["Value_UCErrorCode_NA"] = "#N/A";
        this["Value_UCErrorCode_Name"] = "#NAME?";
        this["Value_UCErrorCode_Null"] = "#NULL!";
        this["Value_UCErrorCode_Num"] = "#NUM!";
        this["Value_UCErrorCode_Ok"] = "#OK";
        this["Value_UCErrorCode_Reference"] = "#REF!";
        this["Value_UCErrorCode_Unknown"] = "#N/A";
        this["Value_UCErrorCode_Value"] = "#VALUE!";
        this["WorkbookColorInfo_Automatic_Description"] = "Automat";
        this["WorkbookColorInfo_Description"] = "ColorInfo: {0}";
        this["WorkbookColorInfo_WithTint_Description"] = "ColorInfo: {0}, Tint: {1}";
        this["WorksheetShapeSerialization_GroupName"] = "Grup";
        this["WorksheetShapeSerialization_ImageName"] = "Imagine";
        this["WorksheetShapeSerialization_ShapeName"] = "Formă";
    }
}
