// igniteui-react-excel/esm2015/lib/ExcelComparisonOperator.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various comparisons which can be used in the [[CustomFilterCondition]].
 * @see [[CustomFilterCondition.comparisonOperator]]
 */
export var ExcelComparisonOperator = /*@__PURE__*/ (function (ExcelComparisonOperator) {
    /**
     * Filter in values which are equal to the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["Equals"] = 0] = "Equals";
    /**
     * Filter in values which are not equal to the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["NotEqual"] = 1] = "NotEqual";
    /**
     * Filter in values which are greater than the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["GreaterThan"] = 2] = "GreaterThan";
    /**
     * Filter in values which are greater than or equal to the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["GreaterThanOrEqual"] = 3] = "GreaterThanOrEqual";
    /**
     * Filter in values which are less than the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["LessThan"] = 4] = "LessThan";
    /**
     * Filter in values which are less than or equal to the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["LessThanOrEqual"] = 5] = "LessThanOrEqual";
    /**
     * Filter in string values which begin with the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["BeginsWith"] = 6] = "BeginsWith";
    /**
     * Filter in string values which do not begin with the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["DoesNotBeginWith"] = 7] = "DoesNotBeginWith";
    /**
     * Filter in string values which ends with the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["EndsWith"] = 8] = "EndsWith";
    /**
     * Filter in string values which do not end with the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["DoesNotEndWith"] = 9] = "DoesNotEndWith";
    /**
     * Filter in string values which contain the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["Contains"] = 10] = "Contains";
    /**
     * Filter in string values which do not contain the comparison value.
     */
    ExcelComparisonOperator[ExcelComparisonOperator["DoesNotContain"] = 11] = "DoesNotContain";
    return ExcelComparisonOperator;
})({});
/**
 * @hidden
 */
export let ExcelComparisonOperator_$type = /*@__PURE__*/ markEnum('ExcelComparisonOperator', 'Equals,0|NotEqual,1|GreaterThan,2|GreaterThanOrEqual,3|LessThan,4|LessThanOrEqual,5|BeginsWith,6|DoesNotBeginWith,7|EndsWith,8|DoesNotEndWith,9|Contains,10|DoesNotContain,11');
