// igniteui-react-excel/esm2015/lib/ExcelLocaleZhHant.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
export class ExcelLocaleZhHant {
    constructor() {
        this["DefaultTotalLabel"] = "總計";
        this["Error_AnchoredReference"] = "嘗試存取 {0} 作為未固定的參考";
        this["Error_ExplicitThis"] = "{0} - 顯式不支援";
        this["Error_ExternalRange"] = "範圍包含外部參考";
        this["Error_FormulaNotSupported"] = "{0} - 嘗試存取不支援公式的參考上的公式";
        this["Error_Internal"] = "方法 {0} 中發生內部錯誤。";
        this["Error_InvalidFunction"] = "{1} 處的函數 '{0}' 無效。";
        this["Error_InvalidFunctionArgCountTooFew"] = "在 {1} 處引用的函數 '{0}' 不包含正確數量的引數。該函數需要最少 {2} 個參數。";
        this["Error_InvalidFunctionArgCountTooMany"] = "在 {1} 處引用的函數 '{0}' 不包含正確數量的引數。引數計數不得超過 {2} 個參數。";
        this["Error_InvalidNumberSystemDigit"] = "基數為 {2} 的數字系統的值 '{1}' 中遇到無效數字 '{0}'。";
        this["Error_InvalidOperand"] = "無效運算元";
        this["Error_InvalidOperatorArgCount"] = "'{0}' 函數必須接受 {1} 引數。";
        this["Error_InvalidReference"] = "無效或不可用的參考";
        this["Error_Location"] = "第 {1} 欄第 {0} 行";
        this["Error_MethodNotSupported"] = "{0} 不支援 {1}。";
        this["Error_NoCurrentRangeElement"] = "無當前範圍元素";
        this["Error_NullFunctionResult"] = "評估 '{0}' 返回空值。";
        this["Error_NullOperand"] = "空運算元";
        this["Error_ParseRef"] = "{0} 處的參考不明。";
        this["Error_RangeFromLast"] = "fromT.Last 未設定";
        this["Error_RangeFromNull"] = "RefRange: from 為空";
        this["Error_RangeFromRefBase"] = "RefRange: from 不是 RefBase : {0}";
        this["Error_RangeLast"] = "RefRange Validate: 最後一個元組 [{0}] != [{1}]";
        this["Error_RangeNormalizeScope"] = "RefRange Normalize: 錯誤範圍 [{0}/{1}]";
        this["Error_RangeToNull"] = "RefRange: 至為空值";
        this["Error_RangeToRefBase"] = "RefRange: to 不是 RefBase : {0}";
        this["Error_RangeTuple"] = "RefRange Validate: 元組 [{0}] !=  [{1}]";
        this["Error_RangeValidateCount"] = "RefRange Validate: 從計數 {0} != 到計數 {1}";
        this["Error_ReadOnlyReference"] = "{0} - 嘗試設定唯讀參考值";
        this["Error_ReferenceNotEumerable"] = "{0} 不是可引用的參考";
        this["Error_RefMalformed"] = "{0} - 格式錯誤的引用 {1}";
        this["Error_RP_ContainsBase"] = "RefParser.Contains - 空基數";
        this["Error_RP_ContainsTarget"] = "RefParser.Contains - 空目標";
        this["Error_RP_EmptyElement"] = "無效參考 {0}。在位置 {1} 遇到空元素。";
        this["Error_RP_EmptyReference"] = "無效參考 {0}。引用為空。";
        this["Error_RP_MalformedExtra"] = "RefParser 格式錯誤的參考: 在 [{1}] 中發現多個 [{0}]";
        this["Error_RP_MalformedUnexpected"] = "RefParser 格式錯誤的參考: 在 [{1}] 中發現意外的 [{0}]";
        this["Error_RP_MergeRelativeBase"] = "RefParser.Merge: 基準是相對的";
        this["Error_RP_MissingCloseParenthesis"] = "無效參考 {0}。缺少右括號。";
        this["Error_RP_MissingEndQuotes"] = "無效參考 {0}。缺少尾引號。";
        this["Error_RT_Expected"] = "RefTuple: 預期 [{1}] 中的 [{0}]";
        this["Error_RT_InvalidScope"] = "無效範圍: {0}。";
        this["Error_RT_InvalidScope_CharactersAfterClosingQuotes"] = "關閉位置 {1} 處 {0} 範圍內的雙引號後遇到的字元。";
        this["Error_RT_InvalidTuple_CharactersAfterScopeEnd"] = "無效元素 {0}。範圍結束後遇到的字元。";
        this["Error_RT_InvalidTuple_MissingCloseParenthesis"] = "無效元素 {0}。缺少右括號。";
        this["Error_RT_InvalidTuple_MissingEndQuotes"] = "無效元素 {0}。缺少尾引號。";
        this["Error_RT_InvalidTuple_NamePortionEmpty"] = "無效元素 {0}。元素名稱部分為空白。";
        this["Error_RT_InvalidTuple_ScopeIndexLarge"] = "無效元素 {0}。索引 {1} 無效: {2}。";
        this["Error_RT_InvalidTuple_UnescapedCharacter"] = "{1} 在位置 {2} 遇到未轉義的 {0} 字元。";
        this["Error_RT_NullRefName"] = "RefTuple: null RefName";
        this["Error_RT_Unexpected"] = "RefTuple: [{1}] 中的意外 [{0}]";
        this["Error_ScopeAllNotLast"] = "{0} - 全部作用域必須出現在最後一個元素 {1} 中";
        this["Error_UCErrorCode_Div"] = "除以零";
        this["Error_UCErrorCode_Fail"] = "失敗";
        this["Error_UCErrorCode_NA"] = "不適用";
        this["Error_UCErrorCode_Name"] = "無效名稱";
        this["Error_UCErrorCode_Null"] = "空值";
        this["Error_UCErrorCode_Num"] = "函數或公式中的無效數值。";
        this["Error_UCErrorCode_Ok"] = "成功";
        this["Error_UCErrorCode_Reference"] = "無效或不可用的參考";
        this["Error_UCErrorCode_Unknown"] = "無效代碼";
        this["Error_UCErrorCode_Value"] = "引數或運算元的類型不正確。";
        this["Error_UnexpectedScope"] = "{0} - 父級 {1} 的意外範圍";
        this["Error_UnknownRange"] = "未知範圍類型";
        this["Error_UnknownRefType"] = "{0} - 未知的 RefType {1}";
        this["Error_UnknownScope"] = "{0} - 未知範圍 {1}";
        this["Error_UnresolvedExternal"] = "未解析的外部參考";
        this["Func_abs_Arg_0"] = "number";
        this["Func_abs_ArgDesc_0"] = "求取絕對值的實數。";
        this["Func_abs_Category"] = "Math";
        this["Func_abs_CategoryURL"] = "Math";
        this["Func_abs_Desc"] = "返回某個數字的絕對值。數字的絕對值是沒有符號的數字。";
        this["Func_acos_Arg_0"] = "number";
        this["Func_acos_ArgDesc_0"] = "求取的角度的餘弦值，必須為 -1 至 1。";
        this["Func_acos_Category"] = "Math";
        this["Func_acos_CategoryURL"] = "Math";
        this["Func_acos_Desc"] = "返回數字的反餘弦或反餘弦。反餘弦是餘弦為指定數的角度。返回的角度為弧度，範圍為 0 至 pi。要將結果從弧度轉換為度數，請乘以 180 / PI()。";
        this["Func_acosh_Arg_0"] = "number";
        this["Func_acosh_ArgDesc_0"] = "大於 1 的實數，可找到反雙曲餘弦值。";
        this["Func_acosh_Category"] = "Math";
        this["Func_acosh_CategoryURL"] = "Math";
        this["Func_acosh_Desc"] = "返回指定數字的反雙曲餘弦值。反雙曲餘弦是其雙曲餘弦為指定數的角度。";
        this["Func_and_Arg_0"] = "logical{0}";
        this["Func_and_ArgDesc_0"] = "邏輯值 (true 或 false)。";
        this["Func_and_Category"] = "Logical";
        this["Func_and_CategoryURL"] = "Logical";
        this["Func_and_Desc"] = "如果所有值均為 true，則返回 true; 如果一個或多個值均為 false，則返回 false。";
        this["Func_asin_Arg_0"] = "number";
        this["Func_asin_ArgDesc_0"] = "求取的角度的正弦，必須為 -1 至 1 弧度。";
        this["Func_asin_Category"] = "Math";
        this["Func_asin_CategoryURL"] = "Math";
        this["Func_asin_Desc"] = "返回數字的反正弦或反正弦。反正弦是其正弦為指定數字的角度。返回的角度為弧度，範圍為 -pi/2 至 pi/2。要以度表示反正弦，請將結果乘以 180 / PI()。";
        this["Func_asinh_Arg_0"] = "number";
        this["Func_asinh_ArgDesc_0"] = "尋找反雙曲正弦的實數。";
        this["Func_asinh_Category"] = "Math";
        this["Func_asinh_CategoryURL"] = "Math";
        this["Func_asinh_Desc"] = "返回指定數字的反雙曲正弦值。反雙曲正弦是雙曲正弦為指定數的角度。";
        this["Func_atan_Arg_0"] = "number";
        this["Func_atan_ArgDesc_0"] = "求取的角度的切線。";
        this["Func_atan_Category"] = "Math";
        this["Func_atan_CategoryURL"] = "Math";
        this["Func_atan_Desc"] = "返回數字的反正切或反正切。反正切是其切線為指定數字的角度。返回的角度為弧度，範圍為 -pi/2 至 pi/2。要以度表示反正切，請將結果乘以 180 / PI()。";
        this["Func_atan2_Arg_0"] = "x";
        this["Func_atan2_Arg_1"] = "y";
        this["Func_atan2_ArgDesc_0"] = "點的 x 座標。";
        this["Func_atan2_ArgDesc_1"] = "點的 y 座標。";
        this["Func_atan2_Category"] = "Math";
        this["Func_atan2_CategoryURL"] = "Math";
        this["Func_atan2_Desc"] = "返回指定 x 和 y 座標的反正切或反正切。反正切是從x軸到包含原點 (0，0) 和座標為 (x_num，y_num) 的點的線的夾角。角度以 -pi 和 pi 之間的弧度表示，不包括 -pi。";
        this["Func_atanh_Arg_0"] = "number";
        this["Func_atanh_ArgDesc_0"] = "求取反雙曲正切的 -1 和 1 之間的實數。";
        this["Func_atanh_Category"] = "Math";
        this["Func_atanh_CategoryURL"] = "Math";
        this["Func_atanh_Desc"] = "返回指定數字的反雙曲正切值。反雙曲正切是其雙曲正切為指定數的角度。";
        this["Func_avedev_Arg_0"] = "number{0}";
        this["Func_avedev_ArgDesc_0"] = "Number1，number2，...是一個或多個數值或數值引用。";
        this["Func_avedev_Category"] = "Statistical";
        this["Func_avedev_CategoryURL"] = "Statistical";
        this["Func_avedev_Desc"] = "返回與所有數值平均值的平均值偏差。";
        this["Func_average_Arg_0"] = "number{0}";
        this["Func_average_ArgDesc_0"] = "一個數字或數字範圍，可與其他引數一起尋找平均值。";
        this["Func_average_Category"] = "Statistical";
        this["Func_average_CategoryURL"] = "Statistical";
        this["Func_average_Desc"] = "返回一個或多個值的平均值 (算術平均值)。";
        this["Func_bin2dec_Arg_0"] = "Number";
        this["Func_bin2dec_ArgDesc_0"] = "重新預設二進制數的文字表達式。";
        this["Func_bin2dec_Category"] = "Engineering";
        this["Func_bin2dec_CategoryURL"] = "Engineering";
        this["Func_bin2dec_Desc"] = "返回二進制格式中數字的文字表示形式的十進制值。";
        this["Func_bin2hex_Arg_0"] = "Number";
        this["Func_bin2hex_Arg_1"] = "Places";
        this["Func_bin2hex_ArgDesc_0"] = "重新預設二進制數的文字表達式。";
        this["Func_bin2hex_ArgDesc_1"] = "用於表示值的字元數。如果未指定，將使用代表該值的最少字元數。這允許您指定前導零。";
        this["Func_bin2hex_Category"] = "Engineering";
        this["Func_bin2hex_CategoryURL"] = "Engineering";
        this["Func_bin2hex_Desc"] = "在十六進制數字系統中，將表示二進制數的文字表達式轉換為表示相同數字的文字表達式。";
        this["Func_bin2oct_Arg_0"] = "Number";
        this["Func_bin2oct_Arg_1"] = "Places";
        this["Func_bin2oct_ArgDesc_0"] = "重新預設二進制數的文字表達式。";
        this["Func_bin2oct_ArgDesc_1"] = "用於表示值的字元數。如果未指定，將使用代表該值的最少字元數。這允許您指定前導零。";
        this["Func_bin2oct_Category"] = "Engineering";
        this["Func_bin2oct_CategoryURL"] = "Engineering";
        this["Func_bin2oct_Desc"] = "在八進制數字系統中，將表示二進制數的文字表達式轉換為表示相同數字的文字表達式。";
        this["Func_binomdist_Arg_0"] = "Number";
        this["Func_binomdist_Arg_1"] = "試用版";
        this["Func_binomdist_Arg_2"] = "機率";
        this["Func_binomdist_Arg_3"] = "累計";
        this["Func_binomdist_ArgDesc_0"] = "試用成功次數。";
        this["Func_binomdist_ArgDesc_1"] = "獨立試驗的次數。";
        this["Func_binomdist_ArgDesc_2"] = "每次試驗成功的概率。";
        this["Func_binomdist_ArgDesc_3"] = "決定功能形式的邏輯值。如果為真，函數將返回累積分佈，該累積分佈是最多有 \"數字\" 個成功的概率，否則它將返回概率質量，即有 \"個數\" 成功的概率。";
        this["Func_binomdist_Category"] = "Statistical";
        this["Func_binomdist_CategoryURL"] = "Statistical";
        this["Func_binomdist_Desc"] = "返回給定一組項目和從該組中選擇的若干項目的可能組合的數量 (重複)。";
        this["Func_CategoryURL_Template"] = "ms-help://INFRAGISTICS_HELP/NetAdvantagev{2}{3}/Win/WinElements/WinCalcManager_{0}_Functions.html#FUNC_{1}";
        this["Func_ceiling_Arg_0"] = "number";
        this["Func_ceiling_Arg_1"] = "significance";
        this["Func_ceiling_ArgDesc_0"] = "要捨入的數字。";
        this["Func_ceiling_ArgDesc_1"] = "四捨五入到的倍數。如果省略，則假定為 1。";
        this["Func_ceiling_Category"] = "Math";
        this["Func_ceiling_CategoryURL"] = "Math";
        this["Func_ceiling_Desc"] = "返回大於或等於指定數字的重要性的最小倍數 (預設為 1)。";
        this["Func_char_Arg_0"] = "charCode";
        this["Func_char_ArgDesc_0"] = "要轉換為字元的數字。";
        this["Func_char_Category"] = "TextAndData";
        this["Func_char_CategoryURL"] = "Text_and_Data";
        this["Func_char_Desc"] = "從您的電腦的字元集中返回由代碼編號指定的字元。";
        this["Func_choose_Arg_0"] = "IndexNum";
        this["Func_choose_Arg_1"] = "值 {0}";
        this["Func_choose_ArgDesc_0"] = "從以 1 為基礎的索引，到要使用的值的其餘引數。";
        this["Func_choose_ArgDesc_1"] = "可以根據 IndexNum 值選擇一個值。";
        this["Func_choose_Category"] = "LookupAndReference";
        this["Func_choose_CategoryURL"] = "Lookup_And_Reference";
        this["Func_choose_Desc"] = "根據第一個引數中的數字，返回其引數中提供的值之一。";
        this["Func_code_Arg_0"] = "text";
        this["Func_code_ArgDesc_0"] = "將評估其第一個字元的文字表達式。";
        this["Func_code_Category"] = "TextAndData";
        this["Func_code_CategoryURL"] = "Text_and_Data";
        this["Func_code_Desc"] = "使用您的電腦使用的字元集，返回文字表達式中第一個字元的數字代碼。";
        this["Func_combin_Arg_0"] = "Number";
        this["Func_combin_Arg_1"] = "NumberChosen";
        this["Func_combin_ArgDesc_0"] = "項目數。";
        this["Func_combin_ArgDesc_1"] = "每個組合中選擇的項目數。";
        this["Func_combin_Category"] = "Math";
        this["Func_combin_CategoryURL"] = "Math";
        this["Func_combin_Desc"] = "返回給定一組項目和從該組中選擇的項目的可能組合的數量。";
        this["Func_combina_Arg_0"] = "Number";
        this["Func_combina_Arg_1"] = "NumberChosen";
        this["Func_combina_ArgDesc_0"] = "項目數。";
        this["Func_combina_ArgDesc_1"] = "每個組合中選擇的項目數。";
        this["Func_combina_Category"] = "Math";
        this["Func_combina_CategoryURL"] = "Math";
        this["Func_combina_Desc"] = "返回給定一組項目和從該組中選擇的若干項目的可能組合的數量 (重複)。";
        this["Func_complex_Arg_0"] = "RealNumber";
        this["Func_complex_Arg_1"] = "ImaginaryNumber";
        this["Func_complex_Arg_2"] = "Suffix";
        this["Func_complex_ArgDesc_0"] = "複數的實係數。";
        this["Func_complex_ArgDesc_1"] = "數字的虛數係數。";
        this["Func_complex_ArgDesc_2"] = "指定要使用的後綴。可接受的值為 \"i\" 或 \"j\"。如果省略，則使用 \"i\"。注意後綴區分大小寫; 不允許使用大寫字母 \"I\" 和 \"J\"，並導致 #VALUE 錯誤。";
        this["Func_complex_Category"] = "Engineering";
        this["Func_complex_CategoryURL"] = "Engineering";
        this["Func_complex_Desc"] = "透過結合實數和虛數，以 \"x + yi\" 或 \"x + yj\" 格式返回以文字表達式表示的複數。";
        this["Func_concatenate_Arg_0"] = "text{0}";
        this["Func_concatenate_ArgDesc_0"] = "文字表達式或一系列文字表達式，將被合併為一個文字表達式。";
        this["Func_concatenate_Category"] = "TextAndData";
        this["Func_concatenate_CategoryURL"] = "Text_and_Data";
        this["Func_concatenate_Desc"] = "將所有引數連結成一個文字表達。";
        this["Func_convert_Arg_0"] = "Number";
        this["Func_convert_Arg_1"] = "FromUnit";
        this["Func_convert_Arg_2"] = "ToUnit";
        this["Func_convert_ArgDesc_0"] = "要轉換的值。";
        this["Func_convert_ArgDesc_1"] = "給出數字的單位。\r\n    \r\n重量和質量單位: \"g\" = 克; \"sg\" = 金屬塊; \"lbm\" = 磅質量 (常衡制); \"u\" = U (原子質量單位); \"ozm\" = 盎司質量 (常衡制)\r\n距離單位: \"m\" = 儀表; \"mi\" = 法定英里; \"Nmi\" = 海里; \"in\" = 英寸; \"ft\" = 英尺; \"yd\" = 碼; \"ang\" = 埃; \"Pica\" = Pica (1/72 英寸)\r\n時間單位: \"yr\" = 天; \"day\" = 天; \"hr\" = 小時; \"mn\" = 分鐘; \"sec\" = 秒\r\n壓力單位: \"Pa\" (或 \"p\") = 帕斯卡; \"atm\" (或 \"at\") = 大氣; \"mmHg\" = 毫米汞柱\r\n力單位: \"N\" = 牛頓; \"dyn\" (或 \"dy\") = 達因; \"lbf\" = 磅力\r\n能源單位: \"J\" = 焦耳; \"e\" = 爾格; \"c\" = 熱力學卡路里; \"cal\" = IT 卡路里; \"eV\" (或 \"ev\") = 電子伏特; \"HPh\" (或 \"hh\") = 馬力小時; \"Wh\" (或 \"wh\") = 瓦特小時; \"flb\" = 英尺磅; \"BTU\" (或 \"btu\") = BTU\r\n功率單位: \"HP\" (或 \"h\") = 馬力; \"W\" (或 \"w\") = 瓦特\r\n磁性單位: \"T\" = 特斯拉; \"ga\" = 高斯\r\n溫度單位: \"C\" (或 \"cel\") = 攝氏度; \"F\" (或 \"fah\") = 華氏度; \"K\" (或 \"kel\") = 開爾文\r\n液體測量單位: \"tsp\" = 茶匙; \"tbs\" = 湯匙; \"oz\" = 液量盎司; \"cup\" = 杯; \"pt\" (或 \"us_pt\") = 美國品脫; \"uk_pt\" = 英國品脫; \"qt\" = 夸脫; \"gal\" = 加侖; \"l\" (或 \"lt\") = 升\r\n\r\n任何度量單位都可以加上以下其中一項作為前綴: exa (\"E\") = 1E+18; peta (\"P\") = 1E+15; tera (\"T\") = 1E+12; giga (\"G\") = 1E+09; mega (\"M\") = 1E+06; kilo (\"k\") = 1E+03; hecto (\"h\") = 1E+02; dekao (\"e\") = 1E+01; deci (\"d\") = 1E-01; centi (\"c\") = 1E-02; milli (\"m\") = 1E-03; micro (\"u\") = 1E-06; nano (\"n\") = 1E-09; pico (\"p\") = 1E-12; femto (\"f\") = 1E-15; atto (\"a\") = 1E-18\r\n    ";
        this["Func_convert_ArgDesc_2"] = "返迴轉換值的單位。\r\n    \r\n重量和質量單位: \"g\" = 克; \"sg\" = 金屬塊; \"lbm\" = 磅質量 (常衡制); \"u\" = U (原子質量單位); \"ozm\" = 盎司質量 (常衡制)\r\n距離單位: \"m\" = 儀表; \"mi\" = 法定英里; \"Nmi\" = 海里; \"in\" = 英寸; \"ft\" = 英尺; \"yd\" = 碼; \"ang\" = 埃; \"Pica\" = Pica (1/72 英寸)\r\n時間單位: \"yr\" = 天; \"day\" = 天; \"hr\" = 小時; \"mn\" = 分鐘; \"sec\" = 秒\r\n壓力單位: \"Pa\" (或 \"p\") = 帕斯卡; \"atm\" (或 \"at\") = 大氣; \"mmHg\" = 毫米汞柱\r\n力單位: \"N\" = 牛頓; \"dyn\" (或 \"dy\") = 達因; \"lbf\" = 磅力\r\n能源單位: \"J\" = 焦耳; \"e\" = 爾格; \"c\" = 熱力學卡路里; \"cal\" = IT 卡路里; \"eV\" (或 \"ev\") = 電子伏特; \"HPh\" (或 \"hh\") = 馬力小時; \"Wh\" (或 \"wh\") = 瓦特小時; \"flb\" = 英尺磅; \"BTU\" (或 \"btu\") = BTU\r\n功率單位: \"HP\" (或 \"h\") = 馬力; \"W\" (或 \"w\") = 瓦特\r\n磁性單位: \"T\" = 特斯拉; \"ga\" = 高斯\r\n溫度單位: \"C\" (或 \"cel\") = 攝氏度; \"F\" (或 \"fah\") = 華氏度; \"K\" (或 \"kel\") = 開爾文\r\n液體測量單位: \"tsp\" = 茶匙; \"tbs\" = 湯匙; \"oz\" = 液量盎司; \"cup\" = 杯; \"pt\" (或 \"us_pt\") = 美國品脫; \"uk_pt\" = 英國品脫; \"qt\" = 夸脫; \"gal\" = 加侖; \"l\" (或 \"lt\") = 升\r\n\r\n任何度量單位都可以加上以下其中一項作為前綴: exa (\"E\") = 1E+18; peta (\"P\") = 1E+15; tera (\"T\") = 1E+12; giga (\"G\") = 1E+09; mega (\"M\") = 1E+06; kilo (\"k\") = 1E+03; hecto (\"h\") = 1E+02; dekao (\"e\") = 1E+01; deci (\"d\") = 1E-01; centi (\"c\") = 1E-02; milli (\"m\") = 1E-03; micro (\"u\") = 1E-06; nano (\"n\") = 1E-09; pico (\"p\") = 1E-12; femto (\"f\") = 1E-15; atto (\"a\") = 1E-18\r\n    ";
        this["Func_convert_Category"] = "Engineering";
        this["Func_convert_CategoryURL"] = "Engineering";
        this["Func_convert_Desc"] = "將一個測量系統的值轉換為另一個測量系統，例如米至英寸或小時至秒。";
        this["Func_cos_Arg_0"] = "number";
        this["Func_cos_ArgDesc_0"] = "求取餘弦的弧度角。";
        this["Func_cos_Category"] = "Math";
        this["Func_cos_CategoryURL"] = "Math";
        this["Func_cos_Desc"] = "返回指定角度的餘弦值。";
        this["Func_cosh_Arg_0"] = "number";
        this["Func_cosh_ArgDesc_0"] = "尋找雙曲餘弦的實數。";
        this["Func_cosh_Category"] = "Math";
        this["Func_cosh_CategoryURL"] = "Math";
        this["Func_cosh_Desc"] = "返回數字的雙曲餘弦值。";
        this["Func_count_Arg_0"] = "value{0}";
        this["Func_count_ArgDesc_0"] = "可能包含數字的參考或參考範圍。";
        this["Func_count_Category"] = "Statistical";
        this["Func_count_CategoryURL"] = "Statistical";
        this["Func_count_Desc"] = "計算包含數字的引用數。";
        this["Func_counta_Arg_0"] = "value{0}";
        this["Func_counta_ArgDesc_0"] = "可能包含數字的參考或參考範圍。";
        this["Func_counta_Category"] = "Statistical";
        this["Func_counta_CategoryURL"] = "Statistical";
        this["Func_counta_Desc"] = "計算非空白參考的數量。";
        this["Func_critbinom_Arg_0"] = "試用版";
        this["Func_critbinom_Arg_1"] = "機率";
        this["Func_critbinom_Arg_2"] = "Alpha";
        this["Func_critbinom_ArgDesc_0"] = "獨立試驗的次數。";
        this["Func_critbinom_ArgDesc_1"] = "每次試驗成功的概率。";
        this["Func_critbinom_ArgDesc_2"] = "標準值。";
        this["Func_critbinom_Category"] = "Statistical";
        this["Func_critbinom_CategoryURL"] = "Statistical";
        this["Func_critbinom_Desc"] = "返回累積二項式分佈大於或等於標準值的最小值。";
        this["Func_date_Arg_0"] = "年";
        this["Func_date_Arg_1"] = "月";
        this["Func_date_Arg_2"] = "天";
        this["Func_date_ArgDesc_0"] = "代表年份的數字。";
        this["Func_date_ArgDesc_1"] = "一個數字，代表一年中的月份。";
        this["Func_date_ArgDesc_2"] = "一個數字，代表當月的日期。";
        this["Func_date_Category"] = "DateAndTime";
        this["Func_date_CategoryURL"] = "Date_and_Time";
        this["Func_date_Desc"] = "返回代表特定日期的序列號。";
        this["Func_dateadd_Arg_0"] = "interval";
        this["Func_dateadd_Arg_1"] = "number";
        this["Func_dateadd_Arg_2"] = "日期";
        this["Func_dateadd_ArgDesc_0"] = "表示要添加時間間隔的文字表達。\r\nd = 天\r\ny = 年\r\nh = 小時\r\nn = 分鐘\r\nm = 月\r\nq = 季度\r\ns = 秒\r\nw = 週\r\nww = 日曆週\r\nyyyy = 年\r\n";
        this["Func_dateadd_ArgDesc_1"] = "要添加的間隔數。該值可以是正數 (以獲取將來的 DateTime) 或負數 (以獲取過去的 DateTime)。當間隔指定小時，分鐘或秒時，它可以包含小數部分。對於其他間隔值，將忽略 Number 的任何小數部分。";
        this["Func_dateadd_ArgDesc_2"] = "要添加間隔的日期。";
        this["Func_dateadd_Category"] = "DateAndTime";
        this["Func_dateadd_CategoryURL"] = "Date_and_Time";
        this["Func_dateadd_Desc"] = "返回已添加指定間隔的日期。";
        this["Func_datediff_Arg_0"] = "interval";
        this["Func_datediff_Arg_1"] = "date1";
        this["Func_datediff_Arg_2"] = "date2";
        this["Func_datediff_ArgDesc_0"] = "表示時間間隔的文字表達式，用作 date1 和 date2 之差的單位。\r\nd = 天\r\ny = 天\r\nh = 小時\r\nn = 分鐘\r\nm = 月\r\nq = 季度\r\ns = 秒\r\nw = 週\r\nww = 日曆週\r\nyyyy = 年\r\n";
        this["Func_datediff_ArgDesc_1"] = "差異計算中使用的第一個日期。";
        this["Func_datediff_ArgDesc_2"] = "差異計算中使用的第二個日期。";
        this["Func_datediff_Category"] = "DateAndTime";
        this["Func_datediff_CategoryURL"] = "Date_and_Time";
        this["Func_datediff_Desc"] = "返回一個值，該值指示兩個 DateTime 值之間的間隔數。";
        this["Func_datevalue_Arg_0"] = "date_text";
        this["Func_datevalue_ArgDesc_0"] = "表示日期的文字表達。";
        this["Func_datevalue_Category"] = "DateAndTime";
        this["Func_datevalue_CategoryURL"] = "Date_and_Time";
        this["Func_datevalue_Desc"] = "返回指定日期文字的日期值。";
        this["Func_day_Arg_0"] = "Date_value";
        this["Func_day_ArgDesc_0"] = "用於求取日期的日期值。";
        this["Func_day_Category"] = "DateAndTime";
        this["Func_day_CategoryURL"] = "Date_and_Time";
        this["Func_day_Desc"] = "返回日期值 (某月某天)。";
        this["Func_days360_Arg_0"] = "start_date";
        this["Func_days360_Arg_1"] = "end_date";
        this["Func_days360_Arg_2"] = "method";
        this["Func_days360_ArgDesc_0"] = "天數範圍的開始日期。如果在結束日期之後，該函數將返回負數。";
        this["Func_days360_ArgDesc_1"] = "天數範圍的結束日期。如果此日期早於開始日期，則該函數將返回負數。";
        this["Func_days360_ArgDesc_2"] = "邏輯值，指定在計算中使用美國 (false) 或歐洲 (true) 方法。";
        this["Func_days360_Category"] = "DateAndTime";
        this["Func_days360_CategoryURL"] = "Date_and_Time";
        this["Func_days360_Desc"] = "返回基於 360 天年份的兩個日期之間的天數。";
        this["Func_db_Arg_0"] = "cost";
        this["Func_db_Arg_1"] = "salvage";
        this["Func_db_Arg_2"] = "life";
        this["Func_db_Arg_3"] = "時期";
        this["Func_db_Arg_4"] = "月";
        this["Func_db_ArgDesc_0"] = "資產的初始成本。";
        this["Func_db_ArgDesc_1"] = "折舊結束時的值 (有時稱為資產的殘值)。";
        this["Func_db_ArgDesc_2"] = "折舊資產的期間數 (有時稱為資產的使用壽命)。";
        this["Func_db_ArgDesc_3"] = "計算折舊的期間。期間必須使用與生命相同的單位。";
        this["Func_db_ArgDesc_4"] = "第一年的月數。如果省略月份，則假定為 12。";
        this["Func_db_Category"] = "Financial";
        this["Func_db_CategoryURL"] = "Financial";
        this["Func_db_Desc"] = "使用固定遞減餘額法返回指定時期內資產的折舊。";
        this["Func_dbnull_Category"] = "Information";
        this["Func_dbnull_CategoryURL"] = "Information";
        this["Func_dbnull_Desc"] = "返回代表不存在值的內容。";
        this["Func_ddb_Arg_0"] = "cost";
        this["Func_ddb_Arg_1"] = "salvage";
        this["Func_ddb_Arg_2"] = "life";
        this["Func_ddb_Arg_3"] = "時期";
        this["Func_ddb_Arg_4"] = "factor";
        this["Func_ddb_ArgDesc_0"] = "資產的初始成本。";
        this["Func_ddb_ArgDesc_1"] = "折舊結束時的值 (有時稱為資產的殘值)。";
        this["Func_ddb_ArgDesc_2"] = "折舊資產的期間數 (有時稱為資產的使用壽命)。";
        this["Func_ddb_ArgDesc_3"] = "計算折舊的期間。期間必須使用與生命相同的單位。";
        this["Func_ddb_ArgDesc_4"] = "餘額下降的速率。如果省略 factor，則假定為 2 (雙倍餘額遞減法)。";
        this["Func_ddb_Category"] = "Financial";
        this["Func_ddb_CategoryURL"] = "Financial";
        this["Func_ddb_Desc"] = "使用雙倍減少餘額法或指定的其他方法，返回指定時期內資產的折舊。";
        this["Func_dec2bin_Arg_0"] = "Number";
        this["Func_dec2bin_Arg_1"] = "Places";
        this["Func_dec2bin_ArgDesc_0"] = "要轉換的十進制值。";
        this["Func_dec2bin_ArgDesc_1"] = "用於表示值的字元數。如果未指定 places，將使用代表該值的最少字元數。這允許您指定前導零。";
        this["Func_dec2bin_Category"] = "Engineering";
        this["Func_dec2bin_CategoryURL"] = "Engineering";
        this["Func_dec2bin_Desc"] = "返回表示指定十進制值為二進制數的文字表達式。";
        this["Func_dec2hex_Arg_0"] = "Number";
        this["Func_dec2hex_Arg_1"] = "Places";
        this["Func_dec2hex_ArgDesc_0"] = "要轉換的十進制值。";
        this["Func_dec2hex_ArgDesc_1"] = "用於表示值的字元數。如果未指定 places，將使用代表該值的最少字元數。這允許您指定前導零。";
        this["Func_dec2hex_Category"] = "Engineering";
        this["Func_dec2hex_CategoryURL"] = "Engineering";
        this["Func_dec2hex_Desc"] = "返回一個文字表達式，以十六進制數表示指定的十進制值。";
        this["Func_dec2oct_Arg_0"] = "Number";
        this["Func_dec2oct_Arg_1"] = "Places";
        this["Func_dec2oct_ArgDesc_0"] = "要轉換的十進制值。";
        this["Func_dec2oct_ArgDesc_1"] = "用於表示值的字元數。如果未指定 places，將使用代表該值的最少字元數。這允許您指定前導零。";
        this["Func_dec2oct_Category"] = "Engineering";
        this["Func_dec2oct_CategoryURL"] = "Engineering";
        this["Func_dec2oct_Desc"] = "返回一個文字表達式，將指定的十進制值表示為八進制數字。";
        this["Func_degrees_Arg_0"] = "Radians";
        this["Func_degrees_ArgDesc_0"] = "弧度值，將轉換為度。";
        this["Func_degrees_Category"] = "Math";
        this["Func_degrees_CategoryURL"] = "Math";
        this["Func_degrees_Desc"] = "將弧度轉換為度。";
        this["Func_delta_Arg_0"] = "Number1";
        this["Func_delta_Arg_1"] = "Number2";
        this["Func_delta_ArgDesc_0"] = "要比較的第一個數字。";
        this["Func_delta_ArgDesc_1"] = "要比較的第二個數字。如果未指定，將假定為零。";
        this["Func_delta_Category"] = "Engineering";
        this["Func_delta_CategoryURL"] = "Engineering";
        this["Func_delta_Desc"] = "比較兩個數字，如果相等則返回 1，否則返回零。";
        this["Func_dollarde_Arg_0"] = "FractionDollarAmount";
        this["Func_dollarde_Arg_1"] = "分數";
        this["Func_dollarde_ArgDesc_0"] = "以分數表示的美元金額。";
        this["Func_dollarde_ArgDesc_1"] = "該值用作分數的分母。如果此值不是整數，將被截斷。";
        this["Func_dollarde_Category"] = "Financial";
        this["Func_dollarde_CategoryURL"] = "Financial";
        this["Func_dollarde_Desc"] = "將以分數表示的美元金額轉換為以小數表示的美元金額。";
        this["Func_dollarfr_Arg_0"] = "DecimalDollarAmount";
        this["Func_dollarfr_Arg_1"] = "分數";
        this["Func_dollarfr_ArgDesc_0"] = "以小數表示的美元金額。";
        this["Func_dollarfr_ArgDesc_1"] = "該值用作分數的分母。如果此值不是整數，將被截斷。";
        this["Func_dollarfr_Category"] = "Financial";
        this["Func_dollarfr_CategoryURL"] = "Financial";
        this["Func_dollarfr_Desc"] = "將以十進制表示的美元金額轉換為以分數表示的美元金額。";
        this["Func_edate_Arg_0"] = "日期";
        this["Func_edate_Arg_1"] = "numberOfMonths";
        this["Func_edate_ArgDesc_0"] = "開始日期。";
        this["Func_edate_ArgDesc_1"] = "轉移日期的月數。正數表示開始日期之後的日期。負數表示開始日期之前的日期。";
        this["Func_edate_Category"] = "DateAndTime";
        this["Func_edate_CategoryURL"] = "DateAndTime";
        this["Func_edate_Desc"] = "返回開始日期之前或之後指定的月數。";
        this["Func_eomonth_Arg_0"] = "日期";
        this["Func_eomonth_Arg_1"] = "numberOfMonths";
        this["Func_eomonth_ArgDesc_0"] = "開始日期。";
        this["Func_eomonth_ArgDesc_1"] = "轉移日期的月數。正數表示開始日期之後的日期。負數表示開始日期之前的日期。";
        this["Func_eomonth_Category"] = "DateAndTime";
        this["Func_eomonth_CategoryURL"] = "DateAndTime";
        this["Func_eomonth_Desc"] = "返回月末的日期，該日期是開始日期之前或之後的指定月份數。";
        this["Func_errortype_Arg_0"] = "error_value";
        this["Func_errortype_ArgDesc_0"] = "應該找到其識別號碼的錯誤值。";
        this["Func_errortype_Category"] = "Information";
        this["Func_errortype_CategoryURL"] = "Information";
        this["Func_errortype_Desc"] = "返回與錯誤值對應的數字。";
        this["Func_even_Arg_0"] = "number";
        this["Func_even_ArgDesc_0"] = "要捨入的值。";
        this["Func_even_Category"] = "Math";
        this["Func_even_CategoryURL"] = "Math";
        this["Func_even_Desc"] = "將正數向上取整，負數向下取整至最接近的偶數整數。";
        this["Func_exp_Arg_0"] = "number";
        this["Func_exp_ArgDesc_0"] = "對e求乘方的指數。";
        this["Func_exp_Category"] = "Math";
        this["Func_exp_CategoryURL"] = "Math";
        this["Func_exp_Desc"] = "返回e的指定數值的次冪。常數 e 等於 2.71828182845904。";
        this["Func_fact_Arg_0"] = "Number";
        this["Func_fact_ArgDesc_0"] = "將為其計算階乘的正數。";
        this["Func_fact_Category"] = "Math";
        this["Func_fact_CategoryURL"] = "Math";
        this["Func_fact_Desc"] = "返回數字的階乘。";
        this["Func_factdouble_Arg_0"] = "Number";
        this["Func_factdouble_ArgDesc_0"] = "正因數，將為其計算雙階乘。";
        this["Func_factdouble_Category"] = "Math";
        this["Func_factdouble_CategoryURL"] = "Math";
        this["Func_factdouble_Desc"] = "返回數字的雙階乘。";
        this["Func_false_Category"] = "Logical";
        this["Func_false_CategoryURL"] = "Logical";
        this["Func_false_Desc"] = "返回邏輯值 false。";
        this["Func_find_Arg_0"] = "find_text";
        this["Func_find_Arg_1"] = "within_text";
        this["Func_find_Arg_2"] = "start_num";
        this["Func_find_ArgDesc_0"] = "要尋找的文字。";
        this["Func_find_ArgDesc_1"] = "包含要尋找的文字的文字。";
        this["Func_find_ArgDesc_2"] = "從以 1 為基礎的索引字元，開始搜尋。如果省略，則假定為 1。";
        this["Func_find_Category"] = "TextAndData";
        this["Func_find_CategoryURL"] = "Text_and_Data";
        this["Func_find_Desc"] = "返回一個文字表達式在另一個文字表達式中的起始位置，區分大小寫。";
        this["Func_fixed_Arg_0"] = "number";
        this["Func_fixed_Arg_1"] = "decimals";
        this["Func_fixed_Arg_2"] = "no_commas";
        this["Func_fixed_ArgDesc_0"] = "要捨入並轉換為文字的數字。";
        this["Func_fixed_ArgDesc_1"] = "返回值中小數點右邊的位數。如果此值為負數，則\"數字\"將捨入到小數點左邊。預設值為 2。";
        this["Func_fixed_ArgDesc_2"] = "如果逗號不應該在返回值中，則為 True。如果省略，則假定為假。";
        this["Func_fixed_Category"] = "TextAndData";
        this["Func_fixed_CategoryURL"] = "Text_and_Data";
        this["Func_fixed_Desc"] = "將數字四捨五入到指定的小數位數，並以帶或不帶逗號的文字形式返回結果。";
        this["Func_floor_Arg_0"] = "number";
        this["Func_floor_Arg_1"] = "significance";
        this["Func_floor_ArgDesc_0"] = "要捨入的數字。";
        this["Func_floor_ArgDesc_1"] = "四捨五入到的倍數。如果省略，則假定為 1。";
        this["Func_floor_Category"] = "Math";
        this["Func_floor_CategoryURL"] = "Math";
        this["Func_floor_Desc"] = "返回小於或等於指定數字的重要性的最大倍數(預設為1)。";
        this["Func_fv_Arg_0"] = "rate";
        this["Func_fv_Arg_1"] = "nper";
        this["Func_fv_Arg_2"] = "pmt";
        this["Func_fv_Arg_3"] = "pv";
        this["Func_fv_Arg_4"] = "type";
        this["Func_fv_ArgDesc_0"] = "每期利率。";
        this["Func_fv_ArgDesc_1"] = "年金的支付期總數。";
        this["Func_fv_ArgDesc_2"] = "支付的每個期間，在年金的使用期限內無法更改。";
        this["Func_fv_ArgDesc_3"] = "現在價值，或一系列未來還款額的當前總價值。如果省略，則假定為零。";
        this["Func_fv_ArgDesc_4"] = "數字零或一，表示應在期末 (零，預設值) 或是期初 (一) 支付。";
        this["Func_fv_Category"] = "Financial";
        this["Func_fv_CategoryURL"] = "Financial";
        this["Func_fv_Desc"] = "返回根據周期的投資的未來價值。";
        this["Func_gcd_Arg_0"] = "number{0}";
        this["Func_gcd_ArgDesc_0"] = "整數值。非整數值將被截斷。";
        this["Func_gcd_Category"] = "Math";
        this["Func_gcd_CategoryURL"] = "Math";
        this["Func_gcd_Desc"] = "返回一個或多個整數值的最大公約數。";
        this["Func_gestep_Arg_0"] = "Number1";
        this["Func_gestep_Arg_1"] = "Step";
        this["Func_gestep_ArgDesc_0"] = "要比較的第一個數字。";
        this["Func_gestep_ArgDesc_1"] = "要比較的第二個數字。如果未指定，將假定為零。";
        this["Func_gestep_Category"] = "Engineering";
        this["Func_gestep_CategoryURL"] = "Engineering";
        this["Func_gestep_Desc"] = "比較兩個數字，如果第一個數字大於或等於第二個數字，則返回一個，否則返回零。";
        this["Func_hex2bin_Arg_0"] = "Number";
        this["Func_hex2bin_Arg_1"] = "Places";
        this["Func_hex2bin_ArgDesc_0"] = "預設十六進制數字的文字表達。";
        this["Func_hex2bin_ArgDesc_1"] = "用於表示值的字元數。如果未指定，將使用代表該值的最少字元數。這允許您指定前導零。";
        this["Func_hex2bin_Category"] = "Engineering";
        this["Func_hex2bin_CategoryURL"] = "Engineering";
        this["Func_hex2bin_Desc"] = "在二進制數系統中，將代表十六進制數字的文字表達式轉換為代表相同數字的文字表達式。";
        this["Func_hex2dec_Arg_0"] = "Number";
        this["Func_hex2dec_ArgDesc_0"] = "預設十六進制數字的文字表達。";
        this["Func_hex2dec_Category"] = "Engineering";
        this["Func_hex2dec_CategoryURL"] = "Engineering";
        this["Func_hex2dec_Desc"] = "返回十六進制數字系統中數字文字表示形式的十進制值。";
        this["Func_hex2oct_Arg_0"] = "Number";
        this["Func_hex2oct_Arg_1"] = "Places";
        this["Func_hex2oct_ArgDesc_0"] = "預設十六進制數字的文字表達。";
        this["Func_hex2oct_ArgDesc_1"] = "用於表示值的字元數。如果未指定，將使用代表該值的最少字元數。這允許您指定前導零。";
        this["Func_hex2oct_Category"] = "Engineering";
        this["Func_hex2oct_CategoryURL"] = "Engineering";
        this["Func_hex2oct_Desc"] = "將代表十六進制數字的文字表達轉換為代表八進制數字系統中相同數字的文字表達。";
        this["Func_hour_Arg_0"] = "Time_value";
        this["Func_hour_ArgDesc_0"] = "用於求取小時的時間值。";
        this["Func_hour_Category"] = "DateAndTime";
        this["Func_hour_CategoryURL"] = "Date_and_Time";
        this["Func_hour_Desc"] = "返回時間值的小時。";
        this["Func_if_Arg_0"] = "logical_test";
        this["Func_if_Arg_1"] = "value_if_true";
        this["Func_if_Arg_2"] = "value_if_false";
        this["Func_if_ArgDesc_0"] = "要測試的條件。";
        this["Func_if_ArgDesc_1"] = "如果條件評估為 true，則結果。";
        this["Func_if_ArgDesc_2"] = "如果條件評估為 false，則結果。如果省略，則假定為邏輯值 false。";
        this["Func_if_Category"] = "Logical";
        this["Func_if_CategoryURL"] = "Logical";
        this["Func_if_Desc"] = "如果指定條件的計算結果為 true，則返回一個值; 如果指定的條件計算結果為 false，則返回另一個值。";
        this["Func_iferror_Arg_0"] = "value";
        this["Func_iferror_Arg_1"] = "value_if_error";
        this["Func_iferror_ArgDesc_0"] = "不是錯誤時使用的值。";
        this["Func_iferror_ArgDesc_1"] = "如果值引數為錯誤，則使用的值。";
        this["Func_iferror_Category"] = "Logical";
        this["Func_iferror_CategoryURL"] = "Logical";
        this["Func_iferror_Desc"] = "返回兩個值之一，具體取決於第一個值是否為錯誤。";
        this["Func_imabs_Arg_0"] = "ComplexNumber";
        this["Func_imabs_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imabs_Category"] = "Engineering";
        this["Func_imabs_CategoryURL"] = "Engineering";
        this["Func_imabs_Desc"] = "返回複數的絕對值。";
        this["Func_imaginary_Arg_0"] = "ComplexNumber";
        this["Func_imaginary_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imaginary_Category"] = "Engineering";
        this["Func_imaginary_CategoryURL"] = "Engineering";
        this["Func_imaginary_Desc"] = "返回複數的虛數係數。";
        this["Func_imargument_Arg_0"] = "ComplexNumber";
        this["Func_imargument_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imargument_Category"] = "Engineering";
        this["Func_imargument_CategoryURL"] = "Engineering";
        this["Func_imargument_Desc"] = "返回引數 theta，以弧度表示的角度。";
        this["Func_imconjugate_Arg_0"] = "ComplexNumber";
        this["Func_imconjugate_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imconjugate_Category"] = "Engineering";
        this["Func_imconjugate_CategoryURL"] = "Engineering";
        this["Func_imconjugate_Desc"] = "返回複數的共軛。";
        this["Func_imcos_Arg_0"] = "ComplexNumber";
        this["Func_imcos_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imcos_Category"] = "Engineering";
        this["Func_imcos_CategoryURL"] = "Engineering";
        this["Func_imcos_Desc"] = "返回複數的餘弦值。";
        this["Func_imdiv_Arg_0"] = "除數";
        this["Func_imdiv_Arg_1"] = "除數";
        this["Func_imdiv_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imdiv_ArgDesc_1"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imdiv_Category"] = "Engineering";
        this["Func_imdiv_CategoryURL"] = "Engineering";
        this["Func_imdiv_Desc"] = "返回兩個複數(除數/除數)的商。";
        this["Func_imexp_Arg_0"] = "ComplexNumber";
        this["Func_imexp_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imexp_Category"] = "Engineering";
        this["Func_imexp_CategoryURL"] = "Engineering";
        this["Func_imexp_Desc"] = "返回複數的指數。";
        this["Func_imln_Arg_0"] = "ComplexNumber";
        this["Func_imln_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imln_Category"] = "Engineering";
        this["Func_imln_CategoryURL"] = "Engineering";
        this["Func_imln_Desc"] = "返回複數的自然對數。";
        this["Func_imlog10_Arg_0"] = "ComplexNumber";
        this["Func_imlog10_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imlog10_Category"] = "Engineering";
        this["Func_imlog10_CategoryURL"] = "Engineering";
        this["Func_imlog10_Desc"] = "返回一個複數的以 10 為底的對數。";
        this["Func_imlog2_Arg_0"] = "ComplexNumber";
        this["Func_imlog2_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imlog2_Category"] = "Engineering";
        this["Func_imlog2_CategoryURL"] = "Engineering";
        this["Func_imlog2_Desc"] = "返回一個複數的以 2 為底的對數。";
        this["Func_impower_Arg_0"] = "ComplexNumber";
        this["Func_impower_Arg_1"] = "Power";
        this["Func_impower_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_impower_ArgDesc_1"] = "將把 ComplexNumber 提升到此冪。";
        this["Func_impower_Category"] = "Engineering";
        this["Func_impower_CategoryURL"] = "Engineering";
        this["Func_impower_Desc"] = "返回提高到指定冪的複數。";
        this["Func_improduct_Arg_0"] = "ComplexNumber{0}";
        this["Func_improduct_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_improduct_Category"] = "Engineering";
        this["Func_improduct_CategoryURL"] = "Engineering";
        this["Func_improduct_Desc"] = "返回一個或多個複數的乘積。";
        this["Func_imreal_Arg_0"] = "ComplexNumber";
        this["Func_imreal_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imreal_Category"] = "Engineering";
        this["Func_imreal_CategoryURL"] = "Engineering";
        this["Func_imreal_Desc"] = "返回複數的實係數。";
        this["Func_imsin_Arg_0"] = "ComplexNumber";
        this["Func_imsin_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imsin_Category"] = "Engineering";
        this["Func_imsin_CategoryURL"] = "Engineering";
        this["Func_imsin_Desc"] = "返回複數的正弦值。";
        this["Func_imsqrt_Arg_0"] = "ComplexNumber";
        this["Func_imsqrt_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imsqrt_Category"] = "Engineering";
        this["Func_imsqrt_CategoryURL"] = "Engineering";
        this["Func_imsqrt_Desc"] = "返回複數的平方根。";
        this["Func_imsub_Arg_0"] = "ComplexNumber1";
        this["Func_imsub_Arg_1"] = "ComplexNumber2";
        this["Func_imsub_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"，將從中減去 ComplexNumber2。";
        this["Func_imsub_ArgDesc_1"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"，將從 ComplexNumber1 中減去。";
        this["Func_imsub_Category"] = "Engineering";
        this["Func_imsub_CategoryURL"] = "Engineering";
        this["Func_imsub_Desc"] = "返回兩個複數之差。";
        this["Func_imsum_Arg_0"] = "ComplexNumber{0}";
        this["Func_imsum_ArgDesc_0"] = "複數的文字表示形式，格式為 \"x + yi\" 或 \"x + yj\"。";
        this["Func_imsum_Category"] = "Engineering";
        this["Func_imsum_CategoryURL"] = "Engineering";
        this["Func_imsum_Desc"] = "返回一個或多個複數的和。";
        this["Func_info_Arg_0"] = "類型";
        this["Func_info_ArgDesc_0"] = "表示要返回的資訊類型的文字表達式。\r\n\"directory\" = 當前文件夾的完整路徑。\r\n\"osversion\" = 當前作業系統版本以文字表達式來呈現。\r\n\"system\" = 當前作業系統。 Macintosh = \"mac\", Windows = \"pcdos\".\r\n    ";
        this["Func_info_Category"] = "Information";
        this["Func_info_CategoryURL"] = "Information";
        this["Func_info_Desc"] = "返回有關當前操作環境的資訊。";
        this["Func_int_Arg_0"] = "number";
        this["Func_int_ArgDesc_0"] = "實數四捨五入為整數。";
        this["Func_int_Category"] = "Math";
        this["Func_int_CategoryURL"] = "Math";
        this["Func_int_Desc"] = "將數字四捨五入至最接近的整數。";
        this["Func_intrate_Arg_0"] = "settlement";
        this["Func_intrate_Arg_1"] = "maturity";
        this["Func_intrate_Arg_2"] = "investment";
        this["Func_intrate_Arg_3"] = "redemption";
        this["Func_intrate_Arg_4"] = "basis";
        this["Func_intrate_ArgDesc_0"] = "證券的結算日期。";
        this["Func_intrate_ArgDesc_1"] = "證券的到期日。";
        this["Func_intrate_ArgDesc_2"] = "投資於證券的金額。";
        this["Func_intrate_ArgDesc_3"] = "到期時收取的金額。";
        this["Func_intrate_ArgDesc_4"] = "要使用的日計數類型。\r\n0 或省略 = US (NASD) 30/360,\r\n1 =實際/實際,\r\n2 =實際/360,\r\n3 =實際/365,\r\n4 = 歐洲 30/360\r\n    ";
        this["Func_intrate_Category"] = "Financial";
        this["Func_intrate_CategoryURL"] = "Financial";
        this["Func_intrate_Desc"] = "返回完全投資證券的利率。";
        this["Func_ipmt_Arg_0"] = "rate";
        this["Func_ipmt_Arg_1"] = "per";
        this["Func_ipmt_Arg_2"] = "nper";
        this["Func_ipmt_Arg_3"] = "pv";
        this["Func_ipmt_Arg_4"] = "fv";
        this["Func_ipmt_Arg_5"] = "type";
        this["Func_ipmt_ArgDesc_0"] = "每期利率。";
        this["Func_ipmt_ArgDesc_1"] = "週期，必須在 1 到 nper 之間。";
        this["Func_ipmt_ArgDesc_2"] = "年金的支付期總數。";
        this["Func_ipmt_ArgDesc_3"] = "現在的價值，或一系列未來還款額的總價。";
        this["Func_ipmt_ArgDesc_4"] = "末次付款後您希望獲得的終值或現金餘額。如果省略，則假定為零。";
        this["Func_ipmt_ArgDesc_5"] = "數字零或一，表示應在期末 (零，預設值) 或是期初 (一) 支付。";
        this["Func_ipmt_Category"] = "Financial";
        this["Func_ipmt_CategoryURL"] = "Financial";
        this["Func_ipmt_Desc"] = "根據定期，固定還款額和固定利率，返回投資在指定時期內的利息支付。";
        this["Func_irr_Arg_0"] = "values";
        this["Func_irr_Arg_1"] = "guess";
        this["Func_irr_ArgDesc_0"] = "定期發生的還款額 (負值) 和收入 (正值) 值的集合。";
        this["Func_irr_ArgDesc_1"] = "您猜到的數字接近結果。預設值為 0.1 (10%)。";
        this["Func_irr_Category"] = "Financial";
        this["Func_irr_CategoryURL"] = "Financial";
        this["Func_irr_Desc"] = "返回一系列現金流量的內部收益率。";
        this["Func_isblank_Arg_0"] = "value";
        this["Func_isblank_ArgDesc_0"] = "要測試的值。";
        this["Func_isblank_Category"] = "Information";
        this["Func_isblank_CategoryURL"] = "Information";
        this["Func_isblank_Desc"] = "如果值為空白，則傳回 true。";
        this["Func_isdbnull_Arg_0"] = "value";
        this["Func_isdbnull_ArgDesc_0"] = "要測試的值。";
        this["Func_isdbnull_Category"] = "Information";
        this["Func_isdbnull_CategoryURL"] = "Information";
        this["Func_isdbnull_Desc"] = "如果該值代表不存在的值，則傳回 true。";
        this["Func_iserr_Arg_0"] = "value";
        this["Func_iserr_ArgDesc_0"] = "要測試的值。";
        this["Func_iserr_Category"] = "Information";
        this["Func_iserr_CategoryURL"] = "Information";
        this["Func_iserr_Desc"] = "如果該值是除 #N/A 以外的任何錯誤值，則傳回 true。";
        this["Func_iserror_Arg_0"] = "value";
        this["Func_iserror_ArgDesc_0"] = "要測試的值。";
        this["Func_iserror_Category"] = "Information";
        this["Func_iserror_CategoryURL"] = "Information";
        this["Func_iserror_Desc"] = "如果該值為任何錯誤值，則傳回 true。";
        this["Func_iseven_Arg_0"] = "value";
        this["Func_iseven_ArgDesc_0"] = "要測試的值。";
        this["Func_iseven_Category"] = "Information";
        this["Func_iseven_CategoryURL"] = "Information";
        this["Func_iseven_Desc"] = "如果值是偶數，則返回 true，否則返回 false。";
        this["Func_islogical_Arg_0"] = "value";
        this["Func_islogical_ArgDesc_0"] = "要測試的值。";
        this["Func_islogical_Category"] = "Information";
        this["Func_islogical_CategoryURL"] = "Information";
        this["Func_islogical_Desc"] = "如果該值為邏輯值 (true 或 false)，則返回 true。";
        this["Func_isna_Arg_0"] = "value";
        this["Func_isna_ArgDesc_0"] = "要測試的值。";
        this["Func_isna_Category"] = "Information";
        this["Func_isna_CategoryURL"] = "Information";
        this["Func_isna_Desc"] = "如果值是 #N/A (值不可用) 錯誤值，則傳回 true。";
        this["Func_isnontext_Arg_0"] = "value";
        this["Func_isnontext_ArgDesc_0"] = "要測試的值。";
        this["Func_isnontext_Category"] = "Information";
        this["Func_isnontext_CategoryURL"] = "Information";
        this["Func_isnontext_Desc"] = "如果值是非文字的任何項目，則傳回 true。";
        this["Func_isnull_Arg_0"] = "value";
        this["Func_isnull_ArgDesc_0"] = "要測試的值。";
        this["Func_isnull_Category"] = "Information";
        this["Func_isnull_CategoryURL"] = "Information";
        this["Func_isnull_Desc"] = "如果該值不存在，則傳回 true。";
        this["Func_isnumber_Arg_0"] = "value";
        this["Func_isnumber_ArgDesc_0"] = "要測試的值。";
        this["Func_isnumber_Category"] = "Information";
        this["Func_isnumber_CategoryURL"] = "Information";
        this["Func_isnumber_Desc"] = "如果值為數字，則傳回 true。";
        this["Func_isodd_Arg_0"] = "value";
        this["Func_isodd_ArgDesc_0"] = "要測試的值。";
        this["Func_isodd_Category"] = "Information";
        this["Func_isodd_CategoryURL"] = "Information";
        this["Func_isodd_Desc"] = "如果該值為奇數，則返回 true，否則返回 false。";
        this["Func_isref_Arg_0"] = "value";
        this["Func_isref_ArgDesc_0"] = "要測試的值。";
        this["Func_isref_Category"] = "資訊";
        this["Func_isref_CategoryURL"] = "Information";
        this["Func_isref_Desc"] = "如果值為參考值，則傳回 true。";
        this["Func_istext_Arg_0"] = "value";
        this["Func_istext_ArgDesc_0"] = "要測試的值。";
        this["Func_istext_Category"] = "Information";
        this["Func_istext_CategoryURL"] = "Information";
        this["Func_istext_Desc"] = "如果該值為文字值，則傳回 true。";
        this["Func_lcm_Arg_0"] = "number{0}";
        this["Func_lcm_ArgDesc_0"] = "整數值。非整數值將被截斷。";
        this["Func_lcm_Category"] = "Math";
        this["Func_lcm_CategoryURL"] = "Math";
        this["Func_lcm_Desc"] = "返回一個或多個整數值的最小公倍數。";
        this["Func_left_Arg_0"] = "text";
        this["Func_left_Arg_1"] = "num_chars";
        this["Func_left_ArgDesc_0"] = "包含要提取字元的文字表達。";
        this["Func_left_ArgDesc_1"] = "要提取的字元數。";
        this["Func_left_Category"] = "TextAndData";
        this["Func_left_CategoryURL"] = "Text_and_Data";
        this["Func_left_Desc"] = "返回文字表達式中的前一個或多個字元。";
        this["Func_len_Arg_0"] = "text";
        this["Func_len_ArgDesc_0"] = "要尋找其長度的文字表達式。空格算作字元。";
        this["Func_len_Category"] = "TextAndData";
        this["Func_len_CategoryURL"] = "Text_and_Data";
        this["Func_len_Desc"] = "返回文字表達式中的字元數。";
        this["Func_ln_Arg_0"] = "number";
        this["Func_ln_ArgDesc_0"] = "求取自然對數的正實數。";
        this["Func_ln_Category"] = "Math";
        this["Func_ln_CategoryURL"] = "Math";
        this["Func_ln_Desc"] = "返回數字的自然對數。";
        this["Func_log_Arg_0"] = "number";
        this["Func_log_Arg_1"] = "基座";
        this["Func_log_ArgDesc_0"] = "求取對數的正實數。";
        this["Func_log_ArgDesc_1"] = "對數的底數。";
        this["Func_log_Category"] = "Math";
        this["Func_log_CategoryURL"] = "Math";
        this["Func_log_Desc"] = "返回數字的對數至指定的底數。";
        this["Func_log10_Arg_0"] = "number";
        this["Func_log10_ArgDesc_0"] = "求取以 10 為底的對數的正實數。";
        this["Func_log10_Category"] = "Math";
        this["Func_log10_CategoryURL"] = "Math";
        this["Func_log10_Desc"] = "返回一個數的以 10 為底的對數。";
        this["Func_lower_Arg_0"] = "text";
        this["Func_lower_ArgDesc_0"] = "轉換為小寫的文字。";
        this["Func_lower_Category"] = "TextAndData";
        this["Func_lower_CategoryURL"] = "Text_and_Data";
        this["Func_lower_Desc"] = "將文字表達式中的所有大寫字母轉換為小寫。";
        this["Func_max_Arg_0"] = "number{0}";
        this["Func_max_ArgDesc_0"] = "一個或多個數字範圍。";
        this["Func_max_Category"] = "Statistical";
        this["Func_max_CategoryURL"] = "Statistical";
        this["Func_max_Desc"] = "返回一組數字中的最大數字。";
        this["Func_median_Arg_0"] = "number{0}";
        this["Func_median_ArgDesc_0"] = "一個或多個數字範圍。";
        this["Func_median_Category"] = "Statistical";
        this["Func_median_CategoryURL"] = "Statistical";
        this["Func_median_Desc"] = "返回一組數字的中位數。中位數是對數字進行排序時的中間數字，如果指定數字的偶數則是兩個中間數字的平均值。";
        this["Func_mid_Arg_0"] = "text";
        this["Func_mid_Arg_1"] = "start_num";
        this["Func_mid_Arg_2"] = "num_chars";
        this["Func_mid_ArgDesc_0"] = "包含要提取字元的文字表達。";
        this["Func_mid_ArgDesc_1"] = "要提取的第一個字元的以1為基礎的索引。";
        this["Func_mid_ArgDesc_2"] = "要提取的字元數。";
        this["Func_mid_Category"] = "TextAndData";
        this["Func_mid_CategoryURL"] = "Text_and_Data";
        this["Func_mid_Desc"] = "從文字表達式返回指定數量的字元。";
        this["Func_min_Arg_0"] = "number{0}";
        this["Func_min_ArgDesc_0"] = "一個或多個數字範圍。";
        this["Func_min_Category"] = "Statistical";
        this["Func_min_CategoryURL"] = "Statistical";
        this["Func_min_Desc"] = "返回一組數字中的最小數字。";
        this["Func_minute_Arg_0"] = "Time_value";
        this["Func_minute_ArgDesc_0"] = "用於求取分鐘的時間值。";
        this["Func_minute_Category"] = "DateAndTime";
        this["Func_minute_CategoryURL"] = "Date_and_Time";
        this["Func_minute_Desc"] = "返回時間值的分鐘。";
        this["Func_mod_Arg_0"] = "number";
        this["Func_mod_Arg_1"] = "除數";
        this["Func_mod_ArgDesc_0"] = "尋找剩餘數字。";
        this["Func_mod_ArgDesc_1"] = "用來除數的數值。";
        this["Func_mod_Category"] = "Math";
        this["Func_mod_CategoryURL"] = "Math";
        this["Func_mod_Desc"] = "返回除數後的餘數。結果與除數的符號相同。";
        this["Func_month_Arg_0"] = "Date_value";
        this["Func_month_ArgDesc_0"] = "從中求取月份的日期值。";
        this["Func_month_Category"] = "DateAndTime";
        this["Func_month_CategoryURL"] = "Date_and_Time";
        this["Func_month_Desc"] = "返回日期值的月份。";
        this["Func_mround_Arg_0"] = "Number";
        this["Func_mround_Arg_1"] = "多個";
        this["Func_mround_ArgDesc_0"] = "要四捨五入的數字。";
        this["Func_mround_ArgDesc_1"] = "一個數字，表示要四捨五入的倍數。";
        this["Func_mround_Category"] = "Math";
        this["Func_mround_CategoryURL"] = "Math";
        this["Func_mround_Desc"] = "將數字四捨五入至另一個數字的最接近倍數。";
        this["Func_multinomial_Arg_0"] = "number{0}";
        this["Func_multinomial_ArgDesc_0"] = "求取多項式的數字之一。";
        this["Func_multinomial_Category"] = "Math";
        this["Func_multinomial_CategoryURL"] = "Math";
        this["Func_multinomial_Desc"] = "返回一組數字的多項式。多項式是數值總和的階乘與其階乘乘積之比。";
        this["Func_n_Arg_0"] = "數值";
        this["Func_n_ArgDesc_0"] = "要轉換為數字的值。以下是可接受的值:\r\n數字 - 返回數字。\r\n日期 - 返回日期。\r\nTrue - 返回 1。\r\nFalse - 返回 0。\r\n錯誤值-返回錯誤值。\r\n任何其他值 - 返回 0。\r\n    ";
        this["Func_n_Category"] = "Information";
        this["Func_n_CategoryURL"] = "Information";
        this["Func_n_Desc"] = "將值轉換為數字。";
        this["Func_na_Category"] = "Information";
        this["Func_na_CategoryURL"] = "Information";
        this["Func_na_Desc"] = "返回錯誤值 #N/A (不可用)。";
        this["Func_networkdays_Arg_0"] = "startDate";
        this["Func_networkdays_Arg_1"] = "endDate";
        this["Func_networkdays_Arg_2"] = "holidays";
        this["Func_networkdays_ArgDesc_0"] = "開始的日期。";
        this["Func_networkdays_ArgDesc_1"] = "結束日期。";
        this["Func_networkdays_ArgDesc_2"] = "一個或多個參考範圍，包含被視為非工作日的日期。";
        this["Func_networkdays_Category"] = "DateAndTime";
        this["Func_networkdays_CategoryURL"] = "DateAndTime";
        this["Func_networkdays_Desc"] = "返回指定日期之間的整個工作日總數，不包括任何指定假日。";
        this["Func_not_Arg_0"] = "邏輯";
        this["Func_not_ArgDesc_0"] = "邏輯值 (真或假)。";
        this["Func_not_Category"] = "Logical";
        this["Func_not_CategoryURL"] = "Logical";
        this["Func_not_Desc"] = "反轉邏輯值 (將 true 更改為 false，將 false 更改為 true)。";
        this["Func_now_Category"] = "DateAndTime";
        this["Func_now_CategoryURL"] = "Date_and_Time";
        this["Func_now_Desc"] = "返回代表當前日期和時間的值。";
        this["Func_nper_Arg_0"] = "rate";
        this["Func_nper_Arg_1"] = "pmt";
        this["Func_nper_Arg_2"] = "pv";
        this["Func_nper_Arg_3"] = "fv";
        this["Func_nper_Arg_4"] = "type";
        this["Func_nper_ArgDesc_0"] = "每期利率。";
        this["Func_nper_ArgDesc_1"] = "支付的每個期間，在年金的使用期限內無法更改。";
        this["Func_nper_ArgDesc_2"] = "現在價值，或一系列未來還款額的當前總價值。";
        this["Func_nper_ArgDesc_3"] = "末次付款後您希望獲得的終值或現金餘額。";
        this["Func_nper_ArgDesc_4"] = "數字零或一，表示應在期末 (零，預設值) 或是期初 (一) 支付。";
        this["Func_nper_Category"] = "Financial";
        this["Func_nper_CategoryURL"] = "Financial";
        this["Func_nper_Desc"] = "根據定期，固定還款額和固定利率，傳回投資的周期數。";
        this["Func_npv_Arg_0"] = "rate";
        this["Func_npv_Arg_1"] = "value{0}";
        this["Func_npv_ArgDesc_0"] = "一段期間內的折扣率。";
        this["Func_npv_ArgDesc_1"] = "還款額 (負值) 或收入 (正值)。";
        this["Func_npv_Category"] = "Financial";
        this["Func_npv_CategoryURL"] = "Financial";
        this["Func_npv_Desc"] = "使用折現率和一系列未來還款額 (負值) 和收入 (正值) 計算投資的淨現值。";
        this["Func_null_Category"] = "Information";
        this["Func_null_CategoryURL"] = "Information";
        this["Func_null_Desc"] = "返回不存在的值。";
        this["Func_oct2bin_Arg_0"] = "Number";
        this["Func_oct2bin_Arg_1"] = "Places";
        this["Func_oct2bin_ArgDesc_0"] = "預設八進制數字的文字表達。";
        this["Func_oct2bin_ArgDesc_1"] = "用於表示值的字元數。如果未指定，將使用代表該值的最少字元數。這允許您指定前導零。";
        this["Func_oct2bin_Category"] = "Engineering";
        this["Func_oct2bin_CategoryURL"] = "Engineering";
        this["Func_oct2bin_Desc"] = "在二進制數系統中，將代表八進制數字的文字表達式轉換為代表相同數字的文字表達式。";
        this["Func_oct2dec_Arg_0"] = "Number";
        this["Func_oct2dec_ArgDesc_0"] = "預設八進制數字的文字表達。";
        this["Func_oct2dec_Category"] = "Engineering";
        this["Func_oct2dec_CategoryURL"] = "Engineering";
        this["Func_oct2dec_Desc"] = "返回八進制數系統中數字文字表示形式的十進制值。";
        this["Func_oct2hex_Arg_0"] = "Number";
        this["Func_oct2hex_Arg_1"] = "Places";
        this["Func_oct2hex_ArgDesc_0"] = "預設八進制數字的文字表達。";
        this["Func_oct2hex_ArgDesc_1"] = "用於表示值的字元數。如果未指定，將使用代表該值的最少字元數。這允許您指定前導零。";
        this["Func_oct2hex_Category"] = "Engineering";
        this["Func_oct2hex_CategoryURL"] = "Engineering";
        this["Func_oct2hex_Desc"] = "在十六進制數字系統中，將代表八進制數字的文字表達式轉換為代表相同數字的文字表達式。";
        this["Func_odd_Arg_0"] = "number";
        this["Func_odd_ArgDesc_0"] = "要捨入的值。";
        this["Func_odd_Category"] = "Math";
        this["Func_odd_CategoryURL"] = "Math";
        this["Func_odd_Desc"] = "將正數向上取整，負數向下取整至最接近的奇數整數。";
        this["Func_or_Arg_0"] = "logical{0}";
        this["Func_or_ArgDesc_0"] = "邏輯值 (真或假)。";
        this["Func_or_Category"] = "Logical";
        this["Func_or_CategoryURL"] = "Logical";
        this["Func_or_Desc"] = "如果一個或多個值為 true，則返回 true; 如果所有值均為 false，則返回 false。";
        this["Func_pi_Category"] = "Math";
        this["Func_pi_CategoryURL"] = "Math";
        this["Func_pi_Desc"] = "返回常數 pi，即 3.14159265358979。";
        this["Func_pmt_Arg_0"] = "rate";
        this["Func_pmt_Arg_1"] = "nper";
        this["Func_pmt_Arg_2"] = "pv";
        this["Func_pmt_Arg_3"] = "fv";
        this["Func_pmt_Arg_4"] = "type";
        this["Func_pmt_ArgDesc_0"] = "貸款利率。";
        this["Func_pmt_ArgDesc_1"] = "貸款的總付款次數。";
        this["Func_pmt_ArgDesc_2"] = "現值，或一系列未來還款額的現值總額; 也稱為本金。";
        this["Func_pmt_ArgDesc_3"] = "末次付款後您希望獲得的終值或現金餘額。";
        this["Func_pmt_ArgDesc_4"] = "數字零或一，表示應在期末 (零，預設值) 或是期初 (一) 支付。";
        this["Func_pmt_Category"] = "Financial";
        this["Func_pmt_CategoryURL"] = "Financial";
        this["Func_pmt_Desc"] = "根據固定還款額和固定利率計算貸款的還款額。";
        this["Func_power_Arg_0"] = "number";
        this["Func_power_Arg_1"] = "power";
        this["Func_power_ArgDesc_0"] = "要提升為冪的基數。";
        this["Func_power_ArgDesc_1"] = "對基數求乘方的指數。";
        this["Func_power_Category"] = "Math";
        this["Func_power_CategoryURL"] = "Math";
        this["Func_power_Desc"] = "返回數值加冪的結果。";
        this["Func_ppmt_Arg_0"] = "rate";
        this["Func_ppmt_Arg_1"] = "per";
        this["Func_ppmt_Arg_2"] = "nper";
        this["Func_ppmt_Arg_3"] = "pv";
        this["Func_ppmt_Arg_4"] = "fv";
        this["Func_ppmt_Arg_5"] = "type";
        this["Func_ppmt_ArgDesc_0"] = "每期利率。";
        this["Func_ppmt_ArgDesc_1"] = "週期，必須在 1 到 nper 之間。";
        this["Func_ppmt_ArgDesc_2"] = "年金的支付期總數。";
        this["Func_ppmt_ArgDesc_3"] = "現在的價值，或一系列未來還款額的總價值。";
        this["Func_ppmt_ArgDesc_4"] = "末次付款後您希望獲得的終值或現金餘額。";
        this["Func_ppmt_ArgDesc_5"] = "數字零或一，表示應在期末 (零，預設值) 或是期初 (一) 支付。";
        this["Func_ppmt_Category"] = "Financial";
        this["Func_ppmt_CategoryURL"] = "Financial";
        this["Func_ppmt_Desc"] = "根據定期，固定還款額和固定利率，返回指定時期內本金的投資金額。";
        this["Func_product_Arg_0"] = "number{0}";
        this["Func_product_ArgDesc_0"] = "一個要乘以其他數字的數字。";
        this["Func_product_Category"] = "Math";
        this["Func_product_CategoryURL"] = "Math";
        this["Func_product_Desc"] = "將所有作為引數給出的數字相乘並返回乘積。";
        this["Func_pv_Arg_0"] = "rate";
        this["Func_pv_Arg_1"] = "nper";
        this["Func_pv_Arg_2"] = "pmt";
        this["Func_pv_Arg_3"] = "fv";
        this["Func_pv_Arg_4"] = "type";
        this["Func_pv_ArgDesc_0"] = "每期利率。";
        this["Func_pv_ArgDesc_1"] = "年金的支付期總數。";
        this["Func_pv_ArgDesc_2"] = "支付的每個期間，在年金的使用期限內無法更改。";
        this["Func_pv_ArgDesc_3"] = "末次付款後您希望獲得的終值或現金餘額。";
        this["Func_pv_ArgDesc_4"] = "數字零或一，表示應在期末 (零，預設值) 或是期初 (一) 支付。";
        this["Func_pv_Category"] = "Financial";
        this["Func_pv_CategoryURL"] = "Financial";
        this["Func_pv_Desc"] = "返回投資的現值。";
        this["Func_quotient_Arg_0"] = "numerator";
        this["Func_quotient_Arg_1"] = "denominator";
        this["Func_quotient_ArgDesc_0"] = "除法運算的被除數。";
        this["Func_quotient_ArgDesc_1"] = "除法運算的除數。";
        this["Func_quotient_Category"] = "Math";
        this["Func_quotient_CategoryURL"] = "Math";
        this["Func_quotient_Desc"] = "返回除法的整數部分。這將放棄除法的其餘部分。";
        this["Func_radians_Arg_0"] = "Degrees";
        this["Func_radians_ArgDesc_0"] = "以度為單位的數值，將轉換為弧度。";
        this["Func_radians_Category"] = "Math";
        this["Func_radians_CategoryURL"] = "Math";
        this["Func_radians_Desc"] = "將度數轉換為弧度。";
        this["Func_rand_Category"] = "Math";
        this["Func_rand_CategoryURL"] = "Math";
        this["Func_rand_Desc"] = "返回大於或等於0且小於1的均勻分佈的隨機數";
        this["Func_randbetween_Arg_0"] = "底部";
        this["Func_randbetween_Arg_1"] = "頂部";
        this["Func_randbetween_ArgDesc_0"] = "將返回的最小值。";
        this["Func_randbetween_ArgDesc_1"] = "將返回的最大值。";
        this["Func_randbetween_Category"] = "Math";
        this["Func_randbetween_CategoryURL"] = "Math";
        this["Func_randbetween_Desc"] = "在兩個指定數字之間產生一個偽隨機整數。";
        this["Func_rate_Arg_0"] = "NPer";
        this["Func_rate_Arg_1"] = "Pmt";
        this["Func_rate_Arg_2"] = "Pv";
        this["Func_rate_Arg_3"] = "Fv";
        this["Func_rate_Arg_4"] = "類型";
        this["Func_rate_Arg_5"] = "Guess";
        this["Func_rate_ArgDesc_0"] = "年金的支付期總數。";
        this["Func_rate_ArgDesc_1"] = "付款在每個週期進行，並且在年金的使用期內無法更改。";
        this["Func_rate_ArgDesc_2"] = "現在的價值，或一系列未來還款額的總價值。";
        this["Func_rate_ArgDesc_3"] = "末次付款後您希望獲得的終值或現金餘額。如果省略 fv，則假定為 0 (例如，貸款的終值是 0)。例如，如果您想節省 50,000 美元以支付18年內的特殊項目，則 50,000 美元是終值。然後，您可以保守地估計利率，並確定每個月必須節省多少。如果省略 fv，則必須包含 pmt 引數。fv 的預設值為零。";
        this["Func_rate_ArgDesc_4"] = "數字零或一，表示應在期末 (零，預設值) 或是期初 (一) 支付。";
        this["Func_rate_ArgDesc_5"] = "您猜到的數字接近結果。如果省略，則假定為 0.1 (10%)。";
        this["Func_rate_Category"] = "Financial";
        this["Func_rate_CategoryURL"] = "Financial";
        this["Func_rate_Desc"] = "返回年金每個週期的利率。";
        this["Func_replace_Arg_0"] = "Old_text";
        this["Func_replace_Arg_1"] = "Start_num";
        this["Func_replace_Arg_2"] = "Num_chars";
        this["Func_replace_Arg_3"] = "New_text";
        this["Func_replace_ArgDesc_0"] = "替換某些字元的文字表達。";
        this["Func_replace_ArgDesc_1"] = "將 old_text 中以 1 為基礎的字元索引取代為 new_text。";
        this["Func_replace_ArgDesc_2"] = "用 new_text 替換 old_text 中的字元數。";
        this["Func_replace_ArgDesc_3"] = "將會取代 old_text 中字元的文字。";
        this["Func_replace_Category"] = "TextAndData";
        this["Func_replace_CategoryURL"] = "Text_and_Data";
        this["Func_replace_Desc"] = "替換部分文字表達。";
        this["Func_rept_Arg_0"] = "TextValue";
        this["Func_rept_Arg_1"] = "RepeatCount";
        this["Func_rept_ArgDesc_0"] = "要重複的文字。";
        this["Func_rept_ArgDesc_1"] = "重複文字的正次數。";
        this["Func_rept_Category"] = "TextAndData";
        this["Func_rept_CategoryURL"] = "Text_and_Data";
        this["Func_rept_Desc"] = "重複文字特定次數。";
        this["Func_right_Arg_0"] = "text";
        this["Func_right_Arg_1"] = "num_chars";
        this["Func_right_ArgDesc_0"] = "包含要提取字元的文字表達。";
        this["Func_right_ArgDesc_1"] = "要提取的字元數。";
        this["Func_right_Category"] = "TextAndData";
        this["Func_right_CategoryURL"] = "Text_and_Data";
        this["Func_right_Desc"] = "返回文字表達式中的最後一個或多個字元。";
        this["Func_roman_Arg_0"] = "Number";
        this["Func_roman_Arg_1"] = "Form";
        this["Func_roman_ArgDesc_0"] = "要轉換為羅馬數字的數字。";
        this["Func_roman_ArgDesc_1"] = "一個數字或布林值，指示是使用經典羅馬數字還是使用更簡潔的版本。\r\n0  = (預設) 經典。(499 = \"CDXCIX\")\r\n1  = 更簡潔。(499 = \"LDVLIV\")\r\n2  = 更簡潔。(499 = \"XDIX\")\r\n3  = 更簡潔。(499 = \"VDIV\")\r\n4  = 簡化。(499 = \"ID\")\r\nTrue = 經典。\r\nFalse =簡化。\r\n    ";
        this["Func_roman_Category"] = "Math";
        this["Func_roman_CategoryURL"] = "Math";
        this["Func_roman_Desc"] = "將數字轉換為羅馬數字作為文字表達。";
        this["Func_round_Arg_0"] = "number";
        this["Func_round_Arg_1"] = "num_digits";
        this["Func_round_Arg_2"] = "round_Style";
        this["Func_round_ArgDesc_0"] = "要捨入的數字。";
        this["Func_round_ArgDesc_1"] = "要四捨五入的位數。";
        this["Func_round_ArgDesc_2"] = "一個數字，指示要使用的捨入樣式。\r\n0 - 算術捨入 (預設) - 如果捨入的最後一位有效數字為 5，則將數字向上取整數。\r\n1 - 銀行捨入 - 當最後一位有效數字為5時，前一位數字用於確定是向上還是向下取整數。如果數字為偶數，則四捨五入。如果為奇數，則將數字四捨五入。\r\n    ";
        this["Func_round_Category"] = "Math";
        this["Func_round_CategoryURL"] = "Math";
        this["Func_round_Desc"] = "將數字四捨五入至指定位數。";
        this["Func_rounddown_Arg_0"] = "Number";
        this["Func_rounddown_Arg_1"] = "number";
        this["Func_rounddown_ArgDesc_0"] = "要四捨五入的數字。";
        this["Func_rounddown_ArgDesc_1"] = "要捨入的小數位數。正數表示小數點後的位置，負數表示小數點前的位置。";
        this["Func_rounddown_Category"] = "Math";
        this["Func_rounddown_CategoryURL"] = "Math";
        this["Func_rounddown_Desc"] = "將數字四捨五入至指定位數。";
        this["Func_roundup_Arg_0"] = "Number";
        this["Func_roundup_Arg_1"] = "number";
        this["Func_roundup_ArgDesc_0"] = "要四捨五入的數字。";
        this["Func_roundup_ArgDesc_1"] = "要捨入的小數位數。正數表示小數點後的位置，負數表示小數點前的位置。";
        this["Func_roundup_Category"] = "Math";
        this["Func_roundup_CategoryURL"] = "Math";
        this["Func_roundup_Desc"] = "將數字四捨五入至指定位數。";
        this["Func_search_Arg_0"] = "search_text";
        this["Func_search_Arg_1"] = "value";
        this["Func_search_Arg_2"] = "start_index";
        this["Func_search_ArgDesc_0"] = "要尋找值的文字。";
        this["Func_search_ArgDesc_1"] = "尋找 search_text 的文字。";
        this["Func_search_ArgDesc_2"] = "從以 1 為基礎的索引開始搜索 (如果省略，則 start_index 為 1)。";
        this["Func_search_Category"] = "TextAndData";
        this["Func_search_CategoryURL"] = "TextAndData";
        this["Func_search_Desc"] = "返回一個字串在另一個字串中的從以1開始的索引，搜索不區分大小寫。";
        this["Func_searchb_Arg_0"] = "search_text";
        this["Func_searchb_Arg_1"] = "value";
        this["Func_searchb_Arg_2"] = "start_index";
        this["Func_searchb_ArgDesc_0"] = "要尋找值的文字。";
        this["Func_searchb_ArgDesc_1"] = "尋找 search_text 的文字。";
        this["Func_searchb_ArgDesc_2"] = "從以 1 為基礎的索引開始搜索 (如果省略，則 start_index 為 1)。";
        this["Func_searchb_Category"] = "TextAndData";
        this["Func_searchb_CategoryURL"] = "TextAndData";
        this["Func_searchb_Desc"] = "返回一個字串在另一個字串中的從以1開始的索引，搜索不區分大小寫。";
        this["Func_second_Arg_0"] = "Time_value";
        this["Func_second_ArgDesc_0"] = "從中求取秒的時間值。";
        this["Func_second_Category"] = "DateAndTime";
        this["Func_second_CategoryURL"] = "Date_and_Time";
        this["Func_second_Desc"] = "返回時間值的秒數。";
        this["Func_seriessum_Arg_0"] = "InputValue";
        this["Func_seriessum_Arg_1"] = "InitialPower";
        this["Func_seriessum_Arg_2"] = "Step";
        this["Func_seriessum_Arg_3"] = "Coefficients";
        this["Func_seriessum_ArgDesc_0"] = "冪級數的輸入值。";
        this["Func_seriessum_ArgDesc_1"] = "InputValue 將被提高到的初始冪。";
        this["Func_seriessum_ArgDesc_2"] = "該步驟將用於增加系列中每一期的冪。";
        this["Func_seriessum_ArgDesc_3"] = "與 InputValue 的每個連續冪相乘的一組係數。";
        this["Func_seriessum_Category"] = "Math";
        this["Func_seriessum_CategoryURL"] = "Math";
        this["Func_seriessum_Desc"] = "返回冪級數之和。";
        this["Func_sign_Arg_0"] = "Number";
        this["Func_sign_ArgDesc_0"] = "將返回其符號的數字。";
        this["Func_sign_Category"] = "Math";
        this["Func_sign_CategoryURL"] = "Math";
        this["Func_sign_Desc"] = "返回數字的符號 (-1，0 或 1)。";
        this["Func_sin_Arg_0"] = "number";
        this["Func_sin_ArgDesc_0"] = "求取正弦的弧度角。";
        this["Func_sin_Category"] = "Math";
        this["Func_sin_CategoryURL"] = "Math";
        this["Func_sin_Desc"] = "返回指定角度的正弦。";
        this["Func_sinh_Arg_0"] = "number";
        this["Func_sinh_ArgDesc_0"] = "求取雙曲正弦值的實數。";
        this["Func_sinh_Category"] = "Math";
        this["Func_sinh_CategoryURL"] = "Math";
        this["Func_sinh_Desc"] = "返回數字的雙曲正弦值。";
        this["Func_sln_Arg_0"] = "cost";
        this["Func_sln_Arg_1"] = "salvage";
        this["Func_sln_Arg_2"] = "life";
        this["Func_sln_ArgDesc_0"] = "資產的初始成本。";
        this["Func_sln_ArgDesc_1"] = "折舊結束時的值。";
        this["Func_sln_ArgDesc_2"] = "資產折舊的期間數。";
        this["Func_sln_Category"] = "Financial";
        this["Func_sln_CategoryURL"] = "Financial";
        this["Func_sln_Desc"] = "返回資產在一個期間內的直線折舊。";
        this["Func_sqrt_Arg_0"] = "number";
        this["Func_sqrt_ArgDesc_0"] = "求取平方根的數字。";
        this["Func_sqrt_Category"] = "Math";
        this["Func_sqrt_CategoryURL"] = "Math";
        this["Func_sqrt_Desc"] = "返回數字的正平方根。";
        this["Func_sqrtpi_Arg_0"] = "Number";
        this["Func_sqrtpi_ArgDesc_0"] = "要乘以 pi 的數字。";
        this["Func_sqrtpi_Category"] = "Math";
        this["Func_sqrtpi_CategoryURL"] = "Math";
        this["Func_sqrtpi_Desc"] = "將指定數字乘以 pi，然後返回結果的平方根。";
        this["Func_stdev_Arg_0"] = "number{0}";
        this["Func_stdev_ArgDesc_0"] = "用於指定總體樣本的數量或數量範圍。";
        this["Func_stdev_Category"] = "Statistical";
        this["Func_stdev_CategoryURL"] = "Statistical";
        this["Func_stdev_Desc"] = "返回總體樣本的標準偏差。";
        this["Func_subtotal_Arg_0"] = "FunctionNum";
        this["Func_subtotal_Arg_1"] = "Ref{0}";
        this["Func_subtotal_ArgDesc_0"] = "一個數字，代表要計算的函數。將值加 100，可忽略隱藏值。\r\n1 (或 101) - AVERAGE\r\n2 (或 102) - COUNT\r\n4 (或 104) - MAX\r\n5 (或 105) - MIN\r\n6 (或 106) - PRODUCT\r\n7 (或 107) - STDEV\r\n9 (或 109) - SUM\r\n10 (或 110) - VAR\r\n    ";
        this["Func_subtotal_ArgDesc_1"] = "要尋找小計的一個或多個參考。";
        this["Func_subtotal_Category"] = "Math";
        this["Func_subtotal_CategoryURL"] = "Math";
        this["Func_subtotal_Desc"] = "計算其他參考之一中的小計。";
        this["Func_sum_Arg_0"] = "number{0}";
        this["Func_sum_ArgDesc_0"] = "要與其他數字相加的數字。";
        this["Func_sum_Category"] = "Math";
        this["Func_sum_CategoryURL"] = "Math";
        this["Func_sum_Desc"] = "對所有作為引數給出的數字求和，並返回和。";
        this["Func_syd_Arg_0"] = "cost";
        this["Func_syd_Arg_1"] = "salvage";
        this["Func_syd_Arg_2"] = "life";
        this["Func_syd_Arg_3"] = "per";
        this["Func_syd_ArgDesc_0"] = "資產的初始成本。";
        this["Func_syd_ArgDesc_1"] = "折舊結束時的值。";
        this["Func_syd_ArgDesc_2"] = "資產折舊的期間數。";
        this["Func_syd_ArgDesc_3"] = "週期，必須使用與壽命相同的單位。";
        this["Func_syd_Category"] = "Financial";
        this["Func_syd_CategoryURL"] = "Financial";
        this["Func_syd_Desc"] = "返回指定時期內資產折舊年的總和。";
        this["Func_tan_Arg_0"] = "number";
        this["Func_tan_ArgDesc_0"] = "求取切線的角度 (以弧度為單位)。";
        this["Func_tan_Category"] = "Math";
        this["Func_tan_CategoryURL"] = "Math";
        this["Func_tan_Desc"] = "返回指定角度的切線。";
        this["Func_tanh_Arg_0"] = "number";
        this["Func_tanh_ArgDesc_0"] = "求取雙曲正切的角度 (以弧度為單位)。";
        this["Func_tanh_Category"] = "Math";
        this["Func_tanh_CategoryURL"] = "Math";
        this["Func_tanh_Desc"] = "返回指定角度的雙曲正切值。";
        this["Func_text_Arg_0"] = "數值";
        this["Func_text_Arg_1"] = "FormatText";
        this["Func_text_ArgDesc_0"] = "數值格式。";
        this["Func_text_ArgDesc_1"] = "用於格式化數值的數字格式。";
        this["Func_text_Category"] = "TextAndData";
        this["Func_text_CategoryURL"] = "Text_and_Data";
        this["Func_text_Desc"] = "將值轉換為基於指定格式的文字。";
        this["Func_time_Arg_0"] = "小時";
        this["Func_time_Arg_1"] = "分鐘";
        this["Func_time_Arg_2"] = "秒";
        this["Func_time_ArgDesc_0"] = "代表時間的小時數。";
        this["Func_time_ArgDesc_1"] = "一個數字，代表時間的分鐘數。";
        this["Func_time_ArgDesc_2"] = "一個數字，表示時間的秒數。";
        this["Func_time_Category"] = "DateAndTime";
        this["Func_time_CategoryURL"] = "Date_and_Time";
        this["Func_time_Desc"] = "返回代表特定時間值的數字。";
        this["Func_timevalue_Arg_0"] = "time_text";
        this["Func_timevalue_ArgDesc_0"] = "表示時間的文字表達。";
        this["Func_timevalue_Category"] = "DateAndTime";
        this["Func_timevalue_CategoryURL"] = "Date_and_Time";
        this["Func_timevalue_Desc"] = "將代表時間的文字表達轉換為等效時間值。";
        this["Func_today_Category"] = "DateAndTime";
        this["Func_today_CategoryURL"] = "Date_and_Time";
        this["Func_today_Desc"] = "返回當前的日期。";
        this["Func_trim_Arg_0"] = "text";
        this["Func_trim_ArgDesc_0"] = "要從中移除多餘空格的文字。";
        this["Func_trim_Category"] = "TextAndData";
        this["Func_trim_CategoryURL"] = "Text_and_Data";
        this["Func_trim_Desc"] = "從文字中移除所有空格，但字詞之間的單一空格除外。";
        this["Func_true_Category"] = "Logical";
        this["Func_true_CategoryURL"] = "Logical";
        this["Func_true_Desc"] = "返回邏輯值 true。";
        this["Func_trunc_Arg_0"] = "number";
        this["Func_trunc_Arg_1"] = "num_digits";
        this["Func_trunc_ArgDesc_0"] = "要截斷的數字。";
        this["Func_trunc_ArgDesc_1"] = "一個數字，指定截斷的精度。預設值為零。";
        this["Func_trunc_Category"] = "Math";
        this["Func_trunc_CategoryURL"] = "Math";
        this["Func_trunc_Desc"] = "透過移除數字的小數部分，將數字截斷為整數。";
        this["Func_type_Arg_0"] = "value";
        this["Func_type_ArgDesc_0"] = "求取類型的值。";
        this["Func_type_Category"] = "Information";
        this["Func_type_CategoryURL"] = "Information";
        this["Func_type_Desc"] = "返回代表數值類型的數值: 數字 = 1; 文字 = 2; 邏輯值(正確或錯誤) = 4; 錯誤值 = 16; 陣列 = 64。";
        this["Func_upper_Arg_0"] = "text";
        this["Func_upper_ArgDesc_0"] = "轉換為大寫的文字。";
        this["Func_upper_Category"] = "TextAndData";
        this["Func_upper_CategoryURL"] = "Text_and_Data";
        this["Func_upper_Desc"] = "將文字表達式中的所有小寫字母轉換為大寫。";
        this["Func_value_Arg_0"] = "text";
        this["Func_value_ArgDesc_0"] = "要轉換的文字表達。";
        this["Func_value_Category"] = "TextAndData";
        this["Func_value_CategoryURL"] = "Text_and_Data";
        this["Func_value_Desc"] = "將代表數字的文字表達轉換為數字。";
        this["Func_var_Arg_0"] = "number{0}";
        this["Func_var_ArgDesc_0"] = "用於指定總體樣本的數量或數量範圍。";
        this["Func_var_Category"] = "Statistical";
        this["Func_var_CategoryURL"] = "Statistical";
        this["Func_var_Desc"] = "返回總體樣本的方差。";
        this["Func_weekday_Arg_0"] = "日期";
        this["Func_weekday_Arg_1"] = "ReturnType";
        this["Func_weekday_ArgDesc_0"] = "尋找星期數的日期。";
        this["Func_weekday_ArgDesc_1"] = "確定一週的編號方案。\r\n1 (預設) = 週日 (1) 至週六 (7)\r\n2 = 週一 (1) 至週日 (7)\r\n3 = 週一 (0) 至週日 (6)\r\n    ";
        this["Func_weekday_Category"] = "DateAndTime";
        this["Func_weekday_CategoryURL"] = "DateAndTime";
        this["Func_weekday_Desc"] = "返回代表指定日期的星期幾的數字。";
        this["Func_weeknum_Arg_0"] = "日期";
        this["Func_weeknum_Arg_1"] = "returnType";
        this["Func_weeknum_ArgDesc_0"] = "尋找星期數的日期。";
        this["Func_weeknum_ArgDesc_1"] = "確定一週從哪一天開始。\r\n1 (預設) = 週日是一週的第一天。\r\n2 = 星期一是一週的第一天。\r\n    ";
        this["Func_weeknum_Category"] = "DateAndTime";
        this["Func_weeknum_CategoryURL"] = "DateAndTime";
        this["Func_weeknum_Desc"] = "返回指定日期所屬的星期數。";
        this["Func_workday_Arg_0"] = "startDate";
        this["Func_workday_Arg_1"] = "天";
        this["Func_workday_Arg_2"] = "holidays";
        this["Func_workday_ArgDesc_0"] = "開始的日期。";
        this["Func_workday_ArgDesc_1"] = "從開始日期起要抵消的工作天數。";
        this["Func_workday_ArgDesc_2"] = "一個或多個參考範圍，包含被視為非工作日的日期。";
        this["Func_workday_Category"] = "DateAndTime";
        this["Func_workday_CategoryURL"] = "DateAndTime";
        this["Func_workday_Desc"] = "返回從指定日期起指定天數的工作日，不包括任何指定假日。";
        this["Func_year_Arg_0"] = "Date_value";
        this["Func_year_ArgDesc_0"] = "用於求取年份的日期值。";
        this["Func_year_Category"] = "DateAndTime";
        this["Func_year_CategoryURL"] = "Date_and_Time";
        this["Func_year_Desc"] = "返回日期值的年份。";
        this["Func_yearfrac_Arg_0"] = "StartDate";
        this["Func_yearfrac_Arg_1"] = "EndDate";
        this["Func_yearfrac_Arg_2"] = "Basis";
        this["Func_yearfrac_ArgDesc_0"] = "開始日期是該範圍內的開始日期。";
        this["Func_yearfrac_ArgDesc_1"] = "結束日期是該範圍內的結束日期。";
        this["Func_yearfrac_ArgDesc_2"] = "\r\n      基礎確定要使用的天數的基礎。\r\n      0 (預設 = US (NASD) 30/360\r\n\t    1 =實際/實際\r\n\t    2 =實際/360\r\n\t    3 =實際/365\r\n\t    4 = 歐洲 30/360\r\n    ";
        this["Func_yearfrac_Category"] = "DateAndTime";
        this["Func_yearfrac_CategoryURL"] = "DateAndTime";
        this["Func_yearfrac_Desc"] = "返回代表指定開始日期和結束日期之間指定的年份部分的分數。";
        this["GenerateTableColumnName"] = "資料行";
        this["GenerateTableName"] = "表";
        this["LD_Chart_AxisDisplayUnits"] = "Hundreds:Hundreds, Thousands:Thousands, TenThousands:x 10000, HundredThousands:x 100000, Millions:Millions, TenMillions:x 10000000, HundredMillions:x 100000000, ThousandMillions:Billions, MillionMillions:Trillions";
        this["LD_Fallback_ChartEx_Line1"] = "此圖表在您的 Excel 版本中不可用。";
        this["LD_Fallback_ChartEx_Line2"] = "編輯此形狀或將此工作簿儲存為其他檔案格式將永久中斷圖表。";
        this["LE_AllMustHavGuidsIfAnyHasThem"] = "如果任何現有群組具有 Guid，則無法將 GenerateGuidsForGroups 設為 false。";
        this["LE_ArgumentException_AnchorCellFromOtherWorksheet"] = "指定的儲存格與形狀不在同一工作表中。";
        this["LE_ArgumentException_ArrayFormulaMustHaveSingleRegion"] = "陣列公式只能套用至一個儲存格區域。";
        this["LE_ArgumentException_ArrayTooSmall"] = "陣列的大小不足以將所有值複製到其中。";
        this["LE_ArgumentException_CannotAddStandardTableStyle"] = "不能將標準表格樣式添加到 CustomTableStyleCollection。";
        this["LE_ArgumentException_CannotApplyDVRuleToTotalCell"] = "指定參考中的一個或多個儲存格為表格中的總儲存格，無法套用資料驗證規則。";
        this["LE_ArgumentException_CannotCreateEmptyColorInfo"] = "空白不能用於創建 WorkbookColorInfo 實例。";
        this["LE_ArgumentException_CellMustBeOnSameWorksheetAsReferencesCollection"] = "儲存格必須來自與參考集合相同的工作表。";
        this["LE_ArgumentException_CellShiftedOffWorksheet"] = "儲存格已被刪除或移出工作表，不再代表工作表位置。";
        this["LE_ArgumentException_CellsInTableFromOtherWorksheet"] = "指定的區域與資料表不在同一工作表中。";
        this["LE_ArgumentException_CellValueStringLength"] = "分配給儲存格的字串值的長度不得大於{0}。";
        this["LE_ArgumentException_ColumnFormulaAlreadyAppliedToCell"] = "該公式已套用至儲存格。複製它以用作列公式。";
        this["LE_ArgumentException_ColumnFromDifferentWorksheet"] = "該列與此集合不屬於同一工作表。";
        this["LE_ArgumentException_ColumnInputCellFromOtherWorksheet"] = "指定的列輸入儲存格與資料表不在同一工作表上。";
        this["LE_ArgumentException_ColumnNotInTable"] = "該列來自其他表格。";
        this["LE_ArgumentException_ColumnRemovedFromWorksheet"] = "該列已從工作表中移除。";
        this["LE_ArgumentException_ColumnsToRepeatAtLeftOutsideRange"] = "範圍超出工作表中可用列的範圍。";
        this["LE_ArgumentException_CommentTooLong"] = "註釋長度必須少於 255 個字元。";
        this["LE_ArgumentException_CustomViewNameAlreadyExists"] = "名稱為 \"{0}\" 的自訂檢視已存在。";
        this["LE_ArgumentException_DefaultTableStyleNotInWorkbook"] = "指定的表格樣式不在 CustomTableStyles 或 StandardTableStyles 集合中。";
        this["LE_ArgumentException_DisplayTextTooLong"] = "超連結顯示文字必須少於 256 個字元。";
        this["LE_ArgumentException_DuplicateDisplayValue"] = "指定的顯示值已在集合中。";
        this["LE_ArgumentException_DuplicateFixedDateGroup"] = "指定的 FixedDateGroup 已在集合中。";
        this["LE_ArgumentException_DuplicateTableStyle"] = "指定的樣式已經在 CustomTableStyleCollection 中。";
        this["LE_ArgumentException_DuplicateTableStyleName_Existing"] = "指定的名稱與集合中其他樣式的名稱匹配。";
        this["LE_ArgumentException_DuplicateTableStyleName_New"] = "指定的樣式名稱與集合中的另一個樣式相符。";
        this["LE_ArgumentException_DuplicateWorksheetProtectedRangeTitle"] = "標題為 '{0}' 的範圍已存在。";
        this["LE_ArgumentException_DV_ArrayFormulaCannotBeUsed"] = "不能將 ArrayFormula 用作 {0}。";
        this["LE_ArgumentException_DV_CellFromOtherWorksheet"] = "儲存格與資料驗證規則的集合不屬於同一個工作表。";
        this["LE_ArgumentException_DV_ConstraintFormulaMustBeNumberOrReference"] = "{0}必須設定為數字，數字運算式或對單個儲存格的引用。";
        this["LE_ArgumentException_DV_FormulaCannotBeNull"] = "將資料驗證規則套用至工作表時，無法將{0}設為空值。";
        this["LE_ArgumentException_DV_FormulaCannotFindNamedReference"] = "{0} 包含工作簿中不存在的命名參考。";
        this["LE_ArgumentException_DV_FormulaCannotFindWorksheetReference"] = "{0} 包含工作簿中不存在的工作表上的參考。";
        this["LE_ArgumentException_DV_FormulaCannotReferenceOtherWorkbook"] = "{0} 不能包含對外部工作簿的引用。";
        this["LE_ArgumentException_DV_FormulaInvalidForWorkbookFormat"] = "{0} 引用在 {1} 工作簿格式中無效的儲存格。";
        this["LE_ArgumentException_DV_InvalidAddress"] = "指定的地址不是有效的儲存格或區域地址。";
        this["LE_ArgumentException_DV_InvalidExcelDate"] = "指定的日期無法在 Excel 中顯示。";
        this["LE_ArgumentException_DV_InvalidFormula"] = "資料驗證規則中使用了無效的公式。";
        this["LE_ArgumentException_DV_InvalidMessageLength"] = "{0} 不得超過 {1} 個字元。";
        this["LE_ArgumentException_DV_ListFormulaCannotHaveEmptyString"] = "ValuesFormula 中的字串常數不得為空字串。";
        this["LE_ArgumentException_DV_ListFormulaMustBeStringOrReference"] = "必須將 ValuesFormula 設定為字串常數或對同一工作簿中單個儲存格或區域的引用。";
        this["LE_ArgumentException_DV_ListFormulaReferenceMustBeOneDimensional"] = "ValuesFormula 包含具有多個行和列的區域參考。該區域必須包含一行或一列。";
        this["LE_ArgumentException_DV_LowerGreaterThanUpperContraint"] = "下限約束的值必須小於或等於上限約束。";
        this["LE_ArgumentException_DV_MustHaveOneAcceptedValue"] = "必須指定至少一個值作為接受值。";
        this["LE_ArgumentException_DV_ReferencesAlreadyContainDVs"] = "指定的參考資料已包含一個或多個資料驗證規則。";
        this["LE_ArgumentException_DV_ReferencesFromOtherWorksheet"] = "引用集合與資料驗證規則的集合不屬於同一個工作表。";
        this["LE_ArgumentException_DV_RegionFromOtherWorksheet"] = "該區域與資料驗證規則的集合不屬於同一個工作表。";
        this["LE_ArgumentException_DV_RuleAppliedOnOtherWorksheet"] = "資料驗證規則已套用至其他工作表。";
        this["LE_ArgumentException_DV_UpperLessThanLowerContraint"] = "上限約束的值必須大於或等於下限約束。";
        this["LE_ArgumentException_EditRangeAlreadyOwned"] = "該範圍已存在於工作表的 AllowedEditRanges 中。";
        this["LE_ArgumentException_EditRangeHasRangeFromOtherWorksheet"] = "編輯範圍的工作表範圍不同於 AllowedEditRanges。";
        this["LE_ArgumentException_EmptyCustomList"] = "清單中必須至少有一個字串。";
        this["LE_ArgumentException_EndIndexLessThanZero"] = "範圍的結尾必須大於或等於零。";
        this["LE_ArgumentException_FileDoesntContainsWorkbookStream"] = "指定的串流或檔案不包含有效的工作簿。";
        this["LE_ArgumentException_FormulaReferencesInvalidCells"] = "指定的公式引用了工作簿當前檔案格式不支援的儲存格。";
        this["LE_ArgumentException_HashSetFull"] = "雜湊集中的項目過多。";
        this["LE_ArgumentException_HyperlinkAlreadyOwned"] = "超連接已屬於超連接集合。";
        this["LE_ArgumentException_HyperlinkSourceFromOtherWorksheet"] = "超連接具有來自不同工作表的來源儲存格或區域。";
        this["LE_ArgumentException_HyperlinkTargetFromOtherWorkbook"] = "目標來自與超連接所有者不同的工作簿。";
        this["LE_ArgumentException_InfiniteColumnWidth"] = "列寬不能為無窮大。";
        this["LE_ArgumentException_Interval"] = "間隔參數應為 DateInterval 枚舉中的值之一";
        this["LE_ArgumentException_IntervalStr"] = "間隔參數無法轉換為 DateInterval 枚舉中的值之一";
        this["LE_ArgumentException_InvalidCellAddress"] = "指定的地址不是有效的儲存格地址。";
        this["LE_ArgumentException_InvalidCommentPositioningMode"] = "只能將 DontMoveOrSizeWithCells 和 MoveAndSizeWithCells ShapePositioningModes 應用於註釋形狀。";
        this["LE_ArgumentException_InvalidCustomFilterOperandNumber"] = "值不能為無窮大或 NaN。";
        this["LE_ArgumentException_InvalidCustomFilterOperator"] = "當比較值不是字串時，比較運算子不得為 BeginsWith，DoesNotBeginWith，Contains，DoesNotContain，EndsWith 或 DoesNotEndWith。";
        this["LE_ArgumentException_InvalidDatePeriodFilterValue"] = "該值不在範圍內。對於月份的篩選條件類型，必須為 1-12。對於四分之一過濾器類型，必須為 1-4。";
        this["LE_ArgumentException_InvalidDpi"] = "指定的 Dpi 值無效。";
        this["LE_ArgumentException_InvalidExcelDate"] = "該值無法在 Excel 中表示為日期。";
        this["LE_ArgumentException_InvalidFileFormat"] = "無效或無法識別的檔案格式。";
        this["LE_ArgumentException_InvalidFormula"] = "指定的公式無效。";
        this["LE_ArgumentException_InvalidGradientStopColor"] = "漸變停止中無法使用自動和系統顏色。";
        this["LE_ArgumentException_InvalidNamedReferenceName"] = "該名稱不是有效的已命名參考名稱。";
        this["LE_ArgumentException_InvalidPaletteColor_EmptyOrSystem"] = "空白或系統顏色無法添加至調色板。";
        this["LE_ArgumentException_InvalidPaletteColor_NonOpaque"] = "只能將不透明顏色添加至調色板。";
        this["LE_ArgumentException_InvalidReferencesString"] = "參考值的格式不正確。";
        this["LE_ArgumentException_InvalidRegionAddress"] = "指定的地址不是有效的區域地址。";
        this["LE_ArgumentException_InvalidRowOrColumnRange"] = "無效的行或列範圍。";
        this["LE_ArgumentException_InvalidSeriesValues"] = "無效的系列值。必須至少有一個值，且每個值必須是字串或可以轉換為雙精度的物件。";
        this["LE_ArgumentException_InvalidTarget"] = "超連接目標必須是字串，WorksheetCell，WorksheetRegion 或 NamedReference。";
        this["LE_ArgumentException_InvalidTopOrBottomFilterValue"] = "該值必須介於 1 到 500 之間。";
        this["LE_ArgumentException_InvalidWorksheetName"] = "工作表名稱不得包含以下任何字元: \r\n\t\t: \\ / ? * [ ]";
        this["LE_ArgumentException_InvalidWorksheetNameStartingQuote"] = "工作表名稱不能以單引號開頭或結尾。";
        this["LE_ArgumentException_LessThanTwoGradientStops"] = "必須指定至少兩個梯度停點。";
        this["LE_ArgumentException_NamedReferenceNameTooLong"] = "命名參考的名稱不得超過 255 個字元。";
        this["LE_ArgumentException_NaNDefaultColumnWidth"] = "NaN 不能用作預設列寬。";
        this["LE_ArgumentException_NewRegionOverlapsFilterArea"] = "該操作正在嘗試更改您的工作表上的篩選範圍，因此無法完成。要完成此操作，需要移除工作表中的自動篩選條件。";
        this["LE_ArgumentException_NewTableRegionCannotMoveHeaders"] = "新表格區域無法將標題移至新行。";
        this["LE_ArgumentException_NewTableRegionFromWrongWorksheet"] = "指定的區域與表格不在同一工作表中。";
        this["LE_ArgumentException_NewTableRegionMustHaveDataRows"] = "新表格區域必須包含至少一個資料行。";
        this["LE_ArgumentException_NewTableRegionOverlapOld"] = "新表格區域必須與先前表格區域重疊。";
        this["LE_ArgumentException_NewTableRegionOverlapsBlockingValue"] = "表格無法調整大小至指定區域，因為它包含多儲存格陣列公式或資料表格。";
        this["LE_ArgumentException_NewTableRegionOverlapsMergedRegion"] = "無法將表格重新調整大小至指定區域，因為表格包含合併的儲存格。";
        this["LE_ArgumentException_NewTableRegionOverlapsOtherTable"] = "無法調整表格的大小至指定區域，因為它包含另一個表格。";
        this["LE_ArgumentException_NoRegionsInArray"] = "指定的陣列沒有區域。";
        this["LE_ArgumentException_NotEnum"] = "類型 '{0}' 不是枚舉";
        this["LE_ArgumentException_NotSupportedCodePage"] = "不支援的程式碼頁";
        this["LE_ArgumentException_NPer"] = "nPer 不能為 0";
        this["LE_ArgumentException_NPerFunction"] = "NPer 功能的參數組合錯誤";
        this["LE_ArgumentException_PA_PrintAreasCannotOverlap"] = "指定的列印區域與現有列印區域之一重疊。";
        this["LE_ArgumentException_ParentStyleFromOtherWorkbook"] = "指定的樣式與儲存格格式來自不同的工作簿。";
        this["LE_ArgumentException_PB_CantInsertBreakAtIndex"] = "此集合不支援在索引處插入項目。";
        this["LE_ArgumentException_PB_CantSetBreakAtIndex"] = "此集合不支援索引設定項目。";
        this["LE_ArgumentException_PB_MustBeWithinPrintArea"] = "分頁符必須在列印區域內。";
        this["LE_ArgumentException_PB_PageBreakCantBeA1Cell"] = "無法在工作表的左上方插入分頁符。";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeLeftColumn"] = "無法在工作表的左側插入分頁符。";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeTopRow"] = "無法在工作表頂部插入分頁符。";
        this["LE_ArgumentException_PB_PageBreaksCannotAddExistingInstance"] = "集合中已存在指定的實例。";
        this["LE_ArgumentException_PB_PageBreaksCannotOverlap"] = "分頁符不得與工作表上的其他分頁符重疊。";
        this["LE_ArgumentException_PB_PrintAreaMustBeInPrintAreas"] = "分頁符位於不在 PrintAreas 集合中的列印區域中。";
        this["LE_ArgumentException_PB_PrintAreaMustBeOnSameWorksheet"] = "分頁符位於與分頁符集合不在同一工作表上的列印區域中。";
        this["LE_ArgumentException_Pmt"] = "Pmt 不能為 0";
        this["LE_ArgumentException_PrintAreaShiftedOffWorksheet"] = "分頁符位於已移出工作表的列印區域，不再代表有效的儲存格範圍。";
        this["LE_ArgumentException_RangeFromOtherWorksheetThanProtectedRange"] = "指定的範圍來自與編輯範圍不同的工作表。";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheet"] = "該區域必須來自同一工作表。";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheetAsReferencesCollection"] = "該區域必須來自與參考集合相同的工作表。";
        this["LE_ArgumentException_RegionsFromMixedWorksheets"] = "陣列中指定的所有區域都必須來自同一工作表。";
        this["LE_ArgumentException_RegionShiftedOffWorksheet"] = "該區域已移出工作表，不再代表有效的儲存格範圍。";
        this["LE_ArgumentException_RegionsShiftedOffWorksheet"] = "指定的一個或多個區域已移出工作表，不再代表有效的儲存格範圍。";
        this["LE_ArgumentException_RowFromDifferentWorksheet"] = "該行與此集合不屬於同一工作表。";
        this["LE_ArgumentException_RowInputCellFromOtherWorksheet"] = "指定的行輸入儲存格與資料表不在同一工作表上。";
        this["LE_ArgumentException_RowRemovedFromWorksheet"] = "該行已從工作表中移除。";
        this["LE_ArgumentException_SelectedWorksheetFromOtherWorkbook"] = "該工作表不屬於與這些視窗選項相關的工作簿。";
        this["LE_ArgumentException_ShapeCannotBeAdded"] = "無法將指定的形狀添加至形狀集合。";
        this["LE_ArgumentException_SourceFromOtherWorksheetThanHyperlink"] = "指定的來源儲存格或區域來自與超連接不同的工作表。";
        this["LE_ArgumentException_StartIndexLessThanZero"] = "範圍的起點必須大於或等於零。";
        this["LE_ArgumentException_StyleNameAlreadyExists"] = "工作簿中已存在名稱為 \"{0}\" 的樣式。";
        this["LE_ArgumentException_StyleNameTooLong"] = "樣式名稱不得超過 255 個字元。";
        this["LE_ArgumentException_TableColumnRemovedFromTable"] = "表格列已從表格中移除。";
        this["LE_ArgumentException_TableStyleFromOtherWorkbook"] = "指定的 WorksheetTableStyle 與表格不屬於同一工作簿。";
        this["LE_ArgumentException_TargetTooLong"] = "超連結目標地址必須少於 256 個字元。";
        this["LE_ArgumentException_TextLengthGreaterThanMax"] = "形狀或註釋中的文字長度不得大於{0}。";
        this["LE_ArgumentException_ToolTipTooLong"] = "超連結工具提示必須少於 256 個字元。";
        this["LE_ArgumentException_TopLeftWindowBounds"] = "視窗邊界的左上方必須在 -32768 和 32767 之間。";
        this["LE_ArgumentException_WeekdayFirst"] = "WeekdayFirst 參數應介於 FirstDayOfWeek.System 和 FirstDayOfWeek.Saturday 之間";
        this["LE_ArgumentException_WidthHeightWindowBounds"] = "窗口邊界的寬度和高度必須在 0 到 65535 之間。";
        this["LE_ArgumentException_WorksheetNameAlreadyExists"] = "工作簿中已存在名稱為 \"{0}\" 的工作表。";
        this["LE_ArgumentException_WorksheetNameTooLong"] = "工作表名稱的長度必須在 1 到 31 個字元之間。";
        this["LE_ArgumentException_WorksheetProtectedRangeTitleTooLong"] = "編輯範圍的標題不得超過 255 個字元。";
        this["LE_ArgumentException_WorksheetScopeFromOtherWorkbook"] = "該工作表與此集合不屬於同一工作簿。";
        this["LE_ArgumentNullException_AnchorCell"] = "當工作表或群組上存在形狀時，儲存格不能為空值。";
        this["LE_ArgumentNullException_CustomViewName"] = "自訂資料檢視名稱不得為空白或空白。";
        this["LE_ArgumentNullException_Encoder"] = "影像編碼器不得為空值。";
        this["LE_ArgumentNullException_FindNamedReference"] = "要尋找的名稱不能為空值或空白。";
        this["LE_ArgumentNullException_FormulaCantBeNull"] = "命名參考的公式不能為空值或空白。";
        this["LE_ArgumentNullException_HiddenColumn"] = "無法將空列添加至集合。";
        this["LE_ArgumentNullException_HiddenRow"] = "無法將空行添加至集合。";
        this["LE_ArgumentNullException_Image"] = "影像不得為空。";
        this["LE_ArgumentNullException_NamedReferenceNameCantBeNull"] = "命名參考的名稱不得為空值或空白。";
        this["LE_ArgumentNullException_SaveStream"] = "無法將工作簿儲存為空流。";
        this["LE_ArgumentNullException_SelectedWorksheet"] = "工作簿必須具有有效的所選工作表。";
        this["LE_ArgumentNullException_Shape"] = "無法將空形狀添加至集合。";
        this["LE_ArgumentNullException_SourceFont"] = "來源字體不能為 null。";
        this["LE_ArgumentNullException_SourceFormatting"] = "來源格式不能為 null。";
        this["LE_ArgumentNullException_StyleName"] = "樣式名稱不得為 null 或空白。";
        this["LE_ArgumentNullException_UnformattedString"] = "未格式化的字串不得為 null。";
        this["LE_ArgumentNullException_Workbook"] = "工作簿不能為空。";
        this["LE_ArgumentNullException_WorkbookRequiredToResolveThemeColor"] = "必須指定工作簿以解析主題色彩。";
        this["LE_ArgumentNullException_WorksheetName"] = "工作表名稱不得為 null 或空白。";
        this["LE_ArgumentNullException_WorksheetProtectedRangeTitleCantBeNull"] = "編輯範圍的標題不得為 null 或空白。";
        this["LE_ArgumentNullException_WorksheetScope"] = "工作表範圍不能為空。";
        this["LE_ArgumentOutOfRange_RelativeColumnIndex"] = "相對列索引在指定區域之外。";
        this["LE_ArgumentOutOfRange_RelativeRowIndex"] = "相對行索引在指定區域之外。";
        this["LE_ArgumentOutOfRangeException_AnchorPosition"] = "指定位置的百分比之一小於 0。";
        this["LE_ArgumentOutOfRangeException_BadInsertIndex"] = "指定的索引必須大於或等於零且小於或等於現有段落的數目。";
        this["LE_ArgumentOutOfRangeException_BadRemoveAtIndex"] = "索引必須大於零且小於格式化字串中的段落數。";
        this["LE_ArgumentOutOfRangeException_CollectionIndex"] = "索引超出範圍。它必須為非負數且小於集合的大小。";
        this["LE_ArgumentOutOfRangeException_ColumnWidth"] = "列寬必須在 0 到 65535 之間。";
        this["LE_ArgumentOutOfRangeException_DefaultColumnWidth"] = "預設列寬必須在 0 到 65535 之間。";
        this["LE_ArgumentOutOfRangeException_DefaultFontHeight"] = "預設字體高度必須在 20 到 8180 之間。";
        this["LE_ArgumentOutOfRangeException_DefaultRowHeight"] = "預設行高必須介於 -1 和 8192 之間。";
        this["LE_ArgumentOutOfRangeException_DuplicateItemSorted"] = "指定的項目已應用排序條件。";
        this["LE_ArgumentOutOfRangeException_FirstVisibleTabIndex"] = "第一個可見標籤索引必須大於或等於零。";
        this["LE_ArgumentOutOfRangeException_FontHeight"] = "字型高度必須介於 20 和 8180 之間。";
        this["LE_ArgumentOutOfRangeException_GroupAddedToSelf"] = "無法將群組新增至自己的形狀集合。";
        this["LE_ArgumentOutOfRangeException_Indent"] = "縮進級別必須介於 0 和 250 之間。";
        this["LE_ArgumentOutOfRangeException_IndexNegative"] = "索引必須為非負數。";
        this["LE_ArgumentOutOfRangeException_InvalidCollectionIndex"] = "指定的索引在集合中無效。";
        this["LE_ArgumentOutOfRangeException_InvalidColorInfoTint"] = "色彩值必須介於 -1.0 和 1.0 之間。";
        this["LE_ArgumentOutOfRangeException_InvalidColumnCount"] = "{0} 在當前格式中不是有效的列計數。該值必須介於 {1} 和 {2} 之間。";
        this["LE_ArgumentOutOfRangeException_InvalidColumnIndex"] = "{0} 不是當前格式的有效列索引。該值必須介於 {1} 和 {2} 之間。";
        this["LE_ArgumentOutOfRangeException_InvalidGradientStopOffset"] = "梯度停止偏移必須在 0.0 和 1.0 之間。";
        this["LE_ArgumentOutOfRangeException_InvalidRelativeRectangleValueForGradient"] = "矩形漸變的內部矩形的值必須介於 0.0 和 1.0 之間。";
        this["LE_ArgumentOutOfRangeException_InvalidRemoveAtIndex_SortConditions"] = "移除排序條件的索引超出範圍。";
        this["LE_ArgumentOutOfRangeException_InvalidRowCount"] = "{0} 不是當前格式的有效行計數。該值必須介於 {1} 和 {2} 之間。";
        this["LE_ArgumentOutOfRangeException_InvalidRowIndex"] = "{0} 不是當前格式的有效行索引。該值必須介於 {1} 和 {2} 之間。";
        this["LE_ArgumentOutOfRangeException_InvalidTableStyleAreaStripeExtent"] = "{0} 必須介於 1 和 9 之間 (含 1 和 9)。";
        this["LE_ArgumentOutOfRangeException_LeftPaneWidth"] = "指定的窗格寬度超出有效範圍 {0} 和 {1}。";
        this["LE_ArgumentOutOfRangeException_LengthMustBePositive"] = "長度必須大於或等於 1。";
        this["LE_ArgumentOutOfRangeException_MagnificationLevel"] = "檢視放大倍數級別必須介於 10 和 400 之間。";
        this["LE_ArgumentOutOfRangeException_Margins"] = "邊距必須在 0 到 100 之間。";
        this["LE_ArgumentOutOfRangeException_MaxPagesHorizontally"] = "水平最大頁數必須在 0 到 32767 之間。";
        this["LE_ArgumentOutOfRangeException_MaxPagesVertically"] = "垂直最大頁數必須在 0 到 32767 之間。";
        this["LE_ArgumentOutOfRangeException_MaxRecursionIterations"] = "最大遞歸迭代次數必須介於 1 到 32767 之間。";
        this["LE_ArgumentOutOfRangeException_MaxSortConditions"] = "集合中的排序條件不得超過 {0}。";
        this["LE_ArgumentOutOfRangeException_NegativeStartIndex"] = "開始索引不得為負數。";
        this["LE_ArgumentOutOfRangeException_NumberOfCopies"] = "份數必須在 1 到 65535 之間。";
        this["LE_ArgumentOutOfRangeException_Per"] = "每個值應介於 0 和 nPer + 1 之間";
        this["LE_ArgumentOutOfRangeException_Rate"] = "速率應大於 -1";
        this["LE_ArgumentOutOfRangeException_Resolution"] = "分辨率必須在 0 到 65535 之間。";
        this["LE_ArgumentOutOfRangeException_RowHeight"] = "行高必須介於 0 和 8192 之間。";
        this["LE_ArgumentOutOfRangeException_ScalingFactor"] = "縮放比例必須在 10 到 400 之間。";
        this["LE_ArgumentOutOfRangeException_StartPageNumber"] = "起始頁碼必須在 -32765 和 32767 之間。";
        this["LE_ArgumentOutOfRangeException_TabBarWidth"] = "標籤欄的寬度必須介於 0 和 1000 之間。";
        this["LE_ArgumentOutOfRangeException_TopPaneHeight"] = "指定的窗格高度超出有效範圍 {0} 和 {1}。";
        this["LE_AutoColorNotAllowed"] = "顏色不能為 'Auto'。";
        this["LE_Axis_MajorMinorUnit"] = "majorUnit 參數的值必須大於 minorUnit 參數的值。";
        this["LE_Axis_NoCrossAxis"] = "如果在一個 Axis 物件上設定了 CrossAxis 屬性，則必須在所有這些物件上設定。如果您的圖表只有兩個軸，則可以自動確定交叉軸，但前提是在所有Axis物件上未設定 CrossAxis 屬性。";
        this["LE_Biff8SerializerNotLoaded"] = "'{0}' 格式需要未載入的序列化程序，在 JavaScript 中包含在 infragistics.excel_serialization_biff8.js 檔案中";
        this["LE_CategoryAxisBinning_BinWidthAndNumberOfBins"] = "'binWidth' 和 'numberOfBins' 參數是互斥的。兩者都可以未指定，但如果指定了一個，則不得指定另一個。";
        this["LE_ChartGradientFill_EmptyStops"] = "必須包含至少一個梯度檔。";
        this["LE_ChartObject_DifferentChart"] = "正在分配的物件實例與其他圖表相關聯，因此無法分配給該實例。";
        this["LE_ColumnsToRepeatAtLeftAreOutsideAvailableRange"] = "列印選項的列在左側重複，它們超出新格式的可用列範圍。";
        this["LE_ComboChartGroupCollection_DoesNotExist"] = "指定的 ComboChartGroup 在此集合中不存在。";
        this["LE_FillPercentInvalid"] = "值 {0} 不是 {1} 屬性的有效值。";
        this["LE_FormatLimitError_Indent"] = "縮進超過當前格式提供的最大值 {0}。";
        this["LE_FormatLimitError_MacroWorkbook"] = "該工作簿具有 VBA 代碼，無法另存為啟用巨集的工作簿。";
        this["LE_FormatLimitError_MaxColumnIndex"] = "列索引 {0} 大於指定格式支援的 {1} 最大值。";
        this["LE_FormatLimitError_MaxRowIndex"] = "行索引{0}超過此格式支援的 {1} 最大值。";
        this["LE_FormulaParseException_ArrayContainsConstants"] = "陣列只能包含常數運算式。";
        this["LE_FormulaParseException_ArrayHasEmptyFirstRow"] = "公式中的常數陣列在第一行中沒有項目。";
        this["LE_FormulaParseException_ArrayHasMisalignedRows"] = "公式中的常數陣列的行未對齊。陣列中的每一行必須具有相同的列數。";
        this["LE_FormulaParseException_CellReferenceAfterWorkbookName"] = "外部工作簿引用後必須跟隨一個已命名的引用，而不是儲存格引用。";
        this["LE_FormulaParseException_ExtraExpressions"] = "公式結尾處有額外的運算式。";
        this["LE_FormulaParseException_FunctionMissingClosingParen"] = "公式中的函數沒有右括號。";
        this["LE_FormulaParseException_FunctionNestingTooDeep"] = "公式中的函數在其他函數中的嵌套過多。";
        this["LE_FormulaParseException_IncorrectNumberOfArguments"] = "未為公式中的函數指定正確數量的引數。";
        this["LE_FormulaParseException_InvalidArguments"] = "公式中的運算子或函數具有無效引數。";
        this["LE_FormulaParseException_InvalidConditionalFormatFormula"] = "條件格式公式不能使用參考運算子，例如聯合或相交，陣列常數或結構化表格引用。";
        this["LE_FormulaParseException_InvalidErrorValue"] = "指定的錯誤類型無效。它必須是下列值之一:  \"#NULL!\"，\"#DIV/0!\"，\"#VALUE!\"，\"#REF!\"，\"#NAME?\"，\"#NUM!\", 或 \"#N/A\"。";
        this["LE_FormulaParseException_InvalidFileNameInBrackets"] = "方括號之間的檔案名稱無效。";
        this["LE_FormulaParseException_InvalidStructuredTableReference"] = "結構化表格參考不正確。";
        this["LE_FormulaParseException_InvalidWorkbookName"] = "帶引號的工作簿參考不是有效的工作簿檔案名稱。";
        this["LE_FormulaParseException_InvalidWorksheetName"] = "工作表名稱無效。";
        this["LE_FormulaParseException_MacroFunctionNotAllowed"] = "'{0}' 功能僅在命名參考中有效。";
        this["LE_FormulaParseException_Message_PortionWithError"] = "部分錯誤: {0}";
        this["LE_FormulaParseException_MissingArgumentAfterBinary"] = "公式中的二元運算子後面缺少運算元。";
        this["LE_FormulaParseException_MissingArgumentAfterParen"] = "右括號後沒有表達式。";
        this["LE_FormulaParseException_MissingArgumentAfterUnary"] = "公式中的一元運算符後面沒有有效的運算式。";
        this["LE_FormulaParseException_MissingArgumentBeforeBinary"] = "公式中的二元運算子之前缺少運算元。";
        this["LE_FormulaParseException_NamedReferenceRefsNeedSheetName"] = "命名參考公式中的所有儲存格參考都必須標識工作表。";
        this["LE_FormulaParseException_NoElementAfterArraySerapator"] = "公式中的元素分隔符後沒有陣列元素。";
        this["LE_FormulaParseException_NoEqualsSign"] = "公式必須以等號開頭。";
        this["LE_FormulaParseException_NoExclamationAfterWorkbookName"] = "工作簿參考後必須帶有感嘆號。";
        this["LE_FormulaParseException_NoExclamationAfterWorksheetName"] = "工作表參考後面必須帶有感嘆號。";
        this["LE_FormulaParseException_NoExpressions"] = "公式在等號後沒有有效的運算式。";
        this["LE_FormulaParseException_NoFileNameAfterBracket"] = "請在方括號後輸入有效的檔案名稱。";
        this["LE_FormulaParseException_NoValidTermAfterWorkbookName"] = "外部工作簿參考之後必須跟隨一個已命名的參考。";
        this["LE_FormulaParseException_NoValidTermAfterWorksheetName"] = "工作表引用後必須跟隨一個儲存格或命名引用。";
        this["LE_FormulaParseException_NoWorksheetAfterWorkbookName"] = "在方括號中的工作簿檔案名稱之後必須加上工作表名稱。";
        this["LE_FormulaParseException_StringConstantLengthTooLong"] = "指定的公式不得包含超過 {0} 個字元的字串。";
        this["LE_FormulaParseException_TooLong"] = "對於指定的工作簿格式，公式的最大長度不得超過{0}。";
        this["LE_FormulaParseException_UnknownFunction"] = "公式中的函數沒有有效的函數名稱。";
        this["LE_FormulaParseException_UnmatchedOpenBracket"] = "公式中的常數陣列沒有右括號。";
        this["LE_FormulaParseException_UnmatchedOpenParen"] = "公式中的開括號沒有匹配的閉括號。";
        this["LE_FormulaParseException_UnmatchedOpenSquareBracket"] = "工作簿檔案名稱後應加上方括號。";
        this["LE_FormulaParseException_WorkbookNameMissingEndQuote"] = "工作簿參考沒有結尾引號。";
        this["LE_FormulaParseException_WorksheetRangeMissingEndingName"] = "工作表範圍未指定第二個工作表名稱。";
        this["LE_GradientStop_InvalidPosition"] = "必須指定介於 0 和 1 之間的數字。";
        this["LE_IconSetInvalid"] = "'IconSetNoIcon' 不是 IconSet 屬性的有效設定。";
        this["LE_IndexOutOfRangeException_ArrayBounds"] = "索引超出陣列的範圍。";
        this["LE_InvalidEnumArgumentException_DefaultPatternCannotBeUsed"] = "預設 FillPatternStyle 不能用於創建 CellFillPattern。";
        this["LE_InvalidLineWeight"] = "LineWeight 不得小於 0 或大於 1584。";
        this["LE_InvalidOperationException_AnchorCommentBeforeApplyingToCell"] = "註釋必須同時設定 TopLeftCornerCell 和 BottomRightCornerCell，或者在將註釋套用至儲存格之前，它們都必須為空值。";
        this["LE_InvalidOperationException_AnchorShapeBeforeAddingToCollection"] = "將形狀添加到工作表或群組之前，必須同時設定 TopLeftCornerCell 和 BottomRightCornerCell。";
        this["LE_InvalidOperationException_ArrayFormulaAlreadyApplied"] = "陣列公式已套用至儲存格範圍。首先清除現有儲存格範圍。";
        this["LE_InvalidOperationException_ArrayFormulaAppliedInTable"] = "陣列公式無法應用於存在表格的區域。";
        this["LE_InvalidOperationException_ArrayFormulaInMergedCell"] = "陣列公式在合併的儲存格中無效。";
        this["LE_InvalidOperationException_BottomRightAnchorFromOtherWorksheet"] = "將形狀新增至此集合後，會將其放置在與其 BottomRightCornerCell 不同的工作表上。";
        this["LE_InvalidOperationException_BuiltInStyleNameCannotBeChanged"] = "內置樣式的名稱無法更改。";
        this["LE_InvalidOperationException_CannotAddColorsToPaletteDirectly"] = "無法將顏色添加至 WorkbookColorPalette。使用索引器設定特定索引的顏色。";
        this["LE_InvalidOperationException_CannotAddParagraphDirectly"] = "段落不能直接添加。";
        this["LE_InvalidOperationException_CannotAddTableDirectly"] = "無法將 WorksheetTable 實例直接插入集合中。使用添加方法。";
        this["LE_InvalidOperationException_CannotAddTableToRemovedWorksheet"] = "無法將表格新增至已從其擁有的工作簿移除的工作表中。";
        this["LE_InvalidOperationException_CannotApplyFilterWhileHeaderRowIsHidden"] = "隱藏表格標題行時，無法套用篩選條件。";
        this["LE_InvalidOperationException_CannotApplyFilterWhileUIIsHidden"] = "隱藏篩選條件使用者介面時，無法套用篩選條件。";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfData"] = "無法插入 {0}，因為它會將資料移出工作表的底部。";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfObject"] = "無法插入 {0}，因為它將移動形狀或註釋掉工作表的底部。";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitBlockingValue"] = "無法插入 {0}，因為它會移動陣列公式或資料表格中的儲存格。";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitMergedRegion"] = "無法插入 {0}，因為它會移動合併區域中的儲存格。";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitTable"] = "無法插入 {0}，因為它會移動另一個表格中的儲存格。";
        this["LE_InvalidOperationException_CannotInsertTableRow_TableOnBottomOfWorksheet"] = "無法插入 {0}，因為表格位於工作表的底部。";
        this["LE_InvalidOperationException_CannotModifyKeysCollection"] = "無法修改 Keys 集合。";
        this["LE_InvalidOperationException_CannotModifyStandardTableStyle"] = "標準表格樣式無法修改。";
        this["LE_InvalidOperationException_CannotModifyValuesCollection"] = "值集合無法修改。";
        this["LE_InvalidOperationException_CannotMoveDisconnectedWorksheet"] = "無法移動工作表，因為它不屬於工作簿。";
        this["LE_InvalidOperationException_CannotRemoveColorsToPaletteDirectly"] = "無法從 WorkbookColorPalette 中移除顏色。使用重置方法清除所有自訂顏色。";
        this["LE_InvalidOperationException_CannotRemoveNormalStyle"] = "普通樣式無法移除。";
        this["LE_InvalidOperationException_CannotSetParentStyleOnStyle"] = "無法在屬於另一種風格的格式上設定該風格。";
        this["LE_InvalidOperationException_CannotShift_LossOfData"] = "無法將資料移出工作表。";
        this["LE_InvalidOperationException_CannotShift_LossOfObject"] = "無法將物件移離工作表。";
        this["LE_InvalidOperationException_CannotShift_SplitBlockingValue"] = "無法移入陣列公式或資料表格。";
        this["LE_InvalidOperationException_CannotShift_SplitMergedRegion"] = "無法在合併的區域內移動。";
        this["LE_InvalidOperationException_CannotShift_SplitTable"] = "無法移入表格。";
        this["LE_InvalidOperationException_CantAddChartInChart"] = "不能將圖表添加到其他圖表的 Shapes 集合中。";
        this["LE_InvalidOperationException_CantAddCustomView"] = "自訂資料檢視無法直接新增至集合。";
        this["LE_InvalidOperationException_CantAddDataTable"] = "無法直接新增資料表格。";
        this["LE_InvalidOperationException_CantAddMergedRegion"] = "合併的儲存格區域無法直接新增。";
        this["LE_InvalidOperationException_CantAddNamedReference"] = "不能直接添加已命名的引用。";
        this["LE_InvalidOperationException_CantAddWorksheet"] = "無法將工作表直接添加至此集合。";
        this["LE_InvalidOperationException_CantApplyRemovedCustomView"] = "如果自訂檢視已從工作簿的自訂檢視集合中移除，則無法應用。";
        this["LE_InvalidOperationException_CantChangeArrayFormula"] = "陣列公式的儲存格無法變更。";
        this["LE_InvalidOperationException_CantChangeDataTable"] = "資料表的內部儲存格無法變更。";
        this["LE_InvalidOperationException_CantModifyCollection"] = "集合無法修改。";
        this["LE_InvalidOperationException_CantOverlapArrayFormula"] = "合併的儲存格區域不得與陣列公式區域重疊。";
        this["LE_InvalidOperationException_CantOverlapDataTableInterior"] = "如果合併的儲存格區域延伸至資料表格的最上行或最左列之外，則無法與資料表中的儲存格重疊。";
        this["LE_InvalidOperationException_CantSaveEditRangeWithoutRange"] = "工作表的 AllowedEditRanges 必須至少有一個範圍。";
        this["LE_InvalidOperationException_CantSaveWithNoVisibleWorksheets"] = "工作簿中至少必須有一個工作表可見。";
        this["LE_InvalidOperationException_CantSaveWithNoWorksheets"] = "沒有工作表的工作簿無法儲存";
        this["LE_InvalidOperationException_CantSetCircularityErrorDirectly"] = "圓度誤差值無法直接套用至儲存格。";
        this["LE_InvalidOperationException_CantSetDataTableDirectly"] = "無法將資料表設定為儲存格的值。";
        this["LE_InvalidOperationException_CantSetFormulaDirectly"] = "無法直接在儲存格上設定公式，請改用 ApplyFormula 方法或公式實例的 ApplyTo 方法之一。";
        this["LE_InvalidOperationException_CellsInTableMinSize"] = "資料表中的儲存格區域在每個維度上必須至少有兩個儲存格。";
        this["LE_InvalidOperationException_ChartNotSupported"] = "此格式不支援 WorksheetChart 修改。";
        this["LE_InvalidOperationException_ClearUnknownShapeData"] = "未知此形狀的所有資料，請從集合中移除該形狀。";
        this["LE_InvalidOperationException_CollectionLongerThanMaxValue"] = "集合包含的條目數量超過指定的最大值。";
        this["LE_InvalidOperationException_CollectionModifiedWhileEnumerating"] = "枚舉時修改了集合。";
        this["LE_InvalidOperationException_CurrentTableRowAddressNeedsOriginCell"] = "要使用當前行解析結構化表格參考，必須指定原始儲存格。";
        this["LE_InvalidOperationException_CustomViewNoVisibleWorksheets"] = "名為 \"{0}\" 的自訂檢視中至少一個工作表必須可見。";
        this["LE_InvalidOperationException_DataTableAppliedInTable"] = "資料表不得位於有資料表的區域。";
        this["LE_InvalidOperationException_DataTableFormulaCannotBeApplied"] = "資料表格公式無法直接套用至儲存格。";
        this["LE_InvalidOperationException_DataTableInMergedCell"] = "資料表格的內部儲存格不能成為合併儲存格區域的一部分。";
        this["LE_InvalidOperationException_DataTableRemoved"] = "從工作表中移除資料表格後，無法設定行和列輸入儲存格。";
        this["LE_InvalidOperationException_DeprecatedFillColorSetWithGradientFill"] = "在儲存格上設定漸變時，無法設定填充圖案顏色。";
        this["LE_InvalidOperationException_EncryptedWorkbooksNotSupported"] = "正在加載的工作簿已加密。不支援加密的工作簿。";
        this["LE_InvalidOperationException_ExtensionDoesntMatchCurrentFormat"] = "用來儲存檔案的檔案名稱，其副檔名與工作簿的目前格式不符。";
        this["LE_InvalidOperationException_FixedValuesFilterMustAcceptAValue"] = "不能有不允許值的 FixedValuesFilter。";
        this["LE_InvalidOperationException_FormattedStringAlreadyOwned"] = "格式化的字串已經是另一個儲存格或註釋的值。首先將其複製並將複製應用於新的儲存格或形狀。";
        this["LE_InvalidOperationException_FormattedStringNotOwned"] = "未將格式化的字串或文字套用至儲存格或形狀時，無法存取字型屬性。";
        this["LE_InvalidOperationException_FormattedTextAlreadyOwned"] = "格式化的字串已經是其他形狀的值。首先將其複製並將複製應用於新的儲存格或形狀。";
        this["LE_InvalidOperationException_FormulaAlreadyOwned"] = "該公式已套用至其他儲存格。";
        this["LE_InvalidOperationException_FormulaReferencesInvalidCells"] = "該公式引用了工作簿當前檔案格式不支援的儲存格。";
        this["LE_InvalidOperationException_GetBoundsBeforeAnchorsSet"] = "在設定錨定單元之前，形狀沒有邊界。";
        this["LE_InvalidOperationException_HiddenWorksheetCannotBeSelected"] = "無法選擇隱藏的工作表。";
        this["LE_InvalidOperationException_HyperlinkSealed"] = "超連接無法修改。";
        this["LE_InvalidOperationException_ImageDisposed"] = "WorksheetImage 的影像已被處置或無效。";
        this["LE_InvalidOperationException_InputCellsBothNull"] = "列輸入和行輸入儲存格均不能為空。";
        this["LE_InvalidOperationException_InputCellsInTable"] = "輸入的儲存格不能在資料表格儲存格內。";
        this["LE_InvalidOperationException_InputCellsSame"] = "列輸入和行輸入儲存格不得為同一儲存格。";
        this["LE_InvalidOperationException_InvalidCharacterRange"] = "字體覆蓋的字元範圍不再存在於格式化字串中。";
        this["LE_InvalidOperationException_InvalidFormatOptionsInWholeTableArea"] = "指定的 FormatOptions 格式值在 WholeTable 表格區域無效。";
        this["LE_InvalidOperationException_InvalidFormatString"] = "指定的格式字串無效。";
        this["LE_InvalidOperationException_InvalidFormatString_GetTextCall"] = "儲存格的格式字串無效。要在設定 FormatString 時收到通知，請將 Workbook.ValidateFormatStrings 設定為 True。";
        this["LE_InvalidOperationException_InvalidForWorkbookFormat"] = "公式不適用於指定格式。";
        this["LE_InvalidOperationException_InvalidHeaderCellColumnAreaFormatProperty"] = "無法在 HeaderCell 表格列區域格式中設定頂部邊框值。";
        this["LE_InvalidOperationException_InvalidHeaderRowAreaFormatProperty"] = "無法在 HeaderRow 表格區域格式中設定頂部邊框值。";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFontProperty"] = "無法在表格樣式區域設定 {0} 字體值。";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatOptions"] = "指定的 FormatOptions 格式值在表格樣式區域無效。";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatProperty"] = "無法在表格樣式區域設定{0}格式值。";
        this["LE_InvalidOperationException_InvalidTotalsCellColumnAreaFormatProperty"] = "底部邊框值無法在 TotalCell 表格列區域格式中設定。";
        this["LE_InvalidOperationException_InvalidTotalsRowAreaFormatProperty"] = "底部邊框值無法在 TotalsRow 表格區域格式中設定。";
        this["LE_InvalidOperationException_InvalidWholeTableAreaFormatProperty"] = "在 WholeTable 表格區域格式上只能設定邊緣邊框值。";
        this["LE_InvalidOperationException_LastColumnBeforeFirst"] = "區域中的最後一列應大於或等於第一列。";
        this["LE_InvalidOperationException_LastRowBeforeFirst"] = "該區域的最後一行應大於第一行。";
        this["LE_InvalidOperationException_MaxCellFormats"] = "Excel 文件中不同儲存格格式的最大數量為: {0}";
        this["LE_InvalidOperationException_MaxColors"] = "Excel 工作簿中的最大顏色數量為: {0}";
        this["LE_InvalidOperationException_MaxColumns"] = "當前格式的 Excel 工作表中的最大列數為: {0}";
        this["LE_InvalidOperationException_MaxFonts"] = "工作簿中的字體數量不得超過 {0}。";
        this["LE_InvalidOperationException_MaxFormattedStrings"] = "工作簿中唯一字串的數量不得超過 {0}。";
        this["LE_InvalidOperationException_MaxRows"] = "當前格式的 Excel工作表中的最大行數為: {0}";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableLeftColumn"] = "為資料表指定的儲存格的最左列包含合併的儲存格區域，該區域延伸到這些儲存格區域之外。";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopLeftCell"] = "為資料表格指定的儲存格左上角儲存格是合併儲存格的一部分，合併儲存格延伸至這些儲存格區域之外。";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopRow"] = "為資料表指定的儲存格第一行包含合併的儲存格區域，延伸至這些儲存格區域之外。";
        this["LE_InvalidOperationException_MergedCellsAppliedInTable"] = "表格內的儲存格無法合併。";
        this["LE_InvalidOperationException_NamedReferenceNameAlreadyExists"] = "工作表中具有相同範圍的表或命名引用使用名稱 \"{0}\"。";
        this["LE_InvalidOperationException_NoRegionHasBeenSpecified"] = "未指定任何區域。";
        this["LE_InvalidOperationException_NoWorksheetContextToSetShapeBounds"] = "當 shapes 集合未與工作表關聯時，無法調用 Add 的過載。調用採用工作表實例的 Add 的過載。";
        this["LE_InvalidOperationException_OverlappingTable"] = "指定區域中還有另一個表格。";
        this["LE_InvalidOperationException_ReadOnlyFont"] = "字體物件是唯讀的，無法更改。";
        this["LE_InvalidOperationException_ReadOnlyFormat"] = "格式物件為唯讀格式，無法更改。";
        this["LE_InvalidOperationException_RelativeR1C1AddressNeedsOriginCell"] = "要解析相對 R1C1 地址，必須指定原始儲存格。";
        this["LE_InvalidOperationException_ResizedTableMustBeOnWorksheet"] = "表格必須在工作表上以重新調整大小。";
        this["LE_InvalidOperationException_ResolvedFormatCannotBeModified"] = "解析的儲存格格式為唯讀格式，無法更改。";
        this["LE_InvalidOperationException_ShapeCantChangeOrientation"] = "此形狀無法旋轉或翻轉。";
        this["LE_InvalidOperationException_ShapeFillSolidIsImmutable"] = "ShapeFillSolid 是不可變的。取而代之，創建一個新的 ShapeFillSolid。";
        this["LE_InvalidOperationException_ShapeInAnotherCollection"] = "形狀已存在於另一個工作表或形狀群組中。";
        this["LE_InvalidOperationException_ShapeOutlineSolidIsImmutable"] = "ShapeOutlineSolid 是不可變的。取而代之，創建一個新的 ShapeOutlineSolid。";
        this["LE_InvalidOperationException_ShowFilterUIWhileHeaderRowHidden"] = "隱藏標題行時，無法顯示篩選器使用者介面。";
        this["LE_InvalidOperationException_SystemColorsAccessedFromWrongThread"] = "無法從後台線程存取 SystemColors。";
        this["LE_InvalidOperationException_TableReferenceMustBeUsedFromInsideTable"] = "該公式具有結構化表格參考，只能在表格內部使用。";
        this["LE_InvalidOperationException_TableReferenceToMissingTable"] = "該公式引用了不存在的表格。";
        this["LE_InvalidOperationException_TableReferenceToMissingTableColumn"] = "公式引用了不存在的表格列。";
        this["LE_InvalidOperationException_TopLeftAnchorFromOtherWorksheet"] = "將形狀添加至此集合後，會將其放置在與其 TopLeftCornerCell 不同的工作表上。";
        this["LE_InvalidOperationException_TotalsRowCannotBeShownInLastRow"] = "當表格佔據工作表的最後一行時，無法顯示總計行。";
        this["LE_InvalidOperationException_UnknownBIFFFormat"] = "工作簿檔案的格式未知。";
        this["LE_InvalidOperationException_UnsupportedBIFFFormat"] = "工作簿檔案已以 {0} 格式儲存，不支援此格式。";
        this["LE_InvalidOperationException_WorkbookDataViolatesFormatLimits"] = "該工作簿包含的資料將超出指定格式所施加的限制。";
        this["LE_InvalidRegion"] = "無效的區域地址。";
        this["LE_InvalidValueType"] = "'{0}' 不是 ValueType 屬性的有效值。";
        this["LE_LocationNotSingleRowOrColumn"] = "位置參考無效，因為儲存格並非全部位於同一列或同一行。選擇全部在單一列或行中的儲存格。";
        this["LE_LocationOrDataAreaInvalid"] = "位置或資料範圍的參考無效。";
        this["LE_LocationWorksheetMismatch"] = "位置必須來自此工作表。";
        this["LE_NotSingleTargetFormula"] = "只能支援單一目標公式。";
        this["LE_NotSupportedException_CellType"] = "{0} 不是支援的儲存格值類型。";
        this["LE_NotSupportedException_NoPackageFactory"] = "packageFactory 不能為空值。從 Excel2007 工作簿格式加載並使用 Infragistics2.Documents.Excel 組件時，必須提供 IPackageFactory 來處理資料封裝。如果您使用的是 DotNet Framework 3.0 或更高版本，請改用 Infragistics3.Documents.Excel 組件，封裝將由 WindowsBase 類處理。";
        this["LE_OpenPackagingNonConformanceException_ContainsDublinCoreRefinements"] = "IPackagePart 是核心屬性關係，包含對 'dcterms:created' 和 'dcterms:modified' 以外的都柏林核心元素的改進。";
        this["LE_OpenPackagingNonConformanceException_ContainsXmlLanguageAttribute"] = "IPackagePart 是核心屬性關係，並包含包含 'xml:lang' 屬性的元素。";
        this["LE_OpenPackagingNonConformanceException_CorePropertiesRelationshipAlreadyProcessed"] = "IPackagePart 是核心屬性關係，但已為關聯的 IPackage 處理過。";
        this["LE_OpenPackagingNonConformanceException_DuplicatePartName"] = "具有相同名稱的 IPackagePart 已屬於此 IPackage。IPackage 不得包含具有相同名稱的零件。請注意，檢查重複值不區分大小寫。";
        this["LE_OpenPackagingNonConformanceException_UsesMarkupCompatibilityNamespace"] = "IPackagePart 是核心屬性關係，並使用標記相容性命名空間。";
        this["LE_OpenPackagingNonConformanceException_XmlContainsDocumentTypeDefinition"] = "IPackagePart 包含 DTD (文件類型定義) 聲明。Open Packaging Convention 規則禁止在 DTD 聲明中使用 XML 內容。";
        this["LE_OpenPackagingNonConformanceException_XsiTypeAttributeInvalid"] = "IPackagePart 是核心屬性關係，包含包含 'xsi:type' 屬性的元素，或包含必需且不包含 'xsi:type' 屬性的元素。";
        this["LE_OpenXmlSerializerNotLoaded"] = "'{0}' 格式需要未載入的序列化程序，在 JavaScript 中包含在 infragistics.excel_serialization_openxml.js 檔案中";
        this["LE_PercentageValueOutOfRange"] = "百分比值 {0} 必須在 0 到 100 之間。";
        this["LE_PercentileValueOutOfRange"] = "百分數值 {0} 必須在 0 到 100 之間。";
        this["LE_PercentRangeInvalid"] = "{0}必須小於 {1}。";
        this["LE_RowsToRepeatAtTopAreOutsideAvailableRange"] = "列印選項的行會在頂部重複，但超出新格式的可用行範圍。";
        this["LE_Serialization_NoPrimaryCategoryAxis"] = "圖表類型 '{0}' 需要主要類別軸。將主要類別軸新增至圖表的 AxisCollection。";
        this["LE_Serialization_NoPrimaryValueAxis"] = "圖表類型 '{0}' 需要一個主要值軸。將主要值軸新增至圖表的 AxisCollection。";
        this["LE_Series_ChartType_ChartEx"] = "'{0}' 對於 '{1}' 圖表中的一系列 ChartType 無效。";
        this["LE_Series_ChartType_InvalidForComboChart"] = "圖表類型 '{0}' 不適用於組合圖表。";
        this["LE_Series_ChartType_NonCombo"] = "該系列的圖表類型由父圖表的類型 ('{0}') 決定，無法設定。將父圖表的 ChartType 屬性設定為 '{1}'。";
        this["LE_Series_Explosion_OutOfRange"] = "爆炸屬性支援 0 到4 00 (含) 範圍內的值。";
        this["LE_Series_InvalidAxisGroup"] = "無法將AxisGroup 屬性設定為 '{0}'，因為同一軸上存在一系列類型 '{1}'，不能與一系列 '{2}' 共享同一軸。";
        this["LE_Series_InvalidSeriesTypeForComboChart"] = "組合圖表中無法使用一系列類型的 '{0}'。";
        this["LE_Series_NoPrimaryAxis"] = "必須在主軸上繪製至少一個系列。將至少一個系列的 AxisGroup 屬性變更為 'Primary'。";
        this["LE_Series_OwningSeries_NotClusteredColumn"] = "只能將 OwningSeries 屬性設定為帕累托圖表中的叢集列系列。";
        this["LE_Series_OwningSeries_NotParetoLine"] = "只能為帕累托圖表中的線系列設定 OwningSeries 屬性。";
        this["LE_Series_OwningSeries_Same"] = "OwningSeries 無法引用自己。";
        this["LE_Series_SeriesType_ChartEx"] = "SeriesType 值 '{0}' 對 '{1}' 圖表無效。";
        this["LE_Series_SeriesType_NonCombo"] = "Type 屬性僅適用於組合圖表，不能為 '{0}' 類型的圖表設定。";
        this["LE_SetValue_Formula"] = "'{0}' 值類型僅對公式有效。請改用 SetFormula 方法。";
        this["LE_SetValue_NonNumeric"] = "值參數不適用於值類型 '{0}'。使用僅接受值類型的過載。";
        this["LE_SetValue_Numeric"] = "必須為值類型 '{0}' 指定值參數。使用同時採用值和值類型的過載。";
        this["LE_WorksheetChart_ChartType_ChartEx_CannotTransition"] = "ChartType 屬性無法從 '{0}' 更改為 '{1}'。";
        this["LE_WorksheetChart_ChartType_ComboChart_From"] = "組合圖的 ChartType 屬性無法更改為 '{0}'。";
        this["LE_WorksheetChart_ChartType_ComboChart_To"] = "類型 '{0}' 的圖表無法更改為 'Combo'。可以使用 AddChart 方法創建新的組合圖，但不支援將現有圖表更改為組合圖。";
        this["LE_WorksheetChart_ComboChart_NotEnoughSeries"] = "組合圖至少需要 2 個系列。向圖表的 SeriesCollection 添加更多項。";
        this["LE_WorksheetChart_MissingRequiredAxis_StockCharts"] = "'{0}' 圖表需要一級分類軸，一級數值軸，二級分類軸和二級數值軸。";
        this["LE_WorksheetChart_NotEnoughSeries"] = "'{0}' 圖表最少需要 {1} 系列。要使用此圖表類型，您必須向 SeriesCollection 添加更多項目。";
        this["LE_WorksheetChart_Save_Series"] = "'{0}' 圖表必須包含至少一個系列。在嘗試儲存工作簿之前，向圖表中添加更多系列。";
        this["LE_WorksheetChart_Save_Series_Pareto"] = "'Pareto' 圖表必須包含一個 'ColumnClustered' 系列和一個 'Pareto' 系列。在嘗試儲存工作簿之前，向圖表中添加適當的系列。";
        this["LE_WorksheetChart_SetComboChartSourceData_InvalidChartTypes"] = "下列一種或多種指定圖表類型不適用於組合圖表: {0}";
        this["LE_WorksheetChart_SetComboChartSourceData_NotAComboChart"] = "無法調用 SetComboChartSourceData，因為當前圖表類型為 '{0}'。將 ChartType 更改為 'Combo'。";
        this["LE_WorksheetChart_SetComboChartSourceData_SeriesChartTypesNotSupported"] = "對於 '{0}' 圖表，seriesChartTypes 參數必須為 null。";
        this["LE_WorksheetChart_SetSourceData_NotEnoughSeries"] = "組合圖至少需要2個系列。將 'dataRange' 參數更改為至少 2 個系列的範圍。";
        this["LE_WorksheetChart_SetSourceData_TooManyAxes"] = "指定的圖表類型組合所需的軸數超過最大允許數量。";
        this["LE_WorksheetChart_StockChartNotSupported"] = "當前不支援對圖表類型 '{0}' 的編輯。";
        this["LE_WorksheetChart_UnknownChartNotSupported"] = "此圖表是從現有工作表載入的，但無法確定其 ChartType。目前不支援編輯未知圖表類型。";
        this["LER_ArgumentOutOfRangeException_OutlineLevel"] = "大綱等級必須介於0和7之間。";
        this["LER_Exception_KeyNotFound"] = "未找到密鑰。";
        this["LER_Exception_MergedRegionsOverlap"] = "合併的儲存格區域不可重疊。";
        this["TableHeaderRowDescription"] = "標頭行";
        this["TableInsertRowDescription"] = "插入行";
        this["TableTotalsRowDescription"] = "總計行";
        this["Value_UCErrorCode_Div"] = "#DIV/0!";
        this["Value_UCErrorCode_Fail"] = "#FAIL";
        this["Value_UCErrorCode_NA"] = "#N/A";
        this["Value_UCErrorCode_Name"] = "#NAME?";
        this["Value_UCErrorCode_Null"] = "#NULL!";
        this["Value_UCErrorCode_Num"] = "#NUM!";
        this["Value_UCErrorCode_Ok"] = "#OK";
        this["Value_UCErrorCode_Reference"] = "#REF!";
        this["Value_UCErrorCode_Unknown"] = "#N/A";
        this["Value_UCErrorCode_Value"] = "#VALUE!";
        this["WorkbookColorInfo_Automatic_Description"] = "自動";
        this["WorkbookColorInfo_Description"] = "顏色資訊: {0}";
        this["WorkbookColorInfo_WithTint_Description"] = "顏色資訊: {0}，色調: {1}";
        this["WorksheetShapeSerialization_GroupName"] = "組";
        this["WorksheetShapeSerialization_ImageName"] = "圖片";
        this["WorksheetShapeSerialization_ShapeName"] = "形狀";
    }
}
