// igniteui-react-excel/esm2015/lib/ExcelLocaleNb.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
export class ExcelLocaleNb {
    constructor() {
        this["DefaultTotalLabel"] = "Total";
        this["Error_AnchoredReference"] = "Prøv å få tilgang til {0} som en referanse som ikke er avvist";
        this["Error_ExplicitThis"] = "{0} - Eksplisitt Dette støttes ikke";
        this["Error_ExternalRange"] = "Området inneholder ekstern referanse";
        this["Error_FormulaNotSupported"] = "{0} - Prøv å få tilgang til formel på en referanse som ikke støtter formler";
        this["Error_Internal"] = "Det oppstod en intern feil i metode {0}.";
        this["Error_InvalidFunction"] = "Ugyldig funksjon {0} på {1}.";
        this["Error_InvalidFunctionArgCountTooFew"] = "Funksjonen '{0}' referert til {1} inneholder ikke riktig antall argumenter. Funksjonen krever minst {2} parameter(er).";
        this["Error_InvalidFunctionArgCountTooMany"] = "Funksjonen '{0}' referert til {1} inneholder ikke riktig antall argumenter. Argumenttellingen bør ikke overstige {2} parameter(er).";
        this["Error_InvalidNumberSystemDigit"] = "Ugyldig siffer for '{0}' som finnes i verdien '{1}' for tallsystemet med basen på {2}.";
        this["Error_InvalidOperand"] = "Ugyldig operand";
        this["Error_InvalidOperatorArgCount"] = "'{0}'-funksjonen må godta {1} argumenter.";
        this["Error_InvalidReference"] = "Ugyldig eller utilgjengelig referanse";
        this["Error_Location"] = "linje {0} kol {1}";
        this["Error_MethodNotSupported"] = "{0} støtter ikke {1}.";
        this["Error_NoCurrentRangeElement"] = "Ingen nåværende områdeelement";
        this["Error_NullFunctionResult"] = "Evalueringen av '{0}' returnerte en nullverdi.";
        this["Error_NullOperand"] = "Null operand";
        this["Error_ParseRef"] = "Ukjent referanse på {0}.";
        this["Error_RangeFromLast"] = "fromT.Last ikke satt";
        this["Error_RangeFromNull"] = "RefRange: fra er null";
        this["Error_RangeFromRefBase"] = "RefRange: from er ikke en RefBase: {0}";
        this["Error_RangeLast"] = "RefRange Validate: last tuples [{0}] != [{1}]";
        this["Error_RangeNormalizeScope"] = "RefRange Normalize: dårlig omfang [{0}/{1}]";
        this["Error_RangeToNull"] = "RefRange: to er null";
        this["Error_RangeToRefBase"] = "RefRange: to er ikke en RefBase: {0}";
        this["Error_RangeTuple"] = "RefRange Validate: tuples [{0}] !=  [{1}]";
        this["Error_RangeValidateCount"] = "RefRange Validate: from count {0} != to count {1}";
        this["Error_ReadOnlyReference"] = "{0} - Prøv å angi verdi på referanse bare";
        this["Error_ReferenceNotEumerable"] = "{0} er ikke en talløs referanse";
        this["Error_RefMalformed"] = "{0} - misdannet referanse {1}";
        this["Error_RP_ContainsBase"] = "RefParser.Contains - Tom Base";
        this["Error_RP_ContainsTarget"] = "RefParser.Contains - Tom Target";
        this["Error_RP_EmptyElement"] = "Ugyldig referanse {0}. Tomt element på plass på {1}.";
        this["Error_RP_EmptyReference"] = "Ugyldig referanse {0}. Referansen er tom.";
        this["Error_RP_MalformedExtra"] = "RefParser misdannet referanse: Flere [{0}] funnet i [{1}]";
        this["Error_RP_MalformedUnexpected"] = "RefParser misdannet referanse: Uventet [{0}] funnet i [{1}]";
        this["Error_RP_MergeRelativeBase"] = "RefParser.Merge: Base er relativ";
        this["Error_RP_MissingCloseParenthesis"] = "Ugyldig referanse {0}. Mangler nær parentes.";
        this["Error_RP_MissingEndQuotes"] = "Ugyldig referanse {0}. Mangler sluttord.";
        this["Error_RT_Expected"] = "RefTuple: Forventet [{0}] i [{1}]";
        this["Error_RT_InvalidScope"] = "Ugyldig omfang: {0}.";
        this["Error_RT_InvalidScope_CharactersAfterClosingQuotes"] = "Tegn som du oppdaget etter å ha lukket doble anførselstegn i {0} omfanget på stedet {1}.";
        this["Error_RT_InvalidTuple_CharactersAfterScopeEnd"] = "Ugyldig element {0}. Tegn som du oppdaget etter endt omfang.";
        this["Error_RT_InvalidTuple_MissingCloseParenthesis"] = "Ugyldig element {0}. Mangler nær parentes.";
        this["Error_RT_InvalidTuple_MissingEndQuotes"] = "Ugyldig element {0}. Mangler sluttord.";
        this["Error_RT_InvalidTuple_NamePortionEmpty"] = "Ugyldig element {0}. Elementnavn delen er tom.";
        this["Error_RT_InvalidTuple_ScopeIndexLarge"] = "Ugyldig element {0}. Indeksen {1} er ugyldig: {2}.";
        this["Error_RT_InvalidTuple_UnescapedCharacter"] = "Et uutviklet {0} tegn på {1} på stedet {2}.";
        this["Error_RT_NullRefName"] = "RefTuple: null RefName";
        this["Error_RT_Unexpected"] = "RefTuple: uventet [{0}] i [{1}]";
        this["Error_ScopeAllNotLast"] = "{0} - Omfang Alle må vises i siste element {1}";
        this["Error_UCErrorCode_Div"] = "Del på null";
        this["Error_UCErrorCode_Fail"] = "Mislykkes";
        this["Error_UCErrorCode_NA"] = "Ikke aktuelt";
        this["Error_UCErrorCode_Name"] = "Ugyldig navn";
        this["Error_UCErrorCode_Null"] = "Nullverdi";
        this["Error_UCErrorCode_Num"] = "Ugyldige numeriske verdier i funksjonen eller formelen.";
        this["Error_UCErrorCode_Ok"] = "Lykkes";
        this["Error_UCErrorCode_Reference"] = "Ugyldig eller utilgjengelig referanse";
        this["Error_UCErrorCode_Unknown"] = "Ugyldig kode";
        this["Error_UCErrorCode_Value"] = "Feil type argument eller operand.";
        this["Error_UnexpectedScope"] = "{0} - Uventet omfang for foreldre {1}";
        this["Error_UnknownRange"] = "Ukjent områdetype";
        this["Error_UnknownRefType"] = "{0} - Ukjent RefType {1}";
        this["Error_UnknownScope"] = "{0} - Ukjent omfang {1}";
        this["Error_UnresolvedExternal"] = "Uavklart ekstern referanse";
        this["Func_abs_Arg_0"] = "number";
        this["Func_abs_ArgDesc_0"] = "Det virkelige antallet som skal få den absolutte verdien.";
        this["Func_abs_Category"] = "Math";
        this["Func_abs_CategoryURL"] = "Math";
        this["Func_abs_Desc"] = "Returnerer den absolutte verdien av et tall. Den absolutte verdien av et tall er tallet uten dets tegn.";
        this["Func_acos_Arg_0"] = "number";
        this["Func_acos_ArgDesc_0"] = "Kosinusen for vinkelen som skal hentes, som må være fra-1 til 1.";
        this["Func_acos_Category"] = "Math";
        this["Func_acos_CategoryURL"] = "Math";
        this["Func_acos_Desc"] = "Returnerer arccosine, eller invers cosinus, for et tall. Arccosine er vinkelen hvis cosinus er et spesifisert tall. Returvinkelen er gitt i radianer i området null til pi. For å konvertere resultatet fra radianer til grader, multipliser med 180/PI().";
        this["Func_acosh_Arg_0"] = "number";
        this["Func_acosh_ArgDesc_0"] = "Et reelt tall større enn 1 for å finne den inverse hyperboliske cosinus.";
        this["Func_acosh_Category"] = "Math";
        this["Func_acosh_CategoryURL"] = "Math";
        this["Func_acosh_Desc"] = "Returnerer den inverse hyperboliske cosinusen til det gitte tallet. Den inverse hyperboliske cosinus er vinkelen hvis hyperboliske cosinus er et spesifisert tall.";
        this["Func_and_Arg_0"] = "logical{0}";
        this["Func_and_ArgDesc_0"] = "En logisk verdi (sann eller usann).";
        this["Func_and_Category"] = "Logisk";
        this["Func_and_CategoryURL"] = "Logical";
        this["Func_and_Desc"] = "Returnerer true hvis alle verdier er sanne eller returnerer usanne hvis en eller flere verdier er usanne.";
        this["Func_asin_Arg_0"] = "number";
        this["Func_asin_ArgDesc_0"] = "Sinus for vinkelen å få, som må være fra -1 til 1 radianer.";
        this["Func_asin_Category"] = "Math";
        this["Func_asin_CategoryURL"] = "Math";
        this["Func_asin_Desc"] = "Returnerer buehule, eller invers sinus, for et tall. Bueskinnet er vinkelen hvis sinus er det angitte tallet. Returvinkelen er gitt i radianer i området-pi/2 til pi/2. For å uttrykke buehodet i grader, multipliser resultatet med 180/PI().";
        this["Func_asinh_Arg_0"] = "number";
        this["Func_asinh_ArgDesc_0"] = "Et reelt tall for å finne den inverse hyperbolske sinus.";
        this["Func_asinh_Category"] = "Math";
        this["Func_asinh_CategoryURL"] = "Math";
        this["Func_asinh_Desc"] = "Returnerer den inverse hyperbolske sinusen til det gitte tallet. Den omvendte hyperboliske sinus er vinkelen hvis hyperboliske sinus er et spesifisert tall.";
        this["Func_atan_Arg_0"] = "number";
        this["Func_atan_ArgDesc_0"] = "Tangenten til vinkelen å få.";
        this["Func_atan_Category"] = "Math";
        this["Func_atan_CategoryURL"] = "Math";
        this["Func_atan_Desc"] = "Returnerer den arktangente eller omvendte tangenten til et tall. Arktangenten er vinkelen hvis tangent er det angitte tallet. Returvinkelen er gitt i radianer i området-pi/2 til pi/2. For å uttrykke arktangenten i grader, multipliser resultatet med 180/PI ().";
        this["Func_atan2_Arg_0"] = "x";
        this["Func_atan2_Arg_1"] = "y";
        this["Func_atan2_ArgDesc_0"] = "X-koordinaten til punktet.";
        this["Func_atan2_ArgDesc_1"] = "Y-koordinaten til punktet.";
        this["Func_atan2_Category"] = "Math";
        this["Func_atan2_CategoryURL"] = "Math";
        this["Func_atan2_Desc"] = "Returnerer den arktangente eller inverse tangenten til de angitte x- og y-koordinatene. Arktangenten er vinkelen fra x-aksen til en linje som inneholder opprinnelsen (0, 0) og et punkt med koordinater (x_num, y_num). Vinkelen er gitt i radianer mellom-pi og pi, unntatt-pi.";
        this["Func_atanh_Arg_0"] = "number";
        this["Func_atanh_ArgDesc_0"] = "Et reelt tall mellom -1 og 1 for å få den inverse hyperboliske tangenten.";
        this["Func_atanh_Category"] = "Math";
        this["Func_atanh_CategoryURL"] = "Math";
        this["Func_atanh_Desc"] = "Returnerer den inverse hyperbolske tangenten til det gitte tallet. Den omvendte hyperboliske tangenten er vinkelen hvis hyperboliske tangens er et spesifisert tall.";
        this["Func_avedev_Arg_0"] = "number{0}";
        this["Func_avedev_ArgDesc_0"] = "Nummer1, nummer2, ... er en eller flere numeriske verdier eller referanser til numeriske verdier.";
        this["Func_avedev_Category"] = "Statistical";
        this["Func_avedev_CategoryURL"] = "Statistical";
        this["Func_avedev_Desc"] = "Returnerer gjennomsnittlig avvik fra gjennomsnittet av alle numeriske verdier.";
        this["Func_average_Arg_0"] = "number{0}";
        this["Func_average_ArgDesc_0"] = "Et tall eller tallområde for å finne gjennomsnittet for de andre argumentene.";
        this["Func_average_Category"] = "Statistisk";
        this["Func_average_CategoryURL"] = "Statistical";
        this["Func_average_Desc"] = "Returnerer gjennomsnittet (aritmetisk gjennomsnitt) av en eller flere verdier.";
        this["Func_bin2dec_Arg_0"] = "Nummer";
        this["Func_bin2dec_ArgDesc_0"] = "Et tekstuttrykk som representerer et binært tall.";
        this["Func_bin2dec_Category"] = "Engineering";
        this["Func_bin2dec_CategoryURL"] = "Engineering";
        this["Func_bin2dec_Desc"] = "Returnerer desimalverdien til en tekstrepresentasjon av et tall i det binære tallsystemet.";
        this["Func_bin2hex_Arg_0"] = "Nummer";
        this["Func_bin2hex_Arg_1"] = "Places";
        this["Func_bin2hex_ArgDesc_0"] = "Et tekstuttrykk som representerer et binært tall.";
        this["Func_bin2hex_ArgDesc_1"] = "Antall tegn som skal brukes for å representere verdien. Hvis dette ikke er spesifisert, brukes det maksimale antallet tegn som er nødvendig for å representere verdien. Dette lar deg angi ledende nuller.";
        this["Func_bin2hex_Category"] = "Engineering";
        this["Func_bin2hex_CategoryURL"] = "Engineering";
        this["Func_bin2hex_Desc"] = "Konverterer et tekstuttrykk som representerer et binært tall til et tekstuttrykk som representerer det samme tallet i det heksadesimale tallsystemet.";
        this["Func_bin2oct_Arg_0"] = "Nummer";
        this["Func_bin2oct_Arg_1"] = "Places";
        this["Func_bin2oct_ArgDesc_0"] = "Et tekstuttrykk som representerer et binært tall.";
        this["Func_bin2oct_ArgDesc_1"] = "Antall tegn som skal brukes for å representere verdien. Hvis dette ikke er spesifisert, brukes det maksimale antallet tegn som er nødvendig for å representere verdien. Dette lar deg angi ledende nuller.";
        this["Func_bin2oct_Category"] = "Engineering";
        this["Func_bin2oct_CategoryURL"] = "Engineering";
        this["Func_bin2oct_Desc"] = "Konverterer et tekstuttrykk som representerer et binært tall til et tekstuttrykk som representerer det samme tallet i det oktale tallsystemet.";
        this["Func_binomdist_Arg_0"] = "Nummer";
        this["Func_binomdist_Arg_1"] = "Forsøk";
        this["Func_binomdist_Arg_2"] = "Sannsynlighet";
        this["Func_binomdist_Arg_3"] = "Kumulativ";
        this["Func_binomdist_ArgDesc_0"] = "Antall suksesser i forsøk.";
        this["Func_binomdist_ArgDesc_1"] = "Antall uavhengige forsøk.";
        this["Func_binomdist_ArgDesc_2"] = "Sannsynligheten for suksess på hver prøve.";
        this["Func_binomdist_ArgDesc_3"] = "En logisk verdi som bestemmer formen på funksjonen. Hvis true, vil funksjonen returnere den kumulative fordelingen, som er sannsynligheten for at det er maksimalt 'antall' suksesser, ellers vil det returnere sannsynlighetsmassen som er sannsynligheten for at det er 'antall' suksesser.";
        this["Func_binomdist_Category"] = "Statistical";
        this["Func_binomdist_CategoryURL"] = "Statistical";
        this["Func_binomdist_Desc"] = "Returnerer antall mulige kombinasjoner (med repetisjoner) gitt et sett med elementer og et antall valgte elementer fra det settet.";
        this["Func_CategoryURL_Template"] = "ms-help://INFRAGISTICS_HELP/NetAdvantagev{2}{3}/Win/WinElements/WinCalcManager_{0}_Functions.html#FUNC_{1}";
        this["Func_ceiling_Arg_0"] = "number";
        this["Func_ceiling_Arg_1"] = "betydning";
        this["Func_ceiling_ArgDesc_0"] = "Tallet som skal rundes.";
        this["Func_ceiling_ArgDesc_1"] = "Multipelen som skal rundes til. Hvis den utelates, antas dette å være en.";
        this["Func_ceiling_Category"] = "Math";
        this["Func_ceiling_CategoryURL"] = "Math";
        this["Func_ceiling_Desc"] = "Returnerer det minste multiplumet av signifikansen (standard til 1) større enn eller lik det angitte tallet.";
        this["Func_char_Arg_0"] = "charCode";
        this["Func_char_ArgDesc_0"] = "Nummeret som skal konverteres til et tegn.";
        this["Func_char_Category"] = "TextAndData";
        this["Func_char_CategoryURL"] = "Text_and_Data";
        this["Func_char_Desc"] = "Returnerer tegnet som er angitt av kodenummeret fra tegnsettet for datamaskinen din.";
        this["Func_choose_Arg_0"] = "IndexNum";
        this["Func_choose_Arg_1"] = "Value{0}";
        this["Func_choose_ArgDesc_0"] = "Den 1-baserte indeksen i resten av argumentene for verdien som skal brukes.";
        this["Func_choose_ArgDesc_1"] = "En verdi som kan velges avhengig av IndexNum-verdien.";
        this["Func_choose_Category"] = "LookupAndReference";
        this["Func_choose_CategoryURL"] = "Lookup_And_Reference";
        this["Func_choose_Desc"] = "Returnerer en av verdiene i argumentene basert på tallet i det første argumentet.";
        this["Func_code_Arg_0"] = "tekst";
        this["Func_code_ArgDesc_0"] = "Tekstuttrykket som vil få sitt første tegn evaluert.";
        this["Func_code_Category"] = "TextAndData";
        this["Func_code_CategoryURL"] = "Text_and_Data";
        this["Func_code_Desc"] = "Returnerer en numerisk kode for det første tegnet i et tekstuttrykk ved hjelp av tegnsettet som brukes av datamaskinen.";
        this["Func_combin_Arg_0"] = "Nummer";
        this["Func_combin_Arg_1"] = "Antall valgt";
        this["Func_combin_ArgDesc_0"] = "Antall elementer.";
        this["Func_combin_ArgDesc_1"] = "Antall elementer valgt i hver kombinasjon.";
        this["Func_combin_Category"] = "Math";
        this["Func_combin_CategoryURL"] = "Math";
        this["Func_combin_Desc"] = "Returnerer antall mulige kombinasjoner gitt et sett med elementer og et antall valgte elementer fra det settet.";
        this["Func_combina_Arg_0"] = "Nummer";
        this["Func_combina_Arg_1"] = "Antall valgt";
        this["Func_combina_ArgDesc_0"] = "Antall elementer.";
        this["Func_combina_ArgDesc_1"] = "Antall elementer valgt i hver kombinasjon.";
        this["Func_combina_Category"] = "Math";
        this["Func_combina_CategoryURL"] = "Math";
        this["Func_combina_Desc"] = "Returnerer antall mulige kombinasjoner (med repetisjoner) gitt et sett med elementer og et antall valgte elementer fra det settet.";
        this["Func_complex_Arg_0"] = "RealNumber";
        this["Func_complex_Arg_1"] = "ImaginaryNumber";
        this["Func_complex_Arg_2"] = "Suffiks";
        this["Func_complex_ArgDesc_0"] = "Den virkelige koeffisienten for det komplekse tallet.";
        this["Func_complex_ArgDesc_1"] = "Den imaginære koeffisienten for tallet.";
        this["Func_complex_ArgDesc_2"] = "Spesifiserer suffikset som skal brukes. De akseptable verdiene er \"i\" eller \"j\". Hvis jeg utelater det, brukes \"i\". Vær oppmerksom på at suffikset er store og små bokstaver; store bokstaver \"I\" og \"J\" er ikke tillatt og vil resultere i en #VALUE-feil.";
        this["Func_complex_Category"] = "Engineering";
        this["Func_complex_CategoryURL"] = "Engineering";
        this["Func_complex_Desc"] = "Returnerer et komplekst tall representert som et tekstuttrykk i formatet \"x + yi\" eller \"x + yj\" ved å kombinere et reelt og imaginært tall.";
        this["Func_concatenate_Arg_0"] = "text{0}";
        this["Func_concatenate_ArgDesc_0"] = "Et tekstuttrykk eller en rekke tekstuttrykk som skal settes sammen i et enkelt tekstuttrykk.";
        this["Func_concatenate_Category"] = "TextAndData";
        this["Func_concatenate_CategoryURL"] = "Text_and_Data";
        this["Func_concatenate_Desc"] = "Slår sammen alle argumentene i ett tekstuttrykk.";
        this["Func_convert_Arg_0"] = "Nummer";
        this["Func_convert_Arg_1"] = "FromUnit";
        this["Func_convert_Arg_2"] = "ToUnit";
        this["Func_convert_ArgDesc_0"] = "Verdien som skal konverteres.";
        this["Func_convert_ArgDesc_1"] = "Enheten der tallet er oppgitt.\r\n\r\nVekt og masseenheter: \"g\" = gram; \"sg\" = slug; \"lbm\" = pundmasse (avoirdupois); \"u\" = U (atommassenhet); \r\n\"ozm\" = Unse masse (avoirdupois)Avstandsenheter: \"m\" = Meter; \"mi\" = Statuttmile; \"Nmi\" = Nautisk mil; \"in\" = Tommer; \"ft\" = Fot; \"yd\" = Yard; \"ang\" = Angstrom; \"Pica\" = Pica (1/72 tommer)\r\nTidenheter: \"yr\" = År; \"dag\" = Dag; \"hr\" = Time; \"mn\" = minutt; \"sec\" = Andre\r\nTrykk enheter: \"Pa\" (eller \"p\") = Pascal; \"atm\" (eller \"at\") = Atmosfære; \"mmHg \"= mm kvikksølv\r\nTving enheter: \"N\" = Newton; \"dyn\" (eller \"dy\") = Dyne; \"lbf \"= Pundkraft\r\nEnergienheter: \"J\" = Joule; \"e\" = Erg; \"c\" = Termodynamisk kalori; \"cal\" = IT-kalori; \"eV\" (eller \"ev\") = Elektronvolt; \"HPh\" (eller \"hh\") = Hestekrefter; \"Wh\" (eller \"wh\") = Watt-time; \"flb \"= Fotpund; \"BTU\" (eller \"btu\") = BTU\r\nKraftenheter: \"HP\" (eller \"h\") = Hestekrefter; \"W\" (eller \"w\") = Watt\r\nMagentism-enheter: \"T\"= Tesla; \"ga \"= Gauss\r\nTemperaturenheter: \"C\" (eller \"cel\") = Grad Celsius; \"F\" (eller \"fah\") = Grad Fahrenheit; \"K\" (eller \"kel\") = Kelvin\r\nFlytende måleenheter: \"ts\" = Teskje; \"tbs\" = Teskje; \"oz\" = flytende unse; \"cup\" = Cup; \"pt\" (eller \"us_pt\") = amerikansk halvliter; \"uk_pt\" = britisk halvliter; \"qt\" = Quart; \"gal\" = gallon; \"l\" (eller \"lt\") = Liter\r\n\r\nEn hvilken som helst metrisk enhet kan ha et av følgende prefiks: exa (\"E\") = 1E+18; peta (\"P\") = 1E+15; tera (\"T\") = 1E+12; giga (\"G\") = 1E+09; mega (\"M\") = 1E+06; kilo (\"k\") = 1E+03; hekto (\"h\") = 1E+02; dekao (\"e\") = 1E+01; deci (\"d\") = 1E-01; centi (\"c\") = 1E-02; milli (\"m\") = 1E-03; mikro (\"u\") = 1E-06; nano (\"n\") = 1E-09; pico (\"p\") = 1E-12; femto (\"f\") = 1E-15; atto (\"a\") = 1E-18\r\n    ";
        this["Func_convert_ArgDesc_2"] = "Enheten der tallet er oppgitt.\r\n\r\nVekt og masseenheter: \"g\" = gram; \"sg\" = slug; \"lbm\" = pundmasse (avoirdupois); \"u\" = U (atommassenhet); \r\n\"ozm\" = Unse masse (avoirdupois)Avstandsenheter: \"m\" = Meter; \"mi\" = Statuttmile; \"Nmi\" = Nautisk mil; \"in\" = Tommer; \"ft\" = Fot; \"yd\" = Yard; \"ang\" = Angstrom; \"Pica\" = Pica (1/72 tommer)\r\nTidenheter: \"yr\" = År; \"dag\" = Dag; \"hr\" = Time; \"mn\" = minutt; \"sec\" = Andre\r\nTrykk enheter: \"Pa\" (eller \"p\") = Pascal; \"atm\" (eller \"at\") = Atmosfære; \"mmHg \"= mm kvikksølv\r\nTving enheter: \"N\" = Newton; \"dyn\" (eller \"dy\") = Dyne; \"lbf \"= Pundkraft\r\nEnergienheter: \"J\" = Joule; \"e\" = Erg; \"c\" = Termodynamisk kalori; \"cal\" = IT-kalori; \"eV\" (eller \"ev\") = Elektronvolt; \"HPh\" (eller \"hh\") = Hestekrefter; \"Wh\" (eller \"wh\") = Watt-time; \"flb \"= Fotpund; \"BTU\" (eller \"btu\") = BTU\r\nKraftenheter: \"HP\" (eller \"h\") = Hestekrefter; \"W\" (eller \"w\") = Watt\r\nMagentism-enheter: \"T\"= Tesla; \"ga \"= Gauss\r\nTemperaturenheter: \"C\" (eller \"cel\") = Grad Celsius; \"F\" (eller \"fah\") = Grad Fahrenheit; \"K\" (eller \"kel\") = Kelvin\r\nFlytende måleenheter: \"ts\" = Teskje; \"tbs\" = Teskje; \"oz\" = flytende unse; \"cup\" = Cup; \"pt\" (eller \"us_pt\") = amerikansk halvliter; \"uk_pt\" = britisk halvliter; \"qt\" = Quart; \"gal\" = gallon; \"l\" (eller \"lt\") = Liter\r\n\r\nEn hvilken som helst metrisk enhet kan ha et av følgende prefiks: exa (\"E\") = 1E+18; peta (\"P\") = 1E+15; tera (\"T\") = 1E+12; giga (\"G\") = 1E+09; mega (\"M\") = 1E+06; kilo (\"k\") = 1E+03; hekto (\"h\") = 1E+02; dekao (\"e\") = 1E+01; deci (\"d\") = 1E-01; centi (\"c\") = 1E-02; milli (\"m\") = 1E-03; mikro (\"u\") = 1E-06; nano (\"n\") = 1E-09; pico (\"p\") = 1E-12; femto (\"f\") = 1E-15; atto (\"a\") = 1E-18\r\n    ";
        this["Func_convert_Category"] = "Engineering";
        this["Func_convert_CategoryURL"] = "Engineering";
        this["Func_convert_Desc"] = "Konverterer en verdi fra et målesystem til et annet, for eksempel meter til tommer eller timer til sekunder.";
        this["Func_cos_Arg_0"] = "number";
        this["Func_cos_ArgDesc_0"] = "Vinkelen i radianer for å få cosinus.";
        this["Func_cos_Category"] = "Math";
        this["Func_cos_CategoryURL"] = "Math";
        this["Func_cos_Desc"] = "Returnerer cosinus for den gitte vinkelen.";
        this["Func_cosh_Arg_0"] = "number";
        this["Func_cosh_ArgDesc_0"] = "Et reelt tall for å finne den hyperboliske cosinus.";
        this["Func_cosh_Category"] = "Math";
        this["Func_cosh_CategoryURL"] = "Math";
        this["Func_cosh_Desc"] = "Returnerer hyperbolsk cosinus for et tall.";
        this["Func_count_Arg_0"] = "value{0}";
        this["Func_count_ArgDesc_0"] = "En referanse eller referanseområde som kan inneholde tall.";
        this["Func_count_Category"] = "Statistical";
        this["Func_count_CategoryURL"] = "Statistical";
        this["Func_count_Desc"] = "Teller antall referanser som inneholder tall.";
        this["Func_counta_Arg_0"] = "value{0}";
        this["Func_counta_ArgDesc_0"] = "En referanse eller referanseområde som kan inneholde tall.";
        this["Func_counta_Category"] = "Statistical";
        this["Func_counta_CategoryURL"] = "Statistical";
        this["Func_counta_Desc"] = "Teller antall referanser som ikke er tomme.";
        this["Func_critbinom_Arg_0"] = "Forsøk";
        this["Func_critbinom_Arg_1"] = "Sannsynlighet";
        this["Func_critbinom_Arg_2"] = "Alpha";
        this["Func_critbinom_ArgDesc_0"] = "Antall uavhengige forsøk.";
        this["Func_critbinom_ArgDesc_1"] = "Sannsynligheten for suksess på hver prøve.";
        this["Func_critbinom_ArgDesc_2"] = "Kriterieverdien.";
        this["Func_critbinom_Category"] = "Statistical";
        this["Func_critbinom_CategoryURL"] = "Statistical";
        this["Func_critbinom_Desc"] = "Returnerer den minste verdien som den kumulative binomialfordelingen er større enn eller lik en kriterieverdi for.";
        this["Func_date_Arg_0"] = "år";
        this["Func_date_Arg_1"] = "måned";
        this["Func_date_Arg_2"] = "dag";
        this["Func_date_ArgDesc_0"] = "Et tall som representerer året.";
        this["Func_date_ArgDesc_1"] = "Et tall som representerer årets måned.";
        this["Func_date_ArgDesc_2"] = "Et tall som representerer dagen i måneden.";
        this["Func_date_Category"] = "DateAndTime";
        this["Func_date_CategoryURL"] = "Date_and_Time";
        this["Func_date_Desc"] = "Returnerer det sekvensielle serienummeret som representerer en bestemt dato.";
        this["Func_dateadd_Arg_0"] = "interval";
        this["Func_dateadd_Arg_1"] = "number";
        this["Func_dateadd_Arg_2"] = "date";
        this["Func_dateadd_ArgDesc_0"] = "Et tekstuttrykk som representerer tidsintervallet som skal legges til.\r\nd = dag\r\ny = år\r\nh = Time\r\nn = minutt\r\nm = Måned\r\nq = Kvartal\r\ns = Andre\r\nw = Uke\r\nww = Kalenderuke\r\nåååå = år\r\n";
        this["Func_dateadd_ArgDesc_1"] = "Antall intervaller som skal legges til. Denne verdien kan være positiv (for å få en dato i fremtiden) eller negativ (for å få en dato tidligere). Den kan inneholde en brøkdel når Intervall angir timer, minutter eller sekunder. For andre verdier av intervall ignoreres enhver brøkdel av nummer.";
        this["Func_dateadd_ArgDesc_2"] = "Datoen som intervallet skal legges til.";
        this["Func_dateadd_Category"] = "DateAndTime";
        this["Func_dateadd_CategoryURL"] = "Date_and_Time";
        this["Func_dateadd_Desc"] = "Returnerer en dato som det angitte intervallet er lagt til.";
        this["Func_datediff_Arg_0"] = "interval";
        this["Func_datediff_Arg_1"] = "date1";
        this["Func_datediff_Arg_2"] = "date2";
        this["Func_datediff_ArgDesc_0"] = "Et tekstuttrykk som representerer tidsintervallet som skal brukes som differansenhet mellom dato1 og dato2.\r\nd = dag\r\ny = år\r\nh = Time\r\nn = minutt\r\nm = Måned\r\nq = Kvartal\r\ns = Andre\r\nw = Uke\r\nww = Kalenderuke\r\nåååå = år\r\n";
        this["Func_datediff_ArgDesc_1"] = "Den første datoen som skal brukes i differensberegningen.";
        this["Func_datediff_ArgDesc_2"] = "Den andre datoen som skal brukes i differensberegningen.";
        this["Func_datediff_Category"] = "DateAndTime";
        this["Func_datediff_CategoryURL"] = "Date_and_Time";
        this["Func_datediff_Desc"] = "Returnerer en verdi som angir antall intervaller mellom to DateTime-verdier.";
        this["Func_datevalue_Arg_0"] = "date_text";
        this["Func_datevalue_ArgDesc_0"] = "Et tekstuttrykk som representerer en dato.";
        this["Func_datevalue_Category"] = "DateAndTime";
        this["Func_datevalue_CategoryURL"] = "Date_and_Time";
        this["Func_datevalue_Desc"] = "Returnerer datoverdien for den angitte datoteksten.";
        this["Func_day_Arg_0"] = "Date_value";
        this["Func_day_ArgDesc_0"] = "En datoverdi som dagen skal hentes fra.";
        this["Func_day_Category"] = "DateAndTime";
        this["Func_day_CategoryURL"] = "Date_and_Time";
        this["Func_day_Desc"] = "Returnerer dagen i måneden for en datoverdi.";
        this["Func_days360_Arg_0"] = "start_date";
        this["Func_days360_Arg_1"] = "end_date";
        this["Func_days360_Arg_2"] = "method";
        this["Func_days360_ArgDesc_0"] = "Startdato for dager. Hvis dette er etter sluttdatoen, vil funksjonen returnere et negativt tall.";
        this["Func_days360_ArgDesc_1"] = "Sluttdato for dager. Hvis dette er før startdatoen, vil funksjonen returnere et negativt tall.";
        this["Func_days360_ArgDesc_2"] = "En logisk verdi som angir om den amerikanske (falske) eller europeiske (sanne) metoden skal brukes i beregningen.";
        this["Func_days360_Category"] = "DateAndTime";
        this["Func_days360_CategoryURL"] = "Date_and_Time";
        this["Func_days360_Desc"] = "Returnerer antall dager mellom to datoer basert på et 360-dagers år.";
        this["Func_db_Arg_0"] = "koste";
        this["Func_db_Arg_1"] = "berging";
        this["Func_db_Arg_2"] = "liv";
        this["Func_db_Arg_3"] = "periode";
        this["Func_db_Arg_4"] = "måned";
        this["Func_db_ArgDesc_0"] = "Startkostnaden for eiendelen.";
        this["Func_db_ArgDesc_1"] = "Verdien ved slutten av avskrivningen (noen ganger kalt eiendelens bergingsverdi).";
        this["Func_db_ArgDesc_2"] = "Antall perioder som eiendelen avskrives over (noen ganger kalt eiendelens levetid).";
        this["Func_db_ArgDesc_3"] = "Perioden for avskrivning. Perioden må bruke de samme enhetene som livet.";
        this["Func_db_ArgDesc_4"] = "Antall måneder det første året. Hvis måneden er utelatt, antas det å være 12.";
        this["Func_db_Category"] = "Financial";
        this["Func_db_CategoryURL"] = "Financial";
        this["Func_db_Desc"] = "Returnerer avskrivningen på en eiendel for en bestemt periode ved bruk av metoden med fast fallende saldo.";
        this["Func_dbnull_Category"] = "Information";
        this["Func_dbnull_CategoryURL"] = "Information";
        this["Func_dbnull_Desc"] = "Returnerer noe som representerer en ikke-eksisterende verdi.";
        this["Func_ddb_Arg_0"] = "koste";
        this["Func_ddb_Arg_1"] = "berging";
        this["Func_ddb_Arg_2"] = "liv";
        this["Func_ddb_Arg_3"] = "periode";
        this["Func_ddb_Arg_4"] = "faktor";
        this["Func_ddb_ArgDesc_0"] = "Startkostnaden for eiendelen.";
        this["Func_ddb_ArgDesc_1"] = "Verdien ved slutten av avskrivningen (noen ganger kalt eiendelens bergingsverdi).";
        this["Func_ddb_ArgDesc_2"] = "Antall perioder som eiendelen avskrives over (noen ganger kalt eiendelens levetid).";
        this["Func_ddb_ArgDesc_3"] = "Perioden for avskrivning. Perioden må bruke de samme enhetene som livet.";
        this["Func_ddb_ArgDesc_4"] = "Hastigheten balansen synker med. Hvis faktor utelates, antas det å være 2 (metoden med dobbelt fallende saldo).";
        this["Func_ddb_Category"] = "Financial";
        this["Func_ddb_CategoryURL"] = "Financial";
        this["Func_ddb_Desc"] = "Returnerer avskrivningen på en eiendel for en angitt periode ved bruk av dobbelt-fallende saldo-metode eller en annen spesifisert metode.";
        this["Func_dec2bin_Arg_0"] = "Nummer";
        this["Func_dec2bin_Arg_1"] = "Places";
        this["Func_dec2bin_ArgDesc_0"] = "Desimalverdien som skal konverteres.";
        this["Func_dec2bin_ArgDesc_1"] = "Antall tegn som skal brukes for å representere verdien. Hvis steder ikke er spesifisert, brukes det maksimale antallet tegn som er nødvendig for å representere verdien. Dette lar deg angi ledende nuller.";
        this["Func_dec2bin_Category"] = "Engineering";
        this["Func_dec2bin_CategoryURL"] = "Engineering";
        this["Func_dec2bin_Desc"] = "Returnerer et tekstuttrykk som representerer den angitte desimalverdien som et binært tall.";
        this["Func_dec2hex_Arg_0"] = "Nummer";
        this["Func_dec2hex_Arg_1"] = "Places";
        this["Func_dec2hex_ArgDesc_0"] = "Desimalverdien som skal konverteres.";
        this["Func_dec2hex_ArgDesc_1"] = "Antall tegn som skal brukes for å representere verdien. Hvis steder ikke er spesifisert, brukes det maksimale antallet tegn som er nødvendig for å representere verdien. Dette lar deg angi ledende nuller.";
        this["Func_dec2hex_Category"] = "Engineering";
        this["Func_dec2hex_CategoryURL"] = "Engineering";
        this["Func_dec2hex_Desc"] = "Returnerer et tekstuttrykk som representerer den angitte desimalverdien som et heksadesimalt tall.";
        this["Func_dec2oct_Arg_0"] = "Nummer";
        this["Func_dec2oct_Arg_1"] = "Places";
        this["Func_dec2oct_ArgDesc_0"] = "Desimalverdien som skal konverteres.";
        this["Func_dec2oct_ArgDesc_1"] = "Antall tegn som skal brukes for å representere verdien. Hvis steder ikke er spesifisert, brukes det maksimale antallet tegn som er nødvendig for å representere verdien. Dette lar deg angi ledende nuller.";
        this["Func_dec2oct_Category"] = "Engineering";
        this["Func_dec2oct_CategoryURL"] = "Engineering";
        this["Func_dec2oct_Desc"] = "Returnerer et tekstuttrykk som representerer den angitte desimalverdien som et oktaltall.";
        this["Func_degrees_Arg_0"] = "Radianer";
        this["Func_degrees_ArgDesc_0"] = "En verdi i radianer som vil bli konvertert til grader.";
        this["Func_degrees_Category"] = "Math";
        this["Func_degrees_CategoryURL"] = "Math";
        this["Func_degrees_Desc"] = "Konverterer radianer til grader.";
        this["Func_delta_Arg_0"] = "Nummer 1";
        this["Func_delta_Arg_1"] = "Nummer 2";
        this["Func_delta_ArgDesc_0"] = "Det første tallet å sammenligne.";
        this["Func_delta_ArgDesc_1"] = "Det andre tallet å sammenligne. Hvis det ikke er spesifisert, antas det å være null.";
        this["Func_delta_Category"] = "Engineering";
        this["Func_delta_CategoryURL"] = "Engineering";
        this["Func_delta_Desc"] = "Sammenligner to tall og returnerer ett hvis de er like eller null hvis de ikke er det.";
        this["Func_dollarde_Arg_0"] = "FractionDollarAmount";
        this["Func_dollarde_Arg_1"] = "Brøkdel";
        this["Func_dollarde_ArgDesc_0"] = "Dollarbeløpet uttrykt som en brøkdel.";
        this["Func_dollarde_ArgDesc_1"] = "Verdien som brukes som nevner for brøkdelen. Hvis denne verdien ikke er et heltall, blir den avkortet.";
        this["Func_dollarde_Category"] = "Financial";
        this["Func_dollarde_CategoryURL"] = "Financial";
        this["Func_dollarde_Desc"] = "Konverterer et dollarbeløp uttrykt som en brøkdel til et dollarbeløp uttrykt som en desimal.";
        this["Func_dollarfr_Arg_0"] = "DesimalDollarAmount";
        this["Func_dollarfr_Arg_1"] = "Brøkdel";
        this["Func_dollarfr_ArgDesc_0"] = "Dollarbeløpet uttrykt som en desimal.";
        this["Func_dollarfr_ArgDesc_1"] = "Verdien som brukes som nevner for brøkdelen. Hvis denne verdien ikke er et heltall, blir den avkortet.";
        this["Func_dollarfr_Category"] = "Financial";
        this["Func_dollarfr_CategoryURL"] = "Financial";
        this["Func_dollarfr_Desc"] = "Konverterer et dollarbeløp uttrykt som en desimal til et dollarbeløp uttrykt som en brøkdel.";
        this["Func_edate_Arg_0"] = "date";
        this["Func_edate_Arg_1"] = "numberOfMonths";
        this["Func_edate_ArgDesc_0"] = "Startdatoen.";
        this["Func_edate_ArgDesc_1"] = "Antall måneder for å endre datoen. Et positivt tall betyr en dato etter startdatoen. Et negativt tall indikerer en dato før startdatoen.";
        this["Func_edate_Category"] = "DateAndTime";
        this["Func_edate_CategoryURL"] = "DateAndTime";
        this["Func_edate_Desc"] = "Returnerer en dato som er det angitte antallet måneder før eller etter startdatoen.";
        this["Func_eomonth_Arg_0"] = "date";
        this["Func_eomonth_Arg_1"] = "numberOfMonths";
        this["Func_eomonth_ArgDesc_0"] = "Startdatoen.";
        this["Func_eomonth_ArgDesc_1"] = "Antall måneder for å endre datoen. Et positivt tall betyr en dato etter startdatoen. Et negativt tall indikerer en dato før startdatoen.";
        this["Func_eomonth_Category"] = "DateAndTime";
        this["Func_eomonth_CategoryURL"] = "DateAndTime";
        this["Func_eomonth_Desc"] = "Returnerer en dato som er på slutten av måneden som er det angitte antallet måneder før eller etter startdatoen.";
        this["Func_errortype_Arg_0"] = "error_value";
        this["Func_errortype_ArgDesc_0"] = "Feilverdien hvis identifikasjonsnummer skal finnes.";
        this["Func_errortype_Category"] = "Information";
        this["Func_errortype_CategoryURL"] = "Information";
        this["Func_errortype_Desc"] = "Returnerer et tall som tilsvarer en feilverdi.";
        this["Func_even_Arg_0"] = "number";
        this["Func_even_ArgDesc_0"] = "Verdien som skal rundes.";
        this["Func_even_Category"] = "Math";
        this["Func_even_CategoryURL"] = "Math";
        this["Func_even_Desc"] = "Avrunder et positivt tall opp og et negativt tall ned til nærmeste partall heltall.";
        this["Func_exp_Arg_0"] = "number";
        this["Func_exp_ArgDesc_0"] = "Eksponenten søkte å øke e.";
        this["Func_exp_Category"] = "Math";
        this["Func_exp_CategoryURL"] = "Math";
        this["Func_exp_Desc"] = "Returnerer e hevet til effekten av det angitte nummeret. Konstanten e tilsvarer 2,71828182845904.";
        this["Func_fact_Arg_0"] = "Nummer";
        this["Func_fact_ArgDesc_0"] = "Et positivt tall som faktoren skal beregnes for.";
        this["Func_fact_Category"] = "Math";
        this["Func_fact_CategoryURL"] = "Math";
        this["Func_fact_Desc"] = "Returnerer faktoren til et tall.";
        this["Func_factdouble_Arg_0"] = "Nummer";
        this["Func_factdouble_ArgDesc_0"] = "Et positivt tall som dobbeltfaktoriet skal beregnes for.";
        this["Func_factdouble_Category"] = "Math";
        this["Func_factdouble_CategoryURL"] = "Math";
        this["Func_factdouble_Desc"] = "Returnerer den dobbelte faktoren til et tall.";
        this["Func_false_Category"] = "Logical";
        this["Func_false_CategoryURL"] = "Logical";
        this["Func_false_Desc"] = "Returnerer den logiske verdien false.";
        this["Func_find_Arg_0"] = "find_text";
        this["Func_find_Arg_1"] = "within_text";
        this["Func_find_Arg_2"] = "start_num";
        this["Func_find_ArgDesc_0"] = "Teksten å finne.";
        this["Func_find_ArgDesc_1"] = "Teksten som inneholder teksten du vil finne.";
        this["Func_find_ArgDesc_2"] = "Det 1-baserte indekstegnet for å starte søket. Hvis den utelates, antas det å være en.";
        this["Func_find_Category"] = "TextAndData";
        this["Func_find_CategoryURL"] = "Text_and_Data";
        this["Func_find_Desc"] = "Returnerer utgangsposisjonen til ett tekstuttrykk i et annet, med store og små bokstaver.";
        this["Func_fixed_Arg_0"] = "number";
        this["Func_fixed_Arg_1"] = "desimaler";
        this["Func_fixed_Arg_2"] = "no_commas";
        this["Func_fixed_ArgDesc_0"] = "Tallet som skal rundes og konverteres til tekst.";
        this["Func_fixed_ArgDesc_1"] = "Antall sifre til høyre for desimaltegnet i returverdien. Hvis denne verdien er negativ, avrundes 'number' til venstre for desimaltegnet. Standardverdien er 2.";
        this["Func_fixed_ArgDesc_2"] = "Sann hvis kommaer ikke skal være i returverdien. Hvis den utelates, antas den å være usann.";
        this["Func_fixed_Category"] = "TextAndData";
        this["Func_fixed_CategoryURL"] = "Text_and_Data";
        this["Func_fixed_Desc"] = "Avrunder et tall til det angitte antallet desimaler og returnerer resultatet som tekst med eller uten komma.";
        this["Func_floor_Arg_0"] = "number";
        this["Func_floor_Arg_1"] = "betydning";
        this["Func_floor_ArgDesc_0"] = "Tallet som skal rundes.";
        this["Func_floor_ArgDesc_1"] = "Multipelen som skal rundes til. Hvis den utelates, antas dette å være en.";
        this["Func_floor_Category"] = "Math";
        this["Func_floor_CategoryURL"] = "Math";
        this["Func_floor_Desc"] = "Returnerer det største multiplumet av signifikansen (standard til 1) mindre enn eller lik det gitte tallet.";
        this["Func_fv_Arg_0"] = "vurdere";
        this["Func_fv_Arg_1"] = "nper";
        this["Func_fv_Arg_2"] = "pmt";
        this["Func_fv_Arg_3"] = "pv";
        this["Func_fv_Arg_4"] = "type";
        this["Func_fv_ArgDesc_0"] = "Renten per periode.";
        this["Func_fv_ArgDesc_1"] = "Det totale antallet betalingsperioder i en livrente.";
        this["Func_fv_ArgDesc_2"] = "Betalingen foretok hver periode, som ikke kan endres i løpet av livrenten.";
        this["Func_fv_ArgDesc_3"] = "Nåverdien, eller engangsbeløpet som en rekke fremtidige betalinger er verdt akkurat nå. Hvis den er utelatt, antas den å være null.";
        this["Func_fv_ArgDesc_4"] = "Tallet null eller ett som angir om betalinger forfaller ved slutten av perioden (null, standardverdien) eller begynnelsen (en).";
        this["Func_fv_Category"] = "Financial";
        this["Func_fv_CategoryURL"] = "Financial";
        this["Func_fv_Desc"] = "Returnerer den fremtidige verdien av en investering basert på periodisk.";
        this["Func_gcd_Arg_0"] = "number{0}";
        this["Func_gcd_ArgDesc_0"] = "En heltallsverdi. En verdi som ikke er heltall, blir avkortet.";
        this["Func_gcd_Category"] = "Math";
        this["Func_gcd_CategoryURL"] = "Math";
        this["Func_gcd_Desc"] = "Returnerer den største fellesdeleren til en eller flere heltallsverdier.";
        this["Func_gestep_Arg_0"] = "Nummer 1";
        this["Func_gestep_Arg_1"] = "Steg";
        this["Func_gestep_ArgDesc_0"] = "Det første tallet å sammenligne.";
        this["Func_gestep_ArgDesc_1"] = "Det andre tallet å sammenligne. Hvis det ikke er spesifisert, antas det å være null.";
        this["Func_gestep_Category"] = "Engineering";
        this["Func_gestep_CategoryURL"] = "Engineering";
        this["Func_gestep_Desc"] = "Sammenligner to tall og returnerer ett hvis det første tallet er større enn eller lik det andre eller returnerer null hvis ikke.";
        this["Func_hex2bin_Arg_0"] = "Nummer";
        this["Func_hex2bin_Arg_1"] = "Places";
        this["Func_hex2bin_ArgDesc_0"] = "Et tekstuttrykk som representerer et heksadesimalt tall.";
        this["Func_hex2bin_ArgDesc_1"] = "Antall tegn som skal brukes for å representere verdien. Hvis dette ikke er spesifisert, brukes det maksimale antallet tegn som er nødvendig for å representere verdien. Dette lar deg angi ledende nuller.";
        this["Func_hex2bin_Category"] = "Engineering";
        this["Func_hex2bin_CategoryURL"] = "Engineering";
        this["Func_hex2bin_Desc"] = "Konverterer et tekstuttrykk som representerer et heksadesimalt tall til et tekstuttrykk som representerer det samme tallet i det binære tallsystemet.";
        this["Func_hex2dec_Arg_0"] = "Nummer";
        this["Func_hex2dec_ArgDesc_0"] = "Et tekstuttrykk som representerer et heksadesimalt tall.";
        this["Func_hex2dec_Category"] = "Engineering";
        this["Func_hex2dec_CategoryURL"] = "Engineering";
        this["Func_hex2dec_Desc"] = "Returnerer desimalverdien til en tekstrepresentasjon av et tall i det heksadesimale tallsystemet.";
        this["Func_hex2oct_Arg_0"] = "Nummer";
        this["Func_hex2oct_Arg_1"] = "Places";
        this["Func_hex2oct_ArgDesc_0"] = "Et tekstuttrykk som representerer et heksadesimalt tall.";
        this["Func_hex2oct_ArgDesc_1"] = "Antall tegn som skal brukes for å representere verdien. Hvis dette ikke er spesifisert, brukes det maksimale antallet tegn som er nødvendig for å representere verdien. Dette lar deg angi ledende nuller.";
        this["Func_hex2oct_Category"] = "Engineering";
        this["Func_hex2oct_CategoryURL"] = "Engineering";
        this["Func_hex2oct_Desc"] = "Konverterer et tekstuttrykk som representerer et heksadesimalt tall til et tekstuttrykk som representerer det samme tallet i det oktale tallsystemet.";
        this["Func_hour_Arg_0"] = "Time_value";
        this["Func_hour_ArgDesc_0"] = "En tidsverdi for å få timen.";
        this["Func_hour_Category"] = "DateAndTime";
        this["Func_hour_CategoryURL"] = "Date_and_Time";
        this["Func_hour_Desc"] = "Returnerer timen til en tidsverdi.";
        this["Func_if_Arg_0"] = "logical_test";
        this["Func_if_Arg_1"] = "value_if_true";
        this["Func_if_Arg_2"] = "value_if_false";
        this["Func_if_ArgDesc_0"] = "Betingelsen for å teste.";
        this["Func_if_ArgDesc_1"] = "Resultatet hvis tilstanden evalueres til sann.";
        this["Func_if_ArgDesc_2"] = "Resultatet hvis tilstanden evalueres til falsk. Hvis den utelates, antas det å være den logiske verdien false.";
        this["Func_if_Category"] = "Logical";
        this["Func_if_CategoryURL"] = "Logical";
        this["Func_if_Desc"] = "Returnerer én verdi hvis den angitte tilstanden evalueres til sann og en annen verdi hvis den evalueres til usann.";
        this["Func_iferror_Arg_0"] = "verdi";
        this["Func_iferror_Arg_1"] = "value_if_error";
        this["Func_iferror_ArgDesc_0"] = "Verdien som skal brukes når det ikke er en feil.";
        this["Func_iferror_ArgDesc_1"] = "Verdien som skal brukes hvis verdi-argumentet er en feil.";
        this["Func_iferror_Category"] = "Logical";
        this["Func_iferror_CategoryURL"] = "Logical";
        this["Func_iferror_Desc"] = "Returnerer en av to verdier avhengig av om den første verdien er en feil eller ikke.";
        this["Func_imabs_Arg_0"] = "ComplexNumber";
        this["Func_imabs_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imabs_Category"] = "Engineering";
        this["Func_imabs_CategoryURL"] = "Engineering";
        this["Func_imabs_Desc"] = "Returnerer den absolutte verdien av et komplekst tall.";
        this["Func_imaginary_Arg_0"] = "ComplexNumber";
        this["Func_imaginary_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imaginary_Category"] = "Engineering";
        this["Func_imaginary_CategoryURL"] = "Engineering";
        this["Func_imaginary_Desc"] = "Returnerer den imaginære koeffisienten for et komplekst tall.";
        this["Func_imargument_Arg_0"] = "ComplexNumber";
        this["Func_imargument_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imargument_Category"] = "Engineering";
        this["Func_imargument_CategoryURL"] = "Engineering";
        this["Func_imargument_Desc"] = "Returnerer argumentet theta, en vinkel uttrykt i radianer.";
        this["Func_imconjugate_Arg_0"] = "ComplexNumber";
        this["Func_imconjugate_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imconjugate_Category"] = "Engineering";
        this["Func_imconjugate_CategoryURL"] = "Engineering";
        this["Func_imconjugate_Desc"] = "Returnerer konjugatet til et komplekst tall.";
        this["Func_imcos_Arg_0"] = "ComplexNumber";
        this["Func_imcos_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imcos_Category"] = "Engineering";
        this["Func_imcos_CategoryURL"] = "Engineering";
        this["Func_imcos_Desc"] = "Returnerer cosinus for et komplekst tall.";
        this["Func_imdiv_Arg_0"] = "Utbytte";
        this["Func_imdiv_Arg_1"] = "Deler";
        this["Func_imdiv_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imdiv_ArgDesc_1"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imdiv_Category"] = "Engineering";
        this["Func_imdiv_CategoryURL"] = "Engineering";
        this["Func_imdiv_Desc"] = "Returnerer kvoten av to komplekse tall (utbytte / divisor).";
        this["Func_imexp_Arg_0"] = "ComplexNumber";
        this["Func_imexp_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imexp_Category"] = "Engineering";
        this["Func_imexp_CategoryURL"] = "Engineering";
        this["Func_imexp_Desc"] = "Returnerer eksponentiell for et komplekst tall.";
        this["Func_imln_Arg_0"] = "ComplexNumber";
        this["Func_imln_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imln_Category"] = "Engineering";
        this["Func_imln_CategoryURL"] = "Engineering";
        this["Func_imln_Desc"] = "Returnerer den naturlige logaritmen til et komplekst tall.";
        this["Func_imlog10_Arg_0"] = "ComplexNumber";
        this["Func_imlog10_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imlog10_Category"] = "Engineering";
        this["Func_imlog10_CategoryURL"] = "Engineering";
        this["Func_imlog10_Desc"] = "Returnerer base-10-logaritmen til et komplekst tall.";
        this["Func_imlog2_Arg_0"] = "ComplexNumber";
        this["Func_imlog2_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imlog2_Category"] = "Engineering";
        this["Func_imlog2_CategoryURL"] = "Engineering";
        this["Func_imlog2_Desc"] = "Returnerer base-2-logaritmen til et komplekst tall.";
        this["Func_impower_Arg_0"] = "ComplexNumber";
        this["Func_impower_Arg_1"] = "Makt";
        this["Func_impower_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_impower_ArgDesc_1"] = "Kraften som ComplexNumber skal økes til.";
        this["Func_impower_Category"] = "Engineering";
        this["Func_impower_CategoryURL"] = "Engineering";
        this["Func_impower_Desc"] = "Returnerer det komplekse tallet som er hevet til den angitte effekten.";
        this["Func_improduct_Arg_0"] = "ComplexNumber{0}";
        this["Func_improduct_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_improduct_Category"] = "Engineering";
        this["Func_improduct_CategoryURL"] = "Engineering";
        this["Func_improduct_Desc"] = "Returnerer produktet av ett eller flere komplekse tall.";
        this["Func_imreal_Arg_0"] = "ComplexNumber";
        this["Func_imreal_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imreal_Category"] = "Engineering";
        this["Func_imreal_CategoryURL"] = "Engineering";
        this["Func_imreal_Desc"] = "Returnerer den virkelige koeffisienten for et komplekst tall.";
        this["Func_imsin_Arg_0"] = "ComplexNumber";
        this["Func_imsin_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imsin_Category"] = "Engineering";
        this["Func_imsin_CategoryURL"] = "Engineering";
        this["Func_imsin_Desc"] = "Returnerer sinus for et komplekst tall.";
        this["Func_imsqrt_Arg_0"] = "ComplexNumber";
        this["Func_imsqrt_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imsqrt_Category"] = "Engineering";
        this["Func_imsqrt_CategoryURL"] = "Engineering";
        this["Func_imsqrt_Desc"] = "Returnerer kvadratroten til et komplekst tall.";
        this["Func_imsub_Arg_0"] = "KompleksNummer1";
        this["Func_imsub_Arg_1"] = "KompleksNummer2";
        this["Func_imsub_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\" som ComplexNumber2 blir trukket fra.";
        this["Func_imsub_ArgDesc_1"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\" som blir trukket fra ComplexNumber1.";
        this["Func_imsub_Category"] = "Engineering";
        this["Func_imsub_CategoryURL"] = "Engineering";
        this["Func_imsub_Desc"] = "Returnerer forskjellen mellom to komplekse tall.";
        this["Func_imsum_Arg_0"] = "ComplexNumber{0}";
        this["Func_imsum_ArgDesc_0"] = "Tekstrepresentasjonen av et komplekst tall i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imsum_Category"] = "Engineering";
        this["Func_imsum_CategoryURL"] = "Engineering";
        this["Func_imsum_Desc"] = "Returnerer summen av ett eller flere komplekse tall.";
        this["Func_info_Arg_0"] = "Type";
        this["Func_info_ArgDesc_0"] = "Et tekstuttrykk som angir typen informasjon som skal returneres.\r\n\"directory\" = Den fullstendige banen til gjeldende mappe.\r\n\"osversion\" = Den nåværende operativsystemversjonen som tekstuttrykk.\r\n\"system\" = Det nåværende operativsystemet. Macintosh = \"mac\", Windows = \"pcdos\".\r\n    ";
        this["Func_info_Category"] = "Information";
        this["Func_info_CategoryURL"] = "Information";
        this["Func_info_Desc"] = "Returnerer informasjon om gjeldende driftsmiljø.";
        this["Func_int_Arg_0"] = "number";
        this["Func_int_ArgDesc_0"] = "Det virkelige tallet som skal rundes ned til et heltall.";
        this["Func_int_Category"] = "Math";
        this["Func_int_CategoryURL"] = "Math";
        this["Func_int_Desc"] = "Avrunder et tall ned til nærmeste heltall.";
        this["Func_intrate_Arg_0"] = "bosetting";
        this["Func_intrate_Arg_1"] = "modenhet";
        this["Func_intrate_Arg_2"] = "investering";
        this["Func_intrate_Arg_3"] = "forløsning";
        this["Func_intrate_Arg_4"] = "basis";
        this["Func_intrate_ArgDesc_0"] = "Sikkerhetens oppgjørsdato.";
        this["Func_intrate_ArgDesc_1"] = "Verdipapirets løpetid.";
        this["Func_intrate_ArgDesc_2"] = "Beløpet som er investert i sikkerheten.";
        this["Func_intrate_ArgDesc_3"] = "Beløpet som skal mottas ved forfall.";
        this["Func_intrate_ArgDesc_4"] = "Basis for dagtelling som skal brukes.\r\n0 eller utelatt = US (NASD) 30/360,\r\n1 = Faktisk/faktisk,\r\n2 = Faktisk/360,\r\n3 = Faktisk/365,\r\n4 = europeisk 30/360\r\n    ";
        this["Func_intrate_Category"] = "Financial";
        this["Func_intrate_CategoryURL"] = "Financial";
        this["Func_intrate_Desc"] = "Returnerer renten for et fullt investert verdipapir.";
        this["Func_ipmt_Arg_0"] = "vurdere";
        this["Func_ipmt_Arg_1"] = "per";
        this["Func_ipmt_Arg_2"] = "nper";
        this["Func_ipmt_Arg_3"] = "pv";
        this["Func_ipmt_Arg_4"] = "fv";
        this["Func_ipmt_Arg_5"] = "type";
        this["Func_ipmt_ArgDesc_0"] = "Renten per periode.";
        this["Func_ipmt_ArgDesc_1"] = "Perioden, som må være i området fra en til nper.";
        this["Func_ipmt_ArgDesc_2"] = "Det totale antallet betalingsperioder i en livrente.";
        this["Func_ipmt_ArgDesc_3"] = "Nåverdien, eller engangsbeløpet som en rekke fremtidige betalinger er verdt nå.";
        this["Func_ipmt_ArgDesc_4"] = "Den fremtidige verdien, eller en kontantsaldo du vil oppnå etter at den siste betalingen er foretatt. Hvis den er utelatt, antas den å være null.";
        this["Func_ipmt_ArgDesc_5"] = "Tallet null eller ett som angir om betalinger forfaller ved slutten av perioden (null, standardverdien) eller begynnelsen (en).";
        this["Func_ipmt_Category"] = "Financial";
        this["Func_ipmt_CategoryURL"] = "Financial";
        this["Func_ipmt_Desc"] = "Returnerer rentebetalingen for en gitt periode for en investering, basert på periodiske, konstante betalinger og en konstant rente.";
        this["Func_irr_Arg_0"] = "verdier";
        this["Func_irr_Arg_1"] = "Gjett";
        this["Func_irr_ArgDesc_0"] = "En samling av betalinger (negative verdier) og inntekter (positive verdier) som oppstår med jevne mellomrom.";
        this["Func_irr_ArgDesc_1"] = "Et tall du antar er nær resultatet. Standard er 0,1 (10%).";
        this["Func_irr_Category"] = "Finansiell";
        this["Func_irr_CategoryURL"] = "Financial";
        this["Func_irr_Desc"] = "Returnerer den interne avkastningen for en serie kontantstrømmer.";
        this["Func_isblank_Arg_0"] = "verdi";
        this["Func_isblank_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_isblank_Category"] = "Information";
        this["Func_isblank_CategoryURL"] = "Information";
        this["Func_isblank_Desc"] = "Returnerer true hvis verdien er tom.";
        this["Func_isdbnull_Arg_0"] = "verdi";
        this["Func_isdbnull_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_isdbnull_Category"] = "Information";
        this["Func_isdbnull_CategoryURL"] = "Information";
        this["Func_isdbnull_Desc"] = "Returnerer true hvis verdien representerer en ikke-eksisterende verdi.";
        this["Func_iserr_Arg_0"] = "verdi";
        this["Func_iserr_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_iserr_Category"] = "Information";
        this["Func_iserr_CategoryURL"] = "Information";
        this["Func_iserr_Desc"] = "Returnerer true hvis verdien er en feilverdi unntatt #N/A.";
        this["Func_iserror_Arg_0"] = "verdi";
        this["Func_iserror_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_iserror_Category"] = "Information";
        this["Func_iserror_CategoryURL"] = "Information";
        this["Func_iserror_Desc"] = "Returnerer true hvis verdien er en feilverdi.";
        this["Func_iseven_Arg_0"] = "verdi";
        this["Func_iseven_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_iseven_Category"] = "Information";
        this["Func_iseven_CategoryURL"] = "Information";
        this["Func_iseven_Desc"] = "Returnerer true hvis verdien er et partall, ellers returnerer false.";
        this["Func_islogical_Arg_0"] = "verdi";
        this["Func_islogical_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_islogical_Category"] = "Information";
        this["Func_islogical_CategoryURL"] = "Information";
        this["Func_islogical_Desc"] = "Returnerer true hvis verdien er en logisk verdi (true eller false).";
        this["Func_isna_Arg_0"] = "verdi";
        this["Func_isna_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_isna_Category"] = "Information";
        this["Func_isna_CategoryURL"] = "Information";
        this["Func_isna_Desc"] = "Returnerer true hvis verdien er feilverdien #N/A (verdi ikke tilgjengelig).";
        this["Func_isnontext_Arg_0"] = "verdi";
        this["Func_isnontext_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_isnontext_Category"] = "Information";
        this["Func_isnontext_CategoryURL"] = "Information";
        this["Func_isnontext_Desc"] = "Returnerer true hvis verdien er et element som ikke er tekst.";
        this["Func_isnull_Arg_0"] = "verdi";
        this["Func_isnull_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_isnull_Category"] = "Information";
        this["Func_isnull_CategoryURL"] = "Information";
        this["Func_isnull_Desc"] = "Returnerer true hvis verdien ikke finnes.";
        this["Func_isnumber_Arg_0"] = "verdi";
        this["Func_isnumber_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_isnumber_Category"] = "Information";
        this["Func_isnumber_CategoryURL"] = "Information";
        this["Func_isnumber_Desc"] = "Returnerer true hvis verdien er et tall.";
        this["Func_isodd_Arg_0"] = "verdi";
        this["Func_isodd_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_isodd_Category"] = "Information";
        this["Func_isodd_CategoryURL"] = "Information";
        this["Func_isodd_Desc"] = "Returnerer true hvis verdien er et oddetall, ellers returnerer false.";
        this["Func_isref_Arg_0"] = "verdi";
        this["Func_isref_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_isref_Category"] = "Informasjon";
        this["Func_isref_CategoryURL"] = "Information";
        this["Func_isref_Desc"] = "Returnerer true hvis verdien er en referanse.";
        this["Func_istext_Arg_0"] = "verdi";
        this["Func_istext_ArgDesc_0"] = "Verdien som skal testes.";
        this["Func_istext_Category"] = "Information";
        this["Func_istext_CategoryURL"] = "Information";
        this["Func_istext_Desc"] = "Returnerer true hvis verdien er en tekstverdi.";
        this["Func_lcm_Arg_0"] = "number{0}";
        this["Func_lcm_ArgDesc_0"] = "En heltallsverdi. En verdi som ikke er heltall, blir avkortet.";
        this["Func_lcm_Category"] = "Math";
        this["Func_lcm_CategoryURL"] = "Math";
        this["Func_lcm_Desc"] = "Returnerer det minst vanlige multiplumet av en eller flere heltallsverdier.";
        this["Func_left_Arg_0"] = "tekst";
        this["Func_left_Arg_1"] = "num_chars";
        this["Func_left_ArgDesc_0"] = "Tekstuttrykket som inneholder tegnene som skal trekkes ut.";
        this["Func_left_ArgDesc_1"] = "Antall tegn som skal trekkes ut.";
        this["Func_left_Category"] = "TextAndData";
        this["Func_left_CategoryURL"] = "Text_and_Data";
        this["Func_left_Desc"] = "Returnerer det eller de første tegnene i et tekstuttrykk.";
        this["Func_len_Arg_0"] = "tekst";
        this["Func_len_ArgDesc_0"] = "Tekstuttrykket hvis lengde å finne. Mellomrom teller som tegn.";
        this["Func_len_Category"] = "TextAndData";
        this["Func_len_CategoryURL"] = "Text_and_Data";
        this["Func_len_Desc"] = "Returnerer antall tegn i et tekstuttrykk.";
        this["Func_ln_Arg_0"] = "number";
        this["Func_ln_ArgDesc_0"] = "Det positive reelle tallet for å få den naturlige logaritmen.";
        this["Func_ln_Category"] = "Math";
        this["Func_ln_CategoryURL"] = "Math";
        this["Func_ln_Desc"] = "Returnerer den naturlige logaritmen til et tall.";
        this["Func_log_Arg_0"] = "number";
        this["Func_log_Arg_1"] = "utgangspunkt";
        this["Func_log_ArgDesc_0"] = "Det positive reelle tallet for å få logaritmen.";
        this["Func_log_ArgDesc_1"] = "Grunnlaget for logaritmen.";
        this["Func_log_Category"] = "Math";
        this["Func_log_CategoryURL"] = "Math";
        this["Func_log_Desc"] = "Returnerer logaritmen til et tall til den angitte basen.";
        this["Func_log10_Arg_0"] = "number";
        this["Func_log10_ArgDesc_0"] = "Det positive reelle tallet for å få base-10-logaritmen.";
        this["Func_log10_Category"] = "Math";
        this["Func_log10_CategoryURL"] = "Math";
        this["Func_log10_Desc"] = "Returnerer basis-10-logaritmen til et tall.";
        this["Func_lower_Arg_0"] = "tekst";
        this["Func_lower_ArgDesc_0"] = "Teksten som skal konverteres til små bokstaver.";
        this["Func_lower_Category"] = "TextAndData";
        this["Func_lower_CategoryURL"] = "Text_and_Data";
        this["Func_lower_Desc"] = "Konverterer alle store bokstaver i et tekstuttrykk til små bokstaver.";
        this["Func_max_Arg_0"] = "number{0}";
        this["Func_max_ArgDesc_0"] = "Et tall eller tallområde.";
        this["Func_max_Category"] = "Statistical";
        this["Func_max_CategoryURL"] = "Statistical";
        this["Func_max_Desc"] = "Returnerer det største tallet i et sett med tall.";
        this["Func_median_Arg_0"] = "number{0}";
        this["Func_median_ArgDesc_0"] = "Et tall eller tallområde.";
        this["Func_median_Category"] = "Statistical";
        this["Func_median_CategoryURL"] = "Statistical";
        this["Func_median_Desc"] = "Returnerer medianen til et sett med tall. Medianen er mellomtallet når tallene er sortert eller gjennomsnittet av de to mellomtallene hvis det er et partall med spesifiserte tall.";
        this["Func_mid_Arg_0"] = "tekst";
        this["Func_mid_Arg_1"] = "start_num";
        this["Func_mid_Arg_2"] = "num_chars";
        this["Func_mid_ArgDesc_0"] = "Tekstuttrykket som inneholder tegnene som skal trekkes ut.";
        this["Func_mid_ArgDesc_1"] = "Den 1-baserte indeksen til det første tegnet som ble trukket ut.";
        this["Func_mid_ArgDesc_2"] = "Antall tegn som skal trekkes ut.";
        this["Func_mid_Category"] = "TextAndData";
        this["Func_mid_CategoryURL"] = "Text_and_Data";
        this["Func_mid_Desc"] = "Returnerer et bestemt antall tegn fra et tekstuttrykk.";
        this["Func_min_Arg_0"] = "number{0}";
        this["Func_min_ArgDesc_0"] = "Et tall eller tallområde.";
        this["Func_min_Category"] = "Statistical";
        this["Func_min_CategoryURL"] = "Statistical";
        this["Func_min_Desc"] = "Returnerer det minste tallet i et sett med tall.";
        this["Func_minute_Arg_0"] = "Time_value";
        this["Func_minute_ArgDesc_0"] = "En tidsverdi for å få minuttet.";
        this["Func_minute_Category"] = "DateAndTime";
        this["Func_minute_CategoryURL"] = "Date_and_Time";
        this["Func_minute_Desc"] = "Returnerer minuttet for en tidsverdi.";
        this["Func_mod_Arg_0"] = "number";
        this["Func_mod_Arg_1"] = "deler";
        this["Func_mod_ArgDesc_0"] = "Nummeret du vil finne resten for.";
        this["Func_mod_ArgDesc_1"] = "Verdien som tallet skal deles med.";
        this["Func_mod_Category"] = "Math";
        this["Func_mod_CategoryURL"] = "Math";
        this["Func_mod_Desc"] = "Returnerer resten etter at et tall er delt med en divisor. Resultatet har samme tegn som deler.";
        this["Func_month_Arg_0"] = "Date_value";
        this["Func_month_ArgDesc_0"] = "En datoverdi som du kan hente måneden fra.";
        this["Func_month_Category"] = "DateAndTime";
        this["Func_month_CategoryURL"] = "Date_and_Time";
        this["Func_month_Desc"] = "Returnerer måneden for en datoverdi.";
        this["Func_mround_Arg_0"] = "Nummer";
        this["Func_mround_Arg_1"] = "Flere";
        this["Func_mround_ArgDesc_0"] = "Tallet som skal rundes.";
        this["Func_mround_ArgDesc_1"] = "Et tall som angir multiplumet som skal rundes.";
        this["Func_mround_Category"] = "Math";
        this["Func_mround_CategoryURL"] = "Math";
        this["Func_mround_Desc"] = "Avrunder et tall til nærmeste multiplum av et annet tall.";
        this["Func_multinomial_Arg_0"] = "number{0}";
        this["Func_multinomial_ArgDesc_0"] = "Ett av tallene for å få multinomialet.";
        this["Func_multinomial_Category"] = "Math";
        this["Func_multinomial_CategoryURL"] = "Math";
        this["Func_multinomial_Desc"] = "Returnerer multinomialet til et sett med tall. Multinomialet er forholdet mellom faktoren til en sum av verdier til produktet av deres fabrikk.";
        this["Func_n_Arg_0"] = "Verdi";
        this["Func_n_ArgDesc_0"] = "Verdien som skal konverteres til et tall. Følgende er akseptable verdier:\r\nEt tall - returnerer tallet.\r\nEn dato - returnerer datoen.\r\nTrue - returnerer 1.\r\nFalse - returnerer 0.\r\nEn feilverdi - returnerer feilverdien.\r\nEnhver annen verdi - returnerer 0.\r\n    ";
        this["Func_n_Category"] = "Information";
        this["Func_n_CategoryURL"] = "Information";
        this["Func_n_Desc"] = "Konverterer en verdi til et tall.";
        this["Func_na_Category"] = "Information";
        this["Func_na_CategoryURL"] = "Information";
        this["Func_na_Desc"] = "Returnerer feilverdien #N/A (verdi ikke tilgjengelig).";
        this["Func_networkdays_Arg_0"] = "startDate";
        this["Func_networkdays_Arg_1"] = "endDate";
        this["Func_networkdays_Arg_2"] = "holidays";
        this["Func_networkdays_ArgDesc_0"] = "Datoen for start.";
        this["Func_networkdays_ArgDesc_1"] = "Datoen for slutt.";
        this["Func_networkdays_ArgDesc_2"] = "Et område med en eller flere referanser som inneholder datoer som skal betraktes som arbeidsdager.";
        this["Func_networkdays_Category"] = "DateAndTime";
        this["Func_networkdays_CategoryURL"] = "DateAndTime";
        this["Func_networkdays_Desc"] = "Returnerer det totale antallet hele arbeidsdager mellom de angitte datoene, unntatt eventuelle spesifiserte helligdager.";
        this["Func_not_Arg_0"] = "logical";
        this["Func_not_ArgDesc_0"] = "En logisk verdi (sann eller usann).";
        this["Func_not_Category"] = "Logical";
        this["Func_not_CategoryURL"] = "Logical";
        this["Func_not_Desc"] = "Reverserer en logisk verdi (endres true til false og false til true).";
        this["Func_now_Category"] = "DateAndTime";
        this["Func_now_CategoryURL"] = "Date_and_Time";
        this["Func_now_Desc"] = "Returnerer en verdi som representerer gjeldende dato og klokkeslett.";
        this["Func_nper_Arg_0"] = "vurdere";
        this["Func_nper_Arg_1"] = "pmt";
        this["Func_nper_Arg_2"] = "pv";
        this["Func_nper_Arg_3"] = "fv";
        this["Func_nper_Arg_4"] = "type";
        this["Func_nper_ArgDesc_0"] = "Renten per periode.";
        this["Func_nper_ArgDesc_1"] = "Betalingen foretok hver periode, som ikke kan endres i løpet av livrenten.";
        this["Func_nper_ArgDesc_2"] = "Nåverdien, eller engangsbeløpet som en rekke fremtidige betalinger er verdt akkurat nå.";
        this["Func_nper_ArgDesc_3"] = "Den fremtidige verdien, eller en kontantsaldo du vil oppnå etter at den siste betalingen er foretatt.";
        this["Func_nper_ArgDesc_4"] = "Tallet null eller ett som angir om betalinger forfaller ved slutten av perioden (null, standardverdien) eller begynnelsen (en).";
        this["Func_nper_Category"] = "Finansiell";
        this["Func_nper_CategoryURL"] = "Financial";
        this["Func_nper_Desc"] = "Returnerer antall perioder for en investering basert på periodiske, konstante utbetalinger og en konstant rente.";
        this["Func_npv_Arg_0"] = "vurdere";
        this["Func_npv_Arg_1"] = "value{0}";
        this["Func_npv_ArgDesc_0"] = "Rabattraten over en periode.";
        this["Func_npv_ArgDesc_1"] = "En betaling (negativ verdi) eller inntekt (positiv verdi).";
        this["Func_npv_Category"] = "Financial";
        this["Func_npv_CategoryURL"] = "Financial";
        this["Func_npv_Desc"] = "Beregner netto nåverdi av en investering ved å bruke en diskonteringsrente og en serie fremtidige betalinger (negative verdier) og inntekt (positiv verdi).";
        this["Func_null_Category"] = "Information";
        this["Func_null_CategoryURL"] = "Information";
        this["Func_null_Desc"] = "Returnerer en ikke-eksisterende verdi.";
        this["Func_oct2bin_Arg_0"] = "Nummer";
        this["Func_oct2bin_Arg_1"] = "Places";
        this["Func_oct2bin_ArgDesc_0"] = "Et tekstuttrykk som representerer et oktalt tall.";
        this["Func_oct2bin_ArgDesc_1"] = "Antall tegn som skal brukes for å representere verdien. Hvis dette ikke er spesifisert, brukes det maksimale antallet tegn som er nødvendig for å representere verdien. Dette lar deg angi ledende nuller.";
        this["Func_oct2bin_Category"] = "Engineering";
        this["Func_oct2bin_CategoryURL"] = "Engineering";
        this["Func_oct2bin_Desc"] = "Konverterer et tekstuttrykk som representerer et oktaltall til et tekstuttrykk som representerer det samme tallet i det binære tallsystemet.";
        this["Func_oct2dec_Arg_0"] = "Nummer";
        this["Func_oct2dec_ArgDesc_0"] = "Et tekstuttrykk som representerer et oktalt tall.";
        this["Func_oct2dec_Category"] = "Engineering";
        this["Func_oct2dec_CategoryURL"] = "Engineering";
        this["Func_oct2dec_Desc"] = "Returnerer desimalverdien til en tekstrepresentasjon av et tall i det oktale tallsystemet.";
        this["Func_oct2hex_Arg_0"] = "Nummer";
        this["Func_oct2hex_Arg_1"] = "Places";
        this["Func_oct2hex_ArgDesc_0"] = "Et tekstuttrykk som representerer et oktalt tall.";
        this["Func_oct2hex_ArgDesc_1"] = "Antall tegn som skal brukes for å representere verdien. Hvis dette ikke er spesifisert, brukes det maksimale antallet tegn som er nødvendig for å representere verdien. Dette lar deg angi ledende nuller.";
        this["Func_oct2hex_Category"] = "Engineering";
        this["Func_oct2hex_CategoryURL"] = "Engineering";
        this["Func_oct2hex_Desc"] = "Konverterer et tekstuttrykk som representerer et oktaltall til et tekstuttrykk som representerer det samme tallet i det heksadesimale tallsystemet.";
        this["Func_odd_Arg_0"] = "number";
        this["Func_odd_ArgDesc_0"] = "Verdien som skal rundes.";
        this["Func_odd_Category"] = "Math";
        this["Func_odd_CategoryURL"] = "Math";
        this["Func_odd_Desc"] = "Runder et positivt tall opp og et negativt tall ned til nærmeste oddetall.";
        this["Func_or_Arg_0"] = "logical{0}";
        this["Func_or_ArgDesc_0"] = "En logisk verdi (sann eller usann).";
        this["Func_or_Category"] = "Logical";
        this["Func_or_CategoryURL"] = "Logical";
        this["Func_or_Desc"] = "Returnerer true hvis en eller flere verdier er sanne; returnerer false hvis alle verdier er falske.";
        this["Func_pi_Category"] = "Math";
        this["Func_pi_CategoryURL"] = "Math";
        this["Func_pi_Desc"] = "Returnerer konstant pi, som er 3.14159265358979.";
        this["Func_pmt_Arg_0"] = "vurdere";
        this["Func_pmt_Arg_1"] = "nper";
        this["Func_pmt_Arg_2"] = "pv";
        this["Func_pmt_Arg_3"] = "fv";
        this["Func_pmt_Arg_4"] = "type";
        this["Func_pmt_ArgDesc_0"] = "Renten på lånet.";
        this["Func_pmt_ArgDesc_1"] = "Totalt antall betalinger for lånet.";
        this["Func_pmt_ArgDesc_2"] = "Nåverdien, eller det totale beløpet som en serie fremtidige betalinger er verdt nå; også kjent som rektor.";
        this["Func_pmt_ArgDesc_3"] = "Den fremtidige verdien, eller en kontantsaldo du vil oppnå etter at den siste betalingen er foretatt.";
        this["Func_pmt_ArgDesc_4"] = "Tallet null eller ett som angir om betalinger forfaller ved slutten av perioden (null, standardverdien) eller begynnelsen (en).";
        this["Func_pmt_Category"] = "Financial";
        this["Func_pmt_CategoryURL"] = "Financial";
        this["Func_pmt_Desc"] = "Beregner betalingen for et lån basert på faste betalinger og en konstant rente.";
        this["Func_power_Arg_0"] = "number";
        this["Func_power_Arg_1"] = "makt";
        this["Func_power_ArgDesc_0"] = "Grunnnummeret for å heve til en makt.";
        this["Func_power_ArgDesc_1"] = "Eksponenten som basisnummeret er hevet til.";
        this["Func_power_Category"] = "Math";
        this["Func_power_CategoryURL"] = "Math";
        this["Func_power_Desc"] = "Returnerer resultatet av et tall som er hevet til en effekt.";
        this["Func_ppmt_Arg_0"] = "vurdere";
        this["Func_ppmt_Arg_1"] = "per";
        this["Func_ppmt_Arg_2"] = "nper";
        this["Func_ppmt_Arg_3"] = "pv";
        this["Func_ppmt_Arg_4"] = "fv";
        this["Func_ppmt_Arg_5"] = "type";
        this["Func_ppmt_ArgDesc_0"] = "Renten per periode.";
        this["Func_ppmt_ArgDesc_1"] = "Perioden, som må være i området fra en til nper.";
        this["Func_ppmt_ArgDesc_2"] = "Det totale antallet betalingsperioder i en livrente.";
        this["Func_ppmt_ArgDesc_3"] = "Nåverdien, eller det totale beløpet som en serie fremtidige betalinger er verdt nå.";
        this["Func_ppmt_ArgDesc_4"] = "Den fremtidige verdien, eller en kontantsaldo du vil oppnå etter at den siste betalingen er foretatt.";
        this["Func_ppmt_ArgDesc_5"] = "Tallet null eller ett som angir om betalinger forfaller ved slutten av perioden (null, standardverdien) eller begynnelsen (en).";
        this["Func_ppmt_Category"] = "Financial";
        this["Func_ppmt_CategoryURL"] = "Financial";
        this["Func_ppmt_Desc"] = "Returnerer betalingen på hovedstolen for en gitt periode for en investering basert på periodiske, konstante betalinger og en konstant rente.";
        this["Func_product_Arg_0"] = "number{0}";
        this["Func_product_ArgDesc_0"] = "Et tall som skal multipliseres med de andre tallene.";
        this["Func_product_Category"] = "Math";
        this["Func_product_CategoryURL"] = "Math";
        this["Func_product_Desc"] = "Multipliserer alle tallene som er gitt som argumenter og returnerer produktet.";
        this["Func_pv_Arg_0"] = "vurdere";
        this["Func_pv_Arg_1"] = "nper";
        this["Func_pv_Arg_2"] = "pmt";
        this["Func_pv_Arg_3"] = "fv";
        this["Func_pv_Arg_4"] = "type";
        this["Func_pv_ArgDesc_0"] = "Renten per periode.";
        this["Func_pv_ArgDesc_1"] = "Det totale antallet betalingsperioder i en livrente.";
        this["Func_pv_ArgDesc_2"] = "Betalingen foretok hver periode, som ikke kan endres i løpet av livrenten.";
        this["Func_pv_ArgDesc_3"] = "Den fremtidige verdien, eller en kontantsaldo du vil oppnå etter at den siste betalingen er foretatt.";
        this["Func_pv_ArgDesc_4"] = "Tallet null eller ett som angir om betalinger forfaller ved slutten av perioden (null, standardverdien) eller begynnelsen (en).";
        this["Func_pv_Category"] = "Financial";
        this["Func_pv_CategoryURL"] = "Financial";
        this["Func_pv_Desc"] = "Returnerer nåverdien av en investering.";
        this["Func_quotient_Arg_0"] = "teller";
        this["Func_quotient_Arg_1"] = "nevner";
        this["Func_quotient_ArgDesc_0"] = "Utbytte av divisjonsoperasjonen.";
        this["Func_quotient_ArgDesc_1"] = "Deleren av divisjonsoperasjonen.";
        this["Func_quotient_Category"] = "Math";
        this["Func_quotient_CategoryURL"] = "Math";
        this["Func_quotient_Desc"] = "Returnerer hele talldelen av en divisjon. Dette vil forkaste resten av divisjonen.";
        this["Func_radians_Arg_0"] = "Grader";
        this["Func_radians_ArgDesc_0"] = "En verdi i grader som vil bli konvertert til radianer.";
        this["Func_radians_Category"] = "Math";
        this["Func_radians_CategoryURL"] = "Math";
        this["Func_radians_Desc"] = "Konverterer grader til radianer.";
        this["Func_rand_Category"] = "Math";
        this["Func_rand_CategoryURL"] = "Math";
        this["Func_rand_Desc"] = "Returnerer et jevnt fordelt tilfeldig tall større enn eller lik 0 og mindre enn 1";
        this["Func_randbetween_Arg_0"] = "Bunn";
        this["Func_randbetween_Arg_1"] = "Topp";
        this["Func_randbetween_ArgDesc_0"] = "Minste verdi som vil bli returnert.";
        this["Func_randbetween_ArgDesc_1"] = "Maksimal verdi som vil bli returnert.";
        this["Func_randbetween_Category"] = "Math";
        this["Func_randbetween_CategoryURL"] = "Math";
        this["Func_randbetween_Desc"] = "Genererer et pseudoslått heltall mellom to spesifiserte tall.";
        this["Func_rate_Arg_0"] = "NPer";
        this["Func_rate_Arg_1"] = "Pmt";
        this["Func_rate_Arg_2"] = "Pv";
        this["Func_rate_Arg_3"] = "Fv";
        this["Func_rate_Arg_4"] = "Type";
        this["Func_rate_Arg_5"] = "Gjett";
        this["Func_rate_ArgDesc_0"] = "Det totale antallet betalingsperioder i en livrente.";
        this["Func_rate_ArgDesc_1"] = "Betalingen foretok hver periode og kan ikke endres i løpet av livrenten.";
        this["Func_rate_ArgDesc_2"] = "Nåverdien, eller det totale beløpet som en serie fremtidige betalinger er verdt nå.";
        this["Func_rate_ArgDesc_3"] = "Den fremtidige verdien, eller en kontantsaldo du vil oppnå etter at den siste betalingen er foretatt. Hvis fv utelates, antas det å være 0 (den fremtidige verdien av et lån, for eksempel, er 0). For eksempel, hvis du vil spare $50,000 for å betale for et spesialprosjekt på 18 år, er $50,000 den fremtidige verdien. Du kan deretter gjøre et konservativt gjetning til en rente og bestemme hvor mye du må spare hver måned. Hvis fv utelates, må du inkludere pmt-argumentet. Standardverdien for fv er null.";
        this["Func_rate_ArgDesc_4"] = "Tallet null eller ett som angir om betalinger forfaller ved slutten av perioden (null, standardverdien) eller begynnelsen (en).";
        this["Func_rate_ArgDesc_5"] = "Et tall du antar er nær resultatet. Hvis den er utelatt, antas den å være 0,1 (10 %).";
        this["Func_rate_Category"] = "Finansiell";
        this["Func_rate_CategoryURL"] = "Financial";
        this["Func_rate_Desc"] = "Returnerer renten per periode på en livrente.";
        this["Func_replace_Arg_0"] = "Old_text";
        this["Func_replace_Arg_1"] = "Start_num";
        this["Func_replace_Arg_2"] = "Num_chars";
        this["Func_replace_Arg_3"] = "New_text";
        this["Func_replace_ArgDesc_0"] = "Tekstuttrykket der noen tegn skal erstattes.";
        this["Func_replace_ArgDesc_1"] = "Den 1-baserte indeksen til tegnet i old_text som skal erstattes med new_text.";
        this["Func_replace_ArgDesc_2"] = "Antall tegn i old_text som skal erstattes med New_text.";
        this["Func_replace_ArgDesc_3"] = "Teksten som vil erstatte tegnene i old_text.";
        this["Func_replace_Category"] = "TextAndData";
        this["Func_replace_CategoryURL"] = "Text_and_Data";
        this["Func_replace_Desc"] = "Erstatter deler av et tekstuttrykk.";
        this["Func_rept_Arg_0"] = "TextValue";
        this["Func_rept_Arg_1"] = "RepeatCount";
        this["Func_rept_ArgDesc_0"] = "Teksten som skal gjentas.";
        this["Func_rept_ArgDesc_1"] = "Det positive antallet ganger for å gjenta teksten.";
        this["Func_rept_Category"] = "TextAndData";
        this["Func_rept_CategoryURL"] = "Text_and_Data";
        this["Func_rept_Desc"] = "Gjentar tekst et bestemt antall ganger.";
        this["Func_right_Arg_0"] = "tekst";
        this["Func_right_Arg_1"] = "num_chars";
        this["Func_right_ArgDesc_0"] = "Tekstuttrykket som inneholder tegnene som skal trekkes ut.";
        this["Func_right_ArgDesc_1"] = "Antall tegn som skal trekkes ut.";
        this["Func_right_Category"] = "TextAndData";
        this["Func_right_CategoryURL"] = "Text_and_Data";
        this["Func_right_Desc"] = "Returnerer det eller de siste tegnene i et tekstuttrykk.";
        this["Func_roman_Arg_0"] = "Nummer";
        this["Func_roman_Arg_1"] = "Skjema";
        this["Func_roman_ArgDesc_0"] = "Tallet som skal konverteres til romertall.";
        this["Func_roman_ArgDesc_1"] = "Et tall eller en boolsk verdi som angir om du skal bruke klassiske romertall eller en mer kortfattet versjon.\r\n0 = (Standard) Klassisk. (499 = \"CDXCIX\")\r\n1 = Mer konsist. (499 = \"LDVLIV\")\r\n2 = Mer konsist. (499 = \"XDIX\")\r\n3 = Mer konsist. (499 = \"VDIV\")\r\n4 = Forenklet. (499 = \"ID\")\r\nTrue = Klassisk.\r\nFalse = Sorenklet.\r\n    ";
        this["Func_roman_Category"] = "Math";
        this["Func_roman_CategoryURL"] = "Math";
        this["Func_roman_Desc"] = "Konverterer et tall til et romersk tall som et tekstuttrykk.";
        this["Func_round_Arg_0"] = "number";
        this["Func_round_Arg_1"] = "num_digits";
        this["Func_round_Arg_2"] = "round_Style";
        this["Func_round_ArgDesc_0"] = "Tallet som skal rundes.";
        this["Func_round_ArgDesc_1"] = "Antall sifre som nummeret skal rundes til.";
        this["Func_round_ArgDesc_2"] = "Et tall som angir avrundingsstilen som skal brukes.\r\n0 - Aritmetisk avrunding (standard) - Hvis det siste betydelige sifferet i avrundingen er 5, avrundes tallet opp.\r\n1 - Banker's Rounding - Når det siste signifikante sifferet er et 5, brukes det foregående sifferet for å avgjøre om det skal rundes opp eller ned. Hvis tallet er jevnt, avrundes tallet nedover. Hvis det er rart, blir tallet avrundet.\r\n    ";
        this["Func_round_Category"] = "Math";
        this["Func_round_CategoryURL"] = "Math";
        this["Func_round_Desc"] = "Avrunder et tall til et angitt antall sifre.";
        this["Func_rounddown_Arg_0"] = "Nummer";
        this["Func_rounddown_Arg_1"] = "Siffer";
        this["Func_rounddown_ArgDesc_0"] = "Tallet som skal rundes ned.";
        this["Func_rounddown_ArgDesc_1"] = "Antall desimaler som skal rundes. Positive tall angir steder etter desimaltegnet, negative tall angir steder før desimalpunktet.";
        this["Func_rounddown_Category"] = "Math";
        this["Func_rounddown_CategoryURL"] = "Math";
        this["Func_rounddown_Desc"] = "Avrunder et tall ned til det angitte antallet siffer.";
        this["Func_roundup_Arg_0"] = "Nummer";
        this["Func_roundup_Arg_1"] = "Siffer";
        this["Func_roundup_ArgDesc_0"] = "Tallet som skal rundes opp.";
        this["Func_roundup_ArgDesc_1"] = "Antall desimaler som skal rundes. Positive tall angir steder etter desimaltegnet, negative tall angir steder før desimalpunktet.";
        this["Func_roundup_Category"] = "Math";
        this["Func_roundup_CategoryURL"] = "Math";
        this["Func_roundup_Desc"] = "Avrunder et tall opp til det angitte antallet siffer.";
        this["Func_search_Arg_0"] = "search_text";
        this["Func_search_Arg_1"] = "verdi";
        this["Func_search_Arg_2"] = "start_index";
        this["Func_search_ArgDesc_0"] = "Teksten du finner i verdi.";
        this["Func_search_ArgDesc_1"] = "Teksten for å finne search_text.";
        this["Func_search_ArgDesc_2"] = "Den 1-baserte indeksen der søket skal startes (hvis det er utelatt, er startindeksen 1).";
        this["Func_search_Category"] = "TextAndData";
        this["Func_search_CategoryURL"] = "TextAndData";
        this["Func_search_Desc"] = "Returnerer den 1-baserte indeksen til en streng i en annen, og søker etter bokstaver ufølsomt.";
        this["Func_searchb_Arg_0"] = "search_text";
        this["Func_searchb_Arg_1"] = "verdi";
        this["Func_searchb_Arg_2"] = "start_index";
        this["Func_searchb_ArgDesc_0"] = "Teksten du finner i verdi.";
        this["Func_searchb_ArgDesc_1"] = "Teksten for å finne search_text.";
        this["Func_searchb_ArgDesc_2"] = "Den 1-baserte indeksen der søket skal startes (hvis det er utelatt, er startindeksen 1).";
        this["Func_searchb_Category"] = "TextAndData";
        this["Func_searchb_CategoryURL"] = "TextAndData";
        this["Func_searchb_Desc"] = "Returnerer den 1-baserte indeksen til en streng i en annen, og søker etter bokstaver ufølsomt.";
        this["Func_second_Arg_0"] = "Time_value";
        this["Func_second_ArgDesc_0"] = "En tidsverdi som den andre skal hentes fra.";
        this["Func_second_Category"] = "DateAndTime";
        this["Func_second_CategoryURL"] = "Date_and_Time";
        this["Func_second_Desc"] = "Returnerer den andre av en tidsverdi.";
        this["Func_seriessum_Arg_0"] = "InputValue";
        this["Func_seriessum_Arg_1"] = "InitialPower";
        this["Func_seriessum_Arg_2"] = "Steg";
        this["Func_seriessum_Arg_3"] = "Koeffisienter";
        this["Func_seriessum_ArgDesc_0"] = "Inngangsverdien til kraftserien.";
        this["Func_seriessum_ArgDesc_1"] = "Den opprinnelige effekten som InputValue skal økes til.";
        this["Func_seriessum_ArgDesc_2"] = "Trinnet som vil bli brukt til å øke effekten for hvert begrep i serien.";
        this["Func_seriessum_ArgDesc_3"] = "Et sett med koeffisienter som hver påfølgende effekt av InputValue multipliseres med.";
        this["Func_seriessum_Category"] = "Math";
        this["Func_seriessum_CategoryURL"] = "Math";
        this["Func_seriessum_Desc"] = "Returnerer summen av en kraftserie.";
        this["Func_sign_Arg_0"] = "Nummer";
        this["Func_sign_ArgDesc_0"] = "Nummeret hvis tegn vil bli returnert.";
        this["Func_sign_Category"] = "Math";
        this["Func_sign_CategoryURL"] = "Math";
        this["Func_sign_Desc"] = "Returnerer tegnet på et tall (-1, 0 eller 1).";
        this["Func_sin_Arg_0"] = "number";
        this["Func_sin_ArgDesc_0"] = "Vinkelen i radianer for å få sinus.";
        this["Func_sin_Category"] = "Math";
        this["Func_sin_CategoryURL"] = "Math";
        this["Func_sin_Desc"] = "Returnerer sinus for den angitte vinkelen.";
        this["Func_sinh_Arg_0"] = "number";
        this["Func_sinh_ArgDesc_0"] = "Et reelt antall for å få den hyperboliske sinus.";
        this["Func_sinh_Category"] = "Math";
        this["Func_sinh_CategoryURL"] = "Math";
        this["Func_sinh_Desc"] = "Returnerer den hyperboliske sinusen til et tall.";
        this["Func_sln_Arg_0"] = "koste";
        this["Func_sln_Arg_1"] = "berging";
        this["Func_sln_Arg_2"] = "liv";
        this["Func_sln_ArgDesc_0"] = "Startkostnaden for eiendelen.";
        this["Func_sln_ArgDesc_1"] = "Verdien ved slutten av avskrivningen.";
        this["Func_sln_ArgDesc_2"] = "Antall perioder som eiendelen avskrives over.";
        this["Func_sln_Category"] = "Financial";
        this["Func_sln_CategoryURL"] = "Financial";
        this["Func_sln_Desc"] = "Returnerer lineære avskrivninger på en eiendel for en periode.";
        this["Func_sqrt_Arg_0"] = "number";
        this["Func_sqrt_ArgDesc_0"] = "Tallet for å få kvadratroten.";
        this["Func_sqrt_Category"] = "Math";
        this["Func_sqrt_CategoryURL"] = "Math";
        this["Func_sqrt_Desc"] = "Returnerer en positiv kvadratrot av et tall.";
        this["Func_sqrtpi_Arg_0"] = "Nummer";
        this["Func_sqrtpi_ArgDesc_0"] = "Tallet som skal multipliseres med pi.";
        this["Func_sqrtpi_Category"] = "Math";
        this["Func_sqrtpi_CategoryURL"] = "Math";
        this["Func_sqrtpi_Desc"] = "Multipliserer det angitte tallet med pi og returnerer kvadratroten til resultatet.";
        this["Func_stdev_Arg_0"] = "number{0}";
        this["Func_stdev_ArgDesc_0"] = "Et tall eller rekkevidde som angir et utvalg av en populasjon.";
        this["Func_stdev_Category"] = "Statistical";
        this["Func_stdev_CategoryURL"] = "Statistical";
        this["Func_stdev_Desc"] = "Returnerer standardavviket til et populasjonsutvalg.";
        this["Func_subtotal_Arg_0"] = "FunctionNum";
        this["Func_subtotal_Arg_1"] = "Ref{0}";
        this["Func_subtotal_ArgDesc_0"] = "Et tall som representerer funksjonen som skal beregnes. Legg til 100 i verdien for å ignorere skjulte verdier.\r\n1 (eller 101) - AVERAGE\r\n2 (eller 102) - COUNT\r\n4 (eller 104) - MAX\r\n5 (eller 105) - MIN\r\n6 (eller 106) - PRODUCT\r\n7 (eller 107) - STDEV\r\n9 (eller 109) - SUM\r\n10 (eller 110) - VAR\r\n    ";
        this["Func_subtotal_ArgDesc_1"] = "En eller flere referanser for å finne delsummen.";
        this["Func_subtotal_Category"] = "Math";
        this["Func_subtotal_CategoryURL"] = "Math";
        this["Func_subtotal_Desc"] = "Beregner delsummen i en av flere referanser.";
        this["Func_sum_Arg_0"] = "number{0}";
        this["Func_sum_ArgDesc_0"] = "Et tall som skal legges til de andre tallene.";
        this["Func_sum_Category"] = "Math";
        this["Func_sum_CategoryURL"] = "Math";
        this["Func_sum_Desc"] = "Summerer alle tallene som er gitt som argumenter og returnerer summen.";
        this["Func_syd_Arg_0"] = "koste";
        this["Func_syd_Arg_1"] = "berging";
        this["Func_syd_Arg_2"] = "liv";
        this["Func_syd_Arg_3"] = "per";
        this["Func_syd_ArgDesc_0"] = "Startkostnaden for eiendelen.";
        this["Func_syd_ArgDesc_1"] = "Verdien ved slutten av avskrivningen.";
        this["Func_syd_ArgDesc_2"] = "Antall perioder som eiendelen avskrives over.";
        this["Func_syd_ArgDesc_3"] = "Perioden, som må bruke de samme enhetene som livet.";
        this["Func_syd_Category"] = "Financial";
        this["Func_syd_CategoryURL"] = "Financial";
        this["Func_syd_Desc"] = "Returnerer summen av års siffer avskrivninger på en eiendel i en bestemt periode.";
        this["Func_tan_Arg_0"] = "number";
        this["Func_tan_ArgDesc_0"] = "Vinkelen, i radianer, for å få tangenten.";
        this["Func_tan_Category"] = "Math";
        this["Func_tan_CategoryURL"] = "Math";
        this["Func_tan_Desc"] = "Returnerer tangenten til den angitte vinkelen.";
        this["Func_tanh_Arg_0"] = "number";
        this["Func_tanh_ArgDesc_0"] = "Vinkelen, i radianer, for å få den hyperboliske tangenten.";
        this["Func_tanh_Category"] = "Math";
        this["Func_tanh_CategoryURL"] = "Math";
        this["Func_tanh_Desc"] = "Returnerer den hyperboliske tangenten til den angitte vinkelen.";
        this["Func_text_Arg_0"] = "Verdi";
        this["Func_text_Arg_1"] = "FormatText";
        this["Func_text_ArgDesc_0"] = "Den numeriske verdien som skal formateres.";
        this["Func_text_ArgDesc_1"] = "Nummerformatet som verdien skal formateres med.";
        this["Func_text_Category"] = "TextAndData";
        this["Func_text_CategoryURL"] = "Text_and_Data";
        this["Func_text_Desc"] = "Konverterer verdien til tekst basert på det angitte formatet.";
        this["Func_time_Arg_0"] = "time";
        this["Func_time_Arg_1"] = "minutt";
        this["Func_time_Arg_2"] = "sekund";
        this["Func_time_ArgDesc_0"] = "Et tall som representerer tidens time.";
        this["Func_time_ArgDesc_1"] = "Et tall som representerer tidens minutt.";
        this["Func_time_ArgDesc_2"] = "Et tall som representerer tidens andre.";
        this["Func_time_Category"] = "DateAndTime";
        this["Func_time_CategoryURL"] = "Date_and_Time";
        this["Func_time_Desc"] = "Returnerer et tall som representerer en bestemt tidsverdi.";
        this["Func_timevalue_Arg_0"] = "time_text";
        this["Func_timevalue_ArgDesc_0"] = "Et tekstuttrykk som representerer en tid.";
        this["Func_timevalue_Category"] = "DateAndTime";
        this["Func_timevalue_CategoryURL"] = "Date_and_Time";
        this["Func_timevalue_Desc"] = "Konverterer et tekstuttrykk som representerer en tid til den tilsvarende tidsverdien.";
        this["Func_today_Category"] = "DateAndTime";
        this["Func_today_CategoryURL"] = "Date_and_Time";
        this["Func_today_Desc"] = "Returnerer gjeldende dato.";
        this["Func_trim_Arg_0"] = "tekst";
        this["Func_trim_ArgDesc_0"] = "Teksten for å fjerne ekstra mellomrom.";
        this["Func_trim_Category"] = "TextAndData";
        this["Func_trim_CategoryURL"] = "Text_and_Data";
        this["Func_trim_Desc"] = "Fjerner alle mellomrom fra teksten bortsett fra enkeltrom mellom ord.";
        this["Func_true_Category"] = "Logical";
        this["Func_true_CategoryURL"] = "Logical";
        this["Func_true_Desc"] = "Returnerer den logiske verdien true.";
        this["Func_trunc_Arg_0"] = "number";
        this["Func_trunc_Arg_1"] = "num_digits";
        this["Func_trunc_ArgDesc_0"] = "Tallet som skal avkortes.";
        this["Func_trunc_ArgDesc_1"] = "Et tall som angir presisjonen til avkortingen. Standardverdien er null.";
        this["Func_trunc_Category"] = "Math";
        this["Func_trunc_CategoryURL"] = "Math";
        this["Func_trunc_Desc"] = "Avkorter et tall til et heltall ved å fjerne brøkdelen av tallet.";
        this["Func_type_Arg_0"] = "verdi";
        this["Func_type_ArgDesc_0"] = "Verdien for å få typen.";
        this["Func_type_Category"] = "Information";
        this["Func_type_CategoryURL"] = "Information";
        this["Func_type_Desc"] = "Returnerer et tall som representerer verdien av verdien: number = 1; tekst = 2; logisk verdi (sann eller usann) = 4; feil verdi = 16; array = 64.";
        this["Func_upper_Arg_0"] = "tekst";
        this["Func_upper_ArgDesc_0"] = "Teksten som skal konverteres til store bokstaver.";
        this["Func_upper_Category"] = "TextAndData";
        this["Func_upper_CategoryURL"] = "Text_and_Data";
        this["Func_upper_Desc"] = "Konverterer alle små bokstaver i et tekstuttrykk til store bokstaver.";
        this["Func_value_Arg_0"] = "tekst";
        this["Func_value_ArgDesc_0"] = "Tekstuttrykket som skal konverteres.";
        this["Func_value_Category"] = "TextAndData";
        this["Func_value_CategoryURL"] = "Text_and_Data";
        this["Func_value_Desc"] = "Konverterer et tekstuttrykk som representerer et tall til et tall.";
        this["Func_var_Arg_0"] = "number{0}";
        this["Func_var_ArgDesc_0"] = "Et tall eller rekkevidde som angir et utvalg av en populasjon.";
        this["Func_var_Category"] = "Statistical";
        this["Func_var_CategoryURL"] = "Statistical";
        this["Func_var_Desc"] = "Returnerer variansen til et populasjonsutvalg.";
        this["Func_weekday_Arg_0"] = "Dato";
        this["Func_weekday_Arg_1"] = "ReturnType";
        this["Func_weekday_ArgDesc_0"] = "Datoen for å finne ukedagen.";
        this["Func_weekday_ArgDesc_1"] = "Bestemmer hvilket nummereringsskjema som brukes for ukedagene.\r\n1 (standard) = søndag (1) til lørdag (7)\r\n2 = mandag (1) til søndag (7)\r\n3 = mandag (0) til søndag (6)\r\n    ";
        this["Func_weekday_Category"] = "DateAndTime";
        this["Func_weekday_CategoryURL"] = "DateAndTime";
        this["Func_weekday_Desc"] = "Returnerer et tall som representerer ukedagen for den angitte datoen.";
        this["Func_weeknum_Arg_0"] = "date";
        this["Func_weeknum_Arg_1"] = "returnType";
        this["Func_weeknum_ArgDesc_0"] = "Datoen for å finne ukenummeret.";
        this["Func_weeknum_ArgDesc_1"] = "Bestemmer hvilken dag som begynner uken.\r\n1 (standard) = Søndag er den første dagen i uken.\r\n2 = Mandag er den første dagen i uken.\r\n    ";
        this["Func_weeknum_Category"] = "DateAndTime";
        this["Func_weeknum_CategoryURL"] = "DateAndTime";
        this["Func_weeknum_Desc"] = "Returnerer nummeret på uken som den angitte datoen faller inn i.";
        this["Func_workday_Arg_0"] = "startDate";
        this["Func_workday_Arg_1"] = "dager";
        this["Func_workday_Arg_2"] = "holidays";
        this["Func_workday_ArgDesc_0"] = "Datoen for start.";
        this["Func_workday_ArgDesc_1"] = "Antall arbeidsdager som skal forskyves fra startdatoen.";
        this["Func_workday_ArgDesc_2"] = "Et område med en eller flere referanser som inneholder datoer som skal betraktes som arbeidsdager.";
        this["Func_workday_Category"] = "DateAndTime";
        this["Func_workday_CategoryURL"] = "DateAndTime";
        this["Func_workday_Desc"] = "Returnerer en arbeidsdag det angitte antall dager fra den angitte datoen, unntatt eventuelle spesifiserte helligdager.";
        this["Func_year_Arg_0"] = "Date_value";
        this["Func_year_ArgDesc_0"] = "En datoverdi for å få året.";
        this["Func_year_Category"] = "DateAndTime";
        this["Func_year_CategoryURL"] = "Date_and_Time";
        this["Func_year_Desc"] = "Returnerer året for en datoverdi.";
        this["Func_yearfrac_Arg_0"] = "StartDate";
        this["Func_yearfrac_Arg_1"] = "EndDate";
        this["Func_yearfrac_Arg_2"] = "Basis";
        this["Func_yearfrac_ArgDesc_0"] = "Startdato er startdatoen i området.";
        this["Func_yearfrac_ArgDesc_1"] = "Sluttdato er sluttdatoen i området.";
        this["Func_yearfrac_ArgDesc_2"] = "\r\n      Basis bestemmer grunnlaget for dagtellingen som skal brukes.\r\n0 (Standard = USA (NASD) 30/360\r\n1 = Faktisk/faktisk\r\n2 = Faktisk/360\r\n3 = Faktisk/365\r\n4 = Europeisk 30/360\r\n    ";
        this["Func_yearfrac_Category"] = "DateAndTime";
        this["Func_yearfrac_CategoryURL"] = "DateAndTime";
        this["Func_yearfrac_Desc"] = "Returnerer en brøkdel som representerer den delen av et år som er angitt mellom den angitte startdatoen og sluttdatoen.";
        this["GenerateTableColumnName"] = "Kolonne";
        this["GenerateTableName"] = "Tabell";
        this["LD_Chart_AxisDisplayUnits"] = "Hundreds:Hundreds, Thousands:Thousands, TenThousands:x 10000, HundredThousands:x 100000, Millions:Millions, TenMillions:x 10000000, HundredMillions:x 100000000, ThousandMillions:Billions, MillionMillions:Trillions";
        this["LD_Fallback_ChartEx_Line1"] = "Dette diagrammet er ikke tilgjengelig i din versjon av Excel.";
        this["LD_Fallback_ChartEx_Line2"] = "Hvis du redigerer denne formen eller lagrer denne arbeidsboken i et annet filformat, blir diagrammet permanent ødelagt.";
        this["LE_AllMustHavGuidsIfAnyHasThem"] = "Kan ikke sette GenerateGuidsForGroups til false hvis en eksisterende gruppe har en Guid.";
        this["LE_ArgumentException_AnchorCellFromOtherWorksheet"] = "Cellen som er angitt finnes ikke på det samme regnearket som formen.";
        this["LE_ArgumentException_ArrayFormulaMustHaveSingleRegion"] = "En matriseformel kan bare brukes på ett celleområde.";
        this["LE_ArgumentException_ArrayTooSmall"] = "Tabellen er ikke stor nok til at alle verdier kan kopieres til den.";
        this["LE_ArgumentException_CannotAddStandardTableStyle"] = "En standard tabellstil kan ikke legges til i CustomTableStyleCollection.";
        this["LE_ArgumentException_CannotApplyDVRuleToTotalCell"] = "En eller flere celler i den angitte referansen er totale celler i en tabell og kan ikke ha en datavalideringsregel.";
        this["LE_ArgumentException_CannotCreateEmptyColorInfo"] = "Den tomme fargen kan ikke brukes til å opprette en WorkbookColorInfo-forekomst.";
        this["LE_ArgumentException_CellMustBeOnSameWorksheetAsReferencesCollection"] = "Cellen må være fra det samme regnearket som referansesamlingen.";
        this["LE_ArgumentException_CellShiftedOffWorksheet"] = "Cellen har blitt slettet eller flyttet fra regnearket og representerer ikke lenger en regnearkposisjon.";
        this["LE_ArgumentException_CellsInTableFromOtherWorksheet"] = "Området som er angitt er ikke fra det samme regnearket som datatabellen.";
        this["LE_ArgumentException_CellValueStringLength"] = "Strengverdier som er tilordnet en celle, kan ikke ha lengde større enn {0}.";
        this["LE_ArgumentException_ColumnFormulaAlreadyAppliedToCell"] = "Formelen brukes allerede på en celle. Klon den for å bruke den som kolonneformel.";
        this["LE_ArgumentException_ColumnFromDifferentWorksheet"] = "Kolonnen tilhører ikke det samme regnearket som denne samlingen.";
        this["LE_ArgumentException_ColumnInputCellFromOtherWorksheet"] = "Den angitte kolonneinngangscellen eksisterer ikke i det samme regnearket som datatabellen.";
        this["LE_ArgumentException_ColumnNotInTable"] = "Kolonnen er fra en annen tabell.";
        this["LE_ArgumentException_ColumnRemovedFromWorksheet"] = "Kolonnen er fjernet fra regnearket.";
        this["LE_ArgumentException_ColumnsToRepeatAtLeftOutsideRange"] = "Området ligger utenfor området med tilgjengelige kolonner i regnearket.";
        this["LE_ArgumentException_CommentTooLong"] = "Kommentarlengden må være mindre enn 255 tegn.";
        this["LE_ArgumentException_CustomViewNameAlreadyExists"] = "Det finnes allerede en tilpasset visning med navnet \"{0}\".";
        this["LE_ArgumentException_DefaultTableStyleNotInWorkbook"] = "Den angitte tabellstilen er ikke i samlingene CustomTableStyles eller StandardTableStyles.";
        this["LE_ArgumentException_DisplayTextTooLong"] = "Visningsteksten for hyperlinken må inneholde mindre enn 256 tegn.";
        this["LE_ArgumentException_DuplicateDisplayValue"] = "Den angitte visningsverdien er allerede i samlingen.";
        this["LE_ArgumentException_DuplicateFixedDateGroup"] = "Den angitte FixedDateGroup er allerede i samlingen.";
        this["LE_ArgumentException_DuplicateTableStyle"] = "Den angitte stilen er allerede i en CustomTableStyleCollection.";
        this["LE_ArgumentException_DuplicateTableStyleName_Existing"] = "Det angitte navnet som samsvarer med navnet på en annen stil i samlingen.";
        this["LE_ArgumentException_DuplicateTableStyleName_New"] = "Den angitte stilen har et navn som matcher en annen stil i samlingen.";
        this["LE_ArgumentException_DuplicateWorksheetProtectedRangeTitle"] = "Et område med tittelen {0} eksisterer allerede.";
        this["LE_ArgumentException_DV_ArrayFormulaCannotBeUsed"] = "En ArrayFormula kan ikke brukes som {0}.";
        this["LE_ArgumentException_DV_CellFromOtherWorksheet"] = "Cellen tilhører ikke det samme regnearket som samlingen av datavalideringsregler.";
        this["LE_ArgumentException_DV_ConstraintFormulaMustBeNumberOrReference"] = "{0} må settes til et tall, et numerisk uttrykk eller en referanse til en enkelt celle.";
        this["LE_ArgumentException_DV_FormulaCannotBeNull"] = "{0} kan ikke settes til null mens datavalideringsregelen brukes på et regneark.";
        this["LE_ArgumentException_DV_FormulaCannotFindNamedReference"] = "{0} inneholder en navngitt referanse som ikke finnes i arbeidsboken.";
        this["LE_ArgumentException_DV_FormulaCannotFindWorksheetReference"] = "{0} inneholder en referanse på et regneark som ikke finnes i arbeidsboken.";
        this["LE_ArgumentException_DV_FormulaCannotReferenceOtherWorkbook"] = "{0} kan ikke inneholde en referanse til en ekstern arbeidsbok.";
        this["LE_ArgumentException_DV_FormulaInvalidForWorkbookFormat"] = "{0}-referansecellene som ikke er gyldige i {1} arbeidsbokformatet.";
        this["LE_ArgumentException_DV_InvalidAddress"] = "Den angitte adressen er ikke en gyldig celle- eller områdeadresse.";
        this["LE_ArgumentException_DV_InvalidExcelDate"] = "Den angitte datoen kan ikke vises i Excel.";
        this["LE_ArgumentException_DV_InvalidFormula"] = "Ugyldig formel brukt i datavalideringsregelen.";
        this["LE_ArgumentException_DV_InvalidMessageLength"] = "{0} kan ikke inneholde mer enn {1} tegn.";
        this["LE_ArgumentException_DV_ListFormulaCannotHaveEmptyString"] = "Strengkonstanten i ValuesFormula kan ikke være en tom streng.";
        this["LE_ArgumentException_DV_ListFormulaMustBeStringOrReference"] = "ValuesFormula må settes til en strengkonstant eller en referanse til en enkelt celle eller et område i den samme arbeidsboken.";
        this["LE_ArgumentException_DV_ListFormulaReferenceMustBeOneDimensional"] = "ValuesFormula inneholder en områdereferanse som har flere rader og kolonner. Området må inneholde en rad eller en kolonne.";
        this["LE_ArgumentException_DV_LowerGreaterThanUpperContraint"] = "Verdien av den nedre begrensningen må være mindre enn eller lik den øvre begrensningen.";
        this["LE_ArgumentException_DV_MustHaveOneAcceptedValue"] = "Minst én verdi må spesifiseres som en akseptert verdi.";
        this["LE_ArgumentException_DV_ReferencesAlreadyContainDVs"] = "De angitte referansene inneholder allerede en eller flere datavalideringsregler.";
        this["LE_ArgumentException_DV_ReferencesFromOtherWorksheet"] = "Referansesamlingen tilhører ikke det samme regnearket som samlingen av datavalideringsregler.";
        this["LE_ArgumentException_DV_RegionFromOtherWorksheet"] = "Området tilhører ikke det samme regnearket som samlingen av datavalideringsregler.";
        this["LE_ArgumentException_DV_RuleAppliedOnOtherWorksheet"] = "Datavalideringsregelen brukes allerede på et annet regneark.";
        this["LE_ArgumentException_DV_UpperLessThanLowerContraint"] = "Verdien av den øvre grensen må være større enn eller lik den nedre grensen.";
        this["LE_ArgumentException_EditRangeAlreadyOwned"] = "Området er allerede i et regnearks AllowedEditRanges.";
        this["LE_ArgumentException_EditRangeHasRangeFromOtherWorksheet"] = "Redigeringsområdet har et område fra et annet regneark enn AllowedEditRanges.";
        this["LE_ArgumentException_EmptyCustomList"] = "Det må være minst én streng på listen.";
        this["LE_ArgumentException_EndIndexLessThanZero"] = "Siste delen av utvalget må være større enn eller lik null.";
        this["LE_ArgumentException_FileDoesntContainsWorkbookStream"] = "Den angitte strømmen eller filen inneholder ikke en gyldig arbeidsbok.";
        this["LE_ArgumentException_FormulaReferencesInvalidCells"] = "Den angitte formelen refererer til celler som ikke støttes i arbeidsbokens nåværende filformat.";
        this["LE_ArgumentException_HashSetFull"] = "Det er for mange elementer i hash-settet.";
        this["LE_ArgumentException_HyperlinkAlreadyOwned"] = "Hyperlenken tilhører allerede en hyperkoblingssamling.";
        this["LE_ArgumentException_HyperlinkSourceFromOtherWorksheet"] = "Hyperlenken har en kildecelle eller region fra et annet regneark.";
        this["LE_ArgumentException_HyperlinkTargetFromOtherWorkbook"] = "Målet er fra en annen arbeidsbok enn eieren av hyperkoblingen.";
        this["LE_ArgumentException_InfiniteColumnWidth"] = "Kolonnebredden kan ikke være uendelig.";
        this["LE_ArgumentException_Interval"] = "Intervallparameteren bør være en av verdiene i DateInterval-opptellingen";
        this["LE_ArgumentException_IntervalStr"] = "Intervallparameteren kunne ikke konverteres til en av verdiene i DateInterval-opptellingen";
        this["LE_ArgumentException_InvalidCellAddress"] = "Den angitte adressen er ikke en gyldig celleadresse.";
        this["LE_ArgumentException_InvalidCommentPositioningMode"] = "Bare DontMoveOrSizeWithCells og MoveAndSizeWithCells ShapePositioningModes kan brukes på en kommentarform.";
        this["LE_ArgumentException_InvalidCustomFilterOperandNumber"] = "Verdien kan ikke være uendelig eller NaN.";
        this["LE_ArgumentException_InvalidCustomFilterOperator"] = "Når sammenligningsverdien ikke er en streng, kan sammenligningsoperatoren ikke være BeginsWith, DoesNotBeginWith, Contains, DoesNotContain, EndsWith eller DoesNotEndWith.";
        this["LE_ArgumentException_InvalidDatePeriodFilterValue"] = "Verdien er ikke innenfor området. For filtertype Month må den være 1-12. For filtertype Quarter må den være 1-4.";
        this["LE_ArgumentException_InvalidDpi"] = "Den angitte Dpi-verdien er ugyldig.";
        this["LE_ArgumentException_InvalidExcelDate"] = "Verdien kan ikke uttrykkes som en dato i Excel.";
        this["LE_ArgumentException_InvalidFileFormat"] = "Ugyldig eller ukjent filformat.";
        this["LE_ArgumentException_InvalidFormula"] = "Den angitte formelen var ugyldig.";
        this["LE_ArgumentException_InvalidGradientStopColor"] = "Automatiske farger og systemfarger kan ikke brukes i gradientstopp.";
        this["LE_ArgumentException_InvalidNamedReferenceName"] = "Navnet er ikke et gyldig navngitt referansenavn.";
        this["LE_ArgumentException_InvalidPaletteColor_EmptyOrSystem"] = "Tomme farger eller systemfarger kan ikke legges til paletten.";
        this["LE_ArgumentException_InvalidPaletteColor_NonOpaque"] = "Bare ugjennomsiktige farger kan legges til paletten.";
        this["LE_ArgumentException_InvalidReferencesString"] = "Referanseverdien er ikke riktig formatert.";
        this["LE_ArgumentException_InvalidRegionAddress"] = "Den angitte adressen er ikke en gyldig områdeadresse.";
        this["LE_ArgumentException_InvalidRowOrColumnRange"] = "Ugyldig rekke- eller kolonneområde.";
        this["LE_ArgumentException_InvalidSeriesValues"] = "Ugyldige serieverdier. Det må være minst én verdi, og hver verdi må enten være en streng eller et objekt som kan konverteres til en dobbel.";
        this["LE_ArgumentException_InvalidTarget"] = "Hyperkoblingsmålet må være en streng, WorksheetCell, WorksheetRegion eller NamedReference.";
        this["LE_ArgumentException_InvalidTopOrBottomFilterValue"] = "Verdien må være mellom 1 og 500.";
        this["LE_ArgumentException_InvalidWorksheetName"] = "Navnet på regnearket kan ikke inneholde noen av følgende tegn:\r\n: \\ / ? * [ ]";
        this["LE_ArgumentException_InvalidWorksheetNameStartingQuote"] = "Regnearkets navn kan ikke starte eller slutte med et enkelt sitat.";
        this["LE_ArgumentException_LessThanTwoGradientStops"] = "Det må være angitt minst to gradientstopp.";
        this["LE_ArgumentException_NamedReferenceNameTooLong"] = "Navnet på den navngitte referansen kan ikke være mer enn 255 tegn.";
        this["LE_ArgumentException_NaNDefaultColumnWidth"] = "NaN kan ikke brukes som standard kolonnebredde.";
        this["LE_ArgumentException_NewRegionOverlapsFilterArea"] = "Operasjonen prøver å endre et filtrert område i regnearket, og kan ikke fullføres. For å fullføre denne operasjonen må AutoFilters i arket fjernes.";
        this["LE_ArgumentException_NewTableRegionCannotMoveHeaders"] = "Det nye tabellområdet kan ikke flytte overskriftene til en ny rad.";
        this["LE_ArgumentException_NewTableRegionFromWrongWorksheet"] = "Det angitte området er ikke fra det samme regnearket som tabellen.";
        this["LE_ArgumentException_NewTableRegionMustHaveDataRows"] = "Det nye tabellområdet må inneholde minst én datarad.";
        this["LE_ArgumentException_NewTableRegionOverlapOld"] = "Det nye tabellområdet må overlappe med det forrige tabellområdet.";
        this["LE_ArgumentException_NewTableRegionOverlapsBlockingValue"] = "Tabellen kan ikke endres til den angitte området fordi den inneholder en formel for flere celler eller datatabell.";
        this["LE_ArgumentException_NewTableRegionOverlapsMergedRegion"] = "Tabellen kan ikke endres til det angitte området fordi den inneholder en flettet celle.";
        this["LE_ArgumentException_NewTableRegionOverlapsOtherTable"] = "Tabellen kan ikke endres til det angitte området fordi den inneholder en annen tabell.";
        this["LE_ArgumentException_NoRegionsInArray"] = "Den angitte matrisen har ingen regioner.";
        this["LE_ArgumentException_NotEnum"] = "Type '{0}' er ikke et nummer";
        this["LE_ArgumentException_NotSupportedCodePage"] = "Kodeside støttes ikke";
        this["LE_ArgumentException_NPer"] = "nPer kan ikke være 0";
        this["LE_ArgumentException_NPerFunction"] = "Feil kombinasjon av parametere for NPer-funksjonen";
        this["LE_ArgumentException_PA_PrintAreasCannotOverlap"] = "Det angitte utskriftsområdet overlapper med et av de eksisterende utskriftsområdene.";
        this["LE_ArgumentException_ParentStyleFromOtherWorkbook"] = "Den angitte stilen er ikke fra samme arbeidsbok som celleformatet.";
        this["LE_ArgumentException_PB_CantInsertBreakAtIndex"] = "Denne samlingen støtter ikke innsetting av elementer i en indeks.";
        this["LE_ArgumentException_PB_CantSetBreakAtIndex"] = "Denne samlingen støtter ikke stilling av elementer i en indeks.";
        this["LE_ArgumentException_PB_MustBeWithinPrintArea"] = "Sideskiftet må være innenfor utskriftsområdet.";
        this["LE_ArgumentException_PB_PageBreakCantBeA1Cell"] = "Kan ikke sette inn et sideskift øverst til venstre i regnearket.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeLeftColumn"] = "Kan ikke sette inn et sideskift til venstre i regnearket.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeTopRow"] = "Kan ikke sette inn et sideskift øverst i regnearket.";
        this["LE_ArgumentException_PB_PageBreaksCannotAddExistingInstance"] = "Den angitte forekomsten finnes allerede i samlingen.";
        this["LE_ArgumentException_PB_PageBreaksCannotOverlap"] = "Sideskiftet kan ikke overlappe med andre sideskift i regnearket.";
        this["LE_ArgumentException_PB_PrintAreaMustBeInPrintAreas"] = "Sideskiftet er i et utskriftsområde som ikke er i PrintAreas-samlingen.";
        this["LE_ArgumentException_PB_PrintAreaMustBeOnSameWorksheet"] = "Sideskiftet er i et utskriftsområde som ikke er i det samme regnearket som sideskiftsamlingen.";
        this["LE_ArgumentException_Pmt"] = "Pmt kunne ikke være 0";
        this["LE_ArgumentException_PrintAreaShiftedOffWorksheet"] = "Sideskiftet er i et utskriftsområde som har blitt flyttet fra regnearket og som ikke lenger representerer et gyldig celleområde.";
        this["LE_ArgumentException_RangeFromOtherWorksheetThanProtectedRange"] = "Det angitte området er fra et annet regneark enn redigeringsområdet.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheet"] = "Området må være fra det samme regnearket.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheetAsReferencesCollection"] = "Området må være fra det samme regnearket som referansesamlingen.";
        this["LE_ArgumentException_RegionsFromMixedWorksheets"] = "Alle regioner som er angitt i matrisen, må være fra det samme regnearket.";
        this["LE_ArgumentException_RegionShiftedOffWorksheet"] = "Området er flyttet fra regnearket og representerer ikke lenger et gyldig celleområde.";
        this["LE_ArgumentException_RegionsShiftedOffWorksheet"] = "En eller flere spesifiserte regioner er flyttet fra regnearket og representerer ikke lenger et gyldig celleområde.";
        this["LE_ArgumentException_RowFromDifferentWorksheet"] = "Raden tilhører ikke det samme regnearket som denne samlingen.";
        this["LE_ArgumentException_RowInputCellFromOtherWorksheet"] = "Den angitte radinngangscellen eksisterer ikke i det samme regnearket som datatabellen.";
        this["LE_ArgumentException_RowRemovedFromWorksheet"] = "Raden er fjernet fra regnearket.";
        this["LE_ArgumentException_SelectedWorksheetFromOtherWorkbook"] = "Regnearket tilhører ikke arbeidsboken som er knyttet til disse vindusalternativene.";
        this["LE_ArgumentException_ShapeCannotBeAdded"] = "Den angitte formen kan ikke legges til i en formsamling.";
        this["LE_ArgumentException_SourceFromOtherWorksheetThanHyperlink"] = "Den angitte kildecellen eller regionen er fra et annet regneark enn hyperkoblingen.";
        this["LE_ArgumentException_StartIndexLessThanZero"] = "Starten på området må være større enn eller lik null.";
        this["LE_ArgumentException_StyleNameAlreadyExists"] = "En stil med navnet \"{0}\" finnes allerede i arbeidsboken.";
        this["LE_ArgumentException_StyleNameTooLong"] = "Stilnavnet kan ikke inneholde mer enn 255 tegn.";
        this["LE_ArgumentException_TableColumnRemovedFromTable"] = "Tabellkolonnen er fjernet fra tabellen.";
        this["LE_ArgumentException_TableStyleFromOtherWorkbook"] = "Det angitte WorksheetTableStyle tilhører ikke den samme arbeidsboken som tabellen.";
        this["LE_ArgumentException_TargetTooLong"] = "Måladressen for hyperkoblingen må inneholde mindre enn 256 tegn.";
        this["LE_ArgumentException_TextLengthGreaterThanMax"] = "Lengden på teksten i en figur eller kommentar kan ikke være større enn {0}.";
        this["LE_ArgumentException_ToolTipTooLong"] = "Tipset for hyperkoblingsverktøyet må inneholde mindre enn 256 tegn.";
        this["LE_ArgumentException_TopLeftWindowBounds"] = "Venstre grense og toppen på vindusgrensene må være mellom-32768 og 32767.";
        this["LE_ArgumentException_WeekdayFirst"] = "WeekdayFirst-parameteren skal være mellom FirstDayOfWeek.System og FirstDayOfWeek.Saturday";
        this["LE_ArgumentException_WidthHeightWindowBounds"] = "Bredden og høyden på vindusgrensene må være mellom 0 og 65535.";
        this["LE_ArgumentException_WorksheetNameAlreadyExists"] = "Et regneark med navnet \"{0}\" finnes allerede i arbeidsboken.";
        this["LE_ArgumentException_WorksheetNameTooLong"] = "Regnearkets navn må være mellom 1 og 31 tegn langt.";
        this["LE_ArgumentException_WorksheetProtectedRangeTitleTooLong"] = "Tittelen på et redigeringsområde kan ikke være mer enn 255 tegn.";
        this["LE_ArgumentException_WorksheetScopeFromOtherWorkbook"] = "Regnearket tilhører ikke den samme arbeidsboken som denne samlingen.";
        this["LE_ArgumentNullException_AnchorCell"] = "Cellen kan ikke være null når formen finnes på et regneark eller en gruppe.";
        this["LE_ArgumentNullException_CustomViewName"] = "Det egendefinerte visningsnavnet kan ikke være null eller tomt.";
        this["LE_ArgumentNullException_Encoder"] = "Bildekoderen kan ikke være null.";
        this["LE_ArgumentNullException_FindNamedReference"] = "Navnet som skal finnes må ikke være null eller tomt.";
        this["LE_ArgumentNullException_FormulaCantBeNull"] = "Formelen for en navngitt referanse kan ikke være null eller tom.";
        this["LE_ArgumentNullException_HiddenColumn"] = "En null-kolonne kan ikke legges til i samlingen.";
        this["LE_ArgumentNullException_HiddenRow"] = "En null-rad kan ikke legges til i samlingen.";
        this["LE_ArgumentNullException_Image"] = "Bildet kan ikke være null.";
        this["LE_ArgumentNullException_NamedReferenceNameCantBeNull"] = "Navnet på en navngitt referanse kan ikke være null eller tomt.";
        this["LE_ArgumentNullException_SaveStream"] = "Kan ikke lagre arbeidsboken i en nullstrøm.";
        this["LE_ArgumentNullException_SelectedWorksheet"] = "Arbeidsboken må ha et gyldig valgt regneark.";
        this["LE_ArgumentNullException_Shape"] = "En nullform kan ikke legges til i samlingen.";
        this["LE_ArgumentNullException_SourceFont"] = "Kildeskriften kan ikke være null.";
        this["LE_ArgumentNullException_SourceFormatting"] = "Kildeformateringen kan ikke være null.";
        this["LE_ArgumentNullException_StyleName"] = "Stilnavnet kan ikke være null eller tomt.";
        this["LE_ArgumentNullException_UnformattedString"] = "Den uformaterte strengen kan ikke være null.";
        this["LE_ArgumentNullException_Workbook"] = "Arbeidsboken kan ikke være null.";
        this["LE_ArgumentNullException_WorkbookRequiredToResolveThemeColor"] = "En arbeidsbok må spesifiseres for å løse en temafarge.";
        this["LE_ArgumentNullException_WorksheetName"] = "Regnearkets navn kan ikke være null eller tomt.";
        this["LE_ArgumentNullException_WorksheetProtectedRangeTitleCantBeNull"] = "Tittelen på et redigeringsområde kan ikke være null eller tomt.";
        this["LE_ArgumentNullException_WorksheetScope"] = "Regnearkets omfang kan ikke være null.";
        this["LE_ArgumentOutOfRange_RelativeColumnIndex"] = "Den relative kolonneindeksen er utenfor det angitte området.";
        this["LE_ArgumentOutOfRange_RelativeRowIndex"] = "Den relative radindeksen er utenfor det angitte området.";
        this["LE_ArgumentOutOfRangeException_AnchorPosition"] = "En av prosentene på det angitte stedet er mindre enn 0.";
        this["LE_ArgumentOutOfRangeException_BadInsertIndex"] = "Den angitte indeksen må være større enn eller lik null og mindre enn eller lik antallet eksisterende avsnitt.";
        this["LE_ArgumentOutOfRangeException_BadRemoveAtIndex"] = "Indeksen må være større enn null og mindre enn antall avsnitt i den formaterte strengen.";
        this["LE_ArgumentOutOfRangeException_CollectionIndex"] = "Indeksen var utenfor rekkevidde. Den må være positiv og mindre enn størrelsen på samlingen.";
        this["LE_ArgumentOutOfRangeException_ColumnWidth"] = "Kolonnebredden må være mellom 0 og 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultColumnWidth"] = "Standard kolonnebredde må være mellom 0 og 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultFontHeight"] = "Standard skrifthøyde må være mellom 20 og 8180.";
        this["LE_ArgumentOutOfRangeException_DefaultRowHeight"] = "Standard radhøyde må være mellom-1 og 8192.";
        this["LE_ArgumentOutOfRangeException_DuplicateItemSorted"] = "Det angitte elementet har allerede en sorteringsbetingelse.";
        this["LE_ArgumentOutOfRangeException_FirstVisibleTabIndex"] = "Den første synlige faneindeksen må være større enn eller lik null.";
        this["LE_ArgumentOutOfRangeException_FontHeight"] = "Skrifthøyden må være mellom 20 og 8180.";
        this["LE_ArgumentOutOfRangeException_GroupAddedToSelf"] = "En gruppe kan ikke legges til i sin egen samling av former.";
        this["LE_ArgumentOutOfRangeException_Indent"] = "Innrykknivået må være mellom 0 og 250.";
        this["LE_ArgumentOutOfRangeException_IndexNegative"] = "Indeksen må være ikke-negativ.";
        this["LE_ArgumentOutOfRangeException_InvalidCollectionIndex"] = "Den angitte indeksen er ikke gyldig i samlingen.";
        this["LE_ArgumentOutOfRangeException_InvalidColorInfoTint"] = "Fargetonen må være mellom -1,0 og 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnCount"] = "{0} er ikke et gyldig kolonneantall i det nåværende formatet. Verdien må være mellom {1} og {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnIndex"] = "{0} er ikke en gyldig kolonneindeks i det nåværende formatet. Verdien må være mellom {1} og {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidGradientStopOffset"] = "Forskyvningen av gradientstopp må være mellom 0,0 og 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRelativeRectangleValueForGradient"] = "Verdiene for det indre rektangelet til den rektangulære gradienten må være mellom 0,0 og 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRemoveAtIndex_SortConditions"] = "Indeksen der sorteringsbetingelsen skal fjernes, er utenfor området.";
        this["LE_ArgumentOutOfRangeException_InvalidRowCount"] = "{0} er ikke et gyldig radantall i det gjeldende formatet. Verdien må være mellom {1} og {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidRowIndex"] = "{0} er ikke en gyldig radindeks i det nåværende formatet. Verdien må være mellom {1} og {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidTableStyleAreaStripeExtent"] = "{0} må være mellom 1 og 9, inkludert.";
        this["LE_ArgumentOutOfRangeException_LeftPaneWidth"] = "Den angitte rutebredden er utenfor det gyldige området {0} og {1}.";
        this["LE_ArgumentOutOfRangeException_LengthMustBePositive"] = "Lengden må være større enn eller lik en.";
        this["LE_ArgumentOutOfRangeException_MagnificationLevel"] = "Visningsforstørrelsesnivåer må være mellom 10 og 400.";
        this["LE_ArgumentOutOfRangeException_Margins"] = "Marginene må være mellom 0 og 100.";
        this["LE_ArgumentOutOfRangeException_MaxPagesHorizontally"] = "Maksimalt antall sider horisontalt må være mellom 0 og 32767.";
        this["LE_ArgumentOutOfRangeException_MaxPagesVertically"] = "Maksimalt antall sider vertikalt må være mellom 0 og 32767.";
        this["LE_ArgumentOutOfRangeException_MaxRecursionIterations"] = "Maksimal gjentakelse av rekursjoner må være mellom 1 og 32767.";
        this["LE_ArgumentOutOfRangeException_MaxSortConditions"] = "Det kan ikke være mer enn {0} sorteringsbetingelser i samlingen.";
        this["LE_ArgumentOutOfRangeException_NegativeStartIndex"] = "Startindeksen kan ikke være negativ.";
        this["LE_ArgumentOutOfRangeException_NumberOfCopies"] = "Antall kopier må være mellom 1 og 65535.";
        this["LE_ArgumentOutOfRangeException_Per"] = "per verdi skal være mellom 0 og nPer + 1";
        this["LE_ArgumentOutOfRangeException_Rate"] = "Raten bør være større enn-1";
        this["LE_ArgumentOutOfRangeException_Resolution"] = "Oppløsningen må være mellom 0 og 65535.";
        this["LE_ArgumentOutOfRangeException_RowHeight"] = "Radhøyden må være mellom 0 og 8192.";
        this["LE_ArgumentOutOfRangeException_ScalingFactor"] = "Skaleringsfaktoren må være mellom 10 og 400.";
        this["LE_ArgumentOutOfRangeException_StartPageNumber"] = "Startsidenummeret må være mellom-32765 og 32767.";
        this["LE_ArgumentOutOfRangeException_TabBarWidth"] = "Tabulatorbredden må være mellom 0 og 1000.";
        this["LE_ArgumentOutOfRangeException_TopPaneHeight"] = "Rutehøyden som er angitt er utenfor det gyldige området {0} og {1}.";
        this["LE_AutoColorNotAllowed"] = "Fargen kan ikke være 'Auto'.";
        this["LE_Axis_MajorMinorUnit"] = "Verdien til parameteren majorUnit må være større enn verdien til parameteren minorUnit.";
        this["LE_Axis_NoCrossAxis"] = "Hvis CrossAxis-egenskapen er angitt på ett Axis-objekt, må den angis på dem alle. Hvis diagrammet ditt bare har to akser, kan kryssingsaksen bestemmes automatisk, men bare hvis CrossAxis-egenskapen ikke er angitt på alle Axis-objekter.";
        this["LE_Biff8SerializerNotLoaded"] = "'{0}'-formatet krever en serializer som ikke er lastet. I JavaScript finnes dette i filen infragistics.excel_serialization_biff8.js";
        this["LE_CategoryAxisBinning_BinWidthAndNumberOfBins"] = "Parametrene 'binWidth' og 'numberOfBins' utelukker hverandre. Begge kan være uspesifiserte, men hvis den ene er spesifisert, må den andre ikke være det.";
        this["LE_ChartGradientFill_EmptyStops"] = "Må inneholde minst ett gradientstopp.";
        this["LE_ChartObject_DifferentChart"] = "Objektforekomsten som tildeles er knyttet til et annet diagram, og kan ikke tilordnes denne forekomsten.";
        this["LE_ColumnsToRepeatAtLeftAreOutsideAvailableRange"] = "Utskriftsalternativene har kolonner som gjentas til venstre som ligger utenfor området med tilgjengelige kolonner i det nye formatet.";
        this["LE_ComboChartGroupCollection_DoesNotExist"] = "Den angitte ComboChartGroup finnes ikke i denne samlingen.";
        this["LE_FillPercentInvalid"] = "Verdien {0} er ikke en gyldig verdi for {1}-området.";
        this["LE_FormatLimitError_Indent"] = "Innrykket overskrider maksverdien på {0} gitt av det gjeldende formatet.";
        this["LE_FormatLimitError_MacroWorkbook"] = "Arbeidsboken har VBA-kode og kan ikke lagres som en makroaktivert arbeidsbok.";
        this["LE_FormatLimitError_MaxColumnIndex"] = "Kolonneindeksen til {0} er større enn maksimumet på {1} som støttes av det angitte formatet.";
        this["LE_FormatLimitError_MaxRowIndex"] = "Radindeksen for {0} overstiger maksimumet på {1} som støttes av dette formatet.";
        this["LE_FormulaParseException_ArrayContainsConstants"] = "En matrise kan bare inneholde konstante uttrykk.";
        this["LE_FormulaParseException_ArrayHasEmptyFirstRow"] = "En konstant matrise i formelen har ingen elementer i den første raden.";
        this["LE_FormulaParseException_ArrayHasMisalignedRows"] = "En konstant matrise i formelen har feiljusterte rader. Hver rad i en matrise må ha samme antall kolonner.";
        this["LE_FormulaParseException_CellReferenceAfterWorkbookName"] = "En ekstern arbeidsbokreferanse må følges av en navngitt referanse, ikke en cellereferanse.";
        this["LE_FormulaParseException_ExtraExpressions"] = "Formelen har ekstra uttrykk etter seg.";
        this["LE_FormulaParseException_FunctionMissingClosingParen"] = "En funksjon i formelen har ingen avsluttende parentes.";
        this["LE_FormulaParseException_FunctionNestingTooDeep"] = "En funksjon i formelen er nøstet for dypt i andre funksjoner.";
        this["LE_FormulaParseException_IncorrectNumberOfArguments"] = "En funksjon i formelen har ikke fått riktig antall argumenter.";
        this["LE_FormulaParseException_InvalidArguments"] = "En operator eller funksjon i formelen har ugyldige argumenter.";
        this["LE_FormulaParseException_InvalidConditionalFormatFormula"] = "En betinget formatformel kan ikke bruke referanseoperatorer som unioner eller snitt, matrisekonstanter eller strukturerte tabellreferanser.";
        this["LE_FormulaParseException_InvalidErrorValue"] = "Feiltypen som er angitt, er ikke gyldig. Det må være en av følgende verdier: \"#NULL!\", \"#DIV/0!\", \"#VALUE!\", \"#REF!\", \"#NAME?\", \"#NUM!\", Eller \"# N/A\".";
        this["LE_FormulaParseException_InvalidFileNameInBrackets"] = "Et filnavn mellom hakeparenteser var ugyldig.";
        this["LE_FormulaParseException_InvalidStructuredTableReference"] = "Den strukturerte tabellreferansen er feil.";
        this["LE_FormulaParseException_InvalidWorkbookName"] = "Arbeidsbokreferansen i anførselstegn er ikke et gyldig arbeidsbokfilnavn.";
        this["LE_FormulaParseException_InvalidWorksheetName"] = "Regnearkets navn er ugyldig.";
        this["LE_FormulaParseException_MacroFunctionNotAllowed"] = "'{0}'-funksjonene er bare gyldige i en navngitt referanse.";
        this["LE_FormulaParseException_Message_PortionWithError"] = "Del med feil: {0}";
        this["LE_FormulaParseException_MissingArgumentAfterBinary"] = "En binær operator i formelen mangler en operand etter seg.";
        this["LE_FormulaParseException_MissingArgumentAfterParen"] = "Det er ikke noe uttrykk etter en åpen parentes.";
        this["LE_FormulaParseException_MissingArgumentAfterUnary"] = "En unær operator i formelen ble ikke fulgt av et gyldig uttrykk.";
        this["LE_FormulaParseException_MissingArgumentBeforeBinary"] = "En binær operator i formelen mangler en operand før seg.";
        this["LE_FormulaParseException_NamedReferenceRefsNeedSheetName"] = "Alle cellereferanser i en navngitt referanseformel må identifisere regnearket.";
        this["LE_FormulaParseException_NoElementAfterArraySerapator"] = "Det er ingen matrise-element etter en elementskiller i formelen.";
        this["LE_FormulaParseException_NoEqualsSign"] = "Formelen må starte med et likhetstegn.";
        this["LE_FormulaParseException_NoExclamationAfterWorkbookName"] = "Arbeidsbokreferansen må følges av et utropstegn.";
        this["LE_FormulaParseException_NoExclamationAfterWorksheetName"] = "Regnearkets referanse må følges av et utropstegn.";
        this["LE_FormulaParseException_NoExpressions"] = "Formelen har ingen gyldige uttrykk etter likhetstegnet.";
        this["LE_FormulaParseException_NoFileNameAfterBracket"] = "Det ble forventet et gyldig filnavn etter den åpne hakeparentesen.";
        this["LE_FormulaParseException_NoValidTermAfterWorkbookName"] = "En ekstern arbeidsbokreferanse må følges av en navngitt referanse.";
        this["LE_FormulaParseException_NoValidTermAfterWorksheetName"] = "En regnearkreferanse må følges av en celle eller en navngitt referanse.";
        this["LE_FormulaParseException_NoWorksheetAfterWorkbookName"] = "Et arbeidsbokfilnavn i parenteser må følges av et regnearknavn.";
        this["LE_FormulaParseException_StringConstantLengthTooLong"] = "Den angitte formelen kan ikke inneholde en streng som er lengre enn {0} tegn.";
        this["LE_FormulaParseException_TooLong"] = "Maksimal lengde på en formel kan ikke overstige {0} for det angitte arbeidsbokformatet.";
        this["LE_FormulaParseException_UnknownFunction"] = "En funksjon i formelen har ikke et gyldig funksjonsnavn.";
        this["LE_FormulaParseException_UnmatchedOpenBracket"] = "En konstant matrise i formelen har ikke en lukkeparentes.";
        this["LE_FormulaParseException_UnmatchedOpenParen"] = "En åpen parentes i formelen har ikke en matchende avsluttende parentes.";
        this["LE_FormulaParseException_UnmatchedOpenSquareBracket"] = "Det ble forventet en avsluttende hakeparentes etter arbeidsbokfilnavnet.";
        this["LE_FormulaParseException_WorkbookNameMissingEndQuote"] = "Arbeidsbokreferansen har ingen sluttsitat.";
        this["LE_FormulaParseException_WorksheetRangeMissingEndingName"] = "Regnearksområdet angir ikke et annet regnearknavn.";
        this["LE_GradientStop_InvalidPosition"] = "Må angi et tall mellom 0 og 1.";
        this["LE_IconSetInvalid"] = "'IconSetNoIcon' er ikke en gyldig innstilling for IconSet-egenskapen.";
        this["LE_IndexOutOfRangeException_ArrayBounds"] = "Indeksen var utenfor tabellens grenser.";
        this["LE_InvalidEnumArgumentException_DefaultPatternCannotBeUsed"] = "Standard FillPatternStyle kan ikke brukes til å opprette en CellFillPattern.";
        this["LE_InvalidLineWeight"] = "LineWeight må ikke være mindre enn 0 eller større enn 1584.";
        this["LE_InvalidOperationException_AnchorCommentBeforeApplyingToCell"] = "Kommentaren må ha både TopLeftCornerCell og BottomRightCornerCell bestemt, eller de må være begge null før den brukes på en celle.";
        this["LE_InvalidOperationException_AnchorShapeBeforeAddingToCollection"] = "Formen må ha både TopLeftCornerCell og BottomRightCornerCell sett før den legges til i et regneark eller en gruppe.";
        this["LE_InvalidOperationException_ArrayFormulaAlreadyApplied"] = "Matriseformelen er allerede brukt på et celleområde. Fjern først det eksisterende celleområdet.";
        this["LE_InvalidOperationException_ArrayFormulaAppliedInTable"] = "Tabellformler kan ikke brukes på et område der det er en tabell.";
        this["LE_InvalidOperationException_ArrayFormulaInMergedCell"] = "Matriseformler er ikke gyldige i sammenslåtte celler.";
        this["LE_InvalidOperationException_BottomRightAnchorFromOtherWorksheet"] = "Hvis du legger til formen i denne samlingen, plasseres den på et annet regneark enn BottomRightCornerCell.";
        this["LE_InvalidOperationException_BuiltInStyleNameCannotBeChanged"] = "Navnet på en innebygd stil kan ikke endres.";
        this["LE_InvalidOperationException_CannotAddColorsToPaletteDirectly"] = "Farger kan ikke legges til i WorkbookColorPalette. Bruk indekseringen til å angi en farge på en bestemt indeks.";
        this["LE_InvalidOperationException_CannotAddParagraphDirectly"] = "Avsnitt kan ikke legges til direkte.";
        this["LE_InvalidOperationException_CannotAddTableDirectly"] = "WorksheetTable kan ikke settes inn direkte i samlingen. Bruk metoden Legg til.";
        this["LE_InvalidOperationException_CannotAddTableToRemovedWorksheet"] = "Tabeller kan ikke legges til i et regneark som er fjernet fra arbeidsboken du eier.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileHeaderRowIsHidden"] = "Et filter kan ikke brukes mens tabelloverskriftsraden er skjult.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileUIIsHidden"] = "Et filter kan ikke brukes mens filtergrensesnittet er skjult.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfData"] = "{0} kan ikke settes inn fordi det ville flytte data fra bunnen av regnearket.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfObject"] = "{0} kan ikke settes inn fordi det ville forskyve en form eller kommentar fra bunnen av regnearket.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitBlockingValue"] = "{0} kan ikke settes inn fordi det ville flytte celler i en matriseformel eller datatabell.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitMergedRegion"] = "{0} kan ikke settes inn fordi det ville flytte celler i et sammenslått område.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitTable"] = "{0} kan ikke settes inn fordi det ville flytte celler i en annen tabell.";
        this["LE_InvalidOperationException_CannotInsertTableRow_TableOnBottomOfWorksheet"] = "{0} kan ikke settes inn fordi tabellen er nederst i regnearket.";
        this["LE_InvalidOperationException_CannotModifyKeysCollection"] = "Keys-samlingen kan ikke endres.";
        this["LE_InvalidOperationException_CannotModifyStandardTableStyle"] = "Standard tabellstiler kan ikke endres.";
        this["LE_InvalidOperationException_CannotModifyValuesCollection"] = "Verdisamlingen kan ikke endres.";
        this["LE_InvalidOperationException_CannotMoveDisconnectedWorksheet"] = "Regnearket kan ikke flyttes fordi det ikke tilhører en arbeidsbok.";
        this["LE_InvalidOperationException_CannotRemoveColorsToPaletteDirectly"] = "Farger kan ikke fjernes fra WorkbookColorPalette. Bruk Reset-metoden for å fjerne alle egendefinerte farger.";
        this["LE_InvalidOperationException_CannotRemoveNormalStyle"] = "Normal-stilen kan ikke fjernes.";
        this["LE_InvalidOperationException_CannotSetParentStyleOnStyle"] = "Stilen kan ikke settes på et format som tilhører en annen stil.";
        this["LE_InvalidOperationException_CannotShift_LossOfData"] = "Kan ikke flytte data fra arket.";
        this["LE_InvalidOperationException_CannotShift_LossOfObject"] = "Kan ikke flytte objekter fra arket.";
        this["LE_InvalidOperationException_CannotShift_SplitBlockingValue"] = "Kan ikke skifte i en matriseformel eller datatabell.";
        this["LE_InvalidOperationException_CannotShift_SplitMergedRegion"] = "Kan ikke skifte i en sammenslått region.";
        this["LE_InvalidOperationException_CannotShift_SplitTable"] = "Kan ikke skifte i tabellen.";
        this["LE_InvalidOperationException_CantAddChartInChart"] = "Et diagram kan ikke legges til Shapes-samlingen til et annet diagram.";
        this["LE_InvalidOperationException_CantAddCustomView"] = "Egendefinerte visninger kan ikke legges direkte til samlingen.";
        this["LE_InvalidOperationException_CantAddDataTable"] = "Datatabeller kan ikke legges til direkte.";
        this["LE_InvalidOperationException_CantAddMergedRegion"] = "Fusjonerte celleregioner kan ikke legges til direkte.";
        this["LE_InvalidOperationException_CantAddNamedReference"] = "Navngitte referanser kan ikke legges til direkte.";
        this["LE_InvalidOperationException_CantAddWorksheet"] = "Regneark kan ikke legges direkte til denne samlingen.";
        this["LE_InvalidOperationException_CantApplyRemovedCustomView"] = "Den egendefinerte visningen kan ikke brukes hvis den har blitt fjernet fra arbeidsbokens samling av egendefinerte visninger.";
        this["LE_InvalidOperationException_CantChangeArrayFormula"] = "Cellene i en matriseformel kan ikke endres.";
        this["LE_InvalidOperationException_CantChangeDataTable"] = "De indre cellene i en datatabell kan ikke endres.";
        this["LE_InvalidOperationException_CantModifyCollection"] = "Samlingen kan ikke endres.";
        this["LE_InvalidOperationException_CantOverlapArrayFormula"] = "Et sammenslått celleområde kan ikke overlappe et matriseformelområde.";
        this["LE_InvalidOperationException_CantOverlapDataTableInterior"] = "Et sammenslått celleområde kan ikke overlappe cellene i en datatabell hvis den strekker seg utenfor den øverste raden eller kolonnen til venstre i disse cellene.";
        this["LE_InvalidOperationException_CantSaveEditRangeWithoutRange"] = "AllowedEditRanges i regnearkene må ha minst en rekkevidde.";
        this["LE_InvalidOperationException_CantSaveWithNoVisibleWorksheets"] = "Minst ett regneark i arbeidsboken må være synlig.";
        this["LE_InvalidOperationException_CantSaveWithNoWorksheets"] = "En arbeidsbok uten regneark kan ikke lagres";
        this["LE_InvalidOperationException_CantSetCircularityErrorDirectly"] = "Sirkulærfeilverdien kan ikke brukes direkte på celler.";
        this["LE_InvalidOperationException_CantSetDataTableDirectly"] = "Datatabeller kan ikke angis som verdien på en celle.";
        this["LE_InvalidOperationException_CantSetFormulaDirectly"] = "Formler kan ikke angis direkte i en celle, bruk ApplyFormula-metoden eller en av ApplyTo-metodene i formelforekomsten i stedet.";
        this["LE_InvalidOperationException_CellsInTableMinSize"] = "Celleområdet i en datatabell må ha minst to celler i hver dimensjon.";
        this["LE_InvalidOperationException_ChartNotSupported"] = "WorksheetChart-modifiseringer støttes ikke i dette formatet.";
        this["LE_InvalidOperationException_ClearUnknownShapeData"] = "Alle data for denne formen er ukjente. Fjern formen fra samlingen sin i stedet.";
        this["LE_InvalidOperationException_CollectionLongerThanMaxValue"] = "Samlingen inneholder flere oppføringer enn den angitte maksimalverdien tillater.";
        this["LE_InvalidOperationException_CollectionModifiedWhileEnumerating"] = "Samlingen ble endret mens den ble talt opp.";
        this["LE_InvalidOperationException_CurrentTableRowAddressNeedsOriginCell"] = "For å analysere en strukturert tabellreferanse ved hjelp av gjeldende rad, må en opprinnelsescelle spesifiseres.";
        this["LE_InvalidOperationException_CustomViewNoVisibleWorksheets"] = "Minst ett regneark i den egendefinerte visningen med navnet \"{0}\" må være synlig.";
        this["LE_InvalidOperationException_DataTableAppliedInTable"] = "En datatabell kan ikke være i et område der en tabell er tilstede.";
        this["LE_InvalidOperationException_DataTableFormulaCannotBeApplied"] = "En datatabellformel kan ikke brukes direkte på celler.";
        this["LE_InvalidOperationException_DataTableInMergedCell"] = "De indre cellene i en datatabell kan ikke være en del av et sammenslått celleområde.";
        this["LE_InvalidOperationException_DataTableRemoved"] = "Inndatacellene for rad og kolonne kan ikke angis etter at datatabellen er fjernet fra regnearket.";
        this["LE_InvalidOperationException_DeprecatedFillColorSetWithGradientFill"] = "En fyllmønsterfarge kan ikke angis mens en gradient er angitt for cellen.";
        this["LE_InvalidOperationException_EncryptedWorkbooksNotSupported"] = "Arbeidsboken som lastes inn er kryptert. Krypterte arbeidsbøker støttes ikke.";
        this["LE_InvalidOperationException_ExtensionDoesntMatchCurrentFormat"] = "Filnavnet som filen skal lagres som har en utvidelse som ikke samsvarer med gjeldende format for arbeidsboken.";
        this["LE_InvalidOperationException_FixedValuesFilterMustAcceptAValue"] = "Kan ikke ha et FixedValuesFilter som ikke tillater verdier.";
        this["LE_InvalidOperationException_FormattedStringAlreadyOwned"] = "Den formaterte strengen er allerede verdien av en annen celle eller kommentar. Klon den først og legg klonen til i den nye cellen eller formen.";
        this["LE_InvalidOperationException_FormattedStringNotOwned"] = "Fontegenskapene kan ikke nås så lenge den formaterte strengen eller teksten ikke brukes på en celle eller figur.";
        this["LE_InvalidOperationException_FormattedTextAlreadyOwned"] = "Den formaterte strengen er allerede verdien av en annen form. Klon den først og legg klonen på den nye cellen eller formen.";
        this["LE_InvalidOperationException_FormulaAlreadyOwned"] = "Formelen er allerede brukt på en annen celle.";
        this["LE_InvalidOperationException_FormulaReferencesInvalidCells"] = "Formelen refererer til celler som ikke støttes i arbeidsbokens nåværende filformat.";
        this["LE_InvalidOperationException_GetBoundsBeforeAnchorsSet"] = "Formen har ikke grenser før ankercellene er satt.";
        this["LE_InvalidOperationException_HiddenWorksheetCannotBeSelected"] = "Et skjult regneark kan ikke velges.";
        this["LE_InvalidOperationException_HyperlinkSealed"] = "Hyperlenken kan ikke endres.";
        this["LE_InvalidOperationException_ImageDisposed"] = "Et bilde av et WorksheetImage har blitt kastet eller er ugyldig.";
        this["LE_InvalidOperationException_InputCellsBothNull"] = "Kolonneinngangs- og radinngangscellene kan ikke begge være null.";
        this["LE_InvalidOperationException_InputCellsInTable"] = "Inndatacellene kan ikke være innenfor datatabellcellene.";
        this["LE_InvalidOperationException_InputCellsSame"] = "Kolonneinngangs- og radinngangscellene kan ikke være den samme cellen.";
        this["LE_InvalidOperationException_InvalidCharacterRange"] = "Tegnområdet som dekkes av skrifttypen eksisterer ikke lenger i den formaterte strengen.";
        this["LE_InvalidOperationException_InvalidFormatOptionsInWholeTableArea"] = "Den angitte formatverdien for FormatOptions er ugyldig i tabellområdet WholeTable.";
        this["LE_InvalidOperationException_InvalidFormatString"] = "Den angitte formatstrengen er ugyldig.";
        this["LE_InvalidOperationException_InvalidFormatString_GetTextCall"] = "Cellens formatstreng er ugyldig. For å bli varslet om dette når FormatString er satt, sett Workbook.ValidateFormatStrings til True.";
        this["LE_InvalidOperationException_InvalidForWorkbookFormat"] = "Formelen er ikke gyldig for spesifisert format.";
        this["LE_InvalidOperationException_InvalidHeaderCellColumnAreaFormatProperty"] = "De øverste kantverdiene kan ikke angis i formatet for kolonneområdet HeaderCell-tabell.";
        this["LE_InvalidOperationException_InvalidHeaderRowAreaFormatProperty"] = "De øverste kantverdiene kan ikke angis i tabellområdet Format for HeaderRow.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFontProperty"] = "Skriftverdien {0} kan ikke angis på tabellstilområder.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatOptions"] = "Den angitte FormatOptions-formatverdien er ugyldig på tabellstilområder.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatProperty"] = "Verdien for {0}-format kan ikke angis på tabellstilområder.";
        this["LE_InvalidOperationException_InvalidTotalsCellColumnAreaFormatProperty"] = "Verdiene for den nedre grensen kan ikke angis i TotalCell-tabellkolonneområdet.";
        this["LE_InvalidOperationException_InvalidTotalsRowAreaFormatProperty"] = "Verdiene for nedre kant kan ikke angis i TotalsRow-tabellområdet.";
        this["LE_InvalidOperationException_InvalidWholeTableAreaFormatProperty"] = "Bare kantgrenseverdiene kan angis i formatet for hele tabellen.";
        this["LE_InvalidOperationException_LastColumnBeforeFirst"] = "Den siste kolonnen i området skal være større enn eller lik den første kolonnen.";
        this["LE_InvalidOperationException_LastRowBeforeFirst"] = "Den siste raden i området skal være større enn den første raden.";
        this["LE_InvalidOperationException_MaxCellFormats"] = "Maksimalt antall forskjellige celleformater i en excel-fil er: {0}";
        this["LE_InvalidOperationException_MaxColors"] = "Maksimalt antall farger i en Excel-arbeidsbok er: {0}";
        this["LE_InvalidOperationException_MaxColumns"] = "Maksimalt antall kolonner i et Excel-regneark med gjeldende format er: {0}";
        this["LE_InvalidOperationException_MaxFonts"] = "Antall fonter i en arbeidsbok kan ikke overstige {0}.";
        this["LE_InvalidOperationException_MaxFormattedStrings"] = "Antall unike strenger i en arbeidsbok kan ikke overstige {0}.";
        this["LE_InvalidOperationException_MaxRows"] = "Maksimalt antall rader i et Excel-regneark med gjeldende format er: {0}";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableLeftColumn"] = "Den venstre kolonnen med celler spesifisert for datatabellen inneholder et sammenslått celleområde som strekker seg utenfor området til disse cellene.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopLeftCell"] = "Cellen øverst til venstre av cellene spesifisert for datatabellen er en del av en sammenslått celle som strekker seg utenfor området til disse cellene.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopRow"] = "Den øverste raden med celler spesifisert for datatabellen inneholder en sammenslått celleområde som strekker seg utenfor området til disse cellene.";
        this["LE_InvalidOperationException_MergedCellsAppliedInTable"] = "Celler i en tabell kan ikke slås sammen.";
        this["LE_InvalidOperationException_NamedReferenceNameAlreadyExists"] = "Navnet \"{0}\" brukes av en tabell eller navngitt referanse i arbeidsboken med samme omfang.";
        this["LE_InvalidOperationException_NoRegionHasBeenSpecified"] = "Ikke område er spesifisert.";
        this["LE_InvalidOperationException_NoWorksheetContextToSetShapeBounds"] = "Denne overbelastningen av Add kan ikke kalles opp når formsamlingen ikke er knyttet til et regneark. Kall overbelastningen av Add som tar en regnearkforekomst.";
        this["LE_InvalidOperationException_OverlappingTable"] = "Det er en annen tabell i den angitte regionen.";
        this["LE_InvalidOperationException_ReadOnlyFont"] = "Fontobjektet er skrivebeskyttet og kan ikke endres.";
        this["LE_InvalidOperationException_ReadOnlyFormat"] = "Formatobjektet er skrivebeskyttet og kan ikke endres.";
        this["LE_InvalidOperationException_RelativeR1C1AddressNeedsOriginCell"] = "For å analysere en relativ R1C1-adresse må en opprinnelsescelle angis.";
        this["LE_InvalidOperationException_ResizedTableMustBeOnWorksheet"] = "Tabellen må være på et regneark for at man skal kunne endre størrelsen.";
        this["LE_InvalidOperationException_ResolvedFormatCannotBeModified"] = "Det løste celleformatet er skrivebeskyttet og kan ikke endres.";
        this["LE_InvalidOperationException_ShapeCantChangeOrientation"] = "Denne formen kan ikke roteres eller vendes.";
        this["LE_InvalidOperationException_ShapeFillSolidIsImmutable"] = "ShapeFillSolid er uforanderlig. Opprett en ny ShapeFillSolid i stedet.";
        this["LE_InvalidOperationException_ShapeInAnotherCollection"] = "Formen eksisterer allerede på et annet regneark eller en formgruppe.";
        this["LE_InvalidOperationException_ShapeOutlineSolidIsImmutable"] = "ShapeOutlineSolid er uforanderlig. Opprett en ny ShapeOutlineSolid i stedet.";
        this["LE_InvalidOperationException_ShowFilterUIWhileHeaderRowHidden"] = "Filtergrensesnittet kan ikke vises mens overskriftsraden er skjult.";
        this["LE_InvalidOperationException_SystemColorsAccessedFromWrongThread"] = "Får ikke tilgang til SystemColors fra en bakgrunnstråd.";
        this["LE_InvalidOperationException_TableReferenceMustBeUsedFromInsideTable"] = "Formelen har en strukturert tabellreferanse som bare kan brukes fra innsiden av en tabell.";
        this["LE_InvalidOperationException_TableReferenceToMissingTable"] = "Formelen har en referanse til en tabell som ikke eksisterer.";
        this["LE_InvalidOperationException_TableReferenceToMissingTableColumn"] = "Formelen har en referanse til en tabellkolonne som ikke eksisterer.";
        this["LE_InvalidOperationException_TopLeftAnchorFromOtherWorksheet"] = "Hvis du legger til formen i denne samlingen, plasseres den på et annet regneark enn TopLeftCornerCell.";
        this["LE_InvalidOperationException_TotalsRowCannotBeShownInLastRow"] = "Totalraden kan ikke vises når tabellen opptar den siste raden i regnearket.";
        this["LE_InvalidOperationException_UnknownBIFFFormat"] = "Arbeidsbokfilen var i et ukjent BIFF-format.";
        this["LE_InvalidOperationException_UnsupportedBIFFFormat"] = "Arbeidsbokfilen ble lagret i {0}, et format som ikke støttes.";
        this["LE_InvalidOperationException_WorkbookDataViolatesFormatLimits"] = "Arbeidsboken inneholder data som ville ha overskredet grensene som er angitt av det spesifiserte formatet.";
        this["LE_InvalidRegion"] = "Ugyldig regionadresse.";
        this["LE_InvalidValueType"] = "'{0}' er ikke en gyldig verdi for ValueType-egenskapen.";
        this["LE_LocationNotSingleRowOrColumn"] = "Posisjonsreferanse er ikke gyldig fordi cellene ikke alle er i samme kolonne eller rad. Velg celler som alle er i en enkelt kolonne eller rad.";
        this["LE_LocationOrDataAreaInvalid"] = "Referansen for stedet eller dataområdet er ikke gyldig.";
        this["LE_LocationWorksheetMismatch"] = "Plasseringen må være fra dette regnearket.";
        this["LE_NotSingleTargetFormula"] = "Kan bare støtte formler for enkeltmål.";
        this["LE_NotSupportedException_CellType"] = "{0} er ikke en celleverditype som støttes.";
        this["LE_NotSupportedException_NoPackageFactory"] = "packageFactory kan ikke være null. Når du laster fra Excel2007-arbeidsbokformat og bruker Infragistics2.Documents.Excel-enheten, må du oppgi en IPackageFactory for å håndtere pakking av data. Hvis du bruker DotNet Framework 3.0 eller nyere, bruker du Infragistics3.Documents.Excel-enheten i stedet, og emballasjen blir håndtert av WindowsBase-klassen.";
        this["LE_OpenPackagingNonConformanceException_ContainsDublinCoreRefinements"] = "IPackagePart er kjerneegenskapsforholdet, og inneholder forbedringer av andre Dublin Core-elementer enn 'dcterms: created' og 'dcterms: modified'.";
        this["LE_OpenPackagingNonConformanceException_ContainsXmlLanguageAttribute"] = "IPackagePart er kjerneegenskapsforholdet, og inneholder elementer som inneholder attributtet 'xml: lang'.";
        this["LE_OpenPackagingNonConformanceException_CorePropertiesRelationshipAlreadyProcessed"] = "IPackagePart er kjerneegenskapsforholdet, men en er allerede behandlet for den tilhørende IP-pakken.";
        this["LE_OpenPackagingNonConformanceException_DuplicatePartName"] = "En IPackagePart med samme navn tilhører allerede denne IP-pakken. En IP-pakke kan ikke inneholde deler med identiske navn. Vær oppmerksom på at kontrollen etter dupliserte verdier ikke skiller mellom store og små bokstaver.";
        this["LE_OpenPackagingNonConformanceException_UsesMarkupCompatibilityNamespace"] = "IPackagePart er kjerneegenskapsforholdet, og bruker navneområdet Markup Compatibility.";
        this["LE_OpenPackagingNonConformanceException_XmlContainsDocumentTypeDefinition"] = "En IPackagePart inneholdt en DTDerklæring (Document Type Definition). Open Packaging Convention-regler forbyr bruk av XML-innhold med DTD-erklæringer.";
        this["LE_OpenPackagingNonConformanceException_XsiTypeAttributeInvalid"] = "IPackagePart er kjerneegenskapsforholdet, og inneholder elementer som inneholder attributten 'xsi: type', eller inneholder elementer som er påkrevd å inneholde og ikke inneholder attributten 'xsi: type'.";
        this["LE_OpenXmlSerializerNotLoaded"] = "'{0}'-formatet krever en serializer som ikke er lastet. I JavaScript er dette inneholdt i filen infragistics.excel_serialization_openxml.js";
        this["LE_PercentageValueOutOfRange"] = "Prosentverdien {0} må være mellom 0 og 100.";
        this["LE_PercentileValueOutOfRange"] = "Prosentilverdien {0} må være mellom 0 og 100.";
        this["LE_PercentRangeInvalid"] = "{0} må være mindre enn {1}.";
        this["LE_RowsToRepeatAtTopAreOutsideAvailableRange"] = "Utskriftsalternativene har rader som gjentas øverst og som ligger utenfor området med tilgjengelige rader i det nye formatet.";
        this["LE_Serialization_NoPrimaryCategoryAxis"] = "Diagramtypen '{0}' krever en primær kategoriakse. Legg til en primær kategoriakse i diagrammets AxisCollection.";
        this["LE_Serialization_NoPrimaryValueAxis"] = "Diagramtypen '{0}' krever en primærverdiakse. Legg til en primærverdiakse i diagrammets AxisCollection.";
        this["LE_Series_ChartType_ChartEx"] = "{0}' er ugyldig for en serie ChartType i et '{1}'-diagram.";
        this["LE_Series_ChartType_InvalidForComboChart"] = "Karttypen '{0}' er ikke gyldig for bruk i et kombinasjonsdiagram.";
        this["LE_Series_ChartType_NonCombo"] = "Diagramtypen for denne serien bestemmes av typen overordnede diagram ('{0}'), og kan ikke angis. Sett isteden det overordnede diagrammets ChartType til '{1}'.";
        this["LE_Series_Explosion_OutOfRange"] = "Egenskapen Explosion støtter verdier i området 0 til 400, inkludert.";
        this["LE_Series_InvalidAxisGroup"] = "AxisGroup-egenskapen kan ikke settes til '{0}' fordi det finnes en serie av typen '{1}' på samme akse, som ikke kan dele den samme aksen som en serie av '{2}'.";
        this["LE_Series_InvalidSeriesTypeForComboChart"] = "En serie av typen '{0}' kan ikke brukes i et kombinasjonsdiagram.";
        this["LE_Series_NoPrimaryAxis"] = "Minst én serie må plottes på hovedaksen. Endre AxisGroup-egenskapen til 'Primary' for minst én Series.";
        this["LE_Series_OwningSeries_NotClusteredColumn"] = "OwningSeries-egenskapen kan bare settes til en gruppert kolonneserie i et Pareto-diagram.";
        this["LE_Series_OwningSeries_NotParetoLine"] = "OwningSeries-egenskapen kan bare angis for en linjeserie i et Pareto-diagram.";
        this["LE_Series_OwningSeries_Same"] = "OwningSeries kan ikke referere til seg selv.";
        this["LE_Series_SeriesType_ChartEx"] = "SeriesType-verdien '{0}' er ugyldig for et '{1}'-diagram.";
        this["LE_Series_SeriesType_NonCombo"] = "Type-egenskapen gjelder bare for kombinasjonsdiagrammer, og kan ikke angis for et diagram av typen '{0}'.";
        this["LE_SetValue_Formula"] = "Verditypen '{0}' er bare gyldig for formler. Bruk SetFormula-metoden i stedet.";
        this["LE_SetValue_NonNumeric"] = "Verdiparameteren gjelder ikke for verditypen '{0}'. Bruk overbelastningen som bare tar en verditype.";
        this["LE_SetValue_Numeric"] = "Verdiparameteren må spesifiseres for verditypen '{0}'. Bruk overbelastningen som tar både verdi og verditype.";
        this["LE_WorksheetChart_ChartType_ChartEx_CannotTransition"] = "ChartType-egenskapen kan ikke endres fra '{0}' til '{1}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_From"] = "Egenskapen ChartType for et kombinasjonsdiagram kan ikke endres til '{0}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_To"] = "Et diagram av typen '{0}' kan ikke endres til 'Combo'. Et nytt kombinasjonsdiagram kan opprettes ved hjelp av AddChart-metoden, men å endre et eksisterende diagram til et kombinasjonsdiagram støttes ikke.";
        this["LE_WorksheetChart_ComboChart_NotEnoughSeries"] = "Et kombinasjonsdiagram krever minst 2 serier. Legg til flere elementer i diagrammets SeriesCollection.";
        this["LE_WorksheetChart_MissingRequiredAxis_StockCharts"] = "Et '{0}' diagram krever en primær kategoriakse, en primær verdiakse, en sekundær kategoriakse og en sekundær verdiakse.";
        this["LE_WorksheetChart_NotEnoughSeries"] = "Et '{0}' diagram krever minst {1} serier. For å bruke denne diagramtypen må du legge til flere elementer i SeriesCollection.";
        this["LE_WorksheetChart_Save_Series"] = "Et '{0}'-diagram må inneholde minst én serie. Legg til flere serier i diagrammet før du prøver å lagre arbeidsboken.";
        this["LE_WorksheetChart_Save_Series_Pareto"] = "Et 'Pareto'-diagram må inneholde en 'ColumnClustered'-serie og en 'Pareto'-serie. Legg til den aktuelle serien i diagrammet før du prøver å lagre arbeidsboken.";
        this["LE_WorksheetChart_SetComboChartSourceData_InvalidChartTypes"] = "En eller flere av de angitte diagramtypene er ikke gyldige for bruk i et kombinasjonsdiagram: {0}";
        this["LE_WorksheetChart_SetComboChartSourceData_NotAComboChart"] = "SetComboChartSourceData kan ikke kalles fordi den nåværende diagramtypen er '{0}'. Endre ChartType til 'Combo'.";
        this["LE_WorksheetChart_SetComboChartSourceData_SeriesChartTypesNotSupported"] = "Parameteren seriesChartTypes må være null for et '{0}'-diagram.";
        this["LE_WorksheetChart_SetSourceData_NotEnoughSeries"] = "Kombinasjonsdiagrammer krever minst 2 serier. Endre parameteren 'dataRange' til et område som vil resultere i minst 2 serier.";
        this["LE_WorksheetChart_SetSourceData_TooManyAxes"] = "Den angitte kombinasjonen av diagramtyper krever mer enn det maksimalt tillatte antallet akser.";
        this["LE_WorksheetChart_StockChartNotSupported"] = "Redigering av diagramtypen '{0}' støttes foreløpig ikke.";
        this["LE_WorksheetChart_UnknownChartNotSupported"] = "Dette diagrammet ble lastet inn fra et eksisterende regneark, men det kunne ikke fastslås ChartType. Redigering av ukjente diagramtyper støttes foreløpig ikke.";
        this["LER_ArgumentOutOfRangeException_OutlineLevel"] = "Oversiktsnivået må være mellom 0 og 7.";
        this["LER_Exception_KeyNotFound"] = "Nøkkelen ble ikke funnet.";
        this["LER_Exception_MergedRegionsOverlap"] = "Fusjonerte celleområder kan ikke overlappe hverandre.";
        this["TableHeaderRowDescription"] = "overskriftsrad";
        this["TableInsertRowDescription"] = "sett inn rad";
        this["TableTotalsRowDescription"] = "totalrad";
        this["Value_UCErrorCode_Div"] = "#DIV/0!";
        this["Value_UCErrorCode_Fail"] = "#FAIL";
        this["Value_UCErrorCode_NA"] = "#N/A";
        this["Value_UCErrorCode_Name"] = "#NAME?";
        this["Value_UCErrorCode_Null"] = "#NULL!";
        this["Value_UCErrorCode_Num"] = "#NUM!";
        this["Value_UCErrorCode_Ok"] = "#OK";
        this["Value_UCErrorCode_Reference"] = "#REF!";
        this["Value_UCErrorCode_Unknown"] = "#N/A";
        this["Value_UCErrorCode_Value"] = "#VALUE!";
        this["WorkbookColorInfo_Automatic_Description"] = "Automatisk";
        this["WorkbookColorInfo_Description"] = "ColorInfo: {0}";
        this["WorkbookColorInfo_WithTint_Description"] = "ColorInfo: {0}, Tint: {1}";
        this["WorksheetShapeSerialization_GroupName"] = "Gruppe";
        this["WorksheetShapeSerialization_ImageName"] = "Bilde";
        this["WorksheetShapeSerialization_ShapeName"] = "Form";
    }
}
