// igniteui-react-excel/esm2015/lib/ExcelLocalePl.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
export class ExcelLocalePl {
    constructor() {
        this["DefaultTotalLabel"] = "Suma";
        this["Error_AnchoredReference"] = "Spróbuj uzyskać dostęp do {0} jako niezakotwiczonego odwołania";
        this["Error_ExplicitThis"] = "{0} - Jawnie To nie jest obsługiwane";
        this["Error_ExternalRange"] = "Zakres zawiera odwołanie zewnętrzne";
        this["Error_FormulaNotSupported"] = "{0} - Próba uzyskania dostępu do formuły w odwołaniu, które nie obsługuje formuł";
        this["Error_Internal"] = "Wystąpił błąd wewnętrzny w metodzie {0}.";
        this["Error_InvalidFunction"] = "Nieprawidłowa funkcja '{0}' w: {1}.";
        this["Error_InvalidFunctionArgCountTooFew"] = "Funkcja '{0}', do której odwołuje się {1}, nie zawiera poprawnej liczby argumentów. Funkcja wymaga co najmniej {2} parametru(-ów).";
        this["Error_InvalidFunctionArgCountTooMany"] = "Funkcja '{0}', do której odwołuje się {1}, nie zawiera poprawnej liczby argumentów. Liczba argumentów nie powinna przekraczać {2} parametru(-ów).";
        this["Error_InvalidNumberSystemDigit"] = "Napotkano nieprawidłową cyfrę '{0}' w wartości '{1}' w przypadku systemu liczbowego o podstawie {2}.";
        this["Error_InvalidOperand"] = "Nieprawidłowy argument operacji";
        this["Error_InvalidOperatorArgCount"] = "Funkcja '{0}' musi akceptować argumenty {1}.";
        this["Error_InvalidReference"] = "Nieprawidłowe lub niedostępne odwołanie";
        this["Error_Location"] = "wiersz {0}, kolumna {1}";
        this["Error_MethodNotSupported"] = "{0} nie obsługuje {1}.";
        this["Error_NoCurrentRangeElement"] = "Brak aktualnego elementu zakresu";
        this["Error_NullFunctionResult"] = "Ocena '{0}' zwróciła wartość null.";
        this["Error_NullOperand"] = "Zerowy argument operacji";
        this["Error_ParseRef"] = "Nieznane odwołanie pod adresem {0}.";
        this["Error_RangeFromLast"] = "Nie ustawiono fromT.Last";
        this["Error_RangeFromNull"] = "Wartość RefRange: from ma wartość null";
        this["Error_RangeFromRefBase"] = "Wartość RefRange: from nie jest RefBase: {0}";
        this["Error_RangeLast"] = "RefRange Validate: ostatnie spójne kolekcje [{0}] != [{1}]";
        this["Error_RangeNormalizeScope"] = "RefRange Normalize: zły zakres [{0} / {1}]";
        this["Error_RangeToNull"] = "Wartość RefRange: to ma wartość null";
        this["Error_RangeToRefBase"] = "Wartość RefRange: to nie jest RefBase: {0}";
        this["Error_RangeTuple"] = "RefRange Validate: spójne kolekcje [{0}] != [{1}]";
        this["Error_RangeValidateCount"] = "RefRange Validate: from count {0}! = To count {1}";
        this["Error_ReadOnlyReference"] = "{0} - Próba ustawienia wartości w odwołaniu tylko do odczytu";
        this["Error_ReferenceNotEumerable"] = "{0} nie jest wyliczalnym odwołaniem";
        this["Error_RefMalformed"] = "{0} - Błędnie sformułowane odwołanie {1}";
        this["Error_RP_ContainsBase"] = "RefParser.Contains — pusta podstawa";
        this["Error_RP_ContainsTarget"] = "RefParser.Contains — pusty cel";
        this["Error_RP_EmptyElement"] = "Nieprawidłowe odwołanie {0}. Napotkano pusty element w lokalizacji {1}.";
        this["Error_RP_EmptyReference"] = "Nieprawidłowe odwołanie {0}. Odwołanie jest puste.";
        this["Error_RP_MalformedExtra"] = "Błędnie sformułowane odwołanie RefParser: znaleziono wiele elementów [{0}] w [{1}]";
        this["Error_RP_MalformedUnexpected"] = "Błędnie sformułowane odwołanie RefParser: Nieoczekiwany element [{0}] znaleziony w [{1}]";
        this["Error_RP_MergeRelativeBase"] = "RefParser.Merge: podstawa względna";
        this["Error_RP_MissingCloseParenthesis"] = "Nieprawidłowe odwołanie {0}. Brak nawiasu zamykającego.";
        this["Error_RP_MissingEndQuotes"] = "Nieprawidłowe odwołanie {0}. Brak cudzysłowu zamykającego.";
        this["Error_RT_Expected"] = "RefTuple: oczekiwano [{0}] w [{1}]";
        this["Error_RT_InvalidScope"] = "Nieprawidłowy zakres: {0}.";
        this["Error_RT_InvalidScope_CharactersAfterClosingQuotes"] = "Napotkano znaki po zamknięciu podwójnych cudzysłowów w zakresie {0}, w lokalizacji {1}.";
        this["Error_RT_InvalidTuple_CharactersAfterScopeEnd"] = "Nieprawidłowy element {0}. Napotkano znaki za końcem zakresu.";
        this["Error_RT_InvalidTuple_MissingCloseParenthesis"] = "Nieprawidłowy element {0}. Brak nawiasu zamykającego.";
        this["Error_RT_InvalidTuple_MissingEndQuotes"] = "Nieprawidłowy element {0}. Brak cudzysłowu zamykającego.";
        this["Error_RT_InvalidTuple_NamePortionEmpty"] = "Nieprawidłowy element {0}. Część nazwy elementu jest pusta.";
        this["Error_RT_InvalidTuple_ScopeIndexLarge"] = "Nieprawidłowy element {0}. Indeks {1} jest nieprawidłowy: {2}.";
        this["Error_RT_InvalidTuple_UnescapedCharacter"] = "Napotkano znak {0} bez zmiany znaczenia w {1}, w lokalizacji {2}.";
        this["Error_RT_NullRefName"] = "RefTuple: null RefName";
        this["Error_RT_Unexpected"] = "RefTuple: oczekiwano [{0}] w [{1}]";
        this["Error_ScopeAllNotLast"] = "{0} - Zakres Wszystko musi znajdować się w ostatnim elemencie {1}";
        this["Error_UCErrorCode_Div"] = "Dzielenie przez zero";
        this["Error_UCErrorCode_Fail"] = "Niepowodzenie";
        this["Error_UCErrorCode_NA"] = "Nie dotyczy";
        this["Error_UCErrorCode_Name"] = "Nieprawidłowa nazwa";
        this["Error_UCErrorCode_Null"] = "Wartość zerowa";
        this["Error_UCErrorCode_Num"] = "Nieprawidłowe wartości liczbowe w funkcji lub formule.";
        this["Error_UCErrorCode_Ok"] = "Powodzenie";
        this["Error_UCErrorCode_Reference"] = "Nieprawidłowe lub niedostępne odwołanie";
        this["Error_UCErrorCode_Unknown"] = "Nieprawidłowy kod";
        this["Error_UCErrorCode_Value"] = "Nieprawidłowy typ argumentu lub argumentu operacji.";
        this["Error_UnexpectedScope"] = "{0} - Nieoczekiwany zakres elementu nadrzędnego {1}";
        this["Error_UnknownRange"] = "Nieznany typ zakresu";
        this["Error_UnknownRefType"] = "{0} - Nieznane RefType {1}";
        this["Error_UnknownScope"] = "{0} - Nieznany zakres {1}";
        this["Error_UnresolvedExternal"] = "Nierozwiązane odwołanie zewnętrzne";
        this["Func_abs_Arg_0"] = "liczba";
        this["Func_abs_ArgDesc_0"] = "Liczba rzeczywista, dla której ma zostać uzyskana wartość bezwzględna.";
        this["Func_abs_Category"] = "Matematyczne";
        this["Func_abs_CategoryURL"] = "Matematyczne";
        this["Func_abs_Desc"] = "Zwraca wartość bezwzględną liczby. Wartość bezwzględna liczby to liczba bez znaku.";
        this["Func_acos_Arg_0"] = "liczba";
        this["Func_acos_ArgDesc_0"] = "Cosinus szukanego kąta musi się zawierać w przedziale od –1 do 1.";
        this["Func_acos_Category"] = "Matematyczne";
        this["Func_acos_CategoryURL"] = "Matematyczne";
        this["Func_acos_Desc"] = "Zwraca arcus cosinus, czyli odwrotny cosinus liczby. Arcus cosinus to kąt, którego cosinus jest określoną liczbą. Zwrócony kąt jest podawany w radianach jako liczba z zakresu od zera (0) do Pi. Aby skonwertować wynik z radianów na stopnie, należy go pomnożyć przez 180/PI().";
        this["Func_acosh_Arg_0"] = "liczba";
        this["Func_acosh_ArgDesc_0"] = "Liczba rzeczywista większa niż 1, dla której ma zostać znaleziony odwrotny cosinus hiperboliczny.";
        this["Func_acosh_Category"] = "Matematyczne";
        this["Func_acosh_CategoryURL"] = "Matematyczne";
        this["Func_acosh_Desc"] = "Zwraca odwrotny cosinus hiperboliczny podanej liczby. Odwrotny cosinus hiperboliczny to kąt, którego cosinus hiperboliczny jest daną liczbą.";
        this["Func_and_Arg_0"] = "logiczne{0}";
        this["Func_and_ArgDesc_0"] = "Wartość logiczna (PRAWDA lub FAŁSZ).";
        this["Func_and_Category"] = "Logiczne";
        this["Func_and_CategoryURL"] = "Logiczne";
        this["Func_and_Desc"] = "Zwraca wartość PRAWDA, jeśli wszystkie wartości są prawdziwe, lub FAŁSZ, jeśli co najmniej jedna wartość jest fałszywa.";
        this["Func_asin_Arg_0"] = "liczba";
        this["Func_asin_ArgDesc_0"] = "Sinus szukanego kąta musi się zawierać w przedziale od –1 do 1.";
        this["Func_asin_Category"] = "Matematyczne";
        this["Func_asin_CategoryURL"] = "Matematyczne";
        this["Func_asin_Desc"] = "Zwraca arcus sinus, czyli odwrotny sinus, podanej liczby. Arcus sinus to kąt, którego sinus jest podaną liczbą. Zwrócony kąt jest podawany w radianach jako liczba z zakresu od minus Pi/2 do Pi/2. Aby wyrazić arcus sinus w stopniach, należy pomnożyć wynik przez 180/Pi().";
        this["Func_asinh_Arg_0"] = "liczba";
        this["Func_asinh_ArgDesc_0"] = "Liczba rzeczywista, dla której ma zostać znaleziony odwrotny sinus hiperboliczny.";
        this["Func_asinh_Category"] = "Matematyczne";
        this["Func_asinh_CategoryURL"] = "Matematyczne";
        this["Func_asinh_Desc"] = "Zwraca odwrotny sinus hiperboliczny podanej liczby. Odwrotny sinus hiperboliczny to kąt, którego sinus hiperboliczny jest daną liczbą.";
        this["Func_atan_Arg_0"] = "liczba";
        this["Func_atan_ArgDesc_0"] = "Tangens szukanego kąta.";
        this["Func_atan_Category"] = "Matematyczne";
        this["Func_atan_CategoryURL"] = "Matematyczne";
        this["Func_atan_Desc"] = "Zwraca arcus tangens, czyli odwrotny tangens, liczby. Arcus tangens to kąt, którego tangens jest podaną liczbą. Zwrócony kąt jest podawany w radianach jako liczba z zakresu od minus Pi/2 do Pi/2. Aby wyrazić arcus tangens w stopniach, należy pomnożyć wynik przez 180/Pi().";
        this["Func_atan2_Arg_0"] = "x";
        this["Func_atan2_Arg_1"] = "y";
        this["Func_atan2_ArgDesc_0"] = "Współrzędna x punktu.";
        this["Func_atan2_ArgDesc_1"] = "Współrzędna y punktu.";
        this["Func_atan2_Category"] = "Matematyczne";
        this["Func_atan2_CategoryURL"] = "Matematyczne";
        this["Func_atan2_Desc"] = "Zwraca arcus tangens, czyli odwrotny tangens, zadanych współrzędnych x i y. Arcus tangens to kąt zawarty między osią X a linią przechodzącą przez początek układu (0,0) i punkt o współrzędnych (x_liczba, y_liczba). Kąt jest podawany w radianach i przyjmuje wartość z zakresu od minus Pi do Pi, ale bez minus Pi.";
        this["Func_atanh_Arg_0"] = "liczba";
        this["Func_atanh_ArgDesc_0"] = "Liczba rzeczywista z zakresu od –1 do 1, dla której ma zostać uzyskana odwrotny tangens hiperboliczny.";
        this["Func_atanh_Category"] = "Matematyczne";
        this["Func_atanh_CategoryURL"] = "Matematyczne";
        this["Func_atanh_Desc"] = "Zwraca odwrotny tangens hiperboliczny podanej liczby. Odwrotny tangens hiperboliczny to kąt, którego tangens hiperboliczny jest daną liczbą.";
        this["Func_avedev_Arg_0"] = "liczba{0}";
        this["Func_avedev_ArgDesc_0"] = "Liczba1, liczba2, ... co najmniej jedna wartość liczbowa lub odwołanie do wartości liczbowych.";
        this["Func_avedev_Category"] = "Statystyczne";
        this["Func_avedev_CategoryURL"] = "Statystyczne";
        this["Func_avedev_Desc"] = "Zwraca średnie odchylenie od średniej wszystkich wartości liczbowych.";
        this["Func_average_Arg_0"] = "liczba{0}";
        this["Func_average_ArgDesc_0"] = "Liczba lub zakres liczb, dla których należy znaleźć średnią z innymi określonymi argumentami.";
        this["Func_average_Category"] = "Statystyczne";
        this["Func_average_CategoryURL"] = "Statystyczne";
        this["Func_average_Desc"] = "Zwraca średnią (średnią arytmetyczną) jednej lub kilku wartości.";
        this["Func_bin2dec_Arg_0"] = "Liczba";
        this["Func_bin2dec_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące liczbę binarną.";
        this["Func_bin2dec_Category"] = "Inżynieria";
        this["Func_bin2dec_CategoryURL"] = "Inżynieria";
        this["Func_bin2dec_Desc"] = "Zwraca wartość dziesiętną tekstowej reprezentacji liczby w binarnym systemie liczbowym.";
        this["Func_bin2hex_Arg_0"] = "Liczba";
        this["Func_bin2hex_Arg_1"] = "Miejsca";
        this["Func_bin2hex_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące liczbę binarną.";
        this["Func_bin2hex_ArgDesc_1"] = "Liczba znaków używanych do przedstawienia wartości. Jeśli nie zostanie to określone, zostanie użyta minimalna liczba znaków niezbędnych do przedstawienia wartości. Pozwala to określić liczbę zer wiodących.";
        this["Func_bin2hex_Category"] = "Inżynieria";
        this["Func_bin2hex_CategoryURL"] = "Inżynieria";
        this["Func_bin2hex_Desc"] = "Konwertuje wyrażenie tekstowe reprezentujące liczbę binarną na wyrażenie tekstowe reprezentujące tę samą liczbę w szesnastkowym systemie liczbowym.";
        this["Func_bin2oct_Arg_0"] = "Liczba";
        this["Func_bin2oct_Arg_1"] = "Miejsca";
        this["Func_bin2oct_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące liczbę binarną.";
        this["Func_bin2oct_ArgDesc_1"] = "Liczba znaków używanych do przedstawienia wartości. Jeśli nie zostanie to określone, zostanie użyta minimalna liczba znaków niezbędnych do przedstawienia wartości. Pozwala to określić liczbę zer wiodących.";
        this["Func_bin2oct_Category"] = "Inżynieria";
        this["Func_bin2oct_CategoryURL"] = "Inżynieria";
        this["Func_bin2oct_Desc"] = "Konwertuje wyrażenie tekstowe reprezentujące liczbę binarną na wyrażenie tekstowe reprezentujące tę samą liczbę w ósemkowym systemie liczbowym.";
        this["Func_binomdist_Arg_0"] = "Liczba";
        this["Func_binomdist_Arg_1"] = "Liczba prób";
        this["Func_binomdist_Arg_2"] = "Prawdopodobieństwo";
        this["Func_binomdist_Arg_3"] = "Skumulowane";
        this["Func_binomdist_ArgDesc_0"] = "Liczba sukcesów w próbach.";
        this["Func_binomdist_ArgDesc_1"] = "Liczba niezależnych prób.";
        this["Func_binomdist_ArgDesc_2"] = "Prawdopodobieństwo sukcesu w każdej próbie.";
        this["Func_binomdist_ArgDesc_3"] = "Wartość logiczna określająca format funkcji. Jeśli na wartość 'prawda', funkcja zwróci funkcję rozkładu skumulowanego, który określa prawdopodobieństwo, że wystąpi co najwyżej 'liczba' sukcesów; w przeciwnym razie zwróci funkcję masy prawdopodobieństwa, która jest miarą prawdopodobieństwa wystąpienia 'liczby' sukcesów.";
        this["Func_binomdist_Category"] = "Statystyczne";
        this["Func_binomdist_CategoryURL"] = "Statystyczne";
        this["Func_binomdist_Desc"] = "Zwraca liczbę możliwych kombinacji (z powtórzeniami) dla danego zestawu elementów i liczby wybieranych elementów z tego zestawu.";
        this["Func_CategoryURL_Template"] = "ms-help://INFRAGISTICS_HELP/NetAdvantagev{2}{3}/Win/WinElements/WinCalcManager_{0}_Functions.html#FUNC_{1}";
        this["Func_ceiling_Arg_0"] = "liczba";
        this["Func_ceiling_Arg_1"] = "istotność";
        this["Func_ceiling_ArgDesc_0"] = "Liczba do zaokrąglenia.";
        this["Func_ceiling_ArgDesc_1"] = "Wielokrotność, do której należy zaokrąglić. W przypadku pominięcia przyjmowana jest wartość jeden (1).";
        this["Func_ceiling_Category"] = "Matematyczne";
        this["Func_ceiling_CategoryURL"] = "Matematyczne";
        this["Func_ceiling_Desc"] = "Zwraca najmniejszą wielokrotność istotności (domyślnie 1), większą lub równą podanej liczbie.";
        this["Func_char_Arg_0"] = "kodZnaku";
        this["Func_char_ArgDesc_0"] = "Liczba, która ma zostać skonwertowana na znak.";
        this["Func_char_Category"] = "TekstDane";
        this["Func_char_CategoryURL"] = "Text_and_Data";
        this["Func_char_Desc"] = "Zwraca znak określony przez numer w kodzie zestawu znaków używanego na komputerze.";
        this["Func_choose_Arg_0"] = "Numer_indeksu";
        this["Func_choose_Arg_1"] = "Value{0}";
        this["Func_choose_ArgDesc_0"] = "Liczony od 1 indeks wskazujący pozostałe argumenty wartości, która ma zostać użyta.";
        this["Func_choose_ArgDesc_1"] = "Wartość, która może zostać wybrana w zależności od wartości Numer_indeksu.";
        this["Func_choose_Category"] = "WyszukajIodwołaj";
        this["Func_choose_CategoryURL"] = "Lookup_And_Reference";
        this["Func_choose_Desc"] = "Zwraca jedną z wartości podanych jako argument na podstawie liczby w pierwszym argumencie.";
        this["Func_code_Arg_0"] = "tekst";
        this["Func_code_ArgDesc_0"] = "Wyrażenie tekstowe, dla którego zostanie obliczony pierwszy znak.";
        this["Func_code_Category"] = "TekstDane";
        this["Func_code_CategoryURL"] = "Text_and_Data";
        this["Func_code_Desc"] = "Zwraca kod liczbowy pierwszego znaku w wyrażeniu tekstowym, odpowiadający zestawowi znaków używanemu na komputerze.";
        this["Func_combin_Arg_0"] = "Liczba";
        this["Func_combin_Arg_1"] = "NumberChosen";
        this["Func_combin_ArgDesc_0"] = "Liczba elementów.";
        this["Func_combin_ArgDesc_1"] = "Liczba elementów wybranych w każdej kombinacji.";
        this["Func_combin_Category"] = "Matematyczne";
        this["Func_combin_CategoryURL"] = "Matematyczne";
        this["Func_combin_Desc"] = "Zwraca liczbę możliwych kombinacji dla danego zestawu elementów i liczby wybieranych elementów z tego zestawu.";
        this["Func_combina_Arg_0"] = "Liczba";
        this["Func_combina_Arg_1"] = "NumberChosen";
        this["Func_combina_ArgDesc_0"] = "Liczba elementów.";
        this["Func_combina_ArgDesc_1"] = "Liczba elementów wybranych w każdej kombinacji.";
        this["Func_combina_Category"] = "Matematyczne";
        this["Func_combina_CategoryURL"] = "Matematyczne";
        this["Func_combina_Desc"] = "Zwraca liczbę możliwych kombinacji (z powtórzeniami) dla danego zestawu elementów i liczby wybieranych elementów z tego zestawu.";
        this["Func_complex_Arg_0"] = "Część_rzeczywista";
        this["Func_complex_Arg_1"] = "Część_urojona";
        this["Func_complex_Arg_2"] = "Jednostka_urojona";
        this["Func_complex_ArgDesc_0"] = "Część rzeczywista liczby zespolonej.";
        this["Func_complex_ArgDesc_1"] = "Część urojona liczby zespolonej.";
        this["Func_complex_ArgDesc_2"] = "Określa jednostkę urojoną, która ma być używana. Dopuszczalne wartości to 'i' lub 'j'. Jeśli pominięto, używana jest jednostka 'i'. Nalży pamiętać, że w jednostce urojonej rozróżniana jest wielkość liter; duże litery 'I' i 'J' są niedozwolone i spowodują błąd #ARG.";
        this["Func_complex_Category"] = "Inżynieria";
        this["Func_complex_CategoryURL"] = "Inżynieria";
        this["Func_complex_Desc"] = "Zwraca liczbę zespoloną reprezentowaną przez wyrażenie tekstowe w formacie 'x + yi' lub 'x + yj' poprzez połączenie części rzeczywistej i części urojonej.";
        this["Func_concatenate_Arg_0"] = "tekst{0}";
        this["Func_concatenate_ArgDesc_0"] = "Wyrażenie tekstowe lub zakres wyrażeń tekstowych, które mają zostać połączone w jedno wyrażenie tekstowe.";
        this["Func_concatenate_Category"] = "TekstDane";
        this["Func_concatenate_CategoryURL"] = "Text_and_Data";
        this["Func_concatenate_Desc"] = "Łączy wszystkie argumenty w jedno wyrażenie tekstowe.";
        this["Func_convert_Arg_0"] = "Liczba";
        this["Func_convert_Arg_1"] = "Jednostka_wejściowa";
        this["Func_convert_Arg_2"] = "Jednostka_wyjściowa";
        this["Func_convert_ArgDesc_0"] = "Wartość do skonwertowania.";
        this["Func_convert_ArgDesc_1"] = "Jednostka, w której podano liczbę.\r\n    \r\nJednostki masy i ciężaru: 'g' = gram; 'sg' = slug; 'lbm' = funt masy (avoirdupois); 'u' = U (jednostka masy atomowj); 'ozm' = uncja masy (avoirdupois)\r\nJednostki odległości: 'm' = metr; 'mi' = mila lądowa; 'Nmi' = mila morska; 'in' = cal; 'f' = stopa; 'yd' = jard; 'ang' = Angstrem; 'Pica' = Cycero (1/72 cala)\r\nJednostki czasu: 'yr' = rok; 'day' = dzień; 'hr' = godzina; 'mn' = minuta; 'sec' = sekunda\r\nJednostki ciśnienia: 'Pa' (lub 'p') = Pascal; 'atm' (lub 'at') = Atmosfera; 'mmHg' = milimetry słupa rtęci\r\nJednostki siły: 'N' = niuton; 'dyn' (lub 'dy') = Dyna; 'lbf' = funt siły\r\nJednostki energii: 'J' = dżul; 'e' = Erg; 'c' = kaloria termodynamiczna; 'cal' = kaloria międzynarodowa; 'eV' (lub 'ev') = elektronowolt; 'HPh' (lub 'hh') = koniogodzina; 'Wh' (lub 'wh') = watogodzina; 'flb' = stopofunt; 'BTU' (lub 'btu') = Brytyjska jednostka ciepła\r\nJednostki mocy: 'HP' (lub 'h') = Koń mechaniczny; 'W' (lub 'w') = Watt\r\nJednostki pola magnetycznego: 'T' = Tesla; 'ga' = Gauss\r\nJednostki temperatury: 'C' (lub 'cel') = stopień Celsjusza; 'F' (lub 'fah') = stopień Fahrenheita; 'K' (lub 'kel') = Kelwin\r\nJednostki objętości płynów: 'tsp' = Łyżeczka; 'tbs' = Łyżka stołowa; 'oz' = Uncja objętości; 'cup' = Kubek; 'pt' (lub 'us_pt') = Półkwarta amerykańska; 'uk_pt' = Półkwarta brytyjska; 'qt' = Kwarta; 'gal' = Galon; 'l' (lub 'lt') = Litr\r\n\r\nKażda jednostka metryczna może być poprzedzona jednym z następujących prefiksów: eksa ('E') = 1E+18; peta ('P') = 1E+15; tera ('T') = 1E+12; giga ('G') = 1E+09; mega ('M') = 1E+06; kilo ('k') = 1E+03; hekto ('h') = 1E+02; deka ('e') = 1E+01; decy ('d') = 1E-01; centy ('c') = 1E-02; milli ('m') = 1E-03; mikro ('u') = 1E-06; nano ('n') = 1E-09; pico ('p') = 1E-12; femto ('f') = 1E-15; atto ('a') = 1E-18\r\n    ";
        this["Func_convert_ArgDesc_2"] = "Jednostki, w których ma zostać zwrócona skonwertowana wartość.\r\n    \r\nJednostki masy i ciężaru: 'g' = gram; 'sg' = slug; 'lbm' = funt masy (avoirdupois); 'u' = U (jednostka masy atomowj); 'ozm' = uncja masy (avoirdupois)\r\nJednostki odległości: 'm' = metr; 'mi' = mila lądowa; 'Nmi' = mila morska; 'in' = cal; 'f' = stopa; 'yd' = jard; 'ang' = Angstrem; 'Pica' = Cycero (1/72 cala)\r\nJednostki czasu: 'yr' = rok; 'day' = dzień; 'hr' = godzina; 'mn' = minuta; 'sec' = sekunda\r\nJednostki ciśnienia: 'Pa' (lub 'p') = Pascal; 'atm' (lub 'at') = Atmosfera; 'mmHg' = milimetry słupa rtęci\r\nJednostki siły: 'N' = niuton; 'dyn' (lub 'dy') = Dyna; 'lbf' = funt siły\r\nJednostki energii: 'J' = dżul; 'e' = Erg; 'c' = kaloria termodynamiczna; 'cal' = kaloria międzynarodowa; 'eV' (lub 'ev') = elektronowolt; 'HPh' (lub 'hh') = koniogodzina; 'Wh' (lub 'wh') = watogodzina; 'flb' = stopofunt; 'BTU' (lub 'btu') = Brytyjska jednostka ciepła\r\nJednostki mocy: 'HP' (lub 'h') = Koń mechaniczny; 'W' (lub 'w') = Watt\r\nJednostki pola magnetycznego: 'T' = Tesla; 'ga' = Gauss\r\nJednostki temperatury: 'C' (lub 'cel') = stopień Celsjusza; 'F' (lub 'fah') = stopień Fahrenheita; 'K' (lub 'kel') = Kelwin\r\nJednostki objętości płynów: 'tsp' = Łyżeczka; 'tbs' = Łyżka stołowa; 'oz' = Uncja objętości; 'cup' = Kubek; 'pt' (lub 'us_pt') = Półkwarta amerykańska; 'uk_pt' = Półkwarta brytyjska; 'qt' = Kwarta; 'gal' = Galon; 'l' (lub 'lt') = Litr\r\n\r\nKażda jednostka metryczna może być poprzedzona jednym z następujących prefiksów: eksa ('E') = 1E+18; peta ('P') = 1E+15; tera ('T') = 1E+12; giga ('G') = 1E+09; mega ('M') = 1E+06; kilo ('k') = 1E+03; hekto ('h') = 1E+02; deka ('e') = 1E+01; decy ('d') = 1E-01; centy ('c') = 1E-02; milli ('m') = 1E-03; mikro ('u') = 1E-06; nano ('n') = 1E-09; pico ('p') = 1E-12; femto ('f') = 1E-15; atto ('a') = 1E-18\r\n    ";
        this["Func_convert_Category"] = "Inżynieria";
        this["Func_convert_CategoryURL"] = "Inżynieria";
        this["Func_convert_Desc"] = "Konwertuje wartość z jednego systemu miar na inny, na przykład metry na cale lub godziny na sekundy.";
        this["Func_cos_Arg_0"] = "liczba";
        this["Func_cos_ArgDesc_0"] = "Kąt w radianach, dla którego ma zostać uzyskany cosinus.";
        this["Func_cos_Category"] = "Matematyczne";
        this["Func_cos_CategoryURL"] = "Matematyczne";
        this["Func_cos_Desc"] = "Zwraca cosinus podanego kąta.";
        this["Func_cosh_Arg_0"] = "liczba";
        this["Func_cosh_ArgDesc_0"] = "Liczba rzeczywista, dla której ma zostać znaleziony cosinus hiperboliczny.";
        this["Func_cosh_Category"] = "Matematyczne";
        this["Func_cosh_CategoryURL"] = "Matematyczne";
        this["Func_cosh_Desc"] = "Zwraca cosinus hiperboliczny liczby.";
        this["Func_count_Arg_0"] = "wartość{0}";
        this["Func_count_ArgDesc_0"] = "Odwołani lub zakres odwołań, które mogą zawierać liczby.";
        this["Func_count_Category"] = "Statystyczne";
        this["Func_count_CategoryURL"] = "Statystyczne";
        this["Func_count_Desc"] = "Zlicza liczbę odwołań zawierających liczby.";
        this["Func_counta_Arg_0"] = "wartość{0}";
        this["Func_counta_ArgDesc_0"] = "Odwołani lub zakres odwołań, które mogą zawierać liczby.";
        this["Func_counta_Category"] = "Statystyczne";
        this["Func_counta_CategoryURL"] = "Statystyczne";
        this["Func_counta_Desc"] = "Zlicza odwołania, które nie są puste.";
        this["Func_critbinom_Arg_0"] = "Liczba prób";
        this["Func_critbinom_Arg_1"] = "Prawdopodobieństwo";
        this["Func_critbinom_Arg_2"] = "Alfa";
        this["Func_critbinom_ArgDesc_0"] = "Liczba niezależnych prób.";
        this["Func_critbinom_ArgDesc_1"] = "Prawdopodobieństwo sukcesu w każdej próbie.";
        this["Func_critbinom_ArgDesc_2"] = "Wartość kryterium.";
        this["Func_critbinom_Category"] = "Statystyczne";
        this["Func_critbinom_CategoryURL"] = "Statystyczne";
        this["Func_critbinom_Desc"] = "Zwraca najmniejszą wartość, dla której skumulowany rozkład dwumianowy jest większy lub równy wartości kryterium.";
        this["Func_date_Arg_0"] = "rok";
        this["Func_date_Arg_1"] = "miesiąc";
        this["Func_date_Arg_2"] = "dzień";
        this["Func_date_ArgDesc_0"] = "Liczba reprezentująca rok.";
        this["Func_date_ArgDesc_1"] = "Liczba reprezentująca miesiąc roku.";
        this["Func_date_ArgDesc_2"] = "Liczba reprezentująca dzień miesiąca.";
        this["Func_date_Category"] = "DataIGodzina";
        this["Func_date_CategoryURL"] = "Date_and_Time";
        this["Func_date_Desc"] = "Zwraca kolejny numer seryjny, który reprezentuje określoną datę.";
        this["Func_dateadd_Arg_0"] = "interwał";
        this["Func_dateadd_Arg_1"] = "liczba";
        this["Func_dateadd_Arg_2"] = "data";
        this["Func_dateadd_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące przedział czasu do dodania.\r\nd = Dzień\r\ny = Rok\r\nh = Godzina\r\nn = Minuta\r\nm = Miesiąc\r\nq = Kwartał\r\ns = Sekunda\r\nw = Tydzień\r\nww = Tydzień kalendarzowy\r\nyyyy = Rok\r\n";
        this["Func_dateadd_ArgDesc_1"] = "Liczba interwałów do dodania. Ta wartość może być dodatnia (aby uzyskać przyszłą wartość DataGodzina) lub ujemna (aby uzyskać przeszłą wartość DataGodzina). Może zawierać część ułamkową, gdy Interwał określa godziny, minuty lub sekundy. W przypadku innych wartości Interval każda ułamkowa część liczby jest ignorowana.";
        this["Func_dateadd_ArgDesc_2"] = "Data, do której należy dodać interwał.";
        this["Func_dateadd_Category"] = "DataIGodzina";
        this["Func_dateadd_CategoryURL"] = "Date_and_Time";
        this["Func_dateadd_Desc"] = "Zwraca datę, do której dodano określony interwał czasu.";
        this["Func_datediff_Arg_0"] = "interwał";
        this["Func_datediff_Arg_1"] = "date1";
        this["Func_datediff_Arg_2"] = "date2";
        this["Func_datediff_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące przedział czasu, który ma być użyty jako jednostka różnicy między data1 a data2.\r\nd = Dzień\r\ny = Rok\r\nh = Godzina\r\nn = Minuta\r\nm = Miesiąc\r\nq = Kwartał\r\ns = Sekunda\r\nw = Tydzień\r\nww = Tydzień kalendarzowy\r\nyyyy = Rok\r\n";
        this["Func_datediff_ArgDesc_1"] = "Pierwsza data używana do obliczenia różnicy.";
        this["Func_datediff_ArgDesc_2"] = "Druga data używana do obliczenia różnicy.";
        this["Func_datediff_Category"] = "DataIGodzina";
        this["Func_datediff_CategoryURL"] = "Date_and_Time";
        this["Func_datediff_Desc"] = "Zwraca wartość wskazującą liczbę interwałów między dwiema wartościami DataGodzina.";
        this["Func_datevalue_Arg_0"] = "data_tekst";
        this["Func_datevalue_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące datę.";
        this["Func_datevalue_Category"] = "DataIGodzina";
        this["Func_datevalue_CategoryURL"] = "Date_and_Time";
        this["Func_datevalue_Desc"] = "Zwraca wartość daty dla określonego tekstu daty.";
        this["Func_day_Arg_0"] = "Date_value";
        this["Func_day_ArgDesc_0"] = "Wartość daty, z której ma zostać pobrany dzień.";
        this["Func_day_Category"] = "DataIGodzina";
        this["Func_day_CategoryURL"] = "Date_and_Time";
        this["Func_day_Desc"] = "Zwraca dzień miesiąca z wartości daty.";
        this["Func_days360_Arg_0"] = "start_date";
        this["Func_days360_Arg_1"] = "data_końcowa";
        this["Func_days360_Arg_2"] = "metoda";
        this["Func_days360_ArgDesc_0"] = "Data początkowa zakresu dni. Jeśli jest późniejsza niż data końcowa, funkcja zwróci liczbę ujemną.";
        this["Func_days360_ArgDesc_1"] = "Data końcowa zakresu dni. Jeśli jest wcześniejsza niż data początkowa, funkcja zwróci liczbę ujemną.";
        this["Func_days360_ArgDesc_2"] = "Wartość logiczna określająca, czy w obliczeniach należy używać metody amerykańskiej (FAŁSZ), czy europejskiej (PRAWDA).";
        this["Func_days360_Category"] = "DataIGodzina";
        this["Func_days360_CategoryURL"] = "Date_and_Time";
        this["Func_days360_Desc"] = "Zwraca liczbę dni między dwiema datami na podstawie roku obejmującego 360 dni.";
        this["Func_db_Arg_0"] = "koszt";
        this["Func_db_Arg_1"] = "wart. końcowa";
        this["Func_db_Arg_2"] = "czas_życia";
        this["Func_db_Arg_3"] = "okres";
        this["Func_db_Arg_4"] = "miesiąc";
        this["Func_db_ArgDesc_0"] = "Początkowy koszt środka trwałego.";
        this["Func_db_ArgDesc_1"] = "Wartość na koniec okresu amortyzacji (czasami nazywana wartością odzyskaną środka trwałego).";
        this["Func_db_ArgDesc_2"] = "Liczba okresów, przez które jest amortyzowany środek trwały (czasami nazywany czasem życia środka trwałego).";
        this["Func_db_ArgDesc_3"] = "Okres, za który ma zostać obliczona amortyzacja. Okres, który musi używać takich samych jednostek, co trwałość.";
        this["Func_db_ArgDesc_4"] = "Liczba miesięcy w pierwszym roku. Jeśli pominięto miesiąc, przyjmuje się, że wartość wynosi 12.";
        this["Func_db_Category"] = "Finansowe";
        this["Func_db_CategoryURL"] = "Finansowe";
        this["Func_db_Desc"] = "Zwraca amortyzację środka trwałego za podany okres, z zastosowaniem metody salda malejącego.";
        this["Func_dbnull_Category"] = "Informacja";
        this["Func_dbnull_CategoryURL"] = "Informacja";
        this["Func_dbnull_Desc"] = "Zwraca coś, co reprezentuje nieistniejącą wartość.";
        this["Func_ddb_Arg_0"] = "koszt";
        this["Func_ddb_Arg_1"] = "wart. końcowa";
        this["Func_ddb_Arg_2"] = "czas_życia";
        this["Func_ddb_Arg_3"] = "okres";
        this["Func_ddb_Arg_4"] = "współczynnik";
        this["Func_ddb_ArgDesc_0"] = "Początkowy koszt środka trwałego.";
        this["Func_ddb_ArgDesc_1"] = "Wartość na koniec okresu amortyzacji (czasami nazywana wartością odzyskaną środka trwałego).";
        this["Func_ddb_ArgDesc_2"] = "Liczba okresów, przez które jest amortyzowany środek trwały (czasami nazywany czasem życia środka trwałego).";
        this["Func_ddb_ArgDesc_3"] = "Okres, za który ma zostać obliczona amortyzacja. Okres, który musi używać takich samych jednostek, co trwałość.";
        this["Func_ddb_ArgDesc_4"] = "Szybkość, z jaką maleje saldo. W przypadku pominięcia przyjmowana jest wartość 2 (metoda podwójnie malejącego salda).";
        this["Func_ddb_Category"] = "Finansowe";
        this["Func_ddb_CategoryURL"] = "Finansowe";
        this["Func_ddb_Desc"] = "Zwraca amortyzację środka trwałego za podany okres, z zastosowaniem metody podwójnie malejącego salda lub innej wskazanej metody.";
        this["Func_dec2bin_Arg_0"] = "Liczba";
        this["Func_dec2bin_Arg_1"] = "Miejsca";
        this["Func_dec2bin_ArgDesc_0"] = "Wartość dziesiętna do skonwertowania.";
        this["Func_dec2bin_ArgDesc_1"] = "Liczba znaków używanych do przedstawienia wartości. Jeśli miejsca nie zostaną określone, zostanie użyta minimalna liczba znaków niezbędnych do przedstawienia wartości. Pozwala to określić liczbę zer wiodących.";
        this["Func_dec2bin_Category"] = "Inżynieria";
        this["Func_dec2bin_CategoryURL"] = "Inżynieria";
        this["Func_dec2bin_Desc"] = "Zwraca wyrażenie tekstowe reprezentujące podaną wartość dziesiętną jako liczbę binarną.";
        this["Func_dec2hex_Arg_0"] = "Liczba";
        this["Func_dec2hex_Arg_1"] = "Miejsca";
        this["Func_dec2hex_ArgDesc_0"] = "Wartość dziesiętna do skonwertowania.";
        this["Func_dec2hex_ArgDesc_1"] = "Liczba znaków używanych do przedstawienia wartości. Jeśli miejsca nie zostaną określone, zostanie użyta minimalna liczba znaków niezbędnych do przedstawienia wartości. Pozwala to określić liczbę zer wiodących.";
        this["Func_dec2hex_Category"] = "Inżynieria";
        this["Func_dec2hex_CategoryURL"] = "Inżynieria";
        this["Func_dec2hex_Desc"] = "Zwraca wyrażenie tekstowe reprezentujące podaną wartość dziesiętną jako liczbę szesnastkową.";
        this["Func_dec2oct_Arg_0"] = "Liczba";
        this["Func_dec2oct_Arg_1"] = "Miejsca";
        this["Func_dec2oct_ArgDesc_0"] = "Wartość dziesiętna do skonwertowania.";
        this["Func_dec2oct_ArgDesc_1"] = "Liczba znaków używanych do przedstawienia wartości. Jeśli miejsca nie zostaną określone, zostanie użyta minimalna liczba znaków niezbędnych do przedstawienia wartości. Pozwala to określić liczbę zer wiodących.";
        this["Func_dec2oct_Category"] = "Inżynieria";
        this["Func_dec2oct_CategoryURL"] = "Inżynieria";
        this["Func_dec2oct_Desc"] = "Zwraca wyrażenie tekstowe reprezentujące podaną wartość dziesiętną jako liczbę ósemkową.";
        this["Func_degrees_Arg_0"] = "Radiany";
        this["Func_degrees_ArgDesc_0"] = "Wartość w radianach, która ma zostać skonwertowana na stopnie.";
        this["Func_degrees_Category"] = "Matematyczne";
        this["Func_degrees_CategoryURL"] = "Matematyczne";
        this["Func_degrees_Desc"] = "Konwertuje radiany na stopnie.";
        this["Func_delta_Arg_0"] = "Liczba1";
        this["Func_delta_Arg_1"] = "Liczba2";
        this["Func_delta_ArgDesc_0"] = "Pierwsza liczba do porównania.";
        this["Func_delta_ArgDesc_1"] = "Druga liczba do porównania. Jeśli nie zostanie podana, przyjmuje się, że wynosi zero.";
        this["Func_delta_Category"] = "Inżynieria";
        this["Func_delta_CategoryURL"] = "Inżynieria";
        this["Func_delta_Desc"] = "Porównuje dwie liczby i zwraca jeden (1), jeśli są równe, lub zero (0), jeśli nie są.";
        this["Func_dollarde_Arg_0"] = "Wartość_ułamkowa_dolara";
        this["Func_dollarde_Arg_1"] = "Ułamkowe";
        this["Func_dollarde_ArgDesc_0"] = "Kwota w dolarach wyrażona jako ułamek.";
        this["Func_dollarde_ArgDesc_1"] = "Wartość używana jako mianownik ułamka. Jeśli wartość ta nie jest liczbą całkowitą, zostanie obcięta.";
        this["Func_dollarde_Category"] = "Finansowe";
        this["Func_dollarde_CategoryURL"] = "Finansowe";
        this["Func_dollarde_Desc"] = "Konwertuje kwotę w dolarach wyrażoną jako ułamek na kwotę w dolarach wyrażoną jako wartość dziesiętna.";
        this["Func_dollarfr_Arg_0"] = "DecimalDollarAmount";
        this["Func_dollarfr_Arg_1"] = "Ułamkowe";
        this["Func_dollarfr_ArgDesc_0"] = "Kwota w dolarach wyrażona jako wartość dziesiętna.";
        this["Func_dollarfr_ArgDesc_1"] = "Wartość używana jako mianownik ułamka. Jeśli wartość ta nie jest liczbą całkowitą, zostanie obcięta.";
        this["Func_dollarfr_Category"] = "Finansowe";
        this["Func_dollarfr_CategoryURL"] = "Finansowe";
        this["Func_dollarfr_Desc"] = "Konwertuje kwotę w dolarach wyrażoną jako wartość dziesiętną na kwotę w dolarach wyrażoną jako ułamek.";
        this["Func_edate_Arg_0"] = "data";
        this["Func_edate_Arg_1"] = "numberOfMonths";
        this["Func_edate_ArgDesc_0"] = "Data początkowa.";
        this["Func_edate_ArgDesc_1"] = "Liczba miesięcy, o jaką należy przesunąć datę. Liczba dodatnia oznacza datę po dacie początkowej. Liczba ujemna oznacza datę przed datą początkową.";
        this["Func_edate_Category"] = "DataIGodzina";
        this["Func_edate_CategoryURL"] = "DataIGodzina";
        this["Func_edate_Desc"] = "Zwraca datę stanowiącą określoną liczbę miesięcy przed datą lub po dacie początkowej.";
        this["Func_eomonth_Arg_0"] = "data";
        this["Func_eomonth_Arg_1"] = "numberOfMonths";
        this["Func_eomonth_ArgDesc_0"] = "Data początkowa.";
        this["Func_eomonth_ArgDesc_1"] = "Liczba miesięcy, o jaką należy przesunąć datę. Liczba dodatnia oznacza datę po dacie początkowej. Liczba ujemna oznacza datę przed datą początkową.";
        this["Func_eomonth_Category"] = "DataIGodzina";
        this["Func_eomonth_CategoryURL"] = "DataIGodzina";
        this["Func_eomonth_Desc"] = "Zwraca datę przypadającą na koniec miesiąca, czyli określoną liczbę miesięcy przed datą lub po dacie początkowej.";
        this["Func_errortype_Arg_0"] = "error_value";
        this["Func_errortype_ArgDesc_0"] = "Wartość błędu, którego numer identyfikacyjny należy znaleźć.";
        this["Func_errortype_Category"] = "Informacja";
        this["Func_errortype_CategoryURL"] = "Informacja";
        this["Func_errortype_Desc"] = "Zwraca liczbę odpowiadającą wartości błędu.";
        this["Func_even_Arg_0"] = "liczba";
        this["Func_even_ArgDesc_0"] = "Wartość do zaokrąglenia.";
        this["Func_even_Category"] = "Matematyczne";
        this["Func_even_CategoryURL"] = "Matematyczne";
        this["Func_even_Desc"] = "Zaokrągla liczbę dodatnią w górę, a liczbę ujemną w dół do najbliższej parzystej liczby całkowitej.";
        this["Func_exp_Arg_0"] = "liczba";
        this["Func_exp_ArgDesc_0"] = "Wykładnik potęgi, do którego należy podnieść e.";
        this["Func_exp_Category"] = "Matematyczne";
        this["Func_exp_CategoryURL"] = "Matematyczne";
        this["Func_exp_Desc"] = "Zwraca 'e' podniesione do zadanej potęgi. Stała 'e' wynosi 2,71828182845904.";
        this["Func_fact_Arg_0"] = "Liczba";
        this["Func_fact_ArgDesc_0"] = "Liczba dodatnia, dla której ma zostać obliczona silnia.";
        this["Func_fact_Category"] = "Matematyczne";
        this["Func_fact_CategoryURL"] = "Matematyczne";
        this["Func_fact_Desc"] = "Zwraca wartość silni podanej liczby.";
        this["Func_factdouble_Arg_0"] = "Liczba";
        this["Func_factdouble_ArgDesc_0"] = "Liczba dodatnia, dla której ma zostać obliczona podwójna silnia.";
        this["Func_factdouble_Category"] = "Matematyczne";
        this["Func_factdouble_CategoryURL"] = "Matematyczne";
        this["Func_factdouble_Desc"] = "Zwraca podwójną silnię liczby.";
        this["Func_false_Category"] = "Logiczne";
        this["Func_false_CategoryURL"] = "Logiczne";
        this["Func_false_Desc"] = "Zwraca wartość logiczną FAŁSZ.";
        this["Func_find_Arg_0"] = "find_text";
        this["Func_find_Arg_1"] = "within_text";
        this["Func_find_Arg_2"] = "start_num";
        this["Func_find_ArgDesc_0"] = "Tekst do znalezienia.";
        this["Func_find_ArgDesc_1"] = "Tekst zawierający tekst do znalezienia.";
        this["Func_find_ArgDesc_2"] = "Numer znaku, od którego rozpoczyna się wyszukiwanie. W przypadku pominięcia przyjmowana jest wartość jeden (1).";
        this["Func_find_Category"] = "TekstDane";
        this["Func_find_CategoryURL"] = "Text_and_Data";
        this["Func_find_Desc"] = "Zwraca pozycję początkową jednego wyrażenia tekstowego w innym, z uwzględnieniem wielkości liter.";
        this["Func_fixed_Arg_0"] = "liczba";
        this["Func_fixed_Arg_1"] = "decimais";
        this["Func_fixed_Arg_2"] = "no_commas";
        this["Func_fixed_ArgDesc_0"] = "Liczba, która ma zostać zaokrąglona i skonwertowana na tekst.";
        this["Func_fixed_ArgDesc_1"] = "Liczba cyfr po prawej stronie przecinka dziesiętnego w zwracanej wartości. Jeśli ta wartość jest ujemna, 'liczba' jest zaokrąglana do lewej punktu dziesiętnego. Wartość domyślna to 2.";
        this["Func_fixed_ArgDesc_2"] = "PRAWDA, jeśli w zwracanej wartości nie powinny znajdować się przecinki. W przypadku pominięcia przyjmowana jest wartość FAŁSZ.";
        this["Func_fixed_Category"] = "TekstDane";
        this["Func_fixed_CategoryURL"] = "Text_and_Data";
        this["Func_fixed_Desc"] = "Zaokrągla liczbę do podanej liczby miejsc po przecinku i zwraca wynik jako tekst z przecinkami lub bez.";
        this["Func_floor_Arg_0"] = "liczba";
        this["Func_floor_Arg_1"] = "istotność";
        this["Func_floor_ArgDesc_0"] = "Liczba do zaokrąglenia.";
        this["Func_floor_ArgDesc_1"] = "Wielokrotność, do której należy zaokrąglić. W przypadku pominięcia przyjmowana jest wartość jeden (1).";
        this["Func_floor_Category"] = "Matematyczne";
        this["Func_floor_CategoryURL"] = "Matematyczne";
        this["Func_floor_Desc"] = "Zwraca największą wielokrotność istotności (domyślnie 1), mniejszą lub równą podanej liczbie.";
        this["Func_fv_Arg_0"] = "stopa";
        this["Func_fv_Arg_1"] = "nper";
        this["Func_fv_Arg_2"] = "pmt";
        this["Func_fv_Arg_3"] = "pv";
        this["Func_fv_Arg_4"] = "type";
        this["Func_fv_ArgDesc_0"] = "Stopa procentowa dla okresu.";
        this["Func_fv_ArgDesc_1"] = "Całkowita liczba okresów płatności w ramach raty rocznej.";
        this["Func_fv_ArgDesc_2"] = "Płatność dokonywana w każdym okresie, który nie może ulec zmianie w okresie obowiązywania raty rocznej.";
        this["Func_fv_ArgDesc_3"] = "Wartość bieżąca lub kwota ryczałtowa, jaką jest teraz warta seria przyszłych płatności. Jeśli nic nie zostanie wpisane, przyjmowana jest wartość zero (0).";
        this["Func_fv_ArgDesc_4"] = "Wartość zero lub jeden, która wskazuje, czy płatności są należne na koniec okresu (zero, wartość domyślna), czy na początek (jeden).";
        this["Func_fv_Category"] = "Finansowe";
        this["Func_fv_CategoryURL"] = "Finansowe";
        this["Func_fv_Desc"] = "Zwraca przyszłą wartość inwestycji na podstawie okresu.";
        this["Func_gcd_Arg_0"] = "liczba{0}";
        this["Func_gcd_ArgDesc_0"] = "Wartość całkowita. Wartość nie będąca liczbą całkowitą zostanie obcięta.";
        this["Func_gcd_Category"] = "Matematyczne";
        this["Func_gcd_CategoryURL"] = "Matematyczne";
        this["Func_gcd_Desc"] = "Zwraca największy wspólny dzielnik co najmniej jednej wartości całkowitej.";
        this["Func_gestep_Arg_0"] = "Liczba1";
        this["Func_gestep_Arg_1"] = "Krok";
        this["Func_gestep_ArgDesc_0"] = "Pierwsza liczba do porównania.";
        this["Func_gestep_ArgDesc_1"] = "Druga liczba do porównania. Jeśli nie zostanie podana, przyjmuje się, że wynosi zero.";
        this["Func_gestep_Category"] = "Inżynieria";
        this["Func_gestep_CategoryURL"] = "Inżynieria";
        this["Func_gestep_Desc"] = "Porównuje dwie liczby i zwraca jeden (1), jeśli pierwsza liczba jest większa lub równa drugiej; w przeciwnym razie zwraca zero (0).";
        this["Func_hex2bin_Arg_0"] = "Liczba";
        this["Func_hex2bin_Arg_1"] = "Miejsca";
        this["Func_hex2bin_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące liczbę szesnastkową.";
        this["Func_hex2bin_ArgDesc_1"] = "Liczba znaków używanych do przedstawienia wartości. Jeśli nie zostanie to określone, zostanie użyta minimalna liczba znaków niezbędnych do przedstawienia wartości. Pozwala to określić liczbę zer wiodących.";
        this["Func_hex2bin_Category"] = "Inżynieria";
        this["Func_hex2bin_CategoryURL"] = "Inżynieria";
        this["Func_hex2bin_Desc"] = "Konwertuje wyrażenie tekstowe reprezentujące liczbę szesnastkową na wyrażenie tekstowe reprezentujące tę samą liczbę w binarnym systemie liczbowym.";
        this["Func_hex2dec_Arg_0"] = "Liczba";
        this["Func_hex2dec_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące liczbę szesnastkową.";
        this["Func_hex2dec_Category"] = "Inżynieria";
        this["Func_hex2dec_CategoryURL"] = "Inżynieria";
        this["Func_hex2dec_Desc"] = "Zwraca wartość dziesiętną tekstowej reprezentacji liczby w szesnastkowym systemie liczbowym.";
        this["Func_hex2oct_Arg_0"] = "Liczba";
        this["Func_hex2oct_Arg_1"] = "Miejsca";
        this["Func_hex2oct_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące liczbę szesnastkową.";
        this["Func_hex2oct_ArgDesc_1"] = "Liczba znaków używanych do przedstawienia wartości. Jeśli nie zostanie to określone, zostanie użyta minimalna liczba znaków niezbędnych do przedstawienia wartości. Pozwala to określić liczbę zer wiodących.";
        this["Func_hex2oct_Category"] = "Inżynieria";
        this["Func_hex2oct_CategoryURL"] = "Inżynieria";
        this["Func_hex2oct_Desc"] = "Konwertuje wyrażenie tekstowe reprezentujące liczbę szesnastkową na wyrażenie tekstowe reprezentujące tę samą liczbę w ósemkowym systemie liczbowym.";
        this["Func_hour_Arg_0"] = "Time_value";
        this["Func_hour_ArgDesc_0"] = "Wartość czasu, z której ma zostać pobrana godzina.";
        this["Func_hour_Category"] = "DataIGodzina";
        this["Func_hour_CategoryURL"] = "Date_and_Time";
        this["Func_hour_Desc"] = "Zwraca godzinę z wartości czasu.";
        this["Func_if_Arg_0"] = "logical_test";
        this["Func_if_Arg_1"] = "value_if_true";
        this["Func_if_Arg_2"] = "value_if_false";
        this["Func_if_ArgDesc_0"] = "Warunek do przetestowania.";
        this["Func_if_ArgDesc_1"] = "Wynik, jeśli warunek zwraca wartość 'prawda'.";
        this["Func_if_ArgDesc_2"] = "Wynik, jeśli warunek zwraca wartość 'fałsz'. Jeśli nic nie zostanie wpisane, przyjmowana jest wartość logiczna 'fałsz'.";
        this["Func_if_Category"] = "Logiczne";
        this["Func_if_CategoryURL"] = "Logiczne";
        this["Func_if_Desc"] = "Zwraca jedną wartość, jeśli podany warunek ma wartość 'prawda'. Zwraca inną wartość, jeśli wynikiem jest 'fałsz'.";
        this["Func_iferror_Arg_0"] = "wartość";
        this["Func_iferror_Arg_1"] = "value_if_error";
        this["Func_iferror_ArgDesc_0"] = "Wartość, która ma zostać użyta w przypadku braku błędu.";
        this["Func_iferror_ArgDesc_1"] = "Wartość, która ma zostać używa, jeśli argument wartość powoduje błąd.";
        this["Func_iferror_Category"] = "Logiczne";
        this["Func_iferror_CategoryURL"] = "Logiczne";
        this["Func_iferror_Desc"] = "Zwraca jedną z dwóch wartości w zależności od tego, czy pierwsza wartość powoduje błąd, czy nie.";
        this["Func_imabs_Arg_0"] = "Liczba_zespolona";
        this["Func_imabs_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imabs_Category"] = "Inżynieria";
        this["Func_imabs_CategoryURL"] = "Inżynieria";
        this["Func_imabs_Desc"] = "Zwraca wartość bezwzględną liczby zespolonej.";
        this["Func_imaginary_Arg_0"] = "Liczba_zespolona";
        this["Func_imaginary_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imaginary_Category"] = "Inżynieria";
        this["Func_imaginary_CategoryURL"] = "Inżynieria";
        this["Func_imaginary_Desc"] = "Zwraca część urojoną liczby zespolonej.";
        this["Func_imargument_Arg_0"] = "Liczba_zespolona";
        this["Func_imargument_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imargument_Category"] = "Inżynieria";
        this["Func_imargument_CategoryURL"] = "Inżynieria";
        this["Func_imargument_Desc"] = "Zwraca argument theta, kąt wyrażony w radianach.";
        this["Func_imconjugate_Arg_0"] = "Liczba_zespolona";
        this["Func_imconjugate_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imconjugate_Category"] = "Inżynieria";
        this["Func_imconjugate_CategoryURL"] = "Inżynieria";
        this["Func_imconjugate_Desc"] = "Zwraca wartość sprzężoną liczby zespolonej.";
        this["Func_imcos_Arg_0"] = "Liczba_zespolona";
        this["Func_imcos_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imcos_Category"] = "Inżynieria";
        this["Func_imcos_CategoryURL"] = "Inżynieria";
        this["Func_imcos_Desc"] = "Zwraca cosinus liczby zespolonej.";
        this["Func_imdiv_Arg_0"] = "Dzielna";
        this["Func_imdiv_Arg_1"] = "dzielnik";
        this["Func_imdiv_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imdiv_ArgDesc_1"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imdiv_Category"] = "Inżynieria";
        this["Func_imdiv_CategoryURL"] = "Inżynieria";
        this["Func_imdiv_Desc"] = "Zwraca iloraz dwóch liczb zespolonych (Dzielna/Dzielnik).";
        this["Func_imexp_Arg_0"] = "Liczba_zespolona";
        this["Func_imexp_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imexp_Category"] = "Inżynieria";
        this["Func_imexp_CategoryURL"] = "Inżynieria";
        this["Func_imexp_Desc"] = "Zwraca wartość wykładniczą liczby zespolonej.";
        this["Func_imln_Arg_0"] = "Liczba_zespolona";
        this["Func_imln_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imln_Category"] = "Inżynieria";
        this["Func_imln_CategoryURL"] = "Inżynieria";
        this["Func_imln_Desc"] = "Zwraca logarytm naturalny liczby zespolonej.";
        this["Func_imlog10_Arg_0"] = "Liczba_zespolona";
        this["Func_imlog10_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imlog10_Category"] = "Inżynieria";
        this["Func_imlog10_CategoryURL"] = "Inżynieria";
        this["Func_imlog10_Desc"] = "Zwraca logarytm dziesiętny liczby zespolonej.";
        this["Func_imlog2_Arg_0"] = "Liczba_zespolona";
        this["Func_imlog2_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imlog2_Category"] = "Inżynieria";
        this["Func_imlog2_CategoryURL"] = "Inżynieria";
        this["Func_imlog2_Desc"] = "Zwraca logarytm o podstawie 2 liczby zespolonej.";
        this["Func_impower_Arg_0"] = "Liczba_zespolona";
        this["Func_impower_Arg_1"] = "potęga";
        this["Func_impower_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_impower_ArgDesc_1"] = "Potęga, do której ma zostać podniesiona Liczba_zespolona.";
        this["Func_impower_Category"] = "Inżynieria";
        this["Func_impower_CategoryURL"] = "Inżynieria";
        this["Func_impower_Desc"] = "Zwraca liczbę zespoloną podniesioną do podanej potęgi.";
        this["Func_improduct_Arg_0"] = "Liczba_zespolona {0}";
        this["Func_improduct_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_improduct_Category"] = "Inżynieria";
        this["Func_improduct_CategoryURL"] = "Inżynieria";
        this["Func_improduct_Desc"] = "Zwraca iloczyn co najmniej jednej liczby zespolonej.";
        this["Func_imreal_Arg_0"] = "Liczba_zespolona";
        this["Func_imreal_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imreal_Category"] = "Inżynieria";
        this["Func_imreal_CategoryURL"] = "Inżynieria";
        this["Func_imreal_Desc"] = "Zwraca część rzeczywistą liczby zespolonej.";
        this["Func_imsin_Arg_0"] = "Liczba_zespolona";
        this["Func_imsin_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imsin_Category"] = "Inżynieria";
        this["Func_imsin_CategoryURL"] = "Inżynieria";
        this["Func_imsin_Desc"] = "Zwraca sinus liczby zespolonej.";
        this["Func_imsqrt_Arg_0"] = "Liczba_zespolona";
        this["Func_imsqrt_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imsqrt_Category"] = "Inżynieria";
        this["Func_imsqrt_CategoryURL"] = "Inżynieria";
        this["Func_imsqrt_Desc"] = "Zwraca pierwiastek kwadratowy liczby zespolonej.";
        this["Func_imsub_Arg_0"] = "Liczba_zespolona_1";
        this["Func_imsub_Arg_1"] = "Liczba_zespolona_2";
        this["Func_imsub_ArgDesc_0"] = "Tekstowa reprezentacja liczby zespolonej w formacie 'x + yi' lub 'x + yj', od której zostanie odjęta Liczba_zespolona_2.";
        this["Func_imsub_ArgDesc_1"] = "Tekstowa reprezentacja liczby zespolonej w formacie 'x + yi' lub 'x + yj', która zostanie odjęta od Liczby_zespolonej_1.";
        this["Func_imsub_Category"] = "Inżynieria";
        this["Func_imsub_CategoryURL"] = "Inżynieria";
        this["Func_imsub_Desc"] = "Zwraca różnicę między dwiema liczbami zespolonymi.";
        this["Func_imsum_Arg_0"] = "Liczba_zespolona {0}";
        this["Func_imsum_ArgDesc_0"] = "Reprezentacja tekstowa liczby zespolonej w formacie: 'x + yi' lub 'x + yj'.";
        this["Func_imsum_Category"] = "Inżynieria";
        this["Func_imsum_CategoryURL"] = "Inżynieria";
        this["Func_imsum_Desc"] = "Zwraca sumę co najmniej jednej liczby zespolonej.";
        this["Func_info_Arg_0"] = "Typ";
        this["Func_info_ArgDesc_0"] = "Wyrażenie tekstowe określające typ informacji do zwrócenia.\r\n'directory' = pełna ścieżka do bieżącego folderu.\r\n'osversion\" = Bieżąca wersja systemu operacyjnego jako wyrażenie tekstowe.\r\n'system' = bieżący system operacyjny. Macintosh = 'mac', Windows = 'pcdos'.\r\n    ";
        this["Func_info_Category"] = "Informacja";
        this["Func_info_CategoryURL"] = "Informacja";
        this["Func_info_Desc"] = "Zwraca informacje o bieżącym środowisku operacyjnym.";
        this["Func_int_Arg_0"] = "liczba";
        this["Func_int_ArgDesc_0"] = "Liczba rzeczywista, która ma zostać zaokrąglona w dół do liczby całkowitej.";
        this["Func_int_Category"] = "Matematyczne";
        this["Func_int_CategoryURL"] = "Matematyczne";
        this["Func_int_Desc"] = "Zaokrągla liczbę w dół do najbliższej liczby całkowitej.";
        this["Func_intrate_Arg_0"] = "rozliczenie";
        this["Func_intrate_Arg_1"] = "data_spłaty";
        this["Func_intrate_Arg_2"] = "lokata";
        this["Func_intrate_Arg_3"] = "dyskonto";
        this["Func_intrate_Arg_4"] = "podstawa";
        this["Func_intrate_ArgDesc_0"] = "Data rozliczenia papieru wartościowego.";
        this["Func_intrate_ArgDesc_1"] = "Data spłaty papieru wartościowego.";
        this["Func_intrate_ArgDesc_2"] = "Kwota zainwestowana w papier wartościowy.";
        this["Func_intrate_ArgDesc_3"] = "Kwota do otrzymania przy wykupie.";
        this["Func_intrate_ArgDesc_4"] = "Typ podstawy wyliczania dni, który zostanie użyty.\r\n0 lub pominięty = Amerykańska (NASD) 30/360 \r\n1 = Rzeczywiste/rzeczywiste \r\n2 = Rzeczywiste/360 \r\n3 = Rzeczywiste/365 \r\n4 = Europejska 30/360\r\n    ";
        this["Func_intrate_Category"] = "Finansowe";
        this["Func_intrate_CategoryURL"] = "Finansowe";
        this["Func_intrate_Desc"] = "Zwraca stopę procentową dla w papieru wartościowego całkowicie ulokowanego.";
        this["Func_ipmt_Arg_0"] = "stopa";
        this["Func_ipmt_Arg_1"] = "okres";
        this["Func_ipmt_Arg_2"] = "nper";
        this["Func_ipmt_Arg_3"] = "pv";
        this["Func_ipmt_Arg_4"] = "fv";
        this["Func_ipmt_Arg_5"] = "type";
        this["Func_ipmt_ArgDesc_0"] = "Stopa procentowa dla okresu.";
        this["Func_ipmt_ArgDesc_1"] = "Okres, który musi mieścić się w przedziale od 1 do Liczba_rat.";
        this["Func_ipmt_ArgDesc_2"] = "Całkowita liczba okresów płatności w ramach raty rocznej.";
        this["Func_ipmt_ArgDesc_3"] = "Wartość bieżąca lub kwota ryczałtowa, jaką jest teraz warta seria przyszłych płatności.";
        this["Func_ipmt_ArgDesc_4"] = "Wartość przyszła lub saldo gotówkowe, które chcesz osiągnąć po dokonaniu ostatniej płatności. Jeśli nic nie zostanie wpisane, przyjmowana jest wartość zero (0).";
        this["Func_ipmt_ArgDesc_5"] = "Wartość zero lub jeden, która wskazuje, czy płatności są należne na koniec okresu (zero, wartość domyślna), czy na początek (jeden).";
        this["Func_ipmt_Category"] = "Finansowe";
        this["Func_ipmt_CategoryURL"] = "Finansowe";
        this["Func_ipmt_Desc"] = "Zwraca wartość płatności odsetek dla danego okresu inwestycji przy założeniu okresowych, stałych płatności i stałej stopy procentowej.";
        this["Func_irr_Arg_0"] = "wartości";
        this["Func_irr_Arg_1"] = "wynik";
        this["Func_irr_ArgDesc_0"] = "Kolekcja wartości płatności (wartości ujemne) i dochodu (wartości dodatnie), które występują w regularnych odstępach czasu.";
        this["Func_irr_ArgDesc_1"] = "Liczba, która według Ciebie jest bliska wyniku. Wartość domyślna to 0,1 (10%).";
        this["Func_irr_Category"] = "Finansowe";
        this["Func_irr_CategoryURL"] = "Finansowe";
        this["Func_irr_Desc"] = "Zwraca wewnętrzną stopę zwrotu dla serii przepływów środków pieniężnych.";
        this["Func_isblank_Arg_0"] = "wartość";
        this["Func_isblank_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_isblank_Category"] = "Informacja";
        this["Func_isblank_CategoryURL"] = "Informacja";
        this["Func_isblank_Desc"] = "Zwraca wartość PRAWDA, jeśli dana wartość jest pusta.";
        this["Func_isdbnull_Arg_0"] = "wartość";
        this["Func_isdbnull_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_isdbnull_Category"] = "Informacja";
        this["Func_isdbnull_CategoryURL"] = "Informacja";
        this["Func_isdbnull_Desc"] = "Zwraca wartość PRAWDA, jeśli wartość reprezentuje nieistniejącą wartość.";
        this["Func_iserr_Arg_0"] = "wartość";
        this["Func_iserr_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_iserr_Category"] = "Informacja";
        this["Func_iserr_CategoryURL"] = "Informacja";
        this["Func_iserr_Desc"] = "Zwraca wartość PRAWDA, jeśli dana wartość jest wartością błędu, z wyjątkiem #N/D.";
        this["Func_iserror_Arg_0"] = "wartość";
        this["Func_iserror_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_iserror_Category"] = "Informacja";
        this["Func_iserror_CategoryURL"] = "Informacja";
        this["Func_iserror_Desc"] = "Zwraca wartość PRAWDA, jeśli dana wartość jest wartością błędu.";
        this["Func_iseven_Arg_0"] = "wartość";
        this["Func_iseven_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_iseven_Category"] = "Informacja";
        this["Func_iseven_CategoryURL"] = "Informacja";
        this["Func_iseven_Desc"] = "Zwraca wartość PRAWDA, jeśli wartość jest liczbą parzystą; w przeciwnym razie zwraca wartość FAŁSZ.";
        this["Func_islogical_Arg_0"] = "wartość";
        this["Func_islogical_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_islogical_Category"] = "Informacja";
        this["Func_islogical_CategoryURL"] = "Informacja";
        this["Func_islogical_Desc"] = "Zwraca wartość PRAWDA, jeśli dana wartość jest wartością logiczną (PRAWDA lub FAŁSZ).";
        this["Func_isna_Arg_0"] = "wartość";
        this["Func_isna_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_isna_Category"] = "Informacja";
        this["Func_isna_CategoryURL"] = "Informacja";
        this["Func_isna_Desc"] = "Zwraca wartość PRAWDA, jeśli wartość jest wartością błędu #N/D (wartość niedostępna).";
        this["Func_isnontext_Arg_0"] = "wartość";
        this["Func_isnontext_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_isnontext_Category"] = "Informacja";
        this["Func_isnontext_CategoryURL"] = "Informacja";
        this["Func_isnontext_Desc"] = "Zwraca wartość PRAWDA, jeśli wartość jest dowolnym elementem, który nie jest tekstem.";
        this["Func_isnull_Arg_0"] = "wartość";
        this["Func_isnull_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_isnull_Category"] = "Informacja";
        this["Func_isnull_CategoryURL"] = "Informacja";
        this["Func_isnull_Desc"] = "Zwraca wartość PRAWDA, jeśli dana wartość nie istnieje.";
        this["Func_isnumber_Arg_0"] = "wartość";
        this["Func_isnumber_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_isnumber_Category"] = "Informacja";
        this["Func_isnumber_CategoryURL"] = "Informacja";
        this["Func_isnumber_Desc"] = "Zwraca wartość PRAWDA, jeśli dana wartość jest liczbą.";
        this["Func_isodd_Arg_0"] = "wartość";
        this["Func_isodd_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_isodd_Category"] = "Informacja";
        this["Func_isodd_CategoryURL"] = "Informacja";
        this["Func_isodd_Desc"] = "Zwraca wartość PRAWDA, jeśli wartość jest liczbą nieparzystą; w przeciwnym razie zwraca wartość FAŁSZ.";
        this["Func_isref_Arg_0"] = "wartość";
        this["Func_isref_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_isref_Category"] = "Informacja";
        this["Func_isref_CategoryURL"] = "Informacja";
        this["Func_isref_Desc"] = "Zwraca wartość PRAWDA, jeśli dana wartość jest odwołaniem.";
        this["Func_istext_Arg_0"] = "wartość";
        this["Func_istext_ArgDesc_0"] = "Wartość do przetestowania.";
        this["Func_istext_Category"] = "Informacja";
        this["Func_istext_CategoryURL"] = "Informacja";
        this["Func_istext_Desc"] = "Zwraca wartość PRAWDA, jeśli dana wartość jest wartością tekstową.";
        this["Func_lcm_Arg_0"] = "liczba{0}";
        this["Func_lcm_ArgDesc_0"] = "Wartość całkowita. Wartość nie będąca liczbą całkowitą zostanie obcięta.";
        this["Func_lcm_Category"] = "Matematyczne";
        this["Func_lcm_CategoryURL"] = "Matematyczne";
        this["Func_lcm_Desc"] = "Zwraca najmniejszą wspólną wielokrotność co najmniej jednej wartości całkowitej.";
        this["Func_left_Arg_0"] = "tekst";
        this["Func_left_Arg_1"] = "num_chars";
        this["Func_left_ArgDesc_0"] = "Wyrażenie tekstowe zawierające znaki do wyodrębnienia.";
        this["Func_left_ArgDesc_1"] = "Liczba znaków do wyodrębnienia.";
        this["Func_left_Category"] = "TekstDane";
        this["Func_left_CategoryURL"] = "Text_and_Data";
        this["Func_left_Desc"] = "Zwraca pierwszy znak lub pierwsze znaki w wyrażeniu tekstowym.";
        this["Func_len_Arg_0"] = "tekst";
        this["Func_len_ArgDesc_0"] = "Wyrażenie tekstowe, którego długość ma zostać zwrócona. Spacje liczą się jako znaki.";
        this["Func_len_Category"] = "TekstDane";
        this["Func_len_CategoryURL"] = "Text_and_Data";
        this["Func_len_Desc"] = "Zwraca liczbę znaków w wyrażeniu tekstowym.";
        this["Func_ln_Arg_0"] = "liczba";
        this["Func_ln_ArgDesc_0"] = "Dodatnia liczba rzeczywista, której logarytm naturalny ma zostać obliczony.";
        this["Func_ln_Category"] = "Matematyczne";
        this["Func_ln_CategoryURL"] = "Matematyczne";
        this["Func_ln_Desc"] = "Zwraca logarytm naturalny liczby.";
        this["Func_log_Arg_0"] = "liczba";
        this["Func_log_Arg_1"] = "podstawa";
        this["Func_log_ArgDesc_0"] = "Dodatnia liczba rzeczywista, której logarytm ma zostać obliczony.";
        this["Func_log_ArgDesc_1"] = "Podstawa logarytmu.";
        this["Func_log_Category"] = "Matematyczne";
        this["Func_log_CategoryURL"] = "Matematyczne";
        this["Func_log_Desc"] = "Zwraca logarytm liczby o określonej podstawie.";
        this["Func_log10_Arg_0"] = "liczba";
        this["Func_log10_ArgDesc_0"] = "Dodatnia liczba rzeczywista, której logarytm dziesiętny ma zostać obliczony.";
        this["Func_log10_Category"] = "Matematyczne";
        this["Func_log10_CategoryURL"] = "Matematyczne";
        this["Func_log10_Desc"] = "Zwraca logarytm dziesiętny liczby.";
        this["Func_lower_Arg_0"] = "tekst";
        this["Func_lower_ArgDesc_0"] = "Tekst do skonwertowania na małe litery.";
        this["Func_lower_Category"] = "TekstDane";
        this["Func_lower_CategoryURL"] = "Text_and_Data";
        this["Func_lower_Desc"] = "Konwertuje wszystkie wielkie litery w wyrażeniu tekstowym na małe litery.";
        this["Func_max_Arg_0"] = "liczba{0}";
        this["Func_max_ArgDesc_0"] = "Liczba lub zakres liczb.";
        this["Func_max_Category"] = "Statystyczne";
        this["Func_max_CategoryURL"] = "Statystyczne";
        this["Func_max_Desc"] = "Zwraca największą liczbę ze zbioru liczb.";
        this["Func_median_Arg_0"] = "liczba{0}";
        this["Func_median_ArgDesc_0"] = "Liczba lub zakres liczb.";
        this["Func_median_Category"] = "Statystyczne";
        this["Func_median_CategoryURL"] = "Statystyczne";
        this["Func_median_Desc"] = "Zwraca medianę zbioru liczb. Mediana to środkowa liczba zbioru, gdy liczby są posortowane od najmniejszej do największej, lub średnia dwóch środkowych liczb, jeśli łączna liczba elementów zbioru jest parzysta.";
        this["Func_mid_Arg_0"] = "tekst";
        this["Func_mid_Arg_1"] = "start_num";
        this["Func_mid_Arg_2"] = "num_chars";
        this["Func_mid_ArgDesc_0"] = "Wyrażenie tekstowe zawierające znaki do wyodrębnienia.";
        this["Func_mid_ArgDesc_1"] = "Numer kolejny (liczony od 1) pierwszego znaku do wyodrębnienia.";
        this["Func_mid_ArgDesc_2"] = "Liczba znaków do wyodrębnienia.";
        this["Func_mid_Category"] = "TekstDane";
        this["Func_mid_CategoryURL"] = "Text_and_Data";
        this["Func_mid_Desc"] = "Zwraca określoną liczbę znaków z wyrażenia tekstowego.";
        this["Func_min_Arg_0"] = "liczba{0}";
        this["Func_min_ArgDesc_0"] = "Liczba lub zakres liczb.";
        this["Func_min_Category"] = "Statystyczne";
        this["Func_min_CategoryURL"] = "Statystyczne";
        this["Func_min_Desc"] = "Zwraca najmniejszą liczbę ze zbioru liczb.";
        this["Func_minute_Arg_0"] = "Time_value";
        this["Func_minute_ArgDesc_0"] = "Wartość czasu, z której ma zostać pobrana minuta.";
        this["Func_minute_Category"] = "DataIGodzina";
        this["Func_minute_CategoryURL"] = "Date_and_Time";
        this["Func_minute_Desc"] = "Zwraca minutę z wartości czasu.";
        this["Func_mod_Arg_0"] = "liczba";
        this["Func_mod_Arg_1"] = "dzielnik";
        this["Func_mod_ArgDesc_0"] = "Liczba, dla której ma zostać znaleziona reszta z dzielenia.";
        this["Func_mod_ArgDesc_1"] = "Wartość, przez którą należy podzielić liczbę.";
        this["Func_mod_Category"] = "Matematyczne";
        this["Func_mod_CategoryURL"] = "Matematyczne";
        this["Func_mod_Desc"] = "Zwraca resztę po podzieleniu liczby przez dzielnik. Wynik ma ten sam znak, co dzielnik.";
        this["Func_month_Arg_0"] = "Date_value";
        this["Func_month_ArgDesc_0"] = "Wartość daty, z której ma zostać pobrany miesiąc.";
        this["Func_month_Category"] = "DataIGodzina";
        this["Func_month_CategoryURL"] = "Date_and_Time";
        this["Func_month_Desc"] = "Zwraca miesiąc z wartości daty.";
        this["Func_mround_Arg_0"] = "Liczba";
        this["Func_mround_Arg_1"] = "Wielokrotność";
        this["Func_mround_ArgDesc_0"] = "Liczba do zaokrąglenia.";
        this["Func_mround_ArgDesc_1"] = "Liczba wskazująca wielokrotność, do której należy zaokrąglić.";
        this["Func_mround_Category"] = "Matematyczne";
        this["Func_mround_CategoryURL"] = "Matematyczne";
        this["Func_mround_Desc"] = "Zaokrągla liczbę do najbliższej wielokrotności innej liczby.";
        this["Func_multinomial_Arg_0"] = "liczba{0}";
        this["Func_multinomial_ArgDesc_0"] = "Jedna z liczb, dla których ma zostać uzyskany wielomian.";
        this["Func_multinomial_Category"] = "Matematyczne";
        this["Func_multinomial_CategoryURL"] = "Matematyczne";
        this["Func_multinomial_Desc"] = "Zwraca wielomian ze zbioru liczb. Wielomian to stosunek silni sumy wartości do iloczynu silni tych wartości.";
        this["Func_n_Arg_0"] = "Wartość";
        this["Func_n_ArgDesc_0"] = "Wartość, która ma zostać skonwertowana na liczbę. Dopuszczalne wartości:\r\nLiczba - zwraca liczbę.\r\nData - zwraca datę.\r\nPrawda - zwraca 1.\r\nFałsz - zwraca 0.\r\nWartość błędu - zwraca wartość błędu.\r\nDowolna inna wartość - zwraca 0.\r\n    ";
        this["Func_n_Category"] = "Informacja";
        this["Func_n_CategoryURL"] = "Informacja";
        this["Func_n_Desc"] = "Konwertuje wartość na liczbę.";
        this["Func_na_Category"] = "Informacja";
        this["Func_na_CategoryURL"] = "Informacja";
        this["Func_na_Desc"] = "Zwraca wartość błędu #N/D (wartość niedostępna).";
        this["Func_networkdays_Arg_0"] = "dataPoczątkowa";
        this["Func_networkdays_Arg_1"] = "dataKońcowa";
        this["Func_networkdays_Arg_2"] = "święta";
        this["Func_networkdays_ArgDesc_0"] = "Data początkowa.";
        this["Func_networkdays_ArgDesc_1"] = "Data końcowa.";
        this["Func_networkdays_ArgDesc_2"] = "Zakres obejmujący co najmniej jedno odwołanie zawierające daty, które mają być uważane za dni wolne od pracy.";
        this["Func_networkdays_Category"] = "DataIGodzina";
        this["Func_networkdays_CategoryURL"] = "DataIGodzina";
        this["Func_networkdays_Desc"] = "Zwraca łączną liczbę pełnych dni roboczych między podanymi datami, z wyłączeniem wszelkich dni wolnych od pracy.";
        this["Func_not_Arg_0"] = "Logiczne";
        this["Func_not_ArgDesc_0"] = "Wartość logiczna (PRAWDA lub FAŁSZ).";
        this["Func_not_Category"] = "Logiczne";
        this["Func_not_CategoryURL"] = "Logiczne";
        this["Func_not_Desc"] = "Odwraca wartość logiczną (zmienia PRAWDA na FAŁSZ i odwrotnie).";
        this["Func_now_Category"] = "DataIGodzina";
        this["Func_now_CategoryURL"] = "Date_and_Time";
        this["Func_now_Desc"] = "Zwraca wartość reprezentującą bieżącą datę i godzinę.";
        this["Func_nper_Arg_0"] = "stopa";
        this["Func_nper_Arg_1"] = "pmt";
        this["Func_nper_Arg_2"] = "pv";
        this["Func_nper_Arg_3"] = "fv";
        this["Func_nper_Arg_4"] = "type";
        this["Func_nper_ArgDesc_0"] = "Stopa procentowa dla okresu.";
        this["Func_nper_ArgDesc_1"] = "Płatność dokonywana w każdym okresie, który nie może ulec zmianie w okresie obowiązywania raty rocznej.";
        this["Func_nper_ArgDesc_2"] = "Wartość bieżąca lub kwota ryczałtowa, jaką jest teraz warta seria przyszłych płatności.";
        this["Func_nper_ArgDesc_3"] = "Wartość przyszła lub saldo gotówkowe, które chcesz osiągnąć po dokonaniu ostatniej płatności.";
        this["Func_nper_ArgDesc_4"] = "Wartość zero lub jeden, która wskazuje, czy płatności są należne na koniec okresu (zero, wartość domyślna), czy na początek (jeden).";
        this["Func_nper_Category"] = "Finansowe";
        this["Func_nper_CategoryURL"] = "Finansowe";
        this["Func_nper_Desc"] = "Zwraca liczbę okresów dla inwestycji przy założeniu okresowych, stałych płatności i stałej stopy procentowej.";
        this["Func_npv_Arg_0"] = "stopa";
        this["Func_npv_Arg_1"] = "wartość{0}";
        this["Func_npv_ArgDesc_0"] = "Stopa dyskontowa dla podstawowego okresu płatności.";
        this["Func_npv_ArgDesc_1"] = "Płatność (wartość ujemna) lub wpływ (wartość dodatnia).";
        this["Func_npv_Category"] = "Finansowe";
        this["Func_npv_CategoryURL"] = "Finansowe";
        this["Func_npv_Desc"] = "Oblicza bieżącą wartość netto inwestycji obliczoną przy użyciu stopy dyskontowej i serii przyszłych płatności (wartości ujemne) i wpływów (wartość dodatnia).";
        this["Func_null_Category"] = "Informacja";
        this["Func_null_CategoryURL"] = "Informacja";
        this["Func_null_Desc"] = "Zwraca nieistniejącą wartość.";
        this["Func_oct2bin_Arg_0"] = "Liczba";
        this["Func_oct2bin_Arg_1"] = "Miejsca";
        this["Func_oct2bin_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące liczbę ósemkową.";
        this["Func_oct2bin_ArgDesc_1"] = "Liczba znaków używanych do przedstawienia wartości. Jeśli nie zostanie to określone, zostanie użyta minimalna liczba znaków niezbędnych do przedstawienia wartości. Pozwala to określić liczbę zer wiodących.";
        this["Func_oct2bin_Category"] = "Inżynieria";
        this["Func_oct2bin_CategoryURL"] = "Inżynieria";
        this["Func_oct2bin_Desc"] = "Konwertuje wyrażenie tekstowe reprezentujące liczbę ósemkową na wyrażenie tekstowe reprezentujące tę samą liczbę w binarnym systemie liczbowym.";
        this["Func_oct2dec_Arg_0"] = "Liczba";
        this["Func_oct2dec_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące liczbę ósemkową.";
        this["Func_oct2dec_Category"] = "Inżynieria";
        this["Func_oct2dec_CategoryURL"] = "Inżynieria";
        this["Func_oct2dec_Desc"] = "Zwraca wartość dziesiętną tekstowej reprezentacji liczby w ósemkowym systemie liczbowym.";
        this["Func_oct2hex_Arg_0"] = "Liczba";
        this["Func_oct2hex_Arg_1"] = "Miejsca";
        this["Func_oct2hex_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące liczbę ósemkową.";
        this["Func_oct2hex_ArgDesc_1"] = "Liczba znaków używanych do przedstawienia wartości. Jeśli nie zostanie to określone, zostanie użyta minimalna liczba znaków niezbędnych do przedstawienia wartości. Pozwala to określić liczbę zer wiodących.";
        this["Func_oct2hex_Category"] = "Inżynieria";
        this["Func_oct2hex_CategoryURL"] = "Inżynieria";
        this["Func_oct2hex_Desc"] = "Konwertuje wyrażenie tekstowe reprezentujące liczbę ósemkową na wyrażenie tekstowe reprezentujące tę samą liczbę w szesnastkowym systemie liczbowym.";
        this["Func_odd_Arg_0"] = "liczba";
        this["Func_odd_ArgDesc_0"] = "Wartość do zaokrąglenia.";
        this["Func_odd_Category"] = "Matematyczne";
        this["Func_odd_CategoryURL"] = "Matematyczne";
        this["Func_odd_Desc"] = "Zaokrągla liczbę dodatnią w górę, a liczbę ujemną w dół do najbliższej nieparzystej liczby całkowitej.";
        this["Func_or_Arg_0"] = "logiczne{0}";
        this["Func_or_ArgDesc_0"] = "Wartość logiczna (PRAWDA lub FAŁSZ).";
        this["Func_or_Category"] = "Logiczne";
        this["Func_or_CategoryURL"] = "Logiczne";
        this["Func_or_Desc"] = "Zwraca wartość PRAWDA, jeśli co najmniej jedna wartość jest prawdziwa, lub FAŁSZ, jeśli wszystkie wartości są fałszywe.";
        this["Func_pi_Category"] = "Matematyczne";
        this["Func_pi_CategoryURL"] = "Matematyczne";
        this["Func_pi_Desc"] = "Zwraca stałą matematyczną Pi (liczbę 3,14159265358979).";
        this["Func_pmt_Arg_0"] = "stopa";
        this["Func_pmt_Arg_1"] = "nper";
        this["Func_pmt_Arg_2"] = "pv";
        this["Func_pmt_Arg_3"] = "fv";
        this["Func_pmt_Arg_4"] = "type";
        this["Func_pmt_ArgDesc_0"] = "Stopa procentowa (oprocentowanie) pożyczki.";
        this["Func_pmt_ArgDesc_1"] = "Całkowita liczba płatności (rat) pożyczki.";
        this["Func_pmt_ArgDesc_2"] = "Wartość bieżąca lub całkowita kwota, jaką jest teraz warta seria przyszłych płatności; znana również jako zabezpieczenie.";
        this["Func_pmt_ArgDesc_3"] = "Wartość przyszła lub saldo gotówkowe, które chcesz osiągnąć po dokonaniu ostatniej płatności.";
        this["Func_pmt_ArgDesc_4"] = "Wartość zero lub jeden, która wskazuje, czy płatności są należne na koniec okresu (zero, wartość domyślna), czy na początek (jeden).";
        this["Func_pmt_Category"] = "Finansowe";
        this["Func_pmt_CategoryURL"] = "Finansowe";
        this["Func_pmt_Desc"] = "Oblicza wartość płatności pożyczki przy założeniu stałych płatności i stałej stopy procentowej.";
        this["Func_power_Arg_0"] = "liczba";
        this["Func_power_Arg_1"] = "potęga";
        this["Func_power_ArgDesc_0"] = "Podstawa potęgi, czyli liczba, która ma zostać podniesienia do potęgi.";
        this["Func_power_ArgDesc_1"] = "Wykładnik, do którego podnoszona jest podstawa.";
        this["Func_power_Category"] = "Matematyczne";
        this["Func_power_CategoryURL"] = "Matematyczne";
        this["Func_power_Desc"] = "Zwraca liczbę podniesioną do potęgi.";
        this["Func_ppmt_Arg_0"] = "stopa";
        this["Func_ppmt_Arg_1"] = "okres";
        this["Func_ppmt_Arg_2"] = "nper";
        this["Func_ppmt_Arg_3"] = "pv";
        this["Func_ppmt_Arg_4"] = "fv";
        this["Func_ppmt_Arg_5"] = "type";
        this["Func_ppmt_ArgDesc_0"] = "Stopa procentowa dla okresu.";
        this["Func_ppmt_ArgDesc_1"] = "Okres, który musi mieścić się w przedziale od 1 do Liczba_rat.";
        this["Func_ppmt_ArgDesc_2"] = "Całkowita liczba okresów płatności w ramach raty rocznej.";
        this["Func_ppmt_ArgDesc_3"] = "Wartość bieżąca lub całkowita kwota, jaką jest teraz warta seria przyszłych płatności.";
        this["Func_ppmt_ArgDesc_4"] = "Wartość przyszła lub saldo gotówkowe, które chcesz osiągnąć po dokonaniu ostatniej płatności.";
        this["Func_ppmt_ArgDesc_5"] = "Wartość zero lub jeden, która wskazuje, czy płatności są należne na koniec okresu (zero, wartość domyślna), czy na początek (jeden).";
        this["Func_ppmt_Category"] = "Finansowe";
        this["Func_ppmt_CategoryURL"] = "Finansowe";
        this["Func_ppmt_Desc"] = "Zwraca wartość płatności odsetek z danej lokaty dla danego okresu inwestycji przy założeniu okresowych, stałych płatności i stałej stopy procentowej.";
        this["Func_product_Arg_0"] = "liczba{0}";
        this["Func_product_ArgDesc_0"] = "Liczba do przemnożenia przez inne liczby.";
        this["Func_product_Category"] = "Matematyczne";
        this["Func_product_CategoryURL"] = "Matematyczne";
        this["Func_product_Desc"] = "Mnoży wszystkie liczby podane jako argumenty i zwraca iloczyn.";
        this["Func_pv_Arg_0"] = "stopa";
        this["Func_pv_Arg_1"] = "nper";
        this["Func_pv_Arg_2"] = "pmt";
        this["Func_pv_Arg_3"] = "fv";
        this["Func_pv_Arg_4"] = "type";
        this["Func_pv_ArgDesc_0"] = "Stopa procentowa dla okresu.";
        this["Func_pv_ArgDesc_1"] = "Całkowita liczba okresów płatności w ramach raty rocznej.";
        this["Func_pv_ArgDesc_2"] = "Płatność dokonywana w każdym okresie, który nie może ulec zmianie w okresie obowiązywania raty rocznej.";
        this["Func_pv_ArgDesc_3"] = "Wartość przyszła lub saldo gotówkowe, które chcesz osiągnąć po dokonaniu ostatniej płatności.";
        this["Func_pv_ArgDesc_4"] = "Wartość zero lub jeden, która wskazuje, czy płatności są należne na koniec okresu (zero, wartość domyślna), czy na początek (jeden).";
        this["Func_pv_Category"] = "Finansowe";
        this["Func_pv_CategoryURL"] = "Finansowe";
        this["Func_pv_Desc"] = "Zwraca bieżącą wartość inwestycji.";
        this["Func_quotient_Arg_0"] = "licznik";
        this["Func_quotient_Arg_1"] = "mianownik";
        this["Func_quotient_ArgDesc_0"] = "Dzielna operacji dzielenia.";
        this["Func_quotient_ArgDesc_1"] = "Dzielnik operacji dzielenia.";
        this["Func_quotient_Category"] = "Matematyczne";
        this["Func_quotient_CategoryURL"] = "Matematyczne";
        this["Func_quotient_Desc"] = "Zwraca całkowitą część operacji dzielenia. Spowoduje to odrzucenie pozostałej części operacji dzielenia.";
        this["Func_radians_Arg_0"] = "Stopnie";
        this["Func_radians_ArgDesc_0"] = "Wartość w stopniach, która ma zostać skonwertowana na radiany.";
        this["Func_radians_Category"] = "Matematyczne";
        this["Func_radians_CategoryURL"] = "Matematyczne";
        this["Func_radians_Desc"] = "Konwertuje stopnie na radiany.";
        this["Func_rand_Category"] = "Matematyczne";
        this["Func_rand_CategoryURL"] = "Matematyczne";
        this["Func_rand_Desc"] = "Zwraca liczbę losową o równomiernym rozkładzie, która jest większa lub równa 0 i mniejsza niż 1.";
        this["Func_randbetween_Arg_0"] = "Dół";
        this["Func_randbetween_Arg_1"] = "Góra";
        this["Func_randbetween_ArgDesc_0"] = "Minimalna wartość, która ma zostać zwrócona.";
        this["Func_randbetween_ArgDesc_1"] = "Maksymalna wartość, która ma zostać zwrócona.";
        this["Func_randbetween_Category"] = "Matematyczne";
        this["Func_randbetween_CategoryURL"] = "Matematyczne";
        this["Func_randbetween_Desc"] = "Generuje pseudolosową liczbę całkowitą z zakresu między dwiema podanymi liczbami.";
        this["Func_rate_Arg_0"] = "Liczba_rat";
        this["Func_rate_Arg_1"] = "Spłata";
        this["Func_rate_Arg_2"] = "Wart_obecna";
        this["Func_rate_Arg_3"] = "Wart_przyszła";
        this["Func_rate_Arg_4"] = "Typ";
        this["Func_rate_Arg_5"] = "Guess";
        this["Func_rate_ArgDesc_0"] = "Całkowita liczba okresów płatności w ramach raty rocznej.";
        this["Func_rate_ArgDesc_1"] = "Płatność dokonywana w każdym okresie. Nie może ulec zmianie w okresie obowiązywania raty rocznej.";
        this["Func_rate_ArgDesc_2"] = "Wartość bieżąca lub całkowita kwota, jaką jest teraz warta seria przyszłych płatności.";
        this["Func_rate_ArgDesc_3"] = "Wartość przyszła lub saldo gotówkowe, które chcesz osiągnąć po dokonaniu ostatniej płatności. Jeśli wartość Wart_przyszła nie zostanie wpisana, przyjmuje się, że wynosi 0 (na przykład przyszła wartość pożyczki wynosi 0). Jeśli na przykład chcesz zaoszczędzić 50 000 USD, aby zapłacić za specjalny projekt w ciągu 18 lat, wówczas przyszłą wartością jest 50 000 USD. Możesz wtedy ostrożnie oszacować stopy procentowe i ustalić, ile musisz oszczędzać każdego miesiąca. Jeśli wartość Wart_przyszła nie zostanie wpisana, należy określić argument pmt. Domyślna wartość Wart_przyszła wynosi zero.";
        this["Func_rate_ArgDesc_4"] = "Wartość zero lub jeden, która wskazuje, czy płatności są należne na koniec okresu (zero, wartość domyślna), czy na początek (jeden).";
        this["Func_rate_ArgDesc_5"] = "Liczba, która według Ciebie jest bliska wyniku. Jeśli nic nie zostanie wpisane, przyjmowana jest wartość 0,1 (10%).";
        this["Func_rate_Category"] = "Finansowe";
        this["Func_rate_CategoryURL"] = "Finansowe";
        this["Func_rate_Desc"] = "Zwraca stopę procentową okresu raty rocznej.";
        this["Func_replace_Arg_0"] = "Old_text";
        this["Func_replace_Arg_1"] = "Start_num";
        this["Func_replace_Arg_2"] = "Num_chars";
        this["Func_replace_Arg_3"] = "New_text";
        this["Func_replace_ArgDesc_0"] = "Wyrażenie tekstowe, w którym mają zostać zamienione niektóre znaki.";
        this["Func_replace_ArgDesc_1"] = "Numer kolejny (liczony od 1) znaku w starym_tekście, który ma zostać zastąpiony przez new_text.";
        this["Func_replace_ArgDesc_2"] = "Liczba znaków w starym_tekście do zastąpienia przez nowy_tekst.";
        this["Func_replace_ArgDesc_3"] = "Tekst, który zastąpi znaki w starym_tekście.";
        this["Func_replace_Category"] = "TekstDane";
        this["Func_replace_CategoryURL"] = "Text_and_Data";
        this["Func_replace_Desc"] = "Zastępuje część wyrażenia tekstowego.";
        this["Func_rept_Arg_0"] = "Wartość_tekstowa";
        this["Func_rept_Arg_1"] = "Liczba_powtórzeń";
        this["Func_rept_ArgDesc_0"] = "Tekst do powtórzenia.";
        this["Func_rept_ArgDesc_1"] = "Dodatnia liczba powtórzeń tekstu.";
        this["Func_rept_Category"] = "TekstDane";
        this["Func_rept_CategoryURL"] = "Text_and_Data";
        this["Func_rept_Desc"] = "Powtarza tekst podaną liczbę razy.";
        this["Func_right_Arg_0"] = "tekst";
        this["Func_right_Arg_1"] = "num_chars";
        this["Func_right_ArgDesc_0"] = "Wyrażenie tekstowe zawierające znaki do wyodrębnienia.";
        this["Func_right_ArgDesc_1"] = "Liczba znaków do wyodrębnienia.";
        this["Func_right_Category"] = "TekstDane";
        this["Func_right_CategoryURL"] = "Text_and_Data";
        this["Func_right_Desc"] = "Zwraca ostatni znak lub ostatnie znaki w wyrażeniu tekstowym.";
        this["Func_roman_Arg_0"] = "Liczba";
        this["Func_roman_Arg_1"] = "Format";
        this["Func_roman_ArgDesc_0"] = "Liczba do skonwertowania na liczbę rzymską.";
        this["Func_roman_ArgDesc_1"] = "Liczba lub wartość logiczna wskazująca, czy mają być używane klasyczne liczby rzymskie, czy ich skrócona wersja.\r\n0 = (Domyślnie) Klasyczne. (499 = 'CDXCIX')\r\n1 = Bardziej zwięzły format. (499 = 'LDVLIV')\r\n2 = Bardziej zwięzły format. (499 = 'XDIX')\r\n3 = Bardziej zwięzły format. (499 = 'VDIV')\r\n4 = Uproszczony. (499 = 'ID')\r\nPrawda = Klasyczny.\r\nFałsz = Uproszczony.\r\n    ";
        this["Func_roman_Category"] = "Matematyczne";
        this["Func_roman_CategoryURL"] = "Matematyczne";
        this["Func_roman_Desc"] = "Konwertuje liczbę na liczbę rzymską zapisaną jako wyrażenie tekstowe.";
        this["Func_round_Arg_0"] = "liczba";
        this["Func_round_Arg_1"] = "num_digits";
        this["Func_round_Arg_2"] = "round_Style";
        this["Func_round_ArgDesc_0"] = "Liczba do zaokrąglenia.";
        this["Func_round_ArgDesc_1"] = "Liczba cyfr, do których należy zaokrąglić liczbę.";
        this["Func_round_ArgDesc_2"] = "Liczba wskazująca styl zaokrąglania, który ma być używany.\r\n0: zaokrąglenie arytmetyczne (domyślne) - jeśli ostatnią znaczącą cyfrą w zaokrągleniu jest 5, liczba jest zaokrąglana w górę.\r\n1: zaokrąglenie bankierskie - jeśli ostatnią znaczącą cyfrą jest 5, poprzednia cyfra jest używana do określenia, czy zaokrąglić w górę, czy w dół. Jeśli cyfra jest parzysta, liczba jest zaokrąglana w dół. Jeśli jest nieparzysta, liczba jest zaokrąglana w górę.\r\n    ";
        this["Func_round_Category"] = "Matematyczne";
        this["Func_round_CategoryURL"] = "Matematyczne";
        this["Func_round_Desc"] = "Zaokrągla liczbę z dokładnością do określonej liczby cyfr.";
        this["Func_rounddown_Arg_0"] = "Liczba";
        this["Func_rounddown_Arg_1"] = "Cyfry";
        this["Func_rounddown_ArgDesc_0"] = "Liczba do zaokrąglenia w dół.";
        this["Func_rounddown_ArgDesc_1"] = "Liczba miejsc dziesiętnych do zaokrąglenia. Liczby dodatnie oznaczają miejsca po przecinku, a liczby ujemne oznaczają miejsca przed przecinkiem.";
        this["Func_rounddown_Category"] = "Matematyczne";
        this["Func_rounddown_CategoryURL"] = "Matematyczne";
        this["Func_rounddown_Desc"] = "Zaokrągla liczbę w dół do podanej liczby cyfr.";
        this["Func_roundup_Arg_0"] = "Liczba";
        this["Func_roundup_Arg_1"] = "Cyfry";
        this["Func_roundup_ArgDesc_0"] = "Liczba do zaokrąglenia w górę.";
        this["Func_roundup_ArgDesc_1"] = "Liczba miejsc dziesiętnych do zaokrąglenia. Liczby dodatnie oznaczają miejsca po przecinku, a liczby ujemne oznaczają miejsca przed przecinkiem.";
        this["Func_roundup_Category"] = "Matematyczne";
        this["Func_roundup_CategoryURL"] = "Matematyczne";
        this["Func_roundup_Desc"] = "Zaokrągla liczbę w górę do podanej liczby cyfr.";
        this["Func_search_Arg_0"] = "search_text";
        this["Func_search_Arg_1"] = "wartość";
        this["Func_search_Arg_2"] = "start_index";
        this["Func_search_ArgDesc_0"] = "Tekst, który ma zostać znaleziony w wartości.";
        this["Func_search_ArgDesc_1"] = "Tekst, w którym ma zostać znaleziony tekst_wyszukiwania.";
        this["Func_search_ArgDesc_2"] = "Numer kolejny znaku (liczony od 1), od którego rozpocznie się wyszukiwanie (jeśli zostanie pominięty, indeks_początkowy ma wartość 1).";
        this["Func_search_Category"] = "TekstDane";
        this["Func_search_CategoryURL"] = "TekstDane";
        this["Func_search_Desc"] = "Zwraca numer kolejny (liczony od 1) jednego ciągu w drugim; wyszukiwanie bez uwzględniania wielkości liter.";
        this["Func_searchb_Arg_0"] = "search_text";
        this["Func_searchb_Arg_1"] = "wartość";
        this["Func_searchb_Arg_2"] = "start_index";
        this["Func_searchb_ArgDesc_0"] = "Tekst, który ma zostać znaleziony w wartości.";
        this["Func_searchb_ArgDesc_1"] = "Tekst, w którym ma zostać znaleziony tekst_wyszukiwania.";
        this["Func_searchb_ArgDesc_2"] = "Numer kolejny znaku (liczony od 1), od którego rozpocznie się wyszukiwanie (jeśli zostanie pominięty, indeks_początkowy ma wartość 1).";
        this["Func_searchb_Category"] = "TekstDane";
        this["Func_searchb_CategoryURL"] = "TekstDane";
        this["Func_searchb_Desc"] = "Zwraca numer kolejny (liczony od 1) jednego ciągu w drugim; wyszukiwanie bez uwzględniania wielkości liter.";
        this["Func_second_Arg_0"] = "Time_value";
        this["Func_second_ArgDesc_0"] = "Wartość czasu, z której ma zostać pobrana sekunda.";
        this["Func_second_Category"] = "DataIGodzina";
        this["Func_second_CategoryURL"] = "Date_and_Time";
        this["Func_second_Desc"] = "Zwraca sekundę z wartości czasu.";
        this["Func_seriessum_Arg_0"] = "Wartość_wejściowa";
        this["Func_seriessum_Arg_1"] = "Potęga_początkowa";
        this["Func_seriessum_Arg_2"] = "Krok";
        this["Func_seriessum_Arg_3"] = "Współczynniki";
        this["Func_seriessum_ArgDesc_0"] = "Wartość wejściowa szeregu potęgowego.";
        this["Func_seriessum_ArgDesc_1"] = "Potęga początkowa, do której zostanie podniesiona Wartość_wejściowa.";
        this["Func_seriessum_ArgDesc_2"] = "Krok, o który wzrasta potęga dla każdego składnika w szeregu.";
        this["Func_seriessum_ArgDesc_3"] = "Zbiór współczynników, przez które mnoży się każdą kolejną potęgę Wartości_wejściowej.";
        this["Func_seriessum_Category"] = "Matematyczne";
        this["Func_seriessum_CategoryURL"] = "Matematyczne";
        this["Func_seriessum_Desc"] = "Zwraca sumę szeregu potęgowego.";
        this["Func_sign_Arg_0"] = "Liczba";
        this["Func_sign_ArgDesc_0"] = "Liczba, której znak zostanie zwrócony.";
        this["Func_sign_Category"] = "Matematyczne";
        this["Func_sign_CategoryURL"] = "Matematyczne";
        this["Func_sign_Desc"] = "Zwraca znak liczby (–1, 0 lub 1).";
        this["Func_sin_Arg_0"] = "liczba";
        this["Func_sin_ArgDesc_0"] = "Kąt w radianach, dla którego ma zostać uzyskany sinus.";
        this["Func_sin_Category"] = "Matematyczne";
        this["Func_sin_CategoryURL"] = "Matematyczne";
        this["Func_sin_Desc"] = "Zwraca sinus podanego kąta.";
        this["Func_sinh_Arg_0"] = "liczba";
        this["Func_sinh_ArgDesc_0"] = "Liczba rzeczywista, dla której ma zostać uzyskany sinus hiperboliczny.";
        this["Func_sinh_Category"] = "Matematyczne";
        this["Func_sinh_CategoryURL"] = "Matematyczne";
        this["Func_sinh_Desc"] = "Zwraca sinus hiperboliczny liczby.";
        this["Func_sln_Arg_0"] = "koszt";
        this["Func_sln_Arg_1"] = "wart. końcowa";
        this["Func_sln_Arg_2"] = "czas_życia";
        this["Func_sln_ArgDesc_0"] = "Początkowy koszt środka trwałego.";
        this["Func_sln_ArgDesc_1"] = "Wartość na koniec okresu amortyzacji.";
        this["Func_sln_ArgDesc_2"] = "Liczba okresów, przez które jest amortyzowany środek trwały.";
        this["Func_sln_Category"] = "Finansowe";
        this["Func_sln_CategoryURL"] = "Finansowe";
        this["Func_sln_Desc"] = "Zwraca amortyzację liniową środka trwałego za jeden okres.";
        this["Func_sqrt_Arg_0"] = "liczba";
        this["Func_sqrt_ArgDesc_0"] = "Liczba, dla której ma zostać uzyskany pierwiastek kwadratowy.";
        this["Func_sqrt_Category"] = "Matematyczne";
        this["Func_sqrt_CategoryURL"] = "Matematyczne";
        this["Func_sqrt_Desc"] = "Zwraca dodatni pierwiastek kwadratowy liczby.";
        this["Func_sqrtpi_Arg_0"] = "Liczba";
        this["Func_sqrtpi_ArgDesc_0"] = "Liczba do pomnożenia przez liczbę Pi.";
        this["Func_sqrtpi_Category"] = "Matematyczne";
        this["Func_sqrtpi_CategoryURL"] = "Matematyczne";
        this["Func_sqrtpi_Desc"] = "Mnoży podaną liczbę przez liczbę Pi i zwraca pierwiastek kwadratowy z wyniku.";
        this["Func_stdev_Arg_0"] = "liczba{0}";
        this["Func_stdev_ArgDesc_0"] = "Liczba lub zakres liczb określających próbkę populacji.";
        this["Func_stdev_Category"] = "Statystyczne";
        this["Func_stdev_CategoryURL"] = "Statystyczne";
        this["Func_stdev_Desc"] = "Zwraca odchylenie standardowe próbki populacji.";
        this["Func_subtotal_Arg_0"] = "Funkcja_nr";
        this["Func_subtotal_Arg_1"] = "Odwołanie{0}";
        this["Func_subtotal_ArgDesc_0"] = "Liczba określająca funkcję, która ma zostać użyta. Dodaj 100 do wartości, aby zignorować ukryte wartości.\r\n1 (lub 101) - ŚREDNIA\r\n2 (lub 102) - ILE.LICZB\r\n4 (lub 104) - MAX\r\n5 (lub 105) - MIN\r\n6 (lub 106) - ILOCZYN\r\n7 (lub 107) - ODCH.STANDARD\r\n9 (lub 109) - SUMA\r\n10 (or 110) - WARIANCJA.PRÓBKI\r\n    ";
        this["Func_subtotal_ArgDesc_1"] = "Co najmniej jedno odwołanie, dla którego ma zostać znaleziona suma częściowa.";
        this["Func_subtotal_Category"] = "Matematyczne";
        this["Func_subtotal_CategoryURL"] = "Matematyczne";
        this["Func_subtotal_Desc"] = "Oblicza sumę częściową w jednym lub kilku odwołaniach.";
        this["Func_sum_Arg_0"] = "liczba{0}";
        this["Func_sum_ArgDesc_0"] = "Liczba, którą należy dodać do innych liczb.";
        this["Func_sum_Category"] = "Matematyczne";
        this["Func_sum_CategoryURL"] = "Matematyczne";
        this["Func_sum_Desc"] = "Sumuje wszystkie liczby podane jako argumenty i zwraca sumę.";
        this["Func_syd_Arg_0"] = "koszt";
        this["Func_syd_Arg_1"] = "wart. końcowa";
        this["Func_syd_Arg_2"] = "czas_życia";
        this["Func_syd_Arg_3"] = "okres";
        this["Func_syd_ArgDesc_0"] = "Początkowy koszt środka trwałego.";
        this["Func_syd_ArgDesc_1"] = "Wartość na koniec okresu amortyzacji.";
        this["Func_syd_ArgDesc_2"] = "Liczba okresów, przez które jest amortyzowany środek trwały.";
        this["Func_syd_ArgDesc_3"] = "Okres - musi być podany w tych samych jednostek, co czas_życia.";
        this["Func_syd_Category"] = "Finansowe";
        this["Func_syd_CategoryURL"] = "Finansowe";
        this["Func_syd_Desc"] = "Zwraca amortyzację środka trwałego za podany okres metodą sumy cyfr wszystkich lat amortyzacji.";
        this["Func_tan_Arg_0"] = "liczba";
        this["Func_tan_ArgDesc_0"] = "Kąt w radianach, dla którego ma zostać uzyskany tangens.";
        this["Func_tan_Category"] = "Matematyczne";
        this["Func_tan_CategoryURL"] = "Matematyczne";
        this["Func_tan_Desc"] = "Zwraca tangens podanego kąta.";
        this["Func_tanh_Arg_0"] = "liczba";
        this["Func_tanh_ArgDesc_0"] = "Kąt w radianach, dla którego ma zostać uzyskany tangens hiperboliczny.";
        this["Func_tanh_Category"] = "Matematyczne";
        this["Func_tanh_CategoryURL"] = "Matematyczne";
        this["Func_tanh_Desc"] = "Zwraca tangens hiperboliczny podanego kąta.";
        this["Func_text_Arg_0"] = "Wartość";
        this["Func_text_Arg_1"] = "Formatuj_tekst";
        this["Func_text_ArgDesc_0"] = "Wartość liczbowa, która ma zostać sformatowana.";
        this["Func_text_ArgDesc_1"] = "Format liczb, za pomocą którego ma zostać sformatowana wartość.";
        this["Func_text_Category"] = "TekstDane";
        this["Func_text_CategoryURL"] = "Text_and_Data";
        this["Func_text_Desc"] = "Konwertuje wartość na tekst na podstawie podanego formatu.";
        this["Func_time_Arg_0"] = "godzina";
        this["Func_time_Arg_1"] = "minuta";
        this["Func_time_Arg_2"] = "sekunda";
        this["Func_time_ArgDesc_0"] = "Liczba reprezentująca godzinę czasu.";
        this["Func_time_ArgDesc_1"] = "Liczba reprezentująca minutę czasu.";
        this["Func_time_ArgDesc_2"] = "Liczba reprezentująca sekundę czasu.";
        this["Func_time_Category"] = "DataIGodzina";
        this["Func_time_CategoryURL"] = "Date_and_Time";
        this["Func_time_Desc"] = "Zwraca liczbę reprezentującą określoną wartość czasu.";
        this["Func_timevalue_Arg_0"] = "time_text";
        this["Func_timevalue_ArgDesc_0"] = "Wyrażenie tekstowe reprezentujące czas.";
        this["Func_timevalue_Category"] = "DataIGodzina";
        this["Func_timevalue_CategoryURL"] = "Date_and_Time";
        this["Func_timevalue_Desc"] = "Konwertuje wyrażenie tekstowe reprezentujące czas na równoważną wartość czasu.";
        this["Func_today_Category"] = "DataIGodzina";
        this["Func_today_CategoryURL"] = "Date_and_Time";
        this["Func_today_Desc"] = "Zwraca bieżącą datę.";
        this["Func_trim_Arg_0"] = "tekst";
        this["Func_trim_ArgDesc_0"] = "Tekst, z którego mają zostać usunięte dodatkowe spacje.";
        this["Func_trim_Category"] = "TekstDane";
        this["Func_trim_CategoryURL"] = "Text_and_Data";
        this["Func_trim_Desc"] = "Usuwa z tekstu wszystkie spacje oprócz pojedynczych spacji między wyrazami.";
        this["Func_true_Category"] = "Logiczne";
        this["Func_true_CategoryURL"] = "Logiczne";
        this["Func_true_Desc"] = "Zwraca wartość logiczną PRAWDA.";
        this["Func_trunc_Arg_0"] = "liczba";
        this["Func_trunc_Arg_1"] = "num_digits";
        this["Func_trunc_ArgDesc_0"] = "Liczba, która ma zostać obcięta.";
        this["Func_trunc_ArgDesc_1"] = "Liczba określająca dokładność obcięcia. Wartość domyślna wynosi zero (0).";
        this["Func_trunc_Category"] = "Matematyczne";
        this["Func_trunc_CategoryURL"] = "Matematyczne";
        this["Func_trunc_Desc"] = "Obcina liczbę do liczby całkowitej, usuwając jej część ułamkową.";
        this["Func_type_Arg_0"] = "wartość";
        this["Func_type_ArgDesc_0"] = "Wartość, dla której ma zostać pobrany typ.";
        this["Func_type_Category"] = "Informacja";
        this["Func_type_CategoryURL"] = "Informacja";
        this["Func_type_Desc"] = "Zwraca liczbę reprezentującą typ wartości: liczba = 1; tekst = 2; wartość logiczna (PRAWDA lub FAŁSZ) = 4; wartość błędu = 16; tablica = 64.";
        this["Func_upper_Arg_0"] = "tekst";
        this["Func_upper_ArgDesc_0"] = "Tekst do skonwertowania na wielkie litery.";
        this["Func_upper_Category"] = "TekstDane";
        this["Func_upper_CategoryURL"] = "Text_and_Data";
        this["Func_upper_Desc"] = "Konwertuje wszystkie małe litery w wyrażeniu tekstowym na wielkie litery.";
        this["Func_value_Arg_0"] = "tekst";
        this["Func_value_ArgDesc_0"] = "Wyrażenie tekstowe do skonwertowania.";
        this["Func_value_Category"] = "TekstDane";
        this["Func_value_CategoryURL"] = "Text_and_Data";
        this["Func_value_Desc"] = "Konwertuje wyrażenie tekstowe reprezentujące liczbę na liczbę.";
        this["Func_var_Arg_0"] = "liczba{0}";
        this["Func_var_ArgDesc_0"] = "Liczba lub zakres liczb określających próbkę populacji.";
        this["Func_var_Category"] = "Statystyczne";
        this["Func_var_CategoryURL"] = "Statystyczne";
        this["Func_var_Desc"] = "Zwraca wariancję próbki populacji.";
        this["Func_weekday_Arg_0"] = "Data";
        this["Func_weekday_Arg_1"] = "ReturnType";
        this["Func_weekday_ArgDesc_0"] = "Data, dla której ma zostać znaleziony numer dnia tygodnia.";
        this["Func_weekday_ArgDesc_1"] = "Określa, który schemat numeracji jest używany dla dni tygodnia.\r\n1 (domyślnie) = Od niedzieli (1) do soboty (7)\r\n2 = Od poniedziałku (1) do niedzieli (7)\r\n3 = Od poniedziałku (0) do niedzieli (6)\r\n    ";
        this["Func_weekday_Category"] = "DataIGodzina";
        this["Func_weekday_CategoryURL"] = "DataIGodzina";
        this["Func_weekday_Desc"] = "Zwraca liczbę reprezentującą dzień tygodnia na podstawie podanej daty.";
        this["Func_weeknum_Arg_0"] = "data";
        this["Func_weeknum_Arg_1"] = "typZwracany";
        this["Func_weeknum_ArgDesc_0"] = "Data, w której ma zostać znaleziony numer tygodnia.";
        this["Func_weeknum_ArgDesc_1"] = "Określa, od którego dnia zaczyna się tydzień.\r\n1 (domyślnie) = Niedziela to pierwszy dzień tygodnia.\r\n2 = Poniedziałek to pierwszy dzień tygodnia.\r\n    ";
        this["Func_weeknum_Category"] = "DataIGodzina";
        this["Func_weeknum_CategoryURL"] = "DataIGodzina";
        this["Func_weeknum_Desc"] = "Zwraca numer tygodnia, w którym przypada określona data.";
        this["Func_workday_Arg_0"] = "dataPoczątkowa";
        this["Func_workday_Arg_1"] = "dni";
        this["Func_workday_Arg_2"] = "święta";
        this["Func_workday_ArgDesc_0"] = "Data początkowa.";
        this["Func_workday_ArgDesc_1"] = "Liczba dni roboczych do odliczenia od daty początkowej.";
        this["Func_workday_ArgDesc_2"] = "Zakres obejmujący co najmniej jedno odwołanie zawierające daty, które mają być uważane za dni wolne od pracy.";
        this["Func_workday_Category"] = "DataIGodzina";
        this["Func_workday_CategoryURL"] = "DataIGodzina";
        this["Func_workday_Desc"] = "Zwraca dzień roboczy przypadający na określoną liczbę dni od określonej daty, z wyłączeniem świąt i dni wolnych.";
        this["Func_year_Arg_0"] = "Date_value";
        this["Func_year_ArgDesc_0"] = "Wartość daty, z której ma zostać pobrany rok.";
        this["Func_year_Category"] = "DataIGodzina";
        this["Func_year_CategoryURL"] = "Date_and_Time";
        this["Func_year_Desc"] = "Zwraca rok z wartości daty.";
        this["Func_yearfrac_Arg_0"] = "StartDate";
        this["Func_yearfrac_Arg_1"] = "EndDate";
        this["Func_yearfrac_Arg_2"] = "Podstawa";
        this["Func_yearfrac_ArgDesc_0"] = "Data początkowa to data rozpoczęcia zakresu.";
        this["Func_yearfrac_ArgDesc_1"] = "Data zakończenia to data końcowa zakresu.";
        this["Func_yearfrac_ArgDesc_2"] = "\r\n      Podstawa określa podstawę używanej liczby dni.\r\n      0 (domyślnie) = Amerykańska (NASD) 30/360\r\n      1 = Rzeczywiste/rzeczywiste\r\n      2 = Rzeczywiste/360\r\n      3 = Rzeczywiste/365\r\n      4 = Europejska 30/360\r\n    ";
        this["Func_yearfrac_Category"] = "DataIGodzina";
        this["Func_yearfrac_CategoryURL"] = "DataIGodzina";
        this["Func_yearfrac_Desc"] = "Zwraca ułamek reprezentujący część roku określoną między określoną datą rozpoczęcia a datą zakończenia.";
        this["GenerateTableColumnName"] = "Kolumna";
        this["GenerateTableName"] = "Tabela";
        this["LD_Chart_AxisDisplayUnits"] = "Setki:Setki, Tysiące:Tysiące, DziesięćTysięcy:x 10000, StoTysięcy:x 100000, Miliony:Miliony, DziesięćMilionów:x 10000000, StoMilionów:x 100000000, TysiącMilionów:Miliardy, MilionMilionów:Biliony";
        this["LD_Fallback_ChartEx_Line1"] = "Ten wykres nie jest dostępny w używanej wersji programu Excel.";
        this["LD_Fallback_ChartEx_Line2"] = "Edycja tego kształtu lub zapisanie tego skoroszytu w innym formacie pliku spowoduje trwałe uszkodzenie wykresu.";
        this["LE_AllMustHavGuidsIfAnyHasThem"] = "Nie można ustawić parametru GenerateGuidsForGroups na 'fałsz', jeśli jakakolwiek istniejąca grupa ma identyfikator Guid.";
        this["LE_ArgumentException_AnchorCellFromOtherWorksheet"] = "Określona komórka nie istnieje w tym samym arkuszu, co kształt.";
        this["LE_ArgumentException_ArrayFormulaMustHaveSingleRegion"] = "Formułę tablicową można zastosować tylko do jednego regionu komórki.";
        this["LE_ArgumentException_ArrayTooSmall"] = "Tablica jest za mała, aby skopiować do niej wszystkie wartości.";
        this["LE_ArgumentException_CannotAddStandardTableStyle"] = "Nie można dodać standardowego stylu tabeli do kolekcji CustomTableStyleCollection.";
        this["LE_ArgumentException_CannotApplyDVRuleToTotalCell"] = "Co najmniej jedna komórka w podanym odwołaniu obejmuje wszystkie komórki w tabeli. Nie można zastosować reguły sprawdzania poprawności danych.";
        this["LE_ArgumentException_CannotCreateEmptyColorInfo"] = "Pustego koloru nie można użyć do utworzenia wystąpienia WorkbookColorInfo.";
        this["LE_ArgumentException_CellMustBeOnSameWorksheetAsReferencesCollection"] = "Komórka musi pochodzić z tego samego arkusza, co kolekcja odwołań.";
        this["LE_ArgumentException_CellShiftedOffWorksheet"] = "Komórka została usunięta lub przesunięta z arkusza i nie reprezentuje już położenia arkusza.";
        this["LE_ArgumentException_CellsInTableFromOtherWorksheet"] = "Określony region nie pochodzi z tego samego arkusza, co tabela danych.";
        this["LE_ArgumentException_CellValueStringLength"] = "Wartości ciągów przypisane do komórki nie mogą mieć długości większej niż {0}.";
        this["LE_ArgumentException_ColumnFormulaAlreadyAppliedToCell"] = "Formuła została już zastosowana do komórki. Sklonuj ją, aby użyć jej jako formuły kolumny.";
        this["LE_ArgumentException_ColumnFromDifferentWorksheet"] = "Kolumna nie należy do tego samego arkusza, co ta kolekcja.";
        this["LE_ArgumentException_ColumnInputCellFromOtherWorksheet"] = "Określona kolumnowa komórka wejściowa nie istnieje w tym samym arkuszu, co tabela danych.";
        this["LE_ArgumentException_ColumnNotInTable"] = "Kolumna pochodzi z innej tabeli.";
        this["LE_ArgumentException_ColumnRemovedFromWorksheet"] = "Kolumna została usunięta z arkusza.";
        this["LE_ArgumentException_ColumnsToRepeatAtLeftOutsideRange"] = "Zakres wykracza poza zakres dostępnych kolumn w arkuszu.";
        this["LE_ArgumentException_CommentTooLong"] = "Długość komentarza musi być mniejsza niż 255 znaków.";
        this["LE_ArgumentException_CustomViewNameAlreadyExists"] = "Widok niestandardowy o nazwie \"{0}\" już istnieje.";
        this["LE_ArgumentException_DefaultTableStyleNotInWorkbook"] = "Podany styl tabeli nie znajduje się w kolekcjach CustomTableStyles ani StandardTableStyles.";
        this["LE_ArgumentException_DisplayTextTooLong"] = "Tekst wyświetlany hiperłącza musi mieć mniej niż 256 znaków.";
        this["LE_ArgumentException_DuplicateDisplayValue"] = "Podana wartość wyświetlana jest już w kolekcji.";
        this["LE_ArgumentException_DuplicateFixedDateGroup"] = "Podany element FixedDateGroup jest już w kolekcji.";
        this["LE_ArgumentException_DuplicateTableStyle"] = "Podany styl znajduje się już w kolekcji CustomTableStyleCollection.";
        this["LE_ArgumentException_DuplicateTableStyleName_Existing"] = "Podana nazwa pasuje do innego stylu w kolekcji.";
        this["LE_ArgumentException_DuplicateTableStyleName_New"] = "Podany styl ma nazwę, która pasuje do innego stylu w kolekcji.";
        this["LE_ArgumentException_DuplicateWorksheetProtectedRangeTitle"] = "Zakres o nazwie '{0}' już istnieje.";
        this["LE_ArgumentException_DV_ArrayFormulaCannotBeUsed"] = "Nie można zastosować ArrayFormula jako {0}.";
        this["LE_ArgumentException_DV_CellFromOtherWorksheet"] = "Komórka nie należy do tego samego arkusza, co kolekcja reguł sprawdzania poprawności danych.";
        this["LE_ArgumentException_DV_ConstraintFormulaMustBeNumberOrReference"] = "{0} musi być liczbą, wyrażeniem liczbowym lub odwołaniem do pojedynczej komórki.";
        this["LE_ArgumentException_DV_FormulaCannotBeNull"] = "Elementu {0} nie można ustawić na null, gdy do arkusza jest stosowana reguła sprawdzania poprawności danych.";
        this["LE_ArgumentException_DV_FormulaCannotFindNamedReference"] = "{0} zawiera nazwane odwołanie, które nie istnieje w skoroszycie.";
        this["LE_ArgumentException_DV_FormulaCannotFindWorksheetReference"] = "{0} zawiera odwołanie do arkusza, który nie istnieje w skoroszycie.";
        this["LE_ArgumentException_DV_FormulaCannotReferenceOtherWorkbook"] = "{0} nie może zawierać odwołania do zewnętrznego skoroszytu.";
        this["LE_ArgumentException_DV_FormulaInvalidForWorkbookFormat"] = "{0} odwołuje się do komórek, które nie są prawidłowe w formacie skoroszytu {1}.";
        this["LE_ArgumentException_DV_InvalidAddress"] = "Podany adres nie jest prawidłowym adresem komórki lub regionu.";
        this["LE_ArgumentException_DV_InvalidExcelDate"] = "Podanej daty nie można przedstawić w programie Excel.";
        this["LE_ArgumentException_DV_InvalidFormula"] = "Nieprawidłowa formuła użyta w regule sprawdzania poprawności danych.";
        this["LE_ArgumentException_DV_InvalidMessageLength"] = "{0} nie może zawierać więcej niż {1} znaków(-i).";
        this["LE_ArgumentException_DV_ListFormulaCannotHaveEmptyString"] = "Stała znakowa w parametrze ValuesFormula nie może być pustym ciągiem.";
        this["LE_ArgumentException_DV_ListFormulaMustBeStringOrReference"] = "Parametr ValuesFormula musi być ustawiony na stałą znakową lub odwołanie do pojedynczej komórki albo regionu w tym samym skoroszycie.";
        this["LE_ArgumentException_DV_ListFormulaReferenceMustBeOneDimensional"] = "Parametr ValuesFormula zawiera odwołanie do regionu, które ma wiele wierszy i kolumn. Region musi zawierać jeden wiersz lub jedną kolumnę.";
        this["LE_ArgumentException_DV_LowerGreaterThanUpperContraint"] = "Wartość dolnego ograniczenia musi być mniejsza lub równa górnemu ograniczeniu.";
        this["LE_ArgumentException_DV_MustHaveOneAcceptedValue"] = "Co najmniej jedna wartość musi być określona jako akceptowana wartość.";
        this["LE_ArgumentException_DV_ReferencesAlreadyContainDVs"] = "Podane odwołania zawierają już co najmniej jedną regułę sprawdzania poprawności danych.";
        this["LE_ArgumentException_DV_ReferencesFromOtherWorksheet"] = "Kolekcja odwołań nie należy do tego samego arkusza, co kolekcja reguł sprawdzania poprawności danych.";
        this["LE_ArgumentException_DV_RegionFromOtherWorksheet"] = "Region nie należy do tego samego arkusza, co kolekcja reguł sprawdzania poprawności danych.";
        this["LE_ArgumentException_DV_RuleAppliedOnOtherWorksheet"] = "Reguła sprawdzania poprawności danych została już zastosowana w innym arkuszu.";
        this["LE_ArgumentException_DV_UpperLessThanLowerContraint"] = "Wartość górnego ograniczenia musi być większa lub równa dolnemu ograniczeniu.";
        this["LE_ArgumentException_EditRangeAlreadyOwned"] = "Zakres znajduje się już w zakresie AllowedEditRanges arkusza.";
        this["LE_ArgumentException_EditRangeHasRangeFromOtherWorksheet"] = "Zakres edycji obejmuje zakres z innego arkusza niż zakres AllowedEditRanges.";
        this["LE_ArgumentException_EmptyCustomList"] = "Lista musi zawierać co najmniej jeden ciąg.";
        this["LE_ArgumentException_EndIndexLessThanZero"] = "Koniec zakresu musi być większy lub równy zero.";
        this["LE_ArgumentException_FileDoesntContainsWorkbookStream"] = "Określony strumień lub plik nie zawiera prawidłowego skoroszytu.";
        this["LE_ArgumentException_FormulaReferencesInvalidCells"] = "Określona formuła odwołuje się do komórek, które nie są obsługiwane w bieżącym formacie pliku skoroszytu.";
        this["LE_ArgumentException_HashSetFull"] = "Zestaw skrótów zawiera za dużo elementów.";
        this["LE_ArgumentException_HyperlinkAlreadyOwned"] = "Hiperłącze należy już do kolekcji hiperłączy.";
        this["LE_ArgumentException_HyperlinkSourceFromOtherWorksheet"] = "Hiperłącze ma komórkę źródłową lub region z innego arkusza.";
        this["LE_ArgumentException_HyperlinkTargetFromOtherWorkbook"] = "Obiekt docelowy pochodzi z innego skoroszytu niż właściciel hiperłącza.";
        this["LE_ArgumentException_InfiniteColumnWidth"] = "Szerokość kolumny nie może być nieskończona.";
        this["LE_ArgumentException_Interval"] = "Parametr Interval powinien być jedną z wartości w wyliczeniu DateInterval";
        this["LE_ArgumentException_IntervalStr"] = "Nie można skonwertować parametru Interval na jedną z wartości w wyliczeniu DateInterval";
        this["LE_ArgumentException_InvalidCellAddress"] = "Podany adres nie jest prawidłowym adresem komórki.";
        this["LE_ArgumentException_InvalidCommentPositioningMode"] = "Do kształtu komentarza można zastosować tylko metody DontMoveOrSizeWithCells i MoveAndSizeWithCells ShapePositioningModes.";
        this["LE_ArgumentException_InvalidCustomFilterOperandNumber"] = "Wartość nie może być nieskończonością ani nie-liczbą (NaN).";
        this["LE_ArgumentException_InvalidCustomFilterOperator"] = "Gdy wartość porównania nie jest ciągiem, operatorem porównania nie może być BeginsWith, DoesNotBeginWith, Contains, DoesNotContain, EndsWith ani DoesNotEndWith.";
        this["LE_ArgumentException_InvalidDatePeriodFilterValue"] = "Wartość nie mieści się w zakresie. W przypadku filtra typu Miesiąc zakres to 1–12. W przypadku filtra typu Kwartał zakres to 1–4.";
        this["LE_ArgumentException_InvalidDpi"] = "Podana wartość Dpi jest nieprawidłowa.";
        this["LE_ArgumentException_InvalidExcelDate"] = "Wartości nie można odzwierciedlić jako daty w programie Excel.";
        this["LE_ArgumentException_InvalidFileFormat"] = "Nieprawidłowy lub nierozpoznany format pliku.";
        this["LE_ArgumentException_InvalidFormula"] = "Podana formuła była nieprawidłowa.";
        this["LE_ArgumentException_InvalidGradientStopColor"] = "W ogranicznikach gradientu nie można używać kolorów automatycznych ani systemowych.";
        this["LE_ArgumentException_InvalidNamedReferenceName"] = "Nazwa nie jest prawidłową nazwą nazwanego odwołania.";
        this["LE_ArgumentException_InvalidPaletteColor_EmptyOrSystem"] = "Do palety nie można dodawać kolorów pustych ani systemowych.";
        this["LE_ArgumentException_InvalidPaletteColor_NonOpaque"] = "Do palety można dodawać wyłączni kolory nieprzezroczyste.";
        this["LE_ArgumentException_InvalidReferencesString"] = "Wartość odwołania jest nieprawidłowo sformatowana.";
        this["LE_ArgumentException_InvalidRegionAddress"] = "Podany adres nie jest prawidłowym adresem regionu.";
        this["LE_ArgumentException_InvalidRowOrColumnRange"] = "Nieprawidłowy wiersz lub zakres kolumn.";
        this["LE_ArgumentException_InvalidSeriesValues"] = "Nieprawidłowe wartości serii. Musi istnieć co najmniej jedna wartość, a każda wartość musi być ciągiem lub obiektem, który można przekonwertować w wartość typu double.";
        this["LE_ArgumentException_InvalidTarget"] = "Obiektem docelowym hiperłącza musi być ciąg, element WorksheetCell, element WorksheetRegion lub element NamedReference.";
        this["LE_ArgumentException_InvalidTopOrBottomFilterValue"] = "Wartość musi należeć do przedziału od 1 do 500.";
        this["LE_ArgumentException_InvalidWorksheetName"] = "Nazwa arkusza nie może zawierać żadnego z następujących znaków: : \\ / ? * [ ]";
        this["LE_ArgumentException_InvalidWorksheetNameStartingQuote"] = "Nazwa arkusza nie może zaczynać się ani kończyć pojedynczym cudzysłowem.";
        this["LE_ArgumentException_LessThanTwoGradientStops"] = "Należy określić co najmniej dwa ograniczniki gradientu.";
        this["LE_ArgumentException_NamedReferenceNameTooLong"] = "Nazwa nazwanego odwołania nie może mieć więcej niż 255 znaków.";
        this["LE_ArgumentException_NaNDefaultColumnWidth"] = "Nie-liczby (NaN) nie można użyć jako domyślnej szerokości kolumny.";
        this["LE_ArgumentException_NewRegionOverlapsFilterArea"] = "Operacja próbuje zmienić filtrowany zakres w arkuszu i nie może zostać ukończona. Aby zakończyć tę operację, należy usunąć z arkusza elementy AutoFilters.";
        this["LE_ArgumentException_NewTableRegionCannotMoveHeaders"] = "Nowy region tabeli nie może przenieść nagłówków do nowego wiersza.";
        this["LE_ArgumentException_NewTableRegionFromWrongWorksheet"] = "Podany region nie pochodzi z tego samego arkusza, co tabela.";
        this["LE_ArgumentException_NewTableRegionMustHaveDataRows"] = "Nowy region tabeli musi zawierać co najmniej jeden wiersz danych.";
        this["LE_ArgumentException_NewTableRegionOverlapOld"] = "Nowy region tabeli musi nakładać się na poprzedni region tabeli.";
        this["LE_ArgumentException_NewTableRegionOverlapsBlockingValue"] = "Nie można zmienić rozmiaru tabeli na określony region, ponieważ zawiera wielokomórkową formułę tablicową lub tabelę danych.";
        this["LE_ArgumentException_NewTableRegionOverlapsMergedRegion"] = "Nie można zmienić rozmiaru tabeli na określony region, ponieważ zawiera komórkę scaloną.";
        this["LE_ArgumentException_NewTableRegionOverlapsOtherTable"] = "Nie można zmienić rozmiaru tabeli na określony region, ponieważ zawiera inną tabelę.";
        this["LE_ArgumentException_NoRegionsInArray"] = "Określona tablica nie zawiera regionów.";
        this["LE_ArgumentException_NotEnum"] = "Typ '{0}' nie jest wyliczeniem";
        this["LE_ArgumentException_NotSupportedCodePage"] = "Nieobsługiwana strona kodowa";
        this["LE_ArgumentException_NPer"] = "Wartość Liczba_rat nie może wynosić 0";
        this["LE_ArgumentException_NPerFunction"] = "Nieprawidłowa kombinacja parametrów funkcji Liczba_rat";
        this["LE_ArgumentException_PA_PrintAreasCannotOverlap"] = "Określony obszar wydruku nakładać się na jeden z istniejących obszarów wydruku.";
        this["LE_ArgumentException_ParentStyleFromOtherWorkbook"] = "Podany styl nie pochodzi z tego samego skoroszytu, co format komórki.";
        this["LE_ArgumentException_PB_CantInsertBreakAtIndex"] = "Ta kolekcja nie obsługuje wstawiania elementów w indeksie.";
        this["LE_ArgumentException_PB_CantSetBreakAtIndex"] = "Ta kolekcja nie obsługuje ustawiania elementów w indeksie.";
        this["LE_ArgumentException_PB_MustBeWithinPrintArea"] = "Podział strony musi znajdować się w obrębie obszaru wydruku.";
        this["LE_ArgumentException_PB_PageBreakCantBeA1Cell"] = "Nie można wstawić podziału strony w lewym górnym rogu arkusza.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeLeftColumn"] = "Nie można wstawić podziału strony po lewej stronie arkusza.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeTopRow"] = "Nie można wstawić podziału strony na górze arkusza.";
        this["LE_ArgumentException_PB_PageBreaksCannotAddExistingInstance"] = "Określone wystąpienie już istnieje w kolekcji.";
        this["LE_ArgumentException_PB_PageBreaksCannotOverlap"] = "Podział strony nie może nakładać się na inne podziały stron w arkuszu.";
        this["LE_ArgumentException_PB_PrintAreaMustBeInPrintAreas"] = "Podział strony znajduje się w obszarze wydruku, którego nie ma w kolekcji PrintAreas.";
        this["LE_ArgumentException_PB_PrintAreaMustBeOnSameWorksheet"] = "Podział strony znajduje się w obszarze wydruku, który nie znajduje się w tym samym arkuszu, co kolekcja podziałów strony.";
        this["LE_ArgumentException_Pmt"] = "Wartość Spłata nie może wynosić 0";
        this["LE_ArgumentException_PrintAreaShiftedOffWorksheet"] = "Podział strony znajduje się w obszarze wydruku, który został przesunięty z arkusza i nie reprezentuje już prawidłowego zakresu komórek.";
        this["LE_ArgumentException_RangeFromOtherWorksheetThanProtectedRange"] = "Podany zakres pochodzi z innego arkusza niż zakres edycji.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheet"] = "Region musi pochodzić z tego samego arkusza.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheetAsReferencesCollection"] = "Region musi pochodzić z tego samego arkusza, co kolekcja odwołań.";
        this["LE_ArgumentException_RegionsFromMixedWorksheets"] = "Wszystkie regiony określone w tablicy muszą pochodzić z tego samego arkusza.";
        this["LE_ArgumentException_RegionShiftedOffWorksheet"] = "Region został przesunięty z arkusza i nie reprezentuje już prawidłowego zakresu komórek.";
        this["LE_ArgumentException_RegionsShiftedOffWorksheet"] = "Co najmniej jeden podany region został usunięty z arkusza i nie reprezentuje już prawidłowego zakresu komórek.";
        this["LE_ArgumentException_RowFromDifferentWorksheet"] = "Wiersz nie należy do tego samego arkusza, co ta kolekcja.";
        this["LE_ArgumentException_RowInputCellFromOtherWorksheet"] = "Określona wierszowa komórka wejściowa nie istnieje w tym samym arkuszu co tabela danych.";
        this["LE_ArgumentException_RowRemovedFromWorksheet"] = "Wiersz został usunięty z arkusza.";
        this["LE_ArgumentException_SelectedWorksheetFromOtherWorkbook"] = "Arkusz nie należy do skoroszytu skojarzonego z tymi opcjami okna.";
        this["LE_ArgumentException_ShapeCannotBeAdded"] = "Określonego kształtu nie można dodać do kolekcji kształtów.";
        this["LE_ArgumentException_SourceFromOtherWorksheetThanHyperlink"] = "Podana komórka źródłowa lub region pochodzi z innego arkusza niż hiperłącze.";
        this["LE_ArgumentException_StartIndexLessThanZero"] = "Początek zakresu musi być większy lub równy zero.";
        this["LE_ArgumentException_StyleNameAlreadyExists"] = "Styl o nazwie \"{0}\" już istnieje w skoroszycie.";
        this["LE_ArgumentException_StyleNameTooLong"] = "Nazwa stylu może zawierać maksymalnie 255 znaków.";
        this["LE_ArgumentException_TableColumnRemovedFromTable"] = "Kolumna tabeli została usunięta z tabeli.";
        this["LE_ArgumentException_TableStyleFromOtherWorkbook"] = "Podany element WorksheetTableStyle nie należy do tego samego skoroszytu, co tabela.";
        this["LE_ArgumentException_TargetTooLong"] = "Adres docelowy hiperłącza musi mieć mniej niż 256 znaków.";
        this["LE_ArgumentException_TextLengthGreaterThanMax"] = "Długość tekstu w kształcie lub komentarzu nie może być większa niż {0}.";
        this["LE_ArgumentException_ToolTipTooLong"] = "Wskazówka dotycząca hiperłącza musi mieć mniej niż 256 znaków.";
        this["LE_ArgumentException_TopLeftWindowBounds"] = "Lewa i górna granica okna muszą należeć do przedziału od -32768 do 32767.";
        this["LE_ArgumentException_WeekdayFirst"] = "Parametr WeekdayFirst powinien być z przedziału od FirstDayOfWeek.System a FirstDayOfWeek.Saturday";
        this["LE_ArgumentException_WidthHeightWindowBounds"] = "Szerokość i wysokość granic okna muszą należeć do przedziału 0 do 65535.";
        this["LE_ArgumentException_WorksheetNameAlreadyExists"] = "Arkusz o nazwie \"{0}\" już istnieje w skoroszycie.";
        this["LE_ArgumentException_WorksheetNameTooLong"] = "Nazwa arkusza musi mieć od 1 do 31 znaków.";
        this["LE_ArgumentException_WorksheetProtectedRangeTitleTooLong"] = "Tytuł zakresu edycji nie może mieć więcej niż 255 znaków.";
        this["LE_ArgumentException_WorksheetScopeFromOtherWorkbook"] = "Arkusz nie należy do tego samego skoroszytu, co ta kolekcja.";
        this["LE_ArgumentNullException_AnchorCell"] = "Komórka nie może mieć wartości null, jeśli kształt istnieje w arkuszu lub grupie.";
        this["LE_ArgumentNullException_CustomViewName"] = "Nazwa widoku niestandardowego nie może mieć wartości null ani być pusta.";
        this["LE_ArgumentNullException_Encoder"] = "Koder obrazu nie może mieć wartości null.";
        this["LE_ArgumentNullException_FindNamedReference"] = "Nazwa do znalezienia nie może mieć wartości null ani być pusta.";
        this["LE_ArgumentNullException_FormulaCantBeNull"] = "Formuła nazwanego odwołania nie może mieć wartości null ani być pusta.";
        this["LE_ArgumentNullException_HiddenColumn"] = "Nie można dodać pustej kolumny do kolekcji.";
        this["LE_ArgumentNullException_HiddenRow"] = "Nie można dodać pustego wiersza do kolekcji.";
        this["LE_ArgumentNullException_Image"] = "Obraz nie może być pusty.";
        this["LE_ArgumentNullException_NamedReferenceNameCantBeNull"] = "Nazwa nazwanego odwołania nie może mieć wartości null ani być pusta.";
        this["LE_ArgumentNullException_SaveStream"] = "Nie można zapisać skoroszytu w strumieniu o wartości null.";
        this["LE_ArgumentNullException_SelectedWorksheet"] = "W skoroszycie musi być prawidłowo wybrany arkusz.";
        this["LE_ArgumentNullException_Shape"] = "Nie można dodać pustego kształtu do kolekcji.";
        this["LE_ArgumentNullException_SourceFont"] = "Czcionka źródłowa nie może mieć wartości null.";
        this["LE_ArgumentNullException_SourceFormatting"] = "Formatowanie źródła nie może mieć wartości null.";
        this["LE_ArgumentNullException_StyleName"] = "Nazwa stylu nie może mieć wartości null ani być pusta.";
        this["LE_ArgumentNullException_UnformattedString"] = "Niesformatowany ciąg nie może mieć wartości null.";
        this["LE_ArgumentNullException_Workbook"] = "Skoroszyt nie może mieć wartości null.";
        this["LE_ArgumentNullException_WorkbookRequiredToResolveThemeColor"] = "Aby rozwiązać kolor z motywem, należy określić skoroszyt.";
        this["LE_ArgumentNullException_WorksheetName"] = "Nazwa arkusza nie może mieć wartości null ani być pusta.";
        this["LE_ArgumentNullException_WorksheetProtectedRangeTitleCantBeNull"] = "Tytuł zakresu edycji nie może mieć wartości null ani być pusty.";
        this["LE_ArgumentNullException_WorksheetScope"] = "Zakres arkusza nie może mieć wartości null.";
        this["LE_ArgumentOutOfRange_RelativeColumnIndex"] = "Względny indeks kolumny znajduje się poza określonym regionem.";
        this["LE_ArgumentOutOfRange_RelativeRowIndex"] = "Względny indeks wiersza znajduje się poza określonym regionem.";
        this["LE_ArgumentOutOfRangeException_AnchorPosition"] = "Jedna z wartości procentowych w określonej lokalizacji jest mniejsza niż 0.";
        this["LE_ArgumentOutOfRangeException_BadInsertIndex"] = "Określony indeks musi być większy lub równy zero i mniejszy lub równy liczbie istniejących akapitów.";
        this["LE_ArgumentOutOfRangeException_BadRemoveAtIndex"] = "Indeks musi być większy od zera i mniejszy niż liczba akapitów w sformatowanym ciągu.";
        this["LE_ArgumentOutOfRangeException_CollectionIndex"] = "Indeks jest poza zakresem. Musi być nieujemny i mniejszy niż rozmiar kolekcji.";
        this["LE_ArgumentOutOfRangeException_ColumnWidth"] = "Szerokość kolumny musi należeć do przedziału od 0 do 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultColumnWidth"] = "Domyślna szerokość kolumny musi należeć do przedziału od 0 do 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultFontHeight"] = "Domyślna wysokość czcionki musi należeć do przedziału od 20 do 8180.";
        this["LE_ArgumentOutOfRangeException_DefaultRowHeight"] = "Domyślna wysokość wiersza musi należeć do przedziału od –1 do 8192.";
        this["LE_ArgumentOutOfRangeException_DuplicateItemSorted"] = "Podany element ma już zastosowany warunek sortowania.";
        this["LE_ArgumentOutOfRangeException_FirstVisibleTabIndex"] = "Indeks pierwszej widocznej karty musi być większy lub równy zero.";
        this["LE_ArgumentOutOfRangeException_FontHeight"] = "Wysokość czcionki musi należeć do przedziału od 20 do 8180.";
        this["LE_ArgumentOutOfRangeException_GroupAddedToSelf"] = "Nie można dodać grupy do jej własnej kolekcji kształtów.";
        this["LE_ArgumentOutOfRangeException_Indent"] = "Poziom wcięcia musi należeć do przedziału od 0 do 250.";
        this["LE_ArgumentOutOfRangeException_IndexNegative"] = "Indeks musi być nieujemny.";
        this["LE_ArgumentOutOfRangeException_InvalidCollectionIndex"] = "Podany indeks jest nieprawidłowy w kolekcji.";
        this["LE_ArgumentOutOfRangeException_InvalidColorInfoTint"] = "Wartość odcienia musi należeć do przedziału od –1,0 do 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnCount"] = "{0} nie jest prawidłową liczbą kolumn w bieżącym formacie. Wartość musi należeć do przedziału od {1} do {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnIndex"] = "{0} nie jest prawidłowym indeksem kolumny w bieżącym formacie. Wartość musi należeć do przedziału od {1} do {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidGradientStopOffset"] = "Przesunięcie ogranicznika gradientu musi należeć do przedziału od 0,0 do 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRelativeRectangleValueForGradient"] = "Wartości wewnętrznego prostokąta gradientu muszą należeć do przedziału od 0,0 do 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRemoveAtIndex_SortConditions"] = "Indeks, w którym należy usunąć warunek sortowania, jest poza zakresem.";
        this["LE_ArgumentOutOfRangeException_InvalidRowCount"] = "{0} nie jest prawidłową liczbą wierszy w bieżącym formacie. Wartość musi należeć do przedziału od {1} do {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidRowIndex"] = "{0} nie jest prawidłowym indeksem wiersza w bieżącym formacie. Wartość musi należeć do przedziału od {1} do {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidTableStyleAreaStripeExtent"] = "Wartość {0} musi należeć do przedziału od 1 do 9 włącznie.";
        this["LE_ArgumentOutOfRangeException_LeftPaneWidth"] = "Określona szerokość okienka jest poza prawidłowym zakresem {0} i {1}.";
        this["LE_ArgumentOutOfRangeException_LengthMustBePositive"] = "Długość musi być równa co najmniej jeden (1).";
        this["LE_ArgumentOutOfRangeException_MagnificationLevel"] = "Poziom powiększenia widoku musi należeć do przedziału od 10 do 400.";
        this["LE_ArgumentOutOfRangeException_Margins"] = "Marginesy muszą należeć do przedziału 0 do 100.";
        this["LE_ArgumentOutOfRangeException_MaxPagesHorizontally"] = "Maksymalna liczba stron w poziomie musi należeć do przedziału od 0 do 32767.";
        this["LE_ArgumentOutOfRangeException_MaxPagesVertically"] = "Maksymalna liczba stron w pionie musi należeć do przedziału od 0 do 32767.";
        this["LE_ArgumentOutOfRangeException_MaxRecursionIterations"] = "Maksymalna liczba iteracji rekursji musi należeć do przedziału od 1 do 32767.";
        this["LE_ArgumentOutOfRangeException_MaxSortConditions"] = "Maksymalna liczba warunków sortowania w kolekcji wynosi {0}.";
        this["LE_ArgumentOutOfRangeException_NegativeStartIndex"] = "Indeks początkowy nie może być ujemny.";
        this["LE_ArgumentOutOfRangeException_NumberOfCopies"] = "Liczba kopii musi należeć do przedziału od 1 do 65535.";
        this["LE_ArgumentOutOfRangeException_Per"] = "Wartość per powinna być z przedziału od 0 do Liczba_rat + 1";
        this["LE_ArgumentOutOfRangeException_Rate"] = "Stawka powinna być większa niż –1";
        this["LE_ArgumentOutOfRangeException_Resolution"] = "Rozdzielczość musi należeć do przedziału od 0 do 65535.";
        this["LE_ArgumentOutOfRangeException_RowHeight"] = "Wysokość wiersza musi należeć do przedziału od 0 do 8192.";
        this["LE_ArgumentOutOfRangeException_ScalingFactor"] = "Współczynnik skalowania musi należeć do przedziału od 10 do 400.";
        this["LE_ArgumentOutOfRangeException_StartPageNumber"] = "Numer strony początkowej musi należeć do przedziału od -32765 do 32767.";
        this["LE_ArgumentOutOfRangeException_TabBarWidth"] = "Szerokość paska kart musi należeć do przedziału od 0 do 1000.";
        this["LE_ArgumentOutOfRangeException_TopPaneHeight"] = "Określona wysokość okienka jest poza prawidłowym zakresem {0} i {1}.";
        this["LE_AutoColorNotAllowed"] = "Kolor nie może mieć wartości 'Auto'.";
        this["LE_Axis_MajorMinorUnit"] = "Wartość parametru majorUnit musi być większa niż wartość parametru minorUnit.";
        this["LE_Axis_NoCrossAxis"] = "Jeśli właściwość CrossAxis jest ustawiona na jednym obiekcie Axis, należy ją ustawić na wszystkie z nich. Jeśli wykres ma tylko dwie osie, oś przecinającą można określić automatycznie, ale tylko wtedy, gdy właściwość CrossAxis pozostaje nieustawiona we wszystkich obiektach Axis.";
        this["LE_Biff8SerializerNotLoaded"] = "Format '{0}' wymaga programu szeregującego, który nie jest załadowany. W przypadku JavaScript jest to zawarte w pliku infragistics.excel_serialization_biff8.js.";
        this["LE_CategoryAxisBinning_BinWidthAndNumberOfBins"] = "Parametry 'binWidth' i 'numberOfBins' wzajemnie się wykluczają. Oba mogą być nieokreślone, ale jeśli określono jeden, drugiego nie można.";
        this["LE_ChartGradientFill_EmptyStops"] = "Musi zawierać co najmniej jeden ogranicznik gradientu.";
        this["LE_ChartObject_DifferentChart"] = "Przypisywane wystąpienie obiektu jest skojarzone z innym wykresem i nie można go przypisać do tego wystąpienia.";
        this["LE_ColumnsToRepeatAtLeftAreOutsideAvailableRange"] = "Opcje drukowania obejmują kolumny, które powtarzają się po lewej stronie i są poza zakresem dostępnych kolumn w nowym formacie.";
        this["LE_ComboChartGroupCollection_DoesNotExist"] = "Podany element ComboChartGroup nie istnieje w tej kolekcji.";
        this["LE_FillPercentInvalid"] = "Wartość {0} nie jest prawidłową wartością właściwości {1}.";
        this["LE_FormatLimitError_Indent"] = "Wcięcie przekracza maksymalną wartość {0} podaną w bieżącym formacie.";
        this["LE_FormatLimitError_MacroWorkbook"] = "Skoroszyt zawiera kod VBA i nie można go zapisać jako skoroszytu z obsługą makr.";
        this["LE_FormatLimitError_MaxColumnIndex"] = "Indeks kolumny {0} jest większy niż maksymalny limit {1} obsługiwany przez określony format.";
        this["LE_FormatLimitError_MaxRowIndex"] = "Indeks wiersza {0} przekracza maksymalny limit {1} obsługiwany przez ten format.";
        this["LE_FormulaParseException_ArrayContainsConstants"] = "Tablica może zawierać tylko wyrażenia stałe.";
        this["LE_FormulaParseException_ArrayHasEmptyFirstRow"] = "Stała tablica w formule nie zawiera żadnych elementów w pierwszym wierszu.";
        this["LE_FormulaParseException_ArrayHasMisalignedRows"] = "Stała tablica w formule zawiera nierówne wiersze. Każdy wiersz w tablicy musi mieć taką samą liczbę kolumn.";
        this["LE_FormulaParseException_CellReferenceAfterWorkbookName"] = "Po odwołaniu do zewnętrznego skoroszytu musi następować nazwane odwołanie, a nie odwołanie do komórki.";
        this["LE_FormulaParseException_ExtraExpressions"] = "Formuła zawiera dodatkowe wyrażenia na końcu.";
        this["LE_FormulaParseException_FunctionMissingClosingParen"] = "Funkcja w formule nie ma nawiasu zamykającego.";
        this["LE_FormulaParseException_FunctionNestingTooDeep"] = "Funkcja w formule jest zbyt głęboko zagnieżdżona w innych funkcjach.";
        this["LE_FormulaParseException_IncorrectNumberOfArguments"] = "Funkcja w formule nie ma prawidłowej liczby argumentów.";
        this["LE_FormulaParseException_InvalidArguments"] = "Operator lub funkcja w formule zawiera nieprawidłowe argumenty.";
        this["LE_FormulaParseException_InvalidConditionalFormatFormula"] = "Formuła formatu warunkowego nie może używać operatorów odwołań, takich jak sumy lub przecięcia, stałe tablicowe i odwołania do tabel strukturalnych.";
        this["LE_FormulaParseException_InvalidErrorValue"] = "Podany typ błędu jest nieprawidłowy.  Musi mieć jedną z następujących wartości: '#NULL!', '#DIV/0!', '#VALUE!', '#REF!', '#NAME?', '#NUM!' lub '#N/A'.";
        this["LE_FormulaParseException_InvalidFileNameInBrackets"] = "Nazwa pliku w nawiasach kwadratowych była nieprawidłowa.";
        this["LE_FormulaParseException_InvalidStructuredTableReference"] = "Strukturalne odwołanie do tabeli jest nieprawidłowe.";
        this["LE_FormulaParseException_InvalidWorkbookName"] = "Umieszczone w cudzysłowie odwołanie do skoroszytu nie jest prawidłową nazwą pliku skoroszytu.";
        this["LE_FormulaParseException_InvalidWorksheetName"] = "Nazwa arkusza jest nieprawidłowa.";
        this["LE_FormulaParseException_MacroFunctionNotAllowed"] = "Funkcje '{0}' są prawidłowe wyłącznie w nazwanym odwołaniu.";
        this["LE_FormulaParseException_Message_PortionWithError"] = "Część z błędem: {0}";
        this["LE_FormulaParseException_MissingArgumentAfterBinary"] = "Brak argumentu operacji za operatorem binarnym w formule.";
        this["LE_FormulaParseException_MissingArgumentAfterParen"] = "Po otwartym nawiasie nie ma wyrażenia.";
        this["LE_FormulaParseException_MissingArgumentAfterUnary"] = "Brak prawidłowego wyrażenia po operatorze jednoargumentowym w formule.";
        this["LE_FormulaParseException_MissingArgumentBeforeBinary"] = "Brak argumentu operacji przed operatorem binarnym w formule.";
        this["LE_FormulaParseException_NamedReferenceRefsNeedSheetName"] = "Wszystkie odwołania do komórek w nazwanej formule odniesienia muszą identyfikować arkusz.";
        this["LE_FormulaParseException_NoElementAfterArraySerapator"] = "W formule po separatorze elementów nie występuje element tablicy.";
        this["LE_FormulaParseException_NoEqualsSign"] = "Formuła musi się zaczynać od znaku równości.";
        this["LE_FormulaParseException_NoExclamationAfterWorkbookName"] = "Po odwołaniu do skoroszytu należy umieścić wykrzyknik.";
        this["LE_FormulaParseException_NoExclamationAfterWorksheetName"] = "Po odwołaniu do arkusza należy umieścić wykrzyknik.";
        this["LE_FormulaParseException_NoExpressions"] = "Formuła nie zawiera prawidłowych wyrażeń po znaku równości.";
        this["LE_FormulaParseException_NoFileNameAfterBracket"] = "Po otwartym nawiasie kwadratowym oczekiwano prawidłowej nazwy pliku.";
        this["LE_FormulaParseException_NoValidTermAfterWorkbookName"] = "Po odwołaniu do skoroszytu zewnętrznego musi następować nazwane odwołanie.";
        this["LE_FormulaParseException_NoValidTermAfterWorksheetName"] = "Po odwołaniu do arkusza musi następować komórka lub nazwane odwołanie.";
        this["LE_FormulaParseException_NoWorksheetAfterWorkbookName"] = "Po nazwie pliku skoroszytu w nawiasach kwadratowych należy podać nazwę arkusza.";
        this["LE_FormulaParseException_StringConstantLengthTooLong"] = "Określona formuła nie może zawierać ciągu dłuższego niż {0} znaków.";
        this["LE_FormulaParseException_TooLong"] = "Dla określonego formatu skoroszytu maksymalna długość formuły nie może przekraczać {0}.";
        this["LE_FormulaParseException_UnknownFunction"] = "Funkcja w formule nie ma prawidłowej nazwy.";
        this["LE_FormulaParseException_UnmatchedOpenBracket"] = "Stała tablica w formule nie ma nawiasu zamykającego.";
        this["LE_FormulaParseException_UnmatchedOpenParen"] = "Do otwartego nawiasu w formule nie pasuje żaden nawias zamykający.";
        this["LE_FormulaParseException_UnmatchedOpenSquareBracket"] = "Po nazwie pliku skoroszytu oczekiwano zamykającego nawiasu kwadratowego.";
        this["LE_FormulaParseException_WorkbookNameMissingEndQuote"] = "Odwołanie do skoroszytu nie ma cudzysłowu zamykającego.";
        this["LE_FormulaParseException_WorksheetRangeMissingEndingName"] = "Zakres arkusza nie określa nazwy drugiego arkusza.";
        this["LE_GradientStop_InvalidPosition"] = "Należy określić liczbę z przedziału od 0 do 1.";
        this["LE_IconSetInvalid"] = "IconSetNoIcon nie jest prawidłowym ustawieniem właściwości IconSet.";
        this["LE_IndexOutOfRangeException_ArrayBounds"] = "Indeks był poza granicami tablicy.";
        this["LE_InvalidEnumArgumentException_DefaultPatternCannotBeUsed"] = "Nie można użyć domyślnego wyliczenia FillPatternStyle do utworzenia klasy CellFillPattern.";
        this["LE_InvalidLineWeight"] = "Wartość LineWeight nie może być mniejsza niż 0 ani większa niż 1584.";
        this["LE_InvalidOperationException_AnchorCommentBeforeApplyingToCell"] = "Zanim komentarz zostanie zastosowany do komórki, musi mieć ustawione zarówno właściwości TopLeftCornerCell, jak i BottomRightCornerCell, lub oba muszą mieć wartość null.";
        this["LE_InvalidOperationException_AnchorShapeBeforeAddingToCollection"] = "Zanim kształt zostanie dodany do arkusza lub grupy musi mieć ustawione elementy TopLeftCornerCell i BottomRightCornerCell.";
        this["LE_InvalidOperationException_ArrayFormulaAlreadyApplied"] = "Formuła tablicowa została już zastosowana do zakresu komórek. Najpierw wyczyść istniejący zakres komórek.";
        this["LE_InvalidOperationException_ArrayFormulaAppliedInTable"] = "Formuł tablicowych nie można stosować do obszaru, w którym znajduje się tabela.";
        this["LE_InvalidOperationException_ArrayFormulaInMergedCell"] = "Formuły tablicowe w scalonych komórkach są nieprawidłowe.";
        this["LE_InvalidOperationException_BottomRightAnchorFromOtherWorksheet"] = "Dodanie kształtu do tej kolekcji spowoduje umieszczenie go w innym arkuszu niż jego element BottomRightCornerCell.";
        this["LE_InvalidOperationException_BuiltInStyleNameCannotBeChanged"] = "Nazwy stylu wbudowanego nie można zmienić.";
        this["LE_InvalidOperationException_CannotAddColorsToPaletteDirectly"] = "Nie można dodawać kolorów do palety WorkbookColorPalette. Użyj indeksatora, aby ustawić kolor o konkretnym indeksie.";
        this["LE_InvalidOperationException_CannotAddParagraphDirectly"] = "Akapitów nie można dodawać bezpośrednio.";
        this["LE_InvalidOperationException_CannotAddTableDirectly"] = "Wystąpień WorksheetTable nie można wstawiać bezpośrednio do kolekcji. Użyj metody Add.";
        this["LE_InvalidOperationException_CannotAddTableToRemovedWorksheet"] = "Tabel nie można dodawać do arkusza, który został usunięty z jego skoroszytu macierzystego.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileHeaderRowIsHidden"] = "Nie można zastosować filtru, gdy wiersz nagłówka tabeli jest ukryty.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileUIIsHidden"] = "Nie można zastosować filtru, gdy interfejs użytkownika filtru jest ukryty.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfData"] = "Nie można wstawić elementu {0}, ponieważ spowodowałoby to przesunięcie danych poza dolną część arkusza.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfObject"] = "Nie można wstawić elementu {0}, ponieważ spowodowałoby to przesunięcie kształtu lub komentarza poza dolną część arkusza.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitBlockingValue"] = "Nie można wstawić elementu {0}, ponieważ spowodowałoby to przesunięcie komórek w formule tablicowej lub tabeli danych.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitMergedRegion"] = "Nie można wstawić elementu {0}, ponieważ spowodowałoby to przesunięcie komórek w scalonym regionie.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitTable"] = "Nie można wstawić elementu {0}, ponieważ spowodowałoby to przesunięcie komórek w innej tabeli.";
        this["LE_InvalidOperationException_CannotInsertTableRow_TableOnBottomOfWorksheet"] = "Nie można wstawić elementu {0}, ponieważ tabela znajduje się na dole arkusza.";
        this["LE_InvalidOperationException_CannotModifyKeysCollection"] = "Nie można modyfikować kolekcji Keys.";
        this["LE_InvalidOperationException_CannotModifyStandardTableStyle"] = "Nie można modyfikować standardowych stylów tabeli.";
        this["LE_InvalidOperationException_CannotModifyValuesCollection"] = "Nie można modyfikować kolekcji Values.";
        this["LE_InvalidOperationException_CannotMoveDisconnectedWorksheet"] = "Nie można przenieść arkusza, ponieważ nie należy on do skoroszytu.";
        this["LE_InvalidOperationException_CannotRemoveColorsToPaletteDirectly"] = "Nie można usuwać kolorów z palety WorkbookColorPalette. Użyj metody Reset, aby wyczyścić wszystkie kolory niestandardowe.";
        this["LE_InvalidOperationException_CannotRemoveNormalStyle"] = "Nie można usunąć stylu Normalny.";
        this["LE_InvalidOperationException_CannotSetParentStyleOnStyle"] = "Nie można ustawić stylu w formacie należącym do innego stylu.";
        this["LE_InvalidOperationException_CannotShift_LossOfData"] = "Nie można przenieść danych poza arkusz.";
        this["LE_InvalidOperationException_CannotShift_LossOfObject"] = "Nie można przesunąć obiektów poza arkusz.";
        this["LE_InvalidOperationException_CannotShift_SplitBlockingValue"] = "Nie można przesunąć w formule tablicowej lub tabeli danych.";
        this["LE_InvalidOperationException_CannotShift_SplitMergedRegion"] = "Nie można przesunąć w scalonym regionie.";
        this["LE_InvalidOperationException_CannotShift_SplitTable"] = "Nie można przesunąć w tabeli.";
        this["LE_InvalidOperationException_CantAddChartInChart"] = "Nie można dodać wykresu do kolekcji kształtów innego wykresu.";
        this["LE_InvalidOperationException_CantAddCustomView"] = "Widoków niestandardowych nie można dodawać bezpośrednio do kolekcji.";
        this["LE_InvalidOperationException_CantAddDataTable"] = "Tabel danych nie można dodawać bezpośrednio.";
        this["LE_InvalidOperationException_CantAddMergedRegion"] = "Regionów komórek scalonych nie można dodawać bezpośrednio.";
        this["LE_InvalidOperationException_CantAddNamedReference"] = "Nazwanych odwołań nie można dodawać bezpośrednio.";
        this["LE_InvalidOperationException_CantAddWorksheet"] = "Arkuszy roboczych nie można dodawać bezpośrednio do tej kolekcji.";
        this["LE_InvalidOperationException_CantApplyRemovedCustomView"] = "Nie można zastosować widoku niestandardowego, jeśli został usunięty z kolekcji widoków niestandardowych skoroszytu.";
        this["LE_InvalidOperationException_CantChangeArrayFormula"] = "Nie można zmienić komórek formuły tablicowej.";
        this["LE_InvalidOperationException_CantChangeDataTable"] = "Nie można zmienić wewnętrznych komórek tabeli danych.";
        this["LE_InvalidOperationException_CantModifyCollection"] = "Nie można modyfikować kolekcji.";
        this["LE_InvalidOperationException_CantOverlapArrayFormula"] = "Region komórek scalonych nie może nakładać się na region formuły tablicowej.";
        this["LE_InvalidOperationException_CantOverlapDataTableInterior"] = "Region komórek scalonych nie może nakładać się na komórki w tabeli danych, jeśli wykracza poza górny wiersz lub skrajną lewą kolumnę tych komórek.";
        this["LE_InvalidOperationException_CantSaveEditRangeWithoutRange"] = "Element AllowedEditRanges arkusza musi mieć co najmniej jeden zakres.";
        this["LE_InvalidOperationException_CantSaveWithNoVisibleWorksheets"] = "Co najmniej jeden arkusz w skoroszycie musi być widoczny.";
        this["LE_InvalidOperationException_CantSaveWithNoWorksheets"] = "Nie można zapisać skoroszytu bez arkuszy";
        this["LE_InvalidOperationException_CantSetCircularityErrorDirectly"] = "Wartości błędu zaokrąglenia nie można zastosować bezpośrednio do komórek.";
        this["LE_InvalidOperationException_CantSetDataTableDirectly"] = "Tabel danych nie można ustawić jako wartości w komórce.";
        this["LE_InvalidOperationException_CantSetFormulaDirectly"] = "Formuł nie można ustawiać bezpośrednio w komórce. Zamiast tego użyj metody ApplyFormula lub jednej z metod ApplyTo wystąpienia formuły.";
        this["LE_InvalidOperationException_CellsInTableMinSize"] = "Region komórek w tabeli danych musi mieć co najmniej dwie komórki w każdym z wymiarów.";
        this["LE_InvalidOperationException_ChartNotSupported"] = "Modyfikacje elementu WorksheetChart nie są obsługiwane w tym formacie.";
        this["LE_InvalidOperationException_ClearUnknownShapeData"] = "Wszystkie dane tego kształtu są nieznane. Zamiast tego usuń kształt z jego kolekcji.";
        this["LE_InvalidOperationException_CollectionLongerThanMaxValue"] = "Liczba wpisów w kolekcji przekracza określoną wartość maksymalną.";
        this["LE_InvalidOperationException_CollectionModifiedWhileEnumerating"] = "Kolekcja została zmodyfikowana podczas wyliczania.";
        this["LE_InvalidOperationException_CurrentTableRowAddressNeedsOriginCell"] = "Aby przeanalizować strukturalne odwołanie do tabeli przy użyciu bieżącego wiersza, należy określić komórkę źródłową.";
        this["LE_InvalidOperationException_CustomViewNoVisibleWorksheets"] = "Co najmniej jeden arkusz w widoku niestandardowym o nazwie \"{0}\" musi być widoczny.";
        this["LE_InvalidOperationException_DataTableAppliedInTable"] = "Tabela danych nie może znajdować się w obszarze, w którym znajduje się tabela.";
        this["LE_InvalidOperationException_DataTableFormulaCannotBeApplied"] = "Formuły tabeli danych nie można zastosować bezpośrednio do komórek.";
        this["LE_InvalidOperationException_DataTableInMergedCell"] = "Wewnętrzne komórki tabeli danych nie mogą być częścią regionu komórek scalonych.";
        this["LE_InvalidOperationException_DataTableRemoved"] = "Nie można ustawić wierszowych i kolumnowych komórek wejściowych po usunięciu tabeli danych z arkusza.";
        this["LE_InvalidOperationException_DeprecatedFillColorSetWithGradientFill"] = "Nie można ustawić koloru wzoru wypełnienia, gdy w komórce jest ustawiony gradient.";
        this["LE_InvalidOperationException_EncryptedWorkbooksNotSupported"] = "Ładowany skoroszyt jest zaszyfrowany. Zaszyfrowane skoroszyty nie są obsługiwane.";
        this["LE_InvalidOperationException_ExtensionDoesntMatchCurrentFormat"] = "Nazwa pliku, w którym ma zostać zapisany plik, ma rozszerzenie niezgodne z bieżącym formatem skoroszytu.";
        this["LE_InvalidOperationException_FixedValuesFilterMustAcceptAValue"] = "Nie może istnieć filtr FixedValuesFilter, który nie zezwala na żadne wartości.";
        this["LE_InvalidOperationException_FormattedStringAlreadyOwned"] = "Sformatowany ciąg jest już wartością innej komórki lub komentarza. Sklonuj go najpierw, a potem zastosuj klon do nowej komórki lub kształtu.";
        this["LE_InvalidOperationException_FormattedStringNotOwned"] = "Nie można uzyskać dostępu do właściwości czcionki, jeśli do komórki lub kształtu nie jest zastosowany sformatowany ciąg lub tekst.";
        this["LE_InvalidOperationException_FormattedTextAlreadyOwned"] = "Sformatowany ciąg jest już wartością innego kształtu. Sklonuj go najpierw, a potem zastosuj klon do nowej komórki lub kształtu.";
        this["LE_InvalidOperationException_FormulaAlreadyOwned"] = "Formuła została już zastosowana do innej komórki.";
        this["LE_InvalidOperationException_FormulaReferencesInvalidCells"] = "Formuła odwołuje się do komórek, które nie są obsługiwane w bieżącym formacie pliku skoroszytu.";
        this["LE_InvalidOperationException_GetBoundsBeforeAnchorsSet"] = "Przed ustawieniem komórek kotwiczących kształt nie ma granic.";
        this["LE_InvalidOperationException_HiddenWorksheetCannotBeSelected"] = "Nie można wybrać ukrytego arkusza.";
        this["LE_InvalidOperationException_HyperlinkSealed"] = "Hiperłącza nie można modyfikować.";
        this["LE_InvalidOperationException_ImageDisposed"] = "Obraz obiektu WorksheetImage został usunięty lub jest nieprawidłowy.";
        this["LE_InvalidOperationException_InputCellsBothNull"] = "Zarówno kolumnowe, jak i wierszowe komórki wejściowe, nie mogą mieć wartości null.";
        this["LE_InvalidOperationException_InputCellsInTable"] = "Komórki wejściowe nie mogą się znajdować w komórkach tabeli danych.";
        this["LE_InvalidOperationException_InputCellsSame"] = "Kolumnowe i wierszowe komórki wejściowe nie mogą być tą samą komórką.";
        this["LE_InvalidOperationException_InvalidCharacterRange"] = "Zakres znaków objętych czcionką już nie istnieje w sformatowanym ciągu.";
        this["LE_InvalidOperationException_InvalidFormatOptionsInWholeTableArea"] = "Podana wartość formatu FormatOptions jest nieprawidłowa w obszarze tabeli WholeTable.";
        this["LE_InvalidOperationException_InvalidFormatString"] = "Podany ciąg formatu jest nieprawidłowy.";
        this["LE_InvalidOperationException_InvalidFormatString_GetTextCall"] = "Ciąg formatu komórki jest nieprawidłowy. Aby otrzymać o tym powiadomienie, gdy jest ustawiony element FormatString, ustaw element Workbook.ValidateFormatStrings na True.";
        this["LE_InvalidOperationException_InvalidForWorkbookFormat"] = "Formuła nie jest prawidłowa dla określonego formatu.";
        this["LE_InvalidOperationException_InvalidHeaderCellColumnAreaFormatProperty"] = "Wartości górnego obramowania nie można ustawić w formacie obszaru kolumn tabeli HeaderCell.";
        this["LE_InvalidOperationException_InvalidHeaderRowAreaFormatProperty"] = "Wartości górnego obramowania nie można ustawić w formacie obszaru tabeli HeaderRow.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFontProperty"] = "W obszarach stylów tabeli nie można ustawić wartości czcionki {0}.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatOptions"] = "Podana wartość formatu FormatOptions jest nieprawidłowa w obszarach stylów tabeli.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatProperty"] = "W obszarach stylów tabeli nie można ustawić wartości formatu {0}.";
        this["LE_InvalidOperationException_InvalidTotalsCellColumnAreaFormatProperty"] = "Wartości dolnego obramowania nie można ustawić w formacie obszaru kolumn tabeli TotalCell.";
        this["LE_InvalidOperationException_InvalidTotalsRowAreaFormatProperty"] = "Wartości dolnego obramowania nie można ustawić w formacie obszaru tabeli TotalsRow.";
        this["LE_InvalidOperationException_InvalidWholeTableAreaFormatProperty"] = "W formacie obszaru tabeli WholeTable można ustawić wyłącznie wartości obramowania krawędzi.";
        this["LE_InvalidOperationException_LastColumnBeforeFirst"] = "Ostatnia kolumna w regionie powinna być większa lub równa pierwszej kolumnie.";
        this["LE_InvalidOperationException_LastRowBeforeFirst"] = "Ostatni wiersz w regionie powinien być większy niż pierwszy wiersz.";
        this["LE_InvalidOperationException_MaxCellFormats"] = "Maksymalna liczba różnych formatów komórek w pliku Excel wynosi: {0}";
        this["LE_InvalidOperationException_MaxColors"] = "Maksymalna liczba kolorów w skoroszycie programu Excel wynosi: {0}";
        this["LE_InvalidOperationException_MaxColumns"] = "Maksymalna liczba kolumn w arkuszu programu Excel w bieżącym formacie to: {0}";
        this["LE_InvalidOperationException_MaxFonts"] = "Liczba czcionek w skoroszycie nie może przekraczać {0}.";
        this["LE_InvalidOperationException_MaxFormattedStrings"] = "Liczba unikatowych ciągów w skoroszycie nie może przekraczać {0}.";
        this["LE_InvalidOperationException_MaxRows"] = "Maksymalna liczba wierszy w arkuszu programu Excel w bieżącym formacie to: {0}";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableLeftColumn"] = "Lewa skrajna kolumna komórek określona dla tabeli danych zawiera region komórek scalonych, który rozciąga się poza obszar tych komórek.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopLeftCell"] = "Lewa górna komórka spośród komórek określonych dla tabeli danych jest częścią komórki scalonej, która rozciąga się poza obszar tych komórek.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopRow"] = "Górny wiersz komórek określonych dla tabeli danych zawiera region komórek scalonych, który rozciąga się poza obszar tych komórek.";
        this["LE_InvalidOperationException_MergedCellsAppliedInTable"] = "Nie można scalić komórek w tabeli.";
        this["LE_InvalidOperationException_NamedReferenceNameAlreadyExists"] = "Nazwa \"{0}\" jest używana przez tabelę lub przez nazwane odwołanie w skoroszycie w tym samym zakresie.";
        this["LE_InvalidOperationException_NoRegionHasBeenSpecified"] = "Nie określono żadnego regionu.";
        this["LE_InvalidOperationException_NoWorksheetContextToSetShapeBounds"] = "Tego przeładowania Add nie można wywołać, gdy kolekcja kształtów nie jest skojarzona z arkuszem. Wywołaj przeładowanie Add, które przyjmuje wystąpienie arkusza.";
        this["LE_InvalidOperationException_OverlappingTable"] = "We wskazanym regionie znajduje się inna tabela.";
        this["LE_InvalidOperationException_ReadOnlyFont"] = "Obiekt czcionki jest przeznaczony tylko do odczytu i nie można go zmienić.";
        this["LE_InvalidOperationException_ReadOnlyFormat"] = "Obiekt formatu jest przeznaczony tylko do odczytu i nie można go zmienić.";
        this["LE_InvalidOperationException_RelativeR1C1AddressNeedsOriginCell"] = "Aby przeanalizować względny adres W1K1, należy określić komórkę źródłową.";
        this["LE_InvalidOperationException_ResizedTableMustBeOnWorksheet"] = "Tabela musi znajdować się w arkuszu, aby można było zmienić jej rozmiar.";
        this["LE_InvalidOperationException_ResolvedFormatCannotBeModified"] = "Format komórki jest przeznaczony tylko do odczytu i nie można go zmienić.";
        this["LE_InvalidOperationException_ShapeCantChangeOrientation"] = "Tego kształtu nie można obracać ani odwracać.";
        this["LE_InvalidOperationException_ShapeFillSolidIsImmutable"] = "Element ShapeFillSolid jest niezmienialny. Zamiast tego utwórz nowy element ShapeFillSolid.";
        this["LE_InvalidOperationException_ShapeInAnotherCollection"] = "Kształt już istnieje w innym arkuszu lub grupie kształtów.";
        this["LE_InvalidOperationException_ShapeOutlineSolidIsImmutable"] = "Element ShapeOutlineSolid jest niezmienialny. Zamiast tego utwórz nowy element ShapeOutlineSolid.";
        this["LE_InvalidOperationException_ShowFilterUIWhileHeaderRowHidden"] = "Nie można wyświetlić interfejsu użytkownika filtru, gdy wiersz nagłówka jest ukryty.";
        this["LE_InvalidOperationException_SystemColorsAccessedFromWrongThread"] = "Nie można uzyskać dostępu do elementu SystemColors z wątku w tle.";
        this["LE_InvalidOperationException_TableReferenceMustBeUsedFromInsideTable"] = "Formuła zawiera strukturalne odwołanie do tabeli, którego można używać wyłącznie z wnętrza tabeli.";
        this["LE_InvalidOperationException_TableReferenceToMissingTable"] = "Formuła zawiera odwołanie do tabeli, która nie istnieje.";
        this["LE_InvalidOperationException_TableReferenceToMissingTableColumn"] = "Formuła zawiera odwołanie do nieistniejącej kolumny tabeli.";
        this["LE_InvalidOperationException_TopLeftAnchorFromOtherWorksheet"] = "Dodanie kształtu do tej kolekcji spowoduje umieszczenie go w innym arkuszu niż jego element TopLeftCornerCell.";
        this["LE_InvalidOperationException_TotalsRowCannotBeShownInLastRow"] = "Nie można wyświetlić wiersza sum, jeśli tabela zajmuje ostatni wiersz w arkuszu.";
        this["LE_InvalidOperationException_UnknownBIFFFormat"] = "Plik skoroszytu był zapisany w nieznanym formacie BIFF.";
        this["LE_InvalidOperationException_UnsupportedBIFFFormat"] = "Plik skoroszytu został zapisany w formacie {0}, który nie jest obsługiwany.";
        this["LE_InvalidOperationException_WorkbookDataViolatesFormatLimits"] = "Skoroszyt zawiera dane, które przekraczałyby limity narzucone przez określony format.";
        this["LE_InvalidRegion"] = "Nieprawidłowy adres regionu.";
        this["LE_InvalidValueType"] = "{0} nie jest prawidłową wartością właściwości ValueType.";
        this["LE_LocationNotSingleRowOrColumn"] = "Odwołanie do lokalizacji jest nieprawidłowe, ponieważ nie wszystkie komórki znajdują się w tej samej kolumnie lub wierszu. Zaznacz komórki, które znajdują się wszystkie w jednej kolumnie lub wierszu.";
        this["LE_LocationOrDataAreaInvalid"] = "Odwołanie do lokalizacji lub zakresu danych jest nieprawidłowe.";
        this["LE_LocationWorksheetMismatch"] = "Lokalizacja musi pochodzić z tego arkusza.";
        this["LE_NotSingleTargetFormula"] = "Obsługuje wyłącznie formuły z pojedynczym obiektem docelowym.";
        this["LE_NotSupportedException_CellType"] = "{0} nie jest obsługiwanym typem wartości komórki.";
        this["LE_NotSupportedException_NoPackageFactory"] = "Element packageFactory nie może mieć wartości null. Podczas ładowania danych ze skoroszytu formatu Excel2007 i korzystania z zestawu Infragistics2.Documents.Excel należy dostarczyć element IPackageFactory do obsługi pakietowania danych. Jeśli używasz środowiska DotNet Framework 3.0 lub nowszego, zamiast tego użyj zestawu Infragistics3.Documents.Excel, aby pakietowanie było obsługiwane przez klasę WindowsBase.";
        this["LE_OpenPackagingNonConformanceException_ContainsDublinCoreRefinements"] = "IPackagePart to podstawowa relacja właściwości. Zawiera udoskonalenia elementów Dublin Core innych niż 'dcterms: created' i 'dcterms: modified'.";
        this["LE_OpenPackagingNonConformanceException_ContainsXmlLanguageAttribute"] = "IPackagePart to podstawowa relacja właściwości. Zawiera elementy, które zawierają atrybut 'xml: lang'.";
        this["LE_OpenPackagingNonConformanceException_CorePropertiesRelationshipAlreadyProcessed"] = "IPackagePart to podstawowa relacja właściwości, ale została już przetworzona dla skojarzonego elementu IPackage.";
        this["LE_OpenPackagingNonConformanceException_DuplicatePartName"] = "Element IPackagePart o tej samej nazwie już należy do tego pakietu IPackage. Pakiet IPackage nie może zawierać elementów o identycznych nazwach. Należy pamiętać, że podczas sprawdzania zduplikowanych wartości wielkość liter nie jest rozróżniana.";
        this["LE_OpenPackagingNonConformanceException_UsesMarkupCompatibilityNamespace"] = "IPackagePart to podstawowa relacja właściwości. Używa przestrzeni nazw zgodności znaczników.";
        this["LE_OpenPackagingNonConformanceException_XmlContainsDocumentTypeDefinition"] = "Element IPackagePart zawierał deklarację DTD (definicja typu dokumentu). Reguły Open Packaging Convention zabraniają używania zawartości XML z deklaracjami DTD.";
        this["LE_OpenPackagingNonConformanceException_XsiTypeAttributeInvalid"] = "IPackagePart to podstawowa relacja właściwości. Zawiera elementy, które zawierają atrybut 'xsi: type', lub elementy, które są wymagane do atrybutu 'xsi: type' i które go nie zawierają.";
        this["LE_OpenXmlSerializerNotLoaded"] = "Format '{0}' wymaga programu szeregującego, który nie jest załadowany. W przypadku JavaScript jest to zawarte w pliku infragistics.excel_serialization_openxml.js.";
        this["LE_PercentageValueOutOfRange"] = "Wartość procentowa {0} musi należeć do przedziału od 0 do 100.";
        this["LE_PercentileValueOutOfRange"] = "Wartość percentyla {0} musi należeć do przedziału od 0 do 100.";
        this["LE_PercentRangeInvalid"] = "Wartość {0} musi być mniejsza niż {1}.";
        this["LE_RowsToRepeatAtTopAreOutsideAvailableRange"] = "Opcje drukowania obejmują wiersze, które powtarzają się u góry i są poza zakresem dostępnych wierszy w nowym formacie.";
        this["LE_Serialization_NoPrimaryCategoryAxis"] = "Typ wykresu '{0}' wymaga głównej osi kategorii. Dodaj główną oś kategorii do kolekcji AxisCollection wykresu.";
        this["LE_Serialization_NoPrimaryValueAxis"] = "Typ wykresu '{0}' wymaga głównej osi wartości. Dodaj główną oś wartości do kolekcji AxisCollection wykresu.";
        this["LE_Series_ChartType_ChartEx"] = "Wartość '{0}' jest nieprawidłowa dla serii ChartType na wykresie '{1}'.";
        this["LE_Series_ChartType_InvalidForComboChart"] = "Typu wykresu '{0}' nie można używać z wykresem kombi.";
        this["LE_Series_ChartType_NonCombo"] = "Typ wykresu dla tej serii jest określany przez typ wykresu nadrzędnego ('{0}') i nie można go ustawić. Zamiast tego ustaw właściwość ChartType wykresu nadrzędnego na '{1}'.";
        this["LE_Series_Explosion_OutOfRange"] = "Właściwość Explosion obsługuje wartości z przedziału od 0 do 400 włącznie.";
        this["LE_Series_InvalidAxisGroup"] = "Nie można ustawić właściwości AxisGroup na '{0}', ponieważ seria typu '{1}' istnieje na osi, która nie może mieć tej samej osi, co seria typu '{2}'.";
        this["LE_Series_InvalidSeriesTypeForComboChart"] = "Serii typu '{0}' nie można używać na wykresie kombi.";
        this["LE_Series_NoPrimaryAxis"] = "Na oś główną należy nanieść co najmniej jedną serię. Zmień właściwość AxisGroup na 'Primary' dla co najmniej jednej serii.";
        this["LE_Series_OwningSeries_NotClusteredColumn"] = "Na wykresie Pareto właściwość OwningSeries można ustawić wyłącznie na serię kolumnowej grupowanej.";
        this["LE_Series_OwningSeries_NotParetoLine"] = "Na wykresie Pareto właściwość OwningSeries można ustawić wyłącznie dla serii liniowej.";
        this["LE_Series_OwningSeries_Same"] = "Właściwość OwningSeries nie może odwoływać się do siebie.";
        this["LE_Series_SeriesType_ChartEx"] = "Wartość SeriesType '{0}' jest nieprawidłowa dla wykresu '{1}'.";
        this["LE_Series_SeriesType_NonCombo"] = "Właściwość Type ma zastosowanie wyłącznie do wykresów kombi i nie można jej ustawić dla wykresu typu '{0}'.";
        this["LE_SetValue_Formula"] = "Typ wartości '{0}' jest prawidłowy wyłącznie w przypadku formuł. Zamiast tego użyj metody SetFormula.";
        this["LE_SetValue_NonNumeric"] = "Parametr wartości nie ma zastosowania do typu wartości '{0}'. Użyj przeładowania, które obsługuje wyłącznie typ wartości.";
        this["LE_SetValue_Numeric"] = "Należy określić parametr wartości dla typu wartości '{0}'. Użyj przeładowania, które obsługuje zarówno wartość, jak i typ wartości.";
        this["LE_WorksheetChart_ChartType_ChartEx_CannotTransition"] = "Nie można zmienić właściwości ChartType z '{0}' na '{1}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_From"] = "Nie można zmienić właściwości ChartType wykresu kombi na '{0}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_To"] = "Wykresu typu '{0}» nie można zmienić na 'Combo'. Nowy wykres kombi można utworzyć za pomocą metody AddChart, ale zmiana istniejącego wykresu na wykres kombi nie jest obsługiwana.";
        this["LE_WorksheetChart_ComboChart_NotEnoughSeries"] = "Wykres kombi wymaga co najmniej 2 serii. Dodaj więcej elementów do kolekcji SeriesCollection wykresu.";
        this["LE_WorksheetChart_MissingRequiredAxis_StockCharts"] = "Wykres '{0}' wymaga głównej osi kategorii, głównej osi wartości, dodatkowej osi kategorii i dodatkowej osi wartości.";
        this["LE_WorksheetChart_NotEnoughSeries"] = "Wykres '{0}' wymaga co najmniej {1} serii/serie. Aby użyć tego typu wykresu, musisz dodać więcej elementów do kolekcji SeriesCollection.";
        this["LE_WorksheetChart_Save_Series"] = "Wykres '{0}' musi zawierać co najmniej jedną serię. Dodaj więcej serii do wykresu, zanim spróbujesz zapisać skoroszyt.";
        this["LE_WorksheetChart_Save_Series_Pareto"] = "Wykres 'Pareto' musi zawierać jedną serię 'ColumnClustered' i jedną serię 'Pareto'. Dodaj odpowiednią serię do wykresu przed próbą zapisania skoroszytu.";
        this["LE_WorksheetChart_SetComboChartSourceData_InvalidChartTypes"] = "Co najmniej jeden z podanych typów wykresów nie może być używany na wykresie kombi: {0}";
        this["LE_WorksheetChart_SetComboChartSourceData_NotAComboChart"] = "Nie można wywołać metody SetComboChartSourceData, ponieważ bieżący typ wykresu to '{0}'. Zmień wartość właściwości ChartType na 'Combo'.";
        this["LE_WorksheetChart_SetComboChartSourceData_SeriesChartTypesNotSupported"] = "W przypadku wykresu '{0}' parametr seriesChartTypes musi mieć wartość null.";
        this["LE_WorksheetChart_SetSourceData_NotEnoughSeries"] = "Wykresy kombi wymagają co najmniej 2 serii. Zmień parametr 'dataRange' na zakres, który zapewni co najmniej 2 serie.";
        this["LE_WorksheetChart_SetSourceData_TooManyAxes"] = "Podana kombinacja typów wykresów wymaga więcej osi niż maksymalna dopuszczalna liczba osi.";
        this["LE_WorksheetChart_StockChartNotSupported"] = "Edycja typu wykresu '{0}' nie jest obecnie obsługiwana.";
        this["LE_WorksheetChart_UnknownChartNotSupported"] = "Ten wykres został załadowany z istniejącego arkusza, ale nie można określić jego właściwości ChartType. Edycja nieznanych typów wykresów nie jest obecnie obsługiwana.";
        this["LER_ArgumentOutOfRangeException_OutlineLevel"] = "Poziom konspektu musi należeć do przedziału od 0 do 7.";
        this["LER_Exception_KeyNotFound"] = "Nie znaleziono klucza.";
        this["LER_Exception_MergedRegionsOverlap"] = "Scalone regiony komórek nie mogą się nakładać.";
        this["TableHeaderRowDescription"] = "wiersz nagłówka";
        this["TableInsertRowDescription"] = "wstaw wiersz";
        this["TableTotalsRowDescription"] = "wiersz sumy";
        this["Value_UCErrorCode_Div"] = "#DIV/0!";
        this["Value_UCErrorCode_Fail"] = "#FAIL";
        this["Value_UCErrorCode_NA"] = "#N/A";
        this["Value_UCErrorCode_Name"] = "#NAME?";
        this["Value_UCErrorCode_Null"] = "#NULL!";
        this["Value_UCErrorCode_Num"] = "#NUM!";
        this["Value_UCErrorCode_Ok"] = "#OK";
        this["Value_UCErrorCode_Reference"] = "#REF!";
        this["Value_UCErrorCode_Unknown"] = "#N/A";
        this["Value_UCErrorCode_Value"] = "#VALUE!";
        this["WorkbookColorInfo_Automatic_Description"] = "Automatycznie";
        this["WorkbookColorInfo_Description"] = "ColorInfo: {0}";
        this["WorkbookColorInfo_WithTint_Description"] = "ColorInfo: {0}, Tint: {1}";
        this["WorksheetShapeSerialization_GroupName"] = "Grupa";
        this["WorksheetShapeSerialization_ImageName"] = "Obraz";
        this["WorksheetShapeSerialization_ShapeName"] = "Kształt";
    }
}
