// igniteui-react-excel/esm2015/lib/ExcelCalcErrorCode.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Enumeration of error codes assigned to [[ExcelCalcErrorValue]].
 */
export var ExcelCalcErrorCode = /*@__PURE__*/ (function (ExcelCalcErrorCode) {
    /**
     * Occurs when an invalid or disconnected reference is encountered while evaluating a formula
     */
    ExcelCalcErrorCode[ExcelCalcErrorCode["Reference"] = 0] = "Reference";
    /**
     * Occurs when the wrong type of argument or operand is used in a formula
     */
    ExcelCalcErrorCode[ExcelCalcErrorCode["Value"] = 1] = "Value";
    /**
     * Occurs when a number is divided by zero (0)
     */
    ExcelCalcErrorCode[ExcelCalcErrorCode["Div"] = 2] = "Div";
    /**
     * Occurs when @NA is entered into a formula
     */
    ExcelCalcErrorCode[ExcelCalcErrorCode["NA"] = 3] = "NA";
    /**
     * Occurs with invalid numeric values in a formula or function
     */
    ExcelCalcErrorCode[ExcelCalcErrorCode["Num"] = 4] = "Num";
    /**
     * Occurs when a circularity formula is used when circularities are not allowed.
     */
    ExcelCalcErrorCode[ExcelCalcErrorCode["Circularity"] = 5] = "Circularity";
    /**
     * Occurs when there is an intersection of two references that do not contain any common cells.
     */
    ExcelCalcErrorCode[ExcelCalcErrorCode["Null"] = 6] = "Null";
    /**
     * Occurs when text in a formula is not recognized.
     */
    ExcelCalcErrorCode[ExcelCalcErrorCode["Name1"] = 7] = "Name1";
    return ExcelCalcErrorCode;
})({});
/**
 * @hidden
 */
export let ExcelCalcErrorCode_$type = /*@__PURE__*/ markEnum('ExcelCalcErrorCode', 'Reference,0|Value,1|Div,2|NA,3|Num,4|Circularity,5|Null,6|Name:Name1,7');
