// igniteui-react-excel/esm2015/lib/ExtendedPropertiesPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ExtendedPropertiesPartManager = /*@__PURE__*/ (() => {
    class ExtendedPropertiesPartManager extends XmlPartManagerBase {
        get_m() {
            return ExtendedPropertiesPartManager.a3;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            return "/docProps/app.xml";
        }
        o(a) {
            switch (a) {
                case 0: return ExtendedPropertiesPartManager.a6;
                case 1: return ExtendedPropertiesPartManager.a5;
                default: return ExtendedPropertiesPartManager.a6;
            }
        }
        get_g() {
            return false;
        }
        get g() {
            return this.get_g();
        }
    }
    ExtendedPropertiesPartManager.$t = /*@__PURE__*/ markType(ExtendedPropertiesPartManager, 'ExtendedPropertiesPartManager', XmlPartManagerBase.$);
    ExtendedPropertiesPartManager.a3 = "application/vnd.openxmlformats-officedocument.extended-properties+xml";
    ExtendedPropertiesPartManager.a6 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/extended-properties";
    ExtendedPropertiesPartManager.a5 = "http://purl.oclc.org/ooxml/officeDocument/relationships/extendedProperties";
    return ExtendedPropertiesPartManager;
})();
