// igniteui-react-excel/esm2015/lib/ExternalEventWorkItem$1.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WorkItem$1 } from "./WorkItem$1";
import { markType } from "igniteui-react-core";
import { WorkItem_ExceptionInfo } from "./WorkItem_ExceptionInfo";
/**
 * @hidden
 */
export let ExternalEventWorkItem$1 = /*@__PURE__*/ (() => {
    class ExternalEventWorkItem$1 extends WorkItem$1 {
        constructor($t) {
            super($t, 2, null, false);
            this.$t = null;
            this.af = false;
            this.ae = null;
            this.$t = $t;
            this.$type = this.$type.specialize(this.$t);
        }
        p(a) {
            if (this.af) {
                return true;
            }
            this.ae = a;
            return false;
        }
        get_t() {
            return this.af == false;
        }
        ag(a, b = null) {
            this.ad = a;
            if (b != null) {
                this.aa(new WorkItem_ExceptionInfo(b));
            }
            this.af = true;
            if (this.ae != null) {
                this.ae.e(this);
            }
        }
    }
    ExternalEventWorkItem$1.$t = /*@__PURE__*/ markType(ExternalEventWorkItem$1, 'ExternalEventWorkItem$1', /*@__PURE__*/ WorkItem$1.$.specialize(0));
    return ExternalEventWorkItem$1;
})();
