// igniteui-react-excel/esm2015/lib/ExternalEventWorkItem.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WorkItem } from "./WorkItem";
import { markType } from "igniteui-react-core";
import { WorkItem_ExceptionInfo } from "./WorkItem_ExceptionInfo";
/**
 * @hidden
 */
export let ExternalEventWorkItem = /*@__PURE__*/ (() => {
    class ExternalEventWorkItem extends WorkItem {
        constructor() {
            super(2, null, false);
            this.ae = false;
            this.ad = null;
        }
        p(a) {
            if (this.ae) {
                return true;
            }
            this.ad = a;
            return false;
        }
        get_t() {
            return this.ae == false;
        }
        af(a = null) {
            if (a != null) {
                this.aa(new WorkItem_ExceptionInfo(a));
            }
            this.ae = true;
            if (this.ad != null) {
                this.ad.e(this);
            }
        }
    }
    ExternalEventWorkItem.$t = /*@__PURE__*/ markType(ExternalEventWorkItem, 'ExternalEventWorkItem', WorkItem.$);
    return ExternalEventWorkItem;
})();
