// igniteui-react-excel/esm2015/lib/ExternalLinkPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ExternalLinkPartManager = /*@__PURE__*/ (() => {
    class ExternalLinkPartManager extends XmlPartManagerBase {
        get_m() {
            return ExternalLinkPartManager.a4;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/externalLinks/externalLink.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return ExternalLinkPartManager.a6;
                case 1: return ExternalLinkPartManager.a5;
                default: return ExternalLinkPartManager.a6;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    ExternalLinkPartManager.$t = /*@__PURE__*/ markType(ExternalLinkPartManager, 'ExternalLinkPartManager', XmlPartManagerBase.$);
    ExternalLinkPartManager.a4 = "application/vnd.openxmlformats-officedocument.spreadsheetml.externalLink+xml";
    ExternalLinkPartManager.a6 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/externalLink";
    ExternalLinkPartManager.a5 = "http://purl.oclc.org/ooxml/officeDocument/relationships/externalLink";
    return ExternalLinkPartManager;
})();
