// igniteui-react-excel/esm2015/lib/FillPatternStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Enumeration for fill pattern styles. Default value is used in property override situations.
 */
export var FillPatternStyle = /*@__PURE__*/ (function (FillPatternStyle) {
    /**
     * Obsolete. Use None instead.
     */
    FillPatternStyle[FillPatternStyle["Default"] = -1] = "Default";
    /**
     * No fill pattern.
     */
    FillPatternStyle[FillPatternStyle["None"] = 0] = "None";
    /**
     * Solid fill pattern with fill pattern foreground color.
     */
    FillPatternStyle[FillPatternStyle["Solid"] = 1] = "Solid";
    /**
     * "50% gray" fill pattern.
     */
    FillPatternStyle[FillPatternStyle["Gray50percent"] = 2] = "Gray50percent";
    /**
     * "75% gray" fill pattern.
     */
    FillPatternStyle[FillPatternStyle["Gray75percent"] = 3] = "Gray75percent";
    /**
     * "25% gray" fill pattern.
     */
    FillPatternStyle[FillPatternStyle["Gray25percent"] = 4] = "Gray25percent";
    /**
     * Horizontal stripe fill pattern.
     */
    FillPatternStyle[FillPatternStyle["HorizontalStripe"] = 5] = "HorizontalStripe";
    /**
     * Vertical stripe fill pattern.
     */
    FillPatternStyle[FillPatternStyle["VerticalStripe"] = 6] = "VerticalStripe";
    /**
     * Reverse diagonal stripe fill pattern.
     */
    FillPatternStyle[FillPatternStyle["ReverseDiagonalStripe"] = 7] = "ReverseDiagonalStripe";
    /**
     * Diagonal stripe fill pattern.
     */
    FillPatternStyle[FillPatternStyle["DiagonalStripe"] = 8] = "DiagonalStripe";
    /**
     * Diagonal crosshatch fill pattern.
     */
    FillPatternStyle[FillPatternStyle["DiagonalCrosshatch"] = 9] = "DiagonalCrosshatch";
    /**
     * Thick diagonal crosshatch fill pattern.
     */
    FillPatternStyle[FillPatternStyle["ThickDiagonalCrosshatch"] = 10] = "ThickDiagonalCrosshatch";
    /**
     * Thin horizontal stripe fill pattern.
     */
    FillPatternStyle[FillPatternStyle["ThinHorizontalStripe"] = 11] = "ThinHorizontalStripe";
    /**
     * Thin vertical stripe fill pattern.
     */
    FillPatternStyle[FillPatternStyle["ThinVerticalStripe"] = 12] = "ThinVerticalStripe";
    /**
     * Thin reverse diagonal stripe fill pattern.
     */
    FillPatternStyle[FillPatternStyle["ThinReverseDiagonalStripe"] = 13] = "ThinReverseDiagonalStripe";
    /**
     * Thin diagonal stripe fill pattern.
     */
    FillPatternStyle[FillPatternStyle["ThinDiagonalStripe"] = 14] = "ThinDiagonalStripe";
    /**
     * Thin horizontal crosshatch fill pattern.
     */
    FillPatternStyle[FillPatternStyle["ThinHorizontalCrosshatch"] = 15] = "ThinHorizontalCrosshatch";
    /**
     * Thin diagonal crosshatch fill pattern.
     */
    FillPatternStyle[FillPatternStyle["ThinDiagonalCrosshatch"] = 16] = "ThinDiagonalCrosshatch";
    /**
     * "12.5% gray" fill pattern.
     */
    FillPatternStyle[FillPatternStyle["Gray12percent"] = 17] = "Gray12percent";
    /**
     * "6.25% gray" fill pattern.
     */
    FillPatternStyle[FillPatternStyle["Gray6percent"] = 18] = "Gray6percent";
    return FillPatternStyle;
})({});
/**
 * @hidden
 */
export let FillPatternStyle_$type = /*@__PURE__*/ markEnum('FillPatternStyle', 'Default,-1|None,0|Solid,1|Gray50percent,2|Gray75percent,3|Gray25percent,4|HorizontalStripe,5|VerticalStripe,6|ReverseDiagonalStripe,7|DiagonalStripe,8|DiagonalCrosshatch,9|ThickDiagonalCrosshatch,10|ThinHorizontalStripe,11|ThinVerticalStripe,12|ThinReverseDiagonalStripe,13|ThinDiagonalStripe,14|ThinHorizontalCrosshatch,15|ThinDiagonalCrosshatch,16|Gray12percent,17|Gray6percent,18');
