// igniteui-react-excel/esm2015/lib/FixedDateGroupType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various types, or precisions, of a [[FixedDateGroup]].
 * @see [[FixedDateGroup.type]]
 */
export var FixedDateGroupType = /*@__PURE__*/ (function (FixedDateGroupType) {
    /**
     * The group represents the day in which the [[FixedDateGroup.value]] exists.
     */
    FixedDateGroupType[FixedDateGroupType["Day"] = 0] = "Day";
    /**
     * The group represents the hour in which the [[FixedDateGroup.value]] exists.
     */
    FixedDateGroupType[FixedDateGroupType["Hour"] = 1] = "Hour";
    /**
     * The group represents the minute in which the [[FixedDateGroup.value]] exists.
     */
    FixedDateGroupType[FixedDateGroupType["Minute"] = 2] = "Minute";
    /**
     * The group represents the month in which the [[FixedDateGroup.value]] exists.
     */
    FixedDateGroupType[FixedDateGroupType["Month"] = 3] = "Month";
    /**
     * The group represents the second in which the [[FixedDateGroup.value]] exists.
     */
    FixedDateGroupType[FixedDateGroupType["Second"] = 4] = "Second";
    /**
     * The group represents the year in which the [[FixedDateGroup.value]] exists.
     */
    FixedDateGroupType[FixedDateGroupType["Year"] = 5] = "Year";
    return FixedDateGroupType;
})({});
/**
 * @hidden
 */
export let FixedDateGroupType_$type = /*@__PURE__*/ markEnum('FixedDateGroupType', 'Day,0|Hour,1|Minute,2|Month,3|Second,4|Year,5');
