// igniteui-react-excel/esm2015/lib/FontSuperscriptSubscriptStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Enumeration for font superscript or subscript style. Default value is used in property override situations.
 */
export var FontSuperscriptSubscriptStyle = /*@__PURE__*/ (function (FontSuperscriptSubscriptStyle) {
    /**
     * Use the current default.
     */
    FontSuperscriptSubscriptStyle[FontSuperscriptSubscriptStyle["Default"] = -1] = "Default";
    /**
     * No superscript or subscript style.
     */
    FontSuperscriptSubscriptStyle[FontSuperscriptSubscriptStyle["None"] = 0] = "None";
    /**
     * Superscript style.
     */
    FontSuperscriptSubscriptStyle[FontSuperscriptSubscriptStyle["Superscript"] = 1] = "Superscript";
    /**
     * Subscript style.
     */
    FontSuperscriptSubscriptStyle[FontSuperscriptSubscriptStyle["Subscript"] = 2] = "Subscript";
    return FontSuperscriptSubscriptStyle;
})({});
/**
 * @hidden
 */
export let FontSuperscriptSubscriptStyle_$type = /*@__PURE__*/ markEnum('FontSuperscriptSubscriptStyle', 'Default,-1|None,0|Superscript,1|Subscript,2');
