// igniteui-react-excel/esm2015/lib/FontUnderlineStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Enumeration for font underline styles. Default value is used in property override situations.
 */
export var FontUnderlineStyle = /*@__PURE__*/ (function (FontUnderlineStyle) {
    /**
     * Use the current default.
     */
    FontUnderlineStyle[FontUnderlineStyle["Default"] = -1] = "Default";
    /**
     * No underline style.
     */
    FontUnderlineStyle[FontUnderlineStyle["None"] = 0] = "None";
    /**
     * Single underline style.
     */
    FontUnderlineStyle[FontUnderlineStyle["Single"] = 1] = "Single";
    /**
     * Double underline style.
     */
    FontUnderlineStyle[FontUnderlineStyle["double1"] = 2] = "double1";
    /**
     * Single accounting underline style.
     */
    FontUnderlineStyle[FontUnderlineStyle["SingleAccounting"] = 33] = "SingleAccounting";
    /**
     * Double accounting underline style.
     */
    FontUnderlineStyle[FontUnderlineStyle["DoubleAccounting"] = 34] = "DoubleAccounting";
    return FontUnderlineStyle;
})({});
/**
 * @hidden
 */
export let FontUnderlineStyle_$type = /*@__PURE__*/ markEnum('FontUnderlineStyle', 'Default,-1|None,0|Single,1|Double:double1,2|SingleAccounting,33|DoubleAccounting,34');
