// igniteui-react-excel/esm2015/lib/FormulaParser_PositionInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, markStruct } from "igniteui-react-core";
import { FormulaParser_TextSpan } from "./FormulaParser_TextSpan";
/**
 * @hidden
 */
export let FormulaParser_PositionInfo = /*@__PURE__*/ (() => {
    class FormulaParser_PositionInfo extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.a = new FormulaParser_TextSpan();
            this.c = 0;
            this.e = 0;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        let d = _rest[1];
                        let e = _rest[2];
                        this.c = c;
                        this.e = d;
                        this.a = e;
                    }
                    break;
            }
        }
        get b() {
            return this.a;
        }
        get d() {
            return this.c;
        }
        get f() {
            return this.e;
        }
    }
    FormulaParser_PositionInfo.$t = /*@__PURE__*/ markStruct(FormulaParser_PositionInfo, 'FormulaParser_PositionInfo');
    return FormulaParser_PositionInfo;
})();
