// igniteui-react-excel/esm2015/lib/FormatConditionOperator.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which specify the manner in which a cell value is
 * compared against one or more operands.
 */
export var FormatConditionOperator = /*@__PURE__*/ (function (FormatConditionOperator) {
    /**
     * The condition evaluates to true if the cell value is between
     * the values of the two operands.
     */
    FormatConditionOperator[FormatConditionOperator["Between"] = 0] = "Between";
    /**
     * The condition evaluates to true if the cell value is not between
     * the values of the two operands.
     */
    FormatConditionOperator[FormatConditionOperator["NotBetween"] = 1] = "NotBetween";
    /**
     * The condition evaluates to true if the cell value is equal to
     * the value of the operand.
     */
    FormatConditionOperator[FormatConditionOperator["Equal"] = 2] = "Equal";
    /**
     * The condition evaluates to true if the cell value is not equal to
     * the value of the operand.
     */
    FormatConditionOperator[FormatConditionOperator["NotEqual"] = 3] = "NotEqual";
    /**
     * The condition evaluates to true if the cell value is greater than
     * the value of the operand.
     */
    FormatConditionOperator[FormatConditionOperator["Greater"] = 4] = "Greater";
    /**
     * The condition evaluates to true if the cell value is less than
     * the value of the operand.
     */
    FormatConditionOperator[FormatConditionOperator["Less"] = 5] = "Less";
    /**
     * The condition evaluates to true if the cell value is greater than
     * or equal to the value of the operand.
     */
    FormatConditionOperator[FormatConditionOperator["GreaterEqual"] = 6] = "GreaterEqual";
    /**
     * The condition evaluates to true if the cell value is less than
     * or equal to the value of the operand.
     */
    FormatConditionOperator[FormatConditionOperator["LessEqual"] = 7] = "LessEqual";
    return FormatConditionOperator;
})({});
/**
 * @hidden
 */
export let FormatConditionOperator_$type = /*@__PURE__*/ markEnum('FormatConditionOperator', 'Between,0|NotBetween,1|Equal,2|NotEqual,3|Greater,4|Less,5|GreaterEqual,6|LessEqual,7');
