// igniteui-react-excel/esm2015/lib/FtCmo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let FtCmo = /*@__PURE__*/ (() => {
    class FtCmo extends Base {
        constructor(a, ..._rest) {
            super();
            this.a = 0;
            this.s = 0;
            this.u = 0;
            this.w = 0;
            this.x = 0;
            this.y = 0;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.a = c;
                        if (c == 25) {
                            this.u = 16401;
                        }
                        else {
                            this.u = 24593;
                        }
                    }
                    break;
            }
        }
        get c() {
            return (this.u & 4) == 4;
        }
        get d() {
            return (this.u & 128) == 128;
        }
        get e() {
            return (this.u & 1) == 1;
        }
        get f() {
            return (this.u & 8) == 8;
        }
        get g() {
            return (this.u & 4) == 4;
        }
        get h() {
            return (this.u & 512) == 512;
        }
        get i() {
            return (this.u & 4096) == 4096;
        }
        get j() {
            return (this.u & 256) == 256;
        }
        get t() {
            return this.s;
        }
        set t(a) {
            this.s = a;
        }
        get b() {
            return this.a;
        }
    }
    FtCmo.$t = /*@__PURE__*/ markType(FtCmo, 'FtCmo');
    return FtCmo;
})();
