// igniteui-react-excel/esm2015/lib/FtPioGrbit.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let FtPioGrbit = /*@__PURE__*/ (() => {
    class FtPioGrbit extends Base {
        constructor(a) {
            super();
            this.j = 1;
            if (!a) {
                this.j = (this.j & ~1);
            }
        }
        get a() {
            return (this.j & 512) == 512;
        }
        get b() {
            return (this.j & 1) == 1;
        }
        get c() {
            return (this.j & 128) == 128;
        }
        get d() {
            return (this.j & 16) == 16;
        }
        get e() {
            return (this.j & 2) == 2;
        }
        get f() {
            return (this.j & 256) == 256;
        }
        get g() {
            return (this.j & 8) == 8;
        }
        get i() {
            return (this.j & 32) == 32;
        }
        get h() {
            return (this.j & 4) == 4;
        }
    }
    FtPioGrbit.$t = /*@__PURE__*/ markType(FtPioGrbit, 'FtPioGrbit');
    return FtPioGrbit;
})();
