// igniteui-react-excel/esm2015/lib/GeographicMappingArea.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which define the level of granularity for a geographic map chart.
 */
export var GeographicMappingArea = /*@__PURE__*/ (function (GeographicMappingArea) {
    /**
     * Only regions for which series data is available are displayed.
     */
    GeographicMappingArea[GeographicMappingArea["DataOnly"] = 0] = "DataOnly";
    /**
     * A postal region is displayed.
     */
    GeographicMappingArea[GeographicMappingArea["PostalCode"] = 1] = "PostalCode";
    /**
     * A county is displayed.
     */
    GeographicMappingArea[GeographicMappingArea["County"] = 2] = "County";
    /**
     * A state is displayed.
     */
    GeographicMappingArea[GeographicMappingArea["State"] = 3] = "State";
    /**
     * The entire country or region is displayed, with areas for which no
     * series data is available appearing in a lighter color.
     */
    GeographicMappingArea[GeographicMappingArea["CountryOrRegion"] = 4] = "CountryOrRegion";
    /**
     * Multiple countries or regions are displayed, with areas for which no
     * series data is available appearing in a lighter color.
     */
    GeographicMappingArea[GeographicMappingArea["MultipleCountriesOrRegions"] = 5] = "MultipleCountriesOrRegions";
    /**
     * The entire world map is displayed, with areas for which no
     * series data is available appearing in a lighter color.
     */
    GeographicMappingArea[GeographicMappingArea["World"] = 6] = "World";
    return GeographicMappingArea;
})({});
/**
 * @hidden
 */
export let GeographicMappingArea_$type = /*@__PURE__*/ markEnum('GeographicMappingArea', 'DataOnly,0|PostalCode,1|County,2|State,3|CountryOrRegion,4|MultipleCountriesOrRegions,5|World,6');
