// igniteui-react-excel/esm2015/lib/GeographicMapProjection.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which define the projection for a geographic map chart.
 */
export var GeographicMapProjection = /*@__PURE__*/ (function (GeographicMapProjection) {
    /**
     * Albers equal-area conic projection.
     */
    GeographicMapProjection[GeographicMapProjection["Albers"] = 0] = "Albers";
    /**
     * Mercator cylindrical map projection.
     */
    GeographicMapProjection[GeographicMapProjection["Mercator"] = 1] = "Mercator";
    /**
     * Miller cylindrical map projection.
     */
    GeographicMapProjection[GeographicMapProjection["Miller"] = 2] = "Miller";
    /**
     * Robinson flat map projection.
     */
    GeographicMapProjection[GeographicMapProjection["Robinson"] = 3] = "Robinson";
    return GeographicMapProjection;
})({});
/**
 * @hidden
 */
export let GeographicMapProjection_$type = /*@__PURE__*/ markEnum('GeographicMapProjection', 'Albers,0|Mercator,1|Miller,2|Robinson,3');
