// igniteui-react-excel/esm2015/lib/GetCellTextResult.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, markStruct } from "igniteui-react-core";
/**
 * @hidden
 */
export let GetCellTextResult = /*@__PURE__*/ (() => {
    class GetCellTextResult extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.j = null;
            this.e = 0;
            this.a = 0;
            this.h = null;
            this.b = 0;
            this.g = 0;
            this.f = 0;
            this.i = null;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.j = c;
                        this.e = NaN;
                        this.a = 3;
                        this.h = null;
                        this.b = 0;
                        this.g = -1;
                        this.f = 0;
                        this.i = null;
                    }
                    break;
            }
        }
        get c() {
            return this.a == 1;
        }
        get d() {
            return this.a == 2;
        }
    }
    GetCellTextResult.$t = /*@__PURE__*/ markStruct(GetCellTextResult, 'GetCellTextResult');
    return GetCellTextResult;
})();
