// igniteui-react-excel/esm2015/lib/GetXValuesResult.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, Number_$type, markType } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let GetXValuesResult = /*@__PURE__*/ (() => {
    class GetXValuesResult extends Base {
        constructor(a, b) {
            super();
            this.a = 0;
            this.c = null;
            this.a = b;
            this.c = a;
        }
        b(a, b) {
            let c = new List$1(Number_$type, 2, this.c.count);
            a = 1.7976931348623157E+308;
            b = -1.7976931348623157E+308;
            for (let d = 0; d < this.c.count; d++) {
                let e = this.c._inner[d];
                let f = e.c;
                a = Math.min(a, f);
                b = Math.max(b, f);
                c.add(f);
            }
            return {
                ret: c,
                p0: a,
                p1: b
            };
        }
    }
    GetXValuesResult.$t = /*@__PURE__*/ markType(GetXValuesResult, 'GetXValuesResult');
    return GetXValuesResult;
})();
