// igniteui-react-excel/esm2015/lib/GradientType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which determine the type of gradient applied to a
 * [[ChartGradientFill]].
 */
export var GradientType = /*@__PURE__*/ (function (GradientType) {
    /**
     * The gradient extends in a linear direction.
     */
    GradientType[GradientType["Linear"] = 0] = "Linear";
    /**
     * The gradient extends along a radial curve.
     */
    GradientType[GradientType["Radial"] = 1] = "Radial";
    /**
     * The gradient extends along a rectangular path.
     */
    GradientType[GradientType["Rectangular"] = 2] = "Rectangular";
    /**
     * The gradient direction is determined by the shape's path.
     */
    GradientType[GradientType["Path"] = 3] = "Path";
    return GradientType;
})({});
/**
 * @hidden
 */
export let GradientType_$type = /*@__PURE__*/ markEnum('GradientType', 'Linear,0|Radial,1|Rectangular,2|Path,3');
