// igniteui-react-excel/esm2015/lib/GZipStream.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Stream, markType } from "igniteui-react-core";
import { MemoryStream } from "igniteui-react-core";
import { NotImplementedException } from "igniteui-react-core";
import * as Pako from "pako";
/**
 * @hidden
 */
export let GZipStream = /*@__PURE__*/ (() => {
    class GZipStream extends Stream {
        constructor(a, b) {
            super();
            this._o = null;
            this._entryStream = null;
            this._o = a;
            if (b == 0) {
                let c = new Array(a.length);
                a.read(c, 0, c.length);
                this._entryStream = new MemoryStream(2, (Pako.inflate(c)));
            }
            else {
                throw new NotImplementedException(0);
            }
        }
        get _baseStream() {
            return this._o;
        }
        get_canRead() {
            return true;
        }
        get canRead() {
            return this.get_canRead();
        }
        get_canSeek() {
            return true;
        }
        get canSeek() {
            return this.get_canSeek();
        }
        get_canWrite() {
            return false;
        }
        get canWrite() {
            return this.get_canWrite();
        }
        get_length() {
            return this._entryStream.length;
        }
        get length() {
            return this.get_length();
        }
        get_position() {
            return this._entryStream.position;
        }
        set_position(a) {
            this._entryStream.position = a;
        }
        get position() {
            return this.get_position();
        }
        set position(a) {
            this.set_position(a);
        }
        disposeCore(a) {
            this._entryStream.dispose();
        }
        flush() {
            this._entryStream.flush();
        }
        read(a, b, c) {
            return this._entryStream.read(a, b, c);
        }
        seek(a, b) {
            return this._entryStream.seek(a, b);
        }
        setLength(a) {
            this._entryStream.setLength(a);
        }
        write(a, b, c) {
            this._entryStream.write(a, b, c);
        }
    }
    GZipStream.$t = /*@__PURE__*/ markType(GZipStream, 'GZipStream', Stream.$);
    return GZipStream;
})();
