// igniteui-react-excel/esm2015/lib/HorizontalTextAlignment.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various horizontal text alignment types.
 */
export var HorizontalTextAlignment = /*@__PURE__*/ (function (HorizontalTextAlignment) {
    /**
     * Align text in the center of the line.
     */
    HorizontalTextAlignment[HorizontalTextAlignment["Center"] = 0] = "Center";
    /**
     * Distributes the words across the entire line.
     */
    HorizontalTextAlignment[HorizontalTextAlignment["Distributed"] = 1] = "Distributed";
    /**
     * Align text so it is justified across the line.
     */
    HorizontalTextAlignment[HorizontalTextAlignment["Justified"] = 2] = "Justified";
    /**
     * Aligns the text with an adjusted kashida length for Arabic text.
     */
    HorizontalTextAlignment[HorizontalTextAlignment["JustifiedLow"] = 3] = "JustifiedLow";
    /**
     * Align the text to the left of the line.
     */
    HorizontalTextAlignment[HorizontalTextAlignment["Left"] = 4] = "Left";
    /**
     * Align the text to the right of the line.
     */
    HorizontalTextAlignment[HorizontalTextAlignment["Right"] = 5] = "Right";
    /**
     * Distributes Thai text by treating each character as a word.
     */
    HorizontalTextAlignment[HorizontalTextAlignment["ThaiDistributed"] = 6] = "ThaiDistributed";
    return HorizontalTextAlignment;
})({});
/**
 * @hidden
 */
export let HorizontalTextAlignment_$type = /*@__PURE__*/ markEnum('HorizontalTextAlignment', 'Center,0|Distributed,1|Justified,2|JustifiedLow,3|Left,4|Right,5|ThaiDistributed,6');
