// igniteui-react-excel/esm2015/lib/IEG_LineDashProperties_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Type, Base, markType, toNullable } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
import { CT_DashStop } from "./CT_DashStop";
import { ST_PresetLineDashVal_$type } from "./ST_PresetLineDashVal";
/**
 * @hidden
 */
export let IEG_LineDashProperties_$type = /*@__PURE__*/ new Type(null, 'IEG_LineDashProperties');
/**
 * @hidden
 */
export let CT_DashStopList = /*@__PURE__*/ (() => {
    class CT_DashStopList extends Base {
        constructor() {
            super(...arguments);
            this.a = new List$1(CT_DashStop.$, 0);
        }
        get b() {
            return this.a;
        }
        _accept$2$i($tArg, $tResult, a, b) {
            return a._visitCT_DashStopList$i(this, b);
        }
    }
    CT_DashStopList.$t = /*@__PURE__*/ markType(CT_DashStopList, 'CT_DashStopList', Base.$, [IEG_LineDashProperties_$type]);
    return CT_DashStopList;
})();
/**
 * @hidden
 */
export let CT_PresetLineDashProperties = /*@__PURE__*/ (() => {
    class CT_PresetLineDashProperties extends Base {
        constructor() {
            super(...arguments);
            this.a = toNullable(ST_PresetLineDashVal_$type, null);
        }
        _accept$2$i($tArg, $tResult, a, b) {
            return a._visitCT_PresetLineDashProperties$i(this, b);
        }
    }
    CT_PresetLineDashProperties.$t = /*@__PURE__*/ markType(CT_PresetLineDashProperties, 'CT_PresetLineDashProperties', Base.$, [IEG_LineDashProperties_$type]);
    return CT_PresetLineDashProperties;
})();
/**
 * @hidden
 */
export let IEG_LineDashPropertiesVisitor$2_$type = /*@__PURE__*/ new Type(null, 'IEG_LineDashPropertiesVisitor$2');
