// igniteui-react-excel/esm2015/lib/IEG_LineJoinProperties_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Type, Base, markType, toNullable } from "igniteui-react-core";
import { ST_PositivePercentage } from "./ST_PositivePercentage";
/**
 * @hidden
 */
export let IEG_LineJoinProperties_$type = /*@__PURE__*/ new Type(null, 'IEG_LineJoinProperties');
/**
 * @hidden
 */
export let CT_LineJoinBevel = /*@__PURE__*/ (() => {
    class CT_LineJoinBevel extends Base {
        _accept$2$i($tArg, $tResult, a, b) {
            return a._visitCT_LineJoinBevel$i(this, b);
        }
    }
    CT_LineJoinBevel.$t = /*@__PURE__*/ markType(CT_LineJoinBevel, 'CT_LineJoinBevel', Base.$, [IEG_LineJoinProperties_$type]);
    return CT_LineJoinBevel;
})();
/**
 * @hidden
 */
export let CT_LineJoinMiterProperties = /*@__PURE__*/ (() => {
    class CT_LineJoinMiterProperties extends Base {
        constructor() {
            super(...arguments);
            this.a = toNullable(ST_PositivePercentage.$, null);
        }
        _accept$2$i($tArg, $tResult, a, b) {
            return a._visitCT_LineJoinMiterProperties$i(this, b);
        }
    }
    CT_LineJoinMiterProperties.$t = /*@__PURE__*/ markType(CT_LineJoinMiterProperties, 'CT_LineJoinMiterProperties', Base.$, [IEG_LineJoinProperties_$type]);
    return CT_LineJoinMiterProperties;
})();
/**
 * @hidden
 */
export let CT_LineJoinRound = /*@__PURE__*/ (() => {
    class CT_LineJoinRound extends Base {
        _accept$2$i($tArg, $tResult, a, b) {
            return a._visitCT_LineJoinRound$i(this, b);
        }
    }
    CT_LineJoinRound.$t = /*@__PURE__*/ markType(CT_LineJoinRound, 'CT_LineJoinRound', Base.$, [IEG_LineJoinProperties_$type]);
    return CT_LineJoinRound;
})();
/**
 * @hidden
 */
export let IEG_LineJoinPropertiesVisitor$2_$type = /*@__PURE__*/ new Type(null, 'IEG_LineJoinPropertiesVisitor$2');
