// igniteui-react-excel/esm2015/lib/IEG_Geometry_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Type, Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let IEG_Geometry_$type = /*@__PURE__*/ new Type(null, 'IEG_Geometry');
/**
 * @hidden
 */
export let CT_CustomGeometry2D = /*@__PURE__*/ (() => {
    class CT_CustomGeometry2D extends Base {
        constructor() {
            super(...arguments);
            this.c = null;
            this.d = null;
            this.a = null;
            this.b = null;
            this.e = null;
            this.f = null;
            this.g = null;
        }
        _accept$2$i($tArg, $tResult, a, b) {
            return a._visitCT_CustomGeometry2D$i(this, b);
        }
    }
    CT_CustomGeometry2D.$t = /*@__PURE__*/ markType(CT_CustomGeometry2D, 'CT_CustomGeometry2D', Base.$, [IEG_Geometry_$type]);
    return CT_CustomGeometry2D;
})();
/**
 * @hidden
 */
export let CT_PresetGeometry2D = /*@__PURE__*/ (() => {
    class CT_PresetGeometry2D extends Base {
        constructor() {
            super(...arguments);
            this.a = null;
            this.b = 0;
        }
        _accept$2$i($tArg, $tResult, a, b) {
            return a._visitCT_PresetGeometry2D$i(this, b);
        }
    }
    CT_PresetGeometry2D.$t = /*@__PURE__*/ markType(CT_PresetGeometry2D, 'CT_PresetGeometry2D', Base.$, [IEG_Geometry_$type]);
    return CT_PresetGeometry2D;
})();
/**
 * @hidden
 */
export let IEG_GeometryVisitor$2_$type = /*@__PURE__*/ new Type(null, 'IEG_GeometryVisitor$2');
