// igniteui-react-excel/esm2015/lib/ImageData.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { DocCoreUtils } from "./DocCoreUtils";
/**
 * @hidden
 */
export let ImageData = /*@__PURE__*/ (() => {
    class ImageData extends Base {
        constructor(a, ..._rest) {
            super();
            this.a = null;
            this.c = null;
            this.e = null;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0:
                    {
                        let c = _rest[0];
                        let d = _rest[1];
                        this.a = d;
                        this.c = c;
                    }
                    break;
                case 1:
                    {
                        let c = _rest[0];
                        let d = _rest[1];
                        this.c = c;
                        this.a = DocCoreUtils.c(d);
                    }
                    break;
            }
        }
        get b() {
            return this.a;
        }
        get d() {
            return this.c;
        }
        f(a) {
            a.write(this.a, 0, this.a.length);
        }
    }
    ImageData.$t = /*@__PURE__*/ markType(ImageData, 'ImageData');
    return ImageData;
})();
