// igniteui-react-excel/esm2015/lib/InformationBase_PropertySectionDefinition.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { Guid } from "igniteui-react-core";
/**
 * @hidden
 */
export let InformationBase_PropertySectionDefinition = /*@__PURE__*/ (() => {
    class InformationBase_PropertySectionDefinition extends Base {
        constructor() {
            super(...arguments);
            this.b = new Guid();
            this.e = 0;
            this.c = 0;
            this.d = 0;
            this.a = null;
        }
    }
    InformationBase_PropertySectionDefinition.$t = /*@__PURE__*/ markType(InformationBase_PropertySectionDefinition, 'InformationBase_PropertySectionDefinition');
    return InformationBase_PropertySectionDefinition;
})();
