// igniteui-react-excel/esm2015/lib/InPlaceTransform_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let InPlaceTransform = /*@__PURE__*/ (() => {
    class InPlaceTransform extends Base {
        constructor() {
            super(...arguments);
            this.e = 0;
        }
        static get b() {
            if (InPlaceTransform.a == null) {
                InPlaceTransform.a = new InPlaceTransform_NoTransformImpl();
            }
            return InPlaceTransform.a;
        }
        get c() {
            return this.e > 0;
        }
        f() {
        }
        get_d() {
            return false;
        }
        get d() {
            return this.get_d();
        }
        g(a, b) {
            if (this.c) {
                return;
            }
            this.h(a, b);
        }
        h(a, b) {
        }
        i() {
            if (this.e == 0) {
                return;
            }
            this.e--;
        }
        j(a) {
        }
        k() {
            this.e++;
        }
        l(a, b, c, d) {
            if (this.c) {
                return;
            }
            this.m(a, b, c, d);
        }
    }
    InPlaceTransform.$t = /*@__PURE__*/ markType(InPlaceTransform, 'InPlaceTransform');
    InPlaceTransform.a = null;
    return InPlaceTransform;
})();
/**
 * @hidden
 */
export let InPlaceTransform_NoTransformImpl = /*@__PURE__*/ (() => {
    class InPlaceTransform_NoTransformImpl extends InPlaceTransform {
        m(a, b, c, d) {
        }
    }
    InPlaceTransform_NoTransformImpl.$t = /*@__PURE__*/ markType(InPlaceTransform_NoTransformImpl, 'InPlaceTransform_NoTransformImpl', InPlaceTransform.$);
    return InPlaceTransform_NoTransformImpl;
})();
