// igniteui-react-excel/esm2015/lib/InvalidPackage.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { IPackage_$type } from "./IPackage";
import { InvalidOperationException } from "igniteui-react-core";
/**
 * @hidden
 */
export let InvalidPackage = /*@__PURE__*/ (() => {
    class InvalidPackage extends Base {
        createPart(a, b) {
            throw new InvalidOperationException(0);
        }
        createRelationship(a, b, c, d) {
            throw new InvalidOperationException(0);
        }
        dispose() {
            throw new InvalidOperationException(0);
        }
        getPart(a) {
            throw new InvalidOperationException(0);
        }
        getParts() {
            throw new InvalidOperationException(0);
        }
        getRelationship(a) {
            throw new InvalidOperationException(0);
        }
        getRelationships() {
            throw new InvalidOperationException(0);
        }
        partExists(a) {
            throw new InvalidOperationException(0);
        }
    }
    InvalidPackage.$t = /*@__PURE__*/ markType(InvalidPackage, 'InvalidPackage', Base.$, [IPackage_$type]);
    return InvalidPackage;
})();
