// igniteui-react-excel/esm2015/lib/LegacyShapeTypeData.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let LegacyShapeTypeData = /*@__PURE__*/ (() => {
    class LegacyShapeTypeData extends Base {
        constructor() {
            super(...arguments);
            this.f = null;
            this.e = null;
            this.d = toNullable(Number_$type, null);
            this.g = null;
            this.c = null;
            this.b = null;
            this.a = null;
        }
    }
    LegacyShapeTypeData.$t = /*@__PURE__*/ markType(LegacyShapeTypeData, 'LegacyShapeTypeData');
    return LegacyShapeTypeData;
})();
