// igniteui-react-excel/esm2015/lib/LegacyShapePathData.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Boolean_$type, markType } from "igniteui-react-core";
import { ST_ConnectType_$type } from "./ST_ConnectType";
/**
 * @hidden
 */
export let LegacyShapePathData = /*@__PURE__*/ (() => {
    class LegacyShapePathData extends Base {
        constructor() {
            super(...arguments);
            this.e = toNullable(Boolean_$type, null);
            this.b = toNullable(Boolean_$type, null);
            this.f = toNullable(Boolean_$type, null);
            this.c = toNullable(Boolean_$type, null);
            this.d = toNullable(Boolean_$type, null);
            this.a = toNullable(ST_ConnectType_$type, null);
        }
    }
    LegacyShapePathData.$t = /*@__PURE__*/ markType(LegacyShapePathData, 'LegacyShapePathData');
    return LegacyShapePathData;
})();
