// igniteui-react-excel/esm2015/lib/LegendPosition.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which determine the placement of the chart's
 * [[Legend]].
 */
export var LegendPosition = /*@__PURE__*/ (function (LegendPosition) {
    /**
     * The legend appears at the bottom of the chart.
     */
    LegendPosition[LegendPosition["Bottom"] = 0] = "Bottom";
    /**
     * The legend appears near the left edge of the chart.
     */
    LegendPosition[LegendPosition["Left"] = 1] = "Left";
    /**
     * The legend appears near the right edge of the chart.
     */
    LegendPosition[LegendPosition["Right"] = 2] = "Right";
    /**
     * The legend appears at the top of the chart.
     */
    LegendPosition[LegendPosition["Top"] = 3] = "Top";
    /**
     * The legend appears near the top right corner of the chart.
     */
    LegendPosition[LegendPosition["TopRight"] = 4] = "TopRight";
    /**
     * The legend's size and location is determined by the
     * [[Legend.left]], [[Legend.top]],
     * [[Legend.width]], and [[Legend.height]]
     * properties.
     */
    LegendPosition[LegendPosition["Custom"] = 5] = "Custom";
    /**
     * The legend position is determined automatically by Excel.
     */
    LegendPosition[LegendPosition["Default"] = 6] = "Default";
    /**
     * The legend appears near the top left corner of the chart.
     * Supported only for Histogram, Pareto, Treemap, Sunburst, Box and Whisker, and Waterfall chart types.
     */
    LegendPosition[LegendPosition["TopLeft"] = 7] = "TopLeft";
    /**
     * The legend appears near the bottom left corner of the chart.
     * Supported only for Histogram, Pareto, Treemap, Sunburst, Box and Whisker, and Waterfall chart types.
     */
    LegendPosition[LegendPosition["BottomLeft"] = 8] = "BottomLeft";
    /**
     * The legend appears near the bottom right corner of the chart.
     * Supported only for Histogram, Pareto, Treemap, Sunburst, Box and Whisker, and Waterfall chart types.
     */
    LegendPosition[LegendPosition["BottomRight"] = 9] = "BottomRight";
    return LegendPosition;
})({});
/**
 * @hidden
 */
export let LegendPosition_$type = /*@__PURE__*/ markEnum('LegendPosition', 'Bottom,0|Left,1|Right,2|Top,3|TopRight,4|Custom,5|Default,6|TopLeft,7|BottomLeft,8|BottomRight,9');
