// igniteui-react-excel/esm2015/lib/LfuCache_FrequencyNode$2.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { LinkedListNode } from "./LinkedListNode";
import { LinkedList } from "./LinkedList";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let LfuCache_FrequencyNode$2 = /*@__PURE__*/ (() => {
    class LfuCache_FrequencyNode$2 extends LinkedListNode {
        constructor($tKey, $tValue) {
            super();
            this.$tKey = null;
            this.$tValue = null;
            this.o = 0;
            this.n = null;
            this.$tKey = $tKey;
            this.$tValue = $tValue;
            this.$type = this.$type.specialize(this.$tKey, this.$tValue);
            this.n = new LinkedList();
        }
    }
    LfuCache_FrequencyNode$2.$t = /*@__PURE__*/ markType(LfuCache_FrequencyNode$2, 'LfuCache_FrequencyNode$2', LinkedListNode.$);
    return LfuCache_FrequencyNode$2;
})();
