// igniteui-react-excel/esm2015/lib/LineStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Determines the style of a line, i.e., solid, dashed, dotted, etc.
 * @see [[ChartLine.lineStyle]]
 */
export var LineStyle = /*@__PURE__*/ (function (LineStyle) {
    /**
     * Solid line.
     */
    LineStyle[LineStyle["Solid"] = 0] = "Solid";
    /**
     * Dashed line.
     */
    LineStyle[LineStyle["Dash"] = 1] = "Dash";
    /**
     * Alternating dashes and dots.
     */
    LineStyle[LineStyle["DashDot"] = 2] = "DashDot";
    /**
     * Dotted line.
     */
    LineStyle[LineStyle["Dot"] = 3] = "Dot";
    /**
     * Large dashed line.
     */
    LineStyle[LineStyle["LargeDash"] = 4] = "LargeDash";
    /**
     * Large alternating dashes and dots.
     */
    LineStyle[LineStyle["LargeDashDot"] = 5] = "LargeDashDot";
    /**
     * Large alternating dashes and pairs of dots.
     */
    LineStyle[LineStyle["LargeDashDotDot"] = 6] = "LargeDashDotDot";
    /**
     * No line.
     */
    LineStyle[LineStyle["None"] = 7] = "None";
    /**
     * System-defined dashed line.
     */
    LineStyle[LineStyle["SysDash"] = 8] = "SysDash";
    /**
     * System-defined alternating dashes and dots.
     */
    LineStyle[LineStyle["SysDashDot"] = 9] = "SysDashDot";
    /**
     * System-defined alternating dashes and pairs of dots.
     */
    LineStyle[LineStyle["SysDashDotDot"] = 10] = "SysDashDotDot";
    /**
     * System-defined dotted line.
     */
    LineStyle[LineStyle["SysDot"] = 11] = "SysDot";
    return LineStyle;
})({});
/**
 * @hidden
 */
export let LineStyle_$type = /*@__PURE__*/ markEnum('LineStyle', 'Solid,0|Dash,1|DashDot,2|Dot,3|LargeDash,4|LargeDashDot,5|LargeDashDotDot,6|None,7|SysDash,8|SysDashDot,9|SysDashDotDot,10|SysDot,11');
