// igniteui-react-excel/esm2015/lib/LoadOnDemandTree_ChangeHistoryItemRemove$1.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { LoadOnDemandTree_ChangeHistoryItem$1 } from "./LoadOnDemandTree_ChangeHistoryItem$1";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let LoadOnDemandTree_ChangeHistoryItemRemove$1 = /*@__PURE__*/ (() => {
    class LoadOnDemandTree_ChangeHistoryItemRemove$1 extends LoadOnDemandTree_ChangeHistoryItem$1 {
        constructor($t, a, b) {
            super($t, a, b);
            this.$t = null;
            this.$t = $t;
            this.$type = this.$type.specialize(this.$t);
        }
        get_a() {
            return false;
        }
        get a() {
            return this.get_a();
        }
        g(a) {
            this.b = a;
        }
    }
    LoadOnDemandTree_ChangeHistoryItemRemove$1.$t = /*@__PURE__*/ markType(LoadOnDemandTree_ChangeHistoryItemRemove$1, 'LoadOnDemandTree_ChangeHistoryItemRemove$1', /*@__PURE__*/ LoadOnDemandTree_ChangeHistoryItem$1.$.specialize(0));
    return LoadOnDemandTree_ChangeHistoryItemRemove$1;
})();
