// igniteui-react-excel/esm2015/lib/LoadOnDemandTree_ChangeHistoryItemAdd$1.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { LoadOnDemandTree_ChangeHistoryItem$1 } from "./LoadOnDemandTree_ChangeHistoryItem$1";
import { markType } from "igniteui-react-core";
import { arrayCopy1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let LoadOnDemandTree_ChangeHistoryItemAdd$1 = /*@__PURE__*/ (() => {
    class LoadOnDemandTree_ChangeHistoryItemAdd$1 extends LoadOnDemandTree_ChangeHistoryItem$1 {
        constructor($t, a, b) {
            super($t, a, b);
            this.$t = null;
            this.h = null;
            this.$t = $t;
            this.$type = this.$type.specialize(this.$t);
        }
        get j() {
            return this.h != null;
        }
        get_a() {
            return true;
        }
        get a() {
            return this.get_a();
        }
        get i() {
            if (this.h == null) {
                this.h = new Array(this.c);
            }
            return this.h;
        }
        g(a) {
            this.b = a;
            if (this.h != null) {
                let b = this.h;
                this.h = new Array(this.b);
                arrayCopy1(b, 0, this.h, 0, this.b);
            }
        }
    }
    LoadOnDemandTree_ChangeHistoryItemAdd$1.$t = /*@__PURE__*/ markType(LoadOnDemandTree_ChangeHistoryItemAdd$1, 'LoadOnDemandTree_ChangeHistoryItemAdd$1', /*@__PURE__*/ LoadOnDemandTree_ChangeHistoryItem$1.$.specialize(0));
    return LoadOnDemandTree_ChangeHistoryItemAdd$1;
})();
