// igniteui-react-excel/esm2015/lib/LoadOnDemandTree_ChangeHistoryItem$1.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let LoadOnDemandTree_ChangeHistoryItem$1 = /*@__PURE__*/ (() => {
    class LoadOnDemandTree_ChangeHistoryItem$1 extends Base {
        constructor($t, a, b) {
            super();
            this.$t = null;
            this.f = 0;
            this.b = 0;
            this.$t = $t;
            this.$type = this.$type.specialize(this.$t);
            this.f = a;
            this.b = b;
        }
        d(a, b, c, d) {
            return 0;
        }
        e(a, b, c) {
            return 0;
        }
        get c() {
            return this.b;
        }
    }
    LoadOnDemandTree_ChangeHistoryItem$1.$t = /*@__PURE__*/ markType(LoadOnDemandTree_ChangeHistoryItem$1, 'LoadOnDemandTree_ChangeHistoryItem$1');
    return LoadOnDemandTree_ChangeHistoryItem$1;
})();
