// igniteui-react-excel/esm2015/lib/LruCache_LruCacheNode$2.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { LinkedListNode } from "./LinkedListNode";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let LruCache_LruCacheNode$2 = /*@__PURE__*/ (() => {
    class LruCache_LruCacheNode$2 extends LinkedListNode {
        constructor($tKey, $tValue) {
            super();
            this.$tKey = null;
            this.$tValue = null;
            this.n = null;
            this.o = null;
            this.$tKey = $tKey;
            this.$tValue = $tValue;
            this.$type = this.$type.specialize(this.$tKey, this.$tValue);
        }
    }
    LruCache_LruCacheNode$2.$t = /*@__PURE__*/ markType(LruCache_LruCacheNode$2, 'LruCache_LruCacheNode$2', LinkedListNode.$);
    return LruCache_LruCacheNode$2;
})();
