// igniteui-react-excel/esm2015/lib/MarkerStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which determine the style of the markers displayed by line charts.
 */
export var MarkerStyle = /*@__PURE__*/ (function (MarkerStyle) {
    /**
     * The marker style is determined automatically.
     */
    MarkerStyle[MarkerStyle["Automatic"] = -4105] = "Automatic";
    /**
     * The marker is circular.
     */
    MarkerStyle[MarkerStyle["Circle"] = 8] = "Circle";
    /**
     * The marker is a dash.
     */
    MarkerStyle[MarkerStyle["Dash"] = -4115] = "Dash";
    /**
     * The marker is diamond-shaped.
     */
    MarkerStyle[MarkerStyle["Diamond"] = 2] = "Diamond";
    /**
     * The marker is a dot or bullet.
     */
    MarkerStyle[MarkerStyle["Dot"] = -4118] = "Dot";
    /**
     * No marker is displayed.
     */
    MarkerStyle[MarkerStyle["None"] = -4142] = "None";
    /**
     * The marker appears as a picture.
     */
    MarkerStyle[MarkerStyle["Picture"] = -4147] = "Picture";
    /**
     * The marker appears as a plus sign (+).
     */
    MarkerStyle[MarkerStyle["Plus"] = 9] = "Plus";
    /**
     * The marker appears as a square.
     */
    MarkerStyle[MarkerStyle["Square"] = 1] = "Square";
    /**
     * The marker appears as a star.
     */
    MarkerStyle[MarkerStyle["Star"] = 5] = "Star";
    /**
     * The marker appears as a triangle.
     */
    MarkerStyle[MarkerStyle["Triangle"] = 3] = "Triangle";
    /**
     * The marker appears as an 'X'
     */
    MarkerStyle[MarkerStyle["X"] = -4168] = "X";
    return MarkerStyle;
})({});
/**
 * @hidden
 */
export let MarkerStyle_$type = /*@__PURE__*/ markEnum('MarkerStyle', 'Automatic,-4105|Circle,8|Dash,-4115|Diamond,2|Dot,-4118|None,-4142|Picture,-4147|Plus,9|Square,1|Star,5|Triangle,3|X,-4168');
