// igniteui-react-excel/esm2015/lib/ObjectPool$2.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ObjectPool$2 = /*@__PURE__*/ (() => {
    class ObjectPool$2 extends Base {
        constructor($t, $tArg, a, b) {
            super();
            this.$t = null;
            this.$tArg = null;
            this.a = new Array(1024);
            this.d = null;
            this.c = null;
            this.e = 0;
            this.$t = $t;
            this.$tArg = $tArg;
            this.$type = this.$type.specialize(this.$t, this.$tArg);
            this.d = a;
            this.c = b;
        }
        b(a) {
            if (this.e == 0) {
                return this.d(a);
            }
            let b = this.a[--this.e];
            this.c(b, a);
            return b;
        }
        g(a) {
            if (this.e == 1024) {
                return;
            }
            this.a[this.e++] = a;
        }
    }
    ObjectPool$2.$t = /*@__PURE__*/ markType(ObjectPool$2, 'ObjectPool$2');
    return ObjectPool$2;
})();
