// igniteui-react-excel/esm2015/lib/ObjectModelTablePartManager_CustomFilterContext.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { CustomFilter } from "./CustomFilter";
/**
 * @hidden
 */
export let ObjectModelTablePartManager_CustomFilterContext = /*@__PURE__*/ (() => {
    class ObjectModelTablePartManager_CustomFilterContext extends Base {
        constructor(a) {
            super();
            this.a = 0;
            this.c = null;
            this.d = null;
            this.a = a;
        }
        e(a) {
            if (this.c == null) {
                this.c = a;
                return;
            }
            if (this.d == null) {
                this.d = a;
                return;
            }
        }
        b() {
            if (this.c == null && this.d == null) {
                return null;
            }
            return new CustomFilter(1, null, this.c, this.d, this.a);
        }
    }
    ObjectModelTablePartManager_CustomFilterContext.$t = /*@__PURE__*/ markType(ObjectModelTablePartManager_CustomFilterContext, 'ObjectModelTablePartManager_CustomFilterContext');
    return ObjectModelTablePartManager_CustomFilterContext;
})();
