// igniteui-react-excel/esm2015/lib/OverrideContentTypePart.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { IPackagePart_$type } from "./IPackagePart";
/**
 * @hidden
 */
export let OverrideContentTypePart = /*@__PURE__*/ (() => {
    class OverrideContentTypePart extends Base {
        constructor(a, b) {
            super();
            this.b = null;
            this.a = null;
            this.a = a;
            this.b = b != null ? b : a.contentType;
        }
        get $package() {
            return this.a.$package;
        }
        get uri() {
            return this.a.uri;
        }
        get contentType() {
            return this.b;
        }
        getStream(a, b) {
            return this.a.getStream(a, b);
        }
        createRelationship(a, b, c, d) {
            return this.a.createRelationship(a, b, c, d);
        }
        getRelationships() {
            return this.a.getRelationships();
        }
        getRelationship(a) {
            return this.a.getRelationship(a);
        }
    }
    OverrideContentTypePart.$t = /*@__PURE__*/ markType(OverrideContentTypePart, 'OverrideContentTypePart', Base.$, [IPackagePart_$type]);
    return OverrideContentTypePart;
})();
