// igniteui-react-excel/esm2015/lib/PackagePartExposedStream.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Stream, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let PackagePartExposedStream = /*@__PURE__*/ (() => {
    class PackagePartExposedStream extends Stream {
        constructor(a) {
            super();
            this._o = null;
            this._o = a;
        }
        get_canRead() {
            return this._o.canRead;
        }
        get canRead() {
            return this.get_canRead();
        }
        get_canSeek() {
            return this._o.canSeek;
        }
        get canSeek() {
            return this.get_canSeek();
        }
        get_canWrite() {
            return this._o.canWrite;
        }
        get canWrite() {
            return this.get_canWrite();
        }
        flush() {
            this._o.flush();
        }
        get_length() {
            return this._o.length;
        }
        get length() {
            return this.get_length();
        }
        get_position() {
            return this._o.position;
        }
        set_position(a) {
            this._o.position = a;
        }
        get position() {
            return this.get_position();
        }
        set position(a) {
            this.set_position(a);
        }
        read(a, b, c) {
            return this._o.read(a, b, c);
        }
        seek(a, b) {
            return this._o.seek(a, b);
        }
        setLength(a) {
            this._o.setLength(a);
        }
        write(a, b, c) {
            this._o.write(a, b, c);
        }
    }
    PackagePartExposedStream.$t = /*@__PURE__*/ markType(PackagePartExposedStream, 'PackagePartExposedStream', Stream.$);
    return PackagePartExposedStream;
})();
