// igniteui-react-excel/esm2015/lib/PivotTablePartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let PivotTablePartManager = /*@__PURE__*/ (() => {
    class PivotTablePartManager extends XmlPartManagerBase {
        get_m() {
            return PivotTablePartManager.a4;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/pivotTables/pivotTable.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return PivotTablePartManager.a6;
                case 1: return PivotTablePartManager.a5;
                default: return PivotTablePartManager.a6;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    PivotTablePartManager.$t = /*@__PURE__*/ markType(PivotTablePartManager, 'PivotTablePartManager', XmlPartManagerBase.$);
    PivotTablePartManager.a4 = "application/vnd.openxmlformats-officedocument.spreadsheetml.pivotTable+xml";
    PivotTablePartManager.a6 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/pivotTable";
    PivotTablePartManager.a5 = "http://purl.oclc.org/ooxml/officeDocument/relationships/pivotTable";
    return PivotTablePartManager;
})();
