// igniteui-react-excel/esm2015/lib/PivotCacheDefinitionPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let PivotCacheDefinitionPartManager = /*@__PURE__*/ (() => {
    class PivotCacheDefinitionPartManager extends XmlPartManagerBase {
        get_m() {
            return PivotCacheDefinitionPartManager.a4;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/pivotCache/pivotCacheDefinition.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return PivotCacheDefinitionPartManager.a6;
                case 1: return PivotCacheDefinitionPartManager.a5;
                default: return PivotCacheDefinitionPartManager.a6;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    PivotCacheDefinitionPartManager.$t = /*@__PURE__*/ markType(PivotCacheDefinitionPartManager, 'PivotCacheDefinitionPartManager', XmlPartManagerBase.$);
    PivotCacheDefinitionPartManager.a4 = "application/vnd.openxmlformats-officedocument.spreadsheetml.pivotCacheDefinition+xml";
    PivotCacheDefinitionPartManager.a6 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/pivotCacheDefinition";
    PivotCacheDefinitionPartManager.a5 = "http://purl.oclc.org/ooxml/officeDocument/relationships/pivotCacheDefinition";
    return PivotCacheDefinitionPartManager;
})();
