// igniteui-react-excel/esm2015/lib/PngPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ImagePartManagerBase } from "./ImagePartManagerBase";
import { ImageFormat } from "./ImageFormat";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let PngPartManager = /*@__PURE__*/ (() => {
    class PngPartManager extends ImagePartManagerBase {
        constructor() {
            super();
        }
        get_s() {
            return "/xl/media/image.png";
        }
        get s() {
            return this.get_s();
        }
        get_t() {
            return "/ppt/media/image.png";
        }
        get t() {
            return this.get_t();
        }
        get_u() {
            return "/word/media/image.png";
        }
        get u() {
            return this.get_u();
        }
        get_m() {
            return "image/png";
        }
        get m() {
            return this.get_m();
        }
        r(a) {
            return ImageFormat.g;
        }
    }
    PngPartManager.$t = /*@__PURE__*/ markType(PngPartManager, 'PngPartManager', ImagePartManagerBase.$);
    PngPartManager.y = /*@__PURE__*/ new PngPartManager();
    return PngPartManager;
})();
