// igniteui-react-excel/esm2015/lib/PrinterSettingsPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { Base, typeCast, markType } from "igniteui-react-core";
import { DocCoreUtils } from "./DocCoreUtils";
import { BinaryData } from "./BinaryData";
import { WorkItem } from "./WorkItem";
import { InvalidOperationException } from "igniteui-react-core";
/**
 * @hidden
 */
export let PrinterSettingsPartManager = /*@__PURE__*/ (() => {
    class PrinterSettingsPartManager extends XmlPartManagerBase {
        n(a) {
            switch (a) {
                case 0: return "/xl" + "/printerSettings/printerSettings.bin";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return PrinterSettingsPartManager.a7;
                case 1: return PrinterSettingsPartManager.a6;
                default: return PrinterSettingsPartManager.a7;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
        a() {
            return WorkItem.g(Base.$, new BinaryData(DocCoreUtils.c(this.l)));
        }
        b(a) {
            let b = typeCast(BinaryData.$, a);
            if (b == null) {
                return null;
            }
            this.l.write(b.a, 0, b.a.length);
            return null;
        }
        u(a) {
            throw new InvalidOperationException(0);
        }
    }
    PrinterSettingsPartManager.$t = /*@__PURE__*/ markType(PrinterSettingsPartManager, 'PrinterSettingsPartManager', XmlPartManagerBase.$);
    PrinterSettingsPartManager.a4 = "application/vnd.openxmlformats-officedocument.spreadsheetml.printerSettings";
    PrinterSettingsPartManager.a5 = "application/vnd.openxmlformats-officedocument.wordprocessingml.printerSettings";
    PrinterSettingsPartManager.a7 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/printerSettings";
    PrinterSettingsPartManager.a6 = "http://purl.oclc.org/ooxml/officeDocument/relationships/printerSettings";
    return PrinterSettingsPartManager;
})();
