// igniteui-react-excel/esm2015/lib/ProtectionValidationInfo_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType, toNullable, Number_$type } from "igniteui-react-core";
/**
 * @hidden
 */
export let ProtectionValidationInfo = /*@__PURE__*/ (() => {
    class ProtectionValidationInfo extends Base {
        static c(a) {
            return new ProtectionValidationInfo_XorObfuscation(a);
        }
        static b(a, b, c, d) {
            return new ProtectionValidationInfo_Strict(a, b, c, d);
        }
        static a(a, b, c, d) {
            c = a = null;
            b = null;
            d = 0;
            return {
                ret: null,
                p1: b,
                p2: c,
                p3: d
            };
        }
        e() {
            let a = this.f();
            if (a.hasValue && a.value != 0) {
                return true;
            }
            let b;
            let c, d;
            let e;
            if (((() => { let f = this.d(b, c, d, e); b = f.p0; c = f.p1; d = f.p2; e = f.p3; return f.ret; })())) {
                return true;
            }
            return false;
        }
    }
    ProtectionValidationInfo.$t = /*@__PURE__*/ markType(ProtectionValidationInfo, 'ProtectionValidationInfo');
    return ProtectionValidationInfo;
})();
/**
 * @hidden
 */
export let ProtectionValidationInfo_XorObfuscation = /*@__PURE__*/ (() => {
    class ProtectionValidationInfo_XorObfuscation extends ProtectionValidationInfo {
        constructor(a) {
            super();
            this.g = 0;
            this.g = a;
        }
        d(a, b, c, d) {
            a = null;
            b = c = null;
            d = toNullable(Number_$type, null);
            return {
                ret: false,
                p0: a,
                p1: b,
                p2: c,
                p3: d
            };
        }
        f() {
            return toNullable(Number_$type, this.g);
        }
    }
    ProtectionValidationInfo_XorObfuscation.$t = /*@__PURE__*/ markType(ProtectionValidationInfo_XorObfuscation, 'ProtectionValidationInfo_XorObfuscation', ProtectionValidationInfo.$);
    return ProtectionValidationInfo_XorObfuscation;
})();
/**
 * @hidden
 */
export let ProtectionValidationInfo_Strict = /*@__PURE__*/ (() => {
    class ProtectionValidationInfo_Strict extends ProtectionValidationInfo {
        constructor(a, b, c, d) {
            super();
            this.k = null;
            this.g = null;
            this.h = null;
            this.m = 0;
            this.k = a;
            this.g = b;
            this.h = c;
            this.m = d;
        }
        d(a, b, c, d) {
            a = this.k;
            b = this.g;
            c = this.h;
            d = toNullable(Number_$type, this.m);
            return {
                ret: true,
                p0: a,
                p1: b,
                p2: c,
                p3: d
            };
        }
        f() {
            return toNullable(Number_$type, null);
        }
    }
    ProtectionValidationInfo_Strict.$t = /*@__PURE__*/ markType(ProtectionValidationInfo_Strict, 'ProtectionValidationInfo_Strict', ProtectionValidationInfo.$);
    return ProtectionValidationInfo_Strict;
})();
