// igniteui-react-excel/esm2015/lib/RegionLookupTable_RangeHelper$1.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, runOn, markType } from "igniteui-react-core";
import { HashSet$1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let RegionLookupTable_RangeHelper$1 = /*@__PURE__*/ (() => {
    class RegionLookupTable_RangeHelper$1 extends Base {
        constructor($t) {
            super();
            this.$t = null;
            this.g = 0;
            this.h = 0;
            this.i = 0;
            this.j = 0;
            this.e = null;
            this.d = null;
            this.c = null;
            this.$t = $t;
            this.$type = this.$type.specialize(this.$t);
            this.e = runOn(this, this.b);
            this.d = runOn(this, this.a);
            this.c = new HashSet$1(this.$t, 0);
        }
        k() {
            this.c.clear();
        }
        f(a, b, c, d, e) {
            this.i = b;
            this.g = c;
            this.j = d;
            this.h = e;
            return a ? this.e : this.d;
        }
        a(a) {
            return a.intersectsWithColumns(this.g, this.h) && this.c.add_1(a);
        }
        b(a) {
            return a.intersectsWithRows(this.i, this.j) && this.c.add_1(a);
        }
    }
    RegionLookupTable_RangeHelper$1.$t = /*@__PURE__*/ markType(RegionLookupTable_RangeHelper$1, 'RegionLookupTable_RangeHelper$1');
    return RegionLookupTable_RangeHelper$1;
})();
