// igniteui-react-excel/esm2015/lib/RelativeDateRangeDuration.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various durations which can be filtered by the [[RelativeDateRangeFilter]].
 * @see [[RelativeDateRangeFilter.duration]]
 */
export var RelativeDateRangeDuration = /*@__PURE__*/ (function (RelativeDateRangeDuration) {
    /**
     * The duration of accepted values is one day.
     */
    RelativeDateRangeDuration[RelativeDateRangeDuration["Day"] = 0] = "Day";
    /**
     * The duration of accepted values is one week.
     */
    RelativeDateRangeDuration[RelativeDateRangeDuration["Week"] = 1] = "Week";
    /**
     * The duration of accepted values is one month.
     */
    RelativeDateRangeDuration[RelativeDateRangeDuration["Month"] = 2] = "Month";
    /**
     * The duration of accepted values is one quarter.
     */
    RelativeDateRangeDuration[RelativeDateRangeDuration["Quarter"] = 3] = "Quarter";
    /**
     * The duration of accepted values is one year.
     */
    RelativeDateRangeDuration[RelativeDateRangeDuration["Year"] = 4] = "Year";
    return RelativeDateRangeDuration;
})({});
/**
 * @hidden
 */
export let RelativeDateRangeDuration_$type = /*@__PURE__*/ markEnum('RelativeDateRangeDuration', 'Day,0|Week,1|Month,2|Quarter,3|Year,4');
