// igniteui-react-excel/esm2015/lib/RoundTripControlData.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
import { RoundTripPartData } from "./RoundTripPartData";
/**
 * @hidden
 */
export let RoundTripControlData = /*@__PURE__*/ (() => {
    class RoundTripControlData extends Base {
        constructor(a, b, c) {
            super();
            this.c = new List$1(RoundTripPartData.$, 0);
            this.a = null;
            this.d = null;
            this.b = false;
            this.e = null;
            this.d = a;
            this.e = b;
            this.a = c;
        }
    }
    RoundTripControlData.$t = /*@__PURE__*/ markType(RoundTripControlData, 'RoundTripControlData');
    return RoundTripControlData;
})();
