// igniteui-react-excel/esm2015/lib/SeriesType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Determines the type of [[Series]].
 */
export var SeriesType = /*@__PURE__*/ (function (SeriesType) {
    /**
     * Area; used by both area and area 3D charts.
     */
    SeriesType[SeriesType["Area"] = 0] = "Area";
    /**
     * Bar; used by both bar and bar 3D charts.
     */
    SeriesType[SeriesType["Bar"] = 1] = "Bar";
    /**
     * Line; used by line, line 3D, and stock charts.
     */
    SeriesType[SeriesType["Line"] = 2] = "Line";
    /**
     * Pie; used by pie, pie 3D, doughnut, and 'of pie' charts.
     */
    SeriesType[SeriesType["Pie"] = 3] = "Pie";
    /**
     * Radar.
     */
    SeriesType[SeriesType["Radar"] = 4] = "Radar";
    /**
     * Surface; used by surface and surface 3D charts.
     */
    SeriesType[SeriesType["Surface"] = 5] = "Surface";
    /**
     * Scatter.
     */
    SeriesType[SeriesType["Scatter"] = 6] = "Scatter";
    /**
     * Bubble.
     */
    SeriesType[SeriesType["Bubble"] = 7] = "Bubble";
    return SeriesType;
})({});
/**
 * @hidden
 */
export let SeriesType_$type = /*@__PURE__*/ markEnum('SeriesType', 'Area,0|Bar,1|Line,2|Pie,3|Radar,4|Surface,5|Scatter,6|Bubble,7');
