// igniteui-react-excel/esm2015/lib/SharedStringsPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let SharedStringsPartManager = /*@__PURE__*/ (() => {
    class SharedStringsPartManager extends XmlPartManagerBase {
        get_m() {
            return SharedStringsPartManager.a3;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/sharedStrings.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return SharedStringsPartManager.a6;
                case 1: return SharedStringsPartManager.a5;
                default: return SharedStringsPartManager.a6;
            }
        }
        get_g() {
            return false;
        }
        get g() {
            return this.get_g();
        }
    }
    SharedStringsPartManager.$t = /*@__PURE__*/ markType(SharedStringsPartManager, 'SharedStringsPartManager', XmlPartManagerBase.$);
    SharedStringsPartManager.a3 = "application/vnd.openxmlformats-officedocument.spreadsheetml.sharedStrings+xml";
    SharedStringsPartManager.a6 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/sharedStrings";
    SharedStringsPartManager.a5 = "http://purl.oclc.org/ooxml/officeDocument/relationships/sharedStrings";
    return SharedStringsPartManager;
})();
