// igniteui-react-excel/esm2015/lib/ShiftAddressResult.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ShiftAddressResult = /*@__PURE__*/ (() => {
    class ShiftAddressResult extends Base {
        constructor(a, ..._rest) {
            super();
            this.a = 0;
            this.e = false;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0:
                    {
                        let c = _rest[0];
                        let d = [c, 0];
                        {
                            let f = d[0];
                            let g = d[1];
                            this.e = f;
                            this.a = g;
                        }
                    }
                    break;
                case 1:
                    {
                        let c = _rest[0];
                        let d = _rest[1];
                        this.e = c;
                        this.a = d;
                    }
                    break;
            }
        }
        get b() {
            return this.a;
        }
        get f() {
            return this.e;
        }
        get g() {
            return this.a != 0;
        }
    }
    ShiftAddressResult.$t = /*@__PURE__*/ markType(ShiftAddressResult, 'ShiftAddressResult');
    ShiftAddressResult.d = /*@__PURE__*/ new ShiftAddressResult(0, false);
    ShiftAddressResult.c = /*@__PURE__*/ new ShiftAddressResult(0, true);
    return ShiftAddressResult;
})();
