// igniteui-react-excel/esm2015/lib/SlicerPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let SlicerPartManager = /*@__PURE__*/ (() => {
    class SlicerPartManager extends XmlPartManagerBase {
        get_m() {
            return "application/vnd.ms-excel.slicer+xml";
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/slicers/slicer.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0:
                case 1: return SlicerPartManager.a5;
                default: return SlicerPartManager.a5;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    SlicerPartManager.$t = /*@__PURE__*/ markType(SlicerPartManager, 'SlicerPartManager', XmlPartManagerBase.$);
    SlicerPartManager.a5 = "http://schemas.microsoft.com/office/2007/relationships/slicer";
    return SlicerPartManager;
})();
