// igniteui-react-excel/esm2015/lib/SlicerCachePartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let SlicerCachePartManager = /*@__PURE__*/ (() => {
    class SlicerCachePartManager extends XmlPartManagerBase {
        get_m() {
            return SlicerCachePartManager.a4;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/slicerCaches/slicerCache.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0:
                case 1: return SlicerCachePartManager.a5;
                default: return SlicerCachePartManager.a5;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    SlicerCachePartManager.$t = /*@__PURE__*/ markType(SlicerCachePartManager, 'SlicerCachePartManager', XmlPartManagerBase.$);
    SlicerCachePartManager.a4 = "application/vnd.ms-excel.slicerCache+xml";
    SlicerCachePartManager.a5 = "http://schemas.microsoft.com/office/2007/relationships/slicerCache";
    return SlicerCachePartManager;
})();
