// igniteui-react-excel/esm2015/lib/SortSettings_VisibleRowIndexBlock$1.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let SortSettings_VisibleRowIndexBlock$1 = /*@__PURE__*/ (() => {
    class SortSettings_VisibleRowIndexBlock$1 extends Base {
        constructor($t, a) {
            super();
            this.$t = null;
            this.b = 0;
            this.c = 0;
            this.$t = $t;
            this.$type = this.$type.specialize(this.$t);
            this.c = a;
            this.b = 1;
        }
        get a() {
            return this.c + this.b - 1;
        }
    }
    SortSettings_VisibleRowIndexBlock$1.$t = /*@__PURE__*/ markType(SortSettings_VisibleRowIndexBlock$1, 'SortSettings_VisibleRowIndexBlock$1');
    return SortSettings_VisibleRowIndexBlock$1;
})();
