// igniteui-react-excel/esm2015/lib/SparklineType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Determines the type if sparkline
 * @see [[SparklineGroup.type]]
 */
export var SparklineType = /*@__PURE__*/ (function (SparklineType) {
    /**
     * A line sparkline
     */
    SparklineType[SparklineType["Line"] = 0] = "Line";
    /**
     * A series of vertical columns
     */
    SparklineType[SparklineType["Column"] = 1] = "Column";
    /**
     * Deprecated - same as WinLoss
     */
    SparklineType[SparklineType["Stacked"] = 2] = "Stacked";
    /**
     * Positive and negative values stacked
     */
    SparklineType[SparklineType["WinLoss"] = 2] = "WinLoss";
    return SparklineType;
})({});
/**
 * @hidden
 */
export let SparklineType_$type = /*@__PURE__*/ markEnum('SparklineType', 'Line,0|Column,1|Stacked,2|WinLoss,2');
