// igniteui-react-excel/esm2015/lib/ST_AnimationChartBuildType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, markType, EnumUtil } from "igniteui-react-core";
import { ST_AnimationBuildType_$type } from "./ST_AnimationBuildType";
import { ST_AnimationChartOnlyBuildType_$type } from "./ST_AnimationChartOnlyBuildType";
/**
 * @hidden
 */
export let ST_AnimationChartBuildType = /*@__PURE__*/ (() => {
    class ST_AnimationChartBuildType extends Base {
        constructor() {
            super(...arguments);
            this.c = toNullable(ST_AnimationBuildType_$type, null);
            this.d = toNullable(ST_AnimationChartOnlyBuildType_$type, null);
        }
        get e() {
            return this.c;
        }
        set e(a) {
            this.c = a;
        }
        get f() {
            return this.d;
        }
        set f(a) {
            this.d = a;
        }
        static a(a) {
            return ST_AnimationChartBuildType.b(a);
        }
        g() {
            return this.h();
        }
        static b(a) {
            let b = new ST_AnimationChartBuildType();
            let c;
            let d;
            if (((() => { let e = EnumUtil.tryParse$1(ST_AnimationBuildType_$type, a, true, c); c = e.p2; return e.ret; })())) {
                b.e = toNullable(ST_AnimationBuildType_$type, c);
            }
            else if (((() => { let e = EnumUtil.tryParse$1(ST_AnimationChartOnlyBuildType_$type, a, true, d); d = e.p2; return e.ret; })())) {
                b.f = toNullable(ST_AnimationChartOnlyBuildType_$type, d);
            }
            return b;
        }
        h() {
            return this.e.hasValue ? this.e.toString() : this.f.hasValue ? this.f.toString() : null;
        }
    }
    ST_AnimationChartBuildType.$t = /*@__PURE__*/ markType(ST_AnimationChartBuildType, 'ST_AnimationChartBuildType');
    return ST_AnimationChartBuildType;
})();
