// igniteui-react-excel/esm2015/lib/ST_CellSpans.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, String_$type, markType } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { StringBuilder } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_CellSpans = /*@__PURE__*/ (() => {
    class ST_CellSpans extends Base {
        constructor() {
            super();
            this.c = null;
            this.c = new List$1(String_$type, 0);
        }
        static a(a) {
            return ST_CellSpans.b(a);
        }
        d() {
            return this.e();
        }
        static b(a) {
            let b = new ST_CellSpans();
            b.c = XmlNamespaceDefinitionBase.g(String_$type, a, (c) => c);
            return b;
        }
        e() {
            let a = new StringBuilder(0);
            let b = this.c.count - 1;
            for (let c = 0; c <= b; c++) {
                let d = this.c._inner[c];
                a.l(d);
                if (c < b) {
                    a.l(" ");
                }
            }
            return a.toString();
        }
    }
    ST_CellSpans.$t = /*@__PURE__*/ markType(ST_CellSpans, 'ST_CellSpans');
    return ST_CellSpans;
})();
