// igniteui-react-excel/esm2015/lib/ST_DrawingElementId.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, toNullable, Number_$type, markStruct } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_2, tryParseUInt32_2 } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_DrawingElementId = /*@__PURE__*/ (() => {
    class ST_DrawingElementId extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.g = 0;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.g = c;
                    }
                    break;
            }
        }
        static c(a) {
            return ST_DrawingElementId.a(parseUInt32_2(a, 7, CultureInfo.invariantCulture));
        }
        static f(a) {
            let b;
            if (((() => { let c = tryParseUInt32_2(a, 7, CultureInfo.invariantCulture, b); b = c.p3; return c.ret; })())) {
                return toNullable(ST_DrawingElementId.$, ST_DrawingElementId.a(b));
            }
            return toNullable(ST_DrawingElementId.$, null);
        }
        toString() {
            return this.g.toString();
        }
        static h(a) {
            return a.g;
        }
        static e(a) {
            if (!a.hasValue) {
                return toNullable(Number_$type, null);
            }
            return toNullable(Number_$type, ST_DrawingElementId.h(a.value));
        }
        static a(a) {
            return new ST_DrawingElementId(1, a);
        }
        static d(a) {
            if (!a.hasValue) {
                return toNullable(ST_DrawingElementId.$, null);
            }
            return toNullable(ST_DrawingElementId.$, ST_DrawingElementId.a(a.value));
        }
    }
    ST_DrawingElementId.$t = /*@__PURE__*/ markStruct(ST_DrawingElementId, 'ST_DrawingElementId');
    return ST_DrawingElementId;
})();
