// igniteui-react-excel/esm2015/lib/ST_DecimalNumber.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, toNullable, Number_$type, markStruct } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseInt32_1, tryParseInt32_2 } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_DecimalNumber = /*@__PURE__*/ (() => {
    class ST_DecimalNumber extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.e = 0;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.e = c;
                    }
                    break;
            }
        }
        static d(a) {
            return ST_DecimalNumber.b(parseInt32_1(a, CultureInfo.invariantCulture));
        }
        static j(a) {
            let b;
            if (((() => { let c = tryParseInt32_2(a, 7 | 4, CultureInfo.invariantCulture, b); b = c.p3; return c.ret; })())) {
                return toNullable(ST_DecimalNumber.$, ST_DecimalNumber.b(b));
            }
            return toNullable(ST_DecimalNumber.$, null);
        }
        toString() {
            return this.e.toString();
        }
        static f(a) {
            return a.e;
        }
        static h(a) {
            if (!a.hasValue) {
                return toNullable(Number_$type, null);
            }
            return toNullable(Number_$type, ST_DecimalNumber.f(a.value));
        }
        static b(a) {
            return new ST_DecimalNumber(1, a);
        }
        static i(a) {
            if (!a.hasValue) {
                return toNullable(ST_DecimalNumber.$, null);
            }
            return toNullable(ST_DecimalNumber.$, ST_DecimalNumber.b(a.value));
        }
    }
    ST_DecimalNumber.$t = /*@__PURE__*/ markStruct(ST_DecimalNumber, 'ST_DecimalNumber');
    ST_DecimalNumber.a = /*@__PURE__*/ new ST_DecimalNumber(0);
    return ST_DecimalNumber;
})();
