// igniteui-react-excel/esm2015/lib/ST_FixedAngle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, toNullable, typeCast, markStruct } from "igniteui-react-core";
import { ST_Angle } from "./ST_Angle";
import { DocCoreUtils } from "./DocCoreUtils";
/**
 * @hidden
 */
export let ST_FixedAngle = /*@__PURE__*/ (() => {
    class ST_FixedAngle extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.a = new ST_Angle();
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        DocCoreUtils.aw(ST_FixedAngle.$, c.n(), "value", -5400000, 5400000);
                        this.a = c;
                    }
                    break;
            }
        }
        equals(a) {
            let b = toNullable(ST_FixedAngle.$, typeCast(ST_FixedAngle.$, a));
            if (ST_FixedAngle.k(b, toNullable(ST_FixedAngle.$, null))) {
                return false;
            }
            return ST_FixedAngle.h(this, b.getValueOrDefault());
        }
        getHashCode() {
            return this.a.getHashCode();
        }
        toString() {
            return this.a.toString();
        }
        static d(a) {
            return ST_FixedAngle.e(ST_Angle.a(a));
        }
        static g(a) {
            let b = ST_FixedAngle.r(a);
            if (ST_FixedAngle.k(b, toNullable(ST_FixedAngle.$, null))) {
                return b.getValueOrDefault();
            }
            throw DocCoreUtils.n(ST_FixedAngle.$, a);
        }
        s() {
            return this.a.n();
        }
        static r(a) {
            return ST_FixedAngle.q(ST_Angle.m(a));
        }
        static h(a, b) {
            return ST_Angle.c(a.a, b.a);
        }
        static i(a, b) {
            if (!a.hasValue) {
                return !b.hasValue;
            }
            else if (!b.hasValue) {
                return false;
            }
            return ST_FixedAngle.h(a.value, b.value);
        }
        static j(a, b) {
            return !(ST_FixedAngle.h(a, b));
        }
        static k(a, b) {
            if (!a.hasValue) {
                return b.hasValue;
            }
            else if (!b.hasValue) {
                return true;
            }
            return ST_FixedAngle.j(a.value, b.value);
        }
        static e(a) {
            return new ST_FixedAngle(1, a);
        }
        static q(a) {
            if (!a.hasValue) {
                return toNullable(ST_FixedAngle.$, null);
            }
            return toNullable(ST_FixedAngle.$, ST_FixedAngle.e(a.value));
        }
        static b(a) {
            return a.a;
        }
        static p(a) {
            if (!a.hasValue) {
                return toNullable(ST_Angle.$, null);
            }
            return toNullable(ST_Angle.$, ST_FixedAngle.b(a.value));
        }
    }
    ST_FixedAngle.$t = /*@__PURE__*/ markStruct(ST_FixedAngle, 'ST_FixedAngle');
    return ST_FixedAngle;
})();
