// igniteui-react-excel/esm2015/lib/ST_Guid.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, toNullable, markStruct } from "igniteui-react-core";
import { Guid } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { stringToLocaleUpper } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_Guid = /*@__PURE__*/ (() => {
    class ST_Guid extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.d = new Guid();
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.d = c;
                    }
                    break;
            }
        }
        static c(a) {
            return ST_Guid.a(new Guid(3, a));
        }
        toString() {
            return stringToLocaleUpper(this.d.toString2("B"), CultureInfo.invariantCulture);
        }
        static a(a) {
            return new ST_Guid(1, a);
        }
        static h(a) {
            if (!a.hasValue) {
                return toNullable(ST_Guid.$, null);
            }
            return toNullable(ST_Guid.$, ST_Guid.a(a.value));
        }
        static e(a) {
            return a.d;
        }
        static g(a) {
            if (!a.hasValue) {
                return toNullable(Guid.$, null);
            }
            return toNullable(Guid.$, ST_Guid.e(a.value));
        }
    }
    ST_Guid.$t = /*@__PURE__*/ markStruct(ST_Guid, 'ST_Guid');
    return ST_Guid;
})();
