// igniteui-react-excel/esm2015/lib/ST_LongHexNumber.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, toNullable, markStruct } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseInt32_2, intToString1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_LongHexNumber = /*@__PURE__*/ (() => {
    class ST_LongHexNumber extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.e = 0;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.e = c;
                    }
                    break;
            }
        }
        static d(a) {
            return ST_LongHexNumber.b(parseInt32_2(a, 515, CultureInfo.invariantCulture));
        }
        toString() {
            return intToString1(this.e, "X8", CultureInfo.invariantCulture);
        }
        static b(a) {
            return new ST_LongHexNumber(1, a);
        }
        static f(a) {
            if (!a.hasValue) {
                return toNullable(ST_LongHexNumber.$, null);
            }
            return toNullable(ST_LongHexNumber.$, ST_LongHexNumber.b(a.value));
        }
    }
    ST_LongHexNumber.$t = /*@__PURE__*/ markStruct(ST_LongHexNumber, 'ST_LongHexNumber');
    ST_LongHexNumber.a = /*@__PURE__*/ new ST_LongHexNumber(0);
    return ST_LongHexNumber;
})();
