// igniteui-react-excel/esm2015/lib/ST_OnOff.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, markStruct } from "igniteui-react-core";
import { parseBool } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_OnOff = /*@__PURE__*/ (() => {
    class ST_OnOff extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.d = false;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.d = c;
                    }
                    break;
            }
        }
        static e(a) {
            switch (a) {
                case "1": return true;
                case "0": return false;
            }
            return parseBool(a);
        }
        toString() {
            return this.d ? "1" : "0";
        }
        static g(a) {
            return a ? "1" : "0";
        }
        static b(a) {
            return a ? ST_OnOff.c : ST_OnOff.a;
        }
        f() {
            return this.d;
        }
    }
    ST_OnOff.$t = /*@__PURE__*/ markStruct(ST_OnOff, 'ST_OnOff');
    ST_OnOff.a = /*@__PURE__*/ new ST_OnOff(1, false);
    ST_OnOff.c = /*@__PURE__*/ new ST_OnOff(1, true);
    return ST_OnOff;
})();
