// igniteui-react-excel/esm2015/lib/ST_Sqref.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, String_$type, fromEnum, markType } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { StringBuilder } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_Sqref = /*@__PURE__*/ (() => {
    class ST_Sqref extends Base {
        constructor() {
            super(...arguments);
            this.c = null;
        }
        get d() {
            return this.c;
        }
        set d(a) {
            this.c = a;
        }
        static a(a) {
            return ST_Sqref.b(a);
        }
        e() {
            return this.f();
        }
        static b(a) {
            let b = new ST_Sqref();
            let c = XmlNamespaceDefinitionBase.g(String_$type, a, (d) => d);
            if (c != null) {
                b.d = new List$1(String_$type, 2, c.count);
                for (let d of fromEnum(c)) {
                    b.d.add(d);
                }
            }
            return b;
        }
        f() {
            if (this.d == null) {
                return null;
            }
            let a = new StringBuilder(0);
            let b = this.d.count - 1;
            for (let c = 0; c <= b; c++) {
                let d = this.d._inner[c];
                a.l(d);
                if (c < b) {
                    a.l(" ");
                }
            }
            return a.toString();
        }
    }
    ST_Sqref.$t = /*@__PURE__*/ markType(ST_Sqref, 'ST_Sqref');
    return ST_Sqref;
})();
