// igniteui-react-excel/esm2015/lib/ST_TrueFalseBlank.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, markStruct } from "igniteui-react-core";
import { DocCoreUtils } from "./DocCoreUtils";
import { stringIsNullOrEmpty, stringEmpty } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_TrueFalseBlank = /*@__PURE__*/ (() => {
    class ST_TrueFalseBlank extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.c = false;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.c = c;
                    }
                    break;
            }
        }
        static d(a) {
            if (stringIsNullOrEmpty(a)) {
                return true;
            }
            switch (a) {
                case "t":
                case "True": return true;
                case "f":
                case "False": return false;
            }
            throw DocCoreUtils.n(ST_TrueFalseBlank.$, a);
        }
        toString() {
            if (this.c) {
                return stringEmpty();
            }
            return "False";
        }
        static e(a) {
            return a.toString();
        }
        static f(a) {
            return (a ? ST_TrueFalseBlank.b : ST_TrueFalseBlank.a).toString();
        }
    }
    ST_TrueFalseBlank.$t = /*@__PURE__*/ markStruct(ST_TrueFalseBlank, 'ST_TrueFalseBlank');
    ST_TrueFalseBlank.a = /*@__PURE__*/ new ST_TrueFalseBlank(1, false);
    ST_TrueFalseBlank.b = /*@__PURE__*/ new ST_TrueFalseBlank(1, true);
    return ST_TrueFalseBlank;
})();
