// igniteui-react-excel/esm2015/lib/ST_TextPointUnqualified.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, toNullable, Number_$type, markStruct } from "igniteui-react-core";
import { DocCoreUtils } from "./DocCoreUtils";
import { CultureInfo } from "igniteui-react-core";
import { parseInt32_2, tryParseInt32_2 } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_TextPointUnqualified = /*@__PURE__*/ (() => {
    class ST_TextPointUnqualified extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.d = 0;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        DocCoreUtils.aw(ST_TextPointUnqualified.$, c, "value", -400000, 400000);
                        this.d = c;
                    }
                    break;
            }
        }
        static c(a) {
            return ST_TextPointUnqualified.a(parseInt32_2(a, 7, CultureInfo.invariantCulture));
        }
        static i(a) {
            let b;
            if (((() => { let c = tryParseInt32_2(a, 7, CultureInfo.invariantCulture, b); b = c.p3; return c.ret; })())) {
                return toNullable(ST_TextPointUnqualified.$, ST_TextPointUnqualified.a(b));
            }
            return toNullable(ST_TextPointUnqualified.$, null);
        }
        toString() {
            return this.d.toString();
        }
        static e(a) {
            return a.d;
        }
        static g(a) {
            if (!a.hasValue) {
                return toNullable(Number_$type, null);
            }
            return toNullable(Number_$type, ST_TextPointUnqualified.e(a.value));
        }
        static a(a) {
            return new ST_TextPointUnqualified(1, a);
        }
        static h(a) {
            if (!a.hasValue) {
                return toNullable(ST_TextPointUnqualified.$, null);
            }
            return toNullable(ST_TextPointUnqualified.$, ST_TextPointUnqualified.a(a.value));
        }
    }
    ST_TextPointUnqualified.$t = /*@__PURE__*/ markStruct(ST_TextPointUnqualified, 'ST_TextPointUnqualified');
    return ST_TextPointUnqualified;
})();
