// igniteui-react-excel/esm2015/lib/ST_UnsignedDecimalNumber.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, toNullable, Number_$type, markStruct } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { tryParseUInt64_2 } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_UnsignedDecimalNumber = /*@__PURE__*/ (() => {
    class ST_UnsignedDecimalNumber extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.g = 0;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.g = c;
                    }
                    break;
            }
        }
        static f(a) {
            let b;
            if (((() => { let c = tryParseUInt64_2(a, 7 | 4, CultureInfo.invariantCulture, b); b = c.p3; return c.ret; })())) {
                return toNullable(ST_UnsignedDecimalNumber.$, ST_UnsignedDecimalNumber.b(b));
            }
            return toNullable(ST_UnsignedDecimalNumber.$, null);
        }
        toString() {
            return this.g.toString();
        }
        static h(a) {
            return a.g;
        }
        static e(a) {
            if (!a.hasValue) {
                return toNullable(Number_$type, null);
            }
            return toNullable(Number_$type, ST_UnsignedDecimalNumber.h(a.value));
        }
        static b(a) {
            return new ST_UnsignedDecimalNumber(1, a);
        }
        static d(a) {
            if (!a.hasValue) {
                return toNullable(ST_UnsignedDecimalNumber.$, null);
            }
            return toNullable(ST_UnsignedDecimalNumber.$, ST_UnsignedDecimalNumber.b(a.value));
        }
    }
    ST_UnsignedDecimalNumber.$t = /*@__PURE__*/ markStruct(ST_UnsignedDecimalNumber, 'ST_UnsignedDecimalNumber');
    ST_UnsignedDecimalNumber.a = /*@__PURE__*/ new ST_UnsignedDecimalNumber(0);
    return ST_UnsignedDecimalNumber;
})();
