// igniteui-react-excel/esm2015/lib/ST_UcharHexNumber.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, toNullable, Number_$type, markStruct } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt8_2, intToString1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_UcharHexNumber = /*@__PURE__*/ (() => {
    class ST_UcharHexNumber extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.e = 0;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.e = c;
                    }
                    break;
            }
        }
        static d(a) {
            return ST_UcharHexNumber.b(parseUInt8_2(a, 515, CultureInfo.invariantCulture));
        }
        toString() {
            return intToString1(this.e, "X2", CultureInfo.invariantCulture);
        }
        static b(a) {
            return new ST_UcharHexNumber(1, a);
        }
        static i(a) {
            if (!a.hasValue) {
                return toNullable(ST_UcharHexNumber.$, null);
            }
            return toNullable(ST_UcharHexNumber.$, ST_UcharHexNumber.b(a.value));
        }
        static f(a) {
            return a.e;
        }
        static h(a) {
            if (!a.hasValue) {
                return toNullable(Number_$type, null);
            }
            return toNullable(Number_$type, ST_UcharHexNumber.f(a.value));
        }
    }
    ST_UcharHexNumber.$t = /*@__PURE__*/ markStruct(ST_UcharHexNumber, 'ST_UcharHexNumber');
    ST_UcharHexNumber.a = /*@__PURE__*/ new ST_UcharHexNumber(0);
    return ST_UcharHexNumber;
})();
