// igniteui-react-excel/esm2015/lib/ST_UnsignedShortHex.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, markStruct } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt16_2, intToString1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let ST_UnsignedShortHex = /*@__PURE__*/ (() => {
    class ST_UnsignedShortHex extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.c = 0;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        this.c = c;
                    }
                    break;
            }
        }
        static d(a) {
            return parseUInt16_2(a, 515, CultureInfo.invariantCulture);
        }
        static b(a) {
            return new ST_UnsignedShortHex(1, ST_UnsignedShortHex.d(a));
        }
        toString() {
            return intToString1(this.c, "X4", CultureInfo.invariantCulture);
        }
        e() {
            return this.c;
        }
    }
    ST_UnsignedShortHex.$t = /*@__PURE__*/ markStruct(ST_UnsignedShortHex, 'ST_UnsignedShortHex');
    ST_UnsignedShortHex.a = /*@__PURE__*/ new ST_UnsignedShortHex(0);
    return ST_UnsignedShortHex;
})();
