// igniteui-react-excel/esm2015/lib/StopInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType } from "igniteui-react-core";
import { ColorInfo } from "./ColorInfo";
import { CellFillGradientStop } from "./CellFillGradientStop";
import { WorkbookColorInfo } from "./WorkbookColorInfo";
import { ColorsInternal } from "./ColorsInternal";
/**
 * @hidden
 */
export let StopInfo = /*@__PURE__*/ (() => {
    class StopInfo extends Base {
        constructor() {
            super(...arguments);
            this.f = 0;
            this.b = null;
        }
        get g() {
            return this.f;
        }
        set g(a) {
            this.f = a;
        }
        get c() {
            return this.b;
        }
        set c(a) {
            this.b = a;
        }
        equals(a) {
            if (Base.referenceEquals(this, a)) {
                return true;
            }
            let b = typeCast(StopInfo.$, a);
            if (b == null) {
                return false;
            }
            return Base.equalsStatic(this.f, b.f) && Base.equalsStatic(this.b, b.b);
        }
        getHashCode() {
            let a = (this.f);
            if (this.b != null) {
                a ^= this.b.getHashCode();
            }
            return a;
        }
        static d(a, b) {
            let c = new StopInfo();
            c.c = ColorInfo.a(a, b.colorInfo, 1);
            c.g = b.offset;
            return c;
        }
        a(a) {
            let b = new WorkbookColorInfo(ColorsInternal.a);
            if (this.c != null) {
                b = this.c.c(a);
            }
            return new CellFillGradientStop(b, this.g);
        }
    }
    StopInfo.$t = /*@__PURE__*/ markType(StopInfo, 'StopInfo');
    return StopInfo;
})();
