// igniteui-react-excel/esm2015/lib/StringUtilities.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, callStaticConstructors, markType } from "igniteui-react-core";
import { StringComparer } from "igniteui-react-core";
import { stringCompare2 } from "igniteui-react-core";
/**
 * @hidden
 */
export let StringUtilities = /*@__PURE__*/ (() => {
    class StringUtilities extends Base {
        static staticInit() {
            let a = "This\r\nThat";
            StringUtilities.a = a.indexOf("\n") > 0;
        }
        static get j() {
            return 3;
        }
        static get f() {
            return StringComparer.e;
        }
        static get i() {
            return 2;
        }
        static get e() {
            return StringComparer.d;
        }
        static get g() {
            return StringComparer.f;
        }
        static get h() {
            return StringComparer.g;
        }
        static d(a, b, c, d) {
            return stringCompare2(a, b, c, d);
        }
        static b(a, b) {
            return a.toLowerCase();
        }
        static c(a, b) {
            return a.toUpperCase();
        }
    }
    StringUtilities.$t = /*@__PURE__*/ markType(StringUtilities, 'StringUtilities');
    StringUtilities.a = false;
    return StringUtilities;
})();
(function (v) { callStaticConstructors(); })(true);
