// igniteui-react-excel/esm2015/lib/StyleInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, typeCast, markType } from "igniteui-react-core";
import { WorkbookBuiltInStyle } from "./WorkbookBuiltInStyle";
import { Convert } from "igniteui-react-core";
import { stringEmpty } from "igniteui-react-core";
/**
 * @hidden
 */
export let StyleInfo = /*@__PURE__*/ (() => {
    class StyleInfo extends Base {
        constructor(a, ..._rest) {
            super();
            this.g = toNullable(Number_$type, null);
            this.a = false;
            this.c = false;
            this.i = toNullable(Number_$type, 0);
            this.k = stringEmpty();
            this.f = -1;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        let d = _rest[1];
                        this.l = d.name;
                        this.e = c._a9(d);
                        let e = typeCast(WorkbookBuiltInStyle.$, d);
                        if (e != null) {
                            if (e._ak == 255) {
                                this.j = toNullable(Number_$type, null);
                            }
                            else {
                                this.j = toNullable(Number_$type, Convert.toUInt32(e._ak));
                            }
                            this.h = toNullable(Number_$type, e._ac);
                            this.b = e._ah;
                        }
                        this.d = d._m;
                    }
                    break;
            }
        }
        get h() {
            return this.g;
        }
        set h(a) {
            this.g = a;
        }
        get b() {
            return this.a;
        }
        set b(a) {
            this.a = a;
        }
        get d() {
            return this.c;
        }
        set d(a) {
            this.c = a;
        }
        get j() {
            return this.i;
        }
        set j(a) {
            this.i = a;
        }
        get l() {
            return this.k;
        }
        set l(a) {
            this.k = a;
        }
        get e() {
            return this.f;
        }
        set e(a) {
            this.f = a;
        }
    }
    StyleInfo.$t = /*@__PURE__*/ markType(StyleInfo, 'StyleInfo');
    return StyleInfo;
})();
