// igniteui-react-excel/esm2015/lib/ThemeManagerPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ThemeManagerPartManager = /*@__PURE__*/ (() => {
    class ThemeManagerPartManager extends XmlPartManagerBase {
        get_m() {
            return ThemeManagerPartManager.a3;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0:
                case 1: return "/theme/theme/themeManager.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return ThemeManagerPartManager.a6;
                case 1: return ThemeManagerPartManager.a5;
                default: return ThemeManagerPartManager.a6;
            }
        }
        get_g() {
            return false;
        }
        get g() {
            return this.get_g();
        }
    }
    ThemeManagerPartManager.$t = /*@__PURE__*/ markType(ThemeManagerPartManager, 'ThemeManagerPartManager', XmlPartManagerBase.$);
    ThemeManagerPartManager.a3 = "application/vnd.openxmlformats-officedocument.themeManager+xml";
    ThemeManagerPartManager.a6 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/officeDocument";
    ThemeManagerPartManager.a5 = "http://purl.oclc.org/ooxml/officeDocument/relationships/officeDocument";
    return ThemeManagerPartManager;
})();
