// igniteui-react-excel/esm2015/lib/TickLabelPosition.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which determine the placement of an axis' <see cref="Axis.TickLabels">tick labels</see>.
 */
export var TickLabelPosition = /*@__PURE__*/ (function (TickLabelPosition) {
    /**
     * Tick labels appear near the highest data value.
     */
    TickLabelPosition[TickLabelPosition["High"] = -4127] = "High";
    /**
     * Tick labels appear near the lowest data value.
     */
    TickLabelPosition[TickLabelPosition["Low"] = -4134] = "Low";
    /**
     * Tick labels appear near the axis.
     */
    TickLabelPosition[TickLabelPosition["NextToAxis"] = 4] = "NextToAxis";
    /**
     * Tick labels are not displayed.
     */
    TickLabelPosition[TickLabelPosition["None"] = -4142] = "None";
    return TickLabelPosition;
})({});
/**
 * @hidden
 */
export let TickLabelPosition_$type = /*@__PURE__*/ markEnum('TickLabelPosition', 'High,-4127|Low,-4134|NextToAxis,4|None,-4142');
