// igniteui-react-excel/esm2015/lib/TiffPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ImagePartManagerBase } from "./ImagePartManagerBase";
import { ImageFormat } from "./ImageFormat";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let TiffPartManager = /*@__PURE__*/ (() => {
    class TiffPartManager extends ImagePartManagerBase {
        constructor() {
            super();
        }
        get_s() {
            return "/xl/media/image.tiff";
        }
        get s() {
            return this.get_s();
        }
        get_t() {
            return "/ppt/media/image.tiff";
        }
        get t() {
            return this.get_t();
        }
        get_u() {
            return "/word/media/image.tiff";
        }
        get u() {
            return this.get_u();
        }
        get_m() {
            return "image/tiff";
        }
        get m() {
            return this.get_m();
        }
        r(a) {
            return ImageFormat.h;
        }
    }
    TiffPartManager.$t = /*@__PURE__*/ markType(TiffPartManager, 'TiffPartManager', ImagePartManagerBase.$);
    TiffPartManager.y = /*@__PURE__*/ new TiffPartManager();
    return TiffPartManager;
})();
