// igniteui-react-excel/esm2015/lib/TrendlineType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which define the various types of trendlines
 * supported by a [[WorksheetChart]].
 */
export var TrendlineType = /*@__PURE__*/ (function (TrendlineType) {
    /**
     * The trendline is exponential.
     */
    TrendlineType[TrendlineType["Exponential"] = 0] = "Exponential";
    /**
     * The trendline is linear.
     */
    TrendlineType[TrendlineType["Linear"] = 1] = "Linear";
    /**
     * The trendline is logarithmic.
     */
    TrendlineType[TrendlineType["Logarithmic"] = 2] = "Logarithmic";
    /**
     * The trendline depicts a moving average.
     */
    TrendlineType[TrendlineType["MovingAverage"] = 3] = "MovingAverage";
    /**
     * The trendline depicts a polynomial of order/degree <i>n</i>,
     * when <i>n</i> is specified by the ChartTrendline.PolynomialOrder
     * property.
     */
    TrendlineType[TrendlineType["Polynomial"] = 4] = "Polynomial";
    /**
     * The trendline depicts a power curve.
     */
    TrendlineType[TrendlineType["Power"] = 5] = "Power";
    return TrendlineType;
})({});
/**
 * @hidden
 */
export let TrendlineType_$type = /*@__PURE__*/ markEnum('TrendlineType', 'Exponential,0|Linear,1|Logarithmic,2|MovingAverage,3|Polynomial,4|Power,5');
