// igniteui-react-excel/esm2015/lib/UltraCalcFunctionLarge.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { UltraCalcFunctionSmall } from "./UltraCalcFunctionSmall";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionLarge = /*@__PURE__*/ (() => {
    class UltraCalcFunctionLarge extends BuiltInFunctionBase {
        evaluate(a, b) {
            return UltraCalcFunctionSmall._ak(a, true);
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_name() {
            return "large";
        }
        get name() {
            return this.get_name();
        }
    }
    UltraCalcFunctionLarge.$t = /*@__PURE__*/ markType(UltraCalcFunctionLarge, 'UltraCalcFunctionLarge', BuiltInFunctionBase.$);
    return UltraCalcFunctionLarge;
})();
