// igniteui-react-excel/esm2015/lib/UltraCalcFunctionSqrtPi.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionSqrtPi = /*@__PURE__*/ (() => {
    class UltraCalcFunctionSqrtPi extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let d = c._toDouble3();
            if (d < 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let e = Math.sqrt(Math.PI * d);
            return new ExcelCalcValue(e);
        }
        get_name() {
            return "sqrtpi";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return (b == false);
        }
    }
    UltraCalcFunctionSqrtPi.$t = /*@__PURE__*/ markType(UltraCalcFunctionSqrtPi, 'UltraCalcFunctionSqrtPi', BuiltInFunctionBase.$);
    return UltraCalcFunctionSqrtPi;
})();
