// igniteui-react-excel/esm2015/lib/UltraCalcFunctionCombina.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { UltraCalcFunctionCombin } from "./UltraCalcFunctionCombin";
import { markType } from "igniteui-react-core";
import { truncate, isNaN_ } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionCombina = /*@__PURE__*/ (() => {
    class UltraCalcFunctionCombina extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let d = c._toDecimal2();
            c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let e = c._toDecimal2();
            let f = truncate(d);
            let g = truncate(e);
            if (e < 0 || d < 0 || (g == 0 && f > 0)) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let h = UltraCalcFunctionCombin._am(g, f, true);
            if (isNaN_(h)) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            return new ExcelCalcValue(h);
        }
        get_name() {
            return "combina";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionCombina.$t = /*@__PURE__*/ markType(UltraCalcFunctionCombina, 'UltraCalcFunctionCombina', BuiltInFunctionBase.$);
    return UltraCalcFunctionCombina;
})();
