// igniteui-react-excel/esm2015/lib/UltraCalcFunctionLog.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
import { logBase } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionLog = /*@__PURE__*/ (() => {
    class UltraCalcFunctionLog extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = null;
            if (b == 2) {
                c = a.pop();
            }
            let d = a.pop();
            if (d.isError) {
                return new ExcelCalcValue(d.toErrorValue());
            }
            if (c != null && c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let e, f;
            if (c != null) {
                if (((() => { let g = c._toDouble1(e); e = g.p0; return g.ret; })()) == false) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(1));
                }
            }
            else {
                e = 10;
            }
            if (((() => { let g = d._toDouble1(f); f = g.p0; return g.ret; })()) == false) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(1));
            }
            if (true && f == 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            return new ExcelCalcValue(logBase(f, e));
        }
        get_name() {
            return "log";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionLog.$t = /*@__PURE__*/ markType(UltraCalcFunctionLog, 'UltraCalcFunctionLog', BuiltInFunctionBase.$);
    return UltraCalcFunctionLog;
})();
