// igniteui-react-excel/esm2015/lib/UltraCalcFunctionWeibull.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionWeibull = /*@__PURE__*/ (() => {
    class UltraCalcFunctionWeibull extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            let d = a.pop();
            let e = a.pop();
            let f = a.pop();
            if (f.isError) {
                return new ExcelCalcValue(f.toErrorValue());
            }
            if (e.isError) {
                return new ExcelCalcValue(e.toErrorValue());
            }
            if (d.isError) {
                return new ExcelCalcValue(d.toErrorValue());
            }
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let g = f._toDouble3();
            let h = e._toDouble3();
            let i = d._toDouble3();
            let j = c._toBoolean1();
            if (g < 0 || h <= 0 || i <= 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let k = Math.exp(-Math.pow(g / i, h));
            let l = j ? (1 - k) : h / Math.pow(i, h) * Math.pow(g, h - 1) * k;
            return new ExcelCalcValue(l);
        }
        get_name() {
            return "weibull";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 4;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 4;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionWeibull.$t = /*@__PURE__*/ markType(UltraCalcFunctionWeibull, 'UltraCalcFunctionWeibull', BuiltInFunctionBase.$);
    return UltraCalcFunctionWeibull;
})();
