// igniteui-react-excel/esm2015/lib/UltraCalcFunctionAcot.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
/**
 * @hidden
 */
export let UltraCalcFunctionAcot = /*@__PURE__*/ (() => {
    class UltraCalcFunctionAcot extends BuiltInFunctionBase {
        evaluate(a, b) {
            return CalcFunctionUtilities.h(a, true, (c) => new ExcelCalcValue(Math.PI / 2 - Math.atan(c)));
        }
        get_name() {
            return "acot";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionAcot.$t = /*@__PURE__*/ markType(UltraCalcFunctionAcot, 'UltraCalcFunctionAcot', BuiltInFunctionBase.$);
    return UltraCalcFunctionAcot;
})();
