// igniteui-react-excel/esm2015/lib/UltraCalcFunctionInt.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { truncate } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionInt = /*@__PURE__*/ (() => {
    class UltraCalcFunctionInt extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop()._toDecimal2();
            return new ExcelCalcValue(truncate(Math.floor(c)));
        }
        get_name() {
            return "int";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionInt.$t = /*@__PURE__*/ markType(UltraCalcFunctionInt, 'UltraCalcFunctionInt', BuiltInFunctionBase.$);
    return UltraCalcFunctionInt;
})();
