// igniteui-react-excel/esm2015/lib/UltraCalcFunctionDec2Oct.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { UltraCalcFunctionDec2XBase } from "./UltraCalcFunctionDec2XBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionDec2Oct = /*@__PURE__*/ (() => {
    class UltraCalcFunctionDec2Oct extends UltraCalcFunctionDec2XBase {
        get_name() {
            return "dec2oct";
        }
        get name() {
            return this.get_name();
        }
        get__ap() {
            return -536870912;
        }
        get _ap() {
            return this.get__ap();
        }
        get__al() {
            return 536870911;
        }
        get _al() {
            return this.get__al();
        }
        get__ak() {
            return 8;
        }
        get _ak() {
            return this.get__ak();
        }
    }
    UltraCalcFunctionDec2Oct.$t = /*@__PURE__*/ markType(UltraCalcFunctionDec2Oct, 'UltraCalcFunctionDec2Oct', UltraCalcFunctionDec2XBase.$);
    return UltraCalcFunctionDec2Oct;
})();
