// igniteui-react-excel/esm2015/lib/UltraCalcFunctionErrorType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionErrorType = /*@__PURE__*/ (() => {
    class UltraCalcFunctionErrorType extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c;
            let d = a.pop();
            if (d.isError) {
                let e = d.toErrorValue();
                switch (e.code) {
                    case 6:
                        c = 1;
                        break;
                    case 2:
                        c = 2;
                        break;
                    case 1:
                        c = 3;
                        break;
                    case 0:
                        c = 4;
                        break;
                    case 7:
                        c = 5;
                        break;
                    case 4:
                        c = 6;
                        break;
                    case 3:
                        c = 7;
                        break;
                    default: return new ExcelCalcValue(new ExcelCalcErrorValue(3));
                }
                return new ExcelCalcValue(c);
            }
            else {
                return new ExcelCalcValue(new ExcelCalcErrorValue(3));
            }
        }
        get_name() {
            return "errortype";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionErrorType.$t = /*@__PURE__*/ markType(UltraCalcFunctionErrorType, 'UltraCalcFunctionErrorType', BuiltInFunctionBase.$);
    return UltraCalcFunctionErrorType;
})();
