// igniteui-react-excel/esm2015/lib/UltraCalcFunctionOct2Dec.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { UltraCalcFunctionXBase2Dec } from "./UltraCalcFunctionXBase2Dec";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionOct2Dec = /*@__PURE__*/ (() => {
    class UltraCalcFunctionOct2Dec extends UltraCalcFunctionXBase2Dec {
        get_name() {
            return "oct2dec";
        }
        get name() {
            return this.get_name();
        }
        get__ao() {
            return -536870912;
        }
        get _ao() {
            return this.get__ao();
        }
        get__an() {
            return 536870911;
        }
        get _an() {
            return this.get__an();
        }
        get__ak() {
            return 8;
        }
        get _ak() {
            return this.get__ak();
        }
        get__am() {
            return 10;
        }
        get _am() {
            return this.get__am();
        }
    }
    UltraCalcFunctionOct2Dec.$t = /*@__PURE__*/ markType(UltraCalcFunctionOct2Dec, 'UltraCalcFunctionOct2Dec', UltraCalcFunctionXBase2Dec.$);
    return UltraCalcFunctionOct2Dec;
})();
