// igniteui-react-excel/esm2015/lib/UltraCalcFunctionFact.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
import { truncate, isInfinity } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionFact = /*@__PURE__*/ (() => {
    class UltraCalcFunctionFact extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let d = c._toDecimal2();
            if (d < 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let e = truncate(d);
            let f = UltraCalcFunctionFact._ak(e);
            if (isInfinity(f)) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            return new ExcelCalcValue(f);
        }
        get_name() {
            return "fact";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        static _ak(a, b = 1) {
            if (a == 0) {
                a = 1;
            }
            let c = 1;
            for (let d = a; d > 1; d -= b) {
                c *= d;
            }
            return c;
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionFact.$t = /*@__PURE__*/ markType(UltraCalcFunctionFact, 'UltraCalcFunctionFact', BuiltInFunctionBase.$);
    return UltraCalcFunctionFact;
})();
