// igniteui-react-excel/esm2015/lib/UltraCalcFunctionLower.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { stringToLocaleLower } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionLower = /*@__PURE__*/ (() => {
    class UltraCalcFunctionLower extends BuiltInFunctionBase {
        evaluate(a, b) {
            if (true) {
                let c = a.pop();
                if (c.isError) {
                    return c;
                }
                return new ExcelCalcValue(c.toString().toLowerCase());
            }
            if (a.peek().isString) {
                return new ExcelCalcValue(stringToLocaleLower(a.pop().toString(), a._w));
            }
            else {
                return a.pop();
            }
        }
        get_name() {
            return "lower";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionLower.$t = /*@__PURE__*/ markType(UltraCalcFunctionLower, 'UltraCalcFunctionLower', BuiltInFunctionBase.$);
    return UltraCalcFunctionLower;
})();
