// igniteui-react-excel/esm2015/lib/UltraCalcFunctionDollarDe.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { MathUtilities } from "./MathUtilities";
import { UltraCalcFunctionDollarFr } from "./UltraCalcFunctionDollarFr";
import { markType } from "igniteui-react-core";
import { truncate } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionDollarDe = /*@__PURE__*/ (() => {
    class UltraCalcFunctionDollarDe extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let d = c._toDecimal2();
            if (d < 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let e = truncate(d);
            if (e == 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(2));
            }
            c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let f = c._toDouble3();
            let g = MathUtilities.j(f);
            let h = f - g;
            h /= e;
            let i = UltraCalcFunctionDollarFr._ak(e);
            h *= i;
            f = g + h;
            return new ExcelCalcValue(f);
        }
        get_name() {
            return "dollarde";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionDollarDe.$t = /*@__PURE__*/ markType(UltraCalcFunctionDollarDe, 'UltraCalcFunctionDollarDe', BuiltInFunctionBase.$);
    return UltraCalcFunctionDollarDe;
})();
