// igniteui-react-excel/esm2015/lib/UltraCalcFunctionMod.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionMod = /*@__PURE__*/ (() => {
    class UltraCalcFunctionMod extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            let d = a.pop();
            if (d.isError) {
                return new ExcelCalcValue(d.toErrorValue());
            }
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let e, f;
            if (!((() => { let g = c._toDouble1(e); e = g.p0; return g.ret; })()) || !((() => { let g = d._toDouble1(f); f = g.p0; return g.ret; })())) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            if (e == 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(2));
            }
            else {
                let g = f % e;
                let h = Math.sign(g);
                if (h != 0 && h != Math.sign(e)) {
                    g *= -1;
                }
                return new ExcelCalcValue(g);
            }
        }
        get_name() {
            return "mod";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionMod.$t = /*@__PURE__*/ markType(UltraCalcFunctionMod, 'UltraCalcFunctionMod', BuiltInFunctionBase.$);
    return UltraCalcFunctionMod;
})();
