// igniteui-react-excel/esm2015/lib/UltraCalcFunctionEcmaCeiling.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { MathUtilities } from "./MathUtilities";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionEcmaCeiling = /*@__PURE__*/ (() => {
    class UltraCalcFunctionEcmaCeiling extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = null;
            if (b == 2) {
                c = a.pop();
            }
            let d = a.pop();
            if (d.isError) {
                return new ExcelCalcValue(d.toErrorValue());
            }
            if (c != null && c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let e = d._toDouble3();
            let f = c == null ? 1 : c._toDouble3();
            if (e == 0 || f == 0) {
                return new ExcelCalcValue(0);
            }
            if (f < 0 && e > 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let g = MathUtilities.i(Math.ceil(e / f) * f);
            return new ExcelCalcValue(g);
        }
        get_name() {
            return "ecma.ceiling";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionEcmaCeiling.$t = /*@__PURE__*/ markType(UltraCalcFunctionEcmaCeiling, 'UltraCalcFunctionEcmaCeiling', BuiltInFunctionBase.$);
    return UltraCalcFunctionEcmaCeiling;
})();
