// igniteui-react-excel/esm2015/lib/UltraCalcFunctionOct2Hex.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { UltraCalcFunctionXBase2XBase } from "./UltraCalcFunctionXBase2XBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionOct2Hex = /*@__PURE__*/ (() => {
    class UltraCalcFunctionOct2Hex extends UltraCalcFunctionXBase2XBase {
        get_name() {
            return "oct2hex";
        }
        get name() {
            return this.get_name();
        }
        get__aq() {
            return -536870912;
        }
        get _aq() {
            return this.get__aq();
        }
        get__ap() {
            return 536870911;
        }
        get _ap() {
            return this.get__ap();
        }
        get__ao() {
            return -549755813888;
        }
        get _ao() {
            return this.get__ao();
        }
        get__an() {
            return 0x7FFFFFFFFF;
        }
        get _an() {
            return this.get__an();
        }
        get__al() {
            return 8;
        }
        get _al() {
            return this.get__al();
        }
        get__ak() {
            return 16;
        }
        get _ak() {
            return this.get__ak();
        }
        get__am() {
            return 10;
        }
        get _am() {
            return this.get__am();
        }
    }
    UltraCalcFunctionOct2Hex.$t = /*@__PURE__*/ markType(UltraCalcFunctionOct2Hex, 'UltraCalcFunctionOct2Hex', UltraCalcFunctionXBase2XBase.$);
    return UltraCalcFunctionOct2Hex;
})();
