// igniteui-react-excel/esm2015/lib/UltraCalcFunctionStandardize.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
export let UltraCalcFunctionStandardize = /*@__PURE__*/ (() => {
    class UltraCalcFunctionStandardize extends BuiltInFunctionBase {
        evaluate(a, b) {
            return CalcFunctionUtilities.i(a, true, (c, d, e) => {
                if (e <= 0) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(4));
                }
                return new ExcelCalcValue((c - d) / e);
            });
        }
        get_maxArgs() {
            return 3;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        get_minArgs() {
            return 3;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_name() {
            return "standardize";
        }
        get name() {
            return this.get_name();
        }
    }
    UltraCalcFunctionStandardize.$t = /*@__PURE__*/ markType(UltraCalcFunctionStandardize, 'UltraCalcFunctionStandardize', BuiltInFunctionBase.$);
    return UltraCalcFunctionStandardize;
})();
