// igniteui-react-excel/esm2015/lib/UltraCalcFunctionSkew_P.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { UltraCalcFunctionSkew } from "./UltraCalcFunctionSkew";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionSkew_P = /*@__PURE__*/ (() => {
    class UltraCalcFunctionSkew_P extends BuiltInFunctionBase {
        evaluate(a, b) {
            return UltraCalcFunctionSkew._ak(a, b, false);
        }
        get_name() {
            return "skew.p";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0x7FFFFFFF;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionSkew_P.$t = /*@__PURE__*/ markType(UltraCalcFunctionSkew_P, 'UltraCalcFunctionSkew_P', BuiltInFunctionBase.$);
    return UltraCalcFunctionSkew_P;
})();
