// igniteui-react-excel/esm2015/lib/UltraCalcFunctionSLN.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionSLN = /*@__PURE__*/ (() => {
    class UltraCalcFunctionSLN extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop()._toInt321();
            if (c == 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(2));
            }
            let d, e;
            if (!((() => { let f = a.pop()._toDouble1(d); d = f.p0; return f.ret; })()) || !((() => { let f = a.pop()._toDouble1(e); e = f.p0; return f.ret; })())) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            return new ExcelCalcValue((e - d) / c);
        }
        get_name() {
            return "sln";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 3;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 3;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionSLN.$t = /*@__PURE__*/ markType(UltraCalcFunctionSLN, 'UltraCalcFunctionSLN', BuiltInFunctionBase.$);
    return UltraCalcFunctionSLN;
})();
