// igniteui-react-excel/esm2015/lib/UltraCalcFunctionYear.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionYear = /*@__PURE__*/ (() => {
    class UltraCalcFunctionYear extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            if (true) {
                let d;
                if (c.isNull || (((() => { let e = c._toDouble1(d); d = e.p0; return e.ret; })()) && d == 0)) {
                    let e = a._r;
                    if (e != null && e.dateSystem == 1) {
                        return new ExcelCalcValue(1904);
                    }
                    return new ExcelCalcValue(1900);
                }
            }
            let f = c._toDateTime1();
            return new ExcelCalcValue(f.getFullYear());
        }
        get_name() {
            return "year";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionYear.$t = /*@__PURE__*/ markType(UltraCalcFunctionYear, 'UltraCalcFunctionYear', BuiltInFunctionBase.$);
    return UltraCalcFunctionYear;
})();
