// igniteui-react-excel/esm2015/lib/UltraCalcFunctionStDevBase.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionStDevBase = /*@__PURE__*/ (() => {
    class UltraCalcFunctionStDevBase extends BuiltInFunctionBase {
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0x7FFFFFFF;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return (b == false);
        }
    }
    UltraCalcFunctionStDevBase.$t = /*@__PURE__*/ markType(UltraCalcFunctionStDevBase, 'UltraCalcFunctionStDevBase', BuiltInFunctionBase.$);
    return UltraCalcFunctionStDevBase;
})();
