// igniteui-react-excel/esm2015/lib/UltraCalcFunctionIsFormula.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionIsFormula = /*@__PURE__*/ (() => {
    class UltraCalcFunctionIsFormula extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            let d = false;
            if (c.isReference) {
                let e = c.toReference();
                let f = e.value;
                d = e.formula != null;
            }
            return new ExcelCalcValue(d);
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_name() {
            return "isformula";
        }
        get name() {
            return this.get_name();
        }
    }
    UltraCalcFunctionIsFormula.$t = /*@__PURE__*/ markType(UltraCalcFunctionIsFormula, 'UltraCalcFunctionIsFormula', BuiltInFunctionBase.$);
    return UltraCalcFunctionIsFormula;
})();
