// igniteui-react-excel/esm2015/lib/UltraCalcFunctionSum.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
/**
 * @hidden
 */
export let UltraCalcFunctionSum = /*@__PURE__*/ (() => {
    class UltraCalcFunctionSum extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = 0;
            return CalcFunctionUtilities.f(a, b, CalcFunctionUtilities.b(a, false, false, false), (d) => c += d) || new ExcelCalcValue(c);
        }
        canParameterBeEnumerable(a) {
            return true;
        }
        get_name() {
            return "sum";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0x7FFFFFFF;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return (b == false);
        }
    }
    UltraCalcFunctionSum.$t = /*@__PURE__*/ markType(UltraCalcFunctionSum, 'UltraCalcFunctionSum', BuiltInFunctionBase.$);
    return UltraCalcFunctionSum;
})();
