// igniteui-react-excel/esm2015/lib/UltraCalcFunctionOdd.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
import { ieeeRemainder } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionOdd = /*@__PURE__*/ (() => {
    class UltraCalcFunctionOdd extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c;
            let d = a.pop();
            if (d.isError) {
                return d;
            }
            if (((() => { let e = d._toDouble1(c); c = e.p0; return e.ret; })()) == false) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let e = Math.ceil(Math.abs(c));
            let f = Math.abs(ieeeRemainder(e, 2));
            if (f == 0) {
                e++;
            }
            if (c < 0) {
                e *= -1;
            }
            return new ExcelCalcValue(e);
        }
        get_name() {
            return "odd";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionOdd.$t = /*@__PURE__*/ markType(UltraCalcFunctionOdd, 'UltraCalcFunctionOdd', BuiltInFunctionBase.$);
    return UltraCalcFunctionOdd;
})();
