// igniteui-react-excel/esm2015/lib/UltraCalcFunctionEDate.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { CalcManagerUtilities } from "./CalcManagerUtilities";
import { markType } from "igniteui-react-core";
import { dateGetDate } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionEDate = /*@__PURE__*/ (() => {
    class UltraCalcFunctionEDate extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop()._toDouble3();
            let d = dateGetDate(a.pop()._toDateTime1());
            d = CalcManagerUtilities.b(2, c, d);
            return new ExcelCalcValue(d);
        }
        get_name() {
            return "edate";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionEDate.$t = /*@__PURE__*/ markType(UltraCalcFunctionEDate, 'UltraCalcFunctionEDate', BuiltInFunctionBase.$);
    return UltraCalcFunctionEDate;
})();
