// igniteui-react-excel/esm2015/lib/UltraCalcFunctionVar.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
export let UltraCalcFunctionVar = /*@__PURE__*/ (() => {
    class UltraCalcFunctionVar extends BuiltInFunctionBase {
        evaluate(a, b) {
            return CalcFunctionUtilities.d(a, b, 1229, (c) => UltraCalcFunctionVar._ak(c, true));
        }
        canParameterBeEnumerable(a) {
            return true;
        }
        static _ak(a, b) {
            if (a.count < 2) {
                if (!b && a.count == 1) {
                    return new ExcelCalcValue(0);
                }
                return new ExcelCalcValue(new ExcelCalcErrorValue(2));
            }
            return new ExcelCalcValue(CalcFunctionUtilities.ab(a, b));
        }
        get_name() {
            return "var";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0x7FFFFFFF;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return (b == false);
        }
    }
    UltraCalcFunctionVar.$t = /*@__PURE__*/ markType(UltraCalcFunctionVar, 'UltraCalcFunctionVar', BuiltInFunctionBase.$);
    return UltraCalcFunctionVar;
})();
