// igniteui-react-excel/esm2015/lib/UltraCalcFunctionHex2Bin.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { UltraCalcFunctionXBase2XBase } from "./UltraCalcFunctionXBase2XBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionHex2Bin = /*@__PURE__*/ (() => {
    class UltraCalcFunctionHex2Bin extends UltraCalcFunctionXBase2XBase {
        get_name() {
            return "hex2bin";
        }
        get name() {
            return this.get_name();
        }
        get__aq() {
            return -549755813888;
        }
        get _aq() {
            return this.get__aq();
        }
        get__ap() {
            return 0x7FFFFFFFFF;
        }
        get _ap() {
            return this.get__ap();
        }
        get__ao() {
            return -512;
        }
        get _ao() {
            return this.get__ao();
        }
        get__an() {
            return 511;
        }
        get _an() {
            return this.get__an();
        }
        get__al() {
            return 16;
        }
        get _al() {
            return this.get__al();
        }
        get__ak() {
            return 2;
        }
        get _ak() {
            return this.get__ak();
        }
        get__am() {
            return 10;
        }
        get _am() {
            return this.get__am();
        }
    }
    UltraCalcFunctionHex2Bin.$t = /*@__PURE__*/ markType(UltraCalcFunctionHex2Bin, 'UltraCalcFunctionHex2Bin', UltraCalcFunctionXBase2XBase.$);
    return UltraCalcFunctionHex2Bin;
})();
