// igniteui-react-excel/esm2015/lib/UltraCalcFunctionRate.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
export let UltraCalcFunctionRate = /*@__PURE__*/ (() => {
    class UltraCalcFunctionRate extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = 0;
            let d = 0;
            let e = 0;
            let f = 0;
            let g = 0;
            let h = 0.1;
            if (b < 3 || b > 6) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(1));
            }
            if (b == 6 && !((() => { let i = a.pop()._toDouble1(h); h = i.p0; return i.ret; })())) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            if (b >= 5) {
                g = a.pop()._toInt();
                if (g < 0 || g > 1) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(1));
                }
            }
            if (b >= 4 && !((() => { let i = a.pop()._toDouble1(c); c = i.p0; return i.ret; })())) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            if (b >= 3) {
                if (!((() => { let i = a.pop()._toDouble1(d); d = i.p0; return i.ret; })()) || !((() => { let i = a.pop()._toDouble1(e); e = i.p0; return i.ret; })()) || !((() => { let i = a.pop()._toDouble1(f); f = i.p0; return i.ret; })())) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(4));
                }
                if (f <= 0) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(1));
                }
            }
            let i = (j) => c + e * (1 + j * g) * (Math.pow(1 + j, f) - 1) / j + d * Math.pow(1 + j, f);
            let j = (k) => (e * (f * k * Math.pow(1 + k, f - 1 + g) - Math.pow(1 + k, f) + 1) / (k * k)) + f * d * Math.pow(1 + k, f - 1);
            let k = 1E-08;
            let l = h;
            for (let m = 0; m < 40; m++) {
                let n = i(l);
                let o = j(l);
                if (Math.abs(o) <= 4.94065645841247E-324) {
                    break;
                }
                let p = l - n / o;
                if (Math.abs(p - l) <= k) {
                    return new ExcelCalcValue(p);
                }
                l = p;
            }
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        get_name() {
            return "rate";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 3;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 6;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionRate.$t = /*@__PURE__*/ markType(UltraCalcFunctionRate, 'UltraCalcFunctionRate', BuiltInFunctionBase.$);
    return UltraCalcFunctionRate;
})();
