// igniteui-react-excel/esm2015/lib/UltraCalcFunctionIfs.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { List$1 } from "igniteui-react-core";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionIfs = /*@__PURE__*/ (() => {
    class UltraCalcFunctionIfs extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = new List$1(ExcelCalcValue.$, 0);
            while (b > 0) {
                c.add(a.pop());
                b--;
            }
            for (let d = c.count - 1; d > 0; d -= 2) {
                let e = c._inner[d];
                let f = c._inner[d - 1];
                if (e.isError) {
                    return new ExcelCalcValue(e.toErrorValue());
                }
                if (e._toBoolean1() == true) {
                    return f;
                }
            }
            return new ExcelCalcValue(new ExcelCalcErrorValue(3));
        }
        get_name() {
            return "ifs";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0x7FFFFFFF;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return (b == false);
        }
        get__s() {
            return true;
        }
        _v(a) {
            if (a % 2 == 1) {
                return false;
            }
            return true;
        }
    }
    UltraCalcFunctionIfs.$t = /*@__PURE__*/ markType(UltraCalcFunctionIfs, 'UltraCalcFunctionIfs', BuiltInFunctionBase.$);
    return UltraCalcFunctionIfs;
})();
