// igniteui-react-excel/esm2015/lib/UltraCalcFunctionSumIfs.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { UltraCalcConditionalFunctionBase } from "./UltraCalcConditionalFunctionBase";
/**
 * @hidden
 */
export let UltraCalcFunctionSumIfs = /*@__PURE__*/ (() => {
    class UltraCalcFunctionSumIfs extends BuiltInFunctionBase {
        evaluate(a, b) {
            return UltraCalcConditionalFunctionBase._ak(a, b, 0, (c, d) => new ExcelCalcValue(c), (c, d) => c + d);
        }
        get_name() {
            return "sumifs";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 3;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0x7FFFFFFF;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionSumIfs.$t = /*@__PURE__*/ markType(UltraCalcFunctionSumIfs, 'UltraCalcFunctionSumIfs', BuiltInFunctionBase.$);
    return UltraCalcFunctionSumIfs;
})();
