// igniteui-react-excel/esm2015/lib/UltraCalcFunctionCount.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionCount = /*@__PURE__*/ (() => {
    class UltraCalcFunctionCount extends BuiltInFunctionBase {
        evaluate(a, b) {
            return CalcFunctionUtilities.k(a, b, 1485);
        }
        canParameterBeEnumerable(a) {
            return true;
        }
        get_name() {
            return "count";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0x7FFFFFFF;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionCount.$t = /*@__PURE__*/ markType(UltraCalcFunctionCount, 'UltraCalcFunctionCount', BuiltInFunctionBase.$);
    return UltraCalcFunctionCount;
})();
