// igniteui-react-excel/esm2015/lib/UltraCalcFunctionIfNa.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { typeCast, markType } from "igniteui-react-core";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
export let UltraCalcFunctionIfNa = /*@__PURE__*/ (() => {
    class UltraCalcFunctionIfNa extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            let d = a.pop();
            let e = d._getResolvedValue();
            if (typeCast(ExcelCalcErrorValue.$, e) !== null && e.code == 3) {
                return c;
            }
            return new ExcelCalcValue(e);
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_name() {
            return "ifna";
        }
        get name() {
            return this.get_name();
        }
    }
    UltraCalcFunctionIfNa.$t = /*@__PURE__*/ markType(UltraCalcFunctionIfNa, 'UltraCalcFunctionIfNa', BuiltInFunctionBase.$);
    return UltraCalcFunctionIfNa;
})();
