// igniteui-react-excel/esm2015/lib/UltraCalcFunctionType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionType = /*@__PURE__*/ (() => {
    class UltraCalcFunctionType extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c;
            let d = a.pop();
            if (d.isError) {
                c = 16;
            }
            else if (d.isBoolean) {
                c = 4;
            }
            else if (d.isString) {
                c = 2;
            }
            else if (d.isArray) {
                c = 64;
            }
            else {
                let e;
                if (!((() => { let f = d._toDouble1(e); e = f.p0; return f.ret; })())) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(1));
                }
                c = 1;
            }
            return new ExcelCalcValue(c);
        }
        get_name() {
            return "type";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionType.$t = /*@__PURE__*/ markType(UltraCalcFunctionType, 'UltraCalcFunctionType', BuiltInFunctionBase.$);
    return UltraCalcFunctionType;
})();
