// igniteui-react-excel/esm2015/lib/UltraCalcFunctionN.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { typeCast, Date_$type, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionN = /*@__PURE__*/ (() => {
    class UltraCalcFunctionN extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let d;
            if (c.isBoolean) {
                d = c._toBoolean1() ? 1 : 0;
                return new ExcelCalcValue(d);
            }
            let e = c._getResolvedValue();
            if (typeCast(Date_$type, e) !== null) {
                return new ExcelCalcValue(e);
            }
            if (typeof e === 'string') {
                return new ExcelCalcValue(0);
            }
            let f = ((() => { let g = c._toDouble1(d); d = g.p0; return g.ret; })());
            if (f) {
                return new ExcelCalcValue(d);
            }
            return new ExcelCalcValue(0);
        }
        get_name() {
            return "n";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return (b == false);
        }
    }
    UltraCalcFunctionN.$t = /*@__PURE__*/ markType(UltraCalcFunctionN, 'UltraCalcFunctionN', BuiltInFunctionBase.$);
    return UltraCalcFunctionN;
})();
