// igniteui-react-excel/esm2015/lib/UltraCalcFunctionHex2Dec.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { UltraCalcFunctionXBase2Dec } from "./UltraCalcFunctionXBase2Dec";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionHex2Dec = /*@__PURE__*/ (() => {
    class UltraCalcFunctionHex2Dec extends UltraCalcFunctionXBase2Dec {
        get_name() {
            return "hex2dec";
        }
        get name() {
            return this.get_name();
        }
        get__ao() {
            return -549755813888;
        }
        get _ao() {
            return this.get__ao();
        }
        get__an() {
            return 0x7FFFFFFFFF;
        }
        get _an() {
            return this.get__an();
        }
        get__ak() {
            return 16;
        }
        get _ak() {
            return this.get__ak();
        }
        get__am() {
            return 10;
        }
        get _am() {
            return this.get__am();
        }
    }
    UltraCalcFunctionHex2Dec.$t = /*@__PURE__*/ markType(UltraCalcFunctionHex2Dec, 'UltraCalcFunctionHex2Dec', UltraCalcFunctionXBase2Dec.$);
    return UltraCalcFunctionHex2Dec;
})();
