// igniteui-react-excel/esm2015/lib/UltraCalcFunctionTrunc.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionTrunc = /*@__PURE__*/ (() => {
    class UltraCalcFunctionTrunc extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = 0;
            if (b == 2) {
                let d = a.pop();
                if (d.isError) {
                    return new ExcelCalcValue(d.toErrorValue());
                }
                c = d._toInt321();
            }
            let e;
            let f = a.pop();
            if (f.isError) {
                return new ExcelCalcValue(f.toErrorValue());
            }
            if (!((() => { let g = f._toDouble1(e); e = g.p0; return g.ret; })())) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let g = Math.pow(10, c);
            let h = e * g;
            h = h >= 0 ? Math.floor(h) : Math.ceil(h);
            h /= g;
            return new ExcelCalcValue(h);
        }
        get_name() {
            return "trunc";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionTrunc.$t = /*@__PURE__*/ markType(UltraCalcFunctionTrunc, 'UltraCalcFunctionTrunc', BuiltInFunctionBase.$);
    return UltraCalcFunctionTrunc;
})();
