// igniteui-react-excel/esm2015/lib/UltraCalcFunctionFormulaText.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { ArrayInteriorFormula } from "./ArrayInteriorFormula";
import { typeCast, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionFormulaText = /*@__PURE__*/ (() => {
    class UltraCalcFunctionFormulaText extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (!c.isReference) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(3));
            }
            let d = c.toReference();
            let e = d.value;
            let f = d.formula;
            if (f == null) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(3));
            }
            let g;
            if (typeCast(ArrayInteriorFormula.$, f) !== null) {
                g = "{" + f._c0._toString() + "}";
            }
            else {
                g = f.formulaString;
            }
            return new ExcelCalcValue(g);
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_name() {
            return "formulatext";
        }
        get name() {
            return this.get_name();
        }
    }
    UltraCalcFunctionFormulaText.$t = /*@__PURE__*/ markType(UltraCalcFunctionFormulaText, 'UltraCalcFunctionFormulaText', BuiltInFunctionBase.$);
    return UltraCalcFunctionFormulaText;
})();
