// igniteui-react-excel/esm2015/lib/UltraCalcFunctionStdev.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { UltraCalcFunctionVar } from "./UltraCalcFunctionVar";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
export let UltraCalcFunctionStdev = /*@__PURE__*/ (() => {
    class UltraCalcFunctionStdev extends UltraCalcFunctionVar {
        evaluate(a, b) {
            return CalcFunctionUtilities.d(a, b, 1229, (c) => UltraCalcFunctionStdev._al(c, true));
        }
        canParameterBeEnumerable(a) {
            return true;
        }
        static _al(a, b) {
            if (a.count < 2) {
                if (!b && a.count == 1) {
                    return new ExcelCalcValue(0);
                }
                return new ExcelCalcValue(new ExcelCalcErrorValue(2));
            }
            let c = CalcFunctionUtilities.r(a);
            return new ExcelCalcValue(CalcFunctionUtilities.y(c, a.count, b));
        }
        get_name() {
            return "stdev";
        }
        get_minArgs() {
            return 1;
        }
        get_maxArgs() {
            return 0x7FFFFFFF;
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return (b == false);
        }
    }
    UltraCalcFunctionStdev.$t = /*@__PURE__*/ markType(UltraCalcFunctionStdev, 'UltraCalcFunctionStdev', UltraCalcFunctionVar.$);
    return UltraCalcFunctionStdev;
})();
