// igniteui-react-excel/esm2015/lib/UltraCalcFunctionSumSq.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
/**
 * @hidden
 */
export let UltraCalcFunctionSumSq = /*@__PURE__*/ (() => {
    class UltraCalcFunctionSumSq extends BuiltInFunctionBase {
        evaluate(a, b) {
            return CalcFunctionUtilities.d(a, b, 1229, (c) => {
                if (c.count == 0) {
                    return new ExcelCalcValue(0);
                }
                let d = 0;
                for (let e = 0; e < c.count; e++) {
                    d += (c._inner[e] * c._inner[e]);
                }
                return new ExcelCalcValue(d);
            });
        }
        get_name() {
            return "sumsq";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0x7FFFFFFF;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionSumSq.$t = /*@__PURE__*/ markType(UltraCalcFunctionSumSq, 'UltraCalcFunctionSumSq', BuiltInFunctionBase.$);
    return UltraCalcFunctionSumSq;
})();
