// igniteui-react-excel/esm2015/lib/UltraCalcFunctionSYD.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionSYD = /*@__PURE__*/ (() => {
    class UltraCalcFunctionSYD extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop()._toInt321();
            if (c <= 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let d = a.pop()._toInt321();
            if (d <= 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let e, f;
            if (!((() => { let g = a.pop()._toDouble1(e); e = g.p0; return g.ret; })()) || !((() => { let g = a.pop()._toDouble1(f); f = g.p0; return g.ret; })())) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            return new ExcelCalcValue(((f - e) * (d - c + 1) * 2) / (d * (d + 1)));
        }
        get_name() {
            return "syd";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 4;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 4;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionSYD.$t = /*@__PURE__*/ markType(UltraCalcFunctionSYD, 'UltraCalcFunctionSYD', BuiltInFunctionBase.$);
    return UltraCalcFunctionSYD;
})();
