// igniteui-react-excel/esm2015/lib/UltraCalcFunctionNominal.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { truncate } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionNominal = /*@__PURE__*/ (() => {
    class UltraCalcFunctionNominal extends BuiltInFunctionBase {
        evaluate(a, b) {
            return CalcFunctionUtilities.j(a, false, (c, d) => {
                let e = truncate(d);
                let f = e;
                if (c <= 0 || f < 1) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(4));
                }
                let g = (Math.pow(c + 1, 1 / f) - 1) * f;
                return new ExcelCalcValue(g);
            });
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_name() {
            return "nominal";
        }
        get name() {
            return this.get_name();
        }
    }
    UltraCalcFunctionNominal.$t = /*@__PURE__*/ markType(UltraCalcFunctionNominal, 'UltraCalcFunctionNominal', BuiltInFunctionBase.$);
    return UltraCalcFunctionNominal;
})();
