// igniteui-react-excel/esm2015/lib/UltraCalcFunctionDec2Hex.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { UltraCalcFunctionDec2XBase } from "./UltraCalcFunctionDec2XBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionDec2Hex = /*@__PURE__*/ (() => {
    class UltraCalcFunctionDec2Hex extends UltraCalcFunctionDec2XBase {
        get_name() {
            return "dec2hex";
        }
        get name() {
            return this.get_name();
        }
        get__ap() {
            return -549755813888;
        }
        get _ap() {
            return this.get__ap();
        }
        get__al() {
            return 0x7FFFFFFFFF;
        }
        get _al() {
            return this.get__al();
        }
        get__ak() {
            return 16;
        }
        get _ak() {
            return this.get__ak();
        }
    }
    UltraCalcFunctionDec2Hex.$t = /*@__PURE__*/ markType(UltraCalcFunctionDec2Hex, 'UltraCalcFunctionDec2Hex', UltraCalcFunctionDec2XBase.$);
    return UltraCalcFunctionDec2Hex;
})();
