// igniteui-react-excel/esm2015/lib/UltraCalcFunctionIsErr.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionIsErr = /*@__PURE__*/ (() => {
    class UltraCalcFunctionIsErr extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            return new ExcelCalcValue(c.isError && c.toErrorValue().code != 3);
        }
        get_name() {
            return "iserr";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionIsErr.$t = /*@__PURE__*/ markType(UltraCalcFunctionIsErr, 'UltraCalcFunctionIsErr', BuiltInFunctionBase.$);
    return UltraCalcFunctionIsErr;
})();
