// igniteui-react-excel/esm2015/lib/UltraCalcFunctionSec.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
export let UltraCalcFunctionSec = /*@__PURE__*/ (() => {
    class UltraCalcFunctionSec extends BuiltInFunctionBase {
        evaluate(a, b) {
            return CalcFunctionUtilities.h(a, true, (c) => {
                if (c <= -134217728 || c >= 134217728) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(4));
                }
                return new ExcelCalcValue(1 / Math.cos(c));
            });
        }
        get_name() {
            return "sec";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionSec.$t = /*@__PURE__*/ markType(UltraCalcFunctionSec, 'UltraCalcFunctionSec', BuiltInFunctionBase.$);
    return UltraCalcFunctionSec;
})();
