// igniteui-react-excel/esm2015/lib/UltraCalcFunctionHyperlink.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { HyperlinkResult } from "./HyperlinkResult";
/**
 * @hidden
 */
export let UltraCalcFunctionHyperlink = /*@__PURE__*/ (() => {
    class UltraCalcFunctionHyperlink extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = null;
            if (b == 2) {
                c = a.pop();
            }
            let d = a.pop();
            if (d.isError) {
                return d;
            }
            if (c != null && c.isError) {
                return c;
            }
            let e = d.toString();
            let f = c == null ? e : c.toString();
            return new ExcelCalcValue(new HyperlinkResult(e, f));
        }
        get_name() {
            return "hyperlink";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionHyperlink.$t = /*@__PURE__*/ markType(UltraCalcFunctionHyperlink, 'UltraCalcFunctionHyperlink', BuiltInFunctionBase.$);
    return UltraCalcFunctionHyperlink;
})();
