// igniteui-react-excel/esm2015/lib/UltraCalcFunctionBitRShift.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { UltraCalcFunctionBitOperation } from "./UltraCalcFunctionBitOperation";
/**
 * @hidden
 */
export let UltraCalcFunctionBitRShift = /*@__PURE__*/ (() => {
    class UltraCalcFunctionBitRShift extends BuiltInFunctionBase {
        evaluate(a, b) {
            return CalcFunctionUtilities.j(a, true, (c, d) => UltraCalcFunctionBitOperation._ak(c, -d));
        }
        get_name() {
            return "bitrshift";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionBitRShift.$t = /*@__PURE__*/ markType(UltraCalcFunctionBitRShift, 'UltraCalcFunctionBitRShift', BuiltInFunctionBase.$);
    return UltraCalcFunctionBitRShift;
})();
