// igniteui-react-excel/esm2015/lib/UltraCalcFunctionComplex.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { UltraCalcFunctionComplex_ComplexNumber } from "./UltraCalcFunctionComplex_ComplexNumber";
import { stringCompare1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionComplex = /*@__PURE__*/ (() => {
    class UltraCalcFunctionComplex extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c;
            let d = (b == 3) ? a.pop().toString() : "i";
            if (stringCompare1(d, "i", 4) != 0 && stringCompare1(d, "j", 4) != 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(1));
            }
            c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let e = c._toDouble3();
            c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let f = c._toDouble3();
            let g = new UltraCalcFunctionComplex_ComplexNumber(0, f, e, d);
            let h = g.toString();
            return new ExcelCalcValue(h);
        }
        get_name() {
            return "complex";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 3;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionComplex.$t = /*@__PURE__*/ markType(UltraCalcFunctionComplex, 'UltraCalcFunctionComplex', BuiltInFunctionBase.$);
    return UltraCalcFunctionComplex;
})();
