// igniteui-react-excel/esm2015/lib/UltraCalcFunctionLeft.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionLeft = /*@__PURE__*/ (() => {
    class UltraCalcFunctionLeft extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = 1;
            if (b > 1) {
                let d = a.pop();
                if (d.isError) {
                    return new ExcelCalcValue(d.toErrorValue());
                }
                c = d._toInt321();
                if (c < 0) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(1));
                }
            }
            let e = a.pop();
            if (e.isError) {
                return new ExcelCalcValue(e.toErrorValue());
            }
            let f = e.toString();
            return new ExcelCalcValue(f.substr(0, c > f.length ? f.length : c));
        }
        get_name() {
            return "left";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return (b == false);
        }
    }
    UltraCalcFunctionLeft.$t = /*@__PURE__*/ markType(UltraCalcFunctionLeft, 'UltraCalcFunctionLeft', BuiltInFunctionBase.$);
    return UltraCalcFunctionLeft;
})();
