// igniteui-react-excel/esm2015/lib/UltraCalcFunctionDelta.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionDelta = /*@__PURE__*/ (() => {
    class UltraCalcFunctionDelta extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c;
            let d;
            if (b == 2) {
                c = a.pop();
                if (c.isError) {
                    return new ExcelCalcValue(c.toErrorValue());
                }
                d = c._toDouble3();
            }
            else {
                d = 0;
            }
            c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let e = c._toDouble3();
            let f = (e == d) ? 1 : 0;
            return new ExcelCalcValue(f);
        }
        get_name() {
            return "delta";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return (b == false);
        }
    }
    UltraCalcFunctionDelta.$t = /*@__PURE__*/ markType(UltraCalcFunctionDelta, 'UltraCalcFunctionDelta', BuiltInFunctionBase.$);
    return UltraCalcFunctionDelta;
})();
