// igniteui-react-excel/esm2015/lib/UltraCalcFunctionInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
export let UltraCalcFunctionInfo = /*@__PURE__*/ (() => {
    class UltraCalcFunctionInfo extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let d = c.toString();
            switch (d) {
                case "system": return new ExcelCalcValue("pcdos");
                case "numfile":
                case "origin":
                case "recalc":
                case "release": break;
                case "memavail":
                case "memused":
                case "totmem": break;
            }
            return new ExcelCalcValue(new ExcelCalcErrorValue(3));
        }
        get_name() {
            return "info";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionInfo.$t = /*@__PURE__*/ markType(UltraCalcFunctionInfo, 'UltraCalcFunctionInfo', BuiltInFunctionBase.$);
    return UltraCalcFunctionInfo;
})();
