// igniteui-react-excel/esm2015/lib/UltraCalcFunctionIsDBNull.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionIsDBNull = /*@__PURE__*/ (() => {
    class UltraCalcFunctionIsDBNull extends BuiltInFunctionBase {
        evaluate(a, b) {
            return new ExcelCalcValue(a.pop().isDBNull);
        }
        get_name() {
            return "isdbnull";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionIsDBNull.$t = /*@__PURE__*/ markType(UltraCalcFunctionIsDBNull, 'UltraCalcFunctionIsDBNull', BuiltInFunctionBase.$);
    return UltraCalcFunctionIsDBNull;
})();
