// igniteui-react-excel/esm2015/lib/UltraCalcFunctionNa.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionNa = /*@__PURE__*/ (() => {
    class UltraCalcFunctionNa extends BuiltInFunctionBase {
        evaluate(a, b) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(3));
        }
        get_name() {
            return "na";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 0;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionNa.$t = /*@__PURE__*/ markType(UltraCalcFunctionNa, 'UltraCalcFunctionNa', BuiltInFunctionBase.$);
    return UltraCalcFunctionNa;
})();
