// igniteui-react-excel/esm2015/lib/UltraCalcFunctionNPer.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { FinancialUtils } from "./FinancialUtils";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionNPer = /*@__PURE__*/ (() => {
    class UltraCalcFunctionNPer extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = 0;
            if (b == 5) {
                c = a.pop()._toInt321();
            }
            let d = 0;
            if (b >= 4 && !((() => { let e = a.pop()._toDouble1(d); d = e.p0; return e.ret; })())) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let e, f, g;
            if (!((() => { let h = a.pop()._toDouble1(e); e = h.p0; return h.ret; })()) || !((() => { let h = a.pop()._toDouble1(f); f = h.p0; return h.ret; })()) || f == 0 || !((() => { let h = a.pop()._toDouble1(g); g = h.p0; return h.ret; })())) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            if (g <= -1) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            if (g == 0 && f == 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let h = c;
            let i = FinancialUtils.nPer(g, f, e, d, h);
            return new ExcelCalcValue(i);
        }
        get_name() {
            return "nper";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 3;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 5;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionNPer.$t = /*@__PURE__*/ markType(UltraCalcFunctionNPer, 'UltraCalcFunctionNPer', BuiltInFunctionBase.$);
    return UltraCalcFunctionNPer;
})();
