// igniteui-react-excel/esm2015/lib/UltraCalcFunctionExact.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { stringEquals1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionExact = /*@__PURE__*/ (() => {
    class UltraCalcFunctionExact extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            let d = a.pop();
            if (d.isError) {
                return new ExcelCalcValue(d.toErrorValue());
            }
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let e;
            if (d.isString && c.isString) {
                e = stringEquals1(d.toString(), c.toString(), 0);
            }
            else {
                e = d.isSameValue(c);
            }
            return new ExcelCalcValue(e);
        }
        get_name() {
            return "exact";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionExact.$t = /*@__PURE__*/ markType(UltraCalcFunctionExact, 'UltraCalcFunctionExact', BuiltInFunctionBase.$);
    return UltraCalcFunctionExact;
})();
