// igniteui-react-excel/esm2015/lib/UltraCalcFunctionMultinomial.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { UltraCalcFunctionFact } from "./UltraCalcFunctionFact";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionMultinomial = /*@__PURE__*/ (() => {
    class UltraCalcFunctionMultinomial extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = this._getArguments(a, b, true);
            let d = 0;
            let e = 1;
            for (let f = 0; f < c.length; f++) {
                let g = c[f];
                if (g.isError) {
                    return new ExcelCalcValue(g.toErrorValue());
                }
                let h = g._toInt();
                if (h < 0) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(4));
                }
                d += h;
                e *= UltraCalcFunctionFact._ak(h);
            }
            let i = UltraCalcFunctionFact._ak(d);
            let j = i / e;
            return new ExcelCalcValue(j);
        }
        canParameterBeEnumerable(a) {
            return true;
        }
        get_name() {
            return "multinomial";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0x7FFFFFFF;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return (b == false);
        }
    }
    UltraCalcFunctionMultinomial.$t = /*@__PURE__*/ markType(UltraCalcFunctionMultinomial, 'UltraCalcFunctionMultinomial', BuiltInFunctionBase.$);
    return UltraCalcFunctionMultinomial;
})();
