// igniteui-react-excel/esm2015/lib/UltraCalcFunctionDateAdd.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
import { CalcManagerUtilities } from "./CalcManagerUtilities";
/**
 * @hidden
 */
export let UltraCalcFunctionDateAdd = /*@__PURE__*/ (() => {
    class UltraCalcFunctionDateAdd extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop()._toDateTime1();
            let d;
            if (!((() => { let e = a.pop()._toDouble1(d); d = e.p0; return e.ret; })())) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            let e = a.pop().toString().toLowerCase().trim();
            let f = CalcManagerUtilities.c(e, d, c);
            return new ExcelCalcValue(f);
        }
        get_name() {
            return "dateadd";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 3;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 3;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionDateAdd.$t = /*@__PURE__*/ markType(UltraCalcFunctionDateAdd, 'UltraCalcFunctionDateAdd', BuiltInFunctionBase.$);
    return UltraCalcFunctionDateAdd;
})();
