// igniteui-react-excel/esm2015/lib/UltraCalcFunctionReplace.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
export let UltraCalcFunctionReplace = /*@__PURE__*/ (() => {
    class UltraCalcFunctionReplace extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            let d = a.pop();
            let e = a.pop();
            let f = a.pop();
            if (f.isError) {
                return f;
            }
            if (e.isError) {
                return e;
            }
            if (d.isError) {
                return d;
            }
            if (c.isError) {
                return c;
            }
            let g = c.toString();
            let h = d._toInt321();
            let i = e._toInt321() - 1;
            let j = f.toString();
            if (i < 0 || i > j.length) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(1));
            }
            if (h < 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(1));
            }
            if (h > j.length - i) {
                h = j.length - i;
            }
            let k = j.substr(0, i) + g + j.substr(i + h);
            return new ExcelCalcValue(k);
        }
        get_name() {
            return "replace";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 4;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 4;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionReplace.$t = /*@__PURE__*/ markType(UltraCalcFunctionReplace, 'UltraCalcFunctionReplace', BuiltInFunctionBase.$);
    return UltraCalcFunctionReplace;
})();
