// igniteui-react-excel/esm2015/lib/UltraCalcFunctionDevSq.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
export let UltraCalcFunctionDevSq = /*@__PURE__*/ (() => {
    class UltraCalcFunctionDevSq extends BuiltInFunctionBase {
        evaluate(a, b) {
            return CalcFunctionUtilities.d(a, b, 1229, (c) => {
                if (c.count == 0) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(2));
                }
                else if (c.count == 1) {
                    return new ExcelCalcValue(0);
                }
                let d = CalcFunctionUtilities.r(c);
                return new ExcelCalcValue(d);
            });
        }
        get_name() {
            return "devsq";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0x7FFFFFFF;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionDevSq.$t = /*@__PURE__*/ markType(UltraCalcFunctionDevSq, 'UltraCalcFunctionDevSq', BuiltInFunctionBase.$);
    return UltraCalcFunctionDevSq;
})();
