// igniteui-react-excel/esm2015/lib/UltraCalcFunctionRoundDown.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { MathUtilities } from "./MathUtilities";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionRoundDown = /*@__PURE__*/ (() => {
    class UltraCalcFunctionRoundDown extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            let d = c._toInt();
            let e = a.pop();
            if (e.isError) {
                return new ExcelCalcValue(e.toErrorValue());
            }
            let f = e._toDouble3();
            let g = d;
            let h = Math.pow(10, g);
            f *= h;
            f = MathUtilities.i(f);
            f = MathUtilities.j(f);
            f /= h;
            f = MathUtilities.i(f);
            if (f == -0) {
                f = 0;
            }
            return new ExcelCalcValue(f);
        }
        get_name() {
            return "rounddown";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 2;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 2;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionRoundDown.$t = /*@__PURE__*/ markType(UltraCalcFunctionRoundDown, 'UltraCalcFunctionRoundDown', BuiltInFunctionBase.$);
    return UltraCalcFunctionRoundDown;
})();
