// igniteui-react-excel/esm2015/lib/UltraCalcFunctionAcotH.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
export let UltraCalcFunctionAcotH = /*@__PURE__*/ (() => {
    class UltraCalcFunctionAcotH extends BuiltInFunctionBase {
        evaluate(a, b) {
            return CalcFunctionUtilities.h(a, true, (c) => {
                if (Math.abs(c) <= 1) {
                    return new ExcelCalcValue(new ExcelCalcErrorValue(4));
                }
                return new ExcelCalcValue(Math.log((c + 1) / (c - 1)) / 2);
            });
        }
        get_name() {
            return "acoth";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionAcotH.$t = /*@__PURE__*/ markType(UltraCalcFunctionAcotH, 'UltraCalcFunctionAcotH', BuiltInFunctionBase.$);
    return UltraCalcFunctionAcotH;
})();
