// igniteui-react-excel/esm2015/lib/UltraCalcFunctionColumns.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionColumns = /*@__PURE__*/ (() => {
    class UltraCalcFunctionColumns extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isReference == false && c.isError) {
                return c;
            }
            let d = c._b(false);
            if (d.length != 1) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(0));
            }
            return new ExcelCalcValue(d[0].getLength(0));
        }
        get_name() {
            return "columns";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        _v(a) {
            return false;
        }
    }
    UltraCalcFunctionColumns.$t = /*@__PURE__*/ markType(UltraCalcFunctionColumns, 'UltraCalcFunctionColumns', BuiltInFunctionBase.$);
    return UltraCalcFunctionColumns;
})();
