// igniteui-react-excel/esm2015/lib/UltraCalcFunctionMonth.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { dateGetMonth } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionMonth = /*@__PURE__*/ (() => {
    class UltraCalcFunctionMonth extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            if (true) {
                let d;
                if (c.isNull || (((() => { let e = c._toDouble1(d); d = e.p0; return e.ret; })()) && d == 0)) {
                    return new ExcelCalcValue(1);
                }
            }
            let e = c._toDateTime1();
            return new ExcelCalcValue(dateGetMonth(e));
        }
        get_name() {
            return "month";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        doesParameterAllowIntermediateResultArray(a, b) {
            return true;
        }
    }
    UltraCalcFunctionMonth.$t = /*@__PURE__*/ markType(UltraCalcFunctionMonth, 'UltraCalcFunctionMonth', BuiltInFunctionBase.$);
    return UltraCalcFunctionMonth;
})();
