// igniteui-react-excel/esm2015/lib/UltraCalcFunctionMdeterm.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { MatrixUtilities } from "./MatrixUtilities";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionMdeterm = /*@__PURE__*/ (() => {
    class UltraCalcFunctionMdeterm extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isError) {
                return c;
            }
            let d = c.toArrayProxy();
            let e = d.getLength(0);
            if (e != d.getLength(1)) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(1));
            }
            let f;
            let g = ((() => { let h = MatrixUtilities.b(d, 0, f); f = h.p2; return h.ret; })());
            if (f != null) {
                return new ExcelCalcValue(f);
            }
            return new ExcelCalcValue(MatrixUtilities.g(g, e));
        }
        get_name() {
            return "mdeterm";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionMdeterm.$t = /*@__PURE__*/ markType(UltraCalcFunctionMdeterm, 'UltraCalcFunctionMdeterm', BuiltInFunctionBase.$);
    return UltraCalcFunctionMdeterm;
})();
