// igniteui-react-excel/esm2015/lib/UltraCalcFunctionAreas.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionAreas = /*@__PURE__*/ (() => {
    class UltraCalcFunctionAreas extends BuiltInFunctionBase {
        evaluate(a, b) {
            let c = a.pop();
            if (c.isReference == false) {
                if (c.isError) {
                    return c;
                }
                return new ExcelCalcValue(new ExcelCalcErrorValue(1));
            }
            return new ExcelCalcValue(c._b(false).length);
        }
        get_name() {
            return "areas";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 1;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 1;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
        _v(a) {
            return false;
        }
    }
    UltraCalcFunctionAreas.$t = /*@__PURE__*/ markType(UltraCalcFunctionAreas, 'UltraCalcFunctionAreas', BuiltInFunctionBase.$);
    return UltraCalcFunctionAreas;
})();
