// igniteui-react-excel/esm2015/lib/UltraCalcFunctionToday.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { dateToday } from "igniteui-react-core";
/**
 * @hidden
 */
export let UltraCalcFunctionToday = /*@__PURE__*/ (() => {
    class UltraCalcFunctionToday extends BuiltInFunctionBase {
        evaluate(a, b) {
            return new ExcelCalcValue(dateToday());
        }
        get_name() {
            return "today";
        }
        get name() {
            return this.get_name();
        }
        get_minArgs() {
            return 0;
        }
        get minArgs() {
            return this.get_minArgs();
        }
        get_maxArgs() {
            return 0;
        }
        get maxArgs() {
            return this.get_maxArgs();
        }
    }
    UltraCalcFunctionToday.$t = /*@__PURE__*/ markType(UltraCalcFunctionToday, 'UltraCalcFunctionToday', BuiltInFunctionBase.$);
    return UltraCalcFunctionToday;
})();
