// igniteui-react-excel/esm2015/lib/ValueFormatter_FormatSection_FormulaValueInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, markStruct } from "igniteui-react-core";
/**
 * @hidden
 */
export let ValueFormatter_FormatSection_FormulaValueInfo = /*@__PURE__*/ (() => {
    class ValueFormatter_FormatSection_FormulaValueInfo extends ValueType {
        constructor() {
            super();
            this.e = 0;
            this.a = false;
            this.b = false;
            this.f = 0;
            this.g = 0;
            this.d = '\0';
            this.c = false;
        }
    }
    ValueFormatter_FormatSection_FormulaValueInfo.$t = /*@__PURE__*/ markStruct(ValueFormatter_FormatSection_FormulaValueInfo, 'ValueFormatter_FormatSection_FormulaValueInfo');
    return ValueFormatter_FormatSection_FormulaValueInfo;
})();
