// igniteui-react-excel/esm2015/lib/VbaProjectPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { PartManagerBase } from "./PartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let VbaProjectPartManager = /*@__PURE__*/ (() => {
    class VbaProjectPartManager extends PartManagerBase {
        get_m() {
            return "application/vnd.ms-office.vbaProject";
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/vbaProject.bin";
                case 1: return "/word/vbaProject.bin";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return VbaProjectPartManager.u;
                case 1: return VbaProjectPartManager.u;
                default: return VbaProjectPartManager.u;
            }
        }
        get_g() {
            return false;
        }
        get g() {
            return this.get_g();
        }
    }
    VbaProjectPartManager.$t = /*@__PURE__*/ markType(VbaProjectPartManager, 'VbaProjectPartManager', PartManagerBase.$);
    VbaProjectPartManager.u = "http://schemas.microsoft.com/office/2006/relationships/vbaProject";
    return VbaProjectPartManager;
})();
