// igniteui-react-excel/esm2015/lib/VmlDrawingPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { SgmlReader } from "./SgmlReader";
import { StreamReader } from "igniteui-react-core";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let VmlDrawingPartManager = /*@__PURE__*/ (() => {
    class VmlDrawingPartManager extends XmlPartManagerBase {
        constructor(a) {
            super();
            this.a3 = false;
            this.a3 = a;
        }
        get_m() {
            return VmlDrawingPartManager.a7;
        }
        get m() {
            return this.get_m();
        }
        a1() {
            return ((() => {
                let $ret = new SgmlReader();
                $ret._b7 = new StreamReader(this.l);
                return $ret;
            })());
        }
        n(a) {
            if (this.a3) {
                switch (a) {
                    case 1: return "/theme/drawings/vmlDrawing.vml";
                    default: return null;
                }
            }
            else {
                switch (a) {
                    case 0: return "/xl/drawings/vmlDrawing.vml";
                    case 1: return "/word/drawings/vmlDrawing.vml";
                    default: return null;
                }
            }
        }
        o(a) {
            switch (a) {
                case 0:
                case 1: return VmlDrawingPartManager.a8;
                default: return VmlDrawingPartManager.a8;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    VmlDrawingPartManager.$t = /*@__PURE__*/ markType(VmlDrawingPartManager, 'VmlDrawingPartManager', XmlPartManagerBase.$);
    VmlDrawingPartManager.a7 = "application/vnd.openxmlformats-officedocument.vmlDrawing";
    VmlDrawingPartManager.a8 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/vmlDrawing";
    return VmlDrawingPartManager;
})();
