// igniteui-react-excel/esm2015/lib/WorksheetShapeCollection_SeriesShiftInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ValueType, markStruct } from "igniteui-react-core";
/**
 * @hidden
 */
export let WorksheetShapeCollection_SeriesShiftInfo = /*@__PURE__*/ (() => {
    class WorksheetShapeCollection_SeriesShiftInfo extends ValueType {
        constructor(a, ..._rest) {
            super();
            this.a = null;
            this.c = null;
            this.b = null;
            this.d = null;
            this.e = null;
            a = (a == void 0) ? 0 : a;
            switch (a) {
                case 0: break;
                case 1:
                    {
                        let c = _rest[0];
                        let d = _rest[1];
                        let e = _rest[2];
                        let f = _rest[3];
                        let g = _rest[4];
                        this.a = c;
                        this.c = d;
                        this.b = e;
                        this.d = f;
                        this.e = g;
                    }
                    break;
            }
        }
    }
    WorksheetShapeCollection_SeriesShiftInfo.$t = /*@__PURE__*/ markStruct(WorksheetShapeCollection_SeriesShiftInfo, 'WorksheetShapeCollection_SeriesShiftInfo');
    return WorksheetShapeCollection_SeriesShiftInfo;
})();
