// igniteui-react-excel/esm2015/lib/WorksheetTableStyleArea.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various areas which can have a format applied at the table style level.
 * <p class="note">
 * <b>Note:</b> Only certain properties can be set on the table style are formats. The are as follows:
 * <list type="bullet">
 * <item>[[IWorksheetCellFormat.bottomBorderColorInfo]]</item>
 * <item>[[IWorksheetCellFormat.bottomBorderStyle]]</item>
 * <item>[[IWorksheetCellFormat.fill]]</item>
 * <item>[[IWorksheetCellFormat.leftBorderColorInfo]]</item>
 * <item>[[IWorksheetCellFormat.leftBorderStyle]]</item>
 * <item>[[IWorksheetCellFormat.rightBorderColorInfo]]</item>
 * <item>[[IWorksheetCellFormat.rightBorderStyle]]</item>
 * <item>[[IWorksheetCellFormat.topBorderColorInfo]]</item>
 * <item>[[IWorksheetCellFormat.topBorderStyle]]</item>
 * <item><see cref="IWorkbookFont.Bold">Font.Bold</see></item>
 * <item><see cref="IWorkbookFont.ColorInfo">Font.ColorInfo</see></item>
 * <item><see cref="IWorkbookFont.Italic">Font.Italic</see></item>
 * <item><see cref="IWorkbookFont.Strikeout">Font.Strikeout</see></item>
 * <item><see cref="IWorkbookFont.UnderlineStyle">Font.UnderlineStyle</see></item>
 * </list>
 * Setting any other format or font properties on the table style areas will cause an exception.
 * </p>
 * @see [[WorksheetTableStyle.areaFormats]]
 */
export var WorksheetTableStyleArea = /*@__PURE__*/ (function (WorksheetTableStyleArea) {
    /**
     * The format is applied to the entire table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["WholeTable"] = 0] = "WholeTable";
    /**
     * The format is applied to the odd numbered column stripes in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["ColumnStripe"] = 7] = "ColumnStripe";
    /**
     * The format is applied to the even numbered column stripes in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["AlternateColumnStripe"] = 8] = "AlternateColumnStripe";
    /**
     * The format is applied to the odd numbered row stripes in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["RowStripe"] = 5] = "RowStripe";
    /**
     * The format is applied to the even numbered row stripes in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["AlternateRowStripe"] = 6] = "AlternateRowStripe";
    /**
     * The format is applied to the last column in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["LastColumn"] = 4] = "LastColumn";
    /**
     * The format is applied to the first column in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["FirstColumn"] = 3] = "FirstColumn";
    /**
     * The format is applied to the header row in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["HeaderRow"] = 1] = "HeaderRow";
    /**
     * The format is applied to the totals row in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["TotalRow"] = 2] = "TotalRow";
    /**
     * The format is applied to the first header cell in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["FirstHeaderCell"] = 9] = "FirstHeaderCell";
    /**
     * The format is applied to the last header cell in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["LastHeaderCell"] = 10] = "LastHeaderCell";
    /**
     * The format is applied to the first total cell in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["FirstTotalCell"] = 11] = "FirstTotalCell";
    /**
     * The format is applied to the last total cell in the table.
     */
    WorksheetTableStyleArea[WorksheetTableStyleArea["LastTotalCell"] = 12] = "LastTotalCell";
    return WorksheetTableStyleArea;
})({});
/**
 * @hidden
 */
export let WorksheetTableStyleArea_$type = /*@__PURE__*/ markEnum('WorksheetTableStyleArea', 'WholeTable,0|ColumnStripe,7|AlternateColumnStripe,8|RowStripe,5|AlternateRowStripe,6|LastColumn,4|FirstColumn,3|HeaderRow,1|TotalRow,2|FirstHeaderCell,9|LastHeaderCell,10|FirstTotalCell,11|LastTotalCell,12');
