// igniteui-react-excel/esm2015/lib/WorksheetTableArea.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various areas which can have a format applied at the table level.
 * @see [[WorksheetTable.areaFormats]]
 */
export var WorksheetTableArea = /*@__PURE__*/ (function (WorksheetTableArea) {
    /**
     * The format is applied to the entire table. Only the outer border properties can be set on this area format.
     * Setting any other will cause an exception.
     */
    WorksheetTableArea[WorksheetTableArea["WholeTable"] = 0] = "WholeTable";
    /**
     * The format is applied to the data area of the table.
     */
    WorksheetTableArea[WorksheetTableArea["DataArea"] = 1] = "DataArea";
    /**
     * The format is applied to the header row of the table. All but the top border properties can be set on this
     * area format. Setting the top border properties will cause an exception.
     */
    WorksheetTableArea[WorksheetTableArea["HeaderRow"] = 2] = "HeaderRow";
    /**
     * The format is applied to the totals row of the table. All but the bottom border properties can be set on this
     * area format. Setting the bottom border properties will cause an exception.
     */
    WorksheetTableArea[WorksheetTableArea["TotalsRow"] = 3] = "TotalsRow";
    return WorksheetTableArea;
})({});
/**
 * @hidden
 */
export let WorksheetTableArea_$type = /*@__PURE__*/ markEnum('WorksheetTableArea', 'WholeTable,0|DataArea,1|HeaderRow,2|TotalsRow,3');
