// igniteui-react-excel/esm2015/lib/WorkbookColorTransform.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, Number_$type, toNullable, markType } from "igniteui-react-core";
import { wrapNullable, unwrapNullable, nullableEquals } from "igniteui-react-core";
/**
 * Defines color transforms for a [[WorkbookColorInfo]] instance.
 * <p class="body">
 * This class is immutable.
 * </p>
 * <p class="body">
 * Color transforms are not applicable to all contexts within which a WorkbookColorInfo is used;
 * currently they are only used by <see cref="WorksheetChart">charts</see>.
 * </p>
 * @see [[ChartSolidFill]]
 * @see [[GradientStop]]
 * @see [[ChartGradientFill]]
 */
export let WorkbookColorTransform = /*@__PURE__*/ (() => {
    class WorkbookColorTransform extends Base {
        constructor(..._rest) {
            super();
            this._alpha$i_1 = toNullable(Number_$type, null);
            this._luminanceModulation$i_1 = toNullable(Number_$type, null);
            this._luminanceOffset$i_1 = toNullable(Number_$type, null);
            this._shade$i_1 = toNullable(Number_$type, null);
            let alpha = _rest[0];
            let luminanceModulation = _rest[1];
            let luminanceOffset = _rest[2];
            let shade = _rest[3];
            if (alpha === void 0) {
                alpha = null;
            }
            alpha = wrapNullable(Number_$type, alpha);
            if (luminanceModulation === void 0) {
                luminanceModulation = null;
            }
            luminanceModulation = wrapNullable(Number_$type, luminanceModulation);
            if (luminanceOffset === void 0) {
                luminanceOffset = null;
            }
            luminanceOffset = wrapNullable(Number_$type, luminanceOffset);
            if (shade === void 0) {
                shade = null;
            }
            shade = wrapNullable(Number_$type, shade);
            this._alpha$i = alpha;
            this._luminanceModulation$i = luminanceModulation;
            this._luminanceOffset$i = luminanceOffset;
            this._shade$i = shade;
        }
        /**
         * Defines the value of the alpha channel for the associated
         * [[WorkbookColorInfo]] instance, expressed
         * as a number between 0 and 1.
         * <p class="body">
         * Use this property to control the transparency of the associated color.
         * </p>
         * <p class="body">
         * For example, to make the associated color 50% transparent, assign a value of <b>.5</b> to this property.
         * </p>
         * <p class="body">
         * Assigning a value of null to this property implies that the associated
         * color has no specific alpha channel setting, i.e., the associated color
         * is fully opaque.
         * </p>
         */
        get alpha() {
            return unwrapNullable(this._alpha$i);
        }
        set alpha(a) {
            this._alpha$i = arguments[0];
        }
        /**
         * @hidden
         */
        get _alpha$i() {
            return this._alpha$i_1;
        }
        /**
         * @hidden
         */
        set _alpha$i(a) {
            this._alpha$i_1 = a;
        }
        /**
         * Defines the luminance modulation for the associated
         * [[WorkbookColorInfo]] instance, expressed
         * as a fractional number.
         * <p class="body">
         * Luminance is also referred to as brightness.
         * </p>
         * <p class="body">
         * This property specifies the amount by which the luminance of the associated
         * base color is increased. For example, a value of .5 yields a 50% reduction
         * in luminance. A value of 1.0 implies no change in luminance. A value greater
         * than 1.0 implies an increase in luminance; for example, a value of 2.0 doubles
         * the luminance.
         * </p>
         * <p class="body">
         * Assigning a value of null to this property implies that the associated color has no specific luminance modulation.
         * </p>
         * <p class="body">
         * Luminance modulation is applied by converting the base color from RGB to HSL, then
         * multiplying the luminance channel by the value of this property, then converting
         * back to RGB.
         * </p>
         * @see [[luminanceOffset]]
         */
        get luminanceModulation() {
            return unwrapNullable(this._luminanceModulation$i);
        }
        set luminanceModulation(a) {
            this._luminanceModulation$i = arguments[0];
        }
        /**
         * @hidden
         */
        get _luminanceModulation$i() {
            return this._luminanceModulation$i_1;
        }
        /**
         * @hidden
         */
        set _luminanceModulation$i(a) {
            this._luminanceModulation$i_1 = a;
        }
        /**
         * Defines the luminance offset for the associated
         * [[WorkbookColorInfo]] instance, expressed
         * as a fractional number.
         * <p class="body">
         * Luminance is also referred to as brightness.
         * </p>
         * <p class="body">
         * The luminance offset is applied by converting the base color from RGB to HSL, then
         * the value of this property to the luminance channel, then converting back to RGB.
         * </p>
         * <p class="body">
         * Assigning a value of null to this property implies that the associated color has no specific luminance offset.
         * </p>
         * @see [[luminanceModulation]]
         */
        get luminanceOffset() {
            return unwrapNullable(this._luminanceOffset$i);
        }
        set luminanceOffset(a) {
            this._luminanceOffset$i = arguments[0];
        }
        /**
         * @hidden
         */
        get _luminanceOffset$i() {
            return this._luminanceOffset$i_1;
        }
        /**
         * @hidden
         */
        set _luminanceOffset$i(a) {
            this._luminanceOffset$i_1 = a;
        }
        /**
         * Defines the shading for the associated
         * [[WorkbookColorInfo]] instance, expressed
         * as a fractional number.
         * @see [[alpha]]
         */
        get shade() {
            return unwrapNullable(this._shade$i);
        }
        set shade(a) {
            this._shade$i = arguments[0];
        }
        /**
         * @hidden
         */
        get _shade$i() {
            return this._shade$i_1;
        }
        /**
         * @hidden
         */
        set _shade$i(a) {
            this._shade$i_1 = a;
        }
        /**
         * @hidden
         */
        _b() {
            return this._alpha$i.hasValue || this._luminanceModulation$i.hasValue || this._luminanceOffset$i.hasValue || this._shade$i.hasValue;
        }
        /**
         * @hidden
         */
        static _a(a, b) {
            if (a == null && b == null) {
                return true;
            }
            if (a == null || b == null) {
                return false;
            }
            return nullableEquals(a._alpha$i, b._alpha$i) && nullableEquals(a._luminanceModulation$i, b._luminanceModulation$i) && nullableEquals(a._luminanceOffset$i, b._luminanceOffset$i) && nullableEquals(a._shade$i, b._shade$i);
        }
    }
    WorkbookColorTransform.$t = /*@__PURE__*/ markType(WorkbookColorTransform, 'WorkbookColorTransform');
    /**
     * @hidden
     */
    WorkbookColorTransform._c = 100000;
    return WorkbookColorTransform;
})();
