// igniteui-react-excel/esm2015/lib/WorkbookLoadManagerExcel2007_FontCollection.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, String_$type, Number_$type, markType } from "igniteui-react-core";
import { StringComparer } from "igniteui-react-core";
import { Dictionary$2 } from "igniteui-react-core";
import { ExcelUtils } from "./ExcelUtils";
import { stringIsNullOrEmpty } from "igniteui-react-core";
/**
 * @hidden
 */
export let WorkbookLoadManagerExcel2007_FontCollection = /*@__PURE__*/ (() => {
    class WorkbookLoadManagerExcel2007_FontCollection extends Base {
        constructor() {
            super();
            this.b = null;
            this.c = null;
            this.d = null;
            this.b = new Dictionary$2(String_$type, String_$type, 2, StringComparer.c);
            this.c = new Dictionary$2(Number_$type, String_$type, 0);
        }
        f(a, b) {
            this.b.item(a, b);
            this.c.addItem(this.c.count, a);
        }
        e() {
            let a = null;
            if (ExcelUtils.ax) {
                let b = this.b.tryGetValue("jpan", a);
                a = b.p1;
            }
            if (a != null) {
                return a;
            }
            return this.d;
        }
        a(a, b, c) {
            b = null;
            c = null;
            if (((() => { let d = this.c.tryGetValue(a, b); b = d.p1; return d.ret; })())) {
                c = this.b.item(b);
            }
            return {
                ret: stringIsNullOrEmpty(c) == false,
                p1: b,
                p2: c
            };
        }
    }
    WorkbookLoadManagerExcel2007_FontCollection.$t = /*@__PURE__*/ markType(WorkbookLoadManagerExcel2007_FontCollection, 'WorkbookLoadManagerExcel2007_FontCollection');
    return WorkbookLoadManagerExcel2007_FontCollection;
})();
