// igniteui-react-excel/esm2015/lib/WorksheetColumnWidthUnit.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various units in which a column width can be represented.
 * @see [[Worksheet.getDefaultColumnWidth]]
 * @see [[Worksheet.setDefaultColumnWidth]]
 * @see [[WorksheetColumn.getWidth]]
 * @see [[WorksheetColumn.setWidth]]
 */
export var WorksheetColumnWidthUnit = /*@__PURE__*/ (function (WorksheetColumnWidthUnit) {
    /**
     * The column width is represented in units of the '0' digit character width, including column padding. The digit is
     * measured with the default font for the workbook. The padding is a few pixels on either side of the column plus an
     * additional pixel for the gridline.
     */
    WorksheetColumnWidthUnit[WorksheetColumnWidthUnit["Character"] = 0] = "Character";
    /**
     * The column width is represented in 256ths of the '0' digit character width, including column padding, which means this
     * value will be 256 times the width expressed in Character units. The digit is measured with the default font for the
     * workbook. The padding is a few pixels on either side of the column plus an additional pixel for the gridline. These
     * units are the units in which the [[WorksheetColumn.width]] and [[Worksheet.defaultColumnWidth]]
     * properties are expressed.
     */
    WorksheetColumnWidthUnit[WorksheetColumnWidthUnit["Character256th"] = 1] = "Character256th";
    /**
     * The column width is represented in units of the '0' digit character width, excluding padding. The digit is measured with
     * the default font for the workbook. These units are the units in which Microsoft Excel displays column widths to the user
     * and accepts new column widths from the user in the 'Column Width' dialog.
     */
    WorksheetColumnWidthUnit[WorksheetColumnWidthUnit["CharacterPaddingExcluded"] = 2] = "CharacterPaddingExcluded";
    /**
     * The column width is represented in pixels.
     */
    WorksheetColumnWidthUnit[WorksheetColumnWidthUnit["Pixel"] = 3] = "Pixel";
    /**
     * The column width is represented in points.
     */
    WorksheetColumnWidthUnit[WorksheetColumnWidthUnit["Point"] = 4] = "Point";
    /**
     * The column width is represented in twips (20ths of a point).
     */
    WorksheetColumnWidthUnit[WorksheetColumnWidthUnit["Twip"] = 5] = "Twip";
    return WorksheetColumnWidthUnit;
})({});
/**
 * @hidden
 */
export let WorksheetColumnWidthUnit_$type = /*@__PURE__*/ markEnum('WorksheetColumnWidthUnit', 'Character,0|Character256th,1|CharacterPaddingExcluded,2|Pixel,3|Point,4|Twip,5');
