// igniteui-react-excel/esm2015/lib/WorkItem_WorkItemCatch$1.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WorkItem } from "./WorkItem";
import { typeCast, markType } from "igniteui-react-core";
import { WorkItem_ExceptionInfo } from "./WorkItem_ExceptionInfo";
/**
 * @hidden
 */
export let WorkItem_WorkItemCatch$1 = /*@__PURE__*/ (() => {
    class WorkItem_WorkItemCatch$1 extends WorkItem {
        constructor($tException, a, b, c) {
            super(1, b, true);
            this.$tException = null;
            this.ad = null;
            this.ae = false;
            this.$tException = $tException;
            this.$type = this.$type.specialize(this.$tException);
            this.ad = a;
            this.ae = c;
        }
        get_s() {
            return this.ae;
        }
        p(a) {
            return true;
        }
        u(a) {
            let b = typeCast(this.$tException, a.b);
            if (b != null) {
                try {
                    this.ad(b);
                    a = null;
                }
                catch (c) {
                    a = new WorkItem_ExceptionInfo(c);
                }
                return {
                    ret: true,
                    p0: a
                };
            }
            return {
                ret: false,
                p0: a
            };
        }
    }
    WorkItem_WorkItemCatch$1.$t = /*@__PURE__*/ markType(WorkItem_WorkItemCatch$1, 'WorkItem_WorkItemCatch$1', WorkItem.$);
    return WorkItem_WorkItemCatch$1;
})();
