// igniteui-react-excel/esm2015/lib/WorkbookPasswordInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, toNullable, Number_$type, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let WorkbookPasswordInfo = /*@__PURE__*/ (() => {
    class WorkbookPasswordInfo extends Base {
        constructor(a) {
            super();
            this.i = null;
            this.a = null;
            this.d = false;
            this.c = null;
            this.c = a;
        }
        get f() {
            if (this.b != null) {
                return true;
            }
            return this.i != null;
        }
        get g() {
            return this.d;
        }
        set g(a) {
            if (a != this.d) {
                this.d = a;
                if (this.c != null) {
                    this.c();
                }
            }
        }
        get b() {
            return this.a;
        }
        set b(a) {
            if (a != this.a) {
                this.a = a;
                if (this.d == false) {
                    this.g = true;
                }
                else if (this.f == false) {
                    this.g = false;
                }
            }
        }
        e(a, b, c, d) {
            if (this.a != null) {
                return {
                    ret: ((() => { let e = this.a.d(a, b, c, d); a = e.p0; b = e.p1; c = e.p2; d = e.p3; return e.ret; })()),
                    p0: a,
                    p1: b,
                    p2: c,
                    p3: d
                };
            }
            a = null;
            b = c = null;
            d = toNullable(Number_$type, null);
            return {
                ret: false,
                p0: a,
                p1: b,
                p2: c,
                p3: d
            };
        }
        h() {
            if (this.a != null) {
                return this.a.f();
            }
            return toNullable(Number_$type, null);
        }
        j(a) {
            this.d = a;
        }
        k(a) {
            if (this.g == false) {
                this.b = null;
                this.g = true;
            }
        }
        l() {
            this.i = null;
            this.a = null;
            this.g = false;
        }
    }
    WorkbookPasswordInfo.$t = /*@__PURE__*/ markType(WorkbookPasswordInfo, 'WorkbookPasswordInfo');
    return WorkbookPasswordInfo;
})();
