// igniteui-react-excel/esm2015/lib/WorkbookSaveManagerExcel2003_SaveTokenVisitor.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { FormulaTokenVisitor } from "./FormulaTokenVisitor";
import { WorkbookSaveManagerExcel2003_GetSizeTokenVisitor } from "./WorkbookSaveManagerExcel2003_GetSizeTokenVisitor";
import { markType } from "igniteui-react-core";
import { intSToU } from "igniteui-react-core";
/**
 * @hidden
 */
export let WorkbookSaveManagerExcel2003_SaveTokenVisitor = /*@__PURE__*/ (() => {
    class WorkbookSaveManagerExcel2003_SaveTokenVisitor extends FormulaTokenVisitor {
        constructor() {
            super();
            this.a6 = false;
            this.a1 = null;
            this.a3 = null;
            this.a2 = null;
        }
        e(a) {
        }
        g() {
            this.a6 = false;
            this.a1 = null;
            this.a3 = null;
            this.a2 = null;
            super.g();
        }
        h() {
            let a = this.a1.length;
            for (this.d = 0; this.d < a; this.d++) {
                let b = this.a1[this.d];
                this.a3._write9(b.f);
                b.m(this);
            }
            this.g();
        }
        j(a) {
            this.a7(a.o);
            this.n(a);
        }
        l(a) {
            this.a3._write2(WorkbookSaveManagerExcel2003_SaveTokenVisitor.a0);
        }
        n(a) {
            this.a3._write4(a.x);
        }
        o(a) {
            this.a3._write2(WorkbookSaveManagerExcel2003_SaveTokenVisitor.az);
        }
        p(a) {
            this.a3._write9(a.p);
            this.a3._write17((a.skipTokenIndexes.count));
            let b = this.a2[this.d];
            let c = WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a7(a);
            this.a3._write17((c - 4));
            for (let d = 0; d < a.skipTokenIndexes.count; d++) {
                let e = this.a2[a.skipTokenIndexes._inner[d]];
                this.a3._write17((e - b));
            }
        }
        q(a) {
            this.a3._write9(a.p);
            if (a.skipTokenIndexes.count == 0) {
                this.a3._write17(0);
                return;
            }
            let b = this.a2[this.d];
            let c = b + 4;
            let d = this.a2[a.skipTokenIndexes._inner[0]];
            let e = d + 4;
            this.a3._write17((e - c));
        }
        r(a) {
            this.a3._write9(a.p);
            let b = this.a2[this.d];
            let c = b + 4;
            let d = this.a2[a.t];
            let e = d + 4;
            this.a3._write17((e - c - 1));
        }
        s(a) {
            this.a3._write9(a.p);
            this.a3._write9(a.t);
            this.a3._write9(a.v);
        }
        t(a) {
            this.a3._write9(a.p);
            this.a3._write17(0);
        }
        u(a) {
            this.a3._write9(a.p);
            this.a3._write17(0);
        }
        v(a) {
        }
        w(a) {
            this.a3._write9((a.r ? 1 : 0));
        }
        y(a) {
            this.a3._write9(a.p._y);
        }
        z(a) {
            this.a3._write17(a.p.t);
            this.a3._write17(a.p.r);
        }
        aa(a) {
            this.a3._write17(a.$function._wi);
        }
        ac(a) {
            let b = a.argumentCount;
            if (a.$function._q4) {
                b++;
            }
            this.a3._write9(b);
            this.aa(a);
        }
        ad(a) {
            this.a3._write17(a.p);
        }
        ae(a) {
            this.a3._write18(intSToU(0));
            this.a3._write17(a.p);
        }
        af(a) {
            this.a3._write18(intSToU(0));
            this.a3._write17(a.p);
        }
        ag(a) {
            this.a3._write17(a.p);
        }
        ah(a) {
            this.a3._write18(intSToU(0));
            this.a3._write17(a.p);
        }
        ai(a) {
        }
        aj(a) {
            let b = this.a3.saveManager._b._bb;
            let c = b.u.indexOf(a.u);
            if (c < 0) {
                let d = b.c(a.u.name, this.a3.saveManager._b, false);
                c = b.u.indexOf(d);
            }
            this.a3._write17((c + 1));
            this.a3._write17(0);
        }
        ak(a) {
            this.a7(a.o);
            let b = a.o.e.u.indexOf(a.u);
            this.a3._write17((b + 1));
            this.a3._write17(0);
        }
        al(a) {
            this.a3._write10(a.p);
        }
        an(a) {
        }
        ap(a) {
            this.a7(a.o);
            this.au(a);
        }
        as(a) {
            this.a3._write18(intSToU(0));
        }
        au(a) {
            this.a3._write3(a.y);
        }
        av(a) {
            this.a3._write15(a.p, 0);
        }
        aw(a) {
        }
        ax(a) {
            this.a3._write17(a.p.t);
            this.a3._write17(a.p.r);
        }
        ay(a) {
        }
        static a5(a, b, c, d, e) {
            if (WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4 == null) {
                WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4 = new WorkbookSaveManagerExcel2003_SaveTokenVisitor();
            }
            WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4.b = a;
            WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4.a3 = b;
            WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4.a6 = c;
            WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4.a1 = d;
            WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4.a2 = e;
            return WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4;
        }
        a7(a) {
            if (this.a6) {
                this.a3._write17(a.l);
                this.a3._write17(a.m);
            }
            else {
                let b = this.a3.saveManager._j.indexOf(a);
                this.a3._write17(b);
            }
        }
    }
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.$t = /*@__PURE__*/ markType(WorkbookSaveManagerExcel2003_SaveTokenVisitor, 'WorkbookSaveManagerExcel2003_SaveTokenVisitor', FormulaTokenVisitor.$);
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4 = null;
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.az = /*@__PURE__*/ new Array(7);
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.a0 = /*@__PURE__*/ new Array(8);
    return WorkbookSaveManagerExcel2003_SaveTokenVisitor;
})();
