// igniteui-react-excel/esm2015/lib/WorkbookFormat.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various file formats in which a workbook can be saved.
 */
export var WorkbookFormat = /*@__PURE__*/ (function (WorkbookFormat) {
    /**
     * The Excel 97-2003 BIFF8 file format.
     */
    WorkbookFormat[WorkbookFormat["Excel97To2003"] = 0] = "Excel97To2003";
    /**
     * The Excel 97-2003 Template BIFF8 file format.
     */
    WorkbookFormat[WorkbookFormat["Excel97To2003Template"] = 1] = "Excel97To2003Template";
    /**
     * The Excel 2007 XML file format.
     */
    WorkbookFormat[WorkbookFormat["Excel2007"] = 2] = "Excel2007";
    /**
     * The Excel 2007 Macro-Enabled XML file format.
     */
    WorkbookFormat[WorkbookFormat["Excel2007MacroEnabled"] = 3] = "Excel2007MacroEnabled";
    /**
     * The Excel 2007 Macro-Enabled Template XML file format.
     */
    WorkbookFormat[WorkbookFormat["Excel2007MacroEnabledTemplate"] = 4] = "Excel2007MacroEnabledTemplate";
    /**
     * The Excel 2007 Template XML file format.
     */
    WorkbookFormat[WorkbookFormat["Excel2007Template"] = 5] = "Excel2007Template";
    /**
     * The Strict Open XML file format (ISO/IEC 29500 Strict).
     */
    WorkbookFormat[WorkbookFormat["StrictOpenXml"] = 6] = "StrictOpenXml";
    return WorkbookFormat;
})({});
/**
 * @hidden
 */
export let WorkbookFormat_$type = /*@__PURE__*/ markEnum('WorkbookFormat', 'Excel97To2003,0|Excel97To2003Template,1|Excel2007,2|Excel2007MacroEnabled,3|Excel2007MacroEnabledTemplate,4|Excel2007Template,5|StrictOpenXml,6');
