// igniteui-react-excel/esm2015/lib/WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { FormulaTokenVisitor$1 } from "./FormulaTokenVisitor$1";
import { Number_$type, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let WorkbookSaveManagerExcel2003_GetSizeTokenVisitor = /*@__PURE__*/ (() => {
    class WorkbookSaveManagerExcel2003_GetSizeTokenVisitor extends FormulaTokenVisitor$1 {
        constructor() {
            super(Number_$type);
            this.a6 = false;
            this.a3 = null;
            this.a2 = null;
        }
        d(a) {
            return 0;
        }
        a1() {
            this.a6 = false;
            this.a3 = null;
            this.a2 = null;
            super.a1();
        }
        f(a) {
            if (this.a6) {
                return 13;
            }
            return 11;
        }
        h(a) {
            return 9;
        }
        j(a) {
            return 9;
        }
        k(a) {
            return 8;
        }
        l(a) {
            return WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a7(a);
        }
        m(a) {
            return 4;
        }
        n(a) {
            return 4;
        }
        o(a) {
            return 4;
        }
        p(a) {
            return 4;
        }
        q(a) {
            return 4;
        }
        r(a) {
            return 1;
        }
        s(a) {
            return 2;
        }
        u(a) {
            return 2;
        }
        v(a) {
            return 5;
        }
        w(a) {
            return 3;
        }
        y(a) {
            return 4;
        }
        z(a) {
            return 3;
        }
        aa(a) {
            return 7;
        }
        ab(a) {
            return 7;
        }
        ac(a) {
            return 3;
        }
        ad(a) {
            return 7;
        }
        ae(a) {
            return 1;
        }
        af(a) {
            return 5;
        }
        ag(a) {
            if (this.a6) {
                return 9;
            }
            return 7;
        }
        ah(a) {
            return 9;
        }
        aj(a) {
            return 1;
        }
        al(a) {
            if (this.a6) {
                return 9;
            }
            return 7;
        }
        ao(a) {
            return 5;
        }
        aq(a) {
            return 5;
        }
        ar(a) {
            return (1 + this.a3._getDataSize(this.a2[this.az], 0, a.p));
        }
        as(a) {
            return 0;
        }
        at(a) {
            return 5;
        }
        au(a) {
            return 1;
        }
        static a5(a, b, c, d) {
            if (WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4 == null) {
                WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4 = new WorkbookSaveManagerExcel2003_GetSizeTokenVisitor();
            }
            WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4.c = a;
            WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4.a3 = b;
            WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4.a6 = c;
            WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4.a2 = d;
            return WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4;
        }
        static a7(a) {
            return (6 + (2 * a.skipTokenIndexes.count));
        }
    }
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.$t = /*@__PURE__*/ markType(WorkbookSaveManagerExcel2003_GetSizeTokenVisitor, 'WorkbookSaveManagerExcel2003_GetSizeTokenVisitor', /*@__PURE__*/ FormulaTokenVisitor$1.$.specialize(Number_$type));
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4 = null;
    return WorkbookSaveManagerExcel2003_GetSizeTokenVisitor;
})();
