// igniteui-react-excel/esm2015/lib/WorksheetPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let WorksheetPartManager = /*@__PURE__*/ (() => {
    class WorksheetPartManager extends XmlPartManagerBase {
        get_m() {
            return WorksheetPartManager.a4;
        }
        get m() {
            return this.get_m();
        }
        n(a) {
            switch (a) {
                case 0: return "/xl/worksheets/sheet.xml";
                default: return null;
            }
        }
        o(a) {
            switch (a) {
                case 0: return WorksheetPartManager.a6;
                case 1: return WorksheetPartManager.a5;
                default: return WorksheetPartManager.a6;
            }
        }
        get_g() {
            return true;
        }
        get g() {
            return this.get_g();
        }
    }
    WorksheetPartManager.$t = /*@__PURE__*/ markType(WorksheetPartManager, 'WorksheetPartManager', XmlPartManagerBase.$);
    WorksheetPartManager.a4 = "application/vnd.openxmlformats-officedocument.spreadsheetml.worksheet+xml";
    WorksheetPartManager.a6 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/worksheet";
    WorksheetPartManager.a5 = "http://purl.oclc.org/ooxml/officeDocument/relationships/worksheet";
    return WorksheetPartManager;
})();
