// igniteui-react-excel/esm2015/lib/WorkItem_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, Type, markType } from "igniteui-react-core";
import { WorkItem_ExceptionInfo } from "./WorkItem_ExceptionInfo";
import { InvalidOperationException } from "igniteui-react-core";
import { Queue$1 } from "igniteui-react-core";
/**
 * @hidden
 */
export let WorkItem = /*@__PURE__*/ (() => {
    class WorkItem extends Base {
        constructor(a, b, c) {
            super();
            this.j = null;
            this.n = false;
            this.o = false;
            this.a = null;
            this.b = null;
            this.k = 0;
            this.k = a;
            this.o = c;
            if (b != null) {
                b.ab(this);
            }
        }
        static c(a, b = 0) {
            return new WorkItem_WorkItemSync(a, b, null, true);
        }
        static d(a, b = 0) {
            return new WorkItem_WorkItemAsync(a, b, null, true);
        }
        static e($tResult, a, b = 0) {
            return new WorkItem_WorkItemSync$1($tResult, a, b, null, true);
        }
        static f($tResult, a, b = 0) {
            return new WorkItem_WorkItemAsync$1($tResult, a, b, null, true);
        }
        w(a) {
            try {
                if (this.r) {
                    return;
                }
                if (this.j != null && this.o) {
                    if (((() => { let b = this.u(this.j); this.j = b.p0; return b.ret; })())) {
                        this.ac();
                    }
                    this.r = true;
                }
                else if (this.p(a)) {
                    this.r = true;
                }
            }
            catch (b) {
                this.j = new WorkItem_ExceptionInfo(b);
                this.r = true;
            }
            finally {
                if (this.r) {
                    this.z(a);
                }
            }
        }
        q(a, b) {
            if (a == null) {
                return true;
            }
            a.ab(this);
            if (a.t == false) {
                b.e(a);
            }
            else {
                a.w(b);
            }
            return false;
        }
        static g($t, a) {
            return new WorkItem_WorkItemSync$1($t, () => a, 1, null, true);
        }
        static h($t, a) {
            let b = WorkItem.g($t, a);
            b.w(WorkItemScheduler.b());
            return b;
        }
        x(a) {
            a.ab(null);
            this.j = a.j;
        }
        y() {
            if (this.j == null) {
                return;
            }
            this.j.c();
        }
        z(a) {
            let b = this.a;
            if (b != null) {
                b.x(this);
                if (b.t == false) {
                    a.e(b);
                }
                else {
                    b.w(a);
                }
            }
            else {
                this.y();
            }
        }
        aa(a) {
            this.j = a;
        }
        ab(a) {
            if (this.a == a) {
                return;
            }
            if (a == null) {
                this.a.b = null;
            }
            else {
                if (this.a != null) {
                    throw new InvalidOperationException(1, "This work item has other work already scheduled after it.");
                }
                a.b = this.i;
            }
            this.a = a;
        }
        ac() {
            let a = this;
            let b = this.a;
            while (b != null && b.s) {
                b.x(a);
                b.r = true;
                a = b;
                b = b.a;
            }
        }
        u(a) {
            return {
                ret: false,
                p0: a
            };
        }
        get v() {
            if (this.j == null) {
                return null;
            }
            return this.j.b;
        }
        get_s() {
            return false;
        }
        get s() {
            return this.get_s();
        }
        get r() {
            return this.n;
        }
        set r(a) {
            this.n = a;
        }
        get_t() {
            return false;
        }
        get t() {
            return this.get_t();
        }
        get i() {
            return this.b || this;
        }
        get l() {
            return this.k;
        }
    }
    WorkItem.$t = /*@__PURE__*/ markType(WorkItem, 'WorkItem');
    WorkItem.m = () => {
    };
    return WorkItem;
})();
/**
 * @hidden
 */
export let WorkItem_WorkItemSync = /*@__PURE__*/ (() => {
    class WorkItem_WorkItemSync extends WorkItem {
        constructor(a, b, c, d) {
            super(b, c, d);
            this.ad = null;
            this.ad = a;
        }
        p(a) {
            this.ad();
            return true;
        }
    }
    WorkItem_WorkItemSync.$t = /*@__PURE__*/ markType(WorkItem_WorkItemSync, 'WorkItem_WorkItemSync', WorkItem.$);
    return WorkItem_WorkItemSync;
})();
/**
 * @hidden
 */
export let WorkItem$1 = /*@__PURE__*/ (() => {
    class WorkItem$1 extends WorkItem {
        constructor($tResult, a, b, c) {
            super(a, b, c);
            this.$tResult = null;
            this.ad = null;
            this.$tResult = $tResult;
            this.$type = this.$type.specialize(this.$tResult);
        }
    }
    WorkItem$1.$t = /*@__PURE__*/ markType(WorkItem$1, 'WorkItem$1', WorkItem.$);
    return WorkItem$1;
})();
/**
 * @hidden
 */
export let WorkItem_WorkItemSync$1 = /*@__PURE__*/ (() => {
    class WorkItem_WorkItemSync$1 extends WorkItem$1 {
        constructor($tResult, a, b, c, d) {
            super($tResult, b, c, d);
            this.$tResult = null;
            this.ae = null;
            this.$tResult = $tResult;
            this.$type = this.$type.specialize(this.$tResult);
            this.ae = a;
        }
        p(a) {
            this.ad = this.ae();
            return true;
        }
    }
    WorkItem_WorkItemSync$1.$t = /*@__PURE__*/ markType(WorkItem_WorkItemSync$1, 'WorkItem_WorkItemSync$1', /*@__PURE__*/ WorkItem$1.$.specialize(0));
    return WorkItem_WorkItemSync$1;
})();
/**
 * @hidden
 */
export let WorkItem_WorkItemAsync = /*@__PURE__*/ (() => {
    class WorkItem_WorkItemAsync extends WorkItem {
        constructor(a, b, c, d) {
            super(b, c, d);
            this.ad = false;
            this.ae = null;
            this.ae = a;
        }
        p(a) {
            if (this.ad == false) {
                this.ad = true;
                return this.q(this.ae(), a);
            }
            return true;
        }
    }
    WorkItem_WorkItemAsync.$t = /*@__PURE__*/ markType(WorkItem_WorkItemAsync, 'WorkItem_WorkItemAsync', WorkItem.$);
    return WorkItem_WorkItemAsync;
})();
/**
 * @hidden
 */
export let WorkItem_WorkItemAsync$1 = /*@__PURE__*/ (() => {
    class WorkItem_WorkItemAsync$1 extends WorkItem$1 {
        constructor($tResult, a, b, c, d) {
            super($tResult, b, c, d);
            this.$tResult = null;
            this.af = false;
            this.ag = null;
            this.ae = null;
            this.$tResult = $tResult;
            this.$type = this.$type.specialize(this.$tResult);
            this.ag = a;
        }
        p(a) {
            if (this.af == false) {
                this.af = true;
                let b = this.ag();
                this.ae = b;
                return this.q(b, a);
            }
            return true;
        }
        x(a) {
            if (this.ae != null) {
                if (this.ae.v == null) {
                    this.ad = WorkItemExtensions.m(this.$tResult, this.ae);
                }
                this.ae = null;
            }
            super.x(a);
        }
    }
    WorkItem_WorkItemAsync$1.$t = /*@__PURE__*/ markType(WorkItem_WorkItemAsync$1, 'WorkItem_WorkItemAsync$1', /*@__PURE__*/ WorkItem$1.$.specialize(0));
    return WorkItem_WorkItemAsync$1;
})();
/**
 * @hidden
 */
export let WorkItemScheduler = /*@__PURE__*/ (() => {
    class WorkItemScheduler extends Base {
        static a() {
            return WorkItemScheduler.c();
        }
        static b() {
            return new WorkItemScheduler_SynchronousImpl();
        }
        static c() {
            return new WorkItemScheduler_TimeSlicedImpl();
        }
    }
    WorkItemScheduler.$t = /*@__PURE__*/ markType(WorkItemScheduler, 'WorkItemScheduler');
    return WorkItemScheduler;
})();
/**
 * @hidden
 */
export let WorkItemScheduler_SynchronousImpl = /*@__PURE__*/ (() => {
    class WorkItemScheduler_SynchronousImpl extends WorkItemScheduler {
        constructor() {
            super();
            this.f = null;
            this.f = new Queue$1(WorkItem.$);
        }
        d(a) {
            if (a == null) {
                return;
            }
            this.e(a);
            while (this.f.count != 0) {
                let b = this.f.b();
                if (b != null) {
                    b.w(this);
                }
            }
        }
        e(a) {
            this.f.g(a.i);
        }
    }
    WorkItemScheduler_SynchronousImpl.$t = /*@__PURE__*/ markType(WorkItemScheduler_SynchronousImpl, 'WorkItemScheduler_SynchronousImpl', WorkItemScheduler.$);
    return WorkItemScheduler_SynchronousImpl;
})();
/**
 * @hidden
 */
export let WorkItemScheduler_TimeSlicedImpl = /*@__PURE__*/ (() => {
    class WorkItemScheduler_TimeSlicedImpl extends WorkItemScheduler {
        constructor() {
            super();
            this.f = null;
            this.f = new Queue$1(WorkItem.$);
        }
        d(a) {
            if (a == null) {
                return;
            }
            this.f.g(a.i);
            while (this.f.count != 0) {
                let b = this.f.b();
                if (b != null) {
                    b.w(this);
                }
            }
        }
        e(a) {
            if (a.l == 2) {
                setTimeout(() => this.d(a), 1);
            }
            else {
                this.f.g(a.i);
            }
        }
    }
    WorkItemScheduler_TimeSlicedImpl.$t = /*@__PURE__*/ markType(WorkItemScheduler_TimeSlicedImpl, 'WorkItemScheduler_TimeSlicedImpl', WorkItemScheduler.$);
    return WorkItemScheduler_TimeSlicedImpl;
})();
/**
 * @hidden
 */
export let WorkItemExtensions = /*@__PURE__*/ (() => {
    class WorkItemExtensions extends Base {
        static n(a) {
            WorkItemScheduler.b().d(a);
        }
        static o(a) {
            WorkItemScheduler.a().d(WorkItemExtensions.b(a, (b) => b.y(), 1));
        }
        static m($tResult, a) {
            if (a == null) {
                return Type.getDefaultValue($tResult);
            }
            if (a.r == false) {
                WorkItemScheduler.b().d(a);
            }
            a.y();
            return a.ad;
        }
        static a(a, b, c = 0) {
            return new WorkItem_WorkItemSync(b, c, a, true);
        }
        static b(a, b, c = 0) {
            return new WorkItem_WorkItemSync(() => b(a || WorkItemExtensions_WorkItemCompleted.ad), c, a, false);
        }
        static e($t, a, b, c = 0) {
            return new WorkItem_WorkItemSync(() => b(a || WorkItemExtensions_WorkItemCompleted$1.$t.specialize($t).getStaticFields().ae), c, a, false);
        }
        static c(a, b, c = 0) {
            return new WorkItem_WorkItemAsync(b, c, a, true);
        }
        static d(a, b, c = 0) {
            return new WorkItem_WorkItemAsync(() => b(a || WorkItemExtensions_WorkItemCompleted.ad), c, a, false);
        }
        static f($t, a, b, c = 0) {
            return new WorkItem_WorkItemAsync(() => b(a || WorkItemExtensions_WorkItemCompleted$1.$t.specialize($t).getStaticFields().ae), c, a, false);
        }
        static g($tResult, a, b, c = 0) {
            return new WorkItem_WorkItemSync$1($tResult, b, c, a, true);
        }
        static h($tResult, a, b, c = 0) {
            return new WorkItem_WorkItemSync$1($tResult, () => b(a || WorkItemExtensions_WorkItemCompleted.ad), c, a, false);
        }
        static k($t, $tResult, a, b, c = 0) {
            return new WorkItem_WorkItemSync$1($tResult, () => b(a || WorkItemExtensions_WorkItemCompleted$1.$t.specialize($t).getStaticFields().ae), c, a, false);
        }
        static i($tResult, a, b, c = 0) {
            return new WorkItem_WorkItemAsync$1($tResult, b, c, a, true);
        }
        static j($tResult, a, b, c = 0) {
            return new WorkItem_WorkItemAsync$1($tResult, () => b(a || WorkItemExtensions_WorkItemCompleted.ad), c, a, false);
        }
        static l($t, $tResult, a, b, c = 0) {
            return new WorkItem_WorkItemAsync$1($tResult, () => b(a || WorkItemExtensions_WorkItemCompleted$1.$t.specialize($t).getStaticFields().ae), c, a, false);
        }
    }
    WorkItemExtensions.$t = /*@__PURE__*/ markType(WorkItemExtensions, 'WorkItemExtensions');
    return WorkItemExtensions;
})();
/**
 * @hidden
 */
export let WorkItemExtensions_WorkItemCompleted = /*@__PURE__*/ (() => {
    class WorkItemExtensions_WorkItemCompleted extends WorkItem {
        constructor() {
            super(1, null, true);
            this.r = true;
        }
        p(a) {
            return true;
        }
        x(a) {
            super.x(a);
        }
    }
    WorkItemExtensions_WorkItemCompleted.$t = /*@__PURE__*/ markType(WorkItemExtensions_WorkItemCompleted, 'WorkItemExtensions_WorkItemCompleted', WorkItem.$);
    WorkItemExtensions_WorkItemCompleted.ad = /*@__PURE__*/ new WorkItemExtensions_WorkItemCompleted();
    return WorkItemExtensions_WorkItemCompleted;
})();
/**
 * @hidden
 */
export let WorkItemExtensions_WorkItemCompleted$1 = /*@__PURE__*/ (() => {
    class WorkItemExtensions_WorkItemCompleted$1 extends WorkItem$1 {
        constructor($t) {
            super($t, 1, null, true);
            this.$t = null;
            this.$t = $t;
            this.$type = this.$type.specialize(this.$t);
            this.r = true;
        }
        p(a) {
            return true;
        }
        x(a) {
            super.x(a);
        }
    }
    WorkItemExtensions_WorkItemCompleted$1.$t = /*@__PURE__*/ markType(WorkItemExtensions_WorkItemCompleted$1, 'WorkItemExtensions_WorkItemCompleted$1', /*@__PURE__*/ WorkItem$1.$.specialize(0), null, function () {
        this.ae = new WorkItemExtensions_WorkItemCompleted$1(arguments[0]);
    });
    return WorkItemExtensions_WorkItemCompleted$1;
})();
