// igniteui-react-excel/esm2015/lib/WorksheetElementResizeState.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let WorksheetElementResizeState = /*@__PURE__*/ (() => {
    class WorksheetElementResizeState extends Base {
        constructor(a, b, c, d, e) {
            super();
            this.d = 0;
            this.e = 0;
            this.c = 0;
            this.b = false;
            this.a = false;
            this.d = a;
            this.e = b;
            this.c = c;
            this.b = d;
            this.a = e;
        }
    }
    WorksheetElementResizeState.$t = /*@__PURE__*/ markType(WorksheetElementResizeState, 'WorksheetElementResizeState');
    return WorksheetElementResizeState;
})();
