// igniteui-react-excel/esm2015/lib/WorksheetCellCommentData.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let WorksheetCellCommentData = /*@__PURE__*/ (() => {
    class WorksheetCellCommentData extends Base {
        constructor(a, b, c, d) {
            super();
            this.g = 0;
            this.c = 0;
            this.a = null;
            this.e = 0;
            this.g = b;
            this.c = d;
            this.a = a;
            this.e = c;
        }
        get h() {
            return this.g;
        }
        get d() {
            return this.c;
        }
        get b() {
            return this.a;
        }
        get f() {
            return this.e;
        }
    }
    WorksheetCellCommentData.$t = /*@__PURE__*/ markType(WorksheetCellCommentData, 'WorksheetCellCommentData');
    return WorksheetCellCommentData;
})();
