// igniteui-react-excel/esm2015/lib/WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { FormulaTokenVisitor } from "./FormulaTokenVisitor";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor = /*@__PURE__*/ (() => {
    class WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor extends FormulaTokenVisitor {
        constructor() {
            super();
            this.az = null;
        }
        g() {
            this.az = null;
            super.g();
        }
        o(a) {
            let b = a.q.length;
            let c = a.q[0].length;
            this.az._write9((b - 1));
            this.az._write17((c - 1));
            for (let d = 0; d < c; d++) {
                for (let e = 0; e < b; e++) {
                    let f = a.q[e][d];
                    this.az._write9(f.b);
                    f.g(this.az);
                }
            }
        }
        ae(a) {
            this.az._write17(a.r.length);
            for (let b = 0; b < a.r.length; b++) {
                this.az._write4(a.r[b]);
            }
        }
        static a1(a, b) {
            if (WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor.a0 == null) {
                WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor.a0 = new WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor();
            }
            WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor.a0.b = a;
            WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor.a0.az = b;
            return WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor.a0;
        }
    }
    WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor.$t = /*@__PURE__*/ markType(WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor, 'WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor', FormulaTokenVisitor.$);
    WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor.a0 = null;
    return WorkbookSaveManagerExcel2003_SaveAdditionalDataTokenVisitor;
})();
