// igniteui-react-excel/esm2015/lib/XFPropColorLoaded.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XFPropColorBase } from "./XFPropColorBase";
import { ExcelUtils } from "./ExcelUtils";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let XFPropColorLoaded = /*@__PURE__*/ (() => {
    class XFPropColorLoaded extends XFPropColorBase {
        constructor(a, b) {
            super();
            this.d = null;
            this.e = 0;
            this.e = a;
            this.d = b;
        }
        c(a) {
            return ExcelUtils.ae(a, this.d);
        }
        get_a() {
            return this.e;
        }
        get a() {
            return this.get_a();
        }
    }
    XFPropColorLoaded.$t = /*@__PURE__*/ markType(XFPropColorLoaded, 'XFPropColorLoaded', XFPropColorBase.$);
    return XFPropColorLoaded;
})();
