// igniteui-react-excel/esm2015/lib/XFPropByte.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { XFProp } from "./XFProp";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let XFPropByte = /*@__PURE__*/ (() => {
    class XFPropByte extends XFProp {
        constructor(a, b) {
            super();
            this.c = 0;
            this.d = 0;
            this.c = a;
            this.d = b;
        }
        b(a, b) {
        }
        get_a() {
            return this.c;
        }
        get a() {
            return this.get_a();
        }
        get e() {
            return this.d;
        }
    }
    XFPropByte.$t = /*@__PURE__*/ markType(XFPropByte, 'XFPropByte', XFProp.$);
    return XFPropByte;
})();
