// igniteui-react-charts/esm5/lib/CalloutCollisionMode.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * An enum representing the available collision modes of callout layers.
 */
export var CalloutCollisionMode = /*@__PURE__*/ (function (CalloutCollisionMode) {
    /**
     * Automatically decide the collision strategy.
     */
    CalloutCollisionMode[CalloutCollisionMode["Auto"] = 0] = "Auto";
    /**
     * Use a simulated annealing based collision strategy. This is higher quality, but takes longer, and is performed time-sliced in the background until an acceptable quality is reached.
     */
    CalloutCollisionMode[CalloutCollisionMode["SimulatedAnnealing"] = 1] = "SimulatedAnnealing";
    /**
     * Use a greedy algorithm to avoid collisions. This is cheap and predictable, but of comparatively low quality.
     */
    CalloutCollisionMode[CalloutCollisionMode["Greedy"] = 2] = "Greedy";
    /**
     * Use a greedy algorithm with localized center of mass hints to avoid collisions. This is relatively cheap to perform, compared to the simulated annealing approach, but is of comparatively lower quality.
     */
    CalloutCollisionMode[CalloutCollisionMode["GreedyCenterOfMass"] = 3] = "GreedyCenterOfMass";
    return CalloutCollisionMode;
})({});
/**
 * @hidden
 */
export var CalloutCollisionMode_$type = /*@__PURE__*/ markEnum('CalloutCollisionMode', 'Auto,0|SimulatedAnnealing,1|Greedy,2|GreedyCenterOfMass,3');
