// igniteui-react-charts/esm5/lib/DomainChart.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, INotifyPropertyChanged_$type, BaseError, PropertyChangedEventArgs, Delegate_$type, fromEnum, String_$type, runOn, delegateCombine, EnumUtil, enumGetBox, typeCast, delegateRemove, markType, TypeRegistrar, getInstanceType } from "igniteui-react-core";
import { IDataLegendTarget_$type } from "igniteui-react-core";
import { ArrayExtension } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
import { Brush } from "igniteui-react-core";
import { XamDataChart } from "./XamDataChart";
import { DataSeriesType_$type } from "igniteui-react-core";
import { MarkerType_$type } from "./MarkerType";
import { Color } from "igniteui-react-core";
import { ColorUtil } from "igniteui-react-core";
import { DataToolTipLayerGroupingMode_$type } from "igniteui-react-core";
import { DataTooltipGroupedPositionX_$type } from "igniteui-react-core";
import { DataTooltipGroupedPositionY_$type } from "igniteui-react-core";
import { DataAbbreviationMode_$type } from "igniteui-react-core";
import { DataLegendValueMode_$type } from "igniteui-react-core";
import { Thickness } from "igniteui-react-core";
import { DataLegendHeaderDateMode_$type } from "igniteui-react-core";
import { DataLegendHeaderTimeMode_$type } from "igniteui-react-core";
import { DataLegendSummaryType_$type } from "igniteui-react-core";
import { LegendItemBadgeShape_$type } from "igniteui-react-core";
import { DataLegendUnitsMode_$type } from "igniteui-react-core";
import { DataLegendLabelMode_$type } from "igniteui-react-core";
import { ChartSortDescriptionCollection } from "./ChartSortDescriptionCollection";
import { FilterExpressionCollection } from "igniteui-react-core";
import { ChartSummaryDescriptionCollection } from "./ChartSummaryDescriptionCollection";
import { SortGroupParser } from "igniteui-react-core";
import { SummaryParser } from "igniteui-react-core";
import { BrushCollection } from "igniteui-react-core";
import { Rect } from "igniteui-react-core";
import { HorizontalAlignment_$type } from "igniteui-react-core";
import { UnknownValuePlotting_$type } from "igniteui-react-core";
import { SeriesOutlineMode_$type } from "./SeriesOutlineMode";
import { MarkerOutlineMode_$type } from "./MarkerOutlineMode";
import { MarkerFillMode_$type } from "./MarkerFillMode";
import { MarkerAutomaticBehavior_$type } from "./MarkerAutomaticBehavior";
import { TrendLineType_$type } from "igniteui-react-core";
import { ComputedPlotAreaMarginMode_$type } from "./ComputedPlotAreaMarginMode";
import { SeriesPlotAreaMarginHorizontalMode_$type } from "./SeriesPlotAreaMarginHorizontalMode";
import { SeriesPlotAreaMarginVerticalMode_$type } from "./SeriesPlotAreaMarginVerticalMode";
import { SeriesHighlightingMode_$type } from "./SeriesHighlightingMode";
import { SeriesHighlightingBehavior_$type } from "./SeriesHighlightingBehavior";
import { LegendHighlightingMode_$type } from "./LegendHighlightingMode";
import { LegendItemBadgeMode_$type } from "igniteui-react-core";
import { ToolTipType_$type } from "./ToolTipType";
import { CrosshairsDisplayMode_$type } from "./CrosshairsDisplayMode";
import { SeriesViewerScrollbarMode_$type } from "./SeriesViewerScrollbarMode";
import { SeriesViewerHorizontalScrollbarPosition_$type } from "./SeriesViewerHorizontalScrollbarPosition";
import { SeriesViewerVerticalScrollbarPosition_$type } from "./SeriesViewerVerticalScrollbarPosition";
import { Defaults } from "./Defaults";
import { Series } from "./Series";
import { DeviceUtils } from "igniteui-react-core";
import { Dictionary$2 } from "igniteui-react-core";
import { DataSeriesAdapter } from "igniteui-react-core";
import { ChartSortDescription } from "./ChartSortDescription";
import { ChartSummaryDescription } from "./ChartSummaryDescription";
import { ObservableCollection$1 } from "igniteui-react-core";
import { Random } from "igniteui-react-core";
import { ObservableColorCollection } from "igniteui-react-core";
import { BrushCollectionUtil } from "igniteui-react-core";
import { ChartSeriesEventArgs } from "./ChartSeriesEventArgs";
import { DomainChartSeriesPointerEventArgs } from "./DomainChartSeriesPointerEventArgs";
import { BrushUtil } from "igniteui-react-core";
import { IUnknownValuePlotting_$type } from "igniteui-react-core";
import { ITrendLineSeries_$type } from "./ITrendLineSeries";
import { isNaN_ } from "igniteui-react-core";
import { stringIsNullOrEmpty, stringEmpty, stringStartsWith } from "igniteui-react-core";
/**
 * @hidden
 */
var DomainChart = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(DomainChart, _super);
    function DomainChart() {
        var _this = _super.call(this) || this;
        _this.jt = null;
        _this.jr = null;
        _this.f8 = null;
        _this.g0 = NaN;
        _this.jq = null;
        _this.hd = 0;
        _this.he = 0;
        _this.hf = 0;
        _this.hc = 0;
        _this.g8 = 0;
        _this.ha = 0;
        _this.g9 = 0;
        _this.g7 = 0;
        _this.tr = null;
        _this.ts = null;
        _this.gx = NaN;
        _this.hg = NaN;
        _this.g6 = NaN;
        _this.gk = NaN;
        _this.je = 0;
        _this.dt = null;
        _this.i5 = null;
        _this.fy = false;
        _this.b4 = null;
        _this.kn = null;
        _this.kp = null;
        _this.dl = null;
        _this.gc = null;
        _this.propertyChanged = null;
        _this.jd = 0;
        _this.fs = false;
        _this.bz = null;
        _this.f7 = null;
        _this.b3 = null;
        _this.eu = false;
        _this.br = 0;
        _this.go = NaN;
        _this.gp = NaN;
        _this.gm = DeviceUtils.g(10);
        _this.gn = DeviceUtils.g(10);
        _this.bn = 0;
        _this.bp = 0;
        _this.en = true;
        _this.e = null;
        _this.b = null;
        _this.d = null;
        _this.a = null;
        _this.a7 = 0;
        _this.i9 = -1;
        _this.ja = -1;
        _this.bl = 0;
        _this.kf = stringEmpty();
        _this.eo = true;
        _this.kg = null;
        _this.f = null;
        _this.uz = new Thickness(0, DeviceUtils.g(5));
        _this.ep = true;
        _this.kh = "N/A";
        _this.eq = false;
        _this.u0 = new Thickness(1, DeviceUtils.g(2), DeviceUtils.g(0), DeviceUtils.g(2), DeviceUtils.g(0));
        _this.tm = null;
        _this.d3 = null;
        _this.j9 = null;
        _this.c = null;
        _this.j8 = stringEmpty();
        _this.bb = 0;
        _this.bd = 0;
        _this.ka = null;
        _this.te = null;
        _this.ut = new Thickness(1, DeviceUtils.g(2), DeviceUtils.g(0), DeviceUtils.g(0), DeviceUtils.g(0));
        _this.us = new Thickness(0, DeviceUtils.g(5));
        _this.em = true;
        _this.dv = null;
        _this.th = null;
        _this.dy = null;
        _this.bh = 0;
        _this.kc = null;
        _this.uw = new Thickness(1, DeviceUtils.g(2), DeviceUtils.g(0), DeviceUtils.g(0), DeviceUtils.g(0));
        _this.uv = new Thickness(0, DeviceUtils.g(5));
        _this.tj = null;
        _this.d0 = null;
        _this.kb = null;
        _this.tg = null;
        _this.dx = null;
        _this.kd = null;
        _this.ti = null;
        _this.dz = null;
        _this.ur = new Thickness(0, DeviceUtils.g(0));
        _this.ca = 0;
        _this.bj = 0;
        _this.ke = null;
        _this.tl = null;
        _this.uy = new Thickness(0, DeviceUtils.g(0));
        _this.d2 = null;
        _this.ux = new Thickness(0, DeviceUtils.g(0));
        _this.tk = null;
        _this.d1 = null;
        _this.bf = 0;
        _this.tf = null;
        _this.uu = new Thickness(1, DeviceUtils.g(8), DeviceUtils.g(0), DeviceUtils.g(0), DeviceUtils.g(0));
        _this.dw = null;
        _this.seriesAdded = null;
        _this.seriesRemoved = null;
        _this.seriesPointerEnter = null;
        _this.seriesPointerLeave = null;
        _this.seriesPointerMove = null;
        _this.seriesPointerDown = null;
        _this.seriesPointerUp = null;
        _this.calloutStyleUpdating = null;
        _this.calloutRenderStyleUpdating = null;
        _this.gb = null;
        _this.ee = null;
        _this.eh = false;
        _this.f5 = new Dictionary$2(String_$type, Base.$, 0);
        _this.f6 = new List$1(String_$type, 0);
        _this.as = null;
        _this.ao = null;
        _this.an = null;
        _this.dn = null;
        _this.a0 = null;
        _this.dr = null;
        _this.ds = null;
        _this.e2 = true;
        _this.e1 = true;
        _this.e3 = true;
        _this.e0 = true;
        _this.ar = null;
        _this.aq = null;
        _this.ap = null;
        _this.dp = null;
        _this.az = null;
        _this.kk = null;
        _this.ki = null;
        _this.kj = null;
        _this.kl = null;
        _this.jp = null;
        _this.jo = null;
        _this.h = null;
        _this.g = null;
        _this.t = null;
        _this.w = null;
        _this.r = null;
        _this.s = null;
        _this.b1 = null;
        _this.et = true;
        _this.ew = true;
        _this.ey = false;
        _this.e4 = false;
        _this.ev = false;
        _this.uo = new Rect(0, 0, 0, 1, 1);
        _this.ko = null;
        _this.km = null;
        _this.s5 = 1;
        _this.s4 = 1;
        _this.dj = 0;
        _this.g5 = NaN;
        _this.hb = NaN;
        _this.c1 = 0;
        _this.ci = 0;
        _this.cg = 0;
        _this.gy = NaN;
        _this.gz = NaN;
        _this.cn = null;
        _this.ce = 4;
        _this.u = null;
        _this.v = null;
        _this.jc = 400;
        _this.gj = NaN;
        _this.eg = false;
        _this.x = null;
        _this.dh = 0;
        _this.hh = DeviceUtils.g(1.5);
        _this.ef = true;
        _this.g2 = NaN;
        _this.g4 = NaN;
        _this.g3 = NaN;
        _this.g1 = NaN;
        _this.a3 = 0;
        _this.c3 = 0;
        _this.c5 = 0;
        _this.cz = 0;
        _this.cx = 0;
        _this.b6 = 0;
        _this.cb = 0;
        _this.b8 = 0;
        _this.jf = 7;
        _this.df = 4;
        _this.a5 = 0;
        _this.el = false;
        _this.td = null;
        _this.gl = NaN;
        _this.tc = null;
        _this.ek = true;
        _this.s8 = null;
        _this.ta = null;
        _this.s9 = null;
        _this.tb = null;
        _this.i7 = -1;
        _this.i8 = -1;
        _this.ex = false;
        _this.ez = true;
        _this.er = false;
        _this.tn = null;
        _this.to = null;
        _this.jb = -1;
        _this.ej = false;
        _this.ei = false;
        _this.gg = null;
        _this.ah = null;
        _this.j6 = null;
        _this.j7 = null;
        _this.j5 = "CalloutLabel";
        _this.j4 = "CalloutContent";
        _this.c9 = 0;
        _this.da = 0;
        _this.c7 = 0;
        _this.dd = 0;
        _this.tp = null;
        _this.tq = null;
        _this.gu = NaN;
        _this.gt = NaN;
        _this.gq = NaN;
        _this.gr = NaN;
        _this.gs = NaN;
        _this.gw = NaN;
        _this.gv = NaN;
        _this.es = true;
        _this.tt = null;
        _this.tu = null;
        _this.hl = NaN;
        _this.hk = NaN;
        _this.hi = NaN;
        _this.ho = NaN;
        _this.hj = NaN;
        _this.hn = NaN;
        _this.hm = NaN;
        _this.e5 = true;
        _this.hq = NaN;
        _this.hp = NaN;
        _this.hs = NaN;
        _this.hr = NaN;
        _this.gb = new List$1(Delegate_$type, 0);
        _this.ro();
        _this.q4();
        return _this;
    }
    DomainChart.prototype.provideContainer = function (a) {
        if (a == null) {
            this.r4(function (b) {
                b.destroy();
                b.provideContainer(null);
            });
            return;
        }
        var b = this.jt;
        this.jt = a;
        if (this.ee != null && this.jt != null && b == null) {
            this.jt.setTimeout(this.ee, 0);
            this.ee = null;
        }
        var c = ((function () {
            var $ret = new XamDataChart();
            $ret.ef = true;
            return $ret;
        })());
        c.provideContainer(a);
        this.dataChart = c;
        this.rp();
    };
    DomainChart.prototype.destroy = function () {
        this.provideContainer(null);
    };
    DomainChart.prototype.rp = function () {
        if (this.dataChart == null) {
            return;
        }
        var a = this.dataChart.cu.j.e();
        this.aa = a.a;
        this.ae = a.f;
        this.ac = a.b;
        this.ad = a.c;
        this.ag = a.g;
        this.hv = Defaults.j;
        this.rq();
    };
    DomainChart.prototype.rq = function () {
    };
    DomainChart.prototype.rm = function (a) {
        if (a == null) {
            return;
        }
        a.g9 = this.ir;
        a.g7 = this.ip;
        a.g8 = this.iq;
        a.g6 = this.io;
        a.sv = this.uk;
        a.g3 = this.ij;
        a.g5 = this.il;
        a.g4 = this.ik;
        a.g2 = this.ii;
        a.su = this.uj;
        if (!isNaN_(this.ia)) {
            a.gx = this.ia;
        }
    };
    DomainChart.prototype.rj = function () {
    };
    DomainChart.prototype.rs = function (a, b) {
        a.ho = this.jm;
        a.cf = this.du;
    };
    DomainChart.prototype.sd = function (a, b, c) {
        var _this = this;
        switch (a) {
            case "TitleTopMargin":
                if (this.dataChart != null) {
                    this.dataChart.g9 = this.ir;
                }
                break;
            case "TitleLeftMargin":
                if (this.dataChart != null) {
                    this.dataChart.g7 = this.ip;
                }
                break;
            case "PixelScalingRatio":
                if (this.dataChart != null) {
                    this.r4(function (d) { return d.gx = _this.ia; });
                }
                break;
            case "TitleRightMargin":
                if (this.dataChart != null) {
                    this.dataChart.g8 = this.iq;
                }
                break;
            case "TitleBottomMargin":
                if (this.dataChart != null) {
                    this.dataChart.g6 = this.io;
                }
                break;
            case "TitleTextColor":
                if (this.dataChart != null) {
                    this.dataChart.sv = this.uk;
                }
                break;
            case "SubtitleLeftMargin":
                if (this.dataChart != null) {
                    this.dataChart.g3 = this.ij;
                }
                break;
            case "SubtitleTopMargin":
                if (this.dataChart != null) {
                    this.dataChart.g5 = this.il;
                }
                break;
            case "SubtitleRightMargin":
                if (this.dataChart != null) {
                    this.dataChart.g4 = this.ik;
                }
                break;
            case "SubtitleBottomMargin":
                if (this.dataChart != null) {
                    this.dataChart.g2 = this.ii;
                }
                break;
            case "SubtitleTextColor":
                if (this.dataChart != null) {
                    this.dataChart.su = this.uj;
                }
                break;
            case "LeftMargin":
                if (this.dataChart != null) {
                    this.dataChart.gw = this.h7;
                }
                break;
            case "TopMargin":
                if (this.dataChart != null) {
                    this.dataChart.ha = this.is;
                }
                break;
            case "RightMargin":
                if (this.dataChart != null) {
                    this.dataChart.g1 = this.ih;
                }
                break;
            case "BottomMargin":
                if (this.dataChart != null) {
                    this.dataChart.gj = this.hu;
                }
                break;
            case "TransitionDuration":
                this.rf(function (d) { return d.ho = _this.jm; });
                break;
            case "TransitionEasingFunction":
                this.rf(function (d) { return d.cf = _this.du; });
                break;
        }
    };
    Object.defineProperty(DomainChart.prototype, "tooltipTemplate", {
        get: function () {
            return ArrayExtension.getModulus$1(Base.$, this.tooltipTemplates, 0);
        },
        set: function (a) {
            this.tooltipTemplates = a == null ? null : new List$1(Base.$, 1, [a]);
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "tooltipTemplates", {
        get: function () {
            return this.f8;
        },
        set: function (a) {
            var b = this.tooltipTemplates;
            if (a != b) {
                this.f8 = a;
                this.sc("TooltipTemplates", b, this.tooltipTemplates);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ia", {
        get: function () {
            return this.g0;
        },
        set: function (a) {
            var b = this.g0;
            if (a != b) {
                this.g0 = a;
                this.sc("PixelScalingRatio", b, this.ia);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "toolTip", {
        get: function () {
            return this.jq;
        },
        set: function (a) {
            var b = this.toolTip;
            if (b != a) {
                this.jq = a;
                this.sc("ToolTip", b, this.toolTip);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ip", {
        get: function () {
            return this.hd;
        },
        set: function (a) {
            var b = this.ip;
            if (a != b) {
                this.hd = a;
                this.sc("TitleLeftMargin", b, this.ip);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "iq", {
        get: function () {
            return this.he;
        },
        set: function (a) {
            var b = this.iq;
            if (a != b) {
                this.he = a;
                this.sc("TitleRightMargin", b, this.iq);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ir", {
        get: function () {
            return this.hf;
        },
        set: function (a) {
            var b = this.ir;
            if (a != b) {
                this.hf = a;
                this.sc("TitleTopMargin", b, this.ir);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "io", {
        get: function () {
            return this.hc;
        },
        set: function (a) {
            var b = this.io;
            if (a != b) {
                this.hc = a;
                this.sc("TitleBottomMargin", b, this.io);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ij", {
        get: function () {
            return this.g8;
        },
        set: function (a) {
            var b = this.ij;
            if (a != b) {
                this.g8 = a;
                this.sc("SubtitleLeftMargin", b, this.ij);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "il", {
        get: function () {
            return this.ha;
        },
        set: function (a) {
            var b = this.il;
            if (a != b) {
                this.ha = a;
                this.sc("SubtitleTopMargin", b, this.il);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ik", {
        get: function () {
            return this.g9;
        },
        set: function (a) {
            var b = this.ik;
            if (a != b) {
                this.g9 = a;
                this.sc("SubtitleRightMargin", b, this.ik);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ii", {
        get: function () {
            return this.g7;
        },
        set: function (a) {
            var b = this.ii;
            if (a != b) {
                this.g7 = a;
                this.sc("SubtitleBottomMargin", b, this.ii);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "uj", {
        get: function () {
            return this.tr;
        },
        set: function (a) {
            var b = this.uj;
            if (a != b) {
                this.tr = a;
                this.sc("SubtitleTextColor", b, this.uj);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "uk", {
        get: function () {
            return this.ts;
        },
        set: function (a) {
            var b = this.uk;
            if (a != b) {
                this.ts = a;
                this.sc("TitleTextColor", b, this.uk);
            }
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.get_h7 = function () {
        return this.gx;
    };
    DomainChart.prototype.set_h7 = function (a) {
        var b = this.h7;
        if (a != b) {
            this.gx = a;
            this.sc("LeftMargin", b, this.h7);
        }
    };
    Object.defineProperty(DomainChart.prototype, "h7", {
        get: function () {
            return this.get_h7();
        },
        set: function (a) {
            this.set_h7(a);
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "is", {
        get: function () {
            return this.hg;
        },
        set: function (a) {
            var b = this.is;
            if (a != b) {
                this.hg = a;
                this.sc("TopMargin", b, this.is);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ih", {
        get: function () {
            return this.g6;
        },
        set: function (a) {
            var b = this.ih;
            if (a != b) {
                this.g6 = a;
                this.sc("RightMargin", b, this.ih);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "hu", {
        get: function () {
            return this.gk;
        },
        set: function (a) {
            var b = this.hu;
            if (a != b) {
                this.gk = a;
                this.sc("BottomMargin", b, this.hu);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "jm", {
        get: function () {
            return this.je;
        },
        set: function (a) {
            var b = this.jm;
            if (a != b) {
                this.je = a;
                this.sc("TransitionDuration", b, this.jm);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "du", {
        get: function () {
            return this.dt;
        },
        set: function (a) {
            var b = this.du;
            if (a != b) {
                this.dt = a;
                this.sc("TransitionEasingFunction", b, this.du);
            }
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.rn = function (a) {
        if (this.dataChart == null) {
            return;
        }
        a.kn = this.p2;
        a.kb = this.pr;
    };
    DomainChart.prototype.rt = function (a, b) {
        this.ru(a, b);
    };
    DomainChart.prototype.rk = function () {
    };
    DomainChart.prototype.ru = function (a, b) {
        if (a == null) {
            return;
        }
        if (this.dg == 3) {
            return;
        }
        var toolTipTemplate_ = ArrayExtension.getModulus$1(Base.$, this.tooltipTemplates, b);
        if (toolTipTemplate_ == null) {
            a.et = true;
        }
        else {
            var series_ = a;
            a.et = false;
            series_.externalObject.tooltipTemplate = toolTipTemplate_;
        }
    };
    DomainChart.prototype.s2 = function (a, b, c) {
    };
    DomainChart.prototype.jv = function (a) {
        return a;
        return null;
    };
    DomainChart.prototype.i6 = function (a) {
        return null;
    };
    DomainChart.prototype.se = function (a, b, c) {
        var _this = this;
        switch (a) {
            case "SubtitleTextStyle":
                if (this.dataChart != null) {
                    this.dataChart.kb = this.pr;
                }
                break;
            case "TitleTextStyle":
                if (this.dataChart != null) {
                    this.dataChart.kn = this.p2;
                }
                break;
            case "TooltipTemplate":
                if (this.fy) {
                    return;
                }
                this.fy = true;
                this.tooltipTemplate = this.jv(this.tooltipTemplate);
                this.fy = false;
                if (this.dataChart != null) {
                    this.dataChart.h0 = this.tooltipTemplate;
                }
                this.toolTip = this.tooltipTemplate;
                break;
            case "ToolTip":
                this.rg(function (d, e) { return _this.ru(d, e); });
                break;
            case "TooltipTemplates":
                if (this.tooltipTemplates == null) {
                    break;
                }
                this.fy = true;
                this.rg(function (d, e) {
                    var f = _this.jv(ArrayExtension.getModulus$1(Base.$, _this.tooltipTemplates, e));
                    var series_ = d;
                    var templ_ = f;
                    series_.externalObject.tooltipTemplate = templ_;
                });
                this.fy = false;
                break;
        }
    };
    Object.defineProperty(DomainChart.prototype, "b5", {
        get: function () {
            return this.b4;
        },
        set: function (a) {
            var b = this.b5;
            if (a != b) {
                this.b4 = a;
                this.sc("Widget", b, this.b5);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "pr", {
        get: function () {
            return this.kn;
        },
        set: function (a) {
            var b = this.pr;
            if (a != b) {
                this.kn = a;
                this.sc("SubtitleTextStyle", b, this.pr);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "p2", {
        get: function () {
            return this.kp;
        },
        set: function (a) {
            var b = this.p2;
            if (a != b) {
                this.kp = a;
                this.sc("TitleTextStyle", b, this.p2);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "dataChart", {
        get: function () {
            return this.dl;
        },
        set: function (a) {
            var b = this.dataChart;
            if (b != a) {
                this.dl = a;
                this.sc("DataChart", b, this.dataChart);
            }
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.onDetachedFromUI = function () {
        this.r4(function (a) { return a.onDetachedFromUI(); });
    };
    DomainChart.prototype.onAttachedToUI = function () {
        this.r4(function (a) { return a.onAttachedToUI(); });
    };
    DomainChart.prototype.ro = function () {
        var e_1, _a, e_2, _b;
        this.gc = new List$1(MarkerType_$type, 0);
        this.ba = new DataSeriesAdapter().e;
        this.as = new ChartSortDescriptionCollection();
        this.ao = new ChartSortDescriptionCollection();
        this.an = new ChartSortDescriptionCollection();
        this.a0 = new ChartSummaryDescriptionCollection();
        this.dn = new FilterExpressionCollection();
        this.ay.e = this.ba.sortDescriptions;
        this.at.e = this.ba.groupDescriptions;
        this.dq.syncTarget = this.ba.filterExpressions;
        this.a2.e = this.ba.summaryDescriptions;
        this.dr = new SortGroupParser();
        this.ds = new SummaryParser();
        this.ar = ((function () {
            var $ret = new ChartSortDescriptionCollection();
            $ret.i = true;
            return $ret;
        })());
        this.aq = ((function () {
            var $ret = new ChartSortDescriptionCollection();
            $ret.i = true;
            return $ret;
        })());
        this.ap = ((function () {
            var $ret = new ChartSortDescriptionCollection();
            $ret.i = true;
            return $ret;
        })());
        this.az = ((function () {
            var $ret = new ChartSummaryDescriptionCollection();
            $ret.i = true;
            return $ret;
        })());
        this.dp = ((function () {
            var $ret = new FilterExpressionCollection();
            $ret.shouldDetachOnTargetChange = true;
            return $ret;
        })());
        if (this.e2) {
            if (this.ay.j.count == 0) {
                this.ar.e = this.ba.sortDescriptions;
            }
            if (this.au.j.count == 0) {
                this.aq.e = this.ba.postSortDescriptions;
            }
            this.e2 = false;
        }
        if (this.e1) {
            if (this.at.j.count == 0) {
                this.ap.e = this.ba.groupDescriptions;
            }
            this.e1 = false;
        }
        if (this.e3) {
            if (this.a2.j.count == 0) {
                this.az.e = this.ba.summaryDescriptions;
            }
            this.e3 = false;
        }
        if (this.q != null) {
            this.ba.includedProperties = this.q;
        }
        if (this.o != null) {
            this.ba.excludedProperties = this.o;
        }
        this.ba.supportedSeriesTypes = this.ge.toArray();
        this.ba.rules.clear();
        try {
            for (var _c = __values(fromEnum(this.gd)), _d = _c.next(); !_d.done; _d = _c.next()) {
                var a = _d.value;
                this.ba.rules.add(a);
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_d && !_d.done && (_a = _c.return))
                    _a.call(_c);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        var b = new List$1(String_$type, 0);
        try {
            for (var _e = __values(fromEnum(this.ba.rules)), _f = _e.next(); !_f.done; _f = _e.next()) {
                var c = _f.value;
                b.add(getInstanceType(c).typeName);
            }
        }
        catch (e_2_1) {
            e_2 = { error: e_2_1 };
        }
        finally {
            try {
                if (_f && !_f.done && (_b = _e.return))
                    _b.call(_e);
            }
            finally {
                if (e_2)
                    throw e_2.error;
            }
        }
        var d = this.ba.dataSeries;
        d.collectionChanged = delegateCombine(d.collectionChanged, runOn(this, this.r7));
    };
    DomainChart.prototype.fu = function (a) {
        return a != 24 && this.ge.contains(a);
    };
    DomainChart.prototype.notifyResized = function () {
        this.r4(function (a) { return a.notifyContainerResized(); });
    };
    DomainChart.prototype.r2 = function () {
        this.r4(function (a) {
            for (var b = 0; b < a.series.count; b++) {
                a.series._inner[b].lu();
            }
        });
    };
    DomainChart.prototype.q9 = function () {
        if (this.dataChart == null) {
            return;
        }
        if (this.fs) {
            this.sw();
        }
        this.dataChart.flush();
    };
    DomainChart.prototype.rh = function () {
        if (this.dataChart == null) {
            return;
        }
        this.dataChart.l0();
    };
    DomainChart.prototype.sk = function (a) {
        if (this.propertyChanged != null) {
            this.propertyChanged(this, new PropertyChangedEventArgs(a));
        }
    };
    DomainChart.prototype.rb = function (a) {
        this.rc(function (b, c) { return a(b); });
    };
    DomainChart.prototype.rc = function (a) {
        if (this.dataChart == null) {
            return;
        }
        for (var b = 0; b < this.dataChart.series.count; b++) {
            a(this.dataChart.series._inner[b], b);
        }
    };
    DomainChart.prototype.rf = function (a) {
        this.rg(function (b, c) { return a(b); });
    };
    DomainChart.prototype.rg = function (a) {
        if (this.dataChart == null) {
            return;
        }
        for (var b = 0; b < this.dataChart.series.count; b++) {
            var c = this.dataChart.series._inner[b];
            if (!c.db) {
                a(c, b);
            }
        }
    };
    DomainChart.prototype.rd = function (a) {
        this.re(function (b, c) { return a(b); });
    };
    DomainChart.prototype.re = function (a) {
        if (this.dataChart == null) {
            return;
        }
        for (var b = 0; b < this.dataChart.series.count; b++) {
            var c = this.dataChart.series._inner[b];
            if (c.dc) {
                a(c, b);
            }
        }
    };
    DomainChart.prototype.st = function () {
        this.rb(function (a) { return a.mo(); });
    };
    DomainChart.prototype.r4 = function (a) {
        this.r5(a, true);
    };
    DomainChart.prototype.r5 = function (a, b) {
        this.r6(a, b, null);
    };
    DomainChart.prototype.r6 = function (a, b, c) {
        if (this.dataChart == null) {
            return;
        }
        a(this.dataChart);
    };
    DomainChart.prototype.fv = function (a) {
        if (!this.fu(a.suggestedSeries)) {
            return false;
        }
        var b = a.data;
        if (b == null) {
            return false;
        }
        return true;
    };
    DomainChart.prototype.fn = function () {
        if (this.dataChart == null) {
            return false;
        }
        return true;
    };
    DomainChart.prototype.ny = function (a) {
        var b = "series_" + (this.dataChart.series.count + 1) + "_" + EnumUtil.getName(DataSeriesType_$type, a.suggestedSeries);
        return b;
    };
    DomainChart.prototype.nz = function (a, b) {
        if (this.fq && this.f3) {
            var c = "";
            var d = 5 + Math.round(DomainChart.j2.nextDouble() * 4);
            for (var e = 0; e < d; e++) {
                c = c + "\u2588";
            }
            return c;
        }
        if (stringIsNullOrEmpty(a.title)) {
            var f = EnumUtil.getName(DataSeriesType_$type, b);
            return f + (this.dataChart.series.count + 1);
        }
        else {
            return a.title;
        }
    };
    DomainChart.prototype.nx = function () {
        return stringEmpty();
    };
    DomainChart.prototype.sw = function () {
        var e_3, _a, e_4, _b;
        if (this.dataChart == null) {
            return;
        }
        this.fs = false;
        this.sy();
        if (this.actualItemsSource == null) {
            return;
        }
        var a = new List$1(MarkerType_$type, 0);
        try {
            for (var _c = __values(fromEnum(this.ba.dataSeries)), _d = _c.next(); !_d.done; _d = _c.next()) {
                var b = _d.value;
                a.add(this.ck(b.suggestedMarker));
            }
        }
        catch (e_3_1) {
            e_3 = { error: e_3_1 };
        }
        finally {
            try {
                if (_d && !_d.done && (_a = _c.return))
                    _a.call(_c);
            }
            finally {
                if (e_3)
                    throw e_3.error;
            }
        }
        this.gc = a;
        var c = 0;
        try {
            for (var _e = __values(fromEnum(this.ba.dataSeries)), _f = _e.next(); !_f.done; _f = _e.next()) {
                var d = _f.value;
                if (!this.fv(d)) {
                    continue;
                }
                if (d.index < 0) {
                    d.index = c;
                }
                this.su(d);
                this.sv(d);
                c++;
            }
        }
        catch (e_4_1) {
            e_4 = { error: e_4_1 };
        }
        finally {
            try {
                if (_f && !_f.done && (_b = _e.return))
                    _b.call(_e);
            }
            finally {
                if (e_4)
                    throw e_4.error;
            }
        }
        this.s3();
        this.rc(runOn(this, this.rr));
    };
    DomainChart.prototype.cw = function (seriesType_) {
        return (this.externalObject.createSeries(seriesType_).i);
    };
    DomainChart.prototype.cp = function () {
        return this.cw(28);
    };
    DomainChart.prototype.cr = function () {
        return this.cw(27);
    };
    DomainChart.prototype.ct = function () {
        return this.cw(29);
    };
    DomainChart.prototype.cu = function () {
        return this.cw(25);
    };
    DomainChart.prototype.cq = function () {
        return this.cw(26);
    };
    DomainChart.prototype.cs = function () {
        return this.cw(32);
    };
    Object.defineProperty(DomainChart.prototype, "b0", {
        get: function () {
            if (TypeRegistrar.isRegistered("AnnotationLayerProxy")) {
                return TypeRegistrar.create("AnnotationLayerProxy");
            }
            else {
                throw new BaseError(1, "annotation layer module is not loaded, but is required.");
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "fj", {
        get: function () {
            return (TypeRegistrar.isRegistered("AnnotationLayerProxy"));
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.fl = function (a) {
        if (!this.fj) {
            return false;
        }
        return this.b0.matchesType(a);
    };
    DomainChart.prototype.s3 = function () {
        var a = this.ga("", true);
        if (a != null) {
            for (var b = 0; b < a.count; b++) {
                this.dataChart.series.add(a._inner[b]);
            }
        }
    };
    DomainChart.prototype.fo = function () {
        return this.a6 == 0 || this.a6 == 4 || this.a6 == 2;
    };
    DomainChart.prototype.fp = function () {
        return this.a6 == 0 || this.a6 == 4 || this.a6 == 3;
    };
    DomainChart.prototype.get_fm = function () {
        return this.gg == null && this.ej && (this.j6 == null || this.j7 == null);
    };
    Object.defineProperty(DomainChart.prototype, "fm", {
        get: function () {
            return this.get_fm();
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.ga = function (a, b) {
        var e_5, _a, e_6, _b;
        if (!this.fj) {
            return null;
        }
        var c = new List$1(Series.$, 0);
        if (b) {
            if (this.ej) {
                var d = 1;
                if (this.gg == null) {
                    if (this.j6 != null || this.j7 != null || this.j5 != null || this.j4 != null) {
                        try {
                            for (var _c = __values(fromEnum(this.dataChart.series)), _d = _c.next(); !_d.done; _d = _c.next()) {
                                var e = _d.value;
                                var f = this.cp();
                                this.b0.setTargetSeries(f, e);
                                this.b0.setCollisionChannel(f, "MainChartCollisionChannel");
                                this.s1(f, e, this.ba.dataSeries._inner[d - 1]);
                                f.name = a + getInstanceType(f).typeName + d.toString();
                                c.add(f);
                                d++;
                            }
                        }
                        catch (e_5_1) {
                            e_5 = { error: e_5_1 };
                        }
                        finally {
                            try {
                                if (_d && !_d.done && (_a = _c.return))
                                    _a.call(_c);
                            }
                            finally {
                                if (e_5)
                                    throw e_5.error;
                            }
                        }
                        ;
                    }
                    else {
                        var g = this.cp();
                        this.q3(g, this.f2);
                        g.name = a + getInstanceType(g).typeName + d.toString();
                        c.add(g);
                    }
                }
                else {
                    var h = this.cp();
                    this.s1(h, null, this.ba.dataSeries._inner[d - 1]);
                    h.name = a + getInstanceType(h).typeName + d.toString();
                    c.add(h);
                }
            }
        }
        switch (this.a5) {
            case 0:
            case 4:
            case 3:
            case 2:
                {
                    var i = this.cr();
                    this.b0.setCrosshairThickness(i, this.gl);
                    this.b0.setCrosshairUseInterpolation(i, !this.el);
                    this.b0.setCrosshairHorizontalLineVisibility(i, this.fo() ? 0 : 1);
                    this.b0.setCrosshairHorizontalLineStroke(i, this.tc);
                    this.b0.setCrosshairVerticalLineVisibility(i, this.fp() ? 0 : 1);
                    this.b0.setCrosshairVerticalLineStroke(i, this.td);
                    this.b0.setCrosshairOnAxesEnabled(i, this.ek);
                    this.b0.setCrosshairXAxisBackground(i, this.s8);
                    this.b0.setCrosshairXAxisForeground(i, this.s9);
                    this.b0.setCrosshairYAxisBackground(i, this.ta);
                    this.b0.setCrosshairYAxisForeground(i, this.tb);
                    this.b0.setCrosshairXAxisPrecision(i, this.i7);
                    this.b0.setCrosshairYAxisPrecision(i, this.i8);
                    c.add(i);
                    break;
                }
            case 1:
                {
                    break;
                }
        }
        if (this.er) {
            var j = this.ct();
            this.b0.setFinalValueBackground(j, this.tn);
            this.b0.setFinalValueForeground(j, this.to);
            this.b0.setFinalValuePrecision(j, this.jb);
            c.add(j);
        }
        if (b) {
            switch (this.df) {
                case 1:
                    {
                        var k = this.cu();
                        c.add(k);
                        break;
                    }
                case 2:
                    {
                        var l = this.cq();
                        c.add(l);
                        break;
                    }
                case 4:
                    {
                        var m = this.cs();
                        c.add(m);
                        break;
                    }
                case 3:
                    {
                        this.rb(function (n) { return n.et = false; });
                        break;
                    }
                case 0:
                    {
                        this.rb(function (n) { return n.et = true; });
                        break;
                    }
            }
        }
        try {
            for (var _e = __values(fromEnum(c)), _f = _e.next(); !_f.done; _f = _e.next()) {
                var n = _f.value;
                if (stringIsNullOrEmpty(n.name) == false) {
                    continue;
                }
                n.name = a + getInstanceType(n).typeName;
            }
        }
        catch (e_6_1) {
            e_6 = { error: e_6_1 };
        }
        finally {
            try {
                if (_f && !_f.done && (_b = _e.return))
                    _b.call(_e);
            }
            finally {
                if (e_6)
                    throw e_6.error;
            }
        }
        return c;
    };
    DomainChart.prototype.q3 = function (a, b) {
        this.b0.setIsAutoCalloutBehaviorEnabled(a, true);
        this.b0.setGreedyCollisionMode(a, true);
        this.b0.setUseDedicatedLanes(a, b);
    };
    DomainChart.prototype.s1 = function (a, b, c) {
        a.itemsSource = this.gg != null ? this.gg : b.itemsSource;
        this.b0.setXMemberPath(a, this.j6);
        this.b0.setYMemberPath(a, this.j7);
        this.b0.setLabelMemberPath(a, this.j5);
        this.b0.setContentMemberPath(a, this.j4);
        if (this.ah != null) {
            this.b0.setCalloutsAllowedPositions(a, this.ah);
        }
    };
    DomainChart.prototype.sy = function () {
        this.sz(this.dataChart);
    };
    DomainChart.prototype.sz = function (a) {
        var e_7, _a;
        var _this = this;
        if (a != null && a == this.dataChart) {
            this.sx();
        }
        if (a == null || a.series.count == 0) {
            return;
        }
        this.rb(function (b) {
            if (_this.fl(b) && _this.b0.isCalloutLayer(b)) {
                _this.b0.removeCalloutStyleUpdating(b, runOn(_this, _this.sj));
            }
            if (_this.fl(b) && _this.b0.isCalloutLayer(b)) {
                _this.b0.removeCalloutRenderStyleUpdating(b, runOn(_this, _this.si));
            }
            _this.sr(b);
        });
        try {
            for (var _b = __values(fromEnum(a.series)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var b = _c.value;
                this.s0(b);
            }
        }
        catch (e_7_1) {
            e_7 = { error: e_7_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_7)
                    throw e_7.error;
            }
        }
        a.series.clear();
    };
    DomainChart.prototype.s0 = function (a) {
        a.itemsSource = null;
        a.legend = null;
        a.kx();
    };
    DomainChart.prototype.r7 = function (a, b) {
        if (this.dataChart == null) {
            return;
        }
        var c = b.oldItems == null ? "null" : b.oldItems.count.toString();
        var d = b.newItems == null ? "null" : b.newItems.count.toString();
        if (this.fs) {
            return;
        }
        var e = this.sh(runOn(this, this.q7), this.fs);
        this.fs = e.p1;
    };
    DomainChart.prototype.q7 = function () {
        this.sw();
    };
    DomainChart.prototype.uc = function (a) {
        return ArrayExtension.getModulus$1(Brush.$, this.y, a);
    };
    DomainChart.prototype.uf = function (a) {
        return ArrayExtension.getModulus$1(Brush.$, this.z, a);
    };
    DomainChart.prototype.ud = function (a) {
        return ArrayExtension.getModulus$1(Brush.$, this.ac, a);
    };
    DomainChart.prototype.ue = function (a) {
        return ArrayExtension.getModulus$1(Brush.$, this.ad, a);
    };
    DomainChart.prototype.un = function (a) {
        var b = new Color();
        if (this.y == null || this.y.count == 0) {
            b = ArrayExtension.getModulus$1(Color.$, this.f9, a);
        }
        else {
            var c = ArrayExtension.getModulus$1(Brush.$, this.y, a);
            if (c == null) {
                b = ArrayExtension.getModulus$1(Color.$, this.f9, a);
            }
            else {
                b = ColorUtil.q(c);
            }
        }
        return b;
    };
    DomainChart.prototype.ed = function (a) {
        var b = new ObservableColorCollection();
        b.add(ColorUtil.v(a, 0.5));
        b.add(a);
        return b;
    };
    DomainChart.prototype.ab = function (a) {
        var b = this.ed(a);
        var c = BrushCollectionUtil.a(b);
        return c;
    };
    Object.defineProperty(DomainChart.prototype, "f9", {
        get: function () {
            if (this.f7 != null) {
                return this.f7;
            }
            this.f7 = new List$1(Color.$, 0);
            this.f7.add(ColorUtil.r("#8BDC5C"));
            this.f7.add(ColorUtil.r("#8961A9"));
            this.f7.add(ColorUtil.r("#6DB1FF"));
            this.f7.add(ColorUtil.r("#82E9D9"));
            this.f7.add(ColorUtil.r("#EA3C63"));
            this.f7.add(ColorUtil.r("#735656"));
            this.f7.add(ColorUtil.r("#F8CE4F"));
            this.f7.add(ColorUtil.r("#A8A8B7"));
            this.f7.add(ColorUtil.r("#E051A9"));
            this.f7.add(ColorUtil.r("#FF903B"));
            return this.f7;
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.get_f2 = function () {
        return false;
    };
    Object.defineProperty(DomainChart.prototype, "f2", {
        get: function () {
            return this.get_f2();
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.ju = function () {
        if (this.dataChart == null) {
            return null;
        }
        return this.dataChart.vw();
    };
    DomainChart.prototype.rv = function (a, b) {
    };
    DomainChart.prototype.rw = function (a, b, c) {
    };
    DomainChart.prototype.ck = function (a) {
        if (a == 2) {
            return 3;
        }
        else if (a == 12) {
            return 13;
        }
        else if (a == 6) {
            return 7;
        }
        else if (a == 8) {
            return 9;
        }
        else if (a == 11) {
            return 12;
        }
        else if (a == 1) {
            return 1;
        }
        else if (a == 7) {
            return 8;
        }
        else if (a == 10) {
            return 11;
        }
        else if (a == 4) {
            return 5;
        }
        else if (a == 5) {
            return 6;
        }
        else if (a == 9) {
            return 10;
        }
        else if (a == 3) {
            return 4;
        }
        else if (a == 13) {
            return 0;
        }
        return 2;
    };
    DomainChart.prototype.attachDataLegend = function (a) {
        this.q2(a);
    };
    DomainChart.prototype.q2 = function (a) {
        this.b3 = a;
        if (this.dataChart != null) {
            if (!this.eu) {
                this.eu = true;
                this.dataChart.attachDataLegend(a);
            }
        }
    };
    DomainChart.prototype.detachDataLegend = function (a) {
        this.q6(a);
    };
    DomainChart.prototype.q6 = function (a) {
        if (this.dataChart != null) {
            this.eu = false;
            this.dataChart.detachDataLegend(a);
        }
        this.b3 = null;
    };
    DomainChart.prototype.getDataLegendSeries = function () {
        return this.p();
    };
    DomainChart.prototype.p = function () {
        if (this.dataChart != null) {
            return this.dataChart.getDataLegendSeries();
        }
        else {
            return new Array(0);
        }
    };
    DomainChart.prototype.r1 = function (source_, a, b, c) {
        if (this.ba != null) {
            this.ba.notifySetItem(source_, a, b, c);
        }
        if (this.dataChart != null) {
            this.dataChart.mi(source_, a, b, c);
        }
    };
    DomainChart.prototype.rx = function (source_) {
        if (this.ba != null) {
            this.ba.notifyClearItems(source_);
        }
        if (this.dataChart != null) {
            this.dataChart.l9(source_);
        }
    };
    DomainChart.prototype.ry = function (source_, a, b) {
        if (this.ba != null) {
            this.ba.notifyInsertItem(source_, a, b);
        }
        if (this.dataChart != null) {
            this.dataChart.mg(source_, a, b);
        }
    };
    DomainChart.prototype.rz = function (source_, a, b) {
        if (this.ba != null) {
            this.ba.notifyRemoveItem(source_, a, b);
        }
        if (this.dataChart != null) {
            this.dataChart.mh(source_, a, b);
        }
    };
    Object.defineProperty(DomainChart.prototype, "bs", {
        get: function () {
            return this.br;
        },
        set: function (a) {
            var b = this.br;
            this.br = a;
            if (b != this.br) {
                this.sc("DataToolTipGroupingMode", enumGetBox(DataToolTipLayerGroupingMode_$type, b), enumGetBox(DataToolTipLayerGroupingMode_$type, this.br));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "hy", {
        get: function () {
            return this.go;
        },
        set: function (a) {
            var b = this.go;
            this.go = a;
            if (b != this.go) {
                this.sc("DataToolTipPositionOffsetX", b, this.go);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "hz", {
        get: function () {
            return this.gp;
        },
        set: function (a) {
            var b = this.gp;
            this.gp = a;
            if (b != this.gp) {
                this.sc("DataToolTipPositionOffsetY", b, this.gp);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "hw", {
        get: function () {
            return this.gm;
        },
        set: function (a) {
            var b = this.gm;
            this.gm = a;
            if (b != this.gm) {
                this.sc("DataToolTipDefaultPositionOffsetX", b, this.gm);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "hx", {
        get: function () {
            return this.gn;
        },
        set: function (a) {
            var b = this.gn;
            this.gn = a;
            if (b != this.gn) {
                this.sc("DataToolTipDefaultPositionOffsetY", b, this.gn);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "bo", {
        get: function () {
            return this.bn;
        },
        set: function (a) {
            var b = this.bn;
            this.bn = a;
            if (b != this.bn) {
                this.sc("DataToolTipGroupedPositionModeX", enumGetBox(DataTooltipGroupedPositionX_$type, b), enumGetBox(DataTooltipGroupedPositionX_$type, this.bn));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "bq", {
        get: function () {
            return this.bp;
        },
        set: function (a) {
            var b = this.bp;
            this.bp = a;
            if (b != this.bp) {
                this.sc("DataToolTipGroupedPositionModeY", enumGetBox(DataTooltipGroupedPositionY_$type, b), enumGetBox(DataTooltipGroupedPositionY_$type, this.bp));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "fe", {
        get: function () {
            return this.en;
        },
        set: function (a) {
            var b = this.en;
            this.en = a;
            if (b != this.en) {
                this.sc(DomainChart.mj, b, this.en);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "m", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            var b = this.e;
            this.e = a;
            if (b != this.e) {
                this.sc("DataToolTipIncludedSeries", b, this.e);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "j", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            var b = this.b;
            this.b = a;
            if (b != this.b) {
                this.sc("DataToolTipExcludedSeries", b, this.b);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "l", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            var b = this.d;
            this.d = a;
            if (b != this.d) {
                this.sc("DataToolTipIncludedColumns", b, this.d);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "i", {
        get: function () {
            return this.a;
        },
        set: function (a) {
            var b = this.a;
            this.a = a;
            if (b != this.a) {
                this.sc("DataToolTipExcludedColumns", b, this.a);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "a8", {
        get: function () {
            return this.a7;
        },
        set: function (a) {
            var b = this.a7;
            this.a7 = a;
            if (b != this.a7) {
                this.sc("DataToolTipValueFormatAbbreviation", enumGetBox(DataAbbreviationMode_$type, b), enumGetBox(DataAbbreviationMode_$type, this.a7));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ji", {
        get: function () {
            return this.i9;
        },
        set: function (a) {
            var b = this.i9;
            this.i9 = a;
            if (b != this.i9) {
                this.sc("DataToolTipValueFormatMaxFractions", b, this.i9);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "jj", {
        get: function () {
            return this.ja;
        },
        set: function (a) {
            var b = this.ja;
            this.ja = a;
            if (b != this.ja) {
                this.sc("DataToolTipValueFormatMinFractions", b, this.ja);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "bm", {
        get: function () {
            return this.bl;
        },
        set: function (a) {
            var b = this.bl;
            this.bl = a;
            if (b != this.bl) {
                this.sc("DataToolTipValueFormatMode", enumGetBox(DataLegendValueMode_$type, b), enumGetBox(DataLegendValueMode_$type, this.bl));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "nb", {
        get: function () {
            return this.kf;
        },
        set: function (a) {
            var b = this.kf;
            this.kf = a;
            if (b != this.kf) {
                this.sc("DataToolTipValueFormatCulture", b, this.kf);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ff", {
        get: function () {
            return this.eo;
        },
        set: function (a) {
            var b = this.eo;
            this.eo = a;
            if (b != this.eo) {
                this.sc("DataToolTipValueFormatUseGrouping", b, this.eo);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "nh", {
        get: function () {
            return this.kg;
        },
        set: function (a) {
            var b = this.kg;
            this.kg = a;
            if (b != this.kg) {
                this.sc("DataToolTipValueFormatString", b, this.kg);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "n", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            var b = this.f;
            this.f = a;
            if (b != this.f) {
                this.sc("DataToolTipValueFormatSpecifiers", b, this.f);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "u9", {
        get: function () {
            return this.uz;
        },
        set: function (a) {
            var b = this.uz;
            this.uz = a;
            if (b != this.uz) {
                this.sc("DataToolTipValueRowMargin", b, this.uz);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "fg", {
        get: function () {
            return this.ep;
        },
        set: function (a) {
            var b = this.ep;
            this.ep = a;
            if (b != this.ep) {
                this.sc("DataToolTipValueRowVisible", b, this.ep);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "nq", {
        get: function () {
            return this.kh;
        },
        set: function (a) {
            var b = this.kh;
            this.kh = a;
            if (b != this.kh) {
                this.sc("DataToolTipValueTextWhenMissingData", b, this.kh);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "fh", {
        get: function () {
            return this.eq;
        },
        set: function (a) {
            var b = this.eq;
            this.eq = a;
            if (b != this.eq) {
                this.sc("DataToolTipValueTextUseSeriesColors", b, this.eq);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "va", {
        get: function () {
            return this.u0;
        },
        set: function (a) {
            var b = this.u0;
            this.u0 = a;
            if (b != this.u0) {
                this.sc("DataToolTipValueTextMargin", b, this.u0);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "t9", {
        get: function () {
            return this.tm;
        },
        set: function (a) {
            var b = this.tm;
            this.tm = a;
            if (b != this.tm) {
                this.sc("DataToolTipValueTextColor", b, this.tm);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ec", {
        get: function () {
            return this.d3;
        },
        set: function (a) {
            var b = this.d3;
            this.d3 = a;
            if (b != this.d3) {
                this.sc("DataToolTipValueTextStyle", b, this.d3);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "l1", {
        get: function () {
            return this.j9;
        },
        set: function (a) {
            var b = this.j9;
            this.j9 = a;
            if (b != this.j9) {
                this.sc("DataToolTipHeaderFormatString", b, this.j9);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "k", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            var b = this.c;
            this.c = a;
            if (b != this.c) {
                this.sc("DataToolTipHeaderFormatSpecifiers", b, this.c);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "lx", {
        get: function () {
            return this.j8;
        },
        set: function (a) {
            var b = this.j8;
            this.j8 = a;
            if (b != this.j8) {
                this.sc("DataToolTipHeaderFormatCulture", b, this.j8);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "bc", {
        get: function () {
            return this.bb;
        },
        set: function (a) {
            var b = this.bb;
            this.bb = a;
            if (b != this.bb) {
                this.sc("DataToolTipHeaderFormatDate", enumGetBox(DataLegendHeaderDateMode_$type, b), enumGetBox(DataLegendHeaderDateMode_$type, this.bb));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "be", {
        get: function () {
            return this.bd;
        },
        set: function (a) {
            var b = this.bd;
            this.bd = a;
            if (b != this.bd) {
                this.sc("DataToolTipHeaderFormatTime", enumGetBox(DataLegendHeaderTimeMode_$type, b), enumGetBox(DataLegendHeaderTimeMode_$type, this.bd));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "l6", {
        get: function () {
            return this.ka;
        },
        set: function (a) {
            var b = this.ka;
            this.ka = a;
            if (b != this.ka) {
                this.sc("DataToolTipHeaderText", b, this.ka);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "t1", {
        get: function () {
            return this.te;
        },
        set: function (a) {
            var b = this.te;
            this.te = a;
            if (b != this.te) {
                this.sc("DataToolTipHeaderTextColor", b, this.te);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "u3", {
        get: function () {
            return this.ut;
        },
        set: function (a) {
            var b = this.ut;
            this.ut = a;
            if (b != this.ut) {
                this.sc("DataToolTipHeaderTextMargin", b, this.ut);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "u2", {
        get: function () {
            return this.us;
        },
        set: function (a) {
            var b = this.us;
            this.us = a;
            if (b != this.us) {
                this.sc("DataToolTipHeaderRowMargin", b, this.us);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "fd", {
        get: function () {
            return this.em;
        },
        set: function (a) {
            var b = this.em;
            this.em = a;
            if (b != this.em) {
                this.sc("DataToolTipHeaderRowVisible", b, this.em);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "d4", {
        get: function () {
            return this.dv;
        },
        set: function (a) {
            var b = this.dv;
            this.dv = a;
            if (b != this.dv) {
                this.sc("DataToolTipHeaderTextStyle", b, this.dv);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "t4", {
        get: function () {
            return this.th;
        },
        set: function (a) {
            var b = this.th;
            this.th = a;
            if (b != this.th) {
                this.sc("DataToolTipSummaryTitleTextColor", b, this.th);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "d7", {
        get: function () {
            return this.dy;
        },
        set: function (a) {
            var b = this.dy;
            this.dy = a;
            if (b != this.dy) {
                this.sc("DataToolTipSummaryTitleTextStyle", b, this.dy);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "bi", {
        get: function () {
            return this.bh;
        },
        set: function (a) {
            var b = this.bh;
            this.bh = a;
            if (b != this.bh) {
                this.sc("DataToolTipSummaryType", enumGetBox(DataLegendSummaryType_$type, b), enumGetBox(DataLegendSummaryType_$type, this.bh));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "mp", {
        get: function () {
            return this.kc;
        },
        set: function (a) {
            var b = this.kc;
            this.kc = a;
            if (b != this.kc) {
                this.sc("DataToolTipSummaryTitleText", b, this.kc);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "u6", {
        get: function () {
            return this.uw;
        },
        set: function (a) {
            var b = this.uw;
            this.uw = a;
            if (b != this.uw) {
                this.sc("DataToolTipSummaryTitleTextMargin", b, this.uw);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "u5", {
        get: function () {
            return this.uv;
        },
        set: function (a) {
            var b = this.uv;
            this.uv = a;
            if (b != this.uv) {
                this.sc("DataToolTipSummaryRowMargin", b, this.uv);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "t6", {
        get: function () {
            return this.tj;
        },
        set: function (a) {
            var b = this.tj;
            this.tj = a;
            if (b != this.tj) {
                this.sc("DataToolTipSummaryValueTextColor", b, this.tj);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "d9", {
        get: function () {
            return this.d0;
        },
        set: function (a) {
            var b = this.d0;
            this.d0 = a;
            if (b != this.d0) {
                this.sc("DataToolTipSummaryValueTextStyle", b, this.d0);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "mk", {
        get: function () {
            return this.kb;
        },
        set: function (a) {
            var b = this.kb;
            this.kb = a;
            if (b != this.kb) {
                this.sc("DataToolTipSummaryLabelText", b, this.kb);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "t3", {
        get: function () {
            return this.tg;
        },
        set: function (a) {
            var b = this.tg;
            this.tg = a;
            if (b != this.tg) {
                this.sc("DataToolTipSummaryLabelTextColor", b, this.tg);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "d6", {
        get: function () {
            return this.dx;
        },
        set: function (a) {
            var b = this.dx;
            this.dx = a;
            if (b != this.dx) {
                this.sc("DataToolTipSummaryLabelTextStyle", b, this.dx);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "mv", {
        get: function () {
            return this.kd;
        },
        set: function (a) {
            var b = this.kd;
            this.kd = a;
            if (b != this.kd) {
                this.sc("DataToolTipSummaryUnitsText", b, this.kd);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "t5", {
        get: function () {
            return this.ti;
        },
        set: function (a) {
            var b = this.ti;
            this.ti = a;
            if (b != this.ti) {
                this.sc("DataToolTipSummaryUnitsTextColor", b, this.ti);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "d8", {
        get: function () {
            return this.dz;
        },
        set: function (a) {
            var b = this.dz;
            this.dz = a;
            if (b != this.dz) {
                this.sc("DataToolTipSummaryUnitsTextStyle", b, this.dz);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "u1", {
        get: function () {
            return this.ur;
        },
        set: function (a) {
            var b = this.ur;
            this.ur = a;
            if (b != this.ur) {
                this.sc("DataToolTipBadgeMargin", b, this.ur);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "cc", {
        get: function () {
            return this.ca;
        },
        set: function (a) {
            var b = this.ca;
            this.ca = a;
            if (b != this.ca) {
                this.sc("DataToolTipBadgeShape", enumGetBox(LegendItemBadgeShape_$type, b), enumGetBox(LegendItemBadgeShape_$type, this.ca));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "bk", {
        get: function () {
            return this.bj;
        },
        set: function (a) {
            var b = this.bj;
            this.bj = a;
            if (b != this.bj) {
                this.sc("DataToolTipUnitsDisplayMode", enumGetBox(DataLegendUnitsMode_$type, b), enumGetBox(DataLegendUnitsMode_$type, this.bj));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "m5", {
        get: function () {
            return this.ke;
        },
        set: function (a) {
            var b = this.ke;
            this.ke = a;
            if (b != this.ke) {
                this.sc("DataToolTipUnitsText", b, this.ke);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "t8", {
        get: function () {
            return this.tl;
        },
        set: function (a) {
            var b = this.tl;
            this.tl = a;
            if (b != this.tl) {
                this.sc("DataToolTipUnitsTextColor", b, this.tl);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "u8", {
        get: function () {
            return this.uy;
        },
        set: function (a) {
            var b = this.uy;
            this.uy = a;
            if (b != this.uy) {
                this.sc("DataToolTipUnitsTextMargin", b, this.uy);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "eb", {
        get: function () {
            return this.d2;
        },
        set: function (a) {
            var b = this.d2;
            this.d2 = a;
            if (b != this.d2) {
                this.sc("DataToolTipUnitsTextStyle", b, this.d2);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "u7", {
        get: function () {
            return this.ux;
        },
        set: function (a) {
            var b = this.ux;
            this.ux = a;
            if (b != this.ux) {
                this.sc("DataToolTipTitleTextMargin", b, this.ux);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "t7", {
        get: function () {
            return this.tk;
        },
        set: function (a) {
            var b = this.tk;
            this.tk = a;
            if (b != this.tk) {
                this.sc("DataToolTipTitleTextColor", b, this.tk);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ea", {
        get: function () {
            return this.d1;
        },
        set: function (a) {
            var b = this.d1;
            this.d1 = a;
            if (b != this.d1) {
                this.sc("DataToolTipTitleTextStyle", b, this.d1);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "bg", {
        get: function () {
            return this.bf;
        },
        set: function (a) {
            var b = this.bf;
            this.bf = a;
            if (b != this.bf) {
                this.sc("DataToolTipLabelDisplayMode", enumGetBox(DataLegendLabelMode_$type, b), enumGetBox(DataLegendLabelMode_$type, this.bf));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "t2", {
        get: function () {
            return this.tf;
        },
        set: function (a) {
            var b = this.tf;
            this.tf = a;
            if (b != this.tf) {
                this.sc("DataToolTipLabelTextColor", b, this.tf);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "u4", {
        get: function () {
            return this.uu;
        },
        set: function (a) {
            var b = this.uu;
            this.uu = a;
            if (b != this.uu) {
                this.sc("DataToolTipLabelTextMargin", b, this.uu);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "d5", {
        get: function () {
            return this.dw;
        },
        set: function (a) {
            var b = this.dw;
            this.dw = a;
            if (b != this.dw) {
                this.sc("DataToolTipLabelTextStyle", b, this.dw);
            }
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.sl = function (a) {
        if (this.seriesAdded != null && a != null) {
            this.seriesAdded(this, new ChartSeriesEventArgs(a));
        }
    };
    DomainChart.prototype.sr = function (a) {
        if (this.seriesRemoved != null && a != null) {
            this.seriesRemoved(this, new ChartSeriesEventArgs(a));
        }
    };
    DomainChart.prototype.sn = function (a, b) {
        if (this.seriesPointerEnter != null && a != null) {
            this.seriesPointerEnter(this, new DomainChartSeriesPointerEventArgs(a, b));
        }
    };
    DomainChart.prototype.so = function (a, b) {
        if (this.seriesPointerLeave != null && a != null) {
            this.seriesPointerLeave(this, new DomainChartSeriesPointerEventArgs(a, b));
        }
    };
    DomainChart.prototype.sp = function (a, b) {
        if (this.seriesPointerMove != null && a != null) {
            this.seriesPointerMove(this, new DomainChartSeriesPointerEventArgs(a, b));
        }
    };
    DomainChart.prototype.sm = function (a, b) {
        if (this.seriesPointerDown != null && a != null) {
            this.seriesPointerDown(this, new DomainChartSeriesPointerEventArgs(a, b));
        }
    };
    DomainChart.prototype.sq = function (a, b) {
        if (this.seriesPointerUp != null && a != null) {
            this.seriesPointerUp(this, new DomainChartSeriesPointerEventArgs(a, b));
        }
    };
    DomainChart.prototype.sj = function (a, b) {
        if (this.calloutStyleUpdating != null && a != null) {
            this.calloutStyleUpdating(this, b);
        }
    };
    DomainChart.prototype.si = function (a, b) {
        if (this.calloutRenderStyleUpdating != null && a != null) {
            this.calloutRenderStyleUpdating(this, b);
        }
    };
    DomainChart.prototype.sg = function (a) {
        var _this = this;
        if (!this.gb.contains(a)) {
            this.gb.add(a);
            var b = false;
            var c = this.sh(function () {
                a();
                _this.gb.remove(a);
            }, b);
            b = c.p1;
        }
    };
    DomainChart.prototype.sh = function (a, b) {
        if (!b) {
            b = true;
            if (this.jt != null) {
                this.jt.setTimeout(a, 0);
            }
            else {
                this.ee = a;
            }
        }
        return {
            p1: b
        };
    };
    DomainChart.prototype.get_fw = function () {
        return true;
    };
    Object.defineProperty(DomainChart.prototype, "fw", {
        get: function () {
            return this.get_fw();
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.rl = function (a) {
        var _this = this;
        var b = a == this.dataChart;
        if (b) {
            a.ag = this.y;
        }
        if (b) {
            a.aj = this.z;
        }
        a.ah = this.ac;
        a.ai = this.ad;
        a.be = this.cf;
        a.legend = this.legend;
        a.uk = this.fr;
        a.un = this.fx && this.fw;
        a.e9 = this.f1;
        if (b) {
            a.b5 = this.db;
            a.b6 = this.dc;
            a.go = this.h0;
            a.so = this.ug;
            a.gp = this.h1;
            a.gq = this.h2;
            a.gr = this.h3;
            a.sp = this.uh;
            a.b2 = this.c8;
            a.d5 = this.fk;
            a.gs = this.h4;
            a.gt = this.h5;
            a.gu = this.h6;
            a.hb = this.iu;
            a.sx = this.ul;
            a.hh = this.i0;
            a.hc = this.iv;
            a.hd = this.iw;
            a.sy = this.um;
            a.b7 = this.de;
            a.fj = this.f4;
            a.he = this.ix;
            a.hf = this.iy;
            a.hg = this.iz;
            a.hk = this.i1;
            a.hl = this.i2;
            a.hn = this.i4;
            a.hm = this.i3;
        }
        a.t5 = this.c4;
        a.t8 = this.c6;
        if (this.b3 != null) {
            if (!this.eu) {
                this.q2(this.b3);
            }
        }
        if (a.bv != this.c0 || a.bt != this.cy) {
            a.cu.b4(this.jt);
            a.bv = this.c0;
            a.bt = this.cy;
        }
        a.ty = this.a4;
        if (!isNaN_(this.ic)) {
            a.vp = this.ic;
        }
        if (!isNaN_(this.ie)) {
            a.vr = this.ie;
        }
        if (!isNaN_(this.id)) {
            a.vq = this.id;
        }
        if (!isNaN_(this.ib)) {
            a.vo = this.ib;
        }
        a.bb = this.b7;
        a.seriesMouseEnter = delegateCombine(a.seriesMouseEnter, function (c, d) { return _this.sn(d.series, d.item); });
        a.seriesMouseLeave = delegateCombine(a.seriesMouseLeave, function (c, d) { return _this.so(d.series, d.item); });
        a.seriesMouseMove = delegateCombine(a.seriesMouseMove, function (c, d) { return _this.sp(d.series, d.item); });
        a.seriesMouseLeftButtonDown = delegateCombine(a.seriesMouseLeftButtonDown, function (c, d) { return _this.sm(d.series, d.item); });
        a.seriesMouseLeftButtonUp = delegateCombine(a.seriesMouseLeftButtonUp, function (c, d) { return _this.sq(d.series, d.item); });
        if (b) {
            a.tp = this.uq;
        }
        if (b) {
            a.kg = this.pv;
        }
        a.r8 = this.s7;
        if (b) {
            a.j4 = this.pk;
        }
        a.r7 = this.s6;
        a.ug = this.e6;
        a.ds = this.e7;
        a.gw = this.h7;
        a.g1 = this.ih;
        a.ha = this.is;
        a.gj = this.hu;
        if (a == this.dataChart) {
            this.sw();
        }
        this.rm(a);
        this.rn(a);
    };
    DomainChart.prototype.af = function (a) {
        var e_8, _a;
        if (!this.f0 || !this.f3 || this.itemsSource != null) {
            return a;
        }
        if (a == null) {
            return null;
        }
        var b = new BrushCollection();
        try {
            for (var _b = __values(fromEnum(a)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var c = _c.value;
                var d = BrushUtil.r(c);
                b.add(d);
            }
        }
        catch (e_8_1) {
            e_8 = { error: e_8_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_8)
                    throw e_8.error;
            }
        }
        return b;
    };
    DomainChart.prototype.ui = function (a) {
        if (!this.f0 || !this.f3 || this.itemsSource != null) {
            return a;
        }
        var b = BrushUtil.g(0, 0, 0, 0);
        return b;
    };
    DomainChart.prototype.rr = function (a, b) {
        if (!isNaN_(this.ig)) {
            a.gv = this.ig;
        }
        if (!isNaN_(this.im) && !a.dc) {
            a.g1 = this.im;
        }
        if (!isNaN_(this.hv) && a.dc) {
            a.g1 = this.hv;
        }
        a.legendItemBadgeShape = this.cd;
        a.a9 = this.b9;
        a.bt = this.c2;
        a.m3(this.cj);
        a.m2(this.ch);
        a.gs = this.h8;
        if (!isNaN_(this.h9)) {
            a.m4(this.h9);
        }
        if (!isNaN_(this.ht)) {
            a.f1 = this.ht;
        }
        if (a.isUsableInLegend) {
            a.legend = this.legend;
        }
        if (typeCast(IUnknownValuePlotting_$type, a) !== null) {
            a.unknownValuePlotting = this.dk;
        }
        a.dx = this.ft;
        if (this.fl(a) && this.b0.isCalloutLayer(a)) {
            this.b0.setCalloutStyleUpdatingEventEnabled(a, this.e9);
            this.b0.addCalloutStyleUpdating(a, runOn(this, this.sj));
        }
        if (this.fl(a) && this.b0.isCalloutLayer(a)) {
            this.b0.setCalloutRenderStyleUpdatingEventEnabled(a, this.e9);
            this.b0.addCalloutRenderStyleUpdating(a, runOn(this, this.si));
        }
        if (this.fl(a) && this.b0.isDataToolTipLayer(a)) {
            this.qz(a);
        }
        this.q0(a, b);
        this.q1(a, b);
        this.rs(a, b);
        this.rt(a, b);
        this.sl(a);
    };
    Object.defineProperty(DomainChart.prototype, "e8", {
        get: function () {
            return this.eh;
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.q8 = function () {
        if (!this.eh) {
            this.eh = true;
            this.ri();
        }
    };
    DomainChart.prototype.ri = function () {
        this.rj();
        this.rk();
    };
    DomainChart.prototype.ss = function (a) {
        if (this.eu && this.b3 != null) {
            this.eu = false;
            if (a != null) {
                a.detachDataLegend(this.b3);
            }
        }
        this.sz(a);
    };
    DomainChart.prototype.q1 = function (a, b) {
        var c = typeCast(ITrendLineSeries_$type, a);
        if (c != null) {
            c.trendLineBrush = ArrayExtension.getModulus$1(Brush.$, this.ag, b);
            c.trendLineThickness = this.it;
            c.trendLineType = this.di;
            c.trendLinePeriod = this.jn;
        }
    };
    DomainChart.prototype.cm = function (a, b) {
        if (this.co == null || this.co.count == 0) {
            if (this.gc == null || this.gc.count == 0) {
                return a != null && a.d3 ? 1 : 2;
            }
            var c = ArrayExtension.getModulus$1(MarkerType_$type, this.gc, b);
            if (c == 0 && a != null) {
                c = a.d3 ? 1 : 2;
                if (c == 2) {
                    c = this.cl(a);
                }
            }
            if (c == 2) {
                if (a == null) {
                    return 1;
                }
                else {
                    return 2;
                }
            }
            return c;
        }
        return ArrayExtension.getModulus$1(MarkerType_$type, this.co, b);
    };
    DomainChart.prototype.cl = function (a) {
        return 2;
    };
    DomainChart.prototype.jy = function () {
        if (this.itemsSource != null) {
            return this.itemsSource;
        }
        if (this.f0) {
            return this.gi();
        }
        return null;
    };
    DomainChart.prototype.get_fq = function () {
        if (this.itemsSource == null && this.f0) {
            if (this.gi() != null) {
                return true;
            }
        }
        return false;
    };
    Object.defineProperty(DomainChart.prototype, "fq", {
        get: function () {
            return this.get_fq();
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.gi = function () {
        return null;
    };
    DomainChart.prototype.sc = function (a, b, c) {
        var _this = this;
        if (this.propertyChanged != null) {
            this.propertyChanged(this, new PropertyChangedEventArgs(a));
        }
        switch (a) {
            case "DataChart":
                this.ss(typeCast(XamDataChart.$, b));
                this.rl(typeCast(XamDataChart.$, c));
                break;
            case "ShouldUseSkeletonStyleForMockData":
            case "ShouldDisplayMockData":
                this.actualItemsSource = this.jy();
                this.y = this.af(this.aa);
                this.z = this.af(this.ae);
                break;
            case "ActualBrushes":
                this.r6(function (d) { return d.ag = _this.y; }, true, a);
                break;
            case "ActualOutlines":
                this.r6(function (d) { return d.aj = _this.z; }, true, a);
                break;
            case "ActualItemsSource":
                if (this.f0) {
                    this.y = this.af(this.aa);
                    this.z = this.af(this.ae);
                }
                this.ba.itemsSource = this.actualItemsSource;
                break;
            case "ItemsSource":
                this.actualItemsSource = this.jy();
                break;
            case "IncludedProperties":
                this.ba.includedProperties = this.q;
                break;
            case "ExcludedProperties":
                this.ba.excludedProperties = this.o;
                break;
            case "Brushes":
                this.y = this.af(this.aa);
                break;
            case "Outlines":
                this.z = this.af(this.ae);
                break;
            case "Legend":
                this.r4(function (d) { return d.legend = _this.legend; });
                break;
            case "HighlightingMode":
                this.r4(function (d) { return d.bv = _this.c0; });
                break;
            case "HighlightingBehavior":
                this.r4(function (d) { return d.bt = _this.cy; });
                break;
            case "LegendHighlightingMode":
                this.r4(function (d) { return d.bb = _this.b7; });
                break;
            case "LegendItemBadgeShape":
                this.r4(function (d) { return d.bd = _this.cd; });
                break;
            case "LegendItemBadgeMode":
                this.r4(function (d) { return d.bc = _this.b9; });
                break;
            case "PlotAreaMarginLeft":
                this.r4(function (d) { return d.vp = _this.ic; });
                break;
            case "PlotAreaMarginTop":
                this.r4(function (d) { return d.vr = _this.ie; });
                break;
            case "PlotAreaMarginRight":
                this.r4(function (d) { return d.vq = _this.id; });
                break;
            case "PlotAreaMarginBottom":
                this.r4(function (d) { return d.vo = _this.ib; });
                break;
            case "ComputedPlotAreaMarginMode":
                this.r4(function (d) { return d.ty = _this.a4; });
                break;
            case "IsHorizontalZoomEnabled":
                this.r4(function (d) { return d.uk = _this.fr; });
                break;
            case "IsVerticalZoomEnabled":
                this.r4(function (d) { return d.un = _this.fx; });
                break;
            case "WindowRect":
                this.r5(function (d) { return d.tp = _this.uq; }, false);
                break;
            case "WindowRectMinWidth":
                this.r5(function (d) { return d.hl = _this.i2; }, false);
                break;
            case "WindowRectMinHeight":
                this.r5(function (d) { return d.hk = _this.i1; }, false);
                break;
            case "WindowSizeMinWidth":
                this.r5(function (d) { return d.hn = _this.i4; }, false);
                break;
            case "WindowSizeMinHeight":
                this.r5(function (d) { return d.hm = _this.i3; }, false);
                break;
            case "Title":
                this.r5(function (d) { return d.kg = _this.pv; }, false);
                break;
            case "Subtitle":
                this.r5(function (d) { return d.j4 = _this.pk; }, false);
                break;
            case "TitleAlignment":
                this.r4(function (d) { return d.r8 = _this.s7; });
                break;
            case "SubtitleAlignment":
                this.r4(function (d) { return d.r7 = _this.s6; });
                break;
            case "UnknownValuePlotting":
                this.rb(function (d) {
                    if (typeCast(IUnknownValuePlotting_$type, d) !== null) {
                        d.unknownValuePlotting = _this.dk;
                    }
                });
                break;
            case "IsSeriesHighlightingEnabled":
                this.rb(function (d) { return d.dx = _this.ft; });
                break;
            case "Thickness":
                if (!isNaN_(this.im)) {
                    this.rf(function (d) { return d.g1 = _this.im; });
                }
                break;
            case "OutlineMode":
                this.rf(function (d) { return d.bt = _this.c2; });
                break;
            case "MarkerFillOpacity":
                this.rf(function (d) { return d.gs = _this.h8; });
                break;
            case "MarkerThickness":
                if (!isNaN_(this.h9)) {
                    this.rf(function (d) { return d.m4(_this.h9); });
                }
                break;
            case "MarkerOutlineMode":
                this.rf(function (d) { return d.m3(_this.cj); });
                break;
            case "MarkerFillMode":
                this.rf(function (d) { return d.m2(_this.ch); });
                break;
            case "Resolution":
                if (!isNaN_(this.ig)) {
                    this.rb(function (d) { return d.gv = _this.ig; });
                }
                break;
            case "AreaFillOpacity":
                if (!isNaN_(this.ht)) {
                    this.rb(function (d) { return d.f1 = _this.ht; });
                }
                break;
            case "ShouldAvoidAxisAnnotationCollisions":
                this.ra(function (d) {
                    d.ct = _this.fz;
                    d.cv = _this.fz;
                }, true);
                break;
            case "ShouldPanOnMaximumZoom":
                this.r4(function (d) { return d.e9 = _this.f1; });
                break;
            case "TrendLineBrushes":
            case "TrendLineType":
            case "TrendLineThickness":
            case "TrendLinePeriod":
                this.rc(function (d, e) { return _this.q1(d, e); });
                break;
            case "MarkerMaxCount":
                this.rg(function (d, e) { return _this.q0(d, e); });
                break;
            case "MarkerTypes":
                var d = typeCast(ObservableCollection$1.$.specialize(MarkerType_$type), b);
                if (d != null) {
                    d.collectionChanged = delegateRemove(d.collectionChanged, runOn(this, this.sa));
                }
                var e = typeCast(ObservableCollection$1.$.specialize(MarkerType_$type), c);
                if (e != null) {
                    e.collectionChanged = delegateCombine(e.collectionChanged, runOn(this, this.sa));
                }
                this.rg(function (f, g) { return _this.q0(f, g); });
                break;
            case "MarkerAutomaticBehavior":
                this.r4(function (f) { return f.be = _this.cf; });
                this.rg(function (f, g) { return _this.q0(f, g); });
                break;
            case "MarkerBrushes":
                this.r4(function (f) { return f.ah = _this.ac; });
                break;
            case "MarkerOutlines":
                this.r4(function (f) { return f.ai = _this.ad; });
                break;
            case "AlignsGridLinesToPixels":
                this.r4(function (f) { return f.ug = _this.e6; });
                break;
            case "AnimateSeriesWhenAxisRangeChanges":
                this.r4(function (f) { return f.ds = _this.e7; });
                break;
            case "ToolTipType":
            case "CrosshairsDisplayMode":
            case "CrosshairsSnapToData":
            case "CrosshairsLineVerticalStroke":
            case "CrosshairsLineHorizontalStroke":
            case "CrosshairsLineThickness":
            case "CrosshairsAnnotationXAxisBackground":
            case "CrosshairsAnnotationXAxisTextColor":
            case "CrosshairsAnnotationYAxisBackground":
            case "CrosshairsAnnotationYAxisTextColor":
            case "CrosshairsAnnotationEnabled":
            case "CrosshairsAnnotationXAxisPrecision":
            case "CrosshairsAnnotationYAxisPrecision":
            case "FinalValueAnnotationsVisible":
            case "FinalValueAnnotationsPrecision":
            case "FinalValueAnnotationsTextColor":
            case "FinalValueAnnotationsBackground":
            case "CalloutsVisible":
            case "CalloutsAllowedPositions":
            case "CalloutsItemsSource":
            case "CalloutsXMemberPath":
            case "CalloutsYMemberPath":
            case "CalloutsLabelMemberPath":
            case "CalloutsContentMemberPath":
                var f = this.sh(runOn(this, this.q7), this.fs);
                this.fs = f.p1;
                break;
            case "DataToolTipGroupingMode":
            case "DataToolTipPositionOffsetX":
            case "DataToolTipPositionOffsetY":
            case "DataToolTipDefaultPositionOffsetX":
            case "DataToolTipDefaultPositionOffsetY":
            case "DataToolTipGroupedPositionModeX":
            case "DataToolTipGroupedPositionModeY":
            case DomainChart.mj:
            case "DataToolTipIncludedSeries":
            case "DataToolTipExcludedSeries":
            case "DataToolTipIncludedColumns":
            case "DataToolTipExcludedColumns":
            case "DataToolTipValueFormatAbbreviation":
            case "DataToolTipValueFormatMaxFractions":
            case "DataToolTipValueFormatMinFractions":
            case "DataToolTipValueFormatMode":
            case "DataToolTipValueFormatCulture":
            case "DataToolTipValueFormatUseGrouping":
            case "DataToolTipValueFormatString":
            case "DataToolTipValueFormatSpecifiers":
            case "DataToolTipValueRowMargin":
            case "DataToolTipValueRowVisible":
            case "DataToolTipValueTextWhenMissingData":
            case "DataToolTipValueTextUseSeriesColors":
            case "DataToolTipValueTextMargin":
            case "DataToolTipValueTextColor":
            case "DataToolTipValueTextStyle":
            case "DataToolTipHeaderFormatString":
            case "DataToolTipHeaderFormatSpecifiers":
            case "DataToolTipHeaderFormatCulture":
            case "DataToolTipHeaderFormatDate":
            case "DataToolTipHeaderFormatTime":
            case "DataToolTipHeaderText":
            case "DataToolTipHeaderTextColor":
            case "DataToolTipHeaderTextMargin":
            case "DataToolTipHeaderRowMargin":
            case "DataToolTipHeaderRowVisible":
            case "DataToolTipHeaderTextStyle":
            case "DataToolTipSummaryTitleTextColor":
            case "DataToolTipSummaryTitleTextStyle":
            case "DataToolTipSummaryType":
            case "DataToolTipSummaryTitleText":
            case "DataToolTipSummaryTitleTextMargin":
            case "DataToolTipSummaryRowMargin":
            case "DataToolTipSummaryValueTextColor":
            case "DataToolTipSummaryValueTextStyle":
            case "DataToolTipSummaryLabelText":
            case "DataToolTipSummaryLabelTextColor":
            case "DataToolTipSummaryLabelTextStyle":
            case "DataToolTipSummaryUnitsText":
            case "DataToolTipSummaryUnitsTextColor":
            case "DataToolTipSummaryUnitsTextStyle":
            case "DataToolTipBadgeMargin":
            case "DataToolTipBadgeShape":
            case "DataToolTipUnitsDisplayMode":
            case "DataToolTipUnitsText":
            case "DataToolTipUnitsTextColor":
            case "DataToolTipUnitsTextMargin":
            case "DataToolTipUnitsTextStyle":
            case "DataToolTipTitleTextMargin":
            case "DataToolTipTitleTextColor":
            case "DataToolTipTitleTextStyle":
            case "DataToolTipLabelDisplayMode":
            case "DataToolTipLabelTextColor":
            case "DataToolTipLabelTextMargin":
            case "DataToolTipLabelTextStyle":
                this.r8(a, b, c);
                break;
            case "HorizontalViewScrollbarMode":
                this.r4(function (g) { return g.b5 = _this.db; });
                break;
            case "VerticalViewScrollbarMode":
                this.r4(function (g) { return g.b6 = _this.dc; });
                break;
            case "HorizontalViewScrollbarCornerRadius":
                this.r4(function (g) { return g.go = _this.h0; });
                break;
            case "HorizontalViewScrollbarFill":
                this.r4(function (g) { return g.so = _this.ug; });
                break;
            case "HorizontalViewScrollbarMaxOpacity":
                this.r4(function (g) { return g.gr = _this.h3; });
                break;
            case "HorizontalViewScrollbarOutline":
                this.r4(function (g) { return g.sp = _this.uh; });
                break;
            case "HorizontalViewScrollbarStrokeThickness":
                this.r4(function (g) { return g.gs = _this.h4; });
                break;
            case "VerticalViewScrollbarCornerRadius":
                this.r4(function (g) { return g.hb = _this.iu; });
                break;
            case "VerticalViewScrollbarFill":
                this.r4(function (g) { return g.sx = _this.ul; });
                break;
            case "VerticalViewScrollbarMaxOpacity":
                this.r4(function (g) { return g.hd = _this.iw; });
                break;
            case "VerticalViewScrollbarOutline":
                this.r4(function (g) { return g.sy = _this.um; });
                break;
            case "VerticalViewScrollbarStrokeThickness":
                this.r4(function (g) { return g.he = _this.ix; });
                break;
            case "HorizontalViewScrollbarHeight":
                this.r4(function (g) { return g.gp = _this.h1; });
                break;
            case "HorizontalViewScrollbarInset":
                this.r4(function (g) { return g.gq = _this.h2; });
                break;
            case DomainChart.oc:
                this.r4(function (g) { return g.d5 = _this.fk; });
                break;
            case "HorizontalViewScrollbarTrackStartInset":
                this.r4(function (g) { return g.gu = _this.h6; });
                break;
            case "HorizontalViewScrollbarTrackEndInset":
                this.r4(function (g) { return g.gt = _this.h5; });
                break;
            case "VerticalViewScrollbarPosition":
                this.r4(function (g) { return g.b7 = _this.de; });
                break;
            case "HorizontalViewScrollbarPosition":
                this.r4(function (g) { return g.b2 = _this.c8; });
                break;
        }
        this.sd(a, b, c);
        this.se(a, b, c);
    };
    DomainChart.prototype.r8 = function (a, b, c) {
        var _this = this;
        if (!this.f5.containsKey(a)) {
            this.f6.add(a);
            this.f5.addItem(a, c);
        }
        else {
            this.f5.item(a, c);
        }
        this.rd(function (d) {
            if (_this.b0.isDataToolTipLayer(d)) {
                _this.qz(d);
            }
        });
    };
    DomainChart.prototype.qz = function (a) {
        for (var b = 0; b < this.f6.count; b++) {
            var c = this.f6._inner[b];
            var d = this.f5.item(c);
            if (stringStartsWith(c, "DataToolTip")) {
                c = c.substr(("DataToolTip").length);
            }
            this.b0.updateDataToolTipPropertyByName(a, c, d);
        }
    };
    DomainChart.prototype.sa = function (a, b) {
        var _this = this;
        this.rg(function (c, d) { return _this.q0(c, d); });
    };
    Object.defineProperty(DomainChart.prototype, "ay", {
        get: function () {
            return this.as;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "au", {
        get: function () {
            return this.ao;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "at", {
        get: function () {
            return this.an;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "dq", {
        get: function () {
            return this.dn;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "a2", {
        get: function () {
            return this.a0;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ax", {
        get: function () {
            return this.ar;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "aw", {
        get: function () {
            return this.aq;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "av", {
        get: function () {
            return this.ap;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "a1", {
        get: function () {
            return this.az;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "oj", {
        get: function () {
            return this.kk;
        },
        set: function (a) {
            var _this = this;
            var b = this.kk;
            this.kk = a;
            if (b != a) {
                this.ar.n();
                this.dr.b(this.kk, function (c, d, e) { return _this.ar.j.add(new ChartSortDescription(2, c, d)); });
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "n0", {
        get: function () {
            return this.ki;
        },
        set: function (a) {
            var _this = this;
            var b = this.ki;
            this.ki = a;
            if (b != a) {
                this.aq.n();
                this.dr.b(this.ki, function (c, d, e) { return _this.aq.j.add(new ChartSortDescription(2, c, d)); });
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "oh", {
        get: function () {
            return this.kj;
        },
        set: function (a) {
            var _this = this;
            var b = this.kj;
            this.kj = a;
            if (b != a) {
                this.ap.n();
                this.dr.b(this.kj, function (c, d, e) { return _this.ap.j.add(new ChartSortDescription(2, c, d)); });
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ol", {
        get: function () {
            return this.kl;
        },
        set: function (a) {
            var _this = this;
            var b = this.kl;
            this.kl = a;
            if (b != a) {
                this.az.n();
                this.ds.a(this.kl, function (c, d, e) {
                    return _this.az.j.add(((function () {
                        var $ret = new ChartSummaryDescription(2, c, d);
                        $ret.n = e;
                        return $ret;
                    })()));
                });
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "itemsSource", {
        get: function () {
            return this.jp;
        },
        set: function (a) {
            var b = this.itemsSource;
            if (b != a) {
                this.jp = a;
                this.sc("ItemsSource", b, this.itemsSource);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "actualItemsSource", {
        get: function () {
            return this.jo;
        },
        set: function (a) {
            var b = this.actualItemsSource;
            if (b != a) {
                this.jo = a;
                this.sc("ActualItemsSource", b, this.actualItemsSource);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "q", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            var b = this.h;
            this.h = a;
            if (b != this.h) {
                this.sc("IncludedProperties", b, this.h);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "o", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            var b = this.g;
            this.g = a;
            if (b != this.g) {
                this.sc("ExcludedProperties", b, this.g);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "aa", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            var b = this.aa;
            if (a != b) {
                this.t = a;
                this.sc("Brushes", b, this.aa);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ae", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            var b = this.ae;
            if (a != b) {
                this.w = a;
                this.sc("Outlines", b, this.ae);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "y", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            var b = this.r;
            if (a != b) {
                this.r = a;
                this.sc("ActualBrushes", b, this.y);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "z", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            var b = this.s;
            if (a != b) {
                this.s = a;
                this.sc("ActualOutlines", b, this.z);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "legend", {
        get: function () {
            return this.b1;
        },
        set: function (a) {
            var b = this.legend;
            if (a != b) {
                this.b1 = a;
                this.sc("Legend", b, this.legend);
            }
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.get_fr = function () {
        return this.et;
    };
    DomainChart.prototype.set_fr = function (a) {
        var b = this.fr;
        if (a != b) {
            this.et = a;
            this.sc("IsHorizontalZoomEnabled", b, this.fr);
        }
    };
    Object.defineProperty(DomainChart.prototype, "fr", {
        get: function () {
            return this.get_fr();
        },
        set: function (a) {
            this.set_fr(a);
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.get_fx = function () {
        return this.ew;
    };
    DomainChart.prototype.set_fx = function (a) {
        var b = this.fx;
        if (a != b) {
            this.ew = a;
            this.sc("IsVerticalZoomEnabled", b, this.fx);
        }
    };
    Object.defineProperty(DomainChart.prototype, "fx", {
        get: function () {
            return this.get_fx();
        },
        set: function (a) {
            this.set_fx(a);
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.get_f0 = function () {
        return this.ey;
    };
    DomainChart.prototype.set_f0 = function (a) {
        var b = this.f0;
        if (a != b) {
            this.ey = a;
            this.sc("ShouldDisplayMockData", b, this.f0);
        }
    };
    Object.defineProperty(DomainChart.prototype, "f0", {
        get: function () {
            return this.get_f0();
        },
        set: function (a) {
            this.set_f0(a);
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.get_f3 = function () {
        return this.e4;
    };
    DomainChart.prototype.set_f3 = function (a) {
        var b = this.f3;
        if (a != b) {
            this.e4 = a;
            this.sc("ShouldUseSkeletonStyleForMockData", b, this.f3);
        }
    };
    Object.defineProperty(DomainChart.prototype, "f3", {
        get: function () {
            return this.get_f3();
        },
        set: function (a) {
            this.set_f3(a);
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.get_ft = function () {
        return this.ev;
    };
    DomainChart.prototype.set_ft = function (a) {
        var b = this.ft;
        if (a != b) {
            this.ev = a;
            this.sc("IsSeriesHighlightingEnabled", b, this.ft);
        }
    };
    Object.defineProperty(DomainChart.prototype, "ft", {
        get: function () {
            return this.get_ft();
        },
        set: function (a) {
            this.set_ft(a);
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "uq", {
        get: function () {
            return this.uo;
        },
        set: function (a) {
            var b = this.uq;
            if (Rect.l_op_Inequality(a, b)) {
                this.uo = a;
                this.sc("WindowRect", b, this.uq);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "up", {
        get: function () {
            return this.dataChart != null ? this.dataChart.to : Rect.empty;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "pv", {
        get: function () {
            return this.ko;
        },
        set: function (a) {
            var b = this.pv;
            if (a != b) {
                this.ko = a;
                this.sc("Title", b, this.pv);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "pk", {
        get: function () {
            return this.km;
        },
        set: function (a) {
            var b = this.pk;
            if (a != b) {
                this.km = a;
                this.sc("Subtitle", b, this.pk);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "s7", {
        get: function () {
            return this.s5;
        },
        set: function (a) {
            var b = this.s7;
            if (b != a) {
                this.s5 = a;
                this.sc("TitleAlignment", enumGetBox(HorizontalAlignment_$type, b), enumGetBox(HorizontalAlignment_$type, this.s7));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "s6", {
        get: function () {
            return this.s4;
        },
        set: function (a) {
            var b = this.s6;
            if (b != a) {
                this.s4 = a;
                this.sc("SubtitleAlignment", enumGetBox(HorizontalAlignment_$type, b), enumGetBox(HorizontalAlignment_$type, this.s6));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "dk", {
        get: function () {
            return this.dj;
        },
        set: function (a) {
            var b = this.dk;
            if (a != b) {
                this.dj = a;
                this.sc("UnknownValuePlotting", enumGetBox(UnknownValuePlotting_$type, b), enumGetBox(UnknownValuePlotting_$type, this.dk));
            }
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.get_ig = function () {
        return this.g5;
    };
    DomainChart.prototype.set_ig = function (a) {
        var b = this.ig;
        if (a != b) {
            this.g5 = a;
            this.sc("Resolution", b, this.ig);
        }
    };
    Object.defineProperty(DomainChart.prototype, "ig", {
        get: function () {
            return this.get_ig();
        },
        set: function (a) {
            this.set_ig(a);
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "im", {
        get: function () {
            return this.hb;
        },
        set: function (a) {
            var b = this.im;
            if (a != b) {
                this.hb = a;
                this.sc("Thickness", b, this.im);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "c2", {
        get: function () {
            return this.c1;
        },
        set: function (a) {
            var b = this.c2;
            if (a != b) {
                this.c1 = a;
                this.sc("OutlineMode", enumGetBox(SeriesOutlineMode_$type, b), enumGetBox(SeriesOutlineMode_$type, this.c2));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "cj", {
        get: function () {
            return this.ci;
        },
        set: function (a) {
            var b = this.cj;
            if (a != b) {
                this.ci = a;
                this.sc("MarkerOutlineMode", enumGetBox(MarkerOutlineMode_$type, b), enumGetBox(MarkerOutlineMode_$type, this.cj));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ch", {
        get: function () {
            return this.cg;
        },
        set: function (a) {
            var b = this.ch;
            if (a != b) {
                this.cg = a;
                this.sc("MarkerFillMode", enumGetBox(MarkerFillMode_$type, b), enumGetBox(MarkerFillMode_$type, this.ch));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "h8", {
        get: function () {
            return this.gy;
        },
        set: function (a) {
            var b = this.h8;
            if (a != b) {
                this.gy = a;
                this.sc("MarkerFillOpacity", b, this.h8);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "h9", {
        get: function () {
            return this.gz;
        },
        set: function (a) {
            var b = this.h9;
            if (a != b) {
                this.gz = a;
                this.sc("MarkerThickness", b, this.h9);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "co", {
        get: function () {
            return this.cn;
        },
        set: function (a) {
            var b = this.co;
            if (a != b) {
                this.cn = a;
                this.sc("MarkerTypes", b, this.co);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "cf", {
        get: function () {
            return this.ce;
        },
        set: function (a) {
            var b = this.cf;
            if (a != b) {
                this.ce = a;
                this.sc("MarkerAutomaticBehavior", enumGetBox(MarkerAutomaticBehavior_$type, b), enumGetBox(MarkerAutomaticBehavior_$type, this.cf));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ac", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            var b = this.ac;
            if (a != b) {
                this.u = a;
                this.sc("MarkerBrushes", b, this.ac);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ad", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            var b = this.ad;
            if (!Base.equalsStatic(a, b)) {
                this.v = a;
                this.sc("MarkerOutlines", b, this.ad);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "jl", {
        get: function () {
            return this.jc;
        },
        set: function (a) {
            var b = this.jc;
            if (a != b) {
                this.jc = a;
                this.sc("MarkerMaxCount", b, this.jc);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ht", {
        get: function () {
            return this.gj;
        },
        set: function (a) {
            var b = this.ht;
            if (a != b) {
                this.gj = a;
                this.sc("AreaFillOpacity", b, this.ht);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "e7", {
        get: function () {
            return this.eg;
        },
        set: function (a) {
            var b = this.eg;
            if (a != b) {
                this.eg = a;
                this.sc("AnimateSeriesWhenAxisRangeChanges", b, this.eg);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ag", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            var b = this.ag;
            if (a != b) {
                this.x = a;
                this.sc("TrendLineBrushes", b, this.ag);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "di", {
        get: function () {
            return this.dh;
        },
        set: function (a) {
            var b = this.di;
            if (a != b) {
                this.dh = a;
                this.sc("TrendLineType", enumGetBox(TrendLineType_$type, b), enumGetBox(TrendLineType_$type, this.di));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "it", {
        get: function () {
            return this.hh;
        },
        set: function (a) {
            var b = this.it;
            if (a != b) {
                this.hh = a;
                this.sc("TrendLineThickness", b, this.it);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "e6", {
        get: function () {
            return this.ef;
        },
        set: function (a) {
            var b = this.e6;
            if (a != b) {
                this.ef = a;
                this.sc("AlignsGridLinesToPixels", b, this.e6);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ic", {
        get: function () {
            return this.g2;
        },
        set: function (a) {
            var b = this.g2;
            this.g2 = a;
            if (a != b) {
                this.sc("PlotAreaMarginLeft", b, a);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ie", {
        get: function () {
            return this.g4;
        },
        set: function (a) {
            var b = this.g4;
            this.g4 = a;
            if (a != b) {
                this.sc("PlotAreaMarginTop", b, a);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "id", {
        get: function () {
            return this.g3;
        },
        set: function (a) {
            var b = this.g3;
            this.g3 = a;
            if (a != b) {
                this.sc("PlotAreaMarginRight", b, a);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ib", {
        get: function () {
            return this.g1;
        },
        set: function (a) {
            var b = this.g1;
            this.g1 = a;
            if (a != b) {
                this.sc("PlotAreaMarginBottom", b, a);
            }
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.prototype.sb = function () {
    };
    Object.defineProperty(DomainChart.prototype, "a4", {
        get: function () {
            return this.a3;
        },
        set: function (a) {
            var b = this.a4;
            if (a != b) {
                this.a3 = a;
                this.sc("ComputedPlotAreaMarginMode", enumGetBox(ComputedPlotAreaMarginMode_$type, b), enumGetBox(ComputedPlotAreaMarginMode_$type, this.a4));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "c4", {
        get: function () {
            return this.c3;
        },
        set: function (a) {
            var b = this.c4;
            if (a != b) {
                this.c3 = a;
                this.sc("SeriesPlotAreaMarginHorizontalMode", enumGetBox(SeriesPlotAreaMarginHorizontalMode_$type, b), enumGetBox(SeriesPlotAreaMarginHorizontalMode_$type, this.c4));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "c6", {
        get: function () {
            return this.c5;
        },
        set: function (a) {
            var b = this.c6;
            if (a != b) {
                this.c5 = a;
                this.sc("SeriesPlotAreaMarginVerticalMode", enumGetBox(SeriesPlotAreaMarginVerticalMode_$type, b), enumGetBox(SeriesPlotAreaMarginVerticalMode_$type, this.c6));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "c0", {
        get: function () {
            return this.cz;
        },
        set: function (a) {
            var b = this.c0;
            if (a != b) {
                this.cz = a;
                this.sc("HighlightingMode", enumGetBox(SeriesHighlightingMode_$type, b), enumGetBox(SeriesHighlightingMode_$type, this.c0));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "cy", {
        get: function () {
            return this.cx;
        },
        set: function (a) {
            var b = this.cy;
            if (a != b) {
                this.cx = a;
                this.sc("HighlightingBehavior", enumGetBox(SeriesHighlightingBehavior_$type, b), enumGetBox(SeriesHighlightingBehavior_$type, this.cy));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "b7", {
        get: function () {
            return this.b6;
        },
        set: function (a) {
            var b = this.b7;
            if (a != b) {
                this.b6 = a;
                this.sc("LegendHighlightingMode", enumGetBox(LegendHighlightingMode_$type, b), enumGetBox(LegendHighlightingMode_$type, this.b7));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "cd", {
        get: function () {
            return this.cb;
        },
        set: function (a) {
            var b = this.cd;
            if (a != b) {
                this.cb = a;
                this.sc("LegendItemBadgeShape", enumGetBox(LegendItemBadgeShape_$type, b), enumGetBox(LegendItemBadgeShape_$type, this.cd));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "b9", {
        get: function () {
            return this.b8;
        },
        set: function (a) {
            var b = this.b9;
            if (a != b) {
                this.b8 = a;
                this.sc("LegendItemBadgeMode", enumGetBox(LegendItemBadgeMode_$type, b), enumGetBox(LegendItemBadgeMode_$type, this.b9));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "jn", {
        get: function () {
            return this.jf;
        },
        set: function (a) {
            var b = this.jn;
            if (a != b) {
                this.jf = a;
                this.sc("TrendLinePeriod", b, this.jn);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "dg", {
        get: function () {
            return this.df;
        },
        set: function (a) {
            var b = this.dg;
            if (a != b) {
                this.df = a;
                this.sc("ToolTipType", enumGetBox(ToolTipType_$type, b), enumGetBox(ToolTipType_$type, this.dg));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "a6", {
        get: function () {
            return this.a5;
        },
        set: function (a) {
            var b = this.a6;
            if (a != b) {
                this.a5 = a;
                this.sc("CrosshairsDisplayMode", enumGetBox(CrosshairsDisplayMode_$type, b), enumGetBox(CrosshairsDisplayMode_$type, this.a6));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "fc", {
        get: function () {
            return this.el;
        },
        set: function (a) {
            var b = this.fc;
            if (a != b) {
                this.el = a;
                this.sc("CrosshairsSnapToData", b, this.fc);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "t0", {
        get: function () {
            return this.td;
        },
        set: function (a) {
            var b = this.t0;
            if (a != b) {
                this.td = a;
                this.sc("CrosshairsLineVerticalStroke", b, this.t0);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "hv", {
        get: function () {
            return this.gl;
        },
        set: function (a) {
            var b = this.hv;
            if (a != b) {
                this.gl = a;
                this.sc("CrosshairsLineThickness", b, this.hv);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "tz", {
        get: function () {
            return this.tc;
        },
        set: function (a) {
            var b = this.tz;
            if (a != b) {
                this.tc = a;
                this.sc("CrosshairsLineHorizontalStroke", b, this.tz);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "fb", {
        get: function () {
            return this.ek;
        },
        set: function (a) {
            var b = this.fb;
            if (a != b) {
                this.ek = a;
                this.sc("CrosshairsAnnotationEnabled", b, this.fb);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "tv", {
        get: function () {
            return this.s8;
        },
        set: function (a) {
            var b = this.tv;
            if (a != b) {
                this.s8 = a;
                this.sc("CrosshairsAnnotationXAxisBackground", b, this.tv);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "tx", {
        get: function () {
            return this.ta;
        },
        set: function (a) {
            var b = this.tx;
            if (a != b) {
                this.ta = a;
                this.sc("CrosshairsAnnotationYAxisBackground", b, this.tx);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "tw", {
        get: function () {
            return this.s9;
        },
        set: function (a) {
            var b = this.tw;
            if (a != b) {
                this.s9 = a;
                this.sc("CrosshairsAnnotationXAxisTextColor", b, this.tw);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ty", {
        get: function () {
            return this.tb;
        },
        set: function (a) {
            var b = this.ty;
            if (a != b) {
                this.tb = a;
                this.sc("CrosshairsAnnotationYAxisTextColor", b, this.ty);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "jg", {
        get: function () {
            return this.i7;
        },
        set: function (a) {
            var b = this.jg;
            if (a != b) {
                this.i7 = a;
                this.sc("CrosshairsAnnotationXAxisPrecision", b, this.jg);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "jh", {
        get: function () {
            return this.i8;
        },
        set: function (a) {
            var b = this.jh;
            if (a != b) {
                this.i8 = a;
                this.sc("CrosshairsAnnotationYAxisPrecision", b, this.jh);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "fz", {
        get: function () {
            return this.ex;
        },
        set: function (a) {
            var b = this.fz;
            if (a != b) {
                this.ex = a;
                this.sc("ShouldAvoidAxisAnnotationCollisions", b, this.fz);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "f1", {
        get: function () {
            return this.ez;
        },
        set: function (a) {
            var b = this.ez;
            if (a != b) {
                this.ez = a;
                this.sc("ShouldPanOnMaximumZoom", b, this.ez);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "fi", {
        get: function () {
            return this.er;
        },
        set: function (a) {
            var b = this.fi;
            if (a != b) {
                this.er = a;
                this.sc("FinalValueAnnotationsVisible", b, this.fi);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ua", {
        get: function () {
            return this.tn;
        },
        set: function (a) {
            var b = this.ua;
            if (a != b) {
                this.tn = a;
                this.sc("FinalValueAnnotationsBackground", b, this.ua);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ub", {
        get: function () {
            return this.to;
        },
        set: function (a) {
            var b = this.ub;
            if (a != b) {
                this.to = a;
                this.sc("FinalValueAnnotationsTextColor", b, this.ub);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "jk", {
        get: function () {
            return this.jb;
        },
        set: function (a) {
            var b = this.jk;
            if (a != b) {
                this.jb = a;
                this.sc("FinalValueAnnotationsPrecision", b, this.jk);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "fa", {
        get: function () {
            return this.ej;
        },
        set: function (a) {
            var b = this.fa;
            if (a != b) {
                this.ej = a;
                this.sc("CalloutsVisible", b, this.fa);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "e9", {
        get: function () {
            return this.ei;
        },
        set: function (a) {
            var b = this.e9;
            if (a != b) {
                this.ei = a;
                this.sc("CalloutStyleUpdatingEventEnabled", b, this.e9);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "calloutsItemsSource", {
        get: function () {
            return this.gg;
        },
        set: function (a) {
            var b = this.calloutsItemsSource;
            if (a != b) {
                this.gg = a;
                this.sc("CalloutsItemsSource", b, this.calloutsItemsSource);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ai", {
        get: function () {
            return this.ah;
        },
        set: function (a) {
            var b = this.ai;
            if (a != b) {
                this.ah = a;
                this.sc("CalloutsAllowedPositions", b, this.ai);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "k6", {
        get: function () {
            return this.j6;
        },
        set: function (a) {
            var b = this.k6;
            if (a != b) {
                this.j6 = a;
                this.sc("CalloutsXMemberPath", b, this.k6);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "k8", {
        get: function () {
            return this.j7;
        },
        set: function (a) {
            var b = this.k8;
            if (a != b) {
                this.j7 = a;
                this.sc("CalloutsYMemberPath", b, this.k8);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "k2", {
        get: function () {
            return this.j5;
        },
        set: function (a) {
            var b = this.k2;
            if (a != b) {
                this.j5 = a;
                this.sc("CalloutsLabelMemberPath", b, this.k2);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "kz", {
        get: function () {
            return this.j4;
        },
        set: function (a) {
            var b = this.kz;
            if (a != b) {
                this.j4 = a;
                this.sc("CalloutsContentMemberPath", b, this.kz);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "db", {
        get: function () {
            return this.c9;
        },
        set: function (a) {
            var b = this.c9;
            this.c9 = a;
            if (b != this.c9) {
                this.sc("HorizontalViewScrollbarMode", enumGetBox(SeriesViewerScrollbarMode_$type, b), enumGetBox(SeriesViewerScrollbarMode_$type, this.c9));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "dc", {
        get: function () {
            return this.da;
        },
        set: function (a) {
            var b = this.da;
            this.da = a;
            if (b != this.da) {
                this.sc("VerticalViewScrollbarMode", enumGetBox(SeriesViewerScrollbarMode_$type, b), enumGetBox(SeriesViewerScrollbarMode_$type, this.da));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "c8", {
        get: function () {
            return this.c7;
        },
        set: function (a) {
            var b = this.c7;
            this.c7 = a;
            if (b != this.c7) {
                this.sc("HorizontalViewScrollbarPosition", enumGetBox(SeriesViewerHorizontalScrollbarPosition_$type, b), enumGetBox(SeriesViewerHorizontalScrollbarPosition_$type, this.c7));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "de", {
        get: function () {
            return this.dd;
        },
        set: function (a) {
            var b = this.dd;
            this.dd = a;
            if (b != this.dd) {
                this.sc("VerticalViewScrollbarPosition", enumGetBox(SeriesViewerVerticalScrollbarPosition_$type, b), enumGetBox(SeriesViewerVerticalScrollbarPosition_$type, this.dd));
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ug", {
        get: function () {
            return this.tp;
        },
        set: function (a) {
            var b = this.tp;
            this.tp = a;
            if (b != this.tp) {
                this.sc("HorizontalViewScrollbarFill", b, this.tp);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "uh", {
        get: function () {
            return this.tq;
        },
        set: function (a) {
            var b = this.tq;
            this.tq = a;
            if (b != this.tq) {
                this.sc("HorizontalViewScrollbarOutline", b, this.tq);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "h4", {
        get: function () {
            return this.gu;
        },
        set: function (a) {
            var b = this.gu;
            this.gu = a;
            if (b != this.gu) {
                this.sc("HorizontalViewScrollbarStrokeThickness", b, this.gu);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "h3", {
        get: function () {
            return this.gt;
        },
        set: function (a) {
            var b = this.gt;
            this.gt = a;
            if (b != this.gt) {
                this.sc("HorizontalViewScrollbarMaxOpacity", b, this.gt);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "h0", {
        get: function () {
            return this.gq;
        },
        set: function (a) {
            var b = this.gq;
            this.gq = a;
            if (b != this.gq) {
                this.sc("HorizontalViewScrollbarCornerRadius", b, this.gq);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "h1", {
        get: function () {
            return this.gr;
        },
        set: function (a) {
            var b = this.gr;
            this.gr = a;
            if (b != this.gr) {
                this.sc("HorizontalViewScrollbarHeight", b, this.gr);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "h2", {
        get: function () {
            return this.gs;
        },
        set: function (a) {
            var b = this.gs;
            this.gs = a;
            if (b != this.gs) {
                this.sc("HorizontalViewScrollbarInset", b, this.gs);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "h6", {
        get: function () {
            return this.gw;
        },
        set: function (a) {
            var b = this.gw;
            this.gw = a;
            if (b != this.gw) {
                this.sc("HorizontalViewScrollbarTrackStartInset", b, this.gw);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "h5", {
        get: function () {
            return this.gv;
        },
        set: function (a) {
            var b = this.gv;
            this.gv = a;
            if (b != this.gv) {
                this.sc("HorizontalViewScrollbarTrackEndInset", b, this.gv);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "fk", {
        get: function () {
            return this.es;
        },
        set: function (a) {
            var b = this.es;
            this.es = a;
            if (b != this.es) {
                this.sc(DomainChart.oc, b, this.es);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ul", {
        get: function () {
            return this.tt;
        },
        set: function (a) {
            var b = this.tt;
            this.tt = a;
            if (b != this.tt) {
                this.sc("VerticalViewScrollbarFill", b, this.tt);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "um", {
        get: function () {
            return this.tu;
        },
        set: function (a) {
            var b = this.tu;
            this.tu = a;
            if (b != this.tu) {
                this.sc("VerticalViewScrollbarOutline", b, this.tu);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "ix", {
        get: function () {
            return this.hl;
        },
        set: function (a) {
            var b = this.hl;
            this.hl = a;
            if (b != this.hl) {
                this.sc("VerticalViewScrollbarStrokeThickness", b, this.hl);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "iw", {
        get: function () {
            return this.hk;
        },
        set: function (a) {
            var b = this.hk;
            this.hk = a;
            if (b != this.hk) {
                this.sc("VerticalViewScrollbarMaxOpacity", b, this.hk);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "iu", {
        get: function () {
            return this.hi;
        },
        set: function (a) {
            var b = this.hi;
            this.hi = a;
            if (b != this.hi) {
                this.sc("VerticalViewScrollbarCornerRadius", b, this.hi);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "i0", {
        get: function () {
            return this.ho;
        },
        set: function (a) {
            var b = this.ho;
            this.ho = a;
            if (b != this.ho) {
                this.sc("VerticalViewScrollbarWidth", b, this.ho);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "iv", {
        get: function () {
            return this.hj;
        },
        set: function (a) {
            var b = this.hj;
            this.hj = a;
            if (b != this.hj) {
                this.sc("VerticalViewScrollbarInset", b, this.hj);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "iz", {
        get: function () {
            return this.hn;
        },
        set: function (a) {
            var b = this.hn;
            this.hn = a;
            if (b != this.hn) {
                this.sc("VerticalViewScrollbarTrackStartInset", b, this.hn);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "iy", {
        get: function () {
            return this.hm;
        },
        set: function (a) {
            var b = this.hm;
            this.hm = a;
            if (b != this.hm) {
                this.sc("VerticalViewScrollbarTrackEndInset", b, this.hm);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "f4", {
        get: function () {
            return this.e5;
        },
        set: function (a) {
            var b = this.e5;
            this.e5 = a;
            if (b != this.e5) {
                this.sc(DomainChart.qo, b, this.e5);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "i2", {
        get: function () {
            return this.hq;
        },
        set: function (a) {
            var b = this.hq;
            this.hq = a;
            if (b != this.hq) {
                this.sc("WindowRectMinWidth", b, this.hq);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "i1", {
        get: function () {
            return this.hp;
        },
        set: function (a) {
            var b = this.hp;
            this.hp = a;
            if (b != this.hp) {
                this.sc("WindowRectMinHeight", b, this.hp);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "i4", {
        get: function () {
            return this.hs;
        },
        set: function (a) {
            var b = this.hs;
            this.hs = a;
            if (b != this.hs) {
                this.sc("WindowSizeMinWidth", b, this.hs);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DomainChart.prototype, "i3", {
        get: function () {
            return this.hr;
        },
        set: function (a) {
            var b = this.hr;
            this.hr = a;
            if (b != this.hr) {
                this.sc("WindowSizeMinHeight", b, this.hr);
            }
        },
        enumerable: false,
        configurable: true
    });
    DomainChart.$t = markType(DomainChart, 'DomainChart', Base.$, [INotifyPropertyChanged_$type, IDataLegendTarget_$type]);
    DomainChart.j3 = new Random(0);
    DomainChart.j2 = new Random(0);
    DomainChart.mj = "DataToolTipShouldUpdateWhenSeriesDataChanges";
    DomainChart.oc = "HorizontalViewScrollbarShouldAddAutoTrackInsets";
    DomainChart.qo = "VerticalViewScrollbarShouldAddAutoTrackInsets";
    return DomainChart;
}(Base));
export { DomainChart };
