// igniteui-react-charts/esm5/lib/ScatterBase_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { MarkerSeries } from "./MarkerSeries";
import { ISupportsErrorBars_$type } from "./ISupportsErrorBars";
import { ITrendLineSeries_$type } from "./ITrendLineSeries";
import { ScatterAxisInfoCache } from "./ScatterAxisInfoCache";
import { NumericXAxis } from "./NumericXAxis";
import { DependencyProperty } from "igniteui-react-core";
import { NumericYAxis } from "./NumericYAxis";
import { Series } from "./Series";
import { TrendLineType_$type } from "igniteui-react-core";
import { Brush } from "igniteui-react-core";
import { PenLineCap_$type } from "igniteui-react-core";
import { DoubleCollection } from "igniteui-react-core";
import { CollisionAvoidanceType_$type } from "./CollisionAvoidanceType";
import { ScatterItemSearchMode_$type } from "./ScatterItemSearchMode";
import { ScatterErrorBarSettings } from "./ScatterErrorBarSettings";
import { HighlightingInfo } from "./HighlightingInfo";
import { ScatterFrame } from "./ScatterFrame";
import { Base, enumGetBox, Number_$type, typeCast, fromEnum, Point_$type, runOn, markType, markDep, PointUtil } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
import { SeriesPointOfInterest } from "./SeriesPointOfInterest";
import { ScalerParams } from "./ScalerParams";
import { Rect } from "igniteui-react-core";
import { Marker } from "./Marker";
import { MarkerSeriesView } from "./MarkerSeriesView";
import { IFastItemsSource_$type } from "igniteui-react-core";
import { Clipper } from "igniteui-react-core";
import { OwnedPoint } from "./OwnedPoint";
import { Dictionary$2 } from "igniteui-react-core";
import { AxisRange } from "./AxisRange";
import { GeometryUtil } from "igniteui-react-core";
import { HashPool$2 } from "igniteui-react-core";
import { DataContext } from "igniteui-react-core";
import { TrendResolutionParams } from "./TrendResolutionParams";
import { ScatterTrendLineManager } from "./ScatterTrendLineManager";
import { Thickness } from "igniteui-react-core";
import { Size } from "igniteui-react-core";
import { AssigningScatterStyleEventArgs } from "./AssigningScatterStyleEventArgs";
import { AssigningScatterMarkerStyleEventArgs } from "./AssigningScatterMarkerStyleEventArgs";
import { SeriesRenderManager } from "./SeriesRenderManager";
import { ErrorBarsHelper } from "./ErrorBarsHelper";
import { PathGeometry } from "igniteui-react-core";
import { DataLegendSeriesValueInfo } from "igniteui-react-core";
import { DataLegendSeriesContext } from "igniteui-react-core";
import { PropertyMetadata } from "igniteui-react-core";
import { truncate, isNaN_, isInfinity } from "igniteui-react-core";
import { boxArray$1, arrayShallowClone } from "igniteui-react-core";
import { Path } from "igniteui-react-core";
import { DataTemplateMeasureInfo } from "igniteui-react-core";
import { NumericMarkerManager } from "./NumericMarkerManager";
import { stringIsNullOrEmpty } from "igniteui-react-core";
/**
 * @hidden
 */
var ScatterBase = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ScatterBase, _super);
    function ScatterBase() {
        var _this = _super.call(this) || this;
        _this.r1 = null;
        _this.sf = null;
        _this.sg = null;
        _this.u4 = null;
        _this.u5 = null;
        _this.sa = 1;
        _this.rm = null;
        _this.rn = null;
        _this.r7 = null;
        _this.r9 = null;
        _this.r5 = null;
        _this.sk = false;
        _this.sj = false;
        _this.rx = null;
        _this.ry = null;
        _this.assigningScatterStyle = null;
        _this.assigningScatterMarkerStyle = null;
        _this.se = null;
        _this.r8 = null;
        _this.r4 = null;
        _this.sb = _this.sc();
        _this.se = new ScatterSeriesRenderManager();
        _this.r8 = new ScatterFrame();
        _this.u5 = Rect.empty;
        _this.u4 = Rect.empty;
        return _this;
    }
    ScatterBase.prototype.bx = function () {
        return new ScatterBaseView(this);
    };
    ScatterBase.prototype.kx = function () {
        _super.prototype.kx.call(this);
        this.xAxis = null;
        this.yAxis = null;
    };
    ScatterBase.prototype.l2 = function (a) {
        _super.prototype.l2.call(this, a);
        this.r2 = a;
    };
    ScatterBase.prototype.get_ed = function () {
        return true;
    };
    ScatterBase.prototype.ag = function () {
        return [this.xAxis, this.yAxis];
    };
    ScatterBase.prototype.hy = function (a, b) {
        var c = this.js(b);
        if (c == ScatterBase.$$p[10]) {
            return this.hz(a, b, this.ti);
        }
        if (c == ScatterBase.$$p[14]) {
            return this.hz(a, b, this.tq);
        }
        return null;
    };
    ScatterBase.prototype.c1 = function (a, b, c, d) {
        var _this = this;
        if (!this.hasMarkers) {
            return false;
        }
        var e = this.xAxis;
        var f = new ScalerParams(1, this.cd.s9, this.p3, e.cg);
        f.c = this.getEffectiveViewport1(this.b0);
        var g = this.yAxis;
        var h = new ScalerParams(1, this.cd.s9, this.p3, g.cg);
        h.c = this.getEffectiveViewport1(this.b0);
        a.clear();
        this.qg.cf(function (i) {
            if (i._visibility == 0) {
                var j = new SeriesPointOfInterest();
                var k = i.n;
                var l = i.o;
                if (k < _this.p3.left || k > _this.p3.right || l < _this.p3.top || l > _this.p3.bottom) {
                    return;
                }
                j.b = e.el(k, f);
                j.c = g.el(l, h);
                j.a = i.content;
                a.add(j);
            }
        });
        return true;
    };
    ScatterBase.prototype.bp = function () {
        if (this.bq == 0) {
            if (this.s2 <= 2000) {
                return 1;
            }
            else {
                return _super.prototype.bp.call(this);
            }
        }
        else {
            return _super.prototype.bp.call(this);
        }
    };
    ScatterBase.prototype.p1 = function (a) {
        if (!this.hasMarkers || !this.qj()) {
            return Rect.empty;
        }
        var b = this.qg.by(this.pl(a));
        if (b != null) {
            return this.qg.cp(b);
        }
        return Rect.empty;
    };
    ScatterBase.prototype.ex = function (a, b) {
        if (this.qk(a, b)) {
            return true;
        }
        return false;
    };
    Object.defineProperty(ScatterBase.prototype, "sh", {
        get: function () {
            return this.sf;
        },
        set: function (a) {
            if (this.sf != a) {
                var b = this.sh;
                this.sf = a;
                this.raisePropertyChanged("XColumn", b, this.sh);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ScatterBase.prototype, "si", {
        get: function () {
            return this.sg;
        },
        set: function (a) {
            if (this.sg != a) {
                var b = this.si;
                this.sg = a;
                this.raisePropertyChanged("YColumn", b, this.si);
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ScatterBase.prototype, "u1", {
        get: function () {
            return this.c(ScatterBase.ue);
        },
        set: function (a) {
            this.h(ScatterBase.ue, a);
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ScatterBase.prototype, "GetErrorBarsXAxis", {
        get: function () {
            return this.xAxis;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ScatterBase.prototype, "GetErrorBarsYAxis", {
        get: function () {
            return this.yAxis;
        },
        enumerable: false,
        configurable: true
    });
    ScatterBase.prototype.li = function () {
        _super.prototype.li.call(this);
        if (this.xAxis != null) {
            this.xAxis.h7(false);
        }
        if (this.yAxis != null) {
            this.yAxis.h7(false);
        }
    };
    ScatterBase.prototype.l7 = function (a, b, c, d) {
        _super.prototype.l7.call(this, a, b, c, d);
        if (this.r2.cx.g$b(a, b, c, d)) {
            this.ml(false);
            this.ls();
        }
        switch (b) {
            case "FastItemsSource":
                if (typeCast(IFastItemsSource_$type, c) != null) {
                    c.deregisterColumn(this.sh);
                    c.deregisterColumn(this.si);
                    this.sh = null;
                    this.si = null;
                }
                if (typeCast(IFastItemsSource_$type, d) != null) {
                    this.sh = this.ch(this.ti);
                    this.si = this.ch(this.tq);
                }
                if ((this.yAxis != null && !this.yAxis.c0()) || (this.xAxis != null && !this.xAxis.c0())) {
                    this.ml(false);
                }
                break;
            case ScatterBase.$$p[7]:
                if (c != null) {
                    c.bz(this);
                }
                if (d != null) {
                    d.cs(this);
                }
                if ((this.xAxis != null && !this.xAxis.c0()) || (d == null && c != null)) {
                    this.ml(false);
                }
                this.ls();
                break;
            case ScatterBase.$$p[11]:
                if (c != null) {
                    c.bz(this);
                }
                if (d != null) {
                    d.cs(this);
                }
                if ((this.yAxis != null && !this.yAxis.c0()) || (d == null && c != null)) {
                    this.ml(false);
                }
                this.ls();
                break;
            case ScatterBase.$$p[10]:
                if (this.ck != null) {
                    this.ck.deregisterColumn(this.sh);
                    this.sh = this.ch(this.ti);
                }
                break;
            case "XColumn":
                this.r2.cx.g$i();
                if (this.xAxis != null && !this.xAxis.c0()) {
                    this.ml(false);
                }
                break;
            case ScatterBase.$$p[14]:
                if (this.ck != null) {
                    this.ck.deregisterColumn(this.si);
                    this.si = this.ch(this.tq);
                }
                break;
            case "YColumn":
                this.r2.cx.g$i();
                if (this.yAxis != null && !this.yAxis.c0()) {
                    this.ml(false);
                }
                break;
            case ScatterBase.$$p[5]:
                this.ml(false);
                this.ls();
                break;
            case ScatterBase.$$p[6]:
                this.ml(false);
                this.ls();
                break;
            case "TransitionProgress":
                this.r9.c(this.g2, this.r7, this.r5);
                this.ts();
                try {
                    if (this.cx(this.b0)) {
                        return;
                    }
                    if ((Math.round(this.g2 * 100000) / 100000) == 1) {
                        this.uc(this.r5, this.r2);
                    }
                    else {
                        this.uc(this.r9, this.r2);
                    }
                    if (this.cd != null) {
                        this.cd.oa(this);
                    }
                }
                finally {
                    this.ud();
                }
                break;
            case "TrendLineBrush":
                this.mz();
                break;
            case ScatterBase.$$p[0]:
                if (this.r3 != null) {
                    this.r3.r = this;
                }
                this.ml(false);
                this.ls();
                break;
            case "TrendLineType":
                this.ls();
                break;
            case ScatterBase.$$p[2]:
                this.sk = this.ss();
                this.ml(false);
                break;
            case ScatterBase.$$p[1]:
                this.sj = this.sr();
                this.ml(false);
                break;
            case ScatterBase.$$p[3]:
                this.sb = this.sc();
                break;
        }
    };
    ScatterBase.prototype.ss = function () {
        return this.so;
    };
    ScatterBase.prototype.sr = function () {
        return this.sn;
    };
    ScatterBase.prototype.sm = function (a) {
        if (typeCast(NumericYAxis.$, a) !== null) {
            return true;
        }
        return false;
    };
    ScatterBase.prototype.sl = function (a) {
        if (typeCast(NumericXAxis.$, a) !== null) {
            return true;
        }
        return false;
    };
    ScatterBase.prototype.ud = function () {
        this.u4 = Rect.empty;
        this.u5 = Rect.empty;
    };
    ScatterBase.prototype.ts = function () {
        var a = this.getViewInfo(this.u4, this.u5);
        this.u4 = a.p0;
        this.u5 = a.p1;
    };
    ScatterBase.prototype.sq = function (a, b) {
        if (b != 3) {
            return true;
        }
        if (a == null) {
            return true;
        }
        if (this.ti == a || this.tq == a) {
            return true;
        }
        return false;
    };
    ScatterBase.prototype.k3 = function (a, b, c, d) {
        var e = false;
        if (!this.sq(d, a)) {
            return;
        }
        this.r2.cx.g$e(a, b, c, d);
        if (this.xAxis != null && !this.xAxis.c0()) {
            e = true;
        }
        if (this.yAxis != null && !this.yAxis.c0()) {
            e = true;
        }
        if (e) {
            this.ml(true);
        }
    };
    ScatterBase.prototype.t4 = function (a) {
        this.t5(a, null);
    };
    ScatterBase.prototype.t5 = function (a, b) {
        var e_1, _a, e_2, _b;
        var c = this.sh != null ? this.sh.count : 0;
        var d = this.si != null ? this.si.count : 0;
        var e = Math.min(c, d);
        if (e <= this.s2) {
            a.m.clear();
            var f = new List$1(OwnedPoint.$, 0);
            try {
                for (var _c = __values(fromEnum(a.u.values)), _d = _c.next(); !_d.done; _d = _c.next()) {
                    var g = _d.value;
                    f.add(g);
                }
            }
            catch (e_1_1) {
                e_1 = { error: e_1_1 };
            }
            finally {
                try {
                    if (_d && !_d.done && (_a = _c.return))
                        _a.call(_c);
                }
                finally {
                    if (e_1)
                        throw e_1.error;
                }
            }
            var h_1 = this.ck;
            f.aa(function (i, j) {
                var k = h_1.indexOf(i.a);
                var l = h_1.indexOf(j.a);
                if (k < l) {
                    return -1;
                }
                if (k > l) {
                    return 1;
                }
                return 0;
            });
            if (b != null) {
                b.i = a.m;
            }
            try {
                for (var _e = __values(fromEnum(f)), _f = _e.next(); !_f.done; _f = _e.next()) {
                    var i = _f.value;
                    if (h_1.indexOf(i.a) >= 0) {
                        if (b != null) {
                            b.j(i.c);
                        }
                        else {
                            a.m.add({ $type: Point_$type, x: i.c.x, y: i.c.y });
                        }
                    }
                }
            }
            catch (e_2_1) {
                e_2 = { error: e_2_1 };
            }
            finally {
                try {
                    if (_f && !_f.done && (_b = _e.return))
                        _b.call(_e);
                }
                finally {
                    if (e_2)
                        throw e_2.error;
                }
            }
        }
    };
    ScatterBase.prototype.ap = function (a) {
        if (a != null && a == this.xAxis && this.sh != null) {
            return new AxisRange(this.sh.minimum, this.sh.maximum);
        }
        if (a != null && a == this.yAxis && this.si != null) {
            return new AxisRange(this.si.minimum, this.si.maximum);
        }
        return null;
    };
    ScatterBase.prototype.hv = function (a) {
        var b = this.hf(a);
        if (this.ck != null && b >= 0 && b < this.ck.count) {
            return this.ck.item(b);
        }
        return null;
    };
    Object.defineProperty(ScatterBase.prototype, "sb", {
        get: function () {
            return this.sa;
        },
        set: function (a) {
            var b = this.sa;
            this.sa = a;
            if (b != this.sa) {
                this.raisePropertyChanged("ActualItemSearchMode", enumGetBox(ScatterItemSearchMode_$type, b), enumGetBox(ScatterItemSearchMode_$type, this.sa));
            }
        },
        enumerable: false,
        configurable: true
    });
    ScatterBase.prototype.sc = function () {
        if (this.sd != 0) {
            return this.sd;
        }
        return this.sp ? 4 : 1;
    };
    ScatterBase.prototype.hf = function (a) {
        var b = truncate(Math.round(this.f7(a)));
        return b;
    };
    ScatterBase.prototype.f7 = function (a) {
        switch (this.sb) {
            case 1: return this.sy(a);
            case 3: return this.sz(a);
            case 2: return this.sw(a);
            case 4: return this.sx(a);
        }
        return -1;
    };
    ScatterBase.prototype.sx = function (a) {
        return this.s0(a, false);
    };
    ScatterBase.prototype.sw = function (a) {
        var b = this.pl(a);
        var c = b.x;
        var d = b.y;
        if (this.ck == null || this.xAxis == null || this.yAxis == null || this.sh == null || this.si == null) {
            return -1;
        }
        if (this.ck.count > this.s1) {
            return -1;
        }
        this.tv();
        var e = this.rm;
        var f = this.rn;
        if (e == null || f == null) {
            return -1;
        }
        var g = 1.7976931348623157E+308;
        var h = -1;
        for (var i = 0; i < this.sh.count; i++) {
            var j = e[i];
            var k = f[i];
            var l = (c - j) * (c - j) + (d - k) * (d - k);
            if (l < g) {
                g = l;
                h = i;
            }
        }
        return h;
    };
    ScatterBase.prototype.tv = function () {
        var _this = this;
        var a = this.b0.bv;
        var b = this.b0.bu;
        var c = this.getEffectiveViewport();
        if (this.r1 == null) {
            return;
        }
        var d = new ScalerParams(0, a, b, this.r1.d, c);
        var e = ((function () {
            var $ret = new ScalerParams(0, a, b, _this.r1.e, c);
            $ret.b = _this.ge();
            return $ret;
        })());
        if (this.rm == null || this.rn == null) {
            var f = arrayShallowClone(this.sh.asArray());
            var g = arrayShallowClone(this.si.asArray());
            this.xAxis.hm(boxArray$1(f), 0, f.length, d);
            this.yAxis.hm(boxArray$1(g), 0, g.length, e);
            this.rm = f;
            this.rn = g;
        }
    };
    ScatterBase.prototype.get_sp = function () {
        return false;
    };
    Object.defineProperty(ScatterBase.prototype, "sp", {
        get: function () {
            return this.get_sp();
        },
        enumerable: false,
        configurable: true
    });
    ScatterBase.prototype.s0 = function (a, b) {
        if (!this.sp) {
            return -1;
        }
        var c = { $type: Point_$type, x: 0, y: 0 };
        var d = true;
        var e = this.r6();
        if (e == null) {
            return -1;
        }
        if (e.m == null) {
            return -1;
        }
        if (this.ck == null || this.ck.count > this.s1) {
            return -1;
        }
        this.tv();
        var f = this.rm;
        var g = this.rn;
        if (f == null || g == null) {
            return -1;
        }
        var h = this.pl(a);
        var i = 1.7976931348623157E+308;
        var j = -1;
        for (var k = 0; k < this.rm.length; k++) {
            var l = { $type: Point_$type, x: this.rm[k], y: this.rn[k] };
            if (k > 0) {
                if (!b || e.j.containsKey(this.ck.item(k)) || e.j.containsKey(this.ck.item(k - 1))) {
                    var m = GeometryUtil.d(h, c, l);
                    if (m < i) {
                        i = m;
                        j = k;
                    }
                }
            }
            c = l;
        }
        if (j >= 0) {
            var n = { $type: Point_$type, x: this.rm[j], y: this.rn[j] };
            var o = { $type: Point_$type, x: this.rm[j - 1], y: this.rn[j - 1] };
            if (!b || (e.j.containsKey(this.ck.item(j)) && e.j.containsKey(this.ck.item(j - 1)))) {
                var p = GeometryUtil.h(h, o, n);
                return (j - 1) + p;
            }
            else if (e.j.containsKey(this.ck.item(j))) {
                return j;
            }
            else {
                return j - 1;
            }
        }
        return -1;
    };
    ScatterBase.prototype.sz = function (a) {
        return this.s0(a, true);
    };
    ScatterBase.prototype.sy = function (a) {
        var _this = this;
        var b = null;
        var c = 1.7976931348623157E+308;
        var d = this.pl(a);
        var e = d.x;
        var f = d.y;
        if (this.r2.cy.e > this.s1) {
            return -1;
        }
        this.r2.c5(function (g) {
            if (g._visibility == 0) {
                var h = _this.r2.cp(g);
                var i = h.left + h.width / 2;
                var j = h.top + h.height / 2;
                var k = (e - i) * (e - i) + (f - j) * (f - j);
                if (k < c) {
                    c = k;
                    b = g;
                }
            }
        });
        if (b != null) {
            return b.ak;
        }
        return -1;
    };
    ScatterBase.prototype.pq = function (a, b, c) {
        var _this = this;
        if (this.xAxis == null || this.yAxis == null || this.sh == null || this.si == null) {
            return { $type: Point_$type, x: NaN, y: NaN };
        }
        var d = this.b0.bv;
        var e = this.b0.bu;
        var f = this.getEffectiveViewport();
        var g = new ScalerParams(0, d, e, this.xAxis.cg, f);
        var h = ((function () {
            var $ret = new ScalerParams(0, d, e, _this.yAxis.cg, f);
            $ret.b = _this.ge();
            return $ret;
        })());
        if (b) {
            var i = this.f7(a);
            if (i == -1) {
                return { $type: Point_$type, x: NaN, y: NaN };
            }
            var j = truncate(Math.floor(i));
            var k = truncate(Math.ceil(i));
            if (j < 0) {
                j = 0;
            }
            if (k < 0) {
                k = 0;
            }
            if (k > this.ck.count - 1) {
                k = this.ck.count - 1;
            }
            if (j > this.ck.count - 1) {
                j = this.ck.count - 1;
            }
            if (k == j) {
                var l = this.xAxis.ei(this.sh.item(j), g);
                var m = this.yAxis.ei(this.si.item(j), h);
                return { $type: Point_$type, x: l, y: m };
            }
            var n = i - Math.floor(i);
            var o = this.xAxis.ei(this.sh.item(j), g);
            var p = this.yAxis.ei(this.si.item(j), h);
            var q = this.xAxis.ei(this.sh.item(k), g);
            var r = this.yAxis.ei(this.si.item(k), h);
            var s = q - o;
            var t = r - p;
            return { $type: Point_$type, x: o + s * n, y: p + t * n };
        }
        else {
            var u = this.hf(a);
            if (u == -1) {
                return { $type: Point_$type, x: NaN, y: NaN };
            }
            var v = this.xAxis.ei(this.sh.item(u), g);
            var w = this.yAxis.ei(this.si.item(u), h);
            return { $type: Point_$type, x: v, y: w };
        }
    };
    ScatterBase.prototype.eo = function (a) {
        var _this = this;
        var b = this.cd != null ? this.cd.s9 : Rect.empty;
        b = b.copy();
        var c = this.cd != null ? this.cd.to : Rect.empty;
        var d = new Rect(0, 0, 0, 1, 1);
        var e = this.px(this.b0);
        var f = new ScalerParams(0, d, d, this.xAxis.cg, e);
        var g = ((function () {
            var $ret = new ScalerParams(0, d, d, _this.yAxis.cg, e);
            $ret.b = _this.ge();
            return $ret;
        })());
        var h = !b.isEmpty && !c.isEmpty && this.ck != null ? this.ck.indexOf(a) : -1;
        var i = this.xAxis != null && this.sh != null && h < this.sh.count ? this.xAxis.ei(this.sh.item(h), f) : NaN;
        var j = this.yAxis != null && this.si != null && h < this.si.count ? this.yAxis.ei(this.si.item(h), g) : NaN;
        if (!isNaN_(i)) {
            if (i < b.left + 0.1 * b.width) {
                i = i + 0.4 * b.width;
                b.x = i - 0.5 * b.width;
            }
            if (i > b.right - 0.1 * b.width) {
                i = i - 0.4 * b.width;
                b.x = i - 0.5 * b.width;
            }
        }
        if (!isNaN_(j)) {
            if (j < b.top + 0.1 * b.height) {
                j = j + 0.4 * b.height;
                b.y = j - 0.5 * b.height;
            }
            if (j > b.bottom - 0.1 * b.height) {
                j = j - 0.4 * b.height;
                b.y = j - 0.5 * b.height;
            }
        }
        if (this.b8 != null) {
            this.b8.au(this.cd, b, true);
        }
        return h >= 0;
    };
    ScatterBase.prototype.m8 = function (a, b) {
        this.ml(false);
    };
    ScatterBase.prototype.m9 = function (a, b) {
        this.ml(false);
    };
    ScatterBase.prototype.r6 = function () {
        var a = this.r5;
        if (this.cv()) {
            if (this.ce.g) {
                this.ce.t();
            }
            a = this.r9;
        }
        return a;
    };
    ScatterBase.prototype.tt = function (a, b, c, d, e) {
        var _this = this;
        b.g = new Dictionary$2(Base.$, OwnedPoint.$, 1, c);
        var f = this.getEffectiveViewport1(a);
        var g = this.ck;
        var h;
        var i;
        var j = ((function () {
            var $ret = new ScalerParams(0, d, e, _this.r1.d, f);
            $ret.c = _this.getEffectiveViewport1(a);
            return $ret;
        })());
        var k = ((function () {
            var $ret = new ScalerParams(0, d, e, _this.r1.e, f);
            $ret.c = _this.getEffectiveViewport1(a);
            $ret.b = _this.ge();
            return $ret;
        })());
        var l = false;
        var m = 0;
        if (this.cd != null) {
            if (this.cd.e8()) {
                var n = this.cd.lh(d, j);
                l = n.c;
                m = n.d;
            }
        }
        var _loop_1 = function (o) {
            h = this_1.sh.item(o);
            i = this_1.si.item(o);
            var p = { $type: Point_$type, x: this_1.r1.a.ei(h, j), y: this_1.r1.b.ei(i, k) };
            if (l) {
                if (p.x < e.left || p.x > e.right) {
                    if (p.x + m >= e.left && p.x + m <= e.right) {
                        p = { $type: Point_$type, x: p.x + m, y: i };
                    }
                }
            }
            if (!isInfinity(p.x) && !isInfinity(p.y)) {
                var q_1 = g.item(o);
                if (!b.g.containsKey(q_1)) {
                    var r_1 = { $type: Point_$type, x: h, y: i };
                    var s_1 = { $type: Point_$type, x: p.x, y: p.y };
                    b.g.addItem(q_1, ((function () {
                        var $ret = new OwnedPoint();
                        $ret.a = q_1;
                        $ret.b = r_1;
                        $ret.c = s_1;
                        return $ret;
                    })()));
                }
            }
        };
        var this_1 = this;
        for (var o = 0; o < c; o++) {
            _loop_1(o);
        }
    };
    ScatterBase.prototype.t1 = function (a, b) {
        var _this = this;
        this.rm = null;
        this.rn = null;
        a.j.clear();
        a.n.clear();
        a.h.clear();
        a.l.clear();
        a.i.clear();
        a.k.clear();
        var c = Math.min(this.sh != null ? this.sh.count : 0, this.si != null ? this.si.count : 0);
        var d = b.bv;
        var e = b.bu;
        var f = this.getEffectiveViewport1(b);
        var g = new ScalerParams(0, d, e, this.xAxis.cg, f);
        g.c = this.getEffectiveViewport1(b);
        var h = ((function () {
            var $ret = new ScalerParams(0, d, e, _this.yAxis.cg, f);
            $ret.b = _this.ge();
            return $ret;
        })());
        h.c = this.getEffectiveViewport1(b);
        if (c < 1) {
            return;
        }
        this.r1 = ((function () {
            var $ret = new ScatterAxisInfoCache();
            $ret.a = _this.xAxis;
            $ret.b = _this.yAxis;
            $ret.d = _this.xAxis.cg;
            $ret.e = _this.yAxis.cg;
            $ret.c = _this.ck;
            return $ret;
        })());
        var i = b;
        if (c <= this.s2) {
            this.tt(i, a, c, d, e);
        }
        if (this.qj()) {
            b.cv.l(a.j, this.s2, d, e, this.fv);
        }
        var j = ((function () {
            var $ret = new Clipper(0, e, false);
            $ret.i = a.n;
            return $ret;
        })());
        var k = this.pv(b);
        var l = this.xAxis.el(k.left, g);
        var m = this.xAxis.el(k.right, g);
        b.cx.h(a.n, this.trendLineType, this.sh, this.si, this.trendLinePeriod, function (n) { return _this.xAxis.ei(n, g); }, function (n) { return _this.yAxis.ei(n, h); }, ((function () {
            var $ret = new TrendResolutionParams();
            $ret.b = _this.fv;
            $ret.f = e;
            $ret.g = d;
            return $ret;
        })()), j, l, m);
        this.tz(a, b);
    };
    ScatterBase.prototype.k0 = function (a, b) {
        _super.prototype.k0.call(this, a, b);
        var c = b;
        c.c4(a);
    };
    ScatterBase.prototype.get_dy = function () {
        return true;
    };
    ScatterBase.prototype.rv = function (a, b, c) {
        return null;
    };
    ScatterBase.prototype.rw = function (a, b, c) {
        var d = this.r6();
        var e = this.r2.by(this.pl(b));
        var f = -1;
        if (e == null) {
            if (a != null) {
                f = this.ck.indexOf(a);
            }
            else {
                return null;
            }
        }
        else {
            f = e.ak;
        }
        if (f == -1) {
            return null;
        }
        var g = new HighlightingInfo();
        g.b = this;
        g.h = f;
        g.g = f;
        g.d = false;
        g.e = true;
        if (c != null && c.h == g.h && c.g == g.g) {
            return c;
        }
        return g;
    };
    ScatterBase.prototype.p5 = function (a, b) {
        if (this.qh != 1 && this.qh != 13) {
            var c = this.qg.cq();
            var d = (c.width / 2) + (this.qm / 2);
            var e = (c.height / 2) + (this.qm / 2);
            var f = d;
            var g = d;
            var h = e;
            var i = e;
            if (a == 2 || a == 4) {
                f = 0;
            }
            if (a == 2 || a == 3) {
                g = 0;
            }
            if (b == 2 || b == 3) {
                h = 0;
            }
            if (b == 2 || b == 4) {
                i = 0;
            }
            return new Thickness(1, f, h, g, i);
        }
        return new Thickness(1, 0, 0, 0, 0);
    };
    ScatterBase.prototype.ru = function (a, b, c) {
        var d = new HighlightingInfo();
        d.b = this;
        d.h = 0;
        d.g = this.ck.count - 1;
        d.d = true;
        if (c != null && c.h == d.h && c.g == d.g) {
            return c;
        }
        return d;
    };
    ScatterBase.prototype.aw = function (a, b, c) {
        if (this.c2 && this.eg(this.bk) && !c) {
            var d = this.rv(a, b, this.rx);
            this.rx = d;
            return d;
        }
        else {
            var e = this.ru(a, b, this.rx);
            this.rx = e;
            return e;
        }
    };
    ScatterBase.prototype.ax = function (a, b, c) {
        if (this.eg(this.bk) && !c) {
            var d = this.rw(a, b, this.ry);
            this.ry = d;
            return d;
        }
        else {
            var e = this.ru(a, b, this.ry);
            e.e = true;
            this.ry = e;
            return e;
        }
    };
    ScatterBase.prototype.su = function () {
        return (this.assigningScatterStyle != null && this.sk) || (this.bk != 1);
    };
    ScatterBase.prototype.st = function () {
        return (this.assigningScatterMarkerStyle != null && this.sj) || (this.bk != 1);
    };
    ScatterBase.prototype.uc = function (a, b) {
        var _this = this;
        var c = b.bu;
        this.r1 = ((function () {
            var $ret = new ScatterAxisInfoCache();
            $ret.a = _this.xAxis;
            $ret.b = _this.yAxis;
            $ret.d = _this.xAxis.cg;
            $ret.e = _this.yAxis.cg;
            return $ret;
        })());
        if (this.qj()) {
            b.cv.j(a.j, this.ql);
            if (this.st()) {
                this.tu(b, a);
            }
        }
        b.cm();
        var d = ((function () {
            var $ret = new Clipper(1, NaN, c.bottom, NaN, c.top, false);
            $ret.i = b.cx.g$j.ak;
            return $ret;
        })());
        b.cx.g$h(a.n, d);
        this.t9(a, b);
    };
    ScatterBase.prototype.rp = function (a, b) {
        return this.rq(a, b);
    };
    ScatterBase.prototype.rq = function (a, b) {
        if (b < a || a < 0 || a > this.sh.count || b < 0 || b > this.sh.count) {
            return null;
        }
        var c = new Array((b - a) + 1);
        for (var d = a; d <= b; d++) {
            var e = d;
            c[d - a] = this.ck.item(d);
        }
        return c;
    };
    ScatterBase.prototype.tu = function (a, b) {
        var _this = this;
        a.b3 = true;
        this.se.aw(this, this.st(), runOn(this, this.rp));
        var c = this.se.d;
        var d = c != null;
        var e = this.getEffectiveViewport1(a);
        var f = this.ck == null ? 0 : this.ck.count;
        this.r2.cf(function (g) {
            if (g._visibility == 0) {
                var h = g.ak;
                var i = g.content;
                if (i == null) {
                    i = new DataContext();
                    g.content = i;
                }
                if (d) {
                    _this.tw(h, f, a.q);
                }
                _this.se.ac(g, i);
            }
        });
    };
    ScatterBase.prototype.tx = function (a, b, c) {
        var d = this.bk != 1;
        this.se.ay(this.se.b, a, b, false);
        this.se.ab(this.ft);
        var e = this.se.av;
        var f = null;
        var g = this.ck;
        if (d && this.cd != null) {
            f = this.cd.a7.getHighlightingInfo(this, g, e.o, e.n, false);
            e.b = f;
        }
        e.f = c;
        this.t7(e);
        this.se.z(this, f, c, d, a < 0);
    };
    ScatterBase.prototype.tw = function (a, b, c) {
        var d = this.bk != 1;
        this.se.ay(this.se.d, a, b, true);
        this.se.aa(this.ft);
        var e = this.se.au;
        var f = null;
        var g = this.ck;
        var h = null;
        if (d && this.cd != null) {
            f = this.cd.a7.getHighlightingInfo(this, g, e.o, e.n, true);
            if (this.cd.a7.isCrossSeriesHighlight(this)) {
                h = this.cd.a7.getHighlightingInfo(this, g, e.o, e.n, true);
            }
            e.b = f;
        }
        e.f = c;
        this.t6(e);
        this.se.y(this, f, c, d, e.m, this.cd.a7.totalHighlightingProgress);
    };
    ScatterBase.prototype.t7 = function (a) {
        if (this.assigningScatterStyle != null && this.sk) {
            this.assigningScatterStyle(this, a);
        }
    };
    ScatterBase.prototype.t6 = function (a) {
        if (this.assigningScatterMarkerStyle != null && this.sj) {
            this.assigningScatterMarkerStyle(this, a);
        }
    };
    ScatterBase.prototype.tz = function (a, b) {
        var e_3, _a;
        var c = new ErrorBarsHelper(this, b);
        if (this.r3 == null) {
            return;
        }
        var d = this.r3.u;
        var e = this.r3.v;
        var f = 0;
        var g = 0;
        var h = 0;
        var i = 0;
        try {
            for (var _b = __values(fromEnum(a.j.keys)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var j = _c.value;
                var k = a.j.item(j);
                if (d != null) {
                    switch (d.getCalculatorType()) {
                        case 1:
                            var l = this.t2(k, c, d, h, i, true);
                            h = l.p3;
                            i = l.p4;
                            break;
                        case 0:
                            var m = this.t0(k, c, d, h, i, true);
                            h = m.p3;
                            i = m.p4;
                            break;
                        case 2:
                            var n = this.ty(k, c, d, j, h, i, true);
                            h = n.p4;
                            i = n.p5;
                            break;
                        case 3:
                            var o = this.t3(c, d, f, h, i, true);
                            f = o.p2;
                            h = o.p3;
                            i = o.p4;
                            break;
                        case 4:
                            var p = this.t0(k, c, d, h, i, true);
                            h = p.p3;
                            i = p.p4;
                            break;
                    }
                    ;
                    var q = new OwnedPoint();
                    q.c = c.l(d, this.r1.a, k.c, f);
                    q.a = k.a;
                    a.h.addItem(j, q);
                    a.i.addItem(j, [h, i]);
                }
                if (e != null) {
                    switch (e.getCalculatorType()) {
                        case 1:
                            var r = this.t2(k, c, e, h, i, false);
                            h = r.p3;
                            i = r.p4;
                            break;
                        case 0:
                            var s = this.t0(k, c, e, h, i, false);
                            h = s.p3;
                            i = s.p4;
                            break;
                        case 2:
                            var t = this.ty(k, c, e, j, h, i, false);
                            h = t.p4;
                            i = t.p5;
                            break;
                        case 3:
                            var u = this.t3(c, e, g, h, i, false);
                            g = u.p2;
                            h = u.p3;
                            i = u.p4;
                            break;
                        case 4:
                            var v = this.t0(k, c, e, h, i, false);
                            h = v.p3;
                            i = v.p4;
                            break;
                    }
                    ;
                    var w = new OwnedPoint();
                    w.c = c.m(e, this.r1.b, k.c, g);
                    w.a = k.a;
                    a.l.addItem(j, w);
                    a.k.addItem(j, [h, i]);
                }
            }
        }
        catch (e_3_1) {
            e_3 = { error: e_3_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_3)
                    throw e_3.error;
            }
        }
    };
    ScatterBase.prototype.t2 = function (a, b, c, d, e, f) {
        if (f) {
            var g = void 0;
            var h = void 0, i = void 0;
            i = this.r1.a;
            if (this.r3.s == 0) {
                g = a.c.x;
                h = this.r1.a;
            }
            else {
                g = a.c.y;
                h = this.r1.b;
            }
            var j = b.f(g, c, h, i, d, e);
            d = j.p4;
            e = j.p5;
        }
        else {
            var k = void 0;
            var l = void 0, m = void 0;
            m = this.r1.b;
            if (this.r3.t == 0) {
                k = a.c.x;
                l = this.r1.a;
            }
            else {
                k = a.c.y;
                l = this.r1.b;
            }
            var n = b.f(k, c, l, m, d, e);
            d = n.p4;
            e = n.p5;
        }
        return {
            p3: d,
            p4: e
        };
    };
    ScatterBase.prototype.t0 = function (a, b, c, d, e, f) {
        if (f) {
            var g = b.j(a.c.x, c, this.r1.a, d, e);
            d = g.p3;
            e = g.p4;
        }
        else {
            var h = b.j(a.c.y, c, this.r1.b, d, e);
            d = h.p3;
            e = h.p4;
        }
        return {
            p3: d,
            p4: e
        };
    };
    ScatterBase.prototype.ty = function (a, b, c, d, e, f, g) {
        var h = c.getItemColumn();
        var i = this.ck.indexOf(d);
        if (h != null && i < h.count) {
            var j = h.item(i);
            if (g) {
                var k = b.h(a.c.x, j, this.r1.a, e, f);
                e = k.p3;
                f = k.p4;
            }
            else {
                var l = b.h(a.c.y, j, this.r1.b, e, f);
                e = l.p3;
                f = l.p4;
            }
        }
        else {
            e = NaN;
            f = NaN;
        }
        return {
            p4: e,
            p5: f
        };
    };
    ScatterBase.prototype.t3 = function (a, b, c, d, e, f) {
        var g = a.i(b, c);
        c = g.p1;
        if (f) {
            var h = a.k(c, b, this.r1.a, d, e);
            d = h.p3;
            e = h.p4;
        }
        else {
            var i = a.k(c, b, this.r1.b, d, e);
            d = i.p3;
            e = i.p4;
        }
        return {
            p2: c,
            p3: d,
            p4: e
        };
    };
    ScatterBase.prototype.t9 = function (a, b) {
        if (!b.o() || this.r3 == null) {
            b.c6();
            return;
        }
        this.ua(a, b);
        this.ub(a, b);
    };
    ScatterBase.prototype.ua = function (a, b) {
        var e_4, _a;
        b.c2();
        var c = new ErrorBarsHelper(this, b);
        var d = new PathGeometry();
        var e = this.r3.u;
        try {
            for (var _b = __values(fromEnum(a.j.keys)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var f = _c.value;
                if (e != null && a.i.containsKey(f)) {
                    var g = a.i.item(f)[0];
                    var h = a.i.item(f)[1];
                    if (!isNaN_(g) && !isNaN_(h)) {
                        var i = a.h.item(f).c;
                        if (this.r3.o == 1 || this.r3.o == 2) {
                            c.d(d, i, g, true);
                        }
                        if (this.r3.o == 1 || this.r3.o == 3) {
                            c.d(d, i, h, false);
                        }
                    }
                }
            }
        }
        catch (e_4_1) {
            e_4 = { error: e_4_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_4)
                    throw e_4.error;
            }
        }
        b.c7(d);
    };
    ScatterBase.prototype.ub = function (a, b) {
        var e_5, _a;
        b.c3();
        var c = new ErrorBarsHelper(this, b);
        var d = new PathGeometry();
        var e = this.r3.v;
        try {
            for (var _b = __values(fromEnum(a.j.keys)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var f = _c.value;
                if (e != null && a.k.containsKey(f)) {
                    var g = a.k.item(f)[0];
                    var h = a.k.item(f)[1];
                    if (!isNaN_(g) && !isNaN_(h)) {
                        var i = a.l.item(f).c;
                        if (this.r3.p == 1 || this.r3.p == 2) {
                            c.e(d, i, g, true);
                        }
                        if (this.r3.p == 1 || this.r3.p == 3) {
                            c.e(d, i, h, false);
                        }
                    }
                }
            }
        }
        catch (e_5_1) {
            e_5 = { error: e_5_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_5)
                    throw e_5.error;
            }
        }
    };
    ScatterBase.prototype.fb = function (a, b, c) {
        var d = true;
        if (!_super.prototype.fb.call(this, a, b, c) || b.isEmpty || a.isEmpty || this.xAxis == null || this.yAxis == null || this.xAxis.bh == null || this.yAxis.bh == null || this.sh == null || this.si == null || this.sh.count == 0 || this.si.count == 0 || this.ck == null || this.ck.count != this.sh.count || this.ck.count != this.si.count || this.xAxis.bh == null || this.yAxis.bh == null || this.xAxis.lv == this.xAxis.lu || this.yAxis.lv == this.yAxis.lu) {
            d = false;
        }
        return d;
    };
    ScatterBase.prototype.getViewInfo = function (a, b) {
        if (!this.u4.isEmpty && !this.u5.isEmpty) {
            a = this.u4;
            b = this.u5;
            return {
                p0: a,
                p1: b
            };
        }
        a = this.b0.bu;
        b = this.b0.bv;
        return {
            p0: a,
            p1: b
        };
    };
    ScatterBase.prototype.mm = function (a) {
        this.ts();
        try {
            if (this.cx(this.b0)) {
                return;
            }
            if (this.ck != null && this.ck.count > this.s2) {
                a = false;
            }
            if (this.ep(a) && !this.eu) {
                var b = this.r7;
                if (this.cv()) {
                    if (this.ce.g) {
                        this.ce.t();
                    }
                    this.r7 = this.r9;
                    this.r9 = b;
                }
                else {
                    this.r7 = this.r5;
                    this.r5 = b;
                }
                this.t1(this.r5, this.r2);
                this.mu();
            }
            else {
                if (!this.eu) {
                    this.t1(this.r5, this.r2);
                }
                this.uc(this.r5, this.r2);
            }
        }
        finally {
            this.ud();
        }
    };
    ScatterBase.prototype.mz = function () {
        _super.prototype.mz.call(this);
        if (this.index < 0) {
            return;
        }
        this.r2.da();
    };
    ScatterBase.prototype.mn = function (a, b) {
        _super.prototype.mn.call(this, a, b);
        if (!this.e1) {
            this.b0.a2(b);
            return;
        }
        this.b0.a2(b);
        if (this.cx(this.bz)) {
            return;
        }
        var c = typeCast(ScatterBaseView.$, this.bz);
        if (!this.ev) {
            this.r8 = new ScatterFrame();
            this.t1(this.r8, c);
        }
        this.ev = false;
        this.uc(this.r8, c);
        this.e1 = false;
    };
    ScatterBase.prototype.mk = function (a, b, c, d, e) {
        _super.prototype.mk.call(this, a, b, c, d, e);
        var f = this.a6.a$h.item(d);
        var g = f;
        f.a1(c);
        if (this.cx(g)) {
            return;
        }
        if (this.r4 == null) {
            this.r4 = new ScatterFrame();
        }
        this.t1(this.r4, g);
        this.uc(this.r4, g);
    };
    ScatterBase.prototype.t8 = function (a, b) {
        var e_6, _a, e_7, _b;
        var c = new List$1(Base.$, 0);
        try {
            for (var _c = __values(fromEnum(b.c)), _d = _c.next(); !_d.done; _d = _c.next()) {
                var d = _d.value;
                if (!a.containsKey(d)) {
                    c.add1(d);
                }
            }
        }
        catch (e_6_1) {
            e_6 = { error: e_6_1 };
        }
        finally {
            try {
                if (_d && !_d.done && (_a = _c.return))
                    _a.call(_c);
            }
            finally {
                if (e_6)
                    throw e_6.error;
            }
        }
        try {
            for (var _e = __values(fromEnum(c)), _f = _e.next(); !_f.done; _f = _e.next()) {
                var e = _f.value;
                b.h(e);
            }
        }
        catch (e_7_1) {
            e_7 = { error: e_7_1 };
        }
        finally {
            try {
                if (_f && !_f.done && (_b = _e.return))
                    _b.call(_e);
            }
            finally {
                if (e_7)
                    throw e_7.error;
            }
        }
    };
    ScatterBase.prototype.ro = function (a, b, c, d, e) {
        var _this = this;
        if (c == null || c.length != this.r1.c.count) {
            c = new Array(this.r1.c.count);
            for (var f = 0; f < this.r1.c.count; f++) {
                c[f] = PointUtil.create();
            }
        }
        var g = this.getEffectiveViewport1(a);
        var h = ((function () {
            var $ret = new ScalerParams(0, d, e, _this.xAxis.cg, g);
            $ret.c = _this.getEffectiveViewport1(a);
            return $ret;
        })());
        var i = ((function () {
            var $ret = new ScalerParams(0, d, e, _this.yAxis.cg, g);
            $ret.c = _this.getEffectiveViewport1(a);
            $ret.b = _this.ge();
            return $ret;
        })());
        var j = this.pv(a);
        var k = this.r1.a.el(j.left, h);
        var l = this.r1.a.el(j.right, h);
        var m = this.r1.b.el(j.bottom, i);
        var n = this.r1.b.el(j.top, i);
        if (this.r1.d) {
            var o = k;
            k = l;
            l = o;
        }
        if (this.r1.e) {
            var p = m;
            m = n;
            n = p;
        }
        var q = this.r1;
        var r = q.a;
        var s = q.b;
        var t;
        var u;
        var v = this.sh;
        var w = this.si;
        var x = false;
        var y = 0;
        var z = 0;
        if (this.cd != null) {
            if (this.cd.e8()) {
                var aa = this.cd.lh(d, h);
                x = aa.c;
                y = aa.d;
                z = r.el(0 + y, h);
                var ab = r.el(0, h);
                z -= ab;
            }
        }
        for (var ac = 0; ac < this.r1.c.count; ac++) {
            t = v.item(ac);
            u = w.item(ac);
            if (x) {
                if (t >= k && t <= l && u >= m && u <= n) {
                    c[ac].x = r.ei(t, h);
                    c[ac].y = s.ei(u, i);
                }
                else {
                    if ((t + z) >= k && (t + z) <= l && u >= m && u <= n) {
                        c[ac].x = r.ei(t, h) + y;
                        c[ac].y = s.ei(u, i);
                    }
                    else {
                        c[ac].x = NaN;
                        c[ac].y = NaN;
                    }
                }
            }
            else {
                if (t >= k && t <= l && u >= m && u <= n) {
                    c[ac].x = r.ei(t, h);
                    c[ac].y = s.ei(u, i);
                }
                else {
                    c[ac].x = NaN;
                    c[ac].y = NaN;
                }
            }
        }
        return c;
    };
    ScatterBase.prototype.sv = function (a, b) {
        var e_8, _a;
        if (b == null) {
            b = new List$1(Number_$type, 2, a.e);
        }
        else {
            b.clear();
        }
        var c = 0;
        var d = this.ck;
        try {
            for (var _b = __values(fromEnum(a.c)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                b.add(d.indexOf(e));
            }
        }
        catch (e_8_1) {
            e_8 = { error: e_8_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_8)
                    throw e_8.error;
            }
        }
        return b;
    };
    ScatterBase.prototype.la = function (a) {
        _super.prototype.la.call(this, a);
        this.a8.exportTrendlineData(this, a, this.r2.cx.g$j);
    };
    ScatterBase.prototype.ge = function () {
        return Series.gf(this.si, null);
    };
    ScatterBase.prototype.getDataLegendSeriesContextAt = function (a, b) {
        var c = { $type: Point_$type, x: a, y: b };
        var d = this.hv(c);
        var e = d == null ? NaN : this.gd(d, "XMemberPath");
        var f = d == null ? NaN : this.gd(d, "YMemberPath");
        var g = new DataLegendSeriesValueInfo();
        g.l = e;
        g.a = 13;
        g.p = this.ti != null ? this.ti : "X:";
        g.q = this.te != null ? this.te : "X:";
        g.r = this.tg != null ? this.tg : "";
        var h = new DataLegendSeriesValueInfo();
        h.l = f;
        h.a = 14;
        h.p = this.tq != null ? this.tq : "Y:";
        h.q = this.tm != null ? this.tm : "Y:";
        h.r = this.to != null ? this.to : "";
        var i = new DataLegendSeriesContext();
        i.f = ((function () {
            var $ret = new List$1(DataLegendSeriesValueInfo.$, 0);
            $ret.add(g);
            $ret.add(h);
            return $ret;
        })());
        i.b = this.av();
        return i;
    };
    ScatterBase.$t = markType(ScatterBase, 'ScatterBase', MarkerSeries.$, [ISupportsErrorBars_$type, ITrendLineSeries_$type]);
    ScatterBase.ue = DependencyProperty.i("ActualTrendLineBrush", Brush.$, ScatterBase.$, new PropertyMetadata(2, null, function (a, b) { return a.raisePropertyChanged("ActualTrendLineBrush", b.oldValue, b.newValue); }));
    ScatterBase.$$p = markDep(DependencyProperty, PropertyMetadata, ScatterBase, 'raisePropertyChanged', ['ErrorBarSettings:r3:uf', [ScatterErrorBarSettings.$, null], 'IsCustomScatterMarkerStyleAllowed:sn:ug', [0, false], 'IsCustomScatterStyleAllowed:so:uh', [0, false], 'ItemSearchMode:sd:ui', [ScatterItemSearchMode_$type, enumGetBox(ScatterItemSearchMode_$type, 0)], 'ItemSearchThreshold:s1:uj', [1, 10000], 'MarkerCollisionAvoidance:rt:uk', [CollisionAvoidanceType_$type, enumGetBox(CollisionAvoidanceType_$type, 0)], 'MaximumMarkers:s2:ul', [1, 400], 'XAxis::ut', [NumericXAxis.$, null], 'XMemberAsLegendLabel:te:uu', [2, null], 'XMemberAsLegendUnit:tg:uv', [2, null], 'XMemberPath:ti:uw', [2, null], 'YAxis::ux', [NumericYAxis.$, null], 'YMemberAsLegendLabel:tm:uy', [2, null], 'YMemberAsLegendUnit:to:uz', [2, null], 'YMemberPath:tq:u0', [2, null], 'TrendLineType::ur', [TrendLineType_$type, enumGetBox(TrendLineType_$type, 0)], 'TrendLineThickness::uq', [1, 1.5], 'TrendLinePeriod::up', [1, 7], 'TrendLineBrush::um', [Brush.$, null], 'TrendLineZIndex:s3:us', [1, 1001], 'TrendLineDashArray:u2:un', [DoubleCollection.$], 'TrendLineDashCap:u3:uo', [PenLineCap_$type, enumGetBox(PenLineCap_$type, 0)]]);
    return ScatterBase;
}(MarkerSeries));
export { ScatterBase };
/**
 * @hidden
 */
var ScatterSeriesRenderManager = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ScatterSeriesRenderManager, _super);
    function ScatterSeriesRenderManager() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.au = null;
        _this.av = null;
        return _this;
    }
    ScatterSeriesRenderManager.prototype.c = function () {
        this.au = new AssigningScatterMarkerStyleEventArgs();
        return this.au;
    };
    ScatterSeriesRenderManager.prototype.a = function () {
        this.av = new AssigningScatterStyleEventArgs();
        return this.av;
    };
    ScatterSeriesRenderManager.prototype.aw = function (a, b, c) {
        this.w(a, b, c);
    };
    ScatterSeriesRenderManager.prototype.ax = function (a, b, c) {
        this.x(a, b, c);
    };
    ScatterSeriesRenderManager.prototype.ay = function (a, b, c, d) {
        if (b == -1) {
            a.c = false;
            a.o = 0;
            a.n = c - 1;
        }
        else {
            if (d) {
                a.o = b;
                a.n = b;
            }
            else {
                a.c = false;
                a.o = b;
                a.n = b;
            }
        }
    };
    ScatterSeriesRenderManager.$t = markType(ScatterSeriesRenderManager, 'ScatterSeriesRenderManager', SeriesRenderManager.$);
    return ScatterSeriesRenderManager;
}(SeriesRenderManager));
export { ScatterSeriesRenderManager };
/**
 * @hidden
 */
var ScatterBaseView = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ScatterBaseView, _super);
    function ScatterBaseView(a) {
        var _this = _super.call(this, a) || this;
        _this.cv = null;
        _this.ct = null;
        _this.cw = null;
        _this.c1 = null;
        _this.dc = null;
        _this.dd = null;
        _this.cx = null;
        _this.db = null;
        _this.cy = null;
        _this.cw = a;
        _this.cy = new HashPool$2(Base.$, Marker.$);
        _this.ch(_this.cy);
        _this.cx = new ScatterTrendLineManager();
        return _this;
    }
    ScatterBaseView.prototype.as = function () {
        _super.prototype.as.call(this);
        this.cv = this.cu();
        this.dc = new Path();
        this.dd = new Path();
        if (!this.q) {
            this.cw.s2 = 2000;
        }
    };
    ScatterBaseView.prototype.de = function (a) {
        if (this.db == null) {
            this.db = new DataTemplateMeasureInfo();
            this.db.context = this.k.g();
        }
        this.db.width = a.width;
        this.db.height = a.height;
        this.db.data = a.content;
        var b = a.ah;
        if (b.measure != null) {
            b.measure(this.db);
        }
        return new Size(1, this.db.width, this.db.height);
    };
    ScatterBaseView.prototype.cu = function () {
        var _this = this;
        var a = new NumericMarkerManager(1, function (b) { return _this.cy.item(b); }, function (b) { return _this.cw.r1.c.item(b); }, runOn(this, this.c9), runOn(this, this.cr), runOn(this, this.cz), function () { return _this.cw.rt; });
        a.q = runOn(this, this.de);
        return a;
    };
    ScatterBaseView.prototype.c9 = function (a) {
        this.cw.t8(a, this.cy);
    };
    ScatterBaseView.prototype.c0 = function () {
        return this.b4;
    };
    ScatterBaseView.prototype.cs = function () {
        return this.ct;
    };
    ScatterBaseView.prototype.cr = function () {
        this.ct = this.cw.ro(this, this.cy, this.ct, this.bv, this.bu);
        return this.ct;
    };
    ScatterBaseView.prototype.cz = function () {
        this.c1 = this.cw.sv(this.cy, this.c1);
        return this.c1;
    };
    ScatterBaseView.prototype.c2 = function () {
    };
    ScatterBaseView.prototype.c7 = function (a) {
    };
    ScatterBaseView.prototype.c3 = function () {
    };
    ScatterBaseView.prototype.c8 = function (a) {
    };
    ScatterBaseView.prototype.da = function () {
        if (this.cw.trendLineBrush != null) {
            this.cw.u1 = this.cw.trendLineBrush;
        }
        else {
            this.cw.u1 = this.cw.o3;
        }
    };
    ScatterBaseView.prototype.cf = function (a) {
        this.cy.f(a);
    };
    ScatterBaseView.prototype.c5 = function (a) {
        this.cy.g(a);
    };
    ScatterBaseView.prototype.c6 = function () {
    };
    ScatterBaseView.prototype.a5 = function (a, b) {
        _super.prototype.a5.call(this, a, b);
        if (a.d) {
            if (this.cw.trendLineType != 0 && !b) {
                var c = this.cx.g$j;
                c.ac = this.cw.trendLineThickness;
                c._stroke = this.cw.u1;
                c.ai = this.cw.u2;
                c.ad = this.cw.u3;
                a.u(c);
            }
        }
    };
    ScatterBaseView.prototype.c4 = function (a) {
        if (a) {
            this.c6();
            this.cy.clear();
        }
        this.cx.g$d();
        this.aq();
    };
    ScatterBaseView.prototype.getDefaultTooltipTemplate = function () {
        var a = "<div class='ui-chart-default-tooltip-content'><span";
        var b = this.e.a6.a$r(this.e);
        if (!stringIsNullOrEmpty(b)) {
            a += " style='color:" + b + "'";
        }
        a += ">" + this.cw.title + "</span><br/><span>" + "x: </span><span class='ui-priority-primary'>${item." + this.cw.ti + "}</span><br/><span>" + "y: </span><span class='ui-priority-primary'>${item." + this.cw.tq + "}</span></div>";
        return a;
    };
    ScatterBaseView.$t = markType(ScatterBaseView, 'ScatterBaseView', MarkerSeriesView.$);
    return ScatterBaseView;
}(MarkerSeriesView));
export { ScatterBaseView };
