// igniteui-react-core/esm2015/lib/CalendarResourceStringsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let CalendarResourceStringsDescription = /*@__PURE__*/ (() => {
    class CalendarResourceStringsDescription extends Description {
        constructor() {
            super();
            this.q = null;
            this.s = null;
            this.o = null;
            this.r = null;
            this.p = null;
            this.t = null;
            this.h = null;
            this.l = null;
            this.i = null;
            this.m = null;
            this.j = null;
            this.n = null;
            this.k = null;
            this.u = null;
        }
        get_type() {
            return "CalendarResourceStrings";
        }
        get type() {
            return this.get_type();
        }
        get selectMonth() {
            return this.q;
        }
        set selectMonth(a) {
            this.q = a;
            this.e("SelectMonth");
        }
        get selectYear() {
            return this.s;
        }
        set selectYear(a) {
            this.s = a;
            this.e("SelectYear");
        }
        get selectDate() {
            return this.o;
        }
        set selectDate(a) {
            this.o = a;
            this.e("SelectDate");
        }
        get selectRange() {
            return this.r;
        }
        set selectRange(a) {
            this.r = a;
            this.e("SelectRange");
        }
        get selectedDate() {
            return this.p;
        }
        set selectedDate(a) {
            this.p = a;
            this.e("SelectedDate");
        }
        get startDate() {
            return this.t;
        }
        set startDate(a) {
            this.t = a;
            this.e("StartDate");
        }
        get endDate() {
            return this.h;
        }
        set endDate(a) {
            this.h = a;
            this.e("EndDate");
        }
        get previousMonth() {
            return this.l;
        }
        set previousMonth(a) {
            this.l = a;
            this.e("PreviousMonth");
        }
        get nextMonth() {
            return this.i;
        }
        set nextMonth(a) {
            this.i = a;
            this.e("NextMonth");
        }
        get previousYear() {
            return this.m;
        }
        set previousYear(a) {
            this.m = a;
            this.e("PreviousYear");
        }
        get nextYear() {
            return this.j;
        }
        set nextYear(a) {
            this.j = a;
            this.e("NextYear");
        }
        get previousYears() {
            return this.n;
        }
        set previousYears(a) {
            this.n = a;
            this.e("PreviousYears");
        }
        get nextYears() {
            return this.k;
        }
        set nextYears(a) {
            this.k = a;
            this.e("NextYears");
        }
        get weekLabel() {
            return this.u;
        }
        set weekLabel(a) {
            this.u = a;
            this.e("WeekLabel");
        }
    }
    CalendarResourceStringsDescription.$t = /*@__PURE__*/ markType(CalendarResourceStringsDescription, 'CalendarResourceStringsDescription', Description.$);
    CalendarResourceStringsDescription.__marshalByValue = true;
    CalendarResourceStringsDescription.__marshalByValueAlias = "CalendarResourceStrings";
    return CalendarResourceStringsDescription;
})();
