// igniteui-react-core/esm2015/lib/CategoryYAxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CategoryAxisBaseDescription } from "./CategoryAxisBaseDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let CategoryYAxisDescription = /*@__PURE__*/ (() => {
    class CategoryYAxisDescription extends CategoryAxisBaseDescription {
        constructor() {
            super();
            this.dv = 0;
            this.dt = 0;
            this.dx = 0;
            this.du = 0;
            this.dw = 0;
            this.d3 = null;
            this.d4 = null;
        }
        get_type() {
            return "CategoryYAxis";
        }
        get interval() {
            return this.dv;
        }
        set interval(a) {
            this.dv = a;
            this.e("Interval");
        }
        get actualInterval() {
            return this.dt;
        }
        set actualInterval(a) {
            this.dt = a;
            this.e("ActualInterval");
        }
        get minorInterval() {
            return this.dx;
        }
        set minorInterval(a) {
            this.dx = a;
            this.e("MinorInterval");
        }
        get actualMinorInterval() {
            return this.du;
        }
        set actualMinorInterval(a) {
            this.du = a;
            this.e("ActualMinorInterval");
        }
        get maximumZoomCategories() {
            return this.dw;
        }
        set maximumZoomCategories(a) {
            this.dw = a;
            this.e("ZoomMaximumCategoryRange");
        }
        get actualIntervalChangeRef() {
            return this.d3;
        }
        set actualIntervalChangeRef(a) {
            this.d3 = a;
            this.e("ActualIntervalChangeRef");
        }
        get actualMinorIntervalChangeRef() {
            return this.d4;
        }
        set actualMinorIntervalChangeRef(a) {
            this.d4 = a;
            this.e("ActualMinorIntervalChangeRef");
        }
    }
    CategoryYAxisDescription.$t = /*@__PURE__*/ markType(CategoryYAxisDescription, 'CategoryYAxisDescription', CategoryAxisBaseDescription.$);
    return CategoryYAxisDescription;
})();
