// igniteui-react-core/esm2015/lib/CodeGeneratingComponentRendererModuleEmitter.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, typeCast, markType } from "./type";
import { CodeGeneratingCodeWriter } from "./CodeGeneratingCodeWriter";
import { CodeGeneratingImportManager } from "./CodeGeneratingImportManager";
import { JsonDictionaryArray } from "./JsonDictionaryArray";
import { JsonDictionaryValue } from "./JsonDictionaryValue";
/**
 * @hidden
 */
export let CodeGeneratingComponentRendererModuleEmitter = /*@__PURE__*/ (() => {
    class CodeGeneratingComponentRendererModuleEmitter extends Base {
        constructor(a, b) {
            super();
            this.k = null;
            this.i = null;
            this.b = null;
            this.f = new CodeGeneratingImportManager();
            this.e = new CodeGeneratingImportManager();
            this.a = null;
            this.k = a;
            this.i = b;
            this.b = new CodeGeneratingCodeWriter(this.k);
            this.a = new CodeGeneratingCodeWriter(this.k);
        }
        get j() {
            return this.i;
        }
        get d() {
            return this.b;
        }
        get h() {
            return this.f;
        }
        get g() {
            return this.e;
        }
        get c() {
            return this.a;
        }
        m(a) {
            this.o();
            if (typeCast(JsonDictionaryArray.$, a) !== null) {
                let b = a;
                for (let c = 0; c < b.items.length; c++) {
                    let d = b.items[c];
                    if (typeCast(JsonDictionaryValue.$, d) !== null) {
                        let e = d;
                        if (e != null) {
                            let f = e.value.toString();
                            this.l(f);
                        }
                    }
                }
            }
            this.n();
            this.i.addImportResult("modulesImports", this.h);
            this.i.addContainerResult("modulesRegister", this.d.toString());
            this.i.addImportResult("descriptionImports", this.g);
            this.i.addContainerResult("descriptionRegister", this.c.toString());
        }
        n() {
        }
        l(a) {
        }
        o() {
        }
    }
    CodeGeneratingComponentRendererModuleEmitter.$t = /*@__PURE__*/ markType(CodeGeneratingComponentRendererModuleEmitter, 'CodeGeneratingComponentRendererModuleEmitter');
    return CodeGeneratingComponentRendererModuleEmitter;
})();
