// igniteui-react-core/esm2015/lib/ColumnResizingSeparatorDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { DefinitionBaseDescription } from "./DefinitionBaseDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let ColumnResizingSeparatorDescription = /*@__PURE__*/ (() => {
    class ColumnResizingSeparatorDescription extends DefinitionBaseDescription {
        constructor() {
            super();
            this.b2 = 0;
            this.b3 = 0;
        }
        get_type() {
            return "ColumnResizingSeparator";
        }
        get actualOpacity() {
            return this.b2;
        }
        set actualOpacity(a) {
            this.b2 = a;
            this.e("ActualOpacity");
        }
        get opacity() {
            return this.b3;
        }
        set opacity(a) {
            this.b3 = a;
            this.e("Opacity");
        }
    }
    ColumnResizingSeparatorDescription.$t = /*@__PURE__*/ markType(ColumnResizingSeparatorDescription, 'ColumnResizingSeparatorDescription', DefinitionBaseDescription.$);
    return ColumnResizingSeparatorDescription;
})();
