// igniteui-react-core/esm2015/lib/ColumnCustomFilterConditionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ColumnFilterConditionDescription } from "./ColumnFilterConditionDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let ColumnCustomFilterConditionDescription = /*@__PURE__*/ (() => {
    class ColumnCustomFilterConditionDescription extends ColumnFilterConditionDescription {
        constructor() {
            super();
            this.j = null;
            this.i = null;
            this.g = 0;
        }
        get_type() {
            return "ColumnCustomFilterCondition";
        }
        get valueRef() {
            return this.j;
        }
        set valueRef(a) {
            this.j = a;
            this.e("ValueRef");
        }
        get id() {
            return this.i;
        }
        set id(a) {
            this.i = a;
            this.e("Id");
        }
        get index() {
            return this.g;
        }
        set index(a) {
            this.g = a;
            this.e("Index");
        }
    }
    ColumnCustomFilterConditionDescription.$t = /*@__PURE__*/ markType(ColumnCustomFilterConditionDescription, 'ColumnCustomFilterConditionDescription', ColumnFilterConditionDescription.$);
    return ColumnCustomFilterConditionDescription;
})();
