// igniteui-react-core/esm2015/lib/DataLegendSummaryEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let DataLegendSummaryEventArgsDescription = /*@__PURE__*/ (() => {
    class DataLegendSummaryEventArgsDescription extends Description {
        constructor() {
            super();
            this.f = null;
            this.j = null;
            this.h = 0;
            this.l = null;
            this.k = null;
        }
        get_type() {
            return "DataLegendSummaryEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get columnValues() {
            return this.f;
        }
        set columnValues(a) {
            this.f = a;
            this.e("ColumnValues");
        }
        get columnMemberPath() {
            return this.j;
        }
        set columnMemberPath(a) {
            this.j = a;
            this.e("ColumnMemberPath");
        }
        get summaryValue() {
            return this.h;
        }
        set summaryValue(a) {
            this.h = a;
            this.e("SummaryValue");
        }
        get summaryUnits() {
            return this.l;
        }
        set summaryUnits(a) {
            this.l = a;
            this.e("SummaryUnits");
        }
        get summaryLabel() {
            return this.k;
        }
        set summaryLabel(a) {
            this.k = a;
            this.e("SummaryLabel");
        }
    }
    DataLegendSummaryEventArgsDescription.$t = /*@__PURE__*/ markType(DataLegendSummaryEventArgsDescription, 'DataLegendSummaryEventArgsDescription', Description.$);
    return DataLegendSummaryEventArgsDescription;
})();
