// igniteui-react-core/esm2015/lib/DataLegendSeriesGroup.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "./type";
/**
 * Describes type of data columns to use for the data legend.
 */
export var DataLegendSeriesGroup = /*@__PURE__*/ (function (DataLegendSeriesGroup) {
    /**
     * Specifies the series
     */
    DataLegendSeriesGroup[DataLegendSeriesGroup["Category"] = 0] = "Category";
    /**
     * Specifies the series
     */
    DataLegendSeriesGroup[DataLegendSeriesGroup["Range"] = 1] = "Range";
    /**
     * Specifies the series
     */
    DataLegendSeriesGroup[DataLegendSeriesGroup["Financial"] = 2] = "Financial";
    /**
     * Specifies the series
     */
    DataLegendSeriesGroup[DataLegendSeriesGroup["Indicator"] = 3] = "Indicator";
    /**
     * Specifies the series
     */
    DataLegendSeriesGroup[DataLegendSeriesGroup["Polar"] = 4] = "Polar";
    /**
     * Specifies the series
     */
    DataLegendSeriesGroup[DataLegendSeriesGroup["Radial"] = 5] = "Radial";
    /**
     * Specifies the series
     */
    DataLegendSeriesGroup[DataLegendSeriesGroup["Scatter"] = 6] = "Scatter";
    /**
     * Specifies the series
     */
    DataLegendSeriesGroup[DataLegendSeriesGroup["Stacked"] = 7] = "Stacked";
    /**
     * Specifies the series
     */
    DataLegendSeriesGroup[DataLegendSeriesGroup["Shape"] = 8] = "Shape";
    /**
     * Specifies the series
     */
    DataLegendSeriesGroup[DataLegendSeriesGroup["Geographic"] = 9] = "Geographic";
    return DataLegendSeriesGroup;
})({});
/**
 * @hidden
 */
export let DataLegendSeriesGroup_$type = /*@__PURE__*/ markEnum('DataLegendSeriesGroup', 'Category,0|Range,1|Financial,2|Indicator,3|Polar,4|Radial,5|Scatter,6|Stacked,7|Shape,8|Geographic,9');
