// igniteui-react-core/esm2015/lib/DotNetCodeGeneratingComponentRendererModuleEmitter.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CodeGeneratingComponentRendererModuleEmitter } from "./CodeGeneratingComponentRendererModuleEmitter";
import { Dictionary$2 } from "./Dictionary$2";
import { List$1 } from "./List$1";
import { String_$type, markType } from "./type";
import { stringEndsWith } from "./string";
/**
 * @hidden
 */
export let DotNetCodeGeneratingComponentRendererModuleEmitter = /*@__PURE__*/ (() => {
    class DotNetCodeGeneratingComponentRendererModuleEmitter extends CodeGeneratingComponentRendererModuleEmitter {
        constructor(a, b, c) {
            super(b, c);
            this.p = 0;
            this.s = 0;
            this.q = false;
            this.r = new Dictionary$2(String_$type, List$1.$.specialize(String_$type), 0);
            this.p = a;
        }
        get t() {
            return this.s;
        }
        o() {
            super.o();
            this.s = 0;
        }
        get u() {
            switch (this.p) {
                case 0: return "Xam";
                case 4: return "Ultra";
                case 5: return "Igb";
            }
            return "";
        }
        l(a) {
            super.l(a);
            if (a == "withDescriptions" && this.p != 5) {
                this.q = true;
                return;
            }
            let b = "";
            let c = a.split('/');
            if (c.length == 1) {
                b = c[0];
                this.j.addPackageRef(b);
            }
            else {
                this.j.addPackageRef(c[0]);
                b = c[1];
            }
            if (this.q) {
                if (this.s == 0) {
                    this.c.k("private ComponentRenderer _componentRenderer = null;");
                    this.c.k("public ComponentRenderer Renderer");
                    this.c.k("{");
                    this.c.h();
                    this.c.k("get");
                    this.c.k("{");
                    this.c.h();
                    this.c.k("if (this._componentRenderer == null)");
                    this.c.k("{");
                    this.c.h();
                    this.c.k("this._componentRenderer = new ComponentRenderer();");
                    this.c.k("var context = this._componentRenderer.Context;");
                }
                let d = b;
                if (stringEndsWith(d, "Module")) {
                    d = d.substr(0, d.length - ("Module").length);
                    d += "DescriptionModule";
                }
                this.c.k(d + ".Register(context);");
            }
            this.s++;
        }
        n() {
            super.n();
            if (this.q) {
                this.g.addUsing("using Infragistics.Controls.Description;");
                this.c.f();
                this.c.k("}");
                this.c.f();
                this.c.k("return this._componentRenderer;");
                this.c.k("}");
                this.c.f();
                this.c.k("}");
            }
        }
    }
    DotNetCodeGeneratingComponentRendererModuleEmitter.$t = /*@__PURE__*/ markType(DotNetCodeGeneratingComponentRendererModuleEmitter, 'DotNetCodeGeneratingComponentRendererModuleEmitter', CodeGeneratingComponentRendererModuleEmitter.$);
    return DotNetCodeGeneratingComponentRendererModuleEmitter;
})();
