// igniteui-react-core/esm2015/lib/FunnelSliceClickedEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let FunnelSliceClickedEventArgsDescription = /*@__PURE__*/ (() => {
    class FunnelSliceClickedEventArgsDescription extends Description {
        constructor() {
            super();
            this.h = 0;
            this.f = null;
            this.j = null;
        }
        get_type() {
            return "FunnelSliceClickedEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get index() {
            return this.h;
        }
        set index(a) {
            this.h = a;
            this.e("Index");
        }
        get bounds() {
            return this.f;
        }
        set bounds(a) {
            this.f = a;
            this.e("Bounds");
        }
        get itemRef() {
            return this.j;
        }
        set itemRef(a) {
            this.j = a;
            this.e("ItemRef");
        }
    }
    FunnelSliceClickedEventArgsDescription.$t = /*@__PURE__*/ markType(FunnelSliceClickedEventArgsDescription, 'FunnelSliceClickedEventArgsDescription', Description.$);
    return FunnelSliceClickedEventArgsDescription;
})();
