// igniteui-react-core/esm2015/lib/FunnelDataContextDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let FunnelDataContextDescription = /*@__PURE__*/ (() => {
    class FunnelDataContextDescription extends Description {
        constructor() {
            super();
            this.h = null;
            this.f = 0;
        }
        get_type() {
            return "FunnelDataContext";
        }
        get type() {
            return this.get_type();
        }
        get itemRef() {
            return this.h;
        }
        set itemRef(a) {
            this.h = a;
            this.e("ItemRef");
        }
        get index() {
            return this.f;
        }
        set index(a) {
            this.f = a;
            this.e("Index");
        }
    }
    FunnelDataContextDescription.$t = /*@__PURE__*/ markType(FunnelDataContextDescription, 'FunnelDataContextDescription', Description.$);
    return FunnelDataContextDescription;
})();
