// igniteui-react-core/esm2015/lib/GridCellEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let GridCellEventArgsDescription = /*@__PURE__*/ (() => {
    class GridCellEventArgsDescription extends Description {
        constructor() {
            super();
            this.f = null;
            this.h = null;
            this.l = null;
            this.j = false;
        }
        get_type() {
            return "GridCellEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get cellInfo() {
            return this.f;
        }
        set cellInfo(a) {
            this.f = a;
            this.e("CellInfo");
        }
        get grid() {
            return this.h;
        }
        set grid(a) {
            this.h = a;
            this.e("Grid");
        }
        get button() {
            return this.l;
        }
        set button(a) {
            this.l = a;
            this.e("Button");
        }
        get isDoubleClick() {
            return this.j;
        }
        set isDoubleClick(a) {
            this.j = a;
            this.e("IsDoubleClick");
        }
    }
    GridCellEventArgsDescription.$t = /*@__PURE__*/ markType(GridCellEventArgsDescription, 'GridCellEventArgsDescription', Description.$);
    return GridCellEventArgsDescription;
})();
