// igniteui-react-core/esm2015/lib/GridCustomFilterRequestedEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let GridCustomFilterRequestedEventArgsDescription = /*@__PURE__*/ (() => {
    class GridCustomFilterRequestedEventArgsDescription extends Description {
        constructor() {
            super();
            this.f = null;
            this.j = null;
            this.l = null;
            this.h = null;
        }
        get_type() {
            return "GridCustomFilterRequestedEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get column() {
            return this.f;
        }
        set column(a) {
            this.f = a;
            this.e("Column");
        }
        get filter() {
            return this.j;
        }
        set filter(a) {
            this.j = a;
            this.e("Filter");
        }
        get valueRef() {
            return this.l;
        }
        set valueRef(a) {
            this.l = a;
            this.e("ValueRef");
        }
        get expression() {
            return this.h;
        }
        set expression(a) {
            this.h = a;
            this.e("Expression");
        }
    }
    GridCustomFilterRequestedEventArgsDescription.$t = /*@__PURE__*/ markType(GridCustomFilterRequestedEventArgsDescription, 'GridCustomFilterRequestedEventArgsDescription', Description.$);
    return GridCustomFilterRequestedEventArgsDescription;
})();
