// igniteui-react-core/esm2015/lib/InputDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { InputGroupItemDescription } from "./InputGroupItemDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let InputDescription = /*@__PURE__*/ (() => {
    class InputDescription extends InputGroupItemDescription {
        constructor() {
            super();
            this.ae = null;
            this.y = null;
            this.ap = null;
            this.ar = null;
            this.j = false;
            this.aa = null;
            this.z = null;
            this.as = null;
            this.af = null;
            this.q = 0;
            this.ag = null;
            this.ah = null;
            this.al = null;
            this.aq = null;
            this.aj = null;
            this.ak = null;
            this.u = 0;
            this.ai = null;
            this.ab = null;
            this.k = false;
            this.at = null;
            this.i = false;
            this.l = false;
            this.t = 0;
            this.s = 0;
            this.am = null;
            this.ao = null;
            this.an = null;
            this.ac = null;
            this.ad = null;
        }
        get_type() {
            return "Input";
        }
        get density() {
            return this.ae;
        }
        set density(a) {
            this.ae = a;
            this.e("Density");
        }
        get actualDensity() {
            return this.y;
        }
        set actualDensity(a) {
            this.y = a;
            this.e("ActualDensity");
        }
        get mask() {
            return this.ap;
        }
        set mask(a) {
            this.ap = a;
            this.e("Mask");
        }
        get promptChar() {
            return this.ar;
        }
        set promptChar(a) {
            this.ar = a;
            this.e("PromptChar");
        }
        get includeLiterals() {
            return this.j;
        }
        set includeLiterals(a) {
            this.j = a;
            this.e("IncludeLiterals");
        }
        get actualTextColor() {
            return this.aa;
        }
        set actualTextColor(a) {
            this.aa = a;
            this.e("ActualTextColor");
        }
        get actualHoverTextColor() {
            return this.z;
        }
        set actualHoverTextColor(a) {
            this.z = a;
            this.e("ActualHoverTextColor");
        }
        get textColor() {
            return this.as;
        }
        set textColor(a) {
            this.as = a;
            this.e("TextColor");
        }
        get fontFamily() {
            return this.af;
        }
        set fontFamily(a) {
            this.af = a;
            this.e("FontFamily");
        }
        get fontSize() {
            return this.q;
        }
        set fontSize(a) {
            this.q = a;
            this.e("FontSize");
        }
        get fontStyle() {
            return this.ag;
        }
        set fontStyle(a) {
            this.ag = a;
            this.e("FontStyle");
        }
        get fontWeight() {
            return this.ah;
        }
        set fontWeight(a) {
            this.ah = a;
            this.e("FontWeight");
        }
        get inputType() {
            return this.al;
        }
        set inputType(a) {
            this.al = a;
            this.e("InputType");
        }
        get placeholder() {
            return this.aq;
        }
        set placeholder(a) {
            this.aq = a;
            this.e("Placeholder");
        }
        get hoverTextColor() {
            return this.aj;
        }
        set hoverTextColor(a) {
            this.aj = a;
            this.e("HoverTextColor");
        }
        get id() {
            return this.ak;
        }
        set id(a) {
            this.ak = a;
            this.e("Id");
        }
        get tabIndex() {
            return this.u;
        }
        set tabIndex(a) {
            this.u = a;
            this.e("TabIndex");
        }
        get $for() {
            return this.ai;
        }
        set $for(a) {
            this.ai = a;
            this.e("For");
        }
        get ariaLabel() {
            return this.ab;
        }
        set ariaLabel(a) {
            this.ab = a;
            this.e("AriaLabel");
        }
        get isHover() {
            return this.k;
        }
        set isHover(a) {
            this.k = a;
            this.e("IsHover");
        }
        get value() {
            return this.at;
        }
        set value(a) {
            this.at = a;
            this.e("Value");
        }
        get disabled() {
            return this.i;
        }
        set disabled(a) {
            this.i = a;
            this.e("Disabled");
        }
        get readonly() {
            return this.l;
        }
        set readonly(a) {
            this.l = a;
            this.e("Readonly");
        }
        get selectionStart() {
            return this.t;
        }
        set selectionStart(a) {
            this.t = a;
            this.e("SelectionStart");
        }
        get selectionEnd() {
            return this.s;
        }
        set selectionEnd(a) {
            this.s = a;
            this.e("SelectionEnd");
        }
        get keyDownRef() {
            return this.am;
        }
        set keyDownRef(a) {
            this.am = a;
            this.e("KeyDownRef");
        }
        get keyUpRef() {
            return this.ao;
        }
        set keyUpRef(a) {
            this.ao = a;
            this.e("KeyUpRef");
        }
        get keyPressRef() {
            return this.an;
        }
        set keyPressRef(a) {
            this.an = a;
            this.e("KeyPressRef");
        }
        get changeRef() {
            return this.ac;
        }
        set changeRef(a) {
            this.ac = a;
            this.e("ChangeRef");
        }
        get changingRef() {
            return this.ad;
        }
        set changingRef(a) {
            this.ad = a;
            this.e("ChangingRef");
        }
    }
    InputDescription.$t = /*@__PURE__*/ markType(InputDescription, 'InputDescription', InputGroupItemDescription.$);
    return InputDescription;
})();
