// igniteui-react-core/esm2015/lib/ListPanelTemplateHeightRequestedEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let ListPanelTemplateHeightRequestedEventArgsDescription = /*@__PURE__*/ (() => {
    class ListPanelTemplateHeightRequestedEventArgsDescription extends Description {
        constructor() {
            super();
            this.h = 0;
            this.j = null;
            this.f = 0;
        }
        get_type() {
            return "ListPanelTemplateHeightRequestedEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get dataRow() {
            return this.h;
        }
        set dataRow(a) {
            this.h = a;
            this.e("DataRow");
        }
        get dataItemRef() {
            return this.j;
        }
        set dataItemRef(a) {
            this.j = a;
            this.e("DataItemRef");
        }
        get height() {
            return this.f;
        }
        set height(a) {
            this.f = a;
            this.e("Height");
        }
    }
    ListPanelTemplateHeightRequestedEventArgsDescription.$t = /*@__PURE__*/ markType(ListPanelTemplateHeightRequestedEventArgsDescription, 'ListPanelTemplateHeightRequestedEventArgsDescription', Description.$);
    return ListPanelTemplateHeightRequestedEventArgsDescription;
})();
