// igniteui-react-core/esm2015/lib/LiteralFilterExpressionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { FilterExpressionDescription } from "./FilterExpressionDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let LiteralFilterExpressionDescription = /*@__PURE__*/ (() => {
    class LiteralFilterExpressionDescription extends FilterExpressionDescription {
        constructor() {
            super();
            this.l = null;
            this.j = false;
        }
        get_type() {
            return "LiteralFilterExpression";
        }
        get literalValue() {
            return this.l;
        }
        set literalValue(a) {
            this.l = a;
            this.e("LiteralValue");
        }
        get leaveUnquoted() {
            return this.j;
        }
        set leaveUnquoted(a) {
            this.j = a;
            this.e("LeaveUnquoted");
        }
    }
    LiteralFilterExpressionDescription.$t = /*@__PURE__*/ markType(LiteralFilterExpressionDescription, 'LiteralFilterExpressionDescription', FilterExpressionDescription.$);
    LiteralFilterExpressionDescription.__marshalByValue1 = true;
    LiteralFilterExpressionDescription.__marshalByValueAlias1 = "LiteralFilterExpression";
    return LiteralFilterExpressionDescription;
})();
