// igniteui-react-core/esm2015/lib/NativeUI.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "./type";
import { UIElementPropertyAccessorFactory } from "./UIElementPropertyAccessorFactory";
import { MatrixTransform } from "./MatrixTransform";
import { TransformMatrixHelpler } from "./TransformMatrixHelpler";
/**
 * @hidden
 */
export let NativeUI = /*@__PURE__*/ (() => {
    class NativeUI extends Base {
        static ac(a, b, c) {
            let d = UIElementPropertyAccessorFactory.c.a(a, 0);
            d.c(b, c);
        }
        static j(a, b) {
            let c = UIElementPropertyAccessorFactory.c.a(a, 0);
            return c.b(b);
        }
        static u(a, b) {
            NativeUI.ac(1, a, b);
        }
        static e(a) {
            return NativeUI.j(1, a);
        }
        static k(a, b) {
            NativeUI.ac(2, a, b);
        }
        static af(a) {
            return NativeUI.j(2, a);
        }
        static l(a, b) {
            NativeUI.ac(3, a, b);
        }
        static ag(a) {
            return NativeUI.j(3, a);
        }
        static n(a, b) {
            NativeUI.ac(4, a, b);
        }
        static ak(a) {
            return NativeUI.j(4, a);
        }
        static m(a, b) {
            NativeUI.ac(5, a, b);
        }
        static v(a, b) {
            NativeUI.ac(6, a, b);
        }
        static am(a) {
            return NativeUI.j(6, a);
        }
        static ae(a) {
            return NativeUI.j(5, a);
        }
        static o(a, b) {
            NativeUI.ac(7, a, b);
        }
        static ah(a) {
            return NativeUI.j(7, a);
        }
        static q(a, b) {
            NativeUI.ac(8, a, b);
        }
        static b(a) {
            return NativeUI.j(8, a);
        }
        static ad(a, b) {
            NativeUI.ac(9, a, b);
        }
        static i(a) {
            return NativeUI.j(9, a);
        }
        static r(a, b) {
            NativeUI.ac(10, a, b);
        }
        static c(a) {
            return NativeUI.j(10, a);
        }
        static s(a, b) {
            NativeUI.ac(11, a, b);
        }
        static aa(a, b) {
            NativeUI.ac(12, a, b);
        }
        static d(a) {
            return NativeUI.j(11, a);
        }
        static h(a) {
            return NativeUI.j(12, a);
        }
        static t(a, b) {
            NativeUI.ac(13, a, b);
        }
        static al(a) {
            return NativeUI.j(13, a);
        }
        static ab(a, b) {
            let c = TransformMatrixHelpler.f(b);
            let d = new MatrixTransform();
            d.j = c;
            NativeUI.ac(14, a, d);
        }
        static aj(a) {
            return NativeUI.j(14, a);
        }
        static w(a, b) {
            NativeUI.ac(15, a, b);
        }
        static p(a, b) {
            NativeUI.ac(16, a, b);
        }
        static ai(a) {
            return NativeUI.j(15, a);
        }
        static z(a, b) {
            NativeUI.ac(17, a, b);
        }
        static g(a) {
            return NativeUI.j(17, a);
        }
        static x(a, b) {
            NativeUI.ac(18, a, b);
        }
        static a(a) {
            return NativeUI.j(18, a);
        }
        static y(a, b) {
            NativeUI.ac(19, a, b);
        }
        static f(a) {
            return NativeUI.j(19, a);
        }
    }
    NativeUI.$t = /*@__PURE__*/ markType(NativeUI, 'NativeUI');
    return NativeUI;
})();
