// igniteui-react-core/esm2015/lib/NativeUIGrid.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { NativeUIContainer } from "./NativeUIContainer";
import { markType } from "./type";
/**
 * @hidden
 */
export let NativeUIGrid = /*@__PURE__*/ (() => {
    class NativeUIGrid extends NativeUIContainer {
        get_c() {
            return 7;
        }
        get a3() {
            return this.az.getRowDefinitionsCount(this.u);
        }
        get a0() {
            return this.az.getColumnDefinitionsCount(this.u);
        }
        get az() {
            return this.b;
        }
        a5(a, b, c) {
            this.az.addRowDefinition(this.u, a, b, c);
        }
        bb(a, b, c) {
            this.az.updateRowDefinition(this.u, a, b, c);
        }
        a4(a, b, c) {
            this.az.addColumnDefinition(this.u, a, b, c);
        }
        ba(a, b, c) {
            this.az.updateColumnDefinition(this.u, a, b, c);
        }
        a9(a, b) {
            this.az.setRow(this.u, a, b);
        }
        a8(a, b) {
            this.az.setColumn(this.u, a, b);
        }
        a2(a) {
            return this.az.getRow(this.u, a);
        }
        a1(a) {
            return this.az.getColumn(this.u, a);
        }
        a7(a) {
            this.az.removeRowDefinition(this.u, a);
        }
        a6(a) {
            this.az.removeColumnDefinition(this.u, a);
        }
    }
    NativeUIGrid.$t = /*@__PURE__*/ markType(NativeUIGrid, 'NativeUIGrid', NativeUIContainer.$);
    return NativeUIGrid;
})();
