// igniteui-react-core/esm2015/lib/NativeUICheckbox.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { NativeUIComponent } from "./NativeUIComponent";
import { markType } from "./type";
/**
 * @hidden
 */
export let NativeUICheckbox = /*@__PURE__*/ (() => {
    class NativeUICheckbox extends NativeUIComponent {
        constructor() {
            super(...arguments);
            this.at = null;
        }
        get_c() {
            return 6;
        }
        get av() {
            let a = this.o(4);
            return a;
        }
        set av(a) {
            this.af(4, a);
        }
        get aw() {
            let a = this.o(5);
            return a;
        }
        set aw(a) {
            this.af(5, a);
        }
        get au() {
            return this.at;
        }
        set au(a) {
            let b = this.at;
            this.at = a;
            if (this.at != null) {
                this.x(2, this.at);
            }
            else {
                this.ad(2, b);
            }
        }
    }
    NativeUICheckbox.$t = /*@__PURE__*/ markType(NativeUICheckbox, 'NativeUICheckbox', NativeUIComponent.$);
    return NativeUICheckbox;
})();
