// igniteui-react-core/esm2015/lib/OrdinalTimeXAxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CategoryXAxisDescription } from "./CategoryXAxisDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let OrdinalTimeXAxisDescription = /*@__PURE__*/ (() => {
    class OrdinalTimeXAxisDescription extends CategoryXAxisDescription {
        constructor() {
            super();
            this.ed = null;
            this.ee = null;
            this.d7 = null;
            this.ea = new Date();
            this.d9 = new Date();
        }
        get_type() {
            return "OrdinalTimeXAxis";
        }
        get dateTimeMemberPath() {
            return this.ed;
        }
        set dateTimeMemberPath(a) {
            this.ed = a;
            this.e("DateTimeMemberPath");
        }
        get labellingMode() {
            return this.ee;
        }
        set labellingMode(a) {
            this.ee = a;
            this.e("LabellingMode");
        }
        get labelFormats() {
            return this.d7;
        }
        set labelFormats(a) {
            this.d7 = a;
            this.e("LabelFormats");
        }
        get minimumValue() {
            return this.ea;
        }
        set minimumValue(a) {
            this.ea = a;
            this.e("MinimumValue");
        }
        get maximumValue() {
            return this.d9;
        }
        set maximumValue(a) {
            this.d9 = a;
            this.e("MaximumValue");
        }
    }
    OrdinalTimeXAxisDescription.$t = /*@__PURE__*/ markType(OrdinalTimeXAxisDescription, 'OrdinalTimeXAxisDescription', CategoryXAxisDescription.$);
    return OrdinalTimeXAxisDescription;
})();
