// igniteui-react-core/esm2015/lib/PlotAreaMouseButtonEventArgsDescriptionMetadata_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, String_$type, markType } from "./type";
import { Dictionary$2 } from "./Dictionary$2";
import { PlotAreaMouseButtonEventArgsDescription } from "./PlotAreaMouseButtonEventArgsDescription";
import { PlotAreaMouseEventArgsDescription } from "./PlotAreaMouseEventArgsDescription";
import { SeriesDescriptionMetadata } from "./SeriesDescriptionMetadata";
import { ChartCursorEventArgsDescription } from "./ChartCursorEventArgsDescription";
import { ChartMouseEventArgsDescription } from "./ChartMouseEventArgsDescription";
import { DataChartMouseButtonEventArgsDescription } from "./DataChartMouseButtonEventArgsDescription";
import { SeriesViewerManipulationEventArgsDescriptionMetadata } from "./SeriesViewerManipulationEventArgsDescriptionMetadata";
import { RectChangedEventArgsDescriptionMetadata } from "./RectChangedEventArgsDescriptionMetadata";
import { RefreshCompletedEventArgsDescriptionMetadata } from "./RefreshCompletedEventArgsDescriptionMetadata";
/**
 * @hidden
 */
export let PlotAreaMouseButtonEventArgsDescriptionMetadata = /*@__PURE__*/ (() => {
    class PlotAreaMouseButtonEventArgsDescriptionMetadata extends Base {
        static b(a) {
            if (PlotAreaMouseButtonEventArgsDescriptionMetadata.a == null) {
                PlotAreaMouseButtonEventArgsDescriptionMetadata.a = new Dictionary$2(String_$type, String_$type, 0);
                PlotAreaMouseButtonEventArgsDescriptionMetadata.c(PlotAreaMouseButtonEventArgsDescriptionMetadata.a);
            }
            if (a.g(PlotAreaMouseButtonEventArgsDescriptionMetadata.a)) {
                return;
            }
            a.s(PlotAreaMouseButtonEventArgsDescriptionMetadata.a);
            SeriesViewerDescriptionMetadata.d(a);
        }
        static c(a) {
            a.item("ChartPosition", "Point");
            a.item("PlotAreaPosition", "Point");
            a.item("ManipulationOccurred", "Boolean");
            a.item("Viewer", "ExportedType:SeriesViewer");
        }
        static d(a) {
            PlotAreaMouseButtonEventArgsDescriptionMetadata.b(a);
            a.u("PlotAreaMouseButtonEventArgs", () => new PlotAreaMouseButtonEventArgsDescription());
            a.t("PlotAreaMouseButtonEventArgs", PlotAreaMouseButtonEventArgsDescriptionMetadata.a);
        }
    }
    PlotAreaMouseButtonEventArgsDescriptionMetadata.$t = /*@__PURE__*/ markType(PlotAreaMouseButtonEventArgsDescriptionMetadata, 'PlotAreaMouseButtonEventArgsDescriptionMetadata');
    PlotAreaMouseButtonEventArgsDescriptionMetadata.a = null;
    return PlotAreaMouseButtonEventArgsDescriptionMetadata;
})();
/**
 * @hidden
 */
export let PlotAreaMouseEventArgsDescriptionMetadata = /*@__PURE__*/ (() => {
    class PlotAreaMouseEventArgsDescriptionMetadata extends Base {
        static b(a) {
            if (PlotAreaMouseEventArgsDescriptionMetadata.a == null) {
                PlotAreaMouseEventArgsDescriptionMetadata.a = new Dictionary$2(String_$type, String_$type, 0);
                PlotAreaMouseEventArgsDescriptionMetadata.c(PlotAreaMouseEventArgsDescriptionMetadata.a);
            }
            if (a.g(PlotAreaMouseEventArgsDescriptionMetadata.a)) {
                return;
            }
            a.s(PlotAreaMouseEventArgsDescriptionMetadata.a);
            SeriesViewerDescriptionMetadata.d(a);
        }
        static c(a) {
            a.item("ChartPosition", "Point");
            a.item("PlotAreaPosition", "Point");
            a.item("IsDuringManipulation", "Boolean");
            a.item("Viewer", "ExportedType:SeriesViewer");
        }
        static d(a) {
            PlotAreaMouseEventArgsDescriptionMetadata.b(a);
            a.u("PlotAreaMouseEventArgs", () => new PlotAreaMouseEventArgsDescription());
            a.t("PlotAreaMouseEventArgs", PlotAreaMouseEventArgsDescriptionMetadata.a);
        }
    }
    PlotAreaMouseEventArgsDescriptionMetadata.$t = /*@__PURE__*/ markType(PlotAreaMouseEventArgsDescriptionMetadata, 'PlotAreaMouseEventArgsDescriptionMetadata');
    PlotAreaMouseEventArgsDescriptionMetadata.a = null;
    return PlotAreaMouseEventArgsDescriptionMetadata;
})();
/**
 * @hidden
 */
export let ChartCursorEventArgsDescriptionMetadata = /*@__PURE__*/ (() => {
    class ChartCursorEventArgsDescriptionMetadata extends Base {
        static b(a) {
            if (ChartCursorEventArgsDescriptionMetadata.a == null) {
                ChartCursorEventArgsDescriptionMetadata.a = new Dictionary$2(String_$type, String_$type, 0);
                ChartCursorEventArgsDescriptionMetadata.c(ChartCursorEventArgsDescriptionMetadata.a);
            }
            if (a.g(ChartCursorEventArgsDescriptionMetadata.a)) {
                return;
            }
            a.s(ChartCursorEventArgsDescriptionMetadata.a);
            SeriesDescriptionMetadata.d(a);
            SeriesViewerDescriptionMetadata.d(a);
        }
        static c(a) {
            a.item("ItemRef", "(w:Item,wf:Item)DataRef");
            a.item("Series", "ExportedType:Series");
            a.item("SeriesViewer", "ExportedType:SeriesViewer");
        }
        static d(a) {
            ChartCursorEventArgsDescriptionMetadata.b(a);
            a.u("ChartCursorEventArgs", () => new ChartCursorEventArgsDescription());
            a.t("ChartCursorEventArgs", ChartCursorEventArgsDescriptionMetadata.a);
        }
    }
    ChartCursorEventArgsDescriptionMetadata.$t = /*@__PURE__*/ markType(ChartCursorEventArgsDescriptionMetadata, 'ChartCursorEventArgsDescriptionMetadata');
    ChartCursorEventArgsDescriptionMetadata.a = null;
    return ChartCursorEventArgsDescriptionMetadata;
})();
/**
 * @hidden
 */
export let ChartMouseEventArgsDescriptionMetadata = /*@__PURE__*/ (() => {
    class ChartMouseEventArgsDescriptionMetadata extends Base {
        static b(a) {
            if (ChartMouseEventArgsDescriptionMetadata.a == null) {
                ChartMouseEventArgsDescriptionMetadata.a = new Dictionary$2(String_$type, String_$type, 0);
                ChartMouseEventArgsDescriptionMetadata.c(ChartMouseEventArgsDescriptionMetadata.a);
            }
            if (a.g(ChartMouseEventArgsDescriptionMetadata.a)) {
                return;
            }
            a.s(ChartMouseEventArgsDescriptionMetadata.a);
            SeriesDescriptionMetadata.d(a);
            SeriesViewerDescriptionMetadata.d(a);
        }
        static c(a) {
            a.item("ItemRef", "(w:Item,wf:Item)DataRef");
            a.item("Series", "ExportedType:Series");
            a.item("Chart", "ExportedType:SeriesViewer");
        }
        static d(a) {
            ChartMouseEventArgsDescriptionMetadata.b(a);
            a.u("ChartMouseEventArgs", () => new ChartMouseEventArgsDescription());
            a.t("ChartMouseEventArgs", ChartMouseEventArgsDescriptionMetadata.a);
        }
    }
    ChartMouseEventArgsDescriptionMetadata.$t = /*@__PURE__*/ markType(ChartMouseEventArgsDescriptionMetadata, 'ChartMouseEventArgsDescriptionMetadata');
    ChartMouseEventArgsDescriptionMetadata.a = null;
    return ChartMouseEventArgsDescriptionMetadata;
})();
/**
 * @hidden
 */
export let DataChartMouseButtonEventArgsDescriptionMetadata = /*@__PURE__*/ (() => {
    class DataChartMouseButtonEventArgsDescriptionMetadata extends Base {
        static b(a) {
            if (DataChartMouseButtonEventArgsDescriptionMetadata.a == null) {
                DataChartMouseButtonEventArgsDescriptionMetadata.a = new Dictionary$2(String_$type, String_$type, 0);
                DataChartMouseButtonEventArgsDescriptionMetadata.c(DataChartMouseButtonEventArgsDescriptionMetadata.a);
            }
            if (a.g(DataChartMouseButtonEventArgsDescriptionMetadata.a)) {
                return;
            }
            a.s(DataChartMouseButtonEventArgsDescriptionMetadata.a);
            SeriesDescriptionMetadata.d(a);
            SeriesViewerDescriptionMetadata.d(a);
        }
        static c(a) {
            a.item("Handled", "Boolean");
            a.item("ItemRef", "(w:Item,wf:Item)DataRef");
            a.item("Series", "ExportedType:Series");
            a.item("Chart", "ExportedType:SeriesViewer");
        }
        static d(a) {
            DataChartMouseButtonEventArgsDescriptionMetadata.b(a);
            a.u("DataChartMouseButtonEventArgs", () => new DataChartMouseButtonEventArgsDescription());
            a.t("DataChartMouseButtonEventArgs", DataChartMouseButtonEventArgsDescriptionMetadata.a);
        }
    }
    DataChartMouseButtonEventArgsDescriptionMetadata.$t = /*@__PURE__*/ markType(DataChartMouseButtonEventArgsDescriptionMetadata, 'DataChartMouseButtonEventArgsDescriptionMetadata');
    DataChartMouseButtonEventArgsDescriptionMetadata.a = null;
    return DataChartMouseButtonEventArgsDescriptionMetadata;
})();
/**
 * @hidden
 */
export let SeriesViewerDescriptionMetadata = /*@__PURE__*/ (() => {
    class SeriesViewerDescriptionMetadata extends Base {
        static b(a) {
            if (SeriesViewerDescriptionMetadata.a == null) {
                SeriesViewerDescriptionMetadata.a = new Dictionary$2(String_$type, String_$type, 0);
                SeriesViewerDescriptionMetadata.c(SeriesViewerDescriptionMetadata.a);
            }
            if (a.g(SeriesViewerDescriptionMetadata.a)) {
                return;
            }
            a.s(SeriesViewerDescriptionMetadata.a);
            PlotAreaMouseButtonEventArgsDescriptionMetadata.d(a);
            PlotAreaMouseEventArgsDescriptionMetadata.d(a);
            ChartCursorEventArgsDescriptionMetadata.d(a);
            DataChartMouseButtonEventArgsDescriptionMetadata.d(a);
            ChartMouseEventArgsDescriptionMetadata.d(a);
            SeriesViewerManipulationEventArgsDescriptionMetadata.d(a);
            RectChangedEventArgsDescriptionMetadata.d(a);
            RefreshCompletedEventArgsDescriptionMetadata.d(a);
        }
        static c(a) {
            a.item("Series", "Collection:Series:SeriesCollection:Series");
            a.item("Brushes", "BrushCollection:string");
            a.item("Outlines", "BrushCollection:string");
            a.item("MarkerBrushes", "BrushCollection:string");
            a.item("MarkerOutlines", "BrushCollection:string");
            a.item("MarkerAutomaticBehavior", "ExportedType:string:MarkerAutomaticBehavior");
            a.item("MarkerAutomaticBehavior@names", "CircleSmart;None;SmartIndexed;Indexed;Circle;Triangle;Pyramid;Square;Diamond;Pentagon;Hexagon;Tetragram;Pentagram;Hexagram");
            a.item("CrosshairPoint", "Point");
            a.item("LegendRef", "(w:Legend,wf:Legend)DataRef::object");
            a.item("AnimateSeriesWhenAxisRangeChanges", "Boolean");
            a.item("IsWindowSyncedToVisibleRange", "Boolean");
            a.item("ShouldSimulateHoverMoveCrosshairPoint", "Boolean");
            a.item("HighlightingMode", "ExportedType:string:SeriesHighlightingMode");
            a.item("HighlightingMode@names", "Auto;None;BrightenSpecific;Brighten;FadeOthersSpecific;FadeOthers");
            a.item("HorizontalViewScrollbarMode", "ExportedType:string:SeriesViewerScrollbarMode");
            a.item("HorizontalViewScrollbarMode@names", "None;Fading;FadeToLine;Persistent");
            a.item("VerticalViewScrollbarMode", "ExportedType:string:SeriesViewerScrollbarMode");
            a.item("VerticalViewScrollbarMode@names", "None;Fading;FadeToLine;Persistent");
            a.item("HorizontalViewScrollbarPosition", "ExportedType:string:SeriesViewerHorizontalScrollbarPosition");
            a.item("HorizontalViewScrollbarPosition@names", "Auto;InsideBottom;InsideTop");
            a.item("VerticalViewScrollbarPosition", "ExportedType:string:SeriesViewerVerticalScrollbarPosition");
            a.item("VerticalViewScrollbarPosition@names", "Auto;InsideLeft;InsideRight");
            a.item("HorizontalViewScrollbarFill", "Brush");
            a.item("HorizontalViewScrollbarOutline", "Brush");
            a.item("HorizontalViewScrollbarStrokeThickness", "Number:double");
            a.item("HorizontalViewScrollbarMaxOpacity", "Number:double");
            a.item("HorizontalViewScrollbarCornerRadius", "Number:double");
            a.item("HorizontalViewScrollbarHeight", "Number:double");
            a.item("HorizontalViewScrollbarInset", "Number:double");
            a.item("HorizontalViewScrollbarTrackStartInset", "Number:double");
            a.item("HorizontalViewScrollbarTrackEndInset", "Number:double");
            a.item("HorizontalViewScrollbarShouldAddAutoTrackInsets", "Boolean");
            a.item("VerticalViewScrollbarFill", "Brush");
            a.item("VerticalViewScrollbarOutline", "Brush");
            a.item("VerticalViewScrollbarStrokeThickness", "Number:double");
            a.item("VerticalViewScrollbarMaxOpacity", "Number:double");
            a.item("VerticalViewScrollbarCornerRadius", "Number:double");
            a.item("VerticalViewScrollbarWidth", "Number:double");
            a.item("VerticalViewScrollbarInset", "Number:double");
            a.item("VerticalViewScrollbarTrackStartInset", "Number:double");
            a.item("VerticalViewScrollbarTrackEndInset", "Number:double");
            a.item("VerticalViewScrollbarShouldAddAutoTrackInsets", "Boolean");
            a.item("HighlightingBehavior", "ExportedType:string:SeriesHighlightingBehavior");
            a.item("HighlightingBehavior@names", "Auto;DirectlyOver;NearestItems;NearestItemsRetainMainShapes;NearestItemsAndSeries");
            a.item("LegendHighlightingMode", "ExportedType:string:LegendHighlightingMode");
            a.item("LegendHighlightingMode@names", "Auto;None;MatchSeries");
            a.item("LegendItemBadgeShape", "ExportedType:string:LegendItemBadgeShape");
            a.item("LegendItemBadgeShape@names", "Automatic;Circle;Line;Square;Marker;Bar;Column;Hidden");
            a.item("LegendItemBadgeMode", "ExportedType:string:LegendItemBadgeMode");
            a.item("LegendItemBadgeMode@names", "Simplified;MatchSeries");
            a.item("EffectiveViewport", "Rect");
            a.item("WindowRect", "Rect");
            a.item("UseTiledZooming", "Boolean");
            a.item("PreferHigherResolutionTiles", "Boolean");
            a.item("ZoomTileCacheSize", "Number:int");
            a.item("HighlightingTransitionDuration", "Number:int");
            a.item("ScrollbarsAnimationDuration", "Number:int");
            a.item("IsPagePanningAllowed", "Boolean");
            a.item("ContentHitTestMode", "ExportedType:string:ChartHitTestMode");
            a.item("ContentHitTestMode@names", "Auto;Computational;ColorEncoded;Mixed;MixedFavoringComputational");
            a.item("ActualContentHitTestMode", "ExportedType:string:ChartHitTestMode");
            a.item("ActualContentHitTestMode@names", "Auto;Computational;ColorEncoded;Mixed;MixedFavoringComputational");
            a.item("IsSurfaceInteractionDisabled", "Boolean");
            a.item("WindowResponse", "ExportedType:string:WindowResponse");
            a.item("WindowResponse@names", "Deferred;Immediate");
            a.item("WindowRectMinWidth", "Number:double");
            a.item("WindowRectMinHeight", "Number:double");
            a.item("ActualWindowRectMinWidth", "Number:double");
            a.item("ActualWindowRectMinHeight", "Number:double");
            a.item("WindowSizeMinWidth", "Number:double");
            a.item("WindowSizeMinHeight", "Number:double");
            a.item("SyncChannel", "String");
            a.item("CrosshairVisibility", "ExportedType:string:Visibility");
            a.item("CrosshairVisibility@names", "Visible;Collapsed");
            a.item("HorizontalCrosshairBrush", "Brush");
            a.item("VerticalCrosshairBrush", "Brush");
            a.item("ZoomCoercionMode", "ExportedType:string:ZoomCoercionMode");
            a.item("ZoomCoercionMode@names", "Auto;AxisConstrained;Unconstrained");
            a.item("PlotAreaBackground", "Brush");
            a.item("DefaultInteraction", "ExportedType:string:InteractionState");
            a.item("DefaultInteraction@names", "None;DragZoom;DragPan;Auto");
            a.item("RightButtonDefaultInteraction", "ExportedType:string:InteractionState");
            a.item("RightButtonDefaultInteraction@names", "None;DragZoom;DragPan;Auto");
            a.item("DragModifier", "ExportedType:string:ModifierKeys");
            a.item("DragModifier@names", "None;Alt;Control;Shift;Windows;Apple");
            a.item("PanModifier", "ExportedType:string:ModifierKeys");
            a.item("PanModifier@names", "None;Alt;Control;Shift;Windows;Apple");
            a.item("PreviewRect", "Rect");
            a.item("WindowPositionHorizontal", "Number:double");
            a.item("WindowPositionVertical", "Number:double");
            a.item("ChartTitle", "(w:Title,xam:Title,wf:Title)String");
            a.item("TitleHorizontalAlignment", "ExportedType:string:HorizontalAlignment");
            a.item("TitleHorizontalAlignment@names", "Left;Center;Right;Stretch");
            a.item("SubtitleHorizontalAlignment", "ExportedType:string:HorizontalAlignment");
            a.item("SubtitleHorizontalAlignment@names", "Left;Center;Right;Stretch");
            a.item("TitleTextStyle", "String");
            a.item("SubtitleTextStyle", "String");
            a.item("TitleTextColor", "Brush");
            a.item("SubtitleTextColor", "Brush");
            a.item("TitleTopMargin", "Number:double");
            a.item("TitleLeftMargin", "Number:double");
            a.item("TitleRightMargin", "Number:double");
            a.item("TitleBottomMargin", "Number:double");
            a.item("SubtitleTopMargin", "Number:double");
            a.item("SubtitleLeftMargin", "Number:double");
            a.item("SubtitleRightMargin", "Number:double");
            a.item("SubtitleBottomMargin", "Number:double");
            a.item("Subtitle", "String");
            a.item("TopMargin", "Number:double");
            a.item("LeftMargin", "Number:double");
            a.item("RightMargin", "Number:double");
            a.item("BottomMargin", "Number:double");
            a.item("AutoMarginWidth", "Number:double");
            a.item("AutoMarginHeight", "Number:double");
            a.item("IsAntiAliasingEnabledDuringInteraction", "Boolean");
            a.item("PixelScalingRatio", "Number:double");
            a.item("InteractionPixelScalingRatio", "Number:double");
            a.item("ActualPixelScalingRatio", "Number:double");
            a.item("ActualInteractionPixelScalingRatio", "Number:double");
            a.item("ShouldPanOnMaximumZoom", "Boolean");
            a.item("ActualWindowRect", "Rect");
            a.item("ActualWindowPositionHorizontal", "Number:double");
            a.item("ActualWindowPositionVertical", "Number:double");
            a.item("PlotAreaMouseLeftButtonDownRef", "EventRef::plotAreaMouseLeftButtonDown");
            a.item("PlotAreaMouseLeftButtonUpRef", "EventRef::plotAreaMouseLeftButtonUp");
            a.item("PlotAreaClickedRef", "EventRef::plotAreaClicked");
            a.item("PlotAreaMouseEnterRef", "EventRef::plotAreaMouseEnter");
            a.item("PlotAreaMouseLeaveRef", "EventRef::plotAreaMouseLeave");
            a.item("PlotAreaMouseOverRef", "EventRef::plotAreaMouseOver");
            a.item("SeriesCursorMouseMoveRef", "EventRef::seriesCursorMouseMove");
            a.item("SeriesMouseLeftButtonDownRef", "EventRef::seriesMouseLeftButtonDown");
            a.item("SeriesMouseLeftButtonUpRef", "EventRef::seriesMouseLeftButtonUp");
            a.item("SeriesMouseMoveRef", "EventRef::seriesMouseMove");
            a.item("SeriesMouseEnterRef", "EventRef::seriesMouseEnter");
            a.item("SeriesMouseLeaveRef", "EventRef::seriesMouseLeave");
            a.item("ViewerManipulationStartingRef", "EventRef::viewerManipulationStarting");
            a.item("ViewerManipulationEndingRef", "EventRef::viewerManipulationEnding");
            a.item("WindowRectChangedRef", "EventRef::windowRectChanged");
            a.item("SizeChangedRef", "EventRef::sizeChanged");
            a.item("ActualWindowRectChangedRef", "EventRef::actualWindowRectChanged");
            a.item("GridAreaRectChangedRef", "EventRef::gridAreaRectChanged");
            a.item("RefreshCompletedRef", "EventRef::refreshCompleted");
            SeriesViewerDescriptionMetadata.e(a);
        }
        static d(a) {
            SeriesViewerDescriptionMetadata.b(a);
            a.t("SeriesViewer", SeriesViewerDescriptionMetadata.a);
        }
        static e(a) {
            a.item("Background", "(wf:BackColor/ColorTransform)Brush");
            a.item("TitleFontFamily", "(w:TitleFontFamily/FontFamilyTransform,xam:TitleFontFamily,wf:TitleFontFamily,web:TitleTextStyle/FontFamilyTransform)String");
            a.item("TitleFontSize", "(w:TitleFontSize,xam:TitleFontSize,wf:TitleFontSize,web:TitleTextStyle/FontSizeTransform)Number:double");
            a.item("TitleFontStyle", "(w:TitleFontStyle/FontStyleTransform,xam:TitleTextStyle/FontStyleTransform,wf:TitleTextStyle/FontStyleTransform,web:TitleTextStyle/FontStyleTransform)String");
            a.item("TitleFontWeight", "(w:TitleFontWeight/FontWeightTransform,xam:TitleTextStyle/FontWeightTransform,wf:TitleTextStyle/FontWeightTransform,web:TitleTextStyle/FontWeightTransform)String");
            a.item("SubtitleFontFamily", "(w:SubtitleFontFamily/FontFamilyTransform,xam:SubtitleFontFamily,wf:SubtitleFontFamily,web:SubtitleTextStyle/FontFamilyTransform)String");
            a.item("SubtitleFontSize", "(w:SubtitleFontSize,xam:SubtitleFontSize,wf:SubtitleFontSize,web:SubtitleTextStyle/FontSizeTransform)Number:double");
            a.item("SubtitleFontStyle", "(w:SubtitleFontStyle/FontStyleTransform,xam:SubtitleTextStyle/FontStyleTransform,wf:SubtitleTextStyle/FontStyleTransform,web:SubtitleTextStyle/FontStyleTransform)String");
            a.item("SubtitleFontWeight", "(w:SubtitleFontWeight/FontWeightTransform,xam:SubtitleTextStyle/FontWeightTransform,wf:SubtitleTextStyle/FontWeightTransform,web:SubtitleTextStyle/FontWeightTransform)String");
            a.item("SynchronizeHorizontally", "Boolean");
            a.item("SynchronizeVertically", "Boolean");
        }
    }
    SeriesViewerDescriptionMetadata.$t = /*@__PURE__*/ markType(SeriesViewerDescriptionMetadata, 'SeriesViewerDescriptionMetadata');
    SeriesViewerDescriptionMetadata.a = null;
    return SeriesViewerDescriptionMetadata;
})();
