// igniteui-react-core/esm2015/lib/PropertyEditorDataSourceDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let PropertyEditorDataSourceDescription = /*@__PURE__*/ (() => {
    class PropertyEditorDataSourceDescription extends Description {
        constructor() {
            super();
            this.g = null;
            this.f = null;
        }
        get_type() {
            return "PropertyEditorDataSource";
        }
        get type() {
            return this.get_type();
        }
        get descriptionType() {
            return this.g;
        }
        set descriptionType(a) {
            this.g = a;
            this.e("DescriptionType");
        }
        get contextRef() {
            return this.f;
        }
        set contextRef(a) {
            this.f = a;
            this.e("ContextRef");
        }
    }
    PropertyEditorDataSourceDescription.$t = /*@__PURE__*/ markType(PropertyEditorDataSourceDescription, 'PropertyEditorDataSourceDescription', Description.$);
    return PropertyEditorDataSourceDescription;
})();
