// igniteui-react-core/esm2015/lib/PropertyEditorPropertyDescriptionTargetPropertyUpdatingEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let PropertyEditorPropertyDescriptionTargetPropertyUpdatingEventArgsDescription = /*@__PURE__*/ (() => {
    class PropertyEditorPropertyDescriptionTargetPropertyUpdatingEventArgsDescription extends Description {
        constructor() {
            super();
            this.g = null;
            this.h = null;
            this.f = null;
        }
        get_type() {
            return "PropertyEditorPropertyDescriptionTargetPropertyUpdatingEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get targetRef() {
            return this.g;
        }
        set targetRef(a) {
            this.g = a;
            this.e("TargetRef");
        }
        get valueRef() {
            return this.h;
        }
        set valueRef(a) {
            this.h = a;
            this.e("ValueRef");
        }
        get propertyPath() {
            return this.f;
        }
        set propertyPath(a) {
            this.f = a;
            this.e("PropertyPath");
        }
    }
    PropertyEditorPropertyDescriptionTargetPropertyUpdatingEventArgsDescription.$t = /*@__PURE__*/ markType(PropertyEditorPropertyDescriptionTargetPropertyUpdatingEventArgsDescription, 'PropertyEditorPropertyDescriptionTargetPropertyUpdatingEventArgsDescription', Description.$);
    return PropertyEditorPropertyDescriptionTargetPropertyUpdatingEventArgsDescription;
})();
