// igniteui-react-core/esm2015/lib/StackedFragmentSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let StackedFragmentSeriesDescription = /*@__PURE__*/ (() => {
    class StackedFragmentSeriesDescription extends Description {
        constructor() {
            super();
            this.b6 = null;
            this.b5 = null;
            this.bn = null;
            this.j = false;
            this.f = false;
            this.an = 0;
            this.ac = 0;
            this.cl = null;
            this.bz = null;
            this.m = false;
            this.i = false;
            this.ao = 0;
            this.ad = 0;
            this.ap = 0;
            this.ae = 0;
            this.b9 = null;
            this.bq = null;
            this.ca = null;
            this.br = null;
            this.cb = null;
            this.bs = null;
            this.b8 = null;
            this.b7 = null;
            this.bp = null;
            this.bo = null;
            this.cc = null;
            this.bt = null;
            this.cd = null;
            this.bu = null;
            this.ce = null;
            this.bv = null;
            this.cf = null;
            this.bw = null;
            this.cg = null;
            this.ak = 0;
            this.z = 0;
            this.ch = null;
            this.bx = null;
            this.ag = 0;
            this.v = 0;
            this.ai = 0;
            this.x = 0;
            this.ci = null;
            this.by = null;
            this.ah = 0;
            this.w = 0;
            this.al = 0;
            this.aa = 0;
            this.g = false;
            this.k = false;
            this.am = 0;
            this.ab = 0;
            this.aq = 0;
            this.af = 0;
            this.aj = 0;
            this.y = 0;
            this.bl = null;
            this.cm = null;
            this.bf = 0;
            this.bg = 0;
            this.cn = null;
            this.co = null;
            this.l = false;
            this.cp = null;
            this.h = false;
            this.be = 0;
            this.b2 = null;
            this.b3 = null;
            this.b1 = null;
            this.b0 = null;
            this.bd = 0;
            this.cs = null;
            this.cq = null;
            this.cr = null;
            this.ct = null;
            this.b4 = null;
            this.cj = null;
            this.ck = null;
        }
        get_type() {
            return "StackedFragmentSeries";
        }
        get type() {
            return this.get_type();
        }
        get dataSourceRef() {
            return this.b6;
        }
        set dataSourceRef(a) {
            this.b6 = a;
            this.e("DataSourceRef");
        }
        get brush() {
            return this.b5;
        }
        set brush(a) {
            this.b5 = a;
            this.e("Brush");
        }
        get actualBrush() {
            return this.bn;
        }
        set actualBrush(a) {
            this.bn = a;
            this.e("ActualBrush");
        }
        get isDropShadowEnabled() {
            return this.j;
        }
        set isDropShadowEnabled(a) {
            this.j = a;
            this.e("IsDropShadowEnabled");
        }
        get actualIsDropShadowEnabled() {
            return this.f;
        }
        set actualIsDropShadowEnabled(a) {
            this.f = a;
            this.e("ActualIsDropShadowEnabled");
        }
        get shadowBlur() {
            return this.an;
        }
        set shadowBlur(a) {
            this.an = a;
            this.e("ShadowBlur");
        }
        get actualShadowBlur() {
            return this.ac;
        }
        set actualShadowBlur(a) {
            this.ac = a;
            this.e("ActualShadowBlur");
        }
        get shadowColor() {
            return this.cl;
        }
        set shadowColor(a) {
            this.cl = a;
            this.e("ShadowColor");
        }
        get actualShadowColor() {
            return this.bz;
        }
        set actualShadowColor(a) {
            this.bz = a;
            this.e("ActualShadowColor");
        }
        get useSingleShadow() {
            return this.m;
        }
        set useSingleShadow(a) {
            this.m = a;
            this.e("UseSingleShadow");
        }
        get actualUseSingleShadow() {
            return this.i;
        }
        set actualUseSingleShadow(a) {
            this.i = a;
            this.e("ActualUseSingleShadow");
        }
        get shadowOffsetX() {
            return this.ao;
        }
        set shadowOffsetX(a) {
            this.ao = a;
            this.e("ShadowOffsetX");
        }
        get actualShadowOffsetX() {
            return this.ad;
        }
        set actualShadowOffsetX(a) {
            this.ad = a;
            this.e("ActualShadowOffsetX");
        }
        get shadowOffsetY() {
            return this.ap;
        }
        set shadowOffsetY(a) {
            this.ap = a;
            this.e("ShadowOffsetY");
        }
        get actualShadowOffsetY() {
            return this.ae;
        }
        set actualShadowOffsetY(a) {
            this.ae = a;
            this.e("ActualShadowOffsetY");
        }
        get legendItemBadgeTemplateRef() {
            return this.b9;
        }
        set legendItemBadgeTemplateRef(a) {
            this.b9 = a;
            this.e("LegendItemBadgeTemplateRef");
        }
        get actualLegendItemBadgeTemplateRef() {
            return this.bq;
        }
        set actualLegendItemBadgeTemplateRef(a) {
            this.bq = a;
            this.e("ActualLegendItemBadgeTemplateRef");
        }
        get legendItemTemplateRef() {
            return this.ca;
        }
        set legendItemTemplateRef(a) {
            this.ca = a;
            this.e("LegendItemTemplateRef");
        }
        get actualLegendItemTemplateRef() {
            return this.br;
        }
        set actualLegendItemTemplateRef(a) {
            this.br = a;
            this.e("ActualLegendItemTemplateRef");
        }
        get legendItemVisibility() {
            return this.cb;
        }
        set legendItemVisibility(a) {
            this.cb = a;
            this.e("LegendItemVisibility");
        }
        get actualLegendItemVisibility() {
            return this.bs;
        }
        set actualLegendItemVisibility(a) {
            this.bs = a;
            this.e("ActualLegendItemVisibility");
        }
        get legendItemBadgeShape() {
            return this.b8;
        }
        set legendItemBadgeShape(a) {
            this.b8 = a;
            this.e("LegendItemBadgeShape");
        }
        get legendItemBadgeMode() {
            return this.b7;
        }
        set legendItemBadgeMode(a) {
            this.b7 = a;
            this.e("LegendItemBadgeMode");
        }
        get actualLegendItemBadgeShape() {
            return this.bp;
        }
        set actualLegendItemBadgeShape(a) {
            this.bp = a;
            this.e("ActualLegendItemBadgeShape");
        }
        get actualLegendItemBadgeMode() {
            return this.bo;
        }
        set actualLegendItemBadgeMode(a) {
            this.bo = a;
            this.e("ActualLegendItemBadgeMode");
        }
        get markerBrush() {
            return this.cc;
        }
        set markerBrush(a) {
            this.cc = a;
            this.e("MarkerBrush");
        }
        get actualMarkerBrush() {
            return this.bt;
        }
        set actualMarkerBrush(a) {
            this.bt = a;
            this.e("ActualMarkerBrush");
        }
        get markerOutline() {
            return this.cd;
        }
        set markerOutline(a) {
            this.cd = a;
            this.e("MarkerOutline");
        }
        get actualMarkerOutline() {
            return this.bu;
        }
        set actualMarkerOutline(a) {
            this.bu = a;
            this.e("ActualMarkerOutline");
        }
        get markerTemplateRef() {
            return this.ce;
        }
        set markerTemplateRef(a) {
            this.ce = a;
            this.e("MarkerTemplateRef");
        }
        get actualMarkerTemplateRef() {
            return this.bv;
        }
        set actualMarkerTemplateRef(a) {
            this.bv = a;
            this.e("ActualMarkerTemplateRef");
        }
        get markerType() {
            return this.cf;
        }
        set markerType(a) {
            this.cf = a;
            this.e("MarkerType");
        }
        get actualMarkerType() {
            return this.bw;
        }
        set actualMarkerType(a) {
            this.bw = a;
            this.e("ActualMarkerType");
        }
        get name() {
            return this.cg;
        }
        set name(a) {
            this.cg = a;
            this.e("Name");
        }
        get opacity() {
            return this.ak;
        }
        set opacity(a) {
            this.ak = a;
            this.e("Opacity");
        }
        get actualOpacity() {
            return this.z;
        }
        set actualOpacity(a) {
            this.z = a;
            this.e("ActualOpacity");
        }
        get outline() {
            return this.ch;
        }
        set outline(a) {
            this.ch = a;
            this.e("Outline");
        }
        get actualOutline() {
            return this.bx;
        }
        set actualOutline(a) {
            this.bx = a;
            this.e("ActualOutline");
        }
        get areaFillOpacity() {
            return this.ag;
        }
        set areaFillOpacity(a) {
            this.ag = a;
            this.e("AreaFillOpacity");
        }
        get actualAreaFillOpacity() {
            return this.v;
        }
        set actualAreaFillOpacity(a) {
            this.v = a;
            this.e("ActualAreaFillOpacity");
        }
        get markerFillOpacity() {
            return this.ai;
        }
        set markerFillOpacity(a) {
            this.ai = a;
            this.e("MarkerFillOpacity");
        }
        get actualMarkerFillOpacity() {
            return this.x;
        }
        set actualMarkerFillOpacity(a) {
            this.x = a;
            this.e("ActualMarkerFillOpacity");
        }
        get outlineMode() {
            return this.ci;
        }
        set outlineMode(a) {
            this.ci = a;
            this.e("OutlineMode");
        }
        get actualOutlineMode() {
            return this.by;
        }
        set actualOutlineMode(a) {
            this.by = a;
            this.e("ActualOutlineMode");
        }
        get highlightingFadeOpacity() {
            return this.ah;
        }
        set highlightingFadeOpacity(a) {
            this.ah = a;
            this.e("HighlightingFadeOpacity");
        }
        get actualHighlightingFadeOpacity() {
            return this.w;
        }
        set actualHighlightingFadeOpacity(a) {
            this.w = a;
            this.e("ActualHighlightingFadeOpacity");
        }
        get radiusX() {
            return this.al;
        }
        set radiusX(a) {
            this.al = a;
            this.e("RadiusX");
        }
        get actualRadiusX() {
            return this.aa;
        }
        set actualRadiusX(a) {
            this.aa = a;
            this.e("ActualRadiusX");
        }
        get actualIsSplineShapePartOfRange() {
            return this.g;
        }
        set actualIsSplineShapePartOfRange(a) {
            this.g = a;
            this.e("ActualIsSplineShapePartOfRange");
        }
        get isSplineShapePartOfRange() {
            return this.k;
        }
        set isSplineShapePartOfRange(a) {
            this.k = a;
            this.e("IsSplineShapePartOfRange");
        }
        get radiusY() {
            return this.am;
        }
        set radiusY(a) {
            this.am = a;
            this.e("RadiusY");
        }
        get actualRadiusY() {
            return this.ab;
        }
        set actualRadiusY(a) {
            this.ab = a;
            this.e("ActualRadiusY");
        }
        get thickness() {
            return this.aq;
        }
        set thickness(a) {
            this.aq = a;
            this.e("Thickness");
        }
        get actualThickness() {
            return this.af;
        }
        set actualThickness(a) {
            this.af = a;
            this.e("ActualThickness");
        }
        get markerThickness() {
            return this.aj;
        }
        set markerThickness(a) {
            this.aj = a;
            this.e("MarkerThickness");
        }
        get actualMarkerThickness() {
            return this.y;
        }
        set actualMarkerThickness(a) {
            this.y = a;
            this.e("ActualMarkerThickness");
        }
        get title() {
            return this.bl;
        }
        set title(a) {
            this.bl = a;
            this.e("Title");
        }
        get transitionEasingFunctionRef() {
            return this.cm;
        }
        set transitionEasingFunctionRef(a) {
            this.cm = a;
            this.e("TransitionEasingFunctionRef");
        }
        get transitionDuration() {
            return this.bf;
        }
        set transitionDuration(a) {
            this.bf = a;
            this.e("TransitionDuration");
        }
        get transitionInDuration() {
            return this.bg;
        }
        set transitionInDuration(a) {
            this.bg = a;
            this.e("TransitionInDuration");
        }
        get transitionInEasingFunctionRef() {
            return this.cn;
        }
        set transitionInEasingFunctionRef(a) {
            this.cn = a;
            this.e("TransitionInEasingFunctionRef");
        }
        get transitionInMode() {
            return this.co;
        }
        set transitionInMode(a) {
            this.co = a;
            this.e("TransitionInMode");
        }
        get isTransitionInEnabled() {
            return this.l;
        }
        set isTransitionInEnabled(a) {
            this.l = a;
            this.e("IsTransitionInEnabled");
        }
        get transitionInSpeedType() {
            return this.cp;
        }
        set transitionInSpeedType(a) {
            this.cp = a;
            this.e("TransitionInSpeedType");
        }
        get actualIsTransitionInEnabled() {
            return this.h;
        }
        set actualIsTransitionInEnabled(a) {
            this.h = a;
            this.e("ActualIsTransitionInEnabled");
        }
        get actualTransitionInDuration() {
            return this.be;
        }
        set actualTransitionInDuration(a) {
            this.be = a;
            this.e("ActualTransitionInDuration");
        }
        get actualTransitionInMode() {
            return this.b2;
        }
        set actualTransitionInMode(a) {
            this.b2 = a;
            this.e("ActualTransitionInMode");
        }
        get actualTransitionInSpeedType() {
            return this.b3;
        }
        set actualTransitionInSpeedType(a) {
            this.b3 = a;
            this.e("ActualTransitionInSpeedType");
        }
        get actualTransitionInEasingFunctionRef() {
            return this.b1;
        }
        set actualTransitionInEasingFunctionRef(a) {
            this.b1 = a;
            this.e("ActualTransitionInEasingFunctionRef");
        }
        get actualTransitionEasingFunctionRef() {
            return this.b0;
        }
        set actualTransitionEasingFunctionRef(a) {
            this.b0 = a;
            this.e("ActualTransitionEasingFunctionRef");
        }
        get actualTransitionDuration() {
            return this.bd;
        }
        set actualTransitionDuration(a) {
            this.bd = a;
            this.e("ActualTransitionDuration");
        }
        get valueMemberPath() {
            return this.cs;
        }
        set valueMemberPath(a) {
            this.cs = a;
            this.e("ValueMemberPath");
        }
        get valueMemberAsLegendLabel() {
            return this.cq;
        }
        set valueMemberAsLegendLabel(a) {
            this.cq = a;
            this.e("ValueMemberAsLegendLabel");
        }
        get valueMemberAsLegendUnit() {
            return this.cr;
        }
        set valueMemberAsLegendUnit(a) {
            this.cr = a;
            this.e("ValueMemberAsLegendUnit");
        }
        get visibility() {
            return this.ct;
        }
        set visibility(a) {
            this.ct = a;
            this.e("Visibility");
        }
        get actualVisibility() {
            return this.b4;
        }
        set actualVisibility(a) {
            this.b4 = a;
            this.e("ActualVisibility");
        }
        get parentOrLocalBrush() {
            return this.cj;
        }
        set parentOrLocalBrush(a) {
            this.cj = a;
            this.e("ParentOrLocalBrush");
        }
        get propertyUpdatedRef() {
            return this.ck;
        }
        set propertyUpdatedRef(a) {
            this.ck = a;
            this.e("PropertyUpdatedRef");
        }
    }
    StackedFragmentSeriesDescription.$t = /*@__PURE__*/ markType(StackedFragmentSeriesDescription, 'StackedFragmentSeriesDescription', Description.$);
    return StackedFragmentSeriesDescription;
})();
