// igniteui-react-core/esm2015/lib/SummaryResultDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let SummaryResultDescription = /*@__PURE__*/ (() => {
    class SummaryResultDescription extends Description {
        constructor() {
            super();
            this.l = null;
            this.i = null;
            this.g = 0;
        }
        get_type() {
            return "SummaryResult";
        }
        get type() {
            return this.get_type();
        }
        get groupKeyRef() {
            return this.l;
        }
        set groupKeyRef(a) {
            this.l = a;
            this.e("GroupKeyRef");
        }
        get value() {
            return this.i;
        }
        set value(a) {
            this.i = a;
            this.e("Value");
        }
        get summaryIndex() {
            return this.g;
        }
        set summaryIndex(a) {
            this.g = a;
            this.e("SummaryIndex");
        }
    }
    SummaryResultDescription.$t = /*@__PURE__*/ markType(SummaryResultDescription, 'SummaryResultDescription', Description.$);
    SummaryResultDescription.__marshalByValue = true;
    SummaryResultDescription.__marshalByValueAlias = "DefaultSummaryResult";
    return SummaryResultDescription;
})();
