// igniteui-react-core/esm2015/lib/SummaryChangedEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let SummaryChangedEventArgsDescription = /*@__PURE__*/ (() => {
    class SummaryChangedEventArgsDescription extends Description {
        constructor() {
            super();
            this.h = null;
            this.f = false;
        }
        get_type() {
            return "SummaryChangedEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get iD() {
            return this.h;
        }
        set iD(a) {
            this.h = a;
            this.e("ID");
        }
        get isEnabled() {
            return this.f;
        }
        set isEnabled(a) {
            this.f = a;
            this.e("IsEnabled");
        }
    }
    SummaryChangedEventArgsDescription.$t = /*@__PURE__*/ markType(SummaryChangedEventArgsDescription, 'SummaryChangedEventArgsDescription', Description.$);
    return SummaryChangedEventArgsDescription;
})();
