// igniteui-react-core/esm2015/lib/ToolbarDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let ToolbarDescription = /*@__PURE__*/ (() => {
    class ToolbarDescription extends Description {
        constructor() {
            super();
            this.h = null;
            this.f = null;
            this.g = null;
            this.o = null;
            this.l = 0;
            this.p = null;
            this.q = null;
            this.n = null;
            this.s = null;
            this.t = null;
            this.r = null;
        }
        get_type() {
            return "Toolbar";
        }
        get type() {
            return this.get_type();
        }
        get autoGeneratedActions() {
            return this.h;
        }
        set autoGeneratedActions(a) {
            this.h = a;
            this.e("AutoGeneratedActions");
        }
        get actions() {
            return this.f;
        }
        set actions(a) {
            this.f = a;
            this.e("Actions");
        }
        get actualActions() {
            return this.g;
        }
        set actualActions(a) {
            this.g = a;
            this.e("ActualActions");
        }
        get cellFontFamily() {
            return this.o;
        }
        set cellFontFamily(a) {
            this.o = a;
            this.e("CellFontFamily");
        }
        get cellFontSize() {
            return this.l;
        }
        set cellFontSize(a) {
            this.l = a;
            this.e("CellFontSize");
        }
        get cellFontStyle() {
            return this.p;
        }
        set cellFontStyle(a) {
            this.p = a;
            this.e("CellFontStyle");
        }
        get cellFontWeight() {
            return this.q;
        }
        set cellFontWeight(a) {
            this.q = a;
            this.e("CellFontWeight");
        }
        get backgroundColor() {
            return this.n;
        }
        set backgroundColor(a) {
            this.n = a;
            this.e("BackgroundColor");
        }
        get targetRef() {
            return this.s;
        }
        set targetRef(a) {
            this.s = a;
            this.e("TargetRef");
        }
        get textColor() {
            return this.t;
        }
        set textColor(a) {
            this.t = a;
            this.e("TextColor");
        }
        get orientation() {
            return this.r;
        }
        set orientation(a) {
            this.r = a;
            this.e("Orientation");
        }
    }
    ToolbarDescription.$t = /*@__PURE__*/ markType(ToolbarDescription, 'ToolbarDescription', Description.$);
    return ToolbarDescription;
})();
