// igniteui-react-core/esm2015/lib/ToolActionType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "./type";
/**
 * Represents type for toolbar action
 */
export var ToolActionType = /*@__PURE__*/ (function (ToolActionType) {
    ToolActionType[ToolActionType["Unknown"] = 0] = "Unknown";
    ToolActionType[ToolActionType["GroupHeader"] = 1] = "GroupHeader";
    ToolActionType[ToolActionType["Checkbox"] = 2] = "Checkbox";
    ToolActionType[ToolActionType["IconButton"] = 3] = "IconButton";
    ToolActionType[ToolActionType["NumberInput"] = 4] = "NumberInput";
    ToolActionType[ToolActionType["Separator"] = 5] = "Separator";
    ToolActionType[ToolActionType["IconMenu"] = 6] = "IconMenu";
    return ToolActionType;
})({});
/**
 * @hidden
 */
export let ToolActionType_$type = /*@__PURE__*/ markEnum('ToolActionType', 'Unknown,0|GroupHeader,1|Checkbox,2|IconButton,3|NumberInput,4|Separator,5|IconMenu,6');
