// igniteui-react-core/esm2015/lib/ToolCommandDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let ToolCommandDescription = /*@__PURE__*/ (() => {
    class ToolCommandDescription extends Description {
        constructor() {
            super();
            this.j = null;
            this.f = null;
        }
        get_type() {
            return "ToolCommand";
        }
        get type() {
            return this.get_type();
        }
        get commandId() {
            return this.j;
        }
        set commandId(a) {
            this.j = a;
            this.e("CommandId");
        }
        get argumentsList() {
            return this.f;
        }
        set argumentsList(a) {
            this.f = a;
            this.e("ArgumentsList");
        }
    }
    ToolCommandDescription.$t = /*@__PURE__*/ markType(ToolCommandDescription, 'ToolCommandDescription', Description.$);
    ToolCommandDescription.__marshalByValue = true;
    ToolCommandDescription.__marshalByValueAlias = "ToolCommand";
    return ToolCommandDescription;
})();
