// igniteui-react-core/esm2015/lib/ToolActionPerformedEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let ToolActionPerformedEventArgsDescription = /*@__PURE__*/ (() => {
    class ToolActionPerformedEventArgsDescription extends Description {
        constructor() {
            super();
            this.j = false;
            this.h = null;
            this.f = null;
        }
        get_type() {
            return "ToolActionPerformedEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get isAggregate() {
            return this.j;
        }
        set isAggregate(a) {
            this.j = a;
            this.e("IsAggregate");
        }
        get detail() {
            return this.h;
        }
        set detail(a) {
            this.h = a;
            this.e("Detail");
        }
        get detailCollection() {
            return this.f;
        }
        set detailCollection(a) {
            this.f = a;
            this.e("DetailCollection");
        }
    }
    ToolActionPerformedEventArgsDescription.$t = /*@__PURE__*/ markType(ToolActionPerformedEventArgsDescription, 'ToolActionPerformedEventArgsDescription', Description.$);
    return ToolActionPerformedEventArgsDescription;
})();
