// igniteui-react-core/esm2015/lib/ToolActionInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "./type";
/**
 * @hidden
 */
export let ToolActionInfo = /*@__PURE__*/ (() => {
    class ToolActionInfo extends Base {
        constructor() {
            super();
            this._parentId = null;
            this._commandId = null;
            this._commandArgument = null;
            this._commandArgumentValue = null;
            this._paddingLeft = 0;
            this._paddingRight = 0;
            this._paddingBottom = 0;
            this._paddingTop = 0;
            this._name = null;
            this._label = null;
            this._iconName = null;
            this._iconCollectionName = null;
            this.iconCollectionName = "default";
        }
        get parentId() {
            return this._parentId;
        }
        set parentId(a) {
            this._parentId = a;
        }
        get commandId() {
            return this._commandId;
        }
        set commandId(a) {
            this._commandId = a;
        }
        get commandArgument() {
            return this._commandArgument;
        }
        set commandArgument(a) {
            this._commandArgument = a;
        }
        get commandArgumentValue() {
            return this._commandArgumentValue;
        }
        set commandArgumentValue(a) {
            this._commandArgumentValue = a;
        }
        get paddingLeft() {
            return this._paddingLeft;
        }
        set paddingLeft(a) {
            this._paddingLeft = a;
        }
        get paddingRight() {
            return this._paddingRight;
        }
        set paddingRight(a) {
            this._paddingRight = a;
        }
        get paddingBottom() {
            return this._paddingBottom;
        }
        set paddingBottom(a) {
            this._paddingBottom = a;
        }
        get paddingTop() {
            return this._paddingTop;
        }
        set paddingTop(a) {
            this._paddingTop = a;
        }
        get name() {
            return this._name;
        }
        set name(a) {
            this._name = a;
        }
        get label() {
            return this._label;
        }
        set label(a) {
            this._label = a;
        }
        get iconName() {
            return this._iconName;
        }
        set iconName(a) {
            this._iconName = a;
        }
        get iconCollectionName() {
            return this._iconCollectionName;
        }
        set iconCollectionName(a) {
            this._iconCollectionName = a;
        }
        get_a() {
            return 0;
        }
        get a() {
            return this.get_a();
        }
    }
    ToolActionInfo.$t = /*@__PURE__*/ markType(ToolActionInfo, 'ToolActionInfo');
    return ToolActionInfo;
})();
