// igniteui-react-core/esm2015/lib/ToolCommandArgumentDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let ToolCommandArgumentDescription = /*@__PURE__*/ (() => {
    class ToolCommandArgumentDescription extends Description {
        constructor() {
            super();
            this.j = null;
            this.g = null;
        }
        get_type() {
            return "ToolCommandArgument";
        }
        get type() {
            return this.get_type();
        }
        get argumentName() {
            return this.j;
        }
        set argumentName(a) {
            this.j = a;
            this.e("ArgumentName");
        }
        get value() {
            return this.g;
        }
        set value(a) {
            this.g = a;
            this.e("Value");
        }
    }
    ToolCommandArgumentDescription.$t = /*@__PURE__*/ markType(ToolCommandArgumentDescription, 'ToolCommandArgumentDescription', Description.$);
    ToolCommandArgumentDescription.__marshalByValue = true;
    ToolCommandArgumentDescription.__marshalByValueAlias = "ToolCommandArgument";
    return ToolCommandArgumentDescription;
})();
