// igniteui-react-core/esm2015/lib/WebActivePaneEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebActivePaneEventArgsDescription = /*@__PURE__*/ (() => {
    class WebActivePaneEventArgsDescription extends Description {
        constructor() {
            super();
            this.f = null;
            this.g = null;
        }
        get_type() {
            return "WebActivePaneEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get newPane() {
            return this.f;
        }
        set newPane(a) {
            this.f = a;
            this.e("NewPane");
        }
        get oldPane() {
            return this.g;
        }
        set oldPane(a) {
            this.g = a;
            this.e("OldPane");
        }
    }
    WebActivePaneEventArgsDescription.$t = /*@__PURE__*/ markType(WebActivePaneEventArgsDescription, 'WebActivePaneEventArgsDescription', Description.$);
    return WebActivePaneEventArgsDescription;
})();
