// igniteui-react-core/esm2015/lib/WebDockPaneActionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebPaneDragActionDescription } from "./WebPaneDragActionDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebDockPaneActionDescription = /*@__PURE__*/ (() => {
    class WebDockPaneActionDescription extends WebPaneDragActionDescription {
        constructor() {
            super();
            this.k = null;
            this.g = null;
            this.i = null;
        }
        get_type() {
            return "WebDockPaneAction";
        }
        get actionType() {
            return this.k;
        }
        set actionType(a) {
            this.k = a;
            this.e("ActionType");
        }
        get dockingIndicator() {
            return this.g;
        }
        set dockingIndicator(a) {
            this.g = a;
            this.e("DockingIndicator");
        }
        get targetPane() {
            return this.i;
        }
        set targetPane(a) {
            this.i = a;
            this.e("TargetPane");
        }
    }
    WebDockPaneActionDescription.$t = /*@__PURE__*/ markType(WebDockPaneActionDescription, 'WebDockPaneActionDescription', WebPaneDragActionDescription.$);
    return WebDockPaneActionDescription;
})();
