// igniteui-react-core/esm2015/lib/WebFormDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebFormDescription = /*@__PURE__*/ (() => {
    class WebFormDescription extends Description {
        constructor() {
            super();
            this.f = false;
            this.i = null;
            this.h = null;
        }
        get_type() {
            return "WebForm";
        }
        get type() {
            return this.get_type();
        }
        get novalidate() {
            return this.f;
        }
        set novalidate(a) {
            this.f = a;
            this.e("Novalidate");
        }
        get submitRef() {
            return this.i;
        }
        set submitRef(a) {
            this.i = a;
            this.e("SubmitRef");
        }
        get resetRef() {
            return this.h;
        }
        set resetRef(a) {
            this.h = a;
            this.e("ResetRef");
        }
    }
    WebFormDescription.$t = /*@__PURE__*/ markType(WebFormDescription, 'WebFormDescription', Description.$);
    return WebFormDescription;
})();
