// igniteui-react-core/esm2015/lib/WebFloatPaneActionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebPaneDragActionDescription } from "./WebPaneDragActionDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebFloatPaneActionDescription = /*@__PURE__*/ (() => {
    class WebFloatPaneActionDescription extends WebPaneDragActionDescription {
        constructor() {
            super();
            this.m = null;
            this.g = null;
            this.j = 0;
            this.i = 0;
        }
        get_type() {
            return "WebFloatPaneAction";
        }
        get actionType() {
            return this.m;
        }
        set actionType(a) {
            this.m = a;
            this.e("ActionType");
        }
        get location() {
            return this.g;
        }
        set location(a) {
            this.g = a;
            this.e("Location");
        }
        get width() {
            return this.j;
        }
        set width(a) {
            this.j = a;
            this.e("Width");
        }
        get height() {
            return this.i;
        }
        set height(a) {
            this.i = a;
            this.e("Height");
        }
    }
    WebFloatPaneActionDescription.$t = /*@__PURE__*/ markType(WebFloatPaneActionDescription, 'WebFloatPaneActionDescription', WebPaneDragActionDescription.$);
    return WebFloatPaneActionDescription;
})();
