// igniteui-react-core/esm2015/lib/WebFloatingPaneResizeMoveEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebFloatingPaneResizeEventArgsDescription } from "./WebFloatingPaneResizeEventArgsDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebFloatingPaneResizeMoveEventArgsDescription = /*@__PURE__*/ (() => {
    class WebFloatingPaneResizeMoveEventArgsDescription extends WebFloatingPaneResizeEventArgsDescription {
        constructor() {
            super();
            this.r = 0;
            this.p = 0;
            this.q = 0;
            this.o = 0;
            this.l = null;
            this.k = null;
        }
        get_type() {
            return "WebFloatingPaneResizeMoveEventArgs";
        }
        get oldWidth() {
            return this.r;
        }
        set oldWidth(a) {
            this.r = a;
            this.e("OldWidth");
        }
        get newWidth() {
            return this.p;
        }
        set newWidth(a) {
            this.p = a;
            this.e("NewWidth");
        }
        get oldHeight() {
            return this.q;
        }
        set oldHeight(a) {
            this.q = a;
            this.e("OldHeight");
        }
        get newHeight() {
            return this.o;
        }
        set newHeight(a) {
            this.o = a;
            this.e("NewHeight");
        }
        get oldLocation() {
            return this.l;
        }
        set oldLocation(a) {
            this.l = a;
            this.e("OldLocation");
        }
        get newLocation() {
            return this.k;
        }
        set newLocation(a) {
            this.k = a;
            this.e("NewLocation");
        }
    }
    WebFloatingPaneResizeMoveEventArgsDescription.$t = /*@__PURE__*/ markType(WebFloatingPaneResizeMoveEventArgsDescription, 'WebFloatingPaneResizeMoveEventArgsDescription', WebFloatingPaneResizeEventArgsDescription.$);
    return WebFloatingPaneResizeMoveEventArgsDescription;
})();
