// igniteui-react-core/esm2015/lib/WebFloatingPaneResizeEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebFloatingPaneResizeEventArgsDescription = /*@__PURE__*/ (() => {
    class WebFloatingPaneResizeEventArgsDescription extends Description {
        constructor() {
            super();
            this.f = null;
            this.h = null;
        }
        get_type() {
            return "WebFloatingPaneResizeEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get sourcePane() {
            return this.f;
        }
        set sourcePane(a) {
            this.f = a;
            this.e("SourcePane");
        }
        get resizerLocation() {
            return this.h;
        }
        set resizerLocation(a) {
            this.h = a;
            this.e("ResizerLocation");
        }
    }
    WebFloatingPaneResizeEventArgsDescription.$t = /*@__PURE__*/ markType(WebFloatingPaneResizeEventArgsDescription, 'WebFloatingPaneResizeEventArgsDescription', Description.$);
    return WebFloatingPaneResizeEventArgsDescription;
})();
