// igniteui-react-core/esm2015/lib/WebMoveTabActionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebPaneDragActionDescription } from "./WebPaneDragActionDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebMoveTabActionDescription = /*@__PURE__*/ (() => {
    class WebMoveTabActionDescription extends WebPaneDragActionDescription {
        constructor() {
            super();
            this.k = null;
            this.h = 0;
            this.g = 0;
        }
        get_type() {
            return "WebMoveTabAction";
        }
        get actionType() {
            return this.k;
        }
        set actionType(a) {
            this.k = a;
            this.e("ActionType");
        }
        get oldIndex() {
            return this.h;
        }
        set oldIndex(a) {
            this.h = a;
            this.e("OldIndex");
        }
        get newIndex() {
            return this.g;
        }
        set newIndex(a) {
            this.g = a;
            this.e("NewIndex");
        }
    }
    WebMoveTabActionDescription.$t = /*@__PURE__*/ markType(WebMoveTabActionDescription, 'WebMoveTabActionDescription', WebPaneDragActionDescription.$);
    return WebMoveTabActionDescription;
})();
