// igniteui-react-core/esm2015/lib/WebMoveFloatingPaneActionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebPaneDragActionDescription } from "./WebPaneDragActionDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebMoveFloatingPaneActionDescription = /*@__PURE__*/ (() => {
    class WebMoveFloatingPaneActionDescription extends WebPaneDragActionDescription {
        constructor() {
            super();
            this.k = null;
            this.h = null;
            this.g = null;
        }
        get_type() {
            return "WebMoveFloatingPaneAction";
        }
        get actionType() {
            return this.k;
        }
        set actionType(a) {
            this.k = a;
            this.e("ActionType");
        }
        get oldLocation() {
            return this.h;
        }
        set oldLocation(a) {
            this.h = a;
            this.e("OldLocation");
        }
        get newLocation() {
            return this.g;
        }
        set newLocation(a) {
            this.g = a;
            this.e("NewLocation");
        }
    }
    WebMoveFloatingPaneActionDescription.$t = /*@__PURE__*/ markType(WebMoveFloatingPaneActionDescription, 'WebMoveFloatingPaneActionDescription', WebPaneDragActionDescription.$);
    return WebMoveFloatingPaneActionDescription;
})();
