// igniteui-react-core/esm2015/lib/WebMaskInputBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { WebInputBaseDescription } from "./WebInputBaseDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebMaskInputBaseDescription = /*@__PURE__*/ (() => {
    class WebMaskInputBaseDescription extends WebInputBaseDescription {
        constructor() {
            super();
            this.ae = null;
            this.ac = false;
        }
        get_type() {
            return "WebMaskInputBase";
        }
        get prompt() {
            return this.ae;
        }
        set prompt(a) {
            this.ae = a;
            this.e("Prompt");
        }
        get invalid() {
            return this.ac;
        }
        set invalid(a) {
            this.ac = a;
            this.e("Invalid");
        }
    }
    WebMaskInputBaseDescription.$t = /*@__PURE__*/ markType(WebMaskInputBaseDescription, 'WebMaskInputBaseDescription', WebInputBaseDescription.$);
    return WebMaskInputBaseDescription;
})();
