// igniteui-react-core/esm2015/lib/WebPanePinnedEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPanePinnedEventArgsDescription = /*@__PURE__*/ (() => {
    class WebPanePinnedEventArgsDescription extends Description {
        constructor() {
            super();
            this.h = null;
            this.f = null;
            this.j = false;
            this.l = null;
        }
        get_type() {
            return "WebPanePinnedEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get sourcePane() {
            return this.h;
        }
        set sourcePane(a) {
            this.h = a;
            this.e("SourcePane");
        }
        get panes() {
            return this.f;
        }
        set panes(a) {
            this.f = a;
            this.e("Panes");
        }
        get newValue() {
            return this.j;
        }
        set newValue(a) {
            this.j = a;
            this.e("NewValue");
        }
        get location() {
            return this.l;
        }
        set location(a) {
            this.l = a;
            this.e("Location");
        }
    }
    WebPanePinnedEventArgsDescription.$t = /*@__PURE__*/ markType(WebPanePinnedEventArgsDescription, 'WebPanePinnedEventArgsDescription', Description.$);
    return WebPanePinnedEventArgsDescription;
})();
