// igniteui-react-core/esm2015/lib/WebPaneDragOverEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPaneDragOverEventArgsDescription = /*@__PURE__*/ (() => {
    class WebPaneDragOverEventArgsDescription extends Description {
        constructor() {
            super();
            this.h = null;
            this.f = null;
            this.j = null;
            this.l = false;
        }
        get_type() {
            return "WebPaneDragOverEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get sourcePane() {
            return this.h;
        }
        set sourcePane(a) {
            this.h = a;
            this.e("SourcePane");
        }
        get panes() {
            return this.f;
        }
        set panes(a) {
            this.f = a;
            this.e("Panes");
        }
        get action() {
            return this.j;
        }
        set action(a) {
            this.j = a;
            this.e("Action");
        }
        get isValid() {
            return this.l;
        }
        set isValid(a) {
            this.l = a;
            this.e("IsValid");
        }
    }
    WebPaneDragOverEventArgsDescription.$t = /*@__PURE__*/ markType(WebPaneDragOverEventArgsDescription, 'WebPaneDragOverEventArgsDescription', Description.$);
    return WebPaneDragOverEventArgsDescription;
})();
