// igniteui-react-core/esm2015/lib/WebPaneHeaderConnectionEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebPaneHeaderConnectionEventArgsDescription = /*@__PURE__*/ (() => {
    class WebPaneHeaderConnectionEventArgsDescription extends Description {
        constructor() {
            super();
            this.f = null;
            this.h = null;
        }
        get_type() {
            return "WebPaneHeaderConnectionEventArgs";
        }
        get type() {
            return this.get_type();
        }
        get pane() {
            return this.f;
        }
        set pane(a) {
            this.f = a;
            this.e("Pane");
        }
        get element() {
            return this.h;
        }
        set element(a) {
            this.h = a;
            this.e("Element");
        }
    }
    WebPaneHeaderConnectionEventArgsDescription.$t = /*@__PURE__*/ markType(WebPaneHeaderConnectionEventArgsDescription, 'WebPaneHeaderConnectionEventArgsDescription', Description.$);
    return WebPaneHeaderConnectionEventArgsDescription;
})();
