// igniteui-react-core/esm2015/lib/WebRangeSliderValueDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebRangeSliderValueDescription = /*@__PURE__*/ (() => {
    class WebRangeSliderValueDescription extends Description {
        constructor() {
            super();
            this.f = 0;
            this.g = 0;
        }
        get_type() {
            return "WebRangeSliderValue";
        }
        get type() {
            return this.get_type();
        }
        get lower() {
            return this.f;
        }
        set lower(a) {
            this.f = a;
            this.e("Lower");
        }
        get upper() {
            return this.g;
        }
        set upper(a) {
            this.g = a;
            this.e("Upper");
        }
    }
    WebRangeSliderValueDescription.$t = /*@__PURE__*/ markType(WebRangeSliderValueDescription, 'WebRangeSliderValueDescription', Description.$);
    return WebRangeSliderValueDescription;
})();
