// igniteui-react-core/esm2015/lib/WebToastDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export let WebToastDescription = /*@__PURE__*/ (() => {
    class WebToastDescription extends Description {
        constructor() {
            super();
            this.g = false;
            this.j = 0;
            this.f = false;
        }
        get_type() {
            return "WebToast";
        }
        get type() {
            return this.get_type();
        }
        get open() {
            return this.g;
        }
        set open(a) {
            this.g = a;
            this.e("Open");
        }
        get displayTime() {
            return this.j;
        }
        set displayTime(a) {
            this.j = a;
            this.e("DisplayTime");
        }
        get keepOpen() {
            return this.f;
        }
        set keepOpen(a) {
            this.f = a;
            this.e("KeepOpen");
        }
    }
    WebToastDescription.$t = /*@__PURE__*/ markType(WebToastDescription, 'WebToastDescription', Description.$);
    return WebToastDescription;
})();
