// igniteui-react-core/esm2015/lib/WrapperExpressionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { FilterExpressionDescription } from "./FilterExpressionDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export let WrapperExpressionDescription = /*@__PURE__*/ (() => {
    class WrapperExpressionDescription extends FilterExpressionDescription {
        constructor() {
            super();
            this.j = null;
            this.i = null;
            this.o = null;
        }
        get_type() {
            return "WrapperExpression";
        }
        get innerExpression() {
            return this.j;
        }
        set innerExpression(a) {
            this.j = a;
            this.e("InnerExpression");
        }
        get expression() {
            return this.i;
        }
        set expression(a) {
            this.i = a;
            this.e("Expression");
        }
        get propertyName() {
            return this.o;
        }
        set propertyName(a) {
            this.o = a;
            this.e("PropertyName");
        }
    }
    WrapperExpressionDescription.$t = /*@__PURE__*/ markType(WrapperExpressionDescription, 'WrapperExpressionDescription', FilterExpressionDescription.$);
    WrapperExpressionDescription.__marshalByValue1 = true;
    WrapperExpressionDescription.__marshalByValueAlias1 = "WrapperExpression";
    return WrapperExpressionDescription;
})();
