// igniteui-react-core/esm5/lib/ArcGISOnlineMapImageryDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { GeographicMapImageryDescription } from "./GeographicMapImageryDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var ArcGISOnlineMapImageryDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ArcGISOnlineMapImageryDescription, _super);
    function ArcGISOnlineMapImageryDescription() {
        var _this = _super.call(this) || this;
        _this.ae = null;
        _this.ai = null;
        _this.af = null;
        _this.aj = null;
        _this.aa = false;
        _this.ac = 0;
        _this.ah = null;
        _this.ag = null;
        return _this;
    }
    ArcGISOnlineMapImageryDescription.prototype.get_type = function () {
        return "ArcGISOnlineMapImagery";
    };
    Object.defineProperty(ArcGISOnlineMapImageryDescription.prototype, "mapServerUri", {
        get: function () {
            return this.ae;
        },
        set: function (a) {
            this.ae = a;
            this.e("MapServerUri");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ArcGISOnlineMapImageryDescription.prototype, "userName", {
        get: function () {
            return this.ai;
        },
        set: function (a) {
            this.ai = a;
            this.e("UserName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ArcGISOnlineMapImageryDescription.prototype, "password", {
        get: function () {
            return this.af;
        },
        set: function (a) {
            this.af = a;
            this.e("Password");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ArcGISOnlineMapImageryDescription.prototype, "userToken", {
        get: function () {
            return this.aj;
        },
        set: function (a) {
            this.aj = a;
            this.e("UserToken");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ArcGISOnlineMapImageryDescription.prototype, "isMapPublic", {
        get: function () {
            return this.aa;
        },
        set: function (a) {
            this.aa = a;
            this.e("IsMapPublic");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ArcGISOnlineMapImageryDescription.prototype, "defaultTokenTimeout", {
        get: function () {
            return this.ac;
        },
        set: function (a) {
            this.ac = a;
            this.e("DefaultTokenTimeout");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ArcGISOnlineMapImageryDescription.prototype, "tokenGenerationEndPoint", {
        get: function () {
            return this.ah;
        },
        set: function (a) {
            this.ah = a;
            this.e("TokenGenerationEndPoint");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ArcGISOnlineMapImageryDescription.prototype, "refererUri", {
        get: function () {
            return this.ag;
        },
        set: function (a) {
            this.ag = a;
            this.e("RefererUri");
        },
        enumerable: false,
        configurable: true
    });
    ArcGISOnlineMapImageryDescription.$t = markType(ArcGISOnlineMapImageryDescription, 'ArcGISOnlineMapImageryDescription', GeographicMapImageryDescription.$);
    return ArcGISOnlineMapImageryDescription;
}(GeographicMapImageryDescription));
export { ArcGISOnlineMapImageryDescription };
