// igniteui-react-core/esm5/lib/BingMapsMapImageryDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { GeographicMapImageryDescription } from "./GeographicMapImageryDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var BingMapsMapImageryDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(BingMapsMapImageryDescription, _super);
    function BingMapsMapImageryDescription() {
        var _this = _super.call(this) || this;
        _this.af = false;
        _this.ae = false;
        _this.ao = null;
        _this.aj = null;
        _this.ab = null;
        _this.aa = null;
        _this.al = null;
        _this.ai = null;
        _this.am = null;
        _this.ak = null;
        _this.an = null;
        return _this;
    }
    BingMapsMapImageryDescription.prototype.get_type = function () {
        return "BingMapsMapImagery";
    };
    Object.defineProperty(BingMapsMapImageryDescription.prototype, "isInitialized", {
        get: function () {
            return this.af;
        },
        set: function (a) {
            this.af = a;
            this.e("IsInitialized");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(BingMapsMapImageryDescription.prototype, "isDeferredLoad", {
        get: function () {
            return this.ae;
        },
        set: function (a) {
            this.ae = a;
            this.e("IsDeferredLoad");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(BingMapsMapImageryDescription.prototype, "tilePath", {
        get: function () {
            return this.ao;
        },
        set: function (a) {
            this.ao = a;
            this.e("TilePath");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(BingMapsMapImageryDescription.prototype, "actualTilePath", {
        get: function () {
            return this.aj;
        },
        set: function (a) {
            this.aj = a;
            this.e("ActualTilePath");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(BingMapsMapImageryDescription.prototype, "subDomains", {
        get: function () {
            return this.ab;
        },
        set: function (a) {
            this.ab = a;
            this.e("SubDomains");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(BingMapsMapImageryDescription.prototype, "actualSubDomains", {
        get: function () {
            return this.aa;
        },
        set: function (a) {
            this.aa = a;
            this.e("ActualSubDomains");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(BingMapsMapImageryDescription.prototype, "bingImageryRestUri", {
        get: function () {
            return this.al;
        },
        set: function (a) {
            this.al = a;
            this.e("BingImageryRestUri");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(BingMapsMapImageryDescription.prototype, "actualBingImageryRestUri", {
        get: function () {
            return this.ai;
        },
        set: function (a) {
            this.ai = a;
            this.e("ActualBingImageryRestUri");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(BingMapsMapImageryDescription.prototype, "cultureName", {
        get: function () {
            return this.am;
        },
        set: function (a) {
            this.am = a;
            this.e("CultureName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(BingMapsMapImageryDescription.prototype, "apiKey", {
        get: function () {
            return this.ak;
        },
        set: function (a) {
            this.ak = a;
            this.e("ApiKey");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(BingMapsMapImageryDescription.prototype, "imageryStyle", {
        get: function () {
            return this.an;
        },
        set: function (a) {
            this.an = a;
            this.e("ImageryStyle");
        },
        enumerable: false,
        configurable: true
    });
    BingMapsMapImageryDescription.$t = markType(BingMapsMapImageryDescription, 'BingMapsMapImageryDescription', GeographicMapImageryDescription.$);
    return BingMapsMapImageryDescription;
}(GeographicMapImageryDescription));
export { BingMapsMapImageryDescription };
