// igniteui-react-core/esm5/lib/CalendarResourceStringsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var CalendarResourceStringsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CalendarResourceStringsDescription, _super);
    function CalendarResourceStringsDescription() {
        var _this = _super.call(this) || this;
        _this.q = null;
        _this.s = null;
        _this.o = null;
        _this.r = null;
        _this.p = null;
        _this.t = null;
        _this.h = null;
        _this.l = null;
        _this.i = null;
        _this.m = null;
        _this.j = null;
        _this.n = null;
        _this.k = null;
        _this.u = null;
        return _this;
    }
    CalendarResourceStringsDescription.prototype.get_type = function () {
        return "CalendarResourceStrings";
    };
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "selectMonth", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("SelectMonth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "selectYear", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("SelectYear");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "selectDate", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("SelectDate");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "selectRange", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("SelectRange");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "selectedDate", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("SelectedDate");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "startDate", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
            this.e("StartDate");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "endDate", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("EndDate");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "previousMonth", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("PreviousMonth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "nextMonth", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("NextMonth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "previousYear", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("PreviousYear");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "nextYear", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("NextYear");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "previousYears", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("PreviousYears");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "nextYears", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("NextYears");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarResourceStringsDescription.prototype, "weekLabel", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            this.u = a;
            this.e("WeekLabel");
        },
        enumerable: false,
        configurable: true
    });
    CalendarResourceStringsDescription.$t = markType(CalendarResourceStringsDescription, 'CalendarResourceStringsDescription', Description.$);
    CalendarResourceStringsDescription.__marshalByValue = true;
    CalendarResourceStringsDescription.__marshalByValueAlias = "CalendarResourceStrings";
    return CalendarResourceStringsDescription;
}(Description));
export { CalendarResourceStringsDescription };
