// igniteui-react-core/esm5/lib/CalendarDateDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var CalendarDateDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CalendarDateDescription, _super);
    function CalendarDateDescription() {
        var _this = _super.call(this) || this;
        _this.m = new Date();
        _this.g = false;
        _this.i = false;
        _this.h = false;
        return _this;
    }
    CalendarDateDescription.prototype.get_type = function () {
        return "CalendarDate";
    };
    Object.defineProperty(CalendarDateDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarDateDescription.prototype, "date", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("Date");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarDateDescription.prototype, "isCurrentMonth", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("IsCurrentMonth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarDateDescription.prototype, "isPrevMonth", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("IsPrevMonth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CalendarDateDescription.prototype, "isNextMonth", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("IsNextMonth");
        },
        enumerable: false,
        configurable: true
    });
    CalendarDateDescription.$t = markType(CalendarDateDescription, 'CalendarDateDescription', Description.$);
    CalendarDateDescription.__marshalByValue = true;
    CalendarDateDescription.__marshalByValueAlias = "CalendarDate";
    return CalendarDateDescription;
}(Description));
export { CalendarDateDescription };
