// igniteui-react-core/esm5/lib/CategoryXAxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { CategoryAxisBaseDescription } from "./CategoryAxisBaseDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var CategoryXAxisDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CategoryXAxisDescription, _super);
    function CategoryXAxisDescription() {
        var _this = _super.call(this) || this;
        _this.dw = 0;
        _this.dv = 0;
        _this.dt = 0;
        _this.dx = 0;
        _this.du = 0;
        _this.d3 = null;
        _this.d4 = null;
        return _this;
    }
    CategoryXAxisDescription.prototype.get_type = function () {
        return "CategoryXAxis";
    };
    Object.defineProperty(CategoryXAxisDescription.prototype, "maximumZoomCategories", {
        get: function () {
            return this.dw;
        },
        set: function (a) {
            this.dw = a;
            this.e("ZoomMaximumCategoryRange");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CategoryXAxisDescription.prototype, "interval", {
        get: function () {
            return this.dv;
        },
        set: function (a) {
            this.dv = a;
            this.e("Interval");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CategoryXAxisDescription.prototype, "actualInterval", {
        get: function () {
            return this.dt;
        },
        set: function (a) {
            this.dt = a;
            this.e("ActualInterval");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CategoryXAxisDescription.prototype, "minorInterval", {
        get: function () {
            return this.dx;
        },
        set: function (a) {
            this.dx = a;
            this.e("MinorInterval");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CategoryXAxisDescription.prototype, "actualMinorInterval", {
        get: function () {
            return this.du;
        },
        set: function (a) {
            this.du = a;
            this.e("ActualMinorInterval");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CategoryXAxisDescription.prototype, "actualIntervalChangeRef", {
        get: function () {
            return this.d3;
        },
        set: function (a) {
            this.d3 = a;
            this.e("ActualIntervalChangeRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CategoryXAxisDescription.prototype, "actualMinorIntervalChangeRef", {
        get: function () {
            return this.d4;
        },
        set: function (a) {
            this.d4 = a;
            this.e("ActualMinorIntervalChangeRef");
        },
        enumerable: false,
        configurable: true
    });
    CategoryXAxisDescription.$t = markType(CategoryXAxisDescription, 'CategoryXAxisDescription', CategoryAxisBaseDescription.$);
    return CategoryXAxisDescription;
}(CategoryAxisBaseDescription));
export { CategoryXAxisDescription };
