// igniteui-react-core/esm5/lib/ChartSummaryDescriptionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var ChartSummaryDescriptionDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ChartSummaryDescriptionDescription, _super);
    function ChartSummaryDescriptionDescription() {
        var _this = _super.call(this) || this;
        _this.j = null;
        _this.k = null;
        _this.h = null;
        _this.i = null;
        _this.l = null;
        return _this;
    }
    ChartSummaryDescriptionDescription.prototype.get_type = function () {
        return "ChartSummaryDescription";
    };
    Object.defineProperty(ChartSummaryDescriptionDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ChartSummaryDescriptionDescription.prototype, "field", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Field");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ChartSummaryDescriptionDescription.prototype, "operand", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("Operand");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ChartSummaryDescriptionDescription.prototype, "alias", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Alias");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ChartSummaryDescriptionDescription.prototype, "calculatorDisplayName", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("CalculatorDisplayName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ChartSummaryDescriptionDescription.prototype, "provideCalculatorRef", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("ProvideCalculatorRef");
        },
        enumerable: false,
        configurable: true
    });
    ChartSummaryDescriptionDescription.$t = markType(ChartSummaryDescriptionDescription, 'ChartSummaryDescriptionDescription', Description.$);
    ChartSummaryDescriptionDescription.__marshalByValue = true;
    ChartSummaryDescriptionDescription.__marshalByValueAlias = "ChartSummaryDescription";
    return ChartSummaryDescriptionDescription;
}(Description));
export { ChartSummaryDescriptionDescription };
