// igniteui-react-core/esm5/lib/ColumnDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { DefinitionBaseDescription } from "./DefinitionBaseDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var ColumnDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ColumnDescription, _super);
    function ColumnDescription() {
        var _this = _super.call(this) || this;
        _this.di = 0;
        _this.dk = 0;
        _this.dj = 0;
        _this.dh = 0;
        _this.eg = null;
        _this.ea = null;
        _this.ec = null;
        _this.dy = null;
        _this.ek = null;
        _this.d2 = null;
        _this.ei = null;
        _this.d0 = null;
        _this.ej = null;
        _this.d1 = null;
        _this.ca = null;
        _this.b6 = null;
        _this.c8 = 0;
        _this.co = false;
        _this.ci = false;
        _this.b4 = null;
        _this.b8 = null;
        _this.cc = null;
        _this.cn = false;
        _this.cq = false;
        _this.cp = false;
        _this.cr = false;
        _this.el = null;
        _this.eh = null;
        _this.du = null;
        _this.d3 = null;
        _this.dw = null;
        _this.d5 = null;
        _this.d4 = null;
        _this.dv = null;
        _this.cf = false;
        _this.cj = false;
        _this.ch = false;
        _this.cl = false;
        _this.cg = false;
        _this.ck = false;
        _this.cm = false;
        _this.d6 = null;
        _this.dx = null;
        _this.c7 = 0;
        _this.c5 = 0;
        _this.d7 = null;
        _this.c6 = 0;
        _this.d8 = null;
        _this.d9 = null;
        _this.ee = null;
        _this.ed = null;
        _this.ef = null;
        _this.de = 0;
        _this.dg = 0;
        _this.df = 0;
        _this.dd = 0;
        _this.eb = null;
        _this.b2 = null;
        _this.dz = null;
        return _this;
    }
    ColumnDescription.prototype.get_type = function () {
        return "Column";
    };
    Object.defineProperty(ColumnDescription.prototype, "paddingLeft", {
        get: function () {
            return this.di;
        },
        set: function (a) {
            this.di = a;
            this.e("PaddingLeft");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "paddingTop", {
        get: function () {
            return this.dk;
        },
        set: function (a) {
            this.dk = a;
            this.e("PaddingTop");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "paddingRight", {
        get: function () {
            return this.dj;
        },
        set: function (a) {
            this.dj = a;
            this.e("PaddingRight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "paddingBottom", {
        get: function () {
            return this.dh;
        },
        set: function (a) {
            this.dh = a;
            this.e("PaddingBottom");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "name", {
        get: function () {
            return this.eg;
        },
        set: function (a) {
            this.eg = a;
            this.e("Name");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "field", {
        get: function () {
            return this.ea;
        },
        set: function (a) {
            this.ea = a;
            this.e("Field");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "headerText", {
        get: function () {
            return this.ec;
        },
        set: function (a) {
            this.ec = a;
            this.e("HeaderText");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualHeaderText", {
        get: function () {
            return this.dy;
        },
        set: function (a) {
            this.dy = a;
            this.e("ActualHeaderText");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "selectedBackground", {
        get: function () {
            return this.ek;
        },
        set: function (a) {
            this.ek = a;
            this.e("SelectedBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualSelectedBackground", {
        get: function () {
            return this.d2;
        },
        set: function (a) {
            this.d2 = a;
            this.e("ActualSelectedBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "rowHoverBackground", {
        get: function () {
            return this.ei;
        },
        set: function (a) {
            this.ei = a;
            this.e("RowHoverBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualHoverBackground", {
        get: function () {
            return this.d0;
        },
        set: function (a) {
            this.d0 = a;
            this.e("ActualHoverBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "rowHoverTextColor", {
        get: function () {
            return this.ej;
        },
        set: function (a) {
            this.ej = a;
            this.e("RowHoverTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualRowHoverTextColor", {
        get: function () {
            return this.d1;
        },
        set: function (a) {
            this.d1 = a;
            this.e("ActualRowHoverTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "animationSettings", {
        get: function () {
            return this.ca;
        },
        set: function (a) {
            this.ca = a;
            this.e("AnimationSettings");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "width", {
        get: function () {
            return this.b6;
        },
        set: function (a) {
            this.b6 = a;
            this.e("Width");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "minWidth", {
        get: function () {
            return this.c8;
        },
        set: function (a) {
            this.c8 = a;
            this.e("MinWidth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isFromMarkup", {
        get: function () {
            return this.co;
        },
        set: function (a) {
            this.co = a;
            this.e("IsFromMarkup");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isAutoGenerated", {
        get: function () {
            return this.ci;
        },
        set: function (a) {
            this.ci = a;
            this.e("IsAutoGenerated");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "filter", {
        get: function () {
            return this.b4;
        },
        set: function (a) {
            this.b4 = a;
            this.e("Filter");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "filterExpression", {
        get: function () {
            return this.b8;
        },
        set: function (a) {
            this.b8 = a;
            this.e("FilterExpression");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "header", {
        get: function () {
            return this.cc;
        },
        set: function (a) {
            this.cc = a;
            this.e("Header");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isFilteringEnabled", {
        get: function () {
            return this.cn;
        },
        set: function (a) {
            this.cn = a;
            this.e("IsFilteringEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isResizingEnabled", {
        get: function () {
            return this.cq;
        },
        set: function (a) {
            this.cq = a;
            this.e("IsResizingEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isHidden", {
        get: function () {
            return this.cp;
        },
        set: function (a) {
            this.cp = a;
            this.e("IsHidden");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "shouldRemoveWhenHidden", {
        get: function () {
            return this.cr;
        },
        set: function (a) {
            this.cr = a;
            this.e("ShouldRemoveWhenHidden");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "sortDirection", {
        get: function () {
            return this.el;
        },
        set: function (a) {
            this.el = a;
            this.e("SortDirection");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "pinned", {
        get: function () {
            return this.eh;
        },
        set: function (a) {
            this.eh = a;
            this.e("Pinned");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualColumnOptionsIconAlignment", {
        get: function () {
            return this.du;
        },
        set: function (a) {
            this.du = a;
            this.e("ActualColumnOptionsIconAlignment");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "columnOptionsIconAlignment", {
        get: function () {
            return this.d3;
        },
        set: function (a) {
            this.d3 = a;
            this.e("ColumnOptionsIconAlignment");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualColumnOptionsIconColor", {
        get: function () {
            return this.dw;
        },
        set: function (a) {
            this.dw = a;
            this.e("ActualColumnOptionsIconColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "columnOptionsIconColor", {
        get: function () {
            return this.d5;
        },
        set: function (a) {
            this.d5 = a;
            this.e("ColumnOptionsIconColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "columnOptionsIconBehavior", {
        get: function () {
            return this.d4;
        },
        set: function (a) {
            this.d4 = a;
            this.e("ColumnOptionsIconBehavior");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualColumnOptionsIconBehavior", {
        get: function () {
            return this.dv;
        },
        set: function (a) {
            this.dv = a;
            this.e("ActualColumnOptionsIconBehavior");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualIsColumnOptionsEnabled", {
        get: function () {
            return this.cf;
        },
        set: function (a) {
            this.cf = a;
            this.e("ActualIsColumnOptionsEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isColumnOptionsEnabled", {
        get: function () {
            return this.cj;
        },
        set: function (a) {
            this.cj = a;
            this.e("IsColumnOptionsEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualIsColumnOptionsSummariesEnabled", {
        get: function () {
            return this.ch;
        },
        set: function (a) {
            this.ch = a;
            this.e("ActualIsColumnOptionsSummariesEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isColumnOptionsSummariesEnabled", {
        get: function () {
            return this.cl;
        },
        set: function (a) {
            this.cl = a;
            this.e("IsColumnOptionsSummariesEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualIsColumnOptionsGroupingEnabled", {
        get: function () {
            return this.cg;
        },
        set: function (a) {
            this.cg = a;
            this.e("ActualIsColumnOptionsGroupingEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isColumnOptionsGroupingEnabled", {
        get: function () {
            return this.ck;
        },
        set: function (a) {
            this.ck = a;
            this.e("IsColumnOptionsGroupingEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "isEditable", {
        get: function () {
            return this.cm;
        },
        set: function (a) {
            this.cm = a;
            this.e("IsEditable");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "deletedTextColor", {
        get: function () {
            return this.d6;
        },
        set: function (a) {
            this.d6 = a;
            this.e("DeletedTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualDeletedTextColor", {
        get: function () {
            return this.dx;
        },
        set: function (a) {
            this.dx = a;
            this.e("ActualDeletedTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "editOpacity", {
        get: function () {
            return this.c7;
        },
        set: function (a) {
            this.c7 = a;
            this.e("EditOpacity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualEditOpacity", {
        get: function () {
            return this.c5;
        },
        set: function (a) {
            this.c5 = a;
            this.e("ActualEditOpacity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "editFontFamily", {
        get: function () {
            return this.d7;
        },
        set: function (a) {
            this.d7 = a;
            this.e("EditFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "editFontSize", {
        get: function () {
            return this.c6;
        },
        set: function (a) {
            this.c6 = a;
            this.e("EditFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "editFontStyle", {
        get: function () {
            return this.d8;
        },
        set: function (a) {
            this.d8 = a;
            this.e("EditFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "editFontWeight", {
        get: function () {
            return this.d9;
        },
        set: function (a) {
            this.d9 = a;
            this.e("EditFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "mergedCellMode", {
        get: function () {
            return this.ee;
        },
        set: function (a) {
            this.ee = a;
            this.e("MergedCellMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "mergedCellEvaluationCriteria", {
        get: function () {
            return this.ed;
        },
        set: function (a) {
            this.ed = a;
            this.e("MergedCellEvaluationCriteria");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "mergedCellVerticalAlignment", {
        get: function () {
            return this.ef;
        },
        set: function (a) {
            this.ef = a;
            this.e("MergedCellVerticalAlignment");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "mergedCellPaddingLeft", {
        get: function () {
            return this.de;
        },
        set: function (a) {
            this.de = a;
            this.e("MergedCellPaddingLeft");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "mergedCellPaddingTop", {
        get: function () {
            return this.dg;
        },
        set: function (a) {
            this.dg = a;
            this.e("MergedCellPaddingTop");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "mergedCellPaddingRight", {
        get: function () {
            return this.df;
        },
        set: function (a) {
            this.df = a;
            this.e("MergedCellPaddingRight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "mergedCellPaddingBottom", {
        get: function () {
            return this.dd;
        },
        set: function (a) {
            this.dd = a;
            this.e("MergedCellPaddingBottom");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "filterComparisonType", {
        get: function () {
            return this.eb;
        },
        set: function (a) {
            this.eb = a;
            this.e("FilterComparisonType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "filterOperands", {
        get: function () {
            return this.b2;
        },
        set: function (a) {
            this.b2 = a;
            this.e("FilterOperands");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ColumnDescription.prototype, "actualHeaderTextChangeRef", {
        get: function () {
            return this.dz;
        },
        set: function (a) {
            this.dz = a;
            this.e("ActualHeaderTextChangeRef");
        },
        enumerable: false,
        configurable: true
    });
    ColumnDescription.$t = markType(ColumnDescription, 'ColumnDescription', DefinitionBaseDescription.$);
    ColumnDescription.__canMarshalByValue = true;
    return ColumnDescription;
}(DefinitionBaseDescription));
export { ColumnDescription };
