// igniteui-react-core/esm5/lib/DataGridDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var DataGridDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(DataGridDescription, _super);
    function DataGridDescription() {
        var _this = _super.call(this) || this;
        _this.iu = null;
        _this.g0 = null;
        _this.e4 = null;
        _this.br = false;
        _this.bw = false;
        _this.b1 = false;
        _this.fz = null;
        _this.fr = null;
        _this.by = false;
        _this.hc = null;
        _this.hd = null;
        _this.gf = null;
        _this.df = 0;
        _this.dg = 0;
        _this.dh = 0;
        _this.di = 0;
        _this.bb = null;
        _this.bd = null;
        _this.bf = null;
        _this.a9 = null;
        _this.a3 = null;
        _this.a5 = null;
        _this.ax = null;
        _this.av = null;
        _this.a7 = null;
        _this.h = null;
        _this.j = null;
        _this.f9 = null;
        _this.eh = 0;
        _this.eo = 0;
        _this.dc = 0;
        _this.eg = 0;
        _this.ek = 0;
        _this.f7 = null;
        _this.at = null;
        _this.gr = null;
        _this.gd = null;
        _this.ge = null;
        _this.bm = false;
        _this.bj = false;
        _this.ep = 0;
        _this.bx = false;
        _this.fm = null;
        _this.fx = null;
        _this.e8 = null;
        _this.he = null;
        _this.hb = null;
        _this.fj = null;
        _this.az = null;
        _this.fy = null;
        _this.f2 = null;
        _this.fq = null;
        _this.fp = null;
        _this.fn = null;
        _this.hx = null;
        _this.e0 = null;
        _this.hv = null;
        _this.b2 = false;
        _this.v = null;
        _this.w = null;
        _this.u = null;
        _this.t = null;
        _this.p = null;
        _this.q = null;
        _this.bo = false;
        _this.bu = false;
        _this.bv = false;
        _this.fi = null;
        _this.hp = null;
        _this.h2 = null;
        _this.id = null;
        _this.dk = 0;
        _this.ga = null;
        _this.dd = 0;
        _this.gb = null;
        _this.gc = null;
        _this.de = 0;
        _this.f8 = null;
        _this.h0 = null;
        _this.g8 = null;
        _this.g4 = null;
        _this.a1 = null;
        _this.e6 = null;
        _this.fk = null;
        _this.gx = null;
        _this.gy = null;
        _this.fb = null;
        _this.c5 = 0;
        _this.fc = null;
        _this.fd = null;
        _this.hl = null;
        _this.hq = null;
        _this.hm = null;
        _this.dl = 0;
        _this.hn = null;
        _this.ho = null;
        _this.h3 = null;
        _this.dm = 0;
        _this.h4 = null;
        _this.h5 = null;
        _this.ie = null;
        _this.dp = 0;
        _this.ig = null;
        _this.ih = null;
        _this.ib = null;
        _this.h8 = null;
        _this.dn = 0;
        _this.h9 = null;
        _this.ia = null;
        _this.ik = null;
        _this.dq = 0;
        _this.il = null;
        _this.im = null;
        _this.io = null;
        _this.h6 = null;
        _this.ii = null;
        _this.h7 = null;
        _this.ij = null;
        _this.en = 0;
        _this.el = 0;
        _this.gw = null;
        _this.gq = null;
        _this.gz = null;
        _this.gs = null;
        _this.dj = 0;
        _this.gt = null;
        _this.gu = null;
        _this.hf = null;
        _this.gv = null;
        _this.hi = null;
        _this.hg = null;
        _this.hh = null;
        _this.ej = 0;
        _this.c7 = 0;
        _this.ei = 0;
        _this.c6 = 0;
        _this.f0 = null;
        _this.fs = null;
        _this.b7 = false;
        _this.ft = null;
        _this.fv = null;
        _this.fu = null;
        _this.e5 = null;
        _this.c1 = 0;
        _this.c2 = 0;
        _this.c3 = 0;
        _this.c4 = 0;
        _this.b3 = false;
        _this.c9 = 0;
        _this.c8 = 0;
        _this.da = 0;
        _this.db = 0;
        _this.bn = false;
        _this.s = null;
        _this.n = null;
        _this.m = null;
        _this.o = null;
        _this.g2 = null;
        _this.g1 = null;
        _this.g3 = null;
        _this.x = null;
        _this.l = null;
        _this.k = null;
        _this.y = null;
        _this.ic = null;
        _this.bs = false;
        _this.go = null;
        _this.bt = false;
        _this.gp = null;
        _this.bi = false;
        _this.r = null;
        _this.g = null;
        _this.f = null;
        _this.i = null;
        _this.b6 = false;
        _this.b5 = false;
        _this.b4 = false;
        _this.gg = null;
        _this.gh = null;
        _this.bq = false;
        _this.bp = false;
        _this.g6 = null;
        _this.g5 = null;
        _this.g7 = null;
        _this.gm = null;
        _this.gl = null;
        _this.gi = null;
        _this.cz = new Date();
        _this.bk = false;
        _this.em = 0;
        _this.bz = false;
        _this.b0 = false;
        _this.it = null;
        _this.ip = null;
        _this.iq = null;
        _this.ir = null;
        _this.is = null;
        _this.hk = null;
        _this.hj = null;
        _this.bh = false;
        _this.bl = false;
        _this.gk = null;
        _this.gj = null;
        _this.fg = null;
        _this.fh = null;
        _this.fe = null;
        _this.ff = null;
        _this.e7 = null;
        _this.hz = null;
        _this.gn = null;
        _this.h1 = null;
        _this.fo = null;
        _this.fw = null;
        _this.fl = null;
        _this.f5 = null;
        _this.f6 = null;
        _this.hy = null;
        _this.e2 = null;
        _this.f1 = null;
        _this.ht = null;
        _this.hu = null;
        _this.hs = null;
        _this.hw = null;
        _this.e1 = null;
        _this.hr = null;
        _this.f3 = null;
        _this.fa = null;
        _this.e9 = null;
        _this.ha = null;
        _this.g9 = null;
        _this.f4 = null;
        _this.e3 = null;
        return _this;
    }
    Object.defineProperty(DataGridDescription.prototype, "width", {
        get: function () {
            return this.iu;
        },
        set: function (a) {
            this.iu = a;
            this.e("Width");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "height", {
        get: function () {
            return this.g0;
        },
        set: function (a) {
            this.g0 = a;
            this.e("Height");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "background", {
        get: function () {
            return this.e4;
        },
        set: function (a) {
            this.e4 = a;
            this.e("Background");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isGroupByAreaVisible", {
        get: function () {
            return this.br;
        },
        set: function (a) {
            this.br = a;
            this.e("IsGroupByAreaVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isPagerVisible", {
        get: function () {
            return this.bw;
        },
        set: function (a) {
            this.bw = a;
            this.e("IsPagerVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isToolbarVisible", {
        get: function () {
            return this.b1;
        },
        set: function (a) {
            this.b1 = a;
            this.e("IsToolbarVisible");
        },
        enumerable: false,
        configurable: true
    });
    DataGridDescription.prototype.get_type = function () {
        return "DataGrid";
    };
    Object.defineProperty(DataGridDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnResizingMode", {
        get: function () {
            return this.fz;
        },
        set: function (a) {
            this.fz = a;
            this.e("ColumnResizingMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnMovingMode", {
        get: function () {
            return this.fr;
        },
        set: function (a) {
            this.fr = a;
            this.e("ColumnMovingMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isRowHoverEnabled", {
        get: function () {
            return this.by;
        },
        set: function (a) {
            this.by = a;
            this.e("IsRowHoverEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowHoverBackground", {
        get: function () {
            return this.hc;
        },
        set: function (a) {
            this.hc = a;
            this.e("RowHoverBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowHoverTextColor", {
        get: function () {
            return this.hd;
        },
        set: function (a) {
            this.hd = a;
            this.e("RowHoverTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editRowBorder", {
        get: function () {
            return this.gf;
        },
        set: function (a) {
            this.gf = a;
            this.e("EditRowBorder");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editRowBorderWidthBottom", {
        get: function () {
            return this.df;
        },
        set: function (a) {
            this.df = a;
            this.e("EditRowBorderWidthBottom");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editRowBorderWidthLeft", {
        get: function () {
            return this.dg;
        },
        set: function (a) {
            this.dg = a;
            this.e("EditRowBorderWidthLeft");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editRowBorderWidthRight", {
        get: function () {
            return this.dh;
        },
        set: function (a) {
            this.dh = a;
            this.e("EditRowBorderWidthRight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editRowBorderWidthTop", {
        get: function () {
            return this.di;
        },
        set: function (a) {
            this.di = a;
            this.e("EditRowBorderWidthTop");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "sectionHeader", {
        get: function () {
            return this.bb;
        },
        set: function (a) {
            this.bb = a;
            this.e("SectionHeader");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRowRoot", {
        get: function () {
            return this.bd;
        },
        set: function (a) {
            this.bd = a;
            this.e("SummaryRowRoot");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRowSection", {
        get: function () {
            return this.bf;
        },
        set: function (a) {
            this.bf = a;
            this.e("SummaryRowSection");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowSeparator", {
        get: function () {
            return this.a9;
        },
        set: function (a) {
            this.a9 = a;
            this.e("RowSeparator");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerRowSeparator", {
        get: function () {
            return this.a3;
        },
        set: function (a) {
            this.a3 = a;
            this.e("HeaderRowSeparator");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerSeparator", {
        get: function () {
            return this.a5;
        },
        set: function (a) {
            this.a5 = a;
            this.e("HeaderSeparator");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnResizingSeparator", {
        get: function () {
            return this.ax;
        },
        set: function (a) {
            this.ax = a;
            this.e("ColumnResizingSeparator");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnMovingSeparator", {
        get: function () {
            return this.av;
        },
        set: function (a) {
            this.av = a;
            this.e("ColumnMovingSeparator");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "pinnedAreaSeparator", {
        get: function () {
            return this.a7;
        },
        set: function (a) {
            this.a7 = a;
            this.e("PinnedAreaSeparator");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "autoGenerateDesiredProperties", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("AutoGenerateDesiredProperties");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "dataSourceDesiredProperties", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("DataSourceDesiredProperties");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "density", {
        get: function () {
            return this.f9;
        },
        set: function (a) {
            this.f9 = a;
            this.e("Density");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "actualRowHeight", {
        get: function () {
            return this.eh;
        },
        set: function (a) {
            this.eh = a;
            this.e("ActualRowHeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowHeight", {
        get: function () {
            return this.eo;
        },
        set: function (a) {
            this.eo = a;
            this.e("RowHeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "defaultColumnMinWidth", {
        get: function () {
            return this.dc;
        },
        set: function (a) {
            this.dc = a;
            this.e("DefaultColumnMinWidth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "actualHeaderHeight", {
        get: function () {
            return this.eg;
        },
        set: function (a) {
            this.eg = a;
            this.e("ActualHeaderHeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerHeight", {
        get: function () {
            return this.ek;
        },
        set: function (a) {
            this.ek = a;
            this.e("HeaderHeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "dataSourceRef", {
        get: function () {
            return this.f7;
        },
        set: function (a) {
            this.f7 = a;
            this.e("DataSourceRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "dataSource", {
        get: function () {
            return this.at;
        },
        set: function (a) {
            this.at = a;
            this.e("DataSource");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerClickAction", {
        get: function () {
            return this.gr;
        },
        set: function (a) {
            this.gr = a;
            this.e("HeaderClickAction");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editMode", {
        get: function () {
            return this.gd;
        },
        set: function (a) {
            this.gd = a;
            this.e("EditMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editModeClickAction", {
        get: function () {
            return this.ge;
        },
        set: function (a) {
            this.ge = a;
            this.e("EditModeClickAction");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editOnKeyPress", {
        get: function () {
            return this.bm;
        },
        set: function (a) {
            this.bm = a;
            this.e("EditOnKeyPress");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "autoGenerateColumns", {
        get: function () {
            return this.bj;
        },
        set: function (a) {
            this.bj = a;
            this.e("AutoGenerateColumns");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowSeparatorHeight", {
        get: function () {
            return this.ep;
        },
        set: function (a) {
            this.ep = a;
            this.e("RowSeparatorHeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isPlaceholderRenderingEnabled", {
        get: function () {
            return this.bx;
        },
        set: function (a) {
            this.bx = a;
            this.e("IsPlaceholderRenderingEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnAddingAnimationMode", {
        get: function () {
            return this.fm;
        },
        set: function (a) {
            this.fm = a;
            this.e("ColumnAddingAnimationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnPropertyUpdatingAnimationMode", {
        get: function () {
            return this.fx;
        },
        set: function (a) {
            this.fx = a;
            this.e("ColumnPropertyUpdatingAnimationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellDataLoadedAnimationMode", {
        get: function () {
            return this.e8;
        },
        set: function (a) {
            this.e8 = a;
            this.e("CellDataLoadedAnimationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowSelectionAnimationMode", {
        get: function () {
            return this.he;
        },
        set: function (a) {
            this.he = a;
            this.e("RowSelectionAnimationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowHoverAnimationMode", {
        get: function () {
            return this.hb;
        },
        set: function (a) {
            this.hb = a;
            this.e("RowHoverAnimationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellSelectionAnimationMode", {
        get: function () {
            return this.fj;
        },
        set: function (a) {
            this.fj = a;
            this.e("CellSelectionAnimationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "animationSettings", {
        get: function () {
            return this.az;
        },
        set: function (a) {
            this.az = a;
            this.e("AnimationSettings");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnResizingAnimationMode", {
        get: function () {
            return this.fy;
        },
        set: function (a) {
            this.fy = a;
            this.e("ColumnResizingAnimationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnShowingAnimationMode", {
        get: function () {
            return this.f2;
        },
        set: function (a) {
            this.f2 = a;
            this.e("ColumnShowingAnimationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnMovingAnimationMode", {
        get: function () {
            return this.fq;
        },
        set: function (a) {
            this.fq = a;
            this.e("ColumnMovingAnimationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnHidingAnimationMode", {
        get: function () {
            return this.fp;
        },
        set: function (a) {
            this.fp = a;
            this.e("ColumnHidingAnimationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnExchangingAnimationMode", {
        get: function () {
            return this.fn;
        },
        set: function (a) {
            this.fn = a;
            this.e("ColumnExchangingAnimationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "selectionMode", {
        get: function () {
            return this.hx;
        },
        set: function (a) {
            this.hx = a;
            this.e("SelectionMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "activationMode", {
        get: function () {
            return this.e0;
        },
        set: function (a) {
            this.e0 = a;
            this.e("ActivationMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "selectionBehavior", {
        get: function () {
            return this.hv;
        },
        set: function (a) {
            this.hv = a;
            this.e("SelectionBehavior");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "mouseDragSelectionEnabled", {
        get: function () {
            return this.b2;
        },
        set: function (a) {
            this.b2 = a;
            this.e("MouseDragSelectionEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "selectedItems", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            this.v = a;
            this.e("SelectedItems");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "selectedKeys", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            this.w = a;
            this.e("SelectedKeys");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "selectedCells", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            this.u = a;
            this.e("SelectedCells");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "selectedCellRanges", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
            this.e("SelectedCellRanges");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "pinnedItems", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("PinnedItems");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "pinnedKeys", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("PinnedKeys");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isColumnOptionsEnabled", {
        get: function () {
            return this.bo;
        },
        set: function (a) {
            this.bo = a;
            this.e("IsColumnOptionsEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isGroupRowSticky", {
        get: function () {
            return this.bu;
        },
        set: function (a) {
            this.bu = a;
            this.e("IsGroupRowSticky");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isHeaderSeparatorVisible", {
        get: function () {
            return this.bv;
        },
        set: function (a) {
            this.bv = a;
            this.e("IsHeaderSeparatorVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellSelectedBackground", {
        get: function () {
            return this.fi;
        },
        set: function (a) {
            this.fi = a;
            this.e("CellSelectedBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "sectionHeaderSelectedBackground", {
        get: function () {
            return this.hp;
        },
        set: function (a) {
            this.hp = a;
            this.e("SectionHeaderSelectedBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootBackground", {
        get: function () {
            return this.h2;
        },
        set: function (a) {
            this.h2 = a;
            this.e("SummaryRootBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionBackground", {
        get: function () {
            return this.id;
        },
        set: function (a) {
            this.id = a;
            this.e("SummarySectionBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "pinnedRowOpacity", {
        get: function () {
            return this.dk;
        },
        set: function (a) {
            this.dk = a;
            this.e("PinnedRowOpacity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editFontFamily", {
        get: function () {
            return this.ga;
        },
        set: function (a) {
            this.ga = a;
            this.e("EditFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editFontSize", {
        get: function () {
            return this.dd;
        },
        set: function (a) {
            this.dd = a;
            this.e("EditFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editFontStyle", {
        get: function () {
            return this.gb;
        },
        set: function (a) {
            this.gb = a;
            this.e("EditFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editFontWeight", {
        get: function () {
            return this.gc;
        },
        set: function (a) {
            this.gc = a;
            this.e("EditFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "editOpacity", {
        get: function () {
            return this.de;
        },
        set: function (a) {
            this.de = a;
            this.e("EditOpacity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "deletedTextColor", {
        get: function () {
            return this.f8;
        },
        set: function (a) {
            this.f8 = a;
            this.e("DeletedTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "stickyRowBackground", {
        get: function () {
            return this.h0;
        },
        set: function (a) {
            this.h0 = a;
            this.e("StickyRowBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "pinnedRowBackground", {
        get: function () {
            return this.g8;
        },
        set: function (a) {
            this.g8 = a;
            this.e("PinnedRowBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "lastStickyRowBackground", {
        get: function () {
            return this.g4;
        },
        set: function (a) {
            this.g4 = a;
            this.e("LastStickyRowBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "activeCell", {
        get: function () {
            return this.a1;
        },
        set: function (a) {
            this.a1 = a;
            this.e("ActiveCell");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellBackground", {
        get: function () {
            return this.e6;
        },
        set: function (a) {
            this.e6 = a;
            this.e("CellBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellTextColor", {
        get: function () {
            return this.fk;
        },
        set: function (a) {
            this.fk = a;
            this.e("CellTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerSortIndicatorColor", {
        get: function () {
            return this.gx;
        },
        set: function (a) {
            this.gx = a;
            this.e("HeaderSortIndicatorColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerSortIndicatorStyle", {
        get: function () {
            return this.gy;
        },
        set: function (a) {
            this.gy = a;
            this.e("HeaderSortIndicatorStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellFontFamily", {
        get: function () {
            return this.fb;
        },
        set: function (a) {
            this.fb = a;
            this.e("CellFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellFontSize", {
        get: function () {
            return this.c5;
        },
        set: function (a) {
            this.c5 = a;
            this.e("CellFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellFontStyle", {
        get: function () {
            return this.fc;
        },
        set: function (a) {
            this.fc = a;
            this.e("CellFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellFontWeight", {
        get: function () {
            return this.fd;
        },
        set: function (a) {
            this.fd = a;
            this.e("CellFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "sectionHeaderBackground", {
        get: function () {
            return this.hl;
        },
        set: function (a) {
            this.hl = a;
            this.e("SectionHeaderBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "sectionHeaderTextColor", {
        get: function () {
            return this.hq;
        },
        set: function (a) {
            this.hq = a;
            this.e("SectionHeaderTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "sectionHeaderFontFamily", {
        get: function () {
            return this.hm;
        },
        set: function (a) {
            this.hm = a;
            this.e("SectionHeaderFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "sectionHeaderFontSize", {
        get: function () {
            return this.dl;
        },
        set: function (a) {
            this.dl = a;
            this.e("SectionHeaderFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "sectionHeaderFontStyle", {
        get: function () {
            return this.hn;
        },
        set: function (a) {
            this.hn = a;
            this.e("SectionHeaderFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "sectionHeaderFontWeight", {
        get: function () {
            return this.ho;
        },
        set: function (a) {
            this.ho = a;
            this.e("SectionHeaderFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootLabelFontFamily", {
        get: function () {
            return this.h3;
        },
        set: function (a) {
            this.h3 = a;
            this.e("SummaryRootLabelFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootLabelFontSize", {
        get: function () {
            return this.dm;
        },
        set: function (a) {
            this.dm = a;
            this.e("SummaryRootLabelFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootLabelFontStyle", {
        get: function () {
            return this.h4;
        },
        set: function (a) {
            this.h4 = a;
            this.e("SummaryRootLabelFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootLabelFontWeight", {
        get: function () {
            return this.h5;
        },
        set: function (a) {
            this.h5 = a;
            this.e("SummaryRootLabelFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionLabelFontFamily", {
        get: function () {
            return this.ie;
        },
        set: function (a) {
            this.ie = a;
            this.e("SummarySectionLabelFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionLabelFontSize", {
        get: function () {
            return this.dp;
        },
        set: function (a) {
            this.dp = a;
            this.e("SummarySectionLabelFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionLabelFontStyle", {
        get: function () {
            return this.ig;
        },
        set: function (a) {
            this.ig = a;
            this.e("SummarySectionLabelFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionLabelFontWeight", {
        get: function () {
            return this.ih;
        },
        set: function (a) {
            this.ih = a;
            this.e("SummarySectionLabelFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootValueTextColor", {
        get: function () {
            return this.ib;
        },
        set: function (a) {
            this.ib = a;
            this.e("SummaryRootValueTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootValueFontFamily", {
        get: function () {
            return this.h8;
        },
        set: function (a) {
            this.h8 = a;
            this.e("SummaryRootValueFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootValueFontSize", {
        get: function () {
            return this.dn;
        },
        set: function (a) {
            this.dn = a;
            this.e("SummaryRootValueFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootValueFontStyle", {
        get: function () {
            return this.h9;
        },
        set: function (a) {
            this.h9 = a;
            this.e("SummaryRootValueFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootValueFontWeight", {
        get: function () {
            return this.ia;
        },
        set: function (a) {
            this.ia = a;
            this.e("SummaryRootValueFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionValueFontFamily", {
        get: function () {
            return this.ik;
        },
        set: function (a) {
            this.ik = a;
            this.e("SummarySectionValueFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionValueFontSize", {
        get: function () {
            return this.dq;
        },
        set: function (a) {
            this.dq = a;
            this.e("SummarySectionValueFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionValueFontStyle", {
        get: function () {
            return this.il;
        },
        set: function (a) {
            this.il = a;
            this.e("SummarySectionValueFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionValueFontWeight", {
        get: function () {
            return this.im;
        },
        set: function (a) {
            this.im = a;
            this.e("SummarySectionValueFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionValueTextColor", {
        get: function () {
            return this.io;
        },
        set: function (a) {
            this.io = a;
            this.e("SummarySectionValueTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootLabelTextColor", {
        get: function () {
            return this.h6;
        },
        set: function (a) {
            this.h6 = a;
            this.e("SummaryRootLabelTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionLabelTextColor", {
        get: function () {
            return this.ii;
        },
        set: function (a) {
            this.ii = a;
            this.e("SummarySectionLabelTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryRootSelectedBackground", {
        get: function () {
            return this.h7;
        },
        set: function (a) {
            this.h7 = a;
            this.e("SummaryRootSelectedBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summarySectionSelectedBackground", {
        get: function () {
            return this.ij;
        },
        set: function (a) {
            this.ij = a;
            this.e("SummarySectionSelectedBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "pinnedAreaSeparatorWidth", {
        get: function () {
            return this.en;
        },
        set: function (a) {
            this.en = a;
            this.e("PinnedAreaSeparatorWidth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerSeparatorWidth", {
        get: function () {
            return this.el;
        },
        set: function (a) {
            this.el = a;
            this.e("HeaderSeparatorWidth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerSeparatorBackground", {
        get: function () {
            return this.gw;
        },
        set: function (a) {
            this.gw = a;
            this.e("HeaderSeparatorBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerBackground", {
        get: function () {
            return this.gq;
        },
        set: function (a) {
            this.gq = a;
            this.e("HeaderBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerTextColor", {
        get: function () {
            return this.gz;
        },
        set: function (a) {
            this.gz = a;
            this.e("HeaderTextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerFontFamily", {
        get: function () {
            return this.gs;
        },
        set: function (a) {
            this.gs = a;
            this.e("HeaderFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerFontSize", {
        get: function () {
            return this.dj;
        },
        set: function (a) {
            this.dj = a;
            this.e("HeaderFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerFontStyle", {
        get: function () {
            return this.gt;
        },
        set: function (a) {
            this.gt = a;
            this.e("HeaderFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerFontWeight", {
        get: function () {
            return this.gu;
        },
        set: function (a) {
            this.gu = a;
            this.e("HeaderFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowSeparatorBackground", {
        get: function () {
            return this.hf;
        },
        set: function (a) {
            this.hf = a;
            this.e("RowSeparatorBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "headerRowSeparatorBackground", {
        get: function () {
            return this.gv;
        },
        set: function (a) {
            this.gv = a;
            this.e("HeaderRowSeparatorBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowSeparatorStickyRowBackground", {
        get: function () {
            return this.hi;
        },
        set: function (a) {
            this.hi = a;
            this.e("RowSeparatorStickyRowBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowSeparatorLastStickyRowBackground", {
        get: function () {
            return this.hg;
        },
        set: function (a) {
            this.hg = a;
            this.e("RowSeparatorLastStickyRowBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowSeparatorPinnedRowBackground", {
        get: function () {
            return this.hh;
        },
        set: function (a) {
            this.hh = a;
            this.e("RowSeparatorPinnedRowBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnResizingSeparatorWidth", {
        get: function () {
            return this.ej;
        },
        set: function (a) {
            this.ej = a;
            this.e("ColumnResizingSeparatorWidth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnResizingSeparatorOpacity", {
        get: function () {
            return this.c7;
        },
        set: function (a) {
            this.c7 = a;
            this.e("ColumnResizingSeparatorOpacity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnMovingSeparatorWidth", {
        get: function () {
            return this.ei;
        },
        set: function (a) {
            this.ei = a;
            this.e("ColumnMovingSeparatorWidth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnMovingSeparatorOpacity", {
        get: function () {
            return this.c6;
        },
        set: function (a) {
            this.c6 = a;
            this.e("ColumnMovingSeparatorOpacity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnResizingSeparatorBackground", {
        get: function () {
            return this.f0;
        },
        set: function (a) {
            this.f0 = a;
            this.e("ColumnResizingSeparatorBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnMovingSeparatorBackground", {
        get: function () {
            return this.fs;
        },
        set: function (a) {
            this.fs = a;
            this.e("ColumnMovingSeparatorBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "shiftSectionContent", {
        get: function () {
            return this.b7;
        },
        set: function (a) {
            this.b7 = a;
            this.e("ShiftSectionContent");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnOptionsIconAlignment", {
        get: function () {
            return this.ft;
        },
        set: function (a) {
            this.ft = a;
            this.e("ColumnOptionsIconAlignment");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnOptionsIconColor", {
        get: function () {
            return this.fv;
        },
        set: function (a) {
            this.fv = a;
            this.e("ColumnOptionsIconColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnOptionsIconBehavior", {
        get: function () {
            return this.fu;
        },
        set: function (a) {
            this.fu = a;
            this.e("ColumnOptionsIconBehavior");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "border", {
        get: function () {
            return this.e5;
        },
        set: function (a) {
            this.e5 = a;
            this.e("Border");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "borderWidthBottom", {
        get: function () {
            return this.c1;
        },
        set: function (a) {
            this.c1 = a;
            this.e("BorderWidthBottom");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "borderWidthLeft", {
        get: function () {
            return this.c2;
        },
        set: function (a) {
            this.c2 = a;
            this.e("BorderWidthLeft");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "borderWidthRight", {
        get: function () {
            return this.c3;
        },
        set: function (a) {
            this.c3 = a;
            this.e("BorderWidthRight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "borderWidthTop", {
        get: function () {
            return this.c4;
        },
        set: function (a) {
            this.c4 = a;
            this.e("BorderWidthTop");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "notifyOnAllSelectionChanges", {
        get: function () {
            return this.b3;
        },
        set: function (a) {
            this.b3 = a;
            this.e("NotifyOnAllSelectionChanges");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cornerRadiusBottomRight", {
        get: function () {
            return this.c9;
        },
        set: function (a) {
            this.c9 = a;
            this.e("CornerRadiusBottomRight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cornerRadiusBottomLeft", {
        get: function () {
            return this.c8;
        },
        set: function (a) {
            this.c8 = a;
            this.e("CornerRadiusBottomLeft");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cornerRadiusTopLeft", {
        get: function () {
            return this.da;
        },
        set: function (a) {
            this.da = a;
            this.e("CornerRadiusTopLeft");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cornerRadiusTopRight", {
        get: function () {
            return this.db;
        },
        set: function (a) {
            this.db = a;
            this.e("CornerRadiusTopRight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isActiveCellStyleEnabled", {
        get: function () {
            return this.bn;
        },
        set: function (a) {
            this.bn = a;
            this.e("IsActiveCellStyleEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "responsiveStates", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("ResponsiveStates");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "initialSortDescriptions", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("InitialSortDescriptions");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "initialGroupDescriptions", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("InitialGroupDescriptions");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "initialSummaryDescriptions", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("InitialSummaryDescriptions");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "initialSorts", {
        get: function () {
            return this.g2;
        },
        set: function (a) {
            this.g2 = a;
            this.e("InitialSorts");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "initialGroups", {
        get: function () {
            return this.g1;
        },
        set: function (a) {
            this.g1 = a;
            this.e("InitialGroups");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "initialSummaries", {
        get: function () {
            return this.g3;
        },
        set: function (a) {
            this.g3 = a;
            this.e("InitialSummaries");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "sortDescriptions", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            this.x = a;
            this.e("SortDescriptions");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "groupDescriptions", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("GroupDescriptions");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "filterExpressions", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("FilterExpressions");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryDescriptions", {
        get: function () {
            return this.y;
        },
        set: function (a) {
            this.y = a;
            this.e("SummaryDescriptions");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryScope", {
        get: function () {
            return this.ic;
        },
        set: function (a) {
            this.ic = a;
            this.e("SummaryScope");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isGroupCollapsable", {
        get: function () {
            return this.bs;
        },
        set: function (a) {
            this.bs = a;
            this.e("IsGroupCollapsable");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "groupHeaderDisplayMode", {
        get: function () {
            return this.go;
        },
        set: function (a) {
            this.go = a;
            this.e("GroupHeaderDisplayMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isGroupExpandedDefault", {
        get: function () {
            return this.bt;
        },
        set: function (a) {
            this.bt = a;
            this.e("IsGroupExpandedDefault");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "groupSummaryDisplayMode", {
        get: function () {
            return this.gp;
        },
        set: function (a) {
            this.gp = a;
            this.e("GroupSummaryDisplayMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "autoAcceptEdits", {
        get: function () {
            return this.bi;
        },
        set: function (a) {
            this.bi = a;
            this.e("AutoAcceptEdits");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "primaryKey", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("PrimaryKey");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "actualPrimaryKey", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("ActualPrimaryKey");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "actualColumns", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("ActualColumns");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columns", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("Columns");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "reactsToSortChanges", {
        get: function () {
            return this.b6;
        },
        set: function (a) {
            this.b6 = a;
            this.e("ReactsToSortChanges");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "reactsToGroupChanges", {
        get: function () {
            return this.b5;
        },
        set: function (a) {
            this.b5 = a;
            this.e("ReactsToGroupChanges");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "reactsToFilterChanges", {
        get: function () {
            return this.b4;
        },
        set: function (a) {
            this.b4 = a;
            this.e("ReactsToFilterChanges");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "enterBehavior", {
        get: function () {
            return this.gg;
        },
        set: function (a) {
            this.gg = a;
            this.e("EnterBehavior");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "enterBehaviorAfterEdit", {
        get: function () {
            return this.gh;
        },
        set: function (a) {
            this.gh = a;
            this.e("EnterBehaviorAfterEdit");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isColumnOptionsSummariesEnabled", {
        get: function () {
            return this.bq;
        },
        set: function (a) {
            this.bq = a;
            this.e("IsColumnOptionsSummariesEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isColumnOptionsGroupingEnabled", {
        get: function () {
            return this.bp;
        },
        set: function (a) {
            this.bp = a;
            this.e("IsColumnOptionsGroupingEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "mergedCellMode", {
        get: function () {
            return this.g6;
        },
        set: function (a) {
            this.g6 = a;
            this.e("MergedCellMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "mergedCellEvaluationCriteria", {
        get: function () {
            return this.g5;
        },
        set: function (a) {
            this.g5 = a;
            this.e("MergedCellEvaluationCriteria");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "mergedCellVerticalAlignment", {
        get: function () {
            return this.g7;
        },
        set: function (a) {
            this.g7 = a;
            this.e("MergedCellVerticalAlignment");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "filterUIType", {
        get: function () {
            return this.gm;
        },
        set: function (a) {
            this.gm = a;
            this.e("FilterUIType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "filterLogicalOperator", {
        get: function () {
            return this.gl;
        },
        set: function (a) {
            this.gl = a;
            this.e("FilterLogicalOperator");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "filterComparisonType", {
        get: function () {
            return this.gi;
        },
        set: function (a) {
            this.gi = a;
            this.e("FilterComparisonType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "todayOverride", {
        get: function () {
            return this.cz;
        },
        set: function (a) {
            this.cz = a;
            this.e("TodayOverride");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "clearSelectionOnEscape", {
        get: function () {
            return this.bk;
        },
        set: function (a) {
            this.bk = a;
            this.e("ClearSelectionOnEscape");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "pageSize", {
        get: function () {
            return this.em;
        },
        set: function (a) {
            this.em = a;
            this.e("PageSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isToolbarColumnChooserVisible", {
        get: function () {
            return this.bz;
        },
        set: function (a) {
            this.bz = a;
            this.e("IsToolbarColumnChooserVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "isToolbarColumnPinningVisible", {
        get: function () {
            return this.b0;
        },
        set: function (a) {
            this.b0 = a;
            this.e("IsToolbarColumnPinningVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "toolbarTitle", {
        get: function () {
            return this.it;
        },
        set: function (a) {
            this.it = a;
            this.e("ToolbarTitle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "toolbarColumnChooserText", {
        get: function () {
            return this.ip;
        },
        set: function (a) {
            this.ip = a;
            this.e("ToolbarColumnChooserText");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "toolbarColumnChooserTitle", {
        get: function () {
            return this.iq;
        },
        set: function (a) {
            this.iq = a;
            this.e("ToolbarColumnChooserTitle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "toolbarColumnPinningText", {
        get: function () {
            return this.ir;
        },
        set: function (a) {
            this.ir = a;
            this.e("ToolbarColumnPinningText");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "toolbarColumnPinningTitle", {
        get: function () {
            return this.is;
        },
        set: function (a) {
            this.is = a;
            this.e("ToolbarColumnPinningTitle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "scrollbarStyle", {
        get: function () {
            return this.hk;
        },
        set: function (a) {
            this.hk = a;
            this.e("ScrollbarStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "scrollbarBackground", {
        get: function () {
            return this.hj;
        },
        set: function (a) {
            this.hj = a;
            this.e("ScrollbarBackground");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "allowCopyOperation", {
        get: function () {
            return this.bh;
        },
        set: function (a) {
            this.bh = a;
            this.e("AllowCopyOperation");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "deferEventForRowDragSelection", {
        get: function () {
            return this.bl;
        },
        set: function (a) {
            this.bl = a;
            this.e("DeferEventForRowDragSelection");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "filterExpressionsChangingRef", {
        get: function () {
            return this.gk;
        },
        set: function (a) {
            this.gk = a;
            this.e("FilterExpressionsChangingRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "filterExpressionsChangedRef", {
        get: function () {
            return this.gj;
        },
        set: function (a) {
            this.gj = a;
            this.e("FilterExpressionsChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellPreviewPointerDownRef", {
        get: function () {
            return this.fg;
        },
        set: function (a) {
            this.fg = a;
            this.e("CellPreviewPointerDownRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellPreviewPointerUpRef", {
        get: function () {
            return this.fh;
        },
        set: function (a) {
            this.fh = a;
            this.e("CellPreviewPointerUpRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellPointerDownRef", {
        get: function () {
            return this.fe;
        },
        set: function (a) {
            this.fe = a;
            this.e("CellPointerDownRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellPointerUpRef", {
        get: function () {
            return this.ff;
        },
        set: function (a) {
            this.ff = a;
            this.e("CellPointerUpRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellClickedRef", {
        get: function () {
            return this.e7;
        },
        set: function (a) {
            this.e7 = a;
            this.e("CellClickedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "sortDescriptionsChangedRef", {
        get: function () {
            return this.hz;
        },
        set: function (a) {
            this.hz = a;
            this.e("SortDescriptionsChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "groupDescriptionsChangedRef", {
        get: function () {
            return this.gn;
        },
        set: function (a) {
            this.gn = a;
            this.e("GroupDescriptionsChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "summaryDescriptionsChangedRef", {
        get: function () {
            return this.h1;
        },
        set: function (a) {
            this.h1 = a;
            this.e("SummaryDescriptionsChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnHiddenChangedRef", {
        get: function () {
            return this.fo;
        },
        set: function (a) {
            this.fo = a;
            this.e("ColumnHiddenChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnPinnedChangedRef", {
        get: function () {
            return this.fw;
        },
        set: function (a) {
            this.fw = a;
            this.e("ColumnPinnedChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellValueChangingRef", {
        get: function () {
            return this.fl;
        },
        set: function (a) {
            this.fl = a;
            this.e("CellValueChangingRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "dataCommittedRef", {
        get: function () {
            return this.f5;
        },
        set: function (a) {
            this.f5 = a;
            this.e("DataCommittedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "dataCommittingRef", {
        get: function () {
            return this.f6;
        },
        set: function (a) {
            this.f6 = a;
            this.e("DataCommittingRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "sizeChangedRef", {
        get: function () {
            return this.hy;
        },
        set: function (a) {
            this.hy = a;
            this.e("SizeChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "actualColumnsChangedRef", {
        get: function () {
            return this.e2;
        },
        set: function (a) {
            this.e2 = a;
            this.e("ActualColumnsChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnsAutoGeneratedRef", {
        get: function () {
            return this.f1;
        },
        set: function (a) {
            this.f1 = a;
            this.e("ColumnsAutoGeneratedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "selectedItemsChangedRef", {
        get: function () {
            return this.ht;
        },
        set: function (a) {
            this.ht = a;
            this.e("SelectedItemsChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "selectedKeysChangedRef", {
        get: function () {
            return this.hu;
        },
        set: function (a) {
            this.hu = a;
            this.e("SelectedKeysChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "selectedCellsChangedRef", {
        get: function () {
            return this.hs;
        },
        set: function (a) {
            this.hs = a;
            this.e("SelectedCellsChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "selectionChangedRef", {
        get: function () {
            return this.hw;
        },
        set: function (a) {
            this.hw = a;
            this.e("SelectionChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "activeCellChangedRef", {
        get: function () {
            return this.e1;
        },
        set: function (a) {
            this.e1 = a;
            this.e("ActiveCellChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "selectedCellRangesChangedRef", {
        get: function () {
            return this.hr;
        },
        set: function (a) {
            this.hr = a;
            this.e("SelectedCellRangesChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "columnWidthChangedRef", {
        get: function () {
            return this.f3;
        },
        set: function (a) {
            this.f3 = a;
            this.e("ColumnWidthChangedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellEditStartedRef", {
        get: function () {
            return this.fa;
        },
        set: function (a) {
            this.fa = a;
            this.e("CellEditStartedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "cellEditEndedRef", {
        get: function () {
            return this.e9;
        },
        set: function (a) {
            this.e9 = a;
            this.e("CellEditEndedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowEditStartedRef", {
        get: function () {
            return this.ha;
        },
        set: function (a) {
            this.ha = a;
            this.e("RowEditStartedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "rowEditEndedRef", {
        get: function () {
            return this.g9;
        },
        set: function (a) {
            this.g9 = a;
            this.e("RowEditEndedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "customFilterRequestedRef", {
        get: function () {
            return this.f4;
        },
        set: function (a) {
            this.f4 = a;
            this.e("CustomFilterRequestedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridDescription.prototype, "actualPrimaryKeyChangeRef", {
        get: function () {
            return this.e3;
        },
        set: function (a) {
            this.e3 = a;
            this.e("ActualPrimaryKeyChangeRef");
        },
        enumerable: false,
        configurable: true
    });
    DataGridDescription.$t = markType(DataGridDescription, 'DataGridDescription', Description.$);
    return DataGridDescription;
}(Description));
export { DataGridDescription };
