// igniteui-react-core/esm5/lib/DataGridToolbarDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var DataGridToolbarDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(DataGridToolbarDescription, _super);
    function DataGridToolbarDescription() {
        var _this = _super.call(this) || this;
        _this.ab = null;
        _this.ac = null;
        _this.ae = null;
        _this.n = 0;
        _this.af = null;
        _this.ag = null;
        _this.ad = null;
        _this.t = null;
        _this.aa = null;
        _this.f = false;
        _this.v = null;
        _this.w = null;
        _this.g = false;
        _this.x = null;
        _this.y = null;
        _this.j = 0;
        _this.k = 0;
        _this.l = 0;
        _this.m = 0;
        _this.z = null;
        _this.u = null;
        return _this;
    }
    DataGridToolbarDescription.prototype.get_type = function () {
        return "DataGridToolbar";
    };
    Object.defineProperty(DataGridToolbarDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "targetGridRef", {
        get: function () {
            return this.ab;
        },
        set: function (a) {
            this.ab = a;
            this.e("TargetGridRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "toolbarTitle", {
        get: function () {
            return this.ac;
        },
        set: function (a) {
            this.ac = a;
            this.e("ToolbarTitle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "toolbarTitleFontFamily", {
        get: function () {
            return this.ae;
        },
        set: function (a) {
            this.ae = a;
            this.e("ToolbarTitleFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "toolbarTitleFontSize", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("ToolbarTitleFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "toolbarTitleFontStyle", {
        get: function () {
            return this.af;
        },
        set: function (a) {
            this.af = a;
            this.e("ToolbarTitleFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "toolbarTitleFontWeight", {
        get: function () {
            return this.ag;
        },
        set: function (a) {
            this.ag = a;
            this.e("ToolbarTitleFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "toolbarTitleColor", {
        get: function () {
            return this.ad;
        },
        set: function (a) {
            this.ad = a;
            this.e("ToolbarTitleColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "backgroundColor", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
            this.e("BackgroundColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "dialogBackgroundColor", {
        get: function () {
            return this.aa;
        },
        set: function (a) {
            this.aa = a;
            this.e("DialogBackgroundColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "columnChooser", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("ColumnChooser");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "columnChooserText", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            this.v = a;
            this.e("ColumnChooserText");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "columnChooserTitle", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            this.w = a;
            this.e("ColumnChooserTitle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "columnPinning", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("ColumnPinning");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "columnPinningText", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            this.x = a;
            this.e("ColumnPinningText");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "columnPinningTitle", {
        get: function () {
            return this.y;
        },
        set: function (a) {
            this.y = a;
            this.e("ColumnPinningTitle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "borderWidthBottom", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("BorderWidthBottom");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "borderWidthLeft", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("BorderWidthLeft");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "borderWidthRight", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("BorderWidthRight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "borderWidthTop", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("BorderWidthTop");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "density", {
        get: function () {
            return this.z;
        },
        set: function (a) {
            this.z = a;
            this.e("Density");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridToolbarDescription.prototype, "baseTheme", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            this.u = a;
            this.e("BaseTheme");
        },
        enumerable: false,
        configurable: true
    });
    DataGridToolbarDescription.$t = markType(DataGridToolbarDescription, 'DataGridToolbarDescription', Description.$);
    return DataGridToolbarDescription;
}(Description));
export { DataGridToolbarDescription };
