// igniteui-react-core/esm5/lib/DataGridExpansionIndicatorDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var DataGridExpansionIndicatorDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(DataGridExpansionIndicatorDescription, _super);
    function DataGridExpansionIndicatorDescription() {
        var _this = _super.call(this) || this;
        _this.g = false;
        _this.n = null;
        _this.f = false;
        _this.k = 0;
        _this.j = 0;
        return _this;
    }
    DataGridExpansionIndicatorDescription.prototype.get_type = function () {
        return "DataGridExpansionIndicator";
    };
    Object.defineProperty(DataGridExpansionIndicatorDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridExpansionIndicatorDescription.prototype, "isExpanded", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("IsExpanded");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridExpansionIndicatorDescription.prototype, "iconColor", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("IconColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridExpansionIndicatorDescription.prototype, "isAnimationEnabled", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("IsAnimationEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridExpansionIndicatorDescription.prototype, "pixelScalingRatio", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("PixelScalingRatio");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DataGridExpansionIndicatorDescription.prototype, "actualPixelScalingRatio", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("ActualPixelScalingRatio");
        },
        enumerable: false,
        configurable: true
    });
    DataGridExpansionIndicatorDescription.$t = markType(DataGridExpansionIndicatorDescription, 'DataGridExpansionIndicatorDescription', Description.$);
    return DataGridExpansionIndicatorDescription;
}(Description));
export { DataGridExpansionIndicatorDescription };
