// igniteui-react-core/esm5/lib/FilterOperandDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var FilterOperandDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(FilterOperandDescription, _super);
    function FilterOperandDescription() {
        var _this = _super.call(this) || this;
        _this.n = null;
        _this.j = null;
        _this.m = null;
        _this.g = false;
        _this.k = null;
        _this.l = null;
        return _this;
    }
    FilterOperandDescription.prototype.get_type = function () {
        return "FilterOperand";
    };
    Object.defineProperty(FilterOperandDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterOperandDescription.prototype, "iD", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("ID");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterOperandDescription.prototype, "displayName", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("DisplayName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterOperandDescription.prototype, "icon", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("Icon");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterOperandDescription.prototype, "isInputRequired", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("IsInputRequired");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterOperandDescription.prototype, "editorType", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("EditorType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterOperandDescription.prototype, "filterRequestedRef", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("FilterRequestedRef");
        },
        enumerable: false,
        configurable: true
    });
    FilterOperandDescription.$t = markType(FilterOperandDescription, 'FilterOperandDescription', Description.$);
    FilterOperandDescription.__marshalByValue = true;
    FilterOperandDescription.__marshalByValueAlias = "FilterOperand";
    return FilterOperandDescription;
}(Description));
export { FilterOperandDescription };
