// igniteui-react-core/esm5/lib/FilterCellInfoDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { CellInfoDescription } from "./CellInfoDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var FilterCellInfoDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(FilterCellInfoDescription, _super);
    function FilterCellInfoDescription() {
        var _this = _super.call(this) || this;
        _this.f3 = null;
        _this.fl = null;
        _this.fn = null;
        _this.fj = null;
        _this.f5 = null;
        _this.f4 = null;
        _this.fv = 0;
        _this.fu = 0;
        _this.ft = 0;
        _this.fs = 0;
        _this.fr = 0;
        _this.fq = 0;
        return _this;
    }
    FilterCellInfoDescription.prototype.get_type = function () {
        return "FilterCellInfo";
    };
    Object.defineProperty(FilterCellInfoDescription.prototype, "dataType", {
        get: function () {
            return this.f3;
        },
        set: function (a) {
            this.f3 = a;
            this.e("DataType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterCellInfoDescription.prototype, "filter", {
        get: function () {
            return this.fl;
        },
        set: function (a) {
            this.fl = a;
            this.e("Filter");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterCellInfoDescription.prototype, "filterExpression", {
        get: function () {
            return this.fn;
        },
        set: function (a) {
            this.fn = a;
            this.e("FilterExpression");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterCellInfoDescription.prototype, "filterOperands", {
        get: function () {
            return this.fj;
        },
        set: function (a) {
            this.fj = a;
            this.e("FilterOperands");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterCellInfoDescription.prototype, "operatorType", {
        get: function () {
            return this.f5;
        },
        set: function (a) {
            this.f5 = a;
            this.e("OperatorType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterCellInfoDescription.prototype, "filterValueRef", {
        get: function () {
            return this.f4;
        },
        set: function (a) {
            this.f4 = a;
            this.e("FilterValueRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterCellInfoDescription.prototype, "selectorTop", {
        get: function () {
            return this.fv;
        },
        set: function (a) {
            this.fv = a;
            this.e("SelectorTop");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterCellInfoDescription.prototype, "selectorLeft", {
        get: function () {
            return this.fu;
        },
        set: function (a) {
            this.fu = a;
            this.e("SelectorLeft");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterCellInfoDescription.prototype, "editorTop", {
        get: function () {
            return this.ft;
        },
        set: function (a) {
            this.ft = a;
            this.e("EditorTop");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterCellInfoDescription.prototype, "editorLeft", {
        get: function () {
            return this.fs;
        },
        set: function (a) {
            this.fs = a;
            this.e("EditorLeft");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterCellInfoDescription.prototype, "clearIconTop", {
        get: function () {
            return this.fr;
        },
        set: function (a) {
            this.fr = a;
            this.e("ClearIconTop");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(FilterCellInfoDescription.prototype, "clearIconLeft", {
        get: function () {
            return this.fq;
        },
        set: function (a) {
            this.fq = a;
            this.e("ClearIconLeft");
        },
        enumerable: false,
        configurable: true
    });
    FilterCellInfoDescription.$t = markType(FilterCellInfoDescription, 'FilterCellInfoDescription', CellInfoDescription.$);
    FilterCellInfoDescription.__marshalByValue1 = true;
    FilterCellInfoDescription.__marshalByValueAlias1 = "FilterCellModel";
    return FilterCellInfoDescription;
}(CellInfoDescription));
export { FilterCellInfoDescription };
