// igniteui-react-core/esm5/lib/GetTileImageUriArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var GetTileImageUriArgsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(GetTileImageUriArgsDescription, _super);
    function GetTileImageUriArgsDescription() {
        var _this = _super.call(this) || this;
        _this.f = 0;
        _this.g = 0;
        _this.h = 0;
        _this.l = null;
        return _this;
    }
    GetTileImageUriArgsDescription.prototype.get_type = function () {
        return "GetTileImageUriArgs";
    };
    Object.defineProperty(GetTileImageUriArgsDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GetTileImageUriArgsDescription.prototype, "tileLevel", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("TileLevel");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GetTileImageUriArgsDescription.prototype, "tilePositionX", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("TilePositionX");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GetTileImageUriArgsDescription.prototype, "tilePositionY", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("TilePositionY");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GetTileImageUriArgsDescription.prototype, "tileImageUri", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("TileImageUri");
        },
        enumerable: false,
        configurable: true
    });
    GetTileImageUriArgsDescription.$t = markType(GetTileImageUriArgsDescription, 'GetTileImageUriArgsDescription', Description.$);
    return GetTileImageUriArgsDescription;
}(Description));
export { GetTileImageUriArgsDescription };
