// igniteui-react-core/esm5/lib/GridColumnSummaryOptionsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { GridColumnOptionsSectionBaseDescription } from "./GridColumnOptionsSectionBaseDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var GridColumnSummaryOptionsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(GridColumnSummaryOptionsDescription, _super);
    function GridColumnSummaryOptionsDescription() {
        var _this = _super.call(this) || this;
        _this.at = null;
        _this.au = null;
        _this.as = null;
        _this.aq = 0;
        return _this;
    }
    GridColumnSummaryOptionsDescription.prototype.get_type = function () {
        return "GridColumnSummaryOptions";
    };
    Object.defineProperty(GridColumnSummaryOptionsDescription.prototype, "summaryCaption", {
        get: function () {
            return this.at;
        },
        set: function (a) {
            this.at = a;
            this.e("SummaryCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnSummaryOptionsDescription.prototype, "summaryListDensity", {
        get: function () {
            return this.au;
        },
        set: function (a) {
            this.au = a;
            this.e("SummaryListDensity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnSummaryOptionsDescription.prototype, "actualSummaryListDensity", {
        get: function () {
            return this.as;
        },
        set: function (a) {
            this.as = a;
            this.e("ActualSummaryListDensity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnSummaryOptionsDescription.prototype, "activeCount", {
        get: function () {
            return this.aq;
        },
        set: function (a) {
            this.aq = a;
            this.e("ActiveCount");
        },
        enumerable: false,
        configurable: true
    });
    GridColumnSummaryOptionsDescription.$t = markType(GridColumnSummaryOptionsDescription, 'GridColumnSummaryOptionsDescription', GridColumnOptionsSectionBaseDescription.$);
    return GridColumnSummaryOptionsDescription;
}(GridColumnOptionsSectionBaseDescription));
export { GridColumnSummaryOptionsDescription };
