// igniteui-react-core/esm5/lib/GridColumnFilterOptionsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { GridColumnOptionsSectionBaseDescription } from "./GridColumnOptionsSectionBaseDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var GridColumnFilterOptionsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(GridColumnFilterOptionsDescription, _super);
    function GridColumnFilterOptionsDescription() {
        var _this = _super.call(this) || this;
        _this.ar = null;
        _this.at = null;
        _this.as = null;
        _this.aq = null;
        _this.au = null;
        return _this;
    }
    GridColumnFilterOptionsDescription.prototype.get_type = function () {
        return "GridColumnFilterOptions";
    };
    Object.defineProperty(GridColumnFilterOptionsDescription.prototype, "clearColumnFiltersCaption", {
        get: function () {
            return this.ar;
        },
        set: function (a) {
            this.ar = a;
            this.e("ClearColumnFiltersCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnFilterOptionsDescription.prototype, "filterListPlaceholderText", {
        get: function () {
            return this.at;
        },
        set: function (a) {
            this.at = a;
            this.e("FilterListPlaceholderText");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnFilterOptionsDescription.prototype, "filterListDensity", {
        get: function () {
            return this.as;
        },
        set: function (a) {
            this.as = a;
            this.e("FilterListDensity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnFilterOptionsDescription.prototype, "actualFilterListDensity", {
        get: function () {
            return this.aq;
        },
        set: function (a) {
            this.aq = a;
            this.e("ActualFilterListDensity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnFilterOptionsDescription.prototype, "textFilterCaption", {
        get: function () {
            return this.au;
        },
        set: function (a) {
            this.au = a;
            this.e("TextFilterCaption");
        },
        enumerable: false,
        configurable: true
    });
    GridColumnFilterOptionsDescription.$t = markType(GridColumnFilterOptionsDescription, 'GridColumnFilterOptionsDescription', GridColumnOptionsSectionBaseDescription.$);
    return GridColumnFilterOptionsDescription;
}(GridColumnOptionsSectionBaseDescription));
export { GridColumnFilterOptionsDescription };
