// igniteui-react-core/esm5/lib/GridColumnButtonOptionsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { GridColumnOptionsSectionBaseDescription } from "./GridColumnOptionsSectionBaseDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var GridColumnButtonOptionsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(GridColumnButtonOptionsDescription, _super);
    function GridColumnButtonOptionsDescription() {
        var _this = _super.call(this) || this;
        _this.aq = null;
        _this.as = null;
        _this.ar = null;
        _this.at = null;
        return _this;
    }
    GridColumnButtonOptionsDescription.prototype.get_type = function () {
        return "GridColumnButtonOptions";
    };
    Object.defineProperty(GridColumnButtonOptionsDescription.prototype, "applyButtonCaption", {
        get: function () {
            return this.aq;
        },
        set: function (a) {
            this.aq = a;
            this.e("ApplyButtonCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnButtonOptionsDescription.prototype, "cancelButtonCaption", {
        get: function () {
            return this.as;
        },
        set: function (a) {
            this.as = a;
            this.e("CancelButtonCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnButtonOptionsDescription.prototype, "applyButtonClickRef", {
        get: function () {
            return this.ar;
        },
        set: function (a) {
            this.ar = a;
            this.e("ApplyButtonClickRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnButtonOptionsDescription.prototype, "cancelButtonClickRef", {
        get: function () {
            return this.at;
        },
        set: function (a) {
            this.at = a;
            this.e("CancelButtonClickRef");
        },
        enumerable: false,
        configurable: true
    });
    GridColumnButtonOptionsDescription.$t = markType(GridColumnButtonOptionsDescription, 'GridColumnButtonOptionsDescription', GridColumnOptionsSectionBaseDescription.$);
    return GridColumnButtonOptionsDescription;
}(GridColumnOptionsSectionBaseDescription));
export { GridColumnButtonOptionsDescription };
