// igniteui-react-core/esm5/lib/GridColumnOptionsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { GridColumnOptionsBaseDescription } from "./GridColumnOptionsBaseDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var GridColumnOptionsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(GridColumnOptionsDescription, _super);
    function GridColumnOptionsDescription() {
        var _this = _super.call(this) || this;
        _this.bn = null;
        _this.a6 = 0;
        _this.bo = null;
        _this.bp = null;
        _this.a9 = 0;
        _this.bb = 0;
        _this.ba = 0;
        _this.a8 = 0;
        _this.as = false;
        _this.aw = false;
        _this.au = false;
        _this.av = false;
        _this.at = false;
        _this.ar = false;
        _this.aq = false;
        _this.ax = false;
        _this.b0 = null;
        _this.by = null;
        _this.bz = null;
        _this.bs = null;
        _this.bt = null;
        _this.bu = null;
        _this.bv = null;
        _this.bw = null;
        _this.bx = null;
        _this.bj = null;
        _this.bl = null;
        _this.bm = null;
        _this.b1 = null;
        _this.bh = null;
        _this.bq = null;
        _this.bg = null;
        _this.br = null;
        _this.bi = null;
        _this.bk = null;
        return _this;
    }
    GridColumnOptionsDescription.prototype.get_type = function () {
        return "GridColumnOptions";
    };
    Object.defineProperty(GridColumnOptionsDescription.prototype, "columnNameFontFamily", {
        get: function () {
            return this.bn;
        },
        set: function (a) {
            this.bn = a;
            this.e("ColumnNameFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "columnNameFontSize", {
        get: function () {
            return this.a6;
        },
        set: function (a) {
            this.a6 = a;
            this.e("ColumnNameFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "columnNameFontStyle", {
        get: function () {
            return this.bo;
        },
        set: function (a) {
            this.bo = a;
            this.e("ColumnNameFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "columnNameFontWeight", {
        get: function () {
            return this.bp;
        },
        set: function (a) {
            this.bp = a;
            this.e("ColumnNameFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "paddingLeft", {
        get: function () {
            return this.a9;
        },
        set: function (a) {
            this.a9 = a;
            this.e("PaddingLeft");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "paddingTop", {
        get: function () {
            return this.bb;
        },
        set: function (a) {
            this.bb = a;
            this.e("PaddingTop");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "paddingRight", {
        get: function () {
            return this.ba;
        },
        set: function (a) {
            this.ba = a;
            this.e("PaddingRight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "paddingBottom", {
        get: function () {
            return this.a8;
        },
        set: function (a) {
            this.a8 = a;
            this.e("PaddingBottom");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "headerVisible", {
        get: function () {
            return this.as;
        },
        set: function (a) {
            this.as = a;
            this.e("HeaderVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "sortOptionsVisible", {
        get: function () {
            return this.aw;
        },
        set: function (a) {
            this.aw = a;
            this.e("SortOptionsVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "moveOptionsVisible", {
        get: function () {
            return this.au;
        },
        set: function (a) {
            this.au = a;
            this.e("MoveOptionsVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "pinOptionsVisible", {
        get: function () {
            return this.av;
        },
        set: function (a) {
            this.av = a;
            this.e("PinOptionsVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "hideOptionsVisible", {
        get: function () {
            return this.at;
        },
        set: function (a) {
            this.at = a;
            this.e("HideOptionsVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "groupOptionsVisible", {
        get: function () {
            return this.ar;
        },
        set: function (a) {
            this.ar = a;
            this.e("GroupOptionsVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "filterOptionsVisible", {
        get: function () {
            return this.aq;
        },
        set: function (a) {
            this.aq = a;
            this.e("FilterOptionsVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "summaryOptionsVisible", {
        get: function () {
            return this.ax;
        },
        set: function (a) {
            this.ax = a;
            this.e("SummaryOptionsVisible");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "sortHeaderCaption", {
        get: function () {
            return this.b0;
        },
        set: function (a) {
            this.b0 = a;
            this.e("SortHeaderCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "sortAscendingCaption", {
        get: function () {
            return this.by;
        },
        set: function (a) {
            this.by = a;
            this.e("SortAscendingCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "sortDescendingCaption", {
        get: function () {
            return this.bz;
        },
        set: function (a) {
            this.bz = a;
            this.e("SortDescendingCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "moveHeaderCaption", {
        get: function () {
            return this.bs;
        },
        set: function (a) {
            this.bs = a;
            this.e("MoveHeaderCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "moveLeftCaption", {
        get: function () {
            return this.bt;
        },
        set: function (a) {
            this.bt = a;
            this.e("MoveLeftCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "moveRightCaption", {
        get: function () {
            return this.bu;
        },
        set: function (a) {
            this.bu = a;
            this.e("MoveRightCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "pinHeaderCaption", {
        get: function () {
            return this.bv;
        },
        set: function (a) {
            this.bv = a;
            this.e("PinHeaderCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "pinLeftCaption", {
        get: function () {
            return this.bw;
        },
        set: function (a) {
            this.bw = a;
            this.e("PinLeftCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "pinRightCaption", {
        get: function () {
            return this.bx;
        },
        set: function (a) {
            this.bx = a;
            this.e("PinRightCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "applyFiltersButtonCaption", {
        get: function () {
            return this.bj;
        },
        set: function (a) {
            this.bj = a;
            this.e("ApplyFiltersButtonCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "cancelFiltersButtonCaption", {
        get: function () {
            return this.bl;
        },
        set: function (a) {
            this.bl = a;
            this.e("CancelFiltersButtonCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "clearColumnFiltersCaption", {
        get: function () {
            return this.bm;
        },
        set: function (a) {
            this.bm = a;
            this.e("ClearColumnFiltersCaption");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "summaryListDensity", {
        get: function () {
            return this.b1;
        },
        set: function (a) {
            this.b1 = a;
            this.e("SummaryListDensity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "actualSummaryListDensity", {
        get: function () {
            return this.bh;
        },
        set: function (a) {
            this.bh = a;
            this.e("ActualSummaryListDensity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "filterListDensity", {
        get: function () {
            return this.bq;
        },
        set: function (a) {
            this.bq = a;
            this.e("FilterListDensity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "actualFilterListDensity", {
        get: function () {
            return this.bg;
        },
        set: function (a) {
            this.bg = a;
            this.e("ActualFilterListDensity");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "filterListPlaceholderText", {
        get: function () {
            return this.br;
        },
        set: function (a) {
            this.br = a;
            this.e("FilterListPlaceholderText");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "applyButtonClickRef", {
        get: function () {
            return this.bi;
        },
        set: function (a) {
            this.bi = a;
            this.e("ApplyButtonClickRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(GridColumnOptionsDescription.prototype, "cancelButtonClickRef", {
        get: function () {
            return this.bk;
        },
        set: function (a) {
            this.bk = a;
            this.e("CancelButtonClickRef");
        },
        enumerable: false,
        configurable: true
    });
    GridColumnOptionsDescription.$t = markType(GridColumnOptionsDescription, 'GridColumnOptionsDescription', GridColumnOptionsBaseDescription.$);
    return GridColumnOptionsDescription;
}(GridColumnOptionsBaseDescription));
export { GridColumnOptionsDescription };
