// igniteui-react-core/esm5/lib/JsonDictionaryObject.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { JsonDictionaryItem } from "./JsonDictionaryItem";
import { Dictionary$2 } from "./Dictionary$2";
import { List$1 } from "./List$1";
import { String_$type, fromEnum, markType } from "./type";
/**
 * @hidden
 */
var JsonDictionaryObject = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(JsonDictionaryObject, _super);
    function JsonDictionaryObject() {
        var _this = _super.call(this) || this;
        _this.h = new Dictionary$2(String_$type, JsonDictionaryItem.$, 0);
        return _this;
    }
    JsonDictionaryObject.prototype.i = function (a, b) {
        this.h.item(a, b);
    };
    JsonDictionaryObject.prototype.k = function (a) {
        this.h.removeItem(a);
    };
    JsonDictionaryObject.prototype.j = function () {
        this.h.clear();
    };
    JsonDictionaryObject.prototype.item = function (a, b) {
        if (arguments.length === 2) {
            this.h.item(a, b);
            return b;
        }
        else {
            return this.h.item(a);
        }
    };
    JsonDictionaryObject.prototype.g = function (a) {
        return this.h.containsKey(a);
    };
    JsonDictionaryObject.prototype.e = function () {
        var e_1, _a;
        var a = new List$1(String_$type, 0);
        try {
            for (var _b = __values(fromEnum(this.h.keys)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var b = _c.value;
                a.add(b);
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        return a.toArray();
    };
    JsonDictionaryObject.prototype.d = function (a) {
        a.i("{");
        a.h();
        a.g();
        var b = this.e();
        if (b != null) {
            for (var c = 0; c < b.length; c++) {
                if (c > 0) {
                    a.j(",");
                }
                var d = b[c];
                var e = this.h.item(d);
                a.i("\"" + this.a(d) + "\": ");
                if (e == null) {
                    a.i("null");
                }
                else {
                    e.d(a);
                }
            }
        }
        a.h();
        a.e();
        a.i("}");
    };
    JsonDictionaryObject.$t = markType(JsonDictionaryObject, 'JsonDictionaryObject', JsonDictionaryItem.$);
    return JsonDictionaryObject;
}(JsonDictionaryItem));
export { JsonDictionaryObject };
