// igniteui-react-core/esm5/lib/ListPanelDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var ListPanelDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ListPanelDescription, _super);
    function ListPanelDescription() {
        var _this = _super.call(this) || this;
        _this.u = null;
        _this.f = null;
        _this.n = 0;
        _this.j = 0;
        _this.q = null;
        _this.k = 0;
        _this.r = null;
        _this.s = null;
        _this.p = null;
        _this.x = null;
        _this.h = false;
        _this.z = null;
        _this.t = null;
        _this.y = null;
        _this.w = null;
        _this.v = null;
        return _this;
    }
    ListPanelDescription.prototype.get_type = function () {
        return "ListPanel";
    };
    Object.defineProperty(ListPanelDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "dataSourceRef", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            this.u = a;
            this.e("DataSourceRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "dataSource", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("DataSource");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "rowHeight", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("RowHeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "actualRowHeight", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("ActualRowHeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "cellFontFamily", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("CellFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "cellFontSize", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("CellFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "cellFontStyle", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("CellFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "cellFontWeight", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("CellFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "backgroundColor", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("BackgroundColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "orientation", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            this.x = a;
            this.e("Orientation");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "isCustomRowHeightEnabled", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("IsCustomRowHeightEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "textColor", {
        get: function () {
            return this.z;
        },
        set: function (a) {
            this.z = a;
            this.e("TextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "contentRefreshedRef", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
            this.e("ContentRefreshedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "rowUpdatingRef", {
        get: function () {
            return this.y;
        },
        set: function (a) {
            this.y = a;
            this.e("RowUpdatingRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "itemWidthRequestedRef", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            this.w = a;
            this.e("ItemWidthRequestedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ListPanelDescription.prototype, "itemHeightRequestedRef", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            this.v = a;
            this.e("ItemHeightRequestedRef");
        },
        enumerable: false,
        configurable: true
    });
    ListPanelDescription.$t = markType(ListPanelDescription, 'ListPanelDescription', Description.$);
    return ListPanelDescription;
}(Description));
export { ListPanelDescription };
