// igniteui-react-core/esm5/lib/NativeUIElementBridge.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, markType } from "./type";
import { INativeUIElementBridge_$type } from "./INativeUIElementBridge";
import { Size } from "./Size";
/**
 * @hidden
 */
var NativeUIElementBridge = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(NativeUIElementBridge, _super);
    function NativeUIElementBridge() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.b = 0;
        _this.a = 0;
        return _this;
    }
    NativeUIElementBridge.prototype.getDesiredSize = function (a, b) {
        var c = (a);
        return new Size(1, c.offsetWidth, c.offsetHeight);
    };
    NativeUIElementBridge.prototype.getMethodTarget = function (a) {
        return a;
    };
    NativeUIElementBridge.prototype.getTargetForProperty = function (a, b) {
        return b;
    };
    NativeUIElementBridge.prototype.addHandler = function (a, b, c, d) {
    };
    NativeUIElementBridge.prototype.getHorizontalAlignment = function (a) {
        return this.a;
    };
    NativeUIElementBridge.prototype.getTargetForBuildInEvent = function (a, b) {
        return a;
    };
    NativeUIElementBridge.prototype.getTargetForBuildInProperty = function (a, b) {
        return a;
    };
    NativeUIElementBridge.prototype.getValue = function (a, b) {
        return null;
    };
    NativeUIElementBridge.prototype.getVerticalAlignment = function (a) {
        return this.b;
    };
    NativeUIElementBridge.prototype.removeHandler = function (a, b, c, d) {
    };
    NativeUIElementBridge.prototype.setHorizontalAlignment = function (a, b) {
        this.a = b;
        var c = a;
        switch (b) {
            case 0:
                c.setStyleProperty("justify-self", "start");
                c.setStyleProperty("text-align", "left");
                break;
            case 1:
                c.setStyleProperty("justify-self", "center");
                c.setStyleProperty("text-align", "center");
                break;
            case 2:
                c.setStyleProperty("justify-self", "end");
                c.setStyleProperty("text-align", "right");
                break;
            case 3:
                c.setStyleProperty("justify-self", "stretch");
                c.setStyleProperty("text-align", "center");
                break;
        }
    };
    NativeUIElementBridge.prototype.setValue = function (a, b, c) {
    };
    NativeUIElementBridge.prototype.setVerticalAlignment = function (a, b) {
        this.b = b;
        var c = a;
        switch (b) {
            case 0:
                c.setStyleProperty("align-self", "start");
                c.setStyleProperty("valign", "top");
                break;
            case 1:
                c.setStyleProperty("align-self", "center");
                c.setStyleProperty("valign", "center");
                break;
            case 2:
                c.setStyleProperty("align-self", "end");
                c.setStyleProperty("valign", "bottom");
                break;
            case 3:
                c.setStyleProperty("align-self", "stretch");
                c.setStyleProperty("valign", "center");
                break;
        }
    };
    NativeUIElementBridge.$t = markType(NativeUIElementBridge, 'NativeUIElementBridge', Base.$, [INativeUIElementBridge_$type]);
    return NativeUIElementBridge;
}(Base));
export { NativeUIElementBridge };
