// igniteui-react-core/esm5/lib/OperationFilterExpressionDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { FilterExpressionDescription } from "./FilterExpressionDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var OperationFilterExpressionDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(OperationFilterExpressionDescription, _super);
    function OperationFilterExpressionDescription() {
        var _this = _super.call(this) || this;
        _this.p = null;
        _this.q = null;
        _this.i = null;
        _this.j = null;
        _this.o = null;
        return _this;
    }
    OperationFilterExpressionDescription.prototype.get_type = function () {
        return "OperationFilterExpression";
    };
    Object.defineProperty(OperationFilterExpressionDescription.prototype, "propertyName", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("PropertyName");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OperationFilterExpressionDescription.prototype, "valueRef", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("ValueRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OperationFilterExpressionDescription.prototype, "left", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("Left");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OperationFilterExpressionDescription.prototype, "right", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Right");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OperationFilterExpressionDescription.prototype, "operator", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("Operator");
        },
        enumerable: false,
        configurable: true
    });
    OperationFilterExpressionDescription.$t = markType(OperationFilterExpressionDescription, 'OperationFilterExpressionDescription', FilterExpressionDescription.$);
    OperationFilterExpressionDescription.__marshalByValue1 = true;
    OperationFilterExpressionDescription.__marshalByValueAlias1 = "OperationFilterExpression";
    return OperationFilterExpressionDescription;
}(FilterExpressionDescription));
export { OperationFilterExpressionDescription };
