// igniteui-react-core/esm5/lib/OrdinalTimeXAxisDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { CategoryXAxisDescription } from "./CategoryXAxisDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var OrdinalTimeXAxisDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(OrdinalTimeXAxisDescription, _super);
    function OrdinalTimeXAxisDescription() {
        var _this = _super.call(this) || this;
        _this.ed = null;
        _this.ee = null;
        _this.d7 = null;
        _this.ea = new Date();
        _this.d9 = new Date();
        return _this;
    }
    OrdinalTimeXAxisDescription.prototype.get_type = function () {
        return "OrdinalTimeXAxis";
    };
    Object.defineProperty(OrdinalTimeXAxisDescription.prototype, "dateTimeMemberPath", {
        get: function () {
            return this.ed;
        },
        set: function (a) {
            this.ed = a;
            this.e("DateTimeMemberPath");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OrdinalTimeXAxisDescription.prototype, "labellingMode", {
        get: function () {
            return this.ee;
        },
        set: function (a) {
            this.ee = a;
            this.e("LabellingMode");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OrdinalTimeXAxisDescription.prototype, "labelFormats", {
        get: function () {
            return this.d7;
        },
        set: function (a) {
            this.d7 = a;
            this.e("LabelFormats");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OrdinalTimeXAxisDescription.prototype, "minimumValue", {
        get: function () {
            return this.ea;
        },
        set: function (a) {
            this.ea = a;
            this.e("MinimumValue");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OrdinalTimeXAxisDescription.prototype, "maximumValue", {
        get: function () {
            return this.d9;
        },
        set: function (a) {
            this.d9 = a;
            this.e("MaximumValue");
        },
        enumerable: false,
        configurable: true
    });
    OrdinalTimeXAxisDescription.$t = markType(OrdinalTimeXAxisDescription, 'OrdinalTimeXAxisDescription', CategoryXAxisDescription.$);
    return OrdinalTimeXAxisDescription;
}(CategoryXAxisDescription));
export { OrdinalTimeXAxisDescription };
