// igniteui-react-core/esm5/lib/PanelBridge.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { NativeUIElementBridge } from "./NativeUIElementBridge";
import { List$1 } from "./List$1";
import { NativeUIComponent } from "./NativeUIComponent";
import { markType } from "./type";
/**
 * @hidden
 */
var PanelBridge = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(PanelBridge, _super);
    function PanelBridge() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.d = new List$1(NativeUIComponent.$, 0);
        return _this;
    }
    PanelBridge.prototype.f = function (a, b) {
        this.d.add(b);
        a.append(b.v);
    };
    PanelBridge.prototype.c = function (a, b) {
        return this.d._inner[b];
    };
    PanelBridge.prototype.e = function (a) {
        return a.getChildCount();
    };
    PanelBridge.prototype.g = function (a, b) {
        this.d.remove(b);
        a.removeChild(b.v);
    };
    PanelBridge.prototype.h = function (a, b) {
        this.d.removeAt(b);
        a.getChildAt(b).remove();
    };
    PanelBridge.$t = markType(PanelBridge, 'PanelBridge', NativeUIElementBridge.$);
    return PanelBridge;
}(NativeUIElementBridge));
export { PanelBridge };
