// igniteui-react-core/esm5/lib/PlotAreaMouseEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var PlotAreaMouseEventArgsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(PlotAreaMouseEventArgsDescription, _super);
    function PlotAreaMouseEventArgsDescription() {
        var _this = _super.call(this) || this;
        _this.f = null;
        _this.g = null;
        _this.l = false;
        _this.j = null;
        return _this;
    }
    PlotAreaMouseEventArgsDescription.prototype.get_type = function () {
        return "PlotAreaMouseEventArgs";
    };
    Object.defineProperty(PlotAreaMouseEventArgsDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PlotAreaMouseEventArgsDescription.prototype, "chartPosition", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("ChartPosition");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PlotAreaMouseEventArgsDescription.prototype, "plotAreaPosition", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("PlotAreaPosition");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PlotAreaMouseEventArgsDescription.prototype, "isDuringManipulation", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("IsDuringManipulation");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PlotAreaMouseEventArgsDescription.prototype, "viewer", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Viewer");
        },
        enumerable: false,
        configurable: true
    });
    PlotAreaMouseEventArgsDescription.$t = markType(PlotAreaMouseEventArgsDescription, 'PlotAreaMouseEventArgsDescription', Description.$);
    return PlotAreaMouseEventArgsDescription;
}(Description));
export { PlotAreaMouseEventArgsDescription };
