// igniteui-react-core/esm5/lib/PropertyEditorDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var PropertyEditorDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(PropertyEditorDescription, _super);
    function PropertyEditorDescription() {
        var _this = _super.call(this) || this;
        _this.f = null;
        _this.g = null;
        _this.ad = null;
        _this.af = null;
        _this.r = 0;
        _this.w = null;
        _this.p = 0;
        _this.x = null;
        _this.y = null;
        _this.v = null;
        _this.aa = null;
        _this.ab = null;
        _this.z = null;
        _this.ag = null;
        _this.ac = null;
        _this.j = false;
        _this.l = false;
        _this.k = false;
        _this.u = null;
        _this.ah = null;
        _this.ai = null;
        _this.ae = null;
        _this.t = null;
        return _this;
    }
    PropertyEditorDescription.prototype.get_type = function () {
        return "PropertyEditor";
    };
    Object.defineProperty(PropertyEditorDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "actualDataSource", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("ActualDataSource");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "properties", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Properties");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "filterPlaceholderText", {
        get: function () {
            return this.ad;
        },
        set: function (a) {
            this.ad = a;
            this.e("FilterPlaceholderText");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "searchInputType", {
        get: function () {
            return this.af;
        },
        set: function (a) {
            this.af = a;
            this.e("SearchInputType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "rowHeight", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("RowHeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "cellFontFamily", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            this.w = a;
            this.e("CellFontFamily");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "cellFontSize", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("CellFontSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "cellFontStyle", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            this.x = a;
            this.e("CellFontStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "cellFontWeight", {
        get: function () {
            return this.y;
        },
        set: function (a) {
            this.y = a;
            this.e("CellFontWeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "baseTheme", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            this.v = a;
            this.e("BaseTheme");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "density", {
        get: function () {
            return this.aa;
        },
        set: function (a) {
            this.aa = a;
            this.e("Density");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "descriptionContextRef", {
        get: function () {
            return this.ab;
        },
        set: function (a) {
            this.ab = a;
            this.e("DescriptionContextRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "componentRendererRef", {
        get: function () {
            return this.z;
        },
        set: function (a) {
            this.z = a;
            this.e("ComponentRendererRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "targetRef", {
        get: function () {
            return this.ag;
        },
        set: function (a) {
            this.ag = a;
            this.e("TargetRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "descriptionType", {
        get: function () {
            return this.ac;
        },
        set: function (a) {
            this.ac = a;
            this.e("DescriptionType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "isHorizontal", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("IsHorizontal");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "isWrappingEnabled", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("IsWrappingEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "isIndirectModeEnabled", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("IsIndirectModeEnabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "backgroundColor", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            this.u = a;
            this.e("BackgroundColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "textColor", {
        get: function () {
            return this.ah;
        },
        set: function (a) {
            this.ah = a;
            this.e("TextColor");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "width", {
        get: function () {
            return this.ai;
        },
        set: function (a) {
            this.ai = a;
            this.e("Width");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "height", {
        get: function () {
            return this.ae;
        },
        set: function (a) {
            this.ae = a;
            this.e("Height");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(PropertyEditorDescription.prototype, "background", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
            this.e("Background");
        },
        enumerable: false,
        configurable: true
    });
    PropertyEditorDescription.$t = markType(PropertyEditorDescription, 'PropertyEditorDescription', Description.$);
    return PropertyEditorDescription;
}(Description));
export { PropertyEditorDescription };
