// igniteui-react-core/esm5/lib/StackedSeriesCreatedEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var StackedSeriesCreatedEventArgsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(StackedSeriesCreatedEventArgsDescription, _super);
    function StackedSeriesCreatedEventArgsDescription() {
        var _this = _super.call(this) || this;
        _this.n = null;
        _this.p = null;
        _this.o = null;
        _this.q = null;
        _this.v = null;
        _this.i = 0;
        _this.w = null;
        _this.l = 0;
        _this.x = null;
        _this.r = null;
        _this.s = null;
        _this.f = null;
        _this.t = null;
        _this.u = null;
        _this.h = 0;
        return _this;
    }
    StackedSeriesCreatedEventArgsDescription.prototype.get_type = function () {
        return "StackedSeriesCreatedEventArgs";
    };
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "brush", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("Brush");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "legendItemTemplateRef", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("LegendItemTemplateRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "legendItemBadgeTemplateRef", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("LegendItemBadgeTemplateRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "legendItemVisibility", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("LegendItemVisibility");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "outline", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            this.v = a;
            this.e("Outline");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "thickness", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("Thickness");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "titleRef", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            this.w = a;
            this.e("TitleRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "transitionDuration", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("TransitionDuration");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "transitionEasingFunctionRef", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            this.x = a;
            this.e("TransitionEasingFunctionRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "markerBrush", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("MarkerBrush");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "markerOutline", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("MarkerOutline");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "markerStyle", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("MarkerStyle");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "markerTemplateRef", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
            this.e("MarkerTemplateRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "markerType", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            this.u = a;
            this.e("MarkerType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StackedSeriesCreatedEventArgsDescription.prototype, "markerThickness", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("MarkerThickness");
        },
        enumerable: false,
        configurable: true
    });
    StackedSeriesCreatedEventArgsDescription.$t = markType(StackedSeriesCreatedEventArgsDescription, 'StackedSeriesCreatedEventArgsDescription', Description.$);
    return StackedSeriesCreatedEventArgsDescription;
}(Description));
export { StackedSeriesCreatedEventArgsDescription };
