// igniteui-react-core/esm5/lib/TimeAxisLabelFormatDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var TimeAxisLabelFormatDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(TimeAxisLabelFormatDescription, _super);
    function TimeAxisLabelFormatDescription() {
        var _this = _super.call(this) || this;
        _this.f = 0;
        _this.h = null;
        _this.k = null;
        _this.j = null;
        _this.i = null;
        return _this;
    }
    TimeAxisLabelFormatDescription.prototype.get_type = function () {
        return "TimeAxisLabelFormat";
    };
    Object.defineProperty(TimeAxisLabelFormatDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(TimeAxisLabelFormatDescription.prototype, "range", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Range");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(TimeAxisLabelFormatDescription.prototype, "format", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Format");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(TimeAxisLabelFormatDescription.prototype, "repeatedYearFormat", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("RepeatedYearFormat");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(TimeAxisLabelFormatDescription.prototype, "repeatedMonthFormat", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("RepeatedMonthFormat");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(TimeAxisLabelFormatDescription.prototype, "repeatedDayFormat", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("RepeatedDayFormat");
        },
        enumerable: false,
        configurable: true
    });
    TimeAxisLabelFormatDescription.$t = markType(TimeAxisLabelFormatDescription, 'TimeAxisLabelFormatDescription', Description.$);
    return TimeAxisLabelFormatDescription;
}(Description));
export { TimeAxisLabelFormatDescription };
