// igniteui-react-core/esm5/lib/WebContentPaneDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebDockManagerPaneDescription } from "./WebDockManagerPaneDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebContentPaneDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebContentPaneDescription, _super);
    function WebContentPaneDescription() {
        var _this = _super.call(this) || this;
        _this.ai = null;
        _this.aj = null;
        _this.ae = null;
        _this.ag = null;
        _this.ah = null;
        _this.ak = null;
        _this.al = null;
        _this.af = null;
        _this.z = 0;
        _this.h = false;
        _this.k = false;
        _this.l = false;
        _this.i = false;
        _this.j = false;
        _this.aa = 0;
        _this.p = false;
        _this.am = null;
        _this.o = false;
        _this.m = false;
        _this.n = false;
        return _this;
    }
    WebContentPaneDescription.prototype.get_type = function () {
        return "WebContentPane";
    };
    Object.defineProperty(WebContentPaneDescription.prototype, "id", {
        get: function () {
            return this.ai;
        },
        set: function (a) {
            this.ai = a;
            this.e("Id");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "paneType", {
        get: function () {
            return this.aj;
        },
        set: function (a) {
            this.aj = a;
            this.e("PaneType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "contentId", {
        get: function () {
            return this.ae;
        },
        set: function (a) {
            this.ae = a;
            this.e("ContentId");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "header", {
        get: function () {
            return this.ag;
        },
        set: function (a) {
            this.ag = a;
            this.e("Header");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "headerId", {
        get: function () {
            return this.ah;
        },
        set: function (a) {
            this.ah = a;
            this.e("HeaderId");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "tabHeaderId", {
        get: function () {
            return this.ak;
        },
        set: function (a) {
            this.ak = a;
            this.e("TabHeaderId");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "unpinnedHeaderId", {
        get: function () {
            return this.al;
        },
        set: function (a) {
            this.al = a;
            this.e("UnpinnedHeaderId");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "floatingHeaderId", {
        get: function () {
            return this.af;
        },
        set: function (a) {
            this.af = a;
            this.e("FloatingHeaderId");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "size", {
        get: function () {
            return this.z;
        },
        set: function (a) {
            this.z = a;
            this.e("Size");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "allowClose", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("AllowClose");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "allowMaximize", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("AllowMaximize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "allowPinning", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("AllowPinning");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "allowDocking", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("AllowDocking");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "allowFloating", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("AllowFloating");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "unpinnedSize", {
        get: function () {
            return this.aa;
        },
        set: function (a) {
            this.aa = a;
            this.e("UnpinnedSize");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "isPinned", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("IsPinned");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "unpinnedLocation", {
        get: function () {
            return this.am;
        },
        set: function (a) {
            this.am = a;
            this.e("UnpinnedLocation");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "hidden", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("Hidden");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "disabled", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("Disabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebContentPaneDescription.prototype, "documentOnly", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("DocumentOnly");
        },
        enumerable: false,
        configurable: true
    });
    WebContentPaneDescription.$t = markType(WebContentPaneDescription, 'WebContentPaneDescription', WebDockManagerPaneDescription.$);
    WebContentPaneDescription.__marshalByValue = true;
    WebContentPaneDescription.__marshalByValueAlias = "ContentPane";
    return WebContentPaneDescription;
}(WebDockManagerPaneDescription));
export { WebContentPaneDescription };
