// igniteui-react-core/esm5/lib/WebCalendarDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebCalendarBaseDescription } from "./WebCalendarBaseDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebCalendarDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebCalendarDescription, _super);
    function WebCalendarDescription() {
        var _this = _super.call(this) || this;
        _this.ad = new Date();
        _this.s = null;
        _this.ac = new Date();
        _this.z = false;
        _this.y = false;
        _this.ak = null;
        _this.al = null;
        _this.ag = 0;
        _this.ai = null;
        _this.u = null;
        _this.w = null;
        _this.am = null;
        _this.aj = null;
        return _this;
    }
    WebCalendarDescription.prototype.get_type = function () {
        return "WebCalendar";
    };
    Object.defineProperty(WebCalendarDescription.prototype, "value", {
        get: function () {
            return this.ad;
        },
        set: function (a) {
            this.ad = a;
            this.e("Value");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "values", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("Values");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "activeDate", {
        get: function () {
            return this.ac;
        },
        set: function (a) {
            this.ac = a;
            this.e("ActiveDate");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "hideOutsideDays", {
        get: function () {
            return this.z;
        },
        set: function (a) {
            this.z = a;
            this.e("HideOutsideDays");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "hideHeader", {
        get: function () {
            return this.y;
        },
        set: function (a) {
            this.y = a;
            this.e("HideHeader");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "headerOrientation", {
        get: function () {
            return this.ak;
        },
        set: function (a) {
            this.ak = a;
            this.e("HeaderOrientation");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "orientation", {
        get: function () {
            return this.al;
        },
        set: function (a) {
            this.al = a;
            this.e("Orientation");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "visibleMonths", {
        get: function () {
            return this.ag;
        },
        set: function (a) {
            this.ag = a;
            this.e("VisibleMonths");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "activeView", {
        get: function () {
            return this.ai;
        },
        set: function (a) {
            this.ai = a;
            this.e("ActiveView");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "formatOptions", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            this.u = a;
            this.e("FormatOptions");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "resourceStrings", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            this.w = a;
            this.e("ResourceStrings");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "size", {
        get: function () {
            return this.am;
        },
        set: function (a) {
            this.am = a;
            this.e("Size");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebCalendarDescription.prototype, "changeRef", {
        get: function () {
            return this.aj;
        },
        set: function (a) {
            this.aj = a;
            this.e("ChangeRef");
        },
        enumerable: false,
        configurable: true
    });
    WebCalendarDescription.$t = markType(WebCalendarDescription, 'WebCalendarDescription', WebCalendarBaseDescription.$);
    return WebCalendarDescription;
}(WebCalendarBaseDescription));
export { WebCalendarDescription };
