// igniteui-react-core/esm5/lib/WebDropdownItemDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebDropdownItemDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebDropdownItemDescription, _super);
    function WebDropdownItemDescription() {
        var _this = _super.call(this) || this;
        _this.l = null;
        _this.h = false;
        _this.f = false;
        _this.g = false;
        return _this;
    }
    WebDropdownItemDescription.prototype.get_type = function () {
        return "WebDropdownItem";
    };
    Object.defineProperty(WebDropdownItemDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebDropdownItemDescription.prototype, "value", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("Value");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebDropdownItemDescription.prototype, "selected", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Selected");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebDropdownItemDescription.prototype, "active", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Active");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebDropdownItemDescription.prototype, "disabled", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Disabled");
        },
        enumerable: false,
        configurable: true
    });
    WebDropdownItemDescription.$t = markType(WebDropdownItemDescription, 'WebDropdownItemDescription', Description.$);
    return WebDropdownItemDescription;
}(Description));
export { WebDropdownItemDescription };
