// igniteui-react-core/esm5/lib/WebDocumentHostDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebDockManagerPaneDescription } from "./WebDockManagerPaneDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebDocumentHostDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebDocumentHostDescription, _super);
    function WebDocumentHostDescription() {
        var _this = _super.call(this) || this;
        _this.m = null;
        _this.n = null;
        _this.g = null;
        _this.j = 0;
        return _this;
    }
    WebDocumentHostDescription.prototype.get_type = function () {
        return "WebDocumentHost";
    };
    Object.defineProperty(WebDocumentHostDescription.prototype, "id", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("Id");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebDocumentHostDescription.prototype, "paneType", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("PaneType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebDocumentHostDescription.prototype, "rootPane", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("RootPane");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebDocumentHostDescription.prototype, "size", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Size");
        },
        enumerable: false,
        configurable: true
    });
    WebDocumentHostDescription.$t = markType(WebDocumentHostDescription, 'WebDocumentHostDescription', WebDockManagerPaneDescription.$);
    WebDocumentHostDescription.__marshalByValue = true;
    WebDocumentHostDescription.__marshalByValueAlias = "DocumentHost";
    return WebDocumentHostDescription;
}(WebDockManagerPaneDescription));
export { WebDocumentHostDescription };
