// igniteui-react-core/esm5/lib/WebExpansionPanelDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebExpansionPanelDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebExpansionPanelDescription, _super);
    function WebExpansionPanelDescription() {
        var _this = _super.call(this) || this;
        _this.g = false;
        _this.f = false;
        _this.l = null;
        _this.n = null;
        _this.m = null;
        _this.k = null;
        _this.j = null;
        return _this;
    }
    WebExpansionPanelDescription.prototype.get_type = function () {
        return "WebExpansionPanel";
    };
    Object.defineProperty(WebExpansionPanelDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebExpansionPanelDescription.prototype, "open", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Open");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebExpansionPanelDescription.prototype, "disabled", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Disabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebExpansionPanelDescription.prototype, "indicatorPosition", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("IndicatorPosition");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebExpansionPanelDescription.prototype, "openingRef", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
            this.e("OpeningRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebExpansionPanelDescription.prototype, "openedRef", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("OpenedRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebExpansionPanelDescription.prototype, "closingRef", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("ClosingRef");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebExpansionPanelDescription.prototype, "closedRef", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("ClosedRef");
        },
        enumerable: false,
        configurable: true
    });
    WebExpansionPanelDescription.$t = markType(WebExpansionPanelDescription, 'WebExpansionPanelDescription', Description.$);
    return WebExpansionPanelDescription;
}(Description));
export { WebExpansionPanelDescription };
