// igniteui-react-core/esm5/lib/WebFloatingPaneResizeMoveEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebFloatingPaneResizeEventArgsDescription } from "./WebFloatingPaneResizeEventArgsDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebFloatingPaneResizeMoveEventArgsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebFloatingPaneResizeMoveEventArgsDescription, _super);
    function WebFloatingPaneResizeMoveEventArgsDescription() {
        var _this = _super.call(this) || this;
        _this.r = 0;
        _this.p = 0;
        _this.q = 0;
        _this.o = 0;
        _this.l = null;
        _this.k = null;
        return _this;
    }
    WebFloatingPaneResizeMoveEventArgsDescription.prototype.get_type = function () {
        return "WebFloatingPaneResizeMoveEventArgs";
    };
    Object.defineProperty(WebFloatingPaneResizeMoveEventArgsDescription.prototype, "oldWidth", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("OldWidth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFloatingPaneResizeMoveEventArgsDescription.prototype, "newWidth", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("NewWidth");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFloatingPaneResizeMoveEventArgsDescription.prototype, "oldHeight", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("OldHeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFloatingPaneResizeMoveEventArgsDescription.prototype, "newHeight", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
            this.e("NewHeight");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFloatingPaneResizeMoveEventArgsDescription.prototype, "oldLocation", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("OldLocation");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebFloatingPaneResizeMoveEventArgsDescription.prototype, "newLocation", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("NewLocation");
        },
        enumerable: false,
        configurable: true
    });
    WebFloatingPaneResizeMoveEventArgsDescription.$t = markType(WebFloatingPaneResizeMoveEventArgsDescription, 'WebFloatingPaneResizeMoveEventArgsDescription', WebFloatingPaneResizeEventArgsDescription.$);
    return WebFloatingPaneResizeMoveEventArgsDescription;
}(WebFloatingPaneResizeEventArgsDescription));
export { WebFloatingPaneResizeMoveEventArgsDescription };
