// igniteui-react-core/esm5/lib/WebPaneDragOverEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPaneDragOverEventArgsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPaneDragOverEventArgsDescription, _super);
    function WebPaneDragOverEventArgsDescription() {
        var _this = _super.call(this) || this;
        _this.h = null;
        _this.f = null;
        _this.j = null;
        _this.l = false;
        return _this;
    }
    WebPaneDragOverEventArgsDescription.prototype.get_type = function () {
        return "WebPaneDragOverEventArgs";
    };
    Object.defineProperty(WebPaneDragOverEventArgsDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaneDragOverEventArgsDescription.prototype, "sourcePane", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("SourcePane");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaneDragOverEventArgsDescription.prototype, "panes", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Panes");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaneDragOverEventArgsDescription.prototype, "action", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Action");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPaneDragOverEventArgsDescription.prototype, "isValid", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("IsValid");
        },
        enumerable: false,
        configurable: true
    });
    WebPaneDragOverEventArgsDescription.$t = markType(WebPaneDragOverEventArgsDescription, 'WebPaneDragOverEventArgsDescription', Description.$);
    return WebPaneDragOverEventArgsDescription;
}(Description));
export { WebPaneDragOverEventArgsDescription };
