// igniteui-react-core/esm5/lib/WebProgressBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebProgressBaseDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebProgressBaseDescription, _super);
    function WebProgressBaseDescription() {
        var _this = _super.call(this) || this;
        _this.k = 0;
        _this.l = 0;
        _this.q = null;
        _this.j = 0;
        _this.g = false;
        _this.f = false;
        _this.p = null;
        return _this;
    }
    WebProgressBaseDescription.prototype.get_type = function () {
        return "WebProgressBase";
    };
    Object.defineProperty(WebProgressBaseDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebProgressBaseDescription.prototype, "max", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("Max");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebProgressBaseDescription.prototype, "value", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("Value");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebProgressBaseDescription.prototype, "variant", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("Variant");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebProgressBaseDescription.prototype, "animationDuration", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("AnimationDuration");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebProgressBaseDescription.prototype, "indeterminate", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Indeterminate");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebProgressBaseDescription.prototype, "hideLabel", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("HideLabel");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebProgressBaseDescription.prototype, "labelFormat", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
            this.e("LabelFormat");
        },
        enumerable: false,
        configurable: true
    });
    WebProgressBaseDescription.$t = markType(WebProgressBaseDescription, 'WebProgressBaseDescription', Description.$);
    return WebProgressBaseDescription;
}(Description));
export { WebProgressBaseDescription };
