// igniteui-react-core/esm5/lib/WebPanePinnedEventArgsDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebPanePinnedEventArgsDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebPanePinnedEventArgsDescription, _super);
    function WebPanePinnedEventArgsDescription() {
        var _this = _super.call(this) || this;
        _this.h = null;
        _this.f = null;
        _this.j = false;
        _this.l = null;
        return _this;
    }
    WebPanePinnedEventArgsDescription.prototype.get_type = function () {
        return "WebPanePinnedEventArgs";
    };
    Object.defineProperty(WebPanePinnedEventArgsDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPanePinnedEventArgsDescription.prototype, "sourcePane", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("SourcePane");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPanePinnedEventArgsDescription.prototype, "panes", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Panes");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPanePinnedEventArgsDescription.prototype, "newValue", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("NewValue");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebPanePinnedEventArgsDescription.prototype, "location", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("Location");
        },
        enumerable: false,
        configurable: true
    });
    WebPanePinnedEventArgsDescription.$t = markType(WebPanePinnedEventArgsDescription, 'WebPanePinnedEventArgsDescription', Description.$);
    return WebPanePinnedEventArgsDescription;
}(Description));
export { WebPanePinnedEventArgsDescription };
