// igniteui-react-core/esm5/lib/WebSliderBaseDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebSliderBaseDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebSliderBaseDescription, _super);
    function WebSliderBaseDescription() {
        var _this = _super.call(this) || this;
        _this.t = 0;
        _this.s = 0;
        _this.r = 0;
        _this.x = 0;
        _this.h = false;
        _this.i = false;
        _this.l = false;
        _this.w = 0;
        _this.u = 0;
        _this.v = 0;
        _this.ah = null;
        _this.j = false;
        _this.k = false;
        _this.af = null;
        _this.ai = null;
        _this.f = null;
        _this.ag = null;
        return _this;
    }
    WebSliderBaseDescription.prototype.get_type = function () {
        return "WebSliderBase";
    };
    Object.defineProperty(WebSliderBaseDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "min", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
            this.e("Min");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "max", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
            this.e("Max");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "lowerBound", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("LowerBound");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "upperBound", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            this.x = a;
            this.e("UpperBound");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "disabled", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Disabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "discreteTrack", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("DiscreteTrack");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "hideTooltip", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("HideTooltip");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "step", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            this.w = a;
            this.e("Step");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "primaryTicks", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            this.u = a;
            this.e("PrimaryTicks");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "secondaryTicks", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            this.v = a;
            this.e("SecondaryTicks");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "tickOrientation", {
        get: function () {
            return this.ah;
        },
        set: function (a) {
            this.ah = a;
            this.e("TickOrientation");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "hidePrimaryLabels", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("HidePrimaryLabels");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "hideSecondaryLabels", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("HideSecondaryLabels");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "locale", {
        get: function () {
            return this.af;
        },
        set: function (a) {
            this.af = a;
            this.e("Locale");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "valueFormat", {
        get: function () {
            return this.ai;
        },
        set: function (a) {
            this.ai = a;
            this.e("ValueFormat");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "valueFormatOptions", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("ValueFormatOptions");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebSliderBaseDescription.prototype, "tickLabelRotation", {
        get: function () {
            return this.ag;
        },
        set: function (a) {
            this.ag = a;
            this.e("TickLabelRotation");
        },
        enumerable: false,
        configurable: true
    });
    WebSliderBaseDescription.$t = markType(WebSliderBaseDescription, 'WebSliderBaseDescription', Description.$);
    return WebSliderBaseDescription;
}(Description));
export { WebSliderBaseDescription };
