// igniteui-react-core/esm5/lib/WebTabGroupPaneDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WebDockManagerPaneDescription } from "./WebDockManagerPaneDescription";
import { markType } from "./type";
/**
 * @hidden
 */
var WebTabGroupPaneDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebTabGroupPaneDescription, _super);
    function WebTabGroupPaneDescription() {
        var _this = _super.call(this) || this;
        _this.q = null;
        _this.r = null;
        _this.g = null;
        _this.m = 0;
        _this.l = 0;
        _this.j = false;
        return _this;
    }
    WebTabGroupPaneDescription.prototype.get_type = function () {
        return "WebTabGroupPane";
    };
    Object.defineProperty(WebTabGroupPaneDescription.prototype, "id", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
            this.e("Id");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTabGroupPaneDescription.prototype, "paneType", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("PaneType");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTabGroupPaneDescription.prototype, "panes", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
            this.e("Panes");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTabGroupPaneDescription.prototype, "size", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
            this.e("Size");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTabGroupPaneDescription.prototype, "selectedIndex", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("SelectedIndex");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTabGroupPaneDescription.prototype, "allowEmpty", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("AllowEmpty");
        },
        enumerable: false,
        configurable: true
    });
    WebTabGroupPaneDescription.$t = markType(WebTabGroupPaneDescription, 'WebTabGroupPaneDescription', WebDockManagerPaneDescription.$);
    WebTabGroupPaneDescription.__marshalByValue = true;
    WebTabGroupPaneDescription.__marshalByValueAlias = "TabGroupPane";
    return WebTabGroupPaneDescription;
}(WebDockManagerPaneDescription));
export { WebTabGroupPaneDescription };
