// igniteui-react-core/esm5/lib/WebTreeItemDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
var WebTreeItemDescription = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WebTreeItemDescription, _super);
    function WebTreeItemDescription() {
        var _this = _super.call(this) || this;
        _this.f = null;
        _this.r = 0;
        _this.v = null;
        _this.j = false;
        _this.h = false;
        _this.i = false;
        _this.l = false;
        _this.k = false;
        _this.t = null;
        return _this;
    }
    WebTreeItemDescription.prototype.get_type = function () {
        return "WebTreeItem";
    };
    Object.defineProperty(WebTreeItemDescription.prototype, "type", {
        get: function () {
            return this.get_type();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeItemDescription.prototype, "parent", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
            this.e("Parent");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeItemDescription.prototype, "level", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
            this.e("Level");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeItemDescription.prototype, "label", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            this.v = a;
            this.e("Label");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeItemDescription.prototype, "expanded", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
            this.e("Expanded");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeItemDescription.prototype, "active", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
            this.e("Active");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeItemDescription.prototype, "disabled", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
            this.e("Disabled");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeItemDescription.prototype, "selected", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
            this.e("Selected");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeItemDescription.prototype, "loading", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
            this.e("Loading");
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WebTreeItemDescription.prototype, "value", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
            this.e("Value");
        },
        enumerable: false,
        configurable: true
    });
    WebTreeItemDescription.$t = markType(WebTreeItemDescription, 'WebTreeItemDescription', Description.$);
    return WebTreeItemDescription;
}(Description));
export { WebTreeItemDescription };
