// igniteui-react-excel/esm5/lib/AxisCrosses.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which describe the point at which
 * <see cref="Axis">axes</see> intersect.
 */
export var AxisCrosses = /*@__PURE__*/ (function (AxisCrosses) {
    /**
     * The crossing point is determined automatically.
     */
    AxisCrosses[AxisCrosses["Automatic"] = -4105] = "Automatic";
    /**
     * A custom value is designated as the crossing point, using the
     * [[Axis.crossesAt]] property
     */
    AxisCrosses[AxisCrosses["Custom"] = -4114] = "Custom";
    /**
     * The axes cross at the point coinciding with the axis' maximum value.
     */
    AxisCrosses[AxisCrosses["Maximum"] = 2] = "Maximum";
    /**
     * The axes cross at the point coinciding with the axis' minimum value.
     */
    AxisCrosses[AxisCrosses["Minimum"] = 4] = "Minimum";
    return AxisCrosses;
})({});
/**
 * @hidden
 */
export var AxisCrosses_$type = /*@__PURE__*/ markEnum('AxisCrosses', 'Automatic,-4105|Custom,-4114|Maximum,2|Minimum,4');
