// igniteui-react-excel/esm5/lib/BarShape.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which define the shape used by a bar chart series.
 */
export var BarShape = /*@__PURE__*/ (function (BarShape) {
    /**
     * Square box
     */
    BarShape[BarShape["Box"] = 0] = "Box";
    /**
     * Pyramid-shaped bar extending to the series maximum.
     */
    BarShape[BarShape["PyramidToPoint"] = 1] = "PyramidToPoint";
    /**
     * Truncated pyramid-shaped bar.
     */
    BarShape[BarShape["PyramidToMax"] = 2] = "PyramidToMax";
    /**
     * Cylinder shaped bar.
     */
    BarShape[BarShape["Cylinder"] = 3] = "Cylinder";
    /**
     * Cone-shaped bar extending to the series maximum.
     */
    BarShape[BarShape["ConeToPoint"] = 4] = "ConeToPoint";
    /**
     * Truncated cone-shaped bar.
     */
    BarShape[BarShape["ConeToMax"] = 5] = "ConeToMax";
    return BarShape;
})({});
/**
 * @hidden
 */
export var BarShape_$type = /*@__PURE__*/ markEnum('BarShape', 'Box,0|PyramidToPoint,1|PyramidToMax,2|Cylinder,3|ConeToPoint,4|ConeToMax,5');
