// igniteui-react-excel/esm5/lib/BorderLineStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Determines the style of a border line, i.e., solid, dashed, dotted, etc.
 * @see [[ChartBorder.lineStyle]]
 */
export var BorderLineStyle = /*@__PURE__*/ (function (BorderLineStyle) {
    /**
     * Solid line.
     */
    BorderLineStyle[BorderLineStyle["Solid"] = 0] = "Solid";
    /**
     * Dashed line.
     */
    BorderLineStyle[BorderLineStyle["Dash"] = 1] = "Dash";
    /**
     * Alternating dashes and dots.
     */
    BorderLineStyle[BorderLineStyle["DashDot"] = 2] = "DashDot";
    /**
     * Dotted line.
     */
    BorderLineStyle[BorderLineStyle["Dot"] = 3] = "Dot";
    /**
     * Large dashed line.
     */
    BorderLineStyle[BorderLineStyle["LargeDash"] = 4] = "LargeDash";
    /**
     * Large alternating dashes and dots.
     */
    BorderLineStyle[BorderLineStyle["LargeDashDot"] = 5] = "LargeDashDot";
    /**
     * Large alternating dashes and pairs of dots.
     */
    BorderLineStyle[BorderLineStyle["LargeDashDotDot"] = 6] = "LargeDashDotDot";
    /**
     * No line.
     */
    BorderLineStyle[BorderLineStyle["None"] = 7] = "None";
    /**
     * System-defined dashed line.
     */
    BorderLineStyle[BorderLineStyle["SysDash"] = 8] = "SysDash";
    /**
     * System-defined alternating dashes and dots.
     */
    BorderLineStyle[BorderLineStyle["SysDashDot"] = 9] = "SysDashDot";
    /**
     * System-defined alternating dashes and pairs of dots.
     */
    BorderLineStyle[BorderLineStyle["SysDashDotDot"] = 10] = "SysDashDotDot";
    /**
     * System-defined dotted line.
     */
    BorderLineStyle[BorderLineStyle["SysDot"] = 11] = "SysDot";
    return BorderLineStyle;
})({});
/**
 * @hidden
 */
export var BorderLineStyle_$type = /*@__PURE__*/ markEnum('BorderLineStyle', 'Solid,0|Dash,1|DashDot,2|Dot,3|LargeDash,4|LargeDashDot,5|LargeDashDotDot,6|None,7|SysDash,8|SysDashDot,9|SysDashDotDot,10|SysDot,11');
