// igniteui-react-excel/esm5/lib/CachedLoadPackage.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __generator, __values } from "tslib";
import { Base, NotSupportedException, typeCast, fromEnum, toEnum, markType } from "igniteui-react-core";
import { IPackage_$type } from "./IPackage";
import { IPackagePartEx_$type } from "./IPackagePartEx";
import { CachedLoadPackagePartEx } from "./CachedLoadPackagePartEx";
import { CachedLoadPackagePart } from "./CachedLoadPackagePart";
/**
 * @hidden
 */
var CachedLoadPackage = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CachedLoadPackage, _super);
    function CachedLoadPackage(a, b) {
        var _this = _super.call(this) || this;
        _this._a = null;
        _this._c = false;
        _this._a = a;
        _this._c = b;
        return _this;
    }
    CachedLoadPackage.prototype.createPart = function (a, b) {
        throw new NotSupportedException(0);
    };
    CachedLoadPackage.prototype.createRelationship = function (a, b, c, d) {
        throw new NotSupportedException(0);
    };
    CachedLoadPackage.prototype.dispose = function () {
        if (this._c) {
            this._a.dispose();
        }
    };
    CachedLoadPackage.prototype.getPart = function (a) {
        return this._b(this._a.getPart(a));
    };
    CachedLoadPackage.prototype._b = function (a) {
        var b = typeCast(IPackagePartEx_$type, a);
        if (b != null) {
            return new CachedLoadPackagePartEx(this, b);
        }
        return new CachedLoadPackagePart(this, a);
    };
    CachedLoadPackage.prototype._getParts = function () {
        var _d, _e, a, e_1_1;
        var e_1, _f;
        return __generator(this, function (_g) {
            switch (_g.label) {
                case 0:
                    _g.trys.push([0, 5, 6, 7]);
                    _d = __values(fromEnum(this._a.getParts())), _e = _d.next();
                    _g.label = 1;
                case 1:
                    if (!!_e.done)
                        return [3 /*break*/, 4];
                    a = _e.value;
                    return [4 /*yield*/, this._b(a)];
                case 2:
                    _g.sent();
                    _g.label = 3;
                case 3:
                    _e = _d.next();
                    return [3 /*break*/, 1];
                case 4: return [3 /*break*/, 7];
                case 5:
                    e_1_1 = _g.sent();
                    e_1 = { error: e_1_1 };
                    return [3 /*break*/, 7];
                case 6:
                    try {
                        if (_e && !_e.done && (_f = _d.return))
                            _f.call(_d);
                    }
                    finally {
                        if (e_1)
                            throw e_1.error;
                    }
                    return [7 /*endfinally*/];
                case 7: return [2 /*return*/];
            }
        });
    };
    CachedLoadPackage.prototype.getParts = function () {
        var _this = this;
        return toEnum(function () { return _this._getParts(); });
    };
    CachedLoadPackage.prototype.getRelationship = function (a) {
        return this._a.getRelationship(a);
    };
    CachedLoadPackage.prototype.getRelationships = function () {
        return this._a.getRelationships();
    };
    CachedLoadPackage.prototype.partExists = function (a) {
        return this._a.partExists(a);
    };
    CachedLoadPackage.$t = markType(CachedLoadPackage, 'CachedLoadPackage', Base.$, [IPackage_$type]);
    return CachedLoadPackage;
}(Base));
export { CachedLoadPackage };
