// igniteui-react-excel/esm5/lib/ColorInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, toNullable, Boolean_$type, Number_$type, markType } from "igniteui-react-core";
import { ST_UnsignedIntHex } from "./ST_UnsignedIntHex";
import { WorkbookColorInfo } from "./WorkbookColorInfo";
import { Color } from "igniteui-react-core";
import { WorkbookThemeColorType_$type } from "./WorkbookThemeColorType";
import { ExcelUtils } from "./ExcelUtils";
import { nullableEquals, unwrapNullable, nullableNotEquals, nullableLessThan } from "igniteui-react-core";
import { intSToU, u32BitwiseAnd } from "igniteui-react-core";
/**
 * @hidden
 */
var ColorInfo = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ColorInfo, _super);
    function ColorInfo() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.i = toNullable(Boolean_$type, null);
        _this.j = toNullable(Number_$type, null);
        _this.k = toNullable(ST_UnsignedIntHex.$, null);
        _this.l = toNullable(Number_$type, null);
        _this.g = 0;
        _this.d = null;
        return _this;
    }
    ColorInfo.prototype.equals = function (a) {
        if (Base.referenceEquals(this, a)) {
            return true;
        }
        var b = typeCast(ColorInfo.$, a);
        return b != null && nullableEquals(this.i, b.i) && nullableEquals(this.j, b.j) && Base.equalsStatic(this.k, unwrapNullable(b.k)) && nullableEquals(this.l, b.l) && this.g == b.g;
    };
    ColorInfo.prototype.getHashCode = function () {
        return Base.getHashCodeStatic(this.i) ^ Base.getHashCodeStatic(this.j) ^ Base.getHashCodeStatic(this.k) ^ Base.getHashCodeStatic(this.l) ^ (this.g);
    };
    Object.defineProperty(ColorInfo.prototype, "f", {
        get: function () {
            return this.equals(ColorInfo.b);
        },
        enumerable: false,
        configurable: true
    });
    ColorInfo.a = function (a, b, c) {
        var d = new ColorInfo();
        if (WorkbookColorInfo.l_op_Inequality(b, null)) {
            if (b.isAutomatic) {
                d.i = toNullable(Boolean_$type, true);
            }
            if (b._color$i.hasValue) {
                var e = b._color$i.value;
                if (ExcelUtils.al(e)) {
                    d.j = toNullable(Number_$type, intSToU(a._b._palette$i._m(b, c)));
                }
                else {
                    d.k = toNullable(ST_UnsignedIntHex.$, ST_UnsignedIntHex.b(intSToU(ExcelUtils.cx(e))));
                }
            }
            if (b._themeColorType$i.hasValue) {
                d.l = toNullable(Number_$type, b._themeColorType$i.value);
            }
            if (b._tint$i.hasValue) {
                d.g = b._tint$i.value;
            }
        }
        return d;
    };
    ColorInfo.prototype.c = function (a) {
        if (this.f) {
            return null;
        }
        if (nullableEquals(this.i, true)) {
            return WorkbookColorInfo.automatic;
        }
        var b = toNullable(Number_$type, null);
        if (this.g != 0) {
            b = toNullable(Number_$type, this.g);
        }
        if (a == null) {
            return WorkbookColorInfo.automatic;
        }
        if (nullableNotEquals(this.l, null) && nullableLessThan(this.l, a._b._h.length)) {
            return new WorkbookColorInfo(toNullable(Color.$, null), toNullable(WorkbookThemeColorType_$type, this.l.value), b, true, false);
        }
        if (nullableNotEquals(this.j, null)) {
            return new WorkbookColorInfo(toNullable(Color.$, a._b._palette$i._af(this.j.value)), toNullable(WorkbookThemeColorType_$type, null), b, true, false);
        }
        if (nullableNotEquals(this.k, null)) {
            if ((u32BitwiseAnd(ST_UnsignedIntHex.h(this.k.value), 4278190080)) == 4278190080) {
                return new WorkbookColorInfo(toNullable(Color.$, ExcelUtils.gz(ST_UnsignedIntHex.h(this.k.value))), toNullable(WorkbookThemeColorType_$type, null), b, true, true);
            }
            return null;
        }
        return null;
    };
    ColorInfo.prototype.m = function (a, b) {
        this.i = a.i;
        if (a.d != null) {
            this.d = a.d._aj(b);
        }
        this.j = a.j;
        this.k = a.k;
        this.l = ST_UnsignedIntHex.f(a.k);
        this.g = a.g;
    };
    ColorInfo.$t = markType(ColorInfo, 'ColorInfo');
    ColorInfo.b = new ColorInfo();
    return ColorInfo;
}(Base));
export { ColorInfo };
