// igniteui-react-excel/esm5/lib/CT_AnimationChartBuildProperties.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_AnimationChartBuildType } from "./ST_AnimationChartBuildType";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
var CT_AnimationChartBuildProperties = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_AnimationChartBuildProperties, _super);
    function CT_AnimationChartBuildProperties() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.e = toNullable(Boolean_$type, null);
        return _this;
    }
    Object.defineProperty(CT_AnimationChartBuildProperties.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AnimationChartBuildProperties.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_AnimationChartBuildProperties.g = function (a, b, c) {
        var e_1, _a;
        b = null;
        c = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "bld":
                        b = ST_AnimationChartBuildType.a(d.d);
                        break;
                    case "animBg":
                        c = toNullable(Boolean_$type, ST_OnOff.e(d.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_AnimationChartBuildProperties.h = function (a, b, c, d) {
        if (c === void 0) {
            c = toNullable(Boolean_$type, null);
        }
        if (d === void 0) {
            d = null;
        }
        a.ad.p(b, "a");
        var e = null;
        var f = null;
        if (c.hasValue) {
            f = "animBg";
            e = ST_OnOff.g(c.value);
            a.ad.h(f, e);
        }
        if (d != null) {
            f = "bld";
            e = d.g();
            a.ad.h(f, e);
        }
        ;
    };
    CT_AnimationChartBuildProperties.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_AnimationChartBuildProperties.$t = markType(CT_AnimationChartBuildProperties, 'CT_AnimationChartBuildProperties');
    return CT_AnimationChartBuildProperties;
}(Base));
export { CT_AnimationChartBuildProperties };
