// igniteui-react-excel/esm5/lib/CT_AnimationChartElement.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Number_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_ChartBuildStep_$type } from "./ST_ChartBuildStep";
import { CultureInfo } from "igniteui-react-core";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { parseInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_AnimationChartElement = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_AnimationChartElement, _super);
    function CT_AnimationChartElement() {
        var _this = _super.call(this) || this;
        _this.h = toNullable(Number_$type, null);
        _this.f = toNullable(Number_$type, null);
        _this.b = 0;
        return _this;
    }
    Object.defineProperty(CT_AnimationChartElement.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AnimationChartElement.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AnimationChartElement.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_AnimationChartElement.j = function (a, b, c, d) {
        var e_1, _a;
        b = toNullable(Number_$type, null);
        c = toNullable(Number_$type, null);
        d = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "seriesIdx":
                        b = toNullable(Number_$type, parseInt32_1(e.d, CultureInfo.invariantCulture));
                        break;
                    case "categoryIdx":
                        c = toNullable(Number_$type, parseInt32_1(e.d, CultureInfo.invariantCulture));
                        break;
                    case "bldStep":
                        d = XmlNamespaceDefinitionBase.b(ST_ChartBuildStep_$type, e.d);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_AnimationChartElement.k = function (a, b, c, d, e) {
        a.ad.p(b, "a");
        var f = null;
        var g = null;
        if (c.hasValue) {
            g = "categoryIdx";
            f = c.value.toString();
            a.ad.h(g, f);
        }
        if (d.hasValue) {
            g = "seriesIdx";
            f = d.value.toString();
            a.ad.h(g, f);
        }
        g = "bldStep";
        f = XmlNamespaceDefinitionBase.p(ST_ChartBuildStep_$type, e);
        a.ad.h(g, f);
        ;
    };
    CT_AnimationChartElement.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_AnimationChartElement.$t = markType(CT_AnimationChartElement, 'CT_AnimationChartElement');
    return CT_AnimationChartElement;
}(Base));
export { CT_AnimationChartElement };
