// igniteui-react-excel/esm5/lib/CT_AnimationDgmElement.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_DgmBuildStep_$type } from "./ST_DgmBuildStep";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
var CT_AnimationDgmElement = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_AnimationDgmElement, _super);
    function CT_AnimationDgmElement() {
        var _this = _super.call(this) || this;
        _this.f = null;
        _this.c = toNullable(ST_DgmBuildStep_$type, null);
        return _this;
    }
    Object.defineProperty(CT_AnimationDgmElement.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AnimationDgmElement.prototype, "b", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            this.c = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_AnimationDgmElement.h = function (a, b, c) {
        var e_1, _a;
        b = null;
        c = toNullable(ST_DgmBuildStep_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "id":
                        b = d.d;
                        break;
                    case "bldStep":
                        c = toNullable(ST_DgmBuildStep_$type, XmlNamespaceDefinitionBase.b(ST_DgmBuildStep_$type, d.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_AnimationDgmElement.i = function (a, b, c, d) {
        if (c === void 0) {
            c = toNullable(ST_DgmBuildStep_$type, null);
        }
        if (d === void 0) {
            d = null;
        }
        a.ad.p(b, "a");
        var e = null;
        var f = null;
        if (c.hasValue) {
            f = "bldStep";
            e = XmlNamespaceDefinitionBase.p(ST_DgmBuildStep_$type, c.value);
            a.ad.h(f, e);
        }
        if (d != null) {
            f = "id";
            e = d;
            a.ad.h(f, e);
        }
        ;
    };
    CT_AnimationDgmElement.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_AnimationDgmElement.$t = markType(CT_AnimationDgmElement, 'CT_AnimationDgmElement');
    return CT_AnimationDgmElement;
}(Base));
export { CT_AnimationDgmElement };
