// igniteui-react-excel/esm5/lib/CT_AbsSizeAnchor.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, markType, getInstanceType } from "igniteui-react-core";
import { EG_ObjectChoices } from "./EG_ObjectChoices";
/**
 * @hidden
 */
var CT_AbsSizeAnchor = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_AbsSizeAnchor, _super);
    function CT_AbsSizeAnchor() {
        var _this = _super.call(this) || this;
        _this.r = null;
        _this.c = null;
        _this.o = null;
        _this.k = null;
        _this.m = null;
        _this.g = null;
        _this.a = null;
        _this.e = null;
        _this.i = null;
        return _this;
    }
    Object.defineProperty(CT_AbsSizeAnchor.prototype, "item", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AbsSizeAnchor.prototype, "d", {
        get: function () {
            if (this.c == null) {
                this.c = new EG_ObjectChoices();
            }
            return this.c;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AbsSizeAnchor.prototype, "p", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AbsSizeAnchor.prototype, "l", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AbsSizeAnchor.prototype, "n", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AbsSizeAnchor.prototype, "h", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AbsSizeAnchor.prototype, "b", {
        get: function () {
            return this.a;
        },
        set: function (a) {
            this.a = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AbsSizeAnchor.prototype, "f", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AbsSizeAnchor.prototype, "j", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_AbsSizeAnchor.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_AbsSizeAnchor.$t = markType(CT_AbsSizeAnchor, 'CT_AbsSizeAnchor');
    return CT_AbsSizeAnchor;
}(Base));
export { CT_AbsSizeAnchor };
