// igniteui-react-excel/esm5/lib/CT_AudioCDTime.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Number_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt8_1, parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_AudioCDTime = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_AudioCDTime, _super);
    function CT_AudioCDTime() {
        var _this = _super.call(this) || this;
        _this.b = 0;
        _this.d = toNullable(Number_$type, null);
        return _this;
    }
    Object.defineProperty(CT_AudioCDTime.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AudioCDTime.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_AudioCDTime.g = function (a, b, c) {
        var e_1, _a;
        b = 0;
        c = toNullable(Number_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "track":
                        b = parseUInt8_1(d.d, CultureInfo.invariantCulture);
                        break;
                    case "time":
                        c = toNullable(Number_$type, parseUInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_AudioCDTime.h = function (a, b, c, d) {
        a.ad.p(b, "a");
        var e = null;
        var f = null;
        if (c.hasValue) {
            f = "time";
            e = c.value.toString();
            a.ad.h(f, e);
        }
        f = "track";
        e = d.toString();
        a.ad.h(f, e);
        ;
    };
    CT_AudioCDTime.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_AudioCDTime.$t = markType(CT_AudioCDTime, 'CT_AudioCDTime');
    return CT_AudioCDTime;
}(Base));
export { CT_AudioCDTime };
