// igniteui-react-excel/esm5/lib/CT_AudioCD.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_AudioCDTime } from "./CT_AudioCDTime";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
var CT_AudioCD = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_AudioCD, _super);
    function CT_AudioCD() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.b = null;
        _this.f = null;
        return _this;
    }
    Object.defineProperty(CT_AudioCD.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AudioCD.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_AudioCD.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_AudioCD.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1588:
                this.c = typeCast(CT_AudioCDTime.$, b);
                break;
            case 573:
                this.a = typeCast(CT_AudioCDTime.$, b);
                break;
            case 601:
                this.e = typeCast(CT_OfficeArtExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_AudioCD.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_AudioCD.$t = markType(CT_AudioCD, 'CT_AudioCD', Base.$, [ISupportSetProperty_$type]);
    return CT_AudioCD;
}(Base));
export { CT_AudioCD };
