// igniteui-react-excel/esm5/lib/CT_BlurEffect.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Number_$type, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
import { parseInt64_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_BlurEffect = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_BlurEffect, _super);
    function CT_BlurEffect() {
        var _this = _super.call(this) || this;
        _this.f = toNullable(Number_$type, null);
        _this.d = toNullable(Boolean_$type, null);
        return _this;
    }
    Object.defineProperty(CT_BlurEffect.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_BlurEffect.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_BlurEffect.h = function (a, b, c) {
        var e_1, _a;
        b = toNullable(Number_$type, null);
        c = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "rad":
                        b = toNullable(Number_$type, parseInt64_1(d.d, CultureInfo.invariantCulture));
                        break;
                    case "grow":
                        c = toNullable(Boolean_$type, ST_OnOff.e(d.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_BlurEffect.i = function (a, b, c, d) {
        if (c === void 0) {
            c = toNullable(Boolean_$type, null);
        }
        if (d === void 0) {
            d = toNullable(Number_$type, null);
        }
        a.ad.p(b, "a");
        var e = null;
        var f = null;
        if (c.hasValue) {
            f = "grow";
            e = ST_OnOff.g(c.value);
            a.ad.h(f, e);
        }
        if (d.hasValue) {
            f = "rad";
            e = d.value.toString();
            a.ad.h(f, e);
        }
        ;
    };
    CT_BlurEffect.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_BlurEffect.$t = markType(CT_BlurEffect, 'CT_BlurEffect');
    return CT_BlurEffect;
}(Base));
export { CT_BlurEffect };
