// igniteui-react-excel/esm5/lib/CT_Bevel.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Number_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_BevelPresetType_$type } from "./ST_BevelPresetType";
import { CultureInfo } from "igniteui-react-core";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { parseInt64_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_Bevel = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_Bevel, _super);
    function CT_Bevel() {
        var _this = _super.call(this) || this;
        _this.j = toNullable(Number_$type, null);
        _this.f = toNullable(Number_$type, null);
        _this.h = toNullable(ST_BevelPresetType_$type, null);
        return _this;
    }
    Object.defineProperty(CT_Bevel.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Bevel.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Bevel.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_Bevel.a = function (a) {
        var b = toNullable(Number_$type, null);
        var c = toNullable(Number_$type, null);
        var d = toNullable(ST_BevelPresetType_$type, null);
        var e = CT_Bevel.l(a, b, c, d);
        b = e.p1;
        c = e.p2;
        d = e.p3;
        return ((function () {
            var $ret = new CT_Bevel();
            $ret.e = c;
            $ret.g = d;
            $ret.i = b;
            return $ret;
        })());
    };
    CT_Bevel.l = function (a, b, c, d) {
        var e_1, _a;
        b = toNullable(Number_$type, null);
        c = toNullable(Number_$type, null);
        d = toNullable(ST_BevelPresetType_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "w":
                        b = toNullable(Number_$type, parseInt64_1(e.d, CultureInfo.invariantCulture));
                        break;
                    case "h":
                        c = toNullable(Number_$type, parseInt64_1(e.d, CultureInfo.invariantCulture));
                        break;
                    case "prst":
                        d = toNullable(ST_BevelPresetType_$type, XmlNamespaceDefinitionBase.b(ST_BevelPresetType_$type, e.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_Bevel.m = function (a, b, c, d, e) {
        if (c === void 0) {
            c = toNullable(Number_$type, null);
        }
        if (d === void 0) {
            d = toNullable(ST_BevelPresetType_$type, null);
        }
        if (e === void 0) {
            e = toNullable(Number_$type, null);
        }
        a.ad.p(b, "a");
        var f = null;
        var g = null;
        if (c.hasValue) {
            g = "h";
            f = c.value.toString();
            a.ad.h(g, f);
        }
        if (d.hasValue) {
            g = "prst";
            f = XmlNamespaceDefinitionBase.p(ST_BevelPresetType_$type, d.value);
            a.ad.h(g, f);
        }
        if (e.hasValue) {
            g = "w";
            f = e.value.toString();
            a.ad.h(g, f);
        }
        ;
    };
    CT_Bevel.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_Bevel.$t = markType(CT_Bevel, 'CT_Bevel');
    return CT_Bevel;
}(Base));
export { CT_Bevel };
