// igniteui-react-excel/esm5/lib/CT_BaseStyles.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_ColorScheme } from "./CT_ColorScheme";
import { CT_FontScheme } from "./CT_FontScheme";
import { CT_StyleMatrix } from "./CT_StyleMatrix";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
var CT_BaseStyles = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_BaseStyles, _super);
    function CT_BaseStyles() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.d = null;
        _this.h = null;
        _this.f = null;
        return _this;
    }
    Object.defineProperty(CT_BaseStyles.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_BaseStyles.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_BaseStyles.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_BaseStyles.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_BaseStyles.prototype.setProperty = function (a, b) {
        switch (a) {
            case 286:
                this.a = typeCast(CT_ColorScheme.$, b);
                break;
            case 661:
                this.c = typeCast(CT_FontScheme.$, b);
                break;
            case 657:
                this.g = typeCast(CT_StyleMatrix.$, b);
                break;
            case 601:
                this.e = typeCast(CT_OfficeArtExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_BaseStyles.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_BaseStyles.$t = markType(CT_BaseStyles, 'CT_BaseStyles', Base.$, [ISupportSetProperty_$type]);
    return CT_BaseStyles;
}(Base));
export { CT_BaseStyles };
