// igniteui-react-excel/esm5/lib/CT_ConditionalFormat.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_PivotAreas } from "./CT_PivotAreas";
import { CT_ExtensionList } from "./CT_ExtensionList";
import { ST_Scope_$type } from "./ST_Scope";
import { ST_Type_$type } from "./ST_Type";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_ConditionalFormat = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_ConditionalFormat, _super);
    function CT_ConditionalFormat() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.a = null;
        _this.d = 0;
        _this.f = 0;
        _this.h = 0;
        return _this;
    }
    CT_ConditionalFormat.i = function (a, b, c, d) {
        var e_1, _a;
        b = 0;
        c = 0;
        d = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "scope":
                        b = XmlNamespaceDefinitionBase.b(ST_Scope_$type, e.d);
                        break;
                    case "type":
                        c = XmlNamespaceDefinitionBase.b(ST_Type_$type, e.d);
                        break;
                    case "priority":
                        d = parseUInt32_1(e.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_ConditionalFormat.j = function (a, b, c, d, e) {
        a.ad.p(b, null);
        var f = null;
        var g = null;
        g = "priority";
        f = c.toString();
        a.ad.h(g, f);
        g = "scope";
        f = XmlNamespaceDefinitionBase.p(ST_Scope_$type, d);
        a.ad.h(g, f);
        g = "type";
        f = XmlNamespaceDefinitionBase.p(ST_Type_$type, e);
        a.ad.h(g, f);
    };
    CT_ConditionalFormat.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1204:
                this.b = typeCast(CT_PivotAreas.$, b);
                break;
            case 601:
                this.a = typeCast(CT_ExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_ConditionalFormat.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_ConditionalFormat.$t = markType(CT_ConditionalFormat, 'CT_ConditionalFormat', Base.$, [ISupportSetProperty_$type]);
    return CT_ConditionalFormat;
}(Base));
export { CT_ConditionalFormat };
