// igniteui-react-excel/esm5/lib/CT_CacheHierarchy_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
var CT_CacheHierarchy_SML2009 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_CacheHierarchy_SML2009, _super);
    function CT_CacheHierarchy_SML2009() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.h = toNullable(Boolean_$type, null);
        _this.n = toNullable(Boolean_$type, null);
        _this.j = toNullable(Boolean_$type, null);
        _this.l = toNullable(Boolean_$type, null);
        return _this;
    }
    Object.defineProperty(CT_CacheHierarchy_SML2009.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CacheHierarchy_SML2009.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CacheHierarchy_SML2009.prototype, "m", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CacheHierarchy_SML2009.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CacheHierarchy_SML2009.prototype, "k", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_CacheHierarchy_SML2009.o = function (a, b, c, d, e) {
        var e_1, _a;
        b = toNullable(Boolean_$type, null);
        c = toNullable(Boolean_$type, null);
        d = toNullable(Boolean_$type, null);
        e = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var f = _c.value;
                switch (f.a.e(a.d)) {
                    case "flattenHierarchies":
                        b = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "measuresSet":
                        c = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "hierarchizeDistinct":
                        d = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "ignore":
                        e = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e
        };
    };
    CT_CacheHierarchy_SML2009.p = function (a, b, c, d, e, f) {
        if (c === void 0) {
            c = toNullable(Boolean_$type, null);
        }
        if (d === void 0) {
            d = toNullable(Boolean_$type, null);
        }
        if (e === void 0) {
            e = toNullable(Boolean_$type, null);
        }
        if (f === void 0) {
            f = toNullable(Boolean_$type, null);
        }
        a.ad.p(b, null);
        var g = null;
        var h = null;
        if (c.hasValue) {
            h = "flattenHierarchies";
            g = ST_OnOff.g(c.value);
            a.ad.h(h, g);
        }
        if (d.hasValue) {
            h = "hierarchizeDistinct";
            g = ST_OnOff.g(d.value);
            a.ad.h(h, g);
        }
        if (e.hasValue) {
            h = "ignore";
            g = ST_OnOff.g(e.value);
            a.ad.h(h, g);
        }
        if (f.hasValue) {
            h = "measuresSet";
            g = ST_OnOff.g(f.value);
            a.ad.h(h, g);
        }
        ;
    };
    CT_CacheHierarchy_SML2009.$t = markType(CT_CacheHierarchy_SML2009, 'CT_CacheHierarchy_SML2009');
    return CT_CacheHierarchy_SML2009;
}(Base));
export { CT_CacheHierarchy_SML2009 };
