// igniteui-react-excel/esm5/lib/CT_ColorSchemeAndMapping.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_ColorScheme } from "./CT_ColorScheme";
import { CT_ColorMapping } from "./CT_ColorMapping";
/**
 * @hidden
 */
var CT_ColorSchemeAndMapping = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_ColorSchemeAndMapping, _super);
    function CT_ColorSchemeAndMapping() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.b = null;
        return _this;
    }
    Object.defineProperty(CT_ColorSchemeAndMapping.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorSchemeAndMapping.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_ColorSchemeAndMapping.prototype.setProperty = function (a, b) {
        switch (a) {
            case 286:
                this.c = typeCast(CT_ColorScheme.$, b);
                break;
            case 282:
                this.a = typeCast(CT_ColorMapping.$, b);
                break;
            default: break;
        }
    };
    CT_ColorSchemeAndMapping.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_ColorSchemeAndMapping.$t = markType(CT_ColorSchemeAndMapping, 'CT_ColorSchemeAndMapping', Base.$, [ISupportSetProperty_$type]);
    return CT_ColorSchemeAndMapping;
}(Base));
export { CT_ColorSchemeAndMapping };
