// igniteui-react-excel/esm5/lib/CT_CalculatedMember_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
var CT_CalculatedMember_SML2009 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_CalculatedMember_SML2009, _super);
    function CT_CalculatedMember_SML2009() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.m = null;
        _this.i = toNullable(Boolean_$type, null);
        _this.g = toNullable(Boolean_$type, null);
        _this.k = toNullable(Boolean_$type, null);
        _this.o = null;
        return _this;
    }
    Object.defineProperty(CT_CalculatedMember_SML2009.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CalculatedMember_SML2009.prototype, "l", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CalculatedMember_SML2009.prototype, "h", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CalculatedMember_SML2009.prototype, "f", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CalculatedMember_SML2009.prototype, "j", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CalculatedMember_SML2009.prototype, "n", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_CalculatedMember_SML2009.p = function (a, b, c, d, e, f) {
        var e_1, _a;
        b = null;
        c = toNullable(Boolean_$type, null);
        d = toNullable(Boolean_$type, null);
        e = toNullable(Boolean_$type, null);
        f = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var g = _c.value;
                switch (g.a.e(a.d)) {
                    case "displayFolder":
                        b = g.d;
                        break;
                    case "flattenHierarchies":
                        c = toNullable(Boolean_$type, ST_OnOff.e(g.d));
                        break;
                    case "dynamicSet":
                        d = toNullable(Boolean_$type, ST_OnOff.e(g.d));
                        break;
                    case "hierarchizeDistinct":
                        e = toNullable(Boolean_$type, ST_OnOff.e(g.d));
                        break;
                    case "mdxLong":
                        f = g.d;
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e,
            p5: f
        };
    };
    CT_CalculatedMember_SML2009.q = function (a, b, c, d, e, f, g) {
        if (c === void 0) {
            c = null;
        }
        if (d === void 0) {
            d = toNullable(Boolean_$type, null);
        }
        if (e === void 0) {
            e = toNullable(Boolean_$type, null);
        }
        if (f === void 0) {
            f = toNullable(Boolean_$type, null);
        }
        if (g === void 0) {
            g = null;
        }
        a.ad.p(b, null);
        var h = null;
        var i = null;
        if (c != null) {
            i = "displayFolder";
            h = c;
            a.ad.h(i, h);
        }
        if (d.hasValue) {
            i = "dynamicSet";
            h = ST_OnOff.g(d.value);
            a.ad.h(i, h);
        }
        if (e.hasValue) {
            i = "flattenHierarchies";
            h = ST_OnOff.g(e.value);
            a.ad.h(i, h);
        }
        if (f.hasValue) {
            i = "hierarchizeDistinct";
            h = ST_OnOff.g(f.value);
            a.ad.h(i, h);
        }
        if (g != null) {
            i = "mdxLong";
            h = g;
            a.ad.h(i, h);
        }
        ;
    };
    CT_CalculatedMember_SML2009.$t = markType(CT_CalculatedMember_SML2009, 'CT_CalculatedMember_SML2009');
    return CT_CalculatedMember_SML2009;
}(Base));
export { CT_CalculatedMember_SML2009 };
