// igniteui-react-excel/esm5/lib/CT_CategoryFilterException.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, String_$type, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_ShapeProperties_DML } from "./CT_ShapeProperties_DML";
import { CT_UnsignedInt } from "./CT_UnsignedInt";
import { CT_Boolean } from "./CT_Boolean";
import { CT_Marker_Chart } from "./CT_Marker_Chart";
import { CT_DLbl } from "./CT_DLbl";
/**
 * @hidden
 */
var CT_CategoryFilterException = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_CategoryFilterException, _super);
    function CT_CategoryFilterException() {
        var _this = _super.call(this) || this;
        _this.n = null;
        _this.l = null;
        _this.f = null;
        _this.d = null;
        _this.b = null;
        _this.j = null;
        _this.h = null;
        return _this;
    }
    Object.defineProperty(CT_CategoryFilterException.prototype, "m", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CategoryFilterException.prototype, "k", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CategoryFilterException.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CategoryFilterException.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CategoryFilterException.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CategoryFilterException.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CategoryFilterException.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_CategoryFilterException.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1583:
                this.m = typeCast(String_$type, b);
                break;
            case 1570:
                this.k = typeCast(CT_ShapeProperties_DML.$, b);
                break;
            case 598:
                this.e = typeCast(CT_UnsignedInt.$, b);
                break;
            case 848:
                this.c = typeCast(CT_Boolean.$, b);
                break;
            case 183:
                this.a = typeCast(CT_Boolean.$, b);
                break;
            case 979:
                this.i = typeCast(CT_Marker_Chart.$, b);
                break;
            case 401:
                this.g = typeCast(CT_DLbl.$, b);
                break;
            default: break;
        }
    };
    CT_CategoryFilterException.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_CategoryFilterException.$t = markType(CT_CategoryFilterException, 'CT_CategoryFilterException', Base.$, [ISupportSetProperty_$type]);
    return CT_CategoryFilterException;
}(Base));
export { CT_CategoryFilterException };
