// igniteui-react-excel/esm5/lib/CT_Cfvo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_ExtensionList } from "./CT_ExtensionList";
import { ST_CfvoType_$type } from "./ST_CfvoType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
var CT_Cfvo = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_Cfvo, _super);
    function CT_Cfvo() {
        var _this = _super.call(this) || this;
        _this.a = null;
        _this.b = 0;
        _this.f = null;
        _this.d = toNullable(Boolean_$type, null);
        return _this;
    }
    CT_Cfvo.g = function (a, b, c, d) {
        var e_1, _a;
        b = 0;
        c = null;
        d = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "type":
                        b = XmlNamespaceDefinitionBase.b(ST_CfvoType_$type, e.d);
                        break;
                    case "val":
                        c = e.d;
                        break;
                    case "gte":
                        d = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_Cfvo.h = function (a, b, c, d, e) {
        a.ad.p(b, null);
        var f = null;
        var g = null;
        if (c.hasValue) {
            g = "gte";
            f = ST_OnOff.g(c.value);
            a.ad.h(g, f);
        }
        if (d != null) {
            g = "val";
            f = d;
            a.ad.h(g, f);
        }
        g = "type";
        f = XmlNamespaceDefinitionBase.p(ST_CfvoType_$type, e);
        a.ad.h(g, f);
    };
    CT_Cfvo.prototype.setProperty = function (a, b) {
        switch (a) {
            case 601:
                this.a = typeCast(CT_ExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_Cfvo.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_Cfvo.$t = markType(CT_Cfvo, 'CT_Cfvo', Base.$, [ISupportSetProperty_$type]);
    return CT_Cfvo;
}(Base));
export { CT_Cfvo };
