// igniteui-react-excel/esm5/lib/CT_CustomColor.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { EG_ColorChoice_DML } from "./EG_ColorChoice_DML";
/**
 * @hidden
 */
var CT_CustomColor = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_CustomColor, _super);
    function CT_CustomColor() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.a = null;
        return _this;
    }
    Object.defineProperty(CT_CustomColor.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_CustomColor.prototype, "b", {
        get: function () {
            if (this.a == null) {
                this.a = new EG_ColorChoice_DML();
            }
            return this.a;
        },
        enumerable: false,
        configurable: true
    });
    CT_CustomColor.f = function (a, b) {
        var e_1, _a;
        b = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var c = _c.value;
                switch (c.a.e(a.d)) {
                    case "name":
                        b = c.d;
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b
        };
    };
    CT_CustomColor.g = function (a, b, c) {
        if (c === void 0) {
            c = null;
        }
        a.ad.p(b, "a");
        var d = null;
        var e = null;
        if (c != null) {
            e = "name";
            d = c;
            a.ad.h(e, d);
        }
        ;
    };
    CT_CustomColor.prototype.setProperty = function (a, b) {
        switch (a) {
            default:
                if (this.b.g(a)) {
                    this.b.i(a, b);
                    return;
                }
                break;
        }
    };
    CT_CustomColor.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_CustomColor.$t = markType(CT_CustomColor, 'CT_CustomColor', Base.$, [ISupportSetProperty_$type]);
    return CT_CustomColor;
}(Base));
export { CT_CustomColor };
