// igniteui-react-excel/esm5/lib/CT_ColorScheme.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_Color_DML } from "./CT_Color_DML";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
var CT_ColorScheme = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_ColorScheme, _super);
    function CT_ColorScheme() {
        var _this = _super.call(this) || this;
        _this.n = null;
        _this.v = null;
        _this.p = null;
        _this.x = null;
        _this.b = null;
        _this.d = null;
        _this.f = null;
        _this.h = null;
        _this.j = null;
        _this.l = null;
        _this.t = null;
        _this.r = null;
        _this.z = null;
        _this.ab = null;
        return _this;
    }
    Object.defineProperty(CT_ColorScheme.prototype, "m", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "u", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            this.v = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "o", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "w", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            this.x = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "k", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "s", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "q", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "y", {
        get: function () {
            return this.z;
        },
        set: function (a) {
            this.z = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ColorScheme.prototype, "aa", {
        get: function () {
            return this.ab;
        },
        set: function (a) {
            this.ab = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_ColorScheme.ad = function (a, b) {
        var e_1, _a;
        b = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var c = _c.value;
                switch (c.a.e(a.d)) {
                    case "name":
                        b = c.d;
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b
        };
    };
    CT_ColorScheme.ae = function (a, b, c) {
        a.ad.p(b, "a");
        var d = null;
        var e = null;
        if (c != null) {
            e = "name";
            d = c;
            a.ad.h(e, d);
        }
        ;
    };
    CT_ColorScheme.prototype.setProperty = function (a, b) {
        switch (a) {
            case 503:
                this.m = typeCast(CT_Color_DML.$, b);
                break;
            case 939:
                this.u = typeCast(CT_Color_DML.$, b);
                break;
            case 504:
                this.o = typeCast(CT_Color_DML.$, b);
                break;
            case 940:
                this.w = typeCast(CT_Color_DML.$, b);
                break;
            case 7:
                this.a = typeCast(CT_Color_DML.$, b);
                break;
            case 8:
                this.c = typeCast(CT_Color_DML.$, b);
                break;
            case 9:
                this.e = typeCast(CT_Color_DML.$, b);
                break;
            case 10:
                this.g = typeCast(CT_Color_DML.$, b);
                break;
            case 11:
                this.i = typeCast(CT_Color_DML.$, b);
                break;
            case 12:
                this.k = typeCast(CT_Color_DML.$, b);
                break;
            case 790:
                this.s = typeCast(CT_Color_DML.$, b);
                break;
            case 658:
                this.q = typeCast(CT_Color_DML.$, b);
                break;
            case 601:
                this.y = typeCast(CT_OfficeArtExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_ColorScheme.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_ColorScheme.$t = markType(CT_ColorScheme, 'CT_ColorScheme', Base.$, [ISupportSetProperty_$type]);
    return CT_ColorScheme;
}(Base));
export { CT_ColorScheme };
