// igniteui-react-excel/esm5/lib/CT_ConditionalFormatting_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
var CT_ConditionalFormatting_SML2009 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_ConditionalFormatting_SML2009, _super);
    function CT_ConditionalFormatting_SML2009() {
        var _this = _super.call(this) || this;
        _this.g = null;
        _this.d = null;
        _this.b = null;
        _this.i = toNullable(Boolean_$type, null);
        return _this;
    }
    Object.defineProperty(CT_ConditionalFormatting_SML2009.prototype, "f", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ConditionalFormatting_SML2009.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ConditionalFormatting_SML2009.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ConditionalFormatting_SML2009.prototype, "h", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_ConditionalFormatting_SML2009.j = function (a, b) {
        var e_1, _a;
        b = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var c = _c.value;
                switch (c.a.e(a.d)) {
                    case "pivot":
                        b = toNullable(Boolean_$type, ST_OnOff.e(c.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b
        };
    };
    CT_ConditionalFormatting_SML2009.k = function (a, b, c) {
        if (c === void 0) {
            c = toNullable(Boolean_$type, null);
        }
        a.ad.p(b, null);
        var d = null;
        var e = null;
        if (c.hasValue) {
            e = "pivot";
            d = ST_OnOff.g(c.value);
            a.ad.h(e, d);
        }
        ;
    };
    CT_ConditionalFormatting_SML2009.$t = markType(CT_ConditionalFormatting_SML2009, 'CT_ConditionalFormatting_SML2009');
    return CT_ConditionalFormatting_SML2009;
}(Base));
export { CT_ConditionalFormatting_SML2009 };
