// igniteui-react-excel/esm5/lib/CT_ConditionalFormat_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Number_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_Scope_$type } from "./ST_Scope";
import { ST_Type_$type } from "./ST_Type";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_ConditionalFormat_SML2009 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_ConditionalFormat_SML2009, _super);
    function CT_ConditionalFormat_SML2009() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.b = null;
        _this.j = toNullable(ST_Scope_$type, null);
        _this.l = toNullable(ST_Type_$type, null);
        _this.h = toNullable(Number_$type, null);
        _this.n = null;
        return _this;
    }
    Object.defineProperty(CT_ConditionalFormat_SML2009.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ConditionalFormat_SML2009.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ConditionalFormat_SML2009.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ConditionalFormat_SML2009.prototype, "k", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ConditionalFormat_SML2009.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ConditionalFormat_SML2009.prototype, "m", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_ConditionalFormat_SML2009.o = function (a, b, c, d, e) {
        var e_1, _a;
        b = toNullable(ST_Scope_$type, null);
        c = toNullable(ST_Type_$type, null);
        d = toNullable(Number_$type, null);
        e = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var f = _c.value;
                switch (f.a.e(a.d)) {
                    case "scope":
                        b = toNullable(ST_Scope_$type, XmlNamespaceDefinitionBase.b(ST_Scope_$type, f.d));
                        break;
                    case "type":
                        c = toNullable(ST_Type_$type, XmlNamespaceDefinitionBase.b(ST_Type_$type, f.d));
                        break;
                    case "priority":
                        d = toNullable(Number_$type, parseUInt32_1(f.d, CultureInfo.invariantCulture));
                        break;
                    case "id":
                        e = f.d;
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e
        };
    };
    CT_ConditionalFormat_SML2009.p = function (a, b, c, d, e, f) {
        a.ad.p(b, null);
        var g = null;
        var h = null;
        if (c.hasValue) {
            h = "priority";
            g = c.value.toString();
            a.ad.h(h, g);
        }
        if (d.hasValue) {
            h = "scope";
            g = XmlNamespaceDefinitionBase.p(ST_Scope_$type, d.value);
            a.ad.h(h, g);
        }
        if (e.hasValue) {
            h = "type";
            g = XmlNamespaceDefinitionBase.p(ST_Type_$type, e.value);
            a.ad.h(h, g);
        }
        if (f != null) {
            h = "id";
            g = f;
            a.ad.h(h, g);
        }
        ;
    };
    CT_ConditionalFormat_SML2009.$t = markType(CT_ConditionalFormat_SML2009, 'CT_ConditionalFormat_SML2009');
    return CT_ConditionalFormat_SML2009;
}(Base));
export { CT_ConditionalFormat_SML2009 };
