// igniteui-react-excel/esm5/lib/CT_Cell3D.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_Bevel } from "./CT_Bevel";
import { CT_LightRig } from "./CT_LightRig";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
import { ST_PresetMaterialType_$type } from "./ST_PresetMaterialType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
var CT_Cell3D = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_Cell3D, _super);
    function CT_Cell3D() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.d = null;
        _this.f = null;
        _this.i = toNullable(ST_PresetMaterialType_$type, null);
        return _this;
    }
    Object.defineProperty(CT_Cell3D.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Cell3D.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Cell3D.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Cell3D.prototype, "h", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_Cell3D.k = function (a, b) {
        var e_1, _a;
        b = toNullable(ST_PresetMaterialType_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var c = _c.value;
                switch (c.a.e(a.d)) {
                    case "prstMaterial":
                        b = toNullable(ST_PresetMaterialType_$type, XmlNamespaceDefinitionBase.b(ST_PresetMaterialType_$type, c.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b
        };
    };
    CT_Cell3D.l = function (a, b, c) {
        if (c === void 0) {
            c = toNullable(ST_PresetMaterialType_$type, null);
        }
        a.ad.p(b, "a");
        var d = null;
        var e = null;
        if (c.hasValue) {
            e = "prstMaterial";
            d = XmlNamespaceDefinitionBase.p(ST_PresetMaterialType_$type, c.value);
            a.ad.h(e, d);
        }
        ;
    };
    CT_Cell3D.prototype.setProperty = function (a, b) {
        switch (a) {
            case 108:
                this.a = typeCast(CT_Bevel.$, b);
                break;
            case 896:
                this.c = typeCast(CT_LightRig.$, b);
                break;
            case 601:
                this.e = typeCast(CT_OfficeArtExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_Cell3D.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_Cell3D.$t = markType(CT_Cell3D, 'CT_Cell3D', Base.$, [ISupportSetProperty_$type]);
    return CT_Cell3D;
}(Base));
export { CT_Cell3D };
