// igniteui-react-excel/esm5/lib/CT_DataValidations_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, Number_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_DataValidations_SML2009 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_DataValidations_SML2009, _super);
    function CT_DataValidations_SML2009() {
        var _this = _super.call(this) || this;
        _this.c = null;
        _this.g = toNullable(Boolean_$type, null);
        _this.i = toNullable(Number_$type, null);
        _this.k = toNullable(Number_$type, null);
        _this.e = toNullable(Number_$type, null);
        return _this;
    }
    Object.defineProperty(CT_DataValidations_SML2009.prototype, "b", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            this.c = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_DataValidations_SML2009.prototype, "f", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_DataValidations_SML2009.prototype, "h", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_DataValidations_SML2009.prototype, "j", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_DataValidations_SML2009.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_DataValidations_SML2009.l = function (a, b, c, d, e) {
        var e_1, _a;
        b = toNullable(Boolean_$type, null);
        c = toNullable(Number_$type, null);
        d = toNullable(Number_$type, null);
        e = toNullable(Number_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var f = _c.value;
                switch (f.a.e(a.d)) {
                    case "disablePrompts":
                        b = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "xWindow":
                        c = toNullable(Number_$type, parseUInt32_1(f.d, CultureInfo.invariantCulture));
                        break;
                    case "yWindow":
                        d = toNullable(Number_$type, parseUInt32_1(f.d, CultureInfo.invariantCulture));
                        break;
                    case "count":
                        e = toNullable(Number_$type, parseUInt32_1(f.d, CultureInfo.invariantCulture));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e
        };
    };
    CT_DataValidations_SML2009.m = function (a, b, c, d, e, f) {
        if (c === void 0) {
            c = toNullable(Number_$type, null);
        }
        if (d === void 0) {
            d = toNullable(Boolean_$type, null);
        }
        if (e === void 0) {
            e = toNullable(Number_$type, null);
        }
        if (f === void 0) {
            f = toNullable(Number_$type, null);
        }
        a.ad.p(b, null);
        var g = null;
        var h = null;
        if (c.hasValue) {
            h = "count";
            g = c.value.toString();
            a.ad.h(h, g);
        }
        if (d.hasValue) {
            h = "disablePrompts";
            g = ST_OnOff.g(d.value);
            a.ad.h(h, g);
        }
        if (e.hasValue) {
            h = "xWindow";
            g = e.value.toString();
            a.ad.h(h, g);
        }
        if (f.hasValue) {
            h = "yWindow";
            g = f.value.toString();
            a.ad.h(h, g);
        }
        ;
    };
    CT_DataValidations_SML2009.$t = markType(CT_DataValidations_SML2009, 'CT_DataValidations_SML2009');
    return CT_DataValidations_SML2009;
}(Base));
export { CT_DataValidations_SML2009 };
