// igniteui-react-excel/esm5/lib/CT_DataLabelFieldTableEntry.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, String_$type, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_StrData } from "./CT_StrData";
/**
 * @hidden
 */
var CT_DataLabelFieldTableEntry = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_DataLabelFieldTableEntry, _super);
    function CT_DataLabelFieldTableEntry() {
        var _this = _super.call(this) || this;
        _this.g = null;
        _this.d = null;
        _this.b = null;
        return _this;
    }
    Object.defineProperty(CT_DataLabelFieldTableEntry.prototype, "f", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_DataLabelFieldTableEntry.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_DataLabelFieldTableEntry.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_DataLabelFieldTableEntry.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1787:
                this.f = typeCast(String_$type, b);
                break;
            case 612:
                this.c = typeCast(String_$type, b);
                break;
            case 402:
                this.a = typeCast(CT_StrData.$, b);
                break;
            default: break;
        }
    };
    CT_DataLabelFieldTableEntry.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_DataLabelFieldTableEntry.$t = markType(CT_DataLabelFieldTableEntry, 'CT_DataLabelFieldTableEntry', Base.$, [ISupportSetProperty_$type]);
    return CT_DataLabelFieldTableEntry;
}(Base));
export { CT_DataLabelFieldTableEntry };
