// igniteui-react-excel/esm5/lib/CT_DataLabelVisibilities.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
var CT_DataLabelVisibilities = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_DataLabelVisibilities, _super);
    function CT_DataLabelVisibilities() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.d = toNullable(Boolean_$type, null);
        _this.b = toNullable(Boolean_$type, null);
        _this.f = toNullable(Boolean_$type, null);
        return _this;
    }
    Object.defineProperty(CT_DataLabelVisibilities.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_DataLabelVisibilities.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_DataLabelVisibilities.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_DataLabelVisibilities.h = function (a, b, c, d) {
        var e_1, _a;
        b = toNullable(Boolean_$type, null);
        c = toNullable(Boolean_$type, null);
        d = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "seriesName":
                        b = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    case "categoryName":
                        c = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    case "value":
                        d = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_DataLabelVisibilities.i = function (a, b, c, d, e) {
        if (c === void 0) {
            c = toNullable(Boolean_$type, null);
        }
        if (d === void 0) {
            d = toNullable(Boolean_$type, null);
        }
        if (e === void 0) {
            e = toNullable(Boolean_$type, null);
        }
        a.ad.p(b, "cx");
        var f = null;
        var g = null;
        if (c.hasValue) {
            g = "categoryName";
            f = ST_OnOff.g(c.value);
            a.ad.h(g, f);
        }
        if (d.hasValue) {
            g = "seriesName";
            f = ST_OnOff.g(d.value);
            a.ad.h(g, f);
        }
        if (e.hasValue) {
            g = "value";
            f = ST_OnOff.g(e.value);
            a.ad.h(g, f);
        }
        ;
    };
    CT_DataLabelVisibilities.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_DataLabelVisibilities.$t = markType(CT_DataLabelVisibilities, 'CT_DataLabelVisibilities');
    return CT_DataLabelVisibilities;
}(Base));
export { CT_DataLabelVisibilities };
