// igniteui-react-excel/esm5/lib/CT_Dxf.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_Font } from "./CT_Font";
import { CT_NumFmt } from "./CT_NumFmt";
import { CT_Fill } from "./CT_Fill";
import { CT_CellAlignment } from "./CT_CellAlignment";
import { CT_Border } from "./CT_Border";
import { CT_CellProtection } from "./CT_CellProtection";
import { CT_ExtensionList } from "./CT_ExtensionList";
/**
 * @hidden
 */
var CT_Dxf = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_Dxf, _super);
    function CT_Dxf() {
        var _this = _super.call(this) || this;
        _this.l = null;
        _this.n = null;
        _this.j = null;
        _this.f = null;
        _this.d = null;
        _this.h = null;
        _this.b = null;
        return _this;
    }
    Object.defineProperty(CT_Dxf.prototype, "k", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Dxf.prototype, "m", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Dxf.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Dxf.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Dxf.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Dxf.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Dxf.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_Dxf.prototype.setProperty = function (a, b) {
        switch (a) {
            case 659:
                this.k = typeCast(CT_Font.$, b);
                break;
            case 1088:
                this.m = typeCast(CT_NumFmt.$, b);
                break;
            case 630:
                this.i = typeCast(CT_Fill.$, b);
                break;
            case 27:
                this.e = typeCast(CT_CellAlignment.$, b);
                break;
            case 149:
                this.c = typeCast(CT_Border.$, b);
                break;
            case 1271:
                this.g = typeCast(CT_CellProtection.$, b);
                break;
            case 601:
                this.a = typeCast(CT_ExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_Dxf.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_Dxf.$t = markType(CT_Dxf, 'CT_Dxf', Base.$, [ISupportSetProperty_$type]);
    return CT_Dxf;
}(Base));
export { CT_Dxf };
