// igniteui-react-excel/esm5/lib/CT_FontCollection.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_TextFont } from "./CT_TextFont";
import { CT_SupplementalFont } from "./CT_SupplementalFont";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
var CT_FontCollection = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_FontCollection, _super);
    function CT_FontCollection() {
        var _this = _super.call(this) || this;
        _this.h = null;
        _this.f = null;
        _this.d = null;
        _this.j = null;
        _this.b = null;
        return _this;
    }
    Object.defineProperty(CT_FontCollection.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_FontCollection.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_FontCollection.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_FontCollection.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_FontCollection.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_FontCollection.prototype.setProperty = function (a, b) {
        switch (a) {
            case 875:
                this.g = typeCast(CT_TextFont.$, b);
                break;
            case 551:
                this.e = typeCast(CT_TextFont.$, b);
                break;
            case 363:
                this.c = typeCast(CT_TextFont.$, b);
                break;
            case 659:
                this.i.add(typeCast(CT_SupplementalFont.$, b));
                break;
            case 601:
                this.a = typeCast(CT_OfficeArtExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_FontCollection.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_FontCollection.$t = markType(CT_FontCollection, 'CT_FontCollection', Base.$, [ISupportSetProperty_$type]);
    return CT_FontCollection;
}(Base));
export { CT_FontCollection };
