// igniteui-react-excel/esm5/lib/CT_FillStyleList.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { List$1 } from "igniteui-react-core";
import { ISaveToXLSX_$type } from "./ISaveToXLSX";
/**
 * @hidden
 */
var CT_FillStyleList = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_FillStyleList, _super);
    function CT_FillStyleList() {
        var _this = _super.call(this) || this;
        _this.a = null;
        return _this;
    }
    Object.defineProperty(CT_FillStyleList.prototype, "b", {
        get: function () {
            if (this.a == null) {
                this.a = new List$1(ISaveToXLSX_$type, 0);
            }
            return this.a;
        },
        enumerable: false,
        configurable: true
    });
    CT_FillStyleList.c = function (a, b) {
        b = a > 1000;
        var c = b ? (a - 1001) : (a - 1);
        return {
            ret: c,
            p1: b
        };
    };
    CT_FillStyleList.prototype.setProperty = function (a, b) {
        var c = typeCast(ISaveToXLSX_$type, b);
        if (c == null) {
        }
        else {
            this.b.add(c);
        }
    };
    CT_FillStyleList.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_FillStyleList.$t = markType(CT_FillStyleList, 'CT_FillStyleList', Base.$, [ISupportSetProperty_$type]);
    return CT_FillStyleList;
}(Base));
export { CT_FillStyleList };
