// igniteui-react-excel/esm5/lib/CT_GvmlPicture.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_GvmlPictureNonVisual } from "./CT_GvmlPictureNonVisual";
import { CT_BlipFillProperties_DML } from "./CT_BlipFillProperties_DML";
import { CT_ShapeProperties_DML } from "./CT_ShapeProperties_DML";
import { CT_ShapeStyle } from "./CT_ShapeStyle";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
var CT_GvmlPicture = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GvmlPicture, _super);
    function CT_GvmlPicture() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.b = null;
        _this.h = null;
        _this.j = null;
        _this.f = null;
        return _this;
    }
    Object.defineProperty(CT_GvmlPicture.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GvmlPicture.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GvmlPicture.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GvmlPicture.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GvmlPicture.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_GvmlPicture.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1105:
                this.c = typeCast(CT_GvmlPictureNonVisual.$, b);
                break;
            case 128:
                this.a = typeCast(CT_BlipFillProperties_DML.$, b);
                break;
            case 1570:
                this.g = typeCast(CT_ShapeProperties_DML.$, b);
                break;
            case 1618:
                this.i = typeCast(CT_ShapeStyle.$, b);
                break;
            case 601:
                this.e = typeCast(CT_OfficeArtExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_GvmlPicture.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_GvmlPicture.$t = markType(CT_GvmlPicture, 'CT_GvmlPicture', Base.$, [ISupportSetProperty_$type]);
    return CT_GvmlPicture;
}(Base));
export { CT_GvmlPicture };
