// igniteui-react-excel/esm5/lib/CT_GvmlGraphicFrameNonVisual.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_NonVisualDrawingProps } from "./CT_NonVisualDrawingProps";
import { CT_NonVisualGraphicFrameProperties } from "./CT_NonVisualGraphicFrameProperties";
/**
 * @hidden
 */
var CT_GvmlGraphicFrameNonVisual = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GvmlGraphicFrameNonVisual, _super);
    function CT_GvmlGraphicFrameNonVisual() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.d = null;
        return _this;
    }
    Object.defineProperty(CT_GvmlGraphicFrameNonVisual.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GvmlGraphicFrameNonVisual.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_GvmlGraphicFrameNonVisual.prototype.setProperty = function (a, b) {
        switch (a) {
            case 202:
                this.a = typeCast(CT_NonVisualDrawingProps.$, b);
                break;
            case 199:
                this.c = typeCast(CT_NonVisualGraphicFrameProperties.$, b);
                break;
            default: break;
        }
    };
    CT_GvmlGraphicFrameNonVisual.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_GvmlGraphicFrameNonVisual.$t = markType(CT_GvmlGraphicFrameNonVisual, 'CT_GvmlGraphicFrameNonVisual', Base.$, [ISupportSetProperty_$type]);
    return CT_GvmlGraphicFrameNonVisual;
}(Base));
export { CT_GvmlGraphicFrameNonVisual };
