// igniteui-react-excel/esm5/lib/CT_GvmlConnectorNonVisual.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_NonVisualDrawingProps } from "./CT_NonVisualDrawingProps";
import { CT_NonVisualConnectorProperties } from "./CT_NonVisualConnectorProperties";
/**
 * @hidden
 */
var CT_GvmlConnectorNonVisual = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GvmlConnectorNonVisual, _super);
    function CT_GvmlConnectorNonVisual() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.b = null;
        return _this;
    }
    Object.defineProperty(CT_GvmlConnectorNonVisual.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GvmlConnectorNonVisual.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_GvmlConnectorNonVisual.prototype.setProperty = function (a, b) {
        switch (a) {
            case 202:
                this.c = typeCast(CT_NonVisualDrawingProps.$, b);
                break;
            case 198:
                this.a = typeCast(CT_NonVisualConnectorProperties.$, b);
                break;
            default: break;
        }
    };
    CT_GvmlConnectorNonVisual.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_GvmlConnectorNonVisual.$t = markType(CT_GvmlConnectorNonVisual, 'CT_GvmlConnectorNonVisual', Base.$, [ISupportSetProperty_$type]);
    return CT_GvmlConnectorNonVisual;
}(Base));
export { CT_GvmlConnectorNonVisual };
