// igniteui-react-excel/esm5/lib/CT_GeoDataPointToEntityQuery.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_EntityType_$type } from "./ST_EntityType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
var CT_GeoDataPointToEntityQuery = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GeoDataPointToEntityQuery, _super);
    function CT_GeoDataPointToEntityQuery() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.c = 0;
        _this.e = null;
        return _this;
    }
    Object.defineProperty(CT_GeoDataPointToEntityQuery.prototype, "b", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            this.c = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GeoDataPointToEntityQuery.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_GeoDataPointToEntityQuery.a = function (a) {
        var b = 0;
        var c = null;
        var d = CT_GeoDataPointToEntityQuery.g(a, b, c);
        b = d.p1;
        c = d.p2;
        return ((function () {
            var $ret = new CT_GeoDataPointToEntityQuery();
            $ret.d = c;
            $ret.b = b;
            return $ret;
        })());
    };
    CT_GeoDataPointToEntityQuery.g = function (a, b, c) {
        var e_1, _a;
        b = 0;
        c = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "entityType":
                        b = XmlNamespaceDefinitionBase.b(ST_EntityType_$type, d.d);
                        break;
                    case "entityId":
                        c = d.d;
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_GeoDataPointToEntityQuery.h = function (a, b, c, d) {
        a.ad.p(b, "cx");
        var e = null;
        var f = null;
        if (c != null) {
            f = "entityId";
            e = c;
            a.ad.h(f, e);
        }
        f = "entityType";
        e = XmlNamespaceDefinitionBase.p(ST_EntityType_$type, d);
        a.ad.h(f, e);
        ;
    };
    CT_GeoDataPointToEntityQuery.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_GeoDataPointToEntityQuery.$t = markType(CT_GeoDataPointToEntityQuery, 'CT_GeoDataPointToEntityQuery');
    return CT_GeoDataPointToEntityQuery;
}(Base));
export { CT_GeoDataPointToEntityQuery };
