// igniteui-react-excel/esm5/lib/CT_GvmlGraphicalObjectFrame.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_GvmlGraphicFrameNonVisual } from "./CT_GvmlGraphicFrameNonVisual";
import { CT_GraphicalObject } from "./CT_GraphicalObject";
import { CT_Transform2D } from "./CT_Transform2D";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
var CT_GvmlGraphicalObjectFrame = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GvmlGraphicalObjectFrame, _super);
    function CT_GvmlGraphicalObjectFrame() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.f = null;
        _this.h = null;
        _this.d = null;
        return _this;
    }
    Object.defineProperty(CT_GvmlGraphicalObjectFrame.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GvmlGraphicalObjectFrame.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GvmlGraphicalObjectFrame.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GvmlGraphicalObjectFrame.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_GvmlGraphicalObjectFrame.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1103:
                this.a = typeCast(CT_GvmlGraphicFrameNonVisual.$, b);
                break;
            case 727:
                this.e = typeCast(CT_GraphicalObject.$, b);
                break;
            case 1892:
                this.g = typeCast(CT_Transform2D.$, b);
                break;
            case 601:
                this.c = typeCast(CT_OfficeArtExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_GvmlGraphicalObjectFrame.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_GvmlGraphicalObjectFrame.$t = markType(CT_GvmlGraphicalObjectFrame, 'CT_GvmlGraphicalObjectFrame', Base.$, [ISupportSetProperty_$type]);
    return CT_GvmlGraphicalObjectFrame;
}(Base));
export { CT_GvmlGraphicalObjectFrame };
