// igniteui-react-excel/esm5/lib/CT_GeoDataPointToEntityQueryResult.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_GeoDataPointQuery } from "./CT_GeoDataPointQuery";
import { CT_GeoDataPointToEntityQuery } from "./CT_GeoDataPointToEntityQuery";
/**
 * @hidden
 */
var CT_GeoDataPointToEntityQueryResult = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GeoDataPointToEntityQueryResult, _super);
    function CT_GeoDataPointToEntityQueryResult() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.d = null;
        return _this;
    }
    Object.defineProperty(CT_GeoDataPointToEntityQueryResult.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GeoDataPointToEntityQueryResult.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_GeoDataPointToEntityQueryResult.prototype.setProperty = function (a, b) {
        switch (a) {
            case 710:
                this.a = typeCast(CT_GeoDataPointQuery.$, b);
                break;
            case 711:
                this.c = typeCast(CT_GeoDataPointToEntityQuery.$, b);
                break;
            default: break;
        }
    };
    CT_GeoDataPointToEntityQueryResult.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_GeoDataPointToEntityQueryResult.$t = markType(CT_GeoDataPointToEntityQueryResult, 'CT_GeoDataPointToEntityQueryResult', Base.$, [ISupportSetProperty_$type]);
    return CT_GeoDataPointToEntityQueryResult;
}(Base));
export { CT_GeoDataPointToEntityQueryResult };
