// igniteui-react-excel/esm5/lib/CT_GraphicFrame.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_GraphicFrameNonVisual } from "./CT_GraphicFrameNonVisual";
import { CT_Transform2D } from "./CT_Transform2D";
import { CT_GraphicalObject } from "./CT_GraphicalObject";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
var CT_GraphicFrame = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GraphicFrame, _super);
    function CT_GraphicFrame() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.f = null;
        _this.d = null;
        _this.k = null;
        _this.i = toNullable(Boolean_$type, null);
        return _this;
    }
    Object.defineProperty(CT_GraphicFrame.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GraphicFrame.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GraphicFrame.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GraphicFrame.prototype, "j", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GraphicFrame.prototype, "h", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_GraphicFrame.m = function (a, b, c) {
        var e_1, _a;
        b = null;
        c = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "macro":
                        b = d.d;
                        break;
                    case "fPublished":
                        c = toNullable(Boolean_$type, ST_OnOff.e(d.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_GraphicFrame.n = function (a, b, c, d) {
        if (c === void 0) {
            c = toNullable(Boolean_$type, null);
        }
        if (d === void 0) {
            d = null;
        }
        a.ad.p(b, "cd");
        var e = null;
        var f = null;
        if (c.hasValue) {
            f = "fPublished";
            e = ST_OnOff.g(c.value);
            a.ad.h(f, e);
        }
        if (d != null) {
            f = "macro";
            e = d;
            a.ad.h(f, e);
        }
        ;
    };
    CT_GraphicFrame.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1103:
                this.a = typeCast(CT_GraphicFrameNonVisual.$, b);
                break;
            case 1892:
                this.e = typeCast(CT_Transform2D.$, b);
                break;
            case 727:
                this.c = typeCast(CT_GraphicalObject.$, b);
                break;
            default: break;
        }
    };
    CT_GraphicFrame.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_GraphicFrame.$t = markType(CT_GraphicFrame, 'CT_GraphicFrame', Base.$, [ISupportSetProperty_$type]);
    return CT_GraphicFrame;
}(Base));
export { CT_GraphicFrame };
