// igniteui-react-excel/esm5/lib/CT_GraphicalObject_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, Type, markType, String_$type } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
import { Tuple$2 } from "igniteui-react-core";
import { MarkupCompatibility2006 } from "./MarkupCompatibility2006";
import { ICT_ShapePropertiesOwner_$type } from "./ICT_ShapePropertiesOwner";
import { ICT_ShapeStyleOwner_$type } from "./ICT_ShapeStyleOwner";
import { Tuple } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_GraphicalObject = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GraphicalObject, _super);
    function CT_GraphicalObject() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.a = null;
        return _this;
    }
    CT_GraphicalObject.$t = markType(CT_GraphicalObject, 'CT_GraphicalObject');
    return CT_GraphicalObject;
}(Base));
export { CT_GraphicalObject };
/**
 * @hidden
 */
export var IGraphicalObjectContent_$type = /*@__PURE__*/ new Type(null, 'IGraphicalObjectContent');
/**
 * @hidden
 */
var AlternateGraphicalObjectContent = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(AlternateGraphicalObjectContent, _super);
    function AlternateGraphicalObjectContent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.b = null;
        _this.a = null;
        return _this;
    }
    AlternateGraphicalObjectContent.prototype._accept$2$i = function ($tArg, $tResult, a, b) {
        return a._visitAlternateGraphicalObjectContent$i(this, b);
    };
    AlternateGraphicalObjectContent.prototype.getStrictElementName = function (a) {
        return MarkupCompatibility2006.aq;
    };
    AlternateGraphicalObjectContent.$t = markType(AlternateGraphicalObjectContent, 'AlternateGraphicalObjectContent', Base.$, [IGraphicalObjectContent_$type]);
    return AlternateGraphicalObjectContent;
}(Base));
export { AlternateGraphicalObjectContent };
/**
 * @hidden
 */
var CT_Connector = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_Connector, _super);
    function CT_Connector() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.b = false;
        _this.c = null;
        _this.a = null;
        _this._spPr = null;
        _this._style = null;
        return _this;
    }
    Object.defineProperty(CT_Connector.prototype, "spPr", {
        get: function () {
            return this._spPr;
        },
        set: function (a) {
            this._spPr = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Connector.prototype, "style", {
        get: function () {
            return this._style;
        },
        set: function (a) {
            this._style = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_Connector.prototype._accept$2$i = function ($tArg, $tResult, a, b) {
        return a._visitCT_Connector$i(this, b);
    };
    CT_Connector.prototype.getStrictElementName = function (a) {
        return a._t.aw;
    };
    CT_Connector.$t = markType(CT_Connector, 'CT_Connector', Base.$, [IGraphicalObjectContent_$type, ICT_ShapePropertiesOwner_$type, ICT_ShapeStyleOwner_$type]);
    return CT_Connector;
}(Base));
export { CT_Connector };
/**
 * @hidden
 */
export var IShapesOwner_$type = /*@__PURE__*/ new Type(null, 'IShapesOwner');
/**
 * @hidden
 */
var CT_GraphicalObjectData = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GraphicalObjectData, _super);
    function CT_GraphicalObjectData() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.a = null;
        _this.c = null;
        return _this;
    }
    Object.defineProperty(CT_GraphicalObjectData.prototype, "b", {
        get: function () {
            if (this.a == null) {
                this.a = new List$1(Tuple$2.$.specialize(String_$type, Base.$), 0);
            }
            return this.a;
        },
        set: function (a) {
            this.a = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_GraphicalObjectData.prototype.addShape = function (a, b) {
        this.b.add(Tuple.a(String_$type, Base.$, b.getStrictElementName(a), b));
    };
    CT_GraphicalObjectData.$t = markType(CT_GraphicalObjectData, 'CT_GraphicalObjectData', Base.$, [IShapesOwner_$type]);
    return CT_GraphicalObjectData;
}(Base));
export { CT_GraphicalObjectData };
/**
 * @hidden
 */
var CT_GraphicalObjectFrame = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GraphicalObjectFrame, _super);
    function CT_GraphicalObjectFrame() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.d = false;
        _this.a = null;
        _this.e = null;
        _this.b = null;
        _this.c = null;
        return _this;
    }
    CT_GraphicalObjectFrame.prototype._accept$2$i = function ($tArg, $tResult, a, b) {
        return a._visitCT_GraphicalObjectFrame$i(this, b);
    };
    CT_GraphicalObjectFrame.prototype.getStrictElementName = function (a) {
        return a._t.a4;
    };
    CT_GraphicalObjectFrame.$t = markType(CT_GraphicalObjectFrame, 'CT_GraphicalObjectFrame', Base.$, [IGraphicalObjectContent_$type]);
    return CT_GraphicalObjectFrame;
}(Base));
export { CT_GraphicalObjectFrame };
/**
 * @hidden
 */
var CT_GroupShape = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GroupShape, _super);
    function CT_GroupShape() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.b = null;
        _this.a = null;
        _this.c = new List$1(IGraphicalObjectContent_$type, 0);
        return _this;
    }
    Object.defineProperty(CT_GroupShape.prototype, "d", {
        get: function () {
            return this.c;
        },
        enumerable: false,
        configurable: true
    });
    CT_GroupShape.prototype._accept$2$i = function ($tArg, $tResult, a, b) {
        return a._visitCT_GroupShape$i(this, b);
    };
    CT_GroupShape.prototype.getStrictElementName = function (a) {
        return a._t.a5;
    };
    CT_GroupShape.prototype.addShape = function (a, b) {
        this.d.add(b);
    };
    CT_GroupShape.$t = markType(CT_GroupShape, 'CT_GroupShape', Base.$, [IGraphicalObjectContent_$type, IShapesOwner_$type]);
    return CT_GroupShape;
}(Base));
export { CT_GroupShape };
/**
 * @hidden
 */
var CT_Picture = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_Picture, _super);
    function CT_Picture() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.a = null;
        _this.c = false;
        _this.d = null;
        _this.b = null;
        _this._spPr = null;
        return _this;
    }
    Object.defineProperty(CT_Picture.prototype, "spPr", {
        get: function () {
            return this._spPr;
        },
        set: function (a) {
            this._spPr = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_Picture.prototype._accept$2$i = function ($tArg, $tResult, a, b) {
        return a._visitCT_Picture$i(this, b);
    };
    CT_Picture.prototype.getStrictElementName = function (a) {
        return a._t.bt;
    };
    CT_Picture.$t = markType(CT_Picture, 'CT_Picture', Base.$, [IGraphicalObjectContent_$type, ICT_ShapePropertiesOwner_$type]);
    return CT_Picture;
}(Base));
export { CT_Picture };
/**
 * @hidden
 */
var CT_Shape = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_Shape, _super);
    function CT_Shape() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.c = false;
        _this.d = false;
        _this.e = null;
        _this.f = null;
        _this.a = null;
        _this._spPr = null;
        _this._style = null;
        _this.b = null;
        return _this;
    }
    Object.defineProperty(CT_Shape.prototype, "spPr", {
        get: function () {
            return this._spPr;
        },
        set: function (a) {
            this._spPr = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Shape.prototype, "style", {
        get: function () {
            return this._style;
        },
        set: function (a) {
            this._style = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_Shape.prototype._accept$2$i = function ($tArg, $tResult, a, b) {
        return a._visitCT_Shape$i(this, b);
    };
    CT_Shape.prototype.getStrictElementName = function (a) {
        return a._t.bx;
    };
    CT_Shape.$t = markType(CT_Shape, 'CT_Shape', Base.$, [IGraphicalObjectContent_$type, ICT_ShapePropertiesOwner_$type, ICT_ShapeStyleOwner_$type]);
    return CT_Shape;
}(Base));
export { CT_Shape };
/**
 * @hidden
 */
export var IGraphicalObjectContentVisitor$2_$type = /*@__PURE__*/ new Type(null, 'IGraphicalObjectContentVisitor$2');
