// igniteui-react-excel/esm5/lib/CT_GeoDataEntityQueryResult.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_GeoDataEntityQuery } from "./CT_GeoDataEntityQuery";
import { CT_GeoData } from "./CT_GeoData";
/**
 * @hidden
 */
var CT_GeoDataEntityQueryResult = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GeoDataEntityQueryResult, _super);
    function CT_GeoDataEntityQueryResult() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.b = null;
        return _this;
    }
    Object.defineProperty(CT_GeoDataEntityQueryResult.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GeoDataEntityQueryResult.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_GeoDataEntityQueryResult.prototype.setProperty = function (a, b) {
        switch (a) {
            case 707:
                this.c = typeCast(CT_GeoDataEntityQuery.$, b);
                break;
            case 706:
                this.a = typeCast(CT_GeoData.$, b);
                break;
            default: break;
        }
    };
    CT_GeoDataEntityQueryResult.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_GeoDataEntityQueryResult.$t = markType(CT_GeoDataEntityQueryResult, 'CT_GeoDataEntityQueryResult', Base.$, [ISupportSetProperty_$type]);
    return CT_GeoDataEntityQueryResult;
}(Base));
export { CT_GeoDataEntityQueryResult };
