// igniteui-react-excel/esm5/lib/CT_GeoDataPointQuery.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_EntityType_$type } from "./ST_EntityType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { CultureInfo } from "igniteui-react-core";
import { parseNumber } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_GeoDataPointQuery = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GeoDataPointQuery, _super);
    function CT_GeoDataPointQuery() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.c = 0;
        _this.e = 0;
        _this.g = 0;
        return _this;
    }
    Object.defineProperty(CT_GeoDataPointQuery.prototype, "b", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            this.c = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GeoDataPointQuery.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GeoDataPointQuery.prototype, "f", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_GeoDataPointQuery.a = function (a) {
        var b = 0;
        var c = 0;
        var d = 0;
        var e = CT_GeoDataPointQuery.i(a, b, c, d);
        b = e.p1;
        c = e.p2;
        d = e.p3;
        return ((function () {
            var $ret = new CT_GeoDataPointQuery();
            $ret.b = b;
            $ret.d = c;
            $ret.f = d;
            return $ret;
        })());
    };
    CT_GeoDataPointQuery.i = function (a, b, c, d) {
        var e_1, _a;
        b = 0;
        c = 0;
        d = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "entityType":
                        b = XmlNamespaceDefinitionBase.b(ST_EntityType_$type, e.d);
                        break;
                    case "latitude":
                        c = parseNumber(e.d, CultureInfo.invariantCulture);
                        break;
                    case "longitude":
                        d = parseNumber(e.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_GeoDataPointQuery.j = function (a, b, c, d, e) {
        a.ad.p(b, "cx");
        var f = null;
        var g = null;
        g = "entityType";
        f = XmlNamespaceDefinitionBase.p(ST_EntityType_$type, c);
        a.ad.h(g, f);
        g = "latitude";
        f = d.toString();
        a.ad.h(g, f);
        g = "longitude";
        f = e.toString();
        a.ad.h(g, f);
        ;
    };
    CT_GeoDataPointQuery.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_GeoDataPointQuery.$t = markType(CT_GeoDataPointQuery, 'CT_GeoDataPointQuery');
    return CT_GeoDataPointQuery;
}(Base));
export { CT_GeoDataPointQuery };
