// igniteui-react-excel/esm5/lib/CT_GeoLocationQueryResult.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_GeoLocationQuery } from "./CT_GeoLocationQuery";
import { CT_GeoLocations } from "./CT_GeoLocations";
/**
 * @hidden
 */
var CT_GeoLocationQueryResult = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_GeoLocationQueryResult, _super);
    function CT_GeoLocationQueryResult() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.d = null;
        return _this;
    }
    Object.defineProperty(CT_GeoLocationQueryResult.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_GeoLocationQueryResult.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_GeoLocationQueryResult.prototype.setProperty = function (a, b) {
        switch (a) {
            case 716:
                this.a = typeCast(CT_GeoLocationQuery.$, b);
                break;
            case 719:
                this.c = typeCast(CT_GeoLocations.$, b);
                break;
            default: break;
        }
    };
    CT_GeoLocationQueryResult.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_GeoLocationQueryResult.$t = markType(CT_GeoLocationQueryResult, 'CT_GeoLocationQueryResult', Base.$, [ISupportSetProperty_$type]);
    return CT_GeoLocationQueryResult;
}(Base));
export { CT_GeoLocationQueryResult };
