// igniteui-react-excel/esm5/lib/CT_IgnoredError.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_Sqref } from "./ST_Sqref";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
var CT_IgnoredError = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_IgnoredError, _super);
    function CT_IgnoredError() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.q = toNullable(Boolean_$type, null);
        _this.aa = toNullable(Boolean_$type, null);
        _this.y = toNullable(Boolean_$type, null);
        _this.s = toNullable(Boolean_$type, null);
        _this.u = toNullable(Boolean_$type, null);
        _this.ac = toNullable(Boolean_$type, null);
        _this.o = toNullable(Boolean_$type, null);
        _this.w = toNullable(Boolean_$type, null);
        _this.m = toNullable(Boolean_$type, null);
        return _this;
    }
    Object.defineProperty(CT_IgnoredError.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_IgnoredError.prototype, "p", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_IgnoredError.prototype, "z", {
        get: function () {
            return this.aa;
        },
        set: function (a) {
            this.aa = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_IgnoredError.prototype, "x", {
        get: function () {
            return this.y;
        },
        set: function (a) {
            this.y = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_IgnoredError.prototype, "r", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_IgnoredError.prototype, "t", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            this.u = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_IgnoredError.prototype, "ab", {
        get: function () {
            return this.ac;
        },
        set: function (a) {
            this.ac = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_IgnoredError.prototype, "n", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_IgnoredError.prototype, "v", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            this.w = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_IgnoredError.prototype, "l", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_IgnoredError.ae = function (a, b, c, d, e, f, g, h, i, j, k) {
        var e_1, _a;
        b = null;
        c = toNullable(Boolean_$type, null);
        d = toNullable(Boolean_$type, null);
        e = toNullable(Boolean_$type, null);
        f = toNullable(Boolean_$type, null);
        g = toNullable(Boolean_$type, null);
        h = toNullable(Boolean_$type, null);
        i = toNullable(Boolean_$type, null);
        j = toNullable(Boolean_$type, null);
        k = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var l = _c.value;
                switch (l.a.e(a.d)) {
                    case "sqref":
                        b = ST_Sqref.a(l.d);
                        break;
                    case "evalError":
                        c = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "twoDigitTextYear":
                        d = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "numberStoredAsText":
                        e = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "formula":
                        f = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "formulaRange":
                        g = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "unlockedFormula":
                        h = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "emptyCellReference":
                        i = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "listDataValidation":
                        j = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "calculatedColumn":
                        k = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e,
            p5: f,
            p6: g,
            p7: h,
            p8: i,
            p9: j,
            p10: k
        };
    };
    CT_IgnoredError.af = function (a, b, c, d, e, f, g, h, i, j, k, l) {
        a.ad.p(b, null);
        var m = null;
        var n = null;
        if (c.hasValue) {
            n = "calculatedColumn";
            m = ST_OnOff.g(c.value);
            a.ad.h(n, m);
        }
        if (d.hasValue) {
            n = "emptyCellReference";
            m = ST_OnOff.g(d.value);
            a.ad.h(n, m);
        }
        if (e.hasValue) {
            n = "evalError";
            m = ST_OnOff.g(e.value);
            a.ad.h(n, m);
        }
        if (f.hasValue) {
            n = "formula";
            m = ST_OnOff.g(f.value);
            a.ad.h(n, m);
        }
        if (g.hasValue) {
            n = "formulaRange";
            m = ST_OnOff.g(g.value);
            a.ad.h(n, m);
        }
        if (h.hasValue) {
            n = "listDataValidation";
            m = ST_OnOff.g(h.value);
            a.ad.h(n, m);
        }
        if (i.hasValue) {
            n = "numberStoredAsText";
            m = ST_OnOff.g(i.value);
            a.ad.h(n, m);
        }
        if (j.hasValue) {
            n = "twoDigitTextYear";
            m = ST_OnOff.g(j.value);
            a.ad.h(n, m);
        }
        if (k.hasValue) {
            n = "unlockedFormula";
            m = ST_OnOff.g(k.value);
            a.ad.h(n, m);
        }
        if (l != null) {
            n = "sqref";
            m = l.e();
            a.ad.h(n, m);
        }
    };
    CT_IgnoredError.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_IgnoredError.$t = markType(CT_IgnoredError, 'CT_IgnoredError');
    return CT_IgnoredError;
}(Base));
export { CT_IgnoredError };
