// igniteui-react-excel/esm5/lib/CT_LightRig.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_SphereCoords } from "./CT_SphereCoords";
import { ST_LightRigType_$type } from "./ST_LightRigType";
import { ST_LightRigDirection_$type } from "./ST_LightRigDirection";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
var CT_LightRig = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_LightRig, _super);
    function CT_LightRig() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.f = 0;
        _this.d = 0;
        return _this;
    }
    Object.defineProperty(CT_LightRig.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_LightRig.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_LightRig.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_LightRig.h = function (a, b, c) {
        var e_1, _a;
        b = 0;
        c = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "rig":
                        b = XmlNamespaceDefinitionBase.b(ST_LightRigType_$type, d.d);
                        break;
                    case "dir":
                        c = XmlNamespaceDefinitionBase.b(ST_LightRigDirection_$type, d.d);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_LightRig.i = function (a, b, c, d) {
        a.ad.p(b, "a");
        var e = null;
        var f = null;
        f = "dir";
        e = XmlNamespaceDefinitionBase.p(ST_LightRigDirection_$type, c);
        a.ad.h(f, e);
        f = "rig";
        e = XmlNamespaceDefinitionBase.p(ST_LightRigType_$type, d);
        a.ad.h(f, e);
        ;
    };
    CT_LightRig.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1370:
                this.a = typeCast(CT_SphereCoords.$, b);
                break;
            default: break;
        }
    };
    CT_LightRig.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_LightRig.$t = markType(CT_LightRig, 'CT_LightRig', Base.$, [ISupportSetProperty_$type]);
    return CT_LightRig;
}(Base));
export { CT_LightRig };
