// igniteui-react-excel/esm5/lib/CT_NumFmt_Chart.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
var CT_NumFmt_Chart = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_NumFmt_Chart, _super);
    function CT_NumFmt_Chart() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.d = null;
        _this.b = false;
        return _this;
    }
    Object.defineProperty(CT_NumFmt_Chart.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_NumFmt_Chart.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_NumFmt_Chart.f = function (a, b, c) {
        var e_1, _a;
        b = null;
        c = false;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "formatCode":
                        b = d.d;
                        break;
                    case "sourceLinked":
                        c = ST_OnOff.e(d.d);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_NumFmt_Chart.g = function (a, b, c, d) {
        a.ad.p(b, "c");
        var e = null;
        var f = null;
        if (c != null) {
            f = "formatCode";
            e = c;
            a.ad.h(f, e);
        }
        f = "sourceLinked";
        e = ST_OnOff.g(d);
        a.ad.h(f, e);
        ;
    };
    CT_NumFmt_Chart.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_NumFmt_Chart.$t = markType(CT_NumFmt_Chart, 'CT_NumFmt_Chart');
    return CT_NumFmt_Chart;
}(Base));
export { CT_NumFmt_Chart };
