// igniteui-react-excel/esm5/lib/CT_NumericLevel.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { List$1 } from "igniteui-react-core";
import { CT_NumericValue } from "./CT_NumericValue";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_NumericLevel = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_NumericLevel, _super);
    function CT_NumericLevel() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.i = 0;
        _this.d = null;
        _this.f = null;
        _this.a = null;
        _this.a = new List$1(CT_NumericValue.$, 0);
        return _this;
    }
    Object.defineProperty(CT_NumericLevel.prototype, "h", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_NumericLevel.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_NumericLevel.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_NumericLevel.j = function (a, b, c, d) {
        var e_1, _a;
        b = 0;
        c = null;
        d = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "ptCount":
                        b = parseUInt32_1(e.d, CultureInfo.invariantCulture);
                        break;
                    case "formatCode":
                        c = e.d;
                        break;
                    case "name":
                        d = e.d;
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_NumericLevel.k = function (a, b, c, d, e) {
        a.ad.p(b, "cx");
        var f = null;
        var g = null;
        if (c != null) {
            g = "formatCode";
            f = c;
            a.ad.h(g, f);
        }
        if (d != null) {
            g = "name";
            f = d;
            a.ad.h(g, f);
        }
        g = "ptCount";
        f = e.toString();
        a.ad.h(g, f);
        ;
    };
    CT_NumericLevel.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1277:
                this.a.add(typeCast(CT_NumericValue.$, b));
                break;
            default: break;
        }
    };
    CT_NumericLevel.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_NumericLevel.$t = markType(CT_NumericLevel, 'CT_NumericLevel', Base.$, [ISupportSetProperty_$type]);
    return CT_NumericLevel;
}(Base));
export { CT_NumericLevel };
