// igniteui-react-excel/esm5/lib/CT_OlapSlicerCacheLevelData.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, fromEnum, markType } from "igniteui-react-core";
import { ST_OlapSlicerCacheSortOrder_$type } from "./ST_OlapSlicerCacheSortOrder";
import { ST_SlicerCacheCrossFilter_$type } from "./ST_SlicerCacheCrossFilter";
import { CultureInfo } from "igniteui-react-core";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_OlapSlicerCacheLevelData = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_OlapSlicerCacheLevelData, _super);
    function CT_OlapSlicerCacheLevelData() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.l = null;
        _this.j = null;
        _this.n = 0;
        _this.h = toNullable(ST_OlapSlicerCacheSortOrder_$type, null);
        _this.f = toNullable(ST_SlicerCacheCrossFilter_$type, null);
        return _this;
    }
    Object.defineProperty(CT_OlapSlicerCacheLevelData.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_OlapSlicerCacheLevelData.prototype, "k", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_OlapSlicerCacheLevelData.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_OlapSlicerCacheLevelData.prototype, "m", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_OlapSlicerCacheLevelData.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_OlapSlicerCacheLevelData.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_OlapSlicerCacheLevelData.o = function (a, b, c, d, e, f) {
        var e_1, _a;
        b = null;
        c = null;
        d = 0;
        e = toNullable(ST_OlapSlicerCacheSortOrder_$type, null);
        f = toNullable(ST_SlicerCacheCrossFilter_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var g = _c.value;
                switch (g.a.e(a.d)) {
                    case "uniqueName":
                        b = g.d;
                        break;
                    case "sourceCaption":
                        c = g.d;
                        break;
                    case "count":
                        d = parseUInt32_1(g.d, CultureInfo.invariantCulture);
                        break;
                    case "sortOrder":
                        e = toNullable(ST_OlapSlicerCacheSortOrder_$type, XmlNamespaceDefinitionBase.b(ST_OlapSlicerCacheSortOrder_$type, g.d));
                        break;
                    case "crossFilter":
                        f = toNullable(ST_SlicerCacheCrossFilter_$type, XmlNamespaceDefinitionBase.b(ST_SlicerCacheCrossFilter_$type, g.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e,
            p5: f
        };
    };
    CT_OlapSlicerCacheLevelData.p = function (a, b, c, d, e, f, g) {
        a.ad.p(b, null);
        var h = null;
        var i = null;
        if (c.hasValue) {
            i = "crossFilter";
            h = XmlNamespaceDefinitionBase.p(ST_SlicerCacheCrossFilter_$type, c.value);
            a.ad.h(i, h);
        }
        if (d.hasValue) {
            i = "sortOrder";
            h = XmlNamespaceDefinitionBase.p(ST_OlapSlicerCacheSortOrder_$type, d.value);
            a.ad.h(i, h);
        }
        if (e != null) {
            i = "sourceCaption";
            h = e;
            a.ad.h(i, h);
        }
        i = "count";
        h = f.toString();
        a.ad.h(i, h);
        if (g != null) {
            i = "uniqueName";
            h = g;
            a.ad.h(i, h);
        }
        ;
    };
    CT_OlapSlicerCacheLevelData.$t = markType(CT_OlapSlicerCacheLevelData, 'CT_OlapSlicerCacheLevelData');
    return CT_OlapSlicerCacheLevelData;
}(Base));
export { CT_OlapSlicerCacheLevelData };
