// igniteui-react-excel/esm5/lib/CT_OleItem.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
var CT_OleItem = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_OleItem, _super);
    function CT_OleItem() {
        var _this = _super.call(this) || this;
        _this.g = null;
        _this.e = toNullable(Boolean_$type, null);
        _this.d = toNullable(Boolean_$type, null);
        _this.f = toNullable(Boolean_$type, null);
        return _this;
    }
    CT_OleItem.i = function (a, b, c, d, e) {
        var e_1, _a;
        b = null;
        c = toNullable(Boolean_$type, null);
        d = toNullable(Boolean_$type, null);
        e = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var f = _c.value;
                switch (f.a.e(a.d)) {
                    case "name":
                        b = f.d;
                        break;
                    case "icon":
                        c = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "advise":
                        d = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "preferPic":
                        e = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e
        };
    };
    CT_OleItem.j = function (a, b, c, d, e, f) {
        a.ad.p(b, null);
        var g = null;
        var h = null;
        if (c.hasValue) {
            h = "advise";
            g = ST_OnOff.g(c.value);
            a.ad.h(h, g);
        }
        if (d.hasValue) {
            h = "icon";
            g = ST_OnOff.g(d.value);
            a.ad.h(h, g);
        }
        if (e.hasValue) {
            h = "preferPic";
            g = ST_OnOff.g(e.value);
            a.ad.h(h, g);
        }
        if (f != null) {
            h = "name";
            g = f;
            a.ad.h(h, g);
        }
    };
    CT_OleItem.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_OleItem.$t = markType(CT_OleItem, 'CT_OleItem');
    return CT_OleItem;
}(Base));
export { CT_OleItem };
