// igniteui-react-excel/esm5/lib/CT_OlapSlicerCache.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_OlapSlicerCache = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_OlapSlicerCache, _super);
    function CT_OlapSlicerCache() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.f = null;
        _this.b = null;
        _this.h = 0;
        return _this;
    }
    Object.defineProperty(CT_OlapSlicerCache.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_OlapSlicerCache.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_OlapSlicerCache.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_OlapSlicerCache.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_OlapSlicerCache.i = function (a, b) {
        var e_1, _a;
        b = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var c = _c.value;
                switch (c.a.e(a.d)) {
                    case "pivotCacheId":
                        b = parseUInt32_1(c.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b
        };
    };
    CT_OlapSlicerCache.j = function (a, b, c) {
        a.ad.p(b, null);
        var d = null;
        var e = null;
        e = "pivotCacheId";
        d = c.toString();
        a.ad.h(e, d);
        ;
    };
    CT_OlapSlicerCache.$t = markType(CT_OlapSlicerCache, 'CT_OlapSlicerCache');
    return CT_OlapSlicerCache;
}(Base));
export { CT_OlapSlicerCache };
