// igniteui-react-excel/esm5/lib/CT_OlapSlicerCacheItem.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
/**
 * @hidden
 */
var CT_OlapSlicerCacheItem = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_OlapSlicerCacheItem, _super);
    function CT_OlapSlicerCacheItem() {
        var _this = _super.call(this) || this;
        _this.c = null;
        _this.i = null;
        _this.g = null;
        _this.e = toNullable(Boolean_$type, null);
        return _this;
    }
    Object.defineProperty(CT_OlapSlicerCacheItem.prototype, "b", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            this.c = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_OlapSlicerCacheItem.prototype, "h", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_OlapSlicerCacheItem.prototype, "f", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_OlapSlicerCacheItem.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_OlapSlicerCacheItem.j = function (a, b, c, d) {
        var e_1, _a;
        b = null;
        c = null;
        d = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "n":
                        b = e.d;
                        break;
                    case "c":
                        c = e.d;
                        break;
                    case "nd":
                        d = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_OlapSlicerCacheItem.k = function (a, b, c, d, e) {
        a.ad.p(b, null);
        var f = null;
        var g = null;
        if (c != null) {
            g = "c";
            f = c;
            a.ad.h(g, f);
        }
        if (d.hasValue) {
            g = "nd";
            f = ST_OnOff.g(d.value);
            a.ad.h(g, f);
        }
        if (e != null) {
            g = "n";
            f = e;
            a.ad.h(g, f);
        }
        ;
    };
    CT_OlapSlicerCacheItem.$t = markType(CT_OlapSlicerCacheItem, 'CT_OlapSlicerCacheItem');
    return CT_OlapSlicerCacheItem;
}(Base));
export { CT_OlapSlicerCacheItem };
