// igniteui-react-excel/esm5/lib/CT_PresetTextShape.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { ST_TextShapeType_$type } from "./ST_TextShapeType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { NotImplementedException } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_PresetTextShape = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_PresetTextShape, _super);
    function CT_PresetTextShape() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.b = 0;
        return _this;
    }
    Object.defineProperty(CT_PresetTextShape.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PresetTextShape.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_PresetTextShape.f = function (a, b) {
        var e_1, _a;
        b = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var c = _c.value;
                switch (c.a.e(a.d)) {
                    case "prst":
                        b = XmlNamespaceDefinitionBase.b(ST_TextShapeType_$type, c.d);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b
        };
    };
    CT_PresetTextShape.g = function (a, b, c) {
        a.ad.p(b, "a");
        var d = null;
        var e = null;
        e = "prst";
        d = XmlNamespaceDefinitionBase.p(ST_TextShapeType_$type, c);
        a.ad.h(e, d);
        ;
    };
    CT_PresetTextShape.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_PresetTextShape.prototype.setProperty = function (a, b) {
        throw new NotImplementedException(0);
    };
    CT_PresetTextShape.$t = markType(CT_PresetTextShape, 'CT_PresetTextShape', Base.$, [ISupportSetProperty_$type]);
    return CT_PresetTextShape;
}(Base));
export { CT_PresetTextShape };
