// igniteui-react-excel/esm5/lib/CT_PivotChange.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType } from "igniteui-react-core";
import { ST_AllocationMethod_$type } from "./ST_AllocationMethod";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
var CT_PivotChange = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_PivotChange, _super);
    function CT_PivotChange() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.i = null;
        _this.b = null;
        _this.f = 0;
        _this.k = null;
        return _this;
    }
    Object.defineProperty(CT_PivotChange.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotChange.prototype, "h", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotChange.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotChange.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotChange.prototype, "j", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_PivotChange.l = function (a, b, c) {
        var e_1, _a;
        b = 0;
        c = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "allocationMethod":
                        b = XmlNamespaceDefinitionBase.b(ST_AllocationMethod_$type, d.d);
                        break;
                    case "weightExpression":
                        c = d.d;
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_PivotChange.m = function (a, b, c, d) {
        a.ad.p(b, null);
        var e = null;
        var f = null;
        if (c != null) {
            f = "weightExpression";
            e = c;
            a.ad.h(f, e);
        }
        f = "allocationMethod";
        e = XmlNamespaceDefinitionBase.p(ST_AllocationMethod_$type, d);
        a.ad.h(f, e);
        ;
    };
    CT_PivotChange.$t = markType(CT_PivotChange, 'CT_PivotChange');
    return CT_PivotChange;
}(Base));
export { CT_PivotChange };
