// igniteui-react-excel/esm5/lib/CT_ProtectedRange_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Number_$type, fromEnum, markType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_ProtectedRange_SML2009 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_ProtectedRange_SML2009, _super);
    function CT_ProtectedRange_SML2009() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.l = null;
        _this.f = null;
        _this.h = null;
        _this.n = null;
        _this.d = toNullable(Number_$type, null);
        _this.j = null;
        _this.p = null;
        return _this;
    }
    Object.defineProperty(CT_ProtectedRange_SML2009.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ProtectedRange_SML2009.prototype, "k", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ProtectedRange_SML2009.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ProtectedRange_SML2009.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ProtectedRange_SML2009.prototype, "m", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ProtectedRange_SML2009.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ProtectedRange_SML2009.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ProtectedRange_SML2009.prototype, "o", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_ProtectedRange_SML2009.q = function (a, b, c, d, e, f, g, h) {
        var e_1, _a;
        b = null;
        c = null;
        d = null;
        e = null;
        f = toNullable(Number_$type, null);
        g = null;
        h = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var i = _c.value;
                switch (i.a.e(a.d)) {
                    case "password":
                        b = i.d;
                        break;
                    case "algorithmName":
                        c = i.d;
                        break;
                    case "hashValue":
                        d = i.d;
                        break;
                    case "saltValue":
                        e = i.d;
                        break;
                    case "spinCount":
                        f = toNullable(Number_$type, parseUInt32_1(i.d, CultureInfo.invariantCulture));
                        break;
                    case "name":
                        g = i.d;
                        break;
                    case "securityDescriptor":
                        h = i.d;
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e,
            p5: f,
            p6: g,
            p7: h
        };
    };
    CT_ProtectedRange_SML2009.r = function (a, b, c, d, e, f, g, h, i) {
        a.ad.p(b, null);
        var j = null;
        var k = null;
        if (c != null) {
            k = "algorithmName";
            j = c;
            a.ad.h(k, j);
        }
        if (d != null) {
            k = "hashValue";
            j = d;
            a.ad.h(k, j);
        }
        if (e != null) {
            k = "password";
            j = e;
            a.ad.h(k, j);
        }
        if (f != null) {
            k = "saltValue";
            j = f;
            a.ad.h(k, j);
        }
        if (g != null) {
            k = "securityDescriptor";
            j = g;
            a.ad.h(k, j);
        }
        if (h.hasValue) {
            k = "spinCount";
            j = h.value.toString();
            a.ad.h(k, j);
        }
        if (i != null) {
            k = "name";
            j = i;
            a.ad.h(k, j);
        }
        ;
    };
    CT_ProtectedRange_SML2009.$t = markType(CT_ProtectedRange_SML2009, 'CT_ProtectedRange_SML2009');
    return CT_ProtectedRange_SML2009;
}(Base));
export { CT_ProtectedRange_SML2009 };
