// igniteui-react-excel/esm5/lib/CT_PivotTableDefinition.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { ST_AllocationMethod_$type } from "./ST_AllocationMethod";
import { ST_OnOff } from "./ST_OnOff";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
var CT_PivotTableDefinition = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_PivotTableDefinition, _super);
    function CT_PivotTableDefinition() {
        var _this = _super.call(this) || this;
        _this.m = null;
        _this.k = null;
        _this.b = null;
        _this.w = toNullable(Boolean_$type, null);
        _this.aa = toNullable(Boolean_$type, null);
        _this.s = toNullable(Boolean_$type, null);
        _this.ac = null;
        _this.ae = null;
        _this.u = toNullable(Boolean_$type, null);
        _this.q = toNullable(Boolean_$type, null);
        _this.o = toNullable(ST_AllocationMethod_$type, null);
        _this.ag = null;
        _this.y = toNullable(Boolean_$type, null);
        return _this;
    }
    Object.defineProperty(CT_PivotTableDefinition.prototype, "l", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "j", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "v", {
        get: function () {
            return this.w;
        },
        set: function (a) {
            this.w = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "z", {
        get: function () {
            return this.aa;
        },
        set: function (a) {
            this.aa = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "r", {
        get: function () {
            return this.s;
        },
        set: function (a) {
            this.s = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "ab", {
        get: function () {
            return this.ac;
        },
        set: function (a) {
            this.ac = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "ad", {
        get: function () {
            return this.ae;
        },
        set: function (a) {
            this.ae = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "t", {
        get: function () {
            return this.u;
        },
        set: function (a) {
            this.u = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "p", {
        get: function () {
            return this.q;
        },
        set: function (a) {
            this.q = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "n", {
        get: function () {
            return this.o;
        },
        set: function (a) {
            this.o = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "af", {
        get: function () {
            return this.ag;
        },
        set: function (a) {
            this.ag = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotTableDefinition.prototype, "x", {
        get: function () {
            return this.y;
        },
        set: function (a) {
            this.y = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_PivotTableDefinition.ah = function (a, b, c, d, e, f, g, h, i, j, k) {
        var e_1, _a;
        b = toNullable(Boolean_$type, null);
        c = toNullable(Boolean_$type, null);
        d = toNullable(Boolean_$type, null);
        e = null;
        f = null;
        g = toNullable(Boolean_$type, null);
        h = toNullable(Boolean_$type, null);
        i = toNullable(ST_AllocationMethod_$type, null);
        j = null;
        k = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var l = _c.value;
                switch (l.a.e(a.d)) {
                    case "fillDownLabelsDefault":
                        b = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "visualTotalsForSets":
                        c = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "calculatedMembersInFilters":
                        d = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "altText":
                        e = l.d;
                        break;
                    case "altTextSummary":
                        f = l.d;
                        break;
                    case "enableEdit":
                        g = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "autoApply":
                        h = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    case "allocationMethod":
                        i = toNullable(ST_AllocationMethod_$type, XmlNamespaceDefinitionBase.b(ST_AllocationMethod_$type, l.d));
                        break;
                    case "weightExpression":
                        j = l.d;
                        break;
                    case "hideValuesRow":
                        k = toNullable(Boolean_$type, ST_OnOff.e(l.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e,
            p5: f,
            p6: g,
            p7: h,
            p8: i,
            p9: j,
            p10: k
        };
    };
    CT_PivotTableDefinition.ai = function (a, b, c, d, e, f, g, h, i, j, k, l) {
        if (c === void 0) {
            c = toNullable(ST_AllocationMethod_$type, null);
        }
        if (d === void 0) {
            d = null;
        }
        if (e === void 0) {
            e = null;
        }
        if (f === void 0) {
            f = toNullable(Boolean_$type, null);
        }
        if (g === void 0) {
            g = toNullable(Boolean_$type, null);
        }
        if (h === void 0) {
            h = toNullable(Boolean_$type, null);
        }
        if (i === void 0) {
            i = toNullable(Boolean_$type, null);
        }
        if (j === void 0) {
            j = toNullable(Boolean_$type, null);
        }
        if (k === void 0) {
            k = toNullable(Boolean_$type, null);
        }
        if (l === void 0) {
            l = null;
        }
        a.ad.p(b, null);
        var m = null;
        var n = null;
        if (c.hasValue) {
            n = "allocationMethod";
            m = XmlNamespaceDefinitionBase.p(ST_AllocationMethod_$type, c.value);
            a.ad.h(n, m);
        }
        if (d != null) {
            n = "altText";
            m = d;
            a.ad.h(n, m);
        }
        if (e != null) {
            n = "altTextSummary";
            m = e;
            a.ad.h(n, m);
        }
        if (f.hasValue) {
            n = "autoApply";
            m = ST_OnOff.g(f.value);
            a.ad.h(n, m);
        }
        if (g.hasValue) {
            n = "calculatedMembersInFilters";
            m = ST_OnOff.g(g.value);
            a.ad.h(n, m);
        }
        if (h.hasValue) {
            n = "enableEdit";
            m = ST_OnOff.g(h.value);
            a.ad.h(n, m);
        }
        if (i.hasValue) {
            n = "fillDownLabelsDefault";
            m = ST_OnOff.g(i.value);
            a.ad.h(n, m);
        }
        if (j.hasValue) {
            n = "hideValuesRow";
            m = ST_OnOff.g(j.value);
            a.ad.h(n, m);
        }
        if (k.hasValue) {
            n = "visualTotalsForSets";
            m = ST_OnOff.g(k.value);
            a.ad.h(n, m);
        }
        if (l != null) {
            n = "weightExpression";
            m = l;
            a.ad.h(n, m);
        }
        ;
    };
    CT_PivotTableDefinition.$t = markType(CT_PivotTableDefinition, 'CT_PivotTableDefinition');
    return CT_PivotTableDefinition;
}(Base));
export { CT_PivotTableDefinition };
