// igniteui-react-excel/esm5/lib/CT_Point3D.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseInt64_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_Point3D = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_Point3D, _super);
    function CT_Point3D() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.b = 0;
        _this.d = 0;
        _this.f = 0;
        return _this;
    }
    Object.defineProperty(CT_Point3D.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Point3D.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Point3D.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_Point3D.h = function (a, b, c, d) {
        var e_1, _a;
        b = 0;
        c = 0;
        d = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "x":
                        b = parseInt64_1(e.d, CultureInfo.invariantCulture);
                        break;
                    case "y":
                        c = parseInt64_1(e.d, CultureInfo.invariantCulture);
                        break;
                    case "z":
                        d = parseInt64_1(e.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_Point3D.i = function (a, b, c, d, e) {
        a.ad.p(b, "a");
        var f = null;
        var g = null;
        g = "x";
        f = c.toString();
        a.ad.h(g, f);
        g = "y";
        f = d.toString();
        a.ad.h(g, f);
        g = "z";
        f = e.toString();
        a.ad.h(g, f);
        ;
    };
    CT_Point3D.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_Point3D.$t = markType(CT_Point3D, 'CT_Point3D');
    return CT_Point3D;
}(Base));
export { CT_Point3D };
