// igniteui-react-excel/esm5/lib/CT_PageMargins_Chart.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseNumber } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_PageMargins_Chart = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_PageMargins_Chart, _super);
    function CT_PageMargins_Chart() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.i = 0;
        _this.k = 0;
        _this.m = 0;
        _this.c = 0;
        _this.g = 0;
        _this.e = 0;
        return _this;
    }
    Object.defineProperty(CT_PageMargins_Chart.prototype, "h", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PageMargins_Chart.prototype, "j", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PageMargins_Chart.prototype, "l", {
        get: function () {
            return this.m;
        },
        set: function (a) {
            this.m = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PageMargins_Chart.prototype, "b", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            this.c = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PageMargins_Chart.prototype, "f", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PageMargins_Chart.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_PageMargins_Chart.a = function (a) {
        var b = 0;
        var c = 0;
        var d = 0;
        var e = 0;
        var f = 0;
        var g = 0;
        var h = CT_PageMargins_Chart.o(a, b, c, d, e, f, g);
        b = h.p1;
        c = h.p2;
        d = h.p3;
        e = h.p4;
        f = h.p5;
        g = h.p6;
        return ((function () {
            var $ret = new CT_PageMargins_Chart();
            $ret.b = e;
            $ret.d = g;
            $ret.f = f;
            $ret.h = b;
            $ret.j = c;
            $ret.l = d;
            return $ret;
        })());
    };
    CT_PageMargins_Chart.o = function (a, b, c, d, e, f, g) {
        var e_1, _a;
        b = 0;
        c = 0;
        d = 0;
        e = 0;
        f = 0;
        g = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var h = _c.value;
                switch (h.a.e(a.d)) {
                    case "l":
                        b = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "r":
                        c = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "t":
                        d = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "b":
                        e = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "header":
                        f = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "footer":
                        g = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e,
            p5: f,
            p6: g
        };
    };
    CT_PageMargins_Chart.p = function (a, b, c, d, e, f, g, h) {
        a.ad.p(b, "c");
        var i = null;
        var j = null;
        j = "b";
        i = c.toString();
        a.ad.h(j, i);
        j = "footer";
        i = d.toString();
        a.ad.h(j, i);
        j = "header";
        i = e.toString();
        a.ad.h(j, i);
        j = "l";
        i = f.toString();
        a.ad.h(j, i);
        j = "r";
        i = g.toString();
        a.ad.h(j, i);
        j = "t";
        i = h.toString();
        a.ad.h(j, i);
        ;
    };
    CT_PageMargins_Chart.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_PageMargins_Chart.$t = markType(CT_PageMargins_Chart, 'CT_PageMargins_Chart');
    return CT_PageMargins_Chart;
}(Base));
export { CT_PageMargins_Chart };
