// igniteui-react-excel/esm5/lib/CT_PrintSettings.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_HeaderFooter_Chart } from "./CT_HeaderFooter_Chart";
import { CT_PageMargins_Chart } from "./CT_PageMargins_Chart";
import { CT_PageSetup_Chart } from "./CT_PageSetup_Chart";
import { CT_RelId } from "./CT_RelId";
/**
 * @hidden
 */
var CT_PrintSettings = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_PrintSettings, _super);
    function CT_PrintSettings() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.d = null;
        _this.f = null;
        _this.h = null;
        return _this;
    }
    Object.defineProperty(CT_PrintSettings.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PrintSettings.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PrintSettings.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PrintSettings.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_PrintSettings.prototype.setProperty = function (a, b) {
        switch (a) {
            case 774:
                this.a = typeCast(CT_HeaderFooter_Chart.$, b);
                break;
            case 1162:
                this.c = typeCast(CT_PageMargins_Chart.$, b);
                break;
            case 1164:
                this.e = typeCast(CT_PageSetup_Chart.$, b);
                break;
            case 889:
                this.g = typeCast(CT_RelId.$, b);
                break;
            default: break;
        }
    };
    CT_PrintSettings.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_PrintSettings.$t = markType(CT_PrintSettings, 'CT_PrintSettings', Base.$, [ISupportSetProperty_$type]);
    return CT_PrintSettings;
}(Base));
export { CT_PrintSettings };
