// igniteui-react-excel/esm5/lib/CT_PivotEdit.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, NotSupportedException, markType } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_PivotEdit = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_PivotEdit, _super);
    function CT_PivotEdit() {
        var _this = _super.call(this) || this;
        _this.f = null;
        _this.h = null;
        _this.d = null;
        _this.b = null;
        return _this;
    }
    Object.defineProperty(CT_PivotEdit.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotEdit.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotEdit.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotEdit.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_PivotEdit.i = function (a) {
        throw new NotSupportedException(1, "The corresponding XML element for this class has no attributes, so there is nothing to read.");
    };
    CT_PivotEdit.j = function (a, b) {
        throw new NotSupportedException(1, "The corresponding XML element for this class has no attributes, so there is nothing to write.");
    };
    CT_PivotEdit.$t = markType(CT_PivotEdit, 'CT_PivotEdit');
    return CT_PivotEdit;
}(Base));
export { CT_PivotEdit };
