// igniteui-react-excel/esm5/lib/CT_PivotAreas.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_PivotArea } from "./CT_PivotArea";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_PivotAreas = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_PivotAreas, _super);
    function CT_PivotAreas() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.e = 0;
        return _this;
    }
    Object.defineProperty(CT_PivotAreas.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PivotAreas.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_PivotAreas.f = function (a, b) {
        var e_1, _a;
        b = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var c = _c.value;
                switch (c.a.e(a.d)) {
                    case "count":
                        b = parseUInt32_1(c.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b
        };
    };
    CT_PivotAreas.g = function (a, b, c) {
        a.ad.p(b, null);
        var d = null;
        var e = null;
        e = "count";
        d = c.toString();
        a.ad.h(e, d);
        ;
    };
    CT_PivotAreas.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1203:
                this.a.add(typeCast(CT_PivotArea.$, b));
                break;
            default: break;
        }
    };
    CT_PivotAreas.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_PivotAreas.$t = markType(CT_PivotAreas, 'CT_PivotAreas', Base.$, [ISupportSetProperty_$type]);
    return CT_PivotAreas;
}(Base));
export { CT_PivotAreas };
