// igniteui-react-excel/esm5/lib/CT_PageMargins.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseNumber } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_PageMargins = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_PageMargins, _super);
    function CT_PageMargins() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.h = 0;
        _this.j = 0;
        _this.l = 0;
        _this.b = 0;
        _this.f = 0;
        _this.d = 0;
        return _this;
    }
    Object.defineProperty(CT_PageMargins.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PageMargins.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PageMargins.prototype, "k", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PageMargins.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PageMargins.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_PageMargins.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_PageMargins.n = function (a, b, c, d, e, f, g) {
        var e_1, _a;
        b = 0;
        c = 0;
        d = 0;
        e = 0;
        f = 0;
        g = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var h = _c.value;
                switch (h.a.e(a.d)) {
                    case "l":
                        b = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "r":
                        c = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "t":
                        d = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "b":
                        e = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "header":
                        f = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    case "footer":
                        g = parseNumber(h.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e,
            p5: f,
            p6: g
        };
    };
    CT_PageMargins.o = function (a, b, c, d, e, f, g, h) {
        a.ad.p(b, "cx");
        var i = null;
        var j = null;
        j = "b";
        i = c.toString();
        a.ad.h(j, i);
        j = "footer";
        i = d.toString();
        a.ad.h(j, i);
        j = "header";
        i = e.toString();
        a.ad.h(j, i);
        j = "l";
        i = f.toString();
        a.ad.h(j, i);
        j = "r";
        i = g.toString();
        a.ad.h(j, i);
        j = "t";
        i = h.toString();
        a.ad.h(j, i);
        ;
    };
    CT_PageMargins.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_PageMargins.$t = markType(CT_PageMargins, 'CT_PageMargins');
    return CT_PageMargins;
}(Base));
export { CT_PageMargins };
