// igniteui-react-excel/esm5/lib/CT_Ratio.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseInt64_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_Ratio = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_Ratio, _super);
    function CT_Ratio() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.d = 0;
        _this.b = 0;
        return _this;
    }
    Object.defineProperty(CT_Ratio.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Ratio.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_Ratio.f = function (a, b, c) {
        var e_1, _a;
        b = 0;
        c = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "n":
                        b = parseInt64_1(d.d, CultureInfo.invariantCulture);
                        break;
                    case "d":
                        c = parseInt64_1(d.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_Ratio.g = function (a, b, c, d) {
        a.ad.p(b, "a");
        var e = null;
        var f = null;
        f = "d";
        e = c.toString();
        a.ad.h(f, e);
        f = "n";
        e = d.toString();
        a.ad.h(f, e);
        ;
    };
    CT_Ratio.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_Ratio.$t = markType(CT_Ratio, 'CT_Ratio');
    return CT_Ratio;
}(Base));
export { CT_Ratio };
