// igniteui-react-excel/esm5/lib/CT_StringDimension.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_Formula } from "./CT_Formula";
import { List$1 } from "igniteui-react-core";
import { CT_StringLevel } from "./CT_StringLevel";
import { ST_StringDimensionType_$type } from "./ST_StringDimensionType";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
/**
 * @hidden
 */
var CT_StringDimension = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_StringDimension, _super);
    function CT_StringDimension() {
        var _this = _super.call(this) || this;
        _this.d = 0;
        _this.a = null;
        _this.b = null;
        _this.e = null;
        _this.e = new List$1(CT_StringLevel.$, 0);
        return _this;
    }
    Object.defineProperty(CT_StringDimension.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_StringDimension.g = function (a, b) {
        var e_1, _a;
        b = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var c = _c.value;
                switch (c.a.e(a.d)) {
                    case "type":
                        b = XmlNamespaceDefinitionBase.b(ST_StringDimensionType_$type, c.d);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b
        };
    };
    CT_StringDimension.h = function (a, b, c) {
        a.ad.p(b, "cx");
        var d = null;
        var e = null;
        e = "type";
        d = XmlNamespaceDefinitionBase.p(ST_StringDimensionType_$type, c);
        a.ad.h(e, d);
        ;
    };
    CT_StringDimension.prototype.setProperty = function (a, b) {
        switch (a) {
            case 612:
                this.a = typeCast(CT_Formula.$, b);
                break;
            case 1907:
                this.b = typeCast(CT_Formula.$, b);
                break;
            case 944:
                this.e.add(typeCast(CT_StringLevel.$, b));
                break;
            default: break;
        }
    };
    CT_StringDimension.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_StringDimension.$t = markType(CT_StringDimension, 'CT_StringDimension', Base.$, [ISupportSetProperty_$type]);
    return CT_StringDimension;
}(Base));
export { CT_StringDimension };
