// igniteui-react-excel/esm5/lib/CT_SlicerCacheDefinition.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_SlicerCacheDefinition = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_SlicerCacheDefinition, _super);
    function CT_SlicerCacheDefinition() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.b = null;
        _this.f = null;
        _this.h = null;
        _this.j = null;
        return _this;
    }
    Object.defineProperty(CT_SlicerCacheDefinition.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_SlicerCacheDefinition.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_SlicerCacheDefinition.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_SlicerCacheDefinition.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_SlicerCacheDefinition.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_SlicerCacheDefinition.k = function (a, b, c) {
        var e_1, _a;
        b = null;
        c = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "name":
                        b = d.d;
                        break;
                    case "sourceName":
                        c = d.d;
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_SlicerCacheDefinition.l = function (a, b, c, d) {
        a.ad.p(b, null);
        var e = null;
        var f = null;
        if (c != null) {
            f = "name";
            e = c;
            a.ad.h(f, e);
        }
        if (d != null) {
            f = "sourceName";
            e = d;
            a.ad.h(f, e);
        }
        ;
    };
    CT_SlicerCacheDefinition.$t = markType(CT_SlicerCacheDefinition, 'CT_SlicerCacheDefinition');
    return CT_SlicerCacheDefinition;
}(Base));
export { CT_SlicerCacheDefinition };
