// igniteui-react-excel/esm5/lib/CT_StrVal.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, typeCast, String_$type, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_StrVal = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_StrVal, _super);
    function CT_StrVal() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.c = null;
        _this.e = 0;
        return _this;
    }
    Object.defineProperty(CT_StrVal.prototype, "b", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            this.c = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_StrVal.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_StrVal.f = function (a, b) {
        var e_1, _a;
        b = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var c = _c.value;
                switch (c.a.e(a.d)) {
                    case "idx":
                        b = parseUInt32_1(c.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b
        };
    };
    CT_StrVal.g = function (a, b, c) {
        a.ad.p(b, "c");
        var d = null;
        var e = null;
        e = "idx";
        d = c.toString();
        a.ad.h(e, d);
        ;
    };
    CT_StrVal.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1818:
                this.b = typeCast(String_$type, b);
                break;
            default: break;
        }
    };
    CT_StrVal.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_StrVal.$t = markType(CT_StrVal, 'CT_StrVal', Base.$, [ISupportSetProperty_$type]);
    return CT_StrVal;
}(Base));
export { CT_StrVal };
