// igniteui-react-excel/esm5/lib/CT_StringLevel.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_StringValue } from "./CT_StringValue";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_StringLevel = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_StringLevel, _super);
    function CT_StringLevel() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.g = 0;
        _this.d = null;
        return _this;
    }
    Object.defineProperty(CT_StringLevel.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_StringLevel.prototype, "f", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_StringLevel.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_StringLevel.h = function (a, b, c) {
        var e_1, _a;
        b = 0;
        c = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "ptCount":
                        b = parseUInt32_1(d.d, CultureInfo.invariantCulture);
                        break;
                    case "name":
                        c = d.d;
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_StringLevel.i = function (a, b, c, d) {
        a.ad.p(b, "cx");
        var e = null;
        var f = null;
        if (c != null) {
            f = "name";
            e = c;
            a.ad.h(f, e);
        }
        f = "ptCount";
        e = d.toString();
        a.ad.h(f, e);
        ;
    };
    CT_StringLevel.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1277:
                this.a.add(typeCast(CT_StringValue.$, b));
                break;
            default: break;
        }
    };
    CT_StringLevel.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_StringLevel.$t = markType(CT_StringLevel, 'CT_StringLevel', Base.$, [ISupportSetProperty_$type]);
    return CT_StringLevel;
}(Base));
export { CT_StringLevel };
