// igniteui-react-excel/esm5/lib/CT_Table.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { NotImplementedException } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_Table = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_Table, _super);
    function CT_Table() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.d = null;
        _this.f = null;
        return _this;
    }
    Object.defineProperty(CT_Table.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Table.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Table.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_Table.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_Table.prototype.setProperty = function (a, b) {
        throw new NotImplementedException(0);
    };
    CT_Table.$t = markType(CT_Table, 'CT_Table', Base.$, [ISupportSetProperty_$type]);
    return CT_Table;
}(Base));
export { CT_Table };
