// igniteui-react-excel/esm5/lib/CT_TextAutonumberBullet.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Number_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_TextAutonumberScheme_$type } from "./ST_TextAutonumberScheme";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { CultureInfo } from "igniteui-react-core";
import { parseInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_TextAutonumberBullet = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_TextAutonumberBullet, _super);
    function CT_TextAutonumberBullet() {
        var _this = _super.call(this) || this;
        _this.b = 0;
        _this.e = toNullable(Number_$type, null);
        return _this;
    }
    Object.defineProperty(CT_TextAutonumberBullet.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TextAutonumberBullet.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_TextAutonumberBullet.g = function (a, b, c) {
        var e_1, _a;
        b = 0;
        c = toNullable(Number_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "type":
                        b = XmlNamespaceDefinitionBase.b(ST_TextAutonumberScheme_$type, d.d);
                        break;
                    case "startAt":
                        c = toNullable(Number_$type, parseInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_TextAutonumberBullet.h = function (a, b, c, d) {
        a.ad.p(b, "a");
        var e = null;
        var f = null;
        if (c.hasValue) {
            f = "startAt";
            e = c.value.toString();
            a.ad.h(f, e);
        }
        f = "type";
        e = XmlNamespaceDefinitionBase.p(ST_TextAutonumberScheme_$type, d);
        a.ad.h(f, e);
        ;
    };
    CT_TextAutonumberBullet.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_TextAutonumberBullet.$t = markType(CT_TextAutonumberBullet, 'CT_TextAutonumberBullet');
    return CT_TextAutonumberBullet;
}(Base));
export { CT_TextAutonumberBullet };
