// igniteui-react-excel/esm5/lib/CT_TableStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_TableBackgroundStyle } from "./CT_TableBackgroundStyle";
import { CT_TablePartStyle } from "./CT_TablePartStyle";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
var CT_TableStyle = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_TableStyle, _super);
    function CT_TableStyle() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.ad = null;
        _this.f = null;
        _this.j = null;
        _this.h = null;
        _this.l = null;
        _this.r = null;
        _this.n = null;
        _this.t = null;
        _this.z = null;
        _this.ab = null;
        _this.p = null;
        _this.v = null;
        _this.x = null;
        _this.b = null;
        _this.af = null;
        _this.ah = null;
        return _this;
    }
    Object.defineProperty(CT_TableStyle.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "ac", {
        get: function () {
            return this.ad;
        },
        set: function (a) {
            this.ad = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "k", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "q", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "m", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "s", {
        get: function () {
            return this.t;
        },
        set: function (a) {
            this.t = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "y", {
        get: function () {
            return this.z;
        },
        set: function (a) {
            this.z = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "aa", {
        get: function () {
            return this.ab;
        },
        set: function (a) {
            this.ab = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "o", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "u", {
        get: function () {
            return this.v;
        },
        set: function (a) {
            this.v = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "w", {
        get: function () {
            return this.x;
        },
        set: function (a) {
            this.x = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "ae", {
        get: function () {
            return this.af;
        },
        set: function (a) {
            this.af = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyle.prototype, "ag", {
        get: function () {
            return this.ah;
        },
        set: function (a) {
            this.ah = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_TableStyle.aj = function (a, b, c) {
        var e_1, _a;
        b = null;
        c = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "styleId":
                        b = d.d;
                        break;
                    case "styleName":
                        c = d.d;
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_TableStyle.ak = function (a, b, c, d) {
        a.ad.p(b, "a");
        var e = null;
        var f = null;
        if (c != null) {
            f = "styleId";
            e = c;
            a.ad.h(f, e);
        }
        if (d != null) {
            f = "styleName";
            e = d;
            a.ad.h(f, e);
        }
        ;
    };
    CT_TableStyle.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1680:
                this.c = typeCast(CT_TableBackgroundStyle.$, b);
                break;
            case 1866:
                this.ac = typeCast(CT_TablePartStyle.$, b);
                break;
            case 92:
                this.e = typeCast(CT_TablePartStyle.$, b);
                break;
            case 94:
                this.i = typeCast(CT_TablePartStyle.$, b);
                break;
            case 93:
                this.g = typeCast(CT_TablePartStyle.$, b);
                break;
            case 95:
                this.k = typeCast(CT_TablePartStyle.$, b);
                break;
            case 871:
                this.q = typeCast(CT_TablePartStyle.$, b);
                break;
            case 643:
                this.m = typeCast(CT_TablePartStyle.$, b);
                break;
            case 873:
                this.s = typeCast(CT_TablePartStyle.$, b);
                break;
            case 1437:
                this.y = typeCast(CT_TablePartStyle.$, b);
                break;
            case 1647:
                this.aa = typeCast(CT_TablePartStyle.$, b);
                break;
            case 646:
                this.o = typeCast(CT_TablePartStyle.$, b);
                break;
            case 1052:
                this.u = typeCast(CT_TablePartStyle.$, b);
                break;
            case 1107:
                this.w = typeCast(CT_TablePartStyle.$, b);
                break;
            case 601:
                this.a = typeCast(CT_OfficeArtExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_TableStyle.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_TableStyle.$t = markType(CT_TableStyle, 'CT_TableStyle', Base.$, [ISupportSetProperty_$type]);
    return CT_TableStyle;
}(Base));
export { CT_TableStyle };
