// igniteui-react-excel/esm5/lib/CT_TableStyleCellStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_TableCellBorderStyle } from "./CT_TableCellBorderStyle";
import { CT_Cell3D } from "./CT_Cell3D";
import { EG_ThemeableFillStyle } from "./EG_ThemeableFillStyle";
/**
 * @hidden
 */
var CT_TableStyleCellStyle = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_TableStyleCellStyle, _super);
    function CT_TableStyleCellStyle() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.b = null;
        _this.e = null;
        return _this;
    }
    Object.defineProperty(CT_TableStyleCellStyle.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyleCellStyle.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableStyleCellStyle.prototype, "f", {
        get: function () {
            if (this.e == null) {
                this.e = new EG_ThemeableFillStyle();
            }
            return this.e;
        },
        enumerable: false,
        configurable: true
    });
    CT_TableStyleCellStyle.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1697:
                this.c = typeCast(CT_TableCellBorderStyle.$, b);
                break;
            case 235:
                this.a = typeCast(CT_Cell3D.$, b);
                break;
            default:
                if (this.f.c(a)) {
                    this.f.e(a, b);
                    return;
                }
                break;
        }
    };
    CT_TableStyleCellStyle.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_TableStyleCellStyle.$t = markType(CT_TableStyleCellStyle, 'CT_TableStyleCellStyle', Base.$, [ISupportSetProperty_$type]);
    return CT_TableStyleCellStyle;
}(Base));
export { CT_TableStyleCellStyle };
