// igniteui-react-excel/esm5/lib/CT_TableCell.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Number_$type, Boolean_$type, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_TextBody_DML } from "./CT_TextBody_DML";
import { CT_TableCellProperties } from "./CT_TableCellProperties";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
import { CultureInfo } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
import { parseInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_TableCell = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_TableCell, _super);
    function CT_TableCell() {
        var _this = _super.call(this) || this;
        _this.f = null;
        _this.d = null;
        _this.b = null;
        _this.p = toNullable(Number_$type, null);
        _this.l = toNullable(Number_$type, null);
        _this.n = toNullable(Boolean_$type, null);
        _this.r = toNullable(Boolean_$type, null);
        return _this;
    }
    Object.defineProperty(CT_TableCell.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCell.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCell.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCell.prototype, "o", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCell.prototype, "k", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCell.prototype, "m", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCell.prototype, "q", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_TableCell.t = function (a, b, c, d, e) {
        var e_1, _a;
        b = toNullable(Number_$type, null);
        c = toNullable(Number_$type, null);
        d = toNullable(Boolean_$type, null);
        e = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var f = _c.value;
                switch (f.a.e(a.d)) {
                    case "rowSpan":
                        b = toNullable(Number_$type, parseInt32_1(f.d, CultureInfo.invariantCulture));
                        break;
                    case "gridSpan":
                        c = toNullable(Number_$type, parseInt32_1(f.d, CultureInfo.invariantCulture));
                        break;
                    case "hMerge":
                        d = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    case "vMerge":
                        e = toNullable(Boolean_$type, ST_OnOff.e(f.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e
        };
    };
    CT_TableCell.u = function (a, b, c, d, e, f) {
        if (c === void 0) {
            c = toNullable(Number_$type, null);
        }
        if (d === void 0) {
            d = toNullable(Boolean_$type, null);
        }
        if (e === void 0) {
            e = toNullable(Number_$type, null);
        }
        if (f === void 0) {
            f = toNullable(Boolean_$type, null);
        }
        a.ad.p(b, "a");
        var g = null;
        var h = null;
        if (c.hasValue) {
            h = "gridSpan";
            g = c.value.toString();
            a.ad.h(h, g);
        }
        if (d.hasValue) {
            h = "hMerge";
            g = ST_OnOff.g(d.value);
            a.ad.h(h, g);
        }
        if (e.hasValue) {
            h = "rowSpan";
            g = e.value.toString();
            a.ad.h(h, g);
        }
        if (f.hasValue) {
            h = "vMerge";
            g = ST_OnOff.g(f.value);
            a.ad.h(h, g);
        }
        ;
    };
    CT_TableCell.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1777:
                this.e = typeCast(CT_TextBody_DML.$, b);
                break;
            case 1701:
                this.c = typeCast(CT_TableCellProperties.$, b);
                break;
            case 601:
                this.a = typeCast(CT_OfficeArtExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_TableCell.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_TableCell.$t = markType(CT_TableCell, 'CT_TableCell', Base.$, [ISupportSetProperty_$type]);
    return CT_TableCell;
}(Base));
export { CT_TableCell };
