// igniteui-react-excel/esm5/lib/CT_TablePartStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_TableStyleTextStyle } from "./CT_TableStyleTextStyle";
import { CT_TableStyleCellStyle } from "./CT_TableStyleCellStyle";
/**
 * @hidden
 */
var CT_TablePartStyle = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_TablePartStyle, _super);
    function CT_TablePartStyle() {
        var _this = _super.call(this) || this;
        _this.d = null;
        _this.b = null;
        return _this;
    }
    Object.defineProperty(CT_TablePartStyle.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TablePartStyle.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_TablePartStyle.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1703:
                this.c = typeCast(CT_TableStyleTextStyle.$, b);
                break;
            case 1702:
                this.a = typeCast(CT_TableStyleCellStyle.$, b);
                break;
            default: break;
        }
    };
    CT_TablePartStyle.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_TablePartStyle.$t = markType(CT_TablePartStyle, 'CT_TablePartStyle', Base.$, [ISupportSetProperty_$type]);
    return CT_TablePartStyle;
}(Base));
export { CT_TablePartStyle };
