// igniteui-react-excel/esm5/lib/CT_TableCol.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
import { CultureInfo } from "igniteui-react-core";
import { parseInt64_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_TableCol = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_TableCol, _super);
    function CT_TableCol() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.d = 0;
        return _this;
    }
    Object.defineProperty(CT_TableCol.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCol.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_TableCol.f = function (a, b) {
        var e_1, _a;
        b = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var c = _c.value;
                switch (c.a.e(a.d)) {
                    case "w":
                        b = parseInt64_1(c.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b
        };
    };
    CT_TableCol.g = function (a, b, c) {
        a.ad.p(b, "a");
        var d = null;
        var e = null;
        e = "w";
        d = c.toString();
        a.ad.h(e, d);
        ;
    };
    CT_TableCol.prototype.setProperty = function (a, b) {
        switch (a) {
            case 601:
                this.a = typeCast(CT_OfficeArtExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_TableCol.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_TableCol.$t = markType(CT_TableCol, 'CT_TableCol', Base.$, [ISupportSetProperty_$type]);
    return CT_TableCol;
}(Base));
export { CT_TableCol };
