// igniteui-react-excel/esm5/lib/CT_TabularSlicerCacheItem.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Boolean_$type, fromEnum, markType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_TabularSlicerCacheItem = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_TabularSlicerCacheItem, _super);
    function CT_TabularSlicerCacheItem() {
        var _this = _super.call(this) || this;
        _this.h = 0;
        _this.f = toNullable(Boolean_$type, null);
        _this.d = toNullable(Boolean_$type, null);
        return _this;
    }
    Object.defineProperty(CT_TabularSlicerCacheItem.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TabularSlicerCacheItem.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TabularSlicerCacheItem.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_TabularSlicerCacheItem.i = function (a, b, c, d) {
        var e_1, _a;
        b = 0;
        c = toNullable(Boolean_$type, null);
        d = toNullable(Boolean_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "x":
                        b = parseUInt32_1(e.d, CultureInfo.invariantCulture);
                        break;
                    case "s":
                        c = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    case "nd":
                        d = toNullable(Boolean_$type, ST_OnOff.e(e.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_TabularSlicerCacheItem.j = function (a, b, c, d, e) {
        a.ad.p(b, null);
        var f = null;
        var g = null;
        if (c.hasValue) {
            g = "nd";
            f = ST_OnOff.g(c.value);
            a.ad.h(g, f);
        }
        if (d.hasValue) {
            g = "s";
            f = ST_OnOff.g(d.value);
            a.ad.h(g, f);
        }
        g = "x";
        f = e.toString();
        a.ad.h(g, f);
        ;
    };
    CT_TabularSlicerCacheItem.$t = markType(CT_TabularSlicerCacheItem, 'CT_TabularSlicerCacheItem');
    return CT_TabularSlicerCacheItem;
}(Base));
export { CT_TabularSlicerCacheItem };
