// igniteui-react-excel/esm5/lib/CT_TextTabStop.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Number_$type, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_TextTabAlignType_$type } from "./ST_TextTabAlignType";
import { CultureInfo } from "igniteui-react-core";
import { XmlNamespaceDefinitionBase } from "./XmlNamespaceDefinitionBase";
import { parseInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_TextTabStop = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_TextTabStop, _super);
    function CT_TextTabStop() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.d = toNullable(Number_$type, null);
        _this.b = toNullable(ST_TextTabAlignType_$type, null);
        return _this;
    }
    Object.defineProperty(CT_TextTabStop.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TextTabStop.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_TextTabStop.f = function (a, b, c) {
        var e_1, _a;
        b = toNullable(Number_$type, null);
        c = toNullable(ST_TextTabAlignType_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "pos":
                        b = toNullable(Number_$type, parseInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    case "algn":
                        c = toNullable(ST_TextTabAlignType_$type, XmlNamespaceDefinitionBase.b(ST_TextTabAlignType_$type, d.d));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_TextTabStop.g = function (a, b, c, d) {
        if (c === void 0) {
            c = toNullable(ST_TextTabAlignType_$type, null);
        }
        if (d === void 0) {
            d = toNullable(Number_$type, null);
        }
        a.ad.p(b, "a");
        var e = null;
        var f = null;
        if (c.hasValue) {
            f = "algn";
            e = XmlNamespaceDefinitionBase.p(ST_TextTabAlignType_$type, c.value);
            a.ad.h(f, e);
        }
        if (d.hasValue) {
            f = "pos";
            e = d.value.toString();
            a.ad.h(f, e);
        }
        ;
    };
    CT_TextTabStop.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_TextTabStop.$t = markType(CT_TextTabStop, 'CT_TextTabStop');
    return CT_TextTabStop;
}(Base));
export { CT_TextTabStop };
