// igniteui-react-excel/esm5/lib/CT_TableCellBorderStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_ThemeableLineStyle } from "./CT_ThemeableLineStyle";
import { CT_OfficeArtExtensionList } from "./CT_OfficeArtExtensionList";
/**
 * @hidden
 */
var CT_TableCellBorderStyle = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_TableCellBorderStyle, _super);
    function CT_TableCellBorderStyle() {
        var _this = _super.call(this) || this;
        _this.j = null;
        _this.l = null;
        _this.p = null;
        _this.d = null;
        _this.f = null;
        _this.h = null;
        _this.n = null;
        _this.r = null;
        _this.b = null;
        return _this;
    }
    Object.defineProperty(CT_TableCellBorderStyle.prototype, "i", {
        get: function () {
            return this.j;
        },
        set: function (a) {
            this.j = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCellBorderStyle.prototype, "k", {
        get: function () {
            return this.l;
        },
        set: function (a) {
            this.l = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCellBorderStyle.prototype, "o", {
        get: function () {
            return this.p;
        },
        set: function (a) {
            this.p = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCellBorderStyle.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCellBorderStyle.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCellBorderStyle.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCellBorderStyle.prototype, "m", {
        get: function () {
            return this.n;
        },
        set: function (a) {
            this.n = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCellBorderStyle.prototype, "q", {
        get: function () {
            return this.r;
        },
        set: function (a) {
            this.r = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TableCellBorderStyle.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_TableCellBorderStyle.prototype.setProperty = function (a, b) {
        switch (a) {
            case 887:
                this.i = typeCast(CT_ThemeableLineStyle.$, b);
                break;
            case 1363:
                this.k = typeCast(CT_ThemeableLineStyle.$, b);
                break;
            case 1754:
                this.o = typeCast(CT_ThemeableLineStyle.$, b);
                break;
            case 160:
                this.c = typeCast(CT_ThemeableLineStyle.$, b);
                break;
            case 833:
                this.e = typeCast(CT_ThemeableLineStyle.$, b);
                break;
            case 834:
                this.g = typeCast(CT_ThemeableLineStyle.$, b);
                break;
            case 1749:
                this.m = typeCast(CT_ThemeableLineStyle.$, b);
                break;
            case 1761:
                this.q = typeCast(CT_ThemeableLineStyle.$, b);
                break;
            case 601:
                this.a = typeCast(CT_OfficeArtExtensionList.$, b);
                break;
            default: break;
        }
    };
    CT_TableCellBorderStyle.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_TableCellBorderStyle.$t = markType(CT_TableCellBorderStyle, 'CT_TableCellBorderStyle', Base.$, [ISupportSetProperty_$type]);
    return CT_TableCellBorderStyle;
}(Base));
export { CT_TableCellBorderStyle };
