// igniteui-react-excel/esm5/lib/CT_TupleSet.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, toNullable, Number_$type, fromEnum, markType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_TupleSet = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_TupleSet, _super);
    function CT_TupleSet() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.d = null;
        _this.h = toNullable(Number_$type, null);
        _this.f = toNullable(Number_$type, null);
        return _this;
    }
    Object.defineProperty(CT_TupleSet.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TupleSet.prototype, "c", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            this.d = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TupleSet.prototype, "g", {
        get: function () {
            return this.h;
        },
        set: function (a) {
            this.h = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_TupleSet.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_TupleSet.k = function (a, b, c) {
        var e_1, _a;
        b = toNullable(Number_$type, null);
        c = toNullable(Number_$type, null);
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var d = _c.value;
                switch (d.a.e(a.d)) {
                    case "rowCount":
                        b = toNullable(Number_$type, parseUInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    case "columnCount":
                        c = toNullable(Number_$type, parseUInt32_1(d.d, CultureInfo.invariantCulture));
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c
        };
    };
    CT_TupleSet.l = function (a, b, c, d) {
        if (c === void 0) {
            c = toNullable(Number_$type, null);
        }
        if (d === void 0) {
            d = toNullable(Number_$type, null);
        }
        a.ad.p(b, null);
        var e = null;
        var f = null;
        if (c.hasValue) {
            f = "columnCount";
            e = c.value.toString();
            a.ad.h(f, e);
        }
        if (d.hasValue) {
            f = "rowCount";
            e = d.value.toString();
            a.ad.h(f, e);
        }
        ;
    };
    CT_TupleSet.$t = markType(CT_TupleSet, 'CT_TupleSet');
    return CT_TupleSet;
}(Base));
export { CT_TupleSet };
