// igniteui-react-excel/esm5/lib/CT_ValueColorEndPosition.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_ExtremeValueColorPosition } from "./CT_ExtremeValueColorPosition";
import { CT_NumberColorPosition } from "./CT_NumberColorPosition";
import { CT_PercentageColorPosition } from "./CT_PercentageColorPosition";
/**
 * @hidden
 */
var CT_ValueColorEndPosition = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_ValueColorEndPosition, _super);
    function CT_ValueColorEndPosition() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.a = null;
        _this.b = null;
        _this.c = null;
        return _this;
    }
    CT_ValueColorEndPosition.prototype.setProperty = function (a, b) {
        switch (a) {
            case 1914:
                this.a = typeCast(CT_ExtremeValueColorPosition.$, b);
                break;
            case 1912:
                this.b = typeCast(CT_NumberColorPosition.$, b);
                break;
            case 1913:
                this.c = typeCast(CT_PercentageColorPosition.$, b);
                break;
            default: break;
        }
    };
    CT_ValueColorEndPosition.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_ValueColorEndPosition.$t = markType(CT_ValueColorEndPosition, 'CT_ValueColorEndPosition', Base.$, [ISupportSetProperty_$type]);
    return CT_ValueColorEndPosition;
}(Base));
export { CT_ValueColorEndPosition };
