// igniteui-react-excel/esm5/lib/CT_ValueAxisScaling.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { ST_DoubleOrAutomatic } from "./ST_DoubleOrAutomatic";
import { ST_ValueAxisUnit } from "./ST_ValueAxisUnit";
/**
 * @hidden
 */
var CT_ValueAxisScaling = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_ValueAxisScaling, _super);
    function CT_ValueAxisScaling() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.c = null;
        _this.e = null;
        _this.g = null;
        _this.i = null;
        return _this;
    }
    Object.defineProperty(CT_ValueAxisScaling.prototype, "b", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            this.c = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ValueAxisScaling.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ValueAxisScaling.prototype, "f", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_ValueAxisScaling.prototype, "h", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_ValueAxisScaling.a = function (a) {
        var b = null;
        var c = null;
        var d = null;
        var e = null;
        var f = CT_ValueAxisScaling.k(a, b, c, d, e);
        b = f.p1;
        c = f.p2;
        d = f.p3;
        e = f.p4;
        return ((function () {
            var $ret = new CT_ValueAxisScaling();
            $ret.b = b;
            $ret.d = c;
            $ret.f = d;
            $ret.h = e;
            return $ret;
        })());
    };
    CT_ValueAxisScaling.k = function (a, b, c, d, e) {
        var e_1, _a;
        b = null;
        c = null;
        d = null;
        e = null;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var f = _c.value;
                switch (f.a.e(a.d)) {
                    case "max":
                        b = ST_DoubleOrAutomatic.f(f.d);
                        break;
                    case "min":
                        c = ST_DoubleOrAutomatic.f(f.d);
                        break;
                    case "majorUnit":
                        d = ST_ValueAxisUnit.f(f.d);
                        break;
                    case "minorUnit":
                        e = ST_ValueAxisUnit.f(f.d);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d,
            p4: e
        };
    };
    CT_ValueAxisScaling.l = function (a, b, c, d, e, f) {
        if (c === void 0) {
            c = null;
        }
        if (d === void 0) {
            d = null;
        }
        if (e === void 0) {
            e = null;
        }
        if (f === void 0) {
            f = null;
        }
        a.ad.p(b, "cx");
        var g = null;
        var h = null;
        if (c != null) {
            h = "majorUnit";
            g = c.d();
            a.ad.h(h, g);
        }
        if (d != null) {
            h = "max";
            g = d.d();
            a.ad.h(h, g);
        }
        if (e != null) {
            h = "min";
            g = e.d();
            a.ad.h(h, g);
        }
        if (f != null) {
            h = "minorUnit";
            g = f.d();
            a.ad.h(h, g);
        }
        ;
    };
    CT_ValueAxisScaling.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_ValueAxisScaling.$t = markType(CT_ValueAxisScaling, 'CT_ValueAxisScaling');
    return CT_ValueAxisScaling;
}(Base));
export { CT_ValueAxisScaling };
