// igniteui-react-excel/esm5/lib/CT_Vector3D.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType, getInstanceType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { parseInt64_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_Vector3D = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_Vector3D, _super);
    function CT_Vector3D() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.c = 0;
        _this.e = 0;
        _this.g = 0;
        return _this;
    }
    Object.defineProperty(CT_Vector3D.prototype, "b", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            this.c = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Vector3D.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_Vector3D.prototype, "f", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_Vector3D.a = function (a) {
        var b, c, d;
        var e = CT_Vector3D.i(a, b, c, d);
        b = e.p1;
        c = e.p2;
        d = e.p3;
        return ((function () {
            var $ret = new CT_Vector3D();
            $ret.b = b;
            $ret.d = c;
            $ret.f = d;
            return $ret;
        })());
    };
    CT_Vector3D.i = function (a, b, c, d) {
        var e_1, _a;
        b = 0;
        c = 0;
        d = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "dx":
                        b = parseInt64_1(e.d, CultureInfo.invariantCulture);
                        break;
                    case "dy":
                        c = parseInt64_1(e.d, CultureInfo.invariantCulture);
                        break;
                    case "dz":
                        d = parseInt64_1(e.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_Vector3D.j = function (a, b, c, d, e) {
        a.ad.p(b, "a");
        var f = null;
        var g = null;
        g = "dx";
        f = c.toString();
        a.ad.h(g, f);
        g = "dy";
        f = d.toString();
        a.ad.h(g, f);
        g = "dz";
        f = e.toString();
        a.ad.h(g, f);
        ;
    };
    CT_Vector3D.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_Vector3D.$t = markType(CT_Vector3D, 'CT_Vector3D');
    return CT_Vector3D;
}(Base));
export { CT_Vector3D };
