// igniteui-react-excel/esm5/lib/CT_WholeE2oFormatting.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType, getInstanceType } from "igniteui-react-core";
import { ISupportSetProperty_$type } from "./ISupportSetProperty";
import { CT_LineProperties_DML } from "./CT_LineProperties_DML";
import { EG_EffectProperties } from "./EG_EffectProperties";
/**
 * @hidden
 */
var CT_WholeE2oFormatting = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_WholeE2oFormatting, _super);
    function CT_WholeE2oFormatting() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.c = null;
        return _this;
    }
    Object.defineProperty(CT_WholeE2oFormatting.prototype, "a", {
        get: function () {
            return this.b;
        },
        set: function (a) {
            this.b = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_WholeE2oFormatting.prototype, "d", {
        get: function () {
            if (this.c == null) {
                this.c = new EG_EffectProperties();
            }
            return this.c;
        },
        enumerable: false,
        configurable: true
    });
    CT_WholeE2oFormatting.prototype.setProperty = function (a, b) {
        switch (a) {
            case 918:
                this.a = typeCast(CT_LineProperties_DML.$, b);
                break;
            default:
                if (this.d.c(a)) {
                    this.d.e(a, b);
                    return;
                }
                break;
        }
    };
    CT_WholeE2oFormatting.prototype.toString = function () {
        var a = getInstanceType(this);
        return a != null ? a.typeName : _super.prototype.toString.call(this);
    };
    CT_WholeE2oFormatting.$t = markType(CT_WholeE2oFormatting, 'CT_WholeE2oFormatting', Base.$, [ISupportSetProperty_$type]);
    return CT_WholeE2oFormatting;
}(Base));
export { CT_WholeE2oFormatting };
