// igniteui-react-excel/esm5/lib/CT_WorkbookPr_SML2009.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, fromEnum, markType } from "igniteui-react-core";
import { CultureInfo } from "igniteui-react-core";
import { ST_OnOff } from "./ST_OnOff";
import { parseUInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var CT_WorkbookPr_SML2009 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_WorkbookPr_SML2009, _super);
    function CT_WorkbookPr_SML2009() {
        var _this = _super.call(this) || this;
        _this.i = 0;
        _this.c = false;
        _this.e = 0;
        return _this;
    }
    Object.defineProperty(CT_WorkbookPr_SML2009.prototype, "h", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_WorkbookPr_SML2009.prototype, "b", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            this.c = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(CT_WorkbookPr_SML2009.prototype, "d", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            this.e = a;
        },
        enumerable: false,
        configurable: true
    });
    CT_WorkbookPr_SML2009.j = function (a, b, c, d) {
        var e_1, _a;
        b = 0;
        c = false;
        d = 0;
        try {
            for (var _b = __values(fromEnum(a.ab.u)), _c = _b.next(); !_c.done; _c = _b.next()) {
                var e = _c.value;
                switch (e.a.e(a.d)) {
                    case "defaultImageDpi":
                        b = parseUInt32_1(e.d, CultureInfo.invariantCulture);
                        break;
                    case "discardImageEditData":
                        c = ST_OnOff.e(e.d);
                        break;
                    case "accuracyVersion":
                        d = parseUInt32_1(e.d, CultureInfo.invariantCulture);
                        break;
                    default: break;
                }
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return))
                    _a.call(_b);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        ;
        return {
            p1: b,
            p2: c,
            p3: d
        };
    };
    CT_WorkbookPr_SML2009.k = function (a, b, c, d, e) {
        a.ad.p(b, null);
        var f = null;
        var g = null;
        g = "accuracyVersion";
        f = c.toString();
        a.ad.h(g, f);
        g = "defaultImageDpi";
        f = d.toString();
        a.ad.h(g, f);
        g = "discardImageEditData";
        f = ST_OnOff.g(e);
        a.ad.h(g, f);
        ;
    };
    CT_WorkbookPr_SML2009.$t = markType(CT_WorkbookPr_SML2009, 'CT_WorkbookPr_SML2009');
    return CT_WorkbookPr_SML2009;
}(Base));
export { CT_WorkbookPr_SML2009 };
