// igniteui-react-excel/esm5/lib/DisplayUnit.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which determine the unit of expression for
 * [[Axis]] value labels.
 */
export var DisplayUnit = /*@__PURE__*/ (function (DisplayUnit) {
    /**
     * The raw value is displayed without special formatting.
     */
    DisplayUnit[DisplayUnit["None"] = -4142] = "None";
    /**
     * The [[Axis.displayUnitCustom]] property determines the display unit.
     */
    DisplayUnit[DisplayUnit["Custom"] = -4114] = "Custom";
    /**
     * 100,000,000
     */
    DisplayUnit[DisplayUnit["HundredMillions"] = -8] = "HundredMillions";
    /**
     * 100
     */
    DisplayUnit[DisplayUnit["Hundreds"] = -2] = "Hundreds";
    /**
     * 100,000
     */
    DisplayUnit[DisplayUnit["HundredThousands"] = -5] = "HundredThousands";
    /**
     * 1,000,000,000,000 (trillions)
     */
    DisplayUnit[DisplayUnit["MillionMillions"] = -10] = "MillionMillions";
    /**
     * 1,000,000
     */
    DisplayUnit[DisplayUnit["Millions"] = -6] = "Millions";
    /**
     * 10,000,000
     */
    DisplayUnit[DisplayUnit["TenMillions"] = -7] = "TenMillions";
    /**
     * 10,000
     */
    DisplayUnit[DisplayUnit["TenThousands"] = -4] = "TenThousands";
    /**
     * 1,000,000,000 (billions)
     */
    DisplayUnit[DisplayUnit["ThousandMillions"] = -9] = "ThousandMillions";
    /**
     * 1,000
     */
    DisplayUnit[DisplayUnit["Thousands"] = -3] = "Thousands";
    /**
     * Displays as a percentage. Applicable only to
     * Box and Whisker, Histogram, Pareto, Sunburst, Treemap, and Waterfall chart types;
     * if used for any other chart type, resolved to <i>Hundreds</i>.
     */
    DisplayUnit[DisplayUnit["Percentage"] = 1] = "Percentage";
    return DisplayUnit;
})({});
/**
 * @hidden
 */
export var DisplayUnit_$type = /*@__PURE__*/ markEnum('DisplayUnit', 'None,-4142|Custom,-4114|HundredMillions,-8|Hundreds,-2|HundredThousands,-5|MillionMillions,-10|Millions,-6|TenMillions,-7|TenThousands,-4|ThousandMillions,-9|Thousands,-3|Percentage,1');
