// igniteui-react-excel/esm5/lib/DocumentsCoreLocaleHu.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var DocumentsCoreLocaleHu = /** @class */ /*@__PURE__*/ (function () {
    function DocumentsCoreLocaleHu() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Érvénytelen {0} érték. Az értéknek {1} és {2} között kell lennie.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "A dokumentum titkosított, és csak jelszóval lehet megnyitni.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "A dokumentum nem támogatott titkosítási algoritmussal van titkosítva, és a titkosítás nem oldható fel.";
        this["LE_FormatException_TypeError"] = "Helytelen {0} formátum: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "A titkosított dokumentum megnyitásához használt jelszó helytelen.";
    }
    return DocumentsCoreLocaleHu;
}());
export { DocumentsCoreLocaleHu };
