// igniteui-react-excel/esm5/lib/DocumentsCoreLocaleRo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var DocumentsCoreLocaleRo = /** @class */ /*@__PURE__*/ (function () {
    function DocumentsCoreLocaleRo() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Valoare {0} nevalidă. Trebuie să fie între {1} și {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Documentul este criptat și trebuie deschis cu o parolă.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Documentul este criptat cu un algoritm de criptare neacceptat și nu poate fi decriptat.";
        this["LE_FormatException_TypeError"] = "Format incorect {0}: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "Parola utilizată pentru a deschide documentul criptat este incorectă.";
    }
    return DocumentsCoreLocaleRo;
}());
export { DocumentsCoreLocaleRo };
