// igniteui-react-excel/esm5/lib/DocumentsCoreLocaleTr.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var DocumentsCoreLocaleTr = /** @class */ /*@__PURE__*/ (function () {
    function DocumentsCoreLocaleTr() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Geçersiz {0} değeri. {1} ile {2} arasında olmalıdır.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Belge şifrelenmiştir ve bir parola ile açılmalıdır.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Belge, desteklenmeyen bir şifreleme algoritmasıyla şifrelenmiş ve şifresi çözülemez.";
        this["LE_FormatException_TypeError"] = "Yanlış {0} biçimi: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "Şifrelenmiş belgeyi açmak için kullanılan parola yanlış.";
    }
    return DocumentsCoreLocaleTr;
}());
export { DocumentsCoreLocaleTr };
