// igniteui-react-excel/esm5/lib/DocumentSummaryInformation_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { InformationBase } from "./InformationBase";
import { Dictionary$2 } from "igniteui-react-core";
import { UserDefinedPropertyType_$type } from "./UserDefinedPropertyType";
import { Base, String_$type, fromEnum, typeCast, markType, Number_$type, Boolean_$type, Array_$type, getInstanceType } from "igniteui-react-core";
import { DocCoreUtils } from "./DocCoreUtils";
import { List$1 } from "igniteui-react-core";
import { StringUtilities } from "./StringUtilities";
import { InformationBase_PropertySectionDefinition } from "./InformationBase_PropertySectionDefinition";
import { Guid } from "igniteui-react-core";
import { InformationBase_PropertyDefinition } from "./InformationBase_PropertyDefinition";
import { VariantUtilities } from "./VariantUtilities";
import { intSToU } from "igniteui-react-core";
import { arrayGetRank } from "igniteui-react-core";
/**
 * @hidden
 */
var DocumentSummaryInformation = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(DocumentSummaryInformation, _super);
    function DocumentSummaryInformation() {
        var _this = _super.call(this) || this;
        _this.s = null;
        _this.u = null;
        _this.x = null;
        _this.w = null;
        _this.ab = DocCoreUtils.ae;
        _this.x = new List$1(String_$type, 0);
        _this.w = new Dictionary$2(String_$type, Base.$, 2, StringUtilities.f);
        _this.u = new Dictionary$2(UserDefinedPropertyType_$type, Base.$, 0);
        return _this;
    }
    DocumentSummaryInformation.prototype.get_f = function () {
        return "d5cdd5022e9c101b939708002b2cf9ae";
    };
    Object.defineProperty(DocumentSummaryInformation.prototype, "f", {
        get: function () {
            return this.get_f();
        },
        enumerable: false,
        configurable: true
    });
    DocumentSummaryInformation.prototype.l = function (a) {
        _super.prototype.l.call(this, a);
        if (this.w.count > 0) {
            var b = new InformationBase_PropertySectionDefinition();
            b.b = new Guid(3, "d5cdd5052e9c101b939708002b2cf9ae");
            a.add(b);
        }
    };
    DocumentSummaryInformation.prototype.n = function (a, b) {
        var c = b.b.toString2("N");
        if (c == "d5cdd5052e9c101b939708002b2cf9ae") {
            this.ab = this.i(a, b);
            for (var d = 0; d < b.d; d++) {
                var e = b.a._inner[d];
                if (e.b != 1) {
                    this.ac(a, e);
                }
            }
        }
        else {
            _super.prototype.n.call(this, a, b);
        }
    };
    DocumentSummaryInformation.prototype.p = function () {
        var e_1, _a, e_2, _b;
        _super.prototype.p.call(this);
        var a = new List$1(String_$type, 0);
        try {
            for (var _c = __values(fromEnum(this.w.keys)), _d = _c.next(); !_d.done; _d = _c.next()) {
                var b = _d.value;
                a.add(b);
            }
        }
        catch (e_1_1) {
            e_1 = { error: e_1_1 };
        }
        finally {
            try {
                if (_d && !_d.done && (_a = _c.return))
                    _a.call(_c);
            }
            finally {
                if (e_1)
                    throw e_1.error;
            }
        }
        try {
            for (var _e = __values(fromEnum(this.w.values)), _f = _e.next(); !_f.done; _f = _e.next()) {
                var c = _f.value;
                var d = typeCast(String_$type, c);
                if (d != null) {
                    a.add(d);
                }
            }
        }
        catch (e_2_1) {
            e_2 = { error: e_2_1 };
        }
        finally {
            try {
                if (_f && !_f.done && (_b = _e.return))
                    _b.call(_e);
            }
            finally {
                if (e_2)
                    throw e_2.error;
            }
        }
        this.ab = InformationBase.h(a);
    };
    DocumentSummaryInformation.prototype.r = function (a, b) {
        var e_3, _a;
        var c = b.b.toString2("N");
        if (c == "d5cdd5052e9c101b939708002b2cf9ae") {
            b.d = this.w.count + 3;
            b.a = new List$1(InformationBase_PropertyDefinition.$, 0);
            a.c.position += b.d * 8;
            var d = new InformationBase_PropertyDefinition();
            d.a = a.c.position;
            d.b = 0;
            this.ad(a, d);
            b.a.add(d);
            var e = new InformationBase_PropertyDefinition();
            e.a = a.c.position;
            e.b = 1;
            VariantUtilities.f(a, this.ab, DocCoreUtils.r(this.ab));
            b.a.add(e);
            var f = new InformationBase_PropertyDefinition();
            f.a = a.c.position;
            f.b = -2147483648;
            this.ad(a, f);
            b.a.add(f);
            try {
                for (var _b = __values(fromEnum(this.v)), _c = _b.next(); !_c.done; _c = _b.next()) {
                    var g = _c.value;
                    var h = new InformationBase_PropertyDefinition();
                    h.a = a.c.position;
                    h.b = g.key;
                    VariantUtilities.f(a, this.ab, g.value);
                    b.a.add(h);
                }
            }
            catch (e_3_1) {
                e_3 = { error: e_3_1 };
            }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return))
                        _a.call(_b);
                }
                finally {
                    if (e_3)
                        throw e_3.error;
                }
            }
            for (var i = 0; i < this.x.count; i++) {
                var j = new InformationBase_PropertyDefinition();
                j.a = a.c.position;
                j.b = i + 2;
                this.ad(a, j);
                b.a.add(j);
            }
        }
        else {
            _super.prototype.r.call(this, a, b);
        }
    };
    DocumentSummaryInformation.prototype.ac = function (a, b) {
        a.baseStream.position = b.a;
        switch (b.b) {
            case 0:
                {
                    var c = a.readInt32();
                    this.x = new List$1(String_$type, 2, c);
                    this.w = new Dictionary$2(String_$type, Base.$, 4, c, StringUtilities.f);
                    for (var d = 0; d < this.x.h; d++) {
                        this.x.add(null);
                    }
                    for (var e = 0; e < c; e++) {
                        var f = a.readInt32();
                        this.x._inner[f - 2] = VariantUtilities.c(a, 30, this.ab);
                    }
                    break;
                }
            case 16777218:
            case 16777220:
                this.v.addItem(b.b, VariantUtilities.b(a, this.ab));
                break;
            case -2147483648:
                {
                    VariantUtilities.b(a, this.ab);
                    break;
                }
            default:
                {
                    if ((b.b & 16777216) == 16777216) {
                        this.v.addItem(b.b, VariantUtilities.b(a, this.ab));
                    }
                    else if (b.b - 2 >= 0 && b.b - 2 < this.x.count) {
                        this.w.addItem(this.x._inner[b.b - 2], VariantUtilities.b(a, this.ab));
                    }
                    break;
                }
        }
    };
    DocumentSummaryInformation.prototype.ad = function (a, b) {
        if (b.b == -2147483648) {
            VariantUtilities.g(a, this.ab, intSToU(DocCoreUtils.q), true);
        }
        else if (b.b == 0) {
            a.q(this.x.count);
            for (var c = 0; c < this.x.count; c++) {
                a.q(c + 2);
                VariantUtilities.h(a, 30, this.ab, this.x._inner[c], false);
            }
        }
        else {
            var d = this.w.item(this.x._inner[b.b - 2]);
            VariantUtilities.f(a, this.ab, d);
        }
    };
    Object.defineProperty(DocumentSummaryInformation.prototype, "t", {
        get: function () {
            if (this.s == null) {
                this.s = new DocumentSummaryInformation_DocumentSummaryInformationProperties(this);
            }
            return this.s;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DocumentSummaryInformation.prototype, "v", {
        get: function () {
            return this.u;
        },
        enumerable: false,
        configurable: true
    });
    DocumentSummaryInformation.$t = markType(DocumentSummaryInformation, 'DocumentSummaryInformation', InformationBase.$);
    return DocumentSummaryInformation;
}(InformationBase));
export { DocumentSummaryInformation };
/**
 * @hidden
 */
var DocumentSummaryInformation_DocumentSummaryInformationProperties = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(DocumentSummaryInformation_DocumentSummaryInformationProperties, _super);
    function DocumentSummaryInformation_DocumentSummaryInformationProperties(a) {
        var _this = _super.call(this) || this;
        _this.a = null;
        _this.a = a;
        return _this;
    }
    DocumentSummaryInformation_DocumentSummaryInformationProperties.b = function (a, b) {
        if (1 == a) {
            return false;
        }
        switch (a) {
            case 2:
            case 3:
            case 14:
            case 15: return b == String_$type;
            case 4:
            case 5:
            case 6:
            case 7:
            case 8:
            case 9:
            case 10: return b == Number_$type;
            case 11:
            case 16: return b == Boolean_$type;
            case 12: return b == Array_$type;
            case 13: return b == Array_$type;
            default: return true;
        }
    };
    DocumentSummaryInformation_DocumentSummaryInformationProperties.c = function (a, b) {
        if (1 == a) {
            return false;
        }
        switch (a) {
            case 16777218: return b == String_$type;
            case 16777220: return true;
            default: return true;
        }
    };
    DocumentSummaryInformation_DocumentSummaryInformationProperties.prototype.item = function (a, b) {
        if (arguments.length === 2) {
            var c = a;
            var d = typeCast(Array_$type, b);
            if (DocCoreUtils.e(b) || (d != null && d.length == 0)) {
                if (this.a.b.containsKey(c)) {
                    this.a.b.removeItem(c);
                }
            }
            else {
                if (DocumentSummaryInformation_DocumentSummaryInformationProperties.b(a, getInstanceType(b)) == false) {
                    return;
                }
                if (d != null && arrayGetRank(d) != 1) {
                    return;
                }
                if (this.a.b.containsKey(c)) {
                    this.a.b.item(c, b);
                }
                else {
                    this.a.b.addItem(c, b);
                }
            }
            return b;
        }
        else {
            var b_1 = a;
            if (this.a.b.containsKey(b_1) == false) {
                return null;
            }
            return this.a.b.item(b_1);
        }
    };
    DocumentSummaryInformation_DocumentSummaryInformationProperties.prototype.item1 = function (a, b) {
        if (arguments.length === 2) {
            var c = typeCast(Array_$type, b);
            if (DocCoreUtils.e(b) || (c != null && c.length == 0)) {
                if (this.a.v.containsKey(a)) {
                    this.a.v.removeItem(a);
                }
            }
            else {
                if (DocumentSummaryInformation_DocumentSummaryInformationProperties.c(a, getInstanceType(b)) == false) {
                    return;
                }
                if (c != null && arrayGetRank(c) != 1) {
                    return;
                }
                this.a.v.item(a, b);
            }
            return b;
        }
        else {
            if (this.a.v.containsKey(a) == false) {
                return null;
            }
            return this.a.v.item(a);
        }
    };
    DocumentSummaryInformation_DocumentSummaryInformationProperties.$t = markType(DocumentSummaryInformation_DocumentSummaryInformationProperties, 'DocumentSummaryInformation_DocumentSummaryInformationProperties');
    return DocumentSummaryInformation_DocumentSummaryInformationProperties;
}(Base));
export { DocumentSummaryInformation_DocumentSummaryInformationProperties };
