// igniteui-react-excel/esm5/lib/DocumentsCoreLocaleDe.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var DocumentsCoreLocaleDe = /** @class */ /*@__PURE__*/ (function () {
    function DocumentsCoreLocaleDe() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Ungültiger Wert {0}. Er muss zwischen {1} und {2} liegen.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Das Dokument ist verschlüsselt und muss mit einem Kennwort geöffnet werden.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Das Dokument ist mit einem nicht unterstützten Verschlüsselungsalgorithmus verschlüsselt und kann nicht entschlüsselt werden.";
        this["LE_FormatException_TypeError"] = "Falsches Format {0}: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "Das zum Öffnen des verschlüsselten Dokuments verwendete Kennwort ist falsch.";
    }
    return DocumentsCoreLocaleDe;
}());
export { DocumentsCoreLocaleDe };
