// igniteui-react-excel/esm5/lib/DocumentsCoreLocaleEs.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var DocumentsCoreLocaleEs = /** @class */ /*@__PURE__*/ (function () {
    function DocumentsCoreLocaleEs() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Valor {0} no válido. Debe estar entre {1} y {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "El documento está cifrado y debe abrirse con una contraseña.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "El documento está cifrado con un algoritmo de cifrado no compatible y no se puede descifrar.";
        this["LE_FormatException_TypeError"] = "Formato {0} incorrecto: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "La contraseña utilizada para abrir el documento cifrado es incorrecta.";
    }
    return DocumentsCoreLocaleEs;
}());
export { DocumentsCoreLocaleEs };
