// igniteui-react-excel/esm5/lib/DocumentsCoreLocaleNl.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var DocumentsCoreLocaleNl = /** @class */ /*@__PURE__*/ (function () {
    function DocumentsCoreLocaleNl() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Ongeldige waarde voor {0}. Het moet tussen {1} en {2} liggen.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Het document is gecodeerd en moet met een wachtwoord worden geopend.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Het document is gecodeerd met een niet-ondersteund coderingsalgoritme en kan niet worden gedecodeerd.";
        this["LE_FormatException_TypeError"] = "Onjuiste {0} indeling: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "Het wachtwoord dat wordt gebruikt om het gecodeerde document te openen, is onjuist.";
    }
    return DocumentsCoreLocaleNl;
}());
export { DocumentsCoreLocaleNl };
