// igniteui-react-excel/esm5/lib/DocumentsCoreLocaleZhHans.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var DocumentsCoreLocaleZhHans = /** @class */ /*@__PURE__*/ (function () {
    function DocumentsCoreLocaleZhHans() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "无效的 {0} 值。它必须在 {1} 和 {2} 之间。";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "文档已加密，必须使用密码打开。";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "文档使用不受支持的加密算法加密，无法解密。";
        this["LE_FormatException_TypeError"] = "{0} 格式不正确: {0}。";
        this["LE_InvalidPasswordException_DefaultMessage"] = "用于打开加密文档的密码错误。";
    }
    return DocumentsCoreLocaleZhHans;
}());
export { DocumentsCoreLocaleZhHans };
