// igniteui-react-excel/esm5/lib/DocumentsCoreLocaleDa.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var DocumentsCoreLocaleDa = /** @class */ /*@__PURE__*/ (function () {
    function DocumentsCoreLocaleDa() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Ugyldig {0} værdi.  Den skal være mellem {1} og {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Dokumentet er krypteret og skal åbnes med en adgangskode.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Dokumentet er krypteret med en ikke-understøttet krypteringsalgoritme og kan ikke dekrypteres.";
        this["LE_FormatException_TypeError"] = "Forkert {0} format:  {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "Adgangskoden, der bruges til at åbne det krypterede dokument, er forkert.";
    }
    return DocumentsCoreLocaleDa;
}());
export { DocumentsCoreLocaleDa };
