// igniteui-react-excel/esm5/lib/DocumentsCoreLocaleRu.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var DocumentsCoreLocaleRu = /** @class */ /*@__PURE__*/ (function () {
    function DocumentsCoreLocaleRu() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Invalid {0} value. It must be between {1} and {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "The document is encrypted and must be opened with a password.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "The document is encrypted with an unsupported encryption algorithm and cannot be decrypted.";
        this["LE_FormatException_TypeError"] = "Incorrect {0} format: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "The password used to open the encrypted document is incorrect.";
    }
    return DocumentsCoreLocaleRu;
}());
export { DocumentsCoreLocaleRu };
