// igniteui-react-excel/esm5/lib/DocumentsCoreLocaleBg.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var DocumentsCoreLocaleBg = /** @class */ /*@__PURE__*/ (function () {
    function DocumentsCoreLocaleBg() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Невалидна {0} стойност. Трябва да бъде между {1} и {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Документът е шифрован и трябва да бъде отворен с парола.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Документа е шифрован с неподдържан алгоритъм за шифроване и не може да бъде дешифриран.";
        this["LE_FormatException_TypeError"] = "Невалиден {0} формат: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "Паролата, използвана за отваряне на шифрования документ, е неправилна.";
    }
    return DocumentsCoreLocaleBg;
}());
export { DocumentsCoreLocaleBg };
