// igniteui-react-excel/esm5/lib/DocumentsCoreLocaleSv.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var DocumentsCoreLocaleSv = /** @class */ /*@__PURE__*/ (function () {
    function DocumentsCoreLocaleSv() {
        this["LE_ArgumentOutOfRangeException_ValueError"] = "Ogiltigt {0} värde. Det måste vara mellan {1} och {2}.";
        this["LE_DocumentEncryptedException_DefaultMessage"] = "Dokumentet är krypterat och måste öppnas med ett lösenord.";
        this["LE_EncryptionAlgorithmNotSupportedException_DefaultMessage"] = "Dokumentet krypteras med en krypteringsalgoritm som inte stöds och kan inte dekrypteras.";
        this["LE_FormatException_TypeError"] = "Felaktigt {0} format: {0}.";
        this["LE_InvalidPasswordException_DefaultMessage"] = "Lösenordet som används för att öppna det krypterade dokumentet är felaktigt.";
    }
    return DocumentsCoreLocaleSv;
}());
export { DocumentsCoreLocaleSv };
