// igniteui-react-excel/esm5/lib/EG_Anchor_ChartDrawing.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType } from "igniteui-react-core";
import { CT_RelSizeAnchor } from "./CT_RelSizeAnchor";
import { CT_AbsSizeAnchor } from "./CT_AbsSizeAnchor";
import { NotImplementedException } from "igniteui-react-core";
/**
 * @hidden
 */
var EG_Anchor_ChartDrawing = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(EG_Anchor_ChartDrawing, _super);
    function EG_Anchor_ChartDrawing() {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.a = null;
        return _this;
    }
    EG_Anchor_ChartDrawing.prototype.d = function (a, b, c, d) {
        throw new NotImplementedException(0);
    };
    EG_Anchor_ChartDrawing.prototype.e = function (a, b) {
        switch (a) {
            case 1344:
                this.b = typeCast(CT_RelSizeAnchor.$, b);
                break;
            case 1:
                this.a = typeCast(CT_AbsSizeAnchor.$, b);
                break;
            default: break;
        }
    };
    EG_Anchor_ChartDrawing.prototype.c = function (a) {
        switch (a) {
            case 1344:
            case 1: return true;
        }
        return false;
    };
    EG_Anchor_ChartDrawing.$t = markType(EG_Anchor_ChartDrawing, 'EG_Anchor_ChartDrawing');
    return EG_Anchor_ChartDrawing;
}(Base));
export { EG_Anchor_ChartDrawing };
