// igniteui-react-excel/esm5/lib/EG_Anchor_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, Type, markType } from "igniteui-react-core";
/**
 * @hidden
 */
var EG_Anchor = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(EG_Anchor, _super);
    function EG_Anchor() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.a = null;
        return _this;
    }
    EG_Anchor.$t = markType(EG_Anchor, 'EG_Anchor');
    return EG_Anchor;
}(Base));
export { EG_Anchor };
/**
 * @hidden
 */
var CT_AbsoluteAnchor = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_AbsoluteAnchor, _super);
    function CT_AbsoluteAnchor() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.c = null;
        _this.d = null;
        return _this;
    }
    CT_AbsoluteAnchor.prototype.b = function ($tArg, $tResult, a, b) {
        if (b === void 0) {
            b = Type.getDefaultValue($tArg);
        }
        return a._visitCT_AbsoluteAnchor$i(this, b);
    };
    CT_AbsoluteAnchor.$t = markType(CT_AbsoluteAnchor, 'CT_AbsoluteAnchor', EG_Anchor.$);
    return CT_AbsoluteAnchor;
}(EG_Anchor));
export { CT_AbsoluteAnchor };
/**
 * @hidden
 */
var CT_OneCellAnchor = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_OneCellAnchor, _super);
    function CT_OneCellAnchor() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.d = null;
        _this.c = null;
        return _this;
    }
    CT_OneCellAnchor.prototype.b = function ($tArg, $tResult, a, b) {
        if (b === void 0) {
            b = Type.getDefaultValue($tArg);
        }
        return a._visitCT_OneCellAnchor$i(this, b);
    };
    CT_OneCellAnchor.$t = markType(CT_OneCellAnchor, 'CT_OneCellAnchor', EG_Anchor.$);
    return CT_OneCellAnchor;
}(EG_Anchor));
export { CT_OneCellAnchor };
/**
 * @hidden
 */
var CT_TwoCellAnchor = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(CT_TwoCellAnchor, _super);
    function CT_TwoCellAnchor() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.e = 0;
        _this.c = null;
        _this.d = null;
        return _this;
    }
    CT_TwoCellAnchor.prototype.b = function ($tArg, $tResult, a, b) {
        if (b === void 0) {
            b = Type.getDefaultValue($tArg);
        }
        return a._visitCT_TwoCellAnchor$i(this, b);
    };
    CT_TwoCellAnchor.$t = markType(CT_TwoCellAnchor, 'CT_TwoCellAnchor', EG_Anchor.$);
    return CT_TwoCellAnchor;
}(EG_Anchor));
export { CT_TwoCellAnchor };
/**
 * @hidden
 */
export var IEG_AnchorVisitor$2_$type = /*@__PURE__*/ new Type(null, 'IEG_AnchorVisitor$2');
