// igniteui-react-excel/esm5/lib/EG_ObjectChoices.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType } from "igniteui-react-core";
import { CT_Shape } from "./CT_Shape";
import { CT_GroupShape } from "./CT_GroupShape";
import { CT_GraphicFrame } from "./CT_GraphicFrame";
import { CT_Connector } from "./CT_Connector";
import { CT_Picture } from "./CT_Picture";
import { NotImplementedException } from "igniteui-react-core";
/**
 * @hidden
 */
var EG_ObjectChoices = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(EG_ObjectChoices, _super);
    function EG_ObjectChoices() {
        var _this = _super.call(this) || this;
        _this.e = null;
        _this.c = null;
        _this.a = null;
        _this.b = null;
        _this.d = null;
        return _this;
    }
    EG_ObjectChoices.prototype.g = function (a, b, c, d) {
        throw new NotImplementedException(0);
    };
    EG_ObjectChoices.prototype.h = function (a, b) {
        switch (a) {
            case 1563:
                this.e = typeCast(CT_Shape.$, b);
                break;
            case 753:
                this.c = typeCast(CT_GroupShape.$, b);
                break;
            case 729:
                this.a = typeCast(CT_GraphicFrame.$, b);
                break;
            case 396:
                this.b = typeCast(CT_Connector.$, b);
                break;
            case 1194:
                this.d = typeCast(CT_Picture.$, b);
                break;
            default: break;
        }
    };
    EG_ObjectChoices.prototype.f = function (a) {
        switch (a) {
            case 1563:
            case 753:
            case 729:
            case 396:
            case 1194: return true;
        }
        return false;
    };
    EG_ObjectChoices.$t = markType(EG_ObjectChoices, 'EG_ObjectChoices');
    return EG_ObjectChoices;
}(Base));
export { EG_ObjectChoices };
