// igniteui-react-excel/esm5/lib/EG_ThemeableFontStyles.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType } from "igniteui-react-core";
import { CT_FontCollection } from "./CT_FontCollection";
import { CT_FontReference } from "./CT_FontReference";
import { NotImplementedException } from "igniteui-react-core";
/**
 * @hidden
 */
var EG_ThemeableFontStyles = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(EG_ThemeableFontStyles, _super);
    function EG_ThemeableFontStyles() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.a = null;
        _this.b = null;
        return _this;
    }
    EG_ThemeableFontStyles.prototype.d = function (a, b, c, d) {
        throw new NotImplementedException(0);
    };
    EG_ThemeableFontStyles.prototype.e = function (a, b) {
        switch (a) {
            case 659:
                this.a = typeCast(CT_FontCollection.$, b);
                break;
            case 660:
                this.b = typeCast(CT_FontReference.$, b);
                break;
            default: break;
        }
    };
    EG_ThemeableFontStyles.prototype.c = function (a) {
        switch (a) {
            case 659:
            case 660: return true;
        }
        return false;
    };
    EG_ThemeableFontStyles.$t = markType(EG_ThemeableFontStyles, 'EG_ThemeableFontStyles');
    return EG_ThemeableFontStyles;
}(Base));
export { EG_ThemeableFontStyles };
