// igniteui-react-excel/esm5/lib/EG_ThemeableFillStyle.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, typeCast, markType } from "igniteui-react-core";
import { CT_FillProperties } from "./CT_FillProperties";
import { CT_StyleMatrixReference_DML } from "./CT_StyleMatrixReference_DML";
import { DMLExtensions } from "./DMLExtensions";
import { DrawingMLMain } from "./DrawingMLMain";
/**
 * @hidden
 */
var EG_ThemeableFillStyle = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(EG_ThemeableFillStyle, _super);
    function EG_ThemeableFillStyle() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.a = null;
        _this.b = null;
        return _this;
    }
    EG_ThemeableFillStyle.prototype.d = function (a, b, c, d) {
        if (this.a != null) {
            DMLExtensions.ad(this.a, a, b, c, DrawingMLMain.dx);
        }
        if (this.b != null) {
            DMLExtensions.ba(this.b, a, b, c, DrawingMLMain.d0);
        }
    };
    EG_ThemeableFillStyle.prototype.e = function (a, b) {
        switch (a) {
            case 630:
                this.a = typeCast(CT_FillProperties.$, b);
                break;
            case 635:
                this.b = typeCast(CT_StyleMatrixReference_DML.$, b);
                break;
            default: break;
        }
    };
    EG_ThemeableFillStyle.prototype.c = function (a) {
        switch (a) {
            case 630:
            case 635: return true;
        }
        return false;
    };
    EG_ThemeableFillStyle.$t = markType(EG_ThemeableFillStyle, 'EG_ThemeableFillStyle');
    return EG_ThemeableFillStyle;
}(Base));
export { EG_ThemeableFillStyle };
