// igniteui-react-excel/esm5/lib/ElementPosition.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which determine whether a chart element is positioned
 * automatically or at a specific location in the chart space.
 */
export var ElementPosition = /*@__PURE__*/ (function (ElementPosition) {
    /**
     * The chart element is positioned automatically as appropriate
     * for that element. For example, a [[ChartTitle]] is typically
     * positioned at the top of the chart space, centered horizontally.
     */
    ElementPosition[ElementPosition["Automatic"] = -4105] = "Automatic";
    /**
     * Custom.
     */
    ElementPosition[ElementPosition["Custom"] = -4114] = "Custom";
    /**
     * The chart element is positioned above the plot area, centered horizontally.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["Top"] = 100] = "Top";
    /**
     * The chart element is positioned to the left of the plot area, centered vertically.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["Left"] = 101] = "Left";
    /**
     * The chart element is positioned to the right of the plot area, centered vertically.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["Right"] = 102] = "Right";
    /**
     * The chart element is positioned under the plot area, centered horizontally.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["Bottom"] = 103] = "Bottom";
    /**
     * The chart element is positioned above the plot area, horizontally aligned with the left edge.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["TopLeft"] = 104] = "TopLeft";
    /**
     * The chart element is positioned above the plot area, horizontally aligned with the right edge.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["TopRight"] = 105] = "TopRight";
    /**
     * The chart element is positioned below the plot area, horizontally aligned with the left edge.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["BottomLeft"] = 106] = "BottomLeft";
    /**
     * The chart element is positioned below the plot area, horizontally aligned with the right edge.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["BottomRight"] = 107] = "BottomRight";
    /**
     * The chart element is positioned to the left of the plot area, vertically aligned with the top edge.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["LeftTop"] = 108] = "LeftTop";
    /**
     * The chart element is positioned to the left of the plot area, vertically aligned with the bottom edge.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["LeftBottom"] = 109] = "LeftBottom";
    /**
     * The chart element is positioned to the right of the plot area, vertically aligned with the top edge.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["RightTop"] = 110] = "RightTop";
    /**
     * The chart element is positioned to the right of the plot area, vertically aligned with the bottom edge.
     * Applicable only for <i>BoxAndWhisker</i>, <i>Histogram</i>, <i>Pareto</i>, <i>Sunburst</i>, <i>Treemap</i>, and <i>Waterfall</i> chart types.
     */
    ElementPosition[ElementPosition["RightBottom"] = 111] = "RightBottom";
    return ElementPosition;
})({});
/**
 * @hidden
 */
export var ElementPosition_$type = /*@__PURE__*/ markEnum('ElementPosition', 'Automatic,-4105|Custom,-4114|Top,100|Left,101|Right,102|Bottom,103|TopLeft,104|TopRight,105|BottomLeft,106|BottomRight,107|LeftTop,108|LeftBottom,109|RightTop,110|RightBottom,111');
