// igniteui-react-excel/esm5/lib/EmfPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { ImagePartManagerBase } from "./ImagePartManagerBase";
import { ImageFormat } from "./ImageFormat";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var EmfPartManager = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(EmfPartManager, _super);
    function EmfPartManager() {
        return _super.call(this) || this;
    }
    EmfPartManager.prototype.get_s = function () {
        return "/xl/media/image.emf";
    };
    Object.defineProperty(EmfPartManager.prototype, "s", {
        get: function () {
            return this.get_s();
        },
        enumerable: false,
        configurable: true
    });
    EmfPartManager.prototype.get_t = function () {
        return "/ppt/media/image.emf";
    };
    Object.defineProperty(EmfPartManager.prototype, "t", {
        get: function () {
            return this.get_t();
        },
        enumerable: false,
        configurable: true
    });
    EmfPartManager.prototype.get_u = function () {
        return "/word/media/image.emf";
    };
    Object.defineProperty(EmfPartManager.prototype, "u", {
        get: function () {
            return this.get_u();
        },
        enumerable: false,
        configurable: true
    });
    EmfPartManager.prototype.get_m = function () {
        return "image/x-emf";
    };
    Object.defineProperty(EmfPartManager.prototype, "m", {
        get: function () {
            return this.get_m();
        },
        enumerable: false,
        configurable: true
    });
    EmfPartManager.prototype.r = function (a) {
        return ImageFormat.c;
    };
    EmfPartManager.$t = markType(EmfPartManager, 'EmfPartManager', ImagePartManagerBase.$);
    EmfPartManager.y = new EmfPartManager();
    return EmfPartManager;
}(ImagePartManagerBase));
export { EmfPartManager };
