// igniteui-react-excel/esm5/lib/ExcelLocaleDa.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var ExcelLocaleDa = /** @class */ /*@__PURE__*/ (function () {
    function ExcelLocaleDa() {
        this["DefaultTotalLabel"] = "Total";
        this["Error_AnchoredReference"] = "Forsøg på at få adgang til {0} som en ikke-forankret reference";
        this["Error_ExplicitThis"] = "{0} - Eksplicit Dette understøttes ikke";
        this["Error_ExternalRange"] = "Interval indeholder ekstern reference";
        this["Error_FormulaNotSupported"] = "{0} - Forsøg på at få adgang til formlen på en reference, der ikke understøtter formler";
        this["Error_Internal"] = "Der opstod en intern fejl i metode {0}.";
        this["Error_InvalidFunction"] = "Ugyldig funktion '{0}' ved {1}.";
        this["Error_InvalidFunctionArgCountTooFew"] = "Funktionen '{0}' refereret til {1} indeholder ikke det korrekte antal argumenter. Funktionen kræver mindst {2} parameter (er).";
        this["Error_InvalidFunctionArgCountTooMany"] = "Funktionen '{0}' refereret til {1} indeholder ikke det korrekte antal argumenter. Argumentantalet må ikke overstige {2} parameter (er).";
        this["Error_InvalidNumberSystemDigit"] = "Ugyldigt ciffer af '{0}' fundet i værdien '{1}' for nummersystemet med basen på {2}.";
        this["Error_InvalidOperand"] = "Ugyldig operand";
        this["Error_InvalidOperatorArgCount"] = "Funktionen '{0}' skal acceptere {1} argumenter.";
        this["Error_InvalidReference"] = "Ugyldig eller utilgængelig reference";
        this["Error_Location"] = "linje {0} kol {1}";
        this["Error_MethodNotSupported"] = "{0} understøtter ikke {1}.";
        this["Error_NoCurrentRangeElement"] = "Intet aktuelt intervalelement";
        this["Error_NullFunctionResult"] = "Evalueringen af '{0}' returnerede en nullværdi.";
        this["Error_NullOperand"] = "Null operand";
        this["Error_ParseRef"] = "Ukendt reference på {0}.";
        this["Error_RangeFromLast"] = "fraT.Last ikke indstillet";
        this["Error_RangeFromNull"] = "RefRange: fra er null";
        this["Error_RangeFromRefBase"] = "RefRange: fra er ikke en RefBase: {0}";
        this["Error_RangeLast"] = "RefRange Valider: sidste tupler [{0}] != [{1}]";
        this["Error_RangeNormalizeScope"] = "RefRange Normaliser: dårligt omfang [{0}/{1}]";
        this["Error_RangeToNull"] = "RefRange: til er null";
        this["Error_RangeToRefBase"] = "RefRange: til er ikke en RefBase: {0}";
        this["Error_RangeTuple"] = "RefRange Valider: tupler [{0}] != [{1}]";
        this["Error_RangeValidateCount"] = "RefRange Valider: fra antal {0} != til antal {1}";
        this["Error_ReadOnlyReference"] = "{0} - Forsøg på at indstille værdi på skrivebeskyttet reference";
        this["Error_ReferenceNotEumerable"] = "{0} er ikke en optællelig reference";
        this["Error_RefMalformed"] = "{0} - Forkert udformet reference {1}";
        this["Error_RP_ContainsBase"] = "RefParser.Contains- Tom grundlinje";
        this["Error_RP_ContainsTarget"] = "RefParser.Contains - Tomt mål";
        this["Error_RP_EmptyElement"] = "Ugyldig reference {0}. Tomt element fundet på placering {1}.";
        this["Error_RP_EmptyReference"] = "Ugyldig reference {0}. Reference er tom.";
        this["Error_RP_MalformedExtra"] = "Forkert udformet reference til RefParser: Flere [{0}] fundet i [{1}]";
        this["Error_RP_MalformedUnexpected"] = "Forkert udformet reference til RefParser: Uventet [{0}] fundet i [{1}]";
        this["Error_RP_MergeRelativeBase"] = "RefParser.Merge: Grundlinje er relativ";
        this["Error_RP_MissingCloseParenthesis"] = "Ugyldig reference {0}. Mangler lukkende parentes.";
        this["Error_RP_MissingEndQuotes"] = "Ugyldig reference {0}. Mangler slut anførselstegn.";
        this["Error_RT_Expected"] = "RefTuple: Forventes [{0}] i [{1}]";
        this["Error_RT_InvalidScope"] = "Ugyldigt område: {0}.";
        this["Error_RT_InvalidScope_CharactersAfterClosingQuotes"] = "Der er fundet tegn efter lukkende, dobbelte anførselstegn i {0} område på placering {1}.";
        this["Error_RT_InvalidTuple_CharactersAfterScopeEnd"] = "Ugyldigt element {0}. Der er fundet tegn efter afslutningen af området.";
        this["Error_RT_InvalidTuple_MissingCloseParenthesis"] = "Ugyldigt element {0}. Mangler lukkende parentes.";
        this["Error_RT_InvalidTuple_MissingEndQuotes"] = "Ugyldigt element {0}. Mangler slut anførselstegn.";
        this["Error_RT_InvalidTuple_NamePortionEmpty"] = "Ugyldigt element {0}. Elementnavndelen er tom.";
        this["Error_RT_InvalidTuple_ScopeIndexLarge"] = "Ugyldigt element {0}. Indeks {1} er ugyldigt: {2}.";
        this["Error_RT_InvalidTuple_UnescapedCharacter"] = "Ikke-undsluppet {0} tegn fundet i {1} på placering {2}.";
        this["Error_RT_NullRefName"] = "RefTuple: null RefName";
        this["Error_RT_Unexpected"] = "RefTuple: uventet [{0}] i [{1}]";
        this["Error_ScopeAllNotLast"] = "{0} - Omfang Alle skal vises i sidste element {1}";
        this["Error_UCErrorCode_Div"] = "Opdel med nul";
        this["Error_UCErrorCode_Fail"] = "Mislykket";
        this["Error_UCErrorCode_NA"] = "Ikke anvendelig";
        this["Error_UCErrorCode_Name"] = "Ugyldigt navn";
        this["Error_UCErrorCode_Null"] = "Null værdi";
        this["Error_UCErrorCode_Num"] = "Ugyldige numeriske værdier i funktionen eller formlen.";
        this["Error_UCErrorCode_Ok"] = "Vellykket";
        this["Error_UCErrorCode_Reference"] = "Ugyldig eller utilgængelig reference";
        this["Error_UCErrorCode_Unknown"] = "Ugyldig kode";
        this["Error_UCErrorCode_Value"] = "Forkert argumenttype eller operand.";
        this["Error_UnexpectedScope"] = "{0} - Uventet omfang for overordnet {1}";
        this["Error_UnknownRange"] = "Ukendt intervaltype";
        this["Error_UnknownRefType"] = "{0} - Ukendt RefType {1}";
        this["Error_UnknownScope"] = "{0} - Ukendt omfang {1}";
        this["Error_UnresolvedExternal"] = "Uafklaret ekstern reference";
        this["Func_abs_Arg_0"] = "tal";
        this["Func_abs_ArgDesc_0"] = "Det reelle tal, hvorfra den absolutte værdi skal hentes.";
        this["Func_abs_Category"] = "Matematik";
        this["Func_abs_CategoryURL"] = "Matematik";
        this["Func_abs_Desc"] = "Returnerer den absolutte værdi af et tal. Den absolutte værdi af et tal er tallet uden dets tegn.";
        this["Func_acos_Arg_0"] = "tal";
        this["Func_acos_ArgDesc_0"] = "Cosinus for vinklen, der skal findes, som skal være fra -1 til 1.";
        this["Func_acos_Category"] = "Matematik";
        this["Func_acos_CategoryURL"] = "Matematik";
        this["Func_acos_Desc"] = "Returnerer arccos eller invers cosinus for et tal. Arccosin er den vinkel, hvis cosinus er et angivet antal. Den returnerede vinkel er angivet i radianer i området nul til pi. For at konvertere resultatet fra radianer til grader skal du gange med 180 / PI ().";
        this["Func_acosh_Arg_0"] = "tal";
        this["Func_acosh_ArgDesc_0"] = "Et reelt tal større end 1, til hvilket den inverse hyperbolske cosinus skal findes.";
        this["Func_acosh_Category"] = "Matematik";
        this["Func_acosh_CategoryURL"] = "Matematik";
        this["Func_acosh_Desc"] = "Returnerer det inverse hyperbolske cosinus for det givne tal. Den inverse hyperbolske cosinus er den vinkel, hvis hyperbolske cosinus er et angivet tal.";
        this["Func_and_Arg_0"] = "logisk {0}";
        this["Func_and_ArgDesc_0"] = "En logisk værdi (sand eller falsk).";
        this["Func_and_Category"] = "Logisk";
        this["Func_and_CategoryURL"] = "Logisk";
        this["Func_and_Desc"] = "Returnerer sandt, hvis alle værdier er sande eller returnerer falsk, hvis en eller flere værdier er falske.";
        this["Func_asin_Arg_0"] = "tal";
        this["Func_asin_ArgDesc_0"] = "Sinus for vinklen, der skal findes, som skal være fra -1 til 1 radianer.";
        this["Func_asin_Category"] = "Matematik";
        this["Func_asin_CategoryURL"] = "Matematik";
        this["Func_asin_Desc"] = "Returnerer arcsin eller invers sinus for et tal. Arcsin er den vinkel, hvis sinus er det angivne tal. Den returnerede vinkel er angivet i radianer i området -pi / 2 til pi / 2. For at udtrykke arcsin i grader skal du gange resultatet med 180 / PI ().";
        this["Func_asinh_Arg_0"] = "tal";
        this["Func_asinh_ArgDesc_0"] = "Et reelt tal, hvortil den inverse hyperbolske sinus skal findes.";
        this["Func_asinh_Category"] = "Matematik";
        this["Func_asinh_CategoryURL"] = "Matematik";
        this["Func_asinh_Desc"] = "Returnerer den inverse hyperbolske sinus for det givne tal. Den inverse hyperbolske sinus er den vinkel, hvis hyperbolske sinus er et angivet tal.";
        this["Func_atan_Arg_0"] = "tal";
        this["Func_atan_ArgDesc_0"] = "Tangenten for vinklen, der skal findes";
        this["Func_atan_Category"] = "Matematik";
        this["Func_atan_CategoryURL"] = "Matematik";
        this["Func_atan_Desc"] = "Returnerer arctangenten eller den omvendte tangens for et tal. Arctangenten er den vinkel, hvis tangens er det angivne tal. Den returnerede vinkel er angivet i radianer i området -pi / 2 til pi / 2. For at udtrykke arctangenten i grader skal du gange resultatet med 180 / PI ().";
        this["Func_atan2_Arg_0"] = "x";
        this["Func_atan2_Arg_1"] = "y";
        this["Func_atan2_ArgDesc_0"] = "Punktets x-koordinat.";
        this["Func_atan2_ArgDesc_1"] = "Punktets y-koordinat.";
        this["Func_atan2_Category"] = "Matematik";
        this["Func_atan2_CategoryURL"] = "Matematik";
        this["Func_atan2_Desc"] = "Returnerer arctangenten eller den omvendte tangens for de angivne x- og y-koordinater. Arctangenten er vinklen fra x-aksen til en linje, der indeholder nulpunktet (0, 0) og et punkt med koordinater (x_num, y_num). Vinklen er angivet i radianer mellem -pi og pi, eksklusive -pi.";
        this["Func_atanh_Arg_0"] = "tal";
        this["Func_atanh_ArgDesc_0"] = "Et reelt tal mellem -1 og 1, hvortil den inverse hyperbolske tangens skal findes.";
        this["Func_atanh_Category"] = "Matematik";
        this["Func_atanh_CategoryURL"] = "Matematik";
        this["Func_atanh_Desc"] = "Returnerer den omvendte hyperbolske tangens for det givne tal. Den inverse hyperbolske tangens er den vinkel, hvis hyperbolske tangens er et angivet tal.";
        this["Func_avedev_Arg_0"] = "nummer {0}";
        this["Func_avedev_ArgDesc_0"] = "Tal1, tal2, ... er en eller flere numeriske værdier eller referencer til numeriske værdier.";
        this["Func_avedev_Category"] = "Statistisk";
        this["Func_avedev_CategoryURL"] = "Statistisk";
        this["Func_avedev_Desc"] = "Returnerer den gennemsnitlige afvigelse fra gennemsnittet af alle numeriske værdier.";
        this["Func_average_Arg_0"] = "nummer {0}";
        this["Func_average_ArgDesc_0"] = "Et tal eller en række af tal, for hvilke gennemsnittet kan findes med de andre argumenter.";
        this["Func_average_Category"] = "Statistisk";
        this["Func_average_CategoryURL"] = "Statistisk";
        this["Func_average_Desc"] = "Returnerer gennemsnittet (aritmetisk middelværdi) af en eller flere værdier.";
        this["Func_bin2dec_Arg_0"] = "Nummer";
        this["Func_bin2dec_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer et binært tal.";
        this["Func_bin2dec_Category"] = "Engineering";
        this["Func_bin2dec_CategoryURL"] = "Engineering";
        this["Func_bin2dec_Desc"] = "Returnerer decimalværdien af en tekstrepræsentation af et tal i det binære talsystem.";
        this["Func_bin2hex_Arg_0"] = "Nummer";
        this["Func_bin2hex_Arg_1"] = "Pladser";
        this["Func_bin2hex_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer et binært tal.";
        this["Func_bin2hex_ArgDesc_1"] = "Antallet af tegn, der skal bruges til at repræsentere værdien. Hvis dette ikke er angivet, anvendes det minimale antal tegn, der er nødvendigt for at repræsentere værdien. Dette giver dig mulighed for at angive foranstillede nuller.";
        this["Func_bin2hex_Category"] = "Engineering";
        this["Func_bin2hex_CategoryURL"] = "Engineering";
        this["Func_bin2hex_Desc"] = "Konverterer et tekstudtryk, der repræsenterer et binært tal, til et tekstudtryk, der repræsenterer det samme tal i det hexadecimale talsystem.";
        this["Func_bin2oct_Arg_0"] = "Nummer";
        this["Func_bin2oct_Arg_1"] = "Pladser";
        this["Func_bin2oct_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer et binært tal.";
        this["Func_bin2oct_ArgDesc_1"] = "Antallet af tegn, der skal bruges til at repræsentere værdien. Hvis dette ikke er angivet, anvendes det minimale antal tegn, der er nødvendigt for at repræsentere værdien. Dette giver dig mulighed for at angive foranstillede nuller.";
        this["Func_bin2oct_Category"] = "Engineering";
        this["Func_bin2oct_CategoryURL"] = "Engineering";
        this["Func_bin2oct_Desc"] = "Konverterer et tekstudtryk, der repræsenterer et binært tal, til et tekstudtryk, der repræsenterer det samme tal i det oktale talsystem.";
        this["Func_binomdist_Arg_0"] = "Nummer";
        this["Func_binomdist_Arg_1"] = "Forsøg";
        this["Func_binomdist_Arg_2"] = "Sandsynlighed";
        this["Func_binomdist_Arg_3"] = "Akkumulerede";
        this["Func_binomdist_ArgDesc_0"] = "Antallet af succeser i forsøg.";
        this["Func_binomdist_ArgDesc_1"] = "Antallet af uafhængige forsøg.";
        this["Func_binomdist_ArgDesc_2"] = "Sandsynligheden for succes ved hvert forsøg.";
        this["Func_binomdist_ArgDesc_3"] = "En logisk værdi, der bestemmer funktionens form. Hvis det er sandt, returnerer funktionen den kumulative fordeling, som er sandsynligheden for, at der højst er 'antal' succeser, ellers returnerer den sandsynlighedsmassen, hvilket er sandsynligheden for, at der er 'antal' succeser.";
        this["Func_binomdist_Category"] = "Statistisk";
        this["Func_binomdist_CategoryURL"] = "Statistisk";
        this["Func_binomdist_Desc"] = "Returnerer antallet af mulige kombinationer (med gentagelser) i et givet sæt emner og et antal valgte emner fra dette sæt.";
        this["Func_CategoryURL_Template"] = "ms-help://INFRAGISTICS_HELP/NetAdvantagev{2}{3}/Win/WinElements/WinCalcManager_{0}_Functions.html#FUNC_{1}";
        this["Func_ceiling_Arg_0"] = "tal";
        this["Func_ceiling_Arg_1"] = "betydning";
        this["Func_ceiling_ArgDesc_0"] = "Tallet, der skal afrundes.";
        this["Func_ceiling_ArgDesc_1"] = "Det multipel, der skal afrundes til. Hvis det udelades, antages det at være en.";
        this["Func_ceiling_Category"] = "Matematik";
        this["Func_ceiling_CategoryURL"] = "Matematik";
        this["Func_ceiling_Desc"] = "Returnerer det mindste multiplum af betydningen (er som standard 1) større end eller lig med det angivne tal.";
        this["Func_char_Arg_0"] = "tegnKode";
        this["Func_char_ArgDesc_0"] = "Nummeret, der skal konverteres til et tegn.";
        this["Func_char_Category"] = "TekstOgData";
        this["Func_char_CategoryURL"] = "Text_and_Data";
        this["Func_char_Desc"] = "Returnerer det tegn, der er angivet med kodenummeret fra computerens tegnsæt.";
        this["Func_choose_Arg_0"] = "IndeksTal";
        this["Func_choose_Arg_1"] = "Value{0}";
        this["Func_choose_ArgDesc_0"] = "Det 1-baserede indeks i resten af argumenterne for den værdi, der skal bruges.";
        this["Func_choose_ArgDesc_1"] = "En værdi, der kan vælges afhængigt af IndexTal-værdien.";
        this["Func_choose_Category"] = "OpslagOgReference";
        this["Func_choose_CategoryURL"] = "Lookup_And_Reference";
        this["Func_choose_Desc"] = "Returnerer en af de værdier, der er angivet i dens argumenter, baseret på nummeret i det første argument.";
        this["Func_code_Arg_0"] = "tekst";
        this["Func_code_ArgDesc_0"] = "Tekstudtrykket, der får sit første tegn evalueret.";
        this["Func_code_Category"] = "TekstOgData";
        this["Func_code_CategoryURL"] = "Text_and_Data";
        this["Func_code_Desc"] = "Returnerer en numerisk kode for det første tegn i et tekstudtryk ved hjælp af det tegnsæt, computeren anvender.";
        this["Func_combin_Arg_0"] = "Nummer";
        this["Func_combin_Arg_1"] = "AntalValgt";
        this["Func_combin_ArgDesc_0"] = "Antallet af emner.";
        this["Func_combin_ArgDesc_1"] = "Antallet af emner valgt i hver kombination.";
        this["Func_combin_Category"] = "Matematik";
        this["Func_combin_CategoryURL"] = "Matematik";
        this["Func_combin_Desc"] = "Returnerer antallet af mulige kombinationer fra et givet sæt af emner og et antal valgte emner fra dette sæt.";
        this["Func_combina_Arg_0"] = "Nummer";
        this["Func_combina_Arg_1"] = "AntalValgt";
        this["Func_combina_ArgDesc_0"] = "Antallet af emner.";
        this["Func_combina_ArgDesc_1"] = "Antallet af emner valgt i hver kombination.";
        this["Func_combina_Category"] = "Matematik";
        this["Func_combina_CategoryURL"] = "Matematik";
        this["Func_combina_Desc"] = "Returnerer antallet af mulige kombinationer (med gentagelser) i et givet sæt emner og et antal valgte emner fra dette sæt.";
        this["Func_complex_Arg_0"] = "ReeltTal";
        this["Func_complex_Arg_1"] = "ImaginærtTal";
        this["Func_complex_Arg_2"] = "Suffiks";
        this["Func_complex_ArgDesc_0"] = "Den reelle koefficient for det komplekse tal.";
        this["Func_complex_ArgDesc_1"] = "Den imaginære koefficient for nummeret.";
        this["Func_complex_ArgDesc_2"] = "Angiver det suffiks, der skal bruges. De acceptable værdier er 'i' eller 'j'. Hvis udeladt, bruges 'i'. Bemærk, at suffikset er versalfølsomt; store bogstaver 'I' og 'J' er ikke tilladt og vil resultere i en #VALUE-fejl.";
        this["Func_complex_Category"] = "Engineering";
        this["Func_complex_CategoryURL"] = "Engineering";
        this["Func_complex_Desc"] = "Returnerer et komplekst tal repræsenteret som et tekstudtryk i formatet 'x + yi' eller 'x + yj' ved at kombinere et reelt og et imaginært tal.";
        this["Func_concatenate_Arg_0"] = "tekst {0}";
        this["Func_concatenate_ArgDesc_0"] = "Et tekstudtryk eller en række af tekstudtryk, der skal sammenføjes til et enkelt tekstudtryk.";
        this["Func_concatenate_Category"] = "TekstOgData";
        this["Func_concatenate_CategoryURL"] = "Text_and_Data";
        this["Func_concatenate_Desc"] = "Sammenføjer alle argumenter i et tekstudtryk.";
        this["Func_convert_Arg_0"] = "Nummer";
        this["Func_convert_Arg_1"] = "FraEnhed";
        this["Func_convert_Arg_2"] = "TilEnhed";
        this["Func_convert_ArgDesc_0"] = "Værdien, der skal konverteres.";
        this["Func_convert_ArgDesc_1"] = "Enheden, hvormed tallet er angivet. Vægt og masseenheder: \" g \" = gram; \" sg \" =slug; \" lbm \" = pundmasse (avoirdupois); \" u \" = U (atommasseenhed); \" ozm \" = Ounce masse (avoirdupois). Afstandsenheder: \" m \" = Meter; \" mi \" = Statutemile; \" Nmi \" = Sømil; \" in \" = tomme; \" ft \" = fod; \" yd \" = Yard; \" ang \" = Angstrøm; \" Pica \" = Pica (1/72 in.)\r\nTidenheder: \" år \" = År; \" dag \" = dag; \" hr \" = Time; \" mn \" = minut; \" sek \" = sekund    Trykenheder: \" Pa \" (eller \" p \") = Pascal; \" atm \" (eller \" at \") = Atmosfære; \" mmHg \" = mm kviksølv\r\nKraftenheder: \" N \" = Newton; \" dyn \" (eller \" dy \") = Dyne; \" lbf \" = Pundkraft\r\nEnergienheder: \" J \" = Joule; \" e \" = Erg; \" c \" = termodynamisk kalorier; \" cal \" = IT-kalorie; \" eV \" (eller \" ev \") = Elektronvolt; \" HK \" (eller \" hk \") = hestekræfter; \" Wh \" (eller \" wh \") = Watt-time; \" flb \" = Fod-Pund; \" BTU \" (eller \" btu \") = BTU\r\nEffektenheder: \" HK \" (eller \" hk \") = hestekræfter; \" W \" (eller \" w \") = Watt\r\nMagnetismeenheder: \" T \" = Tesla; \" ga \" = Gauss\r\nTemperaturenheder: \" C \" (eller \" cel \") = Grad Celsius; \" F \" (eller \" fah \") = Grad Fahrenheit; \" K \" (eller \" kel \") = Kelvin\r\nFlydende måleenheder: \" tsk \" = teskefuld; \" tbs \" = Spiseske; \" oz \" = Væske ounce; \" kop \" = kop; \" pt \" (eller \" us_pt \") = US pint; \" uk_pt \" = britisk pint; \" qt \" = kvart; \" gal \" = gallon; \" l \" (eller \" lt \") = liter  Enhver metrisk enhed kan forsynes med et af følgende præfikser: exa (\" E \") = 1E + 18; peta (\" P \") = 1E + 15; tera (\" T \") = 1E + 12; giga (\" G \") = 1E + 09; mega (\" M \") = 1E + 06; kilo (\" k \") = 1E + 03; hekto (\" h \") = 1E + 02; deka (\" e \") = 1E + 01; deci (\" d \") = 1E-01; centi (\" c \") = 1E-02; milli (\" m \") = 1E-03; mikro (\" u \") = 1E-06; nano (\" n \") = 1E-09; pico (\" p \") = 1E-12; femto (\" f \") = 1E-15; atto (\" a \") = 1E-18\r\n    ";
        this["Func_convert_ArgDesc_2"] = "Enhederne, hvor den konverterede værdi skal returneres.\r\n\r\nVægt og masseenheder: \" g \" = gram; \" sg \" = slug; \" lbm \" = pundmasse (avoirdupois); \" u \" = U (atommasseenhed); \" ozm \" = Ounse masse (avoirdupois)\r\nAfstandsenheder: \" m \" = Meter; \" mi \" = Statutemil; \" Nmi \" = Sømil; \" in \" = tomme; \" ft \" = fod; \" yd \" = Yard; \" ang \" = Angstrøm; \" Pica \" = Pica (1/72 in.)\r\nTidenheder: \" år \" = År; \" dag \" = dag; \" hr \" = Time; \" mn \" = minut; \" sek \" = Andet\r\nTrykenheder: \" Pa \" (eller \" p \") = Pascal; \" atm \" (eller \" at \") = Atmosfære; \" mmHg \" = mm kviksølv\r\nTving enheder: \" N \" = Newton; \" dyn \" (eller \" dy \") = Dyne; \" lbf \" = Pundkraft\r\nEnergienheder: \" J \" = Joule; \" e \" = Erg; \" c \" = termodynamisk kalorie; \" cal \" = IT-kalorie; \" eV \" (eller \" ev \") = Elektronvolt; \" HPh \" (eller \" hh \") = hestekræfter-time; \" Wh \" (eller \" wh \") = Watt-time; \" flb \" = Fodpund; \" BTU \" (eller \" btu \") = BTU\r\nEffektenheder: \" HP \" (eller \" h \") = hestekræfter; \" W \" (eller \" w \") = Watt\r\nMagentismenheder: \" T \" = Tesla; \" ga \" = Gauss\r\nTemperaturenheder: \" C \" (eller \" cel \") = Grad Celsius; \" F \" (eller \" fah \") = Grad Fahrenheit; \" K \" (eller \" kel \") = Kelvin\r\nFlydende måleenheder: \" tsk \" = teskefuld; \" tbs \" = Spiseske; \" oz \" = Væske ounce; \" kop \" = kop; \" pt \" (eller \" us_pt \") = US pint; \" uk_pt \" = britisk pint; \" qt \" = kvart; \" gal \" = gallon; \" l \" (eller \" lt \") = liter\r\n\r\nEnhver metrisk enhed kan være forud for en af følgende: exa (\" E \") = 1E + 18; peta (\" P \") = 1E + 15; tera (\" T \") = 1E + 12; giga (\" G \") = 1E + 09; mega (\" M \") = 1E + 06; kilo (\" k \") = 1E + 03; hekto (\" h \") = 1E + 02; dekao (\" e \") = 1E + 01; deci (\" d \") = 1E-01; centi (\" c \") = 1E-02; milli (\" m \") = 1E-03; mikro (\" u \") = 1E-06; nano (\" n \") = 1E-09; pico (\" p \") = 1E-12; femto (\" f \") = 1E-15; atto (\" a \") = 1E-18\r\n    ";
        this["Func_convert_Category"] = "Engineering";
        this["Func_convert_CategoryURL"] = "Engineering";
        this["Func_convert_Desc"] = "Konverterer en værdi fra et målesystem til et andet, f.eks. meter til tommer eller timer til sekunder.";
        this["Func_cos_Arg_0"] = "tal";
        this["Func_cos_ArgDesc_0"] = "Vinklen i radianer, hvortil cosinus skal findes.";
        this["Func_cos_Category"] = "Matematik";
        this["Func_cos_CategoryURL"] = "Matematik";
        this["Func_cos_Desc"] = "Returnerer cosinus for den givne vinkel.";
        this["Func_cosh_Arg_0"] = "tal";
        this["Func_cosh_ArgDesc_0"] = "Et reelt tal, hvortil den hyperbolske cosinus skal findes.";
        this["Func_cosh_Category"] = "Matematik";
        this["Func_cosh_CategoryURL"] = "Matematik";
        this["Func_cosh_Desc"] = "Returnerer et tals hyperbolske cosinus.";
        this["Func_count_Arg_0"] = "værdi {0}";
        this["Func_count_ArgDesc_0"] = "En reference eller række af referencer, der kan indeholde tal.";
        this["Func_count_Category"] = "Statistisk";
        this["Func_count_CategoryURL"] = "Statistisk";
        this["Func_count_Desc"] = "Tæller antallet af referencer, der indeholder tal.";
        this["Func_counta_Arg_0"] = "værdi {0}";
        this["Func_counta_ArgDesc_0"] = "En reference eller række af referencer, der kan indeholde tal.";
        this["Func_counta_Category"] = "Statistisk";
        this["Func_counta_CategoryURL"] = "Statistisk";
        this["Func_counta_Desc"] = "Tæller antallet af referencer, der ikke er tomme.";
        this["Func_critbinom_Arg_0"] = "Forsøg";
        this["Func_critbinom_Arg_1"] = "Sandsynlighed";
        this["Func_critbinom_Arg_2"] = "Alpha";
        this["Func_critbinom_ArgDesc_0"] = "Antallet af uafhængige forsøg.";
        this["Func_critbinom_ArgDesc_1"] = "Sandsynligheden for succes ved hvert forsøg.";
        this["Func_critbinom_ArgDesc_2"] = "Kriterieværdien.";
        this["Func_critbinom_Category"] = "Statistisk";
        this["Func_critbinom_CategoryURL"] = "Statistisk";
        this["Func_critbinom_Desc"] = "Returnerer den mindste værdi, for hvilken den kumulative binomialfordeling er større end eller lig med en kriterieværdi.";
        this["Func_date_Arg_0"] = "år";
        this["Func_date_Arg_1"] = "måned";
        this["Func_date_Arg_2"] = "dag";
        this["Func_date_ArgDesc_0"] = "Et tal, der repræsenterer året.";
        this["Func_date_ArgDesc_1"] = "Et tal, der repræsenterer årets måned.";
        this["Func_date_ArgDesc_2"] = "Et tal, der repræsenterer månedsdagen.";
        this["Func_date_Category"] = "Dato og tid";
        this["Func_date_CategoryURL"] = "Date_and_Time";
        this["Func_date_Desc"] = "Returnerer det fortløbende serienummer, der repræsenterer en bestemt dato.";
        this["Func_dateadd_Arg_0"] = "interval";
        this["Func_dateadd_Arg_1"] = "tal";
        this["Func_dateadd_Arg_2"] = "dato";
        this["Func_dateadd_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer det tidsinterval, der skal tilføjes.\r\nd = dag\r\ny = År\r\nh = time\r\nn = minut\r\nm = måned\r\nq = kvartal\r\ns = sekund\r\nw = Uge\r\nww = Kalenderuge\r\nyyyy = År\r\n";
        this["Func_dateadd_ArgDesc_1"] = "Antallet af intervaller, der skal tilføjes. Denne værdi kan være positiv (for at få DatoTidspunkt i fremtiden) eller negativ (for at få DatoTidspunkt i fortiden). Det kan indeholde en brøkdel, når Interval angiver timer, minutter eller sekunder. For andre intervalværdier ignoreres enhver brøkdel af tal.";
        this["Func_dateadd_ArgDesc_2"] = "Den dato, som intervallet skal tilføjes til.";
        this["Func_dateadd_Category"] = "Dato og tid";
        this["Func_dateadd_CategoryURL"] = "Date_and_Time";
        this["Func_dateadd_Desc"] = "Returnerer en dato, hvortil det angivne interval er tilføjet.";
        this["Func_datediff_Arg_0"] = "interval";
        this["Func_datediff_Arg_1"] = "dato1";
        this["Func_datediff_Arg_2"] = "dato2";
        this["Func_datediff_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer tidsintervallet, der skal bruges som forskellenhed mellem dato1 og dato2.\r\nd = dag\r\ny = År\r\nh = time\r\nn = minut\r\nm = måned\r\nq = kvartal\r\ns = Andet\r\nw = Uge\r\nww = Kalenderuge\r\nyyyy = År\r\n";
        this["Func_datediff_ArgDesc_1"] = "Den første dato, der skal bruges i beregningen af differencen.";
        this["Func_datediff_ArgDesc_2"] = "Den anden dato, der skal bruges i beregningen af differencen.";
        this["Func_datediff_Category"] = "Dato og tid";
        this["Func_datediff_CategoryURL"] = "Date_and_Time";
        this["Func_datediff_Desc"] = "Returnerer en værdi, der angiver antallet af intervaller mellem to værdier for DatoTidspunkt.";
        this["Func_datevalue_Arg_0"] = "date_text";
        this["Func_datevalue_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer en dato.";
        this["Func_datevalue_Category"] = "Dato og tid";
        this["Func_datevalue_CategoryURL"] = "Date_and_Time";
        this["Func_datevalue_Desc"] = "Returnerer datoværdien for den angivne datotekst.";
        this["Func_day_Arg_0"] = "Date_value";
        this["Func_day_ArgDesc_0"] = "En datoværdi, hvorfra dagen skal findes.";
        this["Func_day_Category"] = "Dato og tid";
        this["Func_day_CategoryURL"] = "Date_and_Time";
        this["Func_day_Desc"] = "Returnerer en månedsdag for en datoværdi.";
        this["Func_days360_Arg_0"] = "start_date";
        this["Func_days360_Arg_1"] = "end_date";
        this["Func_days360_Arg_2"] = "metode";
        this["Func_days360_ArgDesc_0"] = "Startdatoen for intervallet af dage. Hvis denne ligger efter slutdatoen, returnerer funktionen et negativt tal.";
        this["Func_days360_ArgDesc_1"] = "Slutdatoen for intervallet af dage. Hvis denne ligger før startdatoen, returnerer funktionen et negativt tal.";
        this["Func_days360_ArgDesc_2"] = "En logisk værdi, der specificerer, om den amerikanske (falsk) eller europæiske (sand) metode skal bruges i beregningen.";
        this["Func_days360_Category"] = "Dato og tid";
        this["Func_days360_CategoryURL"] = "Date_and_Time";
        this["Func_days360_Desc"] = "Returnerer antallet af dage mellem to datoer baseret på et 360-dages år.";
        this["Func_db_Arg_0"] = "omkostning";
        this["Func_db_Arg_1"] = "bjærgning";
        this["Func_db_Arg_2"] = "levetid";
        this["Func_db_Arg_3"] = "periode";
        this["Func_db_Arg_4"] = "måned";
        this["Func_db_ArgDesc_0"] = "Aktivets startomkostning.";
        this["Func_db_ArgDesc_1"] = "Værdien ved afskrivningens afslutning (undertiden kaldet aktivets bjærgningsværdi).";
        this["Func_db_ArgDesc_2"] = "Antallet af perioder, hvor aktivet afskrives (kaldes også aktivets levetid).";
        this["Func_db_ArgDesc_3"] = "Den periode, afskrivningen skal beregnes for. Perioden skal bruge de samme enheder som levetiden.";
        this["Func_db_ArgDesc_4"] = "Antallet af måneder i det første år. Hvis måned udelades, antages den at være 12.";
        this["Func_db_Category"] = "Finansiel";
        this["Func_db_CategoryURL"] = "Finansiel";
        this["Func_db_Desc"] = "Returnerer afskrivningen på et aktiv i en bestemt periode ved hjælp af metoden med fast faldende saldo.";
        this["Func_dbnull_Category"] = "Information";
        this["Func_dbnull_CategoryURL"] = "Information";
        this["Func_dbnull_Desc"] = "Returnerer noget, der repræsenterer en ikke-eksisterende værdi.";
        this["Func_ddb_Arg_0"] = "omkostning";
        this["Func_ddb_Arg_1"] = "bjærgning";
        this["Func_ddb_Arg_2"] = "levetid";
        this["Func_ddb_Arg_3"] = "periode";
        this["Func_ddb_Arg_4"] = "faktor";
        this["Func_ddb_ArgDesc_0"] = "Aktivets startomkostning.";
        this["Func_ddb_ArgDesc_1"] = "Værdien ved afskrivningens afslutning (undertiden kaldet aktivets bjærgningsværdi).";
        this["Func_ddb_ArgDesc_2"] = "Antallet af perioder, hvor aktivet afskrives (kaldes også aktivets levetid).";
        this["Func_ddb_ArgDesc_3"] = "Den periode, afskrivningen skal beregnes for. Perioden skal bruge de samme enheder som levetiden.";
        this["Func_ddb_ArgDesc_4"] = "Den hastighed, hvormed saldoen falder. Hvis faktor udelades, antages det at være 2 (den dobbelte faldende balancemetode).";
        this["Func_ddb_Category"] = "Finansiel";
        this["Func_ddb_CategoryURL"] = "Finansiel";
        this["Func_ddb_Desc"] = "Returnerer afskrivningen på et aktiv i en bestemt periode ved hjælp af den dobbelte faldende balancemetode eller en anden specificeret metode.";
        this["Func_dec2bin_Arg_0"] = "Nummer";
        this["Func_dec2bin_Arg_1"] = "Pladser";
        this["Func_dec2bin_ArgDesc_0"] = "Den decimalværdi, der skal konverteres.";
        this["Func_dec2bin_ArgDesc_1"] = "Antallet af tegn, der skal bruges til at repræsentere værdien. Hvis der ikke er angivet pladser, anvendes det minimum antal tegn, der er nødvendigt for at repræsentere værdien. Dette giver dig mulighed for at angive foranstillede nuller.";
        this["Func_dec2bin_Category"] = "Engineering";
        this["Func_dec2bin_CategoryURL"] = "Engineering";
        this["Func_dec2bin_Desc"] = "Returnerer et tekstudtryk, der repræsenterer den angivne decimalværdi som et binært tal.";
        this["Func_dec2hex_Arg_0"] = "Nummer";
        this["Func_dec2hex_Arg_1"] = "Pladser";
        this["Func_dec2hex_ArgDesc_0"] = "Den decimalværdi, der skal konverteres.";
        this["Func_dec2hex_ArgDesc_1"] = "Antallet af tegn, der skal bruges til at repræsentere værdien. Hvis der ikke er angivet pladser, anvendes det minimum antal tegn, der er nødvendigt for at repræsentere værdien. Dette giver dig mulighed for at angive foranstillede nuller.";
        this["Func_dec2hex_Category"] = "Engineering";
        this["Func_dec2hex_CategoryURL"] = "Engineering";
        this["Func_dec2hex_Desc"] = "Returnerer et tekstudtryk, der repræsenterer den angivne decimalværdi som et hexadecimalt tal.";
        this["Func_dec2oct_Arg_0"] = "Nummer";
        this["Func_dec2oct_Arg_1"] = "Pladser";
        this["Func_dec2oct_ArgDesc_0"] = "Den decimalværdi, der skal konverteres.";
        this["Func_dec2oct_ArgDesc_1"] = "Antallet af tegn, der skal bruges til at repræsentere værdien. Hvis der ikke er angivet pladser, anvendes det minimum antal tegn, der er nødvendigt for at repræsentere værdien. Dette giver dig mulighed for at angive foranstillede nuller.";
        this["Func_dec2oct_Category"] = "Engineering";
        this["Func_dec2oct_CategoryURL"] = "Engineering";
        this["Func_dec2oct_Desc"] = "Returnerer et tekstudtryk, der repræsenterer den angivne decimalværdi som et oktalt tal.";
        this["Func_degrees_Arg_0"] = "Radianer";
        this["Func_degrees_ArgDesc_0"] = "En værdi i radianer, der konverteres til grader.";
        this["Func_degrees_Category"] = "Matematik";
        this["Func_degrees_CategoryURL"] = "Matematik";
        this["Func_degrees_Desc"] = "Konverterer radianer til grader.";
        this["Func_delta_Arg_0"] = "Tal 1";
        this["Func_delta_Arg_1"] = "Tal 2";
        this["Func_delta_ArgDesc_0"] = "Det første tal, der skal sammenlignes.";
        this["Func_delta_ArgDesc_1"] = "Det andet tal, der skal sammenlignes. Hvis det ikke er angivet, antages det at være nul.";
        this["Func_delta_Category"] = "Engineering";
        this["Func_delta_CategoryURL"] = "Engineering";
        this["Func_delta_Desc"] = "Sammenligner to tal og returnerer et, hvis de er ens eller nul, hvis de ikke er.";
        this["Func_dollarde_Arg_0"] = "BrøkDollarBeløb";
        this["Func_dollarde_Arg_1"] = "Brøk";
        this["Func_dollarde_ArgDesc_0"] = "Dollarbeløbet udtrykt som en brøk.";
        this["Func_dollarde_ArgDesc_1"] = "Den værdi, der bruges som nævneren for brøken. Hvis denne værdi ikke er et heltal, afkortes den.";
        this["Func_dollarde_Category"] = "Finansiel";
        this["Func_dollarde_CategoryURL"] = "Finansiel";
        this["Func_dollarde_Desc"] = "Konverterer et dollarbeløb udtrykt som en brøk til et dollarbeløb udtrykt som en decimal.";
        this["Func_dollarfr_Arg_0"] = "DecimalDollarBeløb";
        this["Func_dollarfr_Arg_1"] = "Brøk";
        this["Func_dollarfr_ArgDesc_0"] = "Dollarbeløbet udtrykt som en decimal.";
        this["Func_dollarfr_ArgDesc_1"] = "Den værdi, der bruges som nævneren for brøken. Hvis denne værdi ikke er et heltal, afkortes den.";
        this["Func_dollarfr_Category"] = "Finansiel";
        this["Func_dollarfr_CategoryURL"] = "Finansiel";
        this["Func_dollarfr_Desc"] = "Konverterer et dollarbeløb udtrykt som en decimal til et dollarbeløb udtrykt som en brøk.";
        this["Func_edate_Arg_0"] = "dato";
        this["Func_edate_Arg_1"] = "antalMåneder";
        this["Func_edate_ArgDesc_0"] = "Startdatoen.";
        this["Func_edate_ArgDesc_1"] = "Det antal måneder, hvor datoen skal flyttes. Et positivt tal betyder en dato efter startdatoen. Et negativt tal angiver en dato før startdatoen.";
        this["Func_edate_Category"] = "Dato og tid";
        this["Func_edate_CategoryURL"] = "Dato og tid";
        this["Func_edate_Desc"] = "Returnerer en dato, der er det angivne antal måneder før eller efter startdatoen.";
        this["Func_eomonth_Arg_0"] = "dato";
        this["Func_eomonth_Arg_1"] = "antalMåneder";
        this["Func_eomonth_ArgDesc_0"] = "Startdatoen.";
        this["Func_eomonth_ArgDesc_1"] = "Det antal måneder, hvor datoen skal flyttes. Et positivt tal betyder en dato efter startdatoen. Et negativt tal angiver en dato før startdatoen.";
        this["Func_eomonth_Category"] = "Dato og tid";
        this["Func_eomonth_CategoryURL"] = "Dato og tid";
        this["Func_eomonth_Desc"] = "Returnerer en dato, der er i slutningen af den måned, som er det angivne antal måneder før eller efter startdatoen.";
        this["Func_errortype_Arg_0"] = "error_value";
        this["Func_errortype_ArgDesc_0"] = "Fejlværdien, hvis identifikationsnummer skal findes.";
        this["Func_errortype_Category"] = "Information";
        this["Func_errortype_CategoryURL"] = "Information";
        this["Func_errortype_Desc"] = "Returnerer et tal, der svarer til en fejlværdi.";
        this["Func_even_Arg_0"] = "tal";
        this["Func_even_ArgDesc_0"] = "Værdien, der skal afrundes.";
        this["Func_even_Category"] = "Matematik";
        this["Func_even_CategoryURL"] = "Matematik";
        this["Func_even_Desc"] = "Runder et positivt tal op og et negativt tal ned til nærmeste lige heltal.";
        this["Func_exp_Arg_0"] = "tal";
        this["Func_exp_ArgDesc_0"] = "Eksponenten hvortil e skal opløftes";
        this["Func_exp_Category"] = "Matematik";
        this["Func_exp_CategoryURL"] = "Matematik";
        this["Func_exp_Desc"] = "Returnerer e ophævet til det angivne tals potens. Den konstante e er lig med 2,71828182845904.";
        this["Func_fact_Arg_0"] = "Nummer";
        this["Func_fact_ArgDesc_0"] = "Et positivt tal, som fakultetet beregnes for.";
        this["Func_fact_Category"] = "Matematik";
        this["Func_fact_CategoryURL"] = "Matematik";
        this["Func_fact_Desc"] = "Returnerer et tals fakultet";
        this["Func_factdouble_Arg_0"] = "Nummer";
        this["Func_factdouble_ArgDesc_0"] = "Et positivt tal, som det dobbelte fakultet beregnes for.";
        this["Func_factdouble_Category"] = "Matematik";
        this["Func_factdouble_CategoryURL"] = "Matematik";
        this["Func_factdouble_Desc"] = "Returnerer et tals dobbelte fakultet.";
        this["Func_false_Category"] = "Logisk";
        this["Func_false_CategoryURL"] = "Logisk";
        this["Func_false_Desc"] = "Returnerer den logiske værdi falsk.";
        this["Func_find_Arg_0"] = "find_text";
        this["Func_find_Arg_1"] = "within_text";
        this["Func_find_Arg_2"] = "start_num";
        this["Func_find_ArgDesc_0"] = "Teksten, der skal findes";
        this["Func_find_ArgDesc_1"] = "Teksten, der indeholder teksten, der skal findes.";
        this["Func_find_ArgDesc_2"] = "Det 1-baserede indekstegn, som søgningen skal startes med. Hvis den udelades, antages den at være en.";
        this["Func_find_Category"] = "TekstOgData";
        this["Func_find_CategoryURL"] = "Text_and_Data";
        this["Func_find_Desc"] = "Returnerer startpositionen for et tekstudtryk inde i et andet,  store bogstaver.";
        this["Func_fixed_Arg_0"] = "tal";
        this["Func_fixed_Arg_1"] = "decimaler";
        this["Func_fixed_Arg_2"] = "no_commas";
        this["Func_fixed_ArgDesc_0"] = "Antallet, der skal afrundes og konverteres til tekst.";
        this["Func_fixed_ArgDesc_1"] = "Antallet af cifre til højre for decimaltegnet i returværdien. Hvis denne værdi er negativ, afrundes 'tal' til venstre for decimaltegnet. Standardværdien er 2.";
        this["Func_fixed_ArgDesc_2"] = "Sandt, hvis kommaer ikke skulle være i returværdien. Hvis de udelades, antages den at være falsk.";
        this["Func_fixed_Category"] = "TekstOgData";
        this["Func_fixed_CategoryURL"] = "Text_and_Data";
        this["Func_fixed_Desc"] = "Afrunder et tal til det angivne antal decimaler og returnerer resultatet som tekst med eller uden kommaer.";
        this["Func_floor_Arg_0"] = "tal";
        this["Func_floor_Arg_1"] = "betydning";
        this["Func_floor_ArgDesc_0"] = "Tallet, der skal afrundes.";
        this["Func_floor_ArgDesc_1"] = "Det multipel, der skal afrundes til. Hvis det udelades, antages det at være en.";
        this["Func_floor_Category"] = "Matematik";
        this["Func_floor_CategoryURL"] = "Matematik";
        this["Func_floor_Desc"] = "Returnerer det største multiplum af betydningen (er som standard 1) mindre end eller lig med det givne tal.";
        this["Func_fv_Arg_0"] = "sats";
        this["Func_fv_Arg_1"] = "nper";
        this["Func_fv_Arg_2"] = "pmt";
        this["Func_fv_Arg_3"] = "pv";
        this["Func_fv_Arg_4"] = "type";
        this["Func_fv_ArgDesc_0"] = "Rentesatsen pr. periode.";
        this["Func_fv_ArgDesc_1"] = "Det samlede antal betalingsperioder i en annuitet.";
        this["Func_fv_ArgDesc_2"] = "Betalingen, der foretages hver periode, som ikke kan ændres gennem annuitetens løbetid.";
        this["Func_fv_ArgDesc_3"] = "Nuværdien eller værdien af den samlede sum, som en række fremtidige betalinger er værd lige nu. Hvis den udelades, antages den at være nul.";
        this["Func_fv_ArgDesc_4"] = "Tallet nul eller et, der angiver, om betalinger forfalder ved periodens afslutning (nul, standardværdien) eller begyndelsen (et)";
        this["Func_fv_Category"] = "Finansiel";
        this["Func_fv_CategoryURL"] = "Finansiel";
        this["Func_fv_Desc"] = "Returnerer den fremtidige værdi af en investering baseret på periodisk.";
        this["Func_gcd_Arg_0"] = "nummer {0}";
        this["Func_gcd_ArgDesc_0"] = "Et heltal. En ikke-heltal værdi afkortes.";
        this["Func_gcd_Category"] = "Matematik";
        this["Func_gcd_CategoryURL"] = "Matematik";
        this["Func_gcd_Desc"] = "Returnerer den største fællesdeler af en eller flere heltalværdier.";
        this["Func_gestep_Arg_0"] = "Tal 1";
        this["Func_gestep_Arg_1"] = "Trin";
        this["Func_gestep_ArgDesc_0"] = "Det første tal, der skal sammenlignes.";
        this["Func_gestep_ArgDesc_1"] = "Det andet tal, der skal sammenlignes. Hvis det ikke er angivet, antages det at være nul.";
        this["Func_gestep_Category"] = "Engineering";
        this["Func_gestep_CategoryURL"] = "Engineering";
        this["Func_gestep_Desc"] = "Sammenligner to tal og returnerer et, hvis det første tal er større end eller lig med det andet eller returnerer nul, hvis ikke.";
        this["Func_hex2bin_Arg_0"] = "Nummer";
        this["Func_hex2bin_Arg_1"] = "Pladser";
        this["Func_hex2bin_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer et hexadecimalt tal.";
        this["Func_hex2bin_ArgDesc_1"] = "Antallet af tegn, der skal bruges til at repræsentere værdien. Hvis dette ikke er angivet, anvendes det minimale antal tegn, der er nødvendigt for at repræsentere værdien. Dette giver dig mulighed for at angive foranstillede nuller.";
        this["Func_hex2bin_Category"] = "Engineering";
        this["Func_hex2bin_CategoryURL"] = "Engineering";
        this["Func_hex2bin_Desc"] = "Konverterer et tekstudtryk, der repræsenterer et hexadecimalt tal, til et tekstudtryk, der repræsenterer det samme tal i det binære nummersystem.";
        this["Func_hex2dec_Arg_0"] = "Nummer";
        this["Func_hex2dec_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer et hexadecimalt tal.";
        this["Func_hex2dec_Category"] = "Engineering";
        this["Func_hex2dec_CategoryURL"] = "Engineering";
        this["Func_hex2dec_Desc"] = "Returnerer decimalværdien af en tekstrepræsentation af et tal i det hexadecimale talsystem.";
        this["Func_hex2oct_Arg_0"] = "Nummer";
        this["Func_hex2oct_Arg_1"] = "Pladser";
        this["Func_hex2oct_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer et hexadecimalt tal.";
        this["Func_hex2oct_ArgDesc_1"] = "Antallet af tegn, der skal bruges til at repræsentere værdien. Hvis dette ikke er angivet, anvendes det minimale antal tegn, der er nødvendigt for at repræsentere værdien. Dette giver dig mulighed for at angive foranstillede nuller.";
        this["Func_hex2oct_Category"] = "Engineering";
        this["Func_hex2oct_CategoryURL"] = "Engineering";
        this["Func_hex2oct_Desc"] = "Konverterer et tekstudtryk, der repræsenterer et hexadecimalt tal, til et tekstudtryk, der repræsenterer det samme tal i det oktale talsystem.";
        this["Func_hour_Arg_0"] = "Time_value";
        this["Func_hour_ArgDesc_0"] = "En tidsværdi, hvorfra timen skal findes.";
        this["Func_hour_Category"] = "Dato og tid";
        this["Func_hour_CategoryURL"] = "Date_and_Time";
        this["Func_hour_Desc"] = "Returnerer timen for en tidsværdi.";
        this["Func_if_Arg_0"] = "logical_test";
        this["Func_if_Arg_1"] = "value_if_true";
        this["Func_if_Arg_2"] = "value_if_false";
        this["Func_if_ArgDesc_0"] = "Betingelsen til at teste.";
        this["Func_if_ArgDesc_1"] = "Resultatet, hvis tilstanden evalueres til sand.";
        this["Func_if_ArgDesc_2"] = "Resultatet, hvis tilstanden evalueres til falsk. Hvis den udelades, antages den at være den logiske værdi falsk.";
        this["Func_if_Category"] = "Logisk";
        this["Func_if_CategoryURL"] = "Logisk";
        this["Func_if_Desc"] = "Returnerer en værdi, hvis den angivne betingelse evalueres til sand og en anden værdi, hvis den evalueres til falsk.";
        this["Func_iferror_Arg_0"] = "værdi";
        this["Func_iferror_Arg_1"] = "value_if_error";
        this["Func_iferror_ArgDesc_0"] = "Værdien, der skal bruges, når det ikke er en fejl.";
        this["Func_iferror_ArgDesc_1"] = "Værdien, der skal bruges, hvis værdiargumentet er en fejl.";
        this["Func_iferror_Category"] = "Logisk";
        this["Func_iferror_CategoryURL"] = "Logisk";
        this["Func_iferror_Desc"] = "Returnerer en af to værdier, afhængigt af om den første værdi er en fejl eller ej.";
        this["Func_imabs_Arg_0"] = "KomplekstTal";
        this["Func_imabs_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imabs_Category"] = "Engineering";
        this["Func_imabs_CategoryURL"] = "Engineering";
        this["Func_imabs_Desc"] = "Returnerer den absolutte værdi af et komplekst tal.";
        this["Func_imaginary_Arg_0"] = "KomplekstTal";
        this["Func_imaginary_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imaginary_Category"] = "Engineering";
        this["Func_imaginary_CategoryURL"] = "Engineering";
        this["Func_imaginary_Desc"] = "Returnerer den imaginære koefficient for et komplekst tal.";
        this["Func_imargument_Arg_0"] = "KomplekstTal";
        this["Func_imargument_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imargument_Category"] = "Engineering";
        this["Func_imargument_CategoryURL"] = "Engineering";
        this["Func_imargument_Desc"] = "Returnerer argumentet theta, en vinkel udtrykt i radianer.";
        this["Func_imconjugate_Arg_0"] = "KomplekstTal";
        this["Func_imconjugate_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imconjugate_Category"] = "Engineering";
        this["Func_imconjugate_CategoryURL"] = "Engineering";
        this["Func_imconjugate_Desc"] = "Returnerer konjugatet af et komplekst tal.";
        this["Func_imcos_Arg_0"] = "KomplekstTal";
        this["Func_imcos_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imcos_Category"] = "Engineering";
        this["Func_imcos_CategoryURL"] = "Engineering";
        this["Func_imcos_Desc"] = "Returnerer cosinus for et komplekst tal.";
        this["Func_imdiv_Arg_0"] = "Dividend";
        this["Func_imdiv_Arg_1"] = "divisor";
        this["Func_imdiv_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imdiv_ArgDesc_1"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imdiv_Category"] = "Engineering";
        this["Func_imdiv_CategoryURL"] = "Engineering";
        this["Func_imdiv_Desc"] = "Returnerer kvotienten for to komplekse tal (Dividend / Divisor).";
        this["Func_imexp_Arg_0"] = "KomplekstTal";
        this["Func_imexp_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imexp_Category"] = "Engineering";
        this["Func_imexp_CategoryURL"] = "Engineering";
        this["Func_imexp_Desc"] = "Returnerer eksponentialet for et komplekst tal.";
        this["Func_imln_Arg_0"] = "KomplekstTal";
        this["Func_imln_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imln_Category"] = "Engineering";
        this["Func_imln_CategoryURL"] = "Engineering";
        this["Func_imln_Desc"] = "Returnerer den naturlige logaritme for et komplekst tal.";
        this["Func_imlog10_Arg_0"] = "KomplekstTal";
        this["Func_imlog10_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imlog10_Category"] = "Engineering";
        this["Func_imlog10_CategoryURL"] = "Engineering";
        this["Func_imlog10_Desc"] = "Returnerer base-10 logaritmen for et komplekst tal.";
        this["Func_imlog2_Arg_0"] = "KomplekstTal";
        this["Func_imlog2_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imlog2_Category"] = "Engineering";
        this["Func_imlog2_CategoryURL"] = "Engineering";
        this["Func_imlog2_Desc"] = "Returnerer base-2-logaritmen for et komplekst tal.";
        this["Func_impower_Arg_0"] = "KomplekstTal";
        this["Func_impower_Arg_1"] = "potens";
        this["Func_impower_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_impower_ArgDesc_1"] = "Den potens, som KomplekstTal ophæves til.";
        this["Func_impower_Category"] = "Engineering";
        this["Func_impower_CategoryURL"] = "Engineering";
        this["Func_impower_Desc"] = "Returnerer det komplekse tal ophævet til den angivne potens.";
        this["Func_improduct_Arg_0"] = "KomplekstTal{0}";
        this["Func_improduct_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_improduct_Category"] = "Engineering";
        this["Func_improduct_CategoryURL"] = "Engineering";
        this["Func_improduct_Desc"] = "Returnerer produktet af et eller flere komplekse tal.";
        this["Func_imreal_Arg_0"] = "KomplekstTal";
        this["Func_imreal_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imreal_Category"] = "Engineering";
        this["Func_imreal_CategoryURL"] = "Engineering";
        this["Func_imreal_Desc"] = "Returnerer den reelle koefficient for et komplekst tal.";
        this["Func_imsin_Arg_0"] = "KomplekstTal";
        this["Func_imsin_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imsin_Category"] = "Engineering";
        this["Func_imsin_CategoryURL"] = "Engineering";
        this["Func_imsin_Desc"] = "Returnerer sinus for et komplekst tal.";
        this["Func_imsqrt_Arg_0"] = "KomplekstTal";
        this["Func_imsqrt_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imsqrt_Category"] = "Engineering";
        this["Func_imsqrt_CategoryURL"] = "Engineering";
        this["Func_imsqrt_Desc"] = "Returnerer kvadratroden af et komplekst tal.";
        this["Func_imsub_Arg_0"] = "KomplekstTal1";
        this["Func_imsub_Arg_1"] = "KomplekstTal2";
        this["Func_imsub_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj', hvorfra KomplekstTal2 subtraheres.";
        this["Func_imsub_ArgDesc_1"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj', der subtraheres fra KomplekstTal1.";
        this["Func_imsub_Category"] = "Engineering";
        this["Func_imsub_CategoryURL"] = "Engineering";
        this["Func_imsub_Desc"] = "Returnerer differencen mellem to komplekse tal.";
        this["Func_imsum_Arg_0"] = "KomplekstTal{0}";
        this["Func_imsum_ArgDesc_0"] = "Tekstrepræsentationen af et komplekst tal i formatet: 'x + yi' eller 'x + yj'.";
        this["Func_imsum_Category"] = "Engineering";
        this["Func_imsum_CategoryURL"] = "Engineering";
        this["Func_imsum_Desc"] = "Returnerer summen af et eller flere komplekse tal.";
        this["Func_info_Arg_0"] = "Type";
        this["Func_info_ArgDesc_0"] = "Et tekstudtryk, der angiver typen af information, der skal returneres.\r\n\"directory\" = Den fulde sti til den aktuelle mappe.\r\n\"osversion\" = Den aktuelle operativsystemversion som et tekstudtryk.\r\n\"system\" = Det aktuelle operativsystem. Macintosh = \"mac\", Windows = \"pcdos\".\r\n    ";
        this["Func_info_Category"] = "Information";
        this["Func_info_CategoryURL"] = "Information";
        this["Func_info_Desc"] = "Returnerer oplysninger om det aktuelle operativsystem.";
        this["Func_int_Arg_0"] = "tal";
        this["Func_int_ArgDesc_0"] = "Det reelle tal, der skal afrundes til et heltal.";
        this["Func_int_Category"] = "Matematik";
        this["Func_int_CategoryURL"] = "Matematik";
        this["Func_int_Desc"] = "Runder et tal ned til nærmeste heltal.";
        this["Func_intrate_Arg_0"] = "afregning";
        this["Func_intrate_Arg_1"] = "fortaldstid";
        this["Func_intrate_Arg_2"] = "investering";
        this["Func_intrate_Arg_3"] = "indløsning";
        this["Func_intrate_Arg_4"] = "basis";
        this["Func_intrate_ArgDesc_0"] = "Værdipapirets afviklingsdato.";
        this["Func_intrate_ArgDesc_1"] = "Værdipapirets forfaldsdato.";
        this["Func_intrate_ArgDesc_2"] = "Beløbet investeret i værdipapiret.";
        this["Func_intrate_ArgDesc_3"] = "Det beløb, der skal modtages ved forfaldsdato.";
        this["Func_intrate_ArgDesc_4"] = "Den type dagtællingsgrundlag, der skal bruges.\r\n0 eller udeladt = US (NASD) 30/360,\r\n1 = Faktisk / faktisk\r\n2 = Faktisk / 360,\r\n3 = Faktisk / 365,\r\n4 = europæisk 30/360\r\n    ";
        this["Func_intrate_Category"] = "Finansiel";
        this["Func_intrate_CategoryURL"] = "Finansiel";
        this["Func_intrate_Desc"] = "Returnerer renten for et fuldt investeret værdipapir.";
        this["Func_ipmt_Arg_0"] = "sats";
        this["Func_ipmt_Arg_1"] = "pr.";
        this["Func_ipmt_Arg_2"] = "nper";
        this["Func_ipmt_Arg_3"] = "pv";
        this["Func_ipmt_Arg_4"] = "fv";
        this["Func_ipmt_Arg_5"] = "type";
        this["Func_ipmt_ArgDesc_0"] = "Rentesatsen pr. periode.";
        this["Func_ipmt_ArgDesc_1"] = "Perioden, som skal være i intervallet fra en til nper.";
        this["Func_ipmt_ArgDesc_2"] = "Det samlede antal betalingsperioder i en annuitet.";
        this["Func_ipmt_ArgDesc_3"] = "Nuværdien eller den samlede sum, som en række fremtidige betalinger er værd nu.";
        this["Func_ipmt_ArgDesc_4"] = "Den fremtidige værdi eller en kontantsaldo, du vil opnå, efter den sidste betaling er foretaget. Hvis den udelades, antages den at være nul.";
        this["Func_ipmt_ArgDesc_5"] = "Tallet nul eller et, der angiver, om betalinger forfalder ved periodens afslutning (nul, standardværdien) eller begyndelsen (et)";
        this["Func_ipmt_Category"] = "Finansiel";
        this["Func_ipmt_CategoryURL"] = "Finansiel";
        this["Func_ipmt_Desc"] = "Returnerer rentebetalingen for en given periode for en investering baseret på periodiske, konstante betalinger og en konstant rentesats.";
        this["Func_irr_Arg_0"] = "værdier";
        this["Func_irr_Arg_1"] = "gæt";
        this["Func_irr_ArgDesc_0"] = "En samling af betalingsværdier (negative værdier) og indkomstsværdier (positive værdier), der forekommer med jævne mellemrum.";
        this["Func_irr_ArgDesc_1"] = "Et tal, som du gætter, vil komme tæt på resultatet. Standard er 0,1 (10%).";
        this["Func_irr_Category"] = "Finansiel";
        this["Func_irr_CategoryURL"] = "Finansiel";
        this["Func_irr_Desc"] = "Returnerer det interne afkast for en række pengestrømme.";
        this["Func_isblank_Arg_0"] = "værdi";
        this["Func_isblank_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_isblank_Category"] = "Information";
        this["Func_isblank_CategoryURL"] = "Information";
        this["Func_isblank_Desc"] = "Returnerer sandt, hvis værdien er tom.";
        this["Func_isdbnull_Arg_0"] = "værdi";
        this["Func_isdbnull_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_isdbnull_Category"] = "Information";
        this["Func_isdbnull_CategoryURL"] = "Information";
        this["Func_isdbnull_Desc"] = "Returnerer sandt, hvis værdien repræsenterer en ikke-eksisterende værdi.";
        this["Func_iserr_Arg_0"] = "værdi";
        this["Func_iserr_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_iserr_Category"] = "Information";
        this["Func_iserr_CategoryURL"] = "Information";
        this["Func_iserr_Desc"] = "Returnerer sandt, hvis værdien er en hvilken som helst fejlværdi undtagen #N/A.";
        this["Func_iserror_Arg_0"] = "værdi";
        this["Func_iserror_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_iserror_Category"] = "Information";
        this["Func_iserror_CategoryURL"] = "Information";
        this["Func_iserror_Desc"] = "Returnerer sandt, hvis værdien er en hvilken som helst fejlværdi.";
        this["Func_iseven_Arg_0"] = "værdi";
        this["Func_iseven_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_iseven_Category"] = "Information";
        this["Func_iseven_CategoryURL"] = "Information";
        this["Func_iseven_Desc"] = "Returnerer sandt, hvis værdien er et lige tal, ellers returneres falsk.";
        this["Func_islogical_Arg_0"] = "værdi";
        this["Func_islogical_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_islogical_Category"] = "Information";
        this["Func_islogical_CategoryURL"] = "Information";
        this["Func_islogical_Desc"] = "Returnerer sandt, hvis værdien er en logisk værdi (sand eller falsk).";
        this["Func_isna_Arg_0"] = "værdi";
        this["Func_isna_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_isna_Category"] = "Information";
        this["Func_isna_CategoryURL"] = "Information";
        this["Func_isna_Desc"] = "Returnerer sandt, hvis værdien er # N/A (værdi ikke tilgængelig) fejlværdien.";
        this["Func_isnontext_Arg_0"] = "værdi";
        this["Func_isnontext_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_isnontext_Category"] = "Information";
        this["Func_isnontext_CategoryURL"] = "Information";
        this["Func_isnontext_Desc"] = "Returnerer sandt, hvis værdien er et element, der ikke er tekst.";
        this["Func_isnull_Arg_0"] = "værdi";
        this["Func_isnull_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_isnull_Category"] = "Information";
        this["Func_isnull_CategoryURL"] = "Information";
        this["Func_isnull_Desc"] = "Returnerer sandt, hvis værdien ikke eksisterer.";
        this["Func_isnumber_Arg_0"] = "værdi";
        this["Func_isnumber_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_isnumber_Category"] = "Information";
        this["Func_isnumber_CategoryURL"] = "Information";
        this["Func_isnumber_Desc"] = "Returnerer sandt, hvis værdien er et tal.";
        this["Func_isodd_Arg_0"] = "værdi";
        this["Func_isodd_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_isodd_Category"] = "Information";
        this["Func_isodd_CategoryURL"] = "Information";
        this["Func_isodd_Desc"] = "Returnerer sandt, hvis værdien er et ulige tal, ellers returneres falsk.";
        this["Func_isref_Arg_0"] = "værdi";
        this["Func_isref_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_isref_Category"] = "Information";
        this["Func_isref_CategoryURL"] = "Information";
        this["Func_isref_Desc"] = "Returnerer sandt, hvis værdien er en reference.";
        this["Func_istext_Arg_0"] = "værdi";
        this["Func_istext_ArgDesc_0"] = "Værdien, der skal testes.";
        this["Func_istext_Category"] = "Information";
        this["Func_istext_CategoryURL"] = "Information";
        this["Func_istext_Desc"] = "Returnerer sandt, hvis værdien er en tekstværdi.";
        this["Func_lcm_Arg_0"] = "nummer {0}";
        this["Func_lcm_ArgDesc_0"] = "Et heltal. En ikke-heltal værdi afkortes.";
        this["Func_lcm_Category"] = "Matematik";
        this["Func_lcm_CategoryURL"] = "Matematik";
        this["Func_lcm_Desc"] = "Returnerer det mindst almindelige multiplum af en eller flere heltalværdier.";
        this["Func_left_Arg_0"] = "tekst";
        this["Func_left_Arg_1"] = "num_chars";
        this["Func_left_ArgDesc_0"] = "Tekstudtrykket, der indeholder de tegn, der skal uddrages.";
        this["Func_left_ArgDesc_1"] = "Antallet af tegn, der skal uddrages.";
        this["Func_left_Category"] = "TekstOgData";
        this["Func_left_CategoryURL"] = "Text_and_Data";
        this["Func_left_Desc"] = "Returnerer det eller de første tegn i et tekstudtryk.";
        this["Func_len_Arg_0"] = "tekst";
        this["Func_len_ArgDesc_0"] = "Tekstudtryk hvis længde skal findes. Mellemrum tæller som tegn.";
        this["Func_len_Category"] = "TekstOgData";
        this["Func_len_CategoryURL"] = "Text_and_Data";
        this["Func_len_Desc"] = "Returnerer antallet af tegn i et tekstudtryk.";
        this["Func_ln_Arg_0"] = "tal";
        this["Func_ln_ArgDesc_0"] = "Det positive reelle tal, hvortil den naturlige logaritme skal findes.";
        this["Func_ln_Category"] = "Matematik";
        this["Func_ln_CategoryURL"] = "Matematik";
        this["Func_ln_Desc"] = "Returnerer den naturlige logaritme for et tal.";
        this["Func_log_Arg_0"] = "tal";
        this["Func_log_Arg_1"] = "base";
        this["Func_log_ArgDesc_0"] = "Det positive reelle tal, hvortil logaritmen skal findes.";
        this["Func_log_ArgDesc_1"] = "Logaritmens base.";
        this["Func_log_Category"] = "Matematik";
        this["Func_log_CategoryURL"] = "Matematik";
        this["Func_log_Desc"] = "Returnerer logaritmen for et tal til den angivne base.";
        this["Func_log10_Arg_0"] = "tal";
        this["Func_log10_ArgDesc_0"] = "Det positive reelle tal, hvortil base-10 logaritmen skal findes.";
        this["Func_log10_Category"] = "Matematik";
        this["Func_log10_CategoryURL"] = "Matematik";
        this["Func_log10_Desc"] = "Returnerer basis-10-logaritmen for et tal.";
        this["Func_lower_Arg_0"] = "tekst";
        this["Func_lower_ArgDesc_0"] = "Teksten, der skal konverteres til små bogstaver.";
        this["Func_lower_Category"] = "TekstOgData";
        this["Func_lower_CategoryURL"] = "Text_and_Data";
        this["Func_lower_Desc"] = "Konverterer alle store bogstaver i et tekstudtryk til små bogstaver.";
        this["Func_max_Arg_0"] = "nummer {0}";
        this["Func_max_ArgDesc_0"] = "Et tal eller en række af tal.";
        this["Func_max_Category"] = "Statistisk";
        this["Func_max_CategoryURL"] = "Statistisk";
        this["Func_max_Desc"] = "Returnerer det største tal i et sæt tal.";
        this["Func_median_Arg_0"] = "nummer {0}";
        this["Func_median_ArgDesc_0"] = "Et tal eller en række af tal.";
        this["Func_median_Category"] = "Statistisk";
        this["Func_median_CategoryURL"] = "Statistisk";
        this["Func_median_Desc"] = "Returnerer medianen for et sæt tal. Medianen er det midterste tal, når tallene sorteres, eller gennemsnittet af de to midterste tal, hvis der er et lige antal specificerede tal.";
        this["Func_mid_Arg_0"] = "tekst";
        this["Func_mid_Arg_1"] = "start_num";
        this["Func_mid_Arg_2"] = "num_chars";
        this["Func_mid_ArgDesc_0"] = "Tekstudtrykket, der indeholder de tegn, der skal uddrages.";
        this["Func_mid_ArgDesc_1"] = "Det 1-baserede indeks for det første tegn, der skal uddrages.";
        this["Func_mid_ArgDesc_2"] = "Antallet af tegn, der skal uddrages.";
        this["Func_mid_Category"] = "TekstOgData";
        this["Func_mid_CategoryURL"] = "Text_and_Data";
        this["Func_mid_Desc"] = "Returnerer et bestemt antal tegn fra et tekstudtryk.";
        this["Func_min_Arg_0"] = "nummer {0}";
        this["Func_min_ArgDesc_0"] = "Et tal eller en række af tal.";
        this["Func_min_Category"] = "Statistisk";
        this["Func_min_CategoryURL"] = "Statistisk";
        this["Func_min_Desc"] = "Returnerer det mindste tal i et sæt tal.";
        this["Func_minute_Arg_0"] = "Time_value";
        this["Func_minute_ArgDesc_0"] = "En tidsværdi, hvorfra minuttet skal findes.";
        this["Func_minute_Category"] = "Dato og tid";
        this["Func_minute_CategoryURL"] = "Date_and_Time";
        this["Func_minute_Desc"] = "Returnerer minuttet for en tidsværdi.";
        this["Func_mod_Arg_0"] = "tal";
        this["Func_mod_Arg_1"] = "divisor";
        this["Func_mod_ArgDesc_0"] = "Tallet for hvilket resten skal findes.";
        this["Func_mod_ArgDesc_1"] = "Den værdi, hvormed tallet skal divideres.";
        this["Func_mod_Category"] = "Matematik";
        this["Func_mod_CategoryURL"] = "Matematik";
        this["Func_mod_Desc"] = "Returnerer resten efter et tal er divideret med en divisor. Resultatet har det samme tegn som divisoren.";
        this["Func_month_Arg_0"] = "Date_value";
        this["Func_month_ArgDesc_0"] = "En datoværdi, hvorfra måneden skal findes.";
        this["Func_month_Category"] = "Dato og tid";
        this["Func_month_CategoryURL"] = "Date_and_Time";
        this["Func_month_Desc"] = "Returnerer måneden for en datoværdi.";
        this["Func_mround_Arg_0"] = "Nummer";
        this["Func_mround_Arg_1"] = "Mange";
        this["Func_mround_ArgDesc_0"] = "Antallet, der skal afrundes.";
        this["Func_mround_ArgDesc_1"] = "Et tal, der angiver det multiple, som du skal afrunde til.";
        this["Func_mround_Category"] = "Matematik";
        this["Func_mround_CategoryURL"] = "Matematik";
        this["Func_mround_Desc"] = "Afrunder et tal til det nærmeste multiplum af et andet tal.";
        this["Func_multinomial_Arg_0"] = "nummer {0}";
        this["Func_multinomial_ArgDesc_0"] = "Et af de tal, som multinomialet skal findes til.";
        this["Func_multinomial_Category"] = "Matematik";
        this["Func_multinomial_CategoryURL"] = "Matematik";
        this["Func_multinomial_Desc"] = "Returnerer multinomialet for et sæt tal. Multinomialet er forholdet mellem faktoriet af en sum af værdier og produktet af deres fakta.";
        this["Func_n_Arg_0"] = "Værdi";
        this["Func_n_ArgDesc_0"] = "Den værdi, der skal konverteres til et tal. Følgende er acceptable værdier:\r\nEt tal - returnerer tallet.\r\nEn dato - returnerer datoen.\r\nSandt - returnerer 1.\r\nFalsk - returnerer 0.\r\nEn fejlværdi - returnerer fejlværdien.\r\nEnhver anden værdi - returnerer 0.\r\n    ";
        this["Func_n_Category"] = "Information";
        this["Func_n_CategoryURL"] = "Information";
        this["Func_n_Desc"] = "Konverterer en værdi til et tal.";
        this["Func_na_Category"] = "Information";
        this["Func_na_CategoryURL"] = "Information";
        this["Func_na_Desc"] = "Returnerer fejlværdien # N / A (værdi ikke tilgængelig).";
        this["Func_networkdays_Arg_0"] = "startDato";
        this["Func_networkdays_Arg_1"] = "slutDato";
        this["Func_networkdays_Arg_2"] = "helligdage";
        this["Func_networkdays_ArgDesc_0"] = "Datoen, hvorfra du starter.";
        this["Func_networkdays_ArgDesc_1"] = "Datoen, hvor du slutter.";
        this["Func_networkdays_ArgDesc_2"] = "En række af en eller flere referencer, der indeholder datoer, der skal betragtes som ikke-arbejdsdage.";
        this["Func_networkdays_Category"] = "Dato og tid";
        this["Func_networkdays_CategoryURL"] = "Dato og tid";
        this["Func_networkdays_Desc"] = "Returnerer det samlede antal hele arbejdsdage mellem de angivne datoer, ekskl. specificerede helligdage.";
        this["Func_not_Arg_0"] = "Logisk";
        this["Func_not_ArgDesc_0"] = "En logisk værdi (sand eller falsk).";
        this["Func_not_Category"] = "Logisk";
        this["Func_not_CategoryURL"] = "Logisk";
        this["Func_not_Desc"] = "Vender en logisk værdi (ændrer sandt til falsk og falsk til sandt).";
        this["Func_now_Category"] = "Dato og tid";
        this["Func_now_CategoryURL"] = "Date_and_Time";
        this["Func_now_Desc"] = "Returnerer en værdi, der repræsenterer den aktuelle dato og tid.";
        this["Func_nper_Arg_0"] = "sats";
        this["Func_nper_Arg_1"] = "pmt";
        this["Func_nper_Arg_2"] = "pv";
        this["Func_nper_Arg_3"] = "fv";
        this["Func_nper_Arg_4"] = "type";
        this["Func_nper_ArgDesc_0"] = "Rentesatsen pr. periode.";
        this["Func_nper_ArgDesc_1"] = "Betalingen, der foretages hver periode, som ikke kan ændres gennem annuitetens løbetid.";
        this["Func_nper_ArgDesc_2"] = "Nuværdien eller det engangsbeløb, som en række fremtidige betalinger er værd lige nu.";
        this["Func_nper_ArgDesc_3"] = "Den fremtidige værdi eller en kontantsaldo, som du vil opnå efter den sidste betaling er foretaget.";
        this["Func_nper_ArgDesc_4"] = "Tallet nul eller et, der angiver, om betalinger forfalder ved periodens afslutning (nul, standardværdien) eller begyndelsen (et)";
        this["Func_nper_Category"] = "Finansiel";
        this["Func_nper_CategoryURL"] = "Finansiel";
        this["Func_nper_Desc"] = "Returnerer antallet af perioder for en investering baseret på periodiske, konstante betalinger og en konstant rentesats.";
        this["Func_npv_Arg_0"] = "sats";
        this["Func_npv_Arg_1"] = "værdi {0}";
        this["Func_npv_ArgDesc_0"] = "Rabatsatsen over en periode.";
        this["Func_npv_ArgDesc_1"] = "En betaling (negativ værdi) eller indkomst (positiv værdi).";
        this["Func_npv_Category"] = "Finansiel";
        this["Func_npv_CategoryURL"] = "Finansiel";
        this["Func_npv_Desc"] = "Beregner en investerings aktuelle nutidsværdi ved hjælp af en diskonteringssats og en række fremtidige betalinger (negative værdier) og indkomst (positiv værdi).";
        this["Func_null_Category"] = "Information";
        this["Func_null_CategoryURL"] = "Information";
        this["Func_null_Desc"] = "Returnerer en ikke-eksisterende værdi.";
        this["Func_oct2bin_Arg_0"] = "Nummer";
        this["Func_oct2bin_Arg_1"] = "Pladser";
        this["Func_oct2bin_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer et oktalt tal.";
        this["Func_oct2bin_ArgDesc_1"] = "Antallet af tegn, der skal bruges til at repræsentere værdien. Hvis dette ikke er angivet, anvendes det minimale antal tegn, der er nødvendigt for at repræsentere værdien. Dette giver dig mulighed for at angive foranstillede nuller.";
        this["Func_oct2bin_Category"] = "Engineering";
        this["Func_oct2bin_CategoryURL"] = "Engineering";
        this["Func_oct2bin_Desc"] = "Konverterer et tekstudtryk, der repræsenterer et oktalt tal til et tekstudtryk, der repræsenterer det samme tal i det binære nummersystem.";
        this["Func_oct2dec_Arg_0"] = "Nummer";
        this["Func_oct2dec_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer et oktalt tal.";
        this["Func_oct2dec_Category"] = "Engineering";
        this["Func_oct2dec_CategoryURL"] = "Engineering";
        this["Func_oct2dec_Desc"] = "Returnerer decimalværdien af en tekstrepræsentation af et tal i det oktale talsystem.";
        this["Func_oct2hex_Arg_0"] = "Nummer";
        this["Func_oct2hex_Arg_1"] = "Pladser";
        this["Func_oct2hex_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer et oktalt tal.";
        this["Func_oct2hex_ArgDesc_1"] = "Antallet af tegn, der skal bruges til at repræsentere værdien. Hvis dette ikke er angivet, anvendes det minimale antal tegn, der er nødvendigt for at repræsentere værdien. Dette giver dig mulighed for at angive foranstillede nuller.";
        this["Func_oct2hex_Category"] = "Engineering";
        this["Func_oct2hex_CategoryURL"] = "Engineering";
        this["Func_oct2hex_Desc"] = "Konverterer et tekstudtryk, der repræsenterer et oktalt tal til et tekstudtryk, der repræsenterer det samme tal i det hexadecimale talsystem.";
        this["Func_odd_Arg_0"] = "tal";
        this["Func_odd_ArgDesc_0"] = "Værdien, der skal afrundes.";
        this["Func_odd_Category"] = "Matematik";
        this["Func_odd_CategoryURL"] = "Matematik";
        this["Func_odd_Desc"] = "Runder et positivt tal op og et negativt tal ned til nærmeste ulige heltal.";
        this["Func_or_Arg_0"] = "logisk {0}";
        this["Func_or_ArgDesc_0"] = "En logisk værdi (sand eller falsk).";
        this["Func_or_Category"] = "Logisk";
        this["Func_or_CategoryURL"] = "Logisk";
        this["Func_or_Desc"] = "Returnerer sandt, hvis en eller flere værdier er sande; returnerer falsk, hvis alle værdier er falske.";
        this["Func_pi_Category"] = "Matematik";
        this["Func_pi_CategoryURL"] = "Matematik";
        this["Func_pi_Desc"] = "Returnerer den konstante pi, som er 3.14159265358979.";
        this["Func_pmt_Arg_0"] = "sats";
        this["Func_pmt_Arg_1"] = "nper";
        this["Func_pmt_Arg_2"] = "pv";
        this["Func_pmt_Arg_3"] = "fv";
        this["Func_pmt_Arg_4"] = "type";
        this["Func_pmt_ArgDesc_0"] = "Rentesatsen for lånet.";
        this["Func_pmt_ArgDesc_1"] = "Det samlede antal betalinger for lånet.";
        this["Func_pmt_ArgDesc_2"] = "Nuværdien eller det samlede beløb, som en række fremtidige betalinger er værd nu; også kendt som hovedstolen.";
        this["Func_pmt_ArgDesc_3"] = "Den fremtidige værdi eller en kontantsaldo, som du vil opnå efter den sidste betaling er foretaget.";
        this["Func_pmt_ArgDesc_4"] = "Tallet nul eller et, der angiver, om betalinger forfalder ved periodens afslutning (nul, standardværdien) eller begyndelsen (et)";
        this["Func_pmt_Category"] = "Finansiel";
        this["Func_pmt_CategoryURL"] = "Finansiel";
        this["Func_pmt_Desc"] = "Beregner betalingen for et lån baseret på konstante betalinger og en konstant rentesats.";
        this["Func_power_Arg_0"] = "tal";
        this["Func_power_Arg_1"] = "potens";
        this["Func_power_ArgDesc_0"] = "Grundtallet, der skal opløftes til en potens";
        this["Func_power_ArgDesc_1"] = "Eksponenten, som grundtallet opløftes til.";
        this["Func_power_Category"] = "Matematik";
        this["Func_power_CategoryURL"] = "Matematik";
        this["Func_power_Desc"] = "Returnerer resultatet af et tal opløftet til en potens.";
        this["Func_ppmt_Arg_0"] = "sats";
        this["Func_ppmt_Arg_1"] = "pr.";
        this["Func_ppmt_Arg_2"] = "nper";
        this["Func_ppmt_Arg_3"] = "pv";
        this["Func_ppmt_Arg_4"] = "fv";
        this["Func_ppmt_Arg_5"] = "type";
        this["Func_ppmt_ArgDesc_0"] = "Rentesatsen pr. periode.";
        this["Func_ppmt_ArgDesc_1"] = "Perioden, som skal være i intervallet fra en til nper.";
        this["Func_ppmt_ArgDesc_2"] = "Det samlede antal betalingsperioder i en annuitet.";
        this["Func_ppmt_ArgDesc_3"] = "Nuværdien eller det samlede beløb, som en række fremtidige betalinger er værd nu.";
        this["Func_ppmt_ArgDesc_4"] = "Den fremtidige værdi eller en kontantsaldo, som du vil opnå efter den sidste betaling er foretaget.";
        this["Func_ppmt_ArgDesc_5"] = "Tallet nul eller et, der angiver, om betalinger forfalder ved periodens afslutning (nul, standardværdien) eller begyndelsen (et)";
        this["Func_ppmt_Category"] = "Finansiel";
        this["Func_ppmt_CategoryURL"] = "Finansiel";
        this["Func_ppmt_Desc"] = "Returnerer betalingen på hovedstolen i en given periode for en investering baseret på periodiske, konstante betalinger og en konstant rentesats.";
        this["Func_product_Arg_0"] = "nummer {0}";
        this["Func_product_ArgDesc_0"] = "Et tal, der skal multipliceres med de andre tal.";
        this["Func_product_Category"] = "Matematik";
        this["Func_product_CategoryURL"] = "Matematik";
        this["Func_product_Desc"] = "Multipliserer alle de tal, der er angivet som argumenter, og returnerer produktet.";
        this["Func_pv_Arg_0"] = "sats";
        this["Func_pv_Arg_1"] = "nper";
        this["Func_pv_Arg_2"] = "pmt";
        this["Func_pv_Arg_3"] = "fv";
        this["Func_pv_Arg_4"] = "type";
        this["Func_pv_ArgDesc_0"] = "Rentesatsen pr. periode.";
        this["Func_pv_ArgDesc_1"] = "Det samlede antal betalingsperioder i en annuitet.";
        this["Func_pv_ArgDesc_2"] = "Betalingen, der foretages hver periode, som ikke kan ændres gennem annuitetens løbetid.";
        this["Func_pv_ArgDesc_3"] = "Den fremtidige værdi eller en kontantsaldo, som du vil opnå efter den sidste betaling er foretaget.";
        this["Func_pv_ArgDesc_4"] = "Tallet nul eller et, der angiver, om betalinger forfalder ved periodens afslutning (nul, standardværdien) eller begyndelsen (et)";
        this["Func_pv_Category"] = "Finansiel";
        this["Func_pv_CategoryURL"] = "Finansiel";
        this["Func_pv_Desc"] = "Returnerer nutidsværdien af en investering.";
        this["Func_quotient_Arg_0"] = "tæller";
        this["Func_quotient_Arg_1"] = "nævner";
        this["Func_quotient_ArgDesc_0"] = "Divisionshandlingens dividende.";
        this["Func_quotient_ArgDesc_1"] = "Divisionshandlingens divisor.";
        this["Func_quotient_Category"] = "Matematik";
        this["Func_quotient_CategoryURL"] = "Matematik";
        this["Func_quotient_Desc"] = "Returnerer det hele resultat af en division. Dette vil kassere resten fra divisionen.";
        this["Func_radians_Arg_0"] = "Grader";
        this["Func_radians_ArgDesc_0"] = "En værdi i grader, der konverteres til radianer.";
        this["Func_radians_Category"] = "Matematik";
        this["Func_radians_CategoryURL"] = "Matematik";
        this["Func_radians_Desc"] = "Konverterer grader til radianer.";
        this["Func_rand_Category"] = "Matematik";
        this["Func_rand_CategoryURL"] = "Matematik";
        this["Func_rand_Desc"] = "Returnerer et jævnt fordelt tilfældigt tal større end eller lig med 0 og mindre end 1";
        this["Func_randbetween_Arg_0"] = "Bund";
        this["Func_randbetween_Arg_1"] = "Top";
        this["Func_randbetween_ArgDesc_0"] = "Den mindste værdi, der returneres.";
        this["Func_randbetween_ArgDesc_1"] = "Den maksimale værdi, der returneres.";
        this["Func_randbetween_Category"] = "Matematik";
        this["Func_randbetween_CategoryURL"] = "Matematik";
        this["Func_randbetween_Desc"] = "Genererer et pseudorandom-heltal mellem to angivne tal.";
        this["Func_rate_Arg_0"] = "NPer";
        this["Func_rate_Arg_1"] = "Pmt";
        this["Func_rate_Arg_2"] = "Pv";
        this["Func_rate_Arg_3"] = "Fv";
        this["Func_rate_Arg_4"] = "Type";
        this["Func_rate_Arg_5"] = "Guess";
        this["Func_rate_ArgDesc_0"] = "Det samlede antal betalingsperioder i en annuitet.";
        this["Func_rate_ArgDesc_1"] = "Betalingen foretages hver periode og kan ikke ændres i annuitetens løbetid.";
        this["Func_rate_ArgDesc_2"] = "Nuværdien eller det samlede beløb, som en række fremtidige betalinger er værd nu.";
        this["Func_rate_ArgDesc_3"] = "Den fremtidige værdi eller en kontantsaldo, du vil opnå, efter den sidste betaling er foretaget. Hvis fv udelades, antages den at være 0 (den fremtidige værdi af et lån er f.eks. 0). For eksempel, hvis du vil opspare $ 50.000 til at betale for et specielt projekt om 18 år, er $ 50.000 den fremtidige værdi. Derefter kan du lave et konservativt gæt på en rentesats og bestemme, hvor meget du skal spare hver måned. Hvis fv udelades, skal du medtage pmt-argumentet. Standardværdien for fv er nul.";
        this["Func_rate_ArgDesc_4"] = "Tallet nul eller et, der angiver, om betalinger forfalder ved periodens afslutning (nul, standardværdien) eller begyndelsen (et)";
        this["Func_rate_ArgDesc_5"] = "Et tal, som du gætter, er tæt på resultatet. Hvis det udelades, antages det at være 0,1 (10%).";
        this["Func_rate_Category"] = "Finansiel";
        this["Func_rate_CategoryURL"] = "Finansiel";
        this["Func_rate_Desc"] = "Returnerer renten pr. periode for en annuitet.";
        this["Func_replace_Arg_0"] = "Old_text";
        this["Func_replace_Arg_1"] = "Start_num";
        this["Func_replace_Arg_2"] = "Num_chars";
        this["Func_replace_Arg_3"] = "New_text";
        this["Func_replace_ArgDesc_0"] = "Tekstudtrykket, hvor nogle tegn skal erstattes.";
        this["Func_replace_ArgDesc_1"] = "Det 1-baserede indeks for tegnet i gammel_tekst, der skal erstattes med new_text.";
        this["Func_replace_ArgDesc_2"] = "Antallet af tegn i gammel tekst der skal erstattes med ny tekst";
        this["Func_replace_ArgDesc_3"] = "Teksten, der skal erstatte tegnene i gammel_tekst.";
        this["Func_replace_Category"] = "TekstOgData";
        this["Func_replace_CategoryURL"] = "Text_and_Data";
        this["Func_replace_Desc"] = "Erstatter en del af et tekstudtryk.";
        this["Func_rept_Arg_0"] = "Tekstværdi";
        this["Func_rept_Arg_1"] = "GentagTælling";
        this["Func_rept_ArgDesc_0"] = "Teksten, der skal gentages.";
        this["Func_rept_ArgDesc_1"] = "Det positive antal gange, teksten skal gentages.";
        this["Func_rept_Category"] = "TekstOgData";
        this["Func_rept_CategoryURL"] = "Text_and_Data";
        this["Func_rept_Desc"] = "Gentager tekst et bestemt antal gange.";
        this["Func_right_Arg_0"] = "tekst";
        this["Func_right_Arg_1"] = "num_chars";
        this["Func_right_ArgDesc_0"] = "Tekstudtrykket, der indeholder de tegn, der skal uddrages.";
        this["Func_right_ArgDesc_1"] = "Antallet af tegn, der skal uddrages.";
        this["Func_right_Category"] = "TekstOgData";
        this["Func_right_CategoryURL"] = "Text_and_Data";
        this["Func_right_Desc"] = "Returnerer det eller de sidste tegn i et tekstudtryk.";
        this["Func_roman_Arg_0"] = "Nummer";
        this["Func_roman_Arg_1"] = "Form";
        this["Func_roman_ArgDesc_0"] = "Tallet, der skal konverteres til et romertal.";
        this["Func_roman_ArgDesc_1"] = "Et tal eller en boolsk værdi, der angiver, om der skal bruges klassiske romertal eller en mere kortfattet version.\r\n0 = (standard) klassisk. (499 = '' CDXCIX '')\r\n1 = Mere kortfattet. (499 = '' LDVLIV '')\r\n2 = Mere kortfattet. (499 = '' XDIX '')\r\n3 = Mere kortfattet. (499 = '' VDIV '')\r\n4 = forenklet. (499 = '' ID '')\r\nSandt = Klassisk.\r\nFalsk = forenklet.\r\n    ";
        this["Func_roman_Category"] = "Matematik";
        this["Func_roman_CategoryURL"] = "Matematik";
        this["Func_roman_Desc"] = "Konverterer et tal til et romersk tal som et tekstudtryk.";
        this["Func_round_Arg_0"] = "tal";
        this["Func_round_Arg_1"] = "num_digits";
        this["Func_round_Arg_2"] = "round_Style";
        this["Func_round_ArgDesc_0"] = "Tallet, der skal afrundes.";
        this["Func_round_ArgDesc_1"] = "Antallet af cifre, som tallet skal afrundes til.";
        this["Func_round_ArgDesc_2"] = "Et tal, der angiver den afrundingsstil, der skal bruges.\r\n0 - Aritmetisk afrunding (standard) - Hvis det sidste signifikante ciffer i afrundingen er 5, afrundes tallet op.\r\n1 - Bankers afrunding - Når det sidste signifikante ciffer er et 5, bruges det foregående ciffer til at bestemme, om der skal afrundes op eller ned. Hvis cifret er lige, afrundes tallet nedad. Hvis det er ulige, afrundes tallet op.\r\n    ";
        this["Func_round_Category"] = "Matematik";
        this["Func_round_CategoryURL"] = "Matematik";
        this["Func_round_Desc"] = "Afrunder et tal til et specificeret antal cifre.";
        this["Func_rounddown_Arg_0"] = "Nummer";
        this["Func_rounddown_Arg_1"] = "Cifre";
        this["Func_rounddown_ArgDesc_0"] = "Tallet, der skal afrundes.";
        this["Func_rounddown_ArgDesc_1"] = "Det antal decimaler, der skal afrundes. Positive tal angiver pladser efter decimaltegnet, negative tal angiver pladser før decimaltegnet.";
        this["Func_rounddown_Category"] = "Matematik";
        this["Func_rounddown_CategoryURL"] = "Matematik";
        this["Func_rounddown_Desc"] = "Runder et tal ned til det angivne antal cifre.";
        this["Func_roundup_Arg_0"] = "Nummer";
        this["Func_roundup_Arg_1"] = "Cifre";
        this["Func_roundup_ArgDesc_0"] = "Tallet, der skal rundes op.";
        this["Func_roundup_ArgDesc_1"] = "Det antal decimaler, der skal afrundes. Positive tal angiver pladser efter decimaltegnet, negative tal angiver pladser før decimaltegnet.";
        this["Func_roundup_Category"] = "Matematik";
        this["Func_roundup_CategoryURL"] = "Matematik";
        this["Func_roundup_Desc"] = "Runder et tal op til det angivne antal cifre.";
        this["Func_search_Arg_0"] = "search_text";
        this["Func_search_Arg_1"] = "værdi";
        this["Func_search_Arg_2"] = "start_index";
        this["Func_search_ArgDesc_0"] = "Teksten, der skal findes i værdi.";
        this["Func_search_ArgDesc_1"] = "Den tekst hvor søgningstekst search_text.";
        this["Func_search_ArgDesc_2"] = "Det 1-baserede indeks, hvori søgningen skal startes (hvis udeladt er start_indeks 1).";
        this["Func_search_Category"] = "TekstOgData";
        this["Func_search_CategoryURL"] = "TekstOgData";
        this["Func_search_Desc"] = "Returnerer det 1-baserede indeks for en streng inden i en anden og søger ikke-versalfølsomhed.";
        this["Func_searchb_Arg_0"] = "search_text";
        this["Func_searchb_Arg_1"] = "værdi";
        this["Func_searchb_Arg_2"] = "start_index";
        this["Func_searchb_ArgDesc_0"] = "Teksten, der skal findes i værdi.";
        this["Func_searchb_ArgDesc_1"] = "Den tekst hvor søgningstekst search_text.";
        this["Func_searchb_ArgDesc_2"] = "Det 1-baserede indeks, hvori søgningen skal startes (hvis udeladt er start_indeks 1).";
        this["Func_searchb_Category"] = "TekstOgData";
        this["Func_searchb_CategoryURL"] = "TekstOgData";
        this["Func_searchb_Desc"] = "Returnerer det 1-baserede indeks for en streng inden i en anden og søger ikke-versalfølsomhed.";
        this["Func_second_Arg_0"] = "Time_value";
        this["Func_second_ArgDesc_0"] = "En tidsværdi, hvorfra sekundet skal findes.";
        this["Func_second_Category"] = "Dato og tid";
        this["Func_second_CategoryURL"] = "Date_and_Time";
        this["Func_second_Desc"] = "Returnerer sekundet for en tidsværdi.";
        this["Func_seriessum_Arg_0"] = "Inputværdi";
        this["Func_seriessum_Arg_1"] = "IndledendePotens";
        this["Func_seriessum_Arg_2"] = "Trin";
        this["Func_seriessum_Arg_3"] = "Koefficienter";
        this["Func_seriessum_ArgDesc_0"] = "Inputværdien til potensrækken.";
        this["Func_seriessum_ArgDesc_1"] = "Den indledende potens, som InputVærdien ophæves til.";
        this["Func_seriessum_ArgDesc_2"] = "Det trin, der vil blive brugt til at øge potensen for hver periode i rækken.";
        this["Func_seriessum_ArgDesc_3"] = "Et sæt koefficienter, som hver efterfølgende potens af InputVærdi ganges med.";
        this["Func_seriessum_Category"] = "Matematik";
        this["Func_seriessum_CategoryURL"] = "Matematik";
        this["Func_seriessum_Desc"] = "Returnerer summen af en potensrække.";
        this["Func_sign_Arg_0"] = "Nummer";
        this["Func_sign_ArgDesc_0"] = "Tallet, hvis tegn vil blive returneret.";
        this["Func_sign_Category"] = "Matematik";
        this["Func_sign_CategoryURL"] = "Matematik";
        this["Func_sign_Desc"] = "Returnerer tegnet på et tal (-1, 0 eller 1).";
        this["Func_sin_Arg_0"] = "tal";
        this["Func_sin_ArgDesc_0"] = "Vinklen i radianer, hvortil sinus skal findes.";
        this["Func_sin_Category"] = "Matematik";
        this["Func_sin_CategoryURL"] = "Matematik";
        this["Func_sin_Desc"] = "Returnerer sinus for den angivne vinkel.";
        this["Func_sinh_Arg_0"] = "tal";
        this["Func_sinh_ArgDesc_0"] = "Et reelt antal, hvortil den hyperbolske sinus skal findes.";
        this["Func_sinh_Category"] = "Matematik";
        this["Func_sinh_CategoryURL"] = "Matematik";
        this["Func_sinh_Desc"] = "Returnerer et tals hyperbolske sinus.";
        this["Func_sln_Arg_0"] = "omkostning";
        this["Func_sln_Arg_1"] = "bjærgning";
        this["Func_sln_Arg_2"] = "levetid";
        this["Func_sln_ArgDesc_0"] = "Aktivets startomkostning.";
        this["Func_sln_ArgDesc_1"] = "Værdien ved afskrivningens afslutning.";
        this["Func_sln_ArgDesc_2"] = "Antallet af perioder, hvor aktivet afskrives.";
        this["Func_sln_Category"] = "Finansiel";
        this["Func_sln_CategoryURL"] = "Finansiel";
        this["Func_sln_Desc"] = "Returnerer den lineære afskrivning af et aktiv i en periode.";
        this["Func_sqrt_Arg_0"] = "tal";
        this["Func_sqrt_ArgDesc_0"] = "Det tal, som kvadratroden skal findes til.";
        this["Func_sqrt_Category"] = "Matematik";
        this["Func_sqrt_CategoryURL"] = "Matematik";
        this["Func_sqrt_Desc"] = "Returnerer en positiv kvadratrod af et tal.";
        this["Func_sqrtpi_Arg_0"] = "Nummer";
        this["Func_sqrtpi_ArgDesc_0"] = "Det tal, der skal ganges med pi.";
        this["Func_sqrtpi_Category"] = "Matematik";
        this["Func_sqrtpi_CategoryURL"] = "Matematik";
        this["Func_sqrtpi_Desc"] = "Multipliserer det angivne tal med pi og returnerer kvadratroden af resultatet.";
        this["Func_stdev_Arg_0"] = "nummer {0}";
        this["Func_stdev_ArgDesc_0"] = "Et tal eller et talinterval, der angiver en stikprøve af en population.";
        this["Func_stdev_Category"] = "Statistisk";
        this["Func_stdev_CategoryURL"] = "Statistisk";
        this["Func_stdev_Desc"] = "Returnerer standardafvigelsen for en populationsprøve.";
        this["Func_subtotal_Arg_0"] = "FunktionsTal";
        this["Func_subtotal_Arg_1"] = "Ref{0}";
        this["Func_subtotal_ArgDesc_0"] = "Et tal, der repræsenterer den funktion, der skal beregnes. Føj 100 til værdien for at ignorere skjulte værdier. 1 (eller 101) - MIDDEL\r\n2 (eller 102) - TÆL\r\n4 (eller 104) - MAX\r\n5 (eller 105) - MIN\r\n6 (eller 106) - PRODUKT\r\n7 (eller 107) - STDAFV\r\n9 (eller 109) - SUM\r\n10 (eller 110) - VAR\r\n    ";
        this["Func_subtotal_ArgDesc_1"] = "En eller flere referencer, som subtotalen skal findes til.";
        this["Func_subtotal_Category"] = "Matematik";
        this["Func_subtotal_CategoryURL"] = "Matematik";
        this["Func_subtotal_Desc"] = "Beregner subtotalen i en af flere referencer.";
        this["Func_sum_Arg_0"] = "nummer {0}";
        this["Func_sum_ArgDesc_0"] = "Et tal, der skal tilføjes til de andre numre.";
        this["Func_sum_Category"] = "Matematik";
        this["Func_sum_CategoryURL"] = "Matematik";
        this["Func_sum_Desc"] = "Summerer alle de tal, der er angivet som argumenter, og returnerer summen.";
        this["Func_syd_Arg_0"] = "omkostning";
        this["Func_syd_Arg_1"] = "bjærgning";
        this["Func_syd_Arg_2"] = "levetid";
        this["Func_syd_Arg_3"] = "pr.";
        this["Func_syd_ArgDesc_0"] = "Aktivets startomkostning.";
        this["Func_syd_ArgDesc_1"] = "Værdien ved afskrivningens afslutning.";
        this["Func_syd_ArgDesc_2"] = "Antallet af perioder, hvor aktivet afskrives.";
        this["Func_syd_ArgDesc_3"] = "Perioden, der skal bruge de samme enheder som levetid.";
        this["Func_syd_Category"] = "Finansiel";
        this["Func_syd_CategoryURL"] = "Finansiel";
        this["Func_syd_Desc"] = "Returnerer årets cifre afskrivning på et aktiv i en bestemt periode.";
        this["Func_tan_Arg_0"] = "tal";
        this["Func_tan_ArgDesc_0"] = "Vinklen i radianer, hvortil tangenten skal findes.";
        this["Func_tan_Category"] = "Matematik";
        this["Func_tan_CategoryURL"] = "Matematik";
        this["Func_tan_Desc"] = "Returnerer tangenten for den angivne vinkel.";
        this["Func_tanh_Arg_0"] = "tal";
        this["Func_tanh_ArgDesc_0"] = "Vinklen i radianer, hvortil den hyperbolske tangens skal findes.";
        this["Func_tanh_Category"] = "Matematik";
        this["Func_tanh_CategoryURL"] = "Matematik";
        this["Func_tanh_Desc"] = "Returnerer den hyperbolske tangens for den angivne vinkel.";
        this["Func_text_Arg_0"] = "Værdi";
        this["Func_text_Arg_1"] = "FormatTekst";
        this["Func_text_ArgDesc_0"] = "Den numeriske værdi, der skal formateres.";
        this["Func_text_ArgDesc_1"] = "Talformatet hvormed værdien skal formateres.";
        this["Func_text_Category"] = "TekstOgData";
        this["Func_text_CategoryURL"] = "Text_and_Data";
        this["Func_text_Desc"] = "Konverterer værdien til tekst baseret på det angivne format.";
        this["Func_time_Arg_0"] = "time";
        this["Func_time_Arg_1"] = "minut";
        this["Func_time_Arg_2"] = "sekund";
        this["Func_time_ArgDesc_0"] = "Et tal, der repræsenterer tidens time.";
        this["Func_time_ArgDesc_1"] = "Et tal, der repræsenterer tidens minut.";
        this["Func_time_ArgDesc_2"] = "Et tal, der repræsenterer tidens sekund.";
        this["Func_time_Category"] = "Dato og tid";
        this["Func_time_CategoryURL"] = "Date_and_Time";
        this["Func_time_Desc"] = "Returnerer et tal, der repræsenterer en bestemt tidsværdi.";
        this["Func_timevalue_Arg_0"] = "time_text";
        this["Func_timevalue_ArgDesc_0"] = "Et tekstudtryk, der repræsenterer en tid.";
        this["Func_timevalue_Category"] = "Dato og tid";
        this["Func_timevalue_CategoryURL"] = "Date_and_Time";
        this["Func_timevalue_Desc"] = "Konverterer et tekstudtryk, der repræsenterer et tidspunkt til dets ækvivalente tidsværdi.";
        this["Func_today_Category"] = "Dato og tid";
        this["Func_today_CategoryURL"] = "Date_and_Time";
        this["Func_today_Desc"] = "Returnerer den aktuelle dato.";
        this["Func_trim_Arg_0"] = "tekst";
        this["Func_trim_ArgDesc_0"] = "Teksten, hvorfra ekstra mellemrum fjernes.";
        this["Func_trim_Category"] = "TekstOgData";
        this["Func_trim_CategoryURL"] = "Text_and_Data";
        this["Func_trim_Desc"] = "Fjerner alle mellemrum fra tekst undtagen enkelt mellemrum mellem ord.";
        this["Func_true_Category"] = "Logisk";
        this["Func_true_CategoryURL"] = "Logisk";
        this["Func_true_Desc"] = "Returnerer den logiske værdi sand.";
        this["Func_trunc_Arg_0"] = "tal";
        this["Func_trunc_Arg_1"] = "num_digits";
        this["Func_trunc_ArgDesc_0"] = "Tallet, der skal forkortes.";
        this["Func_trunc_ArgDesc_1"] = "Et tal, der specificerer trunkeringens nøjagtighed. Standardværdien er nul.";
        this["Func_trunc_Category"] = "Matematik";
        this["Func_trunc_CategoryURL"] = "Matematik";
        this["Func_trunc_Desc"] = "Afkorter et tal til et heltal ved at fjerne brøkdelen af tallet.";
        this["Func_type_Arg_0"] = "værdi";
        this["Func_type_ArgDesc_0"] = "Værdien, hvorfra typen skal findes.";
        this["Func_type_Category"] = "Information";
        this["Func_type_CategoryURL"] = "Information";
        this["Func_type_Desc"] = "Returnerer et tal, der repræsenterer typen af værdi: tal = 1; tekst = 2; logisk værdi (sand eller falsk) = 4; fejlværdi = 16; matrix = 64.";
        this["Func_upper_Arg_0"] = "tekst";
        this["Func_upper_ArgDesc_0"] = "Teksten, der skal konverteres til store bogstaver.";
        this["Func_upper_Category"] = "TekstOgData";
        this["Func_upper_CategoryURL"] = "Text_and_Data";
        this["Func_upper_Desc"] = "Konverterer alle små bogstaver i et tekstudtryk til store bogstaver.";
        this["Func_value_Arg_0"] = "tekst";
        this["Func_value_ArgDesc_0"] = "Tekstudtrykket, der skal konverteres.";
        this["Func_value_Category"] = "TekstOgData";
        this["Func_value_CategoryURL"] = "Text_and_Data";
        this["Func_value_Desc"] = "Konverterer et tekstudtryk, der repræsenterer et tal til et tal.";
        this["Func_var_Arg_0"] = "nummer {0}";
        this["Func_var_ArgDesc_0"] = "Et tal eller et talinterval, der angiver en stikprøve af en population.";
        this["Func_var_Category"] = "Statistisk";
        this["Func_var_CategoryURL"] = "Statistisk";
        this["Func_var_Desc"] = "Returnerer variansen fra en populationsprøve.";
        this["Func_weekday_Arg_0"] = "Dato";
        this["Func_weekday_Arg_1"] = "ReturnType";
        this["Func_weekday_ArgDesc_0"] = "Datoen, hvortil ugedagens nummer skal findes.";
        this["Func_weekday_ArgDesc_1"] = "Bestemmer, hvilket nummereringsskema der bruges til ugedagene.\r\n1 (standard) = søndag (1) til lørdag (7)\r\n2 = mandag (1) til søndag (7)\r\n3 = mandag (0) til søndag (6)\r\n    ";
        this["Func_weekday_Category"] = "Dato og tid";
        this["Func_weekday_CategoryURL"] = "Dato og tid";
        this["Func_weekday_Desc"] = "Returnerer et nummer, der repræsenterer ugedagen for den angivne dato.";
        this["Func_weeknum_Arg_0"] = "dato";
        this["Func_weeknum_Arg_1"] = "returType";
        this["Func_weeknum_ArgDesc_0"] = "Datoen, hvortil ugenummeret skal findes.";
        this["Func_weeknum_ArgDesc_1"] = "Bestemmer, hvilken dag der begynder ugen.\r\n1 (standard) = Søndag er den første dag i ugen.\r\n2 = mandag er den første dag i ugen.\r\n    ";
        this["Func_weeknum_Category"] = "Dato og tid";
        this["Func_weeknum_CategoryURL"] = "Dato og tid";
        this["Func_weeknum_Desc"] = "Returnerer ugenummeret, i hvilken den angivne dato falder.";
        this["Func_workday_Arg_0"] = "startDato";
        this["Func_workday_Arg_1"] = "dage";
        this["Func_workday_Arg_2"] = "helligdage";
        this["Func_workday_ArgDesc_0"] = "Datoen, hvorfra du starter.";
        this["Func_workday_ArgDesc_1"] = "Antallet af arbejdsdage, der skal forskydes fra startdatoen.";
        this["Func_workday_ArgDesc_2"] = "En række af en eller flere referencer, der indeholder datoer, der skal betragtes som ikke-arbejdsdage.";
        this["Func_workday_Category"] = "Dato og tid";
        this["Func_workday_CategoryURL"] = "Dato og tid";
        this["Func_workday_Desc"] = "Returnerer en arbejdsdag det angivne antal dage fra den angivne dato, ekskl. specificerede helligdage.";
        this["Func_year_Arg_0"] = "Date_value";
        this["Func_year_ArgDesc_0"] = "En datoværdi, hvorfra året skal findes.";
        this["Func_year_Category"] = "Dato og tid";
        this["Func_year_CategoryURL"] = "Date_and_Time";
        this["Func_year_Desc"] = "Returnerer året for en datoværdi.";
        this["Func_yearfrac_Arg_0"] = "StartDate";
        this["Func_yearfrac_Arg_1"] = "EndDate";
        this["Func_yearfrac_Arg_2"] = "Basis";
        this["Func_yearfrac_ArgDesc_0"] = "Startdato er startdatoen i intervallet.";
        this["Func_yearfrac_ArgDesc_1"] = "Slutdato er slutdatoen i intervallet.";
        this["Func_yearfrac_ArgDesc_2"] = "\r\n      Basis bestemmer grundlaget for den dagtælling, der skal bruges.\r\n0 (standard = USA (NASD) 30/360\r\n1 = Faktisk / faktisk\r\n2 = Faktisk / 360\r\n3 = Faktisk / 365\r\n4 = europæisk 30/360\r\n    ";
        this["Func_yearfrac_Category"] = "Dato og tid";
        this["Func_yearfrac_CategoryURL"] = "Dato og tid";
        this["Func_yearfrac_Desc"] = "Returnerer en brøkdel, der repræsenterer den del af et år, der er angivet mellem den angivne startdato og slutdato.";
        this["GenerateTableColumnName"] = "Kolonne";
        this["GenerateTableName"] = "Tabel";
        this["LD_Chart_AxisDisplayUnits"] = "Hundreder: Hundredtusinder: Tusindvis, TiTusindvis: x 10000, HundredTusindvis: x 100000, Millioner: Millioner, TiMillioner: x 10000000, HundredMillioner: x 100000000, Tusind millioner: Milliarder, MillionerMillioner: Billioner";
        this["LD_Fallback_ChartEx_Line1"] = "Dette diagram er ikke tilgængeligt i din version af Excel.";
        this["LD_Fallback_ChartEx_Line2"] = "Hvis du redigerer denne figur eller gemmer denne projektmappe i et andet filformat, ødelægges diagrammet permanent.";
        this["LE_AllMustHavGuidsIfAnyHasThem"] = "GenerateGuidsForGroups kan ikke indstilles til falsk, hvis en eksisterende gruppe har en Guid.";
        this["LE_ArgumentException_AnchorCellFromOtherWorksheet"] = "Den angivne celle findes ikke på det samme regneark som figuren.";
        this["LE_ArgumentException_ArrayFormulaMustHaveSingleRegion"] = "En matrixformel kan kun anvendes på et celleområde.";
        this["LE_ArgumentException_ArrayTooSmall"] = "Matrixen er ikke stort nok til at kopiere alle værdier til den.";
        this["LE_ArgumentException_CannotAddStandardTableStyle"] = "En standard tabeltype kan ikke føjes til CustomTableStyleCollection.";
        this["LE_ArgumentException_CannotApplyDVRuleToTotalCell"] = "En eller flere celler i den angivne reference er absolutte celler i en tabel, og datavalideringsregler kan derfor ikke anvendes derpå.";
        this["LE_ArgumentException_CannotCreateEmptyColorInfo"] = "Den tomme farve kan ikke bruges til at oprette et WorkbookColorInfo-eksempel.";
        this["LE_ArgumentException_CellMustBeOnSameWorksheetAsReferencesCollection"] = "Cellen skal være fra samme regneark som referencesamlingen.";
        this["LE_ArgumentException_CellShiftedOffWorksheet"] = "Cellen er blevet slettet eller flyttet fra regnearket og repræsenterer ikke længere en regnearkposition.";
        this["LE_ArgumentException_CellsInTableFromOtherWorksheet"] = "Den angivne region er ikke fra det samme regneark som datatabellen.";
        this["LE_ArgumentException_CellValueStringLength"] = "Strengværdier, der er tildelt en celle, må ikke have en længde, der er større end {0}.";
        this["LE_ArgumentException_ColumnFormulaAlreadyAppliedToCell"] = "Formlen anvendes allerede på en celle. Klon den for at bruge den som kolonneformel.";
        this["LE_ArgumentException_ColumnFromDifferentWorksheet"] = "Kolonnen hører ikke til det samme regneark som denne samling.";
        this["LE_ArgumentException_ColumnInputCellFromOtherWorksheet"] = "Den angivne kolonneinputscelle findes ikke på det samme regneark som datatabellen.";
        this["LE_ArgumentException_ColumnNotInTable"] = "Kolonnen er fra en anden tabel.";
        this["LE_ArgumentException_ColumnRemovedFromWorksheet"] = "Kolonnen er fjernet fra regnearket.";
        this["LE_ArgumentException_ColumnsToRepeatAtLeftOutsideRange"] = "Området ligger uden for intervallet af tilgængelige kolonner i regnearket.";
        this["LE_ArgumentException_CommentTooLong"] = "Kommentarlængden skal være mindre end 255 tegn.";
        this["LE_ArgumentException_CustomViewNameAlreadyExists"] = "En tilpasset visning med navnet \"{0}\" findes allerede.";
        this["LE_ArgumentException_DefaultTableStyleNotInWorkbook"] = "Den angivne tabeltype findes ikke i samlingerne CustomTableStyles eller StandardTableStyles.";
        this["LE_ArgumentException_DisplayTextTooLong"] = "Displayteksten for hyperlinket skal være mindre end 256 tegn.";
        this["LE_ArgumentException_DuplicateDisplayValue"] = "Den angivne visningsværdi findes allerede i samlingen.";
        this["LE_ArgumentException_DuplicateFixedDateGroup"] = "Den angivne FixedDateGroup er allerede i samlingen.";
        this["LE_ArgumentException_DuplicateTableStyle"] = "Den angivne type findes allerede i en CustomTableStyleCollection.";
        this["LE_ArgumentException_DuplicateTableStyleName_Existing"] = "Det angivne navn, der matcher navnet på en anden type i samlingen.";
        this["LE_ArgumentException_DuplicateTableStyleName_New"] = "Den angivne type har et navn, der matcher en anden type i samlingen.";
        this["LE_ArgumentException_DuplicateWorksheetProtectedRangeTitle"] = "Der findes allerede et interval med titlen '{0}'.";
        this["LE_ArgumentException_DV_ArrayFormulaCannotBeUsed"] = "En Matrixformel kan ikke anvendes som {0}.";
        this["LE_ArgumentException_DV_CellFromOtherWorksheet"] = "Cellen hører ikke til det samme regneark som datavalideringsregelsamlingen.";
        this["LE_ArgumentException_DV_ConstraintFormulaMustBeNumberOrReference"] = "{0} skal indstilles til et tal, et numerisk udtryk eller en reference til en enkelt celle.";
        this["LE_ArgumentException_DV_FormulaCannotBeNull"] = "{0} kan ikke indstilles til null, mens datavalideringsreglen anvendes på et regneark.";
        this["LE_ArgumentException_DV_FormulaCannotFindNamedReference"] = "{0} indeholder en navngivet reference, der ikke findes i projektmappen.";
        this["LE_ArgumentException_DV_FormulaCannotFindWorksheetReference"] = "{0} indeholder en reference på et regneark, der ikke findes i projektmappen.";
        this["LE_ArgumentException_DV_FormulaCannotReferenceOtherWorkbook"] = "{0} kan ikke indeholde en reference til en ekstern projektmappe.";
        this["LE_ArgumentException_DV_FormulaInvalidForWorkbookFormat"] = "{0} refererer til celler, der ikke er gyldige i {1} projektmappeformat.";
        this["LE_ArgumentException_DV_InvalidAddress"] = "Den angivne adresse er ikke en gyldig celle- eller regionadresse.";
        this["LE_ArgumentException_DV_InvalidExcelDate"] = "Den angivne dato kan ikke vises i Excel.";
        this["LE_ArgumentException_DV_InvalidFormula"] = "Ugyldig formel anvendt i datavalideringsreglen.";
        this["LE_ArgumentException_DV_InvalidMessageLength"] = "{0} må ikke være mere end {1} tegn.";
        this["LE_ArgumentException_DV_ListFormulaCannotHaveEmptyString"] = "Strengkonstanten i ValuesFormula kan ikke være en tom streng.";
        this["LE_ArgumentException_DV_ListFormulaMustBeStringOrReference"] = "ValuesFormula skal indstilles til en strengkonstant eller en reference til en enkelt celle eller region i den samme projektmappe.";
        this["LE_ArgumentException_DV_ListFormulaReferenceMustBeOneDimensional"] = "ValuesFormula indeholder en regionsreference, der har flere rækker og kolonner. Regionen skal indeholde en række eller en kolonne.";
        this["LE_ArgumentException_DV_LowerGreaterThanUpperContraint"] = "Værdien af den nederste begrænsning skal være mindre end eller lig med den øvre begrænsning.";
        this["LE_ArgumentException_DV_MustHaveOneAcceptedValue"] = "Mindst en værdi skal angives som en accepteret værdi.";
        this["LE_ArgumentException_DV_ReferencesAlreadyContainDVs"] = "De angivne referencer indeholder allerede en eller flere datavalideringsregler.";
        this["LE_ArgumentException_DV_ReferencesFromOtherWorksheet"] = "Samlingen af referencer hører ikke til det samme regneark som datavalideringsregelsamlingen.";
        this["LE_ArgumentException_DV_RegionFromOtherWorksheet"] = "Regionen tilhører ikke det samme regneark som datavalideringsregelsamlingen.";
        this["LE_ArgumentException_DV_RuleAppliedOnOtherWorksheet"] = "Datavalideringsreglen anvendes allerede på et andet regneark.";
        this["LE_ArgumentException_DV_UpperLessThanLowerContraint"] = "Værdien af den øvre begrænsning skal være større end eller lig med den nedre begrænsning.";
        this["LE_ArgumentException_EditRangeAlreadyOwned"] = "Intervallet er allerede i et regnearks AllowedEditRanges.";
        this["LE_ArgumentException_EditRangeHasRangeFromOtherWorksheet"] = "Redigeringsintervallet har et interval fra et andet regneark end AllowedEditRanges.";
        this["LE_ArgumentException_EmptyCustomList"] = "Der skal være mindst én streng på listen.";
        this["LE_ArgumentException_EndIndexLessThanZero"] = "Slutningen af intervallet skal være større end eller lig med nul.";
        this["LE_ArgumentException_FileDoesntContainsWorkbookStream"] = "Den angivne stream eller fil indeholder ikke en gyldig projektmappe.";
        this["LE_ArgumentException_FormulaReferencesInvalidCells"] = "Den angivne formel refererer til celler, som ikke understøttes i projektmappens aktuelle filformat.";
        this["LE_ArgumentException_HashSetFull"] = "Der er for mange emner i hashtagsættet.";
        this["LE_ArgumentException_HyperlinkAlreadyOwned"] = "Hyperlinket tilhører allerede en samling af hyperlinks.";
        this["LE_ArgumentException_HyperlinkSourceFromOtherWorksheet"] = "Hyperlinket har en kildecelle eller -region fra et andet regneark.";
        this["LE_ArgumentException_HyperlinkTargetFromOtherWorkbook"] = "Målet er fra en anden projektmappe end ejeren af hyperlinket.";
        this["LE_ArgumentException_InfiniteColumnWidth"] = "Kolonnebredden kan ikke være uendelig.";
        this["LE_ArgumentException_Interval"] = "Intervalparameter skal være en af værdierne i DateInterval-optælling";
        this["LE_ArgumentException_IntervalStr"] = "Intervalparameter kunne ikke konverteres til en af værdierne i DateInterval-optælling";
        this["LE_ArgumentException_InvalidCellAddress"] = "Den angivne adresse er ikke en gyldig celleadresse.";
        this["LE_ArgumentException_InvalidCommentPositioningMode"] = "Kun DontMoveOrSizeWithCells og MoveAndSizeWithCells ShapePositioningModes kan anvendes på en kommentarfigur.";
        this["LE_ArgumentException_InvalidCustomFilterOperandNumber"] = "Værdien kan ikke være uendelig eller NaN.";
        this["LE_ArgumentException_InvalidCustomFilterOperator"] = "Når sammenligningsværdien ikke er en streng, kan sammenligningsoperatoren ikke være BeginsWith, DoesNotBeginWith, Contains, DoesNotContain, EndsWith eller DoesNotEndWith.";
        this["LE_ArgumentException_InvalidDatePeriodFilterValue"] = "Værdien er ikke inden for det tilladte interval. For et månedsfilter skal den være 1-12. For et kvartalsfilter skal den være 1-4.";
        this["LE_ArgumentException_InvalidDpi"] = "Den angivne Dpi-værdi er ugyldig.";
        this["LE_ArgumentException_InvalidExcelDate"] = "Værdien kan ikke udtrykkes som en dato i Excel.";
        this["LE_ArgumentException_InvalidFileFormat"] = "Ugyldigt eller ukendt filformat.";
        this["LE_ArgumentException_InvalidFormula"] = "Den angivne formel var ugyldig.";
        this["LE_ArgumentException_InvalidGradientStopColor"] = "Automatiske farver og systemfarver kan ikke bruges i gradientstop.";
        this["LE_ArgumentException_InvalidNamedReferenceName"] = "Navnet er ikke et gyldigt navngivet referencenavn.";
        this["LE_ArgumentException_InvalidPaletteColor_EmptyOrSystem"] = "Tomme eller systemfarver kan ikke føjes til paletten.";
        this["LE_ArgumentException_InvalidPaletteColor_NonOpaque"] = "Kun uigennemsigtige farver kan tilføjes til paletten.";
        this["LE_ArgumentException_InvalidReferencesString"] = "Referenceværdien er ikke korrekt formateret.";
        this["LE_ArgumentException_InvalidRegionAddress"] = "Den angivne adresse er ikke en gyldig regionsadresse.";
        this["LE_ArgumentException_InvalidRowOrColumnRange"] = "Ugyldigt række- eller kolonneointerval.";
        this["LE_ArgumentException_InvalidSeriesValues"] = "Ugyldige serieværdier. Der skal være mindst en værdi, og hver værdi skal enten være en streng eller et objekt, der kan konverteres til en dobbelt.";
        this["LE_ArgumentException_InvalidTarget"] = "Hyperlinkmålet skal være en streng, WorksheetCell, WorksheetRegion eller NamedReference.";
        this["LE_ArgumentException_InvalidTopOrBottomFilterValue"] = "Værdien skal være mellem 1 og 500.";
        this["LE_ArgumentException_InvalidWorksheetName"] = "Regnearkets navn kan ikke indeholde et af følgende tegn: : \\ /? * []";
        this["LE_ArgumentException_InvalidWorksheetNameStartingQuote"] = "Regnearkets navn kan ikke starte eller slutte med et enkelt anførselstegn.";
        this["LE_ArgumentException_LessThanTwoGradientStops"] = "Der skal være angivet mindst to gradientstop.";
        this["LE_ArgumentException_NamedReferenceNameTooLong"] = "Navnet på den navngivne reference kan ikke være mere end 255 tegn.";
        this["LE_ArgumentException_NaNDefaultColumnWidth"] = "NaN kan ikke bruges som standardkolonnebredde.";
        this["LE_ArgumentException_NewRegionOverlapsFilterArea"] = "Handlingen forsøger at ændre et filtreret område på dit regneark og kan ikke udføres. For at fuldføre denne handling skal  AutoFilters i arket fjernes.";
        this["LE_ArgumentException_NewTableRegionCannotMoveHeaders"] = "Den nye tabelregion kan ikke flytte overskrifterne til en ny række.";
        this["LE_ArgumentException_NewTableRegionFromWrongWorksheet"] = "Den angivne region ikke fra det samme regneark som tabellen.";
        this["LE_ArgumentException_NewTableRegionMustHaveDataRows"] = "Den nye tabelregion skal indeholde mindst en datarække.";
        this["LE_ArgumentException_NewTableRegionOverlapOld"] = "Den nye tabelregion skal overlappe den forrige tabelregion.";
        this["LE_ArgumentException_NewTableRegionOverlapsBlockingValue"] = "Tabellen kan ikke ændres til den angivne region, fordi den indeholder en multicelle matrixformel eller datatabel.";
        this["LE_ArgumentException_NewTableRegionOverlapsMergedRegion"] = "Tabellen kan ikke ændres til den angivne region, fordi den inkluderer en flettet celle.";
        this["LE_ArgumentException_NewTableRegionOverlapsOtherTable"] = "Tabellen kan ikke ændres til den angivne region, fordi denne indeholder en anden tabel.";
        this["LE_ArgumentException_NoRegionsInArray"] = "Den angivne matrix har ingen regioner.";
        this["LE_ArgumentException_NotEnum"] = "Type '{0}' er ikke enum";
        this["LE_ArgumentException_NotSupportedCodePage"] = "Ikke understøttet kodeside";
        this["LE_ArgumentException_NPer"] = "nPer kunne ikke være 0";
        this["LE_ArgumentException_NPerFunction"] = "Forkert kombination af parametre til NPer-funktion";
        this["LE_ArgumentException_PA_PrintAreasCannotOverlap"] = "Det angivne udskriftsområde overlapper med et af de eksisterende udskriftsområder.";
        this["LE_ArgumentException_ParentStyleFromOtherWorkbook"] = "Den angivne type er ikke fra den samme projektmappe som celleformatet.";
        this["LE_ArgumentException_PB_CantInsertBreakAtIndex"] = "Denne samling understøtter ikke indsættelse af elementer i et indeks.";
        this["LE_ArgumentException_PB_CantSetBreakAtIndex"] = "Denne samling understøtter ikke indstilling af elementer i et indeks.";
        this["LE_ArgumentException_PB_MustBeWithinPrintArea"] = "Sideskiftet skal være inden for udskriftsområdet.";
        this["LE_ArgumentException_PB_PageBreakCantBeA1Cell"] = "Kan ikke indsætte et sideskift øverst til venstre på regnearket.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeLeftColumn"] = "Kan ikke indsætte et sideskift til venstre for regnearket.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeTopRow"] = "Kan ikke indsætte et sideskift øverst på regnearket.";
        this["LE_ArgumentException_PB_PageBreaksCannotAddExistingInstance"] = "Den angivne forekomst findes allerede i samlingen.";
        this["LE_ArgumentException_PB_PageBreaksCannotOverlap"] = "Sideskiftet kan ikke overlappe med andre sideskift på regnearket.";
        this["LE_ArgumentException_PB_PrintAreaMustBeInPrintAreas"] = "Sideskiftet er i et udskriftsområde, der ikke er i PrintAreas-samlingen.";
        this["LE_ArgumentException_PB_PrintAreaMustBeOnSameWorksheet"] = "Sideskiftet er i et udskriftsområde, der ikke er på det samme regneark som sideskiftsamlingen.";
        this["LE_ArgumentException_Pmt"] = "Pmt kunne ikke være 0";
        this["LE_ArgumentException_PrintAreaShiftedOffWorksheet"] = "Sideskiftet er i et udskriftsområde, der er forskudt fra regnearket og ikke længere repræsenterer et gyldigt celleområde.";
        this["LE_ArgumentException_RangeFromOtherWorksheetThanProtectedRange"] = "Det angivne interval er fra et andet regneark end redigeringsintervallet.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheet"] = "Regionen skal være fra samme regneark.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheetAsReferencesCollection"] = "Regionen skal være fra det samme regneark som referencesamlingen.";
        this["LE_ArgumentException_RegionsFromMixedWorksheets"] = "Alle regioner, der er specificeret i matrixen, skal være fra det samme regneark.";
        this["LE_ArgumentException_RegionShiftedOffWorksheet"] = "Regionen er flyttet fra regnearket og repræsenterer ikke længere et gyldigt celleområde.";
        this["LE_ArgumentException_RegionsShiftedOffWorksheet"] = "En eller flere angivne regioner er blevet flyttet fra regnearket og repræsenterer ikke længere et gyldigt celleområde.";
        this["LE_ArgumentException_RowFromDifferentWorksheet"] = "Ræken hører ikke til det samme regneark som denne samling.";
        this["LE_ArgumentException_RowInputCellFromOtherWorksheet"] = "Den angivne rækkeinputcelle findes ikke på det samme regneark som datatabellen.";
        this["LE_ArgumentException_RowRemovedFromWorksheet"] = "Rækken er fjernet fra regnearket.";
        this["LE_ArgumentException_SelectedWorksheetFromOtherWorkbook"] = "Regnearket hører ikke til den projektmappe, der er knyttet til disse vinduesindstillinger.";
        this["LE_ArgumentException_ShapeCannotBeAdded"] = "Den angivne figur kan ikke føjes til en figursamling.";
        this["LE_ArgumentException_SourceFromOtherWorksheetThanHyperlink"] = "Den angivne kildecelle eller -region er fra et andet regneark end hyperlinket.";
        this["LE_ArgumentException_StartIndexLessThanZero"] = "Starten af intervallet skal være større end eller lig med nul.";
        this["LE_ArgumentException_StyleNameAlreadyExists"] = "En stil med navnet \"{0}\" findes allerede i projektmappen.";
        this["LE_ArgumentException_StyleNameTooLong"] = "Typenavnet kan ikke være mere end 255 tegn.";
        this["LE_ArgumentException_TableColumnRemovedFromTable"] = "Tabelsøjlen er fjernet fra tabellen.";
        this["LE_ArgumentException_TableStyleFromOtherWorkbook"] = "Den angivne WorksheetTableStyle hører ikke til den samme projektmappe som tabellen.";
        this["LE_ArgumentException_TargetTooLong"] = "Adressen til hyperlinkmål skal være mindre end 256 tegn.";
        this["LE_ArgumentException_TextLengthGreaterThanMax"] = "Længden af teksten i en figur eller kommentar må ikke være større end {0}.";
        this["LE_ArgumentException_ToolTipTooLong"] = "Tip til hyperlinkværktøj skal være mindre end 256 tegn.";
        this["LE_ArgumentException_TopLeftWindowBounds"] = "Venstre og øverste del af vinduesgrænserne skal være mellem -32768 og 32767.";
        this["LE_ArgumentException_WeekdayFirst"] = "WeekdayFirst-parameteren skal være mellem FirstDayOfWeek.System og FirstDayOfWeek.Saturday";
        this["LE_ArgumentException_WidthHeightWindowBounds"] = "Vinduesgrænsernes bredde og højde skal være mellem 0 og 65535.";
        this["LE_ArgumentException_WorksheetNameAlreadyExists"] = "Et regneark med navnet \"{0}\" findes allerede i projektmappen.";
        this["LE_ArgumentException_WorksheetNameTooLong"] = "Regnearkets navn skal have en længde på mellem 1 og 31 tegn.";
        this["LE_ArgumentException_WorksheetProtectedRangeTitleTooLong"] = "Titlen på et redigeringsinterval må ikke være mere end 255 tegn.";
        this["LE_ArgumentException_WorksheetScopeFromOtherWorkbook"] = "Regnearket hører ikke til den samme projektmappe som denne samling.";
        this["LE_ArgumentNullException_AnchorCell"] = "Cellen kan ikke være null, når figuren findes på et regneark eller i en gruppe.";
        this["LE_ArgumentNullException_CustomViewName"] = "Navnet på den tilpassede visning kan ikke være null eller tomt.";
        this["LE_ArgumentNullException_Encoder"] = "Billedkoderen kan ikke være null.";
        this["LE_ArgumentNullException_FindNamedReference"] = "Navnet, der skal findes, kan ikke være null eller tomt.";
        this["LE_ArgumentNullException_FormulaCantBeNull"] = "Formlen for en navngivet reference kan ikke være null eller tom.";
        this["LE_ArgumentNullException_HiddenColumn"] = "En nulkolonne kan ikke føjes til samlingen.";
        this["LE_ArgumentNullException_HiddenRow"] = "En nullrække kan ikke føjes til samlingen.";
        this["LE_ArgumentNullException_Image"] = "Billedet kan ikke være null.";
        this["LE_ArgumentNullException_NamedReferenceNameCantBeNull"] = "Navnet på en navngivet reference kan ikke være null eller tom.";
        this["LE_ArgumentNullException_SaveStream"] = "Projektmappen kan ikke gemmes i en null-stream.";
        this["LE_ArgumentNullException_SelectedWorksheet"] = "Projektmappen skal have et gyldigt, markeret regneark.";
        this["LE_ArgumentNullException_Shape"] = "En null-figur kan ikke føjes til samlingen.";
        this["LE_ArgumentNullException_SourceFont"] = "Kildeskrifttypen kan ikke være null.";
        this["LE_ArgumentNullException_SourceFormatting"] = "Kildeformateringen kan ikke være null.";
        this["LE_ArgumentNullException_StyleName"] = "Stilnavnet kan ikke være null eller tomt.";
        this["LE_ArgumentNullException_UnformattedString"] = "Den uformaterede streng kan ikke være null.";
        this["LE_ArgumentNullException_Workbook"] = "Projektmappen kan ikke være null.";
        this["LE_ArgumentNullException_WorkbookRequiredToResolveThemeColor"] = "Der skal angives en projektmappe for at beslutte en temafarve.";
        this["LE_ArgumentNullException_WorksheetName"] = "Regnearkets navn kan ikke være null eller tomt.";
        this["LE_ArgumentNullException_WorksheetProtectedRangeTitleCantBeNull"] = "Titlen på et redigeringsområde kan ikke være null eller tom.";
        this["LE_ArgumentNullException_WorksheetScope"] = "Regnearkets omfang kan ikke være nul.";
        this["LE_ArgumentOutOfRange_RelativeColumnIndex"] = "Det relative kolonneindeks er uden for den angivne region.";
        this["LE_ArgumentOutOfRange_RelativeRowIndex"] = "Det relative rækkeindeks er uden for den angivne region.";
        this["LE_ArgumentOutOfRangeException_AnchorPosition"] = "En af procenterne på det angivne sted er mindre end 0.";
        this["LE_ArgumentOutOfRangeException_BadInsertIndex"] = "Det angivne indeks skal være større end eller lig med nul og mindre end eller lig med antallet af eksisterende afsnit.";
        this["LE_ArgumentOutOfRangeException_BadRemoveAtIndex"] = "Indekset skal være større end nul og mindre end antallet af afsnit i den formaterede streng.";
        this["LE_ArgumentOutOfRangeException_CollectionIndex"] = "Indeks var uden for området. Det skal være ikke-negativt og mindre end størrelsen på samlingen.";
        this["LE_ArgumentOutOfRangeException_ColumnWidth"] = "Kolonnebredden skal være mellem 0 og 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultColumnWidth"] = "Standardkolonnebredden skal være mellem 0 og 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultFontHeight"] = "Standardskrifthøjden skal være mellem 20 og 8180.";
        this["LE_ArgumentOutOfRangeException_DefaultRowHeight"] = "Standard rækkehøjde skal være mellem -1 og 8192.";
        this["LE_ArgumentOutOfRangeException_DuplicateItemSorted"] = "Der er allerede anvendt en sorteringsbetingelse på det angivne element.";
        this["LE_ArgumentOutOfRangeException_FirstVisibleTabIndex"] = "Det første synlige faneindeks skal være større end eller lig med nul.";
        this["LE_ArgumentOutOfRangeException_FontHeight"] = "Skrifttypens højde skal være mellem 20 og 8180.";
        this["LE_ArgumentOutOfRangeException_GroupAddedToSelf"] = "En gruppe kan ikke føjes til sin egen samling af figurer.";
        this["LE_ArgumentOutOfRangeException_Indent"] = "Indrykningsniveauet skal være mellem 0 og 250.";
        this["LE_ArgumentOutOfRangeException_IndexNegative"] = "Indekset skal være ikke-negativt.";
        this["LE_ArgumentOutOfRangeException_InvalidCollectionIndex"] = "Det angivne indeks er ikke gyldigt i samlingen.";
        this["LE_ArgumentOutOfRangeException_InvalidColorInfoTint"] = "Farvetonen skal være mellem -1,0 og 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnCount"] = "{0} er ikke et gyldigt antal kolonner i det aktuelle format. Værdien skal være mellem {1} og {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnIndex"] = "{0} er ikke et gyldigt kolonneindeks i det aktuelle format. Værdien skal være mellem {1} og {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidGradientStopOffset"] = "Gradientstopforskydningen skal være mellem 0,0 og 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRelativeRectangleValueForGradient"] = "Værdierne for det indre rektangel i den rektangulære gradient skal være mellem 0,0 og 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRemoveAtIndex_SortConditions"] = "Det indeks, hvorfra sorteringsbetingelsen skal fjernes, er uden for intervallet.";
        this["LE_ArgumentOutOfRangeException_InvalidRowCount"] = "{0} er ikke et gyldigt rækkeantal i det aktuelle format. Værdien skal være mellem {1} og {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidRowIndex"] = "{0} er ikke et gyldigt rækkeindeks i det aktuelle format. Værdien skal være mellem {1} og {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidTableStyleAreaStripeExtent"] = "{0} skal være mellem 1 og 9 inklusive.";
        this["LE_ArgumentOutOfRangeException_LeftPaneWidth"] = "Den angivne rudebredde ligger uden for det gyldige interval på {0} og {1}.";
        this["LE_ArgumentOutOfRangeException_LengthMustBePositive"] = "Længden skal være større end eller lig med en.";
        this["LE_ArgumentOutOfRangeException_MagnificationLevel"] = "Visningsforstørrelsesniveauer skal være mellem 10 og 400.";
        this["LE_ArgumentOutOfRangeException_Margins"] = "Margenerne skal være mellem 0 og 100.";
        this["LE_ArgumentOutOfRangeException_MaxPagesHorizontally"] = "Det maksimale antal sider vandret skal være mellem 0 og 32767.";
        this["LE_ArgumentOutOfRangeException_MaxPagesVertically"] = "Det maksimale antal sider lodret skal være mellem 0 og 32767.";
        this["LE_ArgumentOutOfRangeException_MaxRecursionIterations"] = "Antallet af rekursionsgentagelser skal maksimalt være mellem 1 og 32767.";
        this["LE_ArgumentOutOfRangeException_MaxSortConditions"] = "Der kan ikke være mere end {0} sorteringsbetingelser i samlingen.";
        this["LE_ArgumentOutOfRangeException_NegativeStartIndex"] = "Startindekset kan ikke være negativt.";
        this["LE_ArgumentOutOfRangeException_NumberOfCopies"] = "Antallet af kopier skal være mellem 1 og 65535.";
        this["LE_ArgumentOutOfRangeException_Per"] = "pr. værdi skal være mellem 0 og nPer + 1";
        this["LE_ArgumentOutOfRangeException_Rate"] = "Satsen skal være større end -1";
        this["LE_ArgumentOutOfRangeException_Resolution"] = "Opløsningen skal være mellem 0 og 65535.";
        this["LE_ArgumentOutOfRangeException_RowHeight"] = "Rækkehøjden skal være mellem 0 og 8192.";
        this["LE_ArgumentOutOfRangeException_ScalingFactor"] = "Skaleringsfaktoren skal være mellem 10 og 400.";
        this["LE_ArgumentOutOfRangeException_StartPageNumber"] = "Startsidens nummer skal være mellem -32765 og 32767.";
        this["LE_ArgumentOutOfRangeException_TabBarWidth"] = "Fanebladets bredde skal være mellem 0 og 1000.";
        this["LE_ArgumentOutOfRangeException_TopPaneHeight"] = "Den angivne rudehøjde ligger uden for det gyldige interval på {0} og {1}.";
        this["LE_AutoColorNotAllowed"] = "Farven kan ikke være 'Auto'.";
        this["LE_Axis_MajorMinorUnit"] = "Værdien af majorUnit-parameteren skal være større end værdien for minorUnit-parameteren.";
        this["LE_Axis_NoCrossAxis"] = "Hvis CrossAxis-egenskaben er indstillet på et akse-objekt, skal den indstilles på dem alle. Hvis dit diagram kun har to akser, kan krydsaksen bestemmes automatisk, men kun hvis CrossAxis- egenskaben ikke er indstillet på alle Axis-objekter.";
        this["LE_Biff8SerializerNotLoaded"] = "Formatet '{0}' kræver en serieomkoder, der ikke er indlæst, i JavaScript er dette indeholdt i filen infragistics.excel_serialization_biff8.js";
        this["LE_CategoryAxisBinning_BinWidthAndNumberOfBins"] = "BinWidth'- og 'numberOfBins'-parametrene er gensidigt uafhængige. Begge kan være uspecificerede, men hvis den ene er specificeret, behøver den anden ikke være det.";
        this["LE_ChartGradientFill_EmptyStops"] = "Skal indeholde mindst et gradientstop.";
        this["LE_ChartObject_DifferentChart"] = "Den objektforekomst, der tildeles, er knyttet til et andet diagram og kan ikke tildeles denne forekomst.";
        this["LE_ColumnsToRepeatAtLeftAreOutsideAvailableRange"] = "Udskriftsindstillingerne har kolonner, der gentages til venstre, som ligger uden for intervallet af tilgængelige kolonner i det nye format.";
        this["LE_ComboChartGroupCollection_DoesNotExist"] = "Den angivne ComboChartGroup findes ikke i denne samling.";
        this["LE_FillPercentInvalid"] = "Værdien {0} er ikke en gyldig værdi for egenskaben {1}.";
        this["LE_FormatLimitError_Indent"] = "Indrykket overstiger den maksimale værdi på {0}, der er angivet i det aktuelle format.";
        this["LE_FormatLimitError_MacroWorkbook"] = "Projektmappen har VBA-kode og kan ikke gemmes som en makroaktiveret projektmappe.";
        this["LE_FormatLimitError_MaxColumnIndex"] = "Kolonneindekset på {0} er større end det maksimale {1}, der understøttes af det angivne format.";
        this["LE_FormatLimitError_MaxRowIndex"] = "Rækkeindekset på {0} overstiger det maksimale {1}, der understøttes af dette format.";
        this["LE_FormulaParseException_ArrayContainsConstants"] = "En matrix kan kun indeholde konstante udtryk.";
        this["LE_FormulaParseException_ArrayHasEmptyFirstRow"] = "En konstant matrix i formlen har ingen elementer i den første række.";
        this["LE_FormulaParseException_ArrayHasMisalignedRows"] = "En konstant matrix i formlen har forkert justerede rækker. Hver række i en matrix skal have det samme antal kolonner.";
        this["LE_FormulaParseException_CellReferenceAfterWorkbookName"] = "En ekstern projektmappereference skal følges af en navngivet reference og ikke en cellehenvisning.";
        this["LE_FormulaParseException_ExtraExpressions"] = "Formlen har ekstra udtryk efter den er slut.";
        this["LE_FormulaParseException_FunctionMissingClosingParen"] = "En funktion i formlen har ingen afsluttende parentes.";
        this["LE_FormulaParseException_FunctionNestingTooDeep"] = "En funktion i formlen er indlejret for dybt i andre funktioner.";
        this["LE_FormulaParseException_IncorrectNumberOfArguments"] = "En funktion i formlen har ikke fået det korrekte antal argumenter.";
        this["LE_FormulaParseException_InvalidArguments"] = "En operatør eller funktion i formlen har ugyldige argumenter.";
        this["LE_FormulaParseException_InvalidConditionalFormatFormula"] = "En formel med betinget format kan ikke bruge referenceoperatorer som grupperinger eller fællesmængder, matrixkonstanter eller strukturerede tabelhenvisninger.";
        this["LE_FormulaParseException_InvalidErrorValue"] = "Den angivne fejltype er ikke gyldig. Det skal være en af følgende værdier: '#NULL!', '# DIV/0!', '#VALUE!', '#REF!', '#NAME?', '#NUM!', Or '#N/A'.";
        this["LE_FormulaParseException_InvalidFileNameInBrackets"] = "Et filnavn mellem firkantede parenteser var ugyldigt.";
        this["LE_FormulaParseException_InvalidStructuredTableReference"] = "Den strukturerede tabelreference er forkert.";
        this["LE_FormulaParseException_InvalidWorkbookName"] = "Projektmappereferencen i anførselstegn er ikke et gyldigt navn på projektmappens fil.";
        this["LE_FormulaParseException_InvalidWorksheetName"] = "Regnearkets navn er ugyldigt.";
        this["LE_FormulaParseException_MacroFunctionNotAllowed"] = "Funktionerne '{0}' er kun gyldige i en navngivet reference.";
        this["LE_FormulaParseException_Message_PortionWithError"] = "Del med fejl: {0}";
        this["LE_FormulaParseException_MissingArgumentAfterBinary"] = "En binær operator i formlen mangler en operand efter den.";
        this["LE_FormulaParseException_MissingArgumentAfterParen"] = "Der er intet udtryk efter en åben parentes.";
        this["LE_FormulaParseException_MissingArgumentAfterUnary"] = "En unær operator i formlen blev ikke efterfulgt af et gyldigt udtryk.";
        this["LE_FormulaParseException_MissingArgumentBeforeBinary"] = "En binær operator i formlen mangler en operand før den.";
        this["LE_FormulaParseException_NamedReferenceRefsNeedSheetName"] = "Alle cellereferencer i en navngivet referenceformel skal identificere regnearket.";
        this["LE_FormulaParseException_NoElementAfterArraySerapator"] = "Der er intet matrix-element efter en element-afgrænser i formlen.";
        this["LE_FormulaParseException_NoEqualsSign"] = "Formlen skal starte med et lighedstegn.";
        this["LE_FormulaParseException_NoExclamationAfterWorkbookName"] = "Projektmappereferencen skal efterfølges af et udråbstegn.";
        this["LE_FormulaParseException_NoExclamationAfterWorksheetName"] = "Regnearksreferencen skal efterfølges af et udråbstegn.";
        this["LE_FormulaParseException_NoExpressions"] = "Formlen har ingen gyldige udtryk efter lighedstegnet.";
        this["LE_FormulaParseException_NoFileNameAfterBracket"] = "Der kræves et gyldigt filnavn efter den åbne, firkantede parentes.";
        this["LE_FormulaParseException_NoValidTermAfterWorkbookName"] = "En ekstern projektmappereference skal følges af en navngivet reference.";
        this["LE_FormulaParseException_NoValidTermAfterWorksheetName"] = "En regnearkreference skal efterfølges af en celle eller navngivet reference.";
        this["LE_FormulaParseException_NoWorksheetAfterWorkbookName"] = "En projektmappes filnavn i firkantede parenteser skal efterfølges af et regnearknavn.";
        this["LE_FormulaParseException_StringConstantLengthTooLong"] = "Den angivne formel må ikke indeholde en streng, der er længere end {0} tegn.";
        this["LE_FormulaParseException_TooLong"] = "Den maksimale længde af en formel kan ikke overstige {0} for det angivne projektmappeformat.";
        this["LE_FormulaParseException_UnknownFunction"] = "En funktion i formlen har ikke et gyldigt funktionsnavn.";
        this["LE_FormulaParseException_UnmatchedOpenBracket"] = "En konstant matrix i formlen har ikke en lukkende parentes.";
        this["LE_FormulaParseException_UnmatchedOpenParen"] = "En åbnende parentes i formlen har ikke en matchende lukkende parentes.";
        this["LE_FormulaParseException_UnmatchedOpenSquareBracket"] = "Der kræves en lukkende, firkantet parentes efter projektmappens navn.";
        this["LE_FormulaParseException_WorkbookNameMissingEndQuote"] = "Arbejdsmappereferencen mangler afsluttende anførselstegn.";
        this["LE_FormulaParseException_WorksheetRangeMissingEndingName"] = "Regnearkområdet specificerer ikke et andet regnearknavn.";
        this["LE_GradientStop_InvalidPosition"] = "Skal angive et tal mellem 0 og 1.";
        this["LE_IconSetInvalid"] = "'IconSetNoIcon' er ikke en gyldig indstilling for egenskaben IconSet.";
        this["LE_IndexOutOfRangeException_ArrayBounds"] = "Indeks var uden for matrixens grænser.";
        this["LE_InvalidEnumArgumentException_DefaultPatternCannotBeUsed"] = "Standard FillPatternStyle kan ikke bruges til at oprette et CellFillPattern.";
        this["LE_InvalidLineWeight"] = "LineWeight må ikke være mindre end 0 eller større end 1584.";
        this["LE_InvalidOperationException_AnchorCommentBeforeApplyingToCell"] = "Kommentaren skal have både TopLeftCornerCell og BottomRightCornerCell indstillet, eller de skal begge være null, før den anvendes på en celle.";
        this["LE_InvalidOperationException_AnchorShapeBeforeAddingToCollection"] = "Figuren skal have både TopLeftCornerCell og BottomRightCornerCell indstillet, før den føjes til et regneark eller en gruppe.";
        this["LE_InvalidOperationException_ArrayFormulaAlreadyApplied"] = "Matrixformlen er allerede anvendt på et celleområde. Ryd det eksisterende celleområde først.";
        this["LE_InvalidOperationException_ArrayFormulaAppliedInTable"] = "Matrixformler kan ikke anvendes på et område, hvor der findes en tabel.";
        this["LE_InvalidOperationException_ArrayFormulaInMergedCell"] = "Matrixformler er ikke gyldige i flettede celler.";
        this["LE_InvalidOperationException_BottomRightAnchorFromOtherWorksheet"] = "Når figuren tilføjes denne samling placeres den på et andet regneark end dens BottomRightCornerCell.";
        this["LE_InvalidOperationException_BuiltInStyleNameCannotBeChanged"] = "Navnet på en indbygget type kan ikke ændres.";
        this["LE_InvalidOperationException_CannotAddColorsToPaletteDirectly"] = "Farver kan ikke føjes til WorkbookColorPalette. Brug indeksøren til at indstille en farve til et bestemt indeks.";
        this["LE_InvalidOperationException_CannotAddParagraphDirectly"] = "Afsnit kan ikke tilføjes direkte.";
        this["LE_InvalidOperationException_CannotAddTableDirectly"] = "WorksheetTable-forekomster kan ikke indsættes direkte i samlingen. Brug metoden Tilføj.";
        this["LE_InvalidOperationException_CannotAddTableToRemovedWorksheet"] = "Tabeller kan ikke føjes til et regneark, der er blevet fjernet fra sin ejerprojektmappe.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileHeaderRowIsHidden"] = "Et filter kan ikke anvendes, mens sidehovedtabelrækken er skjult.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileUIIsHidden"] = "Et filter kan ikke anvendes, mens filterets UI er skjult.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfData"] = "{0} kan ikke indsættes, fordi det vil flytte data fra bunden af regnearket.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfObject"] = "{0} kan ikke indsættes, fordi det vil flytte en figur eller kommentar fra bunden af regnearket.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitBlockingValue"] = "{0} kan ikke indsættes, fordi det vil flytte celler i en matrixformel eller datatabel.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitMergedRegion"] = "{0} kan ikke indsættes, fordi det vil flytte celler i en flettet region.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitTable"] = "{0} kan ikke indsættes, fordi det vil flytte celler i en anden tabel.";
        this["LE_InvalidOperationException_CannotInsertTableRow_TableOnBottomOfWorksheet"] = "{0} kan ikke indsættes, fordi tabellen er i bunden af regnearket.";
        this["LE_InvalidOperationException_CannotModifyKeysCollection"] = "Keys-samlingen kan ikke ændres.";
        this["LE_InvalidOperationException_CannotModifyStandardTableStyle"] = "Standard tabelformater kan ikke ændres.";
        this["LE_InvalidOperationException_CannotModifyValuesCollection"] = "Værdisamlingen kan ikke ændres.";
        this["LE_InvalidOperationException_CannotMoveDisconnectedWorksheet"] = "Regnearket kan ikke flyttes, fordi det ikke hører til en projektmappe.";
        this["LE_InvalidOperationException_CannotRemoveColorsToPaletteDirectly"] = "Farver kan ikke fjernes fra WorkbookColorPalette. Brug Nulstil-metoden til at rydde alle brugerdefinerede farver.";
        this["LE_InvalidOperationException_CannotRemoveNormalStyle"] = "Normaltypen kan ikke fjernes.";
        this["LE_InvalidOperationException_CannotSetParentStyleOnStyle"] = "Typen kan ikke indstilles til et format, der hører til en anden type.";
        this["LE_InvalidOperationException_CannotShift_LossOfData"] = "Kan ikke flytte data fra ark.";
        this["LE_InvalidOperationException_CannotShift_LossOfObject"] = "Kan ikke flytte objekter fra regnearket.";
        this["LE_InvalidOperationException_CannotShift_SplitBlockingValue"] = "Kan ikke flytte i en matrixformel eller datatabel.";
        this["LE_InvalidOperationException_CannotShift_SplitMergedRegion"] = "Kan ikke flytte i en flettet region.";
        this["LE_InvalidOperationException_CannotShift_SplitTable"] = "Kan ikke flytte i tabellen.";
        this["LE_InvalidOperationException_CantAddChartInChart"] = "Et diagram kan ikke føjes til figursamlingen i et andet diagram.";
        this["LE_InvalidOperationException_CantAddCustomView"] = "Brugerdefinerede visninger kan ikke føjes direkte til samlingen.";
        this["LE_InvalidOperationException_CantAddDataTable"] = "Datatabeller kan ikke tilføjes direkte.";
        this["LE_InvalidOperationException_CantAddMergedRegion"] = "Flettede celleregioner kan ikke tilføjes direkte.";
        this["LE_InvalidOperationException_CantAddNamedReference"] = "Navngivne referencer kan ikke tilføjes direkte.";
        this["LE_InvalidOperationException_CantAddWorksheet"] = "Regneark kan ikke føjes direkte til denne samling.";
        this["LE_InvalidOperationException_CantApplyRemovedCustomView"] = "Den brugerdefinerede visning kan ikke anvendes, hvis den er fjernet fra projektmappens samling med tilpasset visning.";
        this["LE_InvalidOperationException_CantChangeArrayFormula"] = "Cellerne i en matrixformel kan ikke ændres.";
        this["LE_InvalidOperationException_CantChangeDataTable"] = "De indre celler i en datatabel kan ikke ændres.";
        this["LE_InvalidOperationException_CantModifyCollection"] = "Samlingen kan ikke ændres.";
        this["LE_InvalidOperationException_CantOverlapArrayFormula"] = "En flettet celleregion kan ikke overlappe en matrixformelregion.";
        this["LE_InvalidOperationException_CantOverlapDataTableInterior"] = "En flettet celleregion kan ikke overlappe cellerne i en datatabel, hvis den strækker sig uden for den øverste række eller kolonnen længst til venstre i disse celler.";
        this["LE_InvalidOperationException_CantSaveEditRangeWithoutRange"] = "Regnearkenes AllowedEditRanges skal have mindst ét interval.";
        this["LE_InvalidOperationException_CantSaveWithNoVisibleWorksheets"] = "Mindst et regneark i projektmappen skal være synligt.";
        this["LE_InvalidOperationException_CantSaveWithNoWorksheets"] = "En projektmappe uden regneark kan ikke gemmes";
        this["LE_InvalidOperationException_CantSetCircularityErrorDirectly"] = "Værdien for cirkularitetsfejl kan ikke anvendes direkte på celler.";
        this["LE_InvalidOperationException_CantSetDataTableDirectly"] = "Datatabeller kan ikke indstilles som værdien på en celle.";
        this["LE_InvalidOperationException_CantSetFormulaDirectly"] = "Formler kan ikke indstilles direkte på en celle, brug i stedet ApplyFormula-metoden eller en af ApplyTo-metoderne i formelforekomsten.";
        this["LE_InvalidOperationException_CellsInTableMinSize"] = "Celleregionen i en datatabel skal have mindst to celler i hver dimension.";
        this["LE_InvalidOperationException_ChartNotSupported"] = "Ændringer i regnearkdiagram understøttes ikke i dette format.";
        this["LE_InvalidOperationException_ClearUnknownShapeData"] = "Alle data for denne form er ukendte. Fjern formen fra dens samling i stedet.";
        this["LE_InvalidOperationException_CollectionLongerThanMaxValue"] = "Samlingen indeholder flere poster, end den angivne maksimale værdi tillader.";
        this["LE_InvalidOperationException_CollectionModifiedWhileEnumerating"] = "Samlingen blev ændret, mens den blev opregnet.";
        this["LE_InvalidOperationException_CurrentTableRowAddressNeedsOriginCell"] = "For at analysere en struktureret tabelreference ved hjælp af den aktuelle række skal der angives en oprindelsescelle.";
        this["LE_InvalidOperationException_CustomViewNoVisibleWorksheets"] = "Mindst et regneark i den brugerdefinerede visning med navnet \"{0}\" skal være synligt.";
        this["LE_InvalidOperationException_DataTableAppliedInTable"] = "En datatabel kan ikke være i et område, hvor der findes en tabel.";
        this["LE_InvalidOperationException_DataTableFormulaCannotBeApplied"] = "En datatabelformel kan ikke anvendes direkte på celler.";
        this["LE_InvalidOperationException_DataTableInMergedCell"] = "De indre celler i en datatabel kan ikke være en del af en flettet celleregion.";
        this["LE_InvalidOperationException_DataTableRemoved"] = "Række- og kolonneinputcellerne kan ikke indstilles, når datatabellen er fjernet fra regnearket.";
        this["LE_InvalidOperationException_DeprecatedFillColorSetWithGradientFill"] = "En fyldmønsterfarve kan ikke indstilles, mens en gradient er indstillet på cellen.";
        this["LE_InvalidOperationException_EncryptedWorkbooksNotSupported"] = "Den projektmappe, der indlæses, er krypteret. Krypterede projektmapper understøttes ikke.";
        this["LE_InvalidOperationException_ExtensionDoesntMatchCurrentFormat"] = "Filnavnet, som filen skal gemmes til, har en udvidelse, der ikke matcher det aktuelle format på projektmappen.";
        this["LE_InvalidOperationException_FixedValuesFilterMustAcceptAValue"] = "Kan ikke have et FixedValuesFilter, som ikke tillader nogen værdier.";
        this["LE_InvalidOperationException_FormattedStringAlreadyOwned"] = "Den formaterede streng er allerede samme værdi som en anden celle eller kommentar. Klon den først, og anvend klonen på den nye celle eller form.";
        this["LE_InvalidOperationException_FormattedStringNotOwned"] = "Der er ikke adgang til skrifttypens egenskaber, mens den formaterede streng eller tekst ikke anvendes på en celle eller form.";
        this["LE_InvalidOperationException_FormattedTextAlreadyOwned"] = "Den formaterede streng er allerede værdien af en anden figur. Klon det først, og anvend klonen på den nye celle eller figur";
        this["LE_InvalidOperationException_FormulaAlreadyOwned"] = "Formlen er allerede anvendt på en anden celle.";
        this["LE_InvalidOperationException_FormulaReferencesInvalidCells"] = "Formlen refererer til celler, som ikke understøttes i projektmappens aktuelle filformat.";
        this["LE_InvalidOperationException_GetBoundsBeforeAnchorsSet"] = "Figuren har ikke grænser, før ankercellerne er indstillet.";
        this["LE_InvalidOperationException_HiddenWorksheetCannotBeSelected"] = "Et skjult regneark kan ikke vælges.";
        this["LE_InvalidOperationException_HyperlinkSealed"] = "Hyperlinket kan ikke ændres.";
        this["LE_InvalidOperationException_ImageDisposed"] = "Et billede af et WorksheetImage er bortskaffet eller er ugyldigt.";
        this["LE_InvalidOperationException_InputCellsBothNull"] = "Kolonneinput- og rækkeinputcellerne kan ikke begge være null.";
        this["LE_InvalidOperationException_InputCellsInTable"] = "Inputcellerne kan ikke være inden for datatabelcellerne.";
        this["LE_InvalidOperationException_InputCellsSame"] = "Kolonneinput- og rækkeinputcellerne kan ikke være den samme celle.";
        this["LE_InvalidOperationException_InvalidCharacterRange"] = "Rækkevidden af tegn, der er dækket af skrifttypen, findes ikke længere i den formaterede streng.";
        this["LE_InvalidOperationException_InvalidFormatOptionsInWholeTableArea"] = "Den angivne FormatOptions-formatværdi er ugyldig i WholeTable-tabelområdet.";
        this["LE_InvalidOperationException_InvalidFormatString"] = "Den angivne formatstreng er ugyldig.";
        this["LE_InvalidOperationException_InvalidFormatString_GetTextCall"] = "Cellens formatstreng er ugyldig. For at få besked om dette, når FormatString er indstillet, skal du indstille Workbook.ValidateFormatStrings til True.";
        this["LE_InvalidOperationException_InvalidForWorkbookFormat"] = "Formlen er ikke gyldig i det angivne format.";
        this["LE_InvalidOperationException_InvalidHeaderCellColumnAreaFormatProperty"] = "Værdierne for den øverste kant kan ikke indstilles i HeaderCell-tabellens kolonneområdeformat.";
        this["LE_InvalidOperationException_InvalidHeaderRowAreaFormatProperty"] = "Værdierne for den øverste kant kan ikke indstilles i HeaderRow-tabelområdeformatet.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFontProperty"] = "Skrifttypeværdien {0} kan ikke indstilles i områder med tabelformat.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatOptions"] = "Den angivne FormatOptions-formatværdi er ugyldig i områder med tabelformat.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatProperty"] = "{0} formatværdien kan ikke indstilles i områder med tabelformat.";
        this["LE_InvalidOperationException_InvalidTotalsCellColumnAreaFormatProperty"] = "Værdierne for den nederste kant kan ikke indstilles i TotalCell-tabellens kolonneområdeformat";
        this["LE_InvalidOperationException_InvalidTotalsRowAreaFormatProperty"] = "De nederste kantværdier kan ikke indstilles i TotalsRow-tabelområdeformatet.";
        this["LE_InvalidOperationException_InvalidWholeTableAreaFormatProperty"] = "Kun kantgrænseværdierne kan indstilles i WholeTable-tabelområdeformatet.";
        this["LE_InvalidOperationException_LastColumnBeforeFirst"] = "Den sidste kolonne i regionen skal være større end eller lig med den første kolonne.";
        this["LE_InvalidOperationException_LastRowBeforeFirst"] = "Den sidste række i regionen skal være større end den første række.";
        this["LE_InvalidOperationException_MaxCellFormats"] = "Det maksimale antal forskellige celleformater i Excel-filen er: {0}";
        this["LE_InvalidOperationException_MaxColors"] = "Det maksimale antal farver i en excel-projektmappe er: {0}";
        this["LE_InvalidOperationException_MaxColumns"] = "Det maksimale antal kolonner i et Excel-regneark med det aktuelle format er: {0}";
        this["LE_InvalidOperationException_MaxFonts"] = "Antallet af skrifttyper i en projektmappe må ikke overstige {0}.";
        this["LE_InvalidOperationException_MaxFormattedStrings"] = "Antallet af unikke strenge i en projektmappe må ikke overstige {0}.";
        this["LE_InvalidOperationException_MaxRows"] = "Det maksimale antal rækker i et Excel-regneark med det aktuelle format er: {0}";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableLeftColumn"] = "Cellekolonnen yderst til venstre, der er angivet til datatabellen, indeholder en flettet celleregion, der strækker sig uden for disse cellers område.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopLeftCell"] = "Den øverste venstre celle af de celler, der er angivet for datatabellen, er en del af en flettet celle, der strækker sig uden for området for disse celler.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopRow"] = "Den øverste række celler, der er angivet for datatabellen, indeholder en flettet celleregion, der strækker sig uden for området for disse celler.";
        this["LE_InvalidOperationException_MergedCellsAppliedInTable"] = "Celler i en tabel kan ikke flettes.";
        this["LE_InvalidOperationException_NamedReferenceNameAlreadyExists"] = "Navnet \"{0}\" bruges af en tabel eller navngivet reference i projektmappen med det samme omfang.";
        this["LE_InvalidOperationException_NoRegionHasBeenSpecified"] = "Der er ikke angivet en region.";
        this["LE_InvalidOperationException_NoWorksheetContextToSetShapeBounds"] = "Denne overbelastning af Tilføj kan anvendes, når figursamlingen ikke er knyttet til et regneark. Anvend overbelastningen af Tilføj, der tager en regnearkseksempel.";
        this["LE_InvalidOperationException_OverlappingTable"] = "Der er en anden tabel i det angivne region";
        this["LE_InvalidOperationException_ReadOnlyFont"] = "Skrifttypen er skrivebeskyttet og kan ikke ændres.";
        this["LE_InvalidOperationException_ReadOnlyFormat"] = "Formatobjektet er skrivebeskyttet og kan ikke ændres.";
        this["LE_InvalidOperationException_RelativeR1C1AddressNeedsOriginCell"] = "For at parse en relativ R1C1-adresse skal der angives en oprindelsescelle.";
        this["LE_InvalidOperationException_ResizedTableMustBeOnWorksheet"] = "Tabellen skal være på et regneark for at ændre størrelsen.";
        this["LE_InvalidOperationException_ResolvedFormatCannotBeModified"] = "Det valgte celleformat er skrivebeskyttet og kan ikke ændres.";
        this["LE_InvalidOperationException_ShapeCantChangeOrientation"] = "Denne figur kan ikke drejes eller vendes.";
        this["LE_InvalidOperationException_ShapeFillSolidIsImmutable"] = "ShapeFillSolid er uforanderlig. Opret en ny ShapeFillSolid i stedet.";
        this["LE_InvalidOperationException_ShapeInAnotherCollection"] = "Figuren findes allerede på et andet regneark eller i en anden figurgruppe.";
        this["LE_InvalidOperationException_ShapeOutlineSolidIsImmutable"] = "ShapeOutlineSolid er uforanderlig.  Opret i stedet en ny ShapeOutlineSolid.";
        this["LE_InvalidOperationException_ShowFilterUIWhileHeaderRowHidden"] = "UI-filteret kan ikke vises, mens sidehovedrækken er skjult.";
        this["LE_InvalidOperationException_SystemColorsAccessedFromWrongThread"] = "Kan ikke få adgang til SystemColors fra en baggrundstråd.";
        this["LE_InvalidOperationException_TableReferenceMustBeUsedFromInsideTable"] = "Formlen har en struktureret tabelreference som kun kan bruges inde fra en tabel.";
        this["LE_InvalidOperationException_TableReferenceToMissingTable"] = "Formlen har en reference til en tabel, der ikke findes";
        this["LE_InvalidOperationException_TableReferenceToMissingTableColumn"] = "Formlen har en reference til en tabelkolonne, der ikke findes.";
        this["LE_InvalidOperationException_TopLeftAnchorFromOtherWorksheet"] = "Når figuren tilføjes denne samling placeres den på et andet regneark end TopLeftCornerCell.";
        this["LE_InvalidOperationException_TotalsRowCannotBeShownInLastRow"] = "Totalrækken kan ikke vises, når tabellen optager den sidste række i regnearket.";
        this["LE_InvalidOperationException_UnknownBIFFFormat"] = "Arbejdsmappefilen var i et ukendt BIFF-format.";
        this["LE_InvalidOperationException_UnsupportedBIFFFormat"] = "Projektmappefilen blev gemt i {0}, et format der ikke understøttes.";
        this["LE_InvalidOperationException_WorkbookDataViolatesFormatLimits"] = "Projektmappen indeholder data, der overskrider grænserne i det angivne format.";
        this["LE_InvalidRegion"] = "Ugyldig regionadresse.";
        this["LE_InvalidValueType"] = "'{0}' er ikke en gyldig værdi for egenskaben ValueType.";
        this["LE_LocationNotSingleRowOrColumn"] = "Placeringsreference er ikke gyldig, fordi cellerne ikke alle er i samme kolonne eller række. Vælg celler, der alle er i en enkelt kolonne eller række.";
        this["LE_LocationOrDataAreaInvalid"] = "Referencen til placeringen eller dataintervallet er ikke gyldig.";
        this["LE_LocationWorksheetMismatch"] = "Placeringen skal være fra dette regneark.";
        this["LE_NotSingleTargetFormula"] = "Kan kun understøtte enkelte målformler.";
        this["LE_NotSupportedException_CellType"] = "{0} er ikke en understøttet celleværditype.";
        this["LE_NotSupportedException_NoPackageFactory"] = "packageFactory kan ikke være null. Når du indlæser fra Excel2007-projektmappeformat og bruger Infragistics2.Documents.Excel-samlingen, skal du angive en IPackageFactory til håndtering af datapakning. Hvis du bruger DotNet Framework 3.0 eller højere, skal du bruge Infragistics3.Documents.Excel-samlingen i stedet, og pakningen håndteres af WindowsBase-klassen.";
        this["LE_OpenPackagingNonConformanceException_ContainsDublinCoreRefinements"] = "IPackagePart er kerneegenskabsforholdet og indeholder andre forbedringer til Dublin Core-elementer end 'dcterms: created' og 'dcterms: modified'.";
        this["LE_OpenPackagingNonConformanceException_ContainsXmlLanguageAttribute"] = "IPackagePart er kerneegenskabsforholdet og indeholder elementer, der indeholder attributten 'xml: lang'.";
        this["LE_OpenPackagingNonConformanceException_CorePropertiesRelationshipAlreadyProcessed"] = "IPackagePart er kerneegenskabsforholdet, men en er allerede blevet behandlet for den tilknyttede IPackage.";
        this["LE_OpenPackagingNonConformanceException_DuplicatePartName"] = "En IPackagePart med samme navn hører allerede til denne IPackage. En IP-pakke kan ikke indeholde dele med identiske navne. Bemærk, at der ikke skelnes mellem store og små bogstaver ved kontrol af identiske værdier.";
        this["LE_OpenPackagingNonConformanceException_UsesMarkupCompatibilityNamespace"] = "IPackagePart er kerneegenskabsforholdet og anvender  Markup Compatibility Namespace..";
        this["LE_OpenPackagingNonConformanceException_XmlContainsDocumentTypeDefinition"] = "En IPackagePart indeholdt en DTD-erklæring (Document Type Definition). Open Packaging Convention-regler forbyder brugen af XML-indhold med DTD-erklæringer.";
        this["LE_OpenPackagingNonConformanceException_XsiTypeAttributeInvalid"] = "IPackagePart er kerneegenskabsforholdet og indeholder elementer, der indeholder attributten 'xsi: type' eller indeholder elementer, der kræves til og ikke indeholder attributten 'xsi: type'.";
        this["LE_OpenXmlSerializerNotLoaded"] = "Formatet '{0}' kræver en serieomkoder, der ikke er indlæst, i JavaScript er dette indeholdt i filen infragistics.excel_serialization_openxml.js";
        this["LE_PercentageValueOutOfRange"] = "Procentværdien {0} skal være mellem 0 og 100.";
        this["LE_PercentileValueOutOfRange"] = "Percentilværdien {0} skal være mellem 0 og 100.";
        this["LE_PercentRangeInvalid"] = "{0} skal være mindre end {1}.";
        this["LE_RowsToRepeatAtTopAreOutsideAvailableRange"] = "Udskriftsindstillingerne har rækker, der gentages øverst, og som ligger uden for intervallet af tilgængelige rækker i det nye format.";
        this["LE_Serialization_NoPrimaryCategoryAxis"] = "Diagramtypen '{0}' kræver en primær kategoriakse. Føj en primær kategoriakse til diagrammets AxisCollection.";
        this["LE_Serialization_NoPrimaryValueAxis"] = "Diagramtypen '{0}' kræver en primær værdiakse. Føj en primær værdiakse til diagrammets AxisCollection.";
        this["LE_Series_ChartType_ChartEx"] = "'{0}' er ugyldig for en ChartType-serie i et '{1}' diagram.";
        this["LE_Series_ChartType_InvalidForComboChart"] = "Diagramtypen '{0}' er ikke gyldig til brug i et kombinationsdiagram.";
        this["LE_Series_ChartType_NonCombo"] = "Diagramtypen for denne serie bestemmes af det overordnede diagrams type ('{0}') og kan ikke indstilles. Indstil det overordnede diagrams ChartType-egenskab til '{1}' i stedet.";
        this["LE_Series_Explosion_OutOfRange"] = "Eksplosionsegenskaben understøtter værdier i området fra 0 til 400 inklusive.";
        this["LE_Series_InvalidAxisGroup"] = "AxisGroup-egenskaben kan ikke indstilles til '{0}', fordi der på den samme akse findes en serie af typen '{1}', som ikke kan dele  akse med en serie af typen '{2}'.";
        this["LE_Series_InvalidSeriesTypeForComboChart"] = "En serie af typen '{0}' kan ikke bruges i et kombinationsdiagram.";
        this["LE_Series_NoPrimaryAxis"] = "Mindst en serie skal afbildes på den primære akse. Skift AxisGroup-egenskaben til 'Primær' i mindst en serie.";
        this["LE_Series_OwningSeries_NotClusteredColumn"] = "OwningSeries-egenskaben kan kun indstilles til en grupperet kolonneserie i et Pareto-diagram.";
        this["LE_Series_OwningSeries_NotParetoLine"] = "OwningSeries-egenskaben kan kun indstilles til en linieserie i et Pareto-diagram.";
        this["LE_Series_OwningSeries_Same"] = "OwningSeries kan ikke referere til sig selv.";
        this["LE_Series_SeriesType_ChartEx"] = "SerieType-værdien '{0}' er ugyldig for et '{1}' - diagram.";
        this["LE_Series_SeriesType_NonCombo"] = "Type-egenskaben kan kun anvendes for kombinationsdiagrammer og kan ikke indstilles til et diagram af typen '{0}'.";
        this["LE_SetValue_Formula"] = "Værditypen '{0}' er kun gyldig for formler. Brug SetFormula-metoden i stedet.";
        this["LE_SetValue_NonNumeric"] = "Værdiparameteren gælder ikke for værditypen '{0}'. Brug den overbelastning, der kun tager en værditype.";
        this["LE_SetValue_Numeric"] = "Værdiparameteren skal angives for værditypen '{0}'. Brug den overbelastning, der tager både værdi og værditype.";
        this["LE_WorksheetChart_ChartType_ChartEx_CannotTransition"] = "ChartType-egenskaben kan ikke ændres fra '{0}' til '{1}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_From"] = "ChartType-egenskaben for et kombinationsdiagram kan ikke ændres til '{0}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_To"] = "Et diagram af typen '{0}' kan ikke ændres til 'Combo'. Et nyt kombinationsdiagram kan oprettes ved hjælp af AddChart-metoden, men ændring af et eksisterende diagram til et kombinationsdiagram understøttes ikke.";
        this["LE_WorksheetChart_ComboChart_NotEnoughSeries"] = "Et kombinationsdiagram kræver mindst to serier. Føj flere elementer til diagrammets SeriesCollection.";
        this["LE_WorksheetChart_MissingRequiredAxis_StockCharts"] = "Et '{0}' diagram kræver en primær kategoriakse, en primær værdiakse, en sekundær kategoriakse og en sekundær værdiakse.";
        this["LE_WorksheetChart_NotEnoughSeries"] = "Et '{0}' diagram kræver mindst {1} serier. For at bruge denne diagramtype skal du føje flere poster til SeriesCollection.";
        this["LE_WorksheetChart_Save_Series"] = "Et '{0}'-diagram skal indeholde mindst én serie. Føj flere serier til diagrammet, før du forsøger at gemme projektmappen.";
        this["LE_WorksheetChart_Save_Series_Pareto"] = "Et 'Pareto'-diagram skal indeholde en' ColumnClustered'-serie og en 'Pareto'-serie. Føj den relevante serie til diagrammet, før du forsøger at gemme projektmappen.";
        this["LE_WorksheetChart_SetComboChartSourceData_InvalidChartTypes"] = "En eller flere af de angivne diagramtyper er ikke gyldige til brug i et kombinationsdiagram: {0}";
        this["LE_WorksheetChart_SetComboChartSourceData_NotAComboChart"] = "SetComboChartSourceData kan ikke kaldes , fordi den aktuelle diagramtype er '{0}'. Skift diagramtype til 'Combo'.";
        this["LE_WorksheetChart_SetComboChartSourceData_SeriesChartTypesNotSupported"] = "Parameteren seriesChartTypes skal være null for et '{0}' - diagram.";
        this["LE_WorksheetChart_SetSourceData_NotEnoughSeries"] = "Kombinationsdiagrammer kræver mindst 2 serier. Skift  'dataRange'-parameteren til et interval, der vil resultere i mindst to serier.";
        this["LE_WorksheetChart_SetSourceData_TooManyAxes"] = "Den angivne kombination af diagramtyper kræver mere end det maksimalt tilladte antal akser.";
        this["LE_WorksheetChart_StockChartNotSupported"] = "Redigering af diagramtypen '{0}' understøttes ikke i øjeblikket.";
        this["LE_WorksheetChart_UnknownChartNotSupported"] = "Dette diagram blev indlæst fra et eksisterende regneark, men dets diagramtype kunne ikke bestemmes. Redigering af ukendte diagramtyper understøttes ikke i øjeblikket.";
        this["LER_ArgumentOutOfRangeException_OutlineLevel"] = "Konturniveauet skal være mellem 0 og 7.";
        this["LER_Exception_KeyNotFound"] = "Nøglen blev ikke fundet.";
        this["LER_Exception_MergedRegionsOverlap"] = "Flettede celleområder kan ikke overlappe hinanden.";
        this["TableHeaderRowDescription"] = "sidehovedrække";
        this["TableInsertRowDescription"] = "indsæt række";
        this["TableTotalsRowDescription"] = "totalrække";
        this["Value_UCErrorCode_Div"] = "#DIV/0!";
        this["Value_UCErrorCode_Fail"] = "#FAIL";
        this["Value_UCErrorCode_NA"] = "#N/A";
        this["Value_UCErrorCode_Name"] = "#NAME?";
        this["Value_UCErrorCode_Null"] = "#NULL!";
        this["Value_UCErrorCode_Num"] = "#NUM!";
        this["Value_UCErrorCode_Ok"] = "#OK";
        this["Value_UCErrorCode_Reference"] = "#REF!";
        this["Value_UCErrorCode_Unknown"] = "#N/A";
        this["Value_UCErrorCode_Value"] = "#VALUE!";
        this["WorkbookColorInfo_Automatic_Description"] = "Automatisk";
        this["WorkbookColorInfo_Description"] = "FarveInfo: {0}";
        this["WorkbookColorInfo_WithTint_Description"] = "FarveInfo: {0}, Farvetone: {1}";
        this["WorksheetShapeSerialization_GroupName"] = "Gruppe";
        this["WorksheetShapeSerialization_ImageName"] = "Billede";
        this["WorksheetShapeSerialization_ShapeName"] = "Figur";
    }
    return ExcelLocaleDa;
}());
export { ExcelLocaleDa };
