// igniteui-react-excel/esm5/lib/ExcelLocaleSv.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var ExcelLocaleSv = /** @class */ /*@__PURE__*/ (function () {
    function ExcelLocaleSv() {
        this["DefaultTotalLabel"] = "Total";
        this["Error_AnchoredReference"] = "Försök att komma åt {0} som en icke-ankrad referens";
        this["Error_ExplicitThis"] = "{0} - Explicit Detta stöds inte";
        this["Error_ExternalRange"] = "Området innehåller extern referens";
        this["Error_FormulaNotSupported"] = "{0} - Försök att komma åt formeln på en referens som inte stöder formler";
        this["Error_Internal"] = "Ett internt fel uppstod i metod {0}.";
        this["Error_InvalidFunction"] = "Ogiltig funktion '{0}' vid {1}.";
        this["Error_InvalidFunctionArgCountTooFew"] = "Funktionen '{0}' som refereras till vid {1} innehåller inte rätt antal argument. Funktionen kräver minst {2} parametrar.";
        this["Error_InvalidFunctionArgCountTooMany"] = "Funktionen '{0}' som refereras till {1} innehåller inte rätt antal argument. Argumentantalet bör inte överstiga {2} parameter (/parametrar).";
        this["Error_InvalidNumberSystemDigit"] = "Ogiltig siffra av '{0}' påträffas i värdet '{1}' för nummersystemet med basen för {2}.";
        this["Error_InvalidOperand"] = "Ogiltig operand";
        this["Error_InvalidOperatorArgCount"] = "Funktionen '{0}' måste acceptera {1} argument.";
        this["Error_InvalidReference"] = "Ogiltig eller otillgänglig referens";
        this["Error_Location"] = "rad {0} kol {1}";
        this["Error_MethodNotSupported"] = "{0} stöder inte {1}.";
        this["Error_NoCurrentRangeElement"] = "Inget aktuellt intervallelement";
        this["Error_NullFunctionResult"] = "Utvärderingen av '{0}' gav ett nollvärde.";
        this["Error_NullOperand"] = "Null-operand";
        this["Error_ParseRef"] = "Okänd referens på {0}.";
        this["Error_RangeFromLast"] = "fromT.Last inte inställt";
        this["Error_RangeFromNull"] = "RefRange: från är null";
        this["Error_RangeFromRefBase"] = "RefRange: från är inte en RefBase: {0}";
        this["Error_RangeLast"] = "RefRange validera: senaste tuplerna [{0}]! = [{1}]";
        this["Error_RangeNormalizeScope"] = "RefRange normalisera: dåligt omfång [{0} / {1}]";
        this["Error_RangeToNull"] = "RefRange: till är null";
        this["Error_RangeToRefBase"] = "RefRange: till är inte en RefBase: {0}";
        this["Error_RangeTuple"] = "RefRange validera: tuplar [{0}] != [{1}]";
        this["Error_RangeValidateCount"] = "RefRange validera: från antal {0}! = Att räkna {1}";
        this["Error_ReadOnlyReference"] = "{0} - Försök att ställa in värde på en enbart läsbar referens";
        this["Error_ReferenceNotEumerable"] = "{0} är inte en uppräkningsbar referens";
        this["Error_RefMalformed"] = "{0} - Felaktig referens {1}";
        this["Error_RP_ContainsBase"] = "RefParser.Contains - Tom bas";
        this["Error_RP_ContainsTarget"] = "RefParser.Contains - Tomt mål";
        this["Error_RP_EmptyElement"] = "Ogiltig referens {0}. Tomt element påträffat på plats {1}.";
        this["Error_RP_EmptyReference"] = "Ogiltig referens {0}. Referensen är tom.";
        this["Error_RP_MalformedExtra"] = "RefParser felaktig referens: Flera [{0}] hittades i [{1}]";
        this["Error_RP_MalformedUnexpected"] = "RefParser felaktig referens: Oväntad [{0}] hittades i [{1}]";
        this["Error_RP_MergeRelativeBase"] = "RefParser.Merge: Basen är relativ";
        this["Error_RP_MissingCloseParenthesis"] = "Ogiltig referens {0}. Saknar avslutande parentes.";
        this["Error_RP_MissingEndQuotes"] = "Ogiltig referens {0}. Avslutande citat saknas.";
        this["Error_RT_Expected"] = "RefTuple: förväntas [{0}] i [{1}]";
        this["Error_RT_InvalidScope"] = "Ogiltigt omfång: {0}.";
        this["Error_RT_InvalidScope_CharactersAfterClosingQuotes"] = "Tecken har påträffats efter att dubbla citat har stängts i {0} omfattning på plats {1}.";
        this["Error_RT_InvalidTuple_CharactersAfterScopeEnd"] = "Ogiltigt element {0}. Tecken har påträffats efter slutet av omfattningen.";
        this["Error_RT_InvalidTuple_MissingCloseParenthesis"] = "Ogiltigt element {0}. Saknar avslutande parentes.";
        this["Error_RT_InvalidTuple_MissingEndQuotes"] = "Ogiltigt element {0}. Avslutande citat saknas.";
        this["Error_RT_InvalidTuple_NamePortionEmpty"] = "Ogiltigt element {0}. Elementnamnsportionen är tom.";
        this["Error_RT_InvalidTuple_ScopeIndexLarge"] = "Ogiltigt element {0}. Index {1} är ogiltigt: {2}.";
        this["Error_RT_InvalidTuple_UnescapedCharacter"] = "Oskyddad {0} karaktär påträffad i {1} på plats {2}.";
        this["Error_RT_NullRefName"] = "RefTuple: null RefName";
        this["Error_RT_Unexpected"] = "RefTuple: oväntat [{0}] i [{1}]";
        this["Error_ScopeAllNotLast"] = "{0} - Omfattning Alla måste visas i det sista elementet {1}";
        this["Error_UCErrorCode_Div"] = "Dela med noll";
        this["Error_UCErrorCode_Fail"] = "Misslyckas";
        this["Error_UCErrorCode_NA"] = "Inte tillämpbar";
        this["Error_UCErrorCode_Name"] = "Ogiltigt namn";
        this["Error_UCErrorCode_Null"] = "Nullvärde";
        this["Error_UCErrorCode_Num"] = "Ogiltiga numeriska värden i funktionen eller formeln.";
        this["Error_UCErrorCode_Ok"] = "Lyckas";
        this["Error_UCErrorCode_Reference"] = "Ogiltig eller otillgänglig referens";
        this["Error_UCErrorCode_Unknown"] = "Ogiltig kod";
        this["Error_UCErrorCode_Value"] = "Felaktig typ av argument eller operand.";
        this["Error_UnexpectedScope"] = "{0} - Oväntat omfång för förälder {1}";
        this["Error_UnknownRange"] = "Okänd räckviddstyp";
        this["Error_UnknownRefType"] = "{0} - Okänd refType {1}";
        this["Error_UnknownScope"] = "{0} - Okänt omfång {1}";
        this["Error_UnresolvedExternal"] = "Olöst extern referens";
        this["Func_abs_Arg_0"] = "nummer";
        this["Func_abs_ArgDesc_0"] = "Det verkliga talet från vilket det absoluta värdet hämtas.";
        this["Func_abs_Category"] = "Matematik";
        this["Func_abs_CategoryURL"] = "Matematik";
        this["Func_abs_Desc"] = "Returnerar det absoluta värdet för ett tal. Det absoluta värdet för ett tal är talet utan dess tecken.";
        this["Func_acos_Arg_0"] = "nummer";
        this["Func_acos_ArgDesc_0"] = "Cosinus för vinkeln som ska hämtas, som måste vara från -1 till 1.";
        this["Func_acos_Category"] = "Matematik";
        this["Func_acos_CategoryURL"] = "Matematik";
        this["Func_acos_Desc"] = "Returnerar arcus cosinus, eller invers cosinus, av ett tal. Arcus cosinus är den vinkel vars cosinus är det angivna talet. Den returnerade vinkeln ges i radianer i området noll till pi. För att konvertera resultatet från radianer till grader, multiplicera med 180 / PI ().";
        this["Func_acosh_Arg_0"] = "nummer";
        this["Func_acosh_ArgDesc_0"] = "Ett reellt tal större än 1 för vilket det inversa hyperboliska cosinusvärdet ska hittas.";
        this["Func_acosh_Category"] = "Matematik";
        this["Func_acosh_CategoryURL"] = "Matematik";
        this["Func_acosh_Desc"] = "Returnerar det inversa hyperboliska cosinusvärdet för det angivna talet. Det inversa hyperboliska cosinusvärdet är den vinkel vars hyperboliska cosinus är det angivna talet.";
        this["Func_and_Arg_0"] = "logisk {0}";
        this["Func_and_ArgDesc_0"] = "Ett logiskt värde (sant eller falskt).";
        this["Func_and_Category"] = "Logisk";
        this["Func_and_CategoryURL"] = "Logisk";
        this["Func_and_Desc"] = "Returnerar sant om alla värden är sanna eller returnerar falskt om ett eller flera värden är falska.";
        this["Func_asin_Arg_0"] = "nummer";
        this["Func_asin_ArgDesc_0"] = "Sinus för som ska hämtas, som måste vara från -1 till 1 radianer.";
        this["Func_asin_Category"] = "Matematik";
        this["Func_asin_CategoryURL"] = "Matematik";
        this["Func_asin_Desc"] = "Returnerar en arcus sinus eller invers sinus för ett tal. Arcus sinus är den vinkel vars sinus är det angivna talet. Den returnerade vinkeln ges i radianer i intervallet -pi / 2 till pi / 2. För att uttrycka arcus sinus i grader, multiplicera resultatet med 180 / PI ().";
        this["Func_asinh_Arg_0"] = "nummer";
        this["Func_asinh_ArgDesc_0"] = "Ett reellt tal för vilket det inversa hyperboliska sinusvärdet ska hittas.";
        this["Func_asinh_Category"] = "Matematik";
        this["Func_asinh_CategoryURL"] = "Matematik";
        this["Func_asinh_Desc"] = "Returnerar det inversa hyperboliska sinusvärdet för det angivna numret. Det inversa hyperboliska sinusvärdet är den vinkel vars hyperboliska sinus är det angivna talet.";
        this["Func_atan_Arg_0"] = "nummer";
        this["Func_atan_ArgDesc_0"] = "Tangenten för vinkeln som ska hämtas.";
        this["Func_atan_Category"] = "Matematik";
        this["Func_atan_CategoryURL"] = "Matematik";
        this["Func_atan_Desc"] = "Returnerar arktangensen eller den inversa tangenten för ett tal. Arktangenten är den vinkel vars tangens är det angivna talet. Den returnerade vinkeln ges i radianer i intervallet -pi / 2 till pi / 2. För att uttrycka arktangenten i grader, multiplicera resultatet med 180 / PI ().";
        this["Func_atan2_Arg_0"] = "x";
        this["Func_atan2_Arg_1"] = "y";
        this["Func_atan2_ArgDesc_0"] = "Punktens x-koordinat.";
        this["Func_atan2_ArgDesc_1"] = "Punktens y-koordinat.";
        this["Func_atan2_Category"] = "Matematik";
        this["Func_atan2_CategoryURL"] = "Matematik";
        this["Func_atan2_Desc"] = "Returnerar arktangenten eller den inversa tangenten för de angivna x- och y-koordinaterna. Arktangenten är vinkeln från x-axeln till en linje som innehåller ursprunget (0, 0) och en punkt med koordinater (x_num, y_num). Vinkeln ges i radianer mellan -pi och pi, exklusive -pi.";
        this["Func_atanh_Arg_0"] = "nummer";
        this["Func_atanh_ArgDesc_0"] = "Ett reellt tal mellan -1 och 1 för vilket den inversa hyperboliska tangenten ska hittas.";
        this["Func_atanh_Category"] = "Matematik";
        this["Func_atanh_CategoryURL"] = "Matematik";
        this["Func_atanh_Desc"] = "Returnerar den inversa hyperboliska tangenten för det angivna numret. Den inversa hyperboliska tangenten är den vinkel vars hyperboliska tangent är det angivna talet.";
        this["Func_avedev_Arg_0"] = "nummer {0}";
        this["Func_avedev_ArgDesc_0"] = "Number1, number2, ... är ett eller flera numeriska värden eller referenser till numeriska värden.";
        this["Func_avedev_Category"] = "Statistisk";
        this["Func_avedev_CategoryURL"] = "Statistisk";
        this["Func_avedev_Desc"] = "Returnerar den genomsnittliga avvikelsen från genomsnittet för alla numeriska värden.";
        this["Func_average_Arg_0"] = "nummer {0}";
        this["Func_average_ArgDesc_0"] = "Ett tal eller intervall av tal för vilka medelvärdet ska hittas med de andra argumenten.";
        this["Func_average_Category"] = "Statistisk";
        this["Func_average_CategoryURL"] = "Statistisk";
        this["Func_average_Desc"] = "Returnerar medelvärdet (aritmetiskt medelvärde) för ett eller flera värden.";
        this["Func_bin2dec_Arg_0"] = "Nummer";
        this["Func_bin2dec_ArgDesc_0"] = "Ett textuttryck som representerar ett binärt tal.";
        this["Func_bin2dec_Category"] = "Teknik";
        this["Func_bin2dec_CategoryURL"] = "Teknik";
        this["Func_bin2dec_Desc"] = "Returnerar decimalvärdet för en textrepresentation av ett tal i det binära nummersystemet.";
        this["Func_bin2hex_Arg_0"] = "Nummer";
        this["Func_bin2hex_Arg_1"] = "Platser";
        this["Func_bin2hex_ArgDesc_0"] = "Ett textuttryck som representerar ett binärt tal.";
        this["Func_bin2hex_ArgDesc_1"] = "Antalet tecken som ska användas för att representera värdet. Om detta inte anges används det maximala antalet tecken som behövs för att representera värdet. Detta låter dig ange ledande nollor.";
        this["Func_bin2hex_Category"] = "Teknik";
        this["Func_bin2hex_CategoryURL"] = "Teknik";
        this["Func_bin2hex_Desc"] = "Konverterar ett textuttryck som representerar ett binärt tal till ett textuttryck som representerar samma nummer i det hexadecimala talsystemet.";
        this["Func_bin2oct_Arg_0"] = "Nummer";
        this["Func_bin2oct_Arg_1"] = "Platser";
        this["Func_bin2oct_ArgDesc_0"] = "Ett textuttryck som representerar ett binärt tal.";
        this["Func_bin2oct_ArgDesc_1"] = "Antalet tecken som ska användas för att representera värdet. Om detta inte anges används det maximala antalet tecken som behövs för att representera värdet. Detta låter dig ange ledande nollor.";
        this["Func_bin2oct_Category"] = "Teknik";
        this["Func_bin2oct_CategoryURL"] = "Teknik";
        this["Func_bin2oct_Desc"] = "Konverterar ett textuttryck som representerar ett binärt tal till ett textuttryck som representerar samma nummer i det oktala talsystemet.";
        this["Func_binomdist_Arg_0"] = "Nummer";
        this["Func_binomdist_Arg_1"] = "Försök";
        this["Func_binomdist_Arg_2"] = "Sannolikhet";
        this["Func_binomdist_Arg_3"] = "Kumulativ";
        this["Func_binomdist_ArgDesc_0"] = "Antalet framgångar i försök.";
        this["Func_binomdist_ArgDesc_1"] = "Antalet oberoende försök.";
        this["Func_binomdist_ArgDesc_2"] = "Sannolikheten för framgång vid varje försök.";
        this["Func_binomdist_ArgDesc_3"] = "Ett logiskt värde som bestämmer funktionens form. Om det är sant kommer funktionen att returnera den kumulativa fördelningen som är sannolikheten för att det är högst \"antal\" framgångar, annars returnerar den sannolikhetsmassan vilket är sannolikheten att det finns \"antal\" framgångar.";
        this["Func_binomdist_Category"] = "Statistisk";
        this["Func_binomdist_CategoryURL"] = "Statistisk";
        this["Func_binomdist_Desc"] = "Returnerar antalet möjliga kombinationer (med repetitioner) som ges en uppsättning objekt och ett antal valda objekt från den uppsättningen.";
        this["Func_CategoryURL_Template"] = "ms-help://INFRAGISTICS_HELP/NetAdvantagev{2}{3}/Win/WinElements/WinCalcManager_{0}_Functions.html#FUNC_{1}";
        this["Func_ceiling_Arg_0"] = "nummer";
        this["Func_ceiling_Arg_1"] = "betydelse";
        this["Func_ceiling_ArgDesc_0"] = "Siffran som ska avrundas.";
        this["Func_ceiling_ArgDesc_1"] = "Multipeln som du ska runda av. Om den utelämnas antas detta vara ett.";
        this["Func_ceiling_Category"] = "Matematik";
        this["Func_ceiling_CategoryURL"] = "Matematik";
        this["Func_ceiling_Desc"] = "Returnerar den minsta multipeln av betydelsen (standard 1) som är större än eller lika med det angivna talet.";
        this["Func_char_Arg_0"] = "charCode";
        this["Func_char_ArgDesc_0"] = "Numret som ska konverteras till ett tecken.";
        this["Func_char_Category"] = "TextAndData";
        this["Func_char_CategoryURL"] = "Text_and_Data";
        this["Func_char_Desc"] = "Returnerar det tecken som anges med kodnumret från teckenuppsättningen för din dator.";
        this["Func_choose_Arg_0"] = "IndexNum";
        this["Func_choose_Arg_1"] = "Value{0}";
        this["Func_choose_ArgDesc_0"] = "Det 1-baserade indexet i resten av argumenten för det värde som ska användas.";
        this["Func_choose_ArgDesc_1"] = "Ett värde som kan väljas beroende på IndexNum-värdet.";
        this["Func_choose_Category"] = "LookupAndReference";
        this["Func_choose_CategoryURL"] = "Lookup_And_Reference";
        this["Func_choose_Desc"] = "Returnerar ett av värdena i dess argument baserat på numret i det första argumentet.";
        this["Func_code_Arg_0"] = "text";
        this["Func_code_ArgDesc_0"] = "Textuttrycket som får sin första karaktär utvärderad.";
        this["Func_code_Category"] = "TextAndData";
        this["Func_code_CategoryURL"] = "Text_and_Data";
        this["Func_code_Desc"] = "Returnerar en numerisk kod för det första tecknet i ett textuttryck med hjälp av teckenuppsättningen som används av din dator.";
        this["Func_combin_Arg_0"] = "Nummer";
        this["Func_combin_Arg_1"] = "NumberChosen";
        this["Func_combin_ArgDesc_0"] = "Antalet objekt.";
        this["Func_combin_ArgDesc_1"] = "Antalet objekt som valts i varje kombination.";
        this["Func_combin_Category"] = "Matematik";
        this["Func_combin_CategoryURL"] = "Matematik";
        this["Func_combin_Desc"] = "Returnerar antalet möjliga kombinationer med en uppsättning objekt och ett antal valda objekt från den uppsättningen.";
        this["Func_combina_Arg_0"] = "Nummer";
        this["Func_combina_Arg_1"] = "NumberChosen";
        this["Func_combina_ArgDesc_0"] = "Antalet objekt.";
        this["Func_combina_ArgDesc_1"] = "Antalet objekt som valts i varje kombination.";
        this["Func_combina_Category"] = "Matematik";
        this["Func_combina_CategoryURL"] = "Matematik";
        this["Func_combina_Desc"] = "Returnerar antalet möjliga kombinationer (med repetitioner) som ges en uppsättning objekt och ett antal valda objekt från den uppsättningen.";
        this["Func_complex_Arg_0"] = "RealNumber";
        this["Func_complex_Arg_1"] = "ImaginaryNumber";
        this["Func_complex_Arg_2"] = "Suffix";
        this["Func_complex_ArgDesc_0"] = "Den reella koefficienten för det komplexa talet.";
        this["Func_complex_ArgDesc_1"] = "Talets imaginära koefficient.";
        this["Func_complex_ArgDesc_2"] = "Anger suffixet som ska användas. De acceptabla värdena är \"i\" eller \"j\". Om detta utelämnas används \"i\". Observera att suffixet är skiftlägeskänsligt; versaler \"I\" och \"J\" är inte tillåtna och kommer att resultera i ett #VALUE-fel.";
        this["Func_complex_Category"] = "Teknik";
        this["Func_complex_CategoryURL"] = "Teknik";
        this["Func_complex_Desc"] = "Returnerar ett komplext tal representerat som ett textuttryck i formatet \"x + yi\" eller \"x + yj\" genom att kombinera ett reellt och imaginärt tal.";
        this["Func_concatenate_Arg_0"] = "sms {0}";
        this["Func_concatenate_ArgDesc_0"] = "Ett textuttryck eller intervall av textuttryck som ska sammanfogas till ett enda textuttryck.";
        this["Func_concatenate_Category"] = "TextAndData";
        this["Func_concatenate_CategoryURL"] = "Text_and_Data";
        this["Func_concatenate_Desc"] = "Sammanfogar alla argument till ett textuttryck.";
        this["Func_convert_Arg_0"] = "Nummer";
        this["Func_convert_Arg_1"] = "FromUnit";
        this["Func_convert_Arg_2"] = "ToUnit";
        this["Func_convert_ArgDesc_0"] = "Värdet som ska konverteras.";
        this["Func_convert_ArgDesc_1"] = "Enheten där numret anges.\r\n\r\nVikt och massenheter: \" g \" = gram; \" sg \" = slug; \" lbm \" = pundmassa (avoirdupois); \" u \" = U (atommasseenhet); \" ozm \" = uns massa (avoirdupois)\r\nAvståndsenheter: \" m \" = mätare; \" mi \" = stadgad mil; \" Nmi \" = sjömil; \" in \" = tum; \" ft \" = Fot; \" yd \" = Yard; \" ang \" = Ångström; \" Pica \" = Pica (1/72 tum)\r\nTidsenheter: \" yr \" = år; \" dag \" = dag; \" hr \" = timme; \" mn \" = minut; \" sek \" = sekund\r\nTryckenheter: \" Pa \" (eller \" p \") = Pascal; \" atm \" (eller \" at \") = Atmosfär; \" mmHg \" = mm kvicksilver\r\nKraft-enheter: \" N \" = Newton; \" dyn \" (eller \" dy \") = Dyne; \" lbf \" = pundkraft\r\nEnergienheter: \" J \" = Joule; \" e \" = Erg; \" c \" = termodynamisk kalori; \" cal \" = IT-kalori; \" eV \" (eller \" ev \") = Elektronvolt; \" HPh \" (eller \" hh \") = Hästkrafttimmar; \" Wh \" (eller \" wh \") = Watt-timme; \" flb \" = Fotpund; \" BTU \" (eller \" btu \") = BTU\r\nKraftenheter: \" HP \" (eller \" h \") = Hästkrafter; \" W \" (eller \" w \") = Watt\r\nMagentism enheter: \" T \" = Tesla; \" ga \" = Gauss\r\nTemperaturenheter: \" C \" (eller \" cel \") = Grad Celsius; \" F \" (eller \" fah \") = Grad Fahrenheit; \" K \" (eller \" kel \") = Kelvin\r\nFlytande måttenheter: \" tsk \"= Tesked; \" tbs \" = matsked; \" oz \" = flytande uns; \" kopp \"= kopp; \" pt \" (eller \" us_pt \") = amerikansk pint; \" uk_pt \" = brittisk pint; \" qt \" = Quart; \" gal \" = gallon; \" l \" (eller \" lt \") = liter\r\n\r\nVilken metrisk enhet som helst kan föregås av något av följande: exa (\" E \") = 1E + 18; peta (\" P \") = 1E + 15; tera (\" T \") = 1E + 12; giga (\" G \") = 1E + 09; mega (\" M \") = 1E + 06; kilo (\" k \") = 1E + 03; hekto (\" h \") = 1E + 02; dekao (\" e \") = 1E + 01; deci (\" d \") = 1E-01; centi (\" c \") = 1E-02; milli (\" m \") = 1E-03; mikro (\" u \") = 1E-06; nano (\" n \") = 1E-09; pico (\" p \") = 1E-12; femto (\" f \") = 1E-15; atto (\" a \") = 1E-18\r\n    ";
        this["Func_convert_ArgDesc_2"] = "De enheter som det konverterade värdet ska returneras i.\r\n\r\nVikt och massa enheter: \" g \" = gram; \" sg \" = slug; \" lbm \" = pundmassa (avoirdupois); \" u \" = U (atommasseenhet); \" ozm \" = uns massa (avoirdupois)\r\nAvståndsenheter: \" m \" = mätare; \" mi \" = stadga mil; \" Nmi \" = sjömil; \" in \" = tum; \" ft \" = Fot; \" yd \" = Yard; \" ang \" = Ångström; \" Pica \" = Pica (1/72 tum)\r\nTidsenheter: \" yr \" = år; \" dag \" = dag; \" hr \" = timme; \" mn \" = minut; \" sek \" = sekund\r\nTryckenheter: \" Pa \" (eller \" p \") = Pascal; \" atm \" (eller \" at \") = Atmosfär; \" mmHg \" = mm kvicksilver\r\nTvinga enheter: \" N \" = Newton; \" dyn \" (eller \" dy \") = Dyne; \" lbf \" = pundkraft\r\nEnergienheter: \" J \" = Joule; \" e \" = Erg; \" c \" = termodynamisk kalori; \" cal \" = IT-kalori; \" eV \" (eller \" ev \") = Elektronvolt; \" HPh \" (eller \" hh \") = Hästkrafttimmar; \" Wh \" (eller \" wh \") = Watt-timme; \" flb \" = Fotpund; \" BTU \" (eller \" btu \") = BTU\r\nKraftenheter: \" HP \" (eller \" h \") = Hästkrafter; \" W \" (eller \" w \") = Watt\r\nMagentism enheter: \" T \" = Tesla; \" ga \" = Gauss\r\nTemperaturenheter: \" C \" (eller \" cel \") = Grad Celsius; \" F \" (eller \" fah \") = Grad Fahrenheit; \" K \" (eller \" kel \") = Kelvin\r\nFlytande måttenheter: \" tsk \"= Tesked; \" tbs \" = matsked; \" oz \" = flytande uns; \" kopp \" = kopp; \" pt \" (eller \" us_pt \") = amerikansk pint; \" uk_pt \" = brittisk pint; \" qt \" = Quart; \" gal \" = gallon; \" l \" (eller \" lt \") = liter\r\n\r\nVilken metrisk enhet som helst kan föregås av något av följande: exa (\" E \") = 1E + 18; peta (\" P \") = 1E + 15; tera (\" T \") = 1E + 12; giga (\" G \") = 1E + 09; mega (\" M \") = 1E + 06; kilo (\" k \") = 1E + 03; hekto (\" h \") = 1E + 02; dekao (\" e \") = 1E + 01; deci (\" d \") = 1E-01; centi (\" c \") = 1E-02; milli (\" m \") = 1E-03; mikro (\" u \") = 1E-06; nano (\" n \") = 1E-09; pico (\" p \") = 1E-12; femto (\" f \") = 1E-15; atto (\" a \") = 1E-18\r\n    ";
        this["Func_convert_Category"] = "Teknik";
        this["Func_convert_CategoryURL"] = "Teknik";
        this["Func_convert_Desc"] = "Konverterar ett värde från ett mätsystem till ett annat, till exempel meter till tum eller timmar till sekunder.";
        this["Func_cos_Arg_0"] = "nummer";
        this["Func_cos_ArgDesc_0"] = "Vinkeln i radianer för vilken cosinus ska fås.";
        this["Func_cos_Category"] = "Matematik";
        this["Func_cos_CategoryURL"] = "Matematik";
        this["Func_cos_Desc"] = "Returnerar cosinus för den angivna vinkeln.";
        this["Func_cosh_Arg_0"] = "nummer";
        this["Func_cosh_ArgDesc_0"] = "Ett reellt tal för vilket ett hyperboliskt cosinus ska hittas.";
        this["Func_cosh_Category"] = "Matematik";
        this["Func_cosh_CategoryURL"] = "Matematik";
        this["Func_cosh_Desc"] = "Returnerar ett tals hyperboliska cosinus.";
        this["Func_count_Arg_0"] = "värde {0}";
        this["Func_count_ArgDesc_0"] = "En referens eller ett referensområde som kan innehålla siffror.";
        this["Func_count_Category"] = "Statistisk";
        this["Func_count_CategoryURL"] = "Statistisk";
        this["Func_count_Desc"] = "Räknar antalet referenser som innehåller nummer.";
        this["Func_counta_Arg_0"] = "värde {0}";
        this["Func_counta_ArgDesc_0"] = "En referens eller ett referensområde som kan innehålla siffror.";
        this["Func_counta_Category"] = "Statistisk";
        this["Func_counta_CategoryURL"] = "Statistisk";
        this["Func_counta_Desc"] = "Räknar antalet referenser som inte är tomma.";
        this["Func_critbinom_Arg_0"] = "Försök";
        this["Func_critbinom_Arg_1"] = "Sannolikhet";
        this["Func_critbinom_Arg_2"] = "Alfa";
        this["Func_critbinom_ArgDesc_0"] = "Antalet oberoende försök.";
        this["Func_critbinom_ArgDesc_1"] = "Sannolikheten för framgång vid varje försök.";
        this["Func_critbinom_ArgDesc_2"] = "Kriterievärdet.";
        this["Func_critbinom_Category"] = "Statistisk";
        this["Func_critbinom_CategoryURL"] = "Statistisk";
        this["Func_critbinom_Desc"] = "Returnerar det minsta värdet för vilket den kumulativa binomialfördelningen är större än eller lika med ett kriterievärde.";
        this["Func_date_Arg_0"] = "år";
        this["Func_date_Arg_1"] = "månad";
        this["Func_date_Arg_2"] = "dag";
        this["Func_date_ArgDesc_0"] = "Ett nummer som representerar året.";
        this["Func_date_ArgDesc_1"] = "Ett nummer som representerar årets månad.";
        this["Func_date_ArgDesc_2"] = "Ett nummer som representerar dagen i månaden.";
        this["Func_date_Category"] = "DatumOchTid";
        this["Func_date_CategoryURL"] = "Date_and_Time";
        this["Func_date_Desc"] = "Returnerar det löpande serienumret som representerar ett visst datum.";
        this["Func_dateadd_Arg_0"] = "intervall";
        this["Func_dateadd_Arg_1"] = "nummer";
        this["Func_dateadd_Arg_2"] = "date";
        this["Func_dateadd_ArgDesc_0"] = "Ett textuttryck som representerar tidsintervallet som ska läggas till.\r\nd = dag\r\ny = år\r\nh = timme\r\nn = minut\r\nm = månad\r\nq = kvartal\r\ns = sekund\r\nw = vecka\r\nww = kalendervecka\r\nyyyy = år\r\n";
        this["Func_dateadd_ArgDesc_1"] = "Antalet intervall som ska läggas till. Detta värde kan vara positivt (för att få en DateTime i framtiden) eller negativt (för att få en DateTime tidigare). Det kan innehålla ett bråk när Intervall avser timmar, minuter eller sekunder. För andra värden av Intervall ignoreras alla delar av Nummer.";
        this["Func_dateadd_ArgDesc_2"] = "Datumet till vilket intervallet ska läggas till.";
        this["Func_dateadd_Category"] = "DatumOchTid";
        this["Func_dateadd_CategoryURL"] = "Date_and_Time";
        this["Func_dateadd_Desc"] = "Returnerar ett datum till vilket det angivna intervallet har lagts till.";
        this["Func_datediff_Arg_0"] = "intervall";
        this["Func_datediff_Arg_1"] = "date1";
        this["Func_datediff_Arg_2"] = "date2";
        this["Func_datediff_ArgDesc_0"] = "Ett textuttryck som representerar tidsintervallet som ska användas som enhet för skillnaden mellan datum1 och datum2.\r\nd = dag\r\ny = år\r\nh = timme\r\nn = minut\r\nm = månad\r\nq = kvartal\r\ns = sekund\r\nw = vecka\r\nww = Kalendervecka\r\nyyyy = år\r\n";
        this["Func_datediff_ArgDesc_1"] = "Det första datumet som ska användas i skillnadsberäkningen.";
        this["Func_datediff_ArgDesc_2"] = "Det andra datumet som ska användas i skillnadsberäkningen.";
        this["Func_datediff_Category"] = "DatumOchTid";
        this["Func_datediff_CategoryURL"] = "Date_and_Time";
        this["Func_datediff_Desc"] = "Returnerar ett värde som anger antalet intervall mellan två DateTime-värden.";
        this["Func_datevalue_Arg_0"] = "date_text";
        this["Func_datevalue_ArgDesc_0"] = "Ett textuttryck som representerar ett datum.";
        this["Func_datevalue_Category"] = "DatumOchTid";
        this["Func_datevalue_CategoryURL"] = "Date_and_Time";
        this["Func_datevalue_Desc"] = "Returnerar datumvärdet för den angivna datumtexten.";
        this["Func_day_Arg_0"] = "Date_value";
        this["Func_day_ArgDesc_0"] = "Ett datumvärde för att få dagen från.";
        this["Func_day_Category"] = "DatumOchTid";
        this["Func_day_CategoryURL"] = "Date_and_Time";
        this["Func_day_Desc"] = "Returnerar dagen i månaden för ett datumvärde.";
        this["Func_days360_Arg_0"] = "start_date";
        this["Func_days360_Arg_1"] = "end_date";
        this["Func_days360_Arg_2"] = "metod";
        this["Func_days360_ArgDesc_0"] = "Startdatum för intervallet av dagar. Om detta är efter slutdatumet returnerar funktionen ett negativt tal.";
        this["Func_days360_ArgDesc_1"] = "Slutdatum för dagar. Om detta är före startdatumet returnerar funktionen ett negativt tal.";
        this["Func_days360_ArgDesc_2"] = "Ett logiskt värde som anger om den amerikanska (falska) eller europeiska (sanna) metoden ska användas i beräkningen.";
        this["Func_days360_Category"] = "DatumOchTid";
        this["Func_days360_CategoryURL"] = "Date_and_Time";
        this["Func_days360_Desc"] = "Returnerar antalet dagar mellan två datum baserat på ett 360-dagarsår.";
        this["Func_db_Arg_0"] = "kostnad";
        this["Func_db_Arg_1"] = "rädda";
        this["Func_db_Arg_2"] = "liv";
        this["Func_db_Arg_3"] = "period";
        this["Func_db_Arg_4"] = "månad";
        this["Func_db_ArgDesc_0"] = "Tillgångens initiala kostnad.";
        this["Func_db_ArgDesc_1"] = "Värdet i slutet av avskrivningen (kallas ibland tillgångens bärgningsvärde).";
        this["Func_db_ArgDesc_2"] = "Antalet perioder under vilka tillgången skrivs av (kallas ibland tillgångens nyttjandeperiod).";
        this["Func_db_ArgDesc_3"] = "Den period för vilken avskrivningarna ska beräknas. Perioden måste använda samma enheter som livet.";
        this["Func_db_ArgDesc_4"] = "Antalet månader det första året. Om månad utelämnas antas den vara 12.";
        this["Func_db_Category"] = "Finansiell";
        this["Func_db_CategoryURL"] = "Finansiell";
        this["Func_db_Desc"] = "Returnerar avskrivningen på en tillgång under en angiven period med hjälp av metoden med fast minskande balans.";
        this["Func_dbnull_Category"] = "Information";
        this["Func_dbnull_CategoryURL"] = "Information";
        this["Func_dbnull_Desc"] = "Returnerar något som representerar ett obefintligt värde.";
        this["Func_ddb_Arg_0"] = "kostnad";
        this["Func_ddb_Arg_1"] = "rädda";
        this["Func_ddb_Arg_2"] = "liv";
        this["Func_ddb_Arg_3"] = "period";
        this["Func_ddb_Arg_4"] = "faktor";
        this["Func_ddb_ArgDesc_0"] = "Tillgångens initiala kostnad.";
        this["Func_ddb_ArgDesc_1"] = "Värdet i slutet av avskrivningen (kallas ibland tillgångens bärgningsvärde).";
        this["Func_ddb_ArgDesc_2"] = "Antalet perioder under vilka tillgången skrivs av (kallas ibland tillgångens nyttjandeperiod).";
        this["Func_ddb_ArgDesc_3"] = "Den period för vilken avskrivningarna ska beräknas. Perioden måste använda samma enheter som livet.";
        this["Func_ddb_ArgDesc_4"] = "Den takt med vilken balansen sjunker. Om faktor utelämnas antas den vara 2 (den dubbelt sjunkande balansmetoden).";
        this["Func_ddb_Category"] = "Finansiell";
        this["Func_ddb_CategoryURL"] = "Finansiell";
        this["Func_ddb_Desc"] = "Returnerar avskrivningen på en tillgång under en angiven period med den dubbelt avtagande balansmetoden eller någon annan angiven metod.";
        this["Func_dec2bin_Arg_0"] = "Nummer";
        this["Func_dec2bin_Arg_1"] = "Platser";
        this["Func_dec2bin_ArgDesc_0"] = "Decimalvärdet som ska konverteras.";
        this["Func_dec2bin_ArgDesc_1"] = "Antalet tecken som ska användas för att representera värdet. Om platser inte anges används det maximala antalet tecken som behövs för att representera värdet. Detta låter dig ange ledande nollor.";
        this["Func_dec2bin_Category"] = "Teknik";
        this["Func_dec2bin_CategoryURL"] = "Teknik";
        this["Func_dec2bin_Desc"] = "Returnerar ett textuttryck som representerar det angivna decimalvärdet som ett binärt tal.";
        this["Func_dec2hex_Arg_0"] = "Nummer";
        this["Func_dec2hex_Arg_1"] = "Platser";
        this["Func_dec2hex_ArgDesc_0"] = "Decimalvärdet som ska konverteras.";
        this["Func_dec2hex_ArgDesc_1"] = "Antalet tecken som ska användas för att representera värdet. Om platser inte anges används det maximala antalet tecken som behövs för att representera värdet. Detta låter dig ange ledande nollor.";
        this["Func_dec2hex_Category"] = "Teknik";
        this["Func_dec2hex_CategoryURL"] = "Teknik";
        this["Func_dec2hex_Desc"] = "Returnerar ett textuttryck som representerar det angivna decimalvärdet som ett hexadecimalt tal.";
        this["Func_dec2oct_Arg_0"] = "Nummer";
        this["Func_dec2oct_Arg_1"] = "Platser";
        this["Func_dec2oct_ArgDesc_0"] = "Decimalvärdet som ska konverteras.";
        this["Func_dec2oct_ArgDesc_1"] = "Antalet tecken som ska användas för att representera värdet. Om platser inte anges används det maximala antalet tecken som behövs för att representera värdet. Detta låter dig ange ledande nollor.";
        this["Func_dec2oct_Category"] = "Teknik";
        this["Func_dec2oct_CategoryURL"] = "Teknik";
        this["Func_dec2oct_Desc"] = "Returnerar ett textuttryck som representerar det angivna decimalvärdet som ett oktalt tal.";
        this["Func_degrees_Arg_0"] = "Radianer";
        this["Func_degrees_ArgDesc_0"] = "Ett värde i radianer som omvandlas till grader.";
        this["Func_degrees_Category"] = "Matematik";
        this["Func_degrees_CategoryURL"] = "Matematik";
        this["Func_degrees_Desc"] = "Konverterar radianer till grader.";
        this["Func_delta_Arg_0"] = "Number1";
        this["Func_delta_Arg_1"] = "Number2";
        this["Func_delta_ArgDesc_0"] = "Det första numret som ska jämföras.";
        this["Func_delta_ArgDesc_1"] = "Det andra numret som ska jämföras. Om det inte anges antas det vara noll.";
        this["Func_delta_Category"] = "Teknik";
        this["Func_delta_CategoryURL"] = "Teknik";
        this["Func_delta_Desc"] = "Jämför två siffror och returnerar ett om de är lika eller noll om de inte är det.";
        this["Func_dollarde_Arg_0"] = "FractionDollarAmount";
        this["Func_dollarde_Arg_1"] = "Fraktion";
        this["Func_dollarde_ArgDesc_0"] = "Dollarbeloppet uttryckt som ett bråk.";
        this["Func_dollarde_ArgDesc_1"] = "Värdet som används som bråkens nämnare. Om detta värde inte är ett heltal trunkeras det.";
        this["Func_dollarde_Category"] = "Finansiell";
        this["Func_dollarde_CategoryURL"] = "Finansiell";
        this["Func_dollarde_Desc"] = "Konverterar ett dollarbelopp uttryckt som ett bråk till ett dollarbelopp uttryckt som ett decimalvärde.";
        this["Func_dollarfr_Arg_0"] = "DecimalDollarAmount";
        this["Func_dollarfr_Arg_1"] = "Fraktion";
        this["Func_dollarfr_ArgDesc_0"] = "Dollarbeloppet uttryckt som ett decimalvärde.";
        this["Func_dollarfr_ArgDesc_1"] = "Värdet som används som bråkens nämnare. Om detta värde inte är ett heltal trunkeras det.";
        this["Func_dollarfr_Category"] = "Finansiell";
        this["Func_dollarfr_CategoryURL"] = "Finansiell";
        this["Func_dollarfr_Desc"] = "Konverterar ett dollarbelopp uttryckt som ett decimalvärde till ett dollarbelopp uttryckt som ett bråk.";
        this["Func_edate_Arg_0"] = "date";
        this["Func_edate_Arg_1"] = "numberOfMonths";
        this["Func_edate_ArgDesc_0"] = "Startdatumet.";
        this["Func_edate_ArgDesc_1"] = "Antalet månader att flytta datumet. Ett positivt tal betyder ett datum efter startdatumet. Ett negativt tal anger ett datum före startdatumet.";
        this["Func_edate_Category"] = "DatumOchTid";
        this["Func_edate_CategoryURL"] = "DatumOchTid";
        this["Func_edate_Desc"] = "Returnerar ett datum som är det angivna antalet månader före eller efter startdatumet.";
        this["Func_eomonth_Arg_0"] = "date";
        this["Func_eomonth_Arg_1"] = "numberOfMonths";
        this["Func_eomonth_ArgDesc_0"] = "Startdatumet.";
        this["Func_eomonth_ArgDesc_1"] = "Antalet månader att flytta datumet. Ett positivt tal betyder ett datum efter startdatumet. Ett negativt tal anger ett datum före startdatumet.";
        this["Func_eomonth_Category"] = "DatumOchTid";
        this["Func_eomonth_CategoryURL"] = "DatumOchTid";
        this["Func_eomonth_Desc"] = "Returnerar ett datum som är i slutet av månaden, vilket är det angivna antalet månader före eller efter startdatumet.";
        this["Func_errortype_Arg_0"] = "error_value";
        this["Func_errortype_ArgDesc_0"] = "Felvärdet vars identifieringsnummer ska hittas.";
        this["Func_errortype_Category"] = "Information";
        this["Func_errortype_CategoryURL"] = "Information";
        this["Func_errortype_Desc"] = "Returnerar ett tal som motsvarar ett felvärde.";
        this["Func_even_Arg_0"] = "nummer";
        this["Func_even_ArgDesc_0"] = "Värdet som ska rundas.";
        this["Func_even_Category"] = "Matematik";
        this["Func_even_CategoryURL"] = "Matematik";
        this["Func_even_Desc"] = "Avrundar ett positivt tal uppåt och ett negativt tal ner till närmaste jämna heltal.";
        this["Func_exp_Arg_0"] = "nummer";
        this["Func_exp_ArgDesc_0"] = "Den tillämpade exponenten som e ska höjas till.";
        this["Func_exp_Category"] = "Matematik";
        this["Func_exp_CategoryURL"] = "Matematik";
        this["Func_exp_Desc"] = "Returnerar e höjt till styrkan för det angivna numret. Konstanten e är lika med 2,71828182845904.";
        this["Func_fact_Arg_0"] = "Nummer";
        this["Func_fact_ArgDesc_0"] = "Ett positivt tal för vilket fakulteten beräknas.";
        this["Func_fact_Category"] = "Matematik";
        this["Func_fact_CategoryURL"] = "Matematik";
        this["Func_fact_Desc"] = "Returnerar ett tals fakultet.";
        this["Func_factdouble_Arg_0"] = "Nummer";
        this["Func_factdouble_ArgDesc_0"] = "Ett positivt tal för vilket den dubbla fakulteten beräknas.";
        this["Func_factdouble_Category"] = "Matematik";
        this["Func_factdouble_CategoryURL"] = "Matematik";
        this["Func_factdouble_Desc"] = "Returnerar den dubbla fakulteten för ett nummer.";
        this["Func_false_Category"] = "Logisk";
        this["Func_false_CategoryURL"] = "Logisk";
        this["Func_false_Desc"] = "Returnerar det logiska värdet falskt.";
        this["Func_find_Arg_0"] = "find_text";
        this["Func_find_Arg_1"] = "within_text";
        this["Func_find_Arg_2"] = "start_num";
        this["Func_find_ArgDesc_0"] = "Texten att hitta.";
        this["Func_find_ArgDesc_1"] = "Texten som innehåller texten att hitta.";
        this["Func_find_ArgDesc_2"] = "Det 1-baserade indextecknet som sökningen ska starta på. Om den utelämnas antas den vara ett.";
        this["Func_find_Category"] = "TextAndData";
        this["Func_find_CategoryURL"] = "Text_and_Data";
        this["Func_find_Desc"] = "Returnerar startpositionen för ett textuttryck i ett annat, skiftlägeskänsligt.";
        this["Func_fixed_Arg_0"] = "nummer";
        this["Func_fixed_Arg_1"] = "decimals";
        this["Func_fixed_Arg_2"] = "no_commas";
        this["Func_fixed_ArgDesc_0"] = "Siffran som ska avrundas och konverteras till text.";
        this["Func_fixed_ArgDesc_1"] = "Antalet siffror till höger om decimalen i returvärdet. Om detta värde är negativt så avrundas \"nummer\" till vänster om decimaltecknet. Standardvärdet är 2.";
        this["Func_fixed_ArgDesc_2"] = "Sant om komma inte ska vara i returvärdet. Om detta utelämnas antas den vara falsk.";
        this["Func_fixed_Category"] = "TextAndData";
        this["Func_fixed_CategoryURL"] = "Text_and_Data";
        this["Func_fixed_Desc"] = "Avrundar ett tal till det angivna antalet decimaler och returnerar resultatet som text med eller utan komma.";
        this["Func_floor_Arg_0"] = "nummer";
        this["Func_floor_Arg_1"] = "betydelse";
        this["Func_floor_ArgDesc_0"] = "Siffran som ska avrundas.";
        this["Func_floor_ArgDesc_1"] = "Multipeln som du ska runda av. Om den utelämnas antas detta vara ett.";
        this["Func_floor_Category"] = "Matematik";
        this["Func_floor_CategoryURL"] = "Matematik";
        this["Func_floor_Desc"] = "Returnerar den största multipeln av signifikansen (standard 1) som är mindre än eller lika med det angivna numret.";
        this["Func_fv_Arg_0"] = "kurs";
        this["Func_fv_Arg_1"] = "nper";
        this["Func_fv_Arg_2"] = "pmt";
        this["Func_fv_Arg_3"] = "pv";
        this["Func_fv_Arg_4"] = "type";
        this["Func_fv_ArgDesc_0"] = "Räntan per period.";
        this["Func_fv_ArgDesc_1"] = "Det totala antalet betalningsperioder i en livränta.";
        this["Func_fv_ArgDesc_2"] = "Betalningen som utförts varje period, som inte kan förändras under livräntans livstid.";
        this["Func_fv_ArgDesc_3"] = "Nuvärdet, eller det schablonbelopp som en serie framtida betalningar är värda just nu. Om den utelämnas antas den vara noll.";
        this["Func_fv_ArgDesc_4"] = "Siffran noll eller en som anger om betalningar ska förfalla i slutet av perioden (noll, standardvärdet) eller början (en).";
        this["Func_fv_Category"] = "Finansiell";
        this["Func_fv_CategoryURL"] = "Finansiell";
        this["Func_fv_Desc"] = "Returnerar framtida värde för en investering baserat på periodiska.";
        this["Func_gcd_Arg_0"] = "nummer {0}";
        this["Func_gcd_ArgDesc_0"] = "Ett heltal. Ett icke-heltalsvärde kommer trunkeras.";
        this["Func_gcd_Category"] = "Matematik";
        this["Func_gcd_CategoryURL"] = "Matematik";
        this["Func_gcd_Desc"] = "Returnerar den största gemensamma delaren av ett eller flera heltal.";
        this["Func_gestep_Arg_0"] = "Number1";
        this["Func_gestep_Arg_1"] = "Steg";
        this["Func_gestep_ArgDesc_0"] = "Det första numret som ska jämföras.";
        this["Func_gestep_ArgDesc_1"] = "Det andra numret som ska jämföras. Om det inte anges antas det vara noll.";
        this["Func_gestep_Category"] = "Teknik";
        this["Func_gestep_CategoryURL"] = "Teknik";
        this["Func_gestep_Desc"] = "Jämför två nummer och returnerar ett om det första numret är större än eller lika med det andra eller returnerar noll om inte.";
        this["Func_hex2bin_Arg_0"] = "Nummer";
        this["Func_hex2bin_Arg_1"] = "Platser";
        this["Func_hex2bin_ArgDesc_0"] = "Ett textuttryck som representerar ett hexadecimalt tal.";
        this["Func_hex2bin_ArgDesc_1"] = "Antalet tecken som ska användas för att representera värdet. Om detta inte anges används det maximala antalet tecken som behövs för att representera värdet. Detta låter dig ange ledande nollor.";
        this["Func_hex2bin_Category"] = "Teknik";
        this["Func_hex2bin_CategoryURL"] = "Teknik";
        this["Func_hex2bin_Desc"] = "Konverterar ett textuttryck som representerar ett hexadecimalt tal till ett textuttryck som representerar samma nummer i det binära nummersystemet.";
        this["Func_hex2dec_Arg_0"] = "Nummer";
        this["Func_hex2dec_ArgDesc_0"] = "Ett textuttryck som representerar ett hexadecimalt tal.";
        this["Func_hex2dec_Category"] = "Teknik";
        this["Func_hex2dec_CategoryURL"] = "Teknik";
        this["Func_hex2dec_Desc"] = "Returnerar decimalvärdet för en textrepresentation av ett tal i det hexadecimala talsystemet.";
        this["Func_hex2oct_Arg_0"] = "Nummer";
        this["Func_hex2oct_Arg_1"] = "Platser";
        this["Func_hex2oct_ArgDesc_0"] = "Ett textuttryck som representerar ett hexadecimalt tal.";
        this["Func_hex2oct_ArgDesc_1"] = "Antalet tecken som ska användas för att representera värdet. Om detta inte anges används det maximala antalet tecken som behövs för att representera värdet. Detta låter dig ange ledande nollor.";
        this["Func_hex2oct_Category"] = "Teknik";
        this["Func_hex2oct_CategoryURL"] = "Teknik";
        this["Func_hex2oct_Desc"] = "Konverterar ett textuttryck som representerar ett hexadecimalt tal till ett textuttryck som representerar samma nummer i det oktala talsystemet.";
        this["Func_hour_Arg_0"] = "Time_value";
        this["Func_hour_ArgDesc_0"] = "Ett tidsvärde från vilket timmen ska hämtas.";
        this["Func_hour_Category"] = "DatumOchTid";
        this["Func_hour_CategoryURL"] = "Date_and_Time";
        this["Func_hour_Desc"] = "Returnerar timmen för ett tidsvärde.";
        this["Func_if_Arg_0"] = "logical_test";
        this["Func_if_Arg_1"] = "value_if_true";
        this["Func_if_Arg_2"] = "value_if_false";
        this["Func_if_ArgDesc_0"] = "Villkoret att testa.";
        this["Func_if_ArgDesc_1"] = "Resultatet om villkoret utvärderas till sant.";
        this["Func_if_ArgDesc_2"] = "Resultatet om villkoret utvärderas till falskt. Om detta utelämnas antas det logiska värdet vara falskt.";
        this["Func_if_Category"] = "Logisk";
        this["Func_if_CategoryURL"] = "Logisk";
        this["Func_if_Desc"] = "Returnerar ett värde om det angivna villkoret utvärderas till sant och ett annat värde om det utvärderas till falskt.";
        this["Func_iferror_Arg_0"] = "värde";
        this["Func_iferror_Arg_1"] = "value_if_error";
        this["Func_iferror_ArgDesc_0"] = "Värdet som ska användas när det inte är ett fel.";
        this["Func_iferror_ArgDesc_1"] = "Värdet som ska användas om värdeargumentet är ett fel.";
        this["Func_iferror_Category"] = "Logisk";
        this["Func_iferror_CategoryURL"] = "Logisk";
        this["Func_iferror_Desc"] = "Returnerar ett av två värden beroende på om det första värdet är ett fel eller inte.";
        this["Func_imabs_Arg_0"] = "ComplexNumber";
        this["Func_imabs_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imabs_Category"] = "Teknik";
        this["Func_imabs_CategoryURL"] = "Teknik";
        this["Func_imabs_Desc"] = "Returnerar det absoluta värdet för ett komplext tal.";
        this["Func_imaginary_Arg_0"] = "ComplexNumber";
        this["Func_imaginary_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imaginary_Category"] = "Teknik";
        this["Func_imaginary_CategoryURL"] = "Teknik";
        this["Func_imaginary_Desc"] = "Returnerar den imaginära koefficienten för ett komplext tal.";
        this["Func_imargument_Arg_0"] = "ComplexNumber";
        this["Func_imargument_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imargument_Category"] = "Teknik";
        this["Func_imargument_CategoryURL"] = "Teknik";
        this["Func_imargument_Desc"] = "Returnerar argumentet theta, en vinkel uttryckt i radianer.";
        this["Func_imconjugate_Arg_0"] = "ComplexNumber";
        this["Func_imconjugate_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imconjugate_Category"] = "Teknik";
        this["Func_imconjugate_CategoryURL"] = "Teknik";
        this["Func_imconjugate_Desc"] = "Returnerar konjugatet för ett komplext tal.";
        this["Func_imcos_Arg_0"] = "ComplexNumber";
        this["Func_imcos_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imcos_Category"] = "Teknik";
        this["Func_imcos_CategoryURL"] = "Teknik";
        this["Func_imcos_Desc"] = "Returnerar cosinus för ett komplext tal.";
        this["Func_imdiv_Arg_0"] = "Dividend";
        this["Func_imdiv_Arg_1"] = "divisor";
        this["Func_imdiv_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imdiv_ArgDesc_1"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imdiv_Category"] = "Teknik";
        this["Func_imdiv_CategoryURL"] = "Teknik";
        this["Func_imdiv_Desc"] = "Returnerar kvoten för två komplexa tal (Dividend / Divisor).";
        this["Func_imexp_Arg_0"] = "ComplexNumber";
        this["Func_imexp_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imexp_Category"] = "Teknik";
        this["Func_imexp_CategoryURL"] = "Teknik";
        this["Func_imexp_Desc"] = "Returnerar exponenten för ett komplext nummer.";
        this["Func_imln_Arg_0"] = "ComplexNumber";
        this["Func_imln_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imln_Category"] = "Teknik";
        this["Func_imln_CategoryURL"] = "Teknik";
        this["Func_imln_Desc"] = "Returnerar den naturliga logaritmen för ett komplext tal.";
        this["Func_imlog10_Arg_0"] = "ComplexNumber";
        this["Func_imlog10_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imlog10_Category"] = "Teknik";
        this["Func_imlog10_CategoryURL"] = "Teknik";
        this["Func_imlog10_Desc"] = "Returnerar bas-10-logaritmen för ett komplext tal.";
        this["Func_imlog2_Arg_0"] = "ComplexNumber";
        this["Func_imlog2_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imlog2_Category"] = "Teknik";
        this["Func_imlog2_CategoryURL"] = "Teknik";
        this["Func_imlog2_Desc"] = "Returnerar bas-2-logaritmen för ett komplext tal.";
        this["Func_impower_Arg_0"] = "ComplexNumber";
        this["Func_impower_Arg_1"] = "exponent";
        this["Func_impower_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_impower_ArgDesc_1"] = "Kraften som ComplexNumber kommer att höjas till.";
        this["Func_impower_Category"] = "Teknik";
        this["Func_impower_CategoryURL"] = "Teknik";
        this["Func_impower_Desc"] = "Returnerar det komplexa tal som höjs till den angivna effekten.";
        this["Func_improduct_Arg_0"] = "ComplexNumber {0}";
        this["Func_improduct_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_improduct_Category"] = "Teknik";
        this["Func_improduct_CategoryURL"] = "Teknik";
        this["Func_improduct_Desc"] = "Returnerar produkten med ett eller flera komplexa nummer.";
        this["Func_imreal_Arg_0"] = "ComplexNumber";
        this["Func_imreal_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imreal_Category"] = "Teknik";
        this["Func_imreal_CategoryURL"] = "Teknik";
        this["Func_imreal_Desc"] = "Returnerar den reella koefficienten för ett komplext tal.";
        this["Func_imsin_Arg_0"] = "ComplexNumber";
        this["Func_imsin_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imsin_Category"] = "Teknik";
        this["Func_imsin_CategoryURL"] = "Teknik";
        this["Func_imsin_Desc"] = "Returnerar sinus för ett komplext tal.";
        this["Func_imsqrt_Arg_0"] = "ComplexNumber";
        this["Func_imsqrt_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imsqrt_Category"] = "Teknik";
        this["Func_imsqrt_CategoryURL"] = "Teknik";
        this["Func_imsqrt_Desc"] = "Returnerar kvadratroten av ett komplext tal.";
        this["Func_imsub_Arg_0"] = "ComplexNumber1";
        this["Func_imsub_Arg_1"] = "ComplexNumber2";
        this["Func_imsub_ArgDesc_0"] = "Textrepresentationen för ett komplext tal i formatet: \"x + yi\" eller \"x + yj\" från vilket ComplexNumber2 kommer att subtraheras.";
        this["Func_imsub_ArgDesc_1"] = "Textrepresentationen för ett komplext tal i formatet: \"x + yi\" eller \"x + yj\" som kommer att subtraheras från ComplexNumber1.";
        this["Func_imsub_Category"] = "Teknik";
        this["Func_imsub_CategoryURL"] = "Teknik";
        this["Func_imsub_Desc"] = "Returnerar skillnaden mellan två komplexa tal.";
        this["Func_imsum_Arg_0"] = "ComplexNumber {0}";
        this["Func_imsum_ArgDesc_0"] = "Textrepresentation av ett komplext tal i formatet: \"x + yi\" eller \"x + yj\".";
        this["Func_imsum_Category"] = "Teknik";
        this["Func_imsum_CategoryURL"] = "Teknik";
        this["Func_imsum_Desc"] = "Returnerar summan av ett eller flera komplexa tal.";
        this["Func_info_Arg_0"] = "Typ";
        this["Func_info_ArgDesc_0"] = "Ett textuttryck som anger vilken typ av information som ska returneras.\r\n\"directory\" = Hela sökvägen till den aktuella mappen.\r\n\"osversion\" = Den nuvarande operativsystemversionen som ett textuttryck.\r\n\"system\" = Det aktuella operativsystemet. Macintosh = \"mac\", Windows = \"pcdos\".\r\n    ";
        this["Func_info_Category"] = "Information";
        this["Func_info_CategoryURL"] = "Information";
        this["Func_info_Desc"] = "Returnerar information om den aktuella driftsmiljön.";
        this["Func_int_Arg_0"] = "nummer";
        this["Func_int_ArgDesc_0"] = "Det verkliga talet som ska avrundas till ett heltal.";
        this["Func_int_Category"] = "Matematik";
        this["Func_int_CategoryURL"] = "Matematik";
        this["Func_int_Desc"] = "Avrundar ett tal till närmaste heltal.";
        this["Func_intrate_Arg_0"] = "lösning";
        this["Func_intrate_Arg_1"] = "mognad";
        this["Func_intrate_Arg_2"] = "investering";
        this["Func_intrate_Arg_3"] = "upprättelse";
        this["Func_intrate_Arg_4"] = "grund";
        this["Func_intrate_ArgDesc_0"] = "Säkerhetens avvecklingsdatum.";
        this["Func_intrate_ArgDesc_1"] = "Säkerhetens förfallodatum.";
        this["Func_intrate_ArgDesc_2"] = "Beloppet som investerats i värdepapperet.";
        this["Func_intrate_ArgDesc_3"] = "Det belopp som ska tas emot vid förfallodagen.";
        this["Func_intrate_ArgDesc_4"] = "Den typ av dagräkning som ska användas.\r\n0 eller utelämnad = US (NASD) 30/360,\r\n1 = Faktiskt / faktiskt,\r\n2 = Faktiskt / 360,\r\n3 = Faktiskt / 365,\r\n4 = Europeisk 30/360\r\n    ";
        this["Func_intrate_Category"] = "Finansiell";
        this["Func_intrate_CategoryURL"] = "Finansiell";
        this["Func_intrate_Desc"] = "Returnerar räntan för ett helt investerat värdepapper.";
        this["Func_ipmt_Arg_0"] = "kurs";
        this["Func_ipmt_Arg_1"] = "per";
        this["Func_ipmt_Arg_2"] = "nper";
        this["Func_ipmt_Arg_3"] = "pv";
        this["Func_ipmt_Arg_4"] = "fv";
        this["Func_ipmt_Arg_5"] = "type";
        this["Func_ipmt_ArgDesc_0"] = "Räntan per period.";
        this["Func_ipmt_ArgDesc_1"] = "Perioden, som måste ligga i intervallet en till nper.";
        this["Func_ipmt_ArgDesc_2"] = "Det totala antalet betalningsperioder i en livränta.";
        this["Func_ipmt_ArgDesc_3"] = "Nuvärdet, eller det engångsbelopp som en serie framtida betalningar är värda nu.";
        this["Func_ipmt_ArgDesc_4"] = "Det framtida värdet eller ett kontantsaldo som du vill uppnå efter den senaste betalningen. Om den utelämnas antas den vara noll.";
        this["Func_ipmt_ArgDesc_5"] = "Siffran noll eller en som anger om betalningar ska förfalla i slutet av perioden (noll, standardvärdet) eller början (en).";
        this["Func_ipmt_Category"] = "Finansiell";
        this["Func_ipmt_CategoryURL"] = "Finansiell";
        this["Func_ipmt_Desc"] = "Returnerar räntebetalningen för en viss period för en investering, baserat på periodiska, konstanta betalningar och en konstant ränta.";
        this["Func_irr_Arg_0"] = "värden";
        this["Func_irr_Arg_1"] = "gissa";
        this["Func_irr_ArgDesc_0"] = "En samling betalningsvärden (negativa värden) och inkomst (positiva värden) som sker med jämna mellanrum.";
        this["Func_irr_ArgDesc_1"] = "Ett nummer som du antar ligger nära resultatet. Standardvärdet är 0,1 (10%).";
        this["Func_irr_Category"] = "Finansiell";
        this["Func_irr_CategoryURL"] = "Finansiell";
        this["Func_irr_Desc"] = "Returnerar den interna avkastningen för en serie kassaflöden.";
        this["Func_isblank_Arg_0"] = "värde";
        this["Func_isblank_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_isblank_Category"] = "Information";
        this["Func_isblank_CategoryURL"] = "Information";
        this["Func_isblank_Desc"] = "Returnerar sant om värdet är tomt.";
        this["Func_isdbnull_Arg_0"] = "värde";
        this["Func_isdbnull_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_isdbnull_Category"] = "Information";
        this["Func_isdbnull_CategoryURL"] = "Information";
        this["Func_isdbnull_Desc"] = "Returnerar true om värdet representerar ett obefintligt värde.";
        this["Func_iserr_Arg_0"] = "värde";
        this["Func_iserr_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_iserr_Category"] = "Information";
        this["Func_iserr_CategoryURL"] = "Information";
        this["Func_iserr_Desc"] = "Returnerar sant om värdet är något felvärde förutom # N / A.";
        this["Func_iserror_Arg_0"] = "värde";
        this["Func_iserror_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_iserror_Category"] = "Information";
        this["Func_iserror_CategoryURL"] = "Information";
        this["Func_iserror_Desc"] = "Returnerar sant om värdet är något felvärde.";
        this["Func_iseven_Arg_0"] = "värde";
        this["Func_iseven_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_iseven_Category"] = "Information";
        this["Func_iseven_CategoryURL"] = "Information";
        this["Func_iseven_Desc"] = "Returnerar sant om värdet är ett jämnt tal, annars returnerar det falskt.";
        this["Func_islogical_Arg_0"] = "värde";
        this["Func_islogical_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_islogical_Category"] = "Information";
        this["Func_islogical_CategoryURL"] = "Information";
        this["Func_islogical_Desc"] = "Returnerar sant om värdet är ett logiskt värde (true eller false).";
        this["Func_isna_Arg_0"] = "värde";
        this["Func_isna_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_isna_Category"] = "Information";
        this["Func_isna_CategoryURL"] = "Information";
        this["Func_isna_Desc"] = "Returnerar sant om värdet är felvärdet # N / A (värde ej tillgängligt).";
        this["Func_isnontext_Arg_0"] = "värde";
        this["Func_isnontext_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_isnontext_Category"] = "Information";
        this["Func_isnontext_CategoryURL"] = "Information";
        this["Func_isnontext_Desc"] = "Returnerar sant om värdet är ett objekt som inte är text.";
        this["Func_isnull_Arg_0"] = "värde";
        this["Func_isnull_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_isnull_Category"] = "Information";
        this["Func_isnull_CategoryURL"] = "Information";
        this["Func_isnull_Desc"] = "Returnerar true om värdet är obefintligt.";
        this["Func_isnumber_Arg_0"] = "värde";
        this["Func_isnumber_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_isnumber_Category"] = "Information";
        this["Func_isnumber_CategoryURL"] = "Information";
        this["Func_isnumber_Desc"] = "Returnerar sant om värdet är ett tal.";
        this["Func_isodd_Arg_0"] = "värde";
        this["Func_isodd_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_isodd_Category"] = "Information";
        this["Func_isodd_CategoryURL"] = "Information";
        this["Func_isodd_Desc"] = "Returnerar sant om värdet är ett udda tal, annars returnerar det falskt.";
        this["Func_isref_Arg_0"] = "värde";
        this["Func_isref_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_isref_Category"] = "Information";
        this["Func_isref_CategoryURL"] = "Information";
        this["Func_isref_Desc"] = "Returnerar sant om värdet är en referens.";
        this["Func_istext_Arg_0"] = "värde";
        this["Func_istext_ArgDesc_0"] = "Värdet som ska testas.";
        this["Func_istext_Category"] = "Information";
        this["Func_istext_CategoryURL"] = "Information";
        this["Func_istext_Desc"] = "Returnerar sant om värdet är ett textvärde.";
        this["Func_lcm_Arg_0"] = "nummer {0}";
        this["Func_lcm_ArgDesc_0"] = "Ett heltal. Ett icke-heltalsvärde kommer trunkeras.";
        this["Func_lcm_Category"] = "Matematik";
        this["Func_lcm_CategoryURL"] = "Matematik";
        this["Func_lcm_Desc"] = "Returnerar den minst vanliga multipeln av ett eller flera heltal.";
        this["Func_left_Arg_0"] = "text";
        this["Func_left_Arg_1"] = "num_chars";
        this["Func_left_ArgDesc_0"] = "Textuttrycket som innehåller de tecken som ska extraheras.";
        this["Func_left_ArgDesc_1"] = "Antalet tecken som ska extraheras.";
        this["Func_left_Category"] = "TextAndData";
        this["Func_left_CategoryURL"] = "Text_and_Data";
        this["Func_left_Desc"] = "Returnerar det första tecknet eller tecknen i ett textuttryck.";
        this["Func_len_Arg_0"] = "text";
        this["Func_len_ArgDesc_0"] = "Textuttrycket vars längd ska hittas. Mellanslag räknas som tecken.";
        this["Func_len_Category"] = "TextAndData";
        this["Func_len_CategoryURL"] = "Text_and_Data";
        this["Func_len_Desc"] = "Returnerar antalet tecken i ett textuttryck.";
        this["Func_ln_Arg_0"] = "nummer";
        this["Func_ln_ArgDesc_0"] = "Det positiva reella talet för vilket man får den naturliga logaritmen.";
        this["Func_ln_Category"] = "Matematik";
        this["Func_ln_CategoryURL"] = "Matematik";
        this["Func_ln_Desc"] = "Returnerar den naturliga logaritmen för ett tal.";
        this["Func_log_Arg_0"] = "nummer";
        this["Func_log_Arg_1"] = "bas";
        this["Func_log_ArgDesc_0"] = "Det positiva reella talet för vilket logaritmen ska hittas.";
        this["Func_log_ArgDesc_1"] = "Logaritmens bas.";
        this["Func_log_Category"] = "Matematik";
        this["Func_log_CategoryURL"] = "Matematik";
        this["Func_log_Desc"] = "Returnerar logaritmen för ett tal till den angivna basen.";
        this["Func_log10_Arg_0"] = "nummer";
        this["Func_log10_ArgDesc_0"] = "Det positiva reella talet för vilket bas-10-logaritmen ska hittas.";
        this["Func_log10_Category"] = "Matematik";
        this["Func_log10_CategoryURL"] = "Matematik";
        this["Func_log10_Desc"] = "Returnerar bas-10-logaritmen för ett tal.";
        this["Func_lower_Arg_0"] = "text";
        this["Func_lower_ArgDesc_0"] = "Texten som ska konverteras till gemener.";
        this["Func_lower_Category"] = "TextAndData";
        this["Func_lower_CategoryURL"] = "Text_and_Data";
        this["Func_lower_Desc"] = "Konverterar alla versaler i ett textuttryck till gemener.";
        this["Func_max_Arg_0"] = "nummer {0}";
        this["Func_max_ArgDesc_0"] = "Ett antal eller intervall av siffror.";
        this["Func_max_Category"] = "Statistisk";
        this["Func_max_CategoryURL"] = "Statistisk";
        this["Func_max_Desc"] = "Returnerar det största numret i en uppsättning siffror.";
        this["Func_median_Arg_0"] = "nummer {0}";
        this["Func_median_ArgDesc_0"] = "Ett antal eller intervall av siffror.";
        this["Func_median_Category"] = "Statistisk";
        this["Func_median_CategoryURL"] = "Statistisk";
        this["Func_median_Desc"] = "Returnerar medianen för en uppsättning siffror. Medianen är mittantalet när siffrorna sorteras eller medelvärdet av de två mellannumren om det finns ett jämnt antal angivna nummer.";
        this["Func_mid_Arg_0"] = "text";
        this["Func_mid_Arg_1"] = "start_num";
        this["Func_mid_Arg_2"] = "num_chars";
        this["Func_mid_ArgDesc_0"] = "Textuttrycket som innehåller de tecken som ska extraheras.";
        this["Func_mid_ArgDesc_1"] = "Det 1-baserade indexet för det första tecknet som extraheras.";
        this["Func_mid_ArgDesc_2"] = "Antalet tecken som ska extraheras.";
        this["Func_mid_Category"] = "TextAndData";
        this["Func_mid_CategoryURL"] = "Text_and_Data";
        this["Func_mid_Desc"] = "Returnerar ett visst antal tecken från ett textuttryck.";
        this["Func_min_Arg_0"] = "nummer {0}";
        this["Func_min_ArgDesc_0"] = "Ett antal eller intervall av siffror.";
        this["Func_min_Category"] = "Statistisk";
        this["Func_min_CategoryURL"] = "Statistisk";
        this["Func_min_Desc"] = "Returnerar det minsta numret i en uppsättning siffror.";
        this["Func_minute_Arg_0"] = "Time_value";
        this["Func_minute_ArgDesc_0"] = "Ett tidsvärde från vilket minuten ska hämtas.";
        this["Func_minute_Category"] = "DatumOchTid";
        this["Func_minute_CategoryURL"] = "Date_and_Time";
        this["Func_minute_Desc"] = "Returnerar minuten för ett tidsvärde.";
        this["Func_mod_Arg_0"] = "nummer";
        this["Func_mod_Arg_1"] = "divisor";
        this["Func_mod_ArgDesc_0"] = "Numret för vilket resten ska hittas.";
        this["Func_mod_ArgDesc_1"] = "Värdet som talet ska delas med.";
        this["Func_mod_Category"] = "Matematik";
        this["Func_mod_CategoryURL"] = "Matematik";
        this["Func_mod_Desc"] = "Returnerar resten efter att ett nummer har delats med en delare. Resultatet har samma tecken som delaren.";
        this["Func_month_Arg_0"] = "Date_value";
        this["Func_month_ArgDesc_0"] = "Ett datumvärde från vilket månaden ska hämtas.";
        this["Func_month_Category"] = "DatumOchTid";
        this["Func_month_CategoryURL"] = "Date_and_Time";
        this["Func_month_Desc"] = "Returnerar månaden för ett datumvärde.";
        this["Func_mround_Arg_0"] = "Nummer";
        this["Func_mround_Arg_1"] = "Flera olika";
        this["Func_mround_ArgDesc_0"] = "Antalet som ska avrundas.";
        this["Func_mround_ArgDesc_1"] = "Ett tal som anger multipeln som du ska runda av.";
        this["Func_mround_Category"] = "Matematik";
        this["Func_mround_CategoryURL"] = "Matematik";
        this["Func_mround_Desc"] = "Avrundar ett tal till närmaste multipel av ett annat nummer.";
        this["Func_multinomial_Arg_0"] = "nummer {0}";
        this["Func_multinomial_ArgDesc_0"] = "Ett av siffrorna för vilka multinomialen ska hittas.";
        this["Func_multinomial_Category"] = "Matematik";
        this["Func_multinomial_CategoryURL"] = "Matematik";
        this["Func_multinomial_Desc"] = "Returnerar multinomialen för en uppsättning siffror. Multinomialen är förhållandet mellan fakulteten av en summa av värden och produkten av deras fakulteter.";
        this["Func_n_Arg_0"] = "Värde";
        this["Func_n_ArgDesc_0"] = "Värdet som ska konverteras till ett tal. Följande är acceptabla värden:\r\nEtt nummer - returnerar numret.\r\nEtt datum - returnerar datumet.\r\nSant - returnerar 1.\r\nFalsk - returnerar 0.\r\nEtt felvärde - returnerar felvärdet.\r\nAlla andra värden - returnerar 0.\r\n    ";
        this["Func_n_Category"] = "Information";
        this["Func_n_CategoryURL"] = "Information";
        this["Func_n_Desc"] = "Konverterar ett värde till ett tal.";
        this["Func_na_Category"] = "Information";
        this["Func_na_CategoryURL"] = "Information";
        this["Func_na_Desc"] = "Returnerar felvärdet # N / A (värde ej tillgängligt).";
        this["Func_networkdays_Arg_0"] = "startDate";
        this["Func_networkdays_Arg_1"] = "endDate";
        this["Func_networkdays_Arg_2"] = "högtider";
        this["Func_networkdays_ArgDesc_0"] = "Datumet från vilket du ska börja.";
        this["Func_networkdays_ArgDesc_1"] = "Datumet då det ska avslutas.";
        this["Func_networkdays_ArgDesc_2"] = "Ett intervall med en eller flera referenser som innehåller datum som ska betraktas som icke-arbetsdagar.";
        this["Func_networkdays_Category"] = "DatumOchTid";
        this["Func_networkdays_CategoryURL"] = "DatumOchTid";
        this["Func_networkdays_Desc"] = "Returnerar det totala antalet hela arbetsdagar mellan de angivna datumen, exklusive angivna helgdagar.";
        this["Func_not_Arg_0"] = "Logisk";
        this["Func_not_ArgDesc_0"] = "Ett logiskt värde (sant eller falskt).";
        this["Func_not_Category"] = "Logisk";
        this["Func_not_CategoryURL"] = "Logisk";
        this["Func_not_Desc"] = "Omvandlar ett logiskt värde (ändrar sant till falskt och falskt till sant).";
        this["Func_now_Category"] = "DatumOchTid";
        this["Func_now_CategoryURL"] = "Date_and_Time";
        this["Func_now_Desc"] = "Returnerar ett värde som representerar aktuellt datum och tid.";
        this["Func_nper_Arg_0"] = "kurs";
        this["Func_nper_Arg_1"] = "pmt";
        this["Func_nper_Arg_2"] = "pv";
        this["Func_nper_Arg_3"] = "fv";
        this["Func_nper_Arg_4"] = "type";
        this["Func_nper_ArgDesc_0"] = "Räntan per period.";
        this["Func_nper_ArgDesc_1"] = "Betalningen som utförts varje period, som inte kan förändras under livräntans livstid.";
        this["Func_nper_ArgDesc_2"] = "Nuvärdet, eller det schablonbelopp som en serie framtida betalningar är värda just nu.";
        this["Func_nper_ArgDesc_3"] = "Det framtida värdet eller ett kontantsaldo som du vill uppnå efter den senaste betalningen.";
        this["Func_nper_ArgDesc_4"] = "Siffran noll eller en som anger om betalningar ska förfalla i slutet av perioden (noll, standardvärdet) eller början (en).";
        this["Func_nper_Category"] = "Finansiell";
        this["Func_nper_CategoryURL"] = "Finansiell";
        this["Func_nper_Desc"] = "Returnerar antalet perioder för en investering baserat på periodiska, konstanta betalningar och en konstant ränta.";
        this["Func_npv_Arg_0"] = "kurs";
        this["Func_npv_Arg_1"] = "värde {0}";
        this["Func_npv_ArgDesc_0"] = "Rabatten över en period.";
        this["Func_npv_ArgDesc_1"] = "En betalning (negativt värde) eller inkomst (positivt värde).";
        this["Func_npv_Category"] = "Finansiell";
        this["Func_npv_CategoryURL"] = "Finansiell";
        this["Func_npv_Desc"] = "Beräknar nuvärdet av en investering med hjälp av en diskonteringsränta och en serie framtida betalningar (negativa värden) och inkomst (positivt värde).";
        this["Func_null_Category"] = "Information";
        this["Func_null_CategoryURL"] = "Information";
        this["Func_null_Desc"] = "Returnerar ett obefintligt värde.";
        this["Func_oct2bin_Arg_0"] = "Nummer";
        this["Func_oct2bin_Arg_1"] = "Platser";
        this["Func_oct2bin_ArgDesc_0"] = "Ett textuttryck som representerar ett oktalt tal.";
        this["Func_oct2bin_ArgDesc_1"] = "Antalet tecken som ska användas för att representera värdet. Om detta inte anges används det maximala antalet tecken som behövs för att representera värdet. Detta låter dig ange ledande nollor.";
        this["Func_oct2bin_Category"] = "Teknik";
        this["Func_oct2bin_CategoryURL"] = "Teknik";
        this["Func_oct2bin_Desc"] = "Konverterar ett textuttryck som representerar ett oktalt tal till ett textuttryck som representerar samma nummer i det binära nummersystemet.";
        this["Func_oct2dec_Arg_0"] = "Nummer";
        this["Func_oct2dec_ArgDesc_0"] = "Ett textuttryck som representerar ett oktalt tal.";
        this["Func_oct2dec_Category"] = "Teknik";
        this["Func_oct2dec_CategoryURL"] = "Teknik";
        this["Func_oct2dec_Desc"] = "Returnerar decimalvärdet för en textrepresentation av ett tal i det oktala talsystemet.";
        this["Func_oct2hex_Arg_0"] = "Nummer";
        this["Func_oct2hex_Arg_1"] = "Platser";
        this["Func_oct2hex_ArgDesc_0"] = "Ett textuttryck som representerar ett oktalt tal.";
        this["Func_oct2hex_ArgDesc_1"] = "Antalet tecken som ska användas för att representera värdet. Om detta inte anges används det maximala antalet tecken som behövs för att representera värdet. Detta låter dig ange ledande nollor.";
        this["Func_oct2hex_Category"] = "Teknik";
        this["Func_oct2hex_CategoryURL"] = "Teknik";
        this["Func_oct2hex_Desc"] = "Konverterar ett textuttryck som representerar ett oktalt tal till ett textuttryck som representerar samma nummer i det hexadecimala talsystemet.";
        this["Func_odd_Arg_0"] = "nummer";
        this["Func_odd_ArgDesc_0"] = "Värdet som ska rundas.";
        this["Func_odd_Category"] = "Matematik";
        this["Func_odd_CategoryURL"] = "Matematik";
        this["Func_odd_Desc"] = "Avrundar ett positivt tal uppåt och ett negativt tal till närmaste udda heltal.";
        this["Func_or_Arg_0"] = "logisk {0}";
        this["Func_or_ArgDesc_0"] = "Ett logiskt värde (sant eller falskt).";
        this["Func_or_Category"] = "Logisk";
        this["Func_or_CategoryURL"] = "Logisk";
        this["Func_or_Desc"] = "Returnerar sant om ett eller flera värden är sanna; returnerar falskt om alla värden är falska.";
        this["Func_pi_Category"] = "Matematik";
        this["Func_pi_CategoryURL"] = "Matematik";
        this["Func_pi_Desc"] = "Returnerar konstanten pi, som är 3.14159265358979.";
        this["Func_pmt_Arg_0"] = "kurs";
        this["Func_pmt_Arg_1"] = "nper";
        this["Func_pmt_Arg_2"] = "pv";
        this["Func_pmt_Arg_3"] = "fv";
        this["Func_pmt_Arg_4"] = "type";
        this["Func_pmt_ArgDesc_0"] = "Räntesatsen för lånet.";
        this["Func_pmt_ArgDesc_1"] = "Det totala antalet betalningar för lånet.";
        this["Func_pmt_ArgDesc_2"] = "Nuvärdet, eller det totala belopp som en serie av framtida betalningar är värda nu; även känd som rektor.";
        this["Func_pmt_ArgDesc_3"] = "Det framtida värdet eller ett kontantsaldo som du vill uppnå efter den senaste betalningen.";
        this["Func_pmt_ArgDesc_4"] = "Siffran noll eller en som anger om betalningar ska förfalla i slutet av perioden (noll, standardvärdet) eller början (en).";
        this["Func_pmt_Category"] = "Finansiell";
        this["Func_pmt_CategoryURL"] = "Finansiell";
        this["Func_pmt_Desc"] = "Beräknar betalningen för ett lån baserat på konstanta betalningar och en konstant ränta.";
        this["Func_power_Arg_0"] = "nummer";
        this["Func_power_Arg_1"] = "exponent";
        this["Func_power_ArgDesc_0"] = "Basnumret som ska höjas till en exponent.";
        this["Func_power_ArgDesc_1"] = "Exponenten till vilken basnumret höjs.";
        this["Func_power_Category"] = "Matematik";
        this["Func_power_CategoryURL"] = "Matematik";
        this["Func_power_Desc"] = "Returnerar resultatet av ett tal som höjs till en effekt.";
        this["Func_ppmt_Arg_0"] = "kurs";
        this["Func_ppmt_Arg_1"] = "per";
        this["Func_ppmt_Arg_2"] = "nper";
        this["Func_ppmt_Arg_3"] = "pv";
        this["Func_ppmt_Arg_4"] = "fv";
        this["Func_ppmt_Arg_5"] = "type";
        this["Func_ppmt_ArgDesc_0"] = "Räntan per period.";
        this["Func_ppmt_ArgDesc_1"] = "Perioden, som måste ligga i intervallet en till nper.";
        this["Func_ppmt_ArgDesc_2"] = "Det totala antalet betalningsperioder i en livränta.";
        this["Func_ppmt_ArgDesc_3"] = "Nuvärdet, eller det totala belopp som en serie av framtida betalningar är värda nu.";
        this["Func_ppmt_ArgDesc_4"] = "Det framtida värdet eller ett kontantsaldo som du vill uppnå efter den senaste betalningen.";
        this["Func_ppmt_ArgDesc_5"] = "Siffran noll eller en som anger om betalningar ska förfalla i slutet av perioden (noll, standardvärdet) eller början (en).";
        this["Func_ppmt_Category"] = "Finansiell";
        this["Func_ppmt_CategoryURL"] = "Finansiell";
        this["Func_ppmt_Desc"] = "Returnerar betalningen på huvudbeloppet för en viss period för en investering baserad på periodiska, konstanta betalningar och en konstant ränta.";
        this["Func_product_Arg_0"] = "nummer {0}";
        this["Func_product_ArgDesc_0"] = "Ett nummer som ska multipliceras med de andra siffrorna.";
        this["Func_product_Category"] = "Matematik";
        this["Func_product_CategoryURL"] = "Matematik";
        this["Func_product_Desc"] = "Multiplicerar alla siffror som ges som argument och returnerar produkten.";
        this["Func_pv_Arg_0"] = "kurs";
        this["Func_pv_Arg_1"] = "nper";
        this["Func_pv_Arg_2"] = "pmt";
        this["Func_pv_Arg_3"] = "fv";
        this["Func_pv_Arg_4"] = "type";
        this["Func_pv_ArgDesc_0"] = "Räntan per period.";
        this["Func_pv_ArgDesc_1"] = "Det totala antalet betalningsperioder i en livränta.";
        this["Func_pv_ArgDesc_2"] = "Betalningen som utförts varje period, som inte kan förändras under livräntans livstid.";
        this["Func_pv_ArgDesc_3"] = "Det framtida värdet eller ett kontantsaldo som du vill uppnå efter den senaste betalningen.";
        this["Func_pv_ArgDesc_4"] = "Siffran noll eller en som anger om betalningar ska förfalla i slutet av perioden (noll, standardvärdet) eller början (en).";
        this["Func_pv_Category"] = "Finansiell";
        this["Func_pv_CategoryURL"] = "Finansiell";
        this["Func_pv_Desc"] = "Returnerar nuvärdet av en investering.";
        this["Func_quotient_Arg_0"] = "täljare";
        this["Func_quotient_Arg_1"] = "nämnare";
        this["Func_quotient_ArgDesc_0"] = "Utdelningen för divisionsoperationen.";
        this["Func_quotient_ArgDesc_1"] = "Delaren av divisionsoperationen.";
        this["Func_quotient_Category"] = "Matematik";
        this["Func_quotient_CategoryURL"] = "Matematik";
        this["Func_quotient_Desc"] = "Returnerar den delen av en division som är ett heltal. Detta kommer att ignorera resten av divisionen.";
        this["Func_radians_Arg_0"] = "Grader";
        this["Func_radians_ArgDesc_0"] = "Ett värde i grader som omvandlas till radianer.";
        this["Func_radians_Category"] = "Matematik";
        this["Func_radians_CategoryURL"] = "Matematik";
        this["Func_radians_Desc"] = "Omvandlar grader till radianer.";
        this["Func_rand_Category"] = "Matematik";
        this["Func_rand_CategoryURL"] = "Matematik";
        this["Func_rand_Desc"] = "Returnerar ett jämnt fördelat slumptal som är större än eller lika med 0 och mindre än 1";
        this["Func_randbetween_Arg_0"] = "Botten";
        this["Func_randbetween_Arg_1"] = "Topp";
        this["Func_randbetween_ArgDesc_0"] = "Minimivärdet som kommer att returneras.";
        this["Func_randbetween_ArgDesc_1"] = "Det maximala värdet som kommer att returneras.";
        this["Func_randbetween_Category"] = "Matematik";
        this["Func_randbetween_CategoryURL"] = "Matematik";
        this["Func_randbetween_Desc"] = "Skapar ett pseudoslumpheltal mellan två angivna tal.";
        this["Func_rate_Arg_0"] = "NPer";
        this["Func_rate_Arg_1"] = "Pmt";
        this["Func_rate_Arg_2"] = "Pv";
        this["Func_rate_Arg_3"] = "Fv";
        this["Func_rate_Arg_4"] = "Typ";
        this["Func_rate_Arg_5"] = "Guess";
        this["Func_rate_ArgDesc_0"] = "Det totala antalet betalningsperioder i en livränta.";
        this["Func_rate_ArgDesc_1"] = "Betalningen gjordes varje period och kan inte ändras under livräntans livstid.";
        this["Func_rate_ArgDesc_2"] = "Nuvärdet, eller det totala belopp som en serie av framtida betalningar är värda nu.";
        this["Func_rate_ArgDesc_3"] = "Det framtida värdet eller ett kontantsaldo som du vill uppnå efter den senaste betalningen. Om fv utelämnas antas det vara 0 (det framtida värdet på ett lån är till exempel 0). Om du till exempel vill spara 50 000 dollar för att betala för ett specialprojekt om 18 år är 50 000 dollar det framtida värdet. Du kan sedan göra en konservativ gissning till en ränta och bestämma hur mycket du måste spara varje månad. Om fv utelämnas måste du inkludera argumentet pmt. Standardvärdet för fv är noll.";
        this["Func_rate_ArgDesc_4"] = "Siffran noll eller en som anger om betalningar ska förfalla i slutet av perioden (noll, standardvärdet) eller början (en).";
        this["Func_rate_ArgDesc_5"] = "Ett nummer som du antar ligger nära resultatet. Om den utelämnas antas den vara 0,1 (10%).";
        this["Func_rate_Category"] = "Finansiell";
        this["Func_rate_CategoryURL"] = "Finansiell";
        this["Func_rate_Desc"] = "Returnerar räntan per livränta.";
        this["Func_replace_Arg_0"] = "Old_text";
        this["Func_replace_Arg_1"] = "Start_num";
        this["Func_replace_Arg_2"] = "Num_chars";
        this["Func_replace_Arg_3"] = "New_text";
        this["Func_replace_ArgDesc_0"] = "Textuttrycket där vissa tecken ska ersättas.";
        this["Func_replace_ArgDesc_1"] = "Det 1-baserade indexet för tecknet i old_text som ska ersättas med new_text.";
        this["Func_replace_ArgDesc_2"] = "Antalet tecken i old_text som ska ersättas med new_text.";
        this["Func_replace_ArgDesc_3"] = "Texten som ersätter tecknen i old_text.";
        this["Func_replace_Category"] = "TextAndData";
        this["Func_replace_CategoryURL"] = "Text_and_Data";
        this["Func_replace_Desc"] = "Ersätter en del av ett textuttryck.";
        this["Func_rept_Arg_0"] = "TextValue";
        this["Func_rept_Arg_1"] = "RepeatCount";
        this["Func_rept_ArgDesc_0"] = "Texten som ska upprepas.";
        this["Func_rept_ArgDesc_1"] = "Det positiva antalet gånger som texten ska upprepas.";
        this["Func_rept_Category"] = "TextAndData";
        this["Func_rept_CategoryURL"] = "Text_and_Data";
        this["Func_rept_Desc"] = "Upprepar text ett visst antal gånger.";
        this["Func_right_Arg_0"] = "text";
        this["Func_right_Arg_1"] = "num_chars";
        this["Func_right_ArgDesc_0"] = "Textuttrycket som innehåller de tecken som ska extraheras.";
        this["Func_right_ArgDesc_1"] = "Antalet tecken som ska extraheras.";
        this["Func_right_Category"] = "TextAndData";
        this["Func_right_CategoryURL"] = "Text_and_Data";
        this["Func_right_Desc"] = "Returnerar det eller de sista tecknen i ett textuttryck.";
        this["Func_roman_Arg_0"] = "Nummer";
        this["Func_roman_Arg_1"] = "Form";
        this["Func_roman_ArgDesc_0"] = "Numret som ska konverteras till romerska siffror.";
        this["Func_roman_ArgDesc_1"] = "Ett tal eller booleskt värde som anger om man ska använda klassiska romerska siffror eller en mer kortfattad version.\r\n0 = (standard) klassisk. (499 = \" CDXCIX \")\r\n1 = Mer kortfattat. (499 = \" LDVLIV \")\r\n2 = Mer kortfattat. (499 = \" XDIX \")\r\n3 = Mer kortfattat. (499 = \" VDIV \")\r\n4 = förenklad. (499 = \" ID \")\r\nSant = Klassisk.\r\nFalskt = Förenklad.\r\n    ";
        this["Func_roman_Category"] = "Matematik";
        this["Func_roman_CategoryURL"] = "Matematik";
        this["Func_roman_Desc"] = "Konverterar ett tal till ett romerskt nummer som ett textuttryck.";
        this["Func_round_Arg_0"] = "nummer";
        this["Func_round_Arg_1"] = "antal_siffror";
        this["Func_round_Arg_2"] = "round_Style";
        this["Func_round_ArgDesc_0"] = "Siffran som ska avrundas.";
        this["Func_round_ArgDesc_1"] = "Antalet siffror som ska rundas av.";
        this["Func_round_ArgDesc_2"] = "Ett nummer som anger vilken avrundningsstil som ska användas.\r\n0 - Aritmetisk avrundning (standard) - Om den sista signifikanta siffran i avrundningen är 5, avrundas numret uppåt.\r\n1 - Bankmansavrundning - När den sista signifikanta siffran är 5 används föregående siffra för att avgöra om den ska rundas upp eller ner. Om siffran är jämn avrundas numret. Om det är udda, avrundas numret uppåt.\r\n    ";
        this["Func_round_Category"] = "Matematik";
        this["Func_round_CategoryURL"] = "Matematik";
        this["Func_round_Desc"] = "Avrundar ett nummer till ett angivet antal siffror.";
        this["Func_rounddown_Arg_0"] = "Nummer";
        this["Func_rounddown_Arg_1"] = "Siffror";
        this["Func_rounddown_ArgDesc_0"] = "Siffran som ska avrundas nedåt.";
        this["Func_rounddown_ArgDesc_1"] = "Antalet decimaler som ska avrundas. Positiva siffror indikerar platser efter decimaltecken, negativa siffror indikerar platser före decimalpunkten.";
        this["Func_rounddown_Category"] = "Matematik";
        this["Func_rounddown_CategoryURL"] = "Matematik";
        this["Func_rounddown_Desc"] = "Rundar ett nummer ned till det angivna antalet siffror.";
        this["Func_roundup_Arg_0"] = "Nummer";
        this["Func_roundup_Arg_1"] = "Siffror";
        this["Func_roundup_ArgDesc_0"] = "Numret som ska avrundas uppåt.";
        this["Func_roundup_ArgDesc_1"] = "Antalet decimaler som ska avrundas. Positiva siffror indikerar platser efter decimaltecken, negativa siffror indikerar platser före decimalpunkten.";
        this["Func_roundup_Category"] = "Matematik";
        this["Func_roundup_CategoryURL"] = "Matematik";
        this["Func_roundup_Desc"] = "Avrundar ett tal upp till det angivna antalet siffror.";
        this["Func_search_Arg_0"] = "search_text";
        this["Func_search_Arg_1"] = "värde";
        this["Func_search_Arg_2"] = "start_index";
        this["Func_search_ArgDesc_0"] = "Texten att hitta i värde.";
        this["Func_search_ArgDesc_1"] = "Den text inom vilken search_text ska finnas.";
        this["Func_search_ArgDesc_2"] = "Det 1-baserade indexet för att starta sökningen (om den utelämnas är start_index 1).";
        this["Func_search_Category"] = "TextAndData";
        this["Func_search_CategoryURL"] = "TextAndData";
        this["Func_search_Desc"] = "Returnerar det 1-baserade indexet för en sträng inom en annan. Söker utan skriftlägeskänslighet.";
        this["Func_searchb_Arg_0"] = "search_text";
        this["Func_searchb_Arg_1"] = "värde";
        this["Func_searchb_Arg_2"] = "start_index";
        this["Func_searchb_ArgDesc_0"] = "Texten att hitta i värde.";
        this["Func_searchb_ArgDesc_1"] = "Den text inom vilken search_text ska finnas.";
        this["Func_searchb_ArgDesc_2"] = "Det 1-baserade indexet för att starta sökningen (om den utelämnas är start_index 1).";
        this["Func_searchb_Category"] = "TextAndData";
        this["Func_searchb_CategoryURL"] = "TextAndData";
        this["Func_searchb_Desc"] = "Returnerar det 1-baserade indexet för en sträng inom en annan. Söker utan skriftlägeskänslighet.";
        this["Func_second_Arg_0"] = "Time_value";
        this["Func_second_ArgDesc_0"] = "Ett tidsvärde från vilket sekunden ska hämtas.";
        this["Func_second_Category"] = "DatumOchTid";
        this["Func_second_CategoryURL"] = "Date_and_Time";
        this["Func_second_Desc"] = "Returnerar sekunden av ett tidsvärde.";
        this["Func_seriessum_Arg_0"] = "InputValue";
        this["Func_seriessum_Arg_1"] = "InitialPower";
        this["Func_seriessum_Arg_2"] = "Steg";
        this["Func_seriessum_Arg_3"] = "Koefficienter";
        this["Func_seriessum_ArgDesc_0"] = "Ingångsvärdet till effektserien.";
        this["Func_seriessum_ArgDesc_1"] = "Den initiala effekten som InputValue höjs till.";
        this["Func_seriessum_ArgDesc_2"] = "Steget som kommer att användas för att öka effekten för varje period i serien.";
        this["Func_seriessum_ArgDesc_3"] = "En uppsättning koefficienter med vilken varje successiv effekt av InputValue multipliceras.";
        this["Func_seriessum_Category"] = "Matematik";
        this["Func_seriessum_CategoryURL"] = "Matematik";
        this["Func_seriessum_Desc"] = "Returnerar summan av en kraftserie.";
        this["Func_sign_Arg_0"] = "Nummer";
        this["Func_sign_ArgDesc_0"] = "Numret vars tecken kommer att returneras.";
        this["Func_sign_Category"] = "Matematik";
        this["Func_sign_CategoryURL"] = "Matematik";
        this["Func_sign_Desc"] = "Returnerar tecknet på ett tal (-1, 0 eller 1).";
        this["Func_sin_Arg_0"] = "nummer";
        this["Func_sin_ArgDesc_0"] = "Vinkeln i radianer för vilken sinus ska hittas.";
        this["Func_sin_Category"] = "Matematik";
        this["Func_sin_CategoryURL"] = "Matematik";
        this["Func_sin_Desc"] = "Returnerar sinus för den angivna vinkeln.";
        this["Func_sinh_Arg_0"] = "nummer";
        this["Func_sinh_ArgDesc_0"] = "Ett reellt tal för vilket hyperboliskt sinus ska hittas.";
        this["Func_sinh_Category"] = "Matematik";
        this["Func_sinh_CategoryURL"] = "Matematik";
        this["Func_sinh_Desc"] = "Returnerar ett tals hyperboliska sinus.";
        this["Func_sln_Arg_0"] = "kostnad";
        this["Func_sln_Arg_1"] = "rädda";
        this["Func_sln_Arg_2"] = "liv";
        this["Func_sln_ArgDesc_0"] = "Tillgångens initiala kostnad.";
        this["Func_sln_ArgDesc_1"] = "Värdet i slutet av avskrivningen.";
        this["Func_sln_ArgDesc_2"] = "Antalet perioder under vilka tillgången skrivs av.";
        this["Func_sln_Category"] = "Finansiell";
        this["Func_sln_CategoryURL"] = "Finansiell";
        this["Func_sln_Desc"] = "Returnerar en tillgångs linjära avskrivning under en period.";
        this["Func_sqrt_Arg_0"] = "nummer";
        this["Func_sqrt_ArgDesc_0"] = "Numret för vilket kvadratroten ska fås.";
        this["Func_sqrt_Category"] = "Matematik";
        this["Func_sqrt_CategoryURL"] = "Matematik";
        this["Func_sqrt_Desc"] = "Returnerar en positiv kvadratrot av ett tal.";
        this["Func_sqrtpi_Arg_0"] = "Nummer";
        this["Func_sqrtpi_ArgDesc_0"] = "Siffran som ska multipliceras med pi.";
        this["Func_sqrtpi_Category"] = "Matematik";
        this["Func_sqrtpi_CategoryURL"] = "Matematik";
        this["Func_sqrtpi_Desc"] = "Multiplicerar det angivna antalet med pi och returnerar kvadratroten av resultatet.";
        this["Func_stdev_Arg_0"] = "nummer {0}";
        this["Func_stdev_ArgDesc_0"] = "Ett nummer eller intervall av nummer som anger ett urval av en population.";
        this["Func_stdev_Category"] = "Statistisk";
        this["Func_stdev_CategoryURL"] = "Statistisk";
        this["Func_stdev_Desc"] = "Returnerar standardavvikelsen för ett populationsprov.";
        this["Func_subtotal_Arg_0"] = "FunctionNum";
        this["Func_subtotal_Arg_1"] = "Ref {0}";
        this["Func_subtotal_ArgDesc_0"] = "Ett tal som representerar funktionen som ska beräknas. Lägg till 100 till värdet för att ignorera dolda värden.\r\n1 (eller 101) - GENOMSNITT\r\n2 (eller 102) - RÄKNING\r\n4 (eller 104) - MAX\r\n5 (eller 105) - MIN\r\n6 (eller 106) - PRODUKT\r\n7 (eller 107) - STDEV\r\n9 (eller 109) - SUM\r\n10 (eller 110) - VAR\r\n    ";
        this["Func_subtotal_ArgDesc_1"] = "En eller flera referenser för att hitta delsumman.";
        this["Func_subtotal_Category"] = "Matematik";
        this["Func_subtotal_CategoryURL"] = "Matematik";
        this["Func_subtotal_Desc"] = "Beräknar delsumman i en av flera referenser.";
        this["Func_sum_Arg_0"] = "nummer {0}";
        this["Func_sum_ArgDesc_0"] = "Ett nummer som ska adderas till de andra siffrorna.";
        this["Func_sum_Category"] = "Matematik";
        this["Func_sum_CategoryURL"] = "Matematik";
        this["Func_sum_Desc"] = "Summerar alla siffror som ges som argument och returnerar summan.";
        this["Func_syd_Arg_0"] = "kostnad";
        this["Func_syd_Arg_1"] = "rädda";
        this["Func_syd_Arg_2"] = "liv";
        this["Func_syd_Arg_3"] = "per";
        this["Func_syd_ArgDesc_0"] = "Tillgångens initiala kostnad.";
        this["Func_syd_ArgDesc_1"] = "Värdet i slutet av avskrivningen.";
        this["Func_syd_ArgDesc_2"] = "Antalet perioder under vilka tillgången skrivs av.";
        this["Func_syd_ArgDesc_3"] = "Perioden, som måste använda samma enheter som livet.";
        this["Func_syd_Category"] = "Finansiell";
        this["Func_syd_CategoryURL"] = "Finansiell";
        this["Func_syd_Desc"] = "Returnerar årets siffer-avskrivningar för en tillgång under en angiven period.";
        this["Func_tan_Arg_0"] = "nummer";
        this["Func_tan_ArgDesc_0"] = "Vinkeln, i radianer, för vilken man kan få tangenten.";
        this["Func_tan_Category"] = "Matematik";
        this["Func_tan_CategoryURL"] = "Matematik";
        this["Func_tan_Desc"] = "Returnerar tangenten för den angivna vinkeln.";
        this["Func_tanh_Arg_0"] = "nummer";
        this["Func_tanh_ArgDesc_0"] = "Vinkeln, i radianer, för vilken man får den hyperboliska tangenten.";
        this["Func_tanh_Category"] = "Matematik";
        this["Func_tanh_CategoryURL"] = "Matematik";
        this["Func_tanh_Desc"] = "Returnerar den hyperboliska tangenten för den angivna vinkeln.";
        this["Func_text_Arg_0"] = "Värde";
        this["Func_text_Arg_1"] = "FormatText";
        this["Func_text_ArgDesc_0"] = "Det numeriska värdet som ska formateras.";
        this["Func_text_ArgDesc_1"] = "Nummerformatet som värdet ska formateras med.";
        this["Func_text_Category"] = "TextAndData";
        this["Func_text_CategoryURL"] = "Text_and_Data";
        this["Func_text_Desc"] = "Konverterar värdet till text baserat på det angivna formatet.";
        this["Func_time_Arg_0"] = "timme";
        this["Func_time_Arg_1"] = "minut";
        this["Func_time_Arg_2"] = "sekund";
        this["Func_time_ArgDesc_0"] = "Ett nummer som representerar tidens timme.";
        this["Func_time_ArgDesc_1"] = "Ett nummer som representerar tidens minut.";
        this["Func_time_ArgDesc_2"] = "Ett nummer som representerar tidens sekund.";
        this["Func_time_Category"] = "DatumOchTid";
        this["Func_time_CategoryURL"] = "Date_and_Time";
        this["Func_time_Desc"] = "Returnerar ett tal som representerar ett visst tidsvärde.";
        this["Func_timevalue_Arg_0"] = "time_text";
        this["Func_timevalue_ArgDesc_0"] = "Ett textuttryck som representerar en tid.";
        this["Func_timevalue_Category"] = "DatumOchTid";
        this["Func_timevalue_CategoryURL"] = "Date_and_Time";
        this["Func_timevalue_Desc"] = "Konverterar ett textuttryck som representerar en tidpunkt till dess motsvarande tidsvärde.";
        this["Func_today_Category"] = "DatumOchTid";
        this["Func_today_CategoryURL"] = "Date_and_Time";
        this["Func_today_Desc"] = "Returnerar det aktuella datumet.";
        this["Func_trim_Arg_0"] = "text";
        this["Func_trim_ArgDesc_0"] = "Texten för att ta bort extra mellanslag.";
        this["Func_trim_Category"] = "TextAndData";
        this["Func_trim_CategoryURL"] = "Text_and_Data";
        this["Func_trim_Desc"] = "Tar bort alla mellanslag från text förutom enstaka mellanrum mellan ord.";
        this["Func_true_Category"] = "Logisk";
        this["Func_true_CategoryURL"] = "Logisk";
        this["Func_true_Desc"] = "Returnerar det logiska värdet sant.";
        this["Func_trunc_Arg_0"] = "nummer";
        this["Func_trunc_Arg_1"] = "antal_siffror";
        this["Func_trunc_ArgDesc_0"] = "Siffran som ska avkortas.";
        this["Func_trunc_ArgDesc_1"] = "Ett nummer som anger avkortningens precision. Standardvärdet är noll.";
        this["Func_trunc_Category"] = "Matematik";
        this["Func_trunc_CategoryURL"] = "Matematik";
        this["Func_trunc_Desc"] = "Avkortar ett tal till ett heltal genom att ta bort bråkdelen från numret.";
        this["Func_type_Arg_0"] = "värde";
        this["Func_type_ArgDesc_0"] = "Värdet från vilket typen ska hittas.";
        this["Func_type_Category"] = "Information";
        this["Func_type_CategoryURL"] = "Information";
        this["Func_type_Desc"] = "Returnerar ett tal som representerar värdetypen: nummer = 1; text = 2; logiskt värde (sant eller falskt) = 4; felvärde = 16; array = 64.";
        this["Func_upper_Arg_0"] = "text";
        this["Func_upper_ArgDesc_0"] = "Texten som ska konverteras till versaler.";
        this["Func_upper_Category"] = "TextAndData";
        this["Func_upper_CategoryURL"] = "Text_and_Data";
        this["Func_upper_Desc"] = "Konverterar alla små bokstäver i ett textuttryck till versaler.";
        this["Func_value_Arg_0"] = "text";
        this["Func_value_ArgDesc_0"] = "Textuttrycket att konvertera.";
        this["Func_value_Category"] = "TextAndData";
        this["Func_value_CategoryURL"] = "Text_and_Data";
        this["Func_value_Desc"] = "Konverterar ett textuttryck som representerar ett tal till ett tal.";
        this["Func_var_Arg_0"] = "nummer {0}";
        this["Func_var_ArgDesc_0"] = "Ett nummer eller intervall av nummer som anger ett urval av en population.";
        this["Func_var_Category"] = "Statistisk";
        this["Func_var_CategoryURL"] = "Statistisk";
        this["Func_var_Desc"] = "Returnerar variansen för ett populationsprov.";
        this["Func_weekday_Arg_0"] = "Datum";
        this["Func_weekday_Arg_1"] = "ReturnType";
        this["Func_weekday_ArgDesc_0"] = "Datumet för vilket veckodagens nummer ska hittas.";
        this["Func_weekday_ArgDesc_1"] = "Bestämmer vilket numreringsschema som används för veckodagarna.\r\n1 (standard) = söndag (1) till lördag (7)\r\n2 = måndag (1) till söndag (7)\r\n3 = måndag (0) till söndag (6)\r\n    ";
        this["Func_weekday_Category"] = "DatumOchTid";
        this["Func_weekday_CategoryURL"] = "DatumOchTid";
        this["Func_weekday_Desc"] = "Returnerar ett nummer som representerar veckodagen för det angivna datumet.";
        this["Func_weeknum_Arg_0"] = "date";
        this["Func_weeknum_Arg_1"] = "returnType";
        this["Func_weeknum_ArgDesc_0"] = "Datumet för att hitta veckonummer.";
        this["Func_weeknum_ArgDesc_1"] = "Bestämmer vilken dag som börjar veckan.\r\n1 (standard) = söndag är den första dagen i veckan.\r\n2 = måndag är den första dagen i veckan.\r\n    ";
        this["Func_weeknum_Category"] = "DatumOchTid";
        this["Func_weeknum_CategoryURL"] = "DatumOchTid";
        this["Func_weeknum_Desc"] = "Returnerar antalet veckor i vilket det angivna datumet faller.";
        this["Func_workday_Arg_0"] = "startDate";
        this["Func_workday_Arg_1"] = "dagar";
        this["Func_workday_Arg_2"] = "högtider";
        this["Func_workday_ArgDesc_0"] = "Datumet från vilket du ska börja.";
        this["Func_workday_ArgDesc_1"] = "Antalet arbetsdagar som ska kompenseras från startdatumet.";
        this["Func_workday_ArgDesc_2"] = "Ett intervall med en eller flera referenser som innehåller datum som ska betraktas som icke-arbetsdagar.";
        this["Func_workday_Category"] = "DatumOchTid";
        this["Func_workday_CategoryURL"] = "DatumOchTid";
        this["Func_workday_Desc"] = "Returnerar en arbetsdag det angivna antalet dagar från det angivna datumet, exklusive eventuella angivna helgdagar.";
        this["Func_year_Arg_0"] = "Date_value";
        this["Func_year_ArgDesc_0"] = "Ett datumvärde från vilket året ska hämtas.";
        this["Func_year_Category"] = "DatumOchTid";
        this["Func_year_CategoryURL"] = "Date_and_Time";
        this["Func_year_Desc"] = "Returnerar året för ett datumvärde.";
        this["Func_yearfrac_Arg_0"] = "StartDate";
        this["Func_yearfrac_Arg_1"] = "EndDate";
        this["Func_yearfrac_Arg_2"] = "Basis";
        this["Func_yearfrac_ArgDesc_0"] = "Startdatum är startdatumet i intervallet.";
        this["Func_yearfrac_ArgDesc_1"] = "Slutdatum är slutdatum i intervallet.";
        this["Func_yearfrac_ArgDesc_2"] = "\r\n      Bas bestämmer grunden för den dagräkning som ska användas.\r\n0 (standard = USA (NASD) 30/360\r\n1 = Faktiskt / faktiskt\r\n2 = Faktiskt / 360\r\n3 = Faktiskt / 365\r\n4 = Europeisk 30/360\r\n    ";
        this["Func_yearfrac_Category"] = "DatumOchTid";
        this["Func_yearfrac_CategoryURL"] = "DatumOchTid";
        this["Func_yearfrac_Desc"] = "Returnerar ett bråk som representerar den del av ett år som anges mellan det angivna startdatumet och slutdatumet.";
        this["GenerateTableColumnName"] = "Kolumn";
        this["GenerateTableName"] = "Tabell";
        this["LD_Chart_AxisDisplayUnits"] = "Hundratals: Hundratals, Tusentals: Tusentals, TioTusentals: x 10000, Hundratusentals: x 100000, Miljoner: Miljoner, TioMiljoner: x 10000000, HundraMiljoner: x 100000000, TusenMiljoner: Miljarder, MiljonerMiljoner: Triljoner";
        this["LD_Fallback_ChartEx_Line1"] = "Diagrammet är inte tillgängligt i din version av Excel.";
        this["LD_Fallback_ChartEx_Line2"] = "Om du redigerar denna form eller sparar den här arbetsboken i ett annat filformat förstörs det här diagrammet permanent.";
        this["LE_AllMustHavGuidsIfAnyHasThem"] = "Kan inte ställa in GenerateGuidsForGroups till false om någon befintlig grupp har en Guide.";
        this["LE_ArgumentException_AnchorCellFromOtherWorksheet"] = "Den angivna cellen finns inte på samma kalkylblad som formen.";
        this["LE_ArgumentException_ArrayFormulaMustHaveSingleRegion"] = "En matrisformel kan bara tillämpas på en cellregion.";
        this["LE_ArgumentException_ArrayTooSmall"] = "Matrisen är inte tillräckligt stor för att kopiera alla värden till den.";
        this["LE_ArgumentException_CannotAddStandardTableStyle"] = "En standardtabellstil kan inte läggas till i CustomTableStyleCollection.";
        this["LE_ArgumentException_CannotApplyDVRuleToTotalCell"] = "En eller flera celler i den angivna referensen är totala celler i en tabell och kan inte ha en datavalideringsregel tillämpad.";
        this["LE_ArgumentException_CannotCreateEmptyColorInfo"] = "Den tomma färgen kan inte användas för att skapa en WorkbookColorInfo-instans.";
        this["LE_ArgumentException_CellMustBeOnSameWorksheetAsReferencesCollection"] = "Cellen måste vara från samma kalkylblad som referenssamlingen.";
        this["LE_ArgumentException_CellShiftedOffWorksheet"] = "Cellen har raderats eller flyttats från kalkylbladet och representerar inte längre en position på kalkylbladet.";
        this["LE_ArgumentException_CellsInTableFromOtherWorksheet"] = "Den angivna regionen kommer inte från samma kalkylblad som datatabellen.";
        this["LE_ArgumentException_CellValueStringLength"] = "Strängvärden som tilldelats en cell får inte ha en längd som är större än {0}.";
        this["LE_ArgumentException_ColumnFormulaAlreadyAppliedToCell"] = "Formeln tillämpas redan på en cell. Klona den för att använda den som kolumnformel.";
        this["LE_ArgumentException_ColumnFromDifferentWorksheet"] = "Kolumnen tillhör inte samma kalkylblad som den här samlingen.";
        this["LE_ArgumentException_ColumnInputCellFromOtherWorksheet"] = "Den angivna kolumninmatningscellen finns inte i samma kalkylblad som datatabellen.";
        this["LE_ArgumentException_ColumnNotInTable"] = "Kolumnen är från en annan tabell.";
        this["LE_ArgumentException_ColumnRemovedFromWorksheet"] = "Kolumnen har tagits bort från kalkylbladet.";
        this["LE_ArgumentException_ColumnsToRepeatAtLeftOutsideRange"] = "Området ligger utanför intervallet för tillgängliga kolumner i kalkylbladet.";
        this["LE_ArgumentException_CommentTooLong"] = "Kommentarlängden måste vara mindre än 255 tecken.";
        this["LE_ArgumentException_CustomViewNameAlreadyExists"] = "En anpassad vy med namnet \"{0}\" finns redan.";
        this["LE_ArgumentException_DefaultTableStyleNotInWorkbook"] = "Den angivna tabellstilen finns inte i CustomTableStyles eller StandardTableStyles-samlingen.";
        this["LE_ArgumentException_DisplayTextTooLong"] = "Visningstexten för hyperlänken måste innehålla mindre än 256 tecken.";
        this["LE_ArgumentException_DuplicateDisplayValue"] = "Det angivna visningsvärdet finns redan i samlingen.";
        this["LE_ArgumentException_DuplicateFixedDateGroup"] = "Den angivna FixedDateGroup finns redan i samlingen.";
        this["LE_ArgumentException_DuplicateTableStyle"] = "Den angivna stilen finns redan i en CustomTableStyleCollection.";
        this["LE_ArgumentException_DuplicateTableStyleName_Existing"] = "Det angivna namnet som matchar namnet på en annan stil i samlingen.";
        this["LE_ArgumentException_DuplicateTableStyleName_New"] = "Den angivna stilen har ett namn som matchar en annan stil i samlingen.";
        this["LE_ArgumentException_DuplicateWorksheetProtectedRangeTitle"] = "Ett intervall med titeln '{0}' finns redan.";
        this["LE_ArgumentException_DV_ArrayFormulaCannotBeUsed"] = "En ArrayFormula kan inte användas som {0}.";
        this["LE_ArgumentException_DV_CellFromOtherWorksheet"] = "Cellen tillhör inte samma kalkylblad som samlingen av datavalideringsregler.";
        this["LE_ArgumentException_DV_ConstraintFormulaMustBeNumberOrReference"] = "{0} måste ange ett nummer, ett numeriskt uttryck eller en hänvisning till en enskild cell.";
        this["LE_ArgumentException_DV_FormulaCannotBeNull"] = "{0} kan inte anges som null medan datavalideringsregeln tillämpas på ett kalkylblad.";
        this["LE_ArgumentException_DV_FormulaCannotFindNamedReference"] = "{0} innehåller en namngiven referens som inte finns i arbetsboken.";
        this["LE_ArgumentException_DV_FormulaCannotFindWorksheetReference"] = "{0} innehåller en referens i ett kalkylblad som inte finns i arbetsboken.";
        this["LE_ArgumentException_DV_FormulaCannotReferenceOtherWorkbook"] = "{0} kan inte innehålla en referens till en extern arbetsbok.";
        this["LE_ArgumentException_DV_FormulaInvalidForWorkbookFormat"] = "De {0} referensceller som inte är giltiga i arbetsboksformatet {1}.";
        this["LE_ArgumentException_DV_InvalidAddress"] = "Den angivna adressen är inte en giltig cell- eller regionadress.";
        this["LE_ArgumentException_DV_InvalidExcelDate"] = "Det angivna datumet kan inte representeras i Excel.";
        this["LE_ArgumentException_DV_InvalidFormula"] = "Ogiltig formel har använts i datavalideringsregeln.";
        this["LE_ArgumentException_DV_InvalidMessageLength"] = "{0} får inte innehålla mer än {1} tecken.";
        this["LE_ArgumentException_DV_ListFormulaCannotHaveEmptyString"] = "Strängkonstanten i ValuesFormula kan inte vara en tom sträng.";
        this["LE_ArgumentException_DV_ListFormulaMustBeStringOrReference"] = "ValuesFormula måste anges som en strängkonstant eller en referens till en enda cell eller region i samma arbetsbok.";
        this["LE_ArgumentException_DV_ListFormulaReferenceMustBeOneDimensional"] = "ValuesFormula innehåller en regionreferens som har flera rader och kolumner. Regionen måste innehålla en rad eller en kolumn.";
        this["LE_ArgumentException_DV_LowerGreaterThanUpperContraint"] = "Värdet på den nedre begränsningen måste vara mindre än eller lika med den övre begränsningen.";
        this["LE_ArgumentException_DV_MustHaveOneAcceptedValue"] = "Minst ett värde måste anges som ett accepterat värde.";
        this["LE_ArgumentException_DV_ReferencesAlreadyContainDVs"] = "De angivna referenserna innehåller redan en eller flera datavalideringsregler.";
        this["LE_ArgumentException_DV_ReferencesFromOtherWorksheet"] = "Referenssamlingen tillhör inte samma kalkylblad som samlingen av datavalideringsregler.";
        this["LE_ArgumentException_DV_RegionFromOtherWorksheet"] = "Regionen tillhör inte samma kalkylblad som samlingen av datavalideringsregler.";
        this["LE_ArgumentException_DV_RuleAppliedOnOtherWorksheet"] = "Datavalideringsregeln tillämpas redan på ett annat kalkylblad.";
        this["LE_ArgumentException_DV_UpperLessThanLowerContraint"] = "Värdet på den övre begränsningen måste vara större än eller lika med den nedre begränsningen.";
        this["LE_ArgumentException_EditRangeAlreadyOwned"] = "Området finns redan i ett kalkylblads AllowedEditRanges.";
        this["LE_ArgumentException_EditRangeHasRangeFromOtherWorksheet"] = "Redigeringsområdet har ett intervall från ett annat kalkylblad än AllowedEditRanges.";
        this["LE_ArgumentException_EmptyCustomList"] = "Det måste finnas minst en sträng i listan.";
        this["LE_ArgumentException_EndIndexLessThanZero"] = "Slutet på intervallet måste vara större än eller lika med noll.";
        this["LE_ArgumentException_FileDoesntContainsWorkbookStream"] = "Den angivna strömmen eller filen innehåller inte en giltig arbetsbok.";
        this["LE_ArgumentException_FormulaReferencesInvalidCells"] = "Den angivna formeln refererar till celler som inte stöds i arbetsbokens nuvarande filformat.";
        this["LE_ArgumentException_HashSetFull"] = "Det finns för många objekt i hashuppsättningen.";
        this["LE_ArgumentException_HyperlinkAlreadyOwned"] = "Hyperlänken tillhör redan en samling hyperlänkar.";
        this["LE_ArgumentException_HyperlinkSourceFromOtherWorksheet"] = "Hyperlänken har en källcell eller region från ett annat kalkylblad.";
        this["LE_ArgumentException_HyperlinkTargetFromOtherWorkbook"] = "Målet kommer från en annan arbetsbok än hyperlänkens ägare.";
        this["LE_ArgumentException_InfiniteColumnWidth"] = "Kolumnbredden kan inte vara oändlig.";
        this["LE_ArgumentException_Interval"] = "Intervallparameter bör vara ett av värdena i DateInterval-uppräkningen";
        this["LE_ArgumentException_IntervalStr"] = "Intervallparametern kunde inte konverteras till ett av värdena i DateInterval-uppräkningen";
        this["LE_ArgumentException_InvalidCellAddress"] = "Den angivna adressen är inte en giltig celladress.";
        this["LE_ArgumentException_InvalidCommentPositioningMode"] = "Endast DontMoveOrSizeWithCells och MoveAndSizeWithCells ShapePositioningModes kan tillämpas på en kommentarform.";
        this["LE_ArgumentException_InvalidCustomFilterOperandNumber"] = "Värdet kan inte vara oändligt eller NaN.";
        this["LE_ArgumentException_InvalidCustomFilterOperator"] = "När jämförelsevärdet inte är en sträng kan jämförelsesoperatören inte vara BeginsWith, DoesNotBeginWith, Contains, DoesNotContain, EndsWith eller DoesNotEndWith.";
        this["LE_ArgumentException_InvalidDatePeriodFilterValue"] = "Värdet ligger inte inom intervallet. För en filtertyp för månader måste värdet vara 1-12. För en filtertyp för kvartal måste den vara 1-4.";
        this["LE_ArgumentException_InvalidDpi"] = "Det angivna Dpi-värdet är ogiltigt.";
        this["LE_ArgumentException_InvalidExcelDate"] = "Värdet kan inte uttryckas som ett datum i Excel.";
        this["LE_ArgumentException_InvalidFileFormat"] = "Ogiltigt eller okänt filformat.";
        this["LE_ArgumentException_InvalidFormula"] = "Den angivna formeln var ogiltig.";
        this["LE_ArgumentException_InvalidGradientStopColor"] = "Automatiska färger och systemfärger kan inte användas i gradientstopp.";
        this["LE_ArgumentException_InvalidNamedReferenceName"] = "Namnet är inte ett giltigt referensnamn.";
        this["LE_ArgumentException_InvalidPaletteColor_EmptyOrSystem"] = "Tomma eller systemfärger kan inte läggas till i paletten.";
        this["LE_ArgumentException_InvalidPaletteColor_NonOpaque"] = "Endast ogenomskinliga färger kan läggas till i paletten.";
        this["LE_ArgumentException_InvalidReferencesString"] = "Referensvärdet är inte korrekt formaterat.";
        this["LE_ArgumentException_InvalidRegionAddress"] = "Den angivna adressen är inte en giltig regionadress.";
        this["LE_ArgumentException_InvalidRowOrColumnRange"] = "Ogiltigt rad- eller kolumnintervall.";
        this["LE_ArgumentException_InvalidSeriesValues"] = "Ogiltiga serievärden. Det måste finnas minst ett värde och varje värde måste antingen vara en sträng eller ett objekt som kan konverteras till en dubbel.";
        this["LE_ArgumentException_InvalidTarget"] = "Hyperlänkmålet måste vara en sträng, WorksheetCell, WorksheetRegion eller NamedReference.";
        this["LE_ArgumentException_InvalidTopOrBottomFilterValue"] = "Värdet måste vara mellan 1 och 500.";
        this["LE_ArgumentException_InvalidWorksheetName"] = "Kalkylbladets namn får inte innehålla något av följande tecken: : \\ /? * []";
        this["LE_ArgumentException_InvalidWorksheetNameStartingQuote"] = "Kalkylbladets namn kan inte börja eller sluta med ett enda citattecken.";
        this["LE_ArgumentException_LessThanTwoGradientStops"] = "Det måste anges minst två gradientstopp.";
        this["LE_ArgumentException_NamedReferenceNameTooLong"] = "Namnet på den namngivna referensen får inte vara mer än 255 tecken.";
        this["LE_ArgumentException_NaNDefaultColumnWidth"] = "NaN kan inte användas som standardkolumnbredd.";
        this["LE_ArgumentException_NewRegionOverlapsFilterArea"] = "Operationen försöker ändra ett filtrerat intervall på kalkylbladet och kan inte slutföras. För att slutföra denna åtgärd måste autofilter i arket tas bort.";
        this["LE_ArgumentException_NewTableRegionCannotMoveHeaders"] = "Den nya tabellregionen kan inte flytta rubrikerna till en ny rad.";
        this["LE_ArgumentException_NewTableRegionFromWrongWorksheet"] = "Den angivna regionen inte från samma kalkylblad som tabellen.";
        this["LE_ArgumentException_NewTableRegionMustHaveDataRows"] = "Den nya tabellregionen måste innehålla minst en datarad.";
        this["LE_ArgumentException_NewTableRegionOverlapOld"] = "Den nya tabellregionen måste överlappa den föregående tabellregionen.";
        this["LE_ArgumentException_NewTableRegionOverlapsBlockingValue"] = "Tabellen kan inte ändras till den angivna regionen eftersom den innehåller en formel eller datatabell med flera celler.";
        this["LE_ArgumentException_NewTableRegionOverlapsMergedRegion"] = "Tabellen kan inte ändras till den angivna regionen eftersom den innehåller en sammanfogad cell.";
        this["LE_ArgumentException_NewTableRegionOverlapsOtherTable"] = "Tabellen kan inte ändras till den angivna regionen eftersom den innehåller en annan tabell.";
        this["LE_ArgumentException_NoRegionsInArray"] = "Den angivna matrisen har inga regioner.";
        this["LE_ArgumentException_NotEnum"] = "Typ '{0}' är inte en uppräkningstyp";
        this["LE_ArgumentException_NotSupportedCodePage"] = "Kodsidan stöds inte";
        this["LE_ArgumentException_NPer"] = "nPer kunde inte vara 0";
        this["LE_ArgumentException_NPerFunction"] = "Fel kombination av parametrar för NPer-funktion";
        this["LE_ArgumentException_PA_PrintAreasCannotOverlap"] = "Det angivna utskriftsområdet överlappar ett av de befintliga utskriftsområdena.";
        this["LE_ArgumentException_ParentStyleFromOtherWorkbook"] = "Den angivna stilen är inte från samma arbetsbok som cellformatet.";
        this["LE_ArgumentException_PB_CantInsertBreakAtIndex"] = "Denna samling stöder inte att infoga objekt i ett index.";
        this["LE_ArgumentException_PB_CantSetBreakAtIndex"] = "Denna samling stöder inte specificeringen av objekt i ett index.";
        this["LE_ArgumentException_PB_MustBeWithinPrintArea"] = "Sidbrytningen måste ligga inom utskriftsområdet.";
        this["LE_ArgumentException_PB_PageBreakCantBeA1Cell"] = "Det går inte att infoga en sidbrytning längst upp till vänster i kalkylbladet.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeLeftColumn"] = "Det går inte att infoga en sidbrytning till vänster om kalkylbladet.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeTopRow"] = "Det går inte att infoga en sidbrytning högst upp i kalkylbladet.";
        this["LE_ArgumentException_PB_PageBreaksCannotAddExistingInstance"] = "Den angivna instansen finns redan i samlingen.";
        this["LE_ArgumentException_PB_PageBreaksCannotOverlap"] = "Sidbrytningen kan inte överlappa andra sidbrytningar i kalkylbladet.";
        this["LE_ArgumentException_PB_PrintAreaMustBeInPrintAreas"] = "Sidbrytningen ligger i ett utskriftsområde som inte finns i PrintAreas-samlingen.";
        this["LE_ArgumentException_PB_PrintAreaMustBeOnSameWorksheet"] = "Sidbrytningen finns inom ett utskriftsområde som inte finns på samma kalkylblad som sidbrytssamlingen.";
        this["LE_ArgumentException_Pmt"] = "Pmt kunde inte vara 0";
        this["LE_ArgumentException_PrintAreaShiftedOffWorksheet"] = "Sidbrytningen ligger i ett utskriftsområde som har flyttats från kalkylbladet och inte längre representerar ett giltigt cellområde.";
        this["LE_ArgumentException_RangeFromOtherWorksheetThanProtectedRange"] = "Det angivna intervallet kommer från ett annat kalkylblad än redigeringsområdet.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheet"] = "Regionen måste vara från samma kalkylblad.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheetAsReferencesCollection"] = "Regionen måste vara från samma kalkylblad som referenssamlingen.";
        this["LE_ArgumentException_RegionsFromMixedWorksheets"] = "Alla regioner som anges i matrisen måste vara från samma kalkylblad.";
        this["LE_ArgumentException_RegionShiftedOffWorksheet"] = "Regionen har flyttats från kalkylbladet och representerar inte längre ett giltigt cellområde.";
        this["LE_ArgumentException_RegionsShiftedOffWorksheet"] = "En eller flera angivna regioner har flyttats från kalkylbladet och representerar inte längre ett giltigt cellområde.";
        this["LE_ArgumentException_RowFromDifferentWorksheet"] = "Raden tillhör inte samma kalkylblad som den här samlingen.";
        this["LE_ArgumentException_RowInputCellFromOtherWorksheet"] = "Den angivna radinmatningscellen finns inte i samma kalkylblad som datatabellen.";
        this["LE_ArgumentException_RowRemovedFromWorksheet"] = "Raden har tagits bort från kalkylbladet.";
        this["LE_ArgumentException_SelectedWorksheetFromOtherWorkbook"] = "Kalkylbladet tillhör inte den arbetsbok som är associerad med dessa fönsteralternativ.";
        this["LE_ArgumentException_ShapeCannotBeAdded"] = "Den angivna formen kan inte läggas till i en formsamling.";
        this["LE_ArgumentException_SourceFromOtherWorksheetThanHyperlink"] = "Den angivna källcellen eller regionen är från ett annat kalkylblad än hyperlänken.";
        this["LE_ArgumentException_StartIndexLessThanZero"] = "Början på intervallet måste vara större än eller lika med noll.";
        this["LE_ArgumentException_StyleNameAlreadyExists"] = "En stil med namnet \"{0}\" finns redan i arbetsboken.";
        this["LE_ArgumentException_StyleNameTooLong"] = "Stilnamnet får inte vara mer än 255 tecken.";
        this["LE_ArgumentException_TableColumnRemovedFromTable"] = "Tabellkolumnen har tagits bort från tabellen.";
        this["LE_ArgumentException_TableStyleFromOtherWorkbook"] = "Den angivna WorksheetTableStyle tillhör inte samma arbetsbok som tabellen.";
        this["LE_ArgumentException_TargetTooLong"] = "Adressen för hyperlänkens mål måste vara mindre än 256 tecken.";
        this["LE_ArgumentException_TextLengthGreaterThanMax"] = "Textens längd i en form eller kommentar får inte vara större än {0}.";
        this["LE_ArgumentException_ToolTipTooLong"] = "Hyperlänkens inforuta måste innehålla mindre än 256 tecken.";
        this["LE_ArgumentException_TopLeftWindowBounds"] = "Den vänstra och övre delen av fönstret måste ligga mellan -32768 och 32767.";
        this["LE_ArgumentException_WeekdayFirst"] = "Parametern WeekdayFirst ska vara mellan FirstDayOfWeek.System och FirstDayOfWeek.Saturday";
        this["LE_ArgumentException_WidthHeightWindowBounds"] = "Fönstrets bredd och höjd måste vara mellan 0 och 65535.";
        this["LE_ArgumentException_WorksheetNameAlreadyExists"] = "Ett kalkylblad med namnet \"{0}\" finns redan i arbetsboken.";
        this["LE_ArgumentException_WorksheetNameTooLong"] = "Kalkylbladets namn måste vara mellan 1 och 31 tecken långt.";
        this["LE_ArgumentException_WorksheetProtectedRangeTitleTooLong"] = "Titeln på ett redigeringsområde kan inte innehålla mer än 255 tecken.";
        this["LE_ArgumentException_WorksheetScopeFromOtherWorkbook"] = "Kalkylbladet tillhör inte samma arbetsbok som den här samlingen.";
        this["LE_ArgumentNullException_AnchorCell"] = "Cellen kan inte vara null när formen finns i ett kalkylblad eller en grupp.";
        this["LE_ArgumentNullException_CustomViewName"] = "Det anpassade vynamnet kan inte vara null eller tomt.";
        this["LE_ArgumentNullException_Encoder"] = "Bildkodaren kan inte vara null.";
        this["LE_ArgumentNullException_FindNamedReference"] = "Namnet som ska hittas får inte vara null eller tomt.";
        this["LE_ArgumentNullException_FormulaCantBeNull"] = "Formeln för en namngiven referens kan inte vara null eller tom.";
        this["LE_ArgumentNullException_HiddenColumn"] = "En null-kolumn kan inte läggas till i samlingen.";
        this["LE_ArgumentNullException_HiddenRow"] = "En null-rad kan inte läggas till i samlingen.";
        this["LE_ArgumentNullException_Image"] = "Bilden kan inte vara null.";
        this["LE_ArgumentNullException_NamedReferenceNameCantBeNull"] = "Namnet på en namngiven referens får inte vara null eller tomt.";
        this["LE_ArgumentNullException_SaveStream"] = "Det går inte att spara arbetsboken till en null-ström.";
        this["LE_ArgumentNullException_SelectedWorksheet"] = "Arbetsboken måste ha ett giltigt valt kalkylblad.";
        this["LE_ArgumentNullException_Shape"] = "En form som är null kan inte läggas till i samlingen.";
        this["LE_ArgumentNullException_SourceFont"] = "Källteckensnittet kan inte vara null.";
        this["LE_ArgumentNullException_SourceFormatting"] = "Källformateringen kan inte vara null.";
        this["LE_ArgumentNullException_StyleName"] = "Stilnamnet får inte vara null eller tomt.";
        this["LE_ArgumentNullException_UnformattedString"] = "Den oformaterade strängen kan inte vara null.";
        this["LE_ArgumentNullException_Workbook"] = "Arbetsboken kan inte vara null.";
        this["LE_ArgumentNullException_WorkbookRequiredToResolveThemeColor"] = "En arbetsbok måste anges för att lösa en temafärg.";
        this["LE_ArgumentNullException_WorksheetName"] = "Kalkylbladets namn får inte vara null eller tomt.";
        this["LE_ArgumentNullException_WorksheetProtectedRangeTitleCantBeNull"] = "Titeln på ett redigeringsområde kan inte vara null eller tom.";
        this["LE_ArgumentNullException_WorksheetScope"] = "Kalkylbladets omfång kan inte vara null.";
        this["LE_ArgumentOutOfRange_RelativeColumnIndex"] = "Det relativa kolumnindexet ligger utanför det angivna området.";
        this["LE_ArgumentOutOfRange_RelativeRowIndex"] = "Det relativa radindexet ligger utanför det angivna området.";
        this["LE_ArgumentOutOfRangeException_AnchorPosition"] = "En av procentsatserna på den angivna platsen är mindre än 0.";
        this["LE_ArgumentOutOfRangeException_BadInsertIndex"] = "Det angivna indexet måste vara större än eller lika med noll och mindre än eller lika med antalet befintliga stycken.";
        this["LE_ArgumentOutOfRangeException_BadRemoveAtIndex"] = "Indexet måste vara större än noll och mindre än antalet stycken i den formaterade strängen.";
        this["LE_ArgumentOutOfRangeException_CollectionIndex"] = "Indexet var utanför intervallet. Det måste vara icke-negativt och mindre än samlingens storlek.";
        this["LE_ArgumentOutOfRangeException_ColumnWidth"] = "Kolumnbredden måste vara mellan 0 och 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultColumnWidth"] = "Standardkolumnbredden måste vara mellan 0 och 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultFontHeight"] = "Standardteckensnittets höjd måste vara mellan 20 och 8180.";
        this["LE_ArgumentOutOfRangeException_DefaultRowHeight"] = "Standardradhöjden måste vara mellan -1 och 8192.";
        this["LE_ArgumentOutOfRangeException_DuplicateItemSorted"] = "Det angivna objektet har redan ett sorteringsvillkor tillämpat.";
        this["LE_ArgumentOutOfRangeException_FirstVisibleTabIndex"] = "Det första synliga flikindexet måste vara större än eller lika med noll.";
        this["LE_ArgumentOutOfRangeException_FontHeight"] = "Teckensnittets höjd måste vara mellan 20 och 8180.";
        this["LE_ArgumentOutOfRangeException_GroupAddedToSelf"] = "En grupp kan inte läggas till i sin egen samling av former.";
        this["LE_ArgumentOutOfRangeException_Indent"] = "Indragningsnivån måste vara mellan 0 och 250.";
        this["LE_ArgumentOutOfRangeException_IndexNegative"] = "Indexet måste vara icke-negativt.";
        this["LE_ArgumentOutOfRangeException_InvalidCollectionIndex"] = "Det angivna indexet är inte giltigt i samlingen.";
        this["LE_ArgumentOutOfRangeException_InvalidColorInfoTint"] = "Tonvärdet måste vara mellan -1,0 och 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnCount"] = "{0} är inte ett giltigt kolumnantal i det aktuella formatet. Värdet måste vara mellan {1} och {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnIndex"] = "{0} är inte ett giltigt kolumnindex i det aktuella formatet. Värdet måste vara mellan {1} och {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidGradientStopOffset"] = "Gradientstoppsförskjutningen måste vara mellan 0,0 och 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRelativeRectangleValueForGradient"] = "Värdena för den inre rektangeln i den rektangulära gradienten måste vara mellan 0,0 och 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRemoveAtIndex_SortConditions"] = "Indexet för att ta bort sorteringsvillkoret ligger utanför intervallet.";
        this["LE_ArgumentOutOfRangeException_InvalidRowCount"] = "{0} är inte ett giltigt radantal i det aktuella formatet. Värdet måste vara mellan {1} och {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidRowIndex"] = "{0} är inte ett giltigt radindex i det aktuella formatet. Värdet måste vara mellan {1} och {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidTableStyleAreaStripeExtent"] = "{0} måste vara mellan 1 och 9 inklusive.";
        this["LE_ArgumentOutOfRangeException_LeftPaneWidth"] = "Den angivna rutans bredd ligger utanför det giltiga intervallet {0} och {1}.";
        this["LE_ArgumentOutOfRangeException_LengthMustBePositive"] = "Längden måste vara större än eller lika med ett.";
        this["LE_ArgumentOutOfRangeException_MagnificationLevel"] = "Visningsförstoringsnivåerna måste vara mellan 10 och 400.";
        this["LE_ArgumentOutOfRangeException_Margins"] = "Marginalerna måste vara mellan 0 och 100.";
        this["LE_ArgumentOutOfRangeException_MaxPagesHorizontally"] = "Maximalt antal sidor horisontellt måste vara mellan 0 och 32767.";
        this["LE_ArgumentOutOfRangeException_MaxPagesVertically"] = "Maximalt antal sidor vertikalt måste vara mellan 0 och 32767.";
        this["LE_ArgumentOutOfRangeException_MaxRecursionIterations"] = "Den maximala rekursions-iterationen måste vara mellan 1 och 32767.";
        this["LE_ArgumentOutOfRangeException_MaxSortConditions"] = "Det kan inte finnas mer än {0} sorteringsvillkor i samlingen.";
        this["LE_ArgumentOutOfRangeException_NegativeStartIndex"] = "Startindexet kan inte vara negativt.";
        this["LE_ArgumentOutOfRangeException_NumberOfCopies"] = "Antal kopior måste vara mellan 1 och 65535.";
        this["LE_ArgumentOutOfRangeException_Per"] = "per värde ska vara mellan 0 och nPer + 1";
        this["LE_ArgumentOutOfRangeException_Rate"] = "Hastigheten bör vara större än -1";
        this["LE_ArgumentOutOfRangeException_Resolution"] = "Upplösningen måste vara mellan 0 och 65535.";
        this["LE_ArgumentOutOfRangeException_RowHeight"] = "Radhöjden måste vara mellan 0 och 8192.";
        this["LE_ArgumentOutOfRangeException_ScalingFactor"] = "Skalningsfaktorn måste vara mellan 10 och 400.";
        this["LE_ArgumentOutOfRangeException_StartPageNumber"] = "Startsidans nummer måste vara mellan -32765 och 32767.";
        this["LE_ArgumentOutOfRangeException_TabBarWidth"] = "Flikfältets bredd måste vara mellan 0 och 1000.";
        this["LE_ArgumentOutOfRangeException_TopPaneHeight"] = "Den angivna rutans höjd ligger utanför det giltiga intervallet {0} och {1}.";
        this["LE_AutoColorNotAllowed"] = "Färgen kan inte vara 'Auto'.";
        this["LE_Axis_MajorMinorUnit"] = "Värdet för parametern majorUnit måste vara större än värdet för parametern minorUnit.";
        this["LE_Axis_NoCrossAxis"] = "Om egenskapen CrossAxis är inställd på ett Axe-objekt måste den ställas in på dem alla. Om diagrammet endast har två axlar kan korsningsaxeln bestämmas automatiskt, men bara om CrossAxis-egenskapen inte är inställd på alla Axis-objekt.";
        this["LE_Biff8SerializerNotLoaded"] = "{0} -formatet kräver en serialiserare som inte är laddad, i JavaScript ingår detta i filen infragistics.excel_serialization_biff8.js";
        this["LE_CategoryAxisBinning_BinWidthAndNumberOfBins"] = "Parametrarna 'binWidth' och 'numberOfBins' är ömsesidigt exklusiva. Båda kan vara ospecificerade, men om en anges får den andra inte anges.";
        this["LE_ChartGradientFill_EmptyStops"] = "Måste innehålla minst ett lutningsstopp.";
        this["LE_ChartObject_DifferentChart"] = "Objektinstansen som tilldelas är associerad med ett annat diagram och kan inte tilldelas den här instansen.";
        this["LE_ColumnsToRepeatAtLeftAreOutsideAvailableRange"] = "Utskriftsalternativen har kolumner som upprepas till vänster och ligger utanför intervallet för tillgängliga kolumner i det nya formatet.";
        this["LE_ComboChartGroupCollection_DoesNotExist"] = "Den angivna ComboChartGroup finns inte i denna samling.";
        this["LE_FillPercentInvalid"] = "Värdet {0} är inte ett giltigt värde för egenskapen {1}.";
        this["LE_FormatLimitError_Indent"] = "Indraget överstiger det maximala värdet på {0} som tillhandahålls av det aktuella formatet.";
        this["LE_FormatLimitError_MacroWorkbook"] = "Arbetsboken har VBA-kod och kan inte sparas som en makroaktiverad arbetsbok.";
        this["LE_FormatLimitError_MaxColumnIndex"] = "Kolumnindex på {0} är större än det maximala {1} som stöds av det angivna formatet.";
        this["LE_FormatLimitError_MaxRowIndex"] = "Radindexet på {0} överstiger det maximala {1} som stöds av detta format.";
        this["LE_FormulaParseException_ArrayContainsConstants"] = "En matris kan bara innehålla konstanta uttryck.";
        this["LE_FormulaParseException_ArrayHasEmptyFirstRow"] = "En konstant matris i formeln har inga objekt i den första raden.";
        this["LE_FormulaParseException_ArrayHasMisalignedRows"] = "En konstant array i formeln har feljusterade rader. Varje rad i en matris måste ha samma antal kolumner.";
        this["LE_FormulaParseException_CellReferenceAfterWorkbookName"] = "En extern arbetsbokreferens måste följas av en namngiven referens, inte en cellreferens.";
        this["LE_FormulaParseException_ExtraExpressions"] = "Formeln har extra uttryck efter slutet av den.";
        this["LE_FormulaParseException_FunctionMissingClosingParen"] = "En funktion i formeln saknar slutparantes.";
        this["LE_FormulaParseException_FunctionNestingTooDeep"] = "En funktion i formeln är kapslad för djupt i andra funktioner.";
        this["LE_FormulaParseException_IncorrectNumberOfArguments"] = "En funktion i formeln har inte fått rätt antal argument.";
        this["LE_FormulaParseException_InvalidArguments"] = "En operator eller funktion i formeln har ogiltiga argument.";
        this["LE_FormulaParseException_InvalidConditionalFormatFormula"] = "En formel med villkorligt format kan inte använda referensoperatorer som föreningar eller korsningar, arraykonstanter eller strukturerade tabellreferenser.";
        this["LE_FormulaParseException_InvalidErrorValue"] = "Den angivna feltypen är inte giltig. Den måste vara ett av följande värden: \"#NULL!\", \"# DIV/0!\", \"#VALUE!\", \"#REF!\", \"#NAME?\", \"#NUM!\", Or \"# N/A \".";
        this["LE_FormulaParseException_InvalidFileNameInBrackets"] = "Ett filnamn mellan hakparenteser var ogiltigt.";
        this["LE_FormulaParseException_InvalidStructuredTableReference"] = "Den strukturerade tabellreferensen är felaktig.";
        this["LE_FormulaParseException_InvalidWorkbookName"] = "Arbetsboksreferensen i citat är inte ett giltigt arbetsnamn.";
        this["LE_FormulaParseException_InvalidWorksheetName"] = "Kalkylbladets namn är ogiltigt.";
        this["LE_FormulaParseException_MacroFunctionNotAllowed"] = "Funktionerna '{0}' är bara giltiga i en namngiven referens.";
        this["LE_FormulaParseException_Message_PortionWithError"] = "Del med fel: {0}";
        this["LE_FormulaParseException_MissingArgumentAfterBinary"] = "En binär operator i formeln saknar en operand efter den.";
        this["LE_FormulaParseException_MissingArgumentAfterParen"] = "Det finns inget uttryck efter en öppen parentes.";
        this["LE_FormulaParseException_MissingArgumentAfterUnary"] = "En unär operator i formeln följdes inte av ett giltigt uttryck.";
        this["LE_FormulaParseException_MissingArgumentBeforeBinary"] = "En binär operator i formeln saknar en operand innan den.";
        this["LE_FormulaParseException_NamedReferenceRefsNeedSheetName"] = "Alla cellreferenser i en namngiven referensformel måste identifiera kalkylbladet.";
        this["LE_FormulaParseException_NoElementAfterArraySerapator"] = "Det finns inget matriselement efter en elementavgränsare i formeln.";
        this["LE_FormulaParseException_NoEqualsSign"] = "Formeln måste börja med ett likhetstecken.";
        this["LE_FormulaParseException_NoExclamationAfterWorkbookName"] = "Arbetsboksreferensen måste följas av ett utropstecken.";
        this["LE_FormulaParseException_NoExclamationAfterWorksheetName"] = "Kalkylbladets referens måste följas av ett utropstecken.";
        this["LE_FormulaParseException_NoExpressions"] = "Formeln har inga giltiga uttryck efter likhetstecknet.";
        this["LE_FormulaParseException_NoFileNameAfterBracket"] = "Ett giltigt filnamn förväntades efter den öppna hakparentesen.";
        this["LE_FormulaParseException_NoValidTermAfterWorkbookName"] = "En extern arbetsbokreferens måste följas av en namngiven referens.";
        this["LE_FormulaParseException_NoValidTermAfterWorksheetName"] = "En kalkylbladshänvisning måste följas av en cell eller namngiven referens.";
        this["LE_FormulaParseException_NoWorksheetAfterWorkbookName"] = "Ett arbetsboksfilnamn inom hakparentes måste följas av ett kalkylarknamn.";
        this["LE_FormulaParseException_StringConstantLengthTooLong"] = "Den angivna formeln får inte innehålla en sträng som är längre än {0} tecken.";
        this["LE_FormulaParseException_TooLong"] = "Den maximala längden på en formel kan inte överstiga {0} för det angivna arbetsbokformatet.";
        this["LE_FormulaParseException_UnknownFunction"] = "En funktion i formeln har inte ett giltigt funktionsnamn.";
        this["LE_FormulaParseException_UnmatchedOpenBracket"] = "En konstant matris i formeln har ingen sluthakparantes.";
        this["LE_FormulaParseException_UnmatchedOpenParen"] = "En öppen parentes i formeln har ingen matchande slutparentes.";
        this["LE_FormulaParseException_UnmatchedOpenSquareBracket"] = "En avslutande hakparentes förväntades efter arbetsbokens filnamn.";
        this["LE_FormulaParseException_WorkbookNameMissingEndQuote"] = "Arbetsboksreferensen har inget slut citat.";
        this["LE_FormulaParseException_WorksheetRangeMissingEndingName"] = "Kalkylområdet specificerar inte ett andra kalkylbladsnamn.";
        this["LE_GradientStop_InvalidPosition"] = "Måste ange ett tal mellan 0 och 1.";
        this["LE_IconSetInvalid"] = "'IconSetNoIcon' är inte en giltig inställning för egenskapen IconSet.";
        this["LE_IndexOutOfRangeException_ArrayBounds"] = "Index var utanför gränserna för matrisen.";
        this["LE_InvalidEnumArgumentException_DefaultPatternCannotBeUsed"] = "Standard FillPatternStyle kan inte användas för att skapa en CellFillPattern.";
        this["LE_InvalidLineWeight"] = "LineWeight får inte vara mindre än 0 eller större än 1584.";
        this["LE_InvalidOperationException_AnchorCommentBeforeApplyingToCell"] = "Kommentaren måste ha både TopLeftCornerCell och BottomRightCornerCell angivna, eller så måste de båda vara null innan kommentaren tillämpas på en cell.";
        this["LE_InvalidOperationException_AnchorShapeBeforeAddingToCollection"] = "Formen måste ha både TopLeftCornerCell och BottomRightCornerCell angivna innan den läggs till i ett kalkylblad eller en grupp.";
        this["LE_InvalidOperationException_ArrayFormulaAlreadyApplied"] = "Matrisformeln tillämpas redan på ett cellintervall. Rensa först det befintliga cellområdet.";
        this["LE_InvalidOperationException_ArrayFormulaAppliedInTable"] = "Matrisformler kan inte tillämpas på ett område där det finns en tabell.";
        this["LE_InvalidOperationException_ArrayFormulaInMergedCell"] = "Array-formler är inte giltiga i sammanfogade celler.";
        this["LE_InvalidOperationException_BottomRightAnchorFromOtherWorksheet"] = "Om du lägger till formen i den här samlingen placeras den på ett annat kalkylblad än dess BottomRightCornerCell.";
        this["LE_InvalidOperationException_BuiltInStyleNameCannotBeChanged"] = "Namnet på en inbyggd stil kan inte ändras.";
        this["LE_InvalidOperationException_CannotAddColorsToPaletteDirectly"] = "Färger kan inte läggas till i WorkbookColorPalette. Använd indexeraren för att ställa in en färg i ett visst index.";
        this["LE_InvalidOperationException_CannotAddParagraphDirectly"] = "Stycken kan inte läggas till direkt.";
        this["LE_InvalidOperationException_CannotAddTableDirectly"] = "WorksheetTable-instanser kan inte infogas direkt i samlingen. Använd metoden Add.";
        this["LE_InvalidOperationException_CannotAddTableToRemovedWorksheet"] = "Tabeller kan inte läggas till i ett kalkylblad som har tagits bort från sin överordnade arbetsbok.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileHeaderRowIsHidden"] = "Ett filter kan inte tillämpas när tabellrubrikraden är dold.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileUIIsHidden"] = "Ett filter kan inte tillämpas medan filtergränssnittet är dolt.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfData"] = "{0} kan inte infogas eftersom det skulle flytta data av botten av kalkylbladet.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfObject"] = "{0} kan inte infogas eftersom det skulle förskjuta en form eller kommentar av kalkylbladets botten.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitBlockingValue"] = "{0} kan inte infogas eftersom det skulle flytta celler i en matrisformel eller datatabell.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitMergedRegion"] = "{0} kan inte infogas eftersom det skulle flytta celler i en sammanfogad region.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitTable"] = "{0} kan inte infogas eftersom det skulle flytta celler i en annan tabell.";
        this["LE_InvalidOperationException_CannotInsertTableRow_TableOnBottomOfWorksheet"] = "{0} kan inte infogas eftersom tabellen befinner sig längst ner i kalkylbladet.";
        this["LE_InvalidOperationException_CannotModifyKeysCollection"] = "Keys-samlingen kan inte ändras.";
        this["LE_InvalidOperationException_CannotModifyStandardTableStyle"] = "Standardtabellformat kan inte ändras.";
        this["LE_InvalidOperationException_CannotModifyValuesCollection"] = "Values-samlingen kan inte ändras.";
        this["LE_InvalidOperationException_CannotMoveDisconnectedWorksheet"] = "Kalkylbladet kan inte flyttas eftersom det inte tillhör en arbetsbok.";
        this["LE_InvalidOperationException_CannotRemoveColorsToPaletteDirectly"] = "Färger kan inte tas bort från WorkbookColorPalette. Använd återställningsmetoden för att rensa alla anpassade färger.";
        this["LE_InvalidOperationException_CannotRemoveNormalStyle"] = "Normalstilen kan inte tas bort.";
        this["LE_InvalidOperationException_CannotSetParentStyleOnStyle"] = "Stilen kan inte ställas in i ett format som tillhör en annan stil.";
        this["LE_InvalidOperationException_CannotShift_LossOfData"] = "Det går inte att flytta data av kalkylbladet.";
        this["LE_InvalidOperationException_CannotShift_LossOfObject"] = "Det går inte att flytta objekt av kalkylbladet.";
        this["LE_InvalidOperationException_CannotShift_SplitBlockingValue"] = "Det går inte att skifta i en matrisformel eller datatabell.";
        this["LE_InvalidOperationException_CannotShift_SplitMergedRegion"] = "Det går inte att flytta i en sammanfogad region.";
        this["LE_InvalidOperationException_CannotShift_SplitTable"] = "Det går inte att flytta i tabellen.";
        this["LE_InvalidOperationException_CantAddChartInChart"] = "Ett diagram kan inte läggas till i Shapes-samlingen av ett annat diagram.";
        this["LE_InvalidOperationException_CantAddCustomView"] = "Anpassade vyer kan inte läggas till direkt i samlingen.";
        this["LE_InvalidOperationException_CantAddDataTable"] = "Datatabeller kan inte läggas till direkt.";
        this["LE_InvalidOperationException_CantAddMergedRegion"] = "Sammanfogade cellerregioner kan inte läggas till direkt.";
        this["LE_InvalidOperationException_CantAddNamedReference"] = "Namngivna referenser kan inte läggas till direkt.";
        this["LE_InvalidOperationException_CantAddWorksheet"] = "Kalkylblad kan inte läggas till direkt i den här samlingen.";
        this["LE_InvalidOperationException_CantApplyRemovedCustomView"] = "Den anpassade vyn kan inte tillämpas om den har tagits bort från arbetsbokens anpassade vysamling.";
        this["LE_InvalidOperationException_CantChangeArrayFormula"] = "Cellerna i en matrisformel kan inte ändras.";
        this["LE_InvalidOperationException_CantChangeDataTable"] = "De inre cellerna i en datatabell kan inte ändras.";
        this["LE_InvalidOperationException_CantModifyCollection"] = "Samlingen kan inte ändras.";
        this["LE_InvalidOperationException_CantOverlapArrayFormula"] = "En sammanslagen cellregion kan inte överlappa en matrisformelregion.";
        this["LE_InvalidOperationException_CantOverlapDataTableInterior"] = "En sammanfogad cellregion kan inte överlappa cellerna i en datatabell om den sträcker sig utanför den översta raden eller kolumnen längst till vänster bland dessa celler.";
        this["LE_InvalidOperationException_CantSaveEditRangeWithoutRange"] = "Kalkylbladets AllowedEditRanges måste innehålla minst ett intervall.";
        this["LE_InvalidOperationException_CantSaveWithNoVisibleWorksheets"] = "Minst ett kalkylblad i arbetsboken måste vara synligt.";
        this["LE_InvalidOperationException_CantSaveWithNoWorksheets"] = "En arbetsbok utan kalkylblad kan inte sparas";
        this["LE_InvalidOperationException_CantSetCircularityErrorDirectly"] = "Felvärdet för cirkularitet kan inte tillämpas direkt på celler.";
        this["LE_InvalidOperationException_CantSetDataTableDirectly"] = "Datatabeller kan inte ställas in som värdet på en cell.";
        this["LE_InvalidOperationException_CantSetFormulaDirectly"] = "Formler kan inte tillämpas på en cell direkt, använd ApplyFormula-metoden eller någon av ApplyTo-metoderna i formelinstansen istället.";
        this["LE_InvalidOperationException_CellsInTableMinSize"] = "Celleregionen i en datatabell måste ha minst två celler i varje dimension.";
        this["LE_InvalidOperationException_ChartNotSupported"] = "Ändringar av WorksheetChart stöds inte i detta format.";
        this["LE_InvalidOperationException_ClearUnknownShapeData"] = "All data för denna form är okänd, ta bort formen från dess samling istället.";
        this["LE_InvalidOperationException_CollectionLongerThanMaxValue"] = "Samlingen innehåller fler inmatningar än det angivna maximivärdet tillåter.";
        this["LE_InvalidOperationException_CollectionModifiedWhileEnumerating"] = "Samlingen modifierades medan den räknades upp.";
        this["LE_InvalidOperationException_CurrentTableRowAddressNeedsOriginCell"] = "För att analysera en strukturerad tabellreferens med den aktuella raden måste en ursprungscell anges.";
        this["LE_InvalidOperationException_CustomViewNoVisibleWorksheets"] = "Minst ett kalkylblad i den anpassade vyn med namnet \"{0}\" måste vara synligt.";
        this["LE_InvalidOperationException_DataTableAppliedInTable"] = "En datatabell kan inte vara i ett område där en tabell finns.";
        this["LE_InvalidOperationException_DataTableFormulaCannotBeApplied"] = "En datatabellformel kan inte tillämpas direkt på celler.";
        this["LE_InvalidOperationException_DataTableInMergedCell"] = "De inre cellerna i en datatabell kan inte vara en del av en sammanfogad cellregion.";
        this["LE_InvalidOperationException_DataTableRemoved"] = "Inmatningscellerna för rad och kolumn kan inte ställas in efter att datatabellen har tagits bort från kalkylbladet.";
        this["LE_InvalidOperationException_DeprecatedFillColorSetWithGradientFill"] = "En fyllnadsmönsterfärg kan inte ställas in medan en lutning är inställd på cellen.";
        this["LE_InvalidOperationException_EncryptedWorkbooksNotSupported"] = "Arbetsboken som laddas är krypterad. Krypterade arbetsböcker stöds inte.";
        this["LE_InvalidOperationException_ExtensionDoesntMatchCurrentFormat"] = "Filnamnet som filen ska sparas till har ett tillägg som inte matchar det aktuella formatet på arbetsboken.";
        this["LE_InvalidOperationException_FixedValuesFilterMustAcceptAValue"] = "Kan inte ha ett FixedValuesFilter som inte tillåter några värden.";
        this["LE_InvalidOperationException_FormattedStringAlreadyOwned"] = "Den formaterade strängen är redan värdet för en annan cell eller kommentar. Klona den först och applicera klonen på den nya cellen eller formen.";
        this["LE_InvalidOperationException_FormattedStringNotOwned"] = "Det går inte att komma åt teckensnittsegenskaperna medan den formaterade strängen eller texten inte tillämpas på en cell eller form.";
        this["LE_InvalidOperationException_FormattedTextAlreadyOwned"] = "Den formaterade strängen är redan värdet på en annan form. Klona den först och applicera klonen på den nya cellen eller formen.";
        this["LE_InvalidOperationException_FormulaAlreadyOwned"] = "Formeln tillämpas redan på en annan cell.";
        this["LE_InvalidOperationException_FormulaReferencesInvalidCells"] = "Formeln refererar till celler som inte stöds i arbetsbokens nuvarande filformat.";
        this["LE_InvalidOperationException_GetBoundsBeforeAnchorsSet"] = "Formen har inga gränser innan ankarcellerna har angetts.";
        this["LE_InvalidOperationException_HiddenWorksheetCannotBeSelected"] = "Ett dolt kalkylblad kan inte väljas.";
        this["LE_InvalidOperationException_HyperlinkSealed"] = "Hyperlänken kan inte ändras.";
        this["LE_InvalidOperationException_ImageDisposed"] = "En bild av ett arbetsblad har tagits bort eller är ogiltigt.";
        this["LE_InvalidOperationException_InputCellsBothNull"] = "Kolumninmatnings- och radinmatningscellerna kan inte båda vara null.";
        this["LE_InvalidOperationException_InputCellsInTable"] = "Inmatningscellerna får inte vara i datatabellcellerna.";
        this["LE_InvalidOperationException_InputCellsSame"] = "Kolumninmatnings- och radinmatningscellerna kan inte vara samma cell.";
        this["LE_InvalidOperationException_InvalidCharacterRange"] = "Teckenintervallet som teckensnittet täcker finns inte längre i den formaterade strängen.";
        this["LE_InvalidOperationException_InvalidFormatOptionsInWholeTableArea"] = "Det angivna FormatOptions-formatvärdet är ogiltigt på WholeTable-tabellområdet.";
        this["LE_InvalidOperationException_InvalidFormatString"] = "Den angivna formatsträngen är ogiltig.";
        this["LE_InvalidOperationException_InvalidFormatString_GetTextCall"] = "Cellens formatsträng är ogiltig. För att meddelas om detta när FormatString är inställt, ställ in Workbook.ValidateFormatStrings till True.";
        this["LE_InvalidOperationException_InvalidForWorkbookFormat"] = "Formeln är inte giltig för angivet format.";
        this["LE_InvalidOperationException_InvalidHeaderCellColumnAreaFormatProperty"] = "Värdena för de övre gränserna kan inte ställas in i HeaderCell-tabellkolumnområdesformatet.";
        this["LE_InvalidOperationException_InvalidHeaderRowAreaFormatProperty"] = "Värdena för de övre gränserna kan inte ställas in i formatet för HeaderRow-tabellområdesformatet.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFontProperty"] = "Teckensnittsvärdet {0} kan inte ställas in i tabellstilområden.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatOptions"] = "Det angivna FormatOptions-formatvärdet är ogiltigt i tabellstilområden.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatProperty"] = "Formatvärdet {0} kan inte ställas in i tabellstilområden.";
        this["LE_InvalidOperationException_InvalidTotalsCellColumnAreaFormatProperty"] = "De nedre gränsvärdena kan inte ställas in i TotalCell-tabellkolumnområdesformatet.";
        this["LE_InvalidOperationException_InvalidTotalsRowAreaFormatProperty"] = "Värdena för de nedre gränserna kan inte ställas in på TotalsRow-tabellområdesformatet.";
        this["LE_InvalidOperationException_InvalidWholeTableAreaFormatProperty"] = "Endast kantgränsvärdena kan ställas in på WholeTable-tabellområdesformatet.";
        this["LE_InvalidOperationException_LastColumnBeforeFirst"] = "Den sista kolumnen i regionen ska vara större än eller lika med den första kolumnen.";
        this["LE_InvalidOperationException_LastRowBeforeFirst"] = "Den sista raden i regionen ska vara större än den första raden.";
        this["LE_InvalidOperationException_MaxCellFormats"] = "Det maximala antalet distinkta cellformat i Excel-filen är: {0}";
        this["LE_InvalidOperationException_MaxColors"] = "Det maximala antalet färger i en Excel-arbetsbok är: {0}";
        this["LE_InvalidOperationException_MaxColumns"] = "Det maximala antalet kolumner i ett Excel-kalkylblad med det aktuella formatet är: {0}";
        this["LE_InvalidOperationException_MaxFonts"] = "Antalet teckensnitt i en arbetsbok får inte överstiga {0}.";
        this["LE_InvalidOperationException_MaxFormattedStrings"] = "Antalet unika strängar i en arbetsbok får inte överstiga {0}.";
        this["LE_InvalidOperationException_MaxRows"] = "Det maximala antalet rader i ett Excel-kalkylblad med det aktuella formatet är: {0}";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableLeftColumn"] = "Cellkolumnen längst till vänster som anges för datatabellen innehåller en sammanfogad cellregion som sträcker sig utanför området för dessa celler.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopLeftCell"] = "Den övre vänstra cellen bland cellerna som anges för datatabellen är en del av en sammanfogad cell som sträcker sig utanför området för dessa celler.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopRow"] = "Den översta raden av celler som anges för datatabellen innehåller en sammanfogad cellregion som sträcker sig utanför området för dessa celler.";
        this["LE_InvalidOperationException_MergedCellsAppliedInTable"] = "Celler i en tabell kan inte fogas samman.";
        this["LE_InvalidOperationException_NamedReferenceNameAlreadyExists"] = "Namnet \"{0}\" används av en tabell eller namngiven referens i arbetsboken med samma omfång.";
        this["LE_InvalidOperationException_NoRegionHasBeenSpecified"] = "Ingen region har specificerats.";
        this["LE_InvalidOperationException_NoWorksheetContextToSetShapeBounds"] = "Denna överbelastning av Add kan inte anropas när figursamlingen inte är associerad med ett kalkylblad. Kalla på överbelastningen av Add som tar en kalkylbladsinstans.";
        this["LE_InvalidOperationException_OverlappingTable"] = "Det finns en annan tabell i det angivna området.";
        this["LE_InvalidOperationException_ReadOnlyFont"] = "Teckensnittsobjektet är skrivskyddat och kan inte ändras.";
        this["LE_InvalidOperationException_ReadOnlyFormat"] = "Formatobjektet är skrivskyddat och kan inte ändras.";
        this["LE_InvalidOperationException_RelativeR1C1AddressNeedsOriginCell"] = "För att analysera en relativ R1C1-adress måste en ursprungscell anges.";
        this["LE_InvalidOperationException_ResizedTableMustBeOnWorksheet"] = "Tabellen måste finnas på ett kalkylblad för att ändra storlek.";
        this["LE_InvalidOperationException_ResolvedFormatCannotBeModified"] = "Det lösta cellformatet är skrivskyddat och kan inte ändras.";
        this["LE_InvalidOperationException_ShapeCantChangeOrientation"] = "Denna form kan inte roteras eller vändas.";
        this["LE_InvalidOperationException_ShapeFillSolidIsImmutable"] = "ShapeFillSolid är oföränderlig. Skapa en ny ShapeFillSolid istället.";
        this["LE_InvalidOperationException_ShapeInAnotherCollection"] = "Formen finns redan i ett annat kalkylblad eller formgrupp.";
        this["LE_InvalidOperationException_ShapeOutlineSolidIsImmutable"] = "ShapeOutlineSolid är oföränderlig. Skapa en ny ShapeOutlineSolid istället.";
        this["LE_InvalidOperationException_ShowFilterUIWhileHeaderRowHidden"] = "Filtergränssnittet kan inte visas medan rubrikraden är dold.";
        this["LE_InvalidOperationException_SystemColorsAccessedFromWrongThread"] = "Kan inte komma åt SystemColors genom en bakgrundstråd.";
        this["LE_InvalidOperationException_TableReferenceMustBeUsedFromInsideTable"] = "Formeln har en strukturerad tabellreferens som endast kan användas inifrån en tabell.";
        this["LE_InvalidOperationException_TableReferenceToMissingTable"] = "Formeln har en hänvisning till en tabell som inte finns.";
        this["LE_InvalidOperationException_TableReferenceToMissingTableColumn"] = "Formeln har en hänvisning till en tabellkolumn som inte finns.";
        this["LE_InvalidOperationException_TopLeftAnchorFromOtherWorksheet"] = "Om du lägger till formen i denna samling placeras den på ett annat kalkylblad än dess TopLeftCornerCell.";
        this["LE_InvalidOperationException_TotalsRowCannotBeShownInLastRow"] = "Totalraden kan inte visas när tabellen upptar den sista raden i kalkylbladet.";
        this["LE_InvalidOperationException_UnknownBIFFFormat"] = "Arbetsboksfilen var i okänt BIFF-format.";
        this["LE_InvalidOperationException_UnsupportedBIFFFormat"] = "Arbetsfilen sparades i {0}, ett format som inte stöds.";
        this["LE_InvalidOperationException_WorkbookDataViolatesFormatLimits"] = "Arbetsboken innehåller data som överskrider gränserna som anges i det angivna formatet.";
        this["LE_InvalidRegion"] = "Ogiltig regionadress.";
        this["LE_InvalidValueType"] = "'{0}' är inte ett giltigt värde för ValueType-egenskapen.";
        this["LE_LocationNotSingleRowOrColumn"] = "Platsreferensen är inte giltig eftersom alla cellerna inte befinner sig på samma kolumn eller rad. Markera celler som alla befinner sig på en och samma kolumn eller rad.";
        this["LE_LocationOrDataAreaInvalid"] = "Referensen för platsen eller dataområdet är inte giltig.";
        this["LE_LocationWorksheetMismatch"] = "Platsen måste vara från detta kalkylblad.";
        this["LE_NotSingleTargetFormula"] = "Kan bara stödja enskilda målformler.";
        this["LE_NotSupportedException_CellType"] = "{0} är inte en celltyp som stöds.";
        this["LE_NotSupportedException_NoPackageFactory"] = "packageFactory kan inte vara null. När du laddar från Excel2007:s arbetsbokformat och använder monteringen Infragistics2.Documents.Excel måste du tillhandahålla en IPackageFactory för att hantera dataförpackningen. Om du använder DotNet Framework 3.0 eller högre, använd Infragistics3.Documents.Excel-montering istället så kommer förpackningen hanteras av WindowsBase-klassen.";
        this["LE_OpenPackagingNonConformanceException_ContainsDublinCoreRefinements"] = "IPackagePart är kärnegenskapsförhållandet och innehåller förbättringar av andra Dublin-kärnelement än 'dcterms: created' och 'dcterms: modified'.";
        this["LE_OpenPackagingNonConformanceException_ContainsXmlLanguageAttribute"] = "IPackagePart är kärnegenskapsförhållandet och innehåller element som innehåller attributet 'xml: lang'.";
        this["LE_OpenPackagingNonConformanceException_CorePropertiesRelationshipAlreadyProcessed"] = "IPackagePart är kärnegenskapsförhållandet, men en har redan bearbetats för tillhörande IPackage.";
        this["LE_OpenPackagingNonConformanceException_DuplicatePartName"] = "En IPackagePart med samma namn tillhör redan detta IPackage. Ett IPackage kan inte innehålla delar med identiska namn. Observera att kontrollen för dubbla värden är skiftlägeskänslig.";
        this["LE_OpenPackagingNonConformanceException_UsesMarkupCompatibilityNamespace"] = "IPackagePart är kärnegenskapsförhållandet och använder namnutrymmet Markup Compatibility.";
        this["LE_OpenPackagingNonConformanceException_XmlContainsDocumentTypeDefinition"] = "En IPackagePart innehöll en DTD (Document Type Definition) -deklaration. Reglerna för Open Packaging Convention förbjuder användningen av XML-innehåll med DTD-deklarationer.";
        this["LE_OpenPackagingNonConformanceException_XsiTypeAttributeInvalid"] = "IPackagePart är kärnegenskapsförhållandet och innehåller element som innehåller attributet 'xsi: type' eller innehåller element som krävs och inte innehåller attributet 'xsi: type'.";
        this["LE_OpenXmlSerializerNotLoaded"] = "Formatet '{0}' kräver en serialiserare som inte är laddad, i JavaScript ingår detta i filen infragistics.excel_serialization_openxml.js";
        this["LE_PercentageValueOutOfRange"] = "Procentandelsvärdet {0} måste vara mellan 0 och 100.";
        this["LE_PercentileValueOutOfRange"] = "Procentilvärdet {0} måste vara mellan 0 och 100.";
        this["LE_PercentRangeInvalid"] = "{0} måste vara mindre än {1}.";
        this["LE_RowsToRepeatAtTopAreOutsideAvailableRange"] = "Utskriftsalternativen har rader som upprepas längst upp och ligger utanför intervallet för tillgängliga rader i det nya formatet.";
        this["LE_Serialization_NoPrimaryCategoryAxis"] = "Diagramtypen '{0}' kräver en primär kategoriaxel. Lägg till en primär kategoriaxel i diagrammets AxisCollection.";
        this["LE_Serialization_NoPrimaryValueAxis"] = "Diagramtypen '{0}' kräver en primär värdeaxel. Lägg till en primärvärdeaxel i diagrammets AxisCollection.";
        this["LE_Series_ChartType_ChartEx"] = "{0} är ogiltig för en serie ChartType i ett '{1}' diagram.";
        this["LE_Series_ChartType_InvalidForComboChart"] = "Diagramtypen '{0}' är inte giltig för användning i ett kombinationsdiagram.";
        this["LE_Series_ChartType_NonCombo"] = "Diagramtypen för denna serie bestäms av överordnad diagramtyp ('{0}') och kan inte ställas in. Ställ istället in det överordnade diagrammets ChartType-egenskap till '{1}'.";
        this["LE_Series_Explosion_OutOfRange"] = "Explosionsegenskapen stöder värden i intervallet 0 till 400 inklusive.";
        this["LE_Series_InvalidAxisGroup"] = "Egenskapen AxisGroup kan inte ställas in på '{0}' eftersom en serie av typen '{1}' finns på samma axel, som inte kan dela samma axel som en serie av typen '{2}'.";
        this["LE_Series_InvalidSeriesTypeForComboChart"] = "En serie av typen '{0}' kan inte användas i ett kombinationsdiagram.";
        this["LE_Series_NoPrimaryAxis"] = "Minst en serie måste plottas på primäraxeln. Ändra AxisGroup-egenskapen till 'Primär' för minst en serie.";
        this["LE_Series_OwningSeries_NotClusteredColumn"] = "Egenskapen OwningSeries kan bara ställas in i en grupperad kolumnserie i ett Pareto-diagram.";
        this["LE_Series_OwningSeries_NotParetoLine"] = "Egenskapen OwningSeries kan bara ställas in för en linjeserie i ett Pareto-diagram.";
        this["LE_Series_OwningSeries_Same"] = "OwningSeries kan inte referera till sig själv.";
        this["LE_Series_SeriesType_ChartEx"] = "SeriesType-värdet '{0}' är ogiltigt för ett '{1}' -diagram.";
        this["LE_Series_SeriesType_NonCombo"] = "Egenskapen Type är endast tillämplig för kombinationsdiagram och kan inte ställas in för ett diagram av typen '{0}'.";
        this["LE_SetValue_Formula"] = "Värdetypen '{0}' är endast giltig för formler. Använd metoden SetFormula istället.";
        this["LE_SetValue_NonNumeric"] = "Värdeparametern är inte tillämplig för värdetypen '{0}'. Använd överbelastningen som bara tar en värdetyp.";
        this["LE_SetValue_Numeric"] = "Värdeparametern måste anges för värdetypen '{0}'. Använd överbelastningen som tar både värdet och värdetypen.";
        this["LE_WorksheetChart_ChartType_ChartEx_CannotTransition"] = "Egenskapen ChartType kan inte ändras från '{0}' till '{1}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_From"] = "Egenskapen ChartType för ett kombinationsdiagram kan inte ändras till '{0}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_To"] = "Ett diagram av typen '{0}' kan inte ändras till 'Combo'. Ett nytt kombinationsdiagram kan skapas med AddChart-metoden, men det går inte att ändra ett befintligt diagram till ett kombinationsdiagram.";
        this["LE_WorksheetChart_ComboChart_NotEnoughSeries"] = "Ett kombinationsdiagram kräver minst två serier. Lägg till fler objekt i diagrammets SeriesCollection.";
        this["LE_WorksheetChart_MissingRequiredAxis_StockCharts"] = "Ett '{0}' -diagram kräver en primär kategoriaxel, en primär värdeaxel, en sekundär kategoriaxel och en sekundär värdeaxel.";
        this["LE_WorksheetChart_NotEnoughSeries"] = "Ett '{0}' -diagram kräver minst {1} serier. För att kunna använda denna diagramtyp måste du lägga till fler objekt i SeriesCollection.";
        this["LE_WorksheetChart_Save_Series"] = "Ett '{0}' - diagram måste innehålla minst en serie. Lägg till fler serier i diagrammet innan du försöker spara arbetsboken.";
        this["LE_WorksheetChart_Save_Series_Pareto"] = "Ett \"Pareto\" -diagram måste innehålla en \"ColumnClustered\" -serie och en \"Pareto\" -serie. Lägg till lämplig serie i diagrammet innan du försöker spara arbetsboken.";
        this["LE_WorksheetChart_SetComboChartSourceData_InvalidChartTypes"] = "En eller flera av de angivna diagramtyperna är inte giltiga för användning i ett kombinationsdiagram: {0}";
        this["LE_WorksheetChart_SetComboChartSourceData_NotAComboChart"] = "SetComboChartSourceData kan inte anropas eftersom den aktuella diagramtypen är '{0}'. Ändra ChartType till 'Combo'.";
        this["LE_WorksheetChart_SetComboChartSourceData_SeriesChartTypesNotSupported"] = "Parametern seriesChartTypes måste vara null för ett {0} -diagram.";
        this["LE_WorksheetChart_SetSourceData_NotEnoughSeries"] = "Kombinationsdiagram kräver minst två serier. Ändra parametern 'dataRange' till ett intervall som resulterar i minst två serier.";
        this["LE_WorksheetChart_SetSourceData_TooManyAxes"] = "Den angivna kombinationen av diagramtyper kräver mer än det maximalt tillåtna antalet axlar.";
        this["LE_WorksheetChart_StockChartNotSupported"] = "Redigering av diagramtypen '{0}' stöds för närvarande inte.";
        this["LE_WorksheetChart_UnknownChartNotSupported"] = "Det här diagrammet laddades från ett befintligt kalkylblad, men dess ChartType kunde inte avläsas. Redigering av okända diagramtyper stöds för närvarande inte.";
        this["LER_ArgumentOutOfRangeException_OutlineLevel"] = "Konturnivån måste vara mellan 0 och 7.";
        this["LER_Exception_KeyNotFound"] = "Nyckeln hittades inte.";
        this["LER_Exception_MergedRegionsOverlap"] = "Sammanfogade cellregioner kan inte överlappa varandra.";
        this["TableHeaderRowDescription"] = "rubrikrad";
        this["TableInsertRowDescription"] = "infoga rad";
        this["TableTotalsRowDescription"] = "totalrad";
        this["Value_UCErrorCode_Div"] = "#DIV/0!";
        this["Value_UCErrorCode_Fail"] = "#FAIL";
        this["Value_UCErrorCode_NA"] = "#N/A";
        this["Value_UCErrorCode_Name"] = "#NAME?";
        this["Value_UCErrorCode_Null"] = "#NULL!";
        this["Value_UCErrorCode_Num"] = "#NUM!";
        this["Value_UCErrorCode_Ok"] = "#OK";
        this["Value_UCErrorCode_Reference"] = "#REF!";
        this["Value_UCErrorCode_Unknown"] = "#N/A";
        this["Value_UCErrorCode_Value"] = "#VALUE!";
        this["WorkbookColorInfo_Automatic_Description"] = "Automatisk";
        this["WorkbookColorInfo_Description"] = "ColorInfo: {0}";
        this["WorkbookColorInfo_WithTint_Description"] = "ColorInfo: {0}, Ton: {1}";
        this["WorksheetShapeSerialization_GroupName"] = "Grupp";
        this["WorksheetShapeSerialization_ImageName"] = "Bild";
        this["WorksheetShapeSerialization_ShapeName"] = "Form";
    }
    return ExcelLocaleSv;
}());
export { ExcelLocaleSv };
