// igniteui-react-excel/esm5/lib/ExcelLocalePt.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
var ExcelLocalePt = /** @class */ /*@__PURE__*/ (function () {
    function ExcelLocalePt() {
        this["DefaultTotalLabel"] = "Total";
        this["Error_AnchoredReference"] = "Tentativa de acesso a {0} como uma referência não ancorada";
        this["Error_ExplicitThis"] = "{0} - Explícito Não suportado";
        this["Error_ExternalRange"] = "Intervalo contém Referência Externa";
        this["Error_FormulaNotSupported"] = "{0} - Tentativa de aceder à fórmula numa referência que não suporta fórmulas";
        this["Error_Internal"] = "Ocorreu um erro interno no método {0}.";
        this["Error_InvalidFunction"] = "Função inválida '{0}' em {1}.";
        this["Error_InvalidFunctionArgCountTooFew"] = "A função '{0}' referenciada em {1} não contém o número correto de argumentos. A função requer um mínimo de {2} parâmetro(s).";
        this["Error_InvalidFunctionArgCountTooMany"] = "A função '{0}' referenciada em {1} não contém o número correto de argumentos. A contagem de argumentos não deve exceder {2} parâmetro(s).";
        this["Error_InvalidNumberSystemDigit"] = "Dígito inválido de '{0}' encontrado no valor '{1}' para o sistema numérico com base em {2}.";
        this["Error_InvalidOperand"] = "Operando inválido";
        this["Error_InvalidOperatorArgCount"] = "A função '{0}' deve aceitar {1} argumentos.";
        this["Error_InvalidReference"] = "Referência inválida ou indisponível";
        this["Error_Location"] = "linha {0} col {1}";
        this["Error_MethodNotSupported"] = "{0} não suporta {1}.";
        this["Error_NoCurrentRangeElement"] = "Nenhum elemento de intervalo atual";
        this["Error_NullFunctionResult"] = "A avaliação de '{0}' devolveu um valor nulo.";
        this["Error_NullOperand"] = "Operando nulo";
        this["Error_ParseRef"] = "Referência desconhecida em {0}.";
        this["Error_RangeFromLast"] = "fromT.Last não definido";
        this["Error_RangeFromNull"] = "RefRange: de é nulo";
        this["Error_RangeFromRefBase"] = "RefRange: de não é um RefBase: {0}";
        this["Error_RangeLast"] = "RefRange Validar: últimas cadeias de identificação [{0}] != [{1}]";
        this["Error_RangeNormalizeScope"] = "RefRange Normalizar: âmbito incorreto [{0}/{1}]";
        this["Error_RangeToNull"] = "RefRange: para é nulo";
        this["Error_RangeToRefBase"] = "RefRange: para não é um RefBase: {0}";
        this["Error_RangeTuple"] = "RefRange Validar: cadeias de identificação [{0}] != [{1}]";
        this["Error_RangeValidateCount"] = "RefRange Validar: de contagem {0} != para contagem {1}";
        this["Error_ReadOnlyReference"] = "{0} - Tentativa de definir valor na referência só de leitura";
        this["Error_ReferenceNotEumerable"] = "{0} não é uma referência enumerável";
        this["Error_RefMalformed"] = "{0} - Referência mal formada {1}";
        this["Error_RP_ContainsBase"] = "RefParser.Contains - Base vazia";
        this["Error_RP_ContainsTarget"] = "RefParser.Contains - Alvo vazio";
        this["Error_RP_EmptyElement"] = "Referência inválida {0}. Elemento vazio encontrado na localização {1}.";
        this["Error_RP_EmptyReference"] = "Referência inválida {0}. A referência está vazia.";
        this["Error_RP_MalformedExtra"] = "RefParser Referência mal formada: Vários [{0}] encontrados em [{1}]";
        this["Error_RP_MalformedUnexpected"] = "RefParser Referência mal formada: Inesperado [{0}] encontrado em [{1}]";
        this["Error_RP_MergeRelativeBase"] = "RefParser.Merge: Base é relativa";
        this["Error_RP_MissingCloseParenthesis"] = "Referência inválida {0}. Falta parêntese de fecho.";
        this["Error_RP_MissingEndQuotes"] = "Referência inválida {0}. Faltam aspas de fecho.";
        this["Error_RT_Expected"] = "RefTuple: Esperado [{0}] em [{1}]";
        this["Error_RT_InvalidScope"] = "Âmbito inválido: {0}.";
        this["Error_RT_InvalidScope_CharactersAfterClosingQuotes"] = "Caracteres encontrados após fechar aspas no âmbito {0} no local {1}.";
        this["Error_RT_InvalidTuple_CharactersAfterScopeEnd"] = "Elemento inválido {0}. Caracteres encontrados no final do âmbito.";
        this["Error_RT_InvalidTuple_MissingCloseParenthesis"] = "Elemento inválido {0}. Falta parêntese de fecho.";
        this["Error_RT_InvalidTuple_MissingEndQuotes"] = "Elemento inválido {0}. Faltam aspas de fecho.";
        this["Error_RT_InvalidTuple_NamePortionEmpty"] = "Elemento inválido {0}. Parte do nome do elemento vazia.";
        this["Error_RT_InvalidTuple_ScopeIndexLarge"] = "Elemento inválido {0}. O índice {1} é inválido: {2}.";
        this["Error_RT_InvalidTuple_UnescapedCharacter"] = "Caractere {0} sem escape encontrado em {1} na localização {2}.";
        this["Error_RT_NullRefName"] = "RefTuple: RefName nulo";
        this["Error_RT_Unexpected"] = "RefTuple: inesperado [{0}] em [{1}]";
        this["Error_ScopeAllNotLast"] = "{0} - Âmbito Todos devem aparecer no último elemento {1}";
        this["Error_UCErrorCode_Div"] = "Dividir por zero";
        this["Error_UCErrorCode_Fail"] = "Falhar";
        this["Error_UCErrorCode_NA"] = "Não aplicável";
        this["Error_UCErrorCode_Name"] = "Nome inválido";
        this["Error_UCErrorCode_Null"] = "Valor nulo";
        this["Error_UCErrorCode_Num"] = "Valores numéricos inválidos na função ou fórmula.";
        this["Error_UCErrorCode_Ok"] = "Sucesso";
        this["Error_UCErrorCode_Reference"] = "Referência inválida ou indisponível";
        this["Error_UCErrorCode_Unknown"] = "Código inválido";
        this["Error_UCErrorCode_Value"] = "Tipo de argumento ou operando incorreto.";
        this["Error_UnexpectedScope"] = "{0} - Âmbito inesperado para Elemento Principal {1}";
        this["Error_UnknownRange"] = "Tipo de intervalo desconhecido";
        this["Error_UnknownRefType"] = "{0} - RefType desconhecido {1}";
        this["Error_UnknownScope"] = "{0} - Âmbito desconhecido {1}";
        this["Error_UnresolvedExternal"] = "Unresolved external reference";
        this["Func_abs_Arg_0"] = "número";
        this["Func_abs_ArgDesc_0"] = "O número real do qual obter o valor absoluto.";
        this["Func_abs_Category"] = "Matemática";
        this["Func_abs_CategoryURL"] = "Matemática";
        this["Func_abs_Desc"] = "Devolve o valor absoluto de um número. O valor absoluto de um número é o número sem o seu sinal.";
        this["Func_acos_Arg_0"] = "número";
        this["Func_acos_ArgDesc_0"] = "O cosseno do ângulo a ser obtido, que deve ser de -1 a 1.";
        this["Func_acos_Category"] = "Matemática";
        this["Func_acos_CategoryURL"] = "Matemática";
        this["Func_acos_Desc"] = "Devolve o arco de cosseno, ou cosseno inverso, de um número. O arco de cosseno é o ângulo cujo cosseno é um número especificado. O ângulo devolvido é dado em radianos no intervalo de zero a pi. Para converter o resultado de radianos em graus, multiplique por 180/PI().";
        this["Func_acosh_Arg_0"] = "número";
        this["Func_acosh_ArgDesc_0"] = "Um número real maior que 1 para o qual encontrar o cosseno hiperbólico inverso.";
        this["Func_acosh_Category"] = "Matemática";
        this["Func_acosh_CategoryURL"] = "Matemática";
        this["Func_acosh_Desc"] = "Devolve o cosseno hiperbólico inverso do número indicado. O cosseno hiperbólico inverso é o ângulo cujo cosseno hiperbólico é o número especificado.";
        this["Func_and_Arg_0"] = "lógico {0}";
        this["Func_and_ArgDesc_0"] = "Um valor lógico (verdadeiro ou falso).";
        this["Func_and_Category"] = "Lógico";
        this["Func_and_CategoryURL"] = "Lógico";
        this["Func_and_Desc"] = "Devolve verdadeiro se todos os valores forem verdadeiros ou devolve falso se um ou mais valores forem falsos.";
        this["Func_asin_Arg_0"] = "número";
        this["Func_asin_ArgDesc_0"] = "O seno do ângulo a ser obtido, que deve ser de -1 a 1 radianos.";
        this["Func_asin_Category"] = "Matemática";
        this["Func_asin_CategoryURL"] = "Matemática";
        this["Func_asin_Desc"] = "Devolve o arco de seno, ou seno inverso, de um número. O arco de seno é o ângulo cujo seno é o número especificado. O ângulo devolvido é dado em radianos no intervalo de -pi/2 a pi/2. Para expressar o arco de seno em graus, multiplique o resultado por 180/PI().";
        this["Func_asinh_Arg_0"] = "número";
        this["Func_asinh_ArgDesc_0"] = "Um número real para o qual encontrar o seno hiperbólico inverso.";
        this["Func_asinh_Category"] = "Matemática";
        this["Func_asinh_CategoryURL"] = "Matemática";
        this["Func_asinh_Desc"] = "Devolve o seno hiperbólico inverso do número indicado. O seno hiperbólico inverso é o ângulo cujo seno hiperbólico é o número especificado.";
        this["Func_atan_Arg_0"] = "número";
        this["Func_atan_ArgDesc_0"] = "A tangente do ângulo a ser obtido.";
        this["Func_atan_Category"] = "Matemática";
        this["Func_atan_CategoryURL"] = "Matemática";
        this["Func_atan_Desc"] = "Devolve o arco de tangente, ou tangente inversa, de um número. O arco de tangente é o ângulo cuja tangente é o número especificado. O ângulo devolvido é dado em radianos na faixa de -pi/2 a pi/2. Para expressar o arco de tangente em graus, multiplique o resultado por 180/PI().";
        this["Func_atan2_Arg_0"] = "x";
        this["Func_atan2_Arg_1"] = "y";
        this["Func_atan2_ArgDesc_0"] = "A coordenada x do ponto.";
        this["Func_atan2_ArgDesc_1"] = "A coordenada y do ponto.";
        this["Func_atan2_Category"] = "Matemática";
        this["Func_atan2_CategoryURL"] = "Matemática";
        this["Func_atan2_Desc"] = "Devolve o arco de tangente, ou tangente inversa, das coordenadas x e y especificadas. O arco de tangente é o ângulo do eixo x para uma linha que contém a origem (0, 0) e um ponto com coordenadas (x_num, y_num). O ângulo é dado em radianos entre -pi e pi, excluindo -pi.";
        this["Func_atanh_Arg_0"] = "número";
        this["Func_atanh_ArgDesc_0"] = "Um número real entre -1 e 1 para o qual obter a tangente hiperbólica inversa.";
        this["Func_atanh_Category"] = "Matemática";
        this["Func_atanh_CategoryURL"] = "Matemática";
        this["Func_atanh_Desc"] = "Devolve a tangente hiperbólica inversa do número indicado. A tangente hiperbólica inversa é o ângulo cuja tangente hiperbólica é um número especificado.";
        this["Func_avedev_Arg_0"] = "número {0}";
        this["Func_avedev_ArgDesc_0"] = "Número1, número2, ... são um ou mais valores numéricos ou referências a valores numéricos.";
        this["Func_avedev_Category"] = "Estatística";
        this["Func_avedev_CategoryURL"] = "Estatística";
        this["Func_avedev_Desc"] = "Devolve o desvio médio da média de todos os valores numéricos.";
        this["Func_average_Arg_0"] = "número {0}";
        this["Func_average_ArgDesc_0"] = "Um número ou intervalo de números para os quais encontrar a média com os outros argumentos.";
        this["Func_average_Category"] = "Estatística";
        this["Func_average_CategoryURL"] = "Estatística";
        this["Func_average_Desc"] = "Devolve a média (média aritmética) de um ou mais valores.";
        this["Func_bin2dec_Arg_0"] = "Número";
        this["Func_bin2dec_ArgDesc_0"] = "Uma expressão de texto que representa um número binário.";
        this["Func_bin2dec_Category"] = "Engenharia";
        this["Func_bin2dec_CategoryURL"] = "Engenharia";
        this["Func_bin2dec_Desc"] = "Devolve o valor decimal de uma representação de texto de um número no sistema numérico binário.";
        this["Func_bin2hex_Arg_0"] = "Número";
        this["Func_bin2hex_Arg_1"] = "Locais";
        this["Func_bin2hex_ArgDesc_0"] = "Uma expressão de texto que representa um número binário.";
        this["Func_bin2hex_ArgDesc_1"] = "O número de caracteres a utilizar para representar o valor. Se não for especificado, será utilizado o número mínimo de caracteres necessário para representar o valor. Isso permite especificar zeros à esquerda.";
        this["Func_bin2hex_Category"] = "Engenharia";
        this["Func_bin2hex_CategoryURL"] = "Engenharia";
        this["Func_bin2hex_Desc"] = "Converte uma expressão de texto que representa um número binário numa expressão de texto que representa o mesmo número no sistema numérico hexadecimal.";
        this["Func_bin2oct_Arg_0"] = "Número";
        this["Func_bin2oct_Arg_1"] = "Locais";
        this["Func_bin2oct_ArgDesc_0"] = "Uma expressão de texto que representa um número binário.";
        this["Func_bin2oct_ArgDesc_1"] = "O número de caracteres a utilizar para representar o valor. Se não for especificado, será utilizado o número mínimo de caracteres necessário para representar o valor. Isso permite especificar zeros à esquerda.";
        this["Func_bin2oct_Category"] = "Engenharia";
        this["Func_bin2oct_CategoryURL"] = "Engenharia";
        this["Func_bin2oct_Desc"] = "Converte uma expressão de texto que representa um número binário numa expressão de texto que representa o mesmo número no sistema numérico octal.";
        this["Func_binomdist_Arg_0"] = "Número";
        this["Func_binomdist_Arg_1"] = "Avaliações";
        this["Func_binomdist_Arg_2"] = "Probabilidade";
        this["Func_binomdist_Arg_3"] = "Acumulado";
        this["Func_binomdist_ArgDesc_0"] = "O número de sucessos nas avaliações.";
        this["Func_binomdist_ArgDesc_1"] = "O número de avaliações independentes.";
        this["Func_binomdist_ArgDesc_2"] = "A probabilidade de sucesso em cada avaliação.";
        this["Func_binomdist_ArgDesc_3"] = "Um valor lógico que determina a forma da função. Se verdadeiro, a função irá devolver a distribuição cumulativa, que é a probabilidade de haver no máximo 'número' sucessos; caso contrário, irá devolver a massa de probabilidade, que é a probabilidade de haver 'número' sucessos.";
        this["Func_binomdist_Category"] = "Estatística";
        this["Func_binomdist_CategoryURL"] = "Estatística";
        this["Func_binomdist_Desc"] = "Devolve o número de combinações possíveis (com repetições) dado um conjunto de itens e um número de itens escolhidos a partir desse conjunto.";
        this["Func_CategoryURL_Template"] = "ms-help://INFRAGISTICS_HELP/NetAdvantagev{2}{3}/Win/WinElements/WinCalcManager_{0}_Functions.html#FUNC_{1}";
        this["Func_ceiling_Arg_0"] = "número";
        this["Func_ceiling_Arg_1"] = "significância";
        this["Func_ceiling_ArgDesc_0"] = "O número a arredondar.";
        this["Func_ceiling_ArgDesc_1"] = "O múltiplo para o qual arredondar. Se omitido, assume-se que seja um.";
        this["Func_ceiling_Category"] = "Matemática";
        this["Func_ceiling_CategoryURL"] = "Matemática";
        this["Func_ceiling_Desc"] = "Devolve o menor múltiplo da significância (o predefinido é 1) maior ou igual ao número especificado.";
        this["Func_char_Arg_0"] = "códigoCar";
        this["Func_char_ArgDesc_0"] = "O número a ser convertido num carácter.";
        this["Func_char_Category"] = "TextAndData";
        this["Func_char_CategoryURL"] = "Text_and_Data";
        this["Func_char_Desc"] = "Devolve o carácter especificado pelo número de código do conjunto de caracteres para o seu computador.";
        this["Func_choose_Arg_0"] = "IndexNum";
        this["Func_choose_Arg_1"] = "Value{0}";
        this["Func_choose_ArgDesc_0"] = "O índice com base em 1 no resto dos argumentos para o valor a usar.";
        this["Func_choose_ArgDesc_1"] = "Um valor que pode ser escolhido dependendo do valor NúmeroÍndice.";
        this["Func_choose_Category"] = "PesquisaEReferência";
        this["Func_choose_CategoryURL"] = "Lookup_And_Reference";
        this["Func_choose_Desc"] = "Devolve um dos valores fornecidos nos seus argumentos com base no número no primeiro argumento.";
        this["Func_code_Arg_0"] = "texto";
        this["Func_code_ArgDesc_0"] = "A expressão de texto que terá o seu primeiro carácter avaliado.";
        this["Func_code_Category"] = "TextAndData";
        this["Func_code_CategoryURL"] = "Text_and_Data";
        this["Func_code_Desc"] = "Devolve um código numérico para o primeiro carácter numa expressão de texto utilizando o conjunto de caracteres utilizado pelo seu computador.";
        this["Func_combin_Arg_0"] = "Número";
        this["Func_combin_Arg_1"] = "NumberChosen";
        this["Func_combin_ArgDesc_0"] = "O número de itens.";
        this["Func_combin_ArgDesc_1"] = "O número de itens escolhidos em cada combinação.";
        this["Func_combin_Category"] = "Matemática";
        this["Func_combin_CategoryURL"] = "Matemática";
        this["Func_combin_Desc"] = "Devolve o número de combinações possíveis, dado um conjunto de itens e um número de itens escolhidos a partir desse conjunto.";
        this["Func_combina_Arg_0"] = "Número";
        this["Func_combina_Arg_1"] = "NumberChosen";
        this["Func_combina_ArgDesc_0"] = "O número de itens.";
        this["Func_combina_ArgDesc_1"] = "O número de itens escolhidos em cada combinação.";
        this["Func_combina_Category"] = "Matemática";
        this["Func_combina_CategoryURL"] = "Matemática";
        this["Func_combina_Desc"] = "Devolve o número de combinações possíveis (com repetições) dado um conjunto de itens e um número de itens escolhidos a partir desse conjunto.";
        this["Func_complex_Arg_0"] = "RealNumber";
        this["Func_complex_Arg_1"] = "ImaginaryNumber";
        this["Func_complex_Arg_2"] = "Sufixo";
        this["Func_complex_ArgDesc_0"] = "O coeficiente real do número complexo.";
        this["Func_complex_ArgDesc_1"] = "O coeficiente imaginário do número.";
        this["Func_complex_ArgDesc_2"] = "Especifica o sufixo a utilizar. Os valores aceitáveis são \"i\" ou \"j\". Se omitido, \"i\" é utilizado. Tenha em consideração que o sufixo diferencia maiúsculas de minúsculas; maiúsculas \"I\" e \"J\" não são permitidas e resultarão num erro de #VALOR.";
        this["Func_complex_Category"] = "Engenharia";
        this["Func_complex_CategoryURL"] = "Engenharia";
        this["Func_complex_Desc"] = "Devolve um número complexo representado como uma expressão de texto no formato \"x + yi\" ou \"x + yj\" combinando um número real e imaginário.";
        this["Func_concatenate_Arg_0"] = "texto {0}";
        this["Func_concatenate_ArgDesc_0"] = "Uma expressão de texto ou intervalo de expressões de texto a serem unidas numa única expressão de texto.";
        this["Func_concatenate_Category"] = "TextAndData";
        this["Func_concatenate_CategoryURL"] = "Text_and_Data";
        this["Func_concatenate_Desc"] = "Junta todos os argumentos numa expressão de texto.";
        this["Func_convert_Arg_0"] = "Número";
        this["Func_convert_Arg_1"] = "FromUnit";
        this["Func_convert_Arg_2"] = "ToUnit";
        this["Func_convert_ArgDesc_0"] = "O valor a converter.";
        this["Func_convert_ArgDesc_1"] = "A unidade na qual o número é dado.\r\n\r\nUnidades de peso e massa: \"g\" = grama; \"sg\" = slug; \"lbm\" = libra-massa (avoirdupois); \"u\" = U (unidade de massa atómica); \"ozm\" = Onça-massa (avoirdupois)\r\nUnidades de distância: \"m\" = Metro; \"mi\" = Milha terrestre; \"Nmi\" = Milha náutica; \"in\" = Polegada; \"ft\" = Pé; \" yd \" = Jarda; \"ang\" = Angstrom; \"Pica\" = Pica (1/72 pol.)\r\nUnidades de tempo: \"yr\" = Ano; \"day\" = Dia; \"hr\" = Hora; \"mn\" = Minuto; \"sec\" = Segundo\r\nUnidades de pressão: \"Pa\" (ou \"p\") = Pascal; \"atm\" (ou \"at\") = Atmosfera; \"mmHg\" = mm de Mercúrio\r\nUnidades de força: \"N\" = Newton; \"dyn\" (ou \"dy\") = Dyne; \"lbf\" = Libra-força\r\nUnidades de energia: \"J\" = Joule; \"e\" = Erg; \"c\" = Caloria termodinâmica; \"cal\" = calorias de TI; \"eV\" (ou \"ev\") = Volt de eletrão; \"HPh\" (ou \"hh\") = Cavalo-hora; \"Wh\" (ou \"wh\") = Watt-hora; \"flb\" = Libra-pé; \"BTU\" (ou \"btu\") = BTU\r\nUnidades de potência: \"HP\" (ou \"h\") = Cavalo-força; \"W\" (ou \"w\") = Watt\r\nUnidades de magentismo: \"T\" = Tesla; \"ga\" = Gauss\r\nUnidades de temperatura: \"C\" (ou \"cel\") = Grau Celsius; \"F\" (ou \"fah\") = Grau Fahrenheit; \"K\" (ou \"kel\") = Kelvin\r\nUnidades de medida de líquidos: \"tsp\" = Colher de chá; \"tbs\" = Colher de sopa; \"oz\" = Onça líquida; \"cup\" = Chávena; \"pt\" (ou \"us_pt\") = Pinto (E.U.A.); \" uk_pt \" = Pinto (R.U.); \"qt\" = Quarto; \"gal\" = Galão; \"l\" (ou \"lt\") = Litro\r\n\r\nQualquer unidade métrica pode ser prefixada com um dos seguintes: exa (\"E\") = 1E + 18; peta (\"P\") = 1E + 15; tera (\"T\") = 1E + 12; giga (\"G\") = 1E + 09; mega (\"M\") = 1E + 06; quilo (\"k\") = 1E + 03; hecto (\"h\") = 1E + 02; deca (\"e\") = 1E + 01; deci (\"d\") = 1E-01; centi (\"c\") = 1E-02; milli (\"m\") = 1E-03; micro (\"u\") = 1E-06; nano (\"n\") = 1E-09; pico (\"p\") = 1E-12; femto (\"f\") = 1E-15; atto (\"a\") = 1E-18\r\n    ";
        this["Func_convert_ArgDesc_2"] = "As unidades na quais devolver o número convertido.\r\n\r\nUnidades de peso e massa: \"g\" = grama; \"sg\" = slug; \"lbm\" = libra-massa (avoirdupois); \"u\" = U (unidade de massa atómica); \"ozm\" = Onça-massa (avoirdupois)\r\nUnidades de distância: \"m\" = Metro; \"mi\" = Milha terrestre; \"Nmi\" = Milha náutica; \"in\" = Polegada; \"ft\" = Pé; \" yd \" = Jarda; \"ang\" = Angstrom; \"Pica\" = Pica (1/72 pol.)\r\nUnidades de tempo: \"yr\" = Ano; \"day\" = Dia; \"hr\" = Hora; \"mn\" = Minuto; \"sec\" = Segundo\r\nUnidades de pressão: \"Pa\" (ou \"p\") = Pascal; \"atm\" (ou \"at\") = Atmosfera; \"mmHg\" = mm de Mercúrio\r\nUnidades de força: \"N\" = Newton; \"dyn\" (ou \"dy\") = Dyne; \"lbf\" = Libra-força\r\nUnidades de energia: \"J\" = Joule; \"e\" = Erg; \"c\" = Caloria termodinâmica; \"cal\" = calorias de TI; \"eV\" (ou \"ev\") = Volt de eletrão; \"HPh\" (ou \"hh\") = Cavalo-hora; \"Wh\" (ou \"wh\") = Watt-hora; \"flb\" = Libra-pé; \"BTU\" (ou \"btu\") = BTU\r\nUnidades de potência: \"HP\" (ou \"h\") = Cavalo-força; \"W\" (ou \"w\") = Watt\r\nUnidades de magentismo: \"T\" = Tesla; \"ga\" = Gauss\r\nUnidades de temperatura: \"C\" (ou \"cel\") = Grau Celsius; \"F\" (ou \"fah\") = Grau Fahrenheit; \"K\" (ou \"kel\") = Kelvin\r\nUnidades de medida de líquidos: \"tsp\" = Colher de chá; \"tbs\" = Colher de sopa; \"oz\" = Onça líquida; \"cup\" = Chávena; \"pt\" (ou \"us_pt\") = Pinto (E.U.A.); \" uk_pt \" = Pinto (R.U.); \"qt\" = Quarto; \"gal\" = Galão; \"l\" (ou \"lt\") = Litro\r\n\r\nQualquer unidade métrica pode ser prefixada com um dos seguintes: exa (\"E\") = 1E + 18; peta (\"P\") = 1E + 15; tera (\"T\") = 1E + 12; giga (\"G\") = 1E + 09; mega (\"M\") = 1E + 06; quilo (\"k\") = 1E + 03; hecto (\"h\") = 1E + 02; deca (\"e\") = 1E + 01; deci (\"d\") = 1E-01; centi (\"c\") = 1E-02; milli (\"m\") = 1E-03; micro (\"u\") = 1E-06; nano (\"n\") = 1E-09; pico (\"p\") = 1E-12; femto (\"f\") = 1E-15; atto (\"a\") = 1E-18\r\n    ";
        this["Func_convert_Category"] = "Engenharia";
        this["Func_convert_CategoryURL"] = "Engenharia";
        this["Func_convert_Desc"] = "Converte um valor de um sistema de medição para outro, como metros em polegadas ou horas em segundos.";
        this["Func_cos_Arg_0"] = "número";
        this["Func_cos_ArgDesc_0"] = "O ângulo em radianos para o qual obter o cosseno.";
        this["Func_cos_Category"] = "Matemática";
        this["Func_cos_CategoryURL"] = "Matemática";
        this["Func_cos_Desc"] = "Devolve o cosseno do ângulo indicado.";
        this["Func_cosh_Arg_0"] = "número";
        this["Func_cosh_ArgDesc_0"] = "Um número real para o qual encontrar o cosseno hiperbólico.";
        this["Func_cosh_Category"] = "Matemática";
        this["Func_cosh_CategoryURL"] = "Matemática";
        this["Func_cosh_Desc"] = "Devolve o cosseno hiperbólico de um número.";
        this["Func_count_Arg_0"] = "valor {0}";
        this["Func_count_ArgDesc_0"] = "Uma referência ou intervalo de referências que podem conter números.";
        this["Func_count_Category"] = "Estatística";
        this["Func_count_CategoryURL"] = "Estatística";
        this["Func_count_Desc"] = "Conta o número de referências que contêm números.";
        this["Func_counta_Arg_0"] = "valor {0}";
        this["Func_counta_ArgDesc_0"] = "Uma referência ou intervalo de referências que podem conter números.";
        this["Func_counta_Category"] = "Estatística";
        this["Func_counta_CategoryURL"] = "Estatística";
        this["Func_counta_Desc"] = "Conta o número de referências que não estão em branco.";
        this["Func_critbinom_Arg_0"] = "Avaliações";
        this["Func_critbinom_Arg_1"] = "Probabilidade";
        this["Func_critbinom_Arg_2"] = "Alfa";
        this["Func_critbinom_ArgDesc_0"] = "O número de avaliações independentes.";
        this["Func_critbinom_ArgDesc_1"] = "A probabilidade de sucesso em cada avaliação.";
        this["Func_critbinom_ArgDesc_2"] = "O valor do critério.";
        this["Func_critbinom_Category"] = "Estatística";
        this["Func_critbinom_CategoryURL"] = "Estatística";
        this["Func_critbinom_Desc"] = "Devolve o valor mais baixo para o qual a distribuição binomial cumulativa é superior ou igual a um valor de critério.";
        this["Func_date_Arg_0"] = "ano";
        this["Func_date_Arg_1"] = "mês";
        this["Func_date_Arg_2"] = "dia";
        this["Func_date_ArgDesc_0"] = "Um número que representa o ano.";
        this["Func_date_ArgDesc_1"] = "Um número que representa o mês do ano.";
        this["Func_date_ArgDesc_2"] = "Um número que representa o dia do mês.";
        this["Func_date_Category"] = "DateAndTime";
        this["Func_date_CategoryURL"] = "Date_and_Time";
        this["Func_date_Desc"] = "Devolve o número de série sequencial que representa uma data específica.";
        this["Func_dateadd_Arg_0"] = "intervalo";
        this["Func_dateadd_Arg_1"] = "número";
        this["Func_dateadd_Arg_2"] = "data";
        this["Func_dateadd_ArgDesc_0"] = "Uma expressão de texto que representa o intervalo de tempo a adicionar.\r\nd = Dia\r\ny = Ano\r\nh = hora\r\nn = Minuto\r\nm = Mês\r\nq = Trimestre\r\ns = Segundo\r\nw = Semana\r\nww = Semana do calendário\r\nyyyy = Ano\r\n";
        this["Func_dateadd_ArgDesc_1"] = "O número de intervalos a serem adicionados. Este valor pode ser positivo (para obter uma DataHora no futuro) ou negativo (para obter uma DataHora no passado). Pode conter uma parte fracional quando Intervalo especifica horas, minutos ou segundos. Para outros valores de Intervalo, qualquer parte fracional de Número é ignorada.";
        this["Func_dateadd_ArgDesc_2"] = "A data à qual o intervalo deve ser adicionado.";
        this["Func_dateadd_Category"] = "DateAndTime";
        this["Func_dateadd_CategoryURL"] = "Date_and_Time";
        this["Func_dateadd_Desc"] = "Devolve uma data à qual o intervalo especificado foi adicionado.";
        this["Func_datediff_Arg_0"] = "intervalo";
        this["Func_datediff_Arg_1"] = "data1";
        this["Func_datediff_Arg_2"] = "data2";
        this["Func_datediff_ArgDesc_0"] = "Uma expressão de texto que representa o intervalo de tempo a ser usado como unidade de diferença entre data1 e data2.\r\nd = Dia\r\ny = Ano\r\nh = Hora\r\nn = Minuto\r\nm = Mês\r\nq = Trimestre\r\ns = Segundo\r\nw = Semana\r\nww = Semana do calendário\r\nyyyy  = Ano\r\n";
        this["Func_datediff_ArgDesc_1"] = "A primeira data a ser usada no cálculo da diferença.";
        this["Func_datediff_ArgDesc_2"] = "A segunda data a ser usada no cálculo da diferença.";
        this["Func_datediff_Category"] = "DateAndTime";
        this["Func_datediff_CategoryURL"] = "Date_and_Time";
        this["Func_datediff_Desc"] = "Devolve um valor que indica o número de intervalos entre dois valores de DataHora.";
        this["Func_datevalue_Arg_0"] = "date_text";
        this["Func_datevalue_ArgDesc_0"] = "Uma expressão de texto que representa uma data.";
        this["Func_datevalue_Category"] = "DateAndTime";
        this["Func_datevalue_CategoryURL"] = "Date_and_Time";
        this["Func_datevalue_Desc"] = "Devolve o valor da data do texto da data especificada.";
        this["Func_day_Arg_0"] = "Date_value";
        this["Func_day_ArgDesc_0"] = "Um valor de data a partir do qual obter o dia.";
        this["Func_day_Category"] = "DateAndTime";
        this["Func_day_CategoryURL"] = "Date_and_Time";
        this["Func_day_Desc"] = "Devolve o dia do mês de um valor de data.";
        this["Func_days360_Arg_0"] = "start_date";
        this["Func_days360_Arg_1"] = "end_date";
        this["Func_days360_Arg_2"] = "método";
        this["Func_days360_ArgDesc_0"] = "A data de início do intervalo de dias. Se tal ocorrer após a data final, a função irá devolver um número negativo.";
        this["Func_days360_ArgDesc_1"] = "A data final do intervalo de dias. Se tal for anterior à data de início, a função irá devolver um número negativo.";
        this["Func_days360_ArgDesc_2"] = "Um valor lógico que especifica se o método E.U.A. (falso) ou europeu (verdadeiro) deve ser utilizado no cálculo.";
        this["Func_days360_Category"] = "DateAndTime";
        this["Func_days360_CategoryURL"] = "Date_and_Time";
        this["Func_days360_Desc"] = "Devolve o número de dias entre duas datas com base num ano de 360 dias.";
        this["Func_db_Arg_0"] = "custo";
        this["Func_db_Arg_1"] = "residual";
        this["Func_db_Arg_2"] = "vida";
        this["Func_db_Arg_3"] = "período";
        this["Func_db_Arg_4"] = "mês";
        this["Func_db_ArgDesc_0"] = "O custo inicial do ativo.";
        this["Func_db_ArgDesc_1"] = "O valor no final da depreciação (às vezes chamado de valor residual do ativo).";
        this["Func_db_ArgDesc_2"] = "O número de períodos durante os quais o ativo é depreciado (às vezes designado vida útil do ativo).";
        this["Func_db_ArgDesc_3"] = "O período para o qual calcular a depreciação. O período deve usar as mesmas unidades que a vida.";
        this["Func_db_ArgDesc_4"] = "O número de meses no primeiro ano. Se o mês for omitido, assume-se que é 12.";
        this["Func_db_Category"] = "Financeiro";
        this["Func_db_CategoryURL"] = "Financeiro";
        this["Func_db_Desc"] = "Devolve a depreciação de um ativo durante um período especificado, utilizando o método de redução fixa do saldo.";
        this["Func_dbnull_Category"] = "Informação";
        this["Func_dbnull_CategoryURL"] = "Informação";
        this["Func_dbnull_Desc"] = "Devolve algo que representa um valor inexistente.";
        this["Func_ddb_Arg_0"] = "custo";
        this["Func_ddb_Arg_1"] = "residual";
        this["Func_ddb_Arg_2"] = "vida";
        this["Func_ddb_Arg_3"] = "período";
        this["Func_ddb_Arg_4"] = "fator";
        this["Func_ddb_ArgDesc_0"] = "O custo inicial do ativo.";
        this["Func_ddb_ArgDesc_1"] = "O valor no final da depreciação (às vezes chamado de valor residual do ativo).";
        this["Func_ddb_ArgDesc_2"] = "O número de períodos durante os quais o ativo é depreciado (às vezes designado vida útil do ativo).";
        this["Func_ddb_ArgDesc_3"] = "O período para o qual calcular a depreciação. O período deve usar as mesmas unidades que a vida.";
        this["Func_ddb_ArgDesc_4"] = "A taxa à qual o saldo diminui. Se o fator for omitido, assume-se que seja 2 (o método de redução dupla do saldo).";
        this["Func_ddb_Category"] = "Financeiro";
        this["Func_ddb_CategoryURL"] = "Financeiro";
        this["Func_ddb_Desc"] = "Devolve a depreciação de um ativo durante um período especificado, utilizando o método de redução dupla do saldo ou outro método especificado.";
        this["Func_dec2bin_Arg_0"] = "Número";
        this["Func_dec2bin_Arg_1"] = "Locais";
        this["Func_dec2bin_ArgDesc_0"] = "O valor decimal a ser convertido.";
        this["Func_dec2bin_ArgDesc_1"] = "O número de caracteres a utilizar para representar o valor. Se locais não for especificado, será utilizado o número mínimo de caracteres necessário para representar o valor. Isso permite especificar zeros à esquerda.";
        this["Func_dec2bin_Category"] = "Engenharia";
        this["Func_dec2bin_CategoryURL"] = "Engenharia";
        this["Func_dec2bin_Desc"] = "Devolve uma expressão de texto que representa o valor decimal especificado como um número binário.";
        this["Func_dec2hex_Arg_0"] = "Número";
        this["Func_dec2hex_Arg_1"] = "Locais";
        this["Func_dec2hex_ArgDesc_0"] = "O valor decimal a ser convertido.";
        this["Func_dec2hex_ArgDesc_1"] = "O número de caracteres a utilizar para representar o valor. Se locais não for especificado, será utilizado o número mínimo de caracteres necessário para representar o valor. Isso permite especificar zeros à esquerda.";
        this["Func_dec2hex_Category"] = "Engenharia";
        this["Func_dec2hex_CategoryURL"] = "Engenharia";
        this["Func_dec2hex_Desc"] = "Devolve uma expressão de texto que representa o valor decimal especificado como um número hexadecimal.";
        this["Func_dec2oct_Arg_0"] = "Número";
        this["Func_dec2oct_Arg_1"] = "Locais";
        this["Func_dec2oct_ArgDesc_0"] = "O valor decimal a ser convertido.";
        this["Func_dec2oct_ArgDesc_1"] = "O número de caracteres a utilizar para representar o valor. Se locais não for especificado, será utilizado o número mínimo de caracteres necessário para representar o valor. Isso permite especificar zeros à esquerda.";
        this["Func_dec2oct_Category"] = "Engenharia";
        this["Func_dec2oct_CategoryURL"] = "Engenharia";
        this["Func_dec2oct_Desc"] = "Devolve uma expressão de texto que representa o valor decimal especificado como um número octal.";
        this["Func_degrees_Arg_0"] = "Radianos";
        this["Func_degrees_ArgDesc_0"] = "Um valor em radianos que será convertido em graus.";
        this["Func_degrees_Category"] = "Matemática";
        this["Func_degrees_CategoryURL"] = "Matemática";
        this["Func_degrees_Desc"] = "Converte radianos em graus.";
        this["Func_delta_Arg_0"] = "Número1";
        this["Func_delta_Arg_1"] = "Número2";
        this["Func_delta_ArgDesc_0"] = "O primeiro número a comparar.";
        this["Func_delta_ArgDesc_1"] = "O segundo número a comparar. Se não for especificado, será assumido como zero.";
        this["Func_delta_Category"] = "Engenharia";
        this["Func_delta_CategoryURL"] = "Engenharia";
        this["Func_delta_Desc"] = "Compara dois números e devolve um, se forem iguais, ou zero, se não forem.";
        this["Func_dollarde_Arg_0"] = "QuantiaFracionalEmDólares";
        this["Func_dollarde_Arg_1"] = "Fração";
        this["Func_dollarde_ArgDesc_0"] = "O valor em dólares expresso como uma fração.";
        this["Func_dollarde_ArgDesc_1"] = "O valor usado como denominador da fração. Se este valor não for um número inteiro, será truncado.";
        this["Func_dollarde_Category"] = "Financeiro";
        this["Func_dollarde_CategoryURL"] = "Financeiro";
        this["Func_dollarde_Desc"] = "Converte um valor em dólares expresso como uma fração num valor em dólares expresso como decimal.";
        this["Func_dollarfr_Arg_0"] = "QuantiaDecimalEmDólares";
        this["Func_dollarfr_Arg_1"] = "Fração";
        this["Func_dollarfr_ArgDesc_0"] = "O valor em dólares expresso como decimal.";
        this["Func_dollarfr_ArgDesc_1"] = "O valor usado como denominador da fração. Se este valor não for um número inteiro, será truncado.";
        this["Func_dollarfr_Category"] = "Financeiro";
        this["Func_dollarfr_CategoryURL"] = "Financeiro";
        this["Func_dollarfr_Desc"] = "Converte um valor em dólares expresso como decimal num valor em dólares expresso em fração.";
        this["Func_edate_Arg_0"] = "data";
        this["Func_edate_Arg_1"] = "numberOfMonths";
        this["Func_edate_ArgDesc_0"] = "A data de início.";
        this["Func_edate_ArgDesc_1"] = "O número de meses para alterar a data. Um número positivo significa uma data após a data de início. Um número negativo indica uma data anterior à data de início.";
        this["Func_edate_Category"] = "DateAndTime";
        this["Func_edate_CategoryURL"] = "DateAndTime";
        this["Func_edate_Desc"] = "Devolve uma data com o número especificado de meses antes ou depois da data de início.";
        this["Func_eomonth_Arg_0"] = "data";
        this["Func_eomonth_Arg_1"] = "numberOfMonths";
        this["Func_eomonth_ArgDesc_0"] = "A data de início.";
        this["Func_eomonth_ArgDesc_1"] = "O número de meses para alterar a data. Um número positivo significa uma data após a data de início. Um número negativo indica uma data anterior à data de início.";
        this["Func_eomonth_Category"] = "DateAndTime";
        this["Func_eomonth_CategoryURL"] = "DateAndTime";
        this["Func_eomonth_Desc"] = "Devolve uma data no final do mês, que é o número especificado de meses antes ou depois da data de início.";
        this["Func_errortype_Arg_0"] = "error_value";
        this["Func_errortype_ArgDesc_0"] = "O valor do erro cujo número de identificação deve ser encontrado.";
        this["Func_errortype_Category"] = "Informação";
        this["Func_errortype_CategoryURL"] = "Informação";
        this["Func_errortype_Desc"] = "Devolve um número correspondente a um valor de erro.";
        this["Func_even_Arg_0"] = "número";
        this["Func_even_ArgDesc_0"] = "O valor a arredondar.";
        this["Func_even_Category"] = "Matemática";
        this["Func_even_CategoryURL"] = "Matemática";
        this["Func_even_Desc"] = "Arredonda um número positivo para cima e um número negativo para baixo para o número par mais próximo.";
        this["Func_exp_Arg_0"] = "número";
        this["Func_exp_ArgDesc_0"] = "O expoente aplicado ao qual aumentar e.";
        this["Func_exp_Category"] = "Matemática";
        this["Func_exp_CategoryURL"] = "Matemática";
        this["Func_exp_Desc"] = "Devolve e elevado à potência do número especificado. A constante e é igual a 2,71828182845904.";
        this["Func_fact_Arg_0"] = "Número";
        this["Func_fact_ArgDesc_0"] = "Um número positivo para o qual o fatorial será calculado.";
        this["Func_fact_Category"] = "Matemática";
        this["Func_fact_CategoryURL"] = "Matemática";
        this["Func_fact_Desc"] = "Devolve o fatorial de um número.";
        this["Func_factdouble_Arg_0"] = "Número";
        this["Func_factdouble_ArgDesc_0"] = "Um número positivo para o qual o fatorial duplo será calculado.";
        this["Func_factdouble_Category"] = "Matemática";
        this["Func_factdouble_CategoryURL"] = "Matemática";
        this["Func_factdouble_Desc"] = "Devolve o fatorial duplo de um número.";
        this["Func_false_Category"] = "Lógico";
        this["Func_false_CategoryURL"] = "Lógico";
        this["Func_false_Desc"] = "Devolve o valor lógico falso.";
        this["Func_find_Arg_0"] = "find_text";
        this["Func_find_Arg_1"] = "within_text";
        this["Func_find_Arg_2"] = "start_num";
        this["Func_find_ArgDesc_0"] = "O texto a encontrar.";
        this["Func_find_ArgDesc_1"] = "O texto que contém o texto a encontrar.";
        this["Func_find_ArgDesc_2"] = "O carácter de índice com base em 1 no qual iniciar a pesquisa. Se omitido, assume-se que seja um.";
        this["Func_find_Category"] = "TextAndData";
        this["Func_find_CategoryURL"] = "Text_and_Data";
        this["Func_find_Desc"] = "Devolve a posição inicial de uma expressão de texto dentro de outra, com distinção entre maiúsculas e minúsculas.";
        this["Func_fixed_Arg_0"] = "número";
        this["Func_fixed_Arg_1"] = "decimais";
        this["Func_fixed_Arg_2"] = "no_commas";
        this["Func_fixed_ArgDesc_0"] = "O número a arredondar e converter em texto.";
        this["Func_fixed_ArgDesc_1"] = "O número de dígitos à direita da vírgula decimal no valor devolvido. Se este valor for negativo, o \"número\" será arredondado para a esquerda da vírgula decimal. O valor predefinido é 2.";
        this["Func_fixed_ArgDesc_2"] = "Verdadeiro se vírgulas não devem estar no valor devolvido. Se omitido, é considerado falso.";
        this["Func_fixed_Category"] = "TextAndData";
        this["Func_fixed_CategoryURL"] = "Text_and_Data";
        this["Func_fixed_Desc"] = "Arredonda um número para o número especificado de casas decimais e devolve o resultado como texto com ou sem vírgulas.";
        this["Func_floor_Arg_0"] = "número";
        this["Func_floor_Arg_1"] = "significância";
        this["Func_floor_ArgDesc_0"] = "O número a arredondar.";
        this["Func_floor_ArgDesc_1"] = "O múltiplo para o qual arredondar. Se omitido, assume-se que seja um.";
        this["Func_floor_Category"] = "Matemática";
        this["Func_floor_CategoryURL"] = "Matemática";
        this["Func_floor_Desc"] = "Devolve o maior múltiplo da significância (o predefinido é 1) menor ou igual ao número indicado.";
        this["Func_fv_Arg_0"] = "taxa";
        this["Func_fv_Arg_1"] = "nper";
        this["Func_fv_Arg_2"] = "pmt";
        this["Func_fv_Arg_3"] = "pv";
        this["Func_fv_Arg_4"] = "type";
        this["Func_fv_ArgDesc_0"] = "A taxa de juros por período.";
        this["Func_fv_ArgDesc_1"] = "O número total de períodos de pagamento numa anuidade.";
        this["Func_fv_ArgDesc_2"] = "O pagamento efetuado a cada período, que não pode ser alterado ao longo da duração da anuidade.";
        this["Func_fv_ArgDesc_3"] = "O valor presente ou a quantia global que uma série de pagamentos futuros vale no momento. Se omitido, assume-se que seja zero.";
        this["Func_fv_ArgDesc_4"] = "O número zero ou um que indica se os pagamentos são devidos no final do período (zero, o valor predefinido) ou no início (um).";
        this["Func_fv_Category"] = "Financeiro";
        this["Func_fv_CategoryURL"] = "Financeiro";
        this["Func_fv_Desc"] = "Devolve o valor futuro de um investimento com base em periódicos.";
        this["Func_gcd_Arg_0"] = "número {0}";
        this["Func_gcd_ArgDesc_0"] = "Um valor inteiro. Um valor não inteiro será truncado.";
        this["Func_gcd_Category"] = "Matemática";
        this["Func_gcd_CategoryURL"] = "Matemática";
        this["Func_gcd_Desc"] = "Devolve o maior divisor comum de um ou mais valores inteiros.";
        this["Func_gestep_Arg_0"] = "Número1";
        this["Func_gestep_Arg_1"] = "Passo";
        this["Func_gestep_ArgDesc_0"] = "O primeiro número a comparar.";
        this["Func_gestep_ArgDesc_1"] = "O segundo número a comparar. Se não for especificado, será assumido como zero.";
        this["Func_gestep_Category"] = "Engenharia";
        this["Func_gestep_CategoryURL"] = "Engenharia";
        this["Func_gestep_Desc"] = "Compara dois números e devolve um, se o primeiro número for maior ou igual ao segundo, ou devolve zero, se não for.";
        this["Func_hex2bin_Arg_0"] = "Número";
        this["Func_hex2bin_Arg_1"] = "Locais";
        this["Func_hex2bin_ArgDesc_0"] = "Uma expressão de texto que representa um número hexadecimal.";
        this["Func_hex2bin_ArgDesc_1"] = "O número de caracteres a utilizar para representar o valor. Se não for especificado, será utilizado o número mínimo de caracteres necessário para representar o valor. Isso permite especificar zeros à esquerda.";
        this["Func_hex2bin_Category"] = "Engenharia";
        this["Func_hex2bin_CategoryURL"] = "Engenharia";
        this["Func_hex2bin_Desc"] = "Converte uma expressão de texto que representa um número hexadecimal numa expressão de texto que representa o mesmo número no sistema numérico binário.";
        this["Func_hex2dec_Arg_0"] = "Número";
        this["Func_hex2dec_ArgDesc_0"] = "Uma expressão de texto que representa um número hexadecimal.";
        this["Func_hex2dec_Category"] = "Engenharia";
        this["Func_hex2dec_CategoryURL"] = "Engenharia";
        this["Func_hex2dec_Desc"] = "Devolve o valor decimal de uma representação de texto de um número no sistema numérico hexadecimal.";
        this["Func_hex2oct_Arg_0"] = "Número";
        this["Func_hex2oct_Arg_1"] = "Locais";
        this["Func_hex2oct_ArgDesc_0"] = "Uma expressão de texto que representa um número hexadecimal.";
        this["Func_hex2oct_ArgDesc_1"] = "O número de caracteres a utilizar para representar o valor. Se não for especificado, será utilizado o número mínimo de caracteres necessário para representar o valor. Isso permite especificar zeros à esquerda.";
        this["Func_hex2oct_Category"] = "Engenharia";
        this["Func_hex2oct_CategoryURL"] = "Engenharia";
        this["Func_hex2oct_Desc"] = "Converte uma expressão de texto que representa um número hexadecimal numa expressão de texto que representa o mesmo número no sistema numérico octal.";
        this["Func_hour_Arg_0"] = "Time_value";
        this["Func_hour_ArgDesc_0"] = "Um valor de tempo a partir do qual obter a hora.";
        this["Func_hour_Category"] = "DateAndTime";
        this["Func_hour_CategoryURL"] = "Date_and_Time";
        this["Func_hour_Desc"] = "Devolve a hora de um valor de tempo.";
        this["Func_if_Arg_0"] = "logical_test";
        this["Func_if_Arg_1"] = "value_if_true";
        this["Func_if_Arg_2"] = "value_if_false";
        this["Func_if_ArgDesc_0"] = "A condição a testar.";
        this["Func_if_ArgDesc_1"] = "O resultado se a condição for avaliada como verdadeira.";
        this["Func_if_ArgDesc_2"] = "O resultado se a condição for avaliada como falsa. Se omitido, é assumido que o valor lógico é falso.";
        this["Func_if_Category"] = "Lógico";
        this["Func_if_CategoryURL"] = "Lógico";
        this["Func_if_Desc"] = "Devolve um valor se a condição especificada for avaliada como verdadeira e outro valor se for avaliada como falsa.";
        this["Func_iferror_Arg_0"] = "valor";
        this["Func_iferror_Arg_1"] = "value_if_error";
        this["Func_iferror_ArgDesc_0"] = "O valor a ser usado quando não houver erro.";
        this["Func_iferror_ArgDesc_1"] = "O valor a ser usado se o argumento do valor for um erro.";
        this["Func_iferror_Category"] = "Lógico";
        this["Func_iferror_CategoryURL"] = "Lógico";
        this["Func_iferror_Desc"] = "Devolve um dos dois valores, dependendo se o primeiro valor é ou não um erro.";
        this["Func_imabs_Arg_0"] = "NúmeroComplexo";
        this["Func_imabs_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imabs_Category"] = "Engenharia";
        this["Func_imabs_CategoryURL"] = "Engenharia";
        this["Func_imabs_Desc"] = "Devolve o valor absoluto de um número complexo.";
        this["Func_imaginary_Arg_0"] = "NúmeroComplexo";
        this["Func_imaginary_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imaginary_Category"] = "Engenharia";
        this["Func_imaginary_CategoryURL"] = "Engenharia";
        this["Func_imaginary_Desc"] = "Devolve o coeficiente imaginário de um número complexo.";
        this["Func_imargument_Arg_0"] = "NúmeroComplexo";
        this["Func_imargument_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imargument_Category"] = "Engenharia";
        this["Func_imargument_CategoryURL"] = "Engenharia";
        this["Func_imargument_Desc"] = "Devolve o argumento teta, um ângulo expresso em radianos.";
        this["Func_imconjugate_Arg_0"] = "NúmeroComplexo";
        this["Func_imconjugate_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imconjugate_Category"] = "Engenharia";
        this["Func_imconjugate_CategoryURL"] = "Engenharia";
        this["Func_imconjugate_Desc"] = "Devolve o conjugado de um número complexo.";
        this["Func_imcos_Arg_0"] = "NúmeroComplexo";
        this["Func_imcos_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imcos_Category"] = "Engenharia";
        this["Func_imcos_CategoryURL"] = "Engenharia";
        this["Func_imcos_Desc"] = "Devolve o cosseno de um número complexo.";
        this["Func_imdiv_Arg_0"] = "Dividendo";
        this["Func_imdiv_Arg_1"] = "Divisor";
        this["Func_imdiv_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imdiv_ArgDesc_1"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imdiv_Category"] = "Engenharia";
        this["Func_imdiv_CategoryURL"] = "Engenharia";
        this["Func_imdiv_Desc"] = "Devolve o quociente de dois números complexos (Dividendo/Divisor).";
        this["Func_imexp_Arg_0"] = "NúmeroComplexo";
        this["Func_imexp_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imexp_Category"] = "Engenharia";
        this["Func_imexp_CategoryURL"] = "Engenharia";
        this["Func_imexp_Desc"] = "Devolve o exponencial de um número complexo.";
        this["Func_imln_Arg_0"] = "NúmeroComplexo";
        this["Func_imln_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imln_Category"] = "Engenharia";
        this["Func_imln_CategoryURL"] = "Engenharia";
        this["Func_imln_Desc"] = "Devolve o logaritmo natural de um número complexo.";
        this["Func_imlog10_Arg_0"] = "NúmeroComplexo";
        this["Func_imlog10_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imlog10_Category"] = "Engenharia";
        this["Func_imlog10_CategoryURL"] = "Engenharia";
        this["Func_imlog10_Desc"] = "Devolve o logaritmo de base 10 de um número complexo.";
        this["Func_imlog2_Arg_0"] = "NúmeroComplexo";
        this["Func_imlog2_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imlog2_Category"] = "Engenharia";
        this["Func_imlog2_CategoryURL"] = "Engenharia";
        this["Func_imlog2_Desc"] = "Devolve o logaritmo de base 2 de um número complexo.";
        this["Func_impower_Arg_0"] = "NúmeroComplexo";
        this["Func_impower_Arg_1"] = "Power";
        this["Func_impower_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_impower_ArgDesc_1"] = "O poder para o qual NúmeroComplexo será aumentado.";
        this["Func_impower_Category"] = "Engenharia";
        this["Func_impower_CategoryURL"] = "Engenharia";
        this["Func_impower_Desc"] = "Devolve o número complexo elevado à potência especificada.";
        this["Func_improduct_Arg_0"] = "NúmeroComplexo {0}";
        this["Func_improduct_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_improduct_Category"] = "Engenharia";
        this["Func_improduct_CategoryURL"] = "Engenharia";
        this["Func_improduct_Desc"] = "Devolve o produto de um ou mais números complexos.";
        this["Func_imreal_Arg_0"] = "NúmeroComplexo";
        this["Func_imreal_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imreal_Category"] = "Engenharia";
        this["Func_imreal_CategoryURL"] = "Engenharia";
        this["Func_imreal_Desc"] = "Devolve o coeficiente real de um número complexo.";
        this["Func_imsin_Arg_0"] = "NúmeroComplexo";
        this["Func_imsin_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imsin_Category"] = "Engenharia";
        this["Func_imsin_CategoryURL"] = "Engenharia";
        this["Func_imsin_Desc"] = "Devolve o seno de um número complexo.";
        this["Func_imsqrt_Arg_0"] = "NúmeroComplexo";
        this["Func_imsqrt_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imsqrt_Category"] = "Engenharia";
        this["Func_imsqrt_CategoryURL"] = "Engenharia";
        this["Func_imsqrt_Desc"] = "Devolve a raiz quadrada de um número complexo.";
        this["Func_imsub_Arg_0"] = "NúmeroComplexo1";
        this["Func_imsub_Arg_1"] = "NúmeroComplexo2";
        this["Func_imsub_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\" a partir do qual o NúmeroComplexo2 será subtraído.";
        this["Func_imsub_ArgDesc_1"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\" que será subtraído do NúmeroComplexo1.";
        this["Func_imsub_Category"] = "Engenharia";
        this["Func_imsub_CategoryURL"] = "Engenharia";
        this["Func_imsub_Desc"] = "Devolve a diferença entre dois números complexos.";
        this["Func_imsum_Arg_0"] = "NúmeroComplexo {0}";
        this["Func_imsum_ArgDesc_0"] = "A representação de texto de um número complexo no formato: \"x + yi\" ou \"x + yj\".";
        this["Func_imsum_Category"] = "Engenharia";
        this["Func_imsum_CategoryURL"] = "Engenharia";
        this["Func_imsum_Desc"] = "Devolve a soma de um ou mais números complexos.";
        this["Func_info_Arg_0"] = "Tipo";
        this["Func_info_ArgDesc_0"] = "Uma expressão de texto que indica o tipo de informação a devolver.\r\n\"directory\" = O caminho completo para a pasta atual.\r\n\"osversion\" = A versão do sistema operativo atual como uma expressão de texto.\r\n\"system\" = O sistema operativo atual. Macintosh = \"mac\", Windows = \"pcdos\".\r\n    ";
        this["Func_info_Category"] = "Informação";
        this["Func_info_CategoryURL"] = "Informação";
        this["Func_info_Desc"] = "Devolve informações sobre o ambiente operativo atual.";
        this["Func_int_Arg_0"] = "número";
        this["Func_int_ArgDesc_0"] = "O número real para arredondar para um número inteiro.";
        this["Func_int_Category"] = "Matemática";
        this["Func_int_CategoryURL"] = "Matemática";
        this["Func_int_Desc"] = "Arredonda um número para baixo até o número inteiro mais próximo.";
        this["Func_intrate_Arg_0"] = "liquidação";
        this["Func_intrate_Arg_1"] = "vencimento";
        this["Func_intrate_Arg_2"] = "investimento";
        this["Func_intrate_Arg_3"] = "resgate";
        this["Func_intrate_Arg_4"] = "base";
        this["Func_intrate_ArgDesc_0"] = "A data de liquidação do título.";
        this["Func_intrate_ArgDesc_1"] = "A data de vencimento do título.";
        this["Func_intrate_ArgDesc_2"] = "A quantia investida no título.";
        this["Func_intrate_ArgDesc_3"] = "O valor a ser recebido no vencimento.";
        this["Func_intrate_ArgDesc_4"] = "O tipo de contagem de dias a utilizar.\r\n0 ou omitido = E.U.A. (NASD) 30/360,\r\n1 = Real/real,\r\n2 = Real/360,\r\n3 = Real/365,\r\n4 = Europeu 30/360\r\n    ";
        this["Func_intrate_Category"] = "Financeiro";
        this["Func_intrate_CategoryURL"] = "Financeiro";
        this["Func_intrate_Desc"] = "Devolve a taxa de juro para um título totalmente investido.";
        this["Func_ipmt_Arg_0"] = "taxa";
        this["Func_ipmt_Arg_1"] = "por";
        this["Func_ipmt_Arg_2"] = "nper";
        this["Func_ipmt_Arg_3"] = "pv";
        this["Func_ipmt_Arg_4"] = "fv";
        this["Func_ipmt_Arg_5"] = "type";
        this["Func_ipmt_ArgDesc_0"] = "A taxa de juros por período.";
        this["Func_ipmt_ArgDesc_1"] = "O período, que deve estar no intervalo de um a nper.";
        this["Func_ipmt_ArgDesc_2"] = "O número total de períodos de pagamento numa anuidade.";
        this["Func_ipmt_ArgDesc_3"] = "O valor presente ou a quantia global que uma série de pagamentos futuros vale agora.";
        this["Func_ipmt_ArgDesc_4"] = "O valor futuro ou um saldo em dinheiro que deseja atingir após o último pagamento. Se omitido, assume-se que seja zero.";
        this["Func_ipmt_ArgDesc_5"] = "O número zero ou um que indica se os pagamentos são devidos no final do período (zero, o valor predefinido) ou no início (um).";
        this["Func_ipmt_Category"] = "Financeiro";
        this["Func_ipmt_CategoryURL"] = "Financeiro";
        this["Func_ipmt_Desc"] = "Devolve o pagamento de juros durante um determinado período para um investimento, com base em pagamentos periódicos e constantes e numa taxa de juros constante.";
        this["Func_irr_Arg_0"] = "valores";
        this["Func_irr_Arg_1"] = "adivinhar";
        this["Func_irr_ArgDesc_0"] = "Uma coleção de valores de pagamento (valores negativos) e rendimento (valores positivos) que ocorrem em intervalos regulares.";
        this["Func_irr_ArgDesc_1"] = "Um número que acha que está próximo do resultado. O padrão é 0,1 (10%).";
        this["Func_irr_Category"] = "Financeiro";
        this["Func_irr_CategoryURL"] = "Financeiro";
        this["Func_irr_Desc"] = "Devolve a taxa interna de devolução para uma série de fluxos de caixa.";
        this["Func_isblank_Arg_0"] = "valor";
        this["Func_isblank_ArgDesc_0"] = "O valor a testar.";
        this["Func_isblank_Category"] = "Informação";
        this["Func_isblank_CategoryURL"] = "Informação";
        this["Func_isblank_Desc"] = "Devolve verdadeiro se o valor estiver em branco.";
        this["Func_isdbnull_Arg_0"] = "valor";
        this["Func_isdbnull_ArgDesc_0"] = "O valor a testar.";
        this["Func_isdbnull_Category"] = "Informação";
        this["Func_isdbnull_CategoryURL"] = "Informação";
        this["Func_isdbnull_Desc"] = "Devolve verdadeiro se o valor representar um valor inexistente.";
        this["Func_iserr_Arg_0"] = "valor";
        this["Func_iserr_ArgDesc_0"] = "O valor a testar.";
        this["Func_iserr_Category"] = "Informação";
        this["Func_iserr_CategoryURL"] = "Informação";
        this["Func_iserr_Desc"] = "Devolve verdadeiro se o valor for qualquer valor de erro, exceto #N/D.";
        this["Func_iserror_Arg_0"] = "valor";
        this["Func_iserror_ArgDesc_0"] = "O valor a testar.";
        this["Func_iserror_Category"] = "Informação";
        this["Func_iserror_CategoryURL"] = "Informação";
        this["Func_iserror_Desc"] = "Devolve verdadeiro se o valor for algum valor de erro.";
        this["Func_iseven_Arg_0"] = "valor";
        this["Func_iseven_ArgDesc_0"] = "O valor a testar.";
        this["Func_iseven_Category"] = "Informação";
        this["Func_iseven_CategoryURL"] = "Informação";
        this["Func_iseven_Desc"] = "Devolve verdadeiro se o valor for um número par, caso contrário devolve falso.";
        this["Func_islogical_Arg_0"] = "valor";
        this["Func_islogical_ArgDesc_0"] = "O valor a testar.";
        this["Func_islogical_Category"] = "Informação";
        this["Func_islogical_CategoryURL"] = "Informação";
        this["Func_islogical_Desc"] = "Devolve verdadeiro se o valor for um valor lógico (verdadeiro ou falso).";
        this["Func_isna_Arg_0"] = "valor";
        this["Func_isna_ArgDesc_0"] = "O valor a testar.";
        this["Func_isna_Category"] = "Informação";
        this["Func_isna_CategoryURL"] = "Informação";
        this["Func_isna_Desc"] = "Devolve verdadeiro se o valor for o valor de erro #N/D (valor não disponível).";
        this["Func_isnontext_Arg_0"] = "valor";
        this["Func_isnontext_ArgDesc_0"] = "O valor a testar.";
        this["Func_isnontext_Category"] = "Informação";
        this["Func_isnontext_CategoryURL"] = "Informação";
        this["Func_isnontext_Desc"] = "Devolve verdadeiro se o valor for qualquer item que não seja texto.";
        this["Func_isnull_Arg_0"] = "valor";
        this["Func_isnull_ArgDesc_0"] = "O valor a testar.";
        this["Func_isnull_Category"] = "Informação";
        this["Func_isnull_CategoryURL"] = "Informação";
        this["Func_isnull_Desc"] = "Devolve verdadeiro se o valor não existir.";
        this["Func_isnumber_Arg_0"] = "valor";
        this["Func_isnumber_ArgDesc_0"] = "O valor a testar.";
        this["Func_isnumber_Category"] = "Informação";
        this["Func_isnumber_CategoryURL"] = "Informação";
        this["Func_isnumber_Desc"] = "Devolve verdadeiro se o valor for um número.";
        this["Func_isodd_Arg_0"] = "valor";
        this["Func_isodd_ArgDesc_0"] = "O valor a testar.";
        this["Func_isodd_Category"] = "Informação";
        this["Func_isodd_CategoryURL"] = "Informação";
        this["Func_isodd_Desc"] = "Devolve verdadeiro se o valor for um número ímpar, caso contrário devolve falso.";
        this["Func_isref_Arg_0"] = "valor";
        this["Func_isref_ArgDesc_0"] = "O valor a testar.";
        this["Func_isref_Category"] = "Informação";
        this["Func_isref_CategoryURL"] = "Informação";
        this["Func_isref_Desc"] = "Devolve verdadeiro se o valor for uma referência.";
        this["Func_istext_Arg_0"] = "valor";
        this["Func_istext_ArgDesc_0"] = "O valor a testar.";
        this["Func_istext_Category"] = "Informação";
        this["Func_istext_CategoryURL"] = "Informação";
        this["Func_istext_Desc"] = "Devolve verdadeiro se o valor for um valor de texto.";
        this["Func_lcm_Arg_0"] = "número {0}";
        this["Func_lcm_ArgDesc_0"] = "Um valor inteiro. Um valor não inteiro será truncado.";
        this["Func_lcm_Category"] = "Matemática";
        this["Func_lcm_CategoryURL"] = "Matemática";
        this["Func_lcm_Desc"] = "Devolve o múltiplo menos comum de um ou mais valores inteiros.";
        this["Func_left_Arg_0"] = "texto";
        this["Func_left_Arg_1"] = "num_chars";
        this["Func_left_ArgDesc_0"] = "A expressão de texto que contém os caracteres a extrair.";
        this["Func_left_ArgDesc_1"] = "O número de caracteres a extrair.";
        this["Func_left_Category"] = "TextAndData";
        this["Func_left_CategoryURL"] = "Text_and_Data";
        this["Func_left_Desc"] = "Devolve o primeiro carácter ou caracteres numa expressão de texto.";
        this["Func_len_Arg_0"] = "texto";
        this["Func_len_ArgDesc_0"] = "A expressão de texto cujo comprimento encontrar. Os espaços contam como caracteres.";
        this["Func_len_Category"] = "TextAndData";
        this["Func_len_CategoryURL"] = "Text_and_Data";
        this["Func_len_Desc"] = "Devolve o número de caracteres numa expressão de texto.";
        this["Func_ln_Arg_0"] = "número";
        this["Func_ln_ArgDesc_0"] = "O número real positivo para o qual obter o logaritmo natural.";
        this["Func_ln_Category"] = "Matemática";
        this["Func_ln_CategoryURL"] = "Matemática";
        this["Func_ln_Desc"] = "Devolve o logaritmo natural de um número.";
        this["Func_log_Arg_0"] = "número";
        this["Func_log_Arg_1"] = "base";
        this["Func_log_ArgDesc_0"] = "O número real positivo para o qual obter o logaritmo.";
        this["Func_log_ArgDesc_1"] = "A base do logaritmo.";
        this["Func_log_Category"] = "Matemática";
        this["Func_log_CategoryURL"] = "Matemática";
        this["Func_log_Desc"] = "Devolve o logaritmo de um número à base especificada.";
        this["Func_log10_Arg_0"] = "número";
        this["Func_log10_ArgDesc_0"] = "O número real positivo para o qual obter o logaritmo da base 10.";
        this["Func_log10_Category"] = "Matemática";
        this["Func_log10_CategoryURL"] = "Matemática";
        this["Func_log10_Desc"] = "Devolve o logaritmo da base 10 de um número.";
        this["Func_lower_Arg_0"] = "texto";
        this["Func_lower_ArgDesc_0"] = "O texto a converter em minúsculas.";
        this["Func_lower_Category"] = "TextAndData";
        this["Func_lower_CategoryURL"] = "Text_and_Data";
        this["Func_lower_Desc"] = "Converte todas as letras maiúsculas numa expressão de texto em minúsculas.";
        this["Func_max_Arg_0"] = "número {0}";
        this["Func_max_ArgDesc_0"] = "Um número ou intervalo de números.";
        this["Func_max_Category"] = "Estatística";
        this["Func_max_CategoryURL"] = "Estatística";
        this["Func_max_Desc"] = "Devolve o maior número de um conjunto de números.";
        this["Func_median_Arg_0"] = "number{0}";
        this["Func_median_ArgDesc_0"] = "Um número ou intervalo de números.";
        this["Func_median_Category"] = "Estatística";
        this["Func_median_CategoryURL"] = "Estatística";
        this["Func_median_Desc"] = "Devolve a mediana de um conjunto de números. A mediana é o número do meio quando os números são ordenados ou a média dos dois números do meio, se houver um número par de números especificados.";
        this["Func_mid_Arg_0"] = "text";
        this["Func_mid_Arg_1"] = "start_num";
        this["Func_mid_Arg_2"] = "num_chars";
        this["Func_mid_ArgDesc_0"] = "A expressão de texto que contém os caracteres a extrair.";
        this["Func_mid_ArgDesc_1"] = "O índice baseado em 1 do primeiro carácter a extrair.";
        this["Func_mid_ArgDesc_2"] = "O número de caracteres a extrair.";
        this["Func_mid_Category"] = "TextAndData";
        this["Func_mid_CategoryURL"] = "Text_and_Data";
        this["Func_mid_Desc"] = "Devolve um número específico de caracteres de uma expressão de texto.";
        this["Func_min_Arg_0"] = "número {0}";
        this["Func_min_ArgDesc_0"] = "Um número ou intervalo de números.";
        this["Func_min_Category"] = "Estatística";
        this["Func_min_CategoryURL"] = "Estatística";
        this["Func_min_Desc"] = "Devolve o menor número num conjunto de números.";
        this["Func_minute_Arg_0"] = "Time_value";
        this["Func_minute_ArgDesc_0"] = "Um valor de tempo a partir do qual obter o minuto.";
        this["Func_minute_Category"] = "DateAndTime";
        this["Func_minute_CategoryURL"] = "Date_and_Time";
        this["Func_minute_Desc"] = "Devolve o minuto de um valor de tempo.";
        this["Func_mod_Arg_0"] = "número";
        this["Func_mod_Arg_1"] = "divisor";
        this["Func_mod_ArgDesc_0"] = "O número para o qual encontrar o restante.";
        this["Func_mod_ArgDesc_1"] = "O valor pelo qual dividir o número.";
        this["Func_mod_Category"] = "Matemática";
        this["Func_mod_CategoryURL"] = "Matemática";
        this["Func_mod_Desc"] = "Devolve o restante depois de um número ser dividido por um divisor. O resultado tem o mesmo sinal que o divisor.";
        this["Func_month_Arg_0"] = "Date_value";
        this["Func_month_ArgDesc_0"] = "Um valor de data a partir do qual obter o mês.";
        this["Func_month_Category"] = "DateAndTime";
        this["Func_month_CategoryURL"] = "Date_and_Time";
        this["Func_month_Desc"] = "Devolve o mês de um valor de data.";
        this["Func_mround_Arg_0"] = "Número";
        this["Func_mround_Arg_1"] = "Vários";
        this["Func_mround_ArgDesc_0"] = "O número a ser arredondado.";
        this["Func_mround_ArgDesc_1"] = "Um número que indica o múltiplo para o qual arredondar.";
        this["Func_mround_Category"] = "Matemática";
        this["Func_mround_CategoryURL"] = "Matemática";
        this["Func_mround_Desc"] = "Arredonda um número para o múltiplo mais próximo de outro número.";
        this["Func_multinomial_Arg_0"] = "número {0}";
        this["Func_multinomial_ArgDesc_0"] = "Um dos números para o qual obter o polinomial.";
        this["Func_multinomial_Category"] = "Matemática";
        this["Func_multinomial_CategoryURL"] = "Matemática";
        this["Func_multinomial_Desc"] = "Devolve o polinomial de um conjunto de números. O polinomial é a razão entre o fatorial de uma soma de valores e o produto dos seus fatoriais.";
        this["Func_n_Arg_0"] = "Valor";
        this["Func_n_ArgDesc_0"] = "O valor a ser convertido num número. Os seguintes são valores aceitáveis:\r\nUm número - devolve o número.\r\nUma data - devolve a data.\r\nVerdadeiro - devolve 1.\r\nFalso - devolve 0.\r\nUm valor de erro - devolve o valor do erro.\r\nQualquer outro valor - devolve 0.\r\n    ";
        this["Func_n_Category"] = "Informação";
        this["Func_n_CategoryURL"] = "Informação";
        this["Func_n_Desc"] = "Converte um valor num número.";
        this["Func_na_Category"] = "Informação";
        this["Func_na_CategoryURL"] = "Informação";
        this["Func_na_Desc"] = "Devolve o valor de erro #N/D (valor não disponível).";
        this["Func_networkdays_Arg_0"] = "startDate";
        this["Func_networkdays_Arg_1"] = "endDate";
        this["Func_networkdays_Arg_2"] = "feriados";
        this["Func_networkdays_ArgDesc_0"] = "A data a partir da qual começar.";
        this["Func_networkdays_ArgDesc_1"] = "A data na qual terminar.";
        this["Func_networkdays_ArgDesc_2"] = "Um intervalo de uma ou mais referências que contêm datas que devem ser consideradas dias não úteis.";
        this["Func_networkdays_Category"] = "DateAndTime";
        this["Func_networkdays_CategoryURL"] = "DateAndTime";
        this["Func_networkdays_Desc"] = "Devolve o número total de dias úteis completos entre as datas especificadas, excluindo os feriados especificados.";
        this["Func_not_Arg_0"] = "lógico";
        this["Func_not_ArgDesc_0"] = "Um valor lógico (verdadeiro ou falso).";
        this["Func_not_Category"] = "Lógico";
        this["Func_not_CategoryURL"] = "Lógico";
        this["Func_not_Desc"] = "Inverte um valor lógico (muda verdadeiro para falso e falso para verdadeiro).";
        this["Func_now_Category"] = "DateAndTime";
        this["Func_now_CategoryURL"] = "Date_and_Time";
        this["Func_now_Desc"] = "Devolve um valor que representa a data e hora atuais.";
        this["Func_nper_Arg_0"] = "taxa";
        this["Func_nper_Arg_1"] = "pmt";
        this["Func_nper_Arg_2"] = "pv";
        this["Func_nper_Arg_3"] = "fv";
        this["Func_nper_Arg_4"] = "type";
        this["Func_nper_ArgDesc_0"] = "A taxa de juros por período.";
        this["Func_nper_ArgDesc_1"] = "O pagamento efetuado a cada período, que não pode ser alterado ao longo da duração da anuidade.";
        this["Func_nper_ArgDesc_2"] = "O valor presente ou o montante fixo que uma série de pagamentos futuros vale no momento.";
        this["Func_nper_ArgDesc_3"] = "O valor futuro ou um saldo em dinheiro que deseja atingir após o último pagamento.";
        this["Func_nper_ArgDesc_4"] = "O número zero ou um que indica se os pagamentos são devidos no final do período (zero, o valor predefinido) ou no início (um).";
        this["Func_nper_Category"] = "Financeiro";
        this["Func_nper_CategoryURL"] = "Financeiro";
        this["Func_nper_Desc"] = "Devolve o número de períodos para um investimento com base em pagamentos periódicos e constantes e numa taxa de juro constante.";
        this["Func_npv_Arg_0"] = "taxa";
        this["Func_npv_Arg_1"] = "valor {0}";
        this["Func_npv_ArgDesc_0"] = "A taxa de desconto ao longo de um período.";
        this["Func_npv_ArgDesc_1"] = "Um pagamento (valor negativo) ou um rendimento (valor positivo).";
        this["Func_npv_Category"] = "Financeiro";
        this["Func_npv_CategoryURL"] = "Financeiro";
        this["Func_npv_Desc"] = "Calcula o valor atual líquido de um investimento usando uma taxa de desconto e uma série de pagamentos futuros (valores negativos) e receita (valor positivo).";
        this["Func_null_Category"] = "Informação";
        this["Func_null_CategoryURL"] = "Informação";
        this["Func_null_Desc"] = "Devolve um valor inexistente.";
        this["Func_oct2bin_Arg_0"] = "Número";
        this["Func_oct2bin_Arg_1"] = "Locais";
        this["Func_oct2bin_ArgDesc_0"] = "Uma expressão de texto que representa um número octal.";
        this["Func_oct2bin_ArgDesc_1"] = "O número de caracteres a utilizar para representar o valor. Se não for especificado, será utilizado o número mínimo de caracteres necessário para representar o valor. Isso permite especificar zeros à esquerda.";
        this["Func_oct2bin_Category"] = "Engenharia";
        this["Func_oct2bin_CategoryURL"] = "Engenharia";
        this["Func_oct2bin_Desc"] = "Converte uma expressão de texto que representa um número octal numa expressão de texto que representa o mesmo número no sistema numérico binário.";
        this["Func_oct2dec_Arg_0"] = "Número";
        this["Func_oct2dec_ArgDesc_0"] = "Uma expressão de texto que representa um número octal.";
        this["Func_oct2dec_Category"] = "Engenharia";
        this["Func_oct2dec_CategoryURL"] = "Engenharia";
        this["Func_oct2dec_Desc"] = "Devolve o valor decimal de uma representação de texto de um número no sistema numérico octal.";
        this["Func_oct2hex_Arg_0"] = "Número";
        this["Func_oct2hex_Arg_1"] = "Locais";
        this["Func_oct2hex_ArgDesc_0"] = "Uma expressão de texto que representa um número octal.";
        this["Func_oct2hex_ArgDesc_1"] = "O número de caracteres a utilizar para representar o valor. Se não for especificado, será utilizado o número mínimo de caracteres necessário para representar o valor. Isso permite especificar zeros à esquerda.";
        this["Func_oct2hex_Category"] = "Engenharia";
        this["Func_oct2hex_CategoryURL"] = "Engenharia";
        this["Func_oct2hex_Desc"] = "Converte uma expressão de texto que representa um número octal numa expressão de texto que representa o mesmo número no sistema numérico hexadecimal.";
        this["Func_odd_Arg_0"] = "número";
        this["Func_odd_ArgDesc_0"] = "O valor a arredondar.";
        this["Func_odd_Category"] = "Matemática";
        this["Func_odd_CategoryURL"] = "Matemática";
        this["Func_odd_Desc"] = "Arredonda um número positivo para cima e um número negativo para baixo até o número inteiro ímpar mais próximo.";
        this["Func_or_Arg_0"] = "lógico {0}";
        this["Func_or_ArgDesc_0"] = "Um valor lógico (verdadeiro ou falso).";
        this["Func_or_Category"] = "Lógico";
        this["Func_or_CategoryURL"] = "Lógico";
        this["Func_or_Desc"] = "Devolve verdadeiro se um ou mais valores forem verdadeiros; devolve falso se todos os valores forem falsos.";
        this["Func_pi_Category"] = "Matemática";
        this["Func_pi_CategoryURL"] = "Matemática";
        this["Func_pi_Desc"] = "Devolve a constante pi, que é 3,14159265358979.";
        this["Func_pmt_Arg_0"] = "taxa";
        this["Func_pmt_Arg_1"] = "nper";
        this["Func_pmt_Arg_2"] = "pv";
        this["Func_pmt_Arg_3"] = "fv";
        this["Func_pmt_Arg_4"] = "type";
        this["Func_pmt_ArgDesc_0"] = "A taxa de juros do empréstimo.";
        this["Func_pmt_ArgDesc_1"] = "O número total de pagamentos para o empréstimo.";
        this["Func_pmt_ArgDesc_2"] = "O valor presente ou o valor total que uma série de pagamentos futuros vale agora; também conhecido como o principal.";
        this["Func_pmt_ArgDesc_3"] = "O valor futuro ou um saldo em dinheiro que deseja atingir após o último pagamento.";
        this["Func_pmt_ArgDesc_4"] = "O número zero ou um que indica se os pagamentos são devidos no final do período (zero, o valor predefinido) ou no início (um).";
        this["Func_pmt_Category"] = "Financeiro";
        this["Func_pmt_CategoryURL"] = "Financeiro";
        this["Func_pmt_Desc"] = "Calcula o pagamento de um empréstimo com base em pagamentos constantes e em uma taxa de juros constante.";
        this["Func_power_Arg_0"] = "número";
        this["Func_power_Arg_1"] = "poder";
        this["Func_power_ArgDesc_0"] = "O número base para elevar a uma potência.";
        this["Func_power_ArgDesc_1"] = "O expoente ao qual o número base é elevado.";
        this["Func_power_Category"] = "Matemática";
        this["Func_power_CategoryURL"] = "Matemática";
        this["Func_power_Desc"] = "Devolve o resultado de um número elevado a uma potência.";
        this["Func_ppmt_Arg_0"] = "taxa";
        this["Func_ppmt_Arg_1"] = "por";
        this["Func_ppmt_Arg_2"] = "nper";
        this["Func_ppmt_Arg_3"] = "pv";
        this["Func_ppmt_Arg_4"] = "fv";
        this["Func_ppmt_Arg_5"] = "type";
        this["Func_ppmt_ArgDesc_0"] = "A taxa de juros por período.";
        this["Func_ppmt_ArgDesc_1"] = "O período, que deve estar no intervalo de um a nper.";
        this["Func_ppmt_ArgDesc_2"] = "O número total de períodos de pagamento numa anuidade.";
        this["Func_ppmt_ArgDesc_3"] = "O valor presente, ou o valor total, que uma série de pagamentos futuros vale agora.";
        this["Func_ppmt_ArgDesc_4"] = "O valor futuro ou um saldo em dinheiro que deseja atingir após o último pagamento.";
        this["Func_ppmt_ArgDesc_5"] = "O número zero ou um que indica se os pagamentos são devidos no final do período (zero, o valor predefinido) ou no início (um).";
        this["Func_ppmt_Category"] = "Financeiro";
        this["Func_ppmt_CategoryURL"] = "Financeiro";
        this["Func_ppmt_Desc"] = "Devolve o pagamento do capital por um determinado período para um investimento com base em pagamentos periódicos e constantes e numa taxa de juros constante.";
        this["Func_product_Arg_0"] = "número {0}";
        this["Func_product_ArgDesc_0"] = "Um número para multiplicar para os outros números.";
        this["Func_product_Category"] = "Matemática";
        this["Func_product_CategoryURL"] = "Matemática";
        this["Func_product_Desc"] = "Multiplica todos os números dados como argumentos e devolve o produto.";
        this["Func_pv_Arg_0"] = "taxa";
        this["Func_pv_Arg_1"] = "nper";
        this["Func_pv_Arg_2"] = "pmt";
        this["Func_pv_Arg_3"] = "fv";
        this["Func_pv_Arg_4"] = "type";
        this["Func_pv_ArgDesc_0"] = "A taxa de juros por período.";
        this["Func_pv_ArgDesc_1"] = "O número total de períodos de pagamento numa anuidade.";
        this["Func_pv_ArgDesc_2"] = "O pagamento efetuado a cada período, que não pode ser alterado ao longo da duração da anuidade.";
        this["Func_pv_ArgDesc_3"] = "O valor futuro ou um saldo em dinheiro que deseja atingir após o último pagamento.";
        this["Func_pv_ArgDesc_4"] = "O número zero ou um que indica se os pagamentos são devidos no final do período (zero, o valor predefinido) ou no início (um).";
        this["Func_pv_Category"] = "Financeiro";
        this["Func_pv_CategoryURL"] = "Financeiro";
        this["Func_pv_Desc"] = "Devolve o valor presente de um investimento.";
        this["Func_quotient_Arg_0"] = "numerador";
        this["Func_quotient_Arg_1"] = "denominador";
        this["Func_quotient_ArgDesc_0"] = "O dividendo da operação de divisão.";
        this["Func_quotient_ArgDesc_1"] = "O divisor da operação de divisão.";
        this["Func_quotient_Category"] = "Matemática";
        this["Func_quotient_CategoryURL"] = "Matemática";
        this["Func_quotient_Desc"] = "Devolve a parte inteira de uma divisão. Isso descartará o restante da divisão.";
        this["Func_radians_Arg_0"] = "Graus";
        this["Func_radians_ArgDesc_0"] = "Um valor em graus que será convertido em radianos.";
        this["Func_radians_Category"] = "Matemática";
        this["Func_radians_CategoryURL"] = "Matemática";
        this["Func_radians_Desc"] = "Converte graus em radianos.";
        this["Func_rand_Category"] = "Matemática";
        this["Func_rand_CategoryURL"] = "Matemática";
        this["Func_rand_Desc"] = "Devolve um número aleatório distribuído de forma igual ou superior a 0 e inferior a 1";
        this["Func_randbetween_Arg_0"] = "Fundo";
        this["Func_randbetween_Arg_1"] = "Topo";
        this["Func_randbetween_ArgDesc_0"] = "O valor mínimo que será devolvido.";
        this["Func_randbetween_ArgDesc_1"] = "O valor máximo que será devolvido.";
        this["Func_randbetween_Category"] = "Matemática";
        this["Func_randbetween_CategoryURL"] = "Matemática";
        this["Func_randbetween_Desc"] = "Gera um número inteiro pseudoaleatório entre dois números especificados.";
        this["Func_rate_Arg_0"] = "NPer";
        this["Func_rate_Arg_1"] = "Pmt";
        this["Func_rate_Arg_2"] = "Pv";
        this["Func_rate_Arg_3"] = "Fv";
        this["Func_rate_Arg_4"] = "Tipo";
        this["Func_rate_Arg_5"] = "Guess";
        this["Func_rate_ArgDesc_0"] = "O número total de períodos de pagamento numa anuidade.";
        this["Func_rate_ArgDesc_1"] = "O pagamento efetuado a cada período e não pode ser alterado ao longo da duração da anuidade.";
        this["Func_rate_ArgDesc_2"] = "O valor presente, ou o valor total, que uma série de pagamentos futuros vale agora.";
        this["Func_rate_ArgDesc_3"] = "O valor futuro ou o saldo em dinheiro que deseja atingir após o último pagamento. Se fv for omitido, assume-se que é 0 (o valor futuro de um empréstimo, por exemplo, é 0). Por exemplo, se quiser poupar 50.000$ para pagar um projeto especial em 18 anos, 50.000$ é o valor futuro. Pode adivinhar de forma conservadora uma taxa de juros e determinar quanto deve poupar em cada mês. Se o fv for omitido, deve incluir o argumento pmt. O valor predefinido para fv é zero.";
        this["Func_rate_ArgDesc_4"] = "O número zero ou um que indica se os pagamentos são devidos no final do período (zero, o valor predefinido) ou no início (um).";
        this["Func_rate_ArgDesc_5"] = "Um número que ache que está próximo do resultado. Se omitido, assume-se que é 0,1 (10%).";
        this["Func_rate_Category"] = "Financeiro";
        this["Func_rate_CategoryURL"] = "Financeiro";
        this["Func_rate_Desc"] = "Devolve a taxa de juro por período de uma anuidade.";
        this["Func_replace_Arg_0"] = "Old_text";
        this["Func_replace_Arg_1"] = "Start_num";
        this["Func_replace_Arg_2"] = "Num_chars";
        this["Func_replace_Arg_3"] = "New_text";
        this["Func_replace_ArgDesc_0"] = "A expressão de texto na qual substituir alguns caracteres.";
        this["Func_replace_ArgDesc_1"] = "O índice com base em 1 do carácter em texto_antigo para substituir por new_text.";
        this["Func_replace_ArgDesc_2"] = "O número de caracteres no texto antigo a substituir por texto novo (_T).";
        this["Func_replace_ArgDesc_3"] = "O texto que substituirá os caracteres em texto antigo (_T).";
        this["Func_replace_Category"] = "TextAndData";
        this["Func_replace_CategoryURL"] = "Text_and_Data";
        this["Func_replace_Desc"] = "Substitui parte de uma expressão de texto.";
        this["Func_rept_Arg_0"] = "TextValue";
        this["Func_rept_Arg_1"] = "RepetirContagem";
        this["Func_rept_ArgDesc_0"] = "O texto a repetir.";
        this["Func_rept_ArgDesc_1"] = "O número positivo de vezes a repetir o texto.";
        this["Func_rept_Category"] = "TextAndData";
        this["Func_rept_CategoryURL"] = "Text_and_Data";
        this["Func_rept_Desc"] = "Repete o texto um número específico de vezes.";
        this["Func_right_Arg_0"] = "texto";
        this["Func_right_Arg_1"] = "num_chars";
        this["Func_right_ArgDesc_0"] = "A expressão de texto que contém os caracteres a extrair.";
        this["Func_right_ArgDesc_1"] = "O número de caracteres a extrair.";
        this["Func_right_Category"] = "TextAndData";
        this["Func_right_CategoryURL"] = "Text_and_Data";
        this["Func_right_Desc"] = "Devolve o último carácter ou caracteres numa expressão de texto.";
        this["Func_roman_Arg_0"] = "Número";
        this["Func_roman_Arg_1"] = "Formulário";
        this["Func_roman_ArgDesc_0"] = "O número a converter em algarismos romanos.";
        this["Func_roman_ArgDesc_1"] = "Um número ou valor booleano que indica se é necessário usar números romanos clássicos ou uma versão mais concisa.\r\n0 = (Predefinição) Clássico. (499 = \" CDXCIX \")\r\n1 = Mais conciso. (499 = \" LDVLIV \")\r\n2 = Mais conciso. (499 = \" XDIX \")\r\n3 = Mais conciso. (499 = \" VDIV \")\r\n4 = Simplificado. (499 = \" ID \")\r\nVerdadeiro = Clássico.\r\nFalso = Simplificado.\r\n    ";
        this["Func_roman_Category"] = "Matemática";
        this["Func_roman_CategoryURL"] = "Matemática";
        this["Func_roman_Desc"] = "Converte um número num número romano como uma expressão de texto.";
        this["Func_round_Arg_0"] = "número";
        this["Func_round_Arg_1"] = "num_digits";
        this["Func_round_Arg_2"] = "round_Style";
        this["Func_round_ArgDesc_0"] = "O número a arredondar.";
        this["Func_round_ArgDesc_1"] = "O número de dígitos para o qual arredondar o número.";
        this["Func_round_ArgDesc_2"] = "Um número que indica o estilo de arredondamento a utilizar.\r\n0 - Arredondamento Aritmético (predefinição) - Se o último dígito significativo no arredondamento for 5, o número é arredondado para cima.\r\n1 - Arredondamento de Banqueiro - Quando o último dígito significativo for 5, o dígito anterior é utilizado para determinar se se deve arredondar para cima ou para baixo. Se o dígito for par, o número é arredondado para baixo. Se for ímpar, o número é arredondado para cima.\r\n    ";
        this["Func_round_Category"] = "Matemática";
        this["Func_round_CategoryURL"] = "Matemática";
        this["Func_round_Desc"] = "Arredonda um número para um número especificado de dígitos.";
        this["Func_rounddown_Arg_0"] = "Número";
        this["Func_rounddown_Arg_1"] = "Dígitos";
        this["Func_rounddown_ArgDesc_0"] = "O número a ser arredondado para baixo.";
        this["Func_rounddown_ArgDesc_1"] = "O número de casas decimais a arredondar. Números positivos indicam casas após o ponto decimal, números negativos indicam casas antes do ponto decimal.";
        this["Func_rounddown_Category"] = "Matemática";
        this["Func_rounddown_CategoryURL"] = "Matemática";
        this["Func_rounddown_Desc"] = "Arredonda um número para baixo, até o número de dígitos especificado.";
        this["Func_roundup_Arg_0"] = "Número";
        this["Func_roundup_Arg_1"] = "Dígitos";
        this["Func_roundup_ArgDesc_0"] = "O número a ser arredondado.";
        this["Func_roundup_ArgDesc_1"] = "O número de casas decimais a arredondar. Números positivos indicam casas após o ponto decimal, números negativos indicam casas antes do ponto decimal.";
        this["Func_roundup_Category"] = "Matemática";
        this["Func_roundup_CategoryURL"] = "Matemática";
        this["Func_roundup_Desc"] = "Arredonda um número para cima, até o número de dígitos especificado.";
        this["Func_search_Arg_0"] = "search_text";
        this["Func_search_Arg_1"] = "valor";
        this["Func_search_Arg_2"] = "start_index";
        this["Func_search_ArgDesc_0"] = "O texto a encontrar em valor.";
        this["Func_search_ArgDesc_1"] = "O texto no qual encontrar o search_text.";
        this["Func_search_ArgDesc_2"] = "O índice com base em 1 no qual iniciar a pesquisa (se omitido, o start_index é 1).";
        this["Func_search_Category"] = "TextAndData";
        this["Func_search_CategoryURL"] = "TextAndData";
        this["Func_search_Desc"] = "Devolve o índice com base em 1 de uma cadeia dentro de outra, pesquisando maiúsculas e minúsculas.";
        this["Func_searchb_Arg_0"] = "search_text";
        this["Func_searchb_Arg_1"] = "valor";
        this["Func_searchb_Arg_2"] = "start_index";
        this["Func_searchb_ArgDesc_0"] = "O texto a encontrar em valor.";
        this["Func_searchb_ArgDesc_1"] = "O texto no qual encontrar o search_text.";
        this["Func_searchb_ArgDesc_2"] = "O índice com base em 1 no qual iniciar a pesquisa (se omitido, o start_index é 1).";
        this["Func_searchb_Category"] = "TextAndData";
        this["Func_searchb_CategoryURL"] = "TextAndData";
        this["Func_searchb_Desc"] = "Devolve o índice com base em 1 de uma cadeia dentro de outra, pesquisando maiúsculas e minúsculas.";
        this["Func_second_Arg_0"] = "Time_value";
        this["Func_second_ArgDesc_0"] = "Um valor de tempo a partir do qual obter o segundo.";
        this["Func_second_Category"] = "DateAndTime";
        this["Func_second_CategoryURL"] = "Date_and_Time";
        this["Func_second_Desc"] = "Devolve o segundo de um valor de tempo.";
        this["Func_seriessum_Arg_0"] = "InputValue";
        this["Func_seriessum_Arg_1"] = "InitialPower";
        this["Func_seriessum_Arg_2"] = "Passo";
        this["Func_seriessum_Arg_3"] = "Coeficientes";
        this["Func_seriessum_ArgDesc_0"] = "O valor de entrada para a série de potências.";
        this["Func_seriessum_ArgDesc_1"] = "A potência inicial para a qual o ValorDeEntrada será aumentado.";
        this["Func_seriessum_ArgDesc_2"] = "O passo que será utilizado para aumentar a potência de cada termo da série.";
        this["Func_seriessum_ArgDesc_3"] = "Um conjunto de coeficientes pelo qual cada potência sucessiva de ValorDeEntrada é multiplicada.";
        this["Func_seriessum_Category"] = "Matemática";
        this["Func_seriessum_CategoryURL"] = "Matemática";
        this["Func_seriessum_Desc"] = "Devolve a soma de uma série de potências.";
        this["Func_sign_Arg_0"] = "Número";
        this["Func_sign_ArgDesc_0"] = "O número cujo sinal será devolvido.";
        this["Func_sign_Category"] = "Matemática";
        this["Func_sign_CategoryURL"] = "Matemática";
        this["Func_sign_Desc"] = "Devolve o sinal de um número (-1, 0 ou 1).";
        this["Func_sin_Arg_0"] = "número";
        this["Func_sin_ArgDesc_0"] = "O ângulo em radianos para o qual obter o seno.";
        this["Func_sin_Category"] = "Matemática";
        this["Func_sin_CategoryURL"] = "Matemática";
        this["Func_sin_Desc"] = "Devolve o seno do ângulo especificado.";
        this["Func_sinh_Arg_0"] = "número";
        this["Func_sinh_ArgDesc_0"] = "Um número real do qual obter o seno hiperbólico.";
        this["Func_sinh_Category"] = "Matemática";
        this["Func_sinh_CategoryURL"] = "Matemática";
        this["Func_sinh_Desc"] = "Devolve o seno hiperbólico de um número.";
        this["Func_sln_Arg_0"] = "custo";
        this["Func_sln_Arg_1"] = "residual";
        this["Func_sln_Arg_2"] = "vida";
        this["Func_sln_ArgDesc_0"] = "O custo inicial do ativo.";
        this["Func_sln_ArgDesc_1"] = "O valor no final da depreciação.";
        this["Func_sln_ArgDesc_2"] = "O número de períodos durante os quais o ativo é depreciado.";
        this["Func_sln_Category"] = "Financeiro";
        this["Func_sln_CategoryURL"] = "Financeiro";
        this["Func_sln_Desc"] = "Devolve a depreciação linear de um ativo durante um período.";
        this["Func_sqrt_Arg_0"] = "número";
        this["Func_sqrt_ArgDesc_0"] = "O número para o qual obter a raiz quadrada.";
        this["Func_sqrt_Category"] = "Matemática";
        this["Func_sqrt_CategoryURL"] = "Matemática";
        this["Func_sqrt_Desc"] = "Devolve uma raiz quadrada positiva de um número.";
        this["Func_sqrtpi_Arg_0"] = "Número";
        this["Func_sqrtpi_ArgDesc_0"] = "O número a ser multiplicado por pi.";
        this["Func_sqrtpi_Category"] = "Matemática";
        this["Func_sqrtpi_CategoryURL"] = "Matemática";
        this["Func_sqrtpi_Desc"] = "Multiplica o número especificado por pi e devolve a raiz quadrada do resultado.";
        this["Func_stdev_Arg_0"] = "número {0}";
        this["Func_stdev_ArgDesc_0"] = "Um número ou intervalo de números que especifica uma amostra de uma população.";
        this["Func_stdev_Category"] = "Estatística";
        this["Func_stdev_CategoryURL"] = "Estatística";
        this["Func_stdev_Desc"] = "Devolve o desvio padrão de uma amostra populacional.";
        this["Func_subtotal_Arg_0"] = "FunctionNum";
        this["Func_subtotal_Arg_1"] = "Ref {0}";
        this["Func_subtotal_ArgDesc_0"] = "Um número que representa a função a ser calculada. Adicione 100 ao valor para ignorar os valores ocultos.\r\n1 (ou 101) - MÉDIA\r\n2 (ou 102) - CONTAGEM\r\n4 (ou 104) - MÁX.\r\n5 (ou 105) - MÍN.\r\n6 (ou 106) - PRODUTO\r\n7 (ou 107) - DESV-PAD\r\n9 (ou 109) - SOMA\r\n10 (ou 110) - VAR.\r\n    ";
        this["Func_subtotal_ArgDesc_1"] = "Uma ou mais referências para as quais encontrar o subtotal.";
        this["Func_subtotal_Category"] = "Matemática";
        this["Func_subtotal_CategoryURL"] = "Matemática";
        this["Func_subtotal_Desc"] = "Calcula o subtotal numa ou mais referências.";
        this["Func_sum_Arg_0"] = "número {0}";
        this["Func_sum_ArgDesc_0"] = "Um número para adicionar aos outros números.";
        this["Func_sum_Category"] = "Matemática";
        this["Func_sum_CategoryURL"] = "Matemática";
        this["Func_sum_Desc"] = "Soma todos os números dados como argumentos e devolve a soma.";
        this["Func_syd_Arg_0"] = "custo";
        this["Func_syd_Arg_1"] = "residual";
        this["Func_syd_Arg_2"] = "vida";
        this["Func_syd_Arg_3"] = "por";
        this["Func_syd_ArgDesc_0"] = "O custo inicial do ativo.";
        this["Func_syd_ArgDesc_1"] = "O valor no final da depreciação.";
        this["Func_syd_ArgDesc_2"] = "O número de períodos durante os quais o ativo é depreciado.";
        this["Func_syd_ArgDesc_3"] = "O período, que deve usar as mesmas unidades que a vida.";
        this["Func_syd_Category"] = "Financeiro";
        this["Func_syd_CategoryURL"] = "Financeiro";
        this["Func_syd_Desc"] = "Devolve a depreciação dos dígitos da soma dos anos de um ativo durante um período especificado.";
        this["Func_tan_Arg_0"] = "número";
        this["Func_tan_ArgDesc_0"] = "O ângulo, em radianos, para o qual obter a tangente.";
        this["Func_tan_Category"] = "Matemática";
        this["Func_tan_CategoryURL"] = "Matemática";
        this["Func_tan_Desc"] = "Devolve a tangente do ângulo especificado.";
        this["Func_tanh_Arg_0"] = "número";
        this["Func_tanh_ArgDesc_0"] = "O ângulo, em radianos, para o qual obter a tangente hiperbólica.";
        this["Func_tanh_Category"] = "Matemática";
        this["Func_tanh_CategoryURL"] = "Matemática";
        this["Func_tanh_Desc"] = "Devolve a tangente hiperbólica do ângulo especificado.";
        this["Func_text_Arg_0"] = "Valor";
        this["Func_text_Arg_1"] = "FormatText";
        this["Func_text_ArgDesc_0"] = "O valor numérico que deve ser formatado.";
        this["Func_text_ArgDesc_1"] = "O formato numérico com o qual formatar o valor.";
        this["Func_text_Category"] = "TextAndData";
        this["Func_text_CategoryURL"] = "Text_and_Data";
        this["Func_text_Desc"] = "Converte o valor em texto com base no formato especificado.";
        this["Func_time_Arg_0"] = "hora";
        this["Func_time_Arg_1"] = "minuto";
        this["Func_time_Arg_2"] = "segundo";
        this["Func_time_ArgDesc_0"] = "Um número que representa uma hora.";
        this["Func_time_ArgDesc_1"] = "Um número que representa um minuto.";
        this["Func_time_ArgDesc_2"] = "Um número que representa um segundo.";
        this["Func_time_Category"] = "DateAndTime";
        this["Func_time_CategoryURL"] = "Date_and_Time";
        this["Func_time_Desc"] = "Devolve um número que representa um valor de tempo específico.";
        this["Func_timevalue_Arg_0"] = "time_text";
        this["Func_timevalue_ArgDesc_0"] = "Uma expressão de texto que representa uma hora.";
        this["Func_timevalue_Category"] = "DateAndTime";
        this["Func_timevalue_CategoryURL"] = "Date_and_Time";
        this["Func_timevalue_Desc"] = "Converte uma expressão de texto que representa uma hora no seu valor de tempo equivalente.";
        this["Func_today_Category"] = "DateAndTime";
        this["Func_today_CategoryURL"] = "Date_and_Time";
        this["Func_today_Desc"] = "Devolve a data atual.";
        this["Func_trim_Arg_0"] = "texto";
        this["Func_trim_ArgDesc_0"] = "O texto do qual remover espaços extras.";
        this["Func_trim_Category"] = "TextAndData";
        this["Func_trim_CategoryURL"] = "Text_and_Data";
        this["Func_trim_Desc"] = "Remove todos os espaços do texto, exceto espaços únicos entre as palavras.";
        this["Func_true_Category"] = "Lógico";
        this["Func_true_CategoryURL"] = "Lógico";
        this["Func_true_Desc"] = "Devolve o valor lógico verdadeiro.";
        this["Func_trunc_Arg_0"] = "número";
        this["Func_trunc_Arg_1"] = "num_digits";
        this["Func_trunc_ArgDesc_0"] = "O número a truncar.";
        this["Func_trunc_ArgDesc_1"] = "Um número que especifica a precisão do truncamento. O valor predefinido é zero.";
        this["Func_trunc_Category"] = "Matemática";
        this["Func_trunc_CategoryURL"] = "Matemática";
        this["Func_trunc_Desc"] = "Trunca um número para um número inteiro removendo a parte fracionária do número.";
        this["Func_type_Arg_0"] = "valor";
        this["Func_type_ArgDesc_0"] = "O valor do qual obter o tipo.";
        this["Func_type_Category"] = "Informação";
        this["Func_type_CategoryURL"] = "Informação";
        this["Func_type_Desc"] = "Devolve um número que representa o tipo de valor: número = 1; texto = 2; valor lógico (verdadeiro ou falso) = 4; valor do erro = 16; matriz = 64.";
        this["Func_upper_Arg_0"] = "texto";
        this["Func_upper_ArgDesc_0"] = "O texto a converter em maiúsculas.";
        this["Func_upper_Category"] = "TextAndData";
        this["Func_upper_CategoryURL"] = "Text_and_Data";
        this["Func_upper_Desc"] = "Converte todas as letras minúsculas numa expressão de texto em maiúsculas.";
        this["Func_value_Arg_0"] = "texto";
        this["Func_value_ArgDesc_0"] = "A expressão de texto a converter.";
        this["Func_value_Category"] = "TextAndData";
        this["Func_value_CategoryURL"] = "Text_and_Data";
        this["Func_value_Desc"] = "Converte num número uma expressão de texto que representa um número.";
        this["Func_var_Arg_0"] = "número {0}";
        this["Func_var_ArgDesc_0"] = "Um número ou intervalo de números que especifica uma amostra de uma população.";
        this["Func_var_Category"] = "Estatística";
        this["Func_var_CategoryURL"] = "Estatística";
        this["Func_var_Desc"] = "Devolve a variação de uma amostra da população.";
        this["Func_weekday_Arg_0"] = "Data";
        this["Func_weekday_Arg_1"] = "ReturnType";
        this["Func_weekday_ArgDesc_0"] = "A data na qual encontrar o número do dia da semana.";
        this["Func_weekday_ArgDesc_1"] = "Determina que esquema de numeração é utilizado para os dias da semana.\r\n1 (predefinição) = Domingo (1) a sábado (7)\r\n2 = Segunda-feira (1) a domingo (7)\r\n3 = Segunda-feira (0) a domingo (6)\r\n    ";
        this["Func_weekday_Category"] = "DateAndTime";
        this["Func_weekday_CategoryURL"] = "DateAndTime";
        this["Func_weekday_Desc"] = "Devolve um número que representa o dia da semana da data especificada.";
        this["Func_weeknum_Arg_0"] = "data";
        this["Func_weeknum_Arg_1"] = "tipoDeRetorno";
        this["Func_weeknum_ArgDesc_0"] = "A data em que encontrar o número da semana.";
        this["Func_weeknum_ArgDesc_1"] = "Determina em que dia começa a semana.\r\n1 (predefinido) = Domingo é o primeiro dia da semana.\r\n2 = Segunda-feira é o primeiro dia da semana.\r\n    ";
        this["Func_weeknum_Category"] = "DateAndTime";
        this["Func_weeknum_CategoryURL"] = "DateAndTime";
        this["Func_weeknum_Desc"] = "Devolve o número da semana em que a data especificada se enquadra.";
        this["Func_workday_Arg_0"] = "dataInício";
        this["Func_workday_Arg_1"] = "dias";
        this["Func_workday_Arg_2"] = "feriados";
        this["Func_workday_ArgDesc_0"] = "A data a partir da qual começar.";
        this["Func_workday_ArgDesc_1"] = "O número de dias úteis a serem compensados a partir da data de início.";
        this["Func_workday_ArgDesc_2"] = "Um intervalo de uma ou mais referências que contêm datas que devem ser consideradas dias não úteis.";
        this["Func_workday_Category"] = "DateAndTime";
        this["Func_workday_CategoryURL"] = "DateAndTime";
        this["Func_workday_Desc"] = "Devolve um dia útil o número de dias especificado a partir da data especificada, excluindo os feriados especificados.";
        this["Func_year_Arg_0"] = "Date_value";
        this["Func_year_ArgDesc_0"] = "Um valor de data a partir do qual obter o ano.";
        this["Func_year_Category"] = "DateAndTime";
        this["Func_year_CategoryURL"] = "Date_and_Time";
        this["Func_year_Desc"] = "Devolve o ano de um valor de data.";
        this["Func_yearfrac_Arg_0"] = "StartDate";
        this["Func_yearfrac_Arg_1"] = "EndDate";
        this["Func_yearfrac_Arg_2"] = "Basis";
        this["Func_yearfrac_ArgDesc_0"] = "Data de início é a data de início no intervalo.";
        this["Func_yearfrac_ArgDesc_1"] = "Data de fim é a data de fim no intervalo.";
        this["Func_yearfrac_ArgDesc_2"] = "\r\n      Base determina que contagem de dias utilizar.\r\n0 (Predefinição = E.U.A. (NASD) 30/360\r\n1 = Real/real\r\n2 = Real/360\r\n3 = Real/365\r\n4 = Europeu 30/360\r\n    ";
        this["Func_yearfrac_Category"] = "DateAndTime";
        this["Func_yearfrac_CategoryURL"] = "DateAndTime";
        this["Func_yearfrac_Desc"] = "Devolve uma fração que representa a parte do ano especificada entre a data de início e de fim especificadas.";
        this["GenerateTableColumnName"] = "Coluna";
        this["GenerateTableName"] = "Tabela";
        this["LD_Chart_AxisDisplayUnits"] = "Centenas:Centenas, Milhares:Milhares, DezenasMilhares:x 10000, CentenasMilhares:x 100000, Milhões:Milhões,DezenasMilhões:x 10000000, CentenasMilhões:x 100000000, MilharesMilhões:MilMilhões, MilhõesMilhões:Biliões";
        this["LD_Fallback_ChartEx_Line1"] = "Este gráfico não está disponível na sua versão do Excel.";
        this["LD_Fallback_ChartEx_Line2"] = "Editar esta forma ou guardar este livro num formato de ficheiro diferente irá interromper permanentemente o gráfico.";
        this["LE_AllMustHavGuidsIfAnyHasThem"] = "Não é possível definir GerarGuidsParaGrupos como Falso se qualquer grupo existente tiver um Guid.";
        this["LE_ArgumentException_AnchorCellFromOtherWorksheet"] = "A célula especificada não existe na mesma folha de cálculo que a forma.";
        this["LE_ArgumentException_ArrayFormulaMustHaveSingleRegion"] = "Uma fórmula de matriz só pode ser aplicada a uma região da célula.";
        this["LE_ArgumentException_ArrayTooSmall"] = "A matriz não é grande o suficiente para copiar todos os valores para ela.";
        this["LE_ArgumentException_CannotAddStandardTableStyle"] = "Um estilo de tabela predefinido não pode ser adicionado à ColeçãoDeEstilosDeTabelaPersonalizada.";
        this["LE_ArgumentException_CannotApplyDVRuleToTotalCell"] = "Uma ou mais células na referência especificada são células totais numa tabela e não podem ter uma regra de validação de dados aplicada.";
        this["LE_ArgumentException_CannotCreateEmptyColorInfo"] = "A cor vazia não pode ser utilizada para criar uma instância de InformaçãoDeCorDoLivro.";
        this["LE_ArgumentException_CellMustBeOnSameWorksheetAsReferencesCollection"] = "A célula deve ser da mesma folha de cálculo que a coleção de referência.";
        this["LE_ArgumentException_CellShiftedOffWorksheet"] = "A célula foi eliminada ou deslocada da folha de cálculo e já não representa uma posição da folha de cálculo.";
        this["LE_ArgumentException_CellsInTableFromOtherWorksheet"] = "A região especificada não é da mesma folha de cálculo que a tabela de dados.";
        this["LE_ArgumentException_CellValueStringLength"] = "Os valores da cadeia atribuídos a uma célula não podem ter um comprimento superior a {0}.";
        this["LE_ArgumentException_ColumnFormulaAlreadyAppliedToCell"] = "A fórmula já está aplicada numa célula. Clone-a para utilizá-la como fórmula da coluna.";
        this["LE_ArgumentException_ColumnFromDifferentWorksheet"] = "A coluna não pertence à mesma folha de cálculo que esta coleção.";
        this["LE_ArgumentException_ColumnInputCellFromOtherWorksheet"] = "A célula de entrada da coluna especificada não existe na mesma folha de cálculo que a tabela de dados.";
        this["LE_ArgumentException_ColumnNotInTable"] = "A coluna é de uma tabela diferente.";
        this["LE_ArgumentException_ColumnRemovedFromWorksheet"] = "A coluna foi removida da folha de cálculo.";
        this["LE_ArgumentException_ColumnsToRepeatAtLeftOutsideRange"] = "O intervalo está fora do intervalo de colunas disponíveis na folha de cálculo.";
        this["LE_ArgumentException_CommentTooLong"] = "O comprimento do comentário deve ter menos de 255 caracteres.";
        this["LE_ArgumentException_CustomViewNameAlreadyExists"] = "Já existe uma vista personalizada com o nome \"{0}\".";
        this["LE_ArgumentException_DefaultTableStyleNotInWorkbook"] = "O estilo de tabela especificado não está nas coleções EstilosDeTabelaPersonalizados ou EstilosDeTabelaPredefinidos.";
        this["LE_ArgumentException_DisplayTextTooLong"] = "O texto de exibição da hiperligação deve ter menos de 256 caracteres.";
        this["LE_ArgumentException_DuplicateDisplayValue"] = "O valor de exibição especificado já está na coleção.";
        this["LE_ArgumentException_DuplicateFixedDateGroup"] = "O GrupoDeDataFixado especificado já está na coleção.";
        this["LE_ArgumentException_DuplicateTableStyle"] = "O estilo especificado já está numa ColeçãoDeEstilosDeTabelaPersonalizada.";
        this["LE_ArgumentException_DuplicateTableStyleName_Existing"] = "O nome especificado que corresponde ao de outro estilo na coleção.";
        this["LE_ArgumentException_DuplicateTableStyleName_New"] = "O estilo especificado tem um nome que corresponde a outro estilo na coleção.";
        this["LE_ArgumentException_DuplicateWorksheetProtectedRangeTitle"] = "Já existe um intervalo com o título '{0}'.";
        this["LE_ArgumentException_DV_ArrayFormulaCannotBeUsed"] = "Uma FórmulaDeMatriz não pode ser aplicada como {0}.";
        this["LE_ArgumentException_DV_CellFromOtherWorksheet"] = "A célula não pertence à mesma folha de cálculo que a coleção de regras de validação de dados.";
        this["LE_ArgumentException_DV_ConstraintFormulaMustBeNumberOrReference"] = "O {0} deve ser definido como um número, uma expressão numérica ou uma referência a uma única célula.";
        this["LE_ArgumentException_DV_FormulaCannotBeNull"] = "O {0} não pode ser definido como nulo enquanto a regra de validação de dados é aplicada a uma folha de cálculo.";
        this["LE_ArgumentException_DV_FormulaCannotFindNamedReference"] = "O {0} contém uma referência com nome que não existe no livro.";
        this["LE_ArgumentException_DV_FormulaCannotFindWorksheetReference"] = "O {0} contém uma referência numa folha de cálculo que não existe no livro.";
        this["LE_ArgumentException_DV_FormulaCannotReferenceOtherWorkbook"] = "O {0} não pode conter uma referência a um livro externo.";
        this["LE_ArgumentException_DV_FormulaInvalidForWorkbookFormat"] = "O {0} faz referência a células que não são válidas no formato de livro {1}.";
        this["LE_ArgumentException_DV_InvalidAddress"] = "O endereço especificado não é um endereço de célula ou região válido.";
        this["LE_ArgumentException_DV_InvalidExcelDate"] = "A data especificada não pode ser representada no Excel.";
        this["LE_ArgumentException_DV_InvalidFormula"] = "Fórmula inválida utilizada na regra de validação de dados.";
        this["LE_ArgumentException_DV_InvalidMessageLength"] = "O {0} não pode ter mais do que {1} caracteres.";
        this["LE_ArgumentException_DV_ListFormulaCannotHaveEmptyString"] = "A constante de cadeia na FórmulaDeValores não pode ser uma cadeia vazia.";
        this["LE_ArgumentException_DV_ListFormulaMustBeStringOrReference"] = "A FórmulaDeValores deve ser definida como uma constante de cadeia ou uma referência a uma única célula ou região no mesmo livro.";
        this["LE_ArgumentException_DV_ListFormulaReferenceMustBeOneDimensional"] = "A FórmulaDeValores contém uma referência de região que possui várias linhas e colunas. A região deve conter uma linha ou uma coluna.";
        this["LE_ArgumentException_DV_LowerGreaterThanUpperContraint"] = "O valor da restrição mais baixa deve ser menor ou igual à restrição superior.";
        this["LE_ArgumentException_DV_MustHaveOneAcceptedValue"] = "Pelo menos um valor deve ser especificado como um valor aceite.";
        this["LE_ArgumentException_DV_ReferencesAlreadyContainDVs"] = "As referências especificadas já contêm uma ou mais regras de validação de dados.";
        this["LE_ArgumentException_DV_ReferencesFromOtherWorksheet"] = "A coleção de referências não pertence à mesma folha de cálculo que a coleção de regras de validação de dados.";
        this["LE_ArgumentException_DV_RegionFromOtherWorksheet"] = "A região não pertence à mesma folha de cálculo que a coleção de regras de validação de dados.";
        this["LE_ArgumentException_DV_RuleAppliedOnOtherWorksheet"] = "A regra de validação de dados já está aplicada a outra folha de cálculo.";
        this["LE_ArgumentException_DV_UpperLessThanLowerContraint"] = "O valor da restrição superior deve ser maior ou igual à restrição mais baixa.";
        this["LE_ArgumentException_EditRangeAlreadyOwned"] = "O intervalo já está nos IntervalosDeEdiçãoPermitidos de uma folha de cálculo.";
        this["LE_ArgumentException_EditRangeHasRangeFromOtherWorksheet"] = "O intervalo de edição possui um intervalo de uma folha de cálculo diferente da dos IntervalosDeEdiçãoPermitidos.";
        this["LE_ArgumentException_EmptyCustomList"] = "Deve haver pelo menos uma cadeia na lista.";
        this["LE_ArgumentException_EndIndexLessThanZero"] = "O final do intervalo deve ser maior ou igual a zero.";
        this["LE_ArgumentException_FileDoesntContainsWorkbookStream"] = "O fluxo ou ficheiro especificado não contém um livro válido.";
        this["LE_ArgumentException_FormulaReferencesInvalidCells"] = "A fórmula especificada faz referência a células que não são suportadas no formato de ficheiro atual do livro.";
        this["LE_ArgumentException_HashSetFull"] = "Existem muitos itens no conjunto de hash.";
        this["LE_ArgumentException_HyperlinkAlreadyOwned"] = "A hiperligação já pertence a uma coleção de hiperligações.";
        this["LE_ArgumentException_HyperlinkSourceFromOtherWorksheet"] = "A hiperligação tem uma célula ou região de origem de uma folha de cálculo diferente.";
        this["LE_ArgumentException_HyperlinkTargetFromOtherWorkbook"] = "O destino é de um livro diferente do do proprietário da hiperligação.";
        this["LE_ArgumentException_InfiniteColumnWidth"] = "A largura da coluna não pode ser infinita.";
        this["LE_ArgumentException_Interval"] = "O parâmetro Intervalo deve ser um dos valores na enumeração IntervaloDeData";
        this["LE_ArgumentException_IntervalStr"] = "O parâmetro Intervalo não pôde ser convertido para um dos valores na enumeração IntervaloDeData";
        this["LE_ArgumentException_InvalidCellAddress"] = "O endereço especificado não é um endereço de célula válido.";
        this["LE_ArgumentException_InvalidCommentPositioningMode"] = "Só os ModosDePosicionamentoDeFormas NãoMoverOuDimensionarCélulas e MoverEDimensionarCélulas podem ser aplicados a uma forma de comentário.";
        this["LE_ArgumentException_InvalidCustomFilterOperandNumber"] = "O valor não pode ser infinito ou NaN.";
        this["LE_ArgumentException_InvalidCustomFilterOperator"] = "Quando o valor de comparação não é uma cadeia, o operador de comparação não pode ser ComeçaCom, NãoComeçaCom, Contém, NãoContém, AcabaCom ou NãoAcabaCom.";
        this["LE_ArgumentException_InvalidDatePeriodFilterValue"] = "O valor não está dentro do intervalo. Para um tipo de filtro Mês, deve ser de 1 a 12. Para um tipo de filtro Trimestre, deve ser 1 a 4.";
        this["LE_ArgumentException_InvalidDpi"] = "O valor Dpi especificado é inválido.";
        this["LE_ArgumentException_InvalidExcelDate"] = "O valor não pode ser expresso como uma data no Excel.";
        this["LE_ArgumentException_InvalidFileFormat"] = "Formato de ficheiro inválido ou não reconhecido.";
        this["LE_ArgumentException_InvalidFormula"] = "A fórmula especificada era inválida.";
        this["LE_ArgumentException_InvalidGradientStopColor"] = "As cores automáticas e do sistema não podem ser utilizadas em gradientes.";
        this["LE_ArgumentException_InvalidNamedReferenceName"] = "O nome não é um nome de referência válido.";
        this["LE_ArgumentException_InvalidPaletteColor_EmptyOrSystem"] = "Cores vazias ou do sistema não podem ser adicionadas à paleta.";
        this["LE_ArgumentException_InvalidPaletteColor_NonOpaque"] = "Apenas cores opacas podem ser adicionadas à paleta.";
        this["LE_ArgumentException_InvalidReferencesString"] = "O valor das referências não está formatado corretamente.";
        this["LE_ArgumentException_InvalidRegionAddress"] = "O endereço especificado não é um endereço de região válido.";
        this["LE_ArgumentException_InvalidRowOrColumnRange"] = "Linha ou intervalo de colunas inválido.";
        this["LE_ArgumentException_InvalidSeriesValues"] = "Valores de série inválidos. Deve haver pelo menos um valor e cada valor deve ser uma cadeia ou um objeto que possa ser convertido num duplo.";
        this["LE_ArgumentException_InvalidTarget"] = "O destino da hiperligação deve ser uma sequência de caracteres, CélulaDeFolhaDeCálculo, RegiãoDeFolhaDeCálculo ou ReferênciaComNome.";
        this["LE_ArgumentException_InvalidTopOrBottomFilterValue"] = "O valor deve estar entre 1 e 500.";
        this["LE_ArgumentException_InvalidWorksheetName"] = "O nome da folha de cálculo não pode conter nenhum dos seguintes caracteres: : \\ /? * []\r\n";
        this["LE_ArgumentException_InvalidWorksheetNameStartingQuote"] = "O nome da folha de cálculo não pode começar ou terminar com uma única citação.";
        this["LE_ArgumentException_LessThanTwoGradientStops"] = "Deve haver pelo menos duas paragens de gradiente especificadas.";
        this["LE_ArgumentException_NamedReferenceNameTooLong"] = "O nome da referência com nome não pode ter mais do que 255 caracteres.";
        this["LE_ArgumentException_NaNDefaultColumnWidth"] = "NaN não pode ser utilizado como a largura predefinida da coluna.";
        this["LE_ArgumentException_NewRegionOverlapsFilterArea"] = "A operação está a tentar alterar um intervalo filtrado na folha de cálculo e não pode ser concluída. Para concluir esta operação, é necessário remover os FiltrosAutomáticos na folha.";
        this["LE_ArgumentException_NewTableRegionCannotMoveHeaders"] = "A nova região da tabela não pode mover os cabeçalhos para uma nova linha.";
        this["LE_ArgumentException_NewTableRegionFromWrongWorksheet"] = "A região especificada não está na mesma folha de cálculo que a tabela.";
        this["LE_ArgumentException_NewTableRegionMustHaveDataRows"] = "A nova região da tabela deve conter pelo menos uma linha de dados.";
        this["LE_ArgumentException_NewTableRegionOverlapOld"] = "A nova região da tabela deve sobrepor-se à região anterior da tabela.";
        this["LE_ArgumentException_NewTableRegionOverlapsBlockingValue"] = "A tabela não pode ser redimensionada para a região especificada porque inclui uma fórmula de matriz com várias células ou tabela de dados.";
        this["LE_ArgumentException_NewTableRegionOverlapsMergedRegion"] = "A tabela não pode ser redimensionada para a região especificada porque inclui uma célula unida.";
        this["LE_ArgumentException_NewTableRegionOverlapsOtherTable"] = "A tabela não pode ser redimensionada para a região especificada porque inclui outra tabela.";
        this["LE_ArgumentException_NoRegionsInArray"] = "A matriz especificada não possui regiões.";
        this["LE_ArgumentException_NotEnum"] = "O tipo '{0}' não é uma enumeração";
        this["LE_ArgumentException_NotSupportedCodePage"] = "Página de códigos não suportada";
        this["LE_ArgumentException_NPer"] = "nPer não pode ser 0";
        this["LE_ArgumentException_NPerFunction"] = "Combinação errada de parâmetros para a função NPer";
        this["LE_ArgumentException_PA_PrintAreasCannotOverlap"] = "A área de impressão especificada sobrepõe-se a uma das áreas de impressão existentes.";
        this["LE_ArgumentException_ParentStyleFromOtherWorkbook"] = "O estilo especificado não é do mesmo livro que o formato da célula.";
        this["LE_ArgumentException_PB_CantInsertBreakAtIndex"] = "Esta coleção não suporta a inserção de itens num índice.";
        this["LE_ArgumentException_PB_CantSetBreakAtIndex"] = "Esta coleção não suporta a definição de itens num índice.";
        this["LE_ArgumentException_PB_MustBeWithinPrintArea"] = "A quebra de página deve estar dentro da área de impressão.";
        this["LE_ArgumentException_PB_PageBreakCantBeA1Cell"] = "Não é possível inserir uma quebra de página na parte superior esquerda da folha de cálculo.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeLeftColumn"] = "Não é possível inserir uma quebra de página no lado esquerdo da folha de cálculo.";
        this["LE_ArgumentException_PB_PageBreakCantBeBeforeTopRow"] = "Não é possível inserir uma quebra de página na parte superior da folha de cálculo.";
        this["LE_ArgumentException_PB_PageBreaksCannotAddExistingInstance"] = "A instância especificada já existe na coleção.";
        this["LE_ArgumentException_PB_PageBreaksCannotOverlap"] = "A quebra de página não pode sobrepor-se a outras quebras de página na folha de cálculo.";
        this["LE_ArgumentException_PB_PrintAreaMustBeInPrintAreas"] = "A quebra de página está numa área de impressão que não está na coleção ÁreasDeImpressão.";
        this["LE_ArgumentException_PB_PrintAreaMustBeOnSameWorksheet"] = "A quebra de página está numa área de impressão que não está na mesma folha de cálculo que a coleção de quebras de página.";
        this["LE_ArgumentException_Pmt"] = "Pmt não pode ser 0";
        this["LE_ArgumentException_PrintAreaShiftedOffWorksheet"] = "A quebra de página está numa área de impressão que foi deslocada da folha de cálculo e já não representa um intervalo válido de células.";
        this["LE_ArgumentException_RangeFromOtherWorksheetThanProtectedRange"] = "O intervalo especificado é de uma folha de cálculo diferente da do intervalo de edição.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheet"] = "A região deve ser da mesma folha de cálculo.";
        this["LE_ArgumentException_RegionMustBeOnSameWorksheetAsReferencesCollection"] = "A região deve ser da mesma folha de cálculo que a coleção de referência.";
        this["LE_ArgumentException_RegionsFromMixedWorksheets"] = "Todas as regiões especificadas na matriz devem ser da mesma folha de cálculo.";
        this["LE_ArgumentException_RegionShiftedOffWorksheet"] = "A região foi deslocada para fora da folha de cálculo e já não representa um intervalo válido de células.";
        this["LE_ArgumentException_RegionsShiftedOffWorksheet"] = "Uma ou mais regiões especificadas foram retiradas da folha de cálculo e já não representam um intervalo válido de células.";
        this["LE_ArgumentException_RowFromDifferentWorksheet"] = "A linha não pertence à mesma folha de cálculo que esta coleção.";
        this["LE_ArgumentException_RowInputCellFromOtherWorksheet"] = "A célula de entrada da linha especificada não existe na mesma folha de cálculo que a tabela de dados.";
        this["LE_ArgumentException_RowRemovedFromWorksheet"] = "A linha foi removida da folha de cálculo.";
        this["LE_ArgumentException_SelectedWorksheetFromOtherWorkbook"] = "A folha de cálculo não pertence ao livro associado a essas opções de janela.";
        this["LE_ArgumentException_ShapeCannotBeAdded"] = "A forma especificada não pode ser adicionada a uma coleção de formas.";
        this["LE_ArgumentException_SourceFromOtherWorksheetThanHyperlink"] = "A célula ou região de origem especificada é de uma folha de cálculo diferente da hiperligação.";
        this["LE_ArgumentException_StartIndexLessThanZero"] = "O início do intervalo deve ser maior ou igual a zero.";
        this["LE_ArgumentException_StyleNameAlreadyExists"] = "Um estilo com o nome \"{0}\" já existe no livro.";
        this["LE_ArgumentException_StyleNameTooLong"] = "O nome do estilo não pode ter mais do que 255 caracteres.";
        this["LE_ArgumentException_TableColumnRemovedFromTable"] = "A coluna da tabela foi removida da tabela.";
        this["LE_ArgumentException_TableStyleFromOtherWorkbook"] = "O EstiloDeTabelaDeFolhaDeCálculo especificado não pertence ao mesmo livro que a tabela.";
        this["LE_ArgumentException_TargetTooLong"] = "O endereço de destino da hiperligação deve ter menos de 256 caracteres.";
        this["LE_ArgumentException_TextLengthGreaterThanMax"] = "O comprimento do texto numa forma ou comentário não pode ser superior a {0}.";
        this["LE_ArgumentException_ToolTipTooLong"] = "A descrição da hiperligação deve ter menos de 256 caracteres.";
        this["LE_ArgumentException_TopLeftWindowBounds"] = "O lado esquerdo e a parte superior dos limites da janela devem estar entre -32768 e 32767.";
        this["LE_ArgumentException_WeekdayFirst"] = "O parâmetro DiaDaSemanaPrimeiro deve estar entre PrimeiroDiaDaSemana.Sistema e PrimeiroDiaDaSemana.Sábado";
        this["LE_ArgumentException_WidthHeightWindowBounds"] = "A largura e a altura dos limites da janela devem estar entre 0 e 65535.";
        this["LE_ArgumentException_WorksheetNameAlreadyExists"] = "Uma folha de cálculo com o nome \"{0}\" já existe no livro.";
        this["LE_ArgumentException_WorksheetNameTooLong"] = "O nome da folha de cálculo deve ter entre 1 e 31 caracteres.";
        this["LE_ArgumentException_WorksheetProtectedRangeTitleTooLong"] = "O título de um intervalo de edição não pode ter mais do que 255 caracteres.";
        this["LE_ArgumentException_WorksheetScopeFromOtherWorkbook"] = "A folha de cálculo não pertence ao mesmo livro que esta coleção.";
        this["LE_ArgumentNullException_AnchorCell"] = "A célula não pode ser nula quando a forma existe numa folha de cálculo ou grupo.";
        this["LE_ArgumentNullException_CustomViewName"] = "O nome da vista personalizada não pode ser nulo ou vazio.";
        this["LE_ArgumentNullException_Encoder"] = "O codificador de imagem não pode ser nulo.";
        this["LE_ArgumentNullException_FindNamedReference"] = "O nome a encontrar não pode ser nulo ou vazio.";
        this["LE_ArgumentNullException_FormulaCantBeNull"] = "A fórmula para uma referência com nome não pode ser nula ou vazia.";
        this["LE_ArgumentNullException_HiddenColumn"] = "Uma coluna nula não pode ser adicionada à coleção.";
        this["LE_ArgumentNullException_HiddenRow"] = "Uma linha nula não pode ser adicionada à coleção.";
        this["LE_ArgumentNullException_Image"] = "A imagem não pode ser nula.";
        this["LE_ArgumentNullException_NamedReferenceNameCantBeNull"] = "O nome de uma referência com nome não pode ser nulo ou vazio.";
        this["LE_ArgumentNullException_SaveStream"] = "Não é possível guardar o livro num fluxo nulo.";
        this["LE_ArgumentNullException_SelectedWorksheet"] = "O livro deve ter uma folha de cálculo selecionada válida.";
        this["LE_ArgumentNullException_Shape"] = "Uma forma nula não pode ser adicionada à coleção.";
        this["LE_ArgumentNullException_SourceFont"] = "O tipo de letra de origem não pode ser nulo.";
        this["LE_ArgumentNullException_SourceFormatting"] = "A formatação de origem não pode ser nula.";
        this["LE_ArgumentNullException_StyleName"] = "O nome do estilo não pode ser nulo ou vazio.";
        this["LE_ArgumentNullException_UnformattedString"] = "A cadeia não formatada não pode ser nula.";
        this["LE_ArgumentNullException_Workbook"] = "O livro não pode ser nulo.";
        this["LE_ArgumentNullException_WorkbookRequiredToResolveThemeColor"] = "Um livro deve ser especificado para resolver uma cor com tema.";
        this["LE_ArgumentNullException_WorksheetName"] = "O nome da folha de cálculo não pode ser nulo ou vazio.";
        this["LE_ArgumentNullException_WorksheetProtectedRangeTitleCantBeNull"] = "O título de um intervalo de edição não pode ser nulo ou vazio.";
        this["LE_ArgumentNullException_WorksheetScope"] = "O âmbito da folha de cálculo não pode ser nulo.";
        this["LE_ArgumentOutOfRange_RelativeColumnIndex"] = "O índice da coluna relativa está fora da região especificada.";
        this["LE_ArgumentOutOfRange_RelativeRowIndex"] = "O índice de linha relativo está fora da região especificada.";
        this["LE_ArgumentOutOfRangeException_AnchorPosition"] = "Uma das percentagens na localização especificada é inferior a 0.";
        this["LE_ArgumentOutOfRangeException_BadInsertIndex"] = "O índice especificado deve ser maior ou igual a zero e menor ou igual ao número de parágrafos existentes.";
        this["LE_ArgumentOutOfRangeException_BadRemoveAtIndex"] = "O índice deve ser maior que zero e menor que o número de parágrafos na cadeia formatada.";
        this["LE_ArgumentOutOfRangeException_CollectionIndex"] = "O índice estava fora do intervalo. Deve ser não negativo e menor que o tamanho da coleção.";
        this["LE_ArgumentOutOfRangeException_ColumnWidth"] = "A largura da coluna deve estar entre 0 e 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultColumnWidth"] = "A largura predefinida da coluna deve estar entre 0 e 65535.";
        this["LE_ArgumentOutOfRangeException_DefaultFontHeight"] = "A altura predefinida do tipo de letra deve estar entre 20 e 8180.";
        this["LE_ArgumentOutOfRangeException_DefaultRowHeight"] = "A altura predefinida da linha deve estar entre -1 e 8192.";
        this["LE_ArgumentOutOfRangeException_DuplicateItemSorted"] = "O item especificado já tem uma condição de ordenação aplicada.";
        this["LE_ArgumentOutOfRangeException_FirstVisibleTabIndex"] = "O primeiro índice de tabulação visível deve ser maior ou igual a zero.";
        this["LE_ArgumentOutOfRangeException_FontHeight"] = "A altura do tipo de letra deve estar entre 20 e 8180.";
        this["LE_ArgumentOutOfRangeException_GroupAddedToSelf"] = "Um grupo não pode ser adicionado à sua própria coleção de formas.";
        this["LE_ArgumentOutOfRangeException_Indent"] = "O nível de indentação deve estar entre 0 e 250.";
        this["LE_ArgumentOutOfRangeException_IndexNegative"] = "O índice não deve ser negativo.";
        this["LE_ArgumentOutOfRangeException_InvalidCollectionIndex"] = "O índice especificado não é válido na coleção.";
        this["LE_ArgumentOutOfRangeException_InvalidColorInfoTint"] = "O valor da tonalidade deve estar entre -1,0 e 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnCount"] = "{0} não é uma contagem de colunas válida no formato atual. O valor deve estar entre {1} e {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidColumnIndex"] = "{0} não é um índice de coluna válido no formato atual. O valor deve estar entre {1} e {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidGradientStopOffset"] = "O deslocamento da paragem do gradiente deve estar entre 0,0 e 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRelativeRectangleValueForGradient"] = "Os valores para o retângulo interno do gradiente retangular devem estar entre 0,0 e 1,0.";
        this["LE_ArgumentOutOfRangeException_InvalidRemoveAtIndex_SortConditions"] = "O índice no qual remover a condição de ordenação está fora do intervalo.";
        this["LE_ArgumentOutOfRangeException_InvalidRowCount"] = "{0} não é uma contagem de linhas válida no formato atual. O valor deve estar entre {1} e {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidRowIndex"] = "{0} não é um índice de linha válido no formato atual. O valor deve estar entre {1} e {2}.";
        this["LE_ArgumentOutOfRangeException_InvalidTableStyleAreaStripeExtent"] = "O {0} deve estar entre 1 e 9, inclusive.";
        this["LE_ArgumentOutOfRangeException_LeftPaneWidth"] = "A largura do painel especificada está fora do intervalo válido de {0} e {1}.";
        this["LE_ArgumentOutOfRangeException_LengthMustBePositive"] = "O comprimento deve ser maior ou igual a um.";
        this["LE_ArgumentOutOfRangeException_MagnificationLevel"] = "Os níveis de ampliação da vista devem estar entre 10 e 400.";
        this["LE_ArgumentOutOfRangeException_Margins"] = "As margens devem estar entre 0 e 100.";
        this["LE_ArgumentOutOfRangeException_MaxPagesHorizontally"] = "As páginas máximas na horizontal devem estar entre 0 e 32767.";
        this["LE_ArgumentOutOfRangeException_MaxPagesVertically"] = "As páginas máximas na vertical devem estar entre 0 e 32767.";
        this["LE_ArgumentOutOfRangeException_MaxRecursionIterations"] = "As iterações máximas de recursão devem estar entre 1 e 32767.";
        this["LE_ArgumentOutOfRangeException_MaxSortConditions"] = "Não pode haver mais do que {0} condições de ordenação na coleção.";
        this["LE_ArgumentOutOfRangeException_NegativeStartIndex"] = "O índice inicial não pode ser negativo.";
        this["LE_ArgumentOutOfRangeException_NumberOfCopies"] = "O número de cópias deve estar entre 1 e 65535.";
        this["LE_ArgumentOutOfRangeException_Per"] = "por valor deve estar entre 0 e nPer + 1";
        this["LE_ArgumentOutOfRangeException_Rate"] = "A taxa deve ser superior a -1";
        this["LE_ArgumentOutOfRangeException_Resolution"] = "A resolução deve estar entre 0 e 65535.";
        this["LE_ArgumentOutOfRangeException_RowHeight"] = "A altura da linha deve estar entre 0 e 8192.";
        this["LE_ArgumentOutOfRangeException_ScalingFactor"] = "O fator de dimensionamento deve estar entre 10 e 400.";
        this["LE_ArgumentOutOfRangeException_StartPageNumber"] = "O número da página inicial deve estar entre -32765 e 32767.";
        this["LE_ArgumentOutOfRangeException_TabBarWidth"] = "A largura da barra de separadores deve estar entre 0 e 1000.";
        this["LE_ArgumentOutOfRangeException_TopPaneHeight"] = "A altura do painel especificada está fora do intervalo válido de {0} e {1}.";
        this["LE_AutoColorNotAllowed"] = "A cor não pode ser 'Auto'.";
        this["LE_Axis_MajorMinorUnit"] = "O valor do parâmetro unidadePrincipal deve ser maior que o valor do parâmetro unidadeSecundária.";
        this["LE_Axis_NoCrossAxis"] = "Se a propriedade EixoCruzado estiver definida num objeto Eixo, ela deverá estar definida em todos eles. Se o gráfico tiver apenas dois eixos, o eixo de cruzamento poderá ser determinado automaticamente, mas apenas se a propriedade EixoCruzado for deixada desmarcada em todos os objetos do Eixo.";
        this["LE_Biff8SerializerNotLoaded"] = "O formato '{0}' requer um serializador que não esteja carregado. No JavaScript, este está contido no ficheiro infragistics.excel_serialization_biff8.js";
        this["LE_CategoryAxisBinning_BinWidthAndNumberOfBins"] = "Os parâmetros 'LarguraDeClasses' e 'NúmeroDeClasses' são mutuamente exclusivos. Ambos podem ser não especificados, mas se um for especificado, o outro não o deve ser.";
        this["LE_ChartGradientFill_EmptyStops"] = "Deve conter pelo menos uma paragem de gradiente.";
        this["LE_ChartObject_DifferentChart"] = "A instância do objeto a ser atribuída está associada a um gráfico diferente e não pode ser atribuída a esta instância.";
        this["LE_ColumnsToRepeatAtLeftAreOutsideAvailableRange"] = "As opções de impressão têm colunas repetidas à esquerda que estão fora do intervalo de colunas disponíveis no novo formato.";
        this["LE_ComboChartGroupCollection_DoesNotExist"] = "O GrupoDoGráficoDeCombinação especificado não existe nesta coleção.";
        this["LE_FillPercentInvalid"] = "O valor {0} não é válido para a propriedade {1}.";
        this["LE_FormatLimitError_Indent"] = "O avanço excede o valor máximo de {0} fornecido pelo formato atual.";
        this["LE_FormatLimitError_MacroWorkbook"] = "O livro tem código VBA e não pode ser guardado como um livro ativado por macro.";
        this["LE_FormatLimitError_MaxColumnIndex"] = "O índice da coluna de {0} é superior ao máximo de {1} suportado pelo formato especificado.";
        this["LE_FormatLimitError_MaxRowIndex"] = "O índice de linha de {0} excede o máximo de {1} suportado por este formato.";
        this["LE_FormulaParseException_ArrayContainsConstants"] = "Uma matriz só pode conter expressões constantes.";
        this["LE_FormulaParseException_ArrayHasEmptyFirstRow"] = "Uma matriz constante na fórmula não tem itens na primeira linha.";
        this["LE_FormulaParseException_ArrayHasMisalignedRows"] = "Uma matriz constante na fórmula tem linhas desalinhadas. Cada linha de uma matriz deve ter o mesmo número de colunas.";
        this["LE_FormulaParseException_CellReferenceAfterWorkbookName"] = "Uma referência de livro externo deve ser seguida de uma referência com nome, não de uma referência de célula.";
        this["LE_FormulaParseException_ExtraExpressions"] = "A fórmula tem expressões extras no final.";
        this["LE_FormulaParseException_FunctionMissingClosingParen"] = "Uma função na fórmula não tem parêntese de fecho.";
        this["LE_FormulaParseException_FunctionNestingTooDeep"] = "Uma função na fórmula está aninhada muito profundamente noutras funções.";
        this["LE_FormulaParseException_IncorrectNumberOfArguments"] = "Uma função na fórmula não recebeu o número correto de argumentos.";
        this["LE_FormulaParseException_InvalidArguments"] = "Um operador ou função na fórmula tem argumentos inválidos.";
        this["LE_FormulaParseException_InvalidConditionalFormatFormula"] = "Uma fórmula de formato condicional não pode utilizar operadores de referência, como uniões ou interseções, constantes de matriz ou referências de tabela estruturada.";
        this["LE_FormulaParseException_InvalidErrorValue"] = "O tipo de erro especificado não é válido. Deve ser um dos seguintes valores: \"#NULL!\", \"#DIV/0!\", \"#VALUE!\", \"#REF!\", \"#NAME?\", \"#NUM!\" Ou \"# N/A\".";
        this["LE_FormulaParseException_InvalidFileNameInBrackets"] = "Um nome de ficheiro entre parênteses retos era inválido.";
        this["LE_FormulaParseException_InvalidStructuredTableReference"] = "A referência da tabela estruturada está incorreta.";
        this["LE_FormulaParseException_InvalidWorkbookName"] = "A referência do livro entre aspas não é um nome de ficheiro válido.";
        this["LE_FormulaParseException_InvalidWorksheetName"] = "O nome da folha de cálculo é inválido.";
        this["LE_FormulaParseException_MacroFunctionNotAllowed"] = "As funções '{0}' são válidas apenas numa referência com nome.";
        this["LE_FormulaParseException_Message_PortionWithError"] = "Parte com erro: {0}";
        this["LE_FormulaParseException_MissingArgumentAfterBinary"] = "Falta um operando depois de um operador binário na fórmula.";
        this["LE_FormulaParseException_MissingArgumentAfterParen"] = "Não há qualquer expressão após um parêntese de abertura.";
        this["LE_FormulaParseException_MissingArgumentAfterUnary"] = "Um operador unário na fórmula não foi seguido por uma expressão válida.";
        this["LE_FormulaParseException_MissingArgumentBeforeBinary"] = "Falta um operando antes de um operador binário na fórmula.";
        this["LE_FormulaParseException_NamedReferenceRefsNeedSheetName"] = "Todas as referências da célula numa fórmula de referência com nome devem identificar a folha de cálculo.";
        this["LE_FormulaParseException_NoElementAfterArraySerapator"] = "Não existe elemento de matriz após um separador de elemento na fórmula.";
        this["LE_FormulaParseException_NoEqualsSign"] = "A fórmula deve começar com um sinal de igual.";
        this["LE_FormulaParseException_NoExclamationAfterWorkbookName"] = "A referência do livro deve ser seguida de um ponto de exclamação.";
        this["LE_FormulaParseException_NoExclamationAfterWorksheetName"] = "A referência da folha de cálculo deve ser seguida de um ponto de exclamação.";
        this["LE_FormulaParseException_NoExpressions"] = "A fórmula não tem expressões válidas após o sinal de igual.";
        this["LE_FormulaParseException_NoFileNameAfterBracket"] = "Um nome de ficheiro válido era esperado após o parêntese reto de abertura.";
        this["LE_FormulaParseException_NoValidTermAfterWorkbookName"] = "Uma referência de livro externo deve ser seguida de uma referência com nome.";
        this["LE_FormulaParseException_NoValidTermAfterWorksheetName"] = "Uma referência de folha de cálculo deve ser seguida de uma célula ou referência com nome.";
        this["LE_FormulaParseException_NoWorksheetAfterWorkbookName"] = "Um nome de ficheiro de livro entre parênteses retos deve ser seguido de um nome de folha de cálculo.";
        this["LE_FormulaParseException_StringConstantLengthTooLong"] = "A fórmula especificada não pode conter uma cadeia com mais de {0} caracteres.";
        this["LE_FormulaParseException_TooLong"] = "O comprimento máximo de uma fórmula não pode exceder {0} para o formato de livro especificado.";
        this["LE_FormulaParseException_UnknownFunction"] = "Uma função na fórmula não tem um nome de função válido.";
        this["LE_FormulaParseException_UnmatchedOpenBracket"] = "Uma matriz constante na fórmula não tem um parêntese reto de fecho.";
        this["LE_FormulaParseException_UnmatchedOpenParen"] = "Um parêntese de abertura na fórmula não tem um parêntese de fecho correspondente.";
        this["LE_FormulaParseException_UnmatchedOpenSquareBracket"] = "Era esperado um parêntese reto de fecho após o nome de ficheiro do livro.";
        this["LE_FormulaParseException_WorkbookNameMissingEndQuote"] = "A referência do livro não tem as aspas fechadas.";
        this["LE_FormulaParseException_WorksheetRangeMissingEndingName"] = "O intervalo da folha de cálculo não especifica um segundo nome para a folha de cálculo.";
        this["LE_GradientStop_InvalidPosition"] = "Deve especificar um número entre 0 e 1.";
        this["LE_IconSetInvalid"] = "'ConjuntoDeíconesSemícones' não é uma definição válida para a propriedade ConjuntoDeícones.";
        this["LE_IndexOutOfRangeException_ArrayBounds"] = "O índice estava fora dos limites da matriz.";
        this["LE_InvalidEnumArgumentException_DefaultPatternCannotBeUsed"] = "O EstiloDePadrãoDePreenchimento predefinido não pode ser utilizado para criar um PadrãoDePreenchimentoDaCélula.";
        this["LE_InvalidLineWeight"] = "A EspessuraDaLinha não deve ser inferior a 0 ou superior a 1584.";
        this["LE_InvalidOperationException_AnchorCommentBeforeApplyingToCell"] = "O comentário deve ter os conjuntos CélulaCantoEsquerdoSuperior e CélulaCantoDireitoInferior, ou ambos devem ser nulos antes de serem aplicados a uma célula.";
        this["LE_InvalidOperationException_AnchorShapeBeforeAddingToCollection"] = "A forma deve ter o conjunto CélulaCantoEsquerdoSuperior e CélulaCantoDireitoInferior antes de ser adicionado a uma folha de cálculo ou grupo.";
        this["LE_InvalidOperationException_ArrayFormulaAlreadyApplied"] = "A fórmula de matriz já está aplicada a um intervalo de células. Limpe o intervalo de células existente primeiro.";
        this["LE_InvalidOperationException_ArrayFormulaAppliedInTable"] = "As fórmulas de matriz não podem ser aplicadas a uma área em que uma tabela está presente.";
        this["LE_InvalidOperationException_ArrayFormulaInMergedCell"] = "As fórmulas de matriz não são válidas em células unidas.";
        this["LE_InvalidOperationException_BottomRightAnchorFromOtherWorksheet"] = "Adicionar a forma a esta coleção irá colocá-la numa folha de cálculo diferente da CélulaCantoDireitoInferior.";
        this["LE_InvalidOperationException_BuiltInStyleNameCannotBeChanged"] = "O nome de um estilo incorporado não pode ser alterado.";
        this["LE_InvalidOperationException_CannotAddColorsToPaletteDirectly"] = "Não é possível adicionar cores ao PaletaDeCoresDoLivro. Utilize o indexador para definir uma cor num índice específico.";
        this["LE_InvalidOperationException_CannotAddParagraphDirectly"] = "Os parágrafos não podem ser adicionados diretamente.";
        this["LE_InvalidOperationException_CannotAddTableDirectly"] = "As instâncias de TabelaDeFolhaDeCálculo não podem ser inseridas diretamente na coleção. Use o método Adicionar.";
        this["LE_InvalidOperationException_CannotAddTableToRemovedWorksheet"] = "As tabelas não podem ser adicionadas a uma folha de cálculo que foi removida do seu próprio livro.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileHeaderRowIsHidden"] = "Um filtro não pode ser aplicado enquanto a linha do cabeçalho da tabela estiver oculta.";
        this["LE_InvalidOperationException_CannotApplyFilterWhileUIIsHidden"] = "Não é possível aplicar um filtro enquanto a IU do filtro está oculta.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfData"] = "{0} não pode ser inserido porque deslocaria os dados para o fundo da folha de cálculo.";
        this["LE_InvalidOperationException_CannotInsertTableRow_LossOfObject"] = "{0} não pode ser inserido porque deslocaria uma forma ou comentaria na parte inferior da folha de cálculo.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitBlockingValue"] = "{0} não pode ser inserido porque deslocaria as células numa fórmula de matriz ou tabela de dados.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitMergedRegion"] = "{0} não pode ser inserido porque deslocaria as células numa região unida.";
        this["LE_InvalidOperationException_CannotInsertTableRow_SplitTable"] = "{0} não pode ser inserido porque deslocaria as células noutra tabela.";
        this["LE_InvalidOperationException_CannotInsertTableRow_TableOnBottomOfWorksheet"] = "{0} não pode ser inserido porque a tabela está na parte inferior da folha de cálculo.";
        this["LE_InvalidOperationException_CannotModifyKeysCollection"] = "A coleção Chaves não pode ser modificada.";
        this["LE_InvalidOperationException_CannotModifyStandardTableStyle"] = "Os estilos predefinidos da tabela não podem ser modificados.";
        this["LE_InvalidOperationException_CannotModifyValuesCollection"] = "A coleção Valores não pode ser modificada.";
        this["LE_InvalidOperationException_CannotMoveDisconnectedWorksheet"] = "A folha de cálculo não pode ser movida porque não pertence a um livro.";
        this["LE_InvalidOperationException_CannotRemoveColorsToPaletteDirectly"] = "As cores não podem ser removidas da PaletaDeCoresDoLivro. Use o método Redefinir para limpar todas as cores personalizadas.";
        this["LE_InvalidOperationException_CannotRemoveNormalStyle"] = "O estilo Normal não pode ser removido.";
        this["LE_InvalidOperationException_CannotSetParentStyleOnStyle"] = "O estilo não pode ser definido num formato que pertença a outro estilo.";
        this["LE_InvalidOperationException_CannotShift_LossOfData"] = "Não é possível deslocar os dados para fora da folha.";
        this["LE_InvalidOperationException_CannotShift_LossOfObject"] = "Não é possível deslocar objetos para fora da folha.";
        this["LE_InvalidOperationException_CannotShift_SplitBlockingValue"] = "Não é possível deslocar numa fórmula de matriz ou tabela de dados.";
        this["LE_InvalidOperationException_CannotShift_SplitMergedRegion"] = "Não é possível deslocar numa região unida.";
        this["LE_InvalidOperationException_CannotShift_SplitTable"] = "Não é possível deslocar numa tabela.";
        this["LE_InvalidOperationException_CantAddChartInChart"] = "Um gráfico não pode ser adicionado à coleção Formas de outro gráfico.";
        this["LE_InvalidOperationException_CantAddCustomView"] = "As vistas personalizadas não podem ser adicionadas diretamente à coleção.";
        this["LE_InvalidOperationException_CantAddDataTable"] = "As tabelas de dados não podem ser adicionadas diretamente.";
        this["LE_InvalidOperationException_CantAddMergedRegion"] = "As regiões de células unidas não podem ser adicionadas diretamente.";
        this["LE_InvalidOperationException_CantAddNamedReference"] = "As referências com nome não podem ser adicionadas diretamente.";
        this["LE_InvalidOperationException_CantAddWorksheet"] = "Não é possível adicionar folhas de cálculo diretamente a esta coleção.";
        this["LE_InvalidOperationException_CantApplyRemovedCustomView"] = "A vista personalizada não pode ser aplicada se tiver sido removida da coleção de vistas personalizadas do livro.";
        this["LE_InvalidOperationException_CantChangeArrayFormula"] = "As células de uma fórmula de matriz não podem ser alteradas.";
        this["LE_InvalidOperationException_CantChangeDataTable"] = "As células internas de uma tabela de dados não podem ser alteradas.";
        this["LE_InvalidOperationException_CantModifyCollection"] = "A coleção não pode ser modificada.";
        this["LE_InvalidOperationException_CantOverlapArrayFormula"] = "Uma região de células unidas não pode sobrepor-se a uma região de fórmula de matriz.";
        this["LE_InvalidOperationException_CantOverlapDataTableInterior"] = "Uma região de células unidas não pode sobrepor as células numa tabela de dados se ela se estender para fora da linha superior ou da coluna mais à esquerda dessas células.";
        this["LE_InvalidOperationException_CantSaveEditRangeWithoutRange"] = "Os IntervalosDeEdiçãoPermitidos das folhas de cálculo devem ter pelo menos um intervalo.";
        this["LE_InvalidOperationException_CantSaveWithNoVisibleWorksheets"] = "Pelo menos uma folha de cálculo no livro deve estar visível.";
        this["LE_InvalidOperationException_CantSaveWithNoWorksheets"] = "Um livro sem folhas de cálculo não pode ser guardado";
        this["LE_InvalidOperationException_CantSetCircularityErrorDirectly"] = "O valor do erro de circularidade não pode ser aplicado diretamente às células.";
        this["LE_InvalidOperationException_CantSetDataTableDirectly"] = "As tabelas de dados não podem ser definidas como o valor numa célula.";
        this["LE_InvalidOperationException_CantSetFormulaDirectly"] = "As fórmulas não podem ser definidas diretamente numa célula; utilize o método AplicarFórmula ou um dos métodos AplicarA da instância da fórmula.";
        this["LE_InvalidOperationException_CellsInTableMinSize"] = "A região de células numa tabela de dados deve ter pelo menos duas células em cada dimensão.";
        this["LE_InvalidOperationException_ChartNotSupported"] = "Não há suporte para modificações no GráficoDeFolhaDeCálculo neste formato.";
        this["LE_InvalidOperationException_ClearUnknownShapeData"] = "Todos os dados para esta forma são desconhecidos; remova-a da sua coleção.";
        this["LE_InvalidOperationException_CollectionLongerThanMaxValue"] = "A coleção contém mais entradas do que o valor máximo especificado permite.";
        this["LE_InvalidOperationException_CollectionModifiedWhileEnumerating"] = "A coleção foi modificada enquanto estava a ser enumerada.";
        this["LE_InvalidOperationException_CurrentTableRowAddressNeedsOriginCell"] = "Para analisar uma referência de tabela estruturada usando a linha atual, uma célula de origem deve ser especificada.";
        this["LE_InvalidOperationException_CustomViewNoVisibleWorksheets"] = "Pelo menos uma folha de cálculo na vista personalizada denominada \"{0}\" deve estar visível.";
        this["LE_InvalidOperationException_DataTableAppliedInTable"] = "Uma tabela de dados não pode estar numa área em que uma tabela está presente.";
        this["LE_InvalidOperationException_DataTableFormulaCannotBeApplied"] = "Uma fórmula da tabela de dados não pode ser aplicada diretamente às células.";
        this["LE_InvalidOperationException_DataTableInMergedCell"] = "As células internas de uma tabela de dados não podem fazer parte de uma região de célula unida.";
        this["LE_InvalidOperationException_DataTableRemoved"] = "As células de entrada de linha e coluna não podem ser definidas após a tabela de dados ter sido removida da folha de cálculo.";
        this["LE_InvalidOperationException_DeprecatedFillColorSetWithGradientFill"] = "Uma cor de padrão de preenchimento não pode ser definida enquanto um gradiente estiver definido na célula.";
        this["LE_InvalidOperationException_EncryptedWorkbooksNotSupported"] = "O livro a ser carregado está encriptado. Os livros encriptados não são suportados.";
        this["LE_InvalidOperationException_ExtensionDoesntMatchCurrentFormat"] = "O nome do ficheiro no qual guardar o ficheiro tem uma extensão que não corresponde ao formato atual do livro.";
        this["LE_InvalidOperationException_FixedValuesFilterMustAcceptAValue"] = "Não pode ter um FiltroDeValoresFixado que não permita valores.";
        this["LE_InvalidOperationException_FormattedStringAlreadyOwned"] = "A cadeia formatada já tem o valor de outra célula ou comentário. Clone-a primeiro e aplique o clone à nova célula ou forma.";
        this["LE_InvalidOperationException_FormattedStringNotOwned"] = "As propriedades do tipo de letra não podem ser acedidas enquanto a cadeia ou texto formatado não é aplicado a uma célula ou forma.";
        this["LE_InvalidOperationException_FormattedTextAlreadyOwned"] = "A cadeia formatada já é o valor de outra forma. Clone-a primeira e aplique o clone à nova célula ou forma.";
        this["LE_InvalidOperationException_FormulaAlreadyOwned"] = "A fórmula já está aplicada a outra célula.";
        this["LE_InvalidOperationException_FormulaReferencesInvalidCells"] = "A fórmula faz referência a células que não são suportadas no formato de ficheiro atual do livro.";
        this["LE_InvalidOperationException_GetBoundsBeforeAnchorsSet"] = "A forma não tem limites antes das células âncoras serem definidas.";
        this["LE_InvalidOperationException_HiddenWorksheetCannotBeSelected"] = "Uma folha de cálculo oculta não pode ser selecionada.";
        this["LE_InvalidOperationException_HyperlinkSealed"] = "A hiperligação não pode ser modificada.";
        this["LE_InvalidOperationException_ImageDisposed"] = "Uma imagem de uma ImagemDeFolhaDeCálculo foi eliminada ou é inválida.";
        this["LE_InvalidOperationException_InputCellsBothNull"] = "As células de entrada da coluna e de entrada da linha não podem ser nulas.";
        this["LE_InvalidOperationException_InputCellsInTable"] = "As células de entrada não podem estar dentro das células da tabela de dados.";
        this["LE_InvalidOperationException_InputCellsSame"] = "As células de entrada da coluna e de entrada da linha não podem ser a mesma célula.";
        this["LE_InvalidOperationException_InvalidCharacterRange"] = "O intervalo de caracteres abrangidos pelo tipo de letra já não existe na cadeia formatada.";
        this["LE_InvalidOperationException_InvalidFormatOptionsInWholeTableArea"] = "O valor do formato OpçõesDeFormato especificado é inválido na área da tabela TabelaInteira.";
        this["LE_InvalidOperationException_InvalidFormatString"] = "A cadeia de formato especificada é inválida.";
        this["LE_InvalidOperationException_InvalidFormatString_GetTextCall"] = "A cadeia de formato da célula é inválida. Para ser notificado quando a CadeiaDeFormato estiver definida, defina ValidarCadeiasDeFormato.Livro como Verdadeiro.";
        this["LE_InvalidOperationException_InvalidForWorkbookFormat"] = "A fórmula não é válida para o formato especificado.";
        this["LE_InvalidOperationException_InvalidHeaderCellColumnAreaFormatProperty"] = "Os valores das margens superiores não podem ser definidos no formato da área da coluna da tabela CélulaDoCabeçalho.";
        this["LE_InvalidOperationException_InvalidHeaderRowAreaFormatProperty"] = "Os valores da margem superior não podem ser definidos no formato da área da tabela LinhaDeCabeçalho.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFontProperty"] = "O valor do tipo de letra {0} não pode ser definido nas áreas de estilo de tabela.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatOptions"] = "O valor do formato OpçõesDeFormato especificado é inválido nas áreas de estilo de tabela.";
        this["LE_InvalidOperationException_InvalidTableStyleAreaFormatProperty"] = "O valor do formato {0} não pode ser definido nas áreas de estilo de tabela.";
        this["LE_InvalidOperationException_InvalidTotalsCellColumnAreaFormatProperty"] = "Os valores da margem inferior não podem ser definidos no formato da área da coluna da tabela CélulaTotal.";
        this["LE_InvalidOperationException_InvalidTotalsRowAreaFormatProperty"] = "Os valores da margem inferior não podem ser definidos no formato da área da tabela LinhaDeTotais.";
        this["LE_InvalidOperationException_InvalidWholeTableAreaFormatProperty"] = "Apenas os valores das margens podem ser definidos no formato da área da tabela TabelaInteira.";
        this["LE_InvalidOperationException_LastColumnBeforeFirst"] = "A última coluna na região deve ser maior ou igual à primeira coluna.";
        this["LE_InvalidOperationException_LastRowBeforeFirst"] = "A última linha na região deve ser maior que a primeira linha.";
        this["LE_InvalidOperationException_MaxCellFormats"] = "O número máximo de formatos de células distintos no ficheiro Excel é: {0}";
        this["LE_InvalidOperationException_MaxColors"] = "O número máximo de cores num livro do Excel é: {0}";
        this["LE_InvalidOperationException_MaxColumns"] = "O número máximo de colunas numa folha de cálculo do Excel com o formato atual é: {0}";
        this["LE_InvalidOperationException_MaxFonts"] = "O número de tipos de letra num livro não pode exceder {0}.";
        this["LE_InvalidOperationException_MaxFormattedStrings"] = "O número de cadeias únicas num livro não pode exceder {0}.";
        this["LE_InvalidOperationException_MaxRows"] = "O número máximo de linhas numa folha de cálculo do Excel com o formato atual é: {0}";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableLeftColumn"] = "A coluna de células mais à esquerda especificada para a tabela de dados contém uma região de células unidas que se estende para fora da área dessas células.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopLeftCell"] = "A célula superior esquerda das células especificadas para a tabela de dados faz parte de uma célula unida que se estende para fora da área dessas células.";
        this["LE_InvalidOperationException_MergedCellCrossesDataTableTopRow"] = "A linha superior de células especificada para a tabela de dados contém uma região de células unidas que se estende para fora da área dessas células.";
        this["LE_InvalidOperationException_MergedCellsAppliedInTable"] = "As células numa tabela não podem ser unidas.";
        this["LE_InvalidOperationException_NamedReferenceNameAlreadyExists"] = "O nome \"{0}\" é utilizado por uma tabela ou referência com nome no livro com o mesmo âmbito.";
        this["LE_InvalidOperationException_NoRegionHasBeenSpecified"] = "Nenhuma região foi especificada.";
        this["LE_InvalidOperationException_NoWorksheetContextToSetShapeBounds"] = "Esta sobrecarga de Adicionar não pode ser chamada quando a coleção de formas não está associada a uma folha de cálculo. Chame a sobrecarga de Adicionar que recebe uma instância da folha de cálculo.";
        this["LE_InvalidOperationException_OverlappingTable"] = "Há outra tabela na região especificada.";
        this["LE_InvalidOperationException_ReadOnlyFont"] = "O objeto do tipo de letra é só de leitura e não pode ser alterado.";
        this["LE_InvalidOperationException_ReadOnlyFormat"] = "O objeto de formatação é só de leitura e não pode ser alterado.";
        this["LE_InvalidOperationException_RelativeR1C1AddressNeedsOriginCell"] = "Para analisar um endereço R1C1 relativo, deve ser especificada uma célula de origem.";
        this["LE_InvalidOperationException_ResizedTableMustBeOnWorksheet"] = "A tabela deve estar numa folha de cálculo para ser redimensionada.";
        this["LE_InvalidOperationException_ResolvedFormatCannotBeModified"] = "O formato da célula resolvido é só de leitura e não pode ser alterado.";
        this["LE_InvalidOperationException_ShapeCantChangeOrientation"] = "Esta forma não pode ser rodada ou invertida.";
        this["LE_InvalidOperationException_ShapeFillSolidIsImmutable"] = "O SólidoDePreenchimentoDeForma é imutável. Crie um novo SólidoDePreenchimentoDeForma.";
        this["LE_InvalidOperationException_ShapeInAnotherCollection"] = "A forma já existe noutra folha de cálculo ou grupo de formas.";
        this["LE_InvalidOperationException_ShapeOutlineSolidIsImmutable"] = "O SólidoDeContornoDaForma é imutável. Crie um novo SólidoDeContornoDaForma.";
        this["LE_InvalidOperationException_ShowFilterUIWhileHeaderRowHidden"] = "A interface de utilizador do filtro não pode ser exibida enquanto a linha do cabeçalho estiver oculta.";
        this["LE_InvalidOperationException_SystemColorsAccessedFromWrongThread"] = "Não é possível aceder a CoresDoSistema a partir de um thread em segundo plano.";
        this["LE_InvalidOperationException_TableReferenceMustBeUsedFromInsideTable"] = "A fórmula tem uma referência de tabela estruturada que só pode ser utilizada a partir de dentro de uma tabela.";
        this["LE_InvalidOperationException_TableReferenceToMissingTable"] = "A fórmula tem uma referência a uma tabela que não existe.";
        this["LE_InvalidOperationException_TableReferenceToMissingTableColumn"] = "A fórmula tem uma referência a uma coluna da tabela que não existe.";
        this["LE_InvalidOperationException_TopLeftAnchorFromOtherWorksheet"] = "Adicionar a forma a esta coleção irá colocá-la numa folha de cálculo diferente da sua CélulaCantoEsquerdoSuperior.";
        this["LE_InvalidOperationException_TotalsRowCannotBeShownInLastRow"] = "A linha de totais não pode ser mostrada quando a tabela ocupa a última linha na folha de cálculo.";
        this["LE_InvalidOperationException_UnknownBIFFFormat"] = "O ficheiro do livro estava num formato BIFF desconhecido.";
        this["LE_InvalidOperationException_UnsupportedBIFFFormat"] = "O ficheiro do livro foi guardado em {0}, um formato não suportado.";
        this["LE_InvalidOperationException_WorkbookDataViolatesFormatLimits"] = "A pasta de trabalho contém dados que excederiam os limites impostos pelo formato especificado.";
        this["LE_InvalidRegion"] = "Endereço de região inválido.";
        this["LE_InvalidValueType"] = "'{0}' não é um valor válido para a propriedade TipoDeValor.";
        this["LE_LocationNotSingleRowOrColumn"] = "A referência de localização não é válida porque as células não estão todas na mesma coluna ou linha. Selecione as células que estão todas numa única coluna ou linha.";
        this["LE_LocationOrDataAreaInvalid"] = "A referência para a localização ou intervalo de dados não é válida.";
        this["LE_LocationWorksheetMismatch"] = "O local deve ser desta folha de cálculo.";
        this["LE_NotSingleTargetFormula"] = "Só pode suportar fórmulas de destino único.";
        this["LE_NotSupportedException_CellType"] = "{0} não é um tipo de valor de célula suportado.";
        this["LE_NotSupportedException_NoPackageFactory"] = "packageFactory não pode ser nulo. Ao carregar a partir do formato de livro do Excel2007, e ao utilizar a assemblagem Infragistics2.Documents.Excel, deve fornecer um IPackageFactory para processar o pacote de dados. Se estiver a utilizar o DotNet Framework 3.0 ou superior, utilize a assemblagem Infragistics3.Documents.Excel e o pacote será processado pela classe WindowsBase.";
        this["LE_OpenPackagingNonConformanceException_ContainsDublinCoreRefinements"] = "O IPackagePart é a relação principal das propriedades e contém refinamentos para os elementos Dublin Core que não 'dcterms: created' e 'dcterms: modified'.";
        this["LE_OpenPackagingNonConformanceException_ContainsXmlLanguageAttribute"] = "O IPackagePart é a relação principal das propriedades e contém elementos que contêm o atributo 'xml: lang'.";
        this["LE_OpenPackagingNonConformanceException_CorePropertiesRelationshipAlreadyProcessed"] = "O IPackagePart é a relação principal das propriedades, mas uma já foi processada para o IPackage associado.";
        this["LE_OpenPackagingNonConformanceException_DuplicatePartName"] = "Um IPackagePart com o mesmo nome já pertence a este IPackage. Um IPackage não pode conter partes com nomes idênticos. Tenha em consideração que a verificação de valores duplicados não diferencia maiúsculas de minúsculas.";
        this["LE_OpenPackagingNonConformanceException_UsesMarkupCompatibilityNamespace"] = "O IPackagePart é a relação principal das propriedades e utiliza o Espaço de nomes de Compatibilidade de Marcação.";
        this["LE_OpenPackagingNonConformanceException_XmlContainsDocumentTypeDefinition"] = "Uma IPackagePart continha uma declaração DTD (Document Type Definition). As regras da Open Packaging Convention proíbem o uso de conteúdo XML com declarações DTD.";
        this["LE_OpenPackagingNonConformanceException_XsiTypeAttributeInvalid"] = "O IPackagePart é a relação principal das propriedades e contém elementos que contêm o atributo 'xsi: type' ou contêm elementos que são necessários e não contêm o atributo 'xsi: type'.";
        this["LE_OpenXmlSerializerNotLoaded"] = "O formato '{0}' requer um serializador que não esteja carregado. No JavaScript, este está contido no ficheiro infragistics.excel_serialization_openxml.js";
        this["LE_PercentageValueOutOfRange"] = "O valor percentual {0} deve estar entre 0 e 100.";
        this["LE_PercentileValueOutOfRange"] = "O valor de percentil {0} deve estar entre 0 e 100.";
        this["LE_PercentRangeInvalid"] = "O {0} deve ser menor que o {1}.";
        this["LE_RowsToRepeatAtTopAreOutsideAvailableRange"] = "As opções de impressão têm linhas que se repetem na parte superior e estão fora do intervalo de linhas disponíveis no novo formato.";
        this["LE_Serialization_NoPrimaryCategoryAxis"] = "O tipo de gráfico '{0}' requer um eixo de categoria principal. Adicione um eixo de categoria principal à ColeçãoDeEixos do gráfico.";
        this["LE_Serialization_NoPrimaryValueAxis"] = "O tipo de gráfico '{0}' requer um eixo de valor primário. Adicione um eixo de valor primário à ColeçãoDeEixos do gráfico.";
        this["LE_Series_ChartType_ChartEx"] = "'{0}' é inválido para um TipoDeGráfico de série num gráfico '{1}'.";
        this["LE_Series_ChartType_InvalidForComboChart"] = "O tipo de gráfico '{0}' não é válido para utilização num gráfico combinado.";
        this["LE_Series_ChartType_NonCombo"] = "O tipo de gráfico para esta série é determinado pelo tipo de gráfico principal ('{0}') e não pode ser definido. Defina a propriedade TipoDeGráfico do gráfico principal para '{1}'.";
        this["LE_Series_Explosion_OutOfRange"] = "A propriedade Explosão suporta valores no intervalo de 0 a 400, inclusive.";
        this["LE_Series_InvalidAxisGroup"] = "A propriedade GrupoDeEixos não pode ser definida como '{0}' porque existe uma série do tipo '{1}' no mesmo eixo, que não pode partilhar o mesmo eixo que uma série do tipo '{2}'.";
        this["LE_Series_InvalidSeriesTypeForComboChart"] = "Uma série do tipo '{0}' não pode ser utilizada num gráfico combinado.";
        this["LE_Series_NoPrimaryAxis"] = "Pelo menos uma série deve ser desenhada no eixo primário. Altere a propriedade GrupoDeEixo para 'Primário' por, pelo menos, uma série.";
        this["LE_Series_OwningSeries_NotClusteredColumn"] = "A propriedade SérieProprietária pode ser definida apenas para uma série de colunas agrupadas num gráfico Pareto.";
        this["LE_Series_OwningSeries_NotParetoLine"] = "A propriedade SérieProprietária pode ser definida apenas para uma série de linhas num gráfico Pareto.";
        this["LE_Series_OwningSeries_Same"] = "A SérieProprietária não pode fazer referência a si própria.";
        this["LE_Series_SeriesType_ChartEx"] = "O valor TipoDeSérie '{0}' é inválido para um gráfico '{1}'.";
        this["LE_Series_SeriesType_NonCombo"] = "A propriedade Tipo é aplicável apenas a gráficos combinados e não pode ser definida para um gráfico do tipo '{0}'.";
        this["LE_SetValue_Formula"] = "O tipo de valor '{0}' só é válido para fórmulas. Utilize o método DefinirFórmula.";
        this["LE_SetValue_NonNumeric"] = "O parâmetro de valor não é aplicável ao tipo de valor '{0}'. Utilize a sobrecarga que aceita apenas um tipo de valor.";
        this["LE_SetValue_Numeric"] = "O parâmetro de valor deve ser especificado para o tipo de valor '{0}'. Utilize a sobrecarga que aceita o valor e o tipo de valor.";
        this["LE_WorksheetChart_ChartType_ChartEx_CannotTransition"] = "A propriedade TipoDeGráfico não pode ser alterada de '{0}' para '{1}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_From"] = "A propriedade TipoDeGráfico de um gráfico combinado não pode ser alterada para '{0}'.";
        this["LE_WorksheetChart_ChartType_ComboChart_To"] = "Um gráfico do tipo '{0}' não pode ser alterado para 'Combinado'. É possível criar um novo gráfico combinado utilizando o método AdicionarGráfico, mas não é possível alterar um gráfico existente para um gráfico combinado.";
        this["LE_WorksheetChart_ComboChart_NotEnoughSeries"] = "Um gráfico combinado requer pelo menos 2 séries. Adicione mais itens à ColeçãoDeSéries do gráfico.";
        this["LE_WorksheetChart_MissingRequiredAxis_StockCharts"] = "Um gráfico '{0}' requer um eixo de categoria primário, um eixo de valor primário, um eixo de categoria secundário e um eixo de valor secundário.";
        this["LE_WorksheetChart_NotEnoughSeries"] = "Um gráfico '{0}' requer um mínimo de {1} séries. Para usar este tipo de gráfico, deve adicionar mais itens à ColeçãoDeSéries.";
        this["LE_WorksheetChart_Save_Series"] = "Um gráfico '{0}' deve conter pelo menos uma série. Adicione mais séries ao gráfico antes de tentar guardar o livro.";
        this["LE_WorksheetChart_Save_Series_Pareto"] = "Um gráfico 'Pareto' deve conter uma série 'ColunasAgrupadas' e uma série 'Pareto'. Adicione a série apropriada ao gráfico antes de tentar guardar o livro.";
        this["LE_WorksheetChart_SetComboChartSourceData_InvalidChartTypes"] = "Um ou mais tipos de gráfico especificados não são válidos para utilização num gráfico combinado: {0}";
        this["LE_WorksheetChart_SetComboChartSourceData_NotAComboChart"] = "DefinirDadosDeOrigemDoGráficoDeCombinação não pode ser chamado porque o tipo de gráfico atual é '{0}'. Altere o TipoDeGráfico para 'Combinado'.";
        this["LE_WorksheetChart_SetComboChartSourceData_SeriesChartTypesNotSupported"] = "O parâmetro tipoDeGráficoDeSérie deve ser nulo para um gráfico '{0}'.";
        this["LE_WorksheetChart_SetSourceData_NotEnoughSeries"] = "Os gráficos combinados requerem um mínimo de 2 séries. Altere o parâmetro 'intervaloDeDados' para um intervalo que resulte em pelo menos 2 séries.";
        this["LE_WorksheetChart_SetSourceData_TooManyAxes"] = "A combinação especificada de tipos de gráfico exige mais do que o número máximo permitido de eixos.";
        this["LE_WorksheetChart_StockChartNotSupported"] = "A edição do tipo de gráfico '{0}' não é atualmente suportada.";
        this["LE_WorksheetChart_UnknownChartNotSupported"] = "Este gráfico foi carregado a partir de uma folha de cálculo existente, mas não é possível determinar o TipoDeGráfico. A edição de tipos de gráficos desconhecidos não é atualmente suportada.";
        this["LER_ArgumentOutOfRangeException_OutlineLevel"] = "O nível de destaque deve estar entre 0 e 7.";
        this["LER_Exception_KeyNotFound"] = "Chave não encontrada.";
        this["LER_Exception_MergedRegionsOverlap"] = "As regiões das células unidas não podem sobrepor-se.";
        this["TableHeaderRowDescription"] = "linha do cabeçalho";
        this["TableInsertRowDescription"] = "inserir linha";
        this["TableTotalsRowDescription"] = "linha de totais";
        this["Value_UCErrorCode_Div"] = "#DIV/0!";
        this["Value_UCErrorCode_Fail"] = "#FAIL";
        this["Value_UCErrorCode_NA"] = "#N/A";
        this["Value_UCErrorCode_Name"] = "#NAME?";
        this["Value_UCErrorCode_Null"] = "#NULL!";
        this["Value_UCErrorCode_Num"] = "#NUM!";
        this["Value_UCErrorCode_Ok"] = "#OK";
        this["Value_UCErrorCode_Reference"] = "#REF!";
        this["Value_UCErrorCode_Unknown"] = "#N/A";
        this["Value_UCErrorCode_Value"] = "#VALUE!";
        this["WorkbookColorInfo_Automatic_Description"] = "Automático";
        this["WorkbookColorInfo_Description"] = "InformaçãoDeCor: {0}";
        this["WorkbookColorInfo_WithTint_Description"] = "InformaçãoDeCor: {0}, Tom: {1}";
        this["WorksheetShapeSerialization_GroupName"] = "Grupo";
        this["WorksheetShapeSerialization_ImageName"] = "Imagem";
        this["WorksheetShapeSerialization_ShapeName"] = "Forma";
    }
    return ExcelLocalePt;
}());
export { ExcelLocalePt };
