// igniteui-react-excel/esm5/lib/ExternalEventWorkItem$1.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WorkItem$1 } from "./WorkItem$1";
import { markType } from "igniteui-react-core";
import { WorkItem_ExceptionInfo } from "./WorkItem_ExceptionInfo";
/**
 * @hidden
 */
var ExternalEventWorkItem$1 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ExternalEventWorkItem$1, _super);
    function ExternalEventWorkItem$1($t) {
        var _this = _super.call(this, $t, 2, null, false) || this;
        _this.$t = null;
        _this.af = false;
        _this.ae = null;
        _this.$t = $t;
        _this.$type = _this.$type.specialize(_this.$t);
        return _this;
    }
    ExternalEventWorkItem$1.prototype.p = function (a) {
        if (this.af) {
            return true;
        }
        this.ae = a;
        return false;
    };
    ExternalEventWorkItem$1.prototype.get_t = function () {
        return this.af == false;
    };
    ExternalEventWorkItem$1.prototype.ag = function (a, b) {
        if (b === void 0) {
            b = null;
        }
        this.ad = a;
        if (b != null) {
            this.aa(new WorkItem_ExceptionInfo(b));
        }
        this.af = true;
        if (this.ae != null) {
            this.ae.e(this);
        }
    };
    ExternalEventWorkItem$1.$t = markType(ExternalEventWorkItem$1, 'ExternalEventWorkItem$1', WorkItem$1.$.specialize(0));
    return ExternalEventWorkItem$1;
}(WorkItem$1));
export { ExternalEventWorkItem$1 };
