// igniteui-react-excel/esm5/lib/ExternalEventWorkItem.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WorkItem } from "./WorkItem";
import { markType } from "igniteui-react-core";
import { WorkItem_ExceptionInfo } from "./WorkItem_ExceptionInfo";
/**
 * @hidden
 */
var ExternalEventWorkItem = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ExternalEventWorkItem, _super);
    function ExternalEventWorkItem() {
        var _this = _super.call(this, 2, null, false) || this;
        _this.ae = false;
        _this.ad = null;
        return _this;
    }
    ExternalEventWorkItem.prototype.p = function (a) {
        if (this.ae) {
            return true;
        }
        this.ad = a;
        return false;
    };
    ExternalEventWorkItem.prototype.get_t = function () {
        return this.ae == false;
    };
    ExternalEventWorkItem.prototype.af = function (a) {
        if (a === void 0) {
            a = null;
        }
        if (a != null) {
            this.aa(new WorkItem_ExceptionInfo(a));
        }
        this.ae = true;
        if (this.ad != null) {
            this.ad.e(this);
        }
    };
    ExternalEventWorkItem.$t = markType(ExternalEventWorkItem, 'ExternalEventWorkItem', WorkItem.$);
    return ExternalEventWorkItem;
}(WorkItem));
export { ExternalEventWorkItem };
