// igniteui-react-excel/esm5/lib/FormatConditionTimePeriod.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which define the manner in which a cell's date/time value is compared
 * when evaluating a [[DateTimeConditionalFormat]].
 */
export var FormatConditionTimePeriod = /*@__PURE__*/ (function (FormatConditionTimePeriod) {
    /**
     * The cell's date/time value is compared to the current date.
     */
    FormatConditionTimePeriod[FormatConditionTimePeriod["Today"] = 0] = "Today";
    /**
     * The cell's date/time value is compared to the day after the current date.
     */
    FormatConditionTimePeriod[FormatConditionTimePeriod["Tomorrow"] = 1] = "Tomorrow";
    /**
     * The cell's date/time value is compared to the day before the current date.
     */
    FormatConditionTimePeriod[FormatConditionTimePeriod["Yesterday"] = 2] = "Yesterday";
    /**
     * The cell's date/time value is compared to the current week, as defined by the
     * [[CalendarWeekRule]].
     */
    FormatConditionTimePeriod[FormatConditionTimePeriod["ThisWeek"] = 3] = "ThisWeek";
    /**
     * The cell's date/time value is compared to the week after the current one, as defined by the
     * [[CalendarWeekRule]].
     */
    FormatConditionTimePeriod[FormatConditionTimePeriod["NextWeek"] = 4] = "NextWeek";
    /**
     * The cell's date/time value is compared to the week before the current one, as defined by the
     * [[CalendarWeekRule]].
     */
    FormatConditionTimePeriod[FormatConditionTimePeriod["LastWeek"] = 5] = "LastWeek";
    /**
     * The cell's date/time value meets the condition if it lies within the range
     * of the past seven days, without regard to the
     * [[CalendarWeekRule]].
     */
    FormatConditionTimePeriod[FormatConditionTimePeriod["LastSevenDays"] = 6] = "LastSevenDays";
    /**
     * The cell's date/time value is compared to the current month.
     */
    FormatConditionTimePeriod[FormatConditionTimePeriod["ThisMonth"] = 7] = "ThisMonth";
    /**
     * The cell's date/time value is compared to the month after the current one.
     */
    FormatConditionTimePeriod[FormatConditionTimePeriod["NextMonth"] = 8] = "NextMonth";
    /**
     * The cell's date/time value is compared to the month before the current one.
     */
    FormatConditionTimePeriod[FormatConditionTimePeriod["LastMonth"] = 9] = "LastMonth";
    return FormatConditionTimePeriod;
})({});
/**
 * @hidden
 */
export var FormatConditionTimePeriod_$type = /*@__PURE__*/ markEnum('FormatConditionTimePeriod', 'Today,0|Tomorrow,1|Yesterday,2|ThisWeek,3|NextWeek,4|LastWeek,5|LastSevenDays,6|ThisMonth,7|NextMonth,8|LastMonth,9');
