// igniteui-react-excel/esm5/lib/FormatConditionValueType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which identify the type of value which defines the criteria for a
 * [[ColorScaleCriterion]] or [[IconCriterion]].
 */
export var FormatConditionValueType = /*@__PURE__*/ (function (FormatConditionValueType) {
    /**
     * The criteria is defined by a numeric value.
     */
    FormatConditionValueType[FormatConditionValueType["Number"] = 0] = "Number";
    /**
     * The criteria is defined by a percentage.
     */
    FormatConditionValueType[FormatConditionValueType["Percentage"] = 1] = "Percentage";
    /**
     * The criteria is defined by a percentile rating.
     */
    FormatConditionValueType[FormatConditionValueType["Percentile"] = 2] = "Percentile";
    /**
     * The criteria is defined by a formula.
     */
    FormatConditionValueType[FormatConditionValueType["Formula"] = 3] = "Formula";
    /**
     * The criteria is defined by the lowest value across the associated range.
     */
    FormatConditionValueType[FormatConditionValueType["LowestValue"] = 4] = "LowestValue";
    /**
     * The criteria is defined by the highest value across the associated range.
     */
    FormatConditionValueType[FormatConditionValueType["HighestValue"] = 5] = "HighestValue";
    /**
     * The shortest data bar is proportional to the lowest value across the associated range.
     * Applicable only to the [[IconCriterion]] class.
     */
    FormatConditionValueType[FormatConditionValueType["AutomaticMinimum"] = 6] = "AutomaticMinimum";
    /**
     * The longest data bar is proportional to the highest value across the associated range.
     * Applicable only to the [[IconCriterion]] class.
     */
    FormatConditionValueType[FormatConditionValueType["AutomaticMaximum"] = 7] = "AutomaticMaximum";
    return FormatConditionValueType;
})({});
/**
 * @hidden
 */
export var FormatConditionValueType_$type = /*@__PURE__*/ markEnum('FormatConditionValueType', 'Number,0|Percentage,1|Percentile,2|Formula,3|LowestValue,4|HighestValue,5|AutomaticMinimum,6|AutomaticMaximum,7');
