// igniteui-react-excel/esm5/lib/FormatConditionAboveBelow.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which specify the manner in which cell values are compared
 * when evaluating a [[AverageConditionalFormat]].
 */
export var FormatConditionAboveBelow = /*@__PURE__*/ (function (FormatConditionAboveBelow) {
    /**
     * The cell value must be above the average for the associated range
     * in order to meet the associated condition.
     */
    FormatConditionAboveBelow[FormatConditionAboveBelow["AboveAverage"] = 0] = "AboveAverage";
    /**
     * The cell value must be below the average for the associated range
     * in order to meet the associated condition.
     */
    FormatConditionAboveBelow[FormatConditionAboveBelow["BelowAverage"] = 1] = "BelowAverage";
    /**
     * The cell value must be equal to or above the average for the associated range
     * in order to meet the associated condition.
     */
    FormatConditionAboveBelow[FormatConditionAboveBelow["EqualAboveAverage"] = 2] = "EqualAboveAverage";
    /**
     * The cell value must be equal to or below the average for the associated range
     * in order to meet the associated condition.
     */
    FormatConditionAboveBelow[FormatConditionAboveBelow["EqualBelowAverage"] = 3] = "EqualBelowAverage";
    /**
     * The cell value must be above the standard deviation for the associated range
     * in order to meet the associated condition.
     */
    FormatConditionAboveBelow[FormatConditionAboveBelow["AboveStandardDeviation"] = 4] = "AboveStandardDeviation";
    /**
     * The cell value must be below the standard deviation for the associated range
     * in order to meet the associated condition.
     */
    FormatConditionAboveBelow[FormatConditionAboveBelow["BelowStandardDeviation"] = 5] = "BelowStandardDeviation";
    return FormatConditionAboveBelow;
})({});
/**
 * @hidden
 */
export var FormatConditionAboveBelow_$type = /*@__PURE__*/ markEnum('FormatConditionAboveBelow', 'AboveAverage,0|BelowAverage,1|EqualAboveAverage,2|EqualBelowAverage,3|AboveStandardDeviation,4|BelowStandardDeviation,5');
