// igniteui-react-excel/esm5/lib/FormatConditionTextOperator.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which specify the text comparison method applied when
 * evaluating a <see cref="ConditionalFormatBase">format condition</see>.
 */
export var FormatConditionTextOperator = /*@__PURE__*/ (function (FormatConditionTextOperator) {
    /**
     * The string being evaluated must begin with the value to which
     * it is being compared in order for the condition to be met.
     */
    FormatConditionTextOperator[FormatConditionTextOperator["BeginsWith"] = 0] = "BeginsWith";
    /**
     * The string being evaluated must end with the value to which
     * it is being compared in order for the condition to be met.
     */
    FormatConditionTextOperator[FormatConditionTextOperator["EndsWith"] = 1] = "EndsWith";
    /**
     * The string being evaluated must match some part of the value to which
     * it is being compared in order for the condition to be met.
     */
    FormatConditionTextOperator[FormatConditionTextOperator["Contains"] = 2] = "Contains";
    /**
     * Negates the [[FormatConditionTextOperator.contains]] value; the string being evaluated
     * must not match any part of the value to which it is being compared in
     * order for the condition to be met.
     */
    FormatConditionTextOperator[FormatConditionTextOperator["DoesNotContain"] = 3] = "DoesNotContain";
    return FormatConditionTextOperator;
})({});
/**
 * @hidden
 */
export var FormatConditionTextOperator_$type = /*@__PURE__*/ markEnum('FormatConditionTextOperator', 'BeginsWith,0|EndsWith,1|Contains,2|DoesNotContain,3');
