// igniteui-react-excel/esm5/lib/GifPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { ImagePartManagerBase } from "./ImagePartManagerBase";
import { ImageFormat } from "./ImageFormat";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var GifPartManager = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(GifPartManager, _super);
    function GifPartManager() {
        return _super.call(this) || this;
    }
    GifPartManager.prototype.get_s = function () {
        return "/xl/media/image.gif";
    };
    Object.defineProperty(GifPartManager.prototype, "s", {
        get: function () {
            return this.get_s();
        },
        enumerable: false,
        configurable: true
    });
    GifPartManager.prototype.get_t = function () {
        return "/ppt/media/image.gif";
    };
    Object.defineProperty(GifPartManager.prototype, "t", {
        get: function () {
            return this.get_t();
        },
        enumerable: false,
        configurable: true
    });
    GifPartManager.prototype.get_u = function () {
        return "/word/media/image.gif";
    };
    Object.defineProperty(GifPartManager.prototype, "u", {
        get: function () {
            return this.get_u();
        },
        enumerable: false,
        configurable: true
    });
    GifPartManager.prototype.get_m = function () {
        return "image/gif";
    };
    Object.defineProperty(GifPartManager.prototype, "m", {
        get: function () {
            return this.get_m();
        },
        enumerable: false,
        configurable: true
    });
    GifPartManager.prototype.r = function (a) {
        return ImageFormat.e;
    };
    GifPartManager.$t = markType(GifPartManager, 'GifPartManager', ImagePartManagerBase.$);
    GifPartManager.y = new GifPartManager();
    return GifPartManager;
}(ImagePartManagerBase));
export { GifPartManager };
