// igniteui-react-excel/esm5/lib/HorizontalCellAlignment.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Enumeration for horizontal alignment styles. Default value is used in property override situations.
 */
export var HorizontalCellAlignment = /*@__PURE__*/ (function (HorizontalCellAlignment) {
    /**
     * Use the current default.
     */
    HorizontalCellAlignment[HorizontalCellAlignment["Default"] = -1] = "Default";
    /**
     * Alignment depends on underlying data type.
     */
    HorizontalCellAlignment[HorizontalCellAlignment["General"] = 0] = "General";
    /**
     * Left alignment.
     */
    HorizontalCellAlignment[HorizontalCellAlignment["Left"] = 1] = "Left";
    /**
     * Centered alignment.
     */
    HorizontalCellAlignment[HorizontalCellAlignment["Center"] = 2] = "Center";
    /**
     * Right alignment.
     */
    HorizontalCellAlignment[HorizontalCellAlignment["Right"] = 3] = "Right";
    /**
     * Repeat cell value to fill whole cell.
     */
    HorizontalCellAlignment[HorizontalCellAlignment["Fill"] = 4] = "Fill";
    /**
     * Justify alignment.
     */
    HorizontalCellAlignment[HorizontalCellAlignment["Justify"] = 5] = "Justify";
    /**
     * Centers the contents of the left-most cell in a center across selection group. All other cells in the center across selection group must be empty. The cells are not merged, and the data may appear to be in a cell other than the left-most cell.
     */
    HorizontalCellAlignment[HorizontalCellAlignment["CenterAcrossSelection"] = 6] = "CenterAcrossSelection";
    /**
     * Distributed alignment.
     */
    HorizontalCellAlignment[HorizontalCellAlignment["Distributed"] = 7] = "Distributed";
    return HorizontalCellAlignment;
})({});
/**
 * @hidden
 */
export var HorizontalCellAlignment_$type = /*@__PURE__*/ markEnum('HorizontalCellAlignment', 'Default,-1|General,0|Left,1|Center,2|Right,3|Fill,4|Justify,5|CenterAcrossSelection,6|Distributed,7');
