// igniteui-react-excel/esm5/lib/InPlaceTransform_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
var InPlaceTransform = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(InPlaceTransform, _super);
    function InPlaceTransform() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.e = 0;
        return _this;
    }
    Object.defineProperty(InPlaceTransform, "b", {
        get: function () {
            if (InPlaceTransform.a == null) {
                InPlaceTransform.a = new InPlaceTransform_NoTransformImpl();
            }
            return InPlaceTransform.a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(InPlaceTransform.prototype, "c", {
        get: function () {
            return this.e > 0;
        },
        enumerable: false,
        configurable: true
    });
    InPlaceTransform.prototype.f = function () {
    };
    InPlaceTransform.prototype.get_d = function () {
        return false;
    };
    Object.defineProperty(InPlaceTransform.prototype, "d", {
        get: function () {
            return this.get_d();
        },
        enumerable: false,
        configurable: true
    });
    InPlaceTransform.prototype.g = function (a, b) {
        if (this.c) {
            return;
        }
        this.h(a, b);
    };
    InPlaceTransform.prototype.h = function (a, b) {
    };
    InPlaceTransform.prototype.i = function () {
        if (this.e == 0) {
            return;
        }
        this.e--;
    };
    InPlaceTransform.prototype.j = function (a) {
    };
    InPlaceTransform.prototype.k = function () {
        this.e++;
    };
    InPlaceTransform.prototype.l = function (a, b, c, d) {
        if (this.c) {
            return;
        }
        this.m(a, b, c, d);
    };
    InPlaceTransform.$t = markType(InPlaceTransform, 'InPlaceTransform');
    InPlaceTransform.a = null;
    return InPlaceTransform;
}(Base));
export { InPlaceTransform };
/**
 * @hidden
 */
var InPlaceTransform_NoTransformImpl = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(InPlaceTransform_NoTransformImpl, _super);
    function InPlaceTransform_NoTransformImpl() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    InPlaceTransform_NoTransformImpl.prototype.m = function (a, b, c, d) {
    };
    InPlaceTransform_NoTransformImpl.$t = markType(InPlaceTransform_NoTransformImpl, 'InPlaceTransform_NoTransformImpl', InPlaceTransform.$);
    return InPlaceTransform_NoTransformImpl;
}(InPlaceTransform));
export { InPlaceTransform_NoTransformImpl };
