// igniteui-react-excel/esm5/lib/InvalidPackage.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, markType } from "igniteui-react-core";
import { IPackage_$type } from "./IPackage";
import { InvalidOperationException } from "igniteui-react-core";
/**
 * @hidden
 */
var InvalidPackage = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(InvalidPackage, _super);
    function InvalidPackage() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    InvalidPackage.prototype.createPart = function (a, b) {
        throw new InvalidOperationException(0);
    };
    InvalidPackage.prototype.createRelationship = function (a, b, c, d) {
        throw new InvalidOperationException(0);
    };
    InvalidPackage.prototype.dispose = function () {
        throw new InvalidOperationException(0);
    };
    InvalidPackage.prototype.getPart = function (a) {
        throw new InvalidOperationException(0);
    };
    InvalidPackage.prototype.getParts = function () {
        throw new InvalidOperationException(0);
    };
    InvalidPackage.prototype.getRelationship = function (a) {
        throw new InvalidOperationException(0);
    };
    InvalidPackage.prototype.getRelationships = function () {
        throw new InvalidOperationException(0);
    };
    InvalidPackage.prototype.partExists = function (a) {
        throw new InvalidOperationException(0);
    };
    InvalidPackage.$t = markType(InvalidPackage, 'InvalidPackage', Base.$, [IPackage_$type]);
    return InvalidPackage;
}(Base));
export { InvalidPackage };
