// igniteui-react-excel/esm5/lib/LfuCache$2.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, Type, markType } from "igniteui-react-core";
import { ArgumentOutOfRangeException } from "igniteui-react-core";
import { Dictionary$2 } from "igniteui-react-core";
import { LfuCache_CacheNode$2 } from "./LfuCache_CacheNode$2";
import { EqualityComparer$1 } from "igniteui-react-core";
import { LinkedList } from "./LinkedList";
import { LfuCache_FrequencyNode$2 } from "./LfuCache_FrequencyNode$2";
/**
 * @hidden
 */
var LfuCache$2 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(LfuCache$2, _super);
    function LfuCache$2($tKey, $tValue, a, b) {
        var _this = _super.call(this) || this;
        _this.$tKey = null;
        _this.$tValue = null;
        _this.e = 0;
        _this.d = null;
        _this.b = null;
        _this.$tKey = $tKey;
        _this.$tValue = $tValue;
        _this.$type = _this.$type.specialize(_this.$tKey, _this.$tValue);
        if (b == null) {
            b = EqualityComparer$1.defaultEqualityComparerValue(_this.$tKey);
        }
        _this.d = new Dictionary$2(_this.$tKey, LfuCache_CacheNode$2.$.specialize(_this.$tKey, _this.$tValue), 2, b);
        _this.b = new LinkedList();
        _this.f = a;
        return _this;
    }
    Object.defineProperty(LfuCache$2.prototype, "f", {
        get: function () {
            return this.e;
        },
        set: function (a) {
            if (a <= 0) {
                throw new ArgumentOutOfRangeException(0);
            }
            this.e = a;
            for (var b = this.d.count - a; b > 0; b--) {
                this.l();
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(LfuCache$2.prototype, "g", {
        get: function () {
            return this.d.count;
        },
        enumerable: false,
        configurable: true
    });
    LfuCache$2.prototype.item = function (a) {
        var _this = this;
        var b;
        var c;
        if (((function () { var d = _this.d.tryGetValue(a, c); c = d.p1; return d.ret; })())) {
            b = c.o;
            this.j(c);
        }
        else {
            b = Type.getDefaultValue(this.$tValue);
        }
        return b;
    };
    LfuCache$2.prototype.h = function (a, b) {
        var _this = this;
        var c;
        if (((function () { var d = _this.d.tryGetValue(a, c); c = d.p1; return d.ret; })())) {
            c.o = b;
            this.j(c);
        }
        else {
            c = new LfuCache_CacheNode$2(this.$tKey, this.$tValue);
            c.n = a;
            c.o = b;
            this.j(c);
            this.d.addItem(a, c);
            if (this.d.count > this.e) {
                this.l();
            }
        }
    };
    LfuCache$2.prototype.i = function () {
        this.d.clear();
        this.b.e();
    };
    LfuCache$2.prototype.j = function (a) {
        var b = a.p;
        if (b == null) {
            b = this.b.d;
            if (b == null || b.o != 1) {
                b = new LfuCache_FrequencyNode$2(this.$tKey, this.$tValue);
                b.o = 1;
                b.l(this.b);
            }
        }
        else {
            var c = b.f;
            if (b.n.c == b.n.d) {
                if (c == null || c.o - 1 > b.o) {
                    b.o++;
                }
                else {
                    a.m();
                    b.m();
                    b = c;
                }
            }
            else {
                if (c == null || c.o - 1 > b.o) {
                    var d = new LfuCache_FrequencyNode$2(this.$tKey, this.$tValue);
                    d.o = b.o + 1;
                    d.h(b);
                    b = d;
                }
                else {
                    b = c;
                }
                a.m();
            }
        }
        a.p = b;
        a.j(b.n);
    };
    LfuCache$2.prototype.c = function (a) {
        var _this = this;
        var b;
        var c;
        b = ((function () { var d = _this.d.tryGetValue(a, c); c = d.p1; return d.ret; })());
        if (b) {
            this.k(c);
        }
        return b;
    };
    LfuCache$2.prototype.k = function (a) {
        this.d.removeItem(a.n);
        a.m();
        var b = a.p;
        if (b.n.c == null) {
            b.m();
        }
    };
    LfuCache$2.prototype.l = function () {
        var a = this.b.d;
        var b = a.n.d;
        this.k(b);
    };
    LfuCache$2.$t = markType(LfuCache$2, 'LfuCache$2');
    return LfuCache$2;
}(Base));
export { LfuCache$2 };
