// igniteui-react-excel/esm5/lib/ObjectModelTablePartManager_TableContext.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends, __values } from "tslib";
import { Base, Number_$type, fromEnum, markType } from "igniteui-react-core";
import { Filter } from "./Filter";
import { Dictionary$2 } from "igniteui-react-core";
import { SortCondition } from "./SortCondition";
import { List$1 } from "igniteui-react-core";
import { KeyValuePair$2 } from "igniteui-react-core";
/**
 * @hidden
 */
var ObjectModelTablePartManager_TableContext = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ObjectModelTablePartManager_TableContext, _super);
    function ObjectModelTablePartManager_TableContext(a) {
        var _this = _super.call(this) || this;
        _this.b = null;
        _this.c = null;
        _this.a = null;
        _this.a = a;
        return _this;
    }
    ObjectModelTablePartManager_TableContext.prototype.d = function (a, b) {
        if (b == null) {
            return;
        }
        if (this.b == null) {
            this.b = new Dictionary$2(Number_$type, Filter.$, 0);
        }
        this.b.item(a, b);
    };
    ObjectModelTablePartManager_TableContext.prototype.e = function (a, b) {
        if (this.c == null) {
            this.c = new List$1(KeyValuePair$2.$.specialize(Number_$type, SortCondition.$), 0);
        }
        this.c.add(new KeyValuePair$2(Number_$type, SortCondition.$, 1, a, b));
    };
    ObjectModelTablePartManager_TableContext.prototype.f = function () {
        var e_1, _a, e_2, _b;
        if (this.b != null) {
            try {
                for (var _c = __values(fromEnum(this.b)), _d = _c.next(); !_d.done; _d = _c.next()) {
                    var a = _d.value;
                    var b = a.key;
                    if (b < 0 || this.a._columns$i.count <= b) {
                        continue;
                    }
                    var c = this.a._columns$i._item(b);
                    if (c == null) {
                        continue;
                    }
                    var d = a.value;
                    d._j(c);
                    c.filter = d;
                }
            }
            catch (e_1_1) {
                e_1 = { error: e_1_1 };
            }
            finally {
                try {
                    if (_d && !_d.done && (_a = _c.return))
                        _a.call(_c);
                }
                finally {
                    if (e_1)
                        throw e_1.error;
                }
            }
        }
        if (this.c != null) {
            try {
                for (var _e = __values(fromEnum(this.c)), _f = _e.next(); !_f.done; _f = _e.next()) {
                    var e = _f.value;
                    var f = e.key - this.a._by._w;
                    if (f < 0 || this.a._columns$i.count <= f) {
                        continue;
                    }
                    var g = this.a._columns$i._item(f);
                    this.a.sortSettings._sortConditions$i._addItem$i(g, e.value);
                }
            }
            catch (e_2_1) {
                e_2 = { error: e_2_1 };
            }
            finally {
                try {
                    if (_f && !_f.done && (_b = _e.return))
                        _b.call(_e);
                }
                finally {
                    if (e_2)
                        throw e_2.error;
                }
            }
        }
    };
    ObjectModelTablePartManager_TableContext.$t = markType(ObjectModelTablePartManager_TableContext, 'ObjectModelTablePartManager_TableContext');
    return ObjectModelTablePartManager_TableContext;
}(Base));
export { ObjectModelTablePartManager_TableContext };
