// igniteui-react-excel/esm5/lib/PackagePartExposedStream.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Stream, markType } from "igniteui-react-core";
/**
 * @hidden
 */
var PackagePartExposedStream = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(PackagePartExposedStream, _super);
    function PackagePartExposedStream(a) {
        var _this = _super.call(this) || this;
        _this._o = null;
        _this._o = a;
        return _this;
    }
    PackagePartExposedStream.prototype.get_canRead = function () {
        return this._o.canRead;
    };
    Object.defineProperty(PackagePartExposedStream.prototype, "canRead", {
        get: function () {
            return this.get_canRead();
        },
        enumerable: false,
        configurable: true
    });
    PackagePartExposedStream.prototype.get_canSeek = function () {
        return this._o.canSeek;
    };
    Object.defineProperty(PackagePartExposedStream.prototype, "canSeek", {
        get: function () {
            return this.get_canSeek();
        },
        enumerable: false,
        configurable: true
    });
    PackagePartExposedStream.prototype.get_canWrite = function () {
        return this._o.canWrite;
    };
    Object.defineProperty(PackagePartExposedStream.prototype, "canWrite", {
        get: function () {
            return this.get_canWrite();
        },
        enumerable: false,
        configurable: true
    });
    PackagePartExposedStream.prototype.flush = function () {
        this._o.flush();
    };
    PackagePartExposedStream.prototype.get_length = function () {
        return this._o.length;
    };
    Object.defineProperty(PackagePartExposedStream.prototype, "length", {
        get: function () {
            return this.get_length();
        },
        enumerable: false,
        configurable: true
    });
    PackagePartExposedStream.prototype.get_position = function () {
        return this._o.position;
    };
    PackagePartExposedStream.prototype.set_position = function (a) {
        this._o.position = a;
    };
    Object.defineProperty(PackagePartExposedStream.prototype, "position", {
        get: function () {
            return this.get_position();
        },
        set: function (a) {
            this.set_position(a);
        },
        enumerable: false,
        configurable: true
    });
    PackagePartExposedStream.prototype.read = function (a, b, c) {
        return this._o.read(a, b, c);
    };
    PackagePartExposedStream.prototype.seek = function (a, b) {
        return this._o.seek(a, b);
    };
    PackagePartExposedStream.prototype.setLength = function (a) {
        this._o.setLength(a);
    };
    PackagePartExposedStream.prototype.write = function (a, b, c) {
        this._o.write(a, b, c);
    };
    PackagePartExposedStream.$t = markType(PackagePartExposedStream, 'PackagePartExposedStream', Stream.$);
    return PackagePartExposedStream;
}(Stream));
export { PackagePartExposedStream };
