// igniteui-react-excel/esm5/lib/PaperSize.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various paper sizes available for printing.
 */
export var PaperSize = /*@__PURE__*/ (function (PaperSize) {
    /**
     * Undefined
     */
    PaperSize[PaperSize["Undefined"] = 0] = "Undefined";
    /**
     * Letter 8 1/2\" x 11\"
     */
    PaperSize[PaperSize["Letter"] = 1] = "Letter";
    /**
     * Letter small 8 1/2\" x 11\"
     */
    PaperSize[PaperSize["LetterSmall"] = 2] = "LetterSmall";
    /**
     * Tabloid 11\" x 17\"
     */
    PaperSize[PaperSize["Tabloid"] = 3] = "Tabloid";
    /**
     * Ledger 17\" x 11\"
     */
    PaperSize[PaperSize["Ledger"] = 4] = "Ledger";
    /**
     * Legal 8 1/2\" x 14\"
     */
    PaperSize[PaperSize["Legal"] = 5] = "Legal";
    /**
     * Statement 5 1/2\" x 8 1/2\"
     */
    PaperSize[PaperSize["Statement"] = 6] = "Statement";
    /**
     * Executive 7 1/4\" x 10 1/2\"
     */
    PaperSize[PaperSize["Executive"] = 7] = "Executive";
    /**
     * A3 297mm x 420mm
     */
    PaperSize[PaperSize["A3"] = 8] = "A3";
    /**
     * A4 210mm x 297mm
     */
    PaperSize[PaperSize["A4"] = 9] = "A4";
    /**
     * A4 small 210mm x 297mm
     */
    PaperSize[PaperSize["A4Small"] = 10] = "A4Small";
    /**
     * A5 148mm x 210mm
     */
    PaperSize[PaperSize["A5"] = 11] = "A5";
    /**
     * B4 (JIS) 257mm x 364mm
     */
    PaperSize[PaperSize["B4JIS"] = 12] = "B4JIS";
    /**
     * B5 (JIS) 182mm x 257mm
     */
    PaperSize[PaperSize["B5JIS"] = 13] = "B5JIS";
    /**
     * Folio 8 1/2\" x 13\"
     */
    PaperSize[PaperSize["Folio"] = 14] = "Folio";
    /**
     * Quarto 215mm x 275mm
     */
    PaperSize[PaperSize["Quarto"] = 15] = "Quarto";
    /**
     * 10x14 10\" x 14\"
     */
    PaperSize[PaperSize["Size10x14"] = 16] = "Size10x14";
    /**
     * 11x17 11\" x 17\"
     */
    PaperSize[PaperSize["Size11x17"] = 17] = "Size11x17";
    /**
     * Note 8 1/2\" x 11\"
     */
    PaperSize[PaperSize["Note"] = 18] = "Note";
    /**
     * Envelope #9 3 7/8\" x 8 7/8\"
     */
    PaperSize[PaperSize["Envelope9"] = 19] = "Envelope9";
    /**
     * Envelope #10 4 1/8\" x 9 1/2\"
     */
    PaperSize[PaperSize["Envelope10"] = 20] = "Envelope10";
    /**
     * Envelope #11 4 1/2\" x 10 3/8\"
     */
    PaperSize[PaperSize["Envelope11"] = 21] = "Envelope11";
    /**
     * Envelope #12 4 3/4\" x 11\"
     */
    PaperSize[PaperSize["Envelope12"] = 22] = "Envelope12";
    /**
     * Envelope #14 5\" x 11 1/2\"
     */
    PaperSize[PaperSize["Envelope14"] = 23] = "Envelope14";
    /**
     * C 17\" x 22\"
     */
    PaperSize[PaperSize["C"] = 24] = "C";
    /**
     * D 22\" x 34\"
     */
    PaperSize[PaperSize["D"] = 25] = "D";
    /**
     * E 34\" x 44\"
     */
    PaperSize[PaperSize["E"] = 26] = "E";
    /**
     * Envelope DL 110mm x 220mm
     */
    PaperSize[PaperSize["EnvelopeDL"] = 27] = "EnvelopeDL";
    /**
     * Envelope C5 162mm x 229mm
     */
    PaperSize[PaperSize["EnvelopeC5"] = 28] = "EnvelopeC5";
    /**
     * Envelope C3 324mm x 458mm
     */
    PaperSize[PaperSize["EnvelopeC3"] = 29] = "EnvelopeC3";
    /**
     * Envelope C4 229mm x 324mm
     */
    PaperSize[PaperSize["EnvelopeC4"] = 30] = "EnvelopeC4";
    /**
     * Envelope C6 114mm x 162mm
     */
    PaperSize[PaperSize["EnvelopeC6"] = 31] = "EnvelopeC6";
    /**
     * Envelope C6/C5 114mm x 229mm
     */
    PaperSize[PaperSize["EnvelopeC6C5"] = 32] = "EnvelopeC6C5";
    /**
     * B4 (ISO) 250mm x 353mm
     */
    PaperSize[PaperSize["B4ISO_1"] = 33] = "B4ISO_1";
    /**
     * B5 (ISO) 176mm x 250mm
     */
    PaperSize[PaperSize["B5ISO"] = 34] = "B5ISO";
    /**
     * B6 (ISO) 125mm x 176mm
     */
    PaperSize[PaperSize["B6ISO"] = 35] = "B6ISO";
    /**
     * Envelope Italy 110mm x 230mm
     */
    PaperSize[PaperSize["EnvelopeItaly"] = 36] = "EnvelopeItaly";
    /**
     * Envelope Monarch 3 7/8\" x 7 1/2\"
     */
    PaperSize[PaperSize["EnvelopeMonarch"] = 37] = "EnvelopeMonarch";
    /**
     * 6 3/4 Envelope 3 5/8\" x 6 1/2\"
     */
    PaperSize[PaperSize["Size634Envelope"] = 38] = "Size634Envelope";
    /**
     * US Standard Fanfold 14 7/8\" x 11\"
     */
    PaperSize[PaperSize["USStandardFanfold"] = 39] = "USStandardFanfold";
    /**
     * German Std. Fanfold 8 1/2\" x 12\"
     */
    PaperSize[PaperSize["GermanStandardFanfold"] = 40] = "GermanStandardFanfold";
    /**
     * German Legal Fanfold 8 1/2\" x 13\"
     */
    PaperSize[PaperSize["GermanLegalFanfold"] = 41] = "GermanLegalFanfold";
    /**
     * B4 (ISO) 250mm x 353mm
     */
    PaperSize[PaperSize["B4ISO_2"] = 42] = "B4ISO_2";
    /**
     * Japanese Postcard 100mm x 148mm
     */
    PaperSize[PaperSize["JapanesePostcard"] = 43] = "JapanesePostcard";
    /**
     * 9x11 9\" x 11\"
     */
    PaperSize[PaperSize["Size9x11"] = 44] = "Size9x11";
    /**
     * 10x11 10\" x 11\"
     */
    PaperSize[PaperSize["Size10x11"] = 45] = "Size10x11";
    /**
     * 15x11 15\" x 11\"
     */
    PaperSize[PaperSize["Size15x11"] = 46] = "Size15x11";
    /**
     * Envelope Invite 220mm x 220mm
     */
    PaperSize[PaperSize["EnvelopeInvite"] = 47] = "EnvelopeInvite";
    /**
     * Letter Extra 9 1/2\" x 12\"
     */
    PaperSize[PaperSize["LetterExtra"] = 50] = "LetterExtra";
    /**
     * Legal Extra 9 1/2\" x 15\"
     */
    PaperSize[PaperSize["LegalExtra"] = 51] = "LegalExtra";
    /**
     * Tabloid Extra 11 11/16\" x 18\"
     */
    PaperSize[PaperSize["TabloidExtra"] = 52] = "TabloidExtra";
    /**
     * A4 Extra 235mm x 322mm
     */
    PaperSize[PaperSize["A4Extra"] = 53] = "A4Extra";
    /**
     * Letter Transverse 8 1/2\" x 11\"
     */
    PaperSize[PaperSize["LetterTransverse"] = 54] = "LetterTransverse";
    /**
     * A4 Transverse 210mm x 297mm
     */
    PaperSize[PaperSize["A4Transverse"] = 55] = "A4Transverse";
    /**
     * Letter Extra Transv. 9 1/2\" x 12\"
     */
    PaperSize[PaperSize["LetterExtraTransverse"] = 56] = "LetterExtraTransverse";
    /**
     * Super A/A4 227mm x 356mm
     */
    PaperSize[PaperSize["SuperAA4"] = 57] = "SuperAA4";
    /**
     * Super B/A3 305mm x 487mm
     */
    PaperSize[PaperSize["SuperBA3"] = 58] = "SuperBA3";
    /**
     * Letter Plus 8 1/2\" x 12 11/16\"
     */
    PaperSize[PaperSize["LetterPlus"] = 59] = "LetterPlus";
    /**
     * A4 Plus 210mm x 330mm
     */
    PaperSize[PaperSize["A4Plus"] = 60] = "A4Plus";
    /**
     * A5 Transverse 148mm x 210mm
     */
    PaperSize[PaperSize["A5Transverse"] = 61] = "A5Transverse";
    /**
     * B5 (JIS) Transverse 182mm x 257mm
     */
    PaperSize[PaperSize["B5JISTransverse"] = 62] = "B5JISTransverse";
    /**
     * A3 Extra 322mm x 445mm
     */
    PaperSize[PaperSize["A3Extra"] = 63] = "A3Extra";
    /**
     * A5 Extra 174mm x 235mm
     */
    PaperSize[PaperSize["A5Extra"] = 64] = "A5Extra";
    /**
     * B5 (ISO) Extra 201mm x 276mm
     */
    PaperSize[PaperSize["B5ISOExtra"] = 65] = "B5ISOExtra";
    /**
     * A2 420mm x 594mm
     */
    PaperSize[PaperSize["A2"] = 66] = "A2";
    /**
     * A3 Transverse 297mm x 420mm
     */
    PaperSize[PaperSize["A3Transverse"] = 67] = "A3Transverse";
    /**
     * A3 Extra Transverse 322mm x 445mm
     */
    PaperSize[PaperSize["A3ExtraTransverse"] = 68] = "A3ExtraTransverse";
    /**
     * Dbl. Japanese Postcard 200mm x 148mm
     */
    PaperSize[PaperSize["DblJapanesePostcard"] = 69] = "DblJapanesePostcard";
    /**
     * A6 105mm x 148mm
     */
    PaperSize[PaperSize["A6"] = 70] = "A6";
    /**
     * Letter Rotated 11\" x 8 1/2\"
     */
    PaperSize[PaperSize["LetterRotated"] = 75] = "LetterRotated";
    /**
     * A3 Rotated 420mm x 297mm
     */
    PaperSize[PaperSize["A3Rotated"] = 76] = "A3Rotated";
    /**
     * A4 Rotated 297mm x 210mm
     */
    PaperSize[PaperSize["A4Rotated"] = 77] = "A4Rotated";
    /**
     * A5 Rotated 210mm x 148mm
     */
    PaperSize[PaperSize["A5Rotated"] = 78] = "A5Rotated";
    /**
     * B4 (JIS) Rotated 364mm x 257mm
     */
    PaperSize[PaperSize["B4JISRotated"] = 79] = "B4JISRotated";
    /**
     * B5 (JIS) Rotated 257mm x 182mm
     */
    PaperSize[PaperSize["B5JISRotated"] = 80] = "B5JISRotated";
    /**
     * Japanese Postcard Rot. 148mm x 100mm
     */
    PaperSize[PaperSize["JapanesePostcardRotated"] = 81] = "JapanesePostcardRotated";
    /**
     * Dbl. Jap. Postcard Rot. 148mm x 200mm
     */
    PaperSize[PaperSize["DblJapanesePostcardRotated"] = 82] = "DblJapanesePostcardRotated";
    /**
     * A6 Rotated 148mm x 105mm
     */
    PaperSize[PaperSize["A6Rotated"] = 83] = "A6Rotated";
    /**
     * B6 (JIS) 128mm x 182mm
     */
    PaperSize[PaperSize["B6JIS"] = 88] = "B6JIS";
    /**
     * B6 (JIS) Rotated 182mm x 128mm
     */
    PaperSize[PaperSize["B6JISRotated"] = 89] = "B6JISRotated";
    /**
     * 12x11 12\" x 11\"
     */
    PaperSize[PaperSize["Size12x11"] = 90] = "Size12x11";
    return PaperSize;
})({});
/**
 * @hidden
 */
export var PaperSize_$type = /*@__PURE__*/ markEnum('PaperSize', 'Undefined,0|Letter,1|LetterSmall,2|Tabloid,3|Ledger,4|Legal,5|Statement,6|Executive,7|A3,8|A4,9|A4Small,10|A5,11|B4JIS,12|B5JIS,13|Folio,14|Quarto,15|Size10x14,16|Size11x17,17|Note,18|Envelope9,19|Envelope10,20|Envelope11,21|Envelope12,22|Envelope14,23|C,24|D,25|E,26|EnvelopeDL,27|EnvelopeC5,28|EnvelopeC3,29|EnvelopeC4,30|EnvelopeC6,31|EnvelopeC6C5,32|B4ISO_1,33|B5ISO,34|B6ISO,35|EnvelopeItaly,36|EnvelopeMonarch,37|Size634Envelope,38|USStandardFanfold,39|GermanStandardFanfold,40|GermanLegalFanfold,41|B4ISO_2,42|JapanesePostcard,43|Size9x11,44|Size10x11,45|Size15x11,46|EnvelopeInvite,47|LetterExtra,50|LegalExtra,51|TabloidExtra,52|A4Extra,53|LetterTransverse,54|A4Transverse,55|LetterExtraTransverse,56|SuperAA4,57|SuperBA3,58|LetterPlus,59|A4Plus,60|A5Transverse,61|B5JISTransverse,62|A3Extra,63|A5Extra,64|B5ISOExtra,65|A2,66|A3Transverse,67|A3ExtraTransverse,68|DblJapanesePostcard,69|A6,70|LetterRotated,75|A3Rotated,76|A4Rotated,77|A5Rotated,78|B4JISRotated,79|B5JISRotated,80|JapanesePostcardRotated,81|DblJapanesePostcardRotated,82|A6Rotated,83|B6JIS,88|B6JISRotated,89|Size12x11,90');
