// igniteui-react-excel/esm5/lib/PredefinedShapeType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the shape types that are predefined in Microsoft Excel.
 */
export var PredefinedShapeType = /*@__PURE__*/ (function (PredefinedShapeType) {
    /**
     * Represents the [[DiamondShape]] shape.
     */
    PredefinedShapeType[PredefinedShapeType["Diamond"] = 4] = "Diamond";
    /**
     * Represents the [[EllipseShape]] shape.
     */
    PredefinedShapeType[PredefinedShapeType["Ellipse"] = 3] = "Ellipse";
    /**
     * Represents the [[HeartShape]] shape.
     */
    PredefinedShapeType[PredefinedShapeType["Heart"] = 74] = "Heart";
    /**
     * Represents the [[IrregularSeal1Shape]] shape.
     */
    PredefinedShapeType[PredefinedShapeType["IrregularSeal1"] = 71] = "IrregularSeal1";
    /**
     * Represents the [[IrregularSeal2Shape]] shape.
     */
    PredefinedShapeType[PredefinedShapeType["IrregularSeal2"] = 72] = "IrregularSeal2";
    /**
     * Represents the [[LightningBoltShape]] shape.
     */
    PredefinedShapeType[PredefinedShapeType["LightningBolt"] = 73] = "LightningBolt";
    /**
     * Represents the [[LineShape]] shape.
     */
    PredefinedShapeType[PredefinedShapeType["Line"] = 20] = "Line";
    /**
     * Represents the [[PentagonShape]] shape.
     */
    PredefinedShapeType[PredefinedShapeType["Pentagon"] = 56] = "Pentagon";
    /**
     * Represents the [[RectangleShape]] shape.
     */
    PredefinedShapeType[PredefinedShapeType["Rectangle"] = 1] = "Rectangle";
    /**
     * Represents the [[RightTriangleShape]] shape.
     */
    PredefinedShapeType[PredefinedShapeType["RightTriangle"] = 6] = "RightTriangle";
    /**
     * Represents the [[StraightConnector1Shape]] shape.
     */
    PredefinedShapeType[PredefinedShapeType["StraightConnector1"] = 32] = "StraightConnector1";
    return PredefinedShapeType;
})({});
/**
 * @hidden
 */
export var PredefinedShapeType_$type = /*@__PURE__*/ markEnum('PredefinedShapeType', 'Diamond,4|Ellipse,3|Heart,74|IrregularSeal1,71|IrregularSeal2,72|LightningBolt,73|Line,20|Pentagon,56|Rectangle,1|RightTriangle,6|StraightConnector1,32');
