// igniteui-react-excel/esm5/lib/PrintErrors.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various ways to print cell errors in a worksheet.
 */
export var PrintErrors = /*@__PURE__*/ (function (PrintErrors) {
    /**
     * Errors are printed just as they are displayed on the worksheet.
     */
    PrintErrors[PrintErrors["PrintAsDisplayed"] = 0] = "PrintAsDisplayed";
    /**
     * Errors are not prints, as though the cells containing them have no value.
     */
    PrintErrors[PrintErrors["DontPrint"] = 1] = "DontPrint";
    /**
     * Errors are printed as two dashes "--".
     */
    PrintErrors[PrintErrors["PrintAsDashes"] = 2] = "PrintAsDashes";
    /**
     * Errors are prints as "#N/A".
     */
    PrintErrors[PrintErrors["PrintAsNA"] = 3] = "PrintAsNA";
    return PrintErrors;
})({});
/**
 * @hidden
 */
export var PrintErrors_$type = /*@__PURE__*/ markEnum('PrintErrors', 'PrintAsDisplayed,0|DontPrint,1|PrintAsDashes,2|PrintAsNA,3');
