// igniteui-react-excel/esm5/lib/ProtectionValidationInfo_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, markType, toNullable, Number_$type } from "igniteui-react-core";
/**
 * @hidden
 */
var ProtectionValidationInfo = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ProtectionValidationInfo, _super);
    function ProtectionValidationInfo() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ProtectionValidationInfo.c = function (a) {
        return new ProtectionValidationInfo_XorObfuscation(a);
    };
    ProtectionValidationInfo.b = function (a, b, c, d) {
        return new ProtectionValidationInfo_Strict(a, b, c, d);
    };
    ProtectionValidationInfo.a = function (a, b, c, d) {
        c = a = null;
        b = null;
        d = 0;
        return {
            ret: null,
            p1: b,
            p2: c,
            p3: d
        };
    };
    ProtectionValidationInfo.prototype.e = function () {
        var _this = this;
        var a = this.f();
        if (a.hasValue && a.value != 0) {
            return true;
        }
        var b;
        var c, d;
        var e;
        if (((function () { var f = _this.d(b, c, d, e); b = f.p0; c = f.p1; d = f.p2; e = f.p3; return f.ret; })())) {
            return true;
        }
        return false;
    };
    ProtectionValidationInfo.$t = markType(ProtectionValidationInfo, 'ProtectionValidationInfo');
    return ProtectionValidationInfo;
}(Base));
export { ProtectionValidationInfo };
/**
 * @hidden
 */
var ProtectionValidationInfo_XorObfuscation = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ProtectionValidationInfo_XorObfuscation, _super);
    function ProtectionValidationInfo_XorObfuscation(a) {
        var _this = _super.call(this) || this;
        _this.g = 0;
        _this.g = a;
        return _this;
    }
    ProtectionValidationInfo_XorObfuscation.prototype.d = function (a, b, c, d) {
        a = null;
        b = c = null;
        d = toNullable(Number_$type, null);
        return {
            ret: false,
            p0: a,
            p1: b,
            p2: c,
            p3: d
        };
    };
    ProtectionValidationInfo_XorObfuscation.prototype.f = function () {
        return toNullable(Number_$type, this.g);
    };
    ProtectionValidationInfo_XorObfuscation.$t = markType(ProtectionValidationInfo_XorObfuscation, 'ProtectionValidationInfo_XorObfuscation', ProtectionValidationInfo.$);
    return ProtectionValidationInfo_XorObfuscation;
}(ProtectionValidationInfo));
export { ProtectionValidationInfo_XorObfuscation };
/**
 * @hidden
 */
var ProtectionValidationInfo_Strict = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ProtectionValidationInfo_Strict, _super);
    function ProtectionValidationInfo_Strict(a, b, c, d) {
        var _this = _super.call(this) || this;
        _this.k = null;
        _this.g = null;
        _this.h = null;
        _this.m = 0;
        _this.k = a;
        _this.g = b;
        _this.h = c;
        _this.m = d;
        return _this;
    }
    ProtectionValidationInfo_Strict.prototype.d = function (a, b, c, d) {
        a = this.k;
        b = this.g;
        c = this.h;
        d = toNullable(Number_$type, this.m);
        return {
            ret: true,
            p0: a,
            p1: b,
            p2: c,
            p3: d
        };
    };
    ProtectionValidationInfo_Strict.prototype.f = function () {
        return toNullable(Number_$type, null);
    };
    ProtectionValidationInfo_Strict.$t = markType(ProtectionValidationInfo_Strict, 'ProtectionValidationInfo_Strict', ProtectionValidationInfo.$);
    return ProtectionValidationInfo_Strict;
}(ProtectionValidationInfo));
export { ProtectionValidationInfo_Strict };
