// igniteui-react-excel/esm5/lib/ProtectionInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, toNullable, Boolean_$type, typeCast, markType } from "igniteui-react-core";
import { Convert } from "igniteui-react-core";
import { XlsxUtilities } from "./XlsxUtilities";
import { unwrapNullable, nullableEquals } from "igniteui-react-core";
/**
 * @hidden
 */
var ProtectionInfo = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(ProtectionInfo, _super);
    function ProtectionInfo() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.g = toNullable(Boolean_$type, null);
        _this.i = toNullable(Boolean_$type, null);
        return _this;
    }
    ProtectionInfo.prototype.equals = function (a) {
        return ProtectionInfo.d(this, typeCast(ProtectionInfo.$, a));
    };
    ProtectionInfo.prototype.getHashCode = function () {
        var a = 0;
        a ^= Convert.toInt321(unwrapNullable(this.g));
        a ^= Convert.toInt321(unwrapNullable(this.i)) << 1;
        return a;
    };
    Object.defineProperty(ProtectionInfo.prototype, "h", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ProtectionInfo.prototype, "j", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(ProtectionInfo.prototype, "e", {
        get: function () {
            return nullableEquals(this.g, null) && nullableEquals(this.i, null);
        },
        enumerable: false,
        configurable: true
    });
    ProtectionInfo.prototype.k = function (a) {
        a._locked$i = XlsxUtilities.a9(this.j);
    };
    ProtectionInfo.a = function (a) {
        return ProtectionInfo.b(a, false);
    };
    ProtectionInfo.b = function (a, b) {
        var c = a._c5;
        if (b == false && nullableEquals(c, true)) {
            return null;
        }
        var d = new ProtectionInfo();
        d.j = toNullable(Boolean_$type, (nullableEquals(c, true)));
        return d;
    };
    ProtectionInfo.d = function (a, b) {
        if (Base.referenceEquals(a, null) && Base.referenceEquals(b, null)) {
            return true;
        }
        if (Base.referenceEquals(a, null) || Base.referenceEquals(b, null)) {
            return false;
        }
        return (nullableEquals(a.g, b.g) && nullableEquals(a.i, b.i));
    };
    ProtectionInfo.$t = markType(ProtectionInfo, 'ProtectionInfo');
    return ProtectionInfo;
}(Base));
export { ProtectionInfo };
