// igniteui-react-excel/esm5/lib/RegionLookupTable_RangeHelper$1.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, runOn, markType } from "igniteui-react-core";
import { HashSet$1 } from "igniteui-react-core";
/**
 * @hidden
 */
var RegionLookupTable_RangeHelper$1 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(RegionLookupTable_RangeHelper$1, _super);
    function RegionLookupTable_RangeHelper$1($t) {
        var _this = _super.call(this) || this;
        _this.$t = null;
        _this.g = 0;
        _this.h = 0;
        _this.i = 0;
        _this.j = 0;
        _this.e = null;
        _this.d = null;
        _this.c = null;
        _this.$t = $t;
        _this.$type = _this.$type.specialize(_this.$t);
        _this.e = runOn(_this, _this.b);
        _this.d = runOn(_this, _this.a);
        _this.c = new HashSet$1(_this.$t, 0);
        return _this;
    }
    RegionLookupTable_RangeHelper$1.prototype.k = function () {
        this.c.clear();
    };
    RegionLookupTable_RangeHelper$1.prototype.f = function (a, b, c, d, e) {
        this.i = b;
        this.g = c;
        this.j = d;
        this.h = e;
        return a ? this.e : this.d;
    };
    RegionLookupTable_RangeHelper$1.prototype.a = function (a) {
        return a.intersectsWithColumns(this.g, this.h) && this.c.add_1(a);
    };
    RegionLookupTable_RangeHelper$1.prototype.b = function (a) {
        return a.intersectsWithRows(this.i, this.j) && this.c.add_1(a);
    };
    RegionLookupTable_RangeHelper$1.$t = markType(RegionLookupTable_RangeHelper$1, 'RegionLookupTable_RangeHelper$1');
    return RegionLookupTable_RangeHelper$1;
}(Base));
export { RegionLookupTable_RangeHelper$1 };
