// igniteui-react-excel/esm5/lib/ScrollBars.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various scroll bar configurations available for the workbook.
 */
export var ScrollBars = /*@__PURE__*/ (function (ScrollBars) {
    /**
     * No scroll bars are shown in Microsoft Excel.
     */
    ScrollBars[ScrollBars["None"] = 0] = "None";
    /**
     * Only the horizontal scroll bar is shown in Microsoft Excel.
     */
    ScrollBars[ScrollBars["Horizontal"] = 1] = "Horizontal";
    /**
     * Only the vertical scroll bar is shown in Microsoft Excel.
     */
    ScrollBars[ScrollBars["Vertical"] = 2] = "Vertical";
    /**
     * Both scroll bars are shown in Microsoft Excel.
     */
    ScrollBars[ScrollBars["Both"] = 3] = "Both";
    return ScrollBars;
})({});
/**
 * @hidden
 */
export var ScrollBars_$type = /*@__PURE__*/ markEnum('ScrollBars', 'None,0|Horizontal,1|Vertical,2|Both,3');
