// igniteui-react-excel/esm5/lib/SgmlReader.js
/*
*
* An XmlReader implementation for loading SGML (including HTML) converting it
* to well formed XML, by adding missing quotes, empty attribute values, ignoring
* duplicate attributes, case folding on tag names, adding missing closing tags
* based on SGML DTD information, and so on.
*
* Copyright (c) 2002 Microsoft Corporation. All rights reserved.
*
* Chris Lovett
*
*/
import { __extends, __read, __spreadArray } from "tslib";
/* MD 2/27/13
 * This software is being re-used under the Microsoft Public License (MS-PL):
 * -----------------------------------------------------------------------------
 * This license governs use of the accompanying software. If you use the software, you accept this license. If you do not accept the
 * license, do not use the software.
 *
 * 1. Definitions
 * The terms "reproduce," "reproduction," "derivative works," and "distribution" have the same meaning here as under U.S. copyright law.
 * A "contribution" is the original software, or any additions or changes to the software. A "contributor" is any person that distributes
 * its contribution under this license. "Licensed patents" are a contributor's patent claims that read directly on its contribution.
 *
 * 2. Grant of Rights
 * (A) Copyright Grant- Subject to the terms of this license, including the license conditions and limitations in section 3, each
 *     contributor grants you a non-exclusive, worldwide, royalty-free copyright license to reproduce its contribution, prepare
 *     derivative works of its contribution, and distribute its contribution or any derivative works that you create.
 * (B) Patent Grant- Subject to the terms of this license, including the license conditions and limitations in section 3, each
 *     contributor grants you a non-exclusive, worldwide, royalty-free license under its licensed patents to make, have made, use,
 *     sell, offer for sale, import, and/or otherwise dispose of its contribution in the software or derivative works of the contribution
 *     in the software.
 *
 * 3. Conditions and Limitations
 * (A) No Trademark License- This license does not grant you rights to use any contributors' name, logo, or trademarks.
 * (B) If you bring a patent claim against any contributor over patents that you claim are infringed by the software, your patent license
 *     from such contributor to the software ends automatically.
 * (C) If you distribute any portion of the software, you must retain all copyright, patent, trademark, and attribution notices that are
 *     present in the software.
 * (D) If you distribute any portion of the software in source code form, you may do so only under this license by including a complete
 *     copy of this license with your distribution. If you distribute any portion of the software in compiled or object code form, you may
 *     only do so under a license that complies with this license.
 * (E) The software is licensed "as-is." You bear the risk of using it. The contributors give no express warranties, guarantees or conditions.
 *     You may have additional consumer rights under your local laws which this license cannot change. To the extent permitted under your
 *     local laws, the contributors exclude the implied warranties of merchantability, fitness for a particular purpose and non-infringement.
 * -----------------------------------------------------------------------------
 *
 * Note: edits have also been made to support namespace prefixes. Those changes are tagged with the following comment:
 * // MD - Added support for namespaces
 *
 * In addition, the public classes have been made internal and the namespace has been changed from Sgml to
 * Infragistics.Documents.Excel.Serialization.Excel2007.Sgml
 *
 * Also, I updated the code to work with Silverlight and WinRT.
 */
import { XmlReader } from "igniteui-react-core";
import { Uri } from "igniteui-react-core";
import { Node } from "./Node";
import { StringUtilitiesSgml } from "./StringUtilitiesSgml";
import { HWStack } from "./HWStack";
import { NameTable } from "igniteui-react-core";
import { XmlNamespaceManager } from "igniteui-react-core";
import { Entity } from "./Entity";
import { markType } from "igniteui-react-core";
import { StringBuilder } from "igniteui-react-core";
import { IndexOutOfRangeException } from "igniteui-react-core";
import { InvalidOperationException } from "igniteui-react-core";
import { StringUtilities } from "./StringUtilities";
import { StringWriter } from "igniteui-react-core";
import { XmlTextWriter } from "igniteui-react-core";
import { stringFormat1, stringCompare1, isLetter, startsWith1 } from "igniteui-react-core";
import { stringEmpty } from "igniteui-react-core";
/**
 * @hidden
 */
var SgmlReader = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(SgmlReader, _super);
    function SgmlReader() {
        var _this = _super.call(this) || this;
        _this._bk = null;
        _this._bc = null;
        _this._bm = 0;
        _this._dh = null;
        _this._b2 = '\0';
        _this._ca = null;
        _this._be = null;
        _this._bg = null;
        _this._a9 = null;
        _this._b3 = 0;
        _this._c0 = null;
        _this._cz = null;
        _this._cy = null;
        _this._b9 = null;
        _this._bn = false;
        _this._bf = null;
        _this._b4 = 0;
        _this._b5 = 0;
        _this._bo = false;
        _this._cp = null;
        _this._ci = null;
        _this._bd = null;
        _this._cm = null;
        _this._b6 = null;
        _this._ct = null;
        _this._cn = null;
        _this._cs = null;
        _this._cf = null;
        _this._de = 0;
        _this._bb = 0;
        _this._b0 = true;
        _this._dg = null;
        _this.c2();
        _this._dh = new NameTable();
        _this._dg = new XmlNamespaceManager(_this._dh);
        return _this;
    }
    Object.defineProperty(SgmlReader.prototype, "_bl", {
        get: function () {
            this._c3(this._c0);
            return this._bk;
        },
        set: function (a) {
            this._bk = a;
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype._c3 = function (a) {
        if (this._bk == null) {
            if (this._bk != null && this._bk.r != null) {
                switch (this._ba) {
                    case 1:
                        this._cp = this._bk.r.toUpperCase();
                        break;
                    case 2:
                        this._cp = this._bk.r.toLowerCase();
                        break;
                    default:
                        this._cp = this._bk.r;
                        break;
                }
                this._bo = StringUtilitiesSgml.a(this._bk.r, "html");
            }
        }
    };
    Object.defineProperty(SgmlReader.prototype, "_cg", {
        get: function () {
            return this._cf;
        },
        set: function (a) {
            this._cf = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SgmlReader.prototype, "_co", {
        get: function () {
            return this._cn;
        },
        set: function (a) {
            this._cn = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SgmlReader.prototype, "_cu", {
        get: function () {
            return this._ct;
        },
        set: function (a) {
            this._ct = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SgmlReader.prototype, "_ck", {
        get: function () {
            return this._cs;
        },
        set: function (a) {
            this._cs = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SgmlReader.prototype, "_b7", {
        get: function () {
            return this._b6;
        },
        set: function (a) {
            this._b6 = a;
            this.c2();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SgmlReader.prototype, "_cw", {
        get: function () {
            return this._cm;
        },
        set: function (a) {
            this._cm = a;
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype._c9 = function (a) {
        this._c0 = new Uri(0, a);
    };
    Object.defineProperty(SgmlReader.prototype, "_cj", {
        get: function () {
            return this._ci;
        },
        set: function (a) {
            this._ci = a;
            this.c2();
            if (this._c0 == null) {
                if (this._ci.indexOf("://") > 0) {
                    this._c0 = new Uri(0, this._ci);
                }
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SgmlReader.prototype, "_b1", {
        get: function () {
            return this._b0;
        },
        set: function (a) {
            this._b0 = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SgmlReader.prototype, "_ba", {
        get: function () {
            return this._bb;
        },
        set: function (a) {
            this._bb = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SgmlReader.prototype, "_b8", {
        get: function () {
            return this._b9;
        },
        set: function (a) {
            this._b9 = a;
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype._c4 = function (a) {
        var b = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            b[_i - 1] = arguments[_i];
        }
        if (this._b8 != null) {
            var c = stringFormat1.apply(void 0, __spreadArray([a], __read(b), false));
            if (this._bd != this._bc) {
                c = c + "    " + this._bc.r();
                this._bd = this._bc;
                this._b8.an("### Error:" + c);
            }
            else {
                var d = "";
                if (this._bc.ae != null) {
                    d = this._bc.ae.absolutePath;
                }
                this._b8.an("### Error in " + d + "#" + this._bc.u + ", line " + this._bc.n + ", position " + this._bc.o + ": " + c);
            }
        }
    };
    SgmlReader.prototype._c5 = function (a, b) {
        this._c4(a, b.toString());
    };
    SgmlReader.prototype.c2 = function () {
        this._bm = 0;
        this._be = new HWStack(10);
        this._bg = this._bi(null, 9, null);
        this._bg.f = false;
        this._cz = new StringBuilder(0);
        this._cy = new StringBuilder(0);
        this._b4 = 0;
        this._bc = null;
        this._b2 = '\0';
        this._ca = null;
        this._a9 = null;
        this._b3 = 0;
        this._bf = null;
        this._b5 = 0;
        this._bn = false;
    };
    SgmlReader.prototype._bi = function (a, b, c) {
        var d = this._be.i();
        if (d == null) {
            d = new Node();
            this._be.item(this._be.c - 1, d);
        }
        d.o(a, b, c);
        this._bg = d;
        return d;
    };
    SgmlReader.prototype._da = function () {
        var a = this._be.c - 1;
        if (a > 0) {
            var b = this._be.item(a - 1);
            this._be.item(a - 1, this._be.item(a));
            this._be.item(a, b);
        }
    };
    SgmlReader.prototype._bh = function (a) {
        var b = this._bi(a.j, a.p, a.k);
        b.c = a.c;
        b.f = a.f;
        b.q = a.q;
        b.l = a.l;
        b.e = a.e;
        b.m(a);
        this._bg = b;
        return b;
    };
    SgmlReader.prototype._c8 = function () {
        if (this._be.c > 1) {
            this._bg = this._be.h();
        }
    };
    SgmlReader.prototype._bj = function () {
        var a = this._be.c - 1;
        if (a > 0) {
            return this._be.item(a);
        }
        return null;
    };
    SgmlReader.prototype.get_a7 = function () {
        if (this._bm == 3) {
            return 2;
        }
        else if (this._bm == 4) {
            return 3;
        }
        else if (this._bm == 2 || this._bm == 7) {
            return 15;
        }
        return this._bg.p;
    };
    Object.defineProperty(SgmlReader.prototype, "a7", {
        get: function () {
            return this.get_a7();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_ai = function () {
        var a = null;
        if (this._bm == 3) {
            a = this._a9.e;
        }
        else if (this._bm == 4) {
            a = null;
        }
        else {
            a = this._bg.j;
        }
        return a;
    };
    Object.defineProperty(SgmlReader.prototype, "ai", {
        get: function () {
            return this.get_ai();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_ag = function () {
        var a = this.ai;
        var b = this.ak;
        if (b.length != 0) {
            return a.substr(b.length + 1);
        }
        return a;
    };
    Object.defineProperty(SgmlReader.prototype, "ag", {
        get: function () {
            return this.get_ag();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_aj = function () {
        if (this._bm == 3 && StringUtilitiesSgml.a(this._a9.e, "xmlns")) {
            return "http://www.w3.org/2000/xmlns/";
        }
        var a = this.ak;
        switch (a) {
            case "xmlns": return "http://www.w3.org/2000/xmlns/";
            case "xml": return "http://www.w3.org/XML/1998/namespace";
            case "": return stringEmpty();
            default: return this.ah(a);
        }
    };
    Object.defineProperty(SgmlReader.prototype, "aj", {
        get: function () {
            return this.get_aj();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_ak = function () {
        var a = this.ai;
        if (a == null) {
            return stringEmpty();
        }
        var b = a.indexOf(':');
        if (b < 0) {
            return stringEmpty();
        }
        return a.substr(0, b);
    };
    Object.defineProperty(SgmlReader.prototype, "ak", {
        get: function () {
            return this.get_ak();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_f = function () {
        if (this._bm == 3 || this._bm == 4) {
            return true;
        }
        return (this._bg.k != null);
    };
    Object.defineProperty(SgmlReader.prototype, "f", {
        get: function () {
            return this.get_f();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_ar = function () {
        if (this._bm == 3 || this._bm == 4) {
            return this._a9.f;
        }
        return this._bg.k;
    };
    Object.defineProperty(SgmlReader.prototype, "ar", {
        get: function () {
            return this.get_ar();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_x = function () {
        if (this._bm == 3) {
            return this._be.c;
        }
        else if (this._bm == 4) {
            return this._be.c + 1;
        }
        return this._be.c - 1;
    };
    Object.defineProperty(SgmlReader.prototype, "x", {
        get: function () {
            return this.get_x();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_z = function () {
        return this._c0 == null ? "" : this._c0.absoluteUri;
    };
    Object.defineProperty(SgmlReader.prototype, "z", {
        get: function () {
            return this.get_z();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_i = function () {
        if (this._bm == 1 || this._bm == 3 || this._bm == 4) {
            return this._bg.f;
        }
        return false;
    };
    Object.defineProperty(SgmlReader.prototype, "i", {
        get: function () {
            return this.get_i();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_h = function () {
        if (this._bm == 3 || this._bm == 4) {
            return this._a9.b;
        }
        return false;
    };
    Object.defineProperty(SgmlReader.prototype, "h", {
        get: function () {
            return this.get_h();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_v = function () {
        if (this._a9 != null) {
            return this._a9.c;
        }
        return '\0';
    };
    Object.defineProperty(SgmlReader.prototype, "v", {
        get: function () {
            return this.get_v();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_a8 = function () {
        for (var a = this._be.c - 1; a > 1; a--) {
            var b = this._be.item(a);
            var c = b.q;
            if (c != 0) {
                return c;
            }
        }
        return 0;
    };
    Object.defineProperty(SgmlReader.prototype, "a8", {
        get: function () {
            return this.get_a8();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_as = function () {
        for (var a = this._be.c - 1; a > 1; a--) {
            var b = this._be.item(a);
            var c = b.l;
            if (c != null) {
                return c;
            }
        }
        return stringEmpty();
    };
    Object.defineProperty(SgmlReader.prototype, "as", {
        get: function () {
            return this.get_as();
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SgmlReader.prototype, "_df", {
        get: function () {
            return this._de;
        },
        set: function (a) {
            this._de = a;
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.get_w = function () {
        if (this._bm == 3 || this._bm == 4) {
            return 0;
        }
        if (this._bg.p == 1 || this._bg.p == 10) {
            return this._bg.h;
        }
        return 0;
    };
    Object.defineProperty(SgmlReader.prototype, "w", {
        get: function () {
            return this.get_w();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.ab = function (a) {
        if (this._bm != 3 && this._bm != 4) {
            var b = this._bg.i(a);
            if (b >= 0) {
                return this.aa(b);
            }
        }
        return null;
    };
    SgmlReader.prototype.ac = function (a, b) {
        return this.ab(a);
    };
    SgmlReader.prototype.aa = function (a) {
        if (this._bm != 3 && this._bm != 4) {
            var b = this._bg.b(a);
            if (b != null) {
                return b.f;
            }
        }
        throw new IndexOutOfRangeException(0);
    };
    SgmlReader.prototype.item = function (a) {
        return this.aa(a);
    };
    SgmlReader.prototype.item1 = function (a) {
        return this.ab(a);
    };
    SgmlReader.prototype.item2 = function (a, b) {
        return this.ac(a, b);
    };
    SgmlReader.prototype.o = function (a) {
        var b = this._bg.i(a);
        if (b >= 0) {
            this.aw(b);
            return true;
        }
        return false;
    };
    SgmlReader.prototype.p = function (a, b) {
        return this.o(a);
    };
    SgmlReader.prototype.aw = function (a) {
        var b = this._bg.b(a);
        if (b != null) {
            this._b3 = a;
            this._a9 = b;
            if (this._bm != 3) {
                this._bg.e = this._bm;
            }
            this._bm = 3;
            return;
        }
        throw new IndexOutOfRangeException(0);
    };
    SgmlReader.prototype.r = function () {
        if (this._bg.h > 0) {
            this.aw(0);
            return true;
        }
        return false;
    };
    SgmlReader.prototype.s = function () {
        if (this._bm != 3 && this._bm != 4) {
            return this.r();
        }
        if (this._b3 < this._bg.h - 1) {
            this.aw(this._b3 + 1);
            return true;
        }
        return false;
    };
    SgmlReader.prototype.q = function () {
        if (this._bm == 3 || this._bm == 4) {
            this._bm = this._bg.e;
            this._a9 = null;
            return true;
        }
        return (this._bg.p == 1);
    };
    Object.defineProperty(SgmlReader.prototype, "_bp", {
        get: function () {
            return this._bo;
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype._cx = function () {
        if (this._bc == null) {
            this._c6();
        }
        return this._bc.ac();
    };
    SgmlReader.prototype._c6 = function () {
        this._c3(this._c0);
        if (this._cj != null) {
            this._bc = new Entity(0, "#document", null, this._ci, this._cm);
        }
        else if (this._b6 != null) {
            this._bc = new Entity(2, "#document", null, this._b6, this._cm);
        }
        else {
            throw new InvalidOperationException(1, "You must specify input either via Href or InputStream properties");
        }
        this._bc.d = this._bp;
        this._bc.ak(null, this._c0);
        if (this._bc.ae != null) {
            this._c0 = this._bc.ae;
        }
        if (this._bc.d && this._bk == null) {
            this._cf = "HTML";
            this._c3(this._c0);
        }
    };
    SgmlReader.prototype.t = function () {
        if (this._bc == null) {
            this._c6();
        }
        var a = this._bm;
        if (this._bg.g) {
            this._bg.g = false;
            this._bg = this._bj();
            this._bm = this._bg.e;
            return true;
        }
        var b = false;
        while (!b) {
            var t1 = this._bm;
            L0: while (true) {
                switch (t1) {
                    case 0:
                        this._bm = 1;
                        this._bc.k();
                        t1 = 1;
                        continue L0;
                    case 11:
                        if (this._bc.b != null) {
                            this._bc.af();
                            this._bc = this._bc.b;
                        }
                        else {
                            return false;
                        }
                        break;
                    case 2:
                        if (this._ca == this._bg.j) {
                            this._c8();
                            this._bm = 1;
                            t1 = 1;
                            continue L0;
                        }
                        this._c8();
                        b = true;
                        break;
                    case 1:
                        if (this._bg.f) {
                            this._c8();
                        }
                        var c = this._bg;
                        b = this._bv();
                        break;
                    case 6:
                        this._c8();
                        this._bm = 1;
                        b = this._by(this._b2);
                        break;
                    case 10:
                        b = this._bx('<');
                        break;
                    case 7:
                        this._c8();
                        if (this._be.c <= this._b4) {
                            this._bm = 1;
                            if (this._bf != null) {
                                this._bh(this._bf);
                                this._bf = null;
                                this._bm = 1;
                            }
                            else if (this._bg.p == 9) {
                                this._bm = 11;
                                t1 = 11;
                                continue L0;
                            }
                        }
                        b = true;
                        break;
                    case 8:
                        b = this._br();
                        break;
                    case 3:
                        t1 = 4;
                        continue L0;
                    case 4:
                        this._bm = 1;
                        t1 = 1;
                        continue L0;
                    case 5:
                        this._c8();
                        t1 = 1;
                        continue L0;
                    case 9:
                        if (this._bz(this._bc.j, false)) {
                            this._bg.p = 13;
                        }
                        b = true;
                        break;
                }
                break;
            }
            if (b && this._bg.p == 13 && this._de == 2) {
                b = false;
            }
            if (!b && this._bm == 11 && this._be.c > 1) {
                this._b4 = 1;
                this._bm = 7;
                this._bg = this._bj();
                return true;
            }
        }
        if (!this._bn && (this.a7 == 1 || this.a7 == 3 || this.a7 == 4)) {
            this._bn = true;
            if (this._bp && (this.a7 != 1 || stringCompare1(this.ag, "html", StringUtilities.j) != 0)) {
                this._bg.e = this._bm;
                var d = this._bi("html", 1, null);
                this._da();
                this._bg = d;
                d.g = true;
                d.f = false;
                this._bm = 1;
            }
            return true;
        }
        return true;
    };
    SgmlReader.prototype._bv = function () {
        var a = this._bc.j;
        if (a == '<') {
            a = this._bc.k();
            return this._by(a);
        }
        else if (a != "\uffff") {
            if (this._bg.c != null && this._bg.c.e.a == 1) {
                this._b2 = '\0';
                this._bm = 8;
                return false;
            }
            else if (this._bz(a, true)) {
                this._bg.p = 13;
            }
            return true;
        }
        this._bm = 11;
        return false;
    };
    SgmlReader.prototype._by = function (a) {
        if (a == '%') {
            return this._bq();
        }
        if (a == '!') {
            a = this._bc.k();
            if (a == '-') {
                return this._bs();
            }
            else if (a == '[') {
                return this._bt();
            }
            else if (a != '_' && !isLetter(a)) {
                var b = this._bc.y(this._cz, "Recovering", ">");
                this._c4("Ignoring invalid markup '<!" + b + ">");
                return false;
            }
            else {
                var c = this._bc.z(this._cz, SgmlReader._ce, false);
                if (c == "DOCTYPE") {
                    this._c7();
                    if (this.ab("SYSTEM") == null && this.ab("PUBLIC") != null) {
                        this._bg.a("SYSTEM", "", '\"', this._bb == 0);
                    }
                    if (this._b0) {
                        return false;
                    }
                    else {
                        this._bg.p = 10;
                        return true;
                    }
                }
                else {
                    this._c4("Invalid declaration '<!{0}...'.  Expecting '<!DOCTYPE' only.", c);
                    this._bc.y(null, "Recovering", ">");
                    return false;
                }
            }
        }
        else if (a == '?') {
            this._bc.k();
            return this._bw();
        }
        else if (a == '/') {
            return this._bu();
        }
        else {
            return this._bx(a);
        }
    };
    SgmlReader.prototype._cr = function (a) {
        var b = this._bc.z(this._cz, a, false);
        switch (this._bb) {
            case 1:
                b = b.toUpperCase();
                break;
            case 2:
                b = b.toLowerCase();
                break;
        }
        return this._dh.b(b);
    };
    SgmlReader.prototype._bx = function (a) {
        var b = null;
        if (this._bm != 10) {
            if (SgmlReader._cv.indexOf(a) >= 0) {
                this._cz.c = 0;
                this._cz.h('<');
                this._bm = 9;
                return false;
            }
            b = this._cr(SgmlReader._cv);
        }
        else {
            this._bm = 1;
        }
        var c = this._bi(b, 1, null);
        c.f = false;
        this._db(c);
        a = this._bc.l();
        this._dg.ac();
        while (a != "\uffff" && a != '>') {
            if (a == '/') {
                this._dg.h();
                c.f = true;
                a = this._bc.k();
                if (a != '>') {
                    this._c5("Expected empty start tag '/>' sequence instead of '{0}'", a);
                    this._bc.y(null, "Recovering", ">");
                    return false;
                }
                break;
            }
            else if (a == '<') {
                this._c4("Start tag '{0}' is missing '>'", b);
                break;
            }
            var d = this._cr(SgmlReader._cb);
            a = this._bc.l();
            if (d == "," || d == "=" || d == ":" || d == ";") {
                continue;
            }
            var e = null;
            var f = '\0';
            if (a == '=' || a == '\"' || a == '\'') {
                if (a == '=') {
                    this._bc.k();
                    a = this._bc.l();
                }
                if (a == '\'' || a == '\"') {
                    f = a;
                    e = this._cq(this._cz, a);
                }
                else if (a != '>') {
                    var g = SgmlReader._cc;
                    e = this._bc.z(this._cz, g, false);
                }
            }
            if (d.length > 0) {
                var h = c.a(d, e, f, this._bb == 0);
                if (h == null) {
                    this._c4("Duplicate attribute '{0}' ignored", d);
                }
                else {
                    this._dc(c, h);
                    if (startsWith1(h.e, "xmlns:", StringUtilities.j)) {
                        var i = h.e.substr(6);
                        this._dg.x(i, h.f);
                    }
                }
            }
            a = this._bc.l();
        }
        if (a == "\uffff") {
            this._bc.aj("Unexpected EOF parsing start tag '{0}'", b);
        }
        else if (a == '>') {
            this._bc.k();
        }
        if (this.x == 1) {
            if (this._b5 == 1) {
                this._bm = 11;
                return false;
            }
            this._b5++;
        }
        this._dd(c);
        return true;
    };
    SgmlReader.prototype._bu = function () {
        this._dg.h();
        this._bm = 2;
        this._bc.k();
        var a = this._cr(SgmlReader._cv);
        var b = this._bc.l();
        if (b != '>') {
            this._c5("Expected empty start tag '/>' sequence instead of '{0}'", b);
            this._bc.y(null, "Recovering", ">");
        }
        this._bc.k();
        this._ca = a;
        var c = (this._bb == 0);
        this._bg = this._be.item(this._be.c - 1);
        for (var d = this._be.c - 1; d > 0; d--) {
            var e = this._be.item(d);
            if (c && stringCompare1(e.j, a, 1) == 0) {
                this._ca = e.j;
                return true;
            }
            else if (e.j == a) {
                return true;
            }
        }
        this._c4("No matching start tag for '</{0}>'", a);
        this._bm = 1;
        return false;
    };
    SgmlReader.prototype._bq = function () {
        var a = "<%" + this._bc.y(this._cz, "AspNet", "%>") + "%>";
        this._bi(null, 4, a);
        return true;
    };
    SgmlReader.prototype._bs = function () {
        var a = this._bc.k();
        if (a != '-') {
            this._c5("Expecting comment '<!--' but found {0}", a);
            this._bc.y(null, "Comment", ">");
            return false;
        }
        var b = this._bc.y(this._cz, "Comment", "-->");
        var c = b.indexOf("--");
        while (c >= 0) {
            var d = c + 2;
            while (d < b.length && b.charAt(d) == '-') {
                d++;
            }
            if (c > 0) {
                b = b.substr(0, c - 1) + "-" + b.substr(d);
            }
            else {
                b = "-" + b.substr(d);
            }
            c = b.indexOf("--");
        }
        if (b.length > 0 && b.charAt(b.length - 1) == '-') {
            b += " ";
        }
        this._bi(null, 8, b);
        return true;
    };
    SgmlReader.prototype._bt = function () {
        var a = this._bc.k();
        a = this._bc.l();
        var b = this._bc.z(this._cz, SgmlReader._cd, false);
        if (b != "CDATA") {
            this._c4("Expecting CDATA but found '{0}'", b);
            this._bc.y(null, "CDATA", ">");
            return false;
        }
        a = this._bc.l();
        if (a != '[') {
            this._c5("Expecting '[' but found '{0}'", a);
            this._bc.y(null, "CDATA", ">");
            return false;
        }
        var c = this._bc.y(this._cz, "CDATA", "]]>");
        this._bi(null, 4, c);
        return true;
    };
    SgmlReader.prototype._c7 = function () {
        var a = this._bc.l();
        var b = this._cr(SgmlReader._ch);
        this._bi(b, 10, null);
        a = this._bc.l();
        if (a != '>') {
            var c = "";
            var d = "";
            var e = "";
            if (a != '[') {
                var f = this._bc.z(this._cz, SgmlReader._ch, false);
                if (f == "PUBLIC") {
                    a = this._bc.l();
                    if (a == '\"' || a == '\'') {
                        d = this._bc.x(this._cz, a);
                        this._bg.a(f, d, a, this._bb == 0);
                    }
                }
                else if (f != "SYSTEM") {
                    this._c4("Unexpected token in DOCTYPE '{0}'", f);
                    this._bc.y(null, "DOCTYPE", ">");
                }
                a = this._bc.l();
                if (a == '\"' || a == '\'') {
                    f = this._dh.b("SYSTEM");
                    e = this._bc.x(this._cz, a);
                    this._bg.a(f, e, a, this._bb == 0);
                }
                a = this._bc.l();
            }
            if (a == '[') {
                c = this._bc.y(this._cz, "Internal Subset", "]");
                this._bg.k = c;
            }
            a = this._bc.l();
            if (a != '>') {
                this._c5("Expecting end of DOCTYPE tag, but found '{0}'", a);
                this._bc.y(null, "DOCTYPE", ">");
            }
            if (this._bk == null) {
                this._cf = b;
                this._cn = d;
                this._ct = e;
                this._cs = c;
                this._c3(this._bc.ae);
            }
        }
        this._bc.k();
    };
    SgmlReader.prototype._bw = function () {
        var a = this._bc.z(this._cz, SgmlReader._cl, false);
        var b = null;
        if (this._bc.j != '?') {
            b = this._bc.y(this._cz, "Processing Instruction", ">");
        }
        else {
            b = this._bc.y(this._cz, "Processing Instruction", ">");
        }
        if (a != "xml") {
            this._bi(this._dh.b(a), 7, b);
            return true;
        }
        return false;
    };
    SgmlReader.prototype._bz = function (a, b) {
        var c = !b || this._bc.g;
        if (b) {
            this._cz.c = 0;
        }
        this._bm = 5;
        while (a != "\uffff") {
            if (a == '<') {
                a = this._bc.k();
                if (a == '/' || a == '!' || a == '?' || isLetter(a)) {
                    this._bm = 6;
                    this._b2 = a;
                    break;
                }
                else {
                    this._cz.h('<');
                    this._cz.h(a);
                    c = false;
                    a = this._bc.k();
                }
            }
            else if (a == '&') {
                this._c1(this._cz, '<');
                c = false;
                a = this._bc.j;
            }
            else {
                if (!this._bc.g) {
                    c = false;
                }
                this._cz.h(a);
                a = this._bc.k();
            }
        }
        var d = this._cz.toString();
        this._bi(null, 3, d);
        return c;
    };
    SgmlReader.prototype._cq = function (a, b) {
        a.c = 0;
        var c = this._bc.k();
        while (c != "\uffff" && c != b) {
            if (c == '&') {
                this._c1(this._cz, b);
                c = this._bc.j;
            }
            else {
                a.h(c);
                c = this._bc.k();
            }
        }
        this._bc.k();
        return a.toString();
    };
    SgmlReader.prototype._br = function () {
        var a = this._bc.g;
        this._cz.c = 0;
        var b = this._bc.j;
        if (this._b2 != '\0') {
            this._c8();
            switch (this._b2) {
                case '!':
                    this._b2 = ' ';
                    return this._bs();
                case '?':
                    this._b2 = ' ';
                    return this._bw();
                case '/':
                    this._bm = 2;
                    return true;
                case ' ': break;
            }
        }
        else {
            b = this._bc.k();
        }
        while (b != "\uffff") {
            if (b == '<') {
                b = this._bc.k();
                if (b == '!') {
                    b = this._bc.k();
                    if (b == '-') {
                        if (a) {
                            this._b2 = ' ';
                            return this._bs();
                        }
                        else {
                            this._b2 = '!';
                            break;
                        }
                    }
                    else {
                        this._cz.h('<');
                        this._cz.h('!');
                        this._cz.h(b);
                        a = false;
                    }
                }
                else if (b == '?') {
                    this._bc.k();
                    if (a) {
                        this._b2 = ' ';
                        return this._bw();
                    }
                    else {
                        this._b2 = '?';
                        break;
                    }
                }
                else if (b == '/') {
                    var c = this._cz.toString();
                    if (this._bu() && this._ca == this._bg.j) {
                        if (a || c == "") {
                            return true;
                        }
                        else {
                            this._b2 = '/';
                            this._cz.c = 0;
                            this._cz.l(c);
                            this._bm = 8;
                            break;
                        }
                    }
                    else {
                        this._cz.c = 0;
                        this._cz.l(c);
                        this._cz.l("</" + this._ca + ">");
                        a = false;
                    }
                }
                else {
                    this._cz.h('<');
                    this._cz.h(b);
                    a = false;
                }
            }
            else {
                if (!this._bc.g && a) {
                    a = false;
                }
                this._cz.h(b);
            }
            b = this._bc.k();
        }
        var d = this._cz.toString();
        this._bi(null, 4, d);
        if (this._b2 == '\0') {
            this._b2 = ' ';
        }
        return true;
    };
    SgmlReader.prototype._c1 = function (a, b) {
        var c = this._bc.k();
        if (c == '#') {
            var d = this._bc.s();
            a.l(d);
            c = this._bc.j;
        }
        else {
            this._cy.c = 0;
            while (c != "\uffff" && (isLetter(c) || c == '_' || c == '-')) {
                this._cy.h(c);
                c = this._bc.k();
            }
            var e = this._cy.toString();
            if (this._bk != null && e != "") {
                var f = this._bk.f(e);
                if (f != null) {
                    if (f.e) {
                        a.l(f.t);
                        if (c != b) {
                            c = this._bc.k();
                        }
                        return;
                    }
                    else {
                        var g = new Entity(0, e, f.w, f.aa, this._bc.v);
                        f.ak(this._bc, new Uri(0, f.aa));
                        this._bc = g;
                        this._bc.k();
                        return;
                    }
                }
                else {
                    this._c4("Undefined entity '{0}'", e);
                }
            }
            a.l("&");
            a.l(e);
            if (c != b) {
                a.h(c);
                c = this._bc.k();
            }
        }
    };
    SgmlReader.prototype.get_d = function () {
        return this._bm == 11;
    };
    Object.defineProperty(SgmlReader.prototype, "d", {
        get: function () {
            return this.get_d();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.au = function () {
        this.disposeCore(true);
    };
    SgmlReader.prototype.disposeCore = function (a) {
        if (a) {
            if (this._bc != null) {
                this._bc.af();
                this._bc = null;
            }
            if (this._b9 != null) {
                this._b9.i();
                this._b9 = null;
            }
        }
    };
    SgmlReader.prototype.get_a3 = function () {
        if (this._bm == 0) {
            return 0;
        }
        else if (this._bm == 11) {
            return 3;
        }
        return 1;
    };
    Object.defineProperty(SgmlReader.prototype, "a3", {
        get: function () {
            return this.get_a3();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.aq = function () {
        if (this._bg.p == 1) {
            this._cz.c = 0;
            while (this.t()) {
                switch (this.a7) {
                    case 4:
                    case 14:
                    case 13:
                    case 3:
                        this._cz.l(this._bg.k);
                        break;
                    default: return this._cz.toString();
                }
            }
            return this._cz.toString();
        }
        return this._bg.k;
    };
    SgmlReader.prototype.ao = function () {
        var a = new StringWriter(0);
        {
            var b = new XmlTextWriter(1, a);
            try {
                switch (this.a7) {
                    case 1:
                        this.t();
                        while (!this.d && this.a7 != 15) {
                            b.ac(this, true);
                        }
                        this.t();
                        break;
                    case 2:
                        a.v(this.ar);
                        break;
                    default: break;
                }
            }
            finally {
                if (b != null) {
                    b.dispose();
                }
            }
        }
        return a.toString();
    };
    SgmlReader.prototype.ap = function () {
        var a = new StringWriter(0);
        {
            var b = new XmlTextWriter(1, a);
            try {
                b.ac(this, true);
            }
            finally {
                if (b != null) {
                    b.dispose();
                }
            }
        }
        return a.toString();
    };
    SgmlReader.prototype.get_a5 = function () {
        return this._dh;
    };
    Object.defineProperty(SgmlReader.prototype, "a5", {
        get: function () {
            return this.get_a5();
        },
        enumerable: false,
        configurable: true
    });
    SgmlReader.prototype.ah = function (a) {
        return this._dg.lookupNamespace(a);
    };
    SgmlReader.prototype.a1 = function () {
        throw new InvalidOperationException(1, "Not on an entity reference.");
    };
    SgmlReader.prototype.u = function () {
        if (this._bm == 3) {
            this._bm = 4;
            return true;
        }
        else if (this._bm == 4) {
            return false;
        }
        throw new InvalidOperationException(1, "Not on an attribute.");
    };
    SgmlReader.prototype._db = function (a) {
        if (this._bk != null) {
            var b = this._bk.d(a.j);
            if (b != null) {
                a.c = b;
                if (b.e.a == 3) {
                    a.f = true;
                }
            }
        }
    };
    SgmlReader.prototype._dc = function (a, b) {
        var c = a.c;
        if (c != null) {
            var d = c.c(b.e);
            if (d != null) {
                b.a = d;
            }
        }
    };
    SgmlReader.prototype._dd = function (a) {
        if (this._bk != null) {
            var b = this._dh.b(a.j.toUpperCase());
            var c = 0;
            var d = this._be.c - 2;
            if (a.c != null) {
                for (c = d; c > 0; c--) {
                    var e = this._be.item(c);
                    if (e.f) {
                        continue;
                    }
                    var f = e.c;
                    if (f != null) {
                        if (f.i == this._bk.r) {
                            break;
                        }
                        if (f.f(b, this._bk)) {
                            break;
                        }
                        else if (!f.g) {
                            break;
                        }
                    }
                    else {
                        break;
                    }
                }
            }
            if (c == 0) {
            }
            else if (c < d) {
                var g = this._be.item(d);
                if (c == d - 1 && b == g.j) {
                }
                else {
                    var h = "";
                    for (var i = d; i >= c + 1; i--) {
                        if (h != "") {
                            h += ",";
                        }
                        var j = this._be.item(i);
                        h += "<" + j.j + ">";
                    }
                    this._c4("Element '{0}' not allowed inside '{1}', closing {2}.", b, g.j, h);
                }
                this._bm = 7;
                this._bf = a;
                this._c8();
                this._b4 = c + 1;
            }
        }
    };
    SgmlReader.$t = markType(SgmlReader, 'SgmlReader', XmlReader.$);
    SgmlReader._ce = " \t\r\n><";
    SgmlReader._cv = " \t\r\n=/><";
    SgmlReader._cb = " \t\r\n='\"/>";
    SgmlReader._cc = " \t\r\n>";
    SgmlReader._cd = "\t\r\n[<>";
    SgmlReader._ch = " \t\r\n>";
    SgmlReader._cl = " \t\r\n?";
    return SgmlReader;
}(XmlReader));
export { SgmlReader };
