// igniteui-react-excel/esm5/lib/StyleInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, toNullable, Number_$type, typeCast, markType } from "igniteui-react-core";
import { WorkbookBuiltInStyle } from "./WorkbookBuiltInStyle";
import { Convert } from "igniteui-react-core";
import { stringEmpty } from "igniteui-react-core";
/**
 * @hidden
 */
var StyleInfo = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(StyleInfo, _super);
    function StyleInfo(a) {
        var _rest = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            _rest[_i - 1] = arguments[_i];
        }
        var _this = _super.call(this) || this;
        _this.g = toNullable(Number_$type, null);
        _this.a = false;
        _this.c = false;
        _this.i = toNullable(Number_$type, 0);
        _this.k = stringEmpty();
        _this.f = -1;
        a = (a == void 0) ? 0 : a;
        switch (a) {
            case 0: break;
            case 1:
                {
                    var c = _rest[0];
                    var d = _rest[1];
                    _this.l = d.name;
                    _this.e = c._a9(d);
                    var e = typeCast(WorkbookBuiltInStyle.$, d);
                    if (e != null) {
                        if (e._ak == 255) {
                            _this.j = toNullable(Number_$type, null);
                        }
                        else {
                            _this.j = toNullable(Number_$type, Convert.toUInt32(e._ak));
                        }
                        _this.h = toNullable(Number_$type, e._ac);
                        _this.b = e._ah;
                    }
                    _this.d = d._m;
                }
                break;
        }
        return _this;
    }
    Object.defineProperty(StyleInfo.prototype, "h", {
        get: function () {
            return this.g;
        },
        set: function (a) {
            this.g = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StyleInfo.prototype, "b", {
        get: function () {
            return this.a;
        },
        set: function (a) {
            this.a = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StyleInfo.prototype, "d", {
        get: function () {
            return this.c;
        },
        set: function (a) {
            this.c = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StyleInfo.prototype, "j", {
        get: function () {
            return this.i;
        },
        set: function (a) {
            this.i = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StyleInfo.prototype, "l", {
        get: function () {
            return this.k;
        },
        set: function (a) {
            this.k = a;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(StyleInfo.prototype, "e", {
        get: function () {
            return this.f;
        },
        set: function (a) {
            this.f = a;
        },
        enumerable: false,
        configurable: true
    });
    StyleInfo.$t = markType(StyleInfo, 'StyleInfo');
    return StyleInfo;
}(Base));
export { StyleInfo };
