// igniteui-react-excel/esm5/lib/TextDirection.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Determines the direction in which text flows.
 */
export var TextDirection = /*@__PURE__*/ (function (TextDirection) {
    /**
     * Text flows along the horizontal plane. Note that under this setting,
     * text rotation is also supported, with the angle of rotation being
     * relative to the horizontal plane.
     */
    TextDirection[TextDirection["Horizontal"] = 0] = "Horizontal";
    /**
     * Text flows along the vertical plane. The plane is defined by rotating
     * the horizontal axis by 90 degrees in the clockwise direction.
     */
    TextDirection[TextDirection["Vertical"] = 1] = "Vertical";
    /**
     * Text flows along the vertical plane. The plane is defined by rotating
     * the horizontal axis by 270 degrees in the clockwise direction.
     */
    TextDirection[TextDirection["Vertical270"] = 2] = "Vertical270";
    /**
     * Text flows along the vertical plane with East Asian vertical orientation.
     */
    TextDirection[TextDirection["EastAsianVertical"] = 3] = "EastAsianVertical";
    /**
     * Text flows along the vertical plane with Mongolian vertical orientation.
     */
    TextDirection[TextDirection["MongolianVertical"] = 4] = "MongolianVertical";
    /**
     * Text flows along the vertical plane with WordArt vertical orientation.
     */
    TextDirection[TextDirection["WordArtVertical"] = 5] = "WordArtVertical";
    /**
     * Similar to <i>WordArtVertical</i>, with a right-to-left text reading direction.
     */
    TextDirection[TextDirection["WordArtVerticalRtl"] = 6] = "WordArtVerticalRtl";
    return TextDirection;
})({});
/**
 * @hidden
 */
export var TextDirection_$type = /*@__PURE__*/ markEnum('TextDirection', 'Horizontal,0|Vertical,1|Vertical270,2|EastAsianVertical,3|MongolianVertical,4|WordArtVertical,5|WordArtVerticalRtl,6');
