// igniteui-react-excel/esm5/lib/TickMark.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which determine the placement of an axis' <see cref="Axis.MajorTickMark">major</see>
 * and <see cref="Axis.MinorTickMark">minor</see> tick marks.
 */
export var TickMark = /*@__PURE__*/ (function (TickMark) {
    /**
     * The tick marks cross the axis line.
     */
    TickMark[TickMark["Cross"] = 4] = "Cross";
    /**
     * The tick marks appear on the side of the axis line
     * closest to the inside of the plot area.
     */
    TickMark[TickMark["Inside"] = 2] = "Inside";
    /**
     * Tick marks are not displayed.
     */
    TickMark[TickMark["None"] = 4142] = "None";
    /**
     * The tick marks appear on the side of the axis line
     * closest to the outside of the plot area.
     */
    TickMark[TickMark["Outside"] = 3] = "Outside";
    return TickMark;
})({});
/**
 * @hidden
 */
export var TickMark_$type = /*@__PURE__*/ markEnum('TickMark', 'Cross,4|Inside,2|None,4142|Outside,3');
