// igniteui-react-excel/esm5/lib/TiffPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { ImagePartManagerBase } from "./ImagePartManagerBase";
import { ImageFormat } from "./ImageFormat";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var TiffPartManager = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(TiffPartManager, _super);
    function TiffPartManager() {
        return _super.call(this) || this;
    }
    TiffPartManager.prototype.get_s = function () {
        return "/xl/media/image.tiff";
    };
    Object.defineProperty(TiffPartManager.prototype, "s", {
        get: function () {
            return this.get_s();
        },
        enumerable: false,
        configurable: true
    });
    TiffPartManager.prototype.get_t = function () {
        return "/ppt/media/image.tiff";
    };
    Object.defineProperty(TiffPartManager.prototype, "t", {
        get: function () {
            return this.get_t();
        },
        enumerable: false,
        configurable: true
    });
    TiffPartManager.prototype.get_u = function () {
        return "/word/media/image.tiff";
    };
    Object.defineProperty(TiffPartManager.prototype, "u", {
        get: function () {
            return this.get_u();
        },
        enumerable: false,
        configurable: true
    });
    TiffPartManager.prototype.get_m = function () {
        return "image/tiff";
    };
    Object.defineProperty(TiffPartManager.prototype, "m", {
        get: function () {
            return this.get_m();
        },
        enumerable: false,
        configurable: true
    });
    TiffPartManager.prototype.r = function (a) {
        return ImageFormat.h;
    };
    TiffPartManager.$t = markType(TiffPartManager, 'TiffPartManager', ImagePartManagerBase.$);
    TiffPartManager.y = new TiffPartManager();
    return TiffPartManager;
}(ImagePartManagerBase));
export { TiffPartManager };
