// igniteui-react-excel/esm5/lib/TopOrBottomFilterType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various filter types available for the [[TopOrBottomFilter]].
 * @see [[TopOrBottomFilter.type]]
 */
export var TopOrBottomFilterType = /*@__PURE__*/ (function (TopOrBottomFilterType) {
    /**
     * Filter in the top N values in the sorted list of values.
     */
    TopOrBottomFilterType[TopOrBottomFilterType["TopValues"] = 0] = "TopValues";
    /**
     * Filter in the bottom N values in the sorted list of values.
     */
    TopOrBottomFilterType[TopOrBottomFilterType["BottomValues"] = 1] = "BottomValues";
    /**
     * Filter in the top N percent of values the sorted list of values.
     */
    TopOrBottomFilterType[TopOrBottomFilterType["TopPercentage"] = 2] = "TopPercentage";
    /**
     * Filter in the bottom N percent of values the sorted list of values.
     */
    TopOrBottomFilterType[TopOrBottomFilterType["BottomPercentage"] = 3] = "BottomPercentage";
    return TopOrBottomFilterType;
})({});
/**
 * @hidden
 */
export var TopOrBottomFilterType_$type = /*@__PURE__*/ markEnum('TopOrBottomFilterType', 'TopValues,0|BottomValues,1|TopPercentage,2|BottomPercentage,3');
