// igniteui-react-excel/esm5/lib/TrendlinePolynomialOrder.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which define the order/degree of the polynomial
 * applied to a [[Trendline]] when the trendline
 * type is <i>Polynomial</i>.
 * @see [[TrendlineType]]
 */
export var TrendlinePolynomialOrder = /*@__PURE__*/ (function (TrendlinePolynomialOrder) {
    /**
     * A second-order/degree polynomial.
     */
    TrendlinePolynomialOrder[TrendlinePolynomialOrder["Second"] = 2] = "Second";
    /**
     * A third-order/degree polynomial.
     */
    TrendlinePolynomialOrder[TrendlinePolynomialOrder["Third"] = 3] = "Third";
    /**
     * A fourth-order/degree polynomial.
     */
    TrendlinePolynomialOrder[TrendlinePolynomialOrder["Fourth"] = 4] = "Fourth";
    /**
     * A fifth-order/degree polynomial.
     */
    TrendlinePolynomialOrder[TrendlinePolynomialOrder["Fifth"] = 5] = "Fifth";
    /**
     * A sixth-order/degree polynomial.
     */
    TrendlinePolynomialOrder[TrendlinePolynomialOrder["Sixth"] = 6] = "Sixth";
    return TrendlinePolynomialOrder;
})({});
/**
 * @hidden
 */
export var TrendlinePolynomialOrder_$type = /*@__PURE__*/ markEnum('TrendlinePolynomialOrder', 'Second,2|Third,3|Fourth,4|Fifth,5|Sixth,6');
