// igniteui-react-excel/esm5/lib/UltraCalcFunctionDateValue.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionDateValue = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionDateValue, _super);
    function UltraCalcFunctionDateValue() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionDateValue.prototype.evaluate = function (a, b) {
        var c = a.pop();
        var d;
        if (((function () { var e = c._aj(c._a0, d); d = e.p1; return e.ret; })()) == false) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(1));
        }
        return new ExcelCalcValue(d);
    };
    UltraCalcFunctionDateValue.prototype.get_name = function () {
        return "datevalue";
    };
    Object.defineProperty(UltraCalcFunctionDateValue.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionDateValue.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionDateValue.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionDateValue.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionDateValue.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionDateValue.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionDateValue.$t = markType(UltraCalcFunctionDateValue, 'UltraCalcFunctionDateValue', BuiltInFunctionBase.$);
    return UltraCalcFunctionDateValue;
}(BuiltInFunctionBase));
export { UltraCalcFunctionDateValue };
