// igniteui-react-excel/esm5/lib/UltraCalcFunctionFactDouble.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { UltraCalcFunctionFact } from "./UltraCalcFunctionFact";
import { markType } from "igniteui-react-core";
import { truncate, isInfinity } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionFactDouble = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionFactDouble, _super);
    function UltraCalcFunctionFactDouble() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionFactDouble.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var d = c._toDecimal2();
        if (d < 0) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        var e = truncate(d);
        var f = UltraCalcFunctionFact._ak(e, 2);
        if (isInfinity(f)) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        return new ExcelCalcValue(f);
    };
    UltraCalcFunctionFactDouble.prototype.get_name = function () {
        return "factdouble";
    };
    Object.defineProperty(UltraCalcFunctionFactDouble.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionFactDouble.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionFactDouble.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionFactDouble.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionFactDouble.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionFactDouble.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionFactDouble.$t = markType(UltraCalcFunctionFactDouble, 'UltraCalcFunctionFactDouble', BuiltInFunctionBase.$);
    return UltraCalcFunctionFactDouble;
}(BuiltInFunctionBase));
export { UltraCalcFunctionFactDouble };
