// igniteui-react-excel/esm5/lib/UltraCalcFunctionTrunc.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionTrunc = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionTrunc, _super);
    function UltraCalcFunctionTrunc() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionTrunc.prototype.evaluate = function (a, b) {
        var c = 0;
        if (b == 2) {
            var d = a.pop();
            if (d.isError) {
                return new ExcelCalcValue(d.toErrorValue());
            }
            c = d._toInt321();
        }
        var e;
        var f = a.pop();
        if (f.isError) {
            return new ExcelCalcValue(f.toErrorValue());
        }
        if (!((function () { var g = f._toDouble1(e); e = g.p0; return g.ret; })())) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        var g = Math.pow(10, c);
        var h = e * g;
        h = h >= 0 ? Math.floor(h) : Math.ceil(h);
        h /= g;
        return new ExcelCalcValue(h);
    };
    UltraCalcFunctionTrunc.prototype.get_name = function () {
        return "trunc";
    };
    Object.defineProperty(UltraCalcFunctionTrunc.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionTrunc.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionTrunc.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionTrunc.prototype.get_maxArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionTrunc.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionTrunc.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionTrunc.$t = markType(UltraCalcFunctionTrunc, 'UltraCalcFunctionTrunc', BuiltInFunctionBase.$);
    return UltraCalcFunctionTrunc;
}(BuiltInFunctionBase));
export { UltraCalcFunctionTrunc };
