// igniteui-react-excel/esm5/lib/UltraCalcFunctionEven.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
import { ieeeRemainder } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionEven = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionEven, _super);
    function UltraCalcFunctionEven() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionEven.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var d;
        if (!((function () { var e = c._toDouble1(d); d = e.p0; return e.ret; })())) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        var e = Math.ceil(Math.abs(d));
        var f = e + Math.abs(ieeeRemainder(e, 2));
        if (d < 0) {
            f *= -1;
        }
        return new ExcelCalcValue(f);
    };
    UltraCalcFunctionEven.prototype.get_name = function () {
        return "even";
    };
    Object.defineProperty(UltraCalcFunctionEven.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionEven.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionEven.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionEven.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionEven.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionEven.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionEven.$t = markType(UltraCalcFunctionEven, 'UltraCalcFunctionEven', BuiltInFunctionBase.$);
    return UltraCalcFunctionEven;
}(BuiltInFunctionBase));
export { UltraCalcFunctionEven };
