// igniteui-react-excel/esm5/lib/UltraCalcFunctionMedian.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { List$1 } from "igniteui-react-core";
import { ExcelCalcFunction } from "./ExcelCalcFunction";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { Number_$type, markType } from "igniteui-react-core";
import { intDivide } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionMedian = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionMedian, _super);
    function UltraCalcFunctionMedian() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionMedian.prototype.evaluate = function (a, b) {
        var c = new List$1(Number_$type, 0);
        if (!ExcelCalcFunction._t(a, b, c, true, true)) {
            return a.pop();
        }
        c.y();
        if (c.count == 0) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        var d;
        var e = intDivide(c.count, 2);
        if (c.count % 2 == 1) {
            d = c._inner[e];
        }
        else {
            var f = (c._inner[e - 1] + c._inner[e]) / 2;
            d = f;
        }
        return new ExcelCalcValue(d);
    };
    UltraCalcFunctionMedian.prototype.canParameterBeEnumerable = function (a) {
        return true;
    };
    UltraCalcFunctionMedian.prototype.get_name = function () {
        return "median";
    };
    Object.defineProperty(UltraCalcFunctionMedian.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMedian.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionMedian.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMedian.prototype.get_maxArgs = function () {
        return 0x7FFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionMedian.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMedian.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionMedian.$t = markType(UltraCalcFunctionMedian, 'UltraCalcFunctionMedian', BuiltInFunctionBase.$);
    return UltraCalcFunctionMedian;
}(BuiltInFunctionBase));
export { UltraCalcFunctionMedian };
