// igniteui-react-excel/esm5/lib/UltraCalcFunctionTrim.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { StringBuilder } from "igniteui-react-core";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionTrim = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionTrim, _super);
    function UltraCalcFunctionTrim() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionTrim.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var d = c.toString().trim();
        var e = new StringBuilder(1, d.length);
        var f = false;
        for (var g = 0; g < d.length; g++) {
            var h = d.charAt(g);
            var i = /\s/i.test(h);
            if (!f || !i) {
                e.h(h);
            }
            f = i;
        }
        return new ExcelCalcValue(e.toString());
    };
    UltraCalcFunctionTrim.prototype.get_name = function () {
        return "trim";
    };
    Object.defineProperty(UltraCalcFunctionTrim.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionTrim.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionTrim.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionTrim.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionTrim.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionTrim.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionTrim.$t = markType(UltraCalcFunctionTrim, 'UltraCalcFunctionTrim', BuiltInFunctionBase.$);
    return UltraCalcFunctionTrim;
}(BuiltInFunctionBase));
export { UltraCalcFunctionTrim };
