// igniteui-react-excel/esm5/lib/UltraCalcFunctionRoman.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { typeCast, Boolean_$type, markType } from "igniteui-react-core";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { UltraCalcFunctionRoman_RomanNumeral } from "./UltraCalcFunctionRoman_RomanNumeral";
/**
 * @hidden
 */
var UltraCalcFunctionRoman = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionRoman, _super);
    function UltraCalcFunctionRoman() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionRoman.prototype.evaluate = function (a, b) {
        var c;
        var d;
        if (b == 2) {
            d = a.pop();
            if (d.isError) {
                return new ExcelCalcValue(d.toErrorValue());
            }
            var e = d._getResolvedValue();
            if (typeCast(Boolean_$type, e) !== null) {
                c = e ? 0 : 4;
            }
            else {
                c = d._toInt();
            }
        }
        else {
            c = 0;
        }
        d = a.pop();
        if (d.isError) {
            return new ExcelCalcValue(d.toErrorValue());
        }
        var f = d._toInt();
        if (f < 0 || f > 3999) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(1));
        }
        var g = new UltraCalcFunctionRoman_RomanNumeral(f);
        return new ExcelCalcValue(g.p(c));
    };
    UltraCalcFunctionRoman.prototype.get_name = function () {
        return "roman";
    };
    Object.defineProperty(UltraCalcFunctionRoman.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionRoman.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionRoman.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionRoman.prototype.get_maxArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionRoman.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionRoman.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionRoman.$t = markType(UltraCalcFunctionRoman, 'UltraCalcFunctionRoman', BuiltInFunctionBase.$);
    return UltraCalcFunctionRoman;
}(BuiltInFunctionBase));
export { UltraCalcFunctionRoman };
