// igniteui-react-excel/esm5/lib/UltraCalcFunctionCountA.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionCountA = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionCountA, _super);
    function UltraCalcFunctionCountA() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionCountA.prototype.evaluate = function (a, b) {
        return CalcFunctionUtilities.k(a, b, 673);
    };
    UltraCalcFunctionCountA.prototype.canParameterBeEnumerable = function (a) {
        return true;
    };
    UltraCalcFunctionCountA.prototype.get_name = function () {
        return "counta";
    };
    Object.defineProperty(UltraCalcFunctionCountA.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionCountA.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionCountA.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionCountA.prototype.get_maxArgs = function () {
        return 0x7FFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionCountA.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionCountA.$t = markType(UltraCalcFunctionCountA, 'UltraCalcFunctionCountA', BuiltInFunctionBase.$);
    return UltraCalcFunctionCountA;
}(BuiltInFunctionBase));
export { UltraCalcFunctionCountA };
