// igniteui-react-excel/esm5/lib/UltraCalcFunctionYear.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionYear = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionYear, _super);
    function UltraCalcFunctionYear() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionYear.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        if (true) {
            var d_1;
            if (c.isNull || (((function () { var e = c._toDouble1(d_1); d_1 = e.p0; return e.ret; })()) && d_1 == 0)) {
                var e = a._r;
                if (e != null && e.dateSystem == 1) {
                    return new ExcelCalcValue(1904);
                }
                return new ExcelCalcValue(1900);
            }
        }
        var f = c._toDateTime1();
        return new ExcelCalcValue(f.getFullYear());
    };
    UltraCalcFunctionYear.prototype.get_name = function () {
        return "year";
    };
    Object.defineProperty(UltraCalcFunctionYear.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionYear.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionYear.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionYear.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionYear.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionYear.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionYear.$t = markType(UltraCalcFunctionYear, 'UltraCalcFunctionYear', BuiltInFunctionBase.$);
    return UltraCalcFunctionYear;
}(BuiltInFunctionBase));
export { UltraCalcFunctionYear };
