// igniteui-react-excel/esm5/lib/UltraCalcFunctionDegrees.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionDegrees = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionDegrees, _super);
    function UltraCalcFunctionDegrees() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionDegrees.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var d = c._toDouble3();
        var e = UltraCalcFunctionDegrees._ak(d);
        return new ExcelCalcValue(e);
    };
    UltraCalcFunctionDegrees.prototype.get_name = function () {
        return "degrees";
    };
    Object.defineProperty(UltraCalcFunctionDegrees.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionDegrees.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionDegrees.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionDegrees.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionDegrees.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionDegrees._ak = function (a) {
        var b = 180 / Math.PI;
        return a * b;
    };
    UltraCalcFunctionDegrees.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionDegrees.$t = markType(UltraCalcFunctionDegrees, 'UltraCalcFunctionDegrees', BuiltInFunctionBase.$);
    return UltraCalcFunctionDegrees;
}(BuiltInFunctionBase));
export { UltraCalcFunctionDegrees };
