// igniteui-react-excel/esm5/lib/UltraCalcFunctionHex2Oct.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { UltraCalcFunctionXBase2XBase } from "./UltraCalcFunctionXBase2XBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionHex2Oct = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionHex2Oct, _super);
    function UltraCalcFunctionHex2Oct() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionHex2Oct.prototype.get_name = function () {
        return "hex2oct";
    };
    Object.defineProperty(UltraCalcFunctionHex2Oct.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Oct.prototype.get__aq = function () {
        return -549755813888;
    };
    Object.defineProperty(UltraCalcFunctionHex2Oct.prototype, "_aq", {
        get: function () {
            return this.get__aq();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Oct.prototype.get__ap = function () {
        return 0x7FFFFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionHex2Oct.prototype, "_ap", {
        get: function () {
            return this.get__ap();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Oct.prototype.get__ao = function () {
        return -536870912;
    };
    Object.defineProperty(UltraCalcFunctionHex2Oct.prototype, "_ao", {
        get: function () {
            return this.get__ao();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Oct.prototype.get__an = function () {
        return 536870911;
    };
    Object.defineProperty(UltraCalcFunctionHex2Oct.prototype, "_an", {
        get: function () {
            return this.get__an();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Oct.prototype.get__al = function () {
        return 16;
    };
    Object.defineProperty(UltraCalcFunctionHex2Oct.prototype, "_al", {
        get: function () {
            return this.get__al();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Oct.prototype.get__ak = function () {
        return 8;
    };
    Object.defineProperty(UltraCalcFunctionHex2Oct.prototype, "_ak", {
        get: function () {
            return this.get__ak();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Oct.prototype.get__am = function () {
        return 10;
    };
    Object.defineProperty(UltraCalcFunctionHex2Oct.prototype, "_am", {
        get: function () {
            return this.get__am();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Oct.$t = markType(UltraCalcFunctionHex2Oct, 'UltraCalcFunctionHex2Oct', UltraCalcFunctionXBase2XBase.$);
    return UltraCalcFunctionHex2Oct;
}(UltraCalcFunctionXBase2XBase));
export { UltraCalcFunctionHex2Oct };
