// igniteui-react-excel/esm5/lib/UltraCalcFunctionLower.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { stringToLocaleLower } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionLower = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionLower, _super);
    function UltraCalcFunctionLower() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionLower.prototype.evaluate = function (a, b) {
        if (true) {
            var c = a.pop();
            if (c.isError) {
                return c;
            }
            return new ExcelCalcValue(c.toString().toLowerCase());
        }
        if (a.peek().isString) {
            return new ExcelCalcValue(stringToLocaleLower(a.pop().toString(), a._w));
        }
        else {
            return a.pop();
        }
    };
    UltraCalcFunctionLower.prototype.get_name = function () {
        return "lower";
    };
    Object.defineProperty(UltraCalcFunctionLower.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionLower.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionLower.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionLower.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionLower.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionLower.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionLower.$t = markType(UltraCalcFunctionLower, 'UltraCalcFunctionLower', BuiltInFunctionBase.$);
    return UltraCalcFunctionLower;
}(BuiltInFunctionBase));
export { UltraCalcFunctionLower };
