// igniteui-react-excel/esm5/lib/UltraCalcFunctionUpper.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { stringToLocaleUpper } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionUpper = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionUpper, _super);
    function UltraCalcFunctionUpper() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionUpper.prototype.evaluate = function (a, b) {
        if (true) {
            var c = a.pop();
            if (c.isError) {
                return c;
            }
            return new ExcelCalcValue(c.toString().toUpperCase());
        }
        if (a.peek().isString) {
            return new ExcelCalcValue(stringToLocaleUpper(a.pop().toString(), a._w));
        }
        else {
            return a.pop();
        }
    };
    UltraCalcFunctionUpper.prototype.get_name = function () {
        return "upper";
    };
    Object.defineProperty(UltraCalcFunctionUpper.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionUpper.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionUpper.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionUpper.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionUpper.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionUpper.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionUpper.$t = markType(UltraCalcFunctionUpper, 'UltraCalcFunctionUpper', BuiltInFunctionBase.$);
    return UltraCalcFunctionUpper;
}(BuiltInFunctionBase));
export { UltraCalcFunctionUpper };
