// igniteui-react-excel/esm5/lib/UltraCalcFunctionSum.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
/**
 * @hidden
 */
var UltraCalcFunctionSum = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionSum, _super);
    function UltraCalcFunctionSum() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionSum.prototype.evaluate = function (a, b) {
        var c = 0;
        return CalcFunctionUtilities.f(a, b, CalcFunctionUtilities.b(a, false, false, false), function (d) { return c += d; }) || new ExcelCalcValue(c);
    };
    UltraCalcFunctionSum.prototype.canParameterBeEnumerable = function (a) {
        return true;
    };
    UltraCalcFunctionSum.prototype.get_name = function () {
        return "sum";
    };
    Object.defineProperty(UltraCalcFunctionSum.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSum.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionSum.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSum.prototype.get_maxArgs = function () {
        return 0x7FFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionSum.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSum.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionSum.$t = markType(UltraCalcFunctionSum, 'UltraCalcFunctionSum', BuiltInFunctionBase.$);
    return UltraCalcFunctionSum;
}(BuiltInFunctionBase));
export { UltraCalcFunctionSum };
