// igniteui-react-excel/esm5/lib/UltraCalcFunctionVar.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
var UltraCalcFunctionVar = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionVar, _super);
    function UltraCalcFunctionVar() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionVar.prototype.evaluate = function (a, b) {
        return CalcFunctionUtilities.d(a, b, 1229, function (c) { return UltraCalcFunctionVar._ak(c, true); });
    };
    UltraCalcFunctionVar.prototype.canParameterBeEnumerable = function (a) {
        return true;
    };
    UltraCalcFunctionVar._ak = function (a, b) {
        if (a.count < 2) {
            if (!b && a.count == 1) {
                return new ExcelCalcValue(0);
            }
            return new ExcelCalcValue(new ExcelCalcErrorValue(2));
        }
        return new ExcelCalcValue(CalcFunctionUtilities.ab(a, b));
    };
    UltraCalcFunctionVar.prototype.get_name = function () {
        return "var";
    };
    Object.defineProperty(UltraCalcFunctionVar.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionVar.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionVar.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionVar.prototype.get_maxArgs = function () {
        return 0x7FFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionVar.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionVar.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionVar.$t = markType(UltraCalcFunctionVar, 'UltraCalcFunctionVar', BuiltInFunctionBase.$);
    return UltraCalcFunctionVar;
}(BuiltInFunctionBase));
export { UltraCalcFunctionVar };
