// igniteui-react-excel/esm5/lib/UltraCalcFunctionSign.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionSign = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionSign, _super);
    function UltraCalcFunctionSign() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionSign.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var d = c._toDouble3();
        var e;
        if (d < 0) {
            e = -1;
        }
        else if (d > 0) {
            e = 1;
        }
        else {
            e = 0;
        }
        return new ExcelCalcValue(e);
    };
    UltraCalcFunctionSign.prototype.get_name = function () {
        return "sign";
    };
    Object.defineProperty(UltraCalcFunctionSign.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSign.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionSign.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSign.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionSign.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSign.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionSign.$t = markType(UltraCalcFunctionSign, 'UltraCalcFunctionSign', BuiltInFunctionBase.$);
    return UltraCalcFunctionSign;
}(BuiltInFunctionBase));
export { UltraCalcFunctionSign };
