// igniteui-react-excel/esm5/lib/UltraCalcFunctionHex2Dec.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { UltraCalcFunctionXBase2Dec } from "./UltraCalcFunctionXBase2Dec";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionHex2Dec = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionHex2Dec, _super);
    function UltraCalcFunctionHex2Dec() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionHex2Dec.prototype.get_name = function () {
        return "hex2dec";
    };
    Object.defineProperty(UltraCalcFunctionHex2Dec.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Dec.prototype.get__ao = function () {
        return -549755813888;
    };
    Object.defineProperty(UltraCalcFunctionHex2Dec.prototype, "_ao", {
        get: function () {
            return this.get__ao();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Dec.prototype.get__an = function () {
        return 0x7FFFFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionHex2Dec.prototype, "_an", {
        get: function () {
            return this.get__an();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Dec.prototype.get__ak = function () {
        return 16;
    };
    Object.defineProperty(UltraCalcFunctionHex2Dec.prototype, "_ak", {
        get: function () {
            return this.get__ak();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Dec.prototype.get__am = function () {
        return 10;
    };
    Object.defineProperty(UltraCalcFunctionHex2Dec.prototype, "_am", {
        get: function () {
            return this.get__am();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Dec.$t = markType(UltraCalcFunctionHex2Dec, 'UltraCalcFunctionHex2Dec', UltraCalcFunctionXBase2Dec.$);
    return UltraCalcFunctionHex2Dec;
}(UltraCalcFunctionXBase2Dec));
export { UltraCalcFunctionHex2Dec };
