// igniteui-react-excel/esm5/lib/UltraCalcFunctionLog10.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
import { log10 } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionLog10 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionLog10, _super);
    function UltraCalcFunctionLog10() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionLog10.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var d;
        if (!((function () { var e = c._toDouble1(d); d = e.p0; return e.ret; })())) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(1));
        }
        if (true && d == 0) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        return new ExcelCalcValue(log10(d));
    };
    UltraCalcFunctionLog10.prototype.get_name = function () {
        return "log10";
    };
    Object.defineProperty(UltraCalcFunctionLog10.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionLog10.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionLog10.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionLog10.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionLog10.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionLog10.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionLog10.$t = markType(UltraCalcFunctionLog10, 'UltraCalcFunctionLog10', BuiltInFunctionBase.$);
    return UltraCalcFunctionLog10;
}(BuiltInFunctionBase));
export { UltraCalcFunctionLog10 };
