// igniteui-react-excel/esm5/lib/UltraCalcFunctionCotH.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
var UltraCalcFunctionCotH = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionCotH, _super);
    function UltraCalcFunctionCotH() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionCotH.prototype.evaluate = function (a, b) {
        return CalcFunctionUtilities.h(a, true, function (c) {
            if (c <= -134217728 || c >= 134217728) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            return new ExcelCalcValue(1 / Math.tanh(c));
        });
    };
    UltraCalcFunctionCotH.prototype.get_name = function () {
        return "coth";
    };
    Object.defineProperty(UltraCalcFunctionCotH.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionCotH.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionCotH.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionCotH.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionCotH.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionCotH.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionCotH.$t = markType(UltraCalcFunctionCotH, 'UltraCalcFunctionCotH', BuiltInFunctionBase.$);
    return UltraCalcFunctionCotH;
}(BuiltInFunctionBase));
export { UltraCalcFunctionCotH };
