// igniteui-react-excel/esm5/lib/UltraCalcFunctionImProduct.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { UltraCalcFunctionComplex_ComplexNumber } from "./UltraCalcFunctionComplex_ComplexNumber";
import { markType } from "igniteui-react-core";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
var UltraCalcFunctionImProduct = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionImProduct, _super);
    function UltraCalcFunctionImProduct() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionImProduct.prototype.evaluate = function (a, b) {
        var c = this._getArguments(a, b, true);
        c.reverse();
        var d = null;
        var e = null;
        for (var g = 0; g < c.length; g++) {
            var f = c[g];
            if (f.isError) {
                return new ExcelCalcValue(f.toErrorValue());
            }
            var h = new UltraCalcFunctionComplex_ComplexNumber(1, a._w, f.toString());
            if (h.y == false) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            if (e == null) {
                e = h.an;
            }
            else if (h.an != e) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            if (d == null) {
                d = h;
            }
            else {
                d = UltraCalcFunctionComplex_ComplexNumber.j(d, h);
            }
        }
        if (d.y == false) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        return new ExcelCalcValue(d.toString());
    };
    UltraCalcFunctionImProduct.prototype.canParameterBeEnumerable = function (a) {
        return true;
    };
    UltraCalcFunctionImProduct.prototype.get_name = function () {
        return "improduct";
    };
    Object.defineProperty(UltraCalcFunctionImProduct.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionImProduct.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionImProduct.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionImProduct.prototype.get_maxArgs = function () {
        return 0x7FFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionImProduct.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionImProduct.$t = markType(UltraCalcFunctionImProduct, 'UltraCalcFunctionImProduct', BuiltInFunctionBase.$);
    return UltraCalcFunctionImProduct;
}(BuiltInFunctionBase));
export { UltraCalcFunctionImProduct };
