// igniteui-react-excel/esm5/lib/UltraCalcFunctionCsc.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
var UltraCalcFunctionCsc = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionCsc, _super);
    function UltraCalcFunctionCsc() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionCsc.prototype.evaluate = function (a, b) {
        return CalcFunctionUtilities.h(a, true, function (c) {
            if (c <= -134217728 || c >= 134217728) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            return new ExcelCalcValue(1 / Math.sin(c));
        });
    };
    UltraCalcFunctionCsc.prototype.get_name = function () {
        return "csc";
    };
    Object.defineProperty(UltraCalcFunctionCsc.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionCsc.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionCsc.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionCsc.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionCsc.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionCsc.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionCsc.$t = markType(UltraCalcFunctionCsc, 'UltraCalcFunctionCsc', BuiltInFunctionBase.$);
    return UltraCalcFunctionCsc;
}(BuiltInFunctionBase));
export { UltraCalcFunctionCsc };
