// igniteui-react-excel/esm5/lib/UltraCalcFunctionMultinomial.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { UltraCalcFunctionFact } from "./UltraCalcFunctionFact";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionMultinomial = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionMultinomial, _super);
    function UltraCalcFunctionMultinomial() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionMultinomial.prototype.evaluate = function (a, b) {
        var c = this._getArguments(a, b, true);
        var d = 0;
        var e = 1;
        for (var f = 0; f < c.length; f++) {
            var g = c[f];
            if (g.isError) {
                return new ExcelCalcValue(g.toErrorValue());
            }
            var h = g._toInt();
            if (h < 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(4));
            }
            d += h;
            e *= UltraCalcFunctionFact._ak(h);
        }
        var i = UltraCalcFunctionFact._ak(d);
        var j = i / e;
        return new ExcelCalcValue(j);
    };
    UltraCalcFunctionMultinomial.prototype.canParameterBeEnumerable = function (a) {
        return true;
    };
    UltraCalcFunctionMultinomial.prototype.get_name = function () {
        return "multinomial";
    };
    Object.defineProperty(UltraCalcFunctionMultinomial.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMultinomial.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionMultinomial.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMultinomial.prototype.get_maxArgs = function () {
        return 0x7FFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionMultinomial.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMultinomial.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionMultinomial.$t = markType(UltraCalcFunctionMultinomial, 'UltraCalcFunctionMultinomial', BuiltInFunctionBase.$);
    return UltraCalcFunctionMultinomial;
}(BuiltInFunctionBase));
export { UltraCalcFunctionMultinomial };
