// igniteui-react-excel/esm5/lib/UltraCalcFunctionMinute.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionMinute = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionMinute, _super);
    function UltraCalcFunctionMinute() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionMinute.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        if (true) {
            var d_1;
            if (c.isNull || (((function () { var e = c._toDouble1(d_1); d_1 = e.p0; return e.ret; })()) && d_1 == 0)) {
                return new ExcelCalcValue(0);
            }
        }
        var e = c._toDateTime1();
        return new ExcelCalcValue(e.getMinutes());
    };
    UltraCalcFunctionMinute.prototype.get_name = function () {
        return "minute";
    };
    Object.defineProperty(UltraCalcFunctionMinute.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMinute.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionMinute.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMinute.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionMinute.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMinute.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionMinute.$t = markType(UltraCalcFunctionMinute, 'UltraCalcFunctionMinute', BuiltInFunctionBase.$);
    return UltraCalcFunctionMinute;
}(BuiltInFunctionBase));
export { UltraCalcFunctionMinute };
