// igniteui-react-excel/esm5/lib/UltraCalcFunctionDay.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { stringIsNullOrEmpty } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionDay = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionDay, _super);
    function UltraCalcFunctionDay() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionDay.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        if (true) {
            var d_1;
            if (c.isNull || (((function () { var e = c._toDouble1(d_1); d_1 = e.p0; return e.ret; })()) && d_1 == 0)) {
                return new ExcelCalcValue(0);
            }
        }
        if (c.isString && stringIsNullOrEmpty(c.toString())) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(1));
        }
        var e = c._toDateTime1();
        return new ExcelCalcValue(e.getDate());
    };
    UltraCalcFunctionDay.prototype.get_name = function () {
        return "day";
    };
    Object.defineProperty(UltraCalcFunctionDay.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionDay.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionDay.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionDay.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionDay.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionDay.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionDay.$t = markType(UltraCalcFunctionDay, 'UltraCalcFunctionDay', BuiltInFunctionBase.$);
    return UltraCalcFunctionDay;
}(BuiltInFunctionBase));
export { UltraCalcFunctionDay };
