// igniteui-react-excel/esm5/lib/UltraCalcFunctionInt.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { truncate } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionInt = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionInt, _super);
    function UltraCalcFunctionInt() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionInt.prototype.evaluate = function (a, b) {
        var c = a.pop()._toDecimal2();
        return new ExcelCalcValue(truncate(Math.floor(c)));
    };
    UltraCalcFunctionInt.prototype.get_name = function () {
        return "int";
    };
    Object.defineProperty(UltraCalcFunctionInt.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionInt.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionInt.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionInt.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionInt.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionInt.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionInt.$t = markType(UltraCalcFunctionInt, 'UltraCalcFunctionInt', BuiltInFunctionBase.$);
    return UltraCalcFunctionInt;
}(BuiltInFunctionBase));
export { UltraCalcFunctionInt };
