// igniteui-react-excel/esm5/lib/UltraCalcFunctionATan2.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionATan2 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionATan2, _super);
    function UltraCalcFunctionATan2() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionATan2.prototype.evaluate = function (a, b) {
        var c = a.pop();
        var d = a.pop();
        if (d.isError) {
            return new ExcelCalcValue(d.toErrorValue());
        }
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var e, f;
        if (!((function () { var g = c._toDouble1(f); f = g.p0; return g.ret; })()) || !((function () { var g = d._toDouble1(e); e = g.p0; return g.ret; })())) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        if (e == 0 && f == 0) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(2));
        }
        return new ExcelCalcValue(Math.atan2(f, e));
    };
    UltraCalcFunctionATan2.prototype.get_name = function () {
        return "atan2";
    };
    Object.defineProperty(UltraCalcFunctionATan2.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionATan2.prototype.get_minArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionATan2.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionATan2.prototype.get_maxArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionATan2.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionATan2.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionATan2.$t = markType(UltraCalcFunctionATan2, 'UltraCalcFunctionATan2', BuiltInFunctionBase.$);
    return UltraCalcFunctionATan2;
}(BuiltInFunctionBase));
export { UltraCalcFunctionATan2 };
