// igniteui-react-excel/esm5/lib/UltraCalcFunctionAverageA.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { UltraCalcFunctionAverage } from "./UltraCalcFunctionAverage";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
/**
 * @hidden
 */
var UltraCalcFunctionAverageA = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionAverageA, _super);
    function UltraCalcFunctionAverageA() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionAverageA.prototype.evaluate = function (a, b) {
        return CalcFunctionUtilities.d(a, b, 1249 & ~1024, UltraCalcFunctionAverage._ak);
    };
    UltraCalcFunctionAverageA.prototype.canParameterBeEnumerable = function (a) {
        return true;
    };
    UltraCalcFunctionAverageA.prototype.get_name = function () {
        return "averagea";
    };
    Object.defineProperty(UltraCalcFunctionAverageA.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionAverageA.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionAverageA.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionAverageA.prototype.get_maxArgs = function () {
        return 0x7FFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionAverageA.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionAverageA.$t = markType(UltraCalcFunctionAverageA, 'UltraCalcFunctionAverageA', BuiltInFunctionBase.$);
    return UltraCalcFunctionAverageA;
}(BuiltInFunctionBase));
export { UltraCalcFunctionAverageA };
