// igniteui-react-excel/esm5/lib/UltraCalcFunctionSqrtPi.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionSqrtPi = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionSqrtPi, _super);
    function UltraCalcFunctionSqrtPi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionSqrtPi.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var d = c._toDouble3();
        if (d < 0) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        var e = Math.sqrt(Math.PI * d);
        return new ExcelCalcValue(e);
    };
    UltraCalcFunctionSqrtPi.prototype.get_name = function () {
        return "sqrtpi";
    };
    Object.defineProperty(UltraCalcFunctionSqrtPi.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSqrtPi.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionSqrtPi.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSqrtPi.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionSqrtPi.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSqrtPi.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionSqrtPi.$t = markType(UltraCalcFunctionSqrtPi, 'UltraCalcFunctionSqrtPi', BuiltInFunctionBase.$);
    return UltraCalcFunctionSqrtPi;
}(BuiltInFunctionBase));
export { UltraCalcFunctionSqrtPi };
