// igniteui-react-excel/esm5/lib/UltraCalcFunctionAcot.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
/**
 * @hidden
 */
var UltraCalcFunctionAcot = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionAcot, _super);
    function UltraCalcFunctionAcot() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionAcot.prototype.evaluate = function (a, b) {
        return CalcFunctionUtilities.h(a, true, function (c) { return new ExcelCalcValue(Math.PI / 2 - Math.atan(c)); });
    };
    UltraCalcFunctionAcot.prototype.get_name = function () {
        return "acot";
    };
    Object.defineProperty(UltraCalcFunctionAcot.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionAcot.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionAcot.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionAcot.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionAcot.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionAcot.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionAcot.$t = markType(UltraCalcFunctionAcot, 'UltraCalcFunctionAcot', BuiltInFunctionBase.$);
    return UltraCalcFunctionAcot;
}(BuiltInFunctionBase));
export { UltraCalcFunctionAcot };
