// igniteui-react-excel/esm5/lib/UltraCalcFunctionMod.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionMod = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionMod, _super);
    function UltraCalcFunctionMod() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionMod.prototype.evaluate = function (a, b) {
        var c = a.pop();
        var d = a.pop();
        if (d.isError) {
            return new ExcelCalcValue(d.toErrorValue());
        }
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var e, f;
        if (!((function () { var g = c._toDouble1(e); e = g.p0; return g.ret; })()) || !((function () { var g = d._toDouble1(f); f = g.p0; return g.ret; })())) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        if (e == 0) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(2));
        }
        else {
            var g = f % e;
            var h = Math.sign(g);
            if (h != 0 && h != Math.sign(e)) {
                g *= -1;
            }
            return new ExcelCalcValue(g);
        }
    };
    UltraCalcFunctionMod.prototype.get_name = function () {
        return "mod";
    };
    Object.defineProperty(UltraCalcFunctionMod.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMod.prototype.get_minArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionMod.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMod.prototype.get_maxArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionMod.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionMod.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionMod.$t = markType(UltraCalcFunctionMod, 'UltraCalcFunctionMod', BuiltInFunctionBase.$);
    return UltraCalcFunctionMod;
}(BuiltInFunctionBase));
export { UltraCalcFunctionMod };
