// igniteui-react-excel/esm5/lib/UltraCalcFunctionTimeValue.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { dateGetTimeOfDay, dateFromTicks } from "igniteui-react-core";
import { timeSpanTicks } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionTimeValue = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionTimeValue, _super);
    function UltraCalcFunctionTimeValue() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionTimeValue.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return c;
        }
        var d = dateGetTimeOfDay(c._toDateTime1());
        if (true) {
            var e = ExcelCalcValue._ay(d);
            return new ExcelCalcValue(e);
        }
        return new ExcelCalcValue(dateFromTicks(timeSpanTicks(d)));
    };
    UltraCalcFunctionTimeValue.prototype.get_name = function () {
        return "timevalue";
    };
    Object.defineProperty(UltraCalcFunctionTimeValue.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionTimeValue.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionTimeValue.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionTimeValue.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionTimeValue.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionTimeValue.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionTimeValue.$t = markType(UltraCalcFunctionTimeValue, 'UltraCalcFunctionTimeValue', BuiltInFunctionBase.$);
    return UltraCalcFunctionTimeValue;
}(BuiltInFunctionBase));
export { UltraCalcFunctionTimeValue };
