// igniteui-react-excel/esm5/lib/UltraCalcFunctionFact.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
import { truncate, isInfinity } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionFact = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionFact, _super);
    function UltraCalcFunctionFact() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionFact.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var d = c._toDecimal2();
        if (d < 0) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        var e = truncate(d);
        var f = UltraCalcFunctionFact._ak(e);
        if (isInfinity(f)) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        return new ExcelCalcValue(f);
    };
    UltraCalcFunctionFact.prototype.get_name = function () {
        return "fact";
    };
    Object.defineProperty(UltraCalcFunctionFact.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionFact.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionFact.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionFact.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionFact.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionFact._ak = function (a, b) {
        if (b === void 0) {
            b = 1;
        }
        if (a == 0) {
            a = 1;
        }
        var c = 1;
        for (var d = a; d > 1; d -= b) {
            c *= d;
        }
        return c;
    };
    UltraCalcFunctionFact.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionFact.$t = markType(UltraCalcFunctionFact, 'UltraCalcFunctionFact', BuiltInFunctionBase.$);
    return UltraCalcFunctionFact;
}(BuiltInFunctionBase));
export { UltraCalcFunctionFact };
