// igniteui-react-excel/esm5/lib/UltraCalcFunctionRept.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { StringBuilder } from "igniteui-react-core";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
import { stringIsNullOrEmpty, stringEmpty } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionRept = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionRept, _super);
    function UltraCalcFunctionRept() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionRept.prototype.evaluate = function (a, b) {
        var c = a.pop();
        var d = a.pop();
        if (d.isError) {
            return new ExcelCalcValue(d.toErrorValue());
        }
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var e = c._toInt321();
        if (e < 0) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(1));
        }
        var f = d.toString();
        if (e == 0 || stringIsNullOrEmpty(f)) {
            return new ExcelCalcValue(stringEmpty());
        }
        var g = f.length * e;
        if (32767 < g) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(1));
        }
        var h = new StringBuilder(1, g);
        for (var i = 0; i < e; i++) {
            h.l(f);
        }
        return new ExcelCalcValue(h.toString());
    };
    UltraCalcFunctionRept.prototype.get_name = function () {
        return "rept";
    };
    Object.defineProperty(UltraCalcFunctionRept.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionRept.prototype.get_minArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionRept.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionRept.prototype.get_maxArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionRept.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionRept.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionRept.$t = markType(UltraCalcFunctionRept, 'UltraCalcFunctionRept', BuiltInFunctionBase.$);
    return UltraCalcFunctionRept;
}(BuiltInFunctionBase));
export { UltraCalcFunctionRept };
