// igniteui-react-excel/esm5/lib/UltraCalcFunctionProduct.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
/**
 * @hidden
 */
var UltraCalcFunctionProduct = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionProduct, _super);
    function UltraCalcFunctionProduct() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionProduct.prototype.evaluate = function (a, b) {
        return CalcFunctionUtilities.d(a, b, 1229, UltraCalcFunctionProduct._ak);
    };
    UltraCalcFunctionProduct.prototype.canParameterBeEnumerable = function (a) {
        return true;
    };
    UltraCalcFunctionProduct._ak = function (a) {
        return new ExcelCalcValue(CalcFunctionUtilities.v(a));
    };
    UltraCalcFunctionProduct.prototype.get_name = function () {
        return "product";
    };
    Object.defineProperty(UltraCalcFunctionProduct.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionProduct.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionProduct.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionProduct.prototype.get_maxArgs = function () {
        return 0x7FFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionProduct.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionProduct.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionProduct.$t = markType(UltraCalcFunctionProduct, 'UltraCalcFunctionProduct', BuiltInFunctionBase.$);
    return UltraCalcFunctionProduct;
}(BuiltInFunctionBase));
export { UltraCalcFunctionProduct };
