// igniteui-react-excel/esm5/lib/UltraCalcFunctionACosh.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionACosh = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionACosh, _super);
    function UltraCalcFunctionACosh() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionACosh.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var d;
        if (!((function () { var e = c._toDouble1(d); d = e.p0; return e.ret; })())) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        if (d < 1) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        var e = Math.log(d + Math.sqrt((d * d) - 1));
        return new ExcelCalcValue(e);
    };
    UltraCalcFunctionACosh.prototype.get_name = function () {
        return "acosh";
    };
    Object.defineProperty(UltraCalcFunctionACosh.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionACosh.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionACosh.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionACosh.prototype.get_maxArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionACosh.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionACosh.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionACosh.$t = markType(UltraCalcFunctionACosh, 'UltraCalcFunctionACosh', BuiltInFunctionBase.$);
    return UltraCalcFunctionACosh;
}(BuiltInFunctionBase));
export { UltraCalcFunctionACosh };
