// igniteui-react-excel/esm5/lib/UltraCalcFunctionSYD.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionSYD = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionSYD, _super);
    function UltraCalcFunctionSYD() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionSYD.prototype.evaluate = function (a, b) {
        var c = a.pop()._toInt321();
        if (c <= 0) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        var d = a.pop()._toInt321();
        if (d <= 0) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        var e, f;
        if (!((function () { var g = a.pop()._toDouble1(e); e = g.p0; return g.ret; })()) || !((function () { var g = a.pop()._toDouble1(f); f = g.p0; return g.ret; })())) {
            return new ExcelCalcValue(new ExcelCalcErrorValue(4));
        }
        return new ExcelCalcValue(((f - e) * (d - c + 1) * 2) / (d * (d + 1)));
    };
    UltraCalcFunctionSYD.prototype.get_name = function () {
        return "syd";
    };
    Object.defineProperty(UltraCalcFunctionSYD.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSYD.prototype.get_minArgs = function () {
        return 4;
    };
    Object.defineProperty(UltraCalcFunctionSYD.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSYD.prototype.get_maxArgs = function () {
        return 4;
    };
    Object.defineProperty(UltraCalcFunctionSYD.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionSYD.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionSYD.$t = markType(UltraCalcFunctionSYD, 'UltraCalcFunctionSYD', BuiltInFunctionBase.$);
    return UltraCalcFunctionSYD;
}(BuiltInFunctionBase));
export { UltraCalcFunctionSYD };
