// igniteui-react-excel/esm5/lib/UltraCalcFunctionStdev.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { UltraCalcFunctionVar } from "./UltraCalcFunctionVar";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
import { CalcFunctionUtilities } from "./CalcFunctionUtilities";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
/**
 * @hidden
 */
var UltraCalcFunctionStdev = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionStdev, _super);
    function UltraCalcFunctionStdev() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionStdev.prototype.evaluate = function (a, b) {
        return CalcFunctionUtilities.d(a, b, 1229, function (c) { return UltraCalcFunctionStdev._al(c, true); });
    };
    UltraCalcFunctionStdev.prototype.canParameterBeEnumerable = function (a) {
        return true;
    };
    UltraCalcFunctionStdev._al = function (a, b) {
        if (a.count < 2) {
            if (!b && a.count == 1) {
                return new ExcelCalcValue(0);
            }
            return new ExcelCalcValue(new ExcelCalcErrorValue(2));
        }
        var c = CalcFunctionUtilities.r(a);
        return new ExcelCalcValue(CalcFunctionUtilities.y(c, a.count, b));
    };
    UltraCalcFunctionStdev.prototype.get_name = function () {
        return "stdev";
    };
    UltraCalcFunctionStdev.prototype.get_minArgs = function () {
        return 1;
    };
    UltraCalcFunctionStdev.prototype.get_maxArgs = function () {
        return 0x7FFFFFFF;
    };
    UltraCalcFunctionStdev.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionStdev.$t = markType(UltraCalcFunctionStdev, 'UltraCalcFunctionStdev', UltraCalcFunctionVar.$);
    return UltraCalcFunctionStdev;
}(UltraCalcFunctionVar));
export { UltraCalcFunctionStdev };
