// igniteui-react-excel/esm5/lib/UltraCalcFunctionGeStep.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionGeStep = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionGeStep, _super);
    function UltraCalcFunctionGeStep() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionGeStep.prototype.evaluate = function (a, b) {
        var c;
        var d;
        if (b == 2) {
            c = a.pop();
            if (c.isError) {
                return new ExcelCalcValue(c.toErrorValue());
            }
            d = c._toDouble3();
        }
        else {
            d = 0;
        }
        c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var e = c._toDouble3();
        var f = (e >= d) ? 1 : 0;
        return new ExcelCalcValue(f);
    };
    UltraCalcFunctionGeStep.prototype.get_name = function () {
        return "gestep";
    };
    Object.defineProperty(UltraCalcFunctionGeStep.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionGeStep.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionGeStep.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionGeStep.prototype.get_maxArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionGeStep.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionGeStep.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionGeStep.$t = markType(UltraCalcFunctionGeStep, 'UltraCalcFunctionGeStep', BuiltInFunctionBase.$);
    return UltraCalcFunctionGeStep;
}(BuiltInFunctionBase));
export { UltraCalcFunctionGeStep };
