// igniteui-react-excel/esm5/lib/UltraCalcFunctionBin2Hex.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { UltraCalcFunctionXBase2XBase } from "./UltraCalcFunctionXBase2XBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionBin2Hex = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionBin2Hex, _super);
    function UltraCalcFunctionBin2Hex() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionBin2Hex.prototype.get_name = function () {
        return "bin2hex";
    };
    Object.defineProperty(UltraCalcFunctionBin2Hex.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionBin2Hex.prototype.get__aq = function () {
        return -512;
    };
    Object.defineProperty(UltraCalcFunctionBin2Hex.prototype, "_aq", {
        get: function () {
            return this.get__aq();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionBin2Hex.prototype.get__ap = function () {
        return 511;
    };
    Object.defineProperty(UltraCalcFunctionBin2Hex.prototype, "_ap", {
        get: function () {
            return this.get__ap();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionBin2Hex.prototype.get__ao = function () {
        return -549755813888;
    };
    Object.defineProperty(UltraCalcFunctionBin2Hex.prototype, "_ao", {
        get: function () {
            return this.get__ao();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionBin2Hex.prototype.get__an = function () {
        return 0x7FFFFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionBin2Hex.prototype, "_an", {
        get: function () {
            return this.get__an();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionBin2Hex.prototype.get__al = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionBin2Hex.prototype, "_al", {
        get: function () {
            return this.get__al();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionBin2Hex.prototype.get__ak = function () {
        return 16;
    };
    Object.defineProperty(UltraCalcFunctionBin2Hex.prototype, "_ak", {
        get: function () {
            return this.get__ak();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionBin2Hex.prototype.get__am = function () {
        return 10;
    };
    Object.defineProperty(UltraCalcFunctionBin2Hex.prototype, "_am", {
        get: function () {
            return this.get__am();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionBin2Hex.$t = markType(UltraCalcFunctionBin2Hex, 'UltraCalcFunctionBin2Hex', UltraCalcFunctionXBase2XBase.$);
    return UltraCalcFunctionBin2Hex;
}(UltraCalcFunctionXBase2XBase));
export { UltraCalcFunctionBin2Hex };
