// igniteui-react-excel/esm5/lib/UltraCalcFunctionRoundDown.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { MathUtilities } from "./MathUtilities";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionRoundDown = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionRoundDown, _super);
    function UltraCalcFunctionRoundDown() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionRoundDown.prototype.evaluate = function (a, b) {
        var c = a.pop();
        if (c.isError) {
            return new ExcelCalcValue(c.toErrorValue());
        }
        var d = c._toInt();
        var e = a.pop();
        if (e.isError) {
            return new ExcelCalcValue(e.toErrorValue());
        }
        var f = e._toDouble3();
        var g = d;
        var h = Math.pow(10, g);
        f *= h;
        f = MathUtilities.i(f);
        f = MathUtilities.j(f);
        f /= h;
        f = MathUtilities.i(f);
        if (f == -0) {
            f = 0;
        }
        return new ExcelCalcValue(f);
    };
    UltraCalcFunctionRoundDown.prototype.get_name = function () {
        return "rounddown";
    };
    Object.defineProperty(UltraCalcFunctionRoundDown.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionRoundDown.prototype.get_minArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionRoundDown.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionRoundDown.prototype.get_maxArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionRoundDown.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionRoundDown.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return true;
    };
    UltraCalcFunctionRoundDown.$t = markType(UltraCalcFunctionRoundDown, 'UltraCalcFunctionRoundDown', BuiltInFunctionBase.$);
    return UltraCalcFunctionRoundDown;
}(BuiltInFunctionBase));
export { UltraCalcFunctionRoundDown };
