// igniteui-react-excel/esm5/lib/UltraCalcFunctionHex2Bin.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { UltraCalcFunctionXBase2XBase } from "./UltraCalcFunctionXBase2XBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionHex2Bin = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionHex2Bin, _super);
    function UltraCalcFunctionHex2Bin() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionHex2Bin.prototype.get_name = function () {
        return "hex2bin";
    };
    Object.defineProperty(UltraCalcFunctionHex2Bin.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Bin.prototype.get__aq = function () {
        return -549755813888;
    };
    Object.defineProperty(UltraCalcFunctionHex2Bin.prototype, "_aq", {
        get: function () {
            return this.get__aq();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Bin.prototype.get__ap = function () {
        return 0x7FFFFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionHex2Bin.prototype, "_ap", {
        get: function () {
            return this.get__ap();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Bin.prototype.get__ao = function () {
        return -512;
    };
    Object.defineProperty(UltraCalcFunctionHex2Bin.prototype, "_ao", {
        get: function () {
            return this.get__ao();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Bin.prototype.get__an = function () {
        return 511;
    };
    Object.defineProperty(UltraCalcFunctionHex2Bin.prototype, "_an", {
        get: function () {
            return this.get__an();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Bin.prototype.get__al = function () {
        return 16;
    };
    Object.defineProperty(UltraCalcFunctionHex2Bin.prototype, "_al", {
        get: function () {
            return this.get__al();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Bin.prototype.get__ak = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionHex2Bin.prototype, "_ak", {
        get: function () {
            return this.get__ak();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Bin.prototype.get__am = function () {
        return 10;
    };
    Object.defineProperty(UltraCalcFunctionHex2Bin.prototype, "_am", {
        get: function () {
            return this.get__am();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionHex2Bin.$t = markType(UltraCalcFunctionHex2Bin, 'UltraCalcFunctionHex2Bin', UltraCalcFunctionXBase2XBase.$);
    return UltraCalcFunctionHex2Bin;
}(UltraCalcFunctionXBase2XBase));
export { UltraCalcFunctionHex2Bin };
