// igniteui-react-excel/esm5/lib/UltraCalcFunctionAnd.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionAnd = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionAnd, _super);
    function UltraCalcFunctionAnd() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionAnd.prototype.evaluate = function (a, b) {
        var c = true;
        var d = false;
        var e = this._getArguments(a, b, true);
        for (var g = 0; g < e.length; g++) {
            var f = e[g];
            if (f.isError) {
                return new ExcelCalcValue(f.toErrorValue());
            }
            try {
                if (!f._toBoolean1()) {
                    c = false;
                }
                d = true;
            }
            catch (h) {
            }
        }
        if (d) {
            return new ExcelCalcValue(c);
        }
        return new ExcelCalcValue(new ExcelCalcErrorValue(1));
    };
    UltraCalcFunctionAnd.prototype.canParameterBeEnumerable = function (a) {
        return true;
    };
    UltraCalcFunctionAnd.prototype.get_name = function () {
        return "and";
    };
    Object.defineProperty(UltraCalcFunctionAnd.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionAnd.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionAnd.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionAnd.prototype.get_maxArgs = function () {
        return 0x7FFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionAnd.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionAnd.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionAnd.$t = markType(UltraCalcFunctionAnd, 'UltraCalcFunctionAnd', BuiltInFunctionBase.$);
    return UltraCalcFunctionAnd;
}(BuiltInFunctionBase));
export { UltraCalcFunctionAnd };
