// igniteui-react-excel/esm5/lib/UltraCalcFunctionLeft.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionLeft = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionLeft, _super);
    function UltraCalcFunctionLeft() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionLeft.prototype.evaluate = function (a, b) {
        var c = 1;
        if (b > 1) {
            var d = a.pop();
            if (d.isError) {
                return new ExcelCalcValue(d.toErrorValue());
            }
            c = d._toInt321();
            if (c < 0) {
                return new ExcelCalcValue(new ExcelCalcErrorValue(1));
            }
        }
        var e = a.pop();
        if (e.isError) {
            return new ExcelCalcValue(e.toErrorValue());
        }
        var f = e.toString();
        return new ExcelCalcValue(f.substr(0, c > f.length ? f.length : c));
    };
    UltraCalcFunctionLeft.prototype.get_name = function () {
        return "left";
    };
    Object.defineProperty(UltraCalcFunctionLeft.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionLeft.prototype.get_minArgs = function () {
        return 1;
    };
    Object.defineProperty(UltraCalcFunctionLeft.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionLeft.prototype.get_maxArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionLeft.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionLeft.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionLeft.$t = markType(UltraCalcFunctionLeft, 'UltraCalcFunctionLeft', BuiltInFunctionBase.$);
    return UltraCalcFunctionLeft;
}(BuiltInFunctionBase));
export { UltraCalcFunctionLeft };
