// igniteui-react-excel/esm5/lib/UltraCalcFunctionIfs.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { BuiltInFunctionBase } from "./BuiltInFunctionBase";
import { ExcelCalcValue } from "./ExcelCalcValue";
import { List$1 } from "igniteui-react-core";
import { ExcelCalcErrorValue } from "./ExcelCalcErrorValue";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var UltraCalcFunctionIfs = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(UltraCalcFunctionIfs, _super);
    function UltraCalcFunctionIfs() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UltraCalcFunctionIfs.prototype.evaluate = function (a, b) {
        var c = new List$1(ExcelCalcValue.$, 0);
        while (b > 0) {
            c.add(a.pop());
            b--;
        }
        for (var d = c.count - 1; d > 0; d -= 2) {
            var e = c._inner[d];
            var f = c._inner[d - 1];
            if (e.isError) {
                return new ExcelCalcValue(e.toErrorValue());
            }
            if (e._toBoolean1() == true) {
                return f;
            }
        }
        return new ExcelCalcValue(new ExcelCalcErrorValue(3));
    };
    UltraCalcFunctionIfs.prototype.get_name = function () {
        return "ifs";
    };
    Object.defineProperty(UltraCalcFunctionIfs.prototype, "name", {
        get: function () {
            return this.get_name();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionIfs.prototype.get_minArgs = function () {
        return 2;
    };
    Object.defineProperty(UltraCalcFunctionIfs.prototype, "minArgs", {
        get: function () {
            return this.get_minArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionIfs.prototype.get_maxArgs = function () {
        return 0x7FFFFFFF;
    };
    Object.defineProperty(UltraCalcFunctionIfs.prototype, "maxArgs", {
        get: function () {
            return this.get_maxArgs();
        },
        enumerable: false,
        configurable: true
    });
    UltraCalcFunctionIfs.prototype.doesParameterAllowIntermediateResultArray = function (a, b) {
        return (b == false);
    };
    UltraCalcFunctionIfs.prototype.get__s = function () {
        return true;
    };
    UltraCalcFunctionIfs.prototype._v = function (a) {
        if (a % 2 == 1) {
            return false;
        }
        return true;
    };
    UltraCalcFunctionIfs.$t = markType(UltraCalcFunctionIfs, 'UltraCalcFunctionIfs', BuiltInFunctionBase.$);
    return UltraCalcFunctionIfs;
}(BuiltInFunctionBase));
export { UltraCalcFunctionIfs };
