// igniteui-react-excel/esm5/lib/VerticalCellAlignment.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Enumeration for vertical alignment styles. Default value is used in property override situations.
 */
export var VerticalCellAlignment = /*@__PURE__*/ (function (VerticalCellAlignment) {
    /**
     * Use the current default.
     */
    VerticalCellAlignment[VerticalCellAlignment["Default"] = -1] = "Default";
    /**
     * Top alignment.
     */
    VerticalCellAlignment[VerticalCellAlignment["Top"] = 0] = "Top";
    /**
     * Center alignment.
     */
    VerticalCellAlignment[VerticalCellAlignment["Center"] = 1] = "Center";
    /**
     * Bottom alignment.
     */
    VerticalCellAlignment[VerticalCellAlignment["Bottom"] = 2] = "Bottom";
    /**
     * Justify alignment.
     */
    VerticalCellAlignment[VerticalCellAlignment["Justify"] = 3] = "Justify";
    /**
     * Distributed alignment.
     */
    VerticalCellAlignment[VerticalCellAlignment["Distributed"] = 4] = "Distributed";
    return VerticalCellAlignment;
})({});
/**
 * @hidden
 */
export var VerticalCellAlignment_$type = /*@__PURE__*/ markEnum('VerticalCellAlignment', 'Default,-1|Top,0|Center,1|Bottom,2|Justify,3|Distributed,4');
