// igniteui-react-excel/esm5/lib/VerticalTitleAlignment.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Determines the horizontal alignment for a title
 */
export var VerticalTitleAlignment = /*@__PURE__*/ (function (VerticalTitleAlignment) {
    /**
     * The title will be aligned on the bottom
     */
    VerticalTitleAlignment[VerticalTitleAlignment["Bottom"] = -4107] = "Bottom";
    /**
     * The title will be centered vertically
     */
    VerticalTitleAlignment[VerticalTitleAlignment["Center"] = -4108] = "Center";
    /**
     * The title will be distributed
     */
    VerticalTitleAlignment[VerticalTitleAlignment["Distributed"] = -4117] = "Distributed";
    /**
     * The title will be justified
     */
    VerticalTitleAlignment[VerticalTitleAlignment["Justify"] = -4130] = "Justify";
    /**
     * The title will be aligned on the top
     */
    VerticalTitleAlignment[VerticalTitleAlignment["Top"] = -4160] = "Top";
    return VerticalTitleAlignment;
})({});
/**
 * @hidden
 */
export var VerticalTitleAlignment_$type = /*@__PURE__*/ markEnum('VerticalTitleAlignment', 'Bottom,-4107|Center,-4108|Distributed,-4117|Justify,-4130|Top,-4160');
