// igniteui-react-excel/esm5/lib/VmlDrawingPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { XmlPartManagerBase } from "./XmlPartManagerBase";
import { SgmlReader } from "./SgmlReader";
import { StreamReader } from "igniteui-react-core";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var VmlDrawingPartManager = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(VmlDrawingPartManager, _super);
    function VmlDrawingPartManager(a) {
        var _this = _super.call(this) || this;
        _this.a3 = false;
        _this.a3 = a;
        return _this;
    }
    VmlDrawingPartManager.prototype.get_m = function () {
        return VmlDrawingPartManager.a7;
    };
    Object.defineProperty(VmlDrawingPartManager.prototype, "m", {
        get: function () {
            return this.get_m();
        },
        enumerable: false,
        configurable: true
    });
    VmlDrawingPartManager.prototype.a1 = function () {
        var _this = this;
        return ((function () {
            var $ret = new SgmlReader();
            $ret._b7 = new StreamReader(_this.l);
            return $ret;
        })());
    };
    VmlDrawingPartManager.prototype.n = function (a) {
        if (this.a3) {
            switch (a) {
                case 1: return "/theme/drawings/vmlDrawing.vml";
                default: return null;
            }
        }
        else {
            switch (a) {
                case 0: return "/xl/drawings/vmlDrawing.vml";
                case 1: return "/word/drawings/vmlDrawing.vml";
                default: return null;
            }
        }
    };
    VmlDrawingPartManager.prototype.o = function (a) {
        switch (a) {
            case 0:
            case 1: return VmlDrawingPartManager.a8;
            default: return VmlDrawingPartManager.a8;
        }
    };
    VmlDrawingPartManager.prototype.get_g = function () {
        return true;
    };
    Object.defineProperty(VmlDrawingPartManager.prototype, "g", {
        get: function () {
            return this.get_g();
        },
        enumerable: false,
        configurable: true
    });
    VmlDrawingPartManager.$t = markType(VmlDrawingPartManager, 'VmlDrawingPartManager', XmlPartManagerBase.$);
    VmlDrawingPartManager.a7 = "application/vnd.openxmlformats-officedocument.vmlDrawing";
    VmlDrawingPartManager.a8 = "http://schemas.openxmlformats.org/officeDocument/2006/relationships/vmlDrawing";
    return VmlDrawingPartManager;
}(XmlPartManagerBase));
export { VmlDrawingPartManager };
