// igniteui-react-excel/esm5/lib/WallType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Constants which determine the type of a [[Wall]],
 * i.e., back, side, or floor. Applicable only for 3-D charts.
 */
export var WallType = /*@__PURE__*/ (function (WallType) {
    /**
     * The back wall, i.e., "behind" the plot area.
     */
    WallType[WallType["Back"] = 0] = "Back";
    /**
     * The side wall. Note that only one side is ever visible,
     * based on the rotation around the Y-axis.
     */
    WallType[WallType["Side"] = 1] = "Side";
    /**
     * The bottom wall, i.e., the floor of the chart.
     */
    WallType[WallType["Floor"] = 2] = "Floor";
    /**
     * Applies to the back and side walls, and the floor.
     */
    WallType[WallType["All"] = 3] = "All";
    return WallType;
})({});
/**
 * @hidden
 */
export var WallType_$type = /*@__PURE__*/ markEnum('WallType', 'Back,0|Side,1|Floor,2|All,3');
