// igniteui-react-excel/esm5/lib/WmfPartManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { ImagePartManagerBase } from "./ImagePartManagerBase";
import { ImageFormat } from "./ImageFormat";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
var WmfPartManager = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WmfPartManager, _super);
    function WmfPartManager() {
        return _super.call(this) || this;
    }
    WmfPartManager.prototype.get_s = function () {
        return "/xl/media/image.wmf";
    };
    Object.defineProperty(WmfPartManager.prototype, "s", {
        get: function () {
            return this.get_s();
        },
        enumerable: false,
        configurable: true
    });
    WmfPartManager.prototype.get_t = function () {
        return "/ppt/media/image.wmf";
    };
    Object.defineProperty(WmfPartManager.prototype, "t", {
        get: function () {
            return this.get_t();
        },
        enumerable: false,
        configurable: true
    });
    WmfPartManager.prototype.get_u = function () {
        return "/word/media/image.wmf";
    };
    Object.defineProperty(WmfPartManager.prototype, "u", {
        get: function () {
            return this.get_u();
        },
        enumerable: false,
        configurable: true
    });
    WmfPartManager.prototype.get_m = function () {
        return "image/x-wmf";
    };
    Object.defineProperty(WmfPartManager.prototype, "m", {
        get: function () {
            return this.get_m();
        },
        enumerable: false,
        configurable: true
    });
    WmfPartManager.prototype.r = function (a) {
        return ImageFormat.i;
    };
    WmfPartManager.$t = markType(WmfPartManager, 'WmfPartManager', ImagePartManagerBase.$);
    WmfPartManager.y = new WmfPartManager();
    return WmfPartManager;
}(ImagePartManagerBase));
export { WmfPartManager };
