// igniteui-react-excel/esm5/lib/WorkbookSaveManagerExcel2003_SaveTokenVisitor.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { FormulaTokenVisitor } from "./FormulaTokenVisitor";
import { WorkbookSaveManagerExcel2003_GetSizeTokenVisitor } from "./WorkbookSaveManagerExcel2003_GetSizeTokenVisitor";
import { markType } from "igniteui-react-core";
import { intSToU } from "igniteui-react-core";
/**
 * @hidden
 */
var WorkbookSaveManagerExcel2003_SaveTokenVisitor = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WorkbookSaveManagerExcel2003_SaveTokenVisitor, _super);
    function WorkbookSaveManagerExcel2003_SaveTokenVisitor() {
        var _this = _super.call(this) || this;
        _this.a6 = false;
        _this.a1 = null;
        _this.a3 = null;
        _this.a2 = null;
        return _this;
    }
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.e = function (a) {
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.g = function () {
        this.a6 = false;
        this.a1 = null;
        this.a3 = null;
        this.a2 = null;
        _super.prototype.g.call(this);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.h = function () {
        var a = this.a1.length;
        for (this.d = 0; this.d < a; this.d++) {
            var b = this.a1[this.d];
            this.a3._write9(b.f);
            b.m(this);
        }
        this.g();
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.j = function (a) {
        this.a7(a.o);
        this.n(a);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.l = function (a) {
        this.a3._write2(WorkbookSaveManagerExcel2003_SaveTokenVisitor.a0);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.n = function (a) {
        this.a3._write4(a.x);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.o = function (a) {
        this.a3._write2(WorkbookSaveManagerExcel2003_SaveTokenVisitor.az);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.p = function (a) {
        this.a3._write9(a.p);
        this.a3._write17((a.skipTokenIndexes.count));
        var b = this.a2[this.d];
        var c = WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a7(a);
        this.a3._write17((c - 4));
        for (var d = 0; d < a.skipTokenIndexes.count; d++) {
            var e = this.a2[a.skipTokenIndexes._inner[d]];
            this.a3._write17((e - b));
        }
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.q = function (a) {
        this.a3._write9(a.p);
        if (a.skipTokenIndexes.count == 0) {
            this.a3._write17(0);
            return;
        }
        var b = this.a2[this.d];
        var c = b + 4;
        var d = this.a2[a.skipTokenIndexes._inner[0]];
        var e = d + 4;
        this.a3._write17((e - c));
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.r = function (a) {
        this.a3._write9(a.p);
        var b = this.a2[this.d];
        var c = b + 4;
        var d = this.a2[a.t];
        var e = d + 4;
        this.a3._write17((e - c - 1));
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.s = function (a) {
        this.a3._write9(a.p);
        this.a3._write9(a.t);
        this.a3._write9(a.v);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.t = function (a) {
        this.a3._write9(a.p);
        this.a3._write17(0);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.u = function (a) {
        this.a3._write9(a.p);
        this.a3._write17(0);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.v = function (a) {
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.w = function (a) {
        this.a3._write9((a.r ? 1 : 0));
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.y = function (a) {
        this.a3._write9(a.p._y);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.z = function (a) {
        this.a3._write17(a.p.t);
        this.a3._write17(a.p.r);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.aa = function (a) {
        this.a3._write17(a.$function._wi);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.ac = function (a) {
        var b = a.argumentCount;
        if (a.$function._q4) {
            b++;
        }
        this.a3._write9(b);
        this.aa(a);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.ad = function (a) {
        this.a3._write17(a.p);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.ae = function (a) {
        this.a3._write18(intSToU(0));
        this.a3._write17(a.p);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.af = function (a) {
        this.a3._write18(intSToU(0));
        this.a3._write17(a.p);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.ag = function (a) {
        this.a3._write17(a.p);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.ah = function (a) {
        this.a3._write18(intSToU(0));
        this.a3._write17(a.p);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.ai = function (a) {
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.aj = function (a) {
        var b = this.a3.saveManager._b._bb;
        var c = b.u.indexOf(a.u);
        if (c < 0) {
            var d = b.c(a.u.name, this.a3.saveManager._b, false);
            c = b.u.indexOf(d);
        }
        this.a3._write17((c + 1));
        this.a3._write17(0);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.ak = function (a) {
        this.a7(a.o);
        var b = a.o.e.u.indexOf(a.u);
        this.a3._write17((b + 1));
        this.a3._write17(0);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.al = function (a) {
        this.a3._write10(a.p);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.an = function (a) {
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.ap = function (a) {
        this.a7(a.o);
        this.au(a);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.as = function (a) {
        this.a3._write18(intSToU(0));
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.au = function (a) {
        this.a3._write3(a.y);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.av = function (a) {
        this.a3._write15(a.p, 0);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.aw = function (a) {
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.ax = function (a) {
        this.a3._write17(a.p.t);
        this.a3._write17(a.p.r);
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.ay = function (a) {
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.a5 = function (a, b, c, d, e) {
        if (WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4 == null) {
            WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4 = new WorkbookSaveManagerExcel2003_SaveTokenVisitor();
        }
        WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4.b = a;
        WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4.a3 = b;
        WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4.a6 = c;
        WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4.a1 = d;
        WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4.a2 = e;
        return WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4;
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.prototype.a7 = function (a) {
        if (this.a6) {
            this.a3._write17(a.l);
            this.a3._write17(a.m);
        }
        else {
            var b = this.a3.saveManager._j.indexOf(a);
            this.a3._write17(b);
        }
    };
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.$t = markType(WorkbookSaveManagerExcel2003_SaveTokenVisitor, 'WorkbookSaveManagerExcel2003_SaveTokenVisitor', FormulaTokenVisitor.$);
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.a4 = null;
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.az = new Array(7);
    WorkbookSaveManagerExcel2003_SaveTokenVisitor.a0 = new Array(8);
    return WorkbookSaveManagerExcel2003_SaveTokenVisitor;
}(FormulaTokenVisitor));
export { WorkbookSaveManagerExcel2003_SaveTokenVisitor };
