// igniteui-react-excel/esm5/lib/WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { FormulaTokenVisitor$1 } from "./FormulaTokenVisitor$1";
import { Number_$type, markType } from "igniteui-react-core";
/**
 * @hidden
 */
var WorkbookSaveManagerExcel2003_GetSizeTokenVisitor = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WorkbookSaveManagerExcel2003_GetSizeTokenVisitor, _super);
    function WorkbookSaveManagerExcel2003_GetSizeTokenVisitor() {
        var _this = _super.call(this, Number_$type) || this;
        _this.a6 = false;
        _this.a3 = null;
        _this.a2 = null;
        return _this;
    }
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.d = function (a) {
        return 0;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.a1 = function () {
        this.a6 = false;
        this.a3 = null;
        this.a2 = null;
        _super.prototype.a1.call(this);
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.f = function (a) {
        if (this.a6) {
            return 13;
        }
        return 11;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.h = function (a) {
        return 9;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.j = function (a) {
        return 9;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.k = function (a) {
        return 8;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.l = function (a) {
        return WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a7(a);
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.m = function (a) {
        return 4;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.n = function (a) {
        return 4;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.o = function (a) {
        return 4;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.p = function (a) {
        return 4;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.q = function (a) {
        return 4;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.r = function (a) {
        return 1;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.s = function (a) {
        return 2;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.u = function (a) {
        return 2;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.v = function (a) {
        return 5;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.w = function (a) {
        return 3;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.y = function (a) {
        return 4;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.z = function (a) {
        return 3;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.aa = function (a) {
        return 7;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.ab = function (a) {
        return 7;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.ac = function (a) {
        return 3;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.ad = function (a) {
        return 7;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.ae = function (a) {
        return 1;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.af = function (a) {
        return 5;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.ag = function (a) {
        if (this.a6) {
            return 9;
        }
        return 7;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.ah = function (a) {
        return 9;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.aj = function (a) {
        return 1;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.al = function (a) {
        if (this.a6) {
            return 9;
        }
        return 7;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.ao = function (a) {
        return 5;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.aq = function (a) {
        return 5;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.ar = function (a) {
        return (1 + this.a3._getDataSize(this.a2[this.az], 0, a.p));
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.as = function (a) {
        return 0;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.at = function (a) {
        return 5;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.prototype.au = function (a) {
        return 1;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a5 = function (a, b, c, d) {
        if (WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4 == null) {
            WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4 = new WorkbookSaveManagerExcel2003_GetSizeTokenVisitor();
        }
        WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4.c = a;
        WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4.a3 = b;
        WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4.a6 = c;
        WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4.a2 = d;
        return WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4;
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a7 = function (a) {
        return (6 + (2 * a.skipTokenIndexes.count));
    };
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.$t = markType(WorkbookSaveManagerExcel2003_GetSizeTokenVisitor, 'WorkbookSaveManagerExcel2003_GetSizeTokenVisitor', FormulaTokenVisitor$1.$.specialize(Number_$type));
    WorkbookSaveManagerExcel2003_GetSizeTokenVisitor.a4 = null;
    return WorkbookSaveManagerExcel2003_GetSizeTokenVisitor;
}(FormulaTokenVisitor$1));
export { WorkbookSaveManagerExcel2003_GetSizeTokenVisitor };
