// igniteui-react-excel/esm5/lib/WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { FormulaTokenVisitor } from "./FormulaTokenVisitor";
import { markType } from "igniteui-react-core";
import { stringIsNullOrEmpty } from "igniteui-react-core";
import { tryParseInt32_1 } from "igniteui-react-core";
/**
 * @hidden
 */
var WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor, _super);
    function WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor() {
        var _this = _super.call(this) || this;
        _this.az = null;
        return _this;
    }
    WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.prototype.g = function () {
        this.az = null;
        _super.prototype.g.call(this);
    };
    WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.prototype.x = function (a) {
        var _this = this;
        var b = a.o;
        if (((function () { var c = _this.a2(b); b = c.p0; return c.ret; })())) {
            a.o = b;
        }
    };
    WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.a1 = function (a, b) {
        if (WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.a0 == null) {
            WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.a0 = new WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor();
        }
        WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.a0.az = b;
        WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.a0.b = a;
        return WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.a0;
    };
    WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.prototype.a2 = function (a) {
        if (a == null || a.i || a.e == null || stringIsNullOrEmpty(a.e.x)) {
            return {
                ret: false,
                p0: a
            };
        }
        var b;
        if (((function () { var c = tryParseInt32_1(a.e.x, b); b = c.p1; return c.ret; })()) == false) {
            return {
                ret: false,
                p0: a
            };
        }
        if (b < 0 || this.az._i.count <= b) {
            return {
                ret: false,
                p0: a
            };
        }
        var c = this.az._i._inner[b];
        a = a.g(c);
        return {
            ret: true,
            p0: a
        };
    };
    WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.$t = markType(WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor, 'WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor', FormulaTokenVisitor.$);
    WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor.a0 = null;
    return WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor;
}(FormulaTokenVisitor));
export { WorkbookLoadManagerExcel2007_UpdateIndexedWorkbookReferencesVisitor };
