// igniteui-react-excel/esm5/lib/WorkbookPasswordInfo.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { Base, toNullable, Number_$type, markType } from "igniteui-react-core";
/**
 * @hidden
 */
var WorkbookPasswordInfo = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WorkbookPasswordInfo, _super);
    function WorkbookPasswordInfo(a) {
        var _this = _super.call(this) || this;
        _this.i = null;
        _this.a = null;
        _this.d = false;
        _this.c = null;
        _this.c = a;
        return _this;
    }
    Object.defineProperty(WorkbookPasswordInfo.prototype, "f", {
        get: function () {
            if (this.b != null) {
                return true;
            }
            return this.i != null;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WorkbookPasswordInfo.prototype, "g", {
        get: function () {
            return this.d;
        },
        set: function (a) {
            if (a != this.d) {
                this.d = a;
                if (this.c != null) {
                    this.c();
                }
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(WorkbookPasswordInfo.prototype, "b", {
        get: function () {
            return this.a;
        },
        set: function (a) {
            if (a != this.a) {
                this.a = a;
                if (this.d == false) {
                    this.g = true;
                }
                else if (this.f == false) {
                    this.g = false;
                }
            }
        },
        enumerable: false,
        configurable: true
    });
    WorkbookPasswordInfo.prototype.e = function (a, b, c, d) {
        var _this = this;
        if (this.a != null) {
            return {
                ret: ((function () { var e = _this.a.d(a, b, c, d); a = e.p0; b = e.p1; c = e.p2; d = e.p3; return e.ret; })()),
                p0: a,
                p1: b,
                p2: c,
                p3: d
            };
        }
        a = null;
        b = c = null;
        d = toNullable(Number_$type, null);
        return {
            ret: false,
            p0: a,
            p1: b,
            p2: c,
            p3: d
        };
    };
    WorkbookPasswordInfo.prototype.h = function () {
        if (this.a != null) {
            return this.a.f();
        }
        return toNullable(Number_$type, null);
    };
    WorkbookPasswordInfo.prototype.j = function (a) {
        this.d = a;
    };
    WorkbookPasswordInfo.prototype.k = function (a) {
        if (this.g == false) {
            this.b = null;
            this.g = true;
        }
    };
    WorkbookPasswordInfo.prototype.l = function () {
        this.i = null;
        this.a = null;
        this.g = false;
    };
    WorkbookPasswordInfo.$t = markType(WorkbookPasswordInfo, 'WorkbookPasswordInfo');
    return WorkbookPasswordInfo;
}(Base));
export { WorkbookPasswordInfo };
