// igniteui-react-excel/esm5/lib/WorkbookThemeColorType.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Represents the various theme colors in a workbook.
 * @see [[WorkbookColorInfo.themeColorType]]
 */
export var WorkbookThemeColorType = /*@__PURE__*/ (function (WorkbookThemeColorType) {
    /**
     * Represents the Text/Background - Light 1 theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["Light1"] = 0] = "Light1";
    /**
     * Represents the Text/Background - Dark 1 theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["Dark1"] = 1] = "Dark1";
    /**
     * Represents the Text/Background - Light 2 theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["Light2"] = 2] = "Light2";
    /**
     * Represents the Text/Background - Dark 2 theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["Dark2"] = 3] = "Dark2";
    /**
     * Represents the Accent 1 theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["Accent1"] = 4] = "Accent1";
    /**
     * Represents the Accent 2 theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["Accent2"] = 5] = "Accent2";
    /**
     * Represents the Accent 3 theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["Accent3"] = 6] = "Accent3";
    /**
     * Represents the Accent 4 theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["Accent4"] = 7] = "Accent4";
    /**
     * Represents the Accent 5 theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["Accent5"] = 8] = "Accent5";
    /**
     * Represents the Accent 6 theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["Accent6"] = 9] = "Accent6";
    /**
     * Represents the Hyperlink theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["Hyperlink"] = 10] = "Hyperlink";
    /**
     * Represents the Followed Hyperlink theme color.
     */
    WorkbookThemeColorType[WorkbookThemeColorType["FollowedHyperlink"] = 11] = "FollowedHyperlink";
    return WorkbookThemeColorType;
})({});
/**
 * @hidden
 */
export var WorkbookThemeColorType_$type = /*@__PURE__*/ markEnum('WorkbookThemeColorType', 'Light1,0|Dark1,1|Light2,2|Dark2,3|Accent1,4|Accent2,5|Accent3,6|Accent4,7|Accent5,8|Accent6,9|Hyperlink,10|FollowedHyperlink,11');
