// igniteui-react-excel/esm5/lib/WorkItem_WorkItemCatch$1.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { __extends } from "tslib";
import { WorkItem } from "./WorkItem";
import { typeCast, markType } from "igniteui-react-core";
import { WorkItem_ExceptionInfo } from "./WorkItem_ExceptionInfo";
/**
 * @hidden
 */
var WorkItem_WorkItemCatch$1 = /** @class */ /*@__PURE__*/ (function (_super) {
    __extends(WorkItem_WorkItemCatch$1, _super);
    function WorkItem_WorkItemCatch$1($tException, a, b, c) {
        var _this = _super.call(this, 1, b, true) || this;
        _this.$tException = null;
        _this.ad = null;
        _this.ae = false;
        _this.$tException = $tException;
        _this.$type = _this.$type.specialize(_this.$tException);
        _this.ad = a;
        _this.ae = c;
        return _this;
    }
    WorkItem_WorkItemCatch$1.prototype.get_s = function () {
        return this.ae;
    };
    WorkItem_WorkItemCatch$1.prototype.p = function (a) {
        return true;
    };
    WorkItem_WorkItemCatch$1.prototype.u = function (a) {
        var b = typeCast(this.$tException, a.b);
        if (b != null) {
            try {
                this.ad(b);
                a = null;
            }
            catch (c) {
                a = new WorkItem_ExceptionInfo(c);
            }
            return {
                ret: true,
                p0: a
            };
        }
        return {
            ret: false,
            p0: a
        };
    };
    WorkItem_WorkItemCatch$1.$t = markType(WorkItem_WorkItemCatch$1, 'WorkItem_WorkItemCatch$1', WorkItem.$);
    return WorkItem_WorkItemCatch$1;
}(WorkItem));
export { WorkItem_WorkItemCatch$1 };
