// igniteui-react-excel/esm5/lib/WorksheetCellFormatOptions.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { markEnum } from "igniteui-react-core";
/**
 * Flagged enumeration which indicates which groups of formatting properties are used in a format.
 * @see [[IWorksheetCellFormat.formatOptions]]
 */
export var WorksheetCellFormatOptions = /*@__PURE__*/ (function (WorksheetCellFormatOptions) {
    /**
     * No format properties are used on the format.
     */
    WorksheetCellFormatOptions[WorksheetCellFormatOptions["None"] = 0] = "None";
    /**
     * The [[IWorksheetCellFormat.formatString]] property is used on the format.
     */
    WorksheetCellFormatOptions[WorksheetCellFormatOptions["ApplyNumberFormatting"] = 1] = "ApplyNumberFormatting";
    /**
     * The [[IWorksheetCellFormat.alignment]], [[IWorksheetCellFormat.indent]], [[IWorksheetCellFormat.rotation]],
     * [[IWorksheetCellFormat.shrinkToFit]], [[IWorksheetCellFormat.verticalAlignment]], and
     * [[IWorksheetCellFormat.wrapText]] properties are used on the format.
     */
    WorksheetCellFormatOptions[WorksheetCellFormatOptions["ApplyAlignmentFormatting"] = 2] = "ApplyAlignmentFormatting";
    /**
     * The [[IWorksheetCellFormat.font]] property is used on the format.
     */
    WorksheetCellFormatOptions[WorksheetCellFormatOptions["ApplyFontFormatting"] = 4] = "ApplyFontFormatting";
    /**
     * The [[IWorksheetCellFormat.bottomBorderColorInfo]], [[IWorksheetCellFormat.bottomBorderStyle]],
     * [[IWorksheetCellFormat.diagonalBorderColorInfo]], [[IWorksheetCellFormat.diagonalBorders]],
     * [[IWorksheetCellFormat.diagonalBorderStyle]], [[IWorksheetCellFormat.leftBorderColorInfo]],
     * [[IWorksheetCellFormat.leftBorderStyle]], [[IWorksheetCellFormat.rightBorderColorInfo]],
     * [[IWorksheetCellFormat.rightBorderStyle]], [[IWorksheetCellFormat.topBorderColorInfo]], and
     * [[IWorksheetCellFormat.topBorderStyle]] properties are used on the format.
     */
    WorksheetCellFormatOptions[WorksheetCellFormatOptions["ApplyBorderFormatting"] = 8] = "ApplyBorderFormatting";
    /**
     * The [[IWorksheetCellFormat.fill]] property is used on the format.
     */
    WorksheetCellFormatOptions[WorksheetCellFormatOptions["ApplyFillFormatting"] = 16] = "ApplyFillFormatting";
    /**
     * The [[IWorksheetCellFormat.locked]] property is used on the format.
     */
    WorksheetCellFormatOptions[WorksheetCellFormatOptions["ApplyProtectionFormatting"] = 32] = "ApplyProtectionFormatting";
    /**
     * All properties are used on the format.
     */
    WorksheetCellFormatOptions[WorksheetCellFormatOptions["All"] = 63] = "All";
    return WorksheetCellFormatOptions;
})({});
/**
 * @hidden
 */
export var WorksheetCellFormatOptions_$type = /*@__PURE__*/ markEnum('WorksheetCellFormatOptions', 'None,0|ApplyNumberFormatting,1|ApplyAlignmentFormatting,2|ApplyFontFormatting,4|ApplyBorderFormatting,8|ApplyFillFormatting,16|ApplyProtectionFormatting,32|All,63');
