// igniteui-react-charts/esm2015/lib/AssigningSeriesStyleEventArgsBase.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let AssigningSeriesStyleEventArgsBase = /*@__PURE__*/ (() => {
    class AssigningSeriesStyleEventArgsBase extends Base {
        constructor() {
            super(...arguments);
            this.o = 0;
            this.n = 0;
            this.h = new Date();
            this.g = new Date();
            this.a = null;
            this.p = null;
            this.q = null;
            this.k = 0;
            this.i = 0;
            this.b = null;
            this.j = 0;
            this.l = 0;
            this.m = 0;
            this.d = false;
            this.c = false;
            this.e = false;
            this.f = false;
        }
    }
    AssigningSeriesStyleEventArgsBase.$t = /*@__PURE__*/ markType(AssigningSeriesStyleEventArgsBase, 'AssigningSeriesStyleEventArgsBase');
    return AssigningSeriesStyleEventArgsBase;
})();
