// igniteui-react-charts/esm2015/lib/CategoryAxisRenderingParameters.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { AxisRenderingParametersBase } from "./AxisRenderingParametersBase";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let CategoryAxisRenderingParameters = /*@__PURE__*/ (() => {
    class CategoryAxisRenderingParameters extends AxisRenderingParametersBase {
        constructor() {
            super(...arguments);
            this.ae = 0;
            this.ab = 0;
            this.ad = false;
            this.af = 0;
            this.ac = false;
        }
    }
    CategoryAxisRenderingParameters.$t = /*@__PURE__*/ markType(CategoryAxisRenderingParameters, 'CategoryAxisRenderingParameters', AxisRenderingParametersBase.$);
    return CategoryAxisRenderingParameters;
})();
