// igniteui-react-charts/esm2015/lib/DataLegendStylingColumnEventArgs.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let DataLegendStylingColumnEventArgs = /*@__PURE__*/ (() => {
    class DataLegendStylingColumnEventArgs extends Base {
        constructor() {
            super(...arguments);
            this.c = 0;
            this.d = 0;
            this.f = null;
            this.b = 0;
            this.h = null;
            this.a = 0;
            this.e = null;
            this.g = null;
            this.i = null;
            this.j = null;
            this.k = null;
            this.l = null;
        }
    }
    DataLegendStylingColumnEventArgs.$t = /*@__PURE__*/ markType(DataLegendStylingColumnEventArgs, 'DataLegendStylingColumnEventArgs');
    return DataLegendStylingColumnEventArgs;
})();
