// igniteui-react-charts/esm2015/lib/DomainChart.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, INotifyPropertyChanged_$type, BaseError, PropertyChangedEventArgs, Delegate_$type, fromEnum, String_$type, runOn, delegateCombine, EnumUtil, enumGetBox, typeCast, delegateRemove, markType, TypeRegistrar, getInstanceType } from "igniteui-react-core";
import { IDataLegendTarget_$type } from "igniteui-react-core";
import { ArrayExtension } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
import { Brush } from "igniteui-react-core";
import { XamDataChart } from "./XamDataChart";
import { DataSeriesType_$type } from "igniteui-react-core";
import { MarkerType_$type } from "./MarkerType";
import { Color } from "igniteui-react-core";
import { ColorUtil } from "igniteui-react-core";
import { DataToolTipLayerGroupingMode_$type } from "igniteui-react-core";
import { DataTooltipGroupedPositionX_$type } from "igniteui-react-core";
import { DataTooltipGroupedPositionY_$type } from "igniteui-react-core";
import { DataAbbreviationMode_$type } from "igniteui-react-core";
import { DataLegendValueMode_$type } from "igniteui-react-core";
import { Thickness } from "igniteui-react-core";
import { DataLegendHeaderDateMode_$type } from "igniteui-react-core";
import { DataLegendHeaderTimeMode_$type } from "igniteui-react-core";
import { DataLegendSummaryType_$type } from "igniteui-react-core";
import { LegendItemBadgeShape_$type } from "igniteui-react-core";
import { DataLegendUnitsMode_$type } from "igniteui-react-core";
import { DataLegendLabelMode_$type } from "igniteui-react-core";
import { ChartSortDescriptionCollection } from "./ChartSortDescriptionCollection";
import { FilterExpressionCollection } from "igniteui-react-core";
import { ChartSummaryDescriptionCollection } from "./ChartSummaryDescriptionCollection";
import { SortGroupParser } from "igniteui-react-core";
import { SummaryParser } from "igniteui-react-core";
import { BrushCollection } from "igniteui-react-core";
import { Rect } from "igniteui-react-core";
import { HorizontalAlignment_$type } from "igniteui-react-core";
import { UnknownValuePlotting_$type } from "igniteui-react-core";
import { SeriesOutlineMode_$type } from "./SeriesOutlineMode";
import { MarkerOutlineMode_$type } from "./MarkerOutlineMode";
import { MarkerFillMode_$type } from "./MarkerFillMode";
import { MarkerAutomaticBehavior_$type } from "./MarkerAutomaticBehavior";
import { TrendLineType_$type } from "igniteui-react-core";
import { ComputedPlotAreaMarginMode_$type } from "./ComputedPlotAreaMarginMode";
import { SeriesPlotAreaMarginHorizontalMode_$type } from "./SeriesPlotAreaMarginHorizontalMode";
import { SeriesPlotAreaMarginVerticalMode_$type } from "./SeriesPlotAreaMarginVerticalMode";
import { SeriesHighlightingMode_$type } from "./SeriesHighlightingMode";
import { SeriesHighlightingBehavior_$type } from "./SeriesHighlightingBehavior";
import { LegendHighlightingMode_$type } from "./LegendHighlightingMode";
import { LegendItemBadgeMode_$type } from "igniteui-react-core";
import { ToolTipType_$type } from "./ToolTipType";
import { CrosshairsDisplayMode_$type } from "./CrosshairsDisplayMode";
import { SeriesViewerScrollbarMode_$type } from "./SeriesViewerScrollbarMode";
import { SeriesViewerHorizontalScrollbarPosition_$type } from "./SeriesViewerHorizontalScrollbarPosition";
import { SeriesViewerVerticalScrollbarPosition_$type } from "./SeriesViewerVerticalScrollbarPosition";
import { Defaults } from "./Defaults";
import { Series } from "./Series";
import { DeviceUtils } from "igniteui-react-core";
import { Dictionary$2 } from "igniteui-react-core";
import { DataSeriesAdapter } from "igniteui-react-core";
import { ChartSortDescription } from "./ChartSortDescription";
import { ChartSummaryDescription } from "./ChartSummaryDescription";
import { ObservableCollection$1 } from "igniteui-react-core";
import { Random } from "igniteui-react-core";
import { ObservableColorCollection } from "igniteui-react-core";
import { BrushCollectionUtil } from "igniteui-react-core";
import { ChartSeriesEventArgs } from "./ChartSeriesEventArgs";
import { DomainChartSeriesPointerEventArgs } from "./DomainChartSeriesPointerEventArgs";
import { BrushUtil } from "igniteui-react-core";
import { IUnknownValuePlotting_$type } from "igniteui-react-core";
import { ITrendLineSeries_$type } from "./ITrendLineSeries";
import { isNaN_ } from "igniteui-react-core";
import { stringIsNullOrEmpty, stringEmpty, stringStartsWith } from "igniteui-react-core";
/**
 * @hidden
 */
export let DomainChart = /*@__PURE__*/ (() => {
    class DomainChart extends Base {
        constructor() {
            super();
            this.jt = null;
            this.jr = null;
            this.f8 = null;
            this.g0 = NaN;
            this.jq = null;
            this.hd = 0;
            this.he = 0;
            this.hf = 0;
            this.hc = 0;
            this.g8 = 0;
            this.ha = 0;
            this.g9 = 0;
            this.g7 = 0;
            this.tr = null;
            this.ts = null;
            this.gx = NaN;
            this.hg = NaN;
            this.g6 = NaN;
            this.gk = NaN;
            this.je = 0;
            this.dt = null;
            this.i5 = null;
            this.fy = false;
            this.b4 = null;
            this.kn = null;
            this.kp = null;
            this.dl = null;
            this.gc = null;
            this.propertyChanged = null;
            this.jd = 0;
            this.fs = false;
            this.bz = null;
            this.f7 = null;
            this.b3 = null;
            this.eu = false;
            this.br = 0;
            this.go = NaN;
            this.gp = NaN;
            this.gm = DeviceUtils.g(10);
            this.gn = DeviceUtils.g(10);
            this.bn = 0;
            this.bp = 0;
            this.en = true;
            this.e = null;
            this.b = null;
            this.d = null;
            this.a = null;
            this.a7 = 0;
            this.i9 = -1;
            this.ja = -1;
            this.bl = 0;
            this.kf = stringEmpty();
            this.eo = true;
            this.kg = null;
            this.f = null;
            this.uz = new Thickness(0, DeviceUtils.g(5));
            this.ep = true;
            this.kh = "N/A";
            this.eq = false;
            this.u0 = new Thickness(1, DeviceUtils.g(2), DeviceUtils.g(0), DeviceUtils.g(2), DeviceUtils.g(0));
            this.tm = null;
            this.d3 = null;
            this.j9 = null;
            this.c = null;
            this.j8 = stringEmpty();
            this.bb = 0;
            this.bd = 0;
            this.ka = null;
            this.te = null;
            this.ut = new Thickness(1, DeviceUtils.g(2), DeviceUtils.g(0), DeviceUtils.g(0), DeviceUtils.g(0));
            this.us = new Thickness(0, DeviceUtils.g(5));
            this.em = true;
            this.dv = null;
            this.th = null;
            this.dy = null;
            this.bh = 0;
            this.kc = null;
            this.uw = new Thickness(1, DeviceUtils.g(2), DeviceUtils.g(0), DeviceUtils.g(0), DeviceUtils.g(0));
            this.uv = new Thickness(0, DeviceUtils.g(5));
            this.tj = null;
            this.d0 = null;
            this.kb = null;
            this.tg = null;
            this.dx = null;
            this.kd = null;
            this.ti = null;
            this.dz = null;
            this.ur = new Thickness(0, DeviceUtils.g(0));
            this.ca = 0;
            this.bj = 0;
            this.ke = null;
            this.tl = null;
            this.uy = new Thickness(0, DeviceUtils.g(0));
            this.d2 = null;
            this.ux = new Thickness(0, DeviceUtils.g(0));
            this.tk = null;
            this.d1 = null;
            this.bf = 0;
            this.tf = null;
            this.uu = new Thickness(1, DeviceUtils.g(8), DeviceUtils.g(0), DeviceUtils.g(0), DeviceUtils.g(0));
            this.dw = null;
            this.seriesAdded = null;
            this.seriesRemoved = null;
            this.seriesPointerEnter = null;
            this.seriesPointerLeave = null;
            this.seriesPointerMove = null;
            this.seriesPointerDown = null;
            this.seriesPointerUp = null;
            this.calloutStyleUpdating = null;
            this.calloutRenderStyleUpdating = null;
            this.gb = null;
            this.ee = null;
            this.eh = false;
            this.f5 = new Dictionary$2(String_$type, Base.$, 0);
            this.f6 = new List$1(String_$type, 0);
            this.as = null;
            this.ao = null;
            this.an = null;
            this.dn = null;
            this.a0 = null;
            this.dr = null;
            this.ds = null;
            this.e2 = true;
            this.e1 = true;
            this.e3 = true;
            this.e0 = true;
            this.ar = null;
            this.aq = null;
            this.ap = null;
            this.dp = null;
            this.az = null;
            this.kk = null;
            this.ki = null;
            this.kj = null;
            this.kl = null;
            this.jp = null;
            this.jo = null;
            this.h = null;
            this.g = null;
            this.t = null;
            this.w = null;
            this.r = null;
            this.s = null;
            this.b1 = null;
            this.et = true;
            this.ew = true;
            this.ey = false;
            this.e4 = false;
            this.ev = false;
            this.uo = new Rect(0, 0, 0, 1, 1);
            this.ko = null;
            this.km = null;
            this.s5 = 1;
            this.s4 = 1;
            this.dj = 0;
            this.g5 = NaN;
            this.hb = NaN;
            this.c1 = 0;
            this.ci = 0;
            this.cg = 0;
            this.gy = NaN;
            this.gz = NaN;
            this.cn = null;
            this.ce = 4;
            this.u = null;
            this.v = null;
            this.jc = 400;
            this.gj = NaN;
            this.eg = false;
            this.x = null;
            this.dh = 0;
            this.hh = DeviceUtils.g(1.5);
            this.ef = true;
            this.g2 = NaN;
            this.g4 = NaN;
            this.g3 = NaN;
            this.g1 = NaN;
            this.a3 = 0;
            this.c3 = 0;
            this.c5 = 0;
            this.cz = 0;
            this.cx = 0;
            this.b6 = 0;
            this.cb = 0;
            this.b8 = 0;
            this.jf = 7;
            this.df = 4;
            this.a5 = 0;
            this.el = false;
            this.td = null;
            this.gl = NaN;
            this.tc = null;
            this.ek = true;
            this.s8 = null;
            this.ta = null;
            this.s9 = null;
            this.tb = null;
            this.i7 = -1;
            this.i8 = -1;
            this.ex = false;
            this.ez = true;
            this.er = false;
            this.tn = null;
            this.to = null;
            this.jb = -1;
            this.ej = false;
            this.ei = false;
            this.gg = null;
            this.ah = null;
            this.j6 = null;
            this.j7 = null;
            this.j5 = "CalloutLabel";
            this.j4 = "CalloutContent";
            this.c9 = 0;
            this.da = 0;
            this.c7 = 0;
            this.dd = 0;
            this.tp = null;
            this.tq = null;
            this.gu = NaN;
            this.gt = NaN;
            this.gq = NaN;
            this.gr = NaN;
            this.gs = NaN;
            this.gw = NaN;
            this.gv = NaN;
            this.es = true;
            this.tt = null;
            this.tu = null;
            this.hl = NaN;
            this.hk = NaN;
            this.hi = NaN;
            this.ho = NaN;
            this.hj = NaN;
            this.hn = NaN;
            this.hm = NaN;
            this.e5 = true;
            this.hq = NaN;
            this.hp = NaN;
            this.hs = NaN;
            this.hr = NaN;
            this.gb = new List$1(Delegate_$type, 0);
            this.ro();
            this.q4();
        }
        provideContainer(a) {
            if (a == null) {
                this.r4((b) => {
                    b.destroy();
                    b.provideContainer(null);
                });
                return;
            }
            let b = this.jt;
            this.jt = a;
            if (this.ee != null && this.jt != null && b == null) {
                this.jt.setTimeout(this.ee, 0);
                this.ee = null;
            }
            let c = ((() => {
                let $ret = new XamDataChart();
                $ret.ef = true;
                return $ret;
            })());
            c.provideContainer(a);
            this.dataChart = c;
            this.rp();
        }
        destroy() {
            this.provideContainer(null);
        }
        rp() {
            if (this.dataChart == null) {
                return;
            }
            let a = this.dataChart.cu.j.e();
            this.aa = a.a;
            this.ae = a.f;
            this.ac = a.b;
            this.ad = a.c;
            this.ag = a.g;
            this.hv = Defaults.j;
            this.rq();
        }
        rq() {
        }
        rm(a) {
            if (a == null) {
                return;
            }
            a.g9 = this.ir;
            a.g7 = this.ip;
            a.g8 = this.iq;
            a.g6 = this.io;
            a.sv = this.uk;
            a.g3 = this.ij;
            a.g5 = this.il;
            a.g4 = this.ik;
            a.g2 = this.ii;
            a.su = this.uj;
            if (!isNaN_(this.ia)) {
                a.gx = this.ia;
            }
        }
        rj() {
        }
        rs(a, b) {
            a.ho = this.jm;
            a.cf = this.du;
        }
        sd(a, b, c) {
            switch (a) {
                case "TitleTopMargin":
                    if (this.dataChart != null) {
                        this.dataChart.g9 = this.ir;
                    }
                    break;
                case "TitleLeftMargin":
                    if (this.dataChart != null) {
                        this.dataChart.g7 = this.ip;
                    }
                    break;
                case "PixelScalingRatio":
                    if (this.dataChart != null) {
                        this.r4((d) => d.gx = this.ia);
                    }
                    break;
                case "TitleRightMargin":
                    if (this.dataChart != null) {
                        this.dataChart.g8 = this.iq;
                    }
                    break;
                case "TitleBottomMargin":
                    if (this.dataChart != null) {
                        this.dataChart.g6 = this.io;
                    }
                    break;
                case "TitleTextColor":
                    if (this.dataChart != null) {
                        this.dataChart.sv = this.uk;
                    }
                    break;
                case "SubtitleLeftMargin":
                    if (this.dataChart != null) {
                        this.dataChart.g3 = this.ij;
                    }
                    break;
                case "SubtitleTopMargin":
                    if (this.dataChart != null) {
                        this.dataChart.g5 = this.il;
                    }
                    break;
                case "SubtitleRightMargin":
                    if (this.dataChart != null) {
                        this.dataChart.g4 = this.ik;
                    }
                    break;
                case "SubtitleBottomMargin":
                    if (this.dataChart != null) {
                        this.dataChart.g2 = this.ii;
                    }
                    break;
                case "SubtitleTextColor":
                    if (this.dataChart != null) {
                        this.dataChart.su = this.uj;
                    }
                    break;
                case "LeftMargin":
                    if (this.dataChart != null) {
                        this.dataChart.gw = this.h7;
                    }
                    break;
                case "TopMargin":
                    if (this.dataChart != null) {
                        this.dataChart.ha = this.is;
                    }
                    break;
                case "RightMargin":
                    if (this.dataChart != null) {
                        this.dataChart.g1 = this.ih;
                    }
                    break;
                case "BottomMargin":
                    if (this.dataChart != null) {
                        this.dataChart.gj = this.hu;
                    }
                    break;
                case "TransitionDuration":
                    this.rf((d) => d.ho = this.jm);
                    break;
                case "TransitionEasingFunction":
                    this.rf((d) => d.cf = this.du);
                    break;
            }
        }
        get tooltipTemplate() {
            return ArrayExtension.getModulus$1(Base.$, this.tooltipTemplates, 0);
        }
        set tooltipTemplate(a) {
            this.tooltipTemplates = a == null ? null : new List$1(Base.$, 1, [a]);
        }
        get tooltipTemplates() {
            return this.f8;
        }
        set tooltipTemplates(a) {
            let b = this.tooltipTemplates;
            if (a != b) {
                this.f8 = a;
                this.sc("TooltipTemplates", b, this.tooltipTemplates);
            }
        }
        get ia() {
            return this.g0;
        }
        set ia(a) {
            let b = this.g0;
            if (a != b) {
                this.g0 = a;
                this.sc("PixelScalingRatio", b, this.ia);
            }
        }
        get toolTip() {
            return this.jq;
        }
        set toolTip(a) {
            let b = this.toolTip;
            if (b != a) {
                this.jq = a;
                this.sc("ToolTip", b, this.toolTip);
            }
        }
        get ip() {
            return this.hd;
        }
        set ip(a) {
            let b = this.ip;
            if (a != b) {
                this.hd = a;
                this.sc("TitleLeftMargin", b, this.ip);
            }
        }
        get iq() {
            return this.he;
        }
        set iq(a) {
            let b = this.iq;
            if (a != b) {
                this.he = a;
                this.sc("TitleRightMargin", b, this.iq);
            }
        }
        get ir() {
            return this.hf;
        }
        set ir(a) {
            let b = this.ir;
            if (a != b) {
                this.hf = a;
                this.sc("TitleTopMargin", b, this.ir);
            }
        }
        get io() {
            return this.hc;
        }
        set io(a) {
            let b = this.io;
            if (a != b) {
                this.hc = a;
                this.sc("TitleBottomMargin", b, this.io);
            }
        }
        get ij() {
            return this.g8;
        }
        set ij(a) {
            let b = this.ij;
            if (a != b) {
                this.g8 = a;
                this.sc("SubtitleLeftMargin", b, this.ij);
            }
        }
        get il() {
            return this.ha;
        }
        set il(a) {
            let b = this.il;
            if (a != b) {
                this.ha = a;
                this.sc("SubtitleTopMargin", b, this.il);
            }
        }
        get ik() {
            return this.g9;
        }
        set ik(a) {
            let b = this.ik;
            if (a != b) {
                this.g9 = a;
                this.sc("SubtitleRightMargin", b, this.ik);
            }
        }
        get ii() {
            return this.g7;
        }
        set ii(a) {
            let b = this.ii;
            if (a != b) {
                this.g7 = a;
                this.sc("SubtitleBottomMargin", b, this.ii);
            }
        }
        get uj() {
            return this.tr;
        }
        set uj(a) {
            let b = this.uj;
            if (a != b) {
                this.tr = a;
                this.sc("SubtitleTextColor", b, this.uj);
            }
        }
        get uk() {
            return this.ts;
        }
        set uk(a) {
            let b = this.uk;
            if (a != b) {
                this.ts = a;
                this.sc("TitleTextColor", b, this.uk);
            }
        }
        get_h7() {
            return this.gx;
        }
        set_h7(a) {
            let b = this.h7;
            if (a != b) {
                this.gx = a;
                this.sc("LeftMargin", b, this.h7);
            }
        }
        get h7() {
            return this.get_h7();
        }
        set h7(a) {
            this.set_h7(a);
        }
        get is() {
            return this.hg;
        }
        set is(a) {
            let b = this.is;
            if (a != b) {
                this.hg = a;
                this.sc("TopMargin", b, this.is);
            }
        }
        get ih() {
            return this.g6;
        }
        set ih(a) {
            let b = this.ih;
            if (a != b) {
                this.g6 = a;
                this.sc("RightMargin", b, this.ih);
            }
        }
        get hu() {
            return this.gk;
        }
        set hu(a) {
            let b = this.hu;
            if (a != b) {
                this.gk = a;
                this.sc("BottomMargin", b, this.hu);
            }
        }
        get jm() {
            return this.je;
        }
        set jm(a) {
            let b = this.jm;
            if (a != b) {
                this.je = a;
                this.sc("TransitionDuration", b, this.jm);
            }
        }
        get du() {
            return this.dt;
        }
        set du(a) {
            let b = this.du;
            if (a != b) {
                this.dt = a;
                this.sc("TransitionEasingFunction", b, this.du);
            }
        }
        rn(a) {
            if (this.dataChart == null) {
                return;
            }
            a.kn = this.p2;
            a.kb = this.pr;
        }
        rt(a, b) {
            this.ru(a, b);
        }
        rk() {
        }
        ru(a, b) {
            if (a == null) {
                return;
            }
            if (this.dg == 3) {
                return;
            }
            let toolTipTemplate_ = ArrayExtension.getModulus$1(Base.$, this.tooltipTemplates, b);
            if (toolTipTemplate_ == null) {
                a.et = true;
            }
            else {
                let series_ = a;
                a.et = false;
                series_.externalObject.tooltipTemplate = toolTipTemplate_;
            }
        }
        s2(a, b, c) {
        }
        jv(a) {
            return a;
            return null;
        }
        i6(a) {
            return null;
        }
        se(a, b, c) {
            switch (a) {
                case "SubtitleTextStyle":
                    if (this.dataChart != null) {
                        this.dataChart.kb = this.pr;
                    }
                    break;
                case "TitleTextStyle":
                    if (this.dataChart != null) {
                        this.dataChart.kn = this.p2;
                    }
                    break;
                case "TooltipTemplate":
                    if (this.fy) {
                        return;
                    }
                    this.fy = true;
                    this.tooltipTemplate = this.jv(this.tooltipTemplate);
                    this.fy = false;
                    if (this.dataChart != null) {
                        this.dataChart.h0 = this.tooltipTemplate;
                    }
                    this.toolTip = this.tooltipTemplate;
                    break;
                case "ToolTip":
                    this.rg((d, e) => this.ru(d, e));
                    break;
                case "TooltipTemplates":
                    if (this.tooltipTemplates == null) {
                        break;
                    }
                    this.fy = true;
                    this.rg((d, e) => {
                        let f = this.jv(ArrayExtension.getModulus$1(Base.$, this.tooltipTemplates, e));
                        let series_ = d;
                        let templ_ = f;
                        series_.externalObject.tooltipTemplate = templ_;
                    });
                    this.fy = false;
                    break;
            }
        }
        get b5() {
            return this.b4;
        }
        set b5(a) {
            let b = this.b5;
            if (a != b) {
                this.b4 = a;
                this.sc("Widget", b, this.b5);
            }
        }
        get pr() {
            return this.kn;
        }
        set pr(a) {
            let b = this.pr;
            if (a != b) {
                this.kn = a;
                this.sc("SubtitleTextStyle", b, this.pr);
            }
        }
        get p2() {
            return this.kp;
        }
        set p2(a) {
            let b = this.p2;
            if (a != b) {
                this.kp = a;
                this.sc("TitleTextStyle", b, this.p2);
            }
        }
        get dataChart() {
            return this.dl;
        }
        set dataChart(a) {
            let b = this.dataChart;
            if (b != a) {
                this.dl = a;
                this.sc("DataChart", b, this.dataChart);
            }
        }
        onDetachedFromUI() {
            this.r4((a) => a.onDetachedFromUI());
        }
        onAttachedToUI() {
            this.r4((a) => a.onAttachedToUI());
        }
        ro() {
            this.gc = new List$1(MarkerType_$type, 0);
            this.ba = new DataSeriesAdapter().e;
            this.as = new ChartSortDescriptionCollection();
            this.ao = new ChartSortDescriptionCollection();
            this.an = new ChartSortDescriptionCollection();
            this.a0 = new ChartSummaryDescriptionCollection();
            this.dn = new FilterExpressionCollection();
            this.ay.e = this.ba.sortDescriptions;
            this.at.e = this.ba.groupDescriptions;
            this.dq.syncTarget = this.ba.filterExpressions;
            this.a2.e = this.ba.summaryDescriptions;
            this.dr = new SortGroupParser();
            this.ds = new SummaryParser();
            this.ar = ((() => {
                let $ret = new ChartSortDescriptionCollection();
                $ret.i = true;
                return $ret;
            })());
            this.aq = ((() => {
                let $ret = new ChartSortDescriptionCollection();
                $ret.i = true;
                return $ret;
            })());
            this.ap = ((() => {
                let $ret = new ChartSortDescriptionCollection();
                $ret.i = true;
                return $ret;
            })());
            this.az = ((() => {
                let $ret = new ChartSummaryDescriptionCollection();
                $ret.i = true;
                return $ret;
            })());
            this.dp = ((() => {
                let $ret = new FilterExpressionCollection();
                $ret.shouldDetachOnTargetChange = true;
                return $ret;
            })());
            if (this.e2) {
                if (this.ay.j.count == 0) {
                    this.ar.e = this.ba.sortDescriptions;
                }
                if (this.au.j.count == 0) {
                    this.aq.e = this.ba.postSortDescriptions;
                }
                this.e2 = false;
            }
            if (this.e1) {
                if (this.at.j.count == 0) {
                    this.ap.e = this.ba.groupDescriptions;
                }
                this.e1 = false;
            }
            if (this.e3) {
                if (this.a2.j.count == 0) {
                    this.az.e = this.ba.summaryDescriptions;
                }
                this.e3 = false;
            }
            if (this.q != null) {
                this.ba.includedProperties = this.q;
            }
            if (this.o != null) {
                this.ba.excludedProperties = this.o;
            }
            this.ba.supportedSeriesTypes = this.ge.toArray();
            this.ba.rules.clear();
            for (let a of fromEnum(this.gd)) {
                this.ba.rules.add(a);
            }
            let b = new List$1(String_$type, 0);
            for (let c of fromEnum(this.ba.rules)) {
                b.add(getInstanceType(c).typeName);
            }
            let d = this.ba.dataSeries;
            d.collectionChanged = delegateCombine(d.collectionChanged, runOn(this, this.r7));
        }
        fu(a) {
            return a != 24 && this.ge.contains(a);
        }
        notifyResized() {
            this.r4((a) => a.notifyContainerResized());
        }
        r2() {
            this.r4((a) => {
                for (let b = 0; b < a.series.count; b++) {
                    a.series._inner[b].lu();
                }
            });
        }
        q9() {
            if (this.dataChart == null) {
                return;
            }
            if (this.fs) {
                this.sw();
            }
            this.dataChart.flush();
        }
        rh() {
            if (this.dataChart == null) {
                return;
            }
            this.dataChart.l0();
        }
        sk(a) {
            if (this.propertyChanged != null) {
                this.propertyChanged(this, new PropertyChangedEventArgs(a));
            }
        }
        rb(a) {
            this.rc((b, c) => a(b));
        }
        rc(a) {
            if (this.dataChart == null) {
                return;
            }
            for (let b = 0; b < this.dataChart.series.count; b++) {
                a(this.dataChart.series._inner[b], b);
            }
        }
        rf(a) {
            this.rg((b, c) => a(b));
        }
        rg(a) {
            if (this.dataChart == null) {
                return;
            }
            for (let b = 0; b < this.dataChart.series.count; b++) {
                let c = this.dataChart.series._inner[b];
                if (!c.db) {
                    a(c, b);
                }
            }
        }
        rd(a) {
            this.re((b, c) => a(b));
        }
        re(a) {
            if (this.dataChart == null) {
                return;
            }
            for (let b = 0; b < this.dataChart.series.count; b++) {
                let c = this.dataChart.series._inner[b];
                if (c.dc) {
                    a(c, b);
                }
            }
        }
        st() {
            this.rb((a) => a.mo());
        }
        r4(a) {
            this.r5(a, true);
        }
        r5(a, b) {
            this.r6(a, b, null);
        }
        r6(a, b, c) {
            if (this.dataChart == null) {
                return;
            }
            a(this.dataChart);
        }
        fv(a) {
            if (!this.fu(a.suggestedSeries)) {
                return false;
            }
            let b = a.data;
            if (b == null) {
                return false;
            }
            return true;
        }
        fn() {
            if (this.dataChart == null) {
                return false;
            }
            return true;
        }
        ny(a) {
            let b = "series_" + (this.dataChart.series.count + 1) + "_" + EnumUtil.getName(DataSeriesType_$type, a.suggestedSeries);
            return b;
        }
        nz(a, b) {
            if (this.fq && this.f3) {
                let c = "";
                let d = 5 + Math.round(DomainChart.j2.nextDouble() * 4);
                for (let e = 0; e < d; e++) {
                    c = c + "\u2588";
                }
                return c;
            }
            if (stringIsNullOrEmpty(a.title)) {
                let f = EnumUtil.getName(DataSeriesType_$type, b);
                return f + (this.dataChart.series.count + 1);
            }
            else {
                return a.title;
            }
        }
        nx() {
            return stringEmpty();
        }
        sw() {
            if (this.dataChart == null) {
                return;
            }
            this.fs = false;
            this.sy();
            if (this.actualItemsSource == null) {
                return;
            }
            let a = new List$1(MarkerType_$type, 0);
            for (let b of fromEnum(this.ba.dataSeries)) {
                a.add(this.ck(b.suggestedMarker));
            }
            this.gc = a;
            let c = 0;
            for (let d of fromEnum(this.ba.dataSeries)) {
                if (!this.fv(d)) {
                    continue;
                }
                if (d.index < 0) {
                    d.index = c;
                }
                this.su(d);
                this.sv(d);
                c++;
            }
            this.s3();
            this.rc(runOn(this, this.rr));
        }
        cw(seriesType_) {
            return (this.externalObject.createSeries(seriesType_).i);
        }
        cp() {
            return this.cw(28);
        }
        cr() {
            return this.cw(27);
        }
        ct() {
            return this.cw(29);
        }
        cu() {
            return this.cw(25);
        }
        cq() {
            return this.cw(26);
        }
        cs() {
            return this.cw(32);
        }
        get b0() {
            if (TypeRegistrar.isRegistered("AnnotationLayerProxy")) {
                return TypeRegistrar.create("AnnotationLayerProxy");
            }
            else {
                throw new BaseError(1, "annotation layer module is not loaded, but is required.");
            }
        }
        get fj() {
            return (TypeRegistrar.isRegistered("AnnotationLayerProxy"));
        }
        fl(a) {
            if (!this.fj) {
                return false;
            }
            return this.b0.matchesType(a);
        }
        s3() {
            let a = this.ga("", true);
            if (a != null) {
                for (let b = 0; b < a.count; b++) {
                    this.dataChart.series.add(a._inner[b]);
                }
            }
        }
        fo() {
            return this.a6 == 0 || this.a6 == 4 || this.a6 == 2;
        }
        fp() {
            return this.a6 == 0 || this.a6 == 4 || this.a6 == 3;
        }
        get_fm() {
            return this.gg == null && this.ej && (this.j6 == null || this.j7 == null);
        }
        get fm() {
            return this.get_fm();
        }
        ga(a, b) {
            if (!this.fj) {
                return null;
            }
            let c = new List$1(Series.$, 0);
            if (b) {
                if (this.ej) {
                    let d = 1;
                    if (this.gg == null) {
                        if (this.j6 != null || this.j7 != null || this.j5 != null || this.j4 != null) {
                            for (let e of fromEnum(this.dataChart.series)) {
                                let f = this.cp();
                                this.b0.setTargetSeries(f, e);
                                this.b0.setCollisionChannel(f, "MainChartCollisionChannel");
                                this.s1(f, e, this.ba.dataSeries._inner[d - 1]);
                                f.name = a + getInstanceType(f).typeName + d.toString();
                                c.add(f);
                                d++;
                            }
                            ;
                        }
                        else {
                            let g = this.cp();
                            this.q3(g, this.f2);
                            g.name = a + getInstanceType(g).typeName + d.toString();
                            c.add(g);
                        }
                    }
                    else {
                        let h = this.cp();
                        this.s1(h, null, this.ba.dataSeries._inner[d - 1]);
                        h.name = a + getInstanceType(h).typeName + d.toString();
                        c.add(h);
                    }
                }
            }
            switch (this.a5) {
                case 0:
                case 4:
                case 3:
                case 2:
                    {
                        let i = this.cr();
                        this.b0.setCrosshairThickness(i, this.gl);
                        this.b0.setCrosshairUseInterpolation(i, !this.el);
                        this.b0.setCrosshairHorizontalLineVisibility(i, this.fo() ? 0 : 1);
                        this.b0.setCrosshairHorizontalLineStroke(i, this.tc);
                        this.b0.setCrosshairVerticalLineVisibility(i, this.fp() ? 0 : 1);
                        this.b0.setCrosshairVerticalLineStroke(i, this.td);
                        this.b0.setCrosshairOnAxesEnabled(i, this.ek);
                        this.b0.setCrosshairXAxisBackground(i, this.s8);
                        this.b0.setCrosshairXAxisForeground(i, this.s9);
                        this.b0.setCrosshairYAxisBackground(i, this.ta);
                        this.b0.setCrosshairYAxisForeground(i, this.tb);
                        this.b0.setCrosshairXAxisPrecision(i, this.i7);
                        this.b0.setCrosshairYAxisPrecision(i, this.i8);
                        c.add(i);
                        break;
                    }
                case 1:
                    {
                        break;
                    }
            }
            if (this.er) {
                let j = this.ct();
                this.b0.setFinalValueBackground(j, this.tn);
                this.b0.setFinalValueForeground(j, this.to);
                this.b0.setFinalValuePrecision(j, this.jb);
                c.add(j);
            }
            if (b) {
                switch (this.df) {
                    case 1:
                        {
                            let k = this.cu();
                            c.add(k);
                            break;
                        }
                    case 2:
                        {
                            let l = this.cq();
                            c.add(l);
                            break;
                        }
                    case 4:
                        {
                            let m = this.cs();
                            c.add(m);
                            break;
                        }
                    case 3:
                        {
                            this.rb((n) => n.et = false);
                            break;
                        }
                    case 0:
                        {
                            this.rb((n) => n.et = true);
                            break;
                        }
                }
            }
            for (let n of fromEnum(c)) {
                if (stringIsNullOrEmpty(n.name) == false) {
                    continue;
                }
                n.name = a + getInstanceType(n).typeName;
            }
            return c;
        }
        q3(a, b) {
            this.b0.setIsAutoCalloutBehaviorEnabled(a, true);
            this.b0.setGreedyCollisionMode(a, true);
            this.b0.setUseDedicatedLanes(a, b);
        }
        s1(a, b, c) {
            a.itemsSource = this.gg != null ? this.gg : b.itemsSource;
            this.b0.setXMemberPath(a, this.j6);
            this.b0.setYMemberPath(a, this.j7);
            this.b0.setLabelMemberPath(a, this.j5);
            this.b0.setContentMemberPath(a, this.j4);
            if (this.ah != null) {
                this.b0.setCalloutsAllowedPositions(a, this.ah);
            }
        }
        sy() {
            this.sz(this.dataChart);
        }
        sz(a) {
            if (a != null && a == this.dataChart) {
                this.sx();
            }
            if (a == null || a.series.count == 0) {
                return;
            }
            this.rb((b) => {
                if (this.fl(b) && this.b0.isCalloutLayer(b)) {
                    this.b0.removeCalloutStyleUpdating(b, runOn(this, this.sj));
                }
                if (this.fl(b) && this.b0.isCalloutLayer(b)) {
                    this.b0.removeCalloutRenderStyleUpdating(b, runOn(this, this.si));
                }
                this.sr(b);
            });
            for (let b of fromEnum(a.series)) {
                this.s0(b);
            }
            a.series.clear();
        }
        s0(a) {
            a.itemsSource = null;
            a.legend = null;
            a.kx();
        }
        r7(a, b) {
            if (this.dataChart == null) {
                return;
            }
            let c = b.oldItems == null ? "null" : b.oldItems.count.toString();
            let d = b.newItems == null ? "null" : b.newItems.count.toString();
            if (this.fs) {
                return;
            }
            let e = this.sh(runOn(this, this.q7), this.fs);
            this.fs = e.p1;
        }
        q7() {
            this.sw();
        }
        uc(a) {
            return ArrayExtension.getModulus$1(Brush.$, this.y, a);
        }
        uf(a) {
            return ArrayExtension.getModulus$1(Brush.$, this.z, a);
        }
        ud(a) {
            return ArrayExtension.getModulus$1(Brush.$, this.ac, a);
        }
        ue(a) {
            return ArrayExtension.getModulus$1(Brush.$, this.ad, a);
        }
        un(a) {
            let b = new Color();
            if (this.y == null || this.y.count == 0) {
                b = ArrayExtension.getModulus$1(Color.$, this.f9, a);
            }
            else {
                let c = ArrayExtension.getModulus$1(Brush.$, this.y, a);
                if (c == null) {
                    b = ArrayExtension.getModulus$1(Color.$, this.f9, a);
                }
                else {
                    b = ColorUtil.q(c);
                }
            }
            return b;
        }
        ed(a) {
            let b = new ObservableColorCollection();
            b.add(ColorUtil.v(a, 0.5));
            b.add(a);
            return b;
        }
        ab(a) {
            let b = this.ed(a);
            let c = BrushCollectionUtil.a(b);
            return c;
        }
        get f9() {
            if (this.f7 != null) {
                return this.f7;
            }
            this.f7 = new List$1(Color.$, 0);
            this.f7.add(ColorUtil.r("#8BDC5C"));
            this.f7.add(ColorUtil.r("#8961A9"));
            this.f7.add(ColorUtil.r("#6DB1FF"));
            this.f7.add(ColorUtil.r("#82E9D9"));
            this.f7.add(ColorUtil.r("#EA3C63"));
            this.f7.add(ColorUtil.r("#735656"));
            this.f7.add(ColorUtil.r("#F8CE4F"));
            this.f7.add(ColorUtil.r("#A8A8B7"));
            this.f7.add(ColorUtil.r("#E051A9"));
            this.f7.add(ColorUtil.r("#FF903B"));
            return this.f7;
        }
        get_f2() {
            return false;
        }
        get f2() {
            return this.get_f2();
        }
        ju() {
            if (this.dataChart == null) {
                return null;
            }
            return this.dataChart.vw();
        }
        rv(a, b) {
        }
        rw(a, b, c) {
        }
        ck(a) {
            if (a == 2) {
                return 3;
            }
            else if (a == 12) {
                return 13;
            }
            else if (a == 6) {
                return 7;
            }
            else if (a == 8) {
                return 9;
            }
            else if (a == 11) {
                return 12;
            }
            else if (a == 1) {
                return 1;
            }
            else if (a == 7) {
                return 8;
            }
            else if (a == 10) {
                return 11;
            }
            else if (a == 4) {
                return 5;
            }
            else if (a == 5) {
                return 6;
            }
            else if (a == 9) {
                return 10;
            }
            else if (a == 3) {
                return 4;
            }
            else if (a == 13) {
                return 0;
            }
            return 2;
        }
        attachDataLegend(a) {
            this.q2(a);
        }
        q2(a) {
            this.b3 = a;
            if (this.dataChart != null) {
                if (!this.eu) {
                    this.eu = true;
                    this.dataChart.attachDataLegend(a);
                }
            }
        }
        detachDataLegend(a) {
            this.q6(a);
        }
        q6(a) {
            if (this.dataChart != null) {
                this.eu = false;
                this.dataChart.detachDataLegend(a);
            }
            this.b3 = null;
        }
        getDataLegendSeries() {
            return this.p();
        }
        p() {
            if (this.dataChart != null) {
                return this.dataChart.getDataLegendSeries();
            }
            else {
                return new Array(0);
            }
        }
        r1(source_, a, b, c) {
            if (this.ba != null) {
                this.ba.notifySetItem(source_, a, b, c);
            }
            if (this.dataChart != null) {
                this.dataChart.mi(source_, a, b, c);
            }
        }
        rx(source_) {
            if (this.ba != null) {
                this.ba.notifyClearItems(source_);
            }
            if (this.dataChart != null) {
                this.dataChart.l9(source_);
            }
        }
        ry(source_, a, b) {
            if (this.ba != null) {
                this.ba.notifyInsertItem(source_, a, b);
            }
            if (this.dataChart != null) {
                this.dataChart.mg(source_, a, b);
            }
        }
        rz(source_, a, b) {
            if (this.ba != null) {
                this.ba.notifyRemoveItem(source_, a, b);
            }
            if (this.dataChart != null) {
                this.dataChart.mh(source_, a, b);
            }
        }
        get bs() {
            return this.br;
        }
        set bs(a) {
            let b = this.br;
            this.br = a;
            if (b != this.br) {
                this.sc("DataToolTipGroupingMode", enumGetBox(DataToolTipLayerGroupingMode_$type, b), enumGetBox(DataToolTipLayerGroupingMode_$type, this.br));
            }
        }
        get hy() {
            return this.go;
        }
        set hy(a) {
            let b = this.go;
            this.go = a;
            if (b != this.go) {
                this.sc("DataToolTipPositionOffsetX", b, this.go);
            }
        }
        get hz() {
            return this.gp;
        }
        set hz(a) {
            let b = this.gp;
            this.gp = a;
            if (b != this.gp) {
                this.sc("DataToolTipPositionOffsetY", b, this.gp);
            }
        }
        get hw() {
            return this.gm;
        }
        set hw(a) {
            let b = this.gm;
            this.gm = a;
            if (b != this.gm) {
                this.sc("DataToolTipDefaultPositionOffsetX", b, this.gm);
            }
        }
        get hx() {
            return this.gn;
        }
        set hx(a) {
            let b = this.gn;
            this.gn = a;
            if (b != this.gn) {
                this.sc("DataToolTipDefaultPositionOffsetY", b, this.gn);
            }
        }
        get bo() {
            return this.bn;
        }
        set bo(a) {
            let b = this.bn;
            this.bn = a;
            if (b != this.bn) {
                this.sc("DataToolTipGroupedPositionModeX", enumGetBox(DataTooltipGroupedPositionX_$type, b), enumGetBox(DataTooltipGroupedPositionX_$type, this.bn));
            }
        }
        get bq() {
            return this.bp;
        }
        set bq(a) {
            let b = this.bp;
            this.bp = a;
            if (b != this.bp) {
                this.sc("DataToolTipGroupedPositionModeY", enumGetBox(DataTooltipGroupedPositionY_$type, b), enumGetBox(DataTooltipGroupedPositionY_$type, this.bp));
            }
        }
        get fe() {
            return this.en;
        }
        set fe(a) {
            let b = this.en;
            this.en = a;
            if (b != this.en) {
                this.sc(DomainChart.mj, b, this.en);
            }
        }
        get m() {
            return this.e;
        }
        set m(a) {
            let b = this.e;
            this.e = a;
            if (b != this.e) {
                this.sc("DataToolTipIncludedSeries", b, this.e);
            }
        }
        get j() {
            return this.b;
        }
        set j(a) {
            let b = this.b;
            this.b = a;
            if (b != this.b) {
                this.sc("DataToolTipExcludedSeries", b, this.b);
            }
        }
        get l() {
            return this.d;
        }
        set l(a) {
            let b = this.d;
            this.d = a;
            if (b != this.d) {
                this.sc("DataToolTipIncludedColumns", b, this.d);
            }
        }
        get i() {
            return this.a;
        }
        set i(a) {
            let b = this.a;
            this.a = a;
            if (b != this.a) {
                this.sc("DataToolTipExcludedColumns", b, this.a);
            }
        }
        get a8() {
            return this.a7;
        }
        set a8(a) {
            let b = this.a7;
            this.a7 = a;
            if (b != this.a7) {
                this.sc("DataToolTipValueFormatAbbreviation", enumGetBox(DataAbbreviationMode_$type, b), enumGetBox(DataAbbreviationMode_$type, this.a7));
            }
        }
        get ji() {
            return this.i9;
        }
        set ji(a) {
            let b = this.i9;
            this.i9 = a;
            if (b != this.i9) {
                this.sc("DataToolTipValueFormatMaxFractions", b, this.i9);
            }
        }
        get jj() {
            return this.ja;
        }
        set jj(a) {
            let b = this.ja;
            this.ja = a;
            if (b != this.ja) {
                this.sc("DataToolTipValueFormatMinFractions", b, this.ja);
            }
        }
        get bm() {
            return this.bl;
        }
        set bm(a) {
            let b = this.bl;
            this.bl = a;
            if (b != this.bl) {
                this.sc("DataToolTipValueFormatMode", enumGetBox(DataLegendValueMode_$type, b), enumGetBox(DataLegendValueMode_$type, this.bl));
            }
        }
        get nb() {
            return this.kf;
        }
        set nb(a) {
            let b = this.kf;
            this.kf = a;
            if (b != this.kf) {
                this.sc("DataToolTipValueFormatCulture", b, this.kf);
            }
        }
        get ff() {
            return this.eo;
        }
        set ff(a) {
            let b = this.eo;
            this.eo = a;
            if (b != this.eo) {
                this.sc("DataToolTipValueFormatUseGrouping", b, this.eo);
            }
        }
        get nh() {
            return this.kg;
        }
        set nh(a) {
            let b = this.kg;
            this.kg = a;
            if (b != this.kg) {
                this.sc("DataToolTipValueFormatString", b, this.kg);
            }
        }
        get n() {
            return this.f;
        }
        set n(a) {
            let b = this.f;
            this.f = a;
            if (b != this.f) {
                this.sc("DataToolTipValueFormatSpecifiers", b, this.f);
            }
        }
        get u9() {
            return this.uz;
        }
        set u9(a) {
            let b = this.uz;
            this.uz = a;
            if (b != this.uz) {
                this.sc("DataToolTipValueRowMargin", b, this.uz);
            }
        }
        get fg() {
            return this.ep;
        }
        set fg(a) {
            let b = this.ep;
            this.ep = a;
            if (b != this.ep) {
                this.sc("DataToolTipValueRowVisible", b, this.ep);
            }
        }
        get nq() {
            return this.kh;
        }
        set nq(a) {
            let b = this.kh;
            this.kh = a;
            if (b != this.kh) {
                this.sc("DataToolTipValueTextWhenMissingData", b, this.kh);
            }
        }
        get fh() {
            return this.eq;
        }
        set fh(a) {
            let b = this.eq;
            this.eq = a;
            if (b != this.eq) {
                this.sc("DataToolTipValueTextUseSeriesColors", b, this.eq);
            }
        }
        get va() {
            return this.u0;
        }
        set va(a) {
            let b = this.u0;
            this.u0 = a;
            if (b != this.u0) {
                this.sc("DataToolTipValueTextMargin", b, this.u0);
            }
        }
        get t9() {
            return this.tm;
        }
        set t9(a) {
            let b = this.tm;
            this.tm = a;
            if (b != this.tm) {
                this.sc("DataToolTipValueTextColor", b, this.tm);
            }
        }
        get ec() {
            return this.d3;
        }
        set ec(a) {
            let b = this.d3;
            this.d3 = a;
            if (b != this.d3) {
                this.sc("DataToolTipValueTextStyle", b, this.d3);
            }
        }
        get l1() {
            return this.j9;
        }
        set l1(a) {
            let b = this.j9;
            this.j9 = a;
            if (b != this.j9) {
                this.sc("DataToolTipHeaderFormatString", b, this.j9);
            }
        }
        get k() {
            return this.c;
        }
        set k(a) {
            let b = this.c;
            this.c = a;
            if (b != this.c) {
                this.sc("DataToolTipHeaderFormatSpecifiers", b, this.c);
            }
        }
        get lx() {
            return this.j8;
        }
        set lx(a) {
            let b = this.j8;
            this.j8 = a;
            if (b != this.j8) {
                this.sc("DataToolTipHeaderFormatCulture", b, this.j8);
            }
        }
        get bc() {
            return this.bb;
        }
        set bc(a) {
            let b = this.bb;
            this.bb = a;
            if (b != this.bb) {
                this.sc("DataToolTipHeaderFormatDate", enumGetBox(DataLegendHeaderDateMode_$type, b), enumGetBox(DataLegendHeaderDateMode_$type, this.bb));
            }
        }
        get be() {
            return this.bd;
        }
        set be(a) {
            let b = this.bd;
            this.bd = a;
            if (b != this.bd) {
                this.sc("DataToolTipHeaderFormatTime", enumGetBox(DataLegendHeaderTimeMode_$type, b), enumGetBox(DataLegendHeaderTimeMode_$type, this.bd));
            }
        }
        get l6() {
            return this.ka;
        }
        set l6(a) {
            let b = this.ka;
            this.ka = a;
            if (b != this.ka) {
                this.sc("DataToolTipHeaderText", b, this.ka);
            }
        }
        get t1() {
            return this.te;
        }
        set t1(a) {
            let b = this.te;
            this.te = a;
            if (b != this.te) {
                this.sc("DataToolTipHeaderTextColor", b, this.te);
            }
        }
        get u3() {
            return this.ut;
        }
        set u3(a) {
            let b = this.ut;
            this.ut = a;
            if (b != this.ut) {
                this.sc("DataToolTipHeaderTextMargin", b, this.ut);
            }
        }
        get u2() {
            return this.us;
        }
        set u2(a) {
            let b = this.us;
            this.us = a;
            if (b != this.us) {
                this.sc("DataToolTipHeaderRowMargin", b, this.us);
            }
        }
        get fd() {
            return this.em;
        }
        set fd(a) {
            let b = this.em;
            this.em = a;
            if (b != this.em) {
                this.sc("DataToolTipHeaderRowVisible", b, this.em);
            }
        }
        get d4() {
            return this.dv;
        }
        set d4(a) {
            let b = this.dv;
            this.dv = a;
            if (b != this.dv) {
                this.sc("DataToolTipHeaderTextStyle", b, this.dv);
            }
        }
        get t4() {
            return this.th;
        }
        set t4(a) {
            let b = this.th;
            this.th = a;
            if (b != this.th) {
                this.sc("DataToolTipSummaryTitleTextColor", b, this.th);
            }
        }
        get d7() {
            return this.dy;
        }
        set d7(a) {
            let b = this.dy;
            this.dy = a;
            if (b != this.dy) {
                this.sc("DataToolTipSummaryTitleTextStyle", b, this.dy);
            }
        }
        get bi() {
            return this.bh;
        }
        set bi(a) {
            let b = this.bh;
            this.bh = a;
            if (b != this.bh) {
                this.sc("DataToolTipSummaryType", enumGetBox(DataLegendSummaryType_$type, b), enumGetBox(DataLegendSummaryType_$type, this.bh));
            }
        }
        get mp() {
            return this.kc;
        }
        set mp(a) {
            let b = this.kc;
            this.kc = a;
            if (b != this.kc) {
                this.sc("DataToolTipSummaryTitleText", b, this.kc);
            }
        }
        get u6() {
            return this.uw;
        }
        set u6(a) {
            let b = this.uw;
            this.uw = a;
            if (b != this.uw) {
                this.sc("DataToolTipSummaryTitleTextMargin", b, this.uw);
            }
        }
        get u5() {
            return this.uv;
        }
        set u5(a) {
            let b = this.uv;
            this.uv = a;
            if (b != this.uv) {
                this.sc("DataToolTipSummaryRowMargin", b, this.uv);
            }
        }
        get t6() {
            return this.tj;
        }
        set t6(a) {
            let b = this.tj;
            this.tj = a;
            if (b != this.tj) {
                this.sc("DataToolTipSummaryValueTextColor", b, this.tj);
            }
        }
        get d9() {
            return this.d0;
        }
        set d9(a) {
            let b = this.d0;
            this.d0 = a;
            if (b != this.d0) {
                this.sc("DataToolTipSummaryValueTextStyle", b, this.d0);
            }
        }
        get mk() {
            return this.kb;
        }
        set mk(a) {
            let b = this.kb;
            this.kb = a;
            if (b != this.kb) {
                this.sc("DataToolTipSummaryLabelText", b, this.kb);
            }
        }
        get t3() {
            return this.tg;
        }
        set t3(a) {
            let b = this.tg;
            this.tg = a;
            if (b != this.tg) {
                this.sc("DataToolTipSummaryLabelTextColor", b, this.tg);
            }
        }
        get d6() {
            return this.dx;
        }
        set d6(a) {
            let b = this.dx;
            this.dx = a;
            if (b != this.dx) {
                this.sc("DataToolTipSummaryLabelTextStyle", b, this.dx);
            }
        }
        get mv() {
            return this.kd;
        }
        set mv(a) {
            let b = this.kd;
            this.kd = a;
            if (b != this.kd) {
                this.sc("DataToolTipSummaryUnitsText", b, this.kd);
            }
        }
        get t5() {
            return this.ti;
        }
        set t5(a) {
            let b = this.ti;
            this.ti = a;
            if (b != this.ti) {
                this.sc("DataToolTipSummaryUnitsTextColor", b, this.ti);
            }
        }
        get d8() {
            return this.dz;
        }
        set d8(a) {
            let b = this.dz;
            this.dz = a;
            if (b != this.dz) {
                this.sc("DataToolTipSummaryUnitsTextStyle", b, this.dz);
            }
        }
        get u1() {
            return this.ur;
        }
        set u1(a) {
            let b = this.ur;
            this.ur = a;
            if (b != this.ur) {
                this.sc("DataToolTipBadgeMargin", b, this.ur);
            }
        }
        get cc() {
            return this.ca;
        }
        set cc(a) {
            let b = this.ca;
            this.ca = a;
            if (b != this.ca) {
                this.sc("DataToolTipBadgeShape", enumGetBox(LegendItemBadgeShape_$type, b), enumGetBox(LegendItemBadgeShape_$type, this.ca));
            }
        }
        get bk() {
            return this.bj;
        }
        set bk(a) {
            let b = this.bj;
            this.bj = a;
            if (b != this.bj) {
                this.sc("DataToolTipUnitsDisplayMode", enumGetBox(DataLegendUnitsMode_$type, b), enumGetBox(DataLegendUnitsMode_$type, this.bj));
            }
        }
        get m5() {
            return this.ke;
        }
        set m5(a) {
            let b = this.ke;
            this.ke = a;
            if (b != this.ke) {
                this.sc("DataToolTipUnitsText", b, this.ke);
            }
        }
        get t8() {
            return this.tl;
        }
        set t8(a) {
            let b = this.tl;
            this.tl = a;
            if (b != this.tl) {
                this.sc("DataToolTipUnitsTextColor", b, this.tl);
            }
        }
        get u8() {
            return this.uy;
        }
        set u8(a) {
            let b = this.uy;
            this.uy = a;
            if (b != this.uy) {
                this.sc("DataToolTipUnitsTextMargin", b, this.uy);
            }
        }
        get eb() {
            return this.d2;
        }
        set eb(a) {
            let b = this.d2;
            this.d2 = a;
            if (b != this.d2) {
                this.sc("DataToolTipUnitsTextStyle", b, this.d2);
            }
        }
        get u7() {
            return this.ux;
        }
        set u7(a) {
            let b = this.ux;
            this.ux = a;
            if (b != this.ux) {
                this.sc("DataToolTipTitleTextMargin", b, this.ux);
            }
        }
        get t7() {
            return this.tk;
        }
        set t7(a) {
            let b = this.tk;
            this.tk = a;
            if (b != this.tk) {
                this.sc("DataToolTipTitleTextColor", b, this.tk);
            }
        }
        get ea() {
            return this.d1;
        }
        set ea(a) {
            let b = this.d1;
            this.d1 = a;
            if (b != this.d1) {
                this.sc("DataToolTipTitleTextStyle", b, this.d1);
            }
        }
        get bg() {
            return this.bf;
        }
        set bg(a) {
            let b = this.bf;
            this.bf = a;
            if (b != this.bf) {
                this.sc("DataToolTipLabelDisplayMode", enumGetBox(DataLegendLabelMode_$type, b), enumGetBox(DataLegendLabelMode_$type, this.bf));
            }
        }
        get t2() {
            return this.tf;
        }
        set t2(a) {
            let b = this.tf;
            this.tf = a;
            if (b != this.tf) {
                this.sc("DataToolTipLabelTextColor", b, this.tf);
            }
        }
        get u4() {
            return this.uu;
        }
        set u4(a) {
            let b = this.uu;
            this.uu = a;
            if (b != this.uu) {
                this.sc("DataToolTipLabelTextMargin", b, this.uu);
            }
        }
        get d5() {
            return this.dw;
        }
        set d5(a) {
            let b = this.dw;
            this.dw = a;
            if (b != this.dw) {
                this.sc("DataToolTipLabelTextStyle", b, this.dw);
            }
        }
        sl(a) {
            if (this.seriesAdded != null && a != null) {
                this.seriesAdded(this, new ChartSeriesEventArgs(a));
            }
        }
        sr(a) {
            if (this.seriesRemoved != null && a != null) {
                this.seriesRemoved(this, new ChartSeriesEventArgs(a));
            }
        }
        sn(a, b) {
            if (this.seriesPointerEnter != null && a != null) {
                this.seriesPointerEnter(this, new DomainChartSeriesPointerEventArgs(a, b));
            }
        }
        so(a, b) {
            if (this.seriesPointerLeave != null && a != null) {
                this.seriesPointerLeave(this, new DomainChartSeriesPointerEventArgs(a, b));
            }
        }
        sp(a, b) {
            if (this.seriesPointerMove != null && a != null) {
                this.seriesPointerMove(this, new DomainChartSeriesPointerEventArgs(a, b));
            }
        }
        sm(a, b) {
            if (this.seriesPointerDown != null && a != null) {
                this.seriesPointerDown(this, new DomainChartSeriesPointerEventArgs(a, b));
            }
        }
        sq(a, b) {
            if (this.seriesPointerUp != null && a != null) {
                this.seriesPointerUp(this, new DomainChartSeriesPointerEventArgs(a, b));
            }
        }
        sj(a, b) {
            if (this.calloutStyleUpdating != null && a != null) {
                this.calloutStyleUpdating(this, b);
            }
        }
        si(a, b) {
            if (this.calloutRenderStyleUpdating != null && a != null) {
                this.calloutRenderStyleUpdating(this, b);
            }
        }
        sg(a) {
            if (!this.gb.contains(a)) {
                this.gb.add(a);
                let b = false;
                let c = this.sh(() => {
                    a();
                    this.gb.remove(a);
                }, b);
                b = c.p1;
            }
        }
        sh(a, b) {
            if (!b) {
                b = true;
                if (this.jt != null) {
                    this.jt.setTimeout(a, 0);
                }
                else {
                    this.ee = a;
                }
            }
            return {
                p1: b
            };
        }
        get_fw() {
            return true;
        }
        get fw() {
            return this.get_fw();
        }
        rl(a) {
            let b = a == this.dataChart;
            if (b) {
                a.ag = this.y;
            }
            if (b) {
                a.aj = this.z;
            }
            a.ah = this.ac;
            a.ai = this.ad;
            a.be = this.cf;
            a.legend = this.legend;
            a.uk = this.fr;
            a.un = this.fx && this.fw;
            a.e9 = this.f1;
            if (b) {
                a.b5 = this.db;
                a.b6 = this.dc;
                a.go = this.h0;
                a.so = this.ug;
                a.gp = this.h1;
                a.gq = this.h2;
                a.gr = this.h3;
                a.sp = this.uh;
                a.b2 = this.c8;
                a.d5 = this.fk;
                a.gs = this.h4;
                a.gt = this.h5;
                a.gu = this.h6;
                a.hb = this.iu;
                a.sx = this.ul;
                a.hh = this.i0;
                a.hc = this.iv;
                a.hd = this.iw;
                a.sy = this.um;
                a.b7 = this.de;
                a.fj = this.f4;
                a.he = this.ix;
                a.hf = this.iy;
                a.hg = this.iz;
                a.hk = this.i1;
                a.hl = this.i2;
                a.hn = this.i4;
                a.hm = this.i3;
            }
            a.t5 = this.c4;
            a.t8 = this.c6;
            if (this.b3 != null) {
                if (!this.eu) {
                    this.q2(this.b3);
                }
            }
            if (a.bv != this.c0 || a.bt != this.cy) {
                a.cu.b4(this.jt);
                a.bv = this.c0;
                a.bt = this.cy;
            }
            a.ty = this.a4;
            if (!isNaN_(this.ic)) {
                a.vp = this.ic;
            }
            if (!isNaN_(this.ie)) {
                a.vr = this.ie;
            }
            if (!isNaN_(this.id)) {
                a.vq = this.id;
            }
            if (!isNaN_(this.ib)) {
                a.vo = this.ib;
            }
            a.bb = this.b7;
            a.seriesMouseEnter = delegateCombine(a.seriesMouseEnter, (c, d) => this.sn(d.series, d.item));
            a.seriesMouseLeave = delegateCombine(a.seriesMouseLeave, (c, d) => this.so(d.series, d.item));
            a.seriesMouseMove = delegateCombine(a.seriesMouseMove, (c, d) => this.sp(d.series, d.item));
            a.seriesMouseLeftButtonDown = delegateCombine(a.seriesMouseLeftButtonDown, (c, d) => this.sm(d.series, d.item));
            a.seriesMouseLeftButtonUp = delegateCombine(a.seriesMouseLeftButtonUp, (c, d) => this.sq(d.series, d.item));
            if (b) {
                a.tp = this.uq;
            }
            if (b) {
                a.kg = this.pv;
            }
            a.r8 = this.s7;
            if (b) {
                a.j4 = this.pk;
            }
            a.r7 = this.s6;
            a.ug = this.e6;
            a.ds = this.e7;
            a.gw = this.h7;
            a.g1 = this.ih;
            a.ha = this.is;
            a.gj = this.hu;
            if (a == this.dataChart) {
                this.sw();
            }
            this.rm(a);
            this.rn(a);
        }
        af(a) {
            if (!this.f0 || !this.f3 || this.itemsSource != null) {
                return a;
            }
            if (a == null) {
                return null;
            }
            let b = new BrushCollection();
            for (let c of fromEnum(a)) {
                let d = BrushUtil.r(c);
                b.add(d);
            }
            return b;
        }
        ui(a) {
            if (!this.f0 || !this.f3 || this.itemsSource != null) {
                return a;
            }
            let b = BrushUtil.g(0, 0, 0, 0);
            return b;
        }
        rr(a, b) {
            if (!isNaN_(this.ig)) {
                a.gv = this.ig;
            }
            if (!isNaN_(this.im) && !a.dc) {
                a.g1 = this.im;
            }
            if (!isNaN_(this.hv) && a.dc) {
                a.g1 = this.hv;
            }
            a.legendItemBadgeShape = this.cd;
            a.a9 = this.b9;
            a.bt = this.c2;
            a.m3(this.cj);
            a.m2(this.ch);
            a.gs = this.h8;
            if (!isNaN_(this.h9)) {
                a.m4(this.h9);
            }
            if (!isNaN_(this.ht)) {
                a.f1 = this.ht;
            }
            if (a.isUsableInLegend) {
                a.legend = this.legend;
            }
            if (typeCast(IUnknownValuePlotting_$type, a) !== null) {
                a.unknownValuePlotting = this.dk;
            }
            a.dx = this.ft;
            if (this.fl(a) && this.b0.isCalloutLayer(a)) {
                this.b0.setCalloutStyleUpdatingEventEnabled(a, this.e9);
                this.b0.addCalloutStyleUpdating(a, runOn(this, this.sj));
            }
            if (this.fl(a) && this.b0.isCalloutLayer(a)) {
                this.b0.setCalloutRenderStyleUpdatingEventEnabled(a, this.e9);
                this.b0.addCalloutRenderStyleUpdating(a, runOn(this, this.si));
            }
            if (this.fl(a) && this.b0.isDataToolTipLayer(a)) {
                this.qz(a);
            }
            this.q0(a, b);
            this.q1(a, b);
            this.rs(a, b);
            this.rt(a, b);
            this.sl(a);
        }
        get e8() {
            return this.eh;
        }
        q8() {
            if (!this.eh) {
                this.eh = true;
                this.ri();
            }
        }
        ri() {
            this.rj();
            this.rk();
        }
        ss(a) {
            if (this.eu && this.b3 != null) {
                this.eu = false;
                if (a != null) {
                    a.detachDataLegend(this.b3);
                }
            }
            this.sz(a);
        }
        q1(a, b) {
            let c = typeCast(ITrendLineSeries_$type, a);
            if (c != null) {
                c.trendLineBrush = ArrayExtension.getModulus$1(Brush.$, this.ag, b);
                c.trendLineThickness = this.it;
                c.trendLineType = this.di;
                c.trendLinePeriod = this.jn;
            }
        }
        cm(a, b) {
            if (this.co == null || this.co.count == 0) {
                if (this.gc == null || this.gc.count == 0) {
                    return a != null && a.d3 ? 1 : 2;
                }
                let c = ArrayExtension.getModulus$1(MarkerType_$type, this.gc, b);
                if (c == 0 && a != null) {
                    c = a.d3 ? 1 : 2;
                    if (c == 2) {
                        c = this.cl(a);
                    }
                }
                if (c == 2) {
                    if (a == null) {
                        return 1;
                    }
                    else {
                        return 2;
                    }
                }
                return c;
            }
            return ArrayExtension.getModulus$1(MarkerType_$type, this.co, b);
        }
        cl(a) {
            return 2;
        }
        jy() {
            if (this.itemsSource != null) {
                return this.itemsSource;
            }
            if (this.f0) {
                return this.gi();
            }
            return null;
        }
        get_fq() {
            if (this.itemsSource == null && this.f0) {
                if (this.gi() != null) {
                    return true;
                }
            }
            return false;
        }
        get fq() {
            return this.get_fq();
        }
        gi() {
            return null;
        }
        sc(a, b, c) {
            if (this.propertyChanged != null) {
                this.propertyChanged(this, new PropertyChangedEventArgs(a));
            }
            switch (a) {
                case "DataChart":
                    this.ss(typeCast(XamDataChart.$, b));
                    this.rl(typeCast(XamDataChart.$, c));
                    break;
                case "ShouldUseSkeletonStyleForMockData":
                case "ShouldDisplayMockData":
                    this.actualItemsSource = this.jy();
                    this.y = this.af(this.aa);
                    this.z = this.af(this.ae);
                    break;
                case "ActualBrushes":
                    this.r6((d) => d.ag = this.y, true, a);
                    break;
                case "ActualOutlines":
                    this.r6((d) => d.aj = this.z, true, a);
                    break;
                case "ActualItemsSource":
                    if (this.f0) {
                        this.y = this.af(this.aa);
                        this.z = this.af(this.ae);
                    }
                    this.ba.itemsSource = this.actualItemsSource;
                    break;
                case "ItemsSource":
                    this.actualItemsSource = this.jy();
                    break;
                case "IncludedProperties":
                    this.ba.includedProperties = this.q;
                    break;
                case "ExcludedProperties":
                    this.ba.excludedProperties = this.o;
                    break;
                case "Brushes":
                    this.y = this.af(this.aa);
                    break;
                case "Outlines":
                    this.z = this.af(this.ae);
                    break;
                case "Legend":
                    this.r4((d) => d.legend = this.legend);
                    break;
                case "HighlightingMode":
                    this.r4((d) => d.bv = this.c0);
                    break;
                case "HighlightingBehavior":
                    this.r4((d) => d.bt = this.cy);
                    break;
                case "LegendHighlightingMode":
                    this.r4((d) => d.bb = this.b7);
                    break;
                case "LegendItemBadgeShape":
                    this.r4((d) => d.bd = this.cd);
                    break;
                case "LegendItemBadgeMode":
                    this.r4((d) => d.bc = this.b9);
                    break;
                case "PlotAreaMarginLeft":
                    this.r4((d) => d.vp = this.ic);
                    break;
                case "PlotAreaMarginTop":
                    this.r4((d) => d.vr = this.ie);
                    break;
                case "PlotAreaMarginRight":
                    this.r4((d) => d.vq = this.id);
                    break;
                case "PlotAreaMarginBottom":
                    this.r4((d) => d.vo = this.ib);
                    break;
                case "ComputedPlotAreaMarginMode":
                    this.r4((d) => d.ty = this.a4);
                    break;
                case "IsHorizontalZoomEnabled":
                    this.r4((d) => d.uk = this.fr);
                    break;
                case "IsVerticalZoomEnabled":
                    this.r4((d) => d.un = this.fx);
                    break;
                case "WindowRect":
                    this.r5((d) => d.tp = this.uq, false);
                    break;
                case "WindowRectMinWidth":
                    this.r5((d) => d.hl = this.i2, false);
                    break;
                case "WindowRectMinHeight":
                    this.r5((d) => d.hk = this.i1, false);
                    break;
                case "WindowSizeMinWidth":
                    this.r5((d) => d.hn = this.i4, false);
                    break;
                case "WindowSizeMinHeight":
                    this.r5((d) => d.hm = this.i3, false);
                    break;
                case "Title":
                    this.r5((d) => d.kg = this.pv, false);
                    break;
                case "Subtitle":
                    this.r5((d) => d.j4 = this.pk, false);
                    break;
                case "TitleAlignment":
                    this.r4((d) => d.r8 = this.s7);
                    break;
                case "SubtitleAlignment":
                    this.r4((d) => d.r7 = this.s6);
                    break;
                case "UnknownValuePlotting":
                    this.rb((d) => {
                        if (typeCast(IUnknownValuePlotting_$type, d) !== null) {
                            d.unknownValuePlotting = this.dk;
                        }
                    });
                    break;
                case "IsSeriesHighlightingEnabled":
                    this.rb((d) => d.dx = this.ft);
                    break;
                case "Thickness":
                    if (!isNaN_(this.im)) {
                        this.rf((d) => d.g1 = this.im);
                    }
                    break;
                case "OutlineMode":
                    this.rf((d) => d.bt = this.c2);
                    break;
                case "MarkerFillOpacity":
                    this.rf((d) => d.gs = this.h8);
                    break;
                case "MarkerThickness":
                    if (!isNaN_(this.h9)) {
                        this.rf((d) => d.m4(this.h9));
                    }
                    break;
                case "MarkerOutlineMode":
                    this.rf((d) => d.m3(this.cj));
                    break;
                case "MarkerFillMode":
                    this.rf((d) => d.m2(this.ch));
                    break;
                case "Resolution":
                    if (!isNaN_(this.ig)) {
                        this.rb((d) => d.gv = this.ig);
                    }
                    break;
                case "AreaFillOpacity":
                    if (!isNaN_(this.ht)) {
                        this.rb((d) => d.f1 = this.ht);
                    }
                    break;
                case "ShouldAvoidAxisAnnotationCollisions":
                    this.ra((d) => {
                        d.ct = this.fz;
                        d.cv = this.fz;
                    }, true);
                    break;
                case "ShouldPanOnMaximumZoom":
                    this.r4((d) => d.e9 = this.f1);
                    break;
                case "TrendLineBrushes":
                case "TrendLineType":
                case "TrendLineThickness":
                case "TrendLinePeriod":
                    this.rc((d, e) => this.q1(d, e));
                    break;
                case "MarkerMaxCount":
                    this.rg((d, e) => this.q0(d, e));
                    break;
                case "MarkerTypes":
                    let d = typeCast(ObservableCollection$1.$.specialize(MarkerType_$type), b);
                    if (d != null) {
                        d.collectionChanged = delegateRemove(d.collectionChanged, runOn(this, this.sa));
                    }
                    let e = typeCast(ObservableCollection$1.$.specialize(MarkerType_$type), c);
                    if (e != null) {
                        e.collectionChanged = delegateCombine(e.collectionChanged, runOn(this, this.sa));
                    }
                    this.rg((f, g) => this.q0(f, g));
                    break;
                case "MarkerAutomaticBehavior":
                    this.r4((f) => f.be = this.cf);
                    this.rg((f, g) => this.q0(f, g));
                    break;
                case "MarkerBrushes":
                    this.r4((f) => f.ah = this.ac);
                    break;
                case "MarkerOutlines":
                    this.r4((f) => f.ai = this.ad);
                    break;
                case "AlignsGridLinesToPixels":
                    this.r4((f) => f.ug = this.e6);
                    break;
                case "AnimateSeriesWhenAxisRangeChanges":
                    this.r4((f) => f.ds = this.e7);
                    break;
                case "ToolTipType":
                case "CrosshairsDisplayMode":
                case "CrosshairsSnapToData":
                case "CrosshairsLineVerticalStroke":
                case "CrosshairsLineHorizontalStroke":
                case "CrosshairsLineThickness":
                case "CrosshairsAnnotationXAxisBackground":
                case "CrosshairsAnnotationXAxisTextColor":
                case "CrosshairsAnnotationYAxisBackground":
                case "CrosshairsAnnotationYAxisTextColor":
                case "CrosshairsAnnotationEnabled":
                case "CrosshairsAnnotationXAxisPrecision":
                case "CrosshairsAnnotationYAxisPrecision":
                case "FinalValueAnnotationsVisible":
                case "FinalValueAnnotationsPrecision":
                case "FinalValueAnnotationsTextColor":
                case "FinalValueAnnotationsBackground":
                case "CalloutsVisible":
                case "CalloutsAllowedPositions":
                case "CalloutsItemsSource":
                case "CalloutsXMemberPath":
                case "CalloutsYMemberPath":
                case "CalloutsLabelMemberPath":
                case "CalloutsContentMemberPath":
                    let f = this.sh(runOn(this, this.q7), this.fs);
                    this.fs = f.p1;
                    break;
                case "DataToolTipGroupingMode":
                case "DataToolTipPositionOffsetX":
                case "DataToolTipPositionOffsetY":
                case "DataToolTipDefaultPositionOffsetX":
                case "DataToolTipDefaultPositionOffsetY":
                case "DataToolTipGroupedPositionModeX":
                case "DataToolTipGroupedPositionModeY":
                case DomainChart.mj:
                case "DataToolTipIncludedSeries":
                case "DataToolTipExcludedSeries":
                case "DataToolTipIncludedColumns":
                case "DataToolTipExcludedColumns":
                case "DataToolTipValueFormatAbbreviation":
                case "DataToolTipValueFormatMaxFractions":
                case "DataToolTipValueFormatMinFractions":
                case "DataToolTipValueFormatMode":
                case "DataToolTipValueFormatCulture":
                case "DataToolTipValueFormatUseGrouping":
                case "DataToolTipValueFormatString":
                case "DataToolTipValueFormatSpecifiers":
                case "DataToolTipValueRowMargin":
                case "DataToolTipValueRowVisible":
                case "DataToolTipValueTextWhenMissingData":
                case "DataToolTipValueTextUseSeriesColors":
                case "DataToolTipValueTextMargin":
                case "DataToolTipValueTextColor":
                case "DataToolTipValueTextStyle":
                case "DataToolTipHeaderFormatString":
                case "DataToolTipHeaderFormatSpecifiers":
                case "DataToolTipHeaderFormatCulture":
                case "DataToolTipHeaderFormatDate":
                case "DataToolTipHeaderFormatTime":
                case "DataToolTipHeaderText":
                case "DataToolTipHeaderTextColor":
                case "DataToolTipHeaderTextMargin":
                case "DataToolTipHeaderRowMargin":
                case "DataToolTipHeaderRowVisible":
                case "DataToolTipHeaderTextStyle":
                case "DataToolTipSummaryTitleTextColor":
                case "DataToolTipSummaryTitleTextStyle":
                case "DataToolTipSummaryType":
                case "DataToolTipSummaryTitleText":
                case "DataToolTipSummaryTitleTextMargin":
                case "DataToolTipSummaryRowMargin":
                case "DataToolTipSummaryValueTextColor":
                case "DataToolTipSummaryValueTextStyle":
                case "DataToolTipSummaryLabelText":
                case "DataToolTipSummaryLabelTextColor":
                case "DataToolTipSummaryLabelTextStyle":
                case "DataToolTipSummaryUnitsText":
                case "DataToolTipSummaryUnitsTextColor":
                case "DataToolTipSummaryUnitsTextStyle":
                case "DataToolTipBadgeMargin":
                case "DataToolTipBadgeShape":
                case "DataToolTipUnitsDisplayMode":
                case "DataToolTipUnitsText":
                case "DataToolTipUnitsTextColor":
                case "DataToolTipUnitsTextMargin":
                case "DataToolTipUnitsTextStyle":
                case "DataToolTipTitleTextMargin":
                case "DataToolTipTitleTextColor":
                case "DataToolTipTitleTextStyle":
                case "DataToolTipLabelDisplayMode":
                case "DataToolTipLabelTextColor":
                case "DataToolTipLabelTextMargin":
                case "DataToolTipLabelTextStyle":
                    this.r8(a, b, c);
                    break;
                case "HorizontalViewScrollbarMode":
                    this.r4((g) => g.b5 = this.db);
                    break;
                case "VerticalViewScrollbarMode":
                    this.r4((g) => g.b6 = this.dc);
                    break;
                case "HorizontalViewScrollbarCornerRadius":
                    this.r4((g) => g.go = this.h0);
                    break;
                case "HorizontalViewScrollbarFill":
                    this.r4((g) => g.so = this.ug);
                    break;
                case "HorizontalViewScrollbarMaxOpacity":
                    this.r4((g) => g.gr = this.h3);
                    break;
                case "HorizontalViewScrollbarOutline":
                    this.r4((g) => g.sp = this.uh);
                    break;
                case "HorizontalViewScrollbarStrokeThickness":
                    this.r4((g) => g.gs = this.h4);
                    break;
                case "VerticalViewScrollbarCornerRadius":
                    this.r4((g) => g.hb = this.iu);
                    break;
                case "VerticalViewScrollbarFill":
                    this.r4((g) => g.sx = this.ul);
                    break;
                case "VerticalViewScrollbarMaxOpacity":
                    this.r4((g) => g.hd = this.iw);
                    break;
                case "VerticalViewScrollbarOutline":
                    this.r4((g) => g.sy = this.um);
                    break;
                case "VerticalViewScrollbarStrokeThickness":
                    this.r4((g) => g.he = this.ix);
                    break;
                case "HorizontalViewScrollbarHeight":
                    this.r4((g) => g.gp = this.h1);
                    break;
                case "HorizontalViewScrollbarInset":
                    this.r4((g) => g.gq = this.h2);
                    break;
                case DomainChart.oc:
                    this.r4((g) => g.d5 = this.fk);
                    break;
                case "HorizontalViewScrollbarTrackStartInset":
                    this.r4((g) => g.gu = this.h6);
                    break;
                case "HorizontalViewScrollbarTrackEndInset":
                    this.r4((g) => g.gt = this.h5);
                    break;
                case "VerticalViewScrollbarPosition":
                    this.r4((g) => g.b7 = this.de);
                    break;
                case "HorizontalViewScrollbarPosition":
                    this.r4((g) => g.b2 = this.c8);
                    break;
            }
            this.sd(a, b, c);
            this.se(a, b, c);
        }
        r8(a, b, c) {
            if (!this.f5.containsKey(a)) {
                this.f6.add(a);
                this.f5.addItem(a, c);
            }
            else {
                this.f5.item(a, c);
            }
            this.rd((d) => {
                if (this.b0.isDataToolTipLayer(d)) {
                    this.qz(d);
                }
            });
        }
        qz(a) {
            for (let b = 0; b < this.f6.count; b++) {
                let c = this.f6._inner[b];
                let d = this.f5.item(c);
                if (stringStartsWith(c, "DataToolTip")) {
                    c = c.substr(("DataToolTip").length);
                }
                this.b0.updateDataToolTipPropertyByName(a, c, d);
            }
        }
        sa(a, b) {
            this.rg((c, d) => this.q0(c, d));
        }
        get ay() {
            return this.as;
        }
        get au() {
            return this.ao;
        }
        get at() {
            return this.an;
        }
        get dq() {
            return this.dn;
        }
        get a2() {
            return this.a0;
        }
        get ax() {
            return this.ar;
        }
        get aw() {
            return this.aq;
        }
        get av() {
            return this.ap;
        }
        get a1() {
            return this.az;
        }
        get oj() {
            return this.kk;
        }
        set oj(a) {
            let b = this.kk;
            this.kk = a;
            if (b != a) {
                this.ar.n();
                this.dr.b(this.kk, (c, d, e) => this.ar.j.add(new ChartSortDescription(2, c, d)));
            }
        }
        get n0() {
            return this.ki;
        }
        set n0(a) {
            let b = this.ki;
            this.ki = a;
            if (b != a) {
                this.aq.n();
                this.dr.b(this.ki, (c, d, e) => this.aq.j.add(new ChartSortDescription(2, c, d)));
            }
        }
        get oh() {
            return this.kj;
        }
        set oh(a) {
            let b = this.kj;
            this.kj = a;
            if (b != a) {
                this.ap.n();
                this.dr.b(this.kj, (c, d, e) => this.ap.j.add(new ChartSortDescription(2, c, d)));
            }
        }
        get ol() {
            return this.kl;
        }
        set ol(a) {
            let b = this.kl;
            this.kl = a;
            if (b != a) {
                this.az.n();
                this.ds.a(this.kl, (c, d, e) => this.az.j.add(((() => {
                    let $ret = new ChartSummaryDescription(2, c, d);
                    $ret.n = e;
                    return $ret;
                })())));
            }
        }
        get itemsSource() {
            return this.jp;
        }
        set itemsSource(a) {
            let b = this.itemsSource;
            if (b != a) {
                this.jp = a;
                this.sc("ItemsSource", b, this.itemsSource);
            }
        }
        get actualItemsSource() {
            return this.jo;
        }
        set actualItemsSource(a) {
            let b = this.actualItemsSource;
            if (b != a) {
                this.jo = a;
                this.sc("ActualItemsSource", b, this.actualItemsSource);
            }
        }
        get q() {
            return this.h;
        }
        set q(a) {
            let b = this.h;
            this.h = a;
            if (b != this.h) {
                this.sc("IncludedProperties", b, this.h);
            }
        }
        get o() {
            return this.g;
        }
        set o(a) {
            let b = this.g;
            this.g = a;
            if (b != this.g) {
                this.sc("ExcludedProperties", b, this.g);
            }
        }
        get aa() {
            return this.t;
        }
        set aa(a) {
            let b = this.aa;
            if (a != b) {
                this.t = a;
                this.sc("Brushes", b, this.aa);
            }
        }
        get ae() {
            return this.w;
        }
        set ae(a) {
            let b = this.ae;
            if (a != b) {
                this.w = a;
                this.sc("Outlines", b, this.ae);
            }
        }
        get y() {
            return this.r;
        }
        set y(a) {
            let b = this.r;
            if (a != b) {
                this.r = a;
                this.sc("ActualBrushes", b, this.y);
            }
        }
        get z() {
            return this.s;
        }
        set z(a) {
            let b = this.s;
            if (a != b) {
                this.s = a;
                this.sc("ActualOutlines", b, this.z);
            }
        }
        get legend() {
            return this.b1;
        }
        set legend(a) {
            let b = this.legend;
            if (a != b) {
                this.b1 = a;
                this.sc("Legend", b, this.legend);
            }
        }
        get_fr() {
            return this.et;
        }
        set_fr(a) {
            let b = this.fr;
            if (a != b) {
                this.et = a;
                this.sc("IsHorizontalZoomEnabled", b, this.fr);
            }
        }
        get fr() {
            return this.get_fr();
        }
        set fr(a) {
            this.set_fr(a);
        }
        get_fx() {
            return this.ew;
        }
        set_fx(a) {
            let b = this.fx;
            if (a != b) {
                this.ew = a;
                this.sc("IsVerticalZoomEnabled", b, this.fx);
            }
        }
        get fx() {
            return this.get_fx();
        }
        set fx(a) {
            this.set_fx(a);
        }
        get_f0() {
            return this.ey;
        }
        set_f0(a) {
            let b = this.f0;
            if (a != b) {
                this.ey = a;
                this.sc("ShouldDisplayMockData", b, this.f0);
            }
        }
        get f0() {
            return this.get_f0();
        }
        set f0(a) {
            this.set_f0(a);
        }
        get_f3() {
            return this.e4;
        }
        set_f3(a) {
            let b = this.f3;
            if (a != b) {
                this.e4 = a;
                this.sc("ShouldUseSkeletonStyleForMockData", b, this.f3);
            }
        }
        get f3() {
            return this.get_f3();
        }
        set f3(a) {
            this.set_f3(a);
        }
        get_ft() {
            return this.ev;
        }
        set_ft(a) {
            let b = this.ft;
            if (a != b) {
                this.ev = a;
                this.sc("IsSeriesHighlightingEnabled", b, this.ft);
            }
        }
        get ft() {
            return this.get_ft();
        }
        set ft(a) {
            this.set_ft(a);
        }
        get uq() {
            return this.uo;
        }
        set uq(a) {
            let b = this.uq;
            if (Rect.l_op_Inequality(a, b)) {
                this.uo = a;
                this.sc("WindowRect", b, this.uq);
            }
        }
        get up() {
            return this.dataChart != null ? this.dataChart.to : Rect.empty;
        }
        get pv() {
            return this.ko;
        }
        set pv(a) {
            let b = this.pv;
            if (a != b) {
                this.ko = a;
                this.sc("Title", b, this.pv);
            }
        }
        get pk() {
            return this.km;
        }
        set pk(a) {
            let b = this.pk;
            if (a != b) {
                this.km = a;
                this.sc("Subtitle", b, this.pk);
            }
        }
        get s7() {
            return this.s5;
        }
        set s7(a) {
            let b = this.s7;
            if (b != a) {
                this.s5 = a;
                this.sc("TitleAlignment", enumGetBox(HorizontalAlignment_$type, b), enumGetBox(HorizontalAlignment_$type, this.s7));
            }
        }
        get s6() {
            return this.s4;
        }
        set s6(a) {
            let b = this.s6;
            if (b != a) {
                this.s4 = a;
                this.sc("SubtitleAlignment", enumGetBox(HorizontalAlignment_$type, b), enumGetBox(HorizontalAlignment_$type, this.s6));
            }
        }
        get dk() {
            return this.dj;
        }
        set dk(a) {
            let b = this.dk;
            if (a != b) {
                this.dj = a;
                this.sc("UnknownValuePlotting", enumGetBox(UnknownValuePlotting_$type, b), enumGetBox(UnknownValuePlotting_$type, this.dk));
            }
        }
        get_ig() {
            return this.g5;
        }
        set_ig(a) {
            let b = this.ig;
            if (a != b) {
                this.g5 = a;
                this.sc("Resolution", b, this.ig);
            }
        }
        get ig() {
            return this.get_ig();
        }
        set ig(a) {
            this.set_ig(a);
        }
        get im() {
            return this.hb;
        }
        set im(a) {
            let b = this.im;
            if (a != b) {
                this.hb = a;
                this.sc("Thickness", b, this.im);
            }
        }
        get c2() {
            return this.c1;
        }
        set c2(a) {
            let b = this.c2;
            if (a != b) {
                this.c1 = a;
                this.sc("OutlineMode", enumGetBox(SeriesOutlineMode_$type, b), enumGetBox(SeriesOutlineMode_$type, this.c2));
            }
        }
        get cj() {
            return this.ci;
        }
        set cj(a) {
            let b = this.cj;
            if (a != b) {
                this.ci = a;
                this.sc("MarkerOutlineMode", enumGetBox(MarkerOutlineMode_$type, b), enumGetBox(MarkerOutlineMode_$type, this.cj));
            }
        }
        get ch() {
            return this.cg;
        }
        set ch(a) {
            let b = this.ch;
            if (a != b) {
                this.cg = a;
                this.sc("MarkerFillMode", enumGetBox(MarkerFillMode_$type, b), enumGetBox(MarkerFillMode_$type, this.ch));
            }
        }
        get h8() {
            return this.gy;
        }
        set h8(a) {
            let b = this.h8;
            if (a != b) {
                this.gy = a;
                this.sc("MarkerFillOpacity", b, this.h8);
            }
        }
        get h9() {
            return this.gz;
        }
        set h9(a) {
            let b = this.h9;
            if (a != b) {
                this.gz = a;
                this.sc("MarkerThickness", b, this.h9);
            }
        }
        get co() {
            return this.cn;
        }
        set co(a) {
            let b = this.co;
            if (a != b) {
                this.cn = a;
                this.sc("MarkerTypes", b, this.co);
            }
        }
        get cf() {
            return this.ce;
        }
        set cf(a) {
            let b = this.cf;
            if (a != b) {
                this.ce = a;
                this.sc("MarkerAutomaticBehavior", enumGetBox(MarkerAutomaticBehavior_$type, b), enumGetBox(MarkerAutomaticBehavior_$type, this.cf));
            }
        }
        get ac() {
            return this.u;
        }
        set ac(a) {
            let b = this.ac;
            if (a != b) {
                this.u = a;
                this.sc("MarkerBrushes", b, this.ac);
            }
        }
        get ad() {
            return this.v;
        }
        set ad(a) {
            let b = this.ad;
            if (!Base.equalsStatic(a, b)) {
                this.v = a;
                this.sc("MarkerOutlines", b, this.ad);
            }
        }
        get jl() {
            return this.jc;
        }
        set jl(a) {
            let b = this.jc;
            if (a != b) {
                this.jc = a;
                this.sc("MarkerMaxCount", b, this.jc);
            }
        }
        get ht() {
            return this.gj;
        }
        set ht(a) {
            let b = this.ht;
            if (a != b) {
                this.gj = a;
                this.sc("AreaFillOpacity", b, this.ht);
            }
        }
        get e7() {
            return this.eg;
        }
        set e7(a) {
            let b = this.eg;
            if (a != b) {
                this.eg = a;
                this.sc("AnimateSeriesWhenAxisRangeChanges", b, this.eg);
            }
        }
        get ag() {
            return this.x;
        }
        set ag(a) {
            let b = this.ag;
            if (a != b) {
                this.x = a;
                this.sc("TrendLineBrushes", b, this.ag);
            }
        }
        get di() {
            return this.dh;
        }
        set di(a) {
            let b = this.di;
            if (a != b) {
                this.dh = a;
                this.sc("TrendLineType", enumGetBox(TrendLineType_$type, b), enumGetBox(TrendLineType_$type, this.di));
            }
        }
        get it() {
            return this.hh;
        }
        set it(a) {
            let b = this.it;
            if (a != b) {
                this.hh = a;
                this.sc("TrendLineThickness", b, this.it);
            }
        }
        get e6() {
            return this.ef;
        }
        set e6(a) {
            let b = this.e6;
            if (a != b) {
                this.ef = a;
                this.sc("AlignsGridLinesToPixels", b, this.e6);
            }
        }
        get ic() {
            return this.g2;
        }
        set ic(a) {
            let b = this.g2;
            this.g2 = a;
            if (a != b) {
                this.sc("PlotAreaMarginLeft", b, a);
            }
        }
        get ie() {
            return this.g4;
        }
        set ie(a) {
            let b = this.g4;
            this.g4 = a;
            if (a != b) {
                this.sc("PlotAreaMarginTop", b, a);
            }
        }
        get id() {
            return this.g3;
        }
        set id(a) {
            let b = this.g3;
            this.g3 = a;
            if (a != b) {
                this.sc("PlotAreaMarginRight", b, a);
            }
        }
        get ib() {
            return this.g1;
        }
        set ib(a) {
            let b = this.g1;
            this.g1 = a;
            if (a != b) {
                this.sc("PlotAreaMarginBottom", b, a);
            }
        }
        sb() {
        }
        get a4() {
            return this.a3;
        }
        set a4(a) {
            let b = this.a4;
            if (a != b) {
                this.a3 = a;
                this.sc("ComputedPlotAreaMarginMode", enumGetBox(ComputedPlotAreaMarginMode_$type, b), enumGetBox(ComputedPlotAreaMarginMode_$type, this.a4));
            }
        }
        get c4() {
            return this.c3;
        }
        set c4(a) {
            let b = this.c4;
            if (a != b) {
                this.c3 = a;
                this.sc("SeriesPlotAreaMarginHorizontalMode", enumGetBox(SeriesPlotAreaMarginHorizontalMode_$type, b), enumGetBox(SeriesPlotAreaMarginHorizontalMode_$type, this.c4));
            }
        }
        get c6() {
            return this.c5;
        }
        set c6(a) {
            let b = this.c6;
            if (a != b) {
                this.c5 = a;
                this.sc("SeriesPlotAreaMarginVerticalMode", enumGetBox(SeriesPlotAreaMarginVerticalMode_$type, b), enumGetBox(SeriesPlotAreaMarginVerticalMode_$type, this.c6));
            }
        }
        get c0() {
            return this.cz;
        }
        set c0(a) {
            let b = this.c0;
            if (a != b) {
                this.cz = a;
                this.sc("HighlightingMode", enumGetBox(SeriesHighlightingMode_$type, b), enumGetBox(SeriesHighlightingMode_$type, this.c0));
            }
        }
        get cy() {
            return this.cx;
        }
        set cy(a) {
            let b = this.cy;
            if (a != b) {
                this.cx = a;
                this.sc("HighlightingBehavior", enumGetBox(SeriesHighlightingBehavior_$type, b), enumGetBox(SeriesHighlightingBehavior_$type, this.cy));
            }
        }
        get b7() {
            return this.b6;
        }
        set b7(a) {
            let b = this.b7;
            if (a != b) {
                this.b6 = a;
                this.sc("LegendHighlightingMode", enumGetBox(LegendHighlightingMode_$type, b), enumGetBox(LegendHighlightingMode_$type, this.b7));
            }
        }
        get cd() {
            return this.cb;
        }
        set cd(a) {
            let b = this.cd;
            if (a != b) {
                this.cb = a;
                this.sc("LegendItemBadgeShape", enumGetBox(LegendItemBadgeShape_$type, b), enumGetBox(LegendItemBadgeShape_$type, this.cd));
            }
        }
        get b9() {
            return this.b8;
        }
        set b9(a) {
            let b = this.b9;
            if (a != b) {
                this.b8 = a;
                this.sc("LegendItemBadgeMode", enumGetBox(LegendItemBadgeMode_$type, b), enumGetBox(LegendItemBadgeMode_$type, this.b9));
            }
        }
        get jn() {
            return this.jf;
        }
        set jn(a) {
            let b = this.jn;
            if (a != b) {
                this.jf = a;
                this.sc("TrendLinePeriod", b, this.jn);
            }
        }
        get dg() {
            return this.df;
        }
        set dg(a) {
            let b = this.dg;
            if (a != b) {
                this.df = a;
                this.sc("ToolTipType", enumGetBox(ToolTipType_$type, b), enumGetBox(ToolTipType_$type, this.dg));
            }
        }
        get a6() {
            return this.a5;
        }
        set a6(a) {
            let b = this.a6;
            if (a != b) {
                this.a5 = a;
                this.sc("CrosshairsDisplayMode", enumGetBox(CrosshairsDisplayMode_$type, b), enumGetBox(CrosshairsDisplayMode_$type, this.a6));
            }
        }
        get fc() {
            return this.el;
        }
        set fc(a) {
            let b = this.fc;
            if (a != b) {
                this.el = a;
                this.sc("CrosshairsSnapToData", b, this.fc);
            }
        }
        get t0() {
            return this.td;
        }
        set t0(a) {
            let b = this.t0;
            if (a != b) {
                this.td = a;
                this.sc("CrosshairsLineVerticalStroke", b, this.t0);
            }
        }
        get hv() {
            return this.gl;
        }
        set hv(a) {
            let b = this.hv;
            if (a != b) {
                this.gl = a;
                this.sc("CrosshairsLineThickness", b, this.hv);
            }
        }
        get tz() {
            return this.tc;
        }
        set tz(a) {
            let b = this.tz;
            if (a != b) {
                this.tc = a;
                this.sc("CrosshairsLineHorizontalStroke", b, this.tz);
            }
        }
        get fb() {
            return this.ek;
        }
        set fb(a) {
            let b = this.fb;
            if (a != b) {
                this.ek = a;
                this.sc("CrosshairsAnnotationEnabled", b, this.fb);
            }
        }
        get tv() {
            return this.s8;
        }
        set tv(a) {
            let b = this.tv;
            if (a != b) {
                this.s8 = a;
                this.sc("CrosshairsAnnotationXAxisBackground", b, this.tv);
            }
        }
        get tx() {
            return this.ta;
        }
        set tx(a) {
            let b = this.tx;
            if (a != b) {
                this.ta = a;
                this.sc("CrosshairsAnnotationYAxisBackground", b, this.tx);
            }
        }
        get tw() {
            return this.s9;
        }
        set tw(a) {
            let b = this.tw;
            if (a != b) {
                this.s9 = a;
                this.sc("CrosshairsAnnotationXAxisTextColor", b, this.tw);
            }
        }
        get ty() {
            return this.tb;
        }
        set ty(a) {
            let b = this.ty;
            if (a != b) {
                this.tb = a;
                this.sc("CrosshairsAnnotationYAxisTextColor", b, this.ty);
            }
        }
        get jg() {
            return this.i7;
        }
        set jg(a) {
            let b = this.jg;
            if (a != b) {
                this.i7 = a;
                this.sc("CrosshairsAnnotationXAxisPrecision", b, this.jg);
            }
        }
        get jh() {
            return this.i8;
        }
        set jh(a) {
            let b = this.jh;
            if (a != b) {
                this.i8 = a;
                this.sc("CrosshairsAnnotationYAxisPrecision", b, this.jh);
            }
        }
        get fz() {
            return this.ex;
        }
        set fz(a) {
            let b = this.fz;
            if (a != b) {
                this.ex = a;
                this.sc("ShouldAvoidAxisAnnotationCollisions", b, this.fz);
            }
        }
        get f1() {
            return this.ez;
        }
        set f1(a) {
            let b = this.ez;
            if (a != b) {
                this.ez = a;
                this.sc("ShouldPanOnMaximumZoom", b, this.ez);
            }
        }
        get fi() {
            return this.er;
        }
        set fi(a) {
            let b = this.fi;
            if (a != b) {
                this.er = a;
                this.sc("FinalValueAnnotationsVisible", b, this.fi);
            }
        }
        get ua() {
            return this.tn;
        }
        set ua(a) {
            let b = this.ua;
            if (a != b) {
                this.tn = a;
                this.sc("FinalValueAnnotationsBackground", b, this.ua);
            }
        }
        get ub() {
            return this.to;
        }
        set ub(a) {
            let b = this.ub;
            if (a != b) {
                this.to = a;
                this.sc("FinalValueAnnotationsTextColor", b, this.ub);
            }
        }
        get jk() {
            return this.jb;
        }
        set jk(a) {
            let b = this.jk;
            if (a != b) {
                this.jb = a;
                this.sc("FinalValueAnnotationsPrecision", b, this.jk);
            }
        }
        get fa() {
            return this.ej;
        }
        set fa(a) {
            let b = this.fa;
            if (a != b) {
                this.ej = a;
                this.sc("CalloutsVisible", b, this.fa);
            }
        }
        get e9() {
            return this.ei;
        }
        set e9(a) {
            let b = this.e9;
            if (a != b) {
                this.ei = a;
                this.sc("CalloutStyleUpdatingEventEnabled", b, this.e9);
            }
        }
        get calloutsItemsSource() {
            return this.gg;
        }
        set calloutsItemsSource(a) {
            let b = this.calloutsItemsSource;
            if (a != b) {
                this.gg = a;
                this.sc("CalloutsItemsSource", b, this.calloutsItemsSource);
            }
        }
        get ai() {
            return this.ah;
        }
        set ai(a) {
            let b = this.ai;
            if (a != b) {
                this.ah = a;
                this.sc("CalloutsAllowedPositions", b, this.ai);
            }
        }
        get k6() {
            return this.j6;
        }
        set k6(a) {
            let b = this.k6;
            if (a != b) {
                this.j6 = a;
                this.sc("CalloutsXMemberPath", b, this.k6);
            }
        }
        get k8() {
            return this.j7;
        }
        set k8(a) {
            let b = this.k8;
            if (a != b) {
                this.j7 = a;
                this.sc("CalloutsYMemberPath", b, this.k8);
            }
        }
        get k2() {
            return this.j5;
        }
        set k2(a) {
            let b = this.k2;
            if (a != b) {
                this.j5 = a;
                this.sc("CalloutsLabelMemberPath", b, this.k2);
            }
        }
        get kz() {
            return this.j4;
        }
        set kz(a) {
            let b = this.kz;
            if (a != b) {
                this.j4 = a;
                this.sc("CalloutsContentMemberPath", b, this.kz);
            }
        }
        get db() {
            return this.c9;
        }
        set db(a) {
            let b = this.c9;
            this.c9 = a;
            if (b != this.c9) {
                this.sc("HorizontalViewScrollbarMode", enumGetBox(SeriesViewerScrollbarMode_$type, b), enumGetBox(SeriesViewerScrollbarMode_$type, this.c9));
            }
        }
        get dc() {
            return this.da;
        }
        set dc(a) {
            let b = this.da;
            this.da = a;
            if (b != this.da) {
                this.sc("VerticalViewScrollbarMode", enumGetBox(SeriesViewerScrollbarMode_$type, b), enumGetBox(SeriesViewerScrollbarMode_$type, this.da));
            }
        }
        get c8() {
            return this.c7;
        }
        set c8(a) {
            let b = this.c7;
            this.c7 = a;
            if (b != this.c7) {
                this.sc("HorizontalViewScrollbarPosition", enumGetBox(SeriesViewerHorizontalScrollbarPosition_$type, b), enumGetBox(SeriesViewerHorizontalScrollbarPosition_$type, this.c7));
            }
        }
        get de() {
            return this.dd;
        }
        set de(a) {
            let b = this.dd;
            this.dd = a;
            if (b != this.dd) {
                this.sc("VerticalViewScrollbarPosition", enumGetBox(SeriesViewerVerticalScrollbarPosition_$type, b), enumGetBox(SeriesViewerVerticalScrollbarPosition_$type, this.dd));
            }
        }
        get ug() {
            return this.tp;
        }
        set ug(a) {
            let b = this.tp;
            this.tp = a;
            if (b != this.tp) {
                this.sc("HorizontalViewScrollbarFill", b, this.tp);
            }
        }
        get uh() {
            return this.tq;
        }
        set uh(a) {
            let b = this.tq;
            this.tq = a;
            if (b != this.tq) {
                this.sc("HorizontalViewScrollbarOutline", b, this.tq);
            }
        }
        get h4() {
            return this.gu;
        }
        set h4(a) {
            let b = this.gu;
            this.gu = a;
            if (b != this.gu) {
                this.sc("HorizontalViewScrollbarStrokeThickness", b, this.gu);
            }
        }
        get h3() {
            return this.gt;
        }
        set h3(a) {
            let b = this.gt;
            this.gt = a;
            if (b != this.gt) {
                this.sc("HorizontalViewScrollbarMaxOpacity", b, this.gt);
            }
        }
        get h0() {
            return this.gq;
        }
        set h0(a) {
            let b = this.gq;
            this.gq = a;
            if (b != this.gq) {
                this.sc("HorizontalViewScrollbarCornerRadius", b, this.gq);
            }
        }
        get h1() {
            return this.gr;
        }
        set h1(a) {
            let b = this.gr;
            this.gr = a;
            if (b != this.gr) {
                this.sc("HorizontalViewScrollbarHeight", b, this.gr);
            }
        }
        get h2() {
            return this.gs;
        }
        set h2(a) {
            let b = this.gs;
            this.gs = a;
            if (b != this.gs) {
                this.sc("HorizontalViewScrollbarInset", b, this.gs);
            }
        }
        get h6() {
            return this.gw;
        }
        set h6(a) {
            let b = this.gw;
            this.gw = a;
            if (b != this.gw) {
                this.sc("HorizontalViewScrollbarTrackStartInset", b, this.gw);
            }
        }
        get h5() {
            return this.gv;
        }
        set h5(a) {
            let b = this.gv;
            this.gv = a;
            if (b != this.gv) {
                this.sc("HorizontalViewScrollbarTrackEndInset", b, this.gv);
            }
        }
        get fk() {
            return this.es;
        }
        set fk(a) {
            let b = this.es;
            this.es = a;
            if (b != this.es) {
                this.sc(DomainChart.oc, b, this.es);
            }
        }
        get ul() {
            return this.tt;
        }
        set ul(a) {
            let b = this.tt;
            this.tt = a;
            if (b != this.tt) {
                this.sc("VerticalViewScrollbarFill", b, this.tt);
            }
        }
        get um() {
            return this.tu;
        }
        set um(a) {
            let b = this.tu;
            this.tu = a;
            if (b != this.tu) {
                this.sc("VerticalViewScrollbarOutline", b, this.tu);
            }
        }
        get ix() {
            return this.hl;
        }
        set ix(a) {
            let b = this.hl;
            this.hl = a;
            if (b != this.hl) {
                this.sc("VerticalViewScrollbarStrokeThickness", b, this.hl);
            }
        }
        get iw() {
            return this.hk;
        }
        set iw(a) {
            let b = this.hk;
            this.hk = a;
            if (b != this.hk) {
                this.sc("VerticalViewScrollbarMaxOpacity", b, this.hk);
            }
        }
        get iu() {
            return this.hi;
        }
        set iu(a) {
            let b = this.hi;
            this.hi = a;
            if (b != this.hi) {
                this.sc("VerticalViewScrollbarCornerRadius", b, this.hi);
            }
        }
        get i0() {
            return this.ho;
        }
        set i0(a) {
            let b = this.ho;
            this.ho = a;
            if (b != this.ho) {
                this.sc("VerticalViewScrollbarWidth", b, this.ho);
            }
        }
        get iv() {
            return this.hj;
        }
        set iv(a) {
            let b = this.hj;
            this.hj = a;
            if (b != this.hj) {
                this.sc("VerticalViewScrollbarInset", b, this.hj);
            }
        }
        get iz() {
            return this.hn;
        }
        set iz(a) {
            let b = this.hn;
            this.hn = a;
            if (b != this.hn) {
                this.sc("VerticalViewScrollbarTrackStartInset", b, this.hn);
            }
        }
        get iy() {
            return this.hm;
        }
        set iy(a) {
            let b = this.hm;
            this.hm = a;
            if (b != this.hm) {
                this.sc("VerticalViewScrollbarTrackEndInset", b, this.hm);
            }
        }
        get f4() {
            return this.e5;
        }
        set f4(a) {
            let b = this.e5;
            this.e5 = a;
            if (b != this.e5) {
                this.sc(DomainChart.qo, b, this.e5);
            }
        }
        get i2() {
            return this.hq;
        }
        set i2(a) {
            let b = this.hq;
            this.hq = a;
            if (b != this.hq) {
                this.sc("WindowRectMinWidth", b, this.hq);
            }
        }
        get i1() {
            return this.hp;
        }
        set i1(a) {
            let b = this.hp;
            this.hp = a;
            if (b != this.hp) {
                this.sc("WindowRectMinHeight", b, this.hp);
            }
        }
        get i4() {
            return this.hs;
        }
        set i4(a) {
            let b = this.hs;
            this.hs = a;
            if (b != this.hs) {
                this.sc("WindowSizeMinWidth", b, this.hs);
            }
        }
        get i3() {
            return this.hr;
        }
        set i3(a) {
            let b = this.hr;
            this.hr = a;
            if (b != this.hr) {
                this.sc("WindowSizeMinHeight", b, this.hr);
            }
        }
    }
    DomainChart.$t = /*@__PURE__*/ markType(DomainChart, 'DomainChart', Base.$, [INotifyPropertyChanged_$type, IDataLegendTarget_$type]);
    DomainChart.j3 = /*@__PURE__*/ new Random(0);
    DomainChart.j2 = /*@__PURE__*/ new Random(0);
    DomainChart.mj = "DataToolTipShouldUpdateWhenSeriesDataChanges";
    DomainChart.oc = "HorizontalViewScrollbarShouldAddAutoTrackInsets";
    DomainChart.qo = "VerticalViewScrollbarShouldAddAutoTrackInsets";
    return DomainChart;
})();
