// igniteui-react-charts/esm2015/lib/FinancialLegend_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { LegendBase } from "./LegendBase";
import { LegendBaseView } from "./LegendBaseView";
import { fromEn, delegateCombine, markType } from "igniteui-react-core";
import { FinancialLegendViewManager } from "./FinancialLegendViewManager";
/**
 * @hidden
 */
export let FinancialLegend = /*@__PURE__*/ (() => {
    class FinancialLegend extends LegendBase {
        constructor() {
            super();
            this.ab = FinancialLegend.$;
            let a = this.at;
            a.collectionChanged = delegateCombine(a.collectionChanged, (a, b) => {
                if (b.oldItems != null) {
                    for (let c of fromEn(b.oldItems)) {
                        this.bb.an(c);
                    }
                }
                if (b.newItems != null) {
                    for (let d of fromEn(b.newItems)) {
                        this.bb.x(d);
                    }
                }
            });
        }
        aj() {
            return new FinancialLegendView(this);
        }
        get bb() {
            return this.ak;
        }
        addChildInOrder(a, b) {
            this.at.add(a);
        }
        get_isFinancial() {
            return true;
        }
    }
    FinancialLegend.$t = /*@__PURE__*/ markType(FinancialLegend, 'FinancialLegend', LegendBase.$);
    return FinancialLegend;
})();
/**
 * @hidden
 */
export let FinancialLegendView = /*@__PURE__*/ (() => {
    class FinancialLegendView extends LegendBaseView {
        constructor(a) {
            super(a);
            this.c = new FinancialLegendViewManager(this);
        }
    }
    FinancialLegendView.$t = /*@__PURE__*/ markType(FinancialLegendView, 'FinancialLegendView', LegendBaseView.$);
    return FinancialLegendView;
})();
