// igniteui-react-charts/esm2015/lib/igr-treemap-node-styling-event-args.js
import { IgrTreemapNodeStyle } from "./igr-treemap-node-style";
import { TreemapNodeStylingEventArgs as TreemapNodeStylingEventArgs_internal } from "./TreemapNodeStylingEventArgs";
import { ensureBool } from "igniteui-react-core";
export class IgrTreemapNodeStylingEventArgs {
    constructor() {
        this._implementation = this.createImplementation();
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    }
    createImplementation() {
        return new TreemapNodeStylingEventArgs_internal();
    }
    /**
     * @hidden
     */
    get i() {
        return this._implementation;
    }
    onImplementationCreated() {
    }
    _provideImplementation(i) {
        this._implementation = i;
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    }
    get value() {
        return this.i.value;
    }
    set value(v) {
        this.i.value = +v;
    }
    get sum() {
        return this.i.sum;
    }
    set sum(v) {
        this.i.sum = +v;
    }
    get customValue() {
        return this.i.customValue;
    }
    set customValue(v) {
        this.i.customValue = v;
    }
    get item() {
        return this.i.item;
    }
    set item(v) {
        this.i.item = v;
    }
    get parentItem() {
        return this.i.parentItem;
    }
    set parentItem(v) {
        this.i.parentItem = v;
    }
    get label() {
        return this.i.label;
    }
    set label(v) {
        this.i.label = v;
    }
    get parentValue() {
        return this.i.parentValue;
    }
    set parentValue(v) {
        this.i.parentValue = +v;
    }
    get parentLabel() {
        return this.i.parentLabel;
    }
    set parentLabel(v) {
        this.i.parentLabel = v;
    }
    get parentSum() {
        return this.i.parentSum;
    }
    set parentSum(v) {
        this.i.parentSum = +v;
    }
    get isParent() {
        return this.i.isParent;
    }
    set isParent(v) {
        this.i.isParent = ensureBool(v);
    }
    get style() {
        if (this.i.style == null) {
            return null;
        }
        if (!this.i.style.externalObject) {
            let e = IgrTreemapNodeStyle._createFromInternal(this.i.style);
            if (e) {
                e._implementation = this.i.style;
            }
            this.i.style.externalObject = e;
        }
        return this.i.style.externalObject;
    }
    set style(v) {
        v == null ? this.i.style = null : this.i.style = v.i;
    }
}
