// igniteui-react-charts/esm2015/lib/igr-final-value-layer.js
import { IgrSeries } from "./igr-series";
import { FinalValueSelectionMode_$type } from "./FinalValueSelectionMode";
import { IgrAnnotationLayer } from "./igr-annotation-layer";
import { FinalValueLayer } from "./FinalValueLayer";
import { ensureEnum, brushToString, stringToBrush } from "igniteui-react-core";
/**
 * Represents an annotation layer that displays crosshair lines that cross through the closest value of the target series under the cursor.
*/
export class IgrFinalValueLayer extends IgrAnnotationLayer {
    createImplementation() {
        return new FinalValueLayer();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    constructor(props) {
        super(props);
    }
    /**
     * Gets whether the series is final value annotation layer.
    */
    get isAnnotationFinalValue() {
        return this.i.da;
    }
    /**
     * Gets or sets the name of the series to target this annotation to. If null, this annotation targets all series simultaneously.
    */
    get targetSeriesName() {
        return this.i.r1;
    }
    set targetSeriesName(v) {
        this.i.r1 = v;
    }
    /**
     * Gets or sets the series to target this annotation to. If null, this annotation targets all series simultaneously.
    */
    get targetSeries() {
        if (this.i.rf == null) {
            return null;
        }
        if (!this.i.rf.externalObject) {
            let e = IgrSeries._createFromInternal(this.i.rf);
            if (e) {
                e._implementation = this.i.rf;
            }
            this.i.rf.externalObject = e;
        }
        return this.i.rf.externalObject;
    }
    set targetSeries(v) {
        if (v != null && this._stylingContainer && v._styling)
            v._styling(this._stylingContainer, this, this);
        v == null ? this.i.rf = null : this.i.rf = v.i;
    }
    /**
     * Gets or sets how to select the final value to annotate.
    */
    get finalValueSelectionMode() {
        return this.i.re;
    }
    set finalValueSelectionMode(v) {
        this.i.re = ensureEnum(FinalValueSelectionMode_$type, v);
    }
    /**
     * Gets or sets the color to use for the axis annotation text. Leave unset for an automatic value.
    */
    get axisAnnotationTextColor() {
        return brushToString(this.i.sl);
    }
    set axisAnnotationTextColor(v) {
        this.i.sl = stringToBrush(v);
    }
    /**
     * Gets or sets the color to use for the axis annotation backing. Leave unset for an automatic value.
    */
    get axisAnnotationBackground() {
        return brushToString(this.i.sj);
    }
    set axisAnnotationBackground(v) {
        this.i.sj = stringToBrush(v);
    }
    /**
     * Gets or sets the corner radius to use for the axis annotation backing. Leave unset for an automatic value.
    */
    get axisAnnotationBackgroundCornerRadius() {
        return this.i.rj;
    }
    set axisAnnotationBackgroundCornerRadius(v) {
        this.i.rj = +v;
    }
    /**
     * Gets or sets the precision to use displaying values for interpolated crosshair positions.
    */
    get axisAnnotationInterpolatedValuePrecision() {
        return this.i.rp;
    }
    set axisAnnotationInterpolatedValuePrecision(v) {
        this.i.rp = +v;
    }
    /**
     * Gets or sets the color to use for the axis annotation outline. Leave unset for an automatic value.
    */
    get axisAnnotationOutline() {
        return brushToString(this.i.sk);
    }
    set axisAnnotationOutline(v) {
        this.i.sk = stringToBrush(v);
    }
    /**
     * Gets or sets the left padding to use withing the axis annotation callout. Leaving this NaN will use an automatic value related to the axis label margins.
    */
    get axisAnnotationPaddingLeft() {
        return this.i.rl;
    }
    set axisAnnotationPaddingLeft(v) {
        this.i.rl = +v;
    }
    /**
     * Gets or sets the padding to use withing the axis annotation callout. Leaving this NaN will use an automatic value related to the axis label margins.
    */
    get axisAnnotationPaddingTop() {
        return this.i.rn;
    }
    set axisAnnotationPaddingTop(v) {
        this.i.rn = +v;
    }
    /**
     * Gets or sets the padding to use withing the axis annotation callout. Leaving this NaN will use an automatic value related to the axis label margins.
    */
    get axisAnnotationPaddingRight() {
        return this.i.rm;
    }
    set axisAnnotationPaddingRight(v) {
        this.i.rm = +v;
    }
    /**
     * Gets or sets the padding to use withing the axis annotation callout. Leaving this NaN will use an automatic value related to the axis label margins.
    */
    get axisAnnotationPaddingBottom() {
        return this.i.rk;
    }
    set axisAnnotationPaddingBottom(v) {
        this.i.rk = +v;
    }
    /**
     * Gets or sets the stroke thickness for the axis annotation backing. Leave unset for an automatic value.
    */
    get axisAnnotationStrokeThickness() {
        return this.i.ro;
    }
    set axisAnnotationStrokeThickness(v) {
        this.i.ro = +v;
    }
    findByName(name) {
        var baseResult = super.findByName(name);
        if (baseResult) {
            return baseResult;
        }
        if (this.targetSeries && this.targetSeries.name && this.targetSeries.name == name) {
            return this.targetSeries;
        }
        return null;
    }
    _styling(container, component, parent) {
        super._styling(container, component, parent);
        this._inStyling = true;
        if (this.targetSeries && this.targetSeries._styling) {
            this.targetSeries._styling(container, component, this);
        }
        this._inStyling = false;
    }
}
