// igniteui-react-charts/esm2015/lib/igr-data-legend.js
import * as React from 'react';
import { XamDataLegend } from "./XamDataLegend";
import { arrayFindByName, ensureEnum, brushToString, stringToBrush, ensureBool, initializePropertiesFromCss, NamePatcher, isValidProp, toSpinal, getModifiedProps } from "igniteui-react-core";
import { FontInfo } from "igniteui-react-core";
import { ReactRenderer } from "igniteui-react-core";
import { Thickness } from "igniteui-react-core";
import { delegateCombine, delegateRemove } from "igniteui-react-core";
import { DataAbbreviationMode_$type } from "igniteui-react-core";
import { DataLegendHeaderTimeMode_$type } from "igniteui-react-core";
import { DataLegendHeaderDateMode_$type } from "igniteui-react-core";
import { DataLegendUnitsMode_$type } from "igniteui-react-core";
import { DataLegendLabelMode_$type } from "igniteui-react-core";
import { DataLegendValueMode_$type } from "igniteui-react-core";
import { DataLegendSummaryType_$type } from "igniteui-react-core";
import { IgrDataLegendSummaryEventArgs } from './igr-data-legend-summary-event-args';
import { IgrDataLegendStylingRowEventArgs } from './igr-data-legend-styling-row-event-args';
import { IgrDataLegendStylingColumnEventArgs } from './igr-data-legend-styling-column-event-args';
import { LegendItemBadgeShape_$type } from "igniteui-react-core";
const DataLegendStylingDefaults = {};
/**
 * Represents data legend that displays information about each series including values and summaries.
 * THIS IS A COMMUNITY TECHNOLOGY PREVIEW, not for use in production.
*/
export class IgrDataLegend extends React.Component {
    constructor(props) {
        super(props);
        this._wrapper = null;
        this._implementation = null;
        this.__p = null;
        this._hasUserValues = new Set();
        this._stylingContainer = null;
        this._stylingParent = null;
        this._inStyling = false;
        this._calculateColumnSummary = null;
        this._calculateColumnSummary_wrapped = null;
        this._styleHeaderRow = null;
        this._styleHeaderRow_wrapped = null;
        this._styleSeriesRow = null;
        this._styleSeriesRow_wrapped = null;
        this._styleSummaryRow = null;
        this._styleSummaryRow_wrapped = null;
        this._styleSeriesColumn = null;
        this._styleSeriesColumn_wrapped = null;
        this._styleSummaryColumn = null;
        this._styleSummaryColumn_wrapped = null;
        if (this._styling) {
            NamePatcher.ensureStylablePatched(Object.getPrototypeOf(this));
        }
        this._getMainRef = this._getMainRef.bind(this);
        this._implementation = this.createImplementation();
        if (document) {
            this._container = document.createElement("div");
            this._container.style.display = "block";
            this._container.style.width = "100%";
            this._container.style.height = "100%";
        }
        var root;
        root = this._container;
        if (this._container != null) {
            root = this._container;
        }
        var ren = new ReactRenderer(root, document, true, DataLegendStylingDefaults);
        this._wrapper = ren;
        var dataLegend = this.i;
        this._dataLegend = dataLegend;
        this._dataLegend.externalObject = this;
        dataLegend.provideContainer(ren);
        dataLegend.notifySizeChanged();
        if (props) {
            this.initializeProperties();
        }
        ren.addSizeWatcher(() => {
            this._dataLegend.notifySizeChanged();
        });
    }
    set height(value) {
        this._height = value;
        if (this._elRef) {
            this._elRef.style.height = value;
            this._dataLegend.notifySizeChanged();
        }
    }
    get height() {
        return this._height;
    }
    set width(value) {
        this._width = value;
        if (this._elRef) {
            this._elRef.style.height = value;
            this._dataLegend.notifySizeChanged();
        }
    }
    //private desiredSizeChanged(width: number, height: number) {
    //    //if (this.width == undefined || this.width == '') {
    //    //    this._elRef.style.width = width + "px";
    //    //}
    //    //if (this.height == undefined || this.height == '') {
    //    //    this._elRef.style.height = height + "px";
    //    //}
    //    //this._elRef.style.minWidth = width + "px";
    //    //this._elRef.style.minHeight = height + "px";
    //    this._dataLegend.notifySizeChanged();
    //}
    get width() {
        return this._width;
    }
    _getMainRef(ref) {
        this._elRef = ref;
    }
    render() {
        let div = React.createElement("div", {
            className: "ig-data-legend igr-data-legend",
            ref: this._getMainRef
        });
        //}
        return div;
    }
    shouldComponentUpdate(nextProps, nextState) {
        const mod = getModifiedProps(this.props, nextProps);
        for (const p of Object.keys(mod)) {
            if (isValidProp(this, p)) {
                this[p] = mod[p];
            }
        }
        return true;
    }
    initializeProperties() {
        for (const p of Object.keys(this.props)) {
            if (isValidProp(this, p)) {
                this[p] = this.props[p];
            }
        }
    }
    // supports angular themes or custom properties set in CSS
    updateStyle() {
        this._styling(this._elRef, this);
    }
    destroy() {
        this._dataLegend.destroy();
        this._wrapper.destroy();
    }
    componentWillUnmount() {
    }
    componentDidMount() {
        this._elRef.style.width = this._width ? this._width : "";
        this._elRef.style.height = this._height ? this._height : "";
        this._elRef.appendChild(this._container);
        this._dataLegend.notifySizeChanged();
        this.initializeContent();
    }
    initializeContent() {
        this.updateStyle();
    }
    createImplementation() {
        return new XamDataLegend();
    }
    get i() {
        return this._implementation;
    }
    static _staticStyling(container, component, parent) {
        let genericPrefix = "";
        let typeName = (component.i ? component.i : component).$type.name;
        if (typeName.indexOf("Xam") === 0) {
            typeName = typeName.substring(3);
        }
        genericPrefix = toSpinal("DataLegend");
        let additionalPrefixes = [];
        let prefix = toSpinal(typeName);
        additionalPrefixes.push(prefix + "-");
        let b = (component.i ? component.i : component).$type.baseType;
        while (b && b.name != "Object" &&
            b.name != "Base" &&
            b.name != "Control" &&
            b.Name != "DependencyObject" &&
            b.Name != "FrameworkElement") {
            typeName = b.name;
            if (typeName.indexOf("Xam") === 0) {
                typeName = typeName.substring(3);
            }
            let basePrefix = toSpinal(typeName);
            additionalPrefixes.push(basePrefix + "-");
            b = b.baseType;
        }
        if (parent) {
            let parentTypeName = parent.i.$type.name;
            if (parentTypeName.indexOf("Xam") === 0) {
                parentTypeName = parentTypeName.substring(3);
            }
            let parentPrefix = toSpinal(parentTypeName);
            additionalPrefixes.push(parentPrefix + "-" + genericPrefix + "-");
            additionalPrefixes.push(parentPrefix + "-" + prefix + "-");
        }
        initializePropertiesFromCss(container, component, genericPrefix + "-", new Set(), false, additionalPrefixes);
    }
    /**
     * Gets or sets the target for the data legend.
    */
    get target() {
        var ret = this.i.target;
        if (!ret) {
            return ret;
        }
        if (ret.externalObject != null) {
            ret = ret.externalObject;
        }
        return ret;
    }
    set target(v) {
        if (v && v._implementation) {
            v = v._implementation;
        }
        this.i.target = v;
    }
    /**
 * Gets whether or not at least one badge is visible based on BadgesShape and setting of a row's IsVisble in StyleSeriesRow event
*/
    get actualBadgesVisible() {
        return this.i.bw;
    }
    set actualBadgesVisible(v) {
        this.i.bw = ensureBool(v);
    }
    /**
     * Gets the actual pixel scaling ratio used to affect the pixel density of the control.
     * A higher scaling ratio will produce crisper visuals at the expense of memory.  Lower values will cause the control to appear blurry.
    */
    get actualPixelScalingRatio() {
        return this.i.cn;
    }
    set actualPixelScalingRatio(v) {
        this.i.cn = +v;
    }
    /**
     * Gets or sets indexes or names of series to include in displaying in the data legend, e.g. "0, 1"
    */
    get includedSeries() {
        return this.i.includedSeries;
    }
    set includedSeries(v) {
        if (v && !Array.isArray(v) && typeof (v) == "string") {
            const re = /\s*(?:,|\s|$)\s*/gm;
            v = v.split(re);
        }
        this.i.includedSeries = v;
    }
    /**
     * Gets or sets indexes or names of series to exclude from displaying in the data legend, e.g. "0, 1"
     * The ExcludedSeries property takes precedences over values of IncludedSeries property
    */
    get excludedSeries() {
        return this.i.excludedSeries;
    }
    set excludedSeries(v) {
        if (v && !Array.isArray(v) && typeof (v) == "string") {
            const re = /\s*(?:,|\s|$)\s*/gm;
            v = v.split(re);
        }
        this.i.excludedSeries = v;
    }
    /**
     * Gets or sets names of data columns or their labels to include in displaying in the data legend, e.g. "High, Low" or "H, L"
     * The ExcludedColumns property takes precedences over values of IncludedColumns property
    */
    get includedColumns() {
        return this.i.includedColumns;
    }
    set includedColumns(v) {
        if (v && !Array.isArray(v) && typeof (v) == "string") {
            const re = /\s*(?:,|\s|$)\s*/gm;
            v = v.split(re);
        }
        this.i.includedColumns = v;
    }
    /**
     * Gets or sets names of data columns or their labels to exclude from displaying in the data legend, e.g. "High, Low" or "H, L"
     * The ExcludedColumns property takes precedences over values of IncludedColumns property
    */
    get excludedColumns() {
        return this.i.excludedColumns;
    }
    set excludedColumns(v) {
        if (v && !Array.isArray(v) && typeof (v) == "string") {
            const re = /\s*(?:,|\s|$)\s*/gm;
            v = v.split(re);
        }
        this.i.excludedColumns = v;
    }
    /**
     * Gets or sets whether the data legend should update when the series data is mutated.
    */
    get shouldUpdateWhenSeriesDataChanges() {
        return this.i.b2;
    }
    set shouldUpdateWhenSeriesDataChanges(v) {
        this.i.b2 = ensureBool(v);
    }
    /**
     * Gets or sets mode for abbreviating large numbers displayed in the legend
     * This property is ignored when setting ValueFormatSpecifiers or ValueFormatString properties
    */
    get valueFormatAbbreviation() {
        return this.i.an;
    }
    set valueFormatAbbreviation(v) {
        this.i.an = ensureEnum(DataAbbreviationMode_$type, v);
    }
    /**
     * Gets or sets maximum digits for formating numbers displayed in the legend
     * This property is ignored when setting ValueFormatSpecifiers or ValueFormatString properties
    */
    get valueFormatMaxFractions() {
        return this.i.ct;
    }
    set valueFormatMaxFractions(v) {
        this.i.ct = +v;
    }
    /**
     * Gets or sets minimum digits for formating numbers displayed in the legend
     * This property is ignored when setting ValueFormatSpecifiers or ValueFormatString properties
    */
    get valueFormatMinFractions() {
        return this.i.cu;
    }
    set valueFormatMinFractions(v) {
        this.i.cu = +v;
    }
    /**
     * Gets or sets the mode for displaying values in the data legend, e.g. Currency ($500.25), Decimal (500.25), Integer (500)
    */
    get valueFormatMode() {
        return this.i.a3;
    }
    set valueFormatMode(v) {
        this.i.a3 = ensureEnum(DataLegendValueMode_$type, v);
    }
    /**
     * Gets or sets globalization culture when displaying values as currencies, e.g. use "en-GB" to display British pound symbol when the ValueFormatMode property is set to 'Currency' mode
     * This property is ignored when setting ValueFormatSpecifiers or ValueFormatString properties
    */
    get valueFormatCulture() {
        return this.i.ex;
    }
    set valueFormatCulture(v) {
        this.i.ex = v;
    }
    /**
     * Gets or sets the currency code used for displaying currency symbol next to values, e.g. USD for $, EUR for €
     * This property is ignored when setting ValueFormatSpecifiers or ValueFormatString properties
    */
    get valueFormatCurrencyCode() {
        return this.i.ez;
    }
    set valueFormatCurrencyCode(v) {
        this.i.ez = v;
    }
    /**
     * Gets or sets whether or not use grouping separator, e.g, 15,000 for 15000
     * This property is ignored when setting ValueFormatSpecifiers or ValueFormatString properties
    */
    get valueFormatUseGrouping() {
        return this.i.b3;
    }
    set valueFormatUseGrouping(v) {
        this.i.b3 = ensureBool(v);
    }
    /**
     * Gets or sets the format string for values displayed in the data legend.
    */
    get valueFormatString() {
        return this.i.e5;
    }
    set valueFormatString(v) {
        this.i.e5 = v;
    }
    /**
     * Gets or sets the format specifiers to use with the ValueFormatString string.
    */
    get valueFormatSpecifiers() {
        return this.i.ah;
    }
    set valueFormatSpecifiers(v) {
        if (v && !Array.isArray(v) && typeof (v) == "string") {
            const re = /\s*(?:,|\s|$)\s*/gm;
            v = v.split(re);
        }
        this.i.ah = v;
    }
    get valueRowMarginBottom() {
        return this.i.ir ? this.i.ir.bottom : NaN;
    }
    set valueRowMarginBottom(v) {
        this.ensureValueRowMargin();
        this.i.ir.bottom = +v;
        this.i.ir = this.i.ir;
    }
    get valueRowMarginLeft() {
        return this.i.ir ? this.i.ir.left : NaN;
    }
    set valueRowMarginLeft(v) {
        this.ensureValueRowMargin();
        this.i.ir.left = +v;
        this.i.ir = this.i.ir;
    }
    get valueRowMarginRight() {
        return this.i.ir ? this.i.ir.right : NaN;
    }
    set valueRowMarginRight(v) {
        this.ensureValueRowMargin();
        this.i.ir.right = +v;
        this.i.ir = this.i.ir;
    }
    get valueRowMarginTop() {
        return this.i.ir ? this.i.ir.top : NaN;
    }
    set valueRowMarginTop(v) {
        this.ensureValueRowMargin();
        this.i.ir.top = +v;
        this.i.ir = this.i.ir;
    }
    ensureValueRowMargin() {
        if (this.i.ir) {
            return;
        }
        this.i.ir = new Thickness(2);
    }
    /**
     * Gets or sets whether to show series rows.
    */
    get valueRowVisible() {
        return this.i.b4;
    }
    set valueRowVisible(v) {
        this.i.b4 = ensureBool(v);
    }
    /**
     * Gets or sets text displayed when data column is missing a value, e.g. "no data"
    */
    get valueTextWhenMissingData() {
        return this.i.fe;
    }
    set valueTextWhenMissingData(v) {
        this.i.fe = v;
    }
    /**
     * Gets or sets whether to use series colors when displaying values in the legend
    */
    get valueTextUseSeriesColors() {
        return this.i.b5;
    }
    set valueTextUseSeriesColors(v) {
        this.i.b5 = ensureBool(v);
    }
    get valueTextMarginBottom() {
        return this.i.is ? this.i.is.bottom : NaN;
    }
    set valueTextMarginBottom(v) {
        this.ensureValueTextMargin();
        this.i.is.bottom = +v;
        this.i.is = this.i.is;
    }
    get valueTextMarginLeft() {
        return this.i.is ? this.i.is.left : NaN;
    }
    set valueTextMarginLeft(v) {
        this.ensureValueTextMargin();
        this.i.is.left = +v;
        this.i.is = this.i.is;
    }
    get valueTextMarginRight() {
        return this.i.is ? this.i.is.right : NaN;
    }
    set valueTextMarginRight(v) {
        this.ensureValueTextMargin();
        this.i.is.right = +v;
        this.i.is = this.i.is;
    }
    get valueTextMarginTop() {
        return this.i.is ? this.i.is.top : NaN;
    }
    set valueTextMarginTop(v) {
        this.ensureValueTextMargin();
        this.i.is.top = +v;
        this.i.is = this.i.is;
    }
    ensureValueTextMargin() {
        if (this.i.is) {
            return;
        }
        this.i.is = new Thickness(2);
    }
    /**
     * Gets or sets the units text color.
    */
    get valueTextColor() {
        return brushToString(this.i.h9);
    }
    set valueTextColor(v) {
        this.i.h9 = stringToBrush(v);
    }
    /**
     * Gets or Sets the style to use for the units text.
    */
    get valueTextStyle() {
        if (this.i.bt == null) {
            return null;
        }
        return this.i.bt.fontString;
    }
    set valueTextStyle(v) {
        let fi = new FontInfo();
        fi.fontString = v;
        this.i.bt = fi;
    }
    /**
     * Gets or sets the format string for header text displayed in the data legend.
    */
    get headerFormatString() {
        return this.i.dc;
    }
    set headerFormatString(v) {
        this.i.dc = v;
    }
    /**
     * Gets or sets the format specifiers to use with the HeaderFormatString string.
    */
    get headerFormatSpecifiers() {
        return this.i.ag;
    }
    set headerFormatSpecifiers(v) {
        if (v && !Array.isArray(v) && typeof (v) == "string") {
            const re = /\s*(?:,|\s|$)\s*/gm;
            v = v.split(re);
        }
        this.i.ag = v;
    }
    /**
     * Gets or sets globalization culture applied to header only when it displays dates that are fetched from date/time axes
     * This property is ignored when setting HeaderFormatSpecifiers or HeaderFormatString properties
    */
    get headerFormatCulture() {
        return this.i.c8;
    }
    set headerFormatCulture(v) {
        this.i.c8 = v;
    }
    /**
     * Gets or sets date format for the header only when it displays dates that are fetched from date/time axes
     * This property is ignored when setting HeaderFormatSpecifiers or HeaderFormatString properties
    */
    get headerFormatDate() {
        return this.i.ap;
    }
    set headerFormatDate(v) {
        this.i.ap = ensureEnum(DataLegendHeaderDateMode_$type, v);
    }
    /**
     * Gets or sets time format for the header only when it displays dates that are fetched from date/time axes
     * This property is ignored when setting HeaderFormatSpecifiers or HeaderFormatString properties
    */
    get headerFormatTime() {
        return this.i.aq;
    }
    set headerFormatTime(v) {
        this.i.aq = ensureEnum(DataLegendHeaderTimeMode_$type, v);
    }
    /**
     * Gets or sets the HeaderText for the data legend.
    */
    get headerText() {
        return this.i.dh;
    }
    set headerText(v) {
        this.i.dh = v;
    }
    /**
     * Gets or sets the header text color.
    */
    get headerTextColor() {
        return brushToString(this.i.h0);
    }
    set headerTextColor(v) {
        this.i.h0 = stringToBrush(v);
    }
    get headerTextMarginBottom() {
        return this.i.ij ? this.i.ij.bottom : NaN;
    }
    set headerTextMarginBottom(v) {
        this.ensureHeaderTextMargin();
        this.i.ij.bottom = +v;
        this.i.ij = this.i.ij;
    }
    get headerTextMarginLeft() {
        return this.i.ij ? this.i.ij.left : NaN;
    }
    set headerTextMarginLeft(v) {
        this.ensureHeaderTextMargin();
        this.i.ij.left = +v;
        this.i.ij = this.i.ij;
    }
    get headerTextMarginRight() {
        return this.i.ij ? this.i.ij.right : NaN;
    }
    set headerTextMarginRight(v) {
        this.ensureHeaderTextMargin();
        this.i.ij.right = +v;
        this.i.ij = this.i.ij;
    }
    get headerTextMarginTop() {
        return this.i.ij ? this.i.ij.top : NaN;
    }
    set headerTextMarginTop(v) {
        this.ensureHeaderTextMargin();
        this.i.ij.top = +v;
        this.i.ij = this.i.ij;
    }
    ensureHeaderTextMargin() {
        if (this.i.ij) {
            return;
        }
        this.i.ij = new Thickness(2);
    }
    get headerRowMarginBottom() {
        return this.i.ii ? this.i.ii.bottom : NaN;
    }
    set headerRowMarginBottom(v) {
        this.ensureHeaderRowMargin();
        this.i.ii.bottom = +v;
        this.i.ii = this.i.ii;
    }
    get headerRowMarginLeft() {
        return this.i.ii ? this.i.ii.left : NaN;
    }
    set headerRowMarginLeft(v) {
        this.ensureHeaderRowMargin();
        this.i.ii.left = +v;
        this.i.ii = this.i.ii;
    }
    get headerRowMarginRight() {
        return this.i.ii ? this.i.ii.right : NaN;
    }
    set headerRowMarginRight(v) {
        this.ensureHeaderRowMargin();
        this.i.ii.right = +v;
        this.i.ii = this.i.ii;
    }
    get headerRowMarginTop() {
        return this.i.ii ? this.i.ii.top : NaN;
    }
    set headerRowMarginTop(v) {
        this.ensureHeaderRowMargin();
        this.i.ii.top = +v;
        this.i.ii = this.i.ii;
    }
    ensureHeaderRowMargin() {
        if (this.i.ii) {
            return;
        }
        this.i.ii = new Thickness(2);
    }
    /**
     * Gets or sets whether to show Header row.
    */
    get headerRowVisible() {
        return this.i.bz;
    }
    set headerRowVisible(v) {
        this.i.bz = ensureBool(v);
    }
    /**
     * Gets or Sets the style to use for the header text.
    */
    get headerTextStyle() {
        if (this.i.bl == null) {
            return null;
        }
        return this.i.bl.fontString;
    }
    set headerTextStyle(v) {
        let fi = new FontInfo();
        fi.fontString = v;
        this.i.bl = fi;
    }
    /**
     * Gets or sets the summary text color.
    */
    get summaryTitleTextColor() {
        return brushToString(this.i.h4);
    }
    set summaryTitleTextColor(v) {
        this.i.h4 = stringToBrush(v);
    }
    /**
     * Gets or Sets the style to use for the summary text.
    */
    get summaryTitleTextStyle() {
        if (this.i.bo == null) {
            return null;
        }
        return this.i.bo.fontString;
    }
    set summaryTitleTextStyle(v) {
        let fi = new FontInfo();
        fi.fontString = v;
        this.i.bo = fi;
    }
    /**
     * Gets or sets the SummaryType for the data legend.
    */
    get summaryType() {
        return this.i.a1;
    }
    set summaryType(v) {
        this.i.a1 = ensureEnum(DataLegendSummaryType_$type, v);
    }
    /**
     * Gets or sets the SummaryTitleText for the data legend.
    */
    get summaryTitleText() {
        return this.i.d8;
    }
    set summaryTitleText(v) {
        this.i.d8 = v;
    }
    get summaryTitleTextMarginBottom() {
        return this.i.im ? this.i.im.bottom : NaN;
    }
    set summaryTitleTextMarginBottom(v) {
        this.ensureSummaryTitleTextMargin();
        this.i.im.bottom = +v;
        this.i.im = this.i.im;
    }
    get summaryTitleTextMarginLeft() {
        return this.i.im ? this.i.im.left : NaN;
    }
    set summaryTitleTextMarginLeft(v) {
        this.ensureSummaryTitleTextMargin();
        this.i.im.left = +v;
        this.i.im = this.i.im;
    }
    get summaryTitleTextMarginRight() {
        return this.i.im ? this.i.im.right : NaN;
    }
    set summaryTitleTextMarginRight(v) {
        this.ensureSummaryTitleTextMargin();
        this.i.im.right = +v;
        this.i.im = this.i.im;
    }
    get summaryTitleTextMarginTop() {
        return this.i.im ? this.i.im.top : NaN;
    }
    set summaryTitleTextMarginTop(v) {
        this.ensureSummaryTitleTextMargin();
        this.i.im.top = +v;
        this.i.im = this.i.im;
    }
    ensureSummaryTitleTextMargin() {
        if (this.i.im) {
            return;
        }
        this.i.im = new Thickness(2);
    }
    get summaryRowMarginBottom() {
        return this.i.il ? this.i.il.bottom : NaN;
    }
    set summaryRowMarginBottom(v) {
        this.ensureSummaryRowMargin();
        this.i.il.bottom = +v;
        this.i.il = this.i.il;
    }
    get summaryRowMarginLeft() {
        return this.i.il ? this.i.il.left : NaN;
    }
    set summaryRowMarginLeft(v) {
        this.ensureSummaryRowMargin();
        this.i.il.left = +v;
        this.i.il = this.i.il;
    }
    get summaryRowMarginRight() {
        return this.i.il ? this.i.il.right : NaN;
    }
    set summaryRowMarginRight(v) {
        this.ensureSummaryRowMargin();
        this.i.il.right = +v;
        this.i.il = this.i.il;
    }
    get summaryRowMarginTop() {
        return this.i.il ? this.i.il.top : NaN;
    }
    set summaryRowMarginTop(v) {
        this.ensureSummaryRowMargin();
        this.i.il.top = +v;
        this.i.il = this.i.il;
    }
    ensureSummaryRowMargin() {
        if (this.i.il) {
            return;
        }
        this.i.il = new Thickness(2);
    }
    /**
     * Gets or sets the units text color.
    */
    get summaryValueTextColor() {
        return brushToString(this.i.h6);
    }
    set summaryValueTextColor(v) {
        this.i.h6 = stringToBrush(v);
    }
    /**
     * Gets or Sets the style to use for the units text.
    */
    get summaryValueTextStyle() {
        if (this.i.bq == null) {
            return null;
        }
        return this.i.bq.fontString;
    }
    set summaryValueTextStyle(v) {
        let fi = new FontInfo();
        fi.fontString = v;
        this.i.bq = fi;
    }
    /**
     * Gets or sets the units text for the data legend.
    */
    get summaryLabelText() {
        return this.i.d3;
    }
    set summaryLabelText(v) {
        this.i.d3 = v;
    }
    /**
     * Gets or sets the units text color.
    */
    get summaryLabelTextColor() {
        return brushToString(this.i.h3);
    }
    set summaryLabelTextColor(v) {
        this.i.h3 = stringToBrush(v);
    }
    /**
     * Gets or Sets the style to use for the units text.
    */
    get summaryLabelTextStyle() {
        if (this.i.bn == null) {
            return null;
        }
        return this.i.bn.fontString;
    }
    set summaryLabelTextStyle(v) {
        let fi = new FontInfo();
        fi.fontString = v;
        this.i.bn = fi;
    }
    /**
     * Gets or sets the units text for the data legend.
    */
    get summaryUnitsText() {
        return this.i.ee;
    }
    set summaryUnitsText(v) {
        this.i.ee = v;
    }
    /**
     * Gets or sets the units text color.
    */
    get summaryUnitsTextColor() {
        return brushToString(this.i.h5);
    }
    set summaryUnitsTextColor(v) {
        this.i.h5 = stringToBrush(v);
    }
    /**
     * Gets or Sets the style to use for the units text.
    */
    get summaryUnitsTextStyle() {
        if (this.i.bp == null) {
            return null;
        }
        return this.i.bp.fontString;
    }
    set summaryUnitsTextStyle(v) {
        let fi = new FontInfo();
        fi.fontString = v;
        this.i.bp = fi;
    }
    get badgeMarginBottom() {
        return this.i.ih ? this.i.ih.bottom : NaN;
    }
    set badgeMarginBottom(v) {
        this.ensureBadgeMargin();
        this.i.ih.bottom = +v;
        this.i.ih = this.i.ih;
    }
    get badgeMarginLeft() {
        return this.i.ih ? this.i.ih.left : NaN;
    }
    set badgeMarginLeft(v) {
        this.ensureBadgeMargin();
        this.i.ih.left = +v;
        this.i.ih = this.i.ih;
    }
    get badgeMarginRight() {
        return this.i.ih ? this.i.ih.right : NaN;
    }
    set badgeMarginRight(v) {
        this.ensureBadgeMargin();
        this.i.ih.right = +v;
        this.i.ih = this.i.ih;
    }
    get badgeMarginTop() {
        return this.i.ih ? this.i.ih.top : NaN;
    }
    set badgeMarginTop(v) {
        this.ensureBadgeMargin();
        this.i.ih.top = +v;
        this.i.ih = this.i.ih;
    }
    ensureBadgeMargin() {
        if (this.i.ih) {
            return;
        }
        this.i.ih = new Thickness(2);
    }
    /**
     * Gets or sets the BadgeShape for the data legend.
    */
    get badgeShape() {
        return this.i.a8;
    }
    set badgeShape(v) {
        this.i.a8 = ensureEnum(LegendItemBadgeShape_$type, v);
    }
    /**
     * Gets or sets horizontal position (between 0.0 and 1.0) in viewport of the target component
    */
    get targetCursorPositionX() {
        return this.i.targetCursorPositionX;
    }
    set targetCursorPositionX(v) {
        this.i.targetCursorPositionX = +v;
    }
    /**
     * Gets or sets vertical position (between 0.0 and 1.0) in viewport of the target component
    */
    get targetCursorPositionY() {
        return this.i.targetCursorPositionY;
    }
    set targetCursorPositionY(v) {
        this.i.targetCursorPositionY = +v;
    }
    /**
     * Gets or sets the UnitsMode for the data legend.
    */
    get unitsDisplayMode() {
        return this.i.a2;
    }
    set unitsDisplayMode(v) {
        this.i.a2 = ensureEnum(DataLegendUnitsMode_$type, v);
    }
    /**
     * Gets or sets the units text for the data legend.
    */
    get unitsText() {
        return this.i.er;
    }
    set unitsText(v) {
        this.i.er = v;
    }
    /**
     * Gets or sets the units text color.
    */
    get unitsTextColor() {
        return brushToString(this.i.h8);
    }
    set unitsTextColor(v) {
        this.i.h8 = stringToBrush(v);
    }
    get unitsTextMarginBottom() {
        return this.i.ip ? this.i.ip.bottom : NaN;
    }
    set unitsTextMarginBottom(v) {
        this.ensureUnitsTextMargin();
        this.i.ip.bottom = +v;
        this.i.ip = this.i.ip;
    }
    get unitsTextMarginLeft() {
        return this.i.ip ? this.i.ip.left : NaN;
    }
    set unitsTextMarginLeft(v) {
        this.ensureUnitsTextMargin();
        this.i.ip.left = +v;
        this.i.ip = this.i.ip;
    }
    get unitsTextMarginRight() {
        return this.i.ip ? this.i.ip.right : NaN;
    }
    set unitsTextMarginRight(v) {
        this.ensureUnitsTextMargin();
        this.i.ip.right = +v;
        this.i.ip = this.i.ip;
    }
    get unitsTextMarginTop() {
        return this.i.ip ? this.i.ip.top : NaN;
    }
    set unitsTextMarginTop(v) {
        this.ensureUnitsTextMargin();
        this.i.ip.top = +v;
        this.i.ip = this.i.ip;
    }
    ensureUnitsTextMargin() {
        if (this.i.ip) {
            return;
        }
        this.i.ip = new Thickness(2);
    }
    /**
     * Gets or Sets the style to use for the units text.
    */
    get unitsTextStyle() {
        if (this.i.bs == null) {
            return null;
        }
        return this.i.bs.fontString;
    }
    set unitsTextStyle(v) {
        let fi = new FontInfo();
        fi.fontString = v;
        this.i.bs = fi;
    }
    get titleTextMarginBottom() {
        return this.i.io ? this.i.io.bottom : NaN;
    }
    set titleTextMarginBottom(v) {
        this.ensureTitleTextMargin();
        this.i.io.bottom = +v;
        this.i.io = this.i.io;
    }
    get titleTextMarginLeft() {
        return this.i.io ? this.i.io.left : NaN;
    }
    set titleTextMarginLeft(v) {
        this.ensureTitleTextMargin();
        this.i.io.left = +v;
        this.i.io = this.i.io;
    }
    get titleTextMarginRight() {
        return this.i.io ? this.i.io.right : NaN;
    }
    set titleTextMarginRight(v) {
        this.ensureTitleTextMargin();
        this.i.io.right = +v;
        this.i.io = this.i.io;
    }
    get titleTextMarginTop() {
        return this.i.io ? this.i.io.top : NaN;
    }
    set titleTextMarginTop(v) {
        this.ensureTitleTextMargin();
        this.i.io.top = +v;
        this.i.io = this.i.io;
    }
    ensureTitleTextMargin() {
        if (this.i.io) {
            return;
        }
        this.i.io = new Thickness(2);
    }
    /**
     * Gets or sets the display text color.
    */
    get titleTextColor() {
        return brushToString(this.i.h7);
    }
    set titleTextColor(v) {
        this.i.h7 = stringToBrush(v);
    }
    /**
     * Gets or Sets the style to use for the display text.
    */
    get titleTextStyle() {
        if (this.i.br == null) {
            return null;
        }
        return this.i.br.fontString;
    }
    set titleTextStyle(v) {
        let fi = new FontInfo();
        fi.fontString = v;
        this.i.br = fi;
    }
    /**
     * Gets or sets the mode for displaying labels before series values in the data legend, e.g. O: H: L: C: for financial series
    */
    get labelDisplayMode() {
        return this.i.ar;
    }
    set labelDisplayMode(v) {
        this.i.ar = ensureEnum(DataLegendLabelMode_$type, v);
    }
    /**
     * Gets or sets the units text color.
    */
    get labelTextColor() {
        return brushToString(this.i.h1);
    }
    set labelTextColor(v) {
        this.i.h1 = stringToBrush(v);
    }
    get labelTextMarginBottom() {
        return this.i.ik ? this.i.ik.bottom : NaN;
    }
    set labelTextMarginBottom(v) {
        this.ensureLabelTextMargin();
        this.i.ik.bottom = +v;
        this.i.ik = this.i.ik;
    }
    get labelTextMarginLeft() {
        return this.i.ik ? this.i.ik.left : NaN;
    }
    set labelTextMarginLeft(v) {
        this.ensureLabelTextMargin();
        this.i.ik.left = +v;
        this.i.ik = this.i.ik;
    }
    get labelTextMarginRight() {
        return this.i.ik ? this.i.ik.right : NaN;
    }
    set labelTextMarginRight(v) {
        this.ensureLabelTextMargin();
        this.i.ik.right = +v;
        this.i.ik = this.i.ik;
    }
    get labelTextMarginTop() {
        return this.i.ik ? this.i.ik.top : NaN;
    }
    set labelTextMarginTop(v) {
        this.ensureLabelTextMargin();
        this.i.ik.top = +v;
        this.i.ik = this.i.ik;
    }
    ensureLabelTextMargin() {
        if (this.i.ik) {
            return;
        }
        this.i.ik = new Thickness(2);
    }
    /**
     * Gets or Sets the style to use for the units text.
    */
    get labelTextStyle() {
        if (this.i.bm == null) {
            return null;
        }
        return this.i.bm.fontString;
    }
    set labelTextStyle(v) {
        let fi = new FontInfo();
        fi.fontString = v;
        this.i.bm = fi;
    }
    /**
     * Gets or sets the scaling value used to affect the pixel density of the control.
     * A higher scaling ratio will produce crisper visuals at the expense of memory.  Lower values will cause the control to appear blurry.
    */
    get pixelScalingRatio() {
        return this.i.pixelScalingRatio;
    }
    set pixelScalingRatio(v) {
        this.i.pixelScalingRatio = +v;
    }
    findByName(name) {
        if (this.valueFormatSpecifiers != null && arrayFindByName(this.valueFormatSpecifiers, name)) {
            return arrayFindByName(this.valueFormatSpecifiers, name);
        }
        if (this.headerFormatSpecifiers != null && arrayFindByName(this.headerFormatSpecifiers, name)) {
            return arrayFindByName(this.headerFormatSpecifiers, name);
        }
        return null;
    }
    get hasUserValues() {
        return this._hasUserValues;
    }
    __m(propertyName) {
        if (!this._inStyling) {
            this._hasUserValues.add(propertyName);
        }
    }
    _styling(container, component, parent) {
        if (this._inStyling) {
            return;
        }
        this._inStyling = true;
        this._stylingContainer = container;
        this._stylingParent = component;
        let genericPrefix = "";
        let typeName = this.i.$type.name;
        if (typeName.indexOf("Xam") === 0) {
            typeName = typeName.substring(3);
        }
        genericPrefix = toSpinal("DataLegend");
        let additionalPrefixes = [];
        let prefix = toSpinal(typeName);
        additionalPrefixes.push(prefix + "-");
        let b = this.i.$type.baseType;
        while (b && b.name != "Object" &&
            b.name != "Base" &&
            b.name != "Control" &&
            b.Name != "DependencyObject" &&
            b.Name != "FrameworkElement") {
            typeName = b.name;
            if (typeName.indexOf("Xam") === 0) {
                typeName = typeName.substring(3);
            }
            let basePrefix = toSpinal(typeName);
            additionalPrefixes.push(basePrefix + "-");
            b = b.baseType;
        }
        if (parent) {
            let parentTypeName = parent.i.$type.name;
            if (parentTypeName.indexOf("Xam") === 0) {
                parentTypeName = parentTypeName.substring(3);
            }
            let parentPrefix = toSpinal(parentTypeName);
            additionalPrefixes.push(parentPrefix + "-" + genericPrefix + "-");
            additionalPrefixes.push(parentPrefix + "-" + prefix + "-");
        }
        initializePropertiesFromCss(container, this, genericPrefix + "-", this.hasUserValues, false, additionalPrefixes);
        if (this._otherStyling) {
            this._otherStyling(container, component, parent);
        }
        this._inStyling = false;
    }
    /**
 * Returns the legend visuals expressed as a serialized string.

*/
    exportSerializedVisualData() {
        let iv = this.i.exportSerializedVisualData();
        return (iv);
    }
    /**
     * Notifies the legend about changes to size
    
    */
    notifySizeChanged() {
        this.i.notifySizeChanged();
    }
    /**
     * Forces any pending refresh to the legend to be finished.
    
    */
    flush() {
        this.i.flush();
    }
    getAbbreviatedNumber(number, mode, minDigits, maxDigits) {
        let iv = this.i.cq(number, (mode == null ? null : mode), minDigits, maxDigits);
        return (iv);
    }
    getAbbreviatedSymbol(number, mode, minDigits, maxDigits) {
        let iv = this.i.c7(number, (mode == null ? null : mode), minDigits, maxDigits);
        return (iv);
    }
    getAbbreviatedString(number, mode, minDigits, maxDigits) {
        let iv = this.i.c6(number, (mode == null ? null : mode), minDigits, maxDigits);
        return (iv);
    }
    /**
     * Event raised when calculating summary value for each column in data legend
    */
    get calculateColumnSummary() {
        return this._calculateColumnSummary;
    }
    set calculateColumnSummary(ev) {
        if (this._calculateColumnSummary_wrapped !== null) {
            this.i.calculateColumnSummary = delegateRemove(this.i.calculateColumnSummary, this._calculateColumnSummary_wrapped);
            this._calculateColumnSummary_wrapped = null;
            this._calculateColumnSummary = null;
        }
        this._calculateColumnSummary = ev;
        this._calculateColumnSummary_wrapped = (o, e) => {
            let outerArgs = new IgrDataLegendSummaryEventArgs();
            outerArgs._provideImplementation(e);
            if (this.beforeCalculateColumnSummary) {
                this.beforeCalculateColumnSummary(this, outerArgs);
            }
            if (this._calculateColumnSummary) {
                this._calculateColumnSummary(this, outerArgs);
            }
        };
        this.i.calculateColumnSummary = delegateCombine(this.i.calculateColumnSummary, this._calculateColumnSummary_wrapped);
        ;
    }
    /**
     * Event raised when styling the header row
    */
    get styleHeaderRow() {
        return this._styleHeaderRow;
    }
    set styleHeaderRow(ev) {
        if (this._styleHeaderRow_wrapped !== null) {
            this.i.styleHeaderRow = delegateRemove(this.i.styleHeaderRow, this._styleHeaderRow_wrapped);
            this._styleHeaderRow_wrapped = null;
            this._styleHeaderRow = null;
        }
        this._styleHeaderRow = ev;
        this._styleHeaderRow_wrapped = (o, e) => {
            let outerArgs = new IgrDataLegendStylingRowEventArgs();
            outerArgs._provideImplementation(e);
            if (this.beforeStyleHeaderRow) {
                this.beforeStyleHeaderRow(this, outerArgs);
            }
            if (this._styleHeaderRow) {
                this._styleHeaderRow(this, outerArgs);
            }
        };
        this.i.styleHeaderRow = delegateCombine(this.i.styleHeaderRow, this._styleHeaderRow_wrapped);
        ;
    }
    /**
     * Event raised when styling title and badge in series rows
    */
    get styleSeriesRow() {
        return this._styleSeriesRow;
    }
    set styleSeriesRow(ev) {
        if (this._styleSeriesRow_wrapped !== null) {
            this.i.styleSeriesRow = delegateRemove(this.i.styleSeriesRow, this._styleSeriesRow_wrapped);
            this._styleSeriesRow_wrapped = null;
            this._styleSeriesRow = null;
        }
        this._styleSeriesRow = ev;
        this._styleSeriesRow_wrapped = (o, e) => {
            let outerArgs = new IgrDataLegendStylingRowEventArgs();
            outerArgs._provideImplementation(e);
            if (this.beforeStyleSeriesRow) {
                this.beforeStyleSeriesRow(this, outerArgs);
            }
            if (this._styleSeriesRow) {
                this._styleSeriesRow(this, outerArgs);
            }
        };
        this.i.styleSeriesRow = delegateCombine(this.i.styleSeriesRow, this._styleSeriesRow_wrapped);
        ;
    }
    /**
     * Event raised when styling title in the summary row
    */
    get styleSummaryRow() {
        return this._styleSummaryRow;
    }
    set styleSummaryRow(ev) {
        if (this._styleSummaryRow_wrapped !== null) {
            this.i.styleSummaryRow = delegateRemove(this.i.styleSummaryRow, this._styleSummaryRow_wrapped);
            this._styleSummaryRow_wrapped = null;
            this._styleSummaryRow = null;
        }
        this._styleSummaryRow = ev;
        this._styleSummaryRow_wrapped = (o, e) => {
            let outerArgs = new IgrDataLegendStylingRowEventArgs();
            outerArgs._provideImplementation(e);
            if (this.beforeStyleSummaryRow) {
                this.beforeStyleSummaryRow(this, outerArgs);
            }
            if (this._styleSummaryRow) {
                this._styleSummaryRow(this, outerArgs);
            }
        };
        this.i.styleSummaryRow = delegateCombine(this.i.styleSummaryRow, this._styleSummaryRow_wrapped);
        ;
    }
    /**
     * Event raised when styling the label, value, and unit columns in series rows
    */
    get styleSeriesColumn() {
        return this._styleSeriesColumn;
    }
    set styleSeriesColumn(ev) {
        if (this._styleSeriesColumn_wrapped !== null) {
            this.i.styleSeriesColumn = delegateRemove(this.i.styleSeriesColumn, this._styleSeriesColumn_wrapped);
            this._styleSeriesColumn_wrapped = null;
            this._styleSeriesColumn = null;
        }
        this._styleSeriesColumn = ev;
        this._styleSeriesColumn_wrapped = (o, e) => {
            let outerArgs = new IgrDataLegendStylingColumnEventArgs();
            outerArgs._provideImplementation(e);
            if (this.beforeStyleSeriesColumn) {
                this.beforeStyleSeriesColumn(this, outerArgs);
            }
            if (this._styleSeriesColumn) {
                this._styleSeriesColumn(this, outerArgs);
            }
        };
        this.i.styleSeriesColumn = delegateCombine(this.i.styleSeriesColumn, this._styleSeriesColumn_wrapped);
        ;
    }
    /**
     * Event raised when styling the label, value, and unit columns in summary row
    */
    get styleSummaryColumn() {
        return this._styleSummaryColumn;
    }
    set styleSummaryColumn(ev) {
        if (this._styleSummaryColumn_wrapped !== null) {
            this.i.styleSummaryColumn = delegateRemove(this.i.styleSummaryColumn, this._styleSummaryColumn_wrapped);
            this._styleSummaryColumn_wrapped = null;
            this._styleSummaryColumn = null;
        }
        this._styleSummaryColumn = ev;
        this._styleSummaryColumn_wrapped = (o, e) => {
            let outerArgs = new IgrDataLegendStylingColumnEventArgs();
            outerArgs._provideImplementation(e);
            if (this.beforeStyleSummaryColumn) {
                this.beforeStyleSummaryColumn(this, outerArgs);
            }
            if (this._styleSummaryColumn) {
                this._styleSummaryColumn(this, outerArgs);
            }
        };
        this.i.styleSummaryColumn = delegateCombine(this.i.styleSummaryColumn, this._styleSummaryColumn_wrapped);
        ;
    }
}
