// igniteui-react-charts/esm2015/lib/igr-callout-series-selecting-event-args.js
import { IgrSeries } from "./igr-series";
import { CalloutSeriesSelectingEventArgs as CalloutSeriesSelectingEventArgs_internal } from "./CalloutSeriesSelectingEventArgs";
/**
 * Represents event arguments for selecting callout layer
*/
export class IgrCalloutSeriesSelectingEventArgs {
    constructor() {
        this._implementation = this.createImplementation();
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    }
    createImplementation() {
        return new CalloutSeriesSelectingEventArgs_internal();
    }
    /**
     * @hidden
     */
    get i() {
        return this._implementation;
    }
    onImplementationCreated() {
    }
    _provideImplementation(i) {
        this._implementation = i;
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    }
    /**
     * Gets data X-value assositated with callout layer
    */
    get xValue() {
        return this.i.c;
    }
    set xValue(v) {
        this.i.c = v;
    }
    /**
     * Gets data Y-value assositated with callout layer
    */
    get yValue() {
        return this.i.d;
    }
    set yValue(v) {
        this.i.d = v;
    }
    /**
     * Gets data item assositated with callout layer
    */
    get item() {
        return this.i.item;
    }
    set item(v) {
        this.i.item = v;
    }
    /**
     * Gets or sets series object assositated with callout layer
    */
    get series() {
        if (this.i.a == null) {
            return null;
        }
        if (!this.i.a.externalObject) {
            let e = IgrSeries._createFromInternal(this.i.a);
            if (e) {
                e._implementation = this.i.a;
            }
            this.i.a.externalObject = e;
        }
        return this.i.a.externalObject;
    }
    set series(v) {
        v == null ? this.i.a = null : this.i.a = v.i;
    }
    /**
     * Gets or sets series name assositated with callout layer
    */
    get seriesName() {
        return this.i.e;
    }
    set seriesName(v) {
        this.i.e = v;
    }
}
