// igniteui-react-charts/esm2015/lib/igr-callout-content-updating-event-args.js
import { CalloutContentUpdatingEventArgs as CalloutContentUpdatingEventArgs_internal } from "./CalloutContentUpdatingEventArgs";
/**
 * Represents event arguments for updating content of callout layer
*/
export class IgrCalloutContentUpdatingEventArgs {
    constructor() {
        this._implementation = this.createImplementation();
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    }
    createImplementation() {
        return new CalloutContentUpdatingEventArgs_internal();
    }
    /**
     * @hidden
     */
    get i() {
        return this._implementation;
    }
    onImplementationCreated() {
    }
    _provideImplementation(i) {
        this._implementation = i;
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    }
    /**
     * Gets data X-value assositated with callout layer
    */
    get xValue() {
        return this.i.c;
    }
    set xValue(v) {
        this.i.c = v;
    }
    /**
     * Gets data Y-value assositated with callout layer
    */
    get yValue() {
        return this.i.d;
    }
    set yValue(v) {
        this.i.d = v;
    }
    /**
     * Gets data item assositated with callout layer
    */
    get item() {
        return this.i.item;
    }
    set item(v) {
        this.i.item = v;
    }
    /**
     * Gets or sets content displayed by callout layer
    */
    get content() {
        return this.i.a;
    }
    set content(v) {
        this.i.a = v;
    }
}
