// igniteui-react-charts/esm2015/lib/igr-series-viewer-manipulation-event-args.js
import { SeriesViewerManipulationEventArgs as SeriesViewerManipulationEventArgs_internal } from "./SeriesViewerManipulationEventArgs";
import { ensureBool } from "igniteui-react-core";
export class IgrSeriesViewerManipulationEventArgs {
    constructor() {
        this._implementation = this.createImplementation();
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    }
    createImplementation() {
        return new SeriesViewerManipulationEventArgs_internal();
    }
    /**
     * @hidden
     */
    get i() {
        return this._implementation;
    }
    onImplementationCreated() {
    }
    _provideImplementation(i) {
        this._implementation = i;
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    }
    get isZoomPan() {
        return this.i.isZoomPan;
    }
    set isZoomPan(v) {
        this.i.isZoomPan = ensureBool(v);
    }
    get isDragZoom() {
        return this.i.isDragZoom;
    }
    set isDragZoom(v) {
        this.i.isDragZoom = ensureBool(v);
    }
}
