// igniteui-react-charts/esm2015/lib/igr-scatter-polyline-series.js
import { IgrStyle } from "igniteui-react-core";
import { IgrShapeSeriesBase } from "./igr-shape-series-base";
import { ScatterPolylineSeries } from "./ScatterPolylineSeries";
/**
 * Series class used for displaying multiple polylines for datasources which contain multiple lists of points.
*/
export class IgrScatterPolylineSeries extends IgrShapeSeriesBase {
    createImplementation() {
        return new ScatterPolylineSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    constructor(props) {
        super(props);
    }
    /**
     * Gets whether the current series shows a polyline shape.
    */
    get isPolyline() {
        return this.i.d9;
    }
    /**
     * The default style to apply to all Shapes in the series.
    */
    get shapeStyle() {
        if (this.i.s1 == null) {
            return null;
        }
        if (!this.i.s1.externalObject) {
            let e = new IgrStyle();
            e._implementation = this.i.s1;
            this.i.s1.externalObject = e;
        }
        return this.i.s1.externalObject;
    }
    set shapeStyle(v) {
        v == null ? this.i.s1 = null : this.i.s1 = v.i;
    }
    findByName(name) {
        var baseResult = super.findByName(name);
        if (baseResult) {
            return baseResult;
        }
        if (this.shapeStyle && this.shapeStyle.name && this.shapeStyle.name == name) {
            return this.shapeStyle;
        }
        return null;
    }
}
