// igniteui-react-charts/esm2015/lib/igr-item-tool-tip-layer.js
import { IgrSeries } from "./igr-series";
import { IgrAnnotationLayer } from "./igr-annotation-layer";
import { ItemToolTipLayer } from "./ItemToolTipLayer";
import { ensureBool } from "igniteui-react-core";
/**
 * Represents an annotation layer that displays tooltips for all target series individually.
*/
export class IgrItemToolTipLayer extends IgrAnnotationLayer {
    createImplementation() {
        return new ItemToolTipLayer();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    constructor(props) {
        super(props);
    }
    /**
     * Gets whether the series is an annotation layer displayed only when hovering over the chart.
    */
    get isAnnotationHoverLayer() {
        return this.i.db;
    }
    /**
     * Gets whether the default tooltip behaviors for the chart are disabled if this layer is present.
    */
    get isDefaultTooltipBehaviorDisabled() {
        return this.i.dl;
    }
    /**
     * Gets or sets the name of the series series to target this annotation to. If null, this annotation targets all series simultaneously.
    */
    get targetSeriesName() {
        return this.i.rj;
    }
    set targetSeriesName(v) {
        this.i.rj = v;
    }
    /**
     * Gets or sets the series to target this annotation to. If null, this annotation targets all series simultaneously.
    */
    get targetSeries() {
        if (this.i.rd == null) {
            return null;
        }
        if (!this.i.rd.externalObject) {
            let e = IgrSeries._createFromInternal(this.i.rd);
            if (e) {
                e._implementation = this.i.rd;
            }
            this.i.rd.externalObject = e;
        }
        return this.i.rd.externalObject;
    }
    set targetSeries(v) {
        if (v != null && this._stylingContainer && v._styling)
            v._styling(this._stylingContainer, this, this);
        v == null ? this.i.rd = null : this.i.rd = v.i;
    }
    /**
     * Gets or sets whether to use value interpolation when drawing the tooltips.
    */
    get useInterpolation() {
        return this.i.rg;
    }
    set useInterpolation(v) {
        this.i.rg = ensureBool(v);
    }
    /**
     * Gets or sets whether to skip past unknown values when searching for series values.
    */
    get skipUnknownValues() {
        return this.i.rf;
    }
    set skipUnknownValues(v) {
        this.i.rf = ensureBool(v);
    }
    findByName(name) {
        var baseResult = super.findByName(name);
        if (baseResult) {
            return baseResult;
        }
        if (this.targetSeries && this.targetSeries.name && this.targetSeries.name == name) {
            return this.targetSeries;
        }
        return null;
    }
    _styling(container, component, parent) {
        super._styling(container, component, parent);
        this._inStyling = true;
        if (this.targetSeries && this.targetSeries._styling) {
            this.targetSeries._styling(container, component, this);
        }
        this._inStyling = false;
    }
    /**
     * Hides any tooltips presented by the layer, if any.
    
    */
    hideToolTips() {
        this.i.lg();
    }
    /**
     * Hides any tooltips presented by the layer, if any.
    
    */
    hideToolTipsImmediate() {
        this.i.lh();
    }
}
