// igniteui-react-charts/esm2015/lib/igr-chart-cursor-event-args.js
import { IgrSeries } from "./igr-series";
/**
 * Provides data for IgxDataChartComponent mouse button related events.
*/
export class IgrChartCursorEventArgs {
    constructor() {
    }
    /**
     * @hidden
     */
    get i() {
        return this._implementation;
    }
    onImplementationCreated() {
    }
    _provideImplementation(i) {
        this._implementation = i;
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    }
    /**
     * Gets the ItemsSource item associated with the current event.
    */
    get item() {
        return this.i.item;
    }
    set item(v) {
        this.i.item = v;
    }
    /**
     * Gets the series associated with the current event.
    */
    get series() {
        if (this.i.series == null) {
            return null;
        }
        if (!this.i.series.externalObject) {
            let e = IgrSeries._createFromInternal(this.i.series);
            if (e) {
                e._implementation = this.i.series;
            }
            this.i.series.externalObject = e;
        }
        return this.i.series.externalObject;
    }
    set series(v) {
        v == null ? this.i.series = null : this.i.series = v.i;
    }
    /**
     * Gets the Chart associated with the current event.
    */
    get seriesViewer() {
        if (this.i.seriesViewer == null) {
            return null;
        }
        return this.i.seriesViewer.externalObject;
    }
    set seriesViewer(v) {
        v == null ? this.i.seriesViewer = null : this.i.seriesViewer = v.i;
    }
    /**
     * Provides a human readable expresion of the event arguments.
    
    */
    toString() {
        let iv = this.i.toString();
        return (iv);
    }
}
