// igniteui-react-charts/esm2015/lib/igr-treemap-node-style-mapping.js
import { TreemapNodeStyleMappingTargetType_$type } from "./TreemapNodeStyleMappingTargetType";
import { TreemapValueMappingMode_$type } from "./TreemapValueMappingMode";
import { IgrTreemapNodeStyle } from "./igr-treemap-node-style";
import { TreemapNodeStyleMapping } from "./TreemapNodeStyleMapping";
import { ensureEnum } from "igniteui-react-core";
/**
 * Represents a style to apply to a set of nodes in the treemap.
*/
export class IgrTreemapNodeStyleMapping extends IgrTreemapNodeStyle {
    createImplementation() {
        return new TreemapNodeStyleMapping();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    constructor(props) {
        super(props);
    }
    /**
     * Gets or sets the unique name of the mapping.
    */
    get targetType() {
        return this.i.bf;
    }
    set targetType(v) {
        this.i.bf = ensureEnum(TreemapNodeStyleMappingTargetType_$type, v);
    }
    /**
     * Gets or sets the unique name of the mapping.
    */
    get name() {
        return this.i.bs;
    }
    set name(v) {
        this.i.bs = v;
    }
    /**
     * Gets or sets the minimum value to which the mapping applies.
    */
    get minimumValue() {
        return this.i.bl;
    }
    set minimumValue(v) {
        this.i.bl = +v;
    }
    /**
     * Gets or sets the maximum value to which the mapping applies.
    */
    get maximumValue() {
        return this.i.bk;
    }
    set maximumValue(v) {
        this.i.bk = +v;
    }
    /**
     * Gets or sets the value to which the mapping applies.
    */
    get value() {
        return this.i.bn;
    }
    set value(v) {
        this.i.bn = v;
    }
    /**
     * Gets or sets the mapping mode to use to evaluate whether the mapping applies.
    */
    get mappingMode() {
        return this.i.bh;
    }
    set mappingMode(v) {
        this.i.bh = ensureEnum(TreemapValueMappingMode_$type, v);
    }
}
