// igniteui-react-charts/esm2015/lib/MarketFacilitationIndexIndicator.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ItemwiseStrategyBasedIndicator } from "./ItemwiseStrategyBasedIndicator";
import { MarketFacilitationIndexIndicatorStrategy } from "./MarketFacilitationIndexIndicatorStrategy";
import { markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let MarketFacilitationIndexIndicator = /*@__PURE__*/ (() => {
    class MarketFacilitationIndexIndicator extends ItemwiseStrategyBasedIndicator {
        constructor() {
            super();
        }
        get_ts() {
            return 1;
        }
        get_u7() {
            return new MarketFacilitationIndexIndicatorStrategy();
        }
        get u7() {
            return this.get_u7();
        }
        get_uy() {
            return MarketFacilitationIndexIndicator.$;
        }
        get uy() {
            return this.get_uy();
        }
    }
    MarketFacilitationIndexIndicator.$t = /*@__PURE__*/ markType(MarketFacilitationIndexIndicator, 'MarketFacilitationIndexIndicator', ItemwiseStrategyBasedIndicator.$);
    return MarketFacilitationIndexIndicator;
})();
