// igniteui-react-charts/esm2015/lib/PolarAxisInfoCache.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let PolarAxisInfoCache = /*@__PURE__*/ (() => {
    class PolarAxisInfoCache extends Base {
        constructor(a, b, c) {
            super();
            this.a = null;
            this.b = null;
            this.e = false;
            this.g = false;
            this.d = false;
            this.f = false;
            this.c = null;
            this.i = 0;
            this.h = 0;
            this.a = a;
            this.b = b;
            this.e = this.a.lh;
            this.d = this.a.cg;
            this.g = this.b.lh;
            this.f = this.b.cg;
            this.i = this.b.nm;
            this.h = this.b.nl;
            this.c = c;
        }
    }
    PolarAxisInfoCache.$t = /*@__PURE__*/ markType(PolarAxisInfoCache, 'PolarAxisInfoCache');
    return PolarAxisInfoCache;
})();
