// igniteui-react-charts/esm2015/lib/SortingTrendLineManager.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { CategoryTrendLineManagerBase } from "./CategoryTrendLineManagerBase";
import { Point_$type, markType } from "igniteui-react-core";
import { List$1 } from "igniteui-react-core";
import { TrendFitCalculator } from "./TrendFitCalculator";
import { TrendAverageCalculator } from "./TrendAverageCalculator";
import { truncate } from "igniteui-react-core";
/**
 * @hidden
 */
export let SortingTrendLineManager = /*@__PURE__*/ (() => {
    class SortingTrendLineManager extends CategoryTrendLineManagerBase {
        constructor(a, b) {
            super();
            this.h = null;
            this.i = null;
            this.h = a;
            this.i = b;
        }
        get_g$a() {
            return 1;
        }
        prepareLineCore(a, b, c, d, e, f, g) {
            let h = g.d * g.c;
            let i = g.e * g.c;
            let j = new List$1(Point_$type, 0);
            if (b == 0) {
                this.a = null;
                this.d.clear();
                return;
            }
            if (this.c(b)) {
                this.d.clear();
                this.a = TrendFitCalculator.a(j, b, g, this.a, c.count, (k) => (k + 1), (k) => c.item(k), (k) => {
                    let l = truncate(Math.floor(k));
                    let m = truncate(Math.ceil(k));
                    let n = k - l;
                    let o;
                    if (m <= i) {
                        o = this.h(l) + n * (this.h(m) - this.h(l));
                    }
                    else {
                        o = this.h(l) + n * (this.h(truncate(i)) - this.h(l));
                    }
                    return e(o);
                }, f, h, i);
            }
            if (this.b(b)) {
                this.a = null;
                TrendAverageCalculator.b(b, this.d, c, d);
                for (let k = g.d; k <= g.e; k += 1) {
                    let l = k * g.c;
                    let m = this.h(l);
                    if (l >= 0 && l < this.d.count) {
                        let n = e(m);
                        let o = f(this.d._inner[l]);
                        j.add({ $type: Point_$type, x: n + g.a, y: o });
                    }
                }
            }
            this.e(j, g, a);
        }
    }
    SortingTrendLineManager.$t = /*@__PURE__*/ markType(SortingTrendLineManager, 'SortingTrendLineManager', CategoryTrendLineManagerBase.$);
    return SortingTrendLineManager;
})();
