// igniteui-react-charts/esm2015/lib/ViewportUtils.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-react-core";
/**
 * @hidden
 */
export let ViewportUtils = /*@__PURE__*/ (() => {
    class ViewportUtils extends Base {
        static c(a, b, c) {
            return c.left + c.width * (a - b.left) / b.width;
        }
        static a(a, b, c) {
            return ((a - c.left) * b.width / c.width) + b.left;
        }
        static f(a, b, c) {
            return c.top + c.height * (a - b.top) / b.height;
        }
        static e(a, b, c) {
            return ((a - c.top) * b.height / c.height) + b.top;
        }
        static d(a, b, c) {
            return c.width * a / b.width;
        }
        static b(a, b, c) {
            return b.width * a / c.width;
        }
        static g(a, b, c) {
            return c.height * a / b.height;
        }
    }
    ViewportUtils.$t = /*@__PURE__*/ markType(ViewportUtils, 'ViewportUtils');
    return ViewportUtils;
})();
